<!DOCTYPE html><html><head><title>Help for package UCSCXenaTools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {UCSCXenaTools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#availTCGA'><p>Get or Check TCGA Available ProjectID, DataType and FileType</p></a></li>
<li><a href='#cohorts'><p>Get cohorts of XenaHub object</p></a></li>
<li><a href='#datasets'><p>Get datasets of XenaHub object</p></a></li>
<li><a href='#downloadTCGA'><p>Easily Download TCGA Data by Several Options</p></a></li>
<li><a href='#fetch'><p>Fetch Data from UCSC Xena Hosts</p></a></li>
<li><a href='#getTCGAdata'><p>Get TCGA Common Data Sets by Project ID and Property</p></a></li>
<li><a href='#hosts'><p>Get hosts of XenaHub object</p></a></li>
<li><a href='#samples'><p>Get Samples of a XenaHub object according to 'by' and 'how' action arguments</p></a></li>
<li><a href='#showTCGA'><p>Show TCGA data structure by Project ID or ALL</p></a></li>
<li><a href='#to_snake'><p>Convert camel case to snake case</p></a></li>
<li><a href='#xena_default_hosts'><p>UCSC Xena Default Hosts</p></a></li>
<li><a href='#XenaBrowse'><p>View Info of Dataset or Cohort at UCSC Xena Website Using Web browser</p></a></li>
<li><a href='#XenaData'><p>Xena Hub Information</p></a></li>
<li><a href='#XenaDataUpdate'><p>Get or Update Newest Data Information of UCSC Xena Data Hubs</p></a></li>
<li><a href='#XenaDownload'><p>Download Datasets from UCSC Xena Hubs</p></a></li>
<li><a href='#XenaFilter'><p>Filter a XenaHub Object</p></a></li>
<li><a href='#XenaGenerate'><p>Generate and Subset a XenaHub Object from 'XenaData'</p></a></li>
<li><a href='#XenaHub'><p>Generate a XenaHub Object</p></a></li>
<li><a href='#XenaHub-class'><p>Class XenaHub</p></a></li>
<li><a href='#XenaPrepare'><p>Prepare (Load) Downloaded Datasets to R</p></a></li>
<li><a href='#XenaQuery'><p>Query URL of Datasets before Downloading</p></a></li>
<li><a href='#XenaQueryProbeMap'><p>Query ProbeMap URL of Datasets</p></a></li>
<li><a href='#XenaScan'><p>Scan all rows according to user input by a regular expression</p></a></li>
<li><a href='#XenaShiny'><p>Xena Shiny App</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Download and Explore Datasets from UCSC Xena Data Hubs</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.8</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shixiang Wang &lt;w_shixiang@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Download and explore datasets from UCSC Xena data hubs, which
    are a collection of UCSC-hosted public databases such as TCGA, ICGC,
    TARGET, GTEx, CCLE, and others.  Databases are normalized so they can
    be combined, linked, filtered, explored and downloaded.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/UCSCXenaTools/">https://docs.ropensci.org/UCSCXenaTools/</a>,
<a href="https://github.com/ropensci/UCSCXenaTools">https://github.com/ropensci/UCSCXenaTools</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/UCSCXenaTools/issues">https://github.com/ropensci/UCSCXenaTools/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>digest, dplyr, httr, jsonlite, magrittr, methods, readr,
rlang, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, DT, knitr, prettydoc, rmarkdown, shiny, shinydashboard,
testthat (&ge; 2.1.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-20 07:53:49 UTC; wsx</td>
</tr>
<tr>
<td>Author:</td>
<td>Shixiang Wang <a href="https://orcid.org/0000-0001-9855-7357"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Xue-Song Liu <a href="https://orcid.org/0000-0002-7736-0077"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Martin Morgan [ctb],
  Christine Stawitz [rev] (Christine reviewed the package for ropensci,
    see &lt;https://github.com/ropensci/software-review/issues/315&gt;),
  Carl Ganz [rev] (Carl reviewed the package for ropensci, see
    &lt;https://github.com/ropensci/software-review/issues/315&gt;)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-20 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic++25+3E+25">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='availTCGA'>Get or Check TCGA Available ProjectID, DataType and FileType</h2><span id='topic+availTCGA'></span>

<h3>Description</h3>

<p>Get or Check TCGA Available ProjectID, DataType and FileType
</p>


<h3>Usage</h3>

<pre><code class='language-R'>availTCGA(which = c("all", "ProjectID", "DataType", "FileType"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="availTCGA_+3A_which">which</code></td>
<td>
<p>a character of <code>c("All", "ProjectID", "DataType", "FileType")</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
availTCGA("all")

</code></pre>

<hr>
<h2 id='cohorts'>Get cohorts of XenaHub object</h2><span id='topic+cohorts'></span>

<h3>Description</h3>

<p>Get cohorts of XenaHub object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohorts(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cohorts_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector contains cohorts
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xe = XenaGenerate(subset = XenaHostNames == "tcgaHub"); cohorts(xe)
</code></pre>

<hr>
<h2 id='datasets'>Get datasets of XenaHub object</h2><span id='topic+datasets'></span>

<h3>Description</h3>

<p>Get datasets of XenaHub object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>datasets(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="datasets_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector contains datasets
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xe = XenaGenerate(subset = XenaHostNames == "tcgaHub"); datasets(xe)
</code></pre>

<hr>
<h2 id='downloadTCGA'>Easily Download TCGA Data by Several Options</h2><span id='topic+downloadTCGA'></span>

<h3>Description</h3>

<p>TCGA is a very useful database and here we provide this function to
download TCGA (include TCGA Pancan) datasets in human-friendly way. Users who are not
familiar with R operation will benefit from this.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downloadTCGA(
  project = NULL,
  data_type = NULL,
  file_type = NULL,
  destdir = tempdir(),
  force = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="downloadTCGA_+3A_project">project</code></td>
<td>
<p>default is <code>NULL</code>. Should be one or more of TCGA project id (character vector) provided by Xena.
See all available project id, please use <code>availTCGA("ProjectID")</code>.</p>
</td></tr>
<tr><td><code id="downloadTCGA_+3A_data_type">data_type</code></td>
<td>
<p>default is <code>NULL</code>. Should be a character vector specify data type.
See all available data types by <code>availTCGA("DataType")</code>.</p>
</td></tr>
<tr><td><code id="downloadTCGA_+3A_file_type">file_type</code></td>
<td>
<p>default is <code>NULL</code>. Should be a character vector specify file type.
See all available file types by <code>availTCGA("FileType")</code>.</p>
</td></tr>
<tr><td><code id="downloadTCGA_+3A_destdir">destdir</code></td>
<td>
<p>specify a location to store download data. Default is system temp directory.</p>
</td></tr>
<tr><td><code id="downloadTCGA_+3A_force">force</code></td>
<td>
<p>logical. if <code>TRUE</code>, force to download data no matter whether files exist.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="downloadTCGA_+3A_...">...</code></td>
<td>
<p>other argument to <code>download.file</code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All availble information about datasets of TCGA can access vis <code>availTCGA()</code> and
check with <code>showTCGA()</code>.
</p>


<h3>Value</h3>

<p>same as <code>XenaDownload()</code> function result.
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+XenaQuery">XenaQuery()</a></code>,
<code><a href="#topic+XenaFilter">XenaFilter()</a></code>,
<code><a href="#topic+XenaDownload">XenaDownload()</a></code>,
<code><a href="#topic+XenaPrepare">XenaPrepare()</a></code>,
<code><a href="#topic+availTCGA">availTCGA()</a></code>,
<code><a href="#topic+showTCGA">showTCGA()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download RNASeq data (use UVM as example)
downloadTCGA(project = "UVM",
                 data_type = "Gene Expression RNASeq",
                 file_type = "IlluminaHiSeq RNASeqV2")

## End(Not run)
</code></pre>

<hr>
<h2 id='fetch'>Fetch Data from UCSC Xena Hosts</h2><span id='topic+fetch'></span><span id='topic+fetch_dense_values'></span><span id='topic+fetch_sparse_values'></span><span id='topic+fetch_dataset_samples'></span><span id='topic+fetch_dataset_identifiers'></span><span id='topic+has_probeMap'></span>

<h3>Description</h3>

<p>When you want to query just data for several genes/samples from UCSC Xena datasets, a better way
is to use these <code>fetch_</code> functions instead of downloading a whole dataset. Details about functions
please see the following sections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch(host, dataset)

fetch_dense_values(
  host,
  dataset,
  identifiers = NULL,
  samples = NULL,
  check = TRUE,
  use_probeMap = FALSE,
  time_limit = 30
)

fetch_sparse_values(host, dataset, genes, samples = NULL, time_limit = 30)

fetch_dataset_samples(host, dataset, limit = NULL)

fetch_dataset_identifiers(host, dataset)

has_probeMap(host, dataset, return_url = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetch_+3A_host">host</code></td>
<td>
<p>a UCSC Xena host, like &quot;https://toil.xenahubs.net&quot;.
All available hosts can be printed by <code><a href="#topic+xena_default_hosts">xena_default_hosts()</a></code>.</p>
</td></tr>
<tr><td><code id="fetch_+3A_dataset">dataset</code></td>
<td>
<p>a UCSC Xena dataset, like &quot;tcga_RSEM_gene_tpm&quot;.
All available datasets can be printed by running <code>XenaData$XenaDatasets</code> or
obtained from <a href="https://xenabrowser.net/datapages/">UCSC Xena datapages</a>.</p>
</td></tr>
<tr><td><code id="fetch_+3A_identifiers">identifiers</code></td>
<td>
<p>Identifiers could be probe (like &quot;ENSG00000000419.12&quot;),
gene (like &quot;TP53&quot;) etc.. If it is <code>NULL</code>, all identifiers in the dataset will be used.</p>
</td></tr>
<tr><td><code id="fetch_+3A_samples">samples</code></td>
<td>
<p>ID of samples, like &quot;TCGA-02-0047-01&quot;.
If it is <code>NULL</code>, all samples in the dataset will be used. However, it is better to download
the whole datasets if you query many samples and genes.</p>
</td></tr>
<tr><td><code id="fetch_+3A_check">check</code></td>
<td>
<p>if <code>TRUE</code>, check whether specified <code>identifiers</code> and <code>samples</code> exist the dataset
(all failed items will be filtered out). However, if <code>FALSE</code>, the code is much faster.</p>
</td></tr>
<tr><td><code id="fetch_+3A_use_probemap">use_probeMap</code></td>
<td>
<p>if <code>TRUE</code>, will check if the dataset has ProbeMap firstly.
When the dataset you want to query has a identifier-to-gene mapping, identifiers can be
gene symbols even the identifiers of dataset are probes or others.</p>
</td></tr>
<tr><td><code id="fetch_+3A_time_limit">time_limit</code></td>
<td>
<p>time limit for getting response in seconds.</p>
</td></tr>
<tr><td><code id="fetch_+3A_genes">genes</code></td>
<td>
<p>gene names.</p>
</td></tr>
<tr><td><code id="fetch_+3A_limit">limit</code></td>
<td>
<p>number of samples, if <code>NULL</code>, return all samples.</p>
</td></tr>
<tr><td><code id="fetch_+3A_return_url">return_url</code></td>
<td>
<p>if <code>TRUE</code>, returns the info of probeMap
instead of a logical value when the result exists.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are three primary data types: dense matrix (samples by probes (or say identifiers)),
sparse (sample, position, variant), and segmented (sample, position, value).
</p>
<p>Dense matrices can be genotypic or phenotypic, it is a sample-by-identifiers matrix.
Phenotypic matrices have associated field metadata (descriptive names, codes, etc.).
Genotypic matricies may have an associated probeMap, which maps probes to genomic locations.
If a matrix has hugo probeMap, the probes themselves are gene names. Otherwise, a probeMap is
used to map a gene location to a set of probes.
</p>


<h3>Value</h3>

<p>a <code>matirx</code> or character vector or a <code>list</code>.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>fetch_dense_values</code>: fetches values from a dense matrix.
</p>
</li>
<li> <p><code>fetch_sparse_values</code>: fetches values from a sparse <code>data.frame</code>.
</p>
</li>
<li> <p><code>fetch_dataset_samples</code>: fetches samples from a dataset
</p>
</li>
<li> <p><code>fetch_dataset_identifiers</code>: fetches identifies from a dataset.
</p>
</li>
<li> <p><code>has_probeMap</code>: checks if a dataset has ProbeMap.
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>library(UCSCXenaTools)

host &lt;- "https://toil.xenahubs.net"
dataset &lt;- "tcga_RSEM_gene_tpm"
samples &lt;- c("TCGA-02-0047-01", "TCGA-02-0055-01", "TCGA-02-2483-01", "TCGA-02-2485-01")
probes &lt;- c("ENSG00000282740.1", "ENSG00000000005.5", "ENSG00000000419.12")
genes &lt;- c("TP53", "RB1", "PIK3CA")


# Fetch samples
fetch_dataset_samples(host, dataset, 2)
# Fetch identifiers
fetch_dataset_identifiers(host, dataset)
# Fetch expression value by probes
fetch_dense_values(host, dataset, probes, samples, check = FALSE)
# Fetch expression value by gene symbol (if the dataset has probeMap)
has_probeMap(host, dataset)
fetch_dense_values(host, dataset, genes, samples, check = FALSE, use_probeMap = TRUE)

</code></pre>

<hr>
<h2 id='getTCGAdata'>Get TCGA Common Data Sets by Project ID and Property</h2><span id='topic+getTCGAdata'></span>

<h3>Description</h3>

<p>This is the most useful function for user to download common
TCGA datasets, it is similar to <code>getFirehoseData</code> function in <code>RTCGAToolbox</code>
package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTCGAdata(
  project = NULL,
  clinical = TRUE,
  download = FALSE,
  forceDownload = FALSE,
  destdir = tempdir(),
  mRNASeq = FALSE,
  mRNAArray = FALSE,
  mRNASeqType = "normalized",
  miRNASeq = FALSE,
  exonRNASeq = FALSE,
  RPPAArray = FALSE,
  ReplicateBaseNormalization = FALSE,
  Methylation = FALSE,
  MethylationType = c("27K", "450K"),
  GeneMutation = FALSE,
  SomaticMutation = FALSE,
  GisticCopyNumber = FALSE,
  Gistic2Threshold = TRUE,
  CopyNumberSegment = FALSE,
  RemoveGermlineCNV = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTCGAdata_+3A_project">project</code></td>
<td>
<p>default is <code>NULL</code>. Should be one or more of TCGA project id (character vector) provided by Xena.
See all available project id, please use <code>availTCGA("ProjectID")</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_clinical">clinical</code></td>
<td>
<p>logical. if <code>TRUE</code>, download clinical information. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_download">download</code></td>
<td>
<p>logical. if <code>TRUE</code>, download data, otherwise return a result list include data
information. Default is <code>FALSE</code>. You can set this to <code>FALSE</code> if you want to check what you will download or
use other function provided by <code>UCSCXenaTools</code> to filter result datasets you want to download.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_forcedownload">forceDownload</code></td>
<td>
<p>logical. if <code>TRUE</code>, force to download files no matter if exist. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_destdir">destdir</code></td>
<td>
<p>specify a location to store download data. Default is system temp directory.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_mrnaseq">mRNASeq</code></td>
<td>
<p>logical. if <code>TRUE</code>, download mRNASeq data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_mrnaarray">mRNAArray</code></td>
<td>
<p>logical. if <code>TRUE</code>, download mRNA microarray data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_mrnaseqtype">mRNASeqType</code></td>
<td>
<p>character vector. Can be one, two or three
in <code>c("normalized", "pancan normalized", "percentile")</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_mirnaseq">miRNASeq</code></td>
<td>
<p>logical. if <code>TRUE</code>, download miRNASeq data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_exonrnaseq">exonRNASeq</code></td>
<td>
<p>logical. if <code>TRUE</code>, download exon RNASeq data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_rppaarray">RPPAArray</code></td>
<td>
<p>logical. if <code>TRUE</code>, download RPPA data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_replicatebasenormalization">ReplicateBaseNormalization</code></td>
<td>
<p>logical. if <code>TRUE</code>, download RPPA data by Replicate Base
Normalization (RBN). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_methylation">Methylation</code></td>
<td>
<p>logical. if <code>TRUE</code>, download DNA Methylation data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_methylationtype">MethylationType</code></td>
<td>
<p>character vector. Can be one or two in <code>c("27K", "450K")</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_genemutation">GeneMutation</code></td>
<td>
<p>logical. if <code>TRUE</code>, download gene mutation data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_somaticmutation">SomaticMutation</code></td>
<td>
<p>logical. if <code>TRUE</code>, download somatic mutation data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_gisticcopynumber">GisticCopyNumber</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Gistic2 Copy Number data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_gistic2threshold">Gistic2Threshold</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Threshold Gistic2 data. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_copynumbersegment">CopyNumberSegment</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Copy Number Segment data. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_removegermlinecnv">RemoveGermlineCNV</code></td>
<td>
<p>logical. if <code>TRUE</code>, download Copy Number Segment data which has removed
germline copy number variation. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="getTCGAdata_+3A_...">...</code></td>
<td>
<p>other argument to <code>download.file</code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>TCGA Common Data Sets are frequently used for biological analysis.
To make easier to achieve these data, this function provide really easy
options to choose datasets and behavior. All availble information about
datasets of TCGA can access vis <code>availTCGA()</code> and check with <code>showTCGA()</code>.
</p>


<h3>Value</h3>

<p>if <code>download=TRUE</code>, return <code>data.frame</code> from <code>XenaDownload</code>,
otherwise return a list including <code>XenaHub</code> object and datasets information
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###### get data, but not download

# 1 choose project and data types you wanna download
getTCGAdata(project = "LUAD", mRNASeq = TRUE, mRNAArray = TRUE,
mRNASeqType = "normalized", miRNASeq = TRUE, exonRNASeq = TRUE,
RPPAArray = TRUE, Methylation = TRUE, MethylationType = "450K",
GeneMutation = TRUE, SomaticMutation = TRUE)

# 2 only choose 'LUAD' and its clinical data
getTCGAdata(project = "LUAD")
## Not run: 
###### download datasets

# 3 download clinical datasets of LUAD and LUSC
getTCGAdata(project = c("LUAD", "LUSC"), clinical = TRUE, download = TRUE)

# 4 download clinical, RPPA and gene mutation datasets of LUAD and LUSC
# getTCGAdata(project = c("LUAD", "LUSC"), clinical = TRUE, RPPAArray = TRUE, GeneMutation = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='hosts'>Get hosts of XenaHub object</h2><span id='topic+hosts'></span>

<h3>Description</h3>

<p>Get hosts of XenaHub object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hosts(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hosts_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector contains hosts
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xe = XenaGenerate(subset = XenaHostNames == "tcgaHub"); hosts(xe)
</code></pre>

<hr>
<h2 id='samples'>Get Samples of a XenaHub object according to 'by' and 'how' action arguments</h2><span id='topic+samples'></span>

<h3>Description</h3>

<p>One is often interested in identifying samples or features present in each data set,
or shared by all data sets, or present in any of several data sets.
Identifying these samples, including samples in arbitrarily chosen data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samples(
  x,
  i = character(),
  by = c("hosts", "cohorts", "datasets"),
  how = c("each", "any", "all")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="samples_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
<tr><td><code id="samples_+3A_i">i</code></td>
<td>
<p>default is a empty character, it is used to specify
the host, cohort or dataset by <code>by</code> option otherwise
info will be automatically extracted by code</p>
</td></tr>
<tr><td><code id="samples_+3A_by">by</code></td>
<td>
<p>a character specify <code>by</code> action</p>
</td></tr>
<tr><td><code id="samples_+3A_how">how</code></td>
<td>
<p>a character specify <code>how</code> action</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list include samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xe = XenaHub(cohorts = "Cancer Cell Line Encyclopedia (CCLE)")
# samples in each dataset, first host
x = samples(xe, by="datasets", how="each")[[1]]
lengths(x)        # data sets in ccle cohort on first (only) host

## End(Not run)
</code></pre>

<hr>
<h2 id='showTCGA'>Show TCGA data structure by Project ID or ALL</h2><span id='topic+showTCGA'></span>

<h3>Description</h3>

<p>This can used to check if data type or file type exist in one or more projects by hand.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showTCGA(project = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showTCGA_+3A_project">project</code></td>
<td>
<p>a character vector. Can be &quot;all&quot; or one or more of TCGA Project IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> including project data structure information.
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+availTCGA">availTCGA()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showTCGA("all")

</code></pre>

<hr>
<h2 id='to_snake'>Convert camel case to snake case</h2><span id='topic+to_snake'></span>

<h3>Description</h3>

<p>Convert camel case to snake case
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_snake(name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to_snake_+3A_name">name</code></td>
<td>
<p>a character vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>same length as <code>name</code> but with snake case
</p>


<h3>Examples</h3>

<pre><code class='language-R'>to_snake("sparseDataRange")
</code></pre>

<hr>
<h2 id='xena_default_hosts'>UCSC Xena Default Hosts</h2><span id='topic+xena_default_hosts'></span>

<h3>Description</h3>

<p>Return Xena default hosts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xena_default_hosts()
</code></pre>


<h3>Value</h3>

<p>A character vector include current defalut hosts
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+XenaHub">XenaHub()</a></code>
</p>

<hr>
<h2 id='XenaBrowse'>View Info of Dataset or Cohort at UCSC Xena Website Using Web browser</h2><span id='topic+XenaBrowse'></span>

<h3>Description</h3>

<p>This will open dataset/cohort link of UCSC Xena
in user's default browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaBrowse(x, type = c("dataset", "cohort"), multiple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaBrowse_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object.</p>
</td></tr>
<tr><td><code id="XenaBrowse_+3A_type">type</code></td>
<td>
<p>one of &quot;dataset&quot; and &quot;cohort&quot;.</p>
</td></tr>
<tr><td><code id="XenaBrowse_+3A_multiple">multiple</code></td>
<td>
<p>if <code>TRUE</code>, browse multiple links instead of throwing error.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
XenaGenerate(subset = XenaHostNames == "tcgaHub") %&gt;%
  XenaFilter(filterDatasets = "clinical") %&gt;%
  XenaFilter(filterDatasets = "LUAD") -&gt; to_browse

</code></pre>

<hr>
<h2 id='XenaData'>Xena Hub Information</h2><span id='topic+XenaData'></span>

<h3>Description</h3>

<p>This <code>data.frame</code> is very useful for selecting datasets fastly and
independent on APIs of UCSC Xena Hubs.
</p>


<h3>Format</h3>

<p>A <code>tibble</code>.
</p>


<h3>Source</h3>

<p>Generated from UCSC Xena Data Hubs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(XenaData)
str(XenaData)
</code></pre>

<hr>
<h2 id='XenaDataUpdate'>Get or Update Newest Data Information of UCSC Xena Data Hubs</h2><span id='topic+XenaDataUpdate'></span>

<h3>Description</h3>

<p>Get or Update Newest Data Information of UCSC Xena Data Hubs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaDataUpdate(saveTolocal = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaDataUpdate_+3A_savetolocal">saveTolocal</code></td>
<td>
<p>logical. Whether save to local R package data directory for permanent use
or Not.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> contains all datasets information of Xena.
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
XenaDataUpdate()
XenaDataUpdate(saveTolocal = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='XenaDownload'>Download Datasets from UCSC Xena Hubs</h2><span id='topic+XenaDownload'></span>

<h3>Description</h3>

<p>Avaliable datasets list: <a href="https://xenabrowser.net/datapages/">https://xenabrowser.net/datapages/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaDownload(
  xquery,
  destdir = tempdir(),
  download_probeMap = FALSE,
  trans_slash = FALSE,
  force = FALSE,
  max_try = 3L,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaDownload_+3A_xquery">xquery</code></td>
<td>
<p>a tibble object generated by <a href="#topic+XenaQuery">XenaQuery</a> function.</p>
</td></tr>
<tr><td><code id="XenaDownload_+3A_destdir">destdir</code></td>
<td>
<p>specify a location to store download data. Default is system temp directory.</p>
</td></tr>
<tr><td><code id="XenaDownload_+3A_download_probemap">download_probeMap</code></td>
<td>
<p>if <code>TRUE</code>, also download ProbeMap data, which used for id mapping.</p>
</td></tr>
<tr><td><code id="XenaDownload_+3A_trans_slash">trans_slash</code></td>
<td>
<p>logical, default is <code>FALSE</code>. If <code>TRUE</code>, transform slash '/' in dataset id
to '__'. This option is for backwards compatibility.</p>
</td></tr>
<tr><td><code id="XenaDownload_+3A_force">force</code></td>
<td>
<p>logical. if <code>TRUE</code>, force to download data no matter whether files exist.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="XenaDownload_+3A_max_try">max_try</code></td>
<td>
<p>time limit to try downloading the data.</p>
</td></tr>
<tr><td><code id="XenaDownload_+3A_...">...</code></td>
<td>
<p>other argument to <code>download.file</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>tibble</code>
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xe = XenaGenerate(subset = XenaHostNames == "tcgaHub")
hosts(xe)
xe_query = XenaQuery(xe)
xe_download = XenaDownload(xe_query)

## End(Not run)
</code></pre>

<hr>
<h2 id='XenaFilter'>Filter a XenaHub Object</h2><span id='topic+XenaFilter'></span>

<h3>Description</h3>

<p>One of main functions in <strong>UCSCXenatools</strong>. It is used to filter
<code>XenaHub</code> object according to cohorts, datasets. All datasets can be found
at <a href="https://xenabrowser.net/datapages/">https://xenabrowser.net/datapages/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaFilter(
  x,
  filterCohorts = NULL,
  filterDatasets = NULL,
  ignore.case = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaFilter_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
<tr><td><code id="XenaFilter_+3A_filtercohorts">filterCohorts</code></td>
<td>
<p>default is <code>NULL</code>. A character used to filter cohorts,
regular expression is supported.</p>
</td></tr>
<tr><td><code id="XenaFilter_+3A_filterdatasets">filterDatasets</code></td>
<td>
<p>default is <code>NULL</code>. A character used to filter datasets,
regular expression is supported.</p>
</td></tr>
<tr><td><code id="XenaFilter_+3A_ignore.case">ignore.case</code></td>
<td>
<p>if <code>FALSE</code>, the pattern matching is case sensitive
and if <code>TRUE</code>, case is ignored during matching.</p>
</td></tr>
<tr><td><code id="XenaFilter_+3A_...">...</code></td>
<td>
<p>other arguments except <code>value</code> passed to <code><a href="base.html#topic+grep">base::grep()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>XenaHub</code> object
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># operate TCGA datasets
xe = XenaGenerate(subset = XenaHostNames == "tcgaHub")
xe
# get all names of clinical data
xe2 = XenaFilter(xe, filterDatasets = "clinical")
datasets(xe2)
</code></pre>

<hr>
<h2 id='XenaGenerate'>Generate and Subset a XenaHub Object from 'XenaData'</h2><span id='topic+XenaGenerate'></span>

<h3>Description</h3>

<p>Generate and Subset a XenaHub Object from 'XenaData'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaGenerate(XenaData = UCSCXenaTools::XenaData, subset = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaGenerate_+3A_xenadata">XenaData</code></td>
<td>
<p>a <code>data.frame</code>. Default is <code>data(XenaData)</code>.
The input of this option can only be <code>data(XenaData)</code> or its subset.</p>
</td></tr>
<tr><td><code id="XenaGenerate_+3A_subset">subset</code></td>
<td>
<p>logical expression indicating elements or rows to keep.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+XenaHub">XenaHub</a> object.
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1 get all datasets
XenaGenerate()
# 2 get TCGA BRCA
XenaGenerate(subset = XenaCohorts == "TCGA Breast Cancer (BRCA)")
# 3 get all datasets containing BRCA
XenaGenerate(subset = grepl("BRCA", XenaCohorts))
</code></pre>

<hr>
<h2 id='XenaHub'>Generate a XenaHub Object</h2><span id='topic+XenaHub'></span>

<h3>Description</h3>

<p>It is used to generate original
<code>XenaHub</code> object according to hosts, cohorts, datasets or hostName.
If these arguments not specified, all hosts and corresponding datasets
will be returned as a <code>XenaHub</code> object. All datasets can be found
at <a href="https://xenabrowser.net/datapages/">https://xenabrowser.net/datapages/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaHub(
  hosts = xena_default_hosts(),
  cohorts = character(),
  datasets = character(),
  hostName = c("publicHub", "tcgaHub", "gdcHub", "icgcHub", "toilHub",
    "pancanAtlasHub", "treehouseHub", "pcawgHub", "atacseqHub", "singlecellHub",
    "kidsfirstHub")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaHub_+3A_hosts">hosts</code></td>
<td>
<p>a character vector specify UCSC Xena hosts, all available hosts can be
found by <code>xena_default_hosts()</code> function. <code>hostName</code> is a more recommend option.</p>
</td></tr>
<tr><td><code id="XenaHub_+3A_cohorts">cohorts</code></td>
<td>
<p>default is empty character vector, all cohorts will be returned.</p>
</td></tr>
<tr><td><code id="XenaHub_+3A_datasets">datasets</code></td>
<td>
<p>default is empty character vector, all datasets will be returned.</p>
</td></tr>
<tr><td><code id="XenaHub_+3A_hostname">hostName</code></td>
<td>
<p>name of host, available options can be accessed by <code>.xena_hosts</code>
This is an easier option for user than <code>hosts</code> option. Note, this option
will overlap <code>hosts</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+XenaHub">XenaHub</a> object
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#1 query all hosts, cohorts and datasets
xe = XenaHub()
xe
#2 query only TCGA hosts
xe = XenaHub(hostName = "tcgaHub")
xe
hosts(xe)     # get hosts
cohorts(xe)   # get cohorts
datasets(xe)  # get datasets
samples(xe)   # get samples

## End(Not run)
</code></pre>

<hr>
<h2 id='XenaHub-class'>Class XenaHub</h2><span id='topic+XenaHub-class'></span><span id='topic+.XenaHub'></span>

<h3>Description</h3>

<p>a S4 class to represent UCSC Xena Data Hubs
</p>


<h3>Slots</h3>


<dl>
<dt><code>hosts</code></dt><dd><p>hosts of data hubs</p>
</dd>
<dt><code>cohorts</code></dt><dd><p>cohorts of data hubs</p>
</dd>
<dt><code>datasets</code></dt><dd><p>datasets of data hubs</p>
</dd>
</dl>

<hr>
<h2 id='XenaPrepare'>Prepare (Load) Downloaded Datasets to R</h2><span id='topic+XenaPrepare'></span>

<h3>Description</h3>

<p>Prepare (Load) Downloaded Datasets to R
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaPrepare(
  objects,
  objectsName = NULL,
  use_chunk = FALSE,
  chunk_size = 100,
  subset_rows = TRUE,
  select_cols = TRUE,
  callback = NULL,
  comment = "#",
  na = c("", "NA", "[Discrepancy]"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaPrepare_+3A_objects">objects</code></td>
<td>
<p>a object of character vector or data.frame. If <code>objects</code> is data.frame,
it should be returned object of <a href="#topic+XenaDownload">XenaDownload</a> function. More easier way is
that objects can be character vector specify local files/directory and download urls.</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_objectsname">objectsName</code></td>
<td>
<p>specify names for elements of return object, i.e. names of list</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_use_chunk">use_chunk</code></td>
<td>
<p>default is <code>FALSE</code>. If you want to select subset of original data, please set it to
<code>TRUE</code> and specify corresponding arguments: <code>chunk_size</code>, <code>select_direction</code>, <code>select_names</code>,
<code>callback</code>.</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_chunk_size">chunk_size</code></td>
<td>
<p>the number of rows to include in each chunk</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_subset_rows">subset_rows</code></td>
<td>
<p>logical expression indicating elements or rows to keep:
missing values are taken as false. <code>x</code> can be a representation of data frame
you wanna do subset operation. Of note, the first colname of most of datasets
in Xena will be set to &quot;sample&quot;, you can use it to select rows.</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_select_cols">select_cols</code></td>
<td>
<p>expression, indicating columns to select from a data frame.
'x' can be a representation of data frame you wanna do subset operation,
e.g. <code>select_cols = colnames(x)[1:3]</code> will keep only first to third column.</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_callback">callback</code></td>
<td>
<p>a function to call on each chunk, default is <code>NULL</code>,
this option will overvide operations of subset_rows and select_cols.</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_comment">comment</code></td>
<td>
<p>a character specify comment rows in files</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_na">na</code></td>
<td>
<p>a character vectory specify <code>NA</code> values in files</p>
</td></tr>
<tr><td><code id="XenaPrepare_+3A_...">...</code></td>
<td>
<p>other arguments transfer to <code>read_tsv</code> function or
<code>read_tsv_chunked</code> function (when <code>use_chunk</code> is <code>TRUE</code>) of <code>readr</code> package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list contains file data, which in way of tibbles
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xe = XenaGenerate(subset = XenaHostNames == "tcgaHub")
hosts(xe)
xe_query = XenaQuery(xe)

xe_download = XenaDownload(xe_query)
dat = XenaPrepare(xe_download)

## End(Not run)
</code></pre>

<hr>
<h2 id='XenaQuery'>Query URL of Datasets before Downloading</h2><span id='topic+XenaQuery'></span>

<h3>Description</h3>

<p>Query URL of Datasets before Downloading
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaQuery(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaQuery_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> contains hosts, datasets and url
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xe = XenaGenerate(subset = XenaHostNames == "tcgaHub")
hosts(xe)
## Not run: 
xe_query = XenaQuery(xe)

## End(Not run)
</code></pre>

<hr>
<h2 id='XenaQueryProbeMap'>Query ProbeMap URL of Datasets</h2><span id='topic+XenaQueryProbeMap'></span>

<h3>Description</h3>

<p>If dataset has no ProbeMap, it will be ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaQueryProbeMap(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaQueryProbeMap_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+XenaHub">XenaHub</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> contains hosts, datasets and url
</p>


<h3>Author(s)</h3>

<p>Shixiang Wang <a href="mailto:w_shixiang@163.com">w_shixiang@163.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xe = XenaGenerate(subset = XenaHostNames == "tcgaHub")
hosts(xe)
## Not run: 
xe_query = XenaQueryProbeMap(xe)

## End(Not run)
</code></pre>

<hr>
<h2 id='XenaScan'>Scan all rows according to user input by a regular expression</h2><span id='topic+XenaScan'></span>

<h3>Description</h3>

<p><code>XenaScan()</code> is a function can be used before <code><a href="#topic+XenaGenerate">XenaGenerate()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaScan(
  XenaData = UCSCXenaTools::XenaData,
  pattern = NULL,
  ignore.case = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XenaScan_+3A_xenadata">XenaData</code></td>
<td>
<p>a <code>data.frame</code>. Default is <code>data(XenaData)</code>.
The input of this option can only be <code>data(XenaData)</code> or its subset.</p>
</td></tr>
<tr><td><code id="XenaScan_+3A_pattern">pattern</code></td>
<td>
<p>character string containing a <a href="base.html#topic+regular+20expression">regular expression</a>
(or character string for <code>fixed = TRUE</code>) to be matched
in the given character vector.  Coerced by
<code><a href="base.html#topic+as.character">as.character</a></code> to a character string if possible.  If a
character vector of length 2 or more is supplied, the first element
is used with a warning.  Missing values are allowed except for
<code>regexpr</code>, <code>gregexpr</code> and <code>regexec</code>.</p>
</td></tr>
<tr><td><code id="XenaScan_+3A_ignore.case">ignore.case</code></td>
<td>
<p>if <code>FALSE</code>, the pattern matching is <em>case
sensitive</em> and if <code>TRUE</code>, case is ignored during matching.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
x1 &lt;- XenaScan(pattern = "Blood")
x2 &lt;- XenaScan(pattern = "LUNG", ignore.case = FALSE)

x1 %&gt;%
  XenaGenerate()
x2 %&gt;%
  XenaGenerate()
</code></pre>

<hr>
<h2 id='XenaShiny'>Xena Shiny App</h2><span id='topic+XenaShiny'></span>

<h3>Description</h3>

<p>Xena Shiny App
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XenaShiny()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
