<!DOCTYPE html><html><head><title>Help for package UKB.COVID19</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {UKB.COVID19}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#comorbidity_asso'><p>Generate comorbidity association result file</p></a></li>
<li><a href='#comorbidity_summary'><p>Create comorbidity summary file</p></a></li>
<li><a href='#covid_example'><p>Provide working directory for UKB.COVID19 example files</p></a></li>
<li><a href='#data_reform'><p>Reform variables</p></a></li>
<li><a href='#log_cov'><p>Perform association tests between phenotype and covariates</p></a></li>
<li><a href='#makeGWASFiles'><p>Generate files for GWAS Software. SAIGE and Plink currently supported.</p></a></li>
<li><a href='#makePhenotypes'><p>Generate COVID-19 phenotypes</p></a></li>
<li><a href='#risk_factor'><p>Generate covariate file</p></a></li>
<li><a href='#sampleQC'><p>Sample QC for genetic analyses</p></a></li>
<li><a href='#variantQC'><p>Variant QC for Genetic Analyses</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>UK Biobank COVID-19 Data Processing and Risk Factor Association
Tests</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Longfei Wang <a href="https://orcid.org/0000-0002-5143-4146"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Longfei Wang &lt;wang.lo@wehi.edu.au&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Process UK Biobank COVID-19 test result data for susceptibility, severity and mortality analyses, perform potential non-genetic COVID-19 risk factor and co-morbidity association tests. Wang et al. (2021) &lt;<a href="https://doi.org/10.5281%2Fzenodo.5174381">doi:10.5281/zenodo.5174381</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>questionr, data.table, tidyverse, magrittr, here, dplyr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/bahlolab/UKB.COVID19">https://github.com/bahlolab/UKB.COVID19</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1.9001</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-25 00:10:27 UTC; wang.lo</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-25 00:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='comorbidity_asso'>Generate comorbidity association result file</h2><span id='topic+comorbidity_asso'></span>

<h3>Description</h3>

<p>Association tests between each co-morbidity and given phenotype (susceptibility, mortality or severity) with the adjustment of covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comorbidity_asso(
  pheno,
  covariates,
  cormorbidity,
  population = "all",
  cov.name = c("sex", "age", "bmi"),
  phe.name,
  ICD10.file
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comorbidity_asso_+3A_pheno">pheno</code></td>
<td>
<p>phenotype dataframe - output from makePheno function</p>
</td></tr>
<tr><td><code id="comorbidity_asso_+3A_covariates">covariates</code></td>
<td>
<p>covariate dataframe - output from risk.factor function.  Optional.</p>
</td></tr>
<tr><td><code id="comorbidity_asso_+3A_cormorbidity">cormorbidity</code></td>
<td>
<p>Comorbidity summary generated from comorbidity.summary.</p>
</td></tr>
<tr><td><code id="comorbidity_asso_+3A_population">population</code></td>
<td>
<p>Choose self-report population/ethnic background group from &quot;all&quot;, white&quot;, &quot;black&quot;, &quot;asian&quot;, &quot;mixed&quot;, or &quot;other&quot;. By default, population=&quot;all&quot;, include all ethnic groups.</p>
</td></tr>
<tr><td><code id="comorbidity_asso_+3A_cov.name">cov.name</code></td>
<td>
<p>Selected covariates names. By default, cov.name=c(&quot;sex&quot;,&quot;age&quot;,&quot;bmi&quot;), covariates are sex age and BMI.</p>
</td></tr>
<tr><td><code id="comorbidity_asso_+3A_phe.name">phe.name</code></td>
<td>
<p>Phenotype name.</p>
</td></tr>
<tr><td><code id="comorbidity_asso_+3A_icd10.file">ICD10.file</code></td>
<td>
<p>The ICD10 code file, which is included in the package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Outputs a comorbidity association test result with OR, 95% CI and p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
comorb.asso &lt;- comorbidity_asso(pheno=phe,
covariates=covar,
cormorbidity=comorb,
population="white",
cov.name=c("sex","age","bmi","SES","smoke","inAgedCare"),
phe.name="hospitalisation",
ICD10.file=covid_example("ICD10.coding19.txt.gz"))

## End(Not run)

</code></pre>

<hr>
<h2 id='comorbidity_summary'>Create comorbidity summary file</h2><span id='topic+comorbidity_summary'></span>

<h3>Description</h3>

<p>summarise disease history records of each individual from the hospital inpatient diagnosis data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>comorbidity_summary(
  ukb.data,
  hesin.file,
  hesin_diag.file,
  primary = FALSE,
  ICD10.file,
  Date.start = NULL,
  Date.end = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="comorbidity_summary_+3A_ukb.data">ukb.data</code></td>
<td>
<p>tab delimited UK Biobank phenotype file, containing sample qc fields (with default UKBiobank codes as column names)</p>
</td></tr>
<tr><td><code id="comorbidity_summary_+3A_hesin.file">hesin.file</code></td>
<td>
<p>Latest hospital inpatient master file.</p>
</td></tr>
<tr><td><code id="comorbidity_summary_+3A_hesin_diag.file">hesin_diag.file</code></td>
<td>
<p>Latest hospital inpatient diagnosis file.</p>
</td></tr>
<tr><td><code id="comorbidity_summary_+3A_primary">primary</code></td>
<td>
<p>TRUE: include primary diagnosis only; FALSE: include all diagnoses.</p>
</td></tr>
<tr><td><code id="comorbidity_summary_+3A_icd10.file">ICD10.file</code></td>
<td>
<p>The ICD10 code file, which is included in the package.</p>
</td></tr>
<tr><td><code id="comorbidity_summary_+3A_date.start">Date.start</code></td>
<td>
<p>Date, dd/mm/yyyy, select the start date of hospital inpatient record period.</p>
</td></tr>
<tr><td><code id="comorbidity_summary_+3A_date.end">Date.end</code></td>
<td>
<p>Date, dd/mm/yyyy, select the end date of hospital inpatient record period.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Outputs comorbidity summary table, named comorbidity_&lt;Date.start&gt;_&lt;Date.end&gt;.RData, including phenotype, non-genetic risk factors and all comorbidities, which will be used in the comorbidity association tests.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
comorb &lt;- comorbidity_summary(ukb.data=covid_example("sim_ukb.tab.gz"),
hesin.file=covid_example("sim_hesin.txt.gz"), 
hesin_diag.file=covid_example("sim_hesin_diag.txt.gz"), 
ICD10.file=covid_example("ICD10.coding19.txt.gz"),
primary = FALSE,
Date.start = "16/03/2020")

## End(Not run)

</code></pre>

<hr>
<h2 id='covid_example'>Provide working directory for UKB.COVID19 example files</h2><span id='topic+covid_example'></span>

<h3>Description</h3>

<p>Provide working directory for UKB.COVID19 example files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid_example(path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="covid_example_+3A_path">path</code></td>
<td>
<p>path to file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Outputs the working directory for UKB.COVID19 example files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>covid_example('results/covariate.txt')

</code></pre>

<hr>
<h2 id='data_reform'>Reform variables</h2><span id='topic+data_reform'></span>

<h3>Description</h3>

<p>Reform variables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_reform(res, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data_reform_+3A_res">res</code></td>
<td>
<p>Merged data of phenotype from makePhenotypes or comorbidity_summary and covariates from risk_factor.</p>
</td></tr>
<tr><td><code id="data_reform_+3A_type">type</code></td>
<td>
<p>Data type: susceptibility, severity, mortality or comorbidity.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Reformed data for association tests using logistic regression models.
</p>

<hr>
<h2 id='log_cov'>Perform association tests between phenotype and covariates</h2><span id='topic+log_cov'></span>

<h3>Description</h3>

<p>Perform association tests between phenotype and covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_cov(pheno, covariates, phe.name, cov.name = c("sex", "age", "bmi"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log_cov_+3A_pheno">pheno</code></td>
<td>
<p>phenotype dataframe - output from makePhenotypes function</p>
</td></tr>
<tr><td><code id="log_cov_+3A_covariates">covariates</code></td>
<td>
<p>covariate dataframe - output from risk_factor function.</p>
</td></tr>
<tr><td><code id="log_cov_+3A_phe.name">phe.name</code></td>
<td>
<p>Phenotype name in the data.</p>
</td></tr>
<tr><td><code id="log_cov_+3A_cov.name">cov.name</code></td>
<td>
<p>Selected covariate names in the data. By default, cov.name=c(&quot;sex&quot;,&quot;age&quot;,&quot;bmi&quot;), covariates include sex, age and BMI.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Outputs association test results with OR, 95% CI, and p-value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
log_cov(pheno=phe, covariates=covar, phe.name="hospitalisation", cov.name=c("sex","age","bmi"))

## End(Not run)
</code></pre>

<hr>
<h2 id='makeGWASFiles'>Generate files for GWAS Software. SAIGE and Plink currently supported.</h2><span id='topic+makeGWASFiles'></span>

<h3>Description</h3>

<p>Generate files for GWAS Software. SAIGE and Plink currently supported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeGWASFiles(
  ukb.data,
  pheno,
  covariates,
  phe.name,
  cov.name = NULL,
  includeSampsFile = NULL,
  software = "SAIGE",
  outDir = "",
  prefix
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeGWASFiles_+3A_ukb.data">ukb.data</code></td>
<td>
<p>tab delimited UK Biobank phenotype file, containing sample qc fields (with default UKBiobank codes as column names)</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_pheno">pheno</code></td>
<td>
<p>phenotype dataframe - output from makePhenotype function</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_covariates">covariates</code></td>
<td>
<p>covariate dataframe - output from risk.factor function.  Optional.</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_phe.name">phe.name</code></td>
<td>
<p>phenotypes to be included in outputted data. multiple phenotypes can be specified as a vector. if null, all phenotypes will be outputted.</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_cov.name">cov.name</code></td>
<td>
<p>covariates to be included in outputted data. Optional. multiple covariates can be specified as a vector. if null, all covariates in file will be outputted</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_includesampsfile">includeSampsFile</code></td>
<td>
<p>list of samples to be included GWAS. File with the first column containing sample IDs to be kept. Can contain other columns. output from sampleQC function may be used.  Optional - if null, all samples will be outputted.</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_software">software</code></td>
<td>
<p>specify &quot;SAIGE&quot; or &quot;plink&quot; - defaults to &quot;SAIGE&quot;</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_outdir">outDir</code></td>
<td>
<p>specify directory to output file</p>
</td></tr>
<tr><td><code id="makeGWASFiles_+3A_prefix">prefix</code></td>
<td>
<p>prefix for file - optional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>outputs file, suitable for reading by chosen GWAS software
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
makeGWASFiles(ukb.data=covid_example("sim_ukb.tab.gz"), 
pheno=phe, 
covariates=covar, 
phe.name="hospitalisation", 
cov.name=NULL, 
includeSampsFile=NULL, 
software="SAIGE", 
outDir=covid_example("results"), 
prefix="hospitalisation")

## End(Not run)

</code></pre>

<hr>
<h2 id='makePhenotypes'>Generate COVID-19 phenotypes</h2><span id='topic+makePhenotypes'></span>

<h3>Description</h3>

<p>Generate COVID-19 phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makePhenotypes(
  ukb.data,
  res.eng,
  res.wal = NULL,
  res.sco = NULL,
  death.file,
  death.cause.file,
  hesin.file,
  hesin_diag.file,
  hesin_oper.file,
  hesin_critical.file,
  code.file,
  pheno.type = "severity",
  Date = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makePhenotypes_+3A_ukb.data">ukb.data</code></td>
<td>
<p>tab delimited UK Biobank phenotype file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_res.eng">res.eng</code></td>
<td>
<p>Latest covid result file/files for England.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_res.wal">res.wal</code></td>
<td>
<p>Latest covid result file/files for Wales. Only available for downloads after April 2021.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_res.sco">res.sco</code></td>
<td>
<p>Latest covid result file/files for Scotland. Only available for downloads after April 2021.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_death.file">death.file</code></td>
<td>
<p>Latest death register file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_death.cause.file">death.cause.file</code></td>
<td>
<p>Latest death cause file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_hesin.file">hesin.file</code></td>
<td>
<p>Latest hospital inpatient master file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_hesin_diag.file">hesin_diag.file</code></td>
<td>
<p>Latest hospital inpatient diagnosis file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_hesin_oper.file">hesin_oper.file</code></td>
<td>
<p>Latest hospital inpatient operation file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_hesin_critical.file">hesin_critical.file</code></td>
<td>
<p>Latest hospital inpatient critical care file.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_code.file">code.file</code></td>
<td>
<p>The operation code file, which is included in the package.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_pheno.type">pheno.type</code></td>
<td>
<p>The phenotype options, which include &quot;susceptibility&quot;, &quot;severity&quot;, and &quot;mortality&quot;.</p>
</td></tr>
<tr><td><code id="makePhenotypes_+3A_date">Date</code></td>
<td>
<p>Date, ddmmyyyy, select the results until a certain date. By default, Date = NULL, the latest hospitalization date.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with phenotypes for COVID-19 susceptibility, severity and mortality.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pheno &lt;- makePhenotypes(ukb.data=covid_example("sim_ukb.tab.gz"),
res.eng=covid_example("sim_result_england.txt.gz"),
death.file=covid_example("sim_death.txt.gz"),
death.cause.file=covid_example("sim_death_cause.txt.gz"),
hesin.file=covid_example("sim_hesin.txt.gz"),
hesin_diag.file=covid_example("sim_hesin_diag.txt.gz"),
hesin_oper.file=covid_example("sim_hesin_oper.txt.gz"),
hesin_critical.file=covid_example("sim_hesin_critical.txt.gz"),
code.file=covid_example("coding240.txt.gz"),
pheno.type = "severity")

## End(Not run)

</code></pre>

<hr>
<h2 id='risk_factor'>Generate covariate file</h2><span id='topic+risk_factor'></span>

<h3>Description</h3>

<p>This function formats and outputs a covariate table, used for input for other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>risk_factor(
  ukb.data,
  ABO.data = NULL,
  hesin.file,
  res.eng,
  res.wal = NULL,
  res.sco = NULL,
  fields = NULL,
  field.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="risk_factor_+3A_ukb.data">ukb.data</code></td>
<td>
<p>tab delimited UK Biobank phenotype file. The file should include fields of gender, year of birth, BMI, ethnic background, SES, and smoking.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_abo.data">ABO.data</code></td>
<td>
<p>Latest yyyymmdd_covid19_misc.txt file.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_hesin.file">hesin.file</code></td>
<td>
<p>Latest yyyymmdd_hesin.txt file.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_res.eng">res.eng</code></td>
<td>
<p>Latest covid result file/files for England.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_res.wal">res.wal</code></td>
<td>
<p>Latest covid result file/files for Wales. Only available for downloads after April 2021.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_res.sco">res.sco</code></td>
<td>
<p>Latest covid result file/files for Scotland. Only available for downloads after April 2021.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_fields">fields</code></td>
<td>
<p>User specified field codes from ukb.data file.</p>
</td></tr>
<tr><td><code id="risk_factor_+3A_field.names">field.names</code></td>
<td>
<p>User specified field names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Outputs a covariate table, used for input for other functions. Automatically returns sex, age at birthday in 2020, SES, self-reported ethnicity, most recently reported BMI, most recently reported pack-years, whether they reside in aged care (based on hospital admissions data, and covid test data) and blood type. Function also allows user to specify fields of interest (field codes, provided by UK Biobank), and allows the users to specify more intuitive names, for selected fields.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
covars &lt;- risk_factor(ukb.data=covid_example("sim_ukb.tab.gz"),
ABO.data=covid_example("sim_covid19_misc.txt.gz"),
hesin.file=covid_example("sim_hesin.txt.gz"),
res.eng=covid_example("sim_result_england.txt.gz"))

## End(Not run)

</code></pre>

<hr>
<h2 id='sampleQC'>Sample QC for genetic analyses</h2><span id='topic+sampleQC'></span>

<h3>Description</h3>

<p>Sample QC for genetic analyses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampleQC(ukb.data, withdrawnFile, ancestry = "all", software = "SAIGE", outDir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampleQC_+3A_ukb.data">ukb.data</code></td>
<td>
<p>tab delimited UK Biobank phenotype file, containing sample qc fields (with default UKBiobank codes as column names)</p>
</td></tr>
<tr><td><code id="sampleQC_+3A_withdrawnfile">withdrawnFile</code></td>
<td>
<p>csv file with withdrawn IDs from UK Biobank</p>
</td></tr>
<tr><td><code id="sampleQC_+3A_ancestry">ancestry</code></td>
<td>
<p>specify &quot;WhiteBritish&quot; or &quot;all&quot; - defaults to &quot;all&quot;</p>
</td></tr>
<tr><td><code id="sampleQC_+3A_software">software</code></td>
<td>
<p>specify &quot;SAIGE&quot; or &quot;plink&quot; - defaults to &quot;SAIGE&quot;</p>
</td></tr>
<tr><td><code id="sampleQC_+3A_outdir">outDir</code></td>
<td>
<p>specify directory for sample QC file and inclusion/exclusion lists</p>
</td></tr>
</table>


<h3>Value</h3>

<p>outputs sample QC file, and sample inclusion / exclusion lists for specified software
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
sampleQC(ukb.data=covid_example("sim_ukb.tab.gz"), 
withdrawnFile=covid_example("sim_withdrawn.csv.gz"), 
ancestry="all", 
software="SAIGE", 
outDir=covid_example("results"))

## End(Not run)

</code></pre>

<hr>
<h2 id='variantQC'>Variant QC for Genetic Analyses</h2><span id='topic+variantQC'></span>

<h3>Description</h3>

<p>Variant QC for Genetic Analyses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variantQC(snpQcFile, mfiDir, mafFilt = 0.001, infoFilt = 0.5, outDir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variantQC_+3A_snpqcfile">snpQcFile</code></td>
<td>
<p>file containing SNP QC info (ukb_snp_qc.txt)</p>
</td></tr>
<tr><td><code id="variantQC_+3A_mfidir">mfiDir</code></td>
<td>
<p>directory where the per chromosome UKBiobank MAF/INFO files (ukb_mfi_chr*_v3.txt) are located</p>
</td></tr>
<tr><td><code id="variantQC_+3A_maffilt">mafFilt</code></td>
<td>
<p>minor allele frequency filter - default 0.001</p>
</td></tr>
<tr><td><code id="variantQC_+3A_infofilt">infoFilt</code></td>
<td>
<p>imputation quality (INFO) score filter - default 0.5</p>
</td></tr>
<tr><td><code id="variantQC_+3A_outdir">outDir</code></td>
<td>
<p>output directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>outputs SNP inclusion lists (SNPID and rsID formats) for given MAF/INFO filters. Also outputs list of SNPs to be used for genetic Relatedness Matrix (GRM) calculations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
variantQC(snpQcFile=covid_example("sim_ukb_snp_qc.txt.gz"), 
mfiDir=covid_example("alleleFreqs"), 
mafFilt=0.001, 
infoFilt=0.5, 
outDir=covid_example("results"))

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
