<!DOCTYPE html><html lang="en"><head><title>Help for package ukbtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ukbtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ukbtools'><p>ukbtools: Manipulate and Explore UK Biobank Data</p></a></li>
<li><a href='#icd10chapters'><p>International Classification of Diseases Revision 10 (ICD-10) chapters</p></a></li>
<li><a href='#icd10codes'><p>International Classification of Diseases Revision 10 (ICD-10) codes</p></a></li>
<li><a href='#icd9chapters'><p>International Classification of Diseases Revision 9 (ICD-9) chapters</p></a></li>
<li><a href='#icd9codes'><p>International Classification of Diseases Revision 9 (ICD-9) codes</p></a></li>
<li><a href='#ukb_centre'><p>Inserts UKB centre names into data</p></a></li>
<li><a href='#ukb_context'><p>Demographics of a UKB sample subset</p></a></li>
<li><a href='#ukb_defunct'><p>Defunct genetic metadata functions</p></a></li>
<li><a href='#ukb_df'><p>Reads a UK Biobank phenotype fileset and returns a single dataset.</p></a></li>
<li><a href='#ukb_df_duplicated_name'><p>Checks for duplicated names within a UKB dataset</p></a></li>
<li><a href='#ukb_df_field'><p>Makes a UKB data-field to variable name table for reference or lookup.</p></a></li>
<li><a href='#ukb_df_full_join'><p>Recursively join a list of UKB datasets</p></a></li>
<li><a href='#ukb_gen_excl'><p>Sample exclusions</p></a></li>
<li><a href='#ukb_gen_excl_to_na'><p>Inserts NA into phenotype for genetic metadata exclusions</p></a></li>
<li><a href='#ukb_gen_het'><p>Heterozygosity outliers</p></a></li>
<li><a href='#ukb_gen_meta'><p>Genetic metadata</p></a></li>
<li><a href='#ukb_gen_pcs'><p>Genetic principal components</p></a></li>
<li><a href='#ukb_gen_read_fam'><p>Reads a PLINK format fam file</p></a></li>
<li><a href='#ukb_gen_read_sample'><p>Reads an Oxford format sample file</p></a></li>
<li><a href='#ukb_gen_rel'><p>Creates a table of related individuals</p></a></li>
<li><a href='#ukb_gen_rel_count'><p>Relatedness count</p></a></li>
<li><a href='#ukb_gen_related_with_data'><p>Subset of the UKB relatedness dataframe with data</p></a></li>
<li><a href='#ukb_gen_samples_to_remove'><p>Related samples (with data on the variable of interest) to remove</p></a></li>
<li><a href='#ukb_gen_sqc_names'><p>Sample QC column names</p></a></li>
<li><a href='#ukb_gen_write_bgenie'><p>Writes a BGENIE format phenotype or covariate file.</p></a></li>
<li><a href='#ukb_gen_write_plink'><p>Writes a PLINK format phenotype or covariate file</p></a></li>
<li><a href='#ukb_gen_write_plink_excl'><p>Writes a PLINK format file for combined exclusions</p></a></li>
<li><a href='#ukb_icd_code_meaning'><p>Retrieves description for a ICD code.</p></a></li>
<li><a href='#ukb_icd_diagnosis'><p>Retrieves diagnoses for an individual.</p></a></li>
<li><a href='#ukb_icd_freq_by'><p>Frequency of an ICD diagnosis by a target variable</p></a></li>
<li><a href='#ukb_icd_keyword'><p>Retrieves diagnoses containing a description.</p></a></li>
<li><a href='#ukb_icd_prevalence'><p>Returns the prevalence for an ICD diagnosis</p></a></li>
<li><a href='#ukbcentre'><p>UKB assessment centre</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.11.3</td>
</tr>
<tr>
<td>Title:</td>
<td>Manipulate and Explore UK Biobank Data</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ken Hanscombe &lt;ken.hanscombe@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of tools to create a UK Biobank <a href="http://www.ukbiobank.ac.uk/">http://www.ukbiobank.ac.uk/</a> dataset from a UKB fileset (.tab, .r, .html),
    visualize primary demographic data for a sample subset, query ICD diagnoses,
    retrieve genetic metadata, read and write standard file formats for genetic analyses.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://kenhanscombe.github.io/ukbtools/">https://kenhanscombe.github.io/ukbtools/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.12), dplyr, purrr, readr, ggplot2, XML,
magrittr, grid, tibble, tidyr, scales, stringr, foreach,
parallel, doParallel</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-15 11:22:55 UTC; ken</td>
</tr>
<tr>
<td>Author:</td>
<td>Ken Hanscombe [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-15 11:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ukbtools'>ukbtools: Manipulate and Explore UK Biobank Data</h2><span id='topic+ukbtools'></span><span id='topic+ukbtools-package'></span>

<h3>Description</h3>

<p>A set of tools to create a <a href="https://www.ukbiobank.ac.uk/">UK Biobank</a> dataset
from a UKB fileset (.tab, .r, .html), visualize primary demographic data for
a sample subset, query ICD diagnoses, retrieve genetic metadata, read and
write standard file formats for genetic analyses.
</p>


<h3>UKB Dataframe</h3>

<p>Functions to wrangle the UKB data into a dataframe with meaningful column
names.
</p>

<ul>
<li> <p><code><a href="#topic+ukb_df">ukb_df</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_df_field">ukb_df_field</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_df_full_join">ukb_df_full_join</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_df_duplicated_name">ukb_df_duplicated_name</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_centre">ukb_centre</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_context">ukb_context</a></code>
</p>
</li></ul>



<h3>Genetic Metadata</h3>

<p>Functions to query the associated genetic sample QC information.
</p>

<ul>
<li> <p><code><a href="#topic+ukb_gen_read_fam">ukb_gen_read_fam</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_read_sample">ukb_gen_read_sample</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_rel_count">ukb_gen_rel_count</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_related_with_data">ukb_gen_related_with_data</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_samples_to_remove">ukb_gen_samples_to_remove</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_sqc_names">ukb_gen_sqc_names</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_write_bgenie">ukb_gen_write_bgenie</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_gen_write_plink">ukb_gen_write_plink</a></code>
</p>
</li></ul>



<h3>Disease Diagnoses</h3>

<p>Functions to query the UKB hospital episodes statistics.
</p>

<ul>
<li> <p><code><a href="#topic+ukb_icd_code_meaning">ukb_icd_code_meaning</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_icd_diagnosis">ukb_icd_diagnosis</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_icd_freq_by">ukb_icd_freq_by</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_icd_keyword">ukb_icd_keyword</a></code>
</p>
</li>
<li> <p><code><a href="#topic+ukb_icd_prevalence">ukb_icd_prevalence</a></code>
</p>
</li></ul>



<h3>Datasets</h3>


<ul>
<li> <p><code><a href="#topic+ukbcentre">ukbcentre</a></code>
</p>
</li>
<li> <p><code><a href="#topic+icd10chapters">icd10chapters</a></code>
</p>
</li>
<li> <p><code><a href="#topic+icd10codes">icd10codes</a></code>
</p>
</li>
<li> <p><code><a href="#topic+icd9chapters">icd9chapters</a></code>
</p>
</li>
<li> <p><code><a href="#topic+icd9codes">icd9codes</a></code>
</p>
</li></ul>


<hr>
<h2 id='icd10chapters'>International Classification of Diseases Revision 10 (ICD-10) chapters</h2><span id='topic+icd10chapters'></span>

<h3>Description</h3>

<p>A dataset containing the ICD-10 chapter titles - a top level description of diagnoses classes (or blocks)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>icd10chapters
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 21 rows and 3 columns.</p>

<hr>
<h2 id='icd10codes'>International Classification of Diseases Revision 10 (ICD-10) codes</h2><span id='topic+icd10codes'></span>

<h3>Description</h3>

<p>A dataset containing the full set ICD-10 diagnoses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>icd10codes
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 18761 rows and 2 columns.</p>

<hr>
<h2 id='icd9chapters'>International Classification of Diseases Revision 9 (ICD-9) chapters</h2><span id='topic+icd9chapters'></span>

<h3>Description</h3>

<p>A dataset containing the ICD-9 chapter titles - a top level description of diagnoses classes (or blocks)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>icd9chapters
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 19 rows and 3 columns.</p>

<hr>
<h2 id='icd9codes'>International Classification of Diseases Revision 9 (ICD-9) codes</h2><span id='topic+icd9codes'></span>

<h3>Description</h3>

<p>A dataset containing the full set ICD-9 diagnoses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>icd9codes
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 13679 rows and 2 columns.</p>

<hr>
<h2 id='ukb_centre'>Inserts UKB centre names into data</h2><span id='topic+ukb_centre'></span>

<h3>Description</h3>

<p>Inserts a column with centre name, <code>ukb_centre</code>, into the supplied data.frame. Useful if your UKB centre variable <code>uk_biobank_assessment_centre_0_0</code> has not been populated with named levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_centre(data, centre.var = "^uk_biobank_assessment_centre.*0_0")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_centre_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_centre_+3A_centre.var">centre.var</code></td>
<td>
<p>The UKB column containing numerically coded assessment centre. The default is a regular expression <code>"^uk_biobank_assessment_centre.*0_0"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with an additional column <code>ukb_centre</code> - UKB assessment centre names
</p>

<hr>
<h2 id='ukb_context'>Demographics of a UKB sample subset</h2><span id='topic+ukb_context'></span>

<h3>Description</h3>

<p>Describes a subset of the UKB sample, relative to a reference subsample, on the <a href="http://biobank.ctsu.ox.ac.uk/crystal/label.cgi?id=1001">UKB primary demographics</a> (sex, age, ethnicity, Townsend deprivation) and assessment centre and current employment status. The &quot;subset&quot; and &quot;reference&quot; samples are defined either by a variable of interest (<code>nonmiss.var</code> - those with data form the &quot;subset&quot; of interest and samples with missing data are the &quot;reference&quot; sample), or a logical vector (<code>subset.var</code> - where <code>TRUE</code> values define the &quot;subset&quot; and <code>FALSE</code> the &quot;reference&quot; samples) . This function is intended as an exploratory data analysis and quality control tool.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_context(data, nonmiss.var = NULL, subset.var = NULL,
  bar.position = "fill", sex.var = "^sex.*0_0",
  age.var = "^age_when_attended_assessment_centre.*0_0",
  socioeconomic.var = "^townsend_deprivation_index_at_recruitment.*0_0",
  ethnicity.var = "^ethnic_background.*0_0",
  employment.var = "^current_employment_status.*0_0",
  centre.var = "^uk_biobank_assessment_centre.*0_0")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_context_+3A_data">data</code></td>
<td>
<p>A UKB dataset constructed with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_nonmiss.var">nonmiss.var</code></td>
<td>
<p>The variable of interest which defines the &quot;subset&quot; (samples with data) and &quot;reference&quot; (samples without data, i.e., NA) samples.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_subset.var">subset.var</code></td>
<td>
<p>A logical vector defining a &quot;subset&quot; (<code>TRUE</code>) and &quot;reference&quot; subset (<code>FALSE</code>). Length must equal the number of rows in your <code>data</code>.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_bar.position">bar.position</code></td>
<td>
<p>This argument is passed to the <code>position</code> in <code>geom_bar</code>. The default value is <code>"fill"</code> which shows reference and subset of interest as proportions of the full dataset. Useful alternatives are <code>"stack"</code> for counts and <code>"dodge"</code> for side-by-side bars.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_sex.var">sex.var</code></td>
<td>
<p>The variable to be used for sex. Default value is the regular expression &quot;^sex.*0_0&quot;.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_age.var">age.var</code></td>
<td>
<p>The variable to be use for age. Default value is the regular expression &quot;^age_when_attended_assessment_centre.*0_0&quot;.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_socioeconomic.var">socioeconomic.var</code></td>
<td>
<p>The variable to be used for socioeconomic status. Default value is deprivation at baseline, the regular expression &quot;^townsend_deprivation_index_at_recruitment.*0_0&quot;.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_ethnicity.var">ethnicity.var</code></td>
<td>
<p>The variable to be used for ethnicity. Default value is the regular expression &quot;^ethnic_background.*0_0&quot;.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_employment.var">employment.var</code></td>
<td>
<p>The variable to be used for employment status. Default value is employment status at baseline &quot;^current_employment_status.*0_0&quot;.</p>
</td></tr>
<tr><td><code id="ukb_context_+3A_centre.var">centre.var</code></td>
<td>
<p>The variable to be used for assessment centre. Default value is the regular expression &quot;^uk_biobank_assessment_centre.*0_0&quot;.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_df">ukb_df</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Compare those with data to those without
ukb_context(my_ukb_data, nonmiss.var = "my_variable_of_interest")

# Define a subset of interest as a logical vector
subgroup_of_interest &lt;- (my_ukb_data$bmi &gt; 40 &amp; my_ukb_data$age &lt; 50)
ukb_context(my_ukb_data, subset.var = subgroup_of_interest)

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_defunct'>Defunct genetic metadata functions</h2><span id='topic+ukb_defunct'></span>

<h3>Description</h3>

<p>The genetic metadata functions were written to retrieve genetic metadata from the phenotype file for the <a href="http://biobank.ctsu.ox.ac.uk/crystal/label.cgi?id=199001">interim genotype release</a>. <strong>The fields retrieved became obselete when the full genotyping results (500K individuals) were released at the end of 2017.</strong> With the release of the full genotyping results, sample QC (ukb_sqc_v2.txt) and marker QC (ukb_snp_qc.txt) data are now supplied as separate files. The contents of these files, along with all other genetic files are described fully in <a href="https://biobank.ctsu.ox.ac.uk/crystal/refer.cgi?id=531">UKB Resource 531</a>.
</p>

<ul>
<li><p> ukb_gen_meta(data)
</p>
</li>
<li><p> ukb_gen_pcs(data)
</p>
</li>
<li><p> ukb_gen_excl(data)
</p>
</li>
<li><p> ukb_gen_rel(data)
</p>
</li>
<li><p> ukb_gen_het(data, all.het = FALSE)
</p>
</li>
<li><p> ukb_gen_excl_to_na(data, x, ukb.id = &quot;eid&quot;, data.frame = FALSE)
</p>
</li>
<li><p> ukb_gen_write_plink_excl(path)
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>ukb_defunct()
</code></pre>


<h3>Details</h3>

<p>See also lists new functionality that works with the files described in <a href="https://biobank.ctsu.ox.ac.uk/crystal/refer.cgi?id=531">UKB Resource 531</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_sqc_names">ukb_gen_sqc_names</a></code>, <code><a href="#topic+ukb_gen_rel_count">ukb_gen_rel_count</a></code>, <code><a href="#topic+ukb_gen_related_with_data">ukb_gen_related_with_data</a></code>, <code><a href="#topic+ukb_gen_samples_to_remove">ukb_gen_samples_to_remove</a></code>
</p>

<hr>
<h2 id='ukb_df'>Reads a UK Biobank phenotype fileset and returns a single dataset.</h2><span id='topic+ukb_df'></span>

<h3>Description</h3>

<p>A UK Biobank <em>fileset</em> includes a <em>.tab</em> file containing the raw data with field codes instead of variable names, an <em>.r</em> (<em>sic</em>) file containing code to read raw data (inserts categorical variable levels and labels), and an <em>.html</em> file containing tables mapping field code to variable name, and labels and levels for categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_df(fileset, path = ".", n_threads = "dt", data.pos = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_df_+3A_fileset">fileset</code></td>
<td>
<p>The prefix for a UKB fileset, e.g., ukbxxxx (for ukbxxxx.tab, ukbxxxx.r, ukbxxxx.html)</p>
</td></tr>
<tr><td><code id="ukb_df_+3A_path">path</code></td>
<td>
<p>The path to the directory containing your UKB fileset. The default value is the current directory.</p>
</td></tr>
<tr><td><code id="ukb_df_+3A_n_threads">n_threads</code></td>
<td>
<p>Either &quot;max&quot; (uses the number of cores, 'parallel::detectCores()'), &quot;dt&quot; (default - uses the data.table default, 'data.table::getDTthreads()'), or a numerical value (in which case n_threads is set to the supplied value, or 'parallel::detectCores()' if it is smaller).</p>
</td></tr>
<tr><td><code id="ukb_df_+3A_data.pos">data.pos</code></td>
<td>
<p>Locates the data in your .html file. The .html file is read into a list; the default value data.pos = 2 indicates the second item in the list. (The first item in the list is the title of the table). You will probably not need to change this value, but if the need arises you can open the .html file in a browser and identify where in the file the data is.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <strong>index</strong> and <strong>array</strong> from the UKB field code are preserved in the variable name, as two numbers separated by underscores at the end of the name e.g. <em>variable_index_array</em>. <strong>index</strong> refers the assessment instance (or visit). <strong>array</strong> captures multiple answers to the same &quot;question&quot;. See UKB documentation for detailed descriptions of <a href="http://biobank.ctsu.ox.ac.uk/crystal/instance.cgi?id=2">index</a> and <a href="http://biobank.ctsu.ox.ac.uk/crystal/help.cgi?cd=array">array</a>.
</p>


<h3>Value</h3>

<p>A dataframe with variable names in snake_case (lowercase and separated by an underscore).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_df_field">ukb_df_field</a></code> <code><a href="#topic+ukb_df_full_join">ukb_df_full_join</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Simply provide the stem of the UKB fileset.
# To read ukb1234.tab, ukb1234.r, ukb1234.html

my_ukb_data &lt;- ukb_df("ukb1234")


If you have multiple UKB filesets, read each then join with your preferred
method (ukb_df_full_join is
a thin wrapper around dplyr::full_join applied recursively with
purrr::reduce).

ukb1234_data &lt;- ukb_df("ukb1234")
ukb2345_data &lt;- ukb_df("ukb2345")
ukb3456_data &lt;- ukb_df("ukb3456")

ukb_df_full_join(ukb1234_data, ukb2345_data, ukb3456_data)

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_df_duplicated_name'>Checks for duplicated names within a UKB dataset</h2><span id='topic+ukb_df_duplicated_name'></span>

<h3>Description</h3>

<p>Checks for duplicated names within a UKB dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_df_duplicated_name(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_df_duplicated_name_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Duplicates *within* a UKB dataset are unlikely to occur, however, <code>ukb_df</code> creates variable names by combining a snake_case descriptor with the variable's **index** and **array**. If an index_array combination is incorrectly repeated in the original UKB data, this will result in a duplicated variable name. . See <code>vignette(topic = "explore-ukb-data", package = "ukbtools")</code> for further details.
</p>


<h3>Value</h3>

<p>Returns a named list of numeric vectors, one for each duplicated variable name. The numeric vectors contain the column indices of duplicates.
</p>

<hr>
<h2 id='ukb_df_field'>Makes a UKB data-field to variable name table for reference or lookup.</h2><span id='topic+ukb_df_field'></span>

<h3>Description</h3>

<p>Makes either a table of Data-Field and description, or a named vector handy for looking up descriptive name by column names in the UKB fileset tab file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_df_field(fileset, path = ".", data.pos = 2, as.lookup = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_df_field_+3A_fileset">fileset</code></td>
<td>
<p>The prefix for a UKB fileset, e.g., ukbxxxx (for ukbxxxx.tab, ukbxxxx.r, ukbxxxx.html)</p>
</td></tr>
<tr><td><code id="ukb_df_field_+3A_path">path</code></td>
<td>
<p>The path to the directory containing your UKB fileset. The default value is the current directory.</p>
</td></tr>
<tr><td><code id="ukb_df_field_+3A_data.pos">data.pos</code></td>
<td>
<p>Locates the data in your .html file. The .html file is read into a list; the default value data.pos = 2 indicates the second item in the list. (The first item in the list is the title of the table). You will probably not need to change this value, but if the need arises you can open the .html file in a browser and identify where in the file the data is.</p>
</td></tr>
<tr><td><code id="ukb_df_field_+3A_as.lookup">as.lookup</code></td>
<td>
<p>If set to TRUE, returns a named <code>vector</code>. The default <code>as.look = FALSE</code> returns a dataframe with columns: field.showcase (as used in the UKB online showcase), field.data (as used in the tab file), name (descriptive name created by <code><a href="#topic+ukb_df">ukb_df</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame with columns <code>field.showcase</code>, <code>field.html</code>, <code>field.tab</code>, <code>names</code>. <code>field.showcase</code> is how the field appears in the online <a href="http://biobank.ctsu.ox.ac.uk/crystal/">UKB showcase</a>; <code>field.html</code> is how the field appears in the html file in your UKB fileset; <code>field.tab</code> is how the field appears in the tab file in your fileset; and <code>names</code> is the descriptive name that <code><a href="#topic+ukb_df">ukb_df</a></code> assigns to the variable. If <code>as.lookup = TRUE</code>, the function returns a named character vector of the descriptive names.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_df">ukb_df</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# UKB field-to-description for ukb1234.tab, ukb1234.r, ukb1234.html

ukb_df_field("ukb1234")

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_df_full_join'>Recursively join a list of UKB datasets</h2><span id='topic+ukb_df_full_join'></span>

<h3>Description</h3>

<p>A thin wrapper around <code>purrr::reduce</code> and <code>dplyr::full_join</code> to merge multiple UKB datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_df_full_join(..., by = "eid")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_df_full_join_+3A_...">...</code></td>
<td>
<p>Supply comma separated unquoted names of to-be-merged UKB datasets (created with <code><a href="#topic+ukb_df">ukb_df</a></code>). Arguments are passed to <code>list</code>.</p>
</td></tr>
<tr><td><code id="ukb_df_full_join_+3A_by">by</code></td>
<td>
<p>Variable used to merge multiple dataframes (default = &quot;eid&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes a comma separated list of unquoted datasets. By explicitly setting the join key to &quot;eid&quot; only (Default value of the <code>by</code> parameter), any additional variables common to any two tables will have &quot;.x&quot; and &quot;.y&quot; appended to their names. If you are satisfied the additional variables are identical to the original, the copies can be safely deleted. For example, if <code>setequal(my_ukb_data$var, my_ukb_data$var.x)</code> is <code>TRUE</code>, then my_ukb_data$var.x can be dropped. A <code>dlyr::full_join</code> is like the set operation union in that all observations from all tables are included, i.e., all samples are included even if they are not included in all datasets.
</p>
<p>NB. <code>ukb_df_full_join</code> will fail if any variable names are repeated **within** a single UKB dataset. This is unlikely to occur, however, <code>ukb_df</code> creates variable names by combining a snake_case descriptor with the variable's **index** and **array**. If an index_array combination is incorrectly repeated, this will result in a duplicated variable. If the join fails, you can use <code><a href="#topic+ukb_df_duplicated_name">ukb_df_duplicated_name</a></code> to find duplicated names. See <code>vignette(topic = "explore-ukb-data", package = "ukbtools")</code> for further details.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_df_duplicated_name">ukb_df_duplicated_name</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# If you have multiple UKB filesets, tidy then merge them.

ukb1234_data &lt;- ukb_df("ukb1234")
ukb2345_data &lt;- ukb_df("ukb2345")
ukb3456_data &lt;- ukb_df("ukb3456")

my_ukb_data &lt;- ukb_df_full_join(ukb1234_data, ukb2345_data, ukb3456_data)

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_gen_excl'>Sample exclusions</h2><span id='topic+ukb_gen_excl'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
</p>
<p>This list of sample exclusions includes UKB's &quot;recommended&quot;, &quot;affymetrix quality control&quot;, and &quot;genotype quality control&quot; exclusions. UKB have published <a href="http://www.ukbiobank.ac.uk/wp-content/uploads/2014/04/UKBiobank_genotyping_QC_documentation-web.pdf">full details of genotyping and quality control</a> for the interim genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_excl(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_excl_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# For a vector of IDs
recommended_excl_ids &lt;- ukb_gen_excl(my_ukb_df)

## End(Not run)
</code></pre>

<hr>
<h2 id='ukb_gen_excl_to_na'>Inserts NA into phenotype for genetic metadata exclusions</h2><span id='topic+ukb_gen_excl_to_na'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
</p>
<p>Replaces data values in a vector (a UKB phenotype) with <code>NA</code> where the sample is to-be-excluded, i.e., is either a UKB recommended exclusion, a heterozygosity outlier, a genetic ethnicity outlier, or a randomly-selected member of a related pair.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_excl_to_na(data, x, ukb.id = "eid", data.frame = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_excl_to_na_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_gen_excl_to_na_+3A_x">x</code></td>
<td>
<p>The phenotype to be updated (as it is named in <code>data</code>) e.g. &quot;height&quot;</p>
</td></tr>
<tr><td><code id="ukb_gen_excl_to_na_+3A_ukb.id">ukb.id</code></td>
<td>
<p>The name of the ID variable in <code>data</code>. Default is &quot;eid&quot;</p>
</td></tr>
<tr><td><code id="ukb_gen_excl_to_na_+3A_data.frame">data.frame</code></td>
<td>
<p>A logical vector indicating whether to return a vector or a data.frame (header: id, meta_excl, pheno, pheno_meta_na) containing the original and updated variable. Default = FALSE returns a vector.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_write_plink_excl">ukb_gen_write_plink_excl</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
my_ukb_data$height_excl_na &lt;- ukb_gen_excl_to_na(my_ukb_data, x = "height")

## End(Not run)
</code></pre>

<hr>
<h2 id='ukb_gen_het'>Heterozygosity outliers</h2><span id='topic+ukb_gen_het'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
</p>
<p>Heterozygosity outliers are typically removed from genetic association analyses. This function returns either a vector of heterozygosity outliers to remove (+/- 3sd from mean heterozygosity), or a data frame with heterozygosity scores for all samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_het(data, all.het = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_het_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_gen_het_+3A_all.het">all.het</code></td>
<td>
<p>Set <code>all.het = TRUE</code> for heterozygosity scores for all samples. By default <code>all.het = FALSE</code> returns a vector of sample IDs for individuals +/-3SD from the mean heterozygosity.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>UKB have published <a href="http://www.ukbiobank.ac.uk/wp-content/uploads/2014/04/UKBiobank_genotyping_QC_documentation-web.pdf">full details of genotyping and quality control</a> for the interim genotype data.
</p>


<h3>Value</h3>

<p>A vector of IDs if <code>all.het = FALSE</code> (default), or a dataframe with ID, heterozygosity and PCA-corrected heterozygosity if <code>all.het = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#' # Heterozygosity outliers (+/-3SD)
outlier_het_ids &lt;- ukb_gen_het(my_ukb_data)

# Retrieve all raw and pca-corrected heterozygosity scores
ukb_het &lt;- ukb_gen_het(my_ukb_data, all.het = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ukb_gen_meta'>Genetic metadata</h2><span id='topic+ukb_gen_meta'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
</p>
<p>UKB have published <a href="http://www.ukbiobank.ac.uk/wp-content/uploads/2014/04/UKBiobank_genotyping_QC_documentation-web.pdf">full details of genotyping and quality control</a> for the interim genotype data. This function retrieves UKB assessment centre codes and assessment centre names, genetic ethnic grouping, genetically-determined sex, missingness, UKB recommended genomic analysis exclusions, BiLeve unrelatedness indicator, and BiLeve Affymetrix and genotype quality control.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_meta(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_meta_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='ukb_gen_pcs'>Genetic principal components</h2><span id='topic+ukb_gen_pcs'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
</p>
<p>These are the principal components derived on the UK Biobank subsample with interim genotype data. UKB have published <a href="http://www.ukbiobank.ac.uk/wp-content/uploads/2014/04/UKBiobank_genotyping_QC_documentation-web.pdf">full details of genotyping and quality control</a> for the interim genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_pcs(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_pcs_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='ukb_gen_read_fam'>Reads a PLINK format fam file</h2><span id='topic+ukb_gen_read_fam'></span>

<h3>Description</h3>

<p>This is wrapper for read_table that reads a basic PLINK fam file. For plink hard-called data, it may be useful to use the fam file ids as a filter for your phenotype and covariate data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_read_fam(file, col.names = c("FID", "IID", "paternalID",
  "maternalID", "sex", "phenotype"), na.strings = "-9")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_read_fam_+3A_file">file</code></td>
<td>
<p>A path to a fam file.</p>
</td></tr>
<tr><td><code id="ukb_gen_read_fam_+3A_col.names">col.names</code></td>
<td>
<p>A character vector of column names. Default: c(&quot;FID&quot;, &quot;IID&quot;, &quot;paternalID&quot;, &quot;maternalID&quot;, &quot;sex&quot;, &quot;phenotype&quot;)</p>
</td></tr>
<tr><td><code id="ukb_gen_read_fam_+3A_na.strings">na.strings</code></td>
<td>
<p>Character vector of strings to use for missing values. Default &quot;-9&quot;. Set this option to character() to indicate no missing values.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_read_sample">ukb_gen_read_sample</a></code> to read a sample file
</p>

<hr>
<h2 id='ukb_gen_read_sample'>Reads an Oxford format sample file</h2><span id='topic+ukb_gen_read_sample'></span>

<h3>Description</h3>

<p>This is a wrapper for <code>read_table</code> that reads an Oxford format .sample file. If you use the unedited sample file as supplied with your genetic data, you should only need to specify the first argument, file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_read_sample(file, col.names = c("id_1", "id_2", "missing"),
  row.skip = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_read_sample_+3A_file">file</code></td>
<td>
<p>A path to a sample file.</p>
</td></tr>
<tr><td><code id="ukb_gen_read_sample_+3A_col.names">col.names</code></td>
<td>
<p>A character vector of column names. Default: c(&quot;id_1&quot;, &quot;id_2&quot;, &quot;missing&quot;)</p>
</td></tr>
<tr><td><code id="ukb_gen_read_sample_+3A_row.skip">row.skip</code></td>
<td>
<p>Number of lines to skip before reading data.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_read_fam">ukb_gen_read_fam</a></code> to read a fam file
</p>

<hr>
<h2 id='ukb_gen_rel'>Creates a table of related individuals</h2><span id='topic+ukb_gen_rel'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
</p>
<p>Makes a data.frame containing all related individuals with columns UKB ID, pair ID, <a href="http://people.virginia.edu/~wc9c/KING/manual.html">KING kinship coefficient</a>, and proportion of alleles IBS = 0. UKB have published <a href="http://www.ukbiobank.ac.uk/wp-content/uploads/2014/04/UKBiobank_genotyping_QC_documentation-web.pdf">full details of genotyping and quality control</a> including details on relatedness calculations for the interim genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_rel(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_rel_+3A_data">data</code></td>
<td>
<p>A UKB dataset created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_rel_count">ukb_gen_rel_count</a></code>
</p>

<hr>
<h2 id='ukb_gen_rel_count'>Relatedness count</h2><span id='topic+ukb_gen_rel_count'></span>

<h3>Description</h3>

<p>Creates a summary count table of the number of individuals and pairs at each degree of relatedness that occurs in the UKB sample, and an optional plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_rel_count(data, plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_rel_count_+3A_data">data</code></td>
<td>
<p>A dataframe of the genetic relatedness data including <a href="http://people.virginia.edu/~wc9c/KING/manual.html">KING kinship coefficient</a>, and proportion of alleles IBS = 0. See Details.</p>
</td></tr>
<tr><td><code id="ukb_gen_rel_count_+3A_plot">plot</code></td>
<td>
<p>Logical indicating whether to plot relatedness figure. Default = FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Use UKB supplied program 'ukbgene' to retrieve genetic relatedness data file ukbA_rel_sP.txt. See <a href="http://biobank.ctsu.ox.ac.uk/crystal/refer.cgi?id=664">UKB Resource 664</a>. The count and plot include individuals with IBS0 &gt;= 0.
</p>


<h3>Value</h3>

<p>If <code>plot = FALSE</code> (default), a count of individuals and pairs at each level of relatedness. If <code>plot = TRUE</code>, reproduces the scatterplot of genetic relatedness against proportion of SNPs shared IBS=0 (each point representing a pair of related UKB individuals) from the <a href="http://www.ukbiobank.ac.uk/wp-content/uploads/2014/04/UKBiobank_genotyping_QC_documentation-web.pdf">genotyping and quality control</a> documentation.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_related_with_data">ukb_gen_related_with_data</a></code>, <code><a href="#topic+ukb_gen_samples_to_remove">ukb_gen_samples_to_remove</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Use UKB supplied program `ukbgene` to retrieve genetic relatedness file ukbA_rel_sP.txt.
See \href{http://biobank.ctsu.ox.ac.uk/crystal/refer.cgi?id=664}{UKB Resource 664}.
With the whitespace delimited file read into R as e.g. ukb_relatedness,
generate a dataframe of counts or a plot as follows:

ukb_gen_rel_count(ukb_relatedness)
ukb_gen_rel_count(ukb_relatedness, plot = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='ukb_gen_related_with_data'>Subset of the UKB relatedness dataframe with data</h2><span id='topic+ukb_gen_related_with_data'></span>

<h3>Description</h3>

<p>Subset of the UKB relatedness dataframe with data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_related_with_data(data, ukb_with_data, cutoff = 0.0884)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_related_with_data_+3A_data">data</code></td>
<td>
<p>The UKB relatedness data as a dataframe (header: ID1, ID2, HetHet, IBS0, Kinship)</p>
</td></tr>
<tr><td><code id="ukb_gen_related_with_data_+3A_ukb_with_data">ukb_with_data</code></td>
<td>
<p>A character vector of ukb eids with data on the phenotype of interest</p>
</td></tr>
<tr><td><code id="ukb_gen_related_with_data_+3A_cutoff">cutoff</code></td>
<td>
<p>KING kingship coefficient cutoff (default 0.0884 includes pairs with greater than 3rd-degree relatedness)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe (header: ID1, ID2, HetHet, IBS0, Kinship) for the subset of individuals with data.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_rel_count">ukb_gen_rel_count</a></code>, <code><a href="#topic+ukb_gen_samples_to_remove">ukb_gen_samples_to_remove</a></code>
</p>

<hr>
<h2 id='ukb_gen_samples_to_remove'>Related samples (with data on the variable of interest) to remove</h2><span id='topic+ukb_gen_samples_to_remove'></span>

<h3>Description</h3>

<p>There are many ways to remove related individuals from phenotypic data for genetic analyses. You could simply exclude all individuals indicated as having &quot;excess relatedness&quot; and include those &quot;used in pca calculation&quot; (these variables are included in the sample QC data, ukb_sqc_v2.txt) - see details. This list is based on the complete dataset, and possibly removes more samples than you need to for your phenotype of interest. Ideally, you want a maximum independent set, i.e., to remove the minimum number of individuals with data on the phenotype of interest, so that no pair exceeds some cutoff for relatedness. <code>ukb_gen_samples_to_remove</code> returns a list of samples to remove in to achieve a maximal set of unrelateds for a given phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_samples_to_remove(data, ukb_with_data, cutoff = 0.0884)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_samples_to_remove_+3A_data">data</code></td>
<td>
<p>The UKB relatedness data as a dataframe (header: ID1, ID2, HetHet, IBS0, Kinship)</p>
</td></tr>
<tr><td><code id="ukb_gen_samples_to_remove_+3A_ukb_with_data">ukb_with_data</code></td>
<td>
<p>A character vector of ukb eids with data on the phenotype of interest</p>
</td></tr>
<tr><td><code id="ukb_gen_samples_to_remove_+3A_cutoff">cutoff</code></td>
<td>
<p>KING kingship coefficient cutoff (default 0.0884 includes pairs with greater than 3rd-degree relatedness)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Trims down the UKB relatedness data before selecting individuals to exclude, using the algorithm: step 1. remove pairs below KING kinship coefficient 0.0884 (3rd-degree or less related, by default. Can be set with <code>cutoff</code> argument), and any pairs if either member does not have data on the phenotype of interest. The user supplies a vector of samples with data. step 2. count the number of &quot;connections&quot; (or relatives) each participant has and add to &quot;samples to exclude&quot; the individual with the most connections. This is the greedy part of the algorithm. step 3. repeat step 2 till all remaining participants only have 1 connection, then add one random member of each remaining pair to &quot;samples to exclude&quot; (adds all those listed under ID2)
</p>
<p><em>Another approach from the UKB email distribution list:</em>
</p>
<p>To: UKB-GENETICS@JISCMAIL.AC.UK
Date:    Wed, 26 Jul 2017 17:06:01 +0100
<strong>Subject: A list of unrelated samples</strong>
</p>
<p>(...) you could use the list of samples which we used to calculate the PCs,
which is a (maximal) subset of unrelated participants after applying some QC
filtering. Please read supplementary Section S3.3.2 for details. You can
find the list of samples using the &quot;used.in.pca.calculation&quot; column in the
sample-QC file (ukb_sqc_v2.txt) (...). Note that this set contains diverse
ancestries. If you take the intersection with the white British ancestry
subset you get ~337,500 unrelated samples.
</p>


<h3>Value</h3>

<p>An integer vector of UKB IDs to remove.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_rel_count">ukb_gen_rel_count</a></code>, <code><a href="#topic+ukb_gen_related_with_data">ukb_gen_related_with_data</a></code>
</p>

<hr>
<h2 id='ukb_gen_sqc_names'>Sample QC column names</h2><span id='topic+ukb_gen_sqc_names'></span>

<h3>Description</h3>

<p>The UKB sample QC file has no header on it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_sqc_names(data, col_names_only = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_sqc_names_+3A_data">data</code></td>
<td>
<p>The UKB ukb_sqc_v2.txt data as dataframe. (Not necessary if column names only are required)</p>
</td></tr>
<tr><td><code id="ukb_gen_sqc_names_+3A_col_names_only">col_names_only</code></td>
<td>
<p>If <code>TRUE</code> returns a character vector of column names (<code>data</code> argument not required). Useful if you would like to supply as header when reading in your sample QC data. If <code>FALSE</code> (Default), returns the supplied dataframe with column names (Checks number of columns in supplied data. See Details.).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>From <a href="https://biobank.ctsu.ox.ac.uk/crystal/refer.cgi?id=531">UKB Resource 531</a>: There are currently 2 versions of this file (UKB ukb_sqc_v2.txt) in circulation. The newer version is described below and contains column headers on the first row. The older (deprecated) version lacks the column headers and has two additional Affymetrix internal values prefixing the columns listed below.
</p>


<h3>Value</h3>

<p>A sample QC dataframe with column names, or a character vector of column names if <code>col_names_only = TRUE</code>.
</p>

<hr>
<h2 id='ukb_gen_write_bgenie'>Writes a BGENIE format phenotype or covariate file.</h2><span id='topic+ukb_gen_write_bgenie'></span>

<h3>Description</h3>

<p>Writes a space-delimited file with a header, missing character set to &quot;-999&quot;, and observations (i.e. UKB subject ids) in sample file order. Use this function to write phenotype and covariate files for downstream genetic analysis in <a href="https://jmarchini.org/bgenie/">BGENIE</a> - the format is the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_write_bgenie(x, ukb.sample, ukb.variables, path, ukb.id = "eid",
  na.strings = "-999")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_write_bgenie_+3A_x">x</code></td>
<td>
<p>A UKB dataset.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_bgenie_+3A_ukb.sample">ukb.sample</code></td>
<td>
<p>A UKB sample file.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_bgenie_+3A_ukb.variables">ukb.variables</code></td>
<td>
<p>A character vector of either the phenotypes for a BGENIE phenotype file, or covariates for a BGENIE covariate file.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_bgenie_+3A_path">path</code></td>
<td>
<p>A path to a file.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_bgenie_+3A_ukb.id">ukb.id</code></td>
<td>
<p>The eid variable name (default = &quot;eid&quot;).</p>
</td></tr>
<tr><td><code id="ukb_gen_write_bgenie_+3A_na.strings">na.strings</code></td>
<td>
<p>Character string to be used for missing value in output file. Default = &quot;-999&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses a <code>dplyr::left_join</code> to the sample file to match sample file order. Any IDs in the sample file not included in the phenotype or covariate data will be missing for all variables selected. See <a href="https://jmarchini.org/bgenie-usage/">BGENIE usage</a> for descriptions of the <code>--pheno</code> and <code>--covar</code> flags to read phenotype and covariate data into BGENIE.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_read_sample">ukb_gen_read_sample</a></code> to read a sample file, <code><a href="#topic+ukb_gen_excl_to_na">ukb_gen_excl_to_na</a></code> to update a phenotype with NAs for samples to-be-excluded based on genetic metadata, and <code><a href="#topic+ukb_gen_write_plink">ukb_gen_write_plink</a></code> to write phenotype and covariate files to PLINK format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Automatically sorts observations to match UKB sample file and writes missing values as -999

my_ukb_sample &lt;- ukb_gen_read_sample("ukb.sample")

ukb_gen_write_bgenie(
   my_ukb_data,
   ukb.sample = my_ukb_sample,
   ukb.variables = c("height", "weight", "iq")
   path = "my_ukb_bgenie.pheno",
)

ukb_gen_write_bgenie(
   my_ukb_data,
   ukb.sample = my_ukb_sample,
   ukb.variables = c("age", "socioeconomic_status", "genetic_pcs")
   path = "my_ukb_bgenie.cov",
)

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_gen_write_plink'>Writes a PLINK format phenotype or covariate file</h2><span id='topic+ukb_gen_write_plink'></span>

<h3>Description</h3>

<p>This function writes a space-delimited file with header, with the obligatory first two columns FID and IID. Use this function to write phenotype and covariate files for downstream genetic analysis in <a href="https://www.cog-genomics.org/plink2">plink</a> - the format is the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_write_plink(x, path, ukb.variables, ukb.id = "eid",
  na.strings = "NA")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_write_plink_+3A_x">x</code></td>
<td>
<p>A UKB dataset.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_plink_+3A_path">path</code></td>
<td>
<p>A path to a file.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_plink_+3A_ukb.variables">ukb.variables</code></td>
<td>
<p>A character vector of either the phenotypes for a PLINK phenotype file, or covariates for a PLINK covariate file.</p>
</td></tr>
<tr><td><code id="ukb_gen_write_plink_+3A_ukb.id">ukb.id</code></td>
<td>
<p>The id variable name (default = &quot;eid&quot;).</p>
</td></tr>
<tr><td><code id="ukb_gen_write_plink_+3A_na.strings">na.strings</code></td>
<td>
<p>String used for missing values. Defaults to NA.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function writes the id variable in your dataset to the first two columns of the output file with the names FID and IID - you do not need to have two id columns in the data.frame passed to the argument <code>x</code>. Use the <code>--pheno-name</code> and <code>--covar-name</code> PLINK flags to select columns by name. See the PLINK documentation for the <code>--pheno</code>, <code>--mpheno</code>, <code>--pheno-name</code>, and <code>--covar</code>, <code>--covar-name</code>, <code>--covar-number</code> flags.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_read_sample">ukb_gen_read_sample</a></code> to read a sample file, and <code><a href="#topic+ukb_gen_write_bgenie">ukb_gen_write_bgenie</a></code> to write phenotype and covariate files to BGENIE format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Automatically inserts FID IID columns required by PLINK

ukb_gen_write_plink(
   my_ukb_data,
   path = "my_ukb_plink.pheno",
   ukb.variables = c("height", "weight", "iq")
)

ukb_gen_write_plink(
   my_ukb_data,
   path = "my_ukb_plink.cov",
   ukb.variables = c("age", "socioeconomic_status", "genetic_pcs")
)

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_gen_write_plink_excl'>Writes a PLINK format file for combined exclusions</h2><span id='topic+ukb_gen_write_plink_excl'></span>

<h3>Description</h3>

<p><strong>Defunct. See help(&quot;ukb_defunct&quot;)</strong>.
Writes a combined exclusions file including UKB recommended exclusions, heterozygosity exclusions (+/- 3*sd from mean), genetic ethnicity exclusions (based on the UKB genetic ethnic grouping variable, field 1002), and relatedness exclusions (a randomly-selected member of each related pair). For exclusion of individuals from a genetic analysis, the PLINK flag <code>--remove</code> accepts a space/tab-delimited text file with family IDs in the first column and within-family IDs in the second column (i.e., FID IID), without a header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_gen_write_plink_excl(path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_gen_write_plink_excl_+3A_path">path</code></td>
<td>
<p>A path to a file.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_gen_meta">ukb_gen_meta</a></code>, <code><a href="#topic+ukb_gen_pcs">ukb_gen_pcs</a></code> which retrieve variables to be included in a covariate file. <code><a href="#topic+ukb_gen_excl_to_na">ukb_gen_excl_to_na</a></code> to update a phenotype with NAs for samples to-be-excluded based on genetic metadata, and <code><a href="#topic+ukb_gen_write_plink">ukb_gen_write_plink</a></code> and <code><a href="#topic+ukb_gen_write_bgenie">ukb_gen_write_bgenie</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Supply name of a file to write PLINK format combined exclusions
ukb_gen_write_plink_excl("combined_exclusions.txt")

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_icd_code_meaning'>Retrieves description for a ICD code.</h2><span id='topic+ukb_icd_code_meaning'></span>

<h3>Description</h3>

<p>Retrieves description for a ICD code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_icd_code_meaning(icd.code, icd.version = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_icd_code_meaning_+3A_icd.code">icd.code</code></td>
<td>
<p>The ICD diagnosis code to be looked up.</p>
</td></tr>
<tr><td><code id="ukb_icd_code_meaning_+3A_icd.version">icd.version</code></td>
<td>
<p>The ICD version (or revision) number, 9 or 10.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_icd_diagnosis">ukb_icd_diagnosis</a></code>, <code><a href="#topic+ukb_icd_keyword">ukb_icd_keyword</a></code>, <code><a href="#topic+ukb_icd_prevalence">ukb_icd_prevalence</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ukb_icd_code_meaning(icd.code = "I74", icd.version = 10)

</code></pre>

<hr>
<h2 id='ukb_icd_diagnosis'>Retrieves diagnoses for an individual.</h2><span id='topic+ukb_icd_diagnosis'></span>

<h3>Description</h3>

<p>Retrieves diagnoses for an individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_icd_diagnosis(data, id, icd.version = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_icd_diagnosis_+3A_data">data</code></td>
<td>
<p>A UKB dataset (or subset) created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_icd_diagnosis_+3A_id">id</code></td>
<td>
<p>An individual's id, i.e., their unique eid reference number.</p>
</td></tr>
<tr><td><code id="ukb_icd_diagnosis_+3A_icd.version">icd.version</code></td>
<td>
<p>The ICD version (or revision) number, 9 or 10.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_df">ukb_df</a></code>, <code><a href="#topic+ukb_icd_code_meaning">ukb_icd_code_meaning</a></code>, <code><a href="#topic+ukb_icd_keyword">ukb_icd_keyword</a></code>, <code><a href="#topic+ukb_icd_prevalence">ukb_icd_prevalence</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ukb_icd_diagnosis(my_ukb_data, id = "123456", icd.version = 10)

## End(Not run)

</code></pre>

<hr>
<h2 id='ukb_icd_freq_by'>Frequency of an ICD diagnosis by a target variable</h2><span id='topic+ukb_icd_freq_by'></span>

<h3>Description</h3>

<p>Produces either a dataframe of diagnosis frequencies or a plot. For a
quantitative reference variable (e.g. BMI), the plot shows frequency of
diagnosis within each group (deciles of the reference
variable by default) at the (max - min) / 2 for
each group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_icd_freq_by(data, reference.var, n.groups = 10,
  icd.code = c("^(I2[0-5])", "^(I6[0-9])",
  "^(J09|J1[0-9]|J2[0-2]|P23|U04)"),
  icd.labels = c("coronary artery disease", "cerebrovascular disease",
  "lower respiratory tract infection"), plot.title = "",
  legend.col = 1, legend.pos = "right", icd.version = 10,
  freq.plot = FALSE, reference.lab = "Reference variable",
  freq.lab = "UKB disease frequency")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_icd_freq_by_+3A_data">data</code></td>
<td>
<p>A UKB dataset (or subset) created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_reference.var">reference.var</code></td>
<td>
<p>UKB ICD frequencies will be calculated by levels of this variable. If continuous, by default it is cut into 10 intervals of approximately equal size (set with n.groups).</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_n.groups">n.groups</code></td>
<td>
<p>Number of approximately equal-sized groups to split a continuous variable into.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_icd.code">icd.code</code></td>
<td>
<p>ICD disease code(s) e.g. &quot;I74&quot;. Use a regular expression to specify a broader set of diagnoses, e.g. &quot;I&quot; captures all Diseases of the circulatory system, I00-I99, &quot;C|D[0-4].&quot; captures all Neoplasms, C00-D49. Default is the WHO top 3 causes of death globally in 2015, see <a href="http://www.who.int/healthinfo/global_burden_disease/GlobalCOD_method_2000_2015.pdf?ua=1">http://www.who.int/healthinfo/global_burden_disease/GlobalCOD_method_2000_2015.pdf?ua=1</a>. Note. If you specify 'icd.codes', you must supply corresponding labels to 'icd.labels'.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_icd.labels">icd.labels</code></td>
<td>
<p>Character vector of ICD labels for the plot legend. Default = V1 to VN.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_plot.title">plot.title</code></td>
<td>
<p>Title for the plot. Default describes the default icd.codes, WHO top 6 cause of death 2015.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_legend.col">legend.col</code></td>
<td>
<p>Number of columns for the legend. (Default = 1).</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_legend.pos">legend.pos</code></td>
<td>
<p>Legend position, default = &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_icd.version">icd.version</code></td>
<td>
<p>The ICD version (or revision) number, 9 or 10.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_freq.plot">freq.plot</code></td>
<td>
<p>If TRUE returns a plot of ICD diagnosis by target variable. If FALSE (default) returns a dataframe.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_reference.lab">reference.lab</code></td>
<td>
<p>An x-axis title for the reference variable.</p>
</td></tr>
<tr><td><code id="ukb_icd_freq_by_+3A_freq.lab">freq.lab</code></td>
<td>
<p>A y-axis title for disease frequency.</p>
</td></tr>
</table>

<hr>
<h2 id='ukb_icd_keyword'>Retrieves diagnoses containing a description.</h2><span id='topic+ukb_icd_keyword'></span>

<h3>Description</h3>

<p>Returns a dataframe of ICD code and descriptions for all entries including any supplied keyword.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_icd_keyword(description, icd.version = 10, ignore.case = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_icd_keyword_+3A_description">description</code></td>
<td>
<p>A character vector of one or more keywords to be looked up in the ICD descriptions, e.g., &quot;cardio&quot;, c(&quot;cardio&quot;, &quot;lymphoma&quot;). Each keyword can be a regular expression, e.g. &quot;lymph*&quot;.</p>
</td></tr>
<tr><td><code id="ukb_icd_keyword_+3A_icd.version">icd.version</code></td>
<td>
<p>The ICD version (or revision) number, 9 or 10. Default = 10.</p>
</td></tr>
<tr><td><code id="ukb_icd_keyword_+3A_ignore.case">ignore.case</code></td>
<td>
<p>If 'TRUE' (default), case is ignored during matching; if 'FALSE', the matching is case sensitive.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_icd_diagnosis">ukb_icd_diagnosis</a></code>, <code><a href="#topic+ukb_icd_code_meaning">ukb_icd_code_meaning</a></code>, <code><a href="#topic+ukb_icd_prevalence">ukb_icd_prevalence</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ukb_icd_keyword("cardio", icd.version = 10)

</code></pre>

<hr>
<h2 id='ukb_icd_prevalence'>Returns the prevalence for an ICD diagnosis</h2><span id='topic+ukb_icd_prevalence'></span>

<h3>Description</h3>

<p>Returns the prevalence for an ICD diagnosis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukb_icd_prevalence(data, icd.code, icd.version = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ukb_icd_prevalence_+3A_data">data</code></td>
<td>
<p>A UKB dataset (or subset) created with <code><a href="#topic+ukb_df">ukb_df</a></code>.</p>
</td></tr>
<tr><td><code id="ukb_icd_prevalence_+3A_icd.code">icd.code</code></td>
<td>
<p>An ICD disease code e.g. &quot;I74&quot;. Use a regular expression to specify a broader set of diagnoses, e.g. &quot;I&quot; captures all Diseases of the circulatory system, I00-I99, &quot;C|D[0-4].&quot; captures all Neoplasms, C00-D49.</p>
</td></tr>
<tr><td><code id="ukb_icd_prevalence_+3A_icd.version">icd.version</code></td>
<td>
<p>The ICD version (or revision) number, 9 or 10. Default = 10.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+ukb_icd_diagnosis">ukb_icd_diagnosis</a></code>, <code><a href="#topic+ukb_icd_code_meaning">ukb_icd_code_meaning</a></code>, <code><a href="#topic+ukb_icd_keyword">ukb_icd_keyword</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ICD-10 code I74, Arterial embolism and thrombosis
ukb_icd_prevalence(my_ukb_data, icd.code = "I74")

# ICD-10 chapter 9, disease block I00I99, Diseases of the circulatory system
ukb_icd_prevalence(my_ukb_data, icd.code = "I")

# ICD-10 chapter 2, C00-D49, Neoplasms
ukb_icd_prevalence(my_ukb_data, icd.code = "C|D[0-4].")

## End(Not run)

</code></pre>

<hr>
<h2 id='ukbcentre'>UKB assessment centre</h2><span id='topic+ukbcentre'></span>

<h3>Description</h3>

<p>A dataset containing the 22 assessment centres (as well as pilot test centre and a revisit centre)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ukbcentre
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 24 rows and 2 columns.</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
