<!DOCTYPE html><html lang="en-US"><head><title>Help for package umiAnalyzer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {umiAnalyzer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addMetaData'><p>Add metaData</p></a></li>
<li><a href='#addUmiSample'><p>Add UMI sample to an existing experiment object</p></a></li>
<li><a href='#AmpliconHeatmap'><p>Amplicon heatmap</p></a></li>
<li><a href='#AmpliconPlot'><p>Generate Amplicon plots</p></a></li>
<li><a href='#BarcodeFamilyHistogram'><p>Consensus depth histograms</p></a></li>
<li><a href='#beta_binom'><p>Beta binomial model</p></a></li>
<li><a href='#callVariants'><p>callVariants using beta binomial distribution</p></a></li>
<li><a href='#createUmiExperiment'><p>Method for creating a UMI experiment object</p></a></li>
<li><a href='#createUMIexperiment_Debarcer'><p>Method for creating a UMI experiment object</p></a></li>
<li><a href='#createUmiSample'><p>createUmiSample</p></a></li>
<li><a href='#createUMIsample_Debarcer'><p>Method for creating a UMIsample object</p></a></li>
<li><a href='#download_template'><p>Download meta data template</p></a></li>
<li><a href='#filterUmiObject'><p>Method for filtering UMIexperiment and sample objects</p></a></li>
<li><a href='#filterVariants'><p>Filter variants based on p values or depth</p></a></li>
<li><a href='#findConsensusReads'><p>Find consensus reads</p>
A function to analyze consensus read tables generated with parseBamFiles or
a UMIexperiment object containing reads.</a></li>
<li><a href='#generateVCF'><p>Generate VCF file from UMI sample or UMI experiment object</p></a></li>
<li><a href='#getFilteredData'><p>Method for retrieving filtered data</p></a></li>
<li><a href='#getMetaData'><p>Retrieve meta data by name.</p></a></li>
<li><a href='#importBedFile'><p>Import bed file</p></a></li>
<li><a href='#importDesign'><p>Import experimental design meta data such as replicates, treatments, categorical variables.</p></a></li>
<li><a href='#mergeAssays'><p>Merge assays</p></a></li>
<li><a href='#parseBamFiles'><p>Function to parse bam files</p></a></li>
<li><a href='#QCplot'><p>Generate QC plots</p></a></li>
<li><a href='#runUmiVisualizer'><p>Function to run the umiVisualizer shiny app</p></a></li>
<li><a href='#saveConsData'><p>Save consensus data</p></a></li>
<li><a href='#simsen'><p>UMIexperiment data generated with SiMSen-Seq</p></a></li>
<li><a href='#timeSeriesGrid'><p>Plot time series data</p></a></li>
<li><a href='#UmiCountsPlot'><p>Plot UMI counts</p></a></li>
<li><a href='#UMIexperiment-class'><p>UMIexperiment class</p></a></li>
<li><a href='#UMIsample-class'><p>UMIsample class</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for Analyzing Sequencing Data with Unique Molecular
Identifiers</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-23</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stefan Filges &lt;stefan.filges@gu.se&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for analyzing sequencing data containing unique
    molecular identifiers generated by 'UMIErrorCorrect'
    (<a href="https://github.com/stahlberggroup/umierrorcorrect">https://github.com/stahlberggroup/umierrorcorrect</a>).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sfilges/umiAnalyzer">https://github.com/sfilges/umiAnalyzer</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sfilges/umiAnalyzer/issues">https://github.com/sfilges/umiAnalyzer/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocManager, dplyr (&ge; 0.7.5), DT (&ge; 0.19), forcats (&ge;
0.5.0), ggplot2 (&ge; 2.2.1), graphics, grDevices, gridExtra (&ge;
2.3), magrittr (&ge; 1.5), methods, pheatmap (&ge; 1.0.12), plotly
(&ge; 4.9.2.1), readr (&ge; 1.1.1), Rsamtools (&ge; 1.32.3), scales
(&ge; 1.1.0), shiny (&ge; 1.7.1), shinydashboard (&ge; 0.7.2),
shinyFiles (&ge; 0.9.0), shinyWidgets (&ge; 0.6.2), stats, stringr
(&ge; 1.4.0), tibble (&ge; 1.4.2), tidyr (&ge; 0.8.1), utils, viridis
(&ge; 0.5.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr (&ge; 1.27), rmarkdown (&ge; 2.1)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-24 21:59:58 UTC; stefa</td>
</tr>
<tr>
<td>Author:</td>
<td>Stefan Filges <a href="https://orcid.org/0000-0002-5994-6699"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gustav Johansson [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-25 08:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='addMetaData'>Add metaData</h2><span id='topic+addMetaData'></span>

<h3>Description</h3>

<p>Add metaData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMetaData(object, attributeName, attributeValue)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addMetaData_+3A_object">object</code></td>
<td>
<p>R object to which meta data should be added</p>
</td></tr>
<tr><td><code id="addMetaData_+3A_attributename">attributeName</code></td>
<td>
<p>Name of the meta data attribute.</p>
</td></tr>
<tr><td><code id="addMetaData_+3A_attributevalue">attributeValue</code></td>
<td>
<p>Meta data to be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)

metaData &lt;- system.file("extdata", "metadata.txt", package = "umiAnalyzer")

simsen &lt;- addMetaData(simsen,'metaData',metaData)


</code></pre>

<hr>
<h2 id='addUmiSample'>Add UMI sample to an existing experiment object</h2><span id='topic+addUmiSample'></span>

<h3>Description</h3>

<p>Add UMI sample to an existing experiment object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addUmiSample(object, sampleName, sampleDir, clearData = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="addUmiSample_+3A_object">object</code></td>
<td>
<p>UMIexperiment object</p>
</td></tr>
<tr><td><code id="addUmiSample_+3A_samplename">sampleName</code></td>
<td>
<p>Name of new sample</p>
</td></tr>
<tr><td><code id="addUmiSample_+3A_sampledir">sampleDir</code></td>
<td>
<p>Directory to new sample</p>
</td></tr>
<tr><td><code id="addUmiSample_+3A_cleardata">clearData</code></td>
<td>
<p>Should other data in UMIexperiment be cleared</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object
</p>

<hr>
<h2 id='AmpliconHeatmap'>Amplicon heatmap</h2><span id='topic+AmpliconHeatmap'></span>

<h3>Description</h3>

<p>Generates a heatmap of mutations with sample clustering using pheatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AmpliconHeatmap(
  object,
  filter.name = "default",
  cut.off = 5,
  left.side = "columns",
  amplicons = NULL,
  samples = NULL,
  abs.count = FALSE,
  font.size = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AmpliconHeatmap_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_filter.name">filter.name</code></td>
<td>
<p>Name of the filter to be plotted.</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_cut.off">cut.off</code></td>
<td>
<p>How many variant reads are necessary to consider a variant above background? Default is 5 reads.</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_left.side">left.side</code></td>
<td>
<p>Show assays or sample on the left side of the heatmap. Default is assays</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_amplicons">amplicons</code></td>
<td>
<p>(Optional) character vector of amplicons to be plotted.</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_samples">samples</code></td>
<td>
<p>(Optional) character vector of samples to be plotted.</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_abs.count">abs.count</code></td>
<td>
<p>Logical. Should absolute counts be used instead of frequencies?</p>
</td></tr>
<tr><td><code id="AmpliconHeatmap_+3A_font.size">font.size</code></td>
<td>
<p>Font size to use for sample labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A graphics object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')
samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)
simsen &lt;- createUmiExperiment(experimentName = 'example',mainDir = main,sampleNames = samples)
simsen &lt;- filterUmiObject(simsen)

hmap &lt;- AmpliconHeatmap(simsen)

## End(Not run)

</code></pre>

<hr>
<h2 id='AmpliconPlot'>Generate Amplicon plots</h2><span id='topic+AmpliconPlot'></span>

<h3>Description</h3>

<p>Plots variant allele frequencies or alternate allele counts for chosen
samples and assays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AmpliconPlot(
  object,
  filter.name = "default",
  cut.off = 5,
  min.count = 0,
  min.vaf = 0,
  amplicons = NULL,
  samples = NULL,
  abs.count = FALSE,
  y_min = 0,
  y_max = NULL,
  theme = "classic",
  option = "default",
  direction = "default",
  plot.text = FALSE,
  plot.ref = TRUE,
  stack.plot = FALSE,
  classic.plot = FALSE,
  fdr = 0.05,
  font.size = 6,
  angle = 45,
  use.caller = FALSE,
  use.plotly = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AmpliconPlot_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_filter.name">filter.name</code></td>
<td>
<p>Name of the filter to be plotted.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_cut.off">cut.off</code></td>
<td>
<p>How many variant reads are necessary to consider a variant above background? Default is 5 reads.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_min.count">min.count</code></td>
<td>
<p>Minimum variants counts to plot, default is 0.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_min.vaf">min.vaf</code></td>
<td>
<p>Minimum variants allele frequency to plot, default is 0.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_amplicons">amplicons</code></td>
<td>
<p>(Optional) character vector of amplicons to be plotted.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_samples">samples</code></td>
<td>
<p>(Optional) character vector of samples to be plotted.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_abs.count">abs.count</code></td>
<td>
<p>Should absolute counts be plotted instead of frequencies? Default is FALSE.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_y_min">y_min</code></td>
<td>
<p>Minimum y-axis value, default is 0</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_y_max">y_max</code></td>
<td>
<p>Maximum y-axis value, default is NULL (autoscale)</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_theme">theme</code></td>
<td>
<p>Plotting theme to use, default is classic.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_option">option</code></td>
<td>
<p>Color palette to use.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_direction">direction</code></td>
<td>
<p>Orientation of the color palette.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_plot.text">plot.text</code></td>
<td>
<p>Should non-references bases be indicated above the bar?</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_plot.ref">plot.ref</code></td>
<td>
<p>If true show reference base instead of position on x-axis.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_stack.plot">stack.plot</code></td>
<td>
<p>Show all variant alleles in a stacked bar plot.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_classic.plot">classic.plot</code></td>
<td>
<p>Show classical debarcer amplicon plot with raw error.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_fdr">fdr</code></td>
<td>
<p>False-discovery-rate cut-off for variants.</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_font.size">font.size</code></td>
<td>
<p>Font size</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_angle">angle</code></td>
<td>
<p>Font angle</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_use.caller">use.caller</code></td>
<td>
<p>Should data from variant caller be used? Default is FALSE</p>
</td></tr>
<tr><td><code id="AmpliconPlot_+3A_use.plotly">use.plotly</code></td>
<td>
<p>Should plotly be used instead of the regular ggplot device? Default is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object containing a ggplot object with the
amplicon plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')
samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)
simsen &lt;- createUmiExperiment(experimentName = 'example',mainDir = main,sampleNames = samples)
simsen &lt;- filterUmiObject(simsen)

amplicon_plot &lt;- AmpliconPlot(simsen)


</code></pre>

<hr>
<h2 id='BarcodeFamilyHistogram'>Consensus depth histograms</h2><span id='topic+BarcodeFamilyHistogram'></span>

<h3>Description</h3>

<p>Generate histograms for the frequency of barcode family depths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BarcodeFamilyHistogram(
  object,
  xMin = 0,
  xMax = 100,
  samples = NULL,
  option = "viridis",
  direction = 1,
  theme = "classic"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BarcodeFamilyHistogram_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object</p>
</td></tr>
<tr><td><code id="BarcodeFamilyHistogram_+3A_xmin">xMin</code></td>
<td>
<p>Minimum consensus family size to plot, default is 0.</p>
</td></tr>
<tr><td><code id="BarcodeFamilyHistogram_+3A_xmax">xMax</code></td>
<td>
<p>Maximum consensus family size to plot. Default is 100.</p>
</td></tr>
<tr><td><code id="BarcodeFamilyHistogram_+3A_samples">samples</code></td>
<td>
<p>List of samples to be shown.</p>
</td></tr>
<tr><td><code id="BarcodeFamilyHistogram_+3A_option">option</code></td>
<td>
<p>Color scheme to use</p>
</td></tr>
<tr><td><code id="BarcodeFamilyHistogram_+3A_direction">direction</code></td>
<td>
<p>If using viridis colors sets the orientation of color scale.</p>
</td></tr>
<tr><td><code id="BarcodeFamilyHistogram_+3A_theme">theme</code></td>
<td>
<p>ggplot theme to use. Defaults to classic.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)
main = system.file('extdata', package = 'umiAnalyzer')
simsen &lt;- createUmiExperiment(main, importBam = TRUE)
barcode_dist &lt;- BarcodeFamilyHistogram(simsen)

</code></pre>

<hr>
<h2 id='beta_binom'>Beta binomial model</h2><span id='topic+beta_binom'></span>

<h3>Description</h3>

<p>Code was obtained from VGAM package function VGAM::rbetabinom.ab. The VGAM
package is available under the GPL-3 license and maintained by
Thomas Yee &lt;t.yee at auckland.ac.nz&gt;. Source code of the function is identical
to rbetabinom.ab, but the function name was changed to beta_binom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta_binom(n, size, shape1, shape2, limit.prob = 0.5, .dontuse.prob = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="beta_binom_+3A_n">n</code></td>
<td>
<p>n</p>
</td></tr>
<tr><td><code id="beta_binom_+3A_size">size</code></td>
<td>
<p>size</p>
</td></tr>
<tr><td><code id="beta_binom_+3A_shape1">shape1</code></td>
<td>
<p>alpha</p>
</td></tr>
<tr><td><code id="beta_binom_+3A_shape2">shape2</code></td>
<td>
<p>beta</p>
</td></tr>
<tr><td><code id="beta_binom_+3A_limit.prob">limit.prob</code></td>
<td>
<p>0.5</p>
</td></tr>
<tr><td><code id="beta_binom_+3A_.dontuse.prob">.dontuse.prob</code></td>
<td>
<p>NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric
</p>


<h3>References</h3>

<p>Yee TW (2015). Vector Generalized Linear and Additive Models: With an Implementation in R. Springer, New York, USA.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beta_binom(10,5, 0.5, 1)
beta_binom(10,2, 0.5, 1)

</code></pre>

<hr>
<h2 id='callVariants'>callVariants using beta binomial distribution</h2><span id='topic+callVariants'></span>

<h3>Description</h3>

<p>Calculate variant p-values using permutation-based testing. A prior is fitted
to model the background error using maximum likelihood estimation of a beta
distribution. The maximum likelihood estimate of the beta distribution is then
used to define the shape of a beta-binomial distribution used to estimate
variant P-Values. This can be interpreted as a probability for a variant to
not have arisen by chance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>callVariants(object, minDepth = 3, minCoverage = 100, computePrior = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="callVariants_+3A_object">object</code></td>
<td>
<p>A UMIErrorCorrect object.</p>
</td></tr>
<tr><td><code id="callVariants_+3A_mindepth">minDepth</code></td>
<td>
<p>Minimum consensus depth required default is 3</p>
</td></tr>
<tr><td><code id="callVariants_+3A_mincoverage">minCoverage</code></td>
<td>
<p>Minimum Coverage to use, default is 100 reads.</p>
</td></tr>
<tr><td><code id="callVariants_+3A_computeprior">computePrior</code></td>
<td>
<p>Should a new distribution be derived from data? Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object containing raw and FDR-adjusted P-Values
</p>


<h3>See Also</h3>

<p><code><a href="#topic+filterVariants">filterVariants</a></code> on how to filter variants.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)
main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)
simsen &lt;- filterUmiObject(simsen)
simsen &lt;- callVariants(simsen, computePrior = FALSE)

</code></pre>

<hr>
<h2 id='createUmiExperiment'>Method for creating a UMI experiment object</h2><span id='topic+createUmiExperiment'></span>

<h3>Description</h3>

<p>Method for creating a UMI experiment object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createUmiExperiment(
  mainDir,
  experimentName = NULL,
  sampleNames = NULL,
  importBam = FALSE,
  as.shiny = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createUmiExperiment_+3A_maindir">mainDir</code></td>
<td>
<p>Main experiment directory</p>
</td></tr>
<tr><td><code id="createUmiExperiment_+3A_experimentname">experimentName</code></td>
<td>
<p>Name of the experiment</p>
</td></tr>
<tr><td><code id="createUmiExperiment_+3A_samplenames">sampleNames</code></td>
<td>
<p>List of sample names. Can be either NULL or list. If NULL all subdirectories of mainDir will be searched.</p>
</td></tr>
<tr><td><code id="createUmiExperiment_+3A_importbam">importBam</code></td>
<td>
<p>Logical. Should bam files be imported on creation? Default is False.</p>
</td></tr>
<tr><td><code id="createUmiExperiment_+3A_as.shiny">as.shiny</code></td>
<td>
<p>Set to TRUE if run within a shiny::withProgress environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class UMIexperiment
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')

samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)

exp1 &lt;- createUmiExperiment(experimentName = 'exp1',mainDir = main,sampleNames = samples)

</code></pre>

<hr>
<h2 id='createUMIexperiment_Debarcer'>Method for creating a UMI experiment object</h2><span id='topic+createUMIexperiment_Debarcer'></span>

<h3>Description</h3>

<p>Method for creating a UMI experiment object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createUMIexperiment_Debarcer(experiment.name, main.dir, dir.names)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createUMIexperiment_Debarcer_+3A_experiment.name">experiment.name</code></td>
<td>
<p>Name of the experiment</p>
</td></tr>
<tr><td><code id="createUMIexperiment_Debarcer_+3A_main.dir">main.dir</code></td>
<td>
<p>Main experiment directory</p>
</td></tr>
<tr><td><code id="createUMIexperiment_Debarcer_+3A_dir.names">dir.names</code></td>
<td>
<p>List of sample names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object
</p>

<hr>
<h2 id='createUmiSample'>createUmiSample</h2><span id='topic+createUmiSample'></span>

<h3>Description</h3>

<p>Method for creating a UMI sample from UMIErrorCorrect output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createUmiSample(sampleName, sampleDir, importBam = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createUmiSample_+3A_samplename">sampleName</code></td>
<td>
<p>UMI sample object name</p>
</td></tr>
<tr><td><code id="createUmiSample_+3A_sampledir">sampleDir</code></td>
<td>
<p>Path to UMI sample folders. Must be a folder generated by UMIErrorCorrect</p>
</td></tr>
<tr><td><code id="createUmiSample_+3A_importbam">importBam</code></td>
<td>
<p>Logical. Should BAM files be imported at object initialization? Default is False.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class UMIsample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')
samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)
s1 &lt;- createUmiSample('s1',sampleDir = paste(main,"/",samples[1],sep=""))

</code></pre>

<hr>
<h2 id='createUMIsample_Debarcer'>Method for creating a UMIsample object</h2><span id='topic+createUMIsample_Debarcer'></span>

<h3>Description</h3>

<p>Method for creating a UMIsample object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createUMIsample_Debarcer(sample.name, sample.dir, cons = "10")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createUMIsample_Debarcer_+3A_sample.name">sample.name</code></td>
<td>
<p>UMI sample object name</p>
</td></tr>
<tr><td><code id="createUMIsample_Debarcer_+3A_sample.dir">sample.dir</code></td>
<td>
<p>Path to UMI sample</p>
</td></tr>
<tr><td><code id="createUMIsample_Debarcer_+3A_cons">cons</code></td>
<td>
<p>Consensus depth. Needs to be string; default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIsample object
</p>

<hr>
<h2 id='download_template'>Download meta data template</h2><span id='topic+download_template'></span>

<h3>Description</h3>

<p>Function for downloading a template file containing metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_template(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="download_template_+3A_object">object</code></td>
<td>
<p>A UMIexperiment object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble containing a metadata template
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)

download_template(simsen)

</code></pre>

<hr>
<h2 id='filterUmiObject'>Method for filtering UMIexperiment and sample objects</h2><span id='topic+filterUmiObject'></span>

<h3>Description</h3>

<p>Method for filtering UMIexperiment and sample objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterUmiObject(
  object,
  name = "default",
  minDepth = 3,
  minCoverage = 100,
  minFreq = 0,
  minCount = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterUmiObject_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object.</p>
</td></tr>
<tr><td><code id="filterUmiObject_+3A_name">name</code></td>
<td>
<p>String. Name of the filter. Default is &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="filterUmiObject_+3A_mindepth">minDepth</code></td>
<td>
<p>Consensus depth to analyze. Default is 3.</p>
</td></tr>
<tr><td><code id="filterUmiObject_+3A_mincoverage">minCoverage</code></td>
<td>
<p>Minimum coverage required for amplicons. Default is 1.</p>
</td></tr>
<tr><td><code id="filterUmiObject_+3A_minfreq">minFreq</code></td>
<td>
<p>Minimum variant allele frequency to keep. Default is 0.</p>
</td></tr>
<tr><td><code id="filterUmiObject_+3A_mincount">minCount</code></td>
<td>
<p>Minimum variant allele count to keep. Default is 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMI sample or UMI experiment object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')

samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)

simsen &lt;- createUmiExperiment(experimentName = 'simsen',mainDir = main,sampleNames = samples)

simsen &lt;- filterUmiObject(simsen)

</code></pre>

<hr>
<h2 id='filterVariants'>Filter variants based on p values or depth</h2><span id='topic+filterVariants'></span>

<h3>Description</h3>

<p>You can filter variants called with the the &quot;callVariants&quot; function based
on adjusted p-value, minimum variant allele count and supply a list
of assays and samples to plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterVariants(
  object,
  p.adjust = 0.2,
  minVarCount = 5,
  amplicons = NULL,
  samples = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filterVariants_+3A_object">object</code></td>
<td>
<p>A UMIexperiment object</p>
</td></tr>
<tr><td><code id="filterVariants_+3A_p.adjust">p.adjust</code></td>
<td>
<p>Numeric. Adjusted p value (FDR). Default is 0.2.</p>
</td></tr>
<tr><td><code id="filterVariants_+3A_minvarcount">minVarCount</code></td>
<td>
<p>Integer. Minimum variant allele count. Default is 5.</p>
</td></tr>
<tr><td><code id="filterVariants_+3A_amplicons">amplicons</code></td>
<td>
<p>NULL or list of assays to plot. NULL uses all.</p>
</td></tr>
<tr><td><code id="filterVariants_+3A_samples">samples</code></td>
<td>
<p>NULL or list of samples to plot. NULL uses all.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object with filtered variants. Can be used to
generate VCF files.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+callVariants">callVariants</a></code> on how to call variants.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umiAnalyzer)
main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)
simsen &lt;- filterUmiObject(simsen)
simsen &lt;- callVariants(simsen, computePrior = FALSE)
simsen &lt;- filterVariants(simsen, p.adjust = 0.05)

## End(Not run)

</code></pre>

<hr>
<h2 id='findConsensusReads'>Find consensus reads
A function to analyze consensus read tables generated with parseBamFiles or
a UMIexperiment object containing reads.</h2><span id='topic+findConsensusReads'></span>

<h3>Description</h3>

<p>Find consensus reads
A function to analyze consensus read tables generated with parseBamFiles or
a UMIexperiment object containing reads.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findConsensusReads(
  object,
  consDepth = 0,
  groupBy = c("none", "sample", "position", "both"),
  pattern = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findConsensusReads_+3A_object">object</code></td>
<td>
<p>Either a tibble generated with parseBamFiles or a UMIexperiment object</p>
</td></tr>
<tr><td><code id="findConsensusReads_+3A_consdepth">consDepth</code></td>
<td>
<p>Minimum consensus depth to keep. Default is 0.</p>
</td></tr>
<tr><td><code id="findConsensusReads_+3A_groupby">groupBy</code></td>
<td>
<p>Should data be grouped by position, sample, both or not at all.</p>
</td></tr>
<tr><td><code id="findConsensusReads_+3A_pattern">pattern</code></td>
<td>
<p>Regular expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(umiAnalyzer)
main &lt;- system.file("extdata", package = "umiAnalyzer")
simsen &lt;- createUmiExperiment(main, importBam = TRUE)

reads &lt;- findConsensusReads(simsen)
reads


</code></pre>

<hr>
<h2 id='generateVCF'>Generate VCF file from UMI sample or UMI experiment object</h2><span id='topic+generateVCF'></span>

<h3>Description</h3>

<p>Generate VCF file from UMI sample or UMI experiment object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateVCF(object, outDir = getwd(), outFile, printAll = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generateVCF_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object</p>
</td></tr>
<tr><td><code id="generateVCF_+3A_outdir">outDir</code></td>
<td>
<p>String. Output directory, defaults to working directory.</p>
</td></tr>
<tr><td><code id="generateVCF_+3A_outfile">outFile</code></td>
<td>
<p>String. Name of the output file</p>
</td></tr>
<tr><td><code id="generateVCF_+3A_printall">printAll</code></td>
<td>
<p>Logical. Should all or only trusted variant be printed?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A VCF file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)

simsen &lt;- filterUmiObject(simsen)

generateVCF(simsen,'simsen.vcf', printAll = FALSE, save = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='getFilteredData'>Method for retrieving filtered data</h2><span id='topic+getFilteredData'></span>

<h3>Description</h3>

<p>Method for retrieving filtered data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFilteredData(
  object,
  name = "default",
  save = FALSE,
  outDir = getwd(),
  fileName = NULL,
  delim = ";"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getFilteredData_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object.</p>
</td></tr>
<tr><td><code id="getFilteredData_+3A_name">name</code></td>
<td>
<p>String. Name of the filter. Default is &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="getFilteredData_+3A_save">save</code></td>
<td>
<p>Logical, should data be saved as csv file? Default is FALSE.</p>
</td></tr>
<tr><td><code id="getFilteredData_+3A_outdir">outDir</code></td>
<td>
<p>Output directory</p>
</td></tr>
<tr><td><code id="getFilteredData_+3A_filename">fileName</code></td>
<td>
<p>Filename to be used, default is the same as 'name'</p>
</td></tr>
<tr><td><code id="getFilteredData_+3A_delim">delim</code></td>
<td>
<p>Character string denoting delimiter to be used, default is ';'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A filtered consensus table, as a tibble.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')

samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)

simsen &lt;- createUmiExperiment(experimentName = 'simsen',mainDir = main,sampleNames = samples)
simsen &lt;- filterUmiObject(simsen)

myfilter &lt;- getFilteredData(simsen)
myfilter

</code></pre>

<hr>
<h2 id='getMetaData'>Retrieve meta data by name.</h2><span id='topic+getMetaData'></span>

<h3>Description</h3>

<p>Retrieve meta data by name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaData(object, attributeName)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMetaData_+3A_object">object</code></td>
<td>
<p>R object from which to get meta data.</p>
</td></tr>
<tr><td><code id="getMetaData_+3A_attributename">attributeName</code></td>
<td>
<p>Name of the meta data attribute.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Metadata
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)

metaData &lt;- system.file("extdata", "metadata.txt", package = "umiAnalyzer")

simsen &lt;- importDesign(object = simsen,file = metaData)
design &lt;- getMetaData(object = simsen, attributeName = "design")
design


</code></pre>

<hr>
<h2 id='importBedFile'>Import bed file</h2><span id='topic+importBedFile'></span>

<h3>Description</h3>

<p>Import bed file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importBedFile(path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="importBedFile_+3A_path">path</code></td>
<td>
<p>path to bed file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table containing genome positions
</p>

<hr>
<h2 id='importDesign'>Import experimental design meta data such as replicates, treatments, categorical variables.</h2><span id='topic+importDesign'></span>

<h3>Description</h3>

<p>Import experimental design meta data such as replicates, treatments, categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importDesign(object, file, delim = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="importDesign_+3A_object">object</code></td>
<td>
<p>UMI.experiment to which to add metadata</p>
</td></tr>
<tr><td><code id="importDesign_+3A_file">file</code></td>
<td>
<p>File containing meta data</p>
</td></tr>
<tr><td><code id="importDesign_+3A_delim">delim</code></td>
<td>
<p>Column separator. Default is NULL (automatically determine delimiter)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)

metaData &lt;- system.file("extdata", "metadata.txt", package = "umiAnalyzer")

simsen &lt;- importDesign(object = simsen,file = metaData)

# Retrieve meta data
design &lt;- getMetaData(object = simsen, attributeName = "design")
design


</code></pre>

<hr>
<h2 id='mergeAssays'>Merge assays</h2><span id='topic+mergeAssays'></span>

<h3>Description</h3>

<p>Merge assays together by name. Requires a name of the new assay and
a list of assays that will be merged.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeAssays(object, name, assay.list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mergeAssays_+3A_object">object</code></td>
<td>
<p>A UMIexperiment object</p>
</td></tr>
<tr><td><code id="mergeAssays_+3A_name">name</code></td>
<td>
<p>Name of the new assay</p>
</td></tr>
<tr><td><code id="mergeAssays_+3A_assay.list">assay.list</code></td>
<td>
<p>List of assays to merge</p>
</td></tr>
</table>


<h3>Value</h3>

<p>merged consensus data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")

simsen &lt;- createUmiExperiment(main)

simsen &lt;- mergeAssays(object = simsen,name = "new",assay.list = c("PIK3CA_123", "PIK3CA_234"))


</code></pre>

<hr>
<h2 id='parseBamFiles'>Function to parse bam files</h2><span id='topic+parseBamFiles'></span>

<h3>Description</h3>

<p>Function to parse bam files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseBamFiles(mainDir, sampleNames = NULL, consDepth = 0, as.shiny = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parseBamFiles_+3A_maindir">mainDir</code></td>
<td>
<p>Directory containing UMIErrorCorrect output folders.</p>
</td></tr>
<tr><td><code id="parseBamFiles_+3A_samplenames">sampleNames</code></td>
<td>
<p>A list of sample names.</p>
</td></tr>
<tr><td><code id="parseBamFiles_+3A_consdepth">consDepth</code></td>
<td>
<p>Only retain consensus reads of at least cons.depth. Default is 0.</p>
</td></tr>
<tr><td><code id="parseBamFiles_+3A_as.shiny">as.shiny</code></td>
<td>
<p>Set to TRUE if run within a shiny::withProgress environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(umiAnalyzer)
main &lt;- system.file("extdata", package = "umiAnalyzer")
simsen &lt;- createUmiExperiment(main)

reads &lt;- parseBamFiles(main, consDepth = 10)


</code></pre>

<hr>
<h2 id='QCplot'>Generate QC plots</h2><span id='topic+QCplot'></span>

<h3>Description</h3>

<p>Visualize the UMI count for each selected assay and sample for a given
consensus depth. This is useful to detect differences in coverage,
especially for multiplexed assays.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QCplot(
  object,
  group.by = "sample",
  plotDepth = 3,
  assays = NULL,
  samples = NULL,
  theme = "classic",
  option = "viridis",
  direction = "default",
  toggle_mean = TRUE,
  center = "mean",
  line_col = "blue",
  angle = 0,
  plotly = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QCplot_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object</p>
</td></tr>
<tr><td><code id="QCplot_+3A_group.by">group.by</code></td>
<td>
<p>String. Which variable should be used as a factor on the x-axis. Default is sample</p>
</td></tr>
<tr><td><code id="QCplot_+3A_plotdepth">plotDepth</code></td>
<td>
<p>Which consensus depth to plot</p>
</td></tr>
<tr><td><code id="QCplot_+3A_assays">assays</code></td>
<td>
<p>(Optional) user-supplied list of assays to plot. Default is all.</p>
</td></tr>
<tr><td><code id="QCplot_+3A_samples">samples</code></td>
<td>
<p>(Optional) user-supplied list of samples to plot. Default is all.</p>
</td></tr>
<tr><td><code id="QCplot_+3A_theme">theme</code></td>
<td>
<p>ggplot theme to use.</p>
</td></tr>
<tr><td><code id="QCplot_+3A_option">option</code></td>
<td>
<p>Color palette to use, either ggplot default or viridis colors.</p>
</td></tr>
<tr><td><code id="QCplot_+3A_direction">direction</code></td>
<td>
<p>If viridis colors are used, choose orientation of color scale.</p>
</td></tr>
<tr><td><code id="QCplot_+3A_toggle_mean">toggle_mean</code></td>
<td>
<p>Show mean or median</p>
</td></tr>
<tr><td><code id="QCplot_+3A_center">center</code></td>
<td>
<p>Choose mean or median</p>
</td></tr>
<tr><td><code id="QCplot_+3A_line_col">line_col</code></td>
<td>
<p>Choose color for mean/median line</p>
</td></tr>
<tr><td><code id="QCplot_+3A_angle">angle</code></td>
<td>
<p>Angle of labels on x-axis.</p>
</td></tr>
<tr><td><code id="QCplot_+3A_plotly">plotly</code></td>
<td>
<p>Should plotly be used for rendering?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')
samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)
simsen &lt;- createUmiExperiment(experimentName = 'example',mainDir = main,sampleNames = samples)

depth_plot &lt;- QCplot(simsen)

</code></pre>

<hr>
<h2 id='runUmiVisualizer'>Function to run the umiVisualizer shiny app</h2><span id='topic+runUmiVisualizer'></span>

<h3>Description</h3>

<p>Function to run the umiVisualizer shiny app
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runUmiVisualizer()
</code></pre>


<h3>Value</h3>

<p>Opens the umiVisualizer app
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umiAnalyzer)

runUmiVisualizer()

## End(Not run)

</code></pre>

<hr>
<h2 id='saveConsData'>Save consensus data</h2><span id='topic+saveConsData'></span>

<h3>Description</h3>

<p>If save is set to TRUE data will be written to a csv file otherwise consensus data will
be returned as a tibble.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>saveConsData(
  object,
  save = FALSE,
  fileName = "consensus_data.csv",
  outDir = getwd(),
  delim = ";"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="saveConsData_+3A_object">object</code></td>
<td>
<p>UMIexperiment object</p>
</td></tr>
<tr><td><code id="saveConsData_+3A_save">save</code></td>
<td>
<p>Logical. Should data be saved to file? Default is FALSE.</p>
</td></tr>
<tr><td><code id="saveConsData_+3A_filename">fileName</code></td>
<td>
<p>String. Name of the file to be saved. Default is 'consensus_data.csv'</p>
</td></tr>
<tr><td><code id="saveConsData_+3A_outdir">outDir</code></td>
<td>
<p>output directory, defaults to working directory</p>
</td></tr>
<tr><td><code id="saveConsData_+3A_delim">delim</code></td>
<td>
<p>Single character string, either ';' or ',' or tab</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')

samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)

example &lt;- createUmiExperiment(experimentName = 'example',mainDir = main,sampleNames = samples)

consensus_data &lt;- saveConsData(object = example)
consensus_data


</code></pre>

<hr>
<h2 id='simsen'>UMIexperiment data generated with SiMSen-Seq</h2><span id='topic+simsen'></span>

<h3>Description</h3>

<p>UMIexperiment data generated with SiMSen-Seq
</p>


<h3>Format</h3>

<p>An object of class <code>"UMIexperiment"</code>
</p>

<hr>
<h2 id='timeSeriesGrid'>Plot time series data</h2><span id='topic+timeSeriesGrid'></span>

<h3>Description</h3>

<p>Function for plotting time series or other meta data. Uses facet wrap to
display user-provided categorical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeSeriesGrid(
  object,
  filter.name = "default",
  cut.off = 5,
  min.count = 0,
  min.vaf = 0,
  amplicons = NULL,
  samples = NULL,
  x_variable = NULL,
  y_variable = "Max Non-ref Allele Frequency",
  columns = "Sample Name",
  rows = "Name",
  color_by = "Name",
  fdr = 0.05,
  use.caller = TRUE,
  bed_positions = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="timeSeriesGrid_+3A_object">object</code></td>
<td>
<p>A consensus data table</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_filter.name">filter.name</code></td>
<td>
<p>&quot;default&quot;</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_cut.off">cut.off</code></td>
<td>
<p>5</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_min.count">min.count</code></td>
<td>
<p>0</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_min.vaf">min.vaf</code></td>
<td>
<p>0</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_amplicons">amplicons</code></td>
<td>
<p>NULL</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_samples">samples</code></td>
<td>
<p>NULL</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_x_variable">x_variable</code></td>
<td>
<p>NULL</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_y_variable">y_variable</code></td>
<td>
<p>&quot;Max Non-ref Allele Frequency&quot;</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_columns">columns</code></td>
<td>
<p>&quot;Sample Name&quot;</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_rows">rows</code></td>
<td>
<p>&quot;Name&quot;</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_color_by">color_by</code></td>
<td>
<p>&quot;Name&quot;</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_fdr">fdr</code></td>
<td>
<p>0.05</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_use.caller">use.caller</code></td>
<td>
<p>TRUE</p>
</td></tr>
<tr><td><code id="timeSeriesGrid_+3A_bed_positions">bed_positions</code></td>
<td>
<p>NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main &lt;- system.file("extdata", package = "umiAnalyzer")
simsen &lt;- createUmiExperiment(main)
simsen &lt;- filterUmiObject(simsen)

metaData &lt;- system.file("extdata", "metadata.txt", package = "umiAnalyzer")
simsen &lt;- importDesign(object = simsen,file = metaData)

bed_dir &lt;- system.file("extdata", "simple.bed", package = "umiAnalyzer")
bed &lt;- importBedFile(path = bed_dir)

time_plot &lt;- timeSeriesGrid(simsen, x_variable = "time", bed_positions = bed)

</code></pre>

<hr>
<h2 id='UmiCountsPlot'>Plot UMI counts</h2><span id='topic+UmiCountsPlot'></span>

<h3>Description</h3>

<p>Visualize the number detected UMI for each consensus depth cut-off. This may
may helpful in choosing the right consensus depth for your analysis, by
checking the number of reads still available for each assay and sample
for your chosen cut-off.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UmiCountsPlot(
  object,
  amplicons = NULL,
  samples = NULL,
  theme = "classic",
  option = "viridis",
  direction = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UmiCountsPlot_+3A_object">object</code></td>
<td>
<p>Requires a UMI sample or UMI experiment object</p>
</td></tr>
<tr><td><code id="UmiCountsPlot_+3A_amplicons">amplicons</code></td>
<td>
<p>(Optional) user-supplied list of assays to plot. Default is all.</p>
</td></tr>
<tr><td><code id="UmiCountsPlot_+3A_samples">samples</code></td>
<td>
<p>(Optional) user-supplied list of samples to plot. Default is all.</p>
</td></tr>
<tr><td><code id="UmiCountsPlot_+3A_theme">theme</code></td>
<td>
<p>Plotting theme, default is classic</p>
</td></tr>
<tr><td><code id="UmiCountsPlot_+3A_option">option</code></td>
<td>
<p>Color palette. Default uses ggplot standard, otherwise viridis options.</p>
</td></tr>
<tr><td><code id="UmiCountsPlot_+3A_direction">direction</code></td>
<td>
<p>If using viridis colors should the scale be inverted or default?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A UMIexperiment object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umiAnalyzer)

main = system.file('extdata', package = 'umiAnalyzer')
samples &lt;- list.dirs(path = main, full.names = FALSE, recursive = FALSE)
simsen &lt;- createUmiExperiment(experimentName = 'example',mainDir = main,sampleNames = samples)
simsen &lt;- filterUmiObject(simsen)

count_plot &lt;- UmiCountsPlot(simsen)

</code></pre>

<hr>
<h2 id='UMIexperiment-class'>UMIexperiment class</h2><span id='topic+UMIexperiment-class'></span><span id='topic+UMIexperiment'></span>

<h3>Description</h3>

<p>The UMIexperiment is the core data object, storing all data and relevant
analysis data associated with your experiment. Each object has number of
slots storing raw data, graphs and processed data.
</p>


<h3>Value</h3>

<p>An object of class UMIexperiment
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code></dt><dd><p>Optional project name for record keeping.</p>
</dd>
<dt><code>cons.data</code></dt><dd><p>The raw consensus data supplied by the user.</p>
</dd>
<dt><code>summary.data</code></dt><dd><p>Summary data from UMIErrorCorrect</p>
</dd>
<dt><code>raw.error</code></dt><dd><p>Cons0 error profile</p>
</dd>
<dt><code>reads</code></dt><dd><p>Consensus reads imported using the parseBamFiles function.</p>
</dd>
<dt><code>meta.data</code></dt><dd><p>Sample data optionally supplied by the user.</p>
</dd>
<dt><code>filters</code></dt><dd><p>A list of filtered cons.data, which can be accessed separately.</p>
</dd>
<dt><code>plots</code></dt><dd><p>A list of generated plots.</p>
</dd>
<dt><code>variants</code></dt><dd><p>Consensus table generated with the umiAnalyzer variant caller.</p>
</dd>
<dt><code>merged.data</code></dt><dd><p>Data generated using the mergeTechnicalReplicates function.</p>
</dd>
</dl>

<hr>
<h2 id='UMIsample-class'>UMIsample class</h2><span id='topic+UMIsample-class'></span><span id='topic+UMIsample'></span>

<h3>Description</h3>

<p>UMIsample class
</p>


<h3>Value</h3>

<p>An object of class UMIsample
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code></dt><dd><p>Sample name</p>
</dd>
<dt><code>cons.data</code></dt><dd><p>Raw consensus data</p>
</dd>
<dt><code>summary.data</code></dt><dd><p>Summary data from UMIErrorCorrect</p>
</dd>
<dt><code>reads</code></dt><dd><p>Consensus reads imported from a bam file.</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
