<!DOCTYPE html><html><head><title>Help for package umx</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {umx}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#umx'><p>Functions for Structural Equation Modeling in OpenMx</p></a></li>
<li><a href='#bucks'><p>Print a money object</p></a></li>
<li><a href='#deg2rad'><p>Convert Degrees to Degrees</p></a></li>
<li><a href='#dl_from_dropbox'><p>dl_from_dropbox</p></a></li>
<li><a href='#docData'><p>Twin data for Direction of causation modelling</p></a></li>
<li><a href='#extractAIC.MxModel'><p>Extract AIC from MxModel</p></a></li>
<li><a href='#fin_interest'><p>Compute the value of a principal &amp; annual deposits at a compound interest over a number of years</p></a></li>
<li><a href='#fin_JustifiedPE'><p>Justified P/E Ratio</p></a></li>
<li><a href='#fin_NI'><p>Compute NI given annual Earnings.</p></a></li>
<li><a href='#fin_percent'><p>Compute the percent change needed to return to the original value after percent off (or on).</p></a></li>
<li><a href='#fin_ticker'><p>Open a ticker in yahoo finance.</p></a></li>
<li><a href='#fin_valuation'><p>Work the valuation of a company</p></a></li>
<li><a href='#Fischbein_wt'><p>Weight data across time.</p></a></li>
<li><a href='#FishersMethod'><p>Fishers Method of combining p-values.</p></a></li>
<li><a href='#geometric_mean'><p>Geometric Mean</p></a></li>
<li><a href='#GFF'><p>Twin data: General Family Functioning, divorce, and well-being.</p></a></li>
<li><a href='#ggAddR'><p>Add a fit statistic to a ggplot</p></a></li>
<li><a href='#harmonic_mean'><p>Harmonic Mean</p></a></li>
<li><a href='#install.OpenMx'><p>Install OpenMx, with choice of builds</p></a></li>
<li><a href='#iqdat'><p>Twin data: IQ measured longitudinally across 4 ages.</p></a></li>
<li><a href='#libs'><p>load libraries</p></a></li>
<li><a href='#loadings'><p>loadings</p>
Generic loadings function to extract factor loadings from exploratory or confirmatory
factor analyses.</a></li>
<li><a href='#loadings.MxModel'><p>Extract factor loadings from an EFA (factor analysis).</p></a></li>
<li><a href='#noNAs'><p>Succinctly select complete rows from a dataframe</p></a></li>
<li><a href='#oddsratio'><p>Compute odds ratio (OR)</p></a></li>
<li><a href='#plot.MxLISRELModel'><p>Create and display a graphical path diagram for a LISREL model.</p></a></li>
<li><a href='#plot.MxModel'><p>Create and display a graphical path diagram for a model.</p></a></li>
<li><a href='#plot.MxModelTwinMaker'><p>Create and display a graphical path diagram for a path-based twin model.</p></a></li>
<li><a href='#plot.percent'><p>Plot a percent change graph</p></a></li>
<li><a href='#power.ACE.test'><p>Test the power of an ACE model to detect paths of interest.</p></a></li>
<li><a href='#print.oddsratio'><p>Print a scale &quot;oddsratio&quot; object</p></a></li>
<li><a href='#print.percent'><p>Print a percent object</p></a></li>
<li><a href='#print.reliability'><p>Print a scale &quot;reliability&quot; object</p></a></li>
<li><a href='#print.RMSEA'><p>Print a RMSEA object</p></a></li>
<li><a href='#prolific_anonymize'><p>Clean up a prolific file for sharing by removing anonymity-compromising columns.</p></a></li>
<li><a href='#prolific_check_ID'><p>Return PIDs in df</p></a></li>
<li><a href='#prolific_read_demog'><p>Read and optionally merge demographics file from prolific academic</p></a></li>
<li><a href='#qm'><p>qm</p></a></li>
<li><a href='#rad2deg'><p>Convert Radians to Degrees</p></a></li>
<li><a href='#reliability'><p>Report coefficient alpha (reliability)</p></a></li>
<li><a href='#residuals.MxModel'><p>Get residuals from an MxModel</p></a></li>
<li><a href='#RMSEA'><p>Generic RMSEA function</p></a></li>
<li><a href='#RMSEA.MxModel'><p>RMSEA function for MxModels</p></a></li>
<li><a href='#RMSEA.summary.mxmodel'><p>RMSEA function for MxModel summary</p></a></li>
<li><a href='#SE_from_p'><p>Compute an SE from a beta and p value</p></a></li>
<li><a href='#tmx_genotypic_effect'><p>Graphical display of genotypic effects.</p></a></li>
<li><a href='#tmx_is.identified'><p>Test if a factor model is identified</p></a></li>
<li><a href='#tmx_show'><p>Show matrices of models in a easy-to-learn-from format.</p></a></li>
<li><a href='#tmx_show.MxMatrix'><p>Show matrices of models in a easy-to-learn-from format.</p></a></li>
<li><a href='#tmx_show.MxModel'><p>Show matrices of RAM models in a easy-to-learn-from format.</p></a></li>
<li><a href='#umx_aggregate'><p>Convenient formula-based cross-tabs &amp; built-in summary functions</p></a></li>
<li><a href='#umx_APA_pval'><p>Round p-values according to APA guidelines</p></a></li>
<li><a href='#umx_apply'><p>umx_apply</p></a></li>
<li><a href='#umx_array_shift'><p>Like the php array_shift function: shifts an item off the beginning of a list</p></a></li>
<li><a href='#umx_as_numeric'><p>umx_as_numeric</p></a></li>
<li><a href='#umx_check'><p>umx_check</p></a></li>
<li><a href='#umx_check_model'><p>Check for required features in an OpenMx.</p></a></li>
<li><a href='#umx_check_names'><p>Check if a request name exists in a dataframe or related object</p></a></li>
<li><a href='#umx_check_OS'><p>umx_check_OS</p></a></li>
<li><a href='#umx_check_parallel'><p>Check if OpenMx is using OpenMP, test cores, and get timings</p></a></li>
<li><a href='#umx_cont_2_quantiles'><p>umx_cont_2_quantiles</p></a></li>
<li><a href='#umx_cor'><p>Report correlations and their p-values</p></a></li>
<li><a href='#umx_explode'><p>Explode a string (Like the php function <code>explode</code>)</p></a></li>
<li><a href='#umx_explode_twin_names'><p>Break twin variable names (BMI_T1, BMI_T2) into base variable names (BMI, &quot;_T&quot;, 1:2)</p></a></li>
<li><a href='#umx_file_load_pseudo'><p>Read in files from pseudocons.</p></a></li>
<li><a href='#umx_find_object'><p>umx_find_object</p></a></li>
<li><a href='#umx_fun_mean_sd'><p>Summarizing functions used in umx_aggregate and for umxAPA</p></a></li>
<li><a href='#umx_get_bracket_addresses'><p>Get bracket-style addresses from an mxMatrix</p></a></li>
<li><a href='#umx_get_checkpoint'><p>Get or set checkpointing for a model</p></a></li>
<li><a href='#umx_get_options'><p>Display umx options</p></a></li>
<li><a href='#umx_grep'><p>Search for text</p></a></li>
<li><a href='#umx_has_been_run'><p>umx_has_been_run</p></a></li>
<li><a href='#umx_has_CIs'><p>umx_has_CIs</p></a></li>
<li><a href='#umx_has_means'><p>umx_has_means</p></a></li>
<li><a href='#umx_has_square_brackets'><p>Check if a label contains square brackets</p></a></li>
<li><a href='#umx_is_class'><p>Check if variables in a dataframe are in a list of classes.</p></a></li>
<li><a href='#umx_is_cov'><p>umx_is_cov</p></a></li>
<li><a href='#umx_is_endogenous'><p>List endogenous variables in a model</p></a></li>
<li><a href='#umx_is_exogenous'><p>umx_is_exogenous</p></a></li>
<li><a href='#umx_is_MxData'><p>Check if an object is an mxData object</p></a></li>
<li><a href='#umx_is_MxMatrix'><p>umx_is_MxMatrix</p></a></li>
<li><a href='#umx_is_MxModel'><p>umx_is_MxModel</p></a></li>
<li><a href='#umx_is_numeric'><p>Check if variables in a dataframe are numeric</p></a></li>
<li><a href='#umx_is_ordered'><p>Test if one or more variables in a dataframe are ordered</p></a></li>
<li><a href='#umx_is_RAM'><p>umx_is_RAM</p></a></li>
<li><a href='#umx_long2wide'><p>Take a long twin-data file and make it wide (one family per row)</p></a></li>
<li><a href='#umx_lower.tri'><p>Get values from lower triangle of a matrix</p></a></li>
<li><a href='#umx_lower2full'><p>Convert lower-only matrix data to full (or enforce symmetry on a full matrix)</p></a></li>
<li><a href='#umx_make'><p>&quot;make&quot; the umx package using devtools: release to CRAN etc.</p></a></li>
<li><a href='#umx_make_fake_data'><p>umx_make_fake_data</p></a></li>
<li><a href='#umx_make_MR_data'><p>Simulate Mendelian Randomization data</p></a></li>
<li><a href='#umx_make_raw_from_cov'><p>Turn a cov matrix into raw data</p></a></li>
<li><a href='#umx_make_sql_from_excel'><p>Convert an excel spreadsheet in a text file on sql statements.</p></a></li>
<li><a href='#umx_make_twin_data_nice'><p>Convert a twin dataset into umx standard format.</p></a></li>
<li><a href='#umx_make_TwinData'><p>Simulate twin data with control over A, C, and E parameters, as well as moderation of A.</p></a></li>
<li><a href='#umx_means'><p>umx_means</p></a></li>
<li><a href='#umx_merge_randomized_columns'><p>umx_merge_randomized_columns</p></a></li>
<li><a href='#umx_move_file'><p>Move files</p></a></li>
<li><a href='#umx_msg'><p>Print the name and compact contents of variable.</p></a></li>
<li><a href='#umx_names'><p>umx_names</p></a></li>
<li><a href='#umx_open'><p>Open a file or folder</p></a></li>
<li><a href='#umx_open_CRAN_page'><p>Open the CRAN page for a package</p></a></li>
<li><a href='#umx_pad'><p>Pad an Object with NAs</p></a></li>
<li><a href='#umx_paste_names'><p>Concatenate base variable names with suffixes to create wide-format variable names (i.e twin-format)</p></a></li>
<li><a href='#umx_polychoric'><p>FIML-based polychoric, polyserial, and Pearson correlations</p></a></li>
<li><a href='#umx_polypairwise'><p>FIML-based Pairwise polychoric, polyserial, and Pearson correlations</p></a></li>
<li><a href='#umx_polytriowise'><p>FIML-based trio-based polychoric, polyserial, and Pearson correlations</p></a></li>
<li><a href='#umx_print'><p>Print tables in a range of formats (markdown default, see <code>umx_set_table_format()</code> for other formats)</p>
or as a web browser table.</a></li>
<li><a href='#umx_r_test'><p>Test the difference between correlations for significance.</p></a></li>
<li><a href='#umx_read_lower'><p>Read lower-triangle of data matrix from console or file</p></a></li>
<li><a href='#umx_rename'><p>umx_rename</p></a></li>
<li><a href='#umx_rename_file'><p>Rename files</p></a></li>
<li><a href='#umx_reorder'><p>Reorder or drop variables from a correlation/covariance matrix.</p></a></li>
<li><a href='#umx_residualize'><p>Easily residualize variables in long or wide dataframes, returning them changed in-place.</p></a></li>
<li><a href='#umx_rot'><p>Rotate a vector</p></a></li>
<li><a href='#umx_round'><p>umx_round</p></a></li>
<li><a href='#umx_scale'><p>Scale data columns, skipping non-scalable columns</p></a></li>
<li><a href='#umx_scale_wide_twin_data'><p>Scale wide twin data</p></a></li>
<li><a href='#umx_score_scale'><p>Score a psychometric scale by summing normal and reversed items.</p></a></li>
<li><a href='#umx_select_valid'><p>Update NA values in one column with valid entries from another</p></a></li>
<li><a href='#umx_set_auto_plot'><p>umx_set_auto_plot</p></a></li>
<li><a href='#umx_set_auto_run'><p>Automatically run models?</p></a></li>
<li><a href='#umx_set_checkpoint'><p>umx_set_checkpoint</p></a></li>
<li><a href='#umx_set_condensed_slots'><p>umx_set_condensed_slots</p></a></li>
<li><a href='#umx_set_cores'><p>umx_set_cores</p></a></li>
<li><a href='#umx_set_data_variance_check'><p>umx_set_data_variance_check</p></a></li>
<li><a href='#umx_set_dollar_symbol'><p>Set the symbol for money</p></a></li>
<li><a href='#umx_set_optimization_options'><p>Set options that affect optimization in OpenMx</p></a></li>
<li><a href='#umx_set_optimizer'><p>Set the optimizer in OpenMx</p></a></li>
<li><a href='#umx_set_plot_file_suffix'><p>Set output suffix used in umx SEM diagram files saved to disk.</p></a></li>
<li><a href='#umx_set_plot_format'><p>Set output format of plots (structural diagrams) in umx</p></a></li>
<li><a href='#umx_set_plot_use_hrbrthemes'><p>Set theme system to use for plots.</p></a></li>
<li><a href='#umx_set_separator'><p>Set the separator</p></a></li>
<li><a href='#umx_set_silent'><p>Turn off most console and summary output from umx</p></a></li>
<li><a href='#umx_set_table_format'><p>umx_set_table_format</p></a></li>
<li><a href='#umx_stack'><p>Stack data like stack() does, with more control.</p></a></li>
<li><a href='#umx_standardize'><p>Return a standardized version of a Structural Model</p></a></li>
<li><a href='#umx_str_chars'><p>Select desired characters from a string</p></a></li>
<li><a href='#umx_str_from_object'><p>Return variable name as a string</p></a></li>
<li><a href='#umx_string_to_algebra'><p>Convert a string to an OpenMx algebra</p></a></li>
<li><a href='#umx_strings2numeric'><p>A wrapper to map columns of strings to numeric.</p></a></li>
<li><a href='#umx_time'><p>umx_time</p></a></li>
<li><a href='#umx_trim'><p>Trim whitespace surrounding a string.</p></a></li>
<li><a href='#umx_var'><p>Get variances from a df that might contain some non-numeric columns</p></a></li>
<li><a href='#umx_wide2long'><p>Change data family data from wide (2 twins per row) to long format.</p></a></li>
<li><a href='#umx_write_to_clipboard'><p>umx_write_to_clipboard</p></a></li>
<li><a href='#umx-deprecated'><p>Deprecated. May already stop() code and ask to be updated. May be dropped entirely in future.</p></a></li>
<li><a href='#umxACE'><p>Build and run a 2-group Cholesky ACE twin model (univariate or multivariate)</p></a></li>
<li><a href='#umxACEcov'><p>Run a Cholesky with covariates that are random (in the expected covariance matrix)</p></a></li>
<li><a href='#umxACEv'><p>Build and run 2-group uni- or multi-variate ACE models based on VARIANCE (not paths).</p></a></li>
<li><a href='#umxAlgebra'><p>A simple wrapper for mxAlgebra with name as the first parameter for more readable compact code.</p></a></li>
<li><a href='#umxAPA'><p>Creates nicely formatted journal-style summaries of models, p-values, data-frames and much more.</p></a></li>
<li><a href='#umxBrownie'><p>A recipe Easter-egg for umx</p></a></li>
<li><a href='#umxCI'><p>Add (and, optionally, run) confidence intervals to a structural model.</p></a></li>
<li><a href='#umxCI_boot'><p>umxCI_boot</p></a></li>
<li><a href='#umxCLPM'><p>Runs cross-lagged panel models</p></a></li>
<li><a href='#umxCompare'><p>Print a comparison table of one or more <code>mxModel()</code>s, formatted nicely.</p></a></li>
<li><a href='#umxConfint'><p>Get confidence intervals from a umx model</p></a></li>
<li><a href='#umxCov2cor'><p>Convert a covariance matrix into a correlation matrix</p></a></li>
<li><a href='#umxCP'><p>umxCP: Build and run a Common Pathway twin model</p></a></li>
<li><a href='#umxDiagnose'><p>Diagnose problems in a model - this is a work in progress.</p></a></li>
<li><a href='#umxDiffMZ'><p>MZ differences method for testing evidence for causality.</p></a></li>
<li><a href='#umxDiscTwin'><p>Intra-pair association in MZ, DZ  twin models. (ALPHA quality!)</p></a></li>
<li><a href='#umxDoC'><p>Build and run a 2-group Direction of Causation twin models.</p></a></li>
<li><a href='#umxDoCp'><p>Make a direction of causation model based on umxPath statements</p></a></li>
<li><a href='#umxEFA'><p>FIML-based Exploratory Factor Analysis (EFA)</p></a></li>
<li><a href='#umxEquate'><p>umxEquate: Equate two or more paths</p></a></li>
<li><a href='#umxExamples'><p>Example code from Twin Research and Human Genetics Paper on umx</p></a></li>
<li><a href='#umxExpCov'><p>Get the expected vcov matrix</p></a></li>
<li><a href='#umxExpMeans'><p>Extract the expected means matrix from an <code>mxModel()</code></p></a></li>
<li><a href='#umxFactor'><p>umxFactor</p></a></li>
<li><a href='#umxFactorScores'><p>Return factor scores from a model as an easily consumable dataframe.</p></a></li>
<li><a href='#umxFitIndices'><p>Get additional fit-indices for a model with umxFitIndices</p></a></li>
<li><a href='#umxFixAll'><p>umxFixAll: Fix all free parameters</p></a></li>
<li><a href='#umxGetLatents'><p>Get the latentVars from a RAM model</p></a></li>
<li><a href='#umxGetManifests'><p>Get the manifestVars from a RAM model</p></a></li>
<li><a href='#umxGetModel'><p>Used to get a RAM submodel by name</p></a></li>
<li><a href='#umxGetParameters'><p>Get parameters from a model, with support for pattern matching!</p></a></li>
<li><a href='#umxGxE'><p>umxGxE: Implements ACE models with moderation of paths, e.g. by SES.</p></a></li>
<li><a href='#umxGxE_window'><p>Implement the moving-window form of GxE analysis.</p></a></li>
<li><a href='#umxGxEbiv'><p>Purcell (2002) Bivariate GxE model: Suitable when twins differ on the moderator.</p></a></li>
<li><a href='#umxHetCor'><p>Create a matrix of correlations for variables of diverse types (binary, ordinal, continuous)</p></a></li>
<li><a href='#umxIP'><p>umxIP: Build and run an Independent Pathway twin model</p></a></li>
<li><a href='#umxJiggle'><p>umxJiggle</p></a></li>
<li><a href='#umxLav2RAM'><p>Convert lavaan string to a umxRAM model</p></a></li>
<li><a href='#umxMatrix'><p>Make a mxMatrix with automatic labels. Also takes name as the first parameter for more readable code.</p></a></li>
<li><a href='#umxMatrixFree'><p>Sets labeled matrix cells to free</p></a></li>
<li><a href='#umxMI'><p>Report modifications which would improve fit.</p></a></li>
<li><a href='#umxModel'><p>Catches users typing umxModel instead of umxRAM.</p></a></li>
<li><a href='#umxModelNames'><p>Return names of models found within a model</p></a></li>
<li><a href='#umxModify'><p>umxModify: Add, set, or drop model paths by label.</p></a></li>
<li><a href='#umxMRDoC'><p>Extends Mendelian randomization with the twin design to test evidence of causality</p></a></li>
<li><a href='#umxParameters'><p>Display path estimates from a model, filtering by name and value.</p></a></li>
<li><a href='#umxParan'><p>A wrapper to make paran easier to use.</p>
Just automates applying <code>complete.cases()</code></a></li>
<li><a href='#umxPath'><p>Easier (and powerful) specification of paths in SEM.</p></a></li>
<li><a href='#umxPlot'><p>Quickly plot y ~ x with a regression line and R^2, and nice labels.</p></a></li>
<li><a href='#umxPlotACE'><p>Make a graphical display of an ACE model</p></a></li>
<li><a href='#umxPlotACEcov'><p>Make a graphical display of an ACE model with covariates.</p></a></li>
<li><a href='#umxPlotACEv'><p>Produce a graphical display of an ACE variance-components twin model</p></a></li>
<li><a href='#umxPlotCP'><p>Draw and display a graphical figure of Common Pathway model</p></a></li>
<li><a href='#umxPlotDoC'><p>Plot a Direction of Causation Model.</p></a></li>
<li><a href='#umxPlotFun'><p>Easily plot functions in R</p></a></li>
<li><a href='#umxPlotGxE'><p>Plot the results of a GxE univariate test for moderation of ACE components.</p></a></li>
<li><a href='#umxPlotGxEbiv'><p>Plot the results of a GxE univariate test for moderation of ACE components.</p></a></li>
<li><a href='#umxPlotIP'><p>Draw a graphical figure for a Independent Pathway model</p></a></li>
<li><a href='#umxPlotSexLim'><p>Draw and display a graphical figure of a Sex limitation model</p></a></li>
<li><a href='#umxPlotSimplex'><p>Draw and display a graphical figure of a simplex model</p></a></li>
<li><a href='#umxPower'><p>Test power to detect specified path values in a model.</p></a></li>
<li><a href='#umxRAM'><p>Build and run path-based SEM models</p></a></li>
<li><a href='#umxRAM2Lav'><p>Convert a RAM model to a lavaan string</p></a></li>
<li><a href='#umxReduce'><p>Reduce models, and report the results.</p></a></li>
<li><a href='#umxReduceACE'><p>Reduce an ACE model.</p></a></li>
<li><a href='#umxReduceGxE'><p>Reduce a GxE model.</p></a></li>
<li><a href='#umxRenameMatrix'><p>Rename a umxMatrix (even in a model)</p></a></li>
<li><a href='#umxRotate'><p>Generic SEM factor model loading rotation function</p></a></li>
<li><a href='#umxRotate.MxModelCP'><p>Rotate a CP solution</p></a></li>
<li><a href='#umxRun'><p>umxRun: Run an mxModel</p></a></li>
<li><a href='#umxSetParameters'><p>Change or fix parameters (e.g. their values, labels, bounds, ..) in a model.</p></a></li>
<li><a href='#umxSexLim'><p>Multivariate sex limitation twin model</p></a></li>
<li><a href='#umxSimplex'><p>Build and run a simplex twin model (not ready for use!)</p></a></li>
<li><a href='#umxSummarizeTwinData'><p>Summarize twin data</p></a></li>
<li><a href='#umxSummary'><p>Shows a compact, publication-style, summary of umx models</p></a></li>
<li><a href='#umxSummary.MxModel'><p>Shows a compact, publication-style, summary of a RAM model</p></a></li>
<li><a href='#umxSummaryACE'><p>Shows a compact, publication-style, summary of a umx Cholesky ACE model</p></a></li>
<li><a href='#umxSummaryACEcov'><p>Present results of a twin ACE-model with covariates in table and graphical forms.</p></a></li>
<li><a href='#umxSummaryACEv'><p>Shows a compact, publication-style, summary of a variance-based Cholesky ACE model.</p></a></li>
<li><a href='#umxSummaryCP'><p>Present the results of a Common-pathway twin model in table and graphical form</p></a></li>
<li><a href='#umxSummaryDoC'><p>Shows a compact, publication-style, summary of a umx Direction of Causation model</p></a></li>
<li><a href='#umxSummaryGxE'><p>Summarize a GxE model</p></a></li>
<li><a href='#umxSummaryGxEbiv'><p>Summarize a bivariate GxE twin model</p></a></li>
<li><a href='#umxSummaryIP'><p>Present the results of an independent-pathway twin model in table and graphical form</p></a></li>
<li><a href='#umxSummaryMRDoC'><p>Present the results of a Mendelian Randomization Direction of Causation Model in a table</p></a></li>
<li><a href='#umxSummarySexLim'><p>Shows a compact, publication-style, summary of a umx Sex Limitation model</p></a></li>
<li><a href='#umxSummarySimplex'><p>Shows a compact, publication-style, summary of a Simplex model.</p></a></li>
<li><a href='#umxSuperModel'><p>Make a multi-group model</p></a></li>
<li><a href='#umxThresholdMatrix'><p>Create the threshold matrix needed for modeling ordinal data.</p></a></li>
<li><a href='#umxTwinMaker'><p>Make a twin model from the model describing just one person</p></a></li>
<li><a href='#umxTwoStage'><p>Build a SEM implementing the instrumental variable design</p></a></li>
<li><a href='#umxUnexplainedCausalNexus'><p>umxUnexplainedCausalNexus</p></a></li>
<li><a href='#umxVersion'><p>Get or print the version of umx, along with detail from OpenMx and general system info.</p></a></li>
<li><a href='#umxWeightedAIC'><p>AIC weight-based conditional probabilities.</p></a></li>
<li><a href='#us_skinfold_data'><p>Anthropometric data on twins</p></a></li>
<li><a href='#xmu_bracket_address2rclabel'><p>Convert a bracket address into an A_rXcX-style label.</p></a></li>
<li><a href='#xmu_cell_is_on'><p>Return whether a cell is in a set location of a matrix</p></a></li>
<li><a href='#xmu_check_levels_identical'><p>xmu_check_levels_identical</p></a></li>
<li><a href='#xmu_check_needs_means'><p>Check data to see if model needs means.</p></a></li>
<li><a href='#xmu_check_variance'><p>Check the minimum variance in data frame</p></a></li>
<li><a href='#xmu_CI_merge'><p>xmu_CI_merge</p></a></li>
<li><a href='#xmu_CI_stash'><p>Stash the CI values of a model as strings in the values of the model</p></a></li>
<li><a href='#xmu_clean_label'><p>Remove illegal characters from labels</p></a></li>
<li><a href='#xmu_data_missing'><p>Drop rows with missing definition variables</p></a></li>
<li><a href='#xmu_data_swap_a_block'><p>Data helper function to swap blocks of data from one set of columns to another.</p></a></li>
<li><a href='#xmu_describe_data_WLS'><p>Determine if a dataset will need statistics for the means if used in a WLS model.</p></a></li>
<li><a href='#xmu_DF_to_mxData_TypeCov'><p>Convert a dataframe into a cov mxData object</p></a></li>
<li><a href='#xmu_dot_define_shapes'><p>Helper to make the list of vars and their shapes for a graphviz string</p></a></li>
<li><a href='#xmu_dot_make_paths'><p>xmu_dot_make_paths (not for end users)</p></a></li>
<li><a href='#xmu_dot_make_residuals'><p>xmu_dot_make_residuals (not for end users)</p></a></li>
<li><a href='#xmu_dot_maker'><p>Internal umx function to help plotting graphviz</p></a></li>
<li><a href='#xmu_dot_mat2dot'><p>Return dot code for paths in a matrix</p></a></li>
<li><a href='#xmu_dot_move_ranks'><p>xmu_dot_move_ranks (not for end users)</p></a></li>
<li><a href='#xmu_dot_rank'><p>Helper to make a graphviz rank string</p></a></li>
<li><a href='#xmu_dot_rank_str'><p>xmu_dot_rank_str (not for end users)</p></a></li>
<li><a href='#xmu_extract_column'><p>Get one or more columns from mzData or regular data.frame</p></a></li>
<li><a href='#xmu_get_CI'><p>Look up and report CIs for free parameters</p></a></li>
<li><a href='#xmu_lavaan_process_group'><p>Process table of paths to model</p></a></li>
<li><a href='#xmu_make_bin_cont_pair_data'><p>Make pairs of  bin &amp; continuous columns to represent censored data</p></a></li>
<li><a href='#xmu_make_mxData'><p>Upgrade a dataframe to an mxData type.</p></a></li>
<li><a href='#xmu_make_TwinSuperModel'><p>Helper to make a basic top, MZ, and DZ model.</p></a></li>
<li><a href='#xmu_match.arg'><p>Select first item in list of options, while being flexible about choices.</p></a></li>
<li><a href='#xmu_name_from_lavaan_str'><p>Find name for model</p></a></li>
<li><a href='#xmu_PadAndPruneForDefVars'><p>Where all data are missing for a twin, add default values for definition variables, allowing the row to be kept</p></a></li>
<li><a href='#xmu_path_regex'><p>Re-name variables umxPaths to twin versions</p></a></li>
<li><a href='#xmu_path2twin'><p>Re-name variables in umxPaths to twin versions</p></a></li>
<li><a href='#xmu_print_algebras'><p>Print algebras from a umx model</p></a></li>
<li><a href='#xmu_rclabel_2_bracket_address'><p>Convert an &quot;A_r1c1&quot;-style label to a bracket address.</p></a></li>
<li><a href='#xmu_safe_run_summary'><p>Safely run and summarize a model</p></a></li>
<li><a href='#xmu_set_sep_from_suffix'><p>Just a helper to cope with deprecated suffix lying around.</p></a></li>
<li><a href='#xmu_show_fit_or_comparison'><p>Show model logLik of model or print comparison table</p></a></li>
<li><a href='#xmu_simplex_corner'><p>Internal function to help building simplex models</p></a></li>
<li><a href='#xmu_standardize_ACE'><p>xmu_standardize_ACE</p></a></li>
<li><a href='#xmu_standardize_ACEcov'><p>xmu_standardize_ACEcov</p></a></li>
<li><a href='#xmu_standardize_ACEv'><p>Standardize an ACE variance components model (ACEv)</p></a></li>
<li><a href='#xmu_standardize_CP'><p>Function to standardize a common pathway model</p></a></li>
<li><a href='#xmu_standardize_IP'><p>non-user: Standardize an IP model</p></a></li>
<li><a href='#xmu_standardize_RAM'><p>Standardize a Structural Model (not for end users)</p></a></li>
<li><a href='#xmu_standardize_SexLim'><p>Standardize a SexLim model</p></a></li>
<li><a href='#xmu_standardize_Simplex'><p>Standardize a Simplex twin model</p></a></li>
<li><a href='#xmu_start_value_list'><p>Make start values</p></a></li>
<li><a href='#xmu_starts'><p>Helper providing boilerplate start values for means and variance in twin models</p></a></li>
<li><a href='#xmu_summary_RAM_group_parameters'><p>Order and group the parameters in a RAM summary</p></a></li>
<li><a href='#xmu_twin_add_WeightMatrices'><p>Add weight matrices to twin models.</p></a></li>
<li><a href='#xmu_twin_check'><p>Check basic aspects of input for twin models.</p></a></li>
<li><a href='#xmu_twin_get_var_names'><p>Not for user: pull variable names from a twin model</p></a></li>
<li><a href='#xmu_twin_make_def_means_mats_and_alg'><p>Make the matrices and algebras for definition-based means models</p></a></li>
<li><a href='#xmu_twin_upgrade_selDvs2SelVars'><p>Upgrade selDVs to selVars</p></a></li>
<li><a href='#xmuHasSquareBrackets'><p>xmuHasSquareBrackets</p></a></li>
<li><a href='#xmuLabel'><p>xmuLabel: Add labels to a RAM model, matrix, or path</p></a></li>
<li><a href='#xmuLabel_Matrix'><p>xmuLabel_Matrix (not a user function)</p></a></li>
<li><a href='#xmuLabel_MATRIX_Model'><p>xmuLabel_MATRIX_Model (not a user function)</p></a></li>
<li><a href='#xmuLabel_RAM_Model'><p>xmuLabel_RAM_Model (not a user function)</p></a></li>
<li><a href='#xmuMakeDeviationThresholdsMatrices'><p>Make a deviation-based mxRAMObjective for ordinal models.</p></a></li>
<li><a href='#xmuMakeOneHeadedPathsFromPathList'><p>xmuMakeOneHeadedPathsFromPathList</p></a></li>
<li><a href='#xmuMakeTwoHeadedPathsFromPathList'><p>xmuMakeTwoHeadedPathsFromPathList</p></a></li>
<li><a href='#xmuMaxLevels'><p>xmuMaxLevels</p></a></li>
<li><a href='#xmuMI'><p>xmuMI (not for end users)</p></a></li>
<li><a href='#xmuMinLevels'><p>xmuMinLevels</p></a></li>
<li><a href='#xmuPropagateLabels'><p>xmuPropagateLabels (not a user function)</p></a></li>
<li><a href='#xmuRAM2Ordinal'><p>xmuRAM2Ordinal</p></a></li>
<li><a href='#xmuTwinSuper_Continuous'><p>Create core of twin model for all-continuous data.</p></a></li>
<li><a href='#xmuTwinSuper_NoBinary'><p>xmuTwinSuper_NoBinary</p></a></li>
<li><a href='#xmuTwinUpgradeMeansToCovariateModel'><p>Not for end-users: Add a means model with covariates to a twin model</p></a></li>
<li><a href='#xmuValues'><p>xmuValues: Set values in RAM model, matrix, or path</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>4.20.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-04-12</td>
</tr>
<tr>
<td>Title:</td>
<td>Structural Equation Modeling and Twin Modeling in R</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Timothy C. Bates &lt;timothy.c.bates@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/tbates/umx#readme">https://github.com/tbates/umx#readme</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Quickly create, run, and report structural equation models, and twin models.
    See '?umx' for help, and umx_open_CRAN_page("umx") for NEWS.
    Timothy C. Bates, Michael C. Neale, Hermine H. Maes, (2019). umx: A library for Structural Equation and Twin Modelling in R.
    Twin Research and Human Genetics, 22, 27-41. &lt;<a href="https://doi.org/10.1017%2Fthg.2019.2">doi:10.1017/thg.2019.2</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), OpenMx (&ge; 2.19.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>cowplot, DiagrammeR, ggplot2, kableExtra, knitr, lavaan, MASS,
Matrix, methods, MuMIn, mvtnorm, nlme, polycor, R2HTML, RCurl,
scales, utils, xtable, gert, hrbrthemes, openxlsx, foreign,
pwr, rmarkdown, psych, psychTools, paran</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cocor, devtools, rhub, spelling, testthat, parallel,
GPArotation</td>
</tr>
<tr>
<td>Enhances:</td>
<td>DiagrammeRsvg, rsvg</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/tbates/umx/issues">https://github.com/tbates/umx/issues</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-10 20:53:13 UTC; tim</td>
</tr>
<tr>
<td>Author:</td>
<td>Timothy C. Bates <a href="https://orcid.org/0000-0002-1153-9007"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Nathan Gillespie [wit],
  Hermine Maes [ctb],
  Michael C. Neale [ctb],
  Joshua N. Pritikin [ctb],
  Luis De Araujo [ctb],
  Brenton Wiernik [ctb],
  Michael Zakharin [wit]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-10 23:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='umx'>Functions for Structural Equation Modeling in OpenMx</h2><span id='topic+umx'></span><span id='topic+umx-package'></span>

<h3>Description</h3>

<p><code>umx</code> allows you to more easily build, run, modify, and report structural models,
building on the OpenMx package.
All core functions are organized into families, so they are easier to find
(so if you know a function similar to what you are looking for, look at other members of its &quot;family&quot; at the bottom of its help file.
</p>
<p>Please cite as: Bates, T. C., Neale, M. C., &amp; Maes, H. H. (2019).
umx: A library for Structural Equation and Twin Modelling in R.
<em>Twin Research and Human Genetics</em>, <strong>22</strong>, 27-41. <a href="https://doi.org/10.1017/thg.2019.2">doi:10.1017/thg.2019.2</a>.
</p>
<p>All the functions have full-featured and well commented examples, some even have <em>figures</em>,
so use the help, even if you think it won't help :-)
Have a look, for example at <code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
<p>Check out NEWS about new features at <code>news(package = "umx")</code>
</p>


<h3>Details</h3>

<p>Introductory working examples are below. You can run all demos with demo(umx)
When I have a vignette, it will be: vignette(&quot;umx&quot;, package = &quot;umx&quot;)
</p>
<p>There is a helpful blog at <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
<p>(Only) if you want the bleeding-edge version:
</p>
<p>devtools::install_github(&quot;tbates/umx&quot;)
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Timothy C. Bates <a href="mailto:timothy.c.bates@gmail.com">timothy.c.bates@gmail.com</a> (<a href="https://orcid.org/0000-0002-1153-9007">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Nathan Gillespie <a href="mailto:nathan.gillespie@vcuhealth.org">nathan.gillespie@vcuhealth.org</a> [witness]
</p>
</li>
<li><p> Hermine Maes <a href="mailto:hmeaes@vcu.edu">hmeaes@vcu.edu</a> [contributor]
</p>
</li>
<li><p> Michael C. Neale <a href="mailto:neale@vcu.edu">neale@vcu.edu</a> [contributor]
</p>
</li>
<li><p> Joshua N. Pritikin <a href="mailto:jpritikin@pobox.com">jpritikin@pobox.com</a> [contributor]
</p>
</li>
<li><p> Luis De Araujo <a href="mailto:ldearaujo@unimelb.edu.au">ldearaujo@unimelb.edu.au</a> [contributor]
</p>
</li>
<li><p> Brenton Wiernik <a href="mailto:wiernik@umn.edu">wiernik@umn.edu</a> [contributor]
</p>
</li>
<li><p> Michael Zakharin <a href="mailto:s1775682@sms.ed.ac.uk">s1775682@sms.ed.ac.uk</a> [witness]
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Bates, T. C., Neale, M. C., &amp; Maes, H. H. (2019). umx: A library for Structural
Equation and Twin Modelling in R. <em>Twin Research and Human Genetics</em>, <strong>22</strong>, 27-41. <a href="https://doi.org/10.1017/thg.2019.2">doi:10.1017/thg.2019.2</a>,
<a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, tutorial: <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/tbates/umx#readme">https://github.com/tbates/umx#readme</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/tbates/umx/issues">https://github.com/tbates/umx/issues</a>
</p>
</li></ul>

<p>Other Core Model Building Functions: 
<code><a href="#topic+umxMatrix">umxMatrix</a>()</code>,
<code><a href="#topic+umxModify">umxModify</a>()</code>,
<code><a href="#topic+umxPath">umxPath</a>()</code>,
<code><a href="#topic+umxRAM">umxRAM</a>()</code>,
<code><a href="#topic+umxSuperModel">umxSuperModel</a>()</code>
</p>
<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxCompare">umxCompare</a>()</code>,
<code><a href="#topic+umxEquate">umxEquate</a>()</code>,
<code><a href="#topic+umxMI">umxMI</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSetParameters">umxSetParameters</a>()</code>,
<code><a href="#topic+umxSummary">umxSummary</a>()</code>
</p>
<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>
</p>
<p>Other Super-easy helpers: 
<code><a href="#topic+umxEFA">umxEFA</a>()</code>,
<code><a href="#topic+umxTwoStage">umxTwoStage</a>()</code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>
</p>
<p>Other Twin Data functions: 
<code><a href="#topic+umx_long2wide">umx_long2wide</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_twin_data_nice">umx_make_twin_data_nice</a>()</code>,
<code><a href="#topic+umx_residualize">umx_residualize</a>()</code>,
<code><a href="#topic+umx_scale_wide_twin_data">umx_scale_wide_twin_data</a>()</code>,
<code><a href="#topic+umx_wide2long">umx_wide2long</a>()</code>
</p>
<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>
</p>
<p>Other Teaching and testing Functions: 
<code><a href="#topic+tmx_genotypic_effect">tmx_genotypic_effect</a>()</code>,
<code><a href="#topic+tmx_is.identified">tmx_is.identified</a>()</code>
</p>
<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>
</p>
<p>Other Check or test: 
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_is_class">umx_is_class</a>()</code>,
<code><a href="#topic+umx_is_endogenous">umx_is_endogenous</a>()</code>,
<code><a href="#topic+umx_is_exogenous">umx_is_exogenous</a>()</code>,
<code><a href="#topic+umx_is_numeric">umx_is_numeric</a>()</code>,
<code><a href="#topic+umx_is_ordered">umx_is_ordered</a>()</code>
</p>
<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>
</p>
<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>
</p>
<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>
</p>
<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>
</p>
<p>Other datasets: 
<code><a href="#topic+Fischbein_wt">Fischbein_wt</a></code>,
<code><a href="#topic+GFF">GFF</a></code>,
<code><a href="#topic+docData">docData</a></code>,
<code><a href="#topic+iqdat">iqdat</a></code>,
<code><a href="#topic+us_skinfold_data">us_skinfold_data</a></code>
</p>
<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require("umx")
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type="cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G"  , fixedAt= 1)
)

# umx added informative labels, created starting values, 
# Ran your model (if autoRun is on), and displayed a brief summary
# including a comparison if you modified a model...!

# umxSummary generates journal-ready fit information.
# We can choose std=T for standardized parameters and can also
# filter out some types of parameter (e.g. means or residuals)

umxSummary(m1, std = TRUE, residuals=FALSE)

# parameters() flexibly retrieves model coefficients. 
# For example just G-loadings greater than |.3| and rounded to 2-digits.
parameters(m1, thresh="above", b=.3, pattern = "G_to.*", digits = 2)

# (The built-in coef works as for lm etc.)
coef(m1)

# ==================
# = Model updating =
# ==================
# umxModify modifies, renames, re-runs, and compares a model
# Can we set the loading of x1 on G to zero? (nope...)
m2 = umxModify(m1, "G_to_x1", name = "no_effect_of_g_on_X1", comparison = TRUE)

# note1: umxSetParameters can do this with some additional flexibility
# note2 "comparison = TRUE" above is the same as calling 
# umxCompare, like this
umxCompare(m1, m2)


# ========================
# = Confidence intervals =
# ========================

# umxSummary() will show these, but you can also use the confint() function
confint(m1) # OpenMx's SE-based confidence intervals


# umxConfint formats everything you need nicely, and allows adding CIs (with parm=)
umxConfint(m1, parm = 'all', run = TRUE) # likelihood-based CIs

# And make a Figure and open in browser
plot(m1, std = TRUE)

# If you just want the .dot code returned set file = NA
plot(m1, std = TRUE, file = NA)

## End(Not run)

</code></pre>

<hr>
<h2 id='bucks'>Print a money object</h2><span id='topic+bucks'></span><span id='topic+print'></span>

<h3>Description</h3>

<p>Print function for &quot;money&quot; objects, e.g. <code><a href="#topic+fin_interest">fin_interest()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bucks(
  x,
  symbol = "$",
  big.mark = ",",
  decimal.mark = ".",
  trim = TRUE,
  largest_with_cents = 1e+05,
  negative_parens = c("hyphen", "minus", "parens"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bucks_+3A_x">x</code></td>
<td>
<p>money object.</p>
</td></tr>
<tr><td><code id="bucks_+3A_symbol">symbol</code></td>
<td>
<p>Default prefix if not set.</p>
</td></tr>
<tr><td><code id="bucks_+3A_big.mark">big.mark</code></td>
<td>
<p>option defaulting to &quot;,&quot;</p>
</td></tr>
<tr><td><code id="bucks_+3A_decimal.mark">decimal.mark</code></td>
<td>
<p>option defaulting to &quot;.&quot;</p>
</td></tr>
<tr><td><code id="bucks_+3A_trim">trim</code></td>
<td>
<p>option defaulting to TRUE</p>
</td></tr>
<tr><td><code id="bucks_+3A_largest_with_cents">largest_with_cents</code></td>
<td>
<p>option defaulting to 1e+05</p>
</td></tr>
<tr><td><code id="bucks_+3A_negative_parens">negative_parens</code></td>
<td>
<p>option defaulting to &quot;hyphen&quot;</p>
</td></tr>
<tr><td><code id="bucks_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods. also cat =F to return string</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> invisible
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_percent">fin_percent()</a></code>, <code><a href="#topic+fin_interest">fin_interest()</a></code>, <code><a href="scales.html#topic+dollar_format">scales::dollar()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>bucks(100 * 1.05^32)
fin_interest(deposits = 20e3, interest = 0.07, yrs = 20)

</code></pre>

<hr>
<h2 id='deg2rad'>Convert Degrees to Degrees</h2><span id='topic+deg2rad'></span>

<h3>Description</h3>

<p>A helper to convert degrees (360 in  a circle) to Rad (<code class="reqn">2\pi</code>
in a circle).
</p>
<p><em>note</em>: R's trig functions, e.g. <code><a href="base.html#topic+sin">sin()</a></code> use Radians for input!
</p>
<p>The formula is radians = <code class="reqn">deg x 180/\pi</code>.
</p>

<ul>
<li><p> 180 Degrees is equal to <code class="reqn">\pi</code> radians.
</p>
</li>
<li><p> 1 Rad = <code class="reqn">180/\pi</code> degrees = ~ 57.296 degrees.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>deg2rad(deg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deg2rad_+3A_deg">deg</code></td>
<td>
<p>The value in degrees you wish to convert to radians</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> value in radians
</p>
</li></ul>



<h3>References</h3>

<p><a href="https://en.wikipedia.org/wiki/Radian">https://en.wikipedia.org/wiki/Radian</a>
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+rad2deg">rad2deg()</a></code>, <code><a href="base.html#topic+sin">sin()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>deg2rad(180) == pi # TRUE!
</code></pre>

<hr>
<h2 id='dl_from_dropbox'>dl_from_dropbox</h2><span id='topic+dl_from_dropbox'></span>

<h3>Description</h3>

<p>Download a file from Dropbox, given either the url, or the name and key
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dl_from_dropbox(x, key = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dl_from_dropbox_+3A_x">x</code></td>
<td>
<p>Either the file name, or full dropbox URL (see example below)</p>
</td></tr>
<tr><td><code id="dl_from_dropbox_+3A_key">key</code></td>
<td>
<p>the code after s/ and before the file name in the dropbox url</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Improvements would include error handling...
</p>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>

<p>- <a href="https://thebiobucket.blogspot.kr/2013/04/download-files-from-dropbox.html">https://thebiobucket.blogspot.kr/2013/04/download-files-from-dropbox.html</a>
</p>


<h3>See Also</h3>

<p>Other File Functions: 
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dl_from_dropbox("https://dl.dropboxusercontent.com/s/7kauod48r9cfhwc/tinytwinData.rda")
dl_from_dropbox("tinytwinData.rda", key = "7kauod48r9cfhwc")

## End(Not run)
</code></pre>

<hr>
<h2 id='docData'>Twin data for Direction of causation modelling</h2><span id='topic+docData'></span>

<h3>Description</h3>

<p>A dataset containing indicators for two traits <code>varA</code> and <code>varB</code>, each measured in MZ and DZ twins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(docData)
</code></pre>


<h3>Format</h3>

<p>A data frame 6 manifests for each of two twins in 1400 families of MZ and DZ twins
</p>


<h3>Details</h3>

<p>It is designed to show off <code><a href="#topic+umxDoC">umxDoC()</a></code> testing the hypothesis <code>varA</code> causes <code>varB</code>, <code>varB</code> causes <code>varA</code>, both cause each other.
</p>

<ul>
<li> <p><em>zygosity</em> &quot;MZFF&quot;, &quot;DZFF&quot;, &quot;MZMM&quot;, or &quot;DZMM&quot;
</p>
</li>
<li> <p><em>varA1_T1</em> Twin one's manifest 1 for varA
</p>
</li>
<li> <p><em>varA2_T1</em> Twin one's manifest 2 for varA
</p>
</li>
<li> <p><em>varA3_T1</em> Twin one's manifest 3 for varA
</p>
</li>
<li> <p><em>varB1_T1</em> Twin one's manifest 1 for varB
</p>
</li>
<li> <p><em>varB2_T1</em> Twin one's manifest 2 for varB
</p>
</li>
<li> <p><em>varB3_T1</em> Twin one's manifest 3 for varB
</p>
</li>
<li> <p><em>varA1_T2</em> Twin two's manifest 1 for varA
</p>
</li>
<li> <p><em>varA2_T2</em> Twin two's manifest 2 for varA
</p>
</li>
<li> <p><em>varA3_T2</em> Twin two's manifest 3 for varA
</p>
</li>
<li> <p><em>varB1_T2</em> Twin two's manifest 1 for varB
</p>
</li>
<li> <p><em>varB2_T2</em> Twin two's manifest 2 for varB
</p>
</li>
<li> <p><em>varB3_T2</em> Twin two's manifest 3 for varB
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> N.A. Gillespie and N.G. Martin (2005). Direction of Causation Models.
In <em>Encyclopedia of Statistics in Behavioral Science</em>, <strong>1</strong>, 496–499. Eds. Brian S. Everitt &amp; David C. Howell
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>, <code><a href="#topic+plot.MxModelDoC">plot.MxModelDoC()</a></code>, <code><a href="#topic+umxSummary.MxModelDoC">umxSummary.MxModelDoC()</a></code>, <code><a href="#topic+umxModify">umxModify()</a></code>
</p>
</li></ul>

<p>Other datasets: 
<code><a href="#topic+Fischbein_wt">Fischbein_wt</a></code>,
<code><a href="#topic+GFF">GFF</a></code>,
<code><a href="#topic+iqdat">iqdat</a></code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+us_skinfold_data">us_skinfold_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(docData)
str(docData)
mzData = subset(docData, zygosity %in% c("MZFF", "MZMM"))
dzData = subset(docData, zygosity %in% c("DZFF", "DZMM"))
par(mfrow = c(1, 2))  # 1 rows and 3 columns
plot(varA1_T2 ~varA1_T1, ylim = c(-4, 4), data = mzData, main="MZ")
tmp = round(cor.test(~varA1_T1 + varA1_T2, data = mzData)$estimate, 2)
text(x=-4, y=3, labels = paste0("r = ", tmp))
plot(varA1_T2 ~varA1_T1, ylim = c(-4, 4), data = dzData, main="DZ")
tmp = round(cor.test(~varA1_T1 + varA1_T2, data = dzData)$estimate, 2)
text(x=-4, y=3, labels = paste0("r = ", tmp))
par(mfrow = c(1, 1))  # back to as it was
</code></pre>

<hr>
<h2 id='extractAIC.MxModel'>Extract AIC from MxModel</h2><span id='topic+extractAIC.MxModel'></span>

<h3>Description</h3>

<p>Returns the AIC for an OpenMx model.
Original Author: Brandmaier
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
extractAIC(fit, scale, k, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractAIC.MxModel_+3A_fit">fit</code></td>
<td>
<p>an fitted <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to get the AIC</p>
</td></tr>
<tr><td><code id="extractAIC.MxModel_+3A_scale">scale</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="extractAIC.MxModel_+3A_k">k</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="extractAIC.MxModel_+3A_...">...</code></td>
<td>
<p>any other parameters (not used)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> AIC value
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://openmx.ssri.psu.edu/thread/931#comment-4858">https://openmx.ssri.psu.edu/thread/931#comment-4858</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+AIC">AIC()</a></code>, <code><a href="#topic+umxCompare">umxCompare()</a></code>, <code><a href="stats.html#topic+logLik">logLik()</a></code>
</p>
</li></ul>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
extractAIC(m1)
# -2.615998
AIC(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='fin_interest'>Compute the value of a principal &amp; annual deposits at a compound interest over a number of years</h2><span id='topic+fin_interest'></span>

<h3>Description</h3>

<p>Allows you to determine the final value of an initial <code>principal</code> (with optional
periodic <code>deposits</code>), over a number of years (<code>yrs</code>) at a given rate of <code>interest</code>.
Principal and deposits are optional. You control compounding periods each year (n) and whether deposits occur at the beginning or end of the year.
The function outputs a nice table of annual returns, formats the total using a user-settable currency <code>symbol</code>. Can also <code>report</code> using a web table.
</p>
<p><em>notes</em>: Graham valuation: fair P/E = 9 + (1.5 * growth%). e.g.  $INTEL fair P/E = 9+.5<em>3 = 10.5 up to  9+2</em>10 = 29
Can move the weighting between a conservative .5 and an optimistic 2 (in terms of how long the growth will last and how low the hurdle rate is)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fin_interest(
  principal = 0,
  deposits = 0,
  inflate = 0,
  interest = 0.05,
  yrs = 10,
  final = NULL,
  n = 12,
  when = "beginning",
  symbol = NULL,
  largest_with_cents = 0,
  baseYear = as.numeric(format(Sys.time(), "%Y")),
  table = TRUE,
  report = c("markdown", "html")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fin_interest_+3A_principal">principal</code></td>
<td>
<p>The initial investment at time 0.</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_deposits">deposits</code></td>
<td>
<p>Optional periodic additional investment each <em>year</em>.</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_inflate">inflate</code></td>
<td>
<p>How much to inflate deposits over time (default = 0)</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_interest">interest</code></td>
<td>
<p>Annual interest rate (default = .05)</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_yrs">yrs</code></td>
<td>
<p>Duration of the investment (default = 10).</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_final">final</code></td>
<td>
<p>if set (default = NULL), returns the rate required to turn principal into final after yrs (principal defaults to 1)</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_n">n</code></td>
<td>
<p>Compounding intervals per year (default = 12 (monthly), 365 for daily)</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_when">when</code></td>
<td>
<p>Deposits made at the &quot;beginning&quot; (of each year) or &quot;end&quot;</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_symbol">symbol</code></td>
<td>
<p>Currency symbol to embed in the result.</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_largest_with_cents">largest_with_cents</code></td>
<td>
<p>Default = 0</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_baseyear">baseYear</code></td>
<td>
<p>Default = current year (for table row labels)</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_table">table</code></td>
<td>
<p>Whether to print a table of annual returns (default TRUE)</p>
</td></tr>
<tr><td><code id="fin_interest_+3A_report">report</code></td>
<td>
<p>&quot;markdown&quot; or &quot;html&quot;,</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Value of balance after yrs of investment.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://en.wikipedia.org/wiki/Compound_interest">https://en.wikipedia.org/wiki/Compound_interest</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol()</a></code>, <code><a href="#topic+fin_percent">fin_percent()</a></code>, <code><a href="#topic+fin_NI">fin_NI()</a></code>, <code><a href="#topic+fin_valuation">fin_valuation()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# 1. Value of a principal after yrs years at 5% return, compounding monthly.
# Report in browser as a nice table of annual returns and formatted totals.
fin_interest(principal = 5000, interest = 0.05, rep= "html")

## End(Not run)

# Report as a nice markdown table
fin_interest(principal = 5000, interest = 0.05, yrs = 10)

umx_set_dollar_symbol("£")
# 2 What rate is needed to increase principal to final value in yrs time?
fin_interest(final = 1.4, yrs=5)
fin_interest(principal = 50, final=200, yrs = 5)

# 3. What's the value of deposits of $100/yr after 10 years at 7% return?
fin_interest(deposits = 100, interest = 0.07, yrs = 10, n = 12)

# 4. What's the value of $20k + $100/yr over 10 years at 7% return?
fin_interest(principal= 20e3, deposits= 100, interest= .07, yrs= 10, symbol="$")

# 5. What is $10,000 invested at the end of each year for 5 years at 6%?
fin_interest(deposits = 10e3, interest = 0.06, yrs = 5, n=1, when= "end")

# 6. What will $20k be worth after 10 years at 15% annually (n=1)?
fin_interest(deposits=20e3, interest = 0.15, yrs = 10, n=1, baseYear=1)
# $466,986

# manual equivalent
sum(20e3*(1.15^(10:1))) # 466985.5

# 7. Annual (rather than monthly) compounding (n=1)
fin_interest(deposits = 100, interest = 0.07, yrs = 10, n=1)

# 8 Interest needed to increase principal to final value in yrs time.
fin_interest(principal = 100, final=200, yrs = 5)

</code></pre>

<hr>
<h2 id='fin_JustifiedPE'>Justified P/E Ratio</h2><span id='topic+fin_JustifiedPE'></span>

<h3>Description</h3>

<p>Compute the Justified P/E of a stock.
Justified P/E = ( (DPS / EPS) * (1 + g)) / (k – g)
DPS is the dividend per share, EPS is the earnings per share,
g is the sustainable growth rate, and k is the required rate of return.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fin_JustifiedPE(
  Dividend = 0.02,
  EPS = 1,
  growthRate = 0.08,
  discountRate = 0.12,
  basePE = 20,
  yrs = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fin_JustifiedPE_+3A_dividend">Dividend</code></td>
<td>
<p>The dividend.</p>
</td></tr>
<tr><td><code id="fin_JustifiedPE_+3A_eps">EPS</code></td>
<td>
<p>The Earnings per Share.</p>
</td></tr>
<tr><td><code id="fin_JustifiedPE_+3A_growthrate">growthRate</code></td>
<td>
<p>The growth rate.</p>
</td></tr>
<tr><td><code id="fin_JustifiedPE_+3A_discountrate">discountRate</code></td>
<td>
<p>Your chosen discount rate.</p>
</td></tr>
<tr><td><code id="fin_JustifiedPE_+3A_basepe">basePE</code></td>
<td>
<p>The base PE.</p>
</td></tr>
<tr><td><code id="fin_JustifiedPE_+3A_yrs">yrs</code></td>
<td>
<p>Years.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A PE that is justified for this stock.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_interest">fin_interest()</a></code>, <code><a href="#topic+fin_percent">fin_percent()</a></code>, <code><a href="#topic+fin_NI">fin_NI()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># fin_JustifiedPE(Dividend= .8, EPS = 2, growthRate = .06, discountRate = .1)

</code></pre>

<hr>
<h2 id='fin_NI'>Compute NI given annual Earnings.</h2><span id='topic+fin_NI'></span>

<h3>Description</h3>

<p>Employees pay contributions at 12%% on annual earnings between £9,568 and £50,270. Above that you pay at 2%%.
Employers pay at 13.8%% on all annual earnings of more than £8,840, although there are different thresholds
for those under the age of 21 and for apprentices under the age of 25.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fin_NI(annualEarnings, symbol = "£")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fin_NI_+3A_annualearnings">annualEarnings</code></td>
<td>
<p>Employee annual earnings.</p>
</td></tr>
<tr><td><code id="fin_NI_+3A_symbol">symbol</code></td>
<td>
<p>Currency symbol to embed in the result.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> NI
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://www.telegraph.co.uk/tax/tax-hacks/politicians-running-scared-long-overdue-national-insurance-overhaul/">https://www.telegraph.co.uk/tax/tax-hacks/politicians-running-scared-long-overdue-national-insurance-overhaul/</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_interest">fin_interest()</a></code>, <code><a href="#topic+fin_percent">fin_percent()</a></code>, <code><a href="#topic+fin_valuation">fin_valuation()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fin_NI(42e3)
fin_NI(142000)

</code></pre>

<hr>
<h2 id='fin_percent'>Compute the percent change needed to return to the original value after percent off (or on).</h2><span id='topic+fin_percent'></span>

<h3>Description</h3>

<p>Determine the percent change needed to &quot;undo&quot; an initial percent change. Has a plot function as well.
If an amount of $100 has 20% added, what percent do we need to drop it by to return to the original value?
</p>
<p><code>fin_percent(20)</code> yields $100 increased by 20% = $120 (Percent to reverse = -17%)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fin_percent(
  percent,
  value = 100,
  symbol = "$",
  digits = 2,
  plot = TRUE,
  logY = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fin_percent_+3A_percent">percent</code></td>
<td>
<p>Change in percent (enter 10 for 10%, not 0.1)</p>
</td></tr>
<tr><td><code id="fin_percent_+3A_value">value</code></td>
<td>
<p>Principal</p>
</td></tr>
<tr><td><code id="fin_percent_+3A_symbol">symbol</code></td>
<td>
<p>value units (default = &quot;$&quot;)</p>
</td></tr>
<tr><td><code id="fin_percent_+3A_digits">digits</code></td>
<td>
<p>Rounding of results (default 2 places)</p>
</td></tr>
<tr><td><code id="fin_percent_+3A_plot">plot</code></td>
<td>
<p>Whether to plot the result (default TRUE)</p>
</td></tr>
<tr><td><code id="fin_percent_+3A_logy">logY</code></td>
<td>
<p>Whether to plot y axis as log (TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> new value and change required to return to baseline.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_interest">fin_interest()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Percent needed to return to original value after 10% taken off
fin_percent(-10)

# Percent needed to return to original value after 10% added on
fin_percent(10)

# Percent needed to return to original value after 50% off 34.50
fin_percent(-50, value = 34.5)
</code></pre>

<hr>
<h2 id='fin_ticker'>Open a ticker in yahoo finance.</h2><span id='topic+fin_ticker'></span>

<h3>Description</h3>

<p>Open a stock ticker, currently in yahoo finance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fin_ticker(ticker = "INTC")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fin_ticker_+3A_ticker">ticker</code></td>
<td>
<p>A stock symbol to look up, e.g., &quot;OXY&quot;</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Open a ticker in a finance site online
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_interest">fin_interest()</a></code>, <code><a href="#topic+fin_percent">fin_percent()</a></code>, <code><a href="#topic+fin_NI">fin_NI()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Open $INTC in yahoo finance.
## Not run: 
fin_ticker("INTC")

## End(Not run)
</code></pre>

<hr>
<h2 id='fin_valuation'>Work the valuation of a company</h2><span id='topic+fin_valuation'></span>

<h3>Description</h3>

<p><code>fin_valuation</code> uses the revenue, operating margin, expenses and PE to compute a market capitalization.
Better to use a more powerful online site.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fin_valuation(
  revenue = 6e+06 * 30000,
  opmargin = 0.08,
  expenses = 0.2,
  PE = 30,
  symbol = "$",
  use = c("B", "M")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fin_valuation_+3A_revenue">revenue</code></td>
<td>
<p>Revenue of the company</p>
</td></tr>
<tr><td><code id="fin_valuation_+3A_opmargin">opmargin</code></td>
<td>
<p>Margin on operating revenue</p>
</td></tr>
<tr><td><code id="fin_valuation_+3A_expenses">expenses</code></td>
<td>
<p>Additional fixed costs</p>
</td></tr>
<tr><td><code id="fin_valuation_+3A_pe">PE</code></td>
<td>
<p>of the company</p>
</td></tr>
<tr><td><code id="fin_valuation_+3A_symbol">symbol</code></td>
<td>
<p>Currency</p>
</td></tr>
<tr><td><code id="fin_valuation_+3A_use">use</code></td>
<td>
<p>reporting values in &quot;B&quot; (billion) or &quot;M&quot; (millions)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Revenue is multiplied by opmargin to get a gross profit. From this the proportion specified in <code>expenses</code> is subtracted
and the resulting earnings turned into a price via the <code>PE</code>
</p>


<h3>Value</h3>


<ul>
<li><p> value
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_interest">fin_interest()</a></code>, <code><a href="#topic+fin_NI">fin_NI()</a></code>, <code><a href="#topic+fin_percent">fin_percent()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fin_valuation(rev=7e9, opmargin=.1, PE=33)
# Market cap =  $18,480,000,000
# (Based on PE= 33, operating Income of $0.70 B, and net income =$0.56B

</code></pre>

<hr>
<h2 id='Fischbein_wt'>Weight data across time.</h2><span id='topic+Fischbein_wt'></span>

<h3>Description</h3>

<p>A dataframe containing correlations of weight for 66 females measured 6 times at 6-month intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fischbein_wt)
</code></pre>


<h3>Format</h3>

<p>A 6*6 correlation matrix based on n = 66 female subjects.
</p>


<h3>Details</h3>


<ul>
<li><p> Weight1: Weight at time 1 (t0)
</p>
</li>
<li><p> Weight2: Weight at time 2 (t0 + 6 months)
</p>
</li>
<li><p> Weight3: Weight at time 3 (t0 + 12 months)
</p>
</li>
<li><p> Weight4: Weight at time 4 (t0 + 18 months)
</p>
</li>
<li><p> Weight5: Weight at time 5 (t0 + 24 months)
</p>
</li>
<li><p> Weight6: Weight at time 6 (t0 + 32 months)
</p>
</li></ul>

<p>Created as follows:
</p>
<div class="sourceCode"><pre>Fischbein_wt = umx_read_lower(file = "", diag = TRUE, names = paste0("Weight", 1:6), ensurePD= TRUE)
1.000
0.985	1.000
0.968	0.981	1.000
0.957	0.970	0.985	1.000
0.932	0.940	0.964	0.975	1.000
0.890	0.897	0.927	0.949	0.973	1.000
</pre></div>


<h3>References</h3>

<p>Fischbein, S. (1977). Intra-pair similarity in physical growth of monozygotic and of dizygotic twins during puberty.
<em>Annals of Human Biology</em>, <strong>4</strong>. 417-430. <a href="https://doi.org/10.1080/03014467700002401">doi:10.1080/03014467700002401</a>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+GFF">GFF</a></code>,
<code><a href="#topic+docData">docData</a></code>,
<code><a href="#topic+iqdat">iqdat</a></code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+us_skinfold_data">us_skinfold_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(Fischbein_wt) # load the data
str(Fischbein_wt) # data.frame
as.matrix(Fischbein_wt) # convert to matrix

## End(Not run)
</code></pre>

<hr>
<h2 id='FishersMethod'>Fishers Method of combining p-values.</h2><span id='topic+FishersMethod'></span>

<h3>Description</h3>

<p><code>FishersMethod</code> implements R.A. Fisher's (1925) method for creating a meta-analytic p-value by combining a
set of p-values from tests of the same hypothesis in independent samples. See also Stouffer's method
for combining Z scores, which allows weighting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FishersMethod(pvalues, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FishersMethod_+3A_pvalues">pvalues</code></td>
<td>
<p>A vector of p-values, e.g. c(.041, .183)</p>
</td></tr>
<tr><td><code id="FishersMethod_+3A_...">...</code></td>
<td>
<p>More p-values if you want to offer them up one by one instead of wrapping in a vector for <code>pvalues</code></p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A meta-analytic p-value
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Fisher, R.A. (1925). <em>Statistical Methods for Research Workers</em>. Oliver and Boyd (Edinburgh). ISBN 0-05-002170-2.
</p>
</li></ul>


<ul>
<li><p> Fisher, R. A (1948). &quot;Questions and answers #14&quot;. <em>The American Statistician</em>. <strong>2</strong>: 30–31. <a href="https://doi.org/10.2307/2681650">doi:10.2307/2681650</a>.
</p>
</li>
<li><p> Stouffer, S. A. and Suchman, E. A. and DeVinney, L. C. and Star, S. A. and Williams, R. M. Jr. (1949) The American Soldier,
Vol. 1 - Adjustment during Army Life. Princeton, Princeton University Press.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>FishersMethod(c(.041, .378))
</code></pre>

<hr>
<h2 id='geometric_mean'>Geometric Mean</h2><span id='topic+geometric_mean'></span>

<h3>Description</h3>

<p>The Geometric mean is the nth-root of the product of <code>n</code> input values.
Common uses include computing economic utility.  That is to say, the utility of
<code>c(1, 2, 10)</code> is </p>
<p style="text-align: center;"><code class="reqn">(1 * 2 * 10)/3</code>
</p>
<p> = 6.6 not 6.3 (the arithmetic mean).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geometric_mean(x, na.rm = c(TRUE, FALSE))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geometric_mean_+3A_x">x</code></td>
<td>
<p>A vector of values.</p>
</td></tr>
<tr><td><code id="geometric_mean_+3A_na.rm">na.rm</code></td>
<td>
<p>remove NAs by default.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Geometric mean of x
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://en.wikipedia.org/wiki/Geometric_mean">https://en.wikipedia.org/wiki/Geometric_mean</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geometric_mean(c(50, 100))

# For a given sum, geometric mean is maximised with equality
geometric_mean(c(75,75))

v = c(1, 149); c(sum(v), geometric_mean(v), mean(v), median(v))
# 150.00000  12.20656  75.00000  75.00000

# Underlying logic
sqrt(50 * 100)

# Alternate form using logs
exp(mean(log(c(50 *100))))

# Reciprocal duality
1/geometric_mean(c(100, 50))
geometric_mean(c(1/100, 1/50))
</code></pre>

<hr>
<h2 id='GFF'>Twin data: General Family Functioning, divorce, and well-being.</h2><span id='topic+GFF'></span>

<h3>Description</h3>

<p>Measures of family functioning, happiness and related variables in twins, and
their brothers and sisters. (see details)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GFF)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows of twin-family data columns.
</p>


<h3>Details</h3>

<p>Several scales in the data are described in  van der Aa et al. (2010).
General Family Functioning (GFF) refers to adolescents' evaluations general family health
vs. pathology. It assesses problem solving, communication, roles within the household,
affection, and control. GFF was assessed with a Dutch translation of the General Functioning
sub-scale of the McMaster Family Assessment Device (FAD) (Epstein et al., 1983).
</p>
<p>Family Conflict (FC) refers to adolescents' evaluations of the amount of openly
expressed anger, aggression, and conflict among family members. Conflict
sub-scale of the Family Environment Scale (FES) (Moos, 1974)
</p>
<p>Quality of life in general (QLg) was assessed with the 10-step Cantril
Ladder from best- to worst-possible life (Cantril, 1965).
</p>

<ul>
<li> <p><em>zyg_6grp</em>: Six-level zygosity: MZMM, DZMM, MZFF, DZFF, DZMF, DZFM
</p>
</li>
<li> <p><em>zyg_2grp</em>: Two-level zygosity measure: 'MZ', 'DZ'
</p>
</li>
<li> <p><em>divorce</em>: Parental divorce status: 0 = No, 1 = Yes
</p>
</li>
<li> <p><em>sex_T1</em>: Sex of twin 1: 0 = &quot;male&quot;, 1 = &quot;female&quot;
</p>
</li>
<li> <p><em>age_T1</em>: Age of twin 1 (years)
</p>
</li>
<li> <p><em>gff_T1</em>: General family functioning for twin 1
</p>
</li>
<li> <p><em>fc_T1</em>: Family conflict sub-scale of the FES
</p>
</li>
<li> <p><em>qol_T1</em>: Quality of life for twin 1
</p>
</li>
<li> <p><em>hap_T1</em>: General happiness for twin 1
</p>
</li>
<li> <p><em>sat_T1</em>: Satisfaction with life for twin 1
</p>
</li>
<li> <p><em>AD_T1</em>: Anxiety and Depression for twin 1
</p>
</li>
<li> <p><em>SOMA_T1</em>: Somatic complaints for twin 1
</p>
</li>
<li> <p><em>SOC_T1</em>: Social problems for twin 1
</p>
</li>
<li> <p><em>THOU_T1</em>: Thought disorder problems for twin 1
</p>
</li>
<li> <p><em>sex_T2</em>: Sex of twin 2
</p>
</li>
<li> <p><em>age_T2</em>: Age of twin 2
</p>
</li>
<li> <p><em>gff_T2</em>: General family functioning for twin 2
</p>
</li>
<li> <p><em>fc_T2</em>: Family conflict sub-scale of the FES
</p>
</li>
<li> <p><em>qol_T2</em>: Quality of life for twin 2
</p>
</li>
<li> <p><em>hap_T2</em>: General happiness for twin 2
</p>
</li>
<li> <p><em>sat_T2</em>: Satisfaction with life for twin 2
</p>
</li>
<li> <p><em>AD_T2</em>: Anxiety and Depression for twin 2
</p>
</li>
<li> <p><em>SOMA_T2</em>: Somatic complaints for twin 2
</p>
</li>
<li> <p><em>SOC_T2</em>: Social problems for twin 2
</p>
</li>
<li> <p><em>THOU_T2</em>: Thought disorder problems for twin 2
</p>
</li>
<li> <p><em>sex_Ta</em>: Sex of sib 1
</p>
</li>
<li> <p><em>age_Ta</em>: Age of sib 1
</p>
</li>
<li> <p><em>gff_Ta</em>: General family functioning for sib 1
</p>
</li>
<li> <p><em>fc_Ta</em>: Family conflict sub-scale of the FES
</p>
</li>
<li> <p><em>qol_Ta</em>: Quality of life for sib 1
</p>
</li>
<li> <p><em>hap_Ta</em>: General happiness for sib 1
</p>
</li>
<li> <p><em>sat_Ta</em>: Satisfaction with life for sib 1
</p>
</li>
<li> <p><em>AD_Ta</em>: Anxiety and Depression for sib 1
</p>
</li>
<li> <p><em>SOMA_Ta</em>: Somatic complaints for sib 1
</p>
</li>
<li> <p><em>SOC_Ta</em>: Social problems for sib 1
</p>
</li>
<li> <p><em>THOU_Ta</em>: Thought disorder problems for sib 1
</p>
</li>
<li> <p><em>sex_Ts</em>: Sex of sib 2
</p>
</li>
<li> <p><em>age_Ts</em>: Age of sib 2
</p>
</li>
<li> <p><em>gff_Ts</em>: General family functioning for sib 2
</p>
</li>
<li> <p><em>fc_Ts</em>: Family conflict sub-scale of the FES
</p>
</li>
<li> <p><em>qol_Ts</em>: Quality of life for sib 2
</p>
</li>
<li> <p><em>hap_Ts</em>: General happiness for sib 2
</p>
</li>
<li> <p><em>sat_Ts</em>: Satisfaction with life for sib 2
</p>
</li>
<li> <p><em>AD_Ts</em>: Anxiety and Depression for sib 2
</p>
</li>
<li> <p><em>SOMA_Ts</em>: Somatic complaints for sib 2
</p>
</li>
<li> <p><em>SOC_Ts</em>: Social problems for sib 2
</p>
</li>
<li> <p><em>THOU_Ts</em>: Thought disorder problems for sib 2
</p>
</li></ul>



<h3>References</h3>

<p>van der Aa, N., Boomsma, D. I., Rebollo-Mesa, I., Hudziak, J. J., &amp; Bartels,
M. (2010). Moderation of genetic factors by parental divorce in adolescents'
evaluations of family functioning and subjective wellbeing. Twin Research
and Human Genetics, <strong>13</strong>, 143-162. <a href="https://doi.org/10.1375/twin.13.2.143">doi:10.1375/twin.13.2.143</a>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+Fischbein_wt">Fischbein_wt</a></code>,
<code><a href="#topic+docData">docData</a></code>,
<code><a href="#topic+iqdat">iqdat</a></code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+us_skinfold_data">us_skinfold_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Twin 1 variables (end in '_T1')
data(GFF)
umx_names(GFF, "1$") # Just variables ending in 1 (twin 1)
str(GFF) # first few rows


m1 = umxACE(selDVs= "gff", sep = "_T",
	mzData = subset(GFF, zyg_2grp == "MZ"), 
	dzData = subset(GFF, zyg_2grp == "DZ")
)

## End(Not run)

</code></pre>

<hr>
<h2 id='ggAddR'>Add a fit statistic to a ggplot</h2><span id='topic+ggAddR'></span>

<h3>Description</h3>

<p>Add a fit statistic to a ggplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggAddR(model, effect = NA, xloc = 8, yloc = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggAddR_+3A_model">model</code></td>
<td>
<p>a statistical model which contains a fit measure.</p>
</td></tr>
<tr><td><code id="ggAddR_+3A_effect">effect</code></td>
<td>
<p>optional hard coded fit/effect.</p>
</td></tr>
<tr><td><code id="ggAddR_+3A_xloc">xloc</code></td>
<td>
<p>x location of R.</p>
</td></tr>
<tr><td><code id="ggAddR_+3A_yloc">yloc</code></td>
<td>
<p>y location of R.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> plot
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxPlot">umxPlot()</a></code>, <code><a href="#topic+umxPlotFun">umxPlotFun()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
m1 = lm(mpg ~ wt, data = mtcars)
p = ggplot2::ggplot(data = mtcars, aes(x = wt, y = mpg))+ geom_point() +geom_smooth()+
ggAddR(m1, effect = NA, xloc=2, yloc= 10); p

## End(Not run)
</code></pre>

<hr>
<h2 id='harmonic_mean'>Harmonic Mean</h2><span id='topic+harmonic_mean'></span>

<h3>Description</h3>

<p>The harmonic mean is the reciprocal of the arithmetic mean of the reciprocals of the input values.
Common uses include computing the mean of ratios, for instance the average P/E ratio in a portfolio.
Also it is the correct mean for averaging speeds weighted for distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harmonic_mean(x, weights = NULL, na.rm = c(TRUE, FALSE))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="harmonic_mean_+3A_x">x</code></td>
<td>
<p>A vector of values to take the harmonic mean for</p>
</td></tr>
<tr><td><code id="harmonic_mean_+3A_weights">weights</code></td>
<td>
<p>Optional vector of weights.</p>
</td></tr>
<tr><td><code id="harmonic_mean_+3A_na.rm">na.rm</code></td>
<td>
<p>remove NAs (default = TRUE).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Harmonic mean of x
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://en.wikipedia.org/wiki/Harmonic_mean">https://en.wikipedia.org/wiki/Harmonic_mean</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Harmonic means are suitable for ratios
tmp = c(33/1, 23/1)
harmonic_mean(tmp)

geometric_mean(tmp)
mean(tmp)

# Example with weights
harmonic_mean(c(33/1, 23/1), weights= c(.2, .8))
# If Jack travels outbound at 1 mph, and returns at 10 miles an hour, what is his average speed?
harmonic_mean(c(1,10)) # 1.81 mph

</code></pre>

<hr>
<h2 id='install.OpenMx'>Install OpenMx, with choice of builds</h2><span id='topic+install.OpenMx'></span><span id='topic+umx_update_OpenMx'></span>

<h3>Description</h3>

<p>You can install OpenMx, including the latest NPSOL-enabled build of OpenMx. Options are:
</p>

<ol>
<li><p> &quot;NPSOL&quot;: Install from our repository (default): This is where we maintain binaries supporting parallel processing and NPSOL.
</p>
</li>
<li><p> &quot;travis&quot;: Install the latest travis built (MacOS only).
</p>
</li>
<li><p> &quot;CRAN&quot;: Install from CRAN.
</p>
</li>
<li><p> &quot;open travis build page&quot;: Open the list of travis builds in a browser window.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>install.OpenMx(
  loc = c("NPSOL", "travis", "CRAN", "open travis build page", "UVa"),
  url = NULL,
  lib,
  repos = getOption("repos")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install.OpenMx_+3A_loc">loc</code></td>
<td>
<p>Version to get default is &quot;NPSOL&quot;. &quot;travis&quot; (latest build),CRAN, list of builds.</p>
</td></tr>
<tr><td><code id="install.OpenMx_+3A_url">url</code></td>
<td>
<p>Custom URL. On Mac, set this to &quot;Finder&quot; and the package selected in the Finder will be installed.</p>
</td></tr>
<tr><td><code id="install.OpenMx_+3A_lib">lib</code></td>
<td>
<p>Where to install the package.</p>
</td></tr>
<tr><td><code id="install.OpenMx_+3A_repos">repos</code></td>
<td>
<p>Which repository to use (ignored currently).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxVersion">umxVersion()</a></code>
</p>
<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
install.OpenMx() # gets the NPSOL version
install.OpenMx("NPSOL") # gets the NPSOL version explicitly
install.OpenMx("CRAN") # Get the latest CRAN version
install.OpenMx("open travis build page") # Open web page of travis builds

## End(Not run)
</code></pre>

<hr>
<h2 id='iqdat'>Twin data: IQ measured longitudinally across 4 ages.</h2><span id='topic+iqdat'></span>

<h3>Description</h3>

<p>Measures of IQ across four ages in 261 pairs of identical twins and 301 pairs of fraternal (DZ) twins. (see details).
It is used as data for the [umxSimplex()] examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(iqdat)
</code></pre>


<h3>Format</h3>

<p>A data frame with 562 rows (twin families). Nine measures on each twin.
</p>


<h3>Details</h3>


<ul>
<li><p> zygosity Zygosity (MZ or DZ)
</p>
</li>
<li><p> IQ_age1_T1 T1 IQ measured at age 1
</p>
</li>
<li><p> IQ_age2_T1 T1 IQ measured at age 2
</p>
</li>
<li><p> IQ_age3_T1 T1 IQ measured at age 3
</p>
</li>
<li><p> IQ_age4_T1 T1 IQ measured at age 4
</p>
</li>
<li><p> IQ_age1_T2 T2 IQ measured at age 1
</p>
</li>
<li><p> IQ_age2_T2 T2 IQ measured at age 2
</p>
</li>
<li><p> IQ_age3_T2 T2 IQ measured at age 3
</p>
</li>
<li><p> IQ_age4_T2 T2 IQ measured at age 4
</p>
</li></ul>



<h3>References</h3>

<p>Boomsma, D. I., Martin, N. G., &amp; Molenaar, P. C. (1989). Factor and simplex models for repeated measures: 
application to two psychomotor measures of alcohol sensitivity in twins. *Behavior Genetics*, **19**, 79-96.
Retrieved from &lt;https://www.ncbi.nlm.nih.gov/pubmed/2712815&gt;
</p>


<h3>See Also</h3>

<p>[umxSimplex()]
</p>
<p>Other datasets: 
<code><a href="#topic+Fischbein_wt">Fischbein_wt</a></code>,
<code><a href="#topic+GFF">GFF</a></code>,
<code><a href="#topic+docData">docData</a></code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+us_skinfold_data">us_skinfold_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(iqdat)
str(iqdat)
par(mfrow = c(1, 3))  # 1 rows and 3 columns
plot(IQ_age4_T1 ~ IQ_age4_T2, ylim = c(50, 150), data = subset(iqdat, zygosity == "MZ"))
plot(IQ_age4_T1 ~ IQ_age4_T2, ylim = c(50, 150), data = subset(iqdat, zygosity == "DZ"))
plot(IQ_age1_T1 ~ IQ_age4_T2, data = subset(iqdat, zygosity == "MZ"))
par(mfrow = c(1, 1))  # back to as it was

## End(Not run)
</code></pre>

<hr>
<h2 id='libs'>load libraries</h2><span id='topic+libs'></span>

<h3>Description</h3>

<p><code>libs</code> allows loading multiple libraries in one call
</p>


<h3>Usage</h3>

<pre><code class='language-R'>libs(..., force.update = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="libs_+3A_...">...</code></td>
<td>
<p>library names as strings, e.g. &quot;pwr&quot;</p>
</td></tr>
<tr><td><code id="libs_+3A_force.update">force.update</code></td>
<td>
<p>install.package even if present (to get new version) FALSE</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> nothing.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="base.html#topic+library">library()</a></code>, <code><a href="utils.html#topic+install.packages">install.packages()</a></code>, <code><a href="utils.html#topic+remove.packages">remove.packages()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
libs("umx", "OpenMx", "car")
libs("umx", c("OpenMx", "car"))
remove.packages()

## End(Not run)
</code></pre>

<hr>
<h2 id='loadings'>loadings
Generic loadings function to extract factor loadings from exploratory or confirmatory
factor analyses.</h2><span id='topic+loadings'></span>

<h3>Description</h3>

<p>See <code><a href="#topic+loadings.MxModel">loadings.MxModel</a></code> to access the loadings of OpenMx EFA models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadings(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadings_+3A_x">x</code></td>
<td>
<p>an object from which to get loadings</p>
</td></tr>
<tr><td><code id="loadings_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Base <code><a href="stats.html#topic+loadings">loadings</a></code> handles <code><a href="stats.html#topic+factanal">factanal()</a></code> objects.
</p>


<h3>Value</h3>


<ul>
<li><p> matrix of loadings
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>

<hr>
<h2 id='loadings.MxModel'>Extract factor loadings from an EFA (factor analysis).</h2><span id='topic+loadings.MxModel'></span>

<h3>Description</h3>

<p>loadings extracts the factor loadings from an EFA (factor analysis) model.
It behaves equivalently to stats::loadings, returning the loadings from an
EFA (factor analysis). However it does not store the rotation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
loadings(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadings.MxModel_+3A_x">x</code></td>
<td>
<p>A RAM model from which to get loadings.</p>
</td></tr>
<tr><td><code id="loadings.MxModel_+3A_...">...</code></td>
<td>
<p>Other parameters (currently unused)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> loadings matrix
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+factanal">factanal()</a></code>, <code><a href="#topic+loadings">loadings()</a></code>
</p>
</li></ul>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
myVars = c("mpg", "disp", "hp", "wt", "qsec")
m1 = umxEFA(name = "test", factors = 2, data = mtcars[, myVars])
loadings(m1)

## End(Not run)

</code></pre>

<hr>
<h2 id='noNAs'>Succinctly select complete rows from a dataframe</h2><span id='topic+noNAs'></span>

<h3>Description</h3>

<p>Succinctly select complete rows from a dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noNAs(df, rows = NULL, cols = NULL, drop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="noNAs_+3A_df">df</code></td>
<td>
<p>an <code><a href="base.html#topic+data.frame">data.frame()</a></code> to select on</p>
</td></tr>
<tr><td><code id="noNAs_+3A_rows">rows</code></td>
<td>
<p>Rows to keep (optional, incomplete rows still discarded)</p>
</td></tr>
<tr><td><code id="noNAs_+3A_cols">cols</code></td>
<td>
<p>Cols to keep</p>
</td></tr>
<tr><td><code id="noNAs_+3A_drop">drop</code></td>
<td>
<p>Whether to return a vector when only 1 column is selected (default TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Complete rows and (optionally) selected columns
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = mtcars
tmp[2,1] = NA
noNAs(tmp, cols="mpg")
noNAs(tmp, cols="mpg", drop = FALSE)
noNAs(tmp) # no Mazda RX4 Wag

</code></pre>

<hr>
<h2 id='oddsratio'>Compute odds ratio (OR)</h2><span id='topic+oddsratio'></span>

<h3>Description</h3>

<p>Returns the odds in each group, and the odds ratio. Takes the cases (n) and total N as a list
of two numbers for each of two groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oddsratio(grp1 = c(n = 3, N = 10), grp2 = c(n = 1, N = 10), alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oddsratio_+3A_grp1">grp1</code></td>
<td>
<p>either odds for group 1, or cases and total N , e.g c(n=3, N=10)</p>
</td></tr>
<tr><td><code id="oddsratio_+3A_grp2">grp2</code></td>
<td>
<p>either odds for group 2, or cases and total N , e.g c(n=1, N=20)</p>
</td></tr>
<tr><td><code id="oddsratio_+3A_alpha">alpha</code></td>
<td>
<p>for CI (default = 0.05)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns a list of odds1, odds2, and OR + CI. Has a pretty-printing method so displays as:
</p>
<div class="sourceCode"><pre>Group 1 odds = 0.43
Group 2 odds = 0.11
          OR = 3.86 CI95[0.160, 3.64]
</pre></div>


<h3>Value</h3>


<ul>
<li><p> List of odds in group 1 and group2, and the resulting OR and CI
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://stats.oarc.ucla.edu/r/dae/logit-regression/">https://stats.oarc.ucla.edu/r/dae/logit-regression/</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_r_test">umx_r_test()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oddsratio(grp1 = c(1, 10), grp2 = c(3, 10))
oddsratio(grp1 = 0.111, grp2 = 0.429)
oddsratio(grp1 = c(3, 10), grp2 = c(1, 10))
oddsratio(grp1 = c(3, 10), grp2 = c(1, 10), alpha = .01)

</code></pre>

<hr>
<h2 id='plot.MxLISRELModel'>Create and display a graphical path diagram for a LISREL model.</h2><span id='topic+plot.MxLISRELModel'></span>

<h3>Description</h3>

<p><code>plot.MxLISRELModel</code> produces SEM diagrams using <code><a href="DiagrammeR.html#topic+DiagrammeR">DiagrammeR::DiagrammeR()</a></code> (or a graphviz application) to create the image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxLISRELModel'
plot(
  x = NA,
  std = FALSE,
  fixed = TRUE,
  means = TRUE,
  digits = 2,
  file = "name",
  labels = c("none", "labels", "both"),
  resid = c("circle", "line", "none"),
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MxLISRELModel_+3A_x">x</code></td>
<td>
<p>A LISREL <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to make a path diagram</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (default = FALSE).</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_fixed">fixed</code></td>
<td>
<p>Whether to show fixed paths (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_means">means</code></td>
<td>
<p>Whether to show means or not (default = TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_digits">digits</code></td>
<td>
<p>The number of decimal places to add to the path coefficients</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_labels">labels</code></td>
<td>
<p>Whether to show labels on the paths. both will show both the parameter and the label. (&quot;both&quot;, &quot;none&quot; or &quot;labels&quot;)</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_resid">resid</code></td>
<td>
<p>How to show residuals and variances default is &quot;circle&quot;. Options are &quot;line&quot; &amp; &quot;none&quot;</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxLISRELModel_+3A_...">...</code></td>
<td>
<p>Optional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Note:</em> By default, plots open in your browser (or plot pane if using RStudio).
</p>
<p><strong>Opening in an external editor/app</strong>
</p>
<p>The underlying format is graphviz.
If you use <code>umx_set_plot_format("graphviz")</code>, figures will open in a graphviz helper app (if installed).
If you use graphviz, we try and use that app, but YOU HAVE TO INSTALL IT!
</p>
<p>On MacOS, you may need to associate the &lsquo;.gv&rsquo; extension with your graphviz app.
Find the .gv file made by plot, get info (cmd-I), then choose &ldquo;open with&rdquo;,
select graphviz.app (or OmniGraffle professional),
then set &ldquo;change all&rdquo;.
</p>
<p>The commercial application &ldquo;OmniGraffle&rdquo; is great for editing these images.
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://en.wikipedia.org/wiki/DOT_(graph_description_language)">https://en.wikipedia.org/wiki/DOT_(graph_description_language)</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_set_plot_format">umx_set_plot_format()</a></code>, <code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot()</a></code>, <code><a href="#topic+umx_set_plot_format">umx_set_plot_format()</a></code>, <code><a href="#topic+plot.MxModel">plot.MxModel()</a></code>, <code><a href="#topic+umxPlotACE">umxPlotACE()</a></code>, <code><a href="#topic+umxPlotCP">umxPlotCP()</a></code>, <code><a href="#topic+umxPlotIP">umxPlotIP()</a></code>, <code><a href="#topic+umxPlotGxE">umxPlotGxE()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># plot()
# TODO get LISREL example model
# Figure out how to map its matrices to plot. Don't do without establishing demand.
</code></pre>

<hr>
<h2 id='plot.MxModel'>Create and display a graphical path diagram for a model.</h2><span id='topic+plot.MxModel'></span><span id='topic+plot'></span>

<h3>Description</h3>

<p><code><a href="#topic+plot">plot()</a></code> produces SEM diagrams in graphviz format, and relies on <code><a href="DiagrammeR.html#topic+DiagrammeR">DiagrammeR()</a></code> (or a
graphviz application) to create the image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
plot(
  x = NA,
  std = FALSE,
  fixed = TRUE,
  means = TRUE,
  digits = 2,
  file = "name",
  labels = c("none", "labels", "both"),
  resid = c("circle", "line", "none"),
  strip_zero = FALSE,
  splines = c("TRUE", "FALSE", "compound", "ortho", "polyline"),
  min = NULL,
  same = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MxModel_+3A_x">x</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to make a path diagram</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (default = FALSE).</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_fixed">fixed</code></td>
<td>
<p>Whether to show fixed paths (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_means">means</code></td>
<td>
<p>Whether to show means or not (default = TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_digits">digits</code></td>
<td>
<p>The number of decimal places to add to the path coefficients</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_labels">labels</code></td>
<td>
<p>Whether to show labels on the paths. &quot;none&quot;, &quot;labels&quot;, or &quot;both&quot; (parameter + label).</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_resid">resid</code></td>
<td>
<p>How to show residuals and variances default is &quot;circle&quot;. Options are &quot;line&quot; &amp; &quot;none&quot;</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = FALSE)</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_splines">splines</code></td>
<td>
<p>Whether to allow lines to curve: defaults to &quot;TRUE&quot; (nb: some models look better with &quot;FALSE&quot;)</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_min">min</code></td>
<td>
<p>optional list of objects to group at the top of the plot. Default (NULL) chooses automatically.</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_same">same</code></td>
<td>
<p>optional list of objects to group at the same rank in the plot. Default (NULL) chooses automatically.</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_max">max</code></td>
<td>
<p>optional list of objects to group at the bottom of the plot. Default (NULL) chooses automatically.</p>
</td></tr>
<tr><td><code id="plot.MxModel_+3A_...">...</code></td>
<td>
<p>Optional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>Note:</em> DiagrammeR is supported out of the box.  By default, plots open in your browser. Other options include pdf SVG etc.
</p>
<p>If you use <code>umx_set_plot_format("graphviz")</code>, graphs will open in a graphviz helper app (if installed).
</p>
<p>The commercial application &ldquo;OmniGraffle&rdquo; is great for editing these images.
On unix and windows, <code><a href="#topic+plot">plot()</a></code> will create a pdf and open it in your default pdf reader.
</p>
<p>If you use graphviz, we try and use that app, but YOU HAVE TO INSTALL IT!
</p>
<p><em>MacOS note</em>: On Mac, we will try and open an app: you may need to associate the &lsquo;.gv&rsquo;
extension with the graphviz app.
Find the .gv file made by plot, get info (cmd-I), then choose &ldquo;open with&rdquo;,
select graphviz.app (or OmniGraffle professional),
then set &ldquo;change all&rdquo;.
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://en.wikipedia.org/wiki/DOT_(graph_description_language)">https://en.wikipedia.org/wiki/DOT_(graph_description_language)</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_set_plot_format">umx_set_plot_format()</a></code>, <code><a href="#topic+plot.MxModel">plot.MxModel()</a></code>, <code><a href="#topic+umxPlotACE">umxPlotACE()</a></code>, <code><a href="#topic+umxPlotCP">umxPlotCP()</a></code>, <code><a href="#topic+umxPlotIP">umxPlotIP()</a></code>, <code><a href="#topic+umxPlotGxE">umxPlotGxE()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
plot(m1)
plot(m1, std = TRUE, resid = "line", digits = 3, strip_zero = FALSE)

# ============================================================
# = With a growth model, demonstrate splines= false to force =
# = straight lines, and move "rank" of intercept object      =
# ============================================================

m1 = umxRAM("grow", data = myGrowthMixtureData,
	umxPath(var = manifests, free = TRUE), 
	umxPath(means = manifests, fixedAt = 0), 
	umxPath(v.m. = c("int","slope")),
	umxPath("int", with = "slope"),
	umxPath("int", to = manifests, fixedAt = 1), 
	umxPath("slope", to = manifests, arrows = 1, fixedAt = c(0,1,2,3,4))
)

plot(m1, means=FALSE, strip=TRUE, splines="FALSE", max="int")

## End(Not run) # end dontrun

</code></pre>

<hr>
<h2 id='plot.MxModelTwinMaker'>Create and display a graphical path diagram for a path-based twin model.</h2><span id='topic+plot.MxModelTwinMaker'></span><span id='topic+umxPlotMxModelTwinMaker'></span>

<h3>Description</h3>

<p>Assumes the model has a group called &quot;MZ&quot; inside.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModelTwinMaker'
plot(
  x = NA,
  std = FALSE,
  fixed = TRUE,
  means = TRUE,
  oneTwin = TRUE,
  sep = "_T",
  digits = 2,
  file = "name",
  labels = c("none", "labels", "both"),
  resid = c("circle", "line", "none"),
  strip_zero = FALSE,
  splines = TRUE,
  min = NULL,
  same = NULL,
  max = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MxModelTwinMaker_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+umxTwinMaker">umxTwinMaker()</a></code> model from which to make a path diagram</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_fixed">fixed</code></td>
<td>
<p>Whether to show fixed paths (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_means">means</code></td>
<td>
<p>Whether to show means or not (default = TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_onetwin">oneTwin</code></td>
<td>
<p>(whether to plot a pair of twins, or just one (default = TRUE)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_sep">sep</code></td>
<td>
<p>The separator for twin variables (&quot;_T&quot;)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_digits">digits</code></td>
<td>
<p>The number of decimal places to add to the path coefficients</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_labels">labels</code></td>
<td>
<p>Whether to show labels on the paths. &quot;none&quot;, &quot;labels&quot;, or &quot;both&quot; (parameter + label).</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_resid">resid</code></td>
<td>
<p>How to show residuals and variances default is &quot;circle&quot;. Options are &quot;line&quot; &amp; &quot;none&quot;</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = FALSE)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_splines">splines</code></td>
<td>
<p>Whether to allow lines to curve: defaults to TRUE (nb: some models look better with FALSE)</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_min">min</code></td>
<td>
<p>optional list of objects to group at the top of the plot. Default (NULL) chooses automatically.</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_same">same</code></td>
<td>
<p>optional list of objects to group at the same rank in the plot. Default (NULL) chooses automatically.</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_max">max</code></td>
<td>
<p>optional list of objects to group at the bottom of the plot. Default (NULL) chooses automatically.</p>
</td></tr>
<tr><td><code id="plot.MxModelTwinMaker_+3A_...">...</code></td>
<td>
<p>Optional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you use umx_set_plot_format(&quot;graphviz&quot;), they will open in a graphviz helper app (if installed).
The commercial application &ldquo;OmniGraffle&rdquo; is great for editing these images.
On unix and windows, <code><a href="#topic+plot">plot()</a></code> will create a pdf and open it in your default pdf reader.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_set_plot_format">umx_set_plot_format()</a></code>, <code><a href="#topic+plot.MxModel">plot.MxModel()</a></code>, <code><a href="#topic+umxPlotACE">umxPlotACE()</a></code>, <code><a href="#topic+umxPlotCP">umxPlotCP()</a></code>, <code><a href="#topic+umxPlotIP">umxPlotIP()</a></code>, <code><a href="#topic+umxPlotGxE">umxPlotGxE()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
# 
# =====================
# = Make an ACE model =
# =====================
# 1. Clean data: Add separator and scale
data(twinData)
tmp = umx_make_twin_data_nice(data=twinData, sep="", zygosity="zygosity", numbering=1:2)
tmp = umx_scale_wide_twin_data(varsToScale= c("wt", "ht"), sep= "_T", data= tmp)
mzData = subset(tmp, zygosity %in%  c("MZFF", "MZMM"))
dzData = subset(tmp, zygosity %in%  c("DZFF", "DZMM"))

# 2. Define paths: You only need the paths for one person:
paths = c(
umxPath(v1m0 = c("a1", 'c1', "e1")),
umxPath(means = c("wt")),
umxPath(c("a1", 'c1', "e1"), to = "wt", values=.2)
)
m1 = umxTwinMaker("test", paths, mzData = mzData, dzData= dzData)
plot(m1, std= TRUE, means= FALSE)
plot(m1, means=FALSE, std=TRUE, strip=TRUE, splines="FALSE", max="intercept")

## End(Not run) # end dontrun

# =================
# = An ACEv model =
# =================
# Not complete

paths = c(
umxPath(v1m0 = c("A1", 'C1', "E1")),
umxPath(v1m0 = c("A2", 'C2', "E2")),
umxPath(v.m0 = c("l1", 'l2')),
umxPath(v.m. = c("wt", "ht")),
umxPath(c("A1", 'C1', "E1"), to = "l1", values= .2),
umxPath(c("A2", 'C2', "E2"), to = "l2", values= .2),
umxPath(c("l1", 'l2'), to = c("wt", "ht"), values= .2)
)

</code></pre>

<hr>
<h2 id='plot.percent'>Plot a percent change graph</h2><span id='topic+plot.percent'></span>

<h3>Description</h3>

<p>Plot method for &quot;percent&quot; objects: e.g. <code><a href="#topic+fin_percent">fin_percent()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'percent'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.percent_+3A_x">x</code></td>
<td>
<p>percent object.</p>
</td></tr>
<tr><td><code id="plot.percent_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> invisible
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_percent">fin_percent()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Percent needed to return to original value after 10% off
fin_percent(-10)
# Percent needed to return to original value after 10% on
tmp = fin_percent(10)
plot(tmp)

# Percent needed to return to original value after 50% off 34.50
fin_percent(-50, value = 34.5, logY = FALSE)

</code></pre>

<hr>
<h2 id='power.ACE.test'>Test the power of an ACE model to detect paths of interest.</h2><span id='topic+power.ACE.test'></span>

<h3>Description</h3>

<p><code>power.ACE.test</code> simulates a univariate ACE model. It
computes power to detect dropping one or more paths (a, c, or a after dropping c), specified in <code style="white-space: pre;">&#8288;drop=&#8288;</code>.
</p>
<p>The interface and functionality of this service are experimental and subject to change.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power.ACE.test(
  AA = 0.5,
  CC = 0,
  EE = NULL,
  DD = NULL,
  update = c("a", "c", "a_after_dropping_c", "d"),
  value = 0,
  n = NULL,
  MZ_DZ_ratio = 1,
  sig.level = 0.05,
  power = 0.8,
  method = c("ncp", "empirical"),
  search = FALSE,
  tryHard = c("yes", "no", "ordinal", "search"),
  digits = 2,
  optimizer = NULL,
  nSim = 4000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power.ACE.test_+3A_aa">AA</code></td>
<td>
<p>Additive genetic variance (Default .5)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_cc">CC</code></td>
<td>
<p>Shared environment variance (Default 0)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_ee">EE</code></td>
<td>
<p>Unique environment variance. Leave NULL (default) to compute an amount summing to 1.</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_dd">DD</code></td>
<td>
<p>Dominance Is set (default= NULL) compute an ADE rather than ACE model (DZr=.25)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_update">update</code></td>
<td>
<p>Component to drop (Default &quot;a&quot;, i.e., drop a)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_value">value</code></td>
<td>
<p>Value to set dropped path to (Default 0)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_n">n</code></td>
<td>
<p>If provided, solve at the given number of MZ+DZ pairs (Default NULL)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_mz_dz_ratio">MZ_DZ_ratio</code></td>
<td>
<p>MZ pairs per DZ pair (Default 1 = equal numbers.)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_sig.level">sig.level</code></td>
<td>
<p>alpha (p-value) Default = 0.05</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_power">power</code></td>
<td>
<p>Default = .8 (80 percent power, equal to 1 - Type II rate)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_method">method</code></td>
<td>
<p>How to estimate power: Default =  use non-centrality parameter (&quot;ncp&quot;). Alternative is &quot;empirical&quot;</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_search">search</code></td>
<td>
<p>Whether to return a search across power or just a point estimate (Default FALSE = point)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_tryhard">tryHard</code></td>
<td>
<p>Whether to tryHard to find a solution (default = &quot;yes&quot;, alternatives are &quot;no&quot;...)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_digits">digits</code></td>
<td>
<p>Rounding for reporting parameters (default 2)</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_optimizer">optimizer</code></td>
<td>
<p>If set, will switch the optimizer.</p>
</td></tr>
<tr><td><code id="power.ACE.test_+3A_nsim">nSim</code></td>
<td>
<p>Total number of pairs to simulate in the models (default = 4000)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Statistical power is the proportion of studies that, over the long run, one should expect to yield a statistically
significant result given certain study characteristics such as sample size (N), the expected effect size (<code class="reqn">\beta</code>),
and the criterion for statistical significance (<code class="reqn">\alpha</code>).
</p>
<p>(with nMZpairs= 2000 and MZ_DZ_ratio*nMZpairs DZ twins.
</p>
<p>A typical target for power is 80%. Much as the accepted critical p-value is .05, this has emerged as a trade off, in this case
of resources required for more powerful studies against the cost of missing a true effect.  People interested in truth
discourage running studies with low power: A study with 20 percent power will fail to detect real effects 80% of the time.
But even with zero power, the Type-I error rate remains a nominal 5% (and with any researcher degrees of freedom, perhaps much more than that).
Low powered research, then, fails to detect true effects, and generates support for random false theories about as often.
This sounds silly, but empirical rates are often as low as 20% (Button, et al., 2013).
</p>
<p>Illustration of  <code class="reqn">\alpha</code>, <code class="reqn">\beta</code>, and power (1-<code class="reqn">\beta</code>):
</p>
<p><img src="../help/figures/power.png" width=50% alt="Figure: power.png" />

</p>


<h3>Value</h3>

<p><code><a href="OpenMx.html#topic+mxPower">OpenMx::mxPower()</a></code> object
</p>


<h3>References</h3>


<ul>
<li><p> Visscher, P.M., Gordon, S., Neale, M.C. (2008). Power of the classical twin design
revisited: II detection of common environmental variance. <em>Twin Res Hum Genet</em>, <strong>11</strong>: 48-54.
<a href="https://doi.org/10.1375/twin.11.1.48">doi:10.1375/twin.11.1.48</a>.
</p>
</li>
<li><p> Button, K. S., Ioannidis, J. P., Mokrysz, C., Nosek, B. A., Flint, J., Robinson, E. S., and Munafo, M. R. (2013).
Power failure: why small sample size undermines the reliability of neuroscience.
<em>Nature Reviews Neuroscience</em>, <strong>14</strong>, 365-376. <a href="https://doi.org/10.1038/nrn3475">doi:10.1038/nrn3475</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxPower">umxPower()</a></code>, <code><a href="OpenMx.html#topic+mxPower">OpenMx::mxPower()</a></code>, <code><a href="#topic+umxACE">umxACE()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# =====================================================
# = N for .8 power to detect a^2 = .5 equal MZ and DZ =
# =====================================================
power.ACE.test(AA = .5, CC = 0, update = "a")
# Suggests n = 84 MZ and 94 DZ pairs.

## Not run: 
# ================================
# = Show power across range of N =
# ================================
power.ACE.test(AA= .5, CC= 0, update = "a", search = TRUE)

# Salutary note: You need well fitting models with correct betas in the data
# for power to be valid.
# tryHard helps ensure this, as does the default nSim= 4000 pair data.
# Power is important to get right, so I recommend using tryHard = "yes" (the default)

# =====================
# = Power to detect C =
# =====================

# 102 of each of MZ and DZ pairs for 80% power (default).
power.ACE.test(AA= .5, CC= .3, update = "c")

# ==========================================
# = Set 'a' to a fixed, but non-zero value =
# ==========================================

power.ACE.test(update= "a", value= sqrt(.2), AA= .5, CC= 0)

# ========================================
# = Drop More than one parameter (A &amp; C) =
# ========================================
# E vs AE: the hypothesis that twins show no familial similarity.
power.ACE.test(update = "a_after_dropping_c", AA= .5, CC= .3)

# ===================================================
# = More power to detect A &gt; 0 when more C present  =
# ===================================================

power.ACE.test(update = "a", AA= .5, CC= .0)
power.ACE.test(update = "a", AA= .5, CC= .3)

# ====================================================
# = More power to detect C &gt; 0 when more A present?  =
# ====================================================

power.ACE.test(update = "c", AA= .0, CC= .5)
power.ACE.test(update = "c", AA= .3, CC= .5)


# ===================================
# = Power with more DZs or more MZs =
# ===================================

# Power about the same: total pairs with 2 MZs per DZ
power.ACE.test(MZ_DZ_ratio= 2/1, update= "a", AA= .3, CC= 0, method="ncp", tryHard="yes")
power.ACE.test(MZ_DZ_ratio= 1/2, update= "a", AA= .3, CC= 0, method="ncp", tryHard="yes")
power.ACE.test(update= "a", AA= .3, CC= 0, method="ncp", tryHard="yes")


# =====================================
# = Compare ncp and empirical methods =
# =====================================

power.ACE.test(update= "a", AA= .5, CC= 0, method = "ncp")
# method = "ncp": For 80% power, you need 166 MZ and 166 DZ pairs
power.ACE.test(update= "a", AA= .5, CC= 0, method= "empirical")
# method= "empirical": For 80% power, you need 154 MZ and 154 DZ pairs

# ====================
# = Show off options =
# ====================
# 1. tryHard

power.ACE.test(update = "a", AA= .5, CC= 0, tryHard= "no")

# 2. toggle optimizer

power.ACE.test(update= "a", AA= .5, CC= 0, optimizer= "SLSQP")

# 3. You can raise or lower the number of pairs used in the true model
#    by varying nSim (twin pairs in the simulated data).

power.ACE.test(update = "a", AA= .5, CC= 0, nSim= 20)


## End(Not run)

</code></pre>

<hr>
<h2 id='print.oddsratio'>Print a scale &quot;oddsratio&quot; object</h2><span id='topic+print.oddsratio'></span>

<h3>Description</h3>

<p>Print method for the <code><a href="#topic+oddsratio">oddsratio()</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'oddsratio'
print(x, digits = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.oddsratio_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+oddsratio">oddsratio()</a></code> result.</p>
</td></tr>
<tr><td><code id="print.oddsratio_+3A_digits">digits</code></td>
<td>
<p>The rounding precision.</p>
</td></tr>
<tr><td><code id="print.oddsratio_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> invisible oddsratio object (x).
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+print">print()</a></code>, <code><a href="#topic+oddsratio">oddsratio()</a></code>,
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>oddsratio(grp1 = c(1, 10), grp2 = c(3, 10))
oddsratio(grp1 = c(3, 10), grp2 = c(1, 10))
oddsratio(grp1 = c(3, 10), grp2 = c(1, 10), alpha = .01)
</code></pre>

<hr>
<h2 id='print.percent'>Print a percent object</h2><span id='topic+print.percent'></span>

<h3>Description</h3>

<p>Print method for &quot;percent&quot; objects: e.g. <code><a href="#topic+fin_percent">fin_percent()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'percent'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.percent_+3A_x">x</code></td>
<td>
<p>percent object.</p>
</td></tr>
<tr><td><code id="print.percent_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> invisible
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+fin_percent">fin_percent()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Percent needed to return to original value after 10% off
fin_percent(-10)
# Percent needed to return to original value after 10% on
fin_percent(10)

# Percent needed to return to original value after 50% off 34.50
fin_percent(-50, value = 34.5)

</code></pre>

<hr>
<h2 id='print.reliability'>Print a scale &quot;reliability&quot; object</h2><span id='topic+print.reliability'></span>

<h3>Description</h3>

<p>Print method for the <code><a href="#topic+reliability">reliability()</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'reliability'
print(x, digits = 4, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.reliability_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+reliability">reliability()</a></code> result.</p>
</td></tr>
<tr><td><code id="print.reliability_+3A_digits">digits</code></td>
<td>
<p>The rounding precision.</p>
</td></tr>
<tr><td><code id="print.reliability_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> invisible reliability object (x)
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+print">print()</a></code>, <code><a href="#topic+reliability">reliability()</a></code>,
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># treat vehicle aspects as items of a test
data(mtcars)
reliability(cov(mtcars))
</code></pre>

<hr>
<h2 id='print.RMSEA'>Print a RMSEA object</h2><span id='topic+print.RMSEA'></span>

<h3>Description</h3>

<p>Print method for &quot;RMSEA&quot; objects: e.g. <code><a href="#topic+RMSEA">RMSEA()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RMSEA'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.RMSEA_+3A_x">x</code></td>
<td>
<p>RMSEA object.</p>
</td></tr>
<tr><td><code id="print.RMSEA_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> invisible
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+RMSEA">RMSEA()</a></code>, <code><a href="#topic+print">print()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type= "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1.0)
)
tmp = summary(m1)
RMSEA(tmp)

## End(Not run)

</code></pre>

<hr>
<h2 id='prolific_anonymize'>Clean up a prolific file for sharing by removing anonymity-compromising columns.</h2><span id='topic+prolific_anonymize'></span>

<h3>Description</h3>

<p>prolific.ac IDs and other columns like IP and lat/long might compromise
subject anonymity when shared.
<code>prolific_anonymize</code> replaces PIDs with a simple numeric sequence, preserving
repeated measures in long data, and removing other columns.
You can delete additional  columns by adding them to <code>extraColumns</code>. It is ideal for use
when sharing data to <a href="https://researchbox.org">https://researchbox.org</a> which enforces anonymization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prolific_anonymize(
  df = NULL,
  PID = "PID",
  extraColumns = NA,
  baseOffset = 10000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prolific_anonymize_+3A_df">df</code></td>
<td>
<p>Existing datafile to anonymize.</p>
</td></tr>
<tr><td><code id="prolific_anonymize_+3A_pid">PID</code></td>
<td>
<p>The prolific ID col name to anonymize</p>
</td></tr>
<tr><td><code id="prolific_anonymize_+3A_extracolumns">extraColumns</code></td>
<td>
<p>Any  extra columns to delete (default NA)</p>
</td></tr>
<tr><td><code id="prolific_anonymize_+3A_baseoffset">baseOffset</code></td>
<td>
<p>The numeric to start renumbering PIDs from (default = 1e4)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> [<a href="base.html#topic+data.frame">data.frame</a>]
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+prolific_check_ID">prolific_check_ID()</a></code>, <code><a href="#topic+prolific_read_demog">prolific_read_demog()</a></code>, <code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns()</a></code>
</p>
</li></ul>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tmp = prolific_anonymize(df, PID = "PID")

## End(Not run)
</code></pre>

<hr>
<h2 id='prolific_check_ID'>Return PIDs in df</h2><span id='topic+prolific_check_ID'></span>

<h3>Description</h3>

<p>Participants may time-out on Prolific, but still complete on Qualtrics. This identifies them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prolific_check_ID(IDs, df, IDcol = "PROLIFIC_PID")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prolific_check_ID_+3A_ids">IDs</code></td>
<td>
<p>Timed-out (or other) IDs to look for.</p>
</td></tr>
<tr><td><code id="prolific_check_ID_+3A_df">df</code></td>
<td>
<p>to search.</p>
</td></tr>
<tr><td><code id="prolific_check_ID_+3A_idcol">IDcol</code></td>
<td>
<p>Name of prolific ID column (default PROLIFIC_PID)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- list of IDs in the dataframe
</p>


<h3>See Also</h3>

<p>- [prolific_read_demog()], [prolific_anonymize()], [umx_merge_randomized_columns()] # [prolific_check_ID()]
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># IDs = c("59d0ec2446447f00011edb063","5a08c9a7f2e3460001edb063f0254")
# prolific_check_ID(IDs, df)
</code></pre>

<hr>
<h2 id='prolific_read_demog'>Read and optionally merge demographics file from prolific academic</h2><span id='topic+prolific_read_demog'></span>

<h3>Description</h3>

<p>prolific academic provides a demographics file. This reads it and merges with your data
using PID and participant_id
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prolific_read_demog(
  file,
  base = "",
  df = NULL,
  by.df = "PROLIFIC_PID",
  by.demog = "Participant.id",
  age = "age",
  sex = "Gender",
  vars = NULL,
  all.df = TRUE,
  all.demog = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prolific_read_demog_+3A_file">file</code></td>
<td>
<p>Path to demographics file.</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_base">base</code></td>
<td>
<p>Optional path to folder, in which case 'file' is just filename.</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_df">df</code></td>
<td>
<p>Existing datafile to merge demographics into (optional)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_by.df">by.df</code></td>
<td>
<p>The ID name in existing df (default = &quot;PROLIFIC_PID&quot;)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_by.demog">by.demog</code></td>
<td>
<p>The ID name in the prolific demographics file (default = &quot;Participant id&quot; was <code>by.demog</code>)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_age">age</code></td>
<td>
<p>Name of age var in demographics file (&quot;age&quot;)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_sex">sex</code></td>
<td>
<p>Name of sex var in demographics file (&quot;Sex&quot;)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_vars">vars</code></td>
<td>
<p>Additional vars to keep from demographics file (WAS age &amp; Sex)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_all.df">all.df</code></td>
<td>
<p>Whether to keep all lines of df (default = TRUE)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_all.demog">all.demog</code></td>
<td>
<p>Whether to keep all lines (people) in the demographics file (default = FALSE)</p>
</td></tr>
<tr><td><code id="prolific_read_demog_+3A_verbose">verbose</code></td>
<td>
<p>Print variable names found in the file.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> [<a href="base.html#topic+data.frame">data.frame</a>]
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+prolific_check_ID">prolific_check_ID()</a></code>, <code><a href="#topic+prolific_anonymize">prolific_anonymize()</a></code>, <code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns()</a></code>
</p>
</li></ul>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fp = "~/Desktop/prolific_export_5f20c3e662e3b6407dcd37a5.csv"
df = prolific_read_demog(fp, sex = "Gender", age = "Age", df = df)
tmp = prolific_read_demog(fp, by.df = "PROLIFIC_PID", vars=c("Ethnicity.simplified"))

## End(Not run)
</code></pre>

<hr>
<h2 id='qm'>qm</h2><span id='topic+qm'></span>

<h3>Description</h3>

<p>Quickmatrix function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qm(..., rowMarker = "|")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qm_+3A_...">...</code></td>
<td>
<p>the components of your matrix</p>
</td></tr>
<tr><td><code id="qm_+3A_rowmarker">rowMarker</code></td>
<td>
<p>mark the end of each row</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- matrix
</p>


<h3>See Also</h3>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simple example
qm(0, 1 |
   2, NA)
## Not run: 
# clever example
M1 = M2 = diag(2)
qm(M1,c(4,5) | c(1,2),M2 | t(1:3))

## End(Not run)
</code></pre>

<hr>
<h2 id='rad2deg'>Convert Radians to Degrees</h2><span id='topic+rad2deg'></span>

<h3>Description</h3>

<p>Just a helper to multiply radians by 180 and divide by <code class="reqn">\pi</code> to get degrees.
</p>
<p><em>note</em>: R's trig functions, e.g. <code><a href="base.html#topic+sin">sin()</a></code> use Radians for input!
</p>
<p>There are <code class="reqn">2\pi</code> radians in a circle.
1 Rad = <code class="reqn">180/\pi</code> degrees = ~ 57.296 degrees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rad2deg(rad)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rad2deg_+3A_rad">rad</code></td>
<td>
<p>The value in Radians you wish to convert</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> value in degrees
</p>
</li></ul>



<h3>References</h3>

<p><a href="https://en.wikipedia.org/wiki/Radian">https://en.wikipedia.org/wiki/Radian</a>
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+deg2rad">deg2rad()</a></code>, <code><a href="base.html#topic+sin">sin()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+umxBrownie">umxBrownie</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rad2deg(pi) #180 degrees
</code></pre>

<hr>
<h2 id='reliability'>Report coefficient alpha (reliability)</h2><span id='topic+reliability'></span>

<h3>Description</h3>

<p>Compute and report Coefficient alpha (extracted from Rcmdr to avoid its dependencies)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reliability(S)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reliability_+3A_s">S</code></td>
<td>
<p>A square, symmetric, numeric covariance matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>

<p>- &lt;https://cran.r-project.org/package=Rcmdr&gt;
</p>


<h3>See Also</h3>

<p>- [umx::print.reliability()],
</p>
<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># treat car data as items of a test
data(mtcars)
reliability(cov(mtcars))
</code></pre>

<hr>
<h2 id='residuals.MxModel'>Get residuals from an MxModel</h2><span id='topic+residuals.MxModel'></span>

<h3>Description</h3>

<p>Return the <code><a href="stats.html#topic+residuals">residuals()</a></code> from an OpenMx RAM model. You can format these (with digits), and suppress small values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
residuals(object, digits = 2, suppress = NULL, reorder = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals.MxModel_+3A_object">object</code></td>
<td>
<p>An fitted <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to get residuals</p>
</td></tr>
<tr><td><code id="residuals.MxModel_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="residuals.MxModel_+3A_suppress">suppress</code></td>
<td>
<p>smallest deviation to print out (default = NULL = show all)</p>
</td></tr>
<tr><td><code id="residuals.MxModel_+3A_reorder">reorder</code></td>
<td>
<p>optionally reorder the variables in the residuals matrix to show patterns</p>
</td></tr>
<tr><td><code id="residuals.MxModel_+3A_...">...</code></td>
<td>
<p>Optional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> matrix of residuals
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1.0)
)

# ===================================
# = Show the residuals of the model =
# ===================================
residuals(m1)
# |   |x1   |x2    |x3   |x4    |x5 |
# |:--|:----|:-----|:----|:-----|:--|
# |x1 |.    |.     |0.01 |.     |.  |
# |x2 |.    |.     |0.01 |-0.01 |.  |
# |x3 |0.01 |0.01  |.    |.     |.  |
# |x4 |.    |-0.01 |.    |.     |.  |
# |x5 |.    |.     |.    |.     |.  |
# [1] "nb: You can zoom in on bad values with, e.g. suppress = .01, which
#      will hide values smaller than this. Use digits = to round"

residuals(m1, digits = 3)
residuals(m1, digits = 3, suppress = .005)
# residuals are returned as an invisible object you can capture in a variable
a = residuals(m1); a

## End(Not run)

</code></pre>

<hr>
<h2 id='RMSEA'>Generic RMSEA function</h2><span id='topic+RMSEA'></span>

<h3>Description</h3>

<p>See <code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel()</a></code> to access the RMSEA of MxModels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RMSEA(x, ci.lower, ci.upper, digits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RMSEA_+3A_x">x</code></td>
<td>
<p>an object from which to get the RMSEA</p>
</td></tr>
<tr><td><code id="RMSEA_+3A_ci.lower">ci.lower</code></td>
<td>
<p>the lower CI to compute</p>
</td></tr>
<tr><td><code id="RMSEA_+3A_ci.upper">ci.upper</code></td>
<td>
<p>the upper CI to compute</p>
</td></tr>
<tr><td><code id="RMSEA_+3A_digits">digits</code></td>
<td>
<p>digits to show</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> RMSEA object containing value (and perhaps a CI)
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>

<hr>
<h2 id='RMSEA.MxModel'>RMSEA function for MxModels</h2><span id='topic+RMSEA.MxModel'></span>

<h3>Description</h3>

<p>Return RMSEA and its confidence interval on a model.
RMSEA(tmp, silent=TRUE)$RMSEA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
RMSEA(x, ci.lower = 0.025, ci.upper = 0.975, digits = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RMSEA.MxModel_+3A_x">x</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to get RMSEA</p>
</td></tr>
<tr><td><code id="RMSEA.MxModel_+3A_ci.lower">ci.lower</code></td>
<td>
<p>the lower CI to compute (only 95%, i.e., .025 supported)</p>
</td></tr>
<tr><td><code id="RMSEA.MxModel_+3A_ci.upper">ci.upper</code></td>
<td>
<p>the upper CI to compute (only 95%, i.e., .975 supported)</p>
</td></tr>
<tr><td><code id="RMSEA.MxModel_+3A_digits">digits</code></td>
<td>
<p>digits to show (default = 3)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> object containing the RMSEA, lower and upper bounds, and p-close
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
RMSEA(m1)

x = RMSEA(m1)
x$RMSEA # 0.0309761

# Raw: needs to be run by umx to get RMSEA
m2 = umxRAM("One Factor", data = demoOneFactor,
	umxPath("G", to = manifests),
	umxPath(v.m. = manifests),
	umxPath(v1m0 = "G")
)
RMSEA(m2)

## End(Not run)
</code></pre>

<hr>
<h2 id='RMSEA.summary.mxmodel'>RMSEA function for MxModel summary</h2><span id='topic+RMSEA.summary.mxmodel'></span>

<h3>Description</h3>

<p>Extract the RMSEA and confidence interval from a model summary and returns it as an RMSEA object.
To report just the RMSEA, you can use <code>RMSEA(model)$RMSEA</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.mxmodel'
RMSEA(x, ci.lower = 0.025, ci.upper = 0.975, digits = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RMSEA.summary.mxmodel_+3A_x">x</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> summary from which to get RMSEA</p>
</td></tr>
<tr><td><code id="RMSEA.summary.mxmodel_+3A_ci.lower">ci.lower</code></td>
<td>
<p>the lower CI to compute (only 95% CI (.025) is implemented)</p>
</td></tr>
<tr><td><code id="RMSEA.summary.mxmodel_+3A_ci.upper">ci.upper</code></td>
<td>
<p>the upper CI to compute (only 95% CI (.975) is implemented)</p>
</td></tr>
<tr><td><code id="RMSEA.summary.mxmodel_+3A_digits">digits</code></td>
<td>
<p>The number of digits to round data (defaults to 3)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> object containing the RMSEA and lower and upper bounds
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/simsem/semTools/wiki/Functions">https://github.com/simsem/semTools/wiki/Functions</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor[1:100,], type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1.0)
)
tmp = summary(m1)
RMSEA(tmp)

## End(Not run)
</code></pre>

<hr>
<h2 id='SE_from_p'>Compute an SE from a beta and p value</h2><span id='topic+SE_from_p'></span>

<h3>Description</h3>

<p><code>SE_from_p</code> takes beta and p, and returns an SE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SE_from_p(beta = NULL, p = NULL, SE = NULL, lower = NULL, upper = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SE_from_p_+3A_beta">beta</code></td>
<td>
<p>The effect size</p>
</td></tr>
<tr><td><code id="SE_from_p_+3A_p">p</code></td>
<td>
<p>The p-value for the effect</p>
</td></tr>
<tr><td><code id="SE_from_p_+3A_se">SE</code></td>
<td>
<p>Standard error</p>
</td></tr>
<tr><td><code id="SE_from_p_+3A_lower">lower</code></td>
<td>
<p>Lower CI</p>
</td></tr>
<tr><td><code id="SE_from_p_+3A_upper">upper</code></td>
<td>
<p>Upper CI</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Standard error
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxAPA">umxAPA()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SE_from_p(beta = .0020, p = .780)
SE_from_p(beta = .0020, p = .01)
SE_from_p(beta = .0020, SE = 0.01)
umxAPA(.0020, p = .01)
</code></pre>

<hr>
<h2 id='tmx_genotypic_effect'>Graphical display of genotypic effects.</h2><span id='topic+tmx_genotypic_effect'></span>

<h3>Description</h3>

<p><code>tmx_genotypic_effect</code> allows you to explore the concept of genotypic effect at a locus. With it,
you can interactively explore the effects of allele frequency, additive variance, and
<a href="https://en.wikipedia.org/wiki/Dominance_(genetics)">dominance</a>.
</p>
<p>This function lets you explore the simplest two–allele system (<code>B</code> and <code>b</code>), with three possible
genotypes, <code>BB</code>, <code>Bb</code>, and <code>bb</code>.
</p>
<p>The point between the two homozygotes is <code>m</code> &ndash; the mean effect of the homozygous genotypes.
</p>
<p>Parameter <code>a</code> is half the measured phenotypic difference between the homozygotes <code>BB</code> and <code>bb</code>.
It corresponds to the additive effect of each additional <code>B</code> allele, relative to the <code>bb</code> phenotype.
</p>
<p>Parameter <code>d</code> is the deviation of the heterozygote <code>Bb</code> phenotype from the homozygote mid-point <code>m</code>.
It corresponds to the non-additive (dominance) effect of the <code>B</code> allele. The heterozygote phenotype
may lie on either side of <code>m</code> and the sign of <code>d</code> will vary accordingly.
</p>
<p><strong>Old system</strong> from book ed 2:
</p>
<p>Adapted from Mather and Jinks, 1977, p. 32). See book issue old-style nomenclature
https://github.com/tbates/BGBook/issues/23
</p>
<p><code>u</code> = Frequency of the dominant allele (now = <code>p</code>).
<code>v</code> = Frequency of the recessive allele (now = <code>q</code>).
</p>
<p><code>m</code> = midpoint between the two homozygotes
<code>d</code> = half the difference between the two homozygote (now <code>a</code>)
</p>
<p><code>h</code> = deviation of the heterozygote from <code>m</code> (now = <code>d</code>)
</p>
<p>New system:
</p>
<p><code>u</code> and <code>v</code> -&gt; <code>p</code> and  <code>q</code>
</p>
<p><code>d</code> and <code>h</code> -&gt; <code>a</code> and <code>d</code>
</p>
<p>See BGBook issue 23
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tmx_genotypic_effect(p = 0.75, q = (1 - p), a = 0.5, d = 0, m = 0, show = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tmx_genotypic_effect_+3A_p">p</code></td>
<td>
<p>The frequency of the B allele (Default .5)</p>
</td></tr>
<tr><td><code id="tmx_genotypic_effect_+3A_q">q</code></td>
<td>
<p>The frequency of the b allele (Default 1-p)</p>
</td></tr>
<tr><td><code id="tmx_genotypic_effect_+3A_a">a</code></td>
<td>
<p>Half the difference between the two homozygote phenotypes (Default .5)</p>
</td></tr>
<tr><td><code id="tmx_genotypic_effect_+3A_d">d</code></td>
<td>
<p>The deviation of the heterozygote from m (Default 0)</p>
</td></tr>
<tr><td><code id="tmx_genotypic_effect_+3A_m">m</code></td>
<td>
<p>The value of the midpoint between the homozygotes (Default 0)</p>
</td></tr>
<tr><td><code id="tmx_genotypic_effect_+3A_show">show</code></td>
<td>
<p>Whether to draw the plot or just return it (Default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optional plot
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Neale, M. C. (2005). Quantitative Genetics. In Encyclopedia of Life Sciences. New York: John Wiley &amp; Sons, Ltd.
<a href="https://vipbg.vcu.edu/vipbg/Articles/2005_book_chapters_ascertainment_corrections.pdf">pdf</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Teaching and testing Functions: 
<code><a href="#topic+tmx_is.identified">tmx_is.identified</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx);


# =========================
# = Pure additivity: d= 0 =
# =========================
tmx_genotypic_effect(p = .5, a = 1, d = 0, m = 0, show = TRUE);

# =============================
# = Complete dominance: a=d=1 =
# =============================
tmx_genotypic_effect(p = .5, q =.5, a = 1, d = 1, m = 0, show = TRUE);

# ===========================
# = Over dominance: a&lt; d =1 =
# ===========================
tmx_genotypic_effect(p = .5, q =.5, a =.5, d = 1, m = 0)

p = tmx_genotypic_effect(p = .5, q = .5, a = 1, d = .5, m = 0, show = TRUE); 
# p = p + ggplot2::geom_point() 
# p + ggplot2::geom_text(hjust = 0, nudge_x = 0.05, label= "x")
# ggsave(paste0(base, "c03_genotypic_effect_by_gene_dose.pdf"), width = 4.6, height = 4.6)


</code></pre>

<hr>
<h2 id='tmx_is.identified'>Test if a factor model is identified</h2><span id='topic+tmx_is.identified'></span>

<h3>Description</h3>

<p>Test if a factor model is identified by establishing if the number of variables is equal too or grater than
the number of model parameters. See also <code><a href="OpenMx.html#topic+mxCheckIdentification">mxCheckIdentification()</a></code> for checking actual models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tmx_is.identified(nVariables, nFactors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tmx_is.identified_+3A_nvariables">nVariables</code></td>
<td>
<p>the number of variables measured.</p>
</td></tr>
<tr><td><code id="tmx_is.identified_+3A_nfactors">nFactors</code></td>
<td>
<p>the number of factors posited.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Binary
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxCheckIdentification">mxCheckIdentification()</a></code>
</p>
</li></ul>

<p>Other Teaching and testing Functions: 
<code><a href="#topic+tmx_genotypic_effect">tmx_genotypic_effect</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmx_is.identified(nVariables = 2, nFactors = 1) # FALSE
tmx_is.identified(nVariables = 3, nFactors = 1) # TRUE
tmx_is.identified(nVariables = 4, nFactors = 2) # FALSE
tmx_is.identified(nVariables = 5, nFactors = 2) # TRUE
</code></pre>

<hr>
<h2 id='tmx_show'>Show matrices of models in a easy-to-learn-from format.</h2><span id='topic+tmx_show'></span>

<h3>Description</h3>

<p>Show matrices of models in a easy-to-learn-from format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tmx_show(
  x,
  what = c("values", "free", "labels", "nonzero_or_free"),
  show = c("free", "fixed", "all"),
  matrices = c("S", "A", "M"),
  digits = 2,
  report = c("html", "markdown"),
  na.print = "",
  zero.print = ".",
  html_font = NULL,
  style = c("paper", "material_dark", "classic", "classic_2", "minimal", "material"),
  bootstrap_options = c("hover", "bordered", "condensed", "responsive"),
  lightable_options = "striped"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tmx_show_+3A_x">x</code></td>
<td>
<p>an object e.g. <code><a href="#topic+umxRAM">umxRAM()</a></code> <code><a href="#topic+umxMatrix">umxMatrix()</a></code> from which to show parameters.</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_what">what</code></td>
<td>
<p>legal options are &quot;values&quot; (default), &quot;free&quot;, or &quot;labels&quot;).</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_show">show</code></td>
<td>
<p>filter on what to show c(&quot;all&quot;, &quot;free&quot;, &quot;fixed&quot;).</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_matrices">matrices</code></td>
<td>
<p>to show  (default is c(&quot;S&quot;, &quot;A&quot;)). &quot;thresholds&quot; in beta.</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_digits">digits</code></td>
<td>
<p>precision to report. Default = round to 2 decimal places.</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_report">report</code></td>
<td>
<p>How to report the results. &quot;html&quot; = open in browser.</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_na.print">na.print</code></td>
<td>
<p>How to display NAs (default = &quot;&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_zero.print">zero.print</code></td>
<td>
<p>How to display 0 values (default = &quot;.&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_html_font">html_font</code></td>
<td>
<p>Default is null. Set (e.g. &quot;Optima&quot;) to override the style's default font.</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_style">style</code></td>
<td>
<p>The style for the table (Defaults to &quot;paper&quot;. Other options are &quot;material_dark&quot;, &quot;classic&quot;, &quot;classic_2&quot;, &quot;minimal&quot;, &quot;material&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_bootstrap_options">bootstrap_options</code></td>
<td>
<p>border etc. Defaults to c(&quot;hover&quot;, &quot;bordered&quot;, &quot;condensed&quot;, &quot;responsive&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show_+3A_lightable_options">lightable_options</code></td>
<td>
<p>Default is &quot;striped&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>

<hr>
<h2 id='tmx_show.MxMatrix'>Show matrices of models in a easy-to-learn-from format.</h2><span id='topic+tmx_show.MxMatrix'></span>

<h3>Description</h3>

<p>Show matrices of models in a easy-to-learn-from format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxMatrix'
tmx_show(
  x,
  what = c("values", "free", "labels", "nonzero_or_free"),
  show = c("free", "fixed", "all"),
  matrices = c("S", "A", "M"),
  digits = 2,
  report = c("html", "markdown"),
  na.print = "",
  zero.print = ".",
  html_font = NULL,
  style = c("paper", "material_dark", "classic", "classic_2", "minimal", "material"),
  bootstrap_options = c("hover", "bordered", "condensed", "responsive"),
  lightable_options = "striped"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tmx_show.MxMatrix_+3A_x">x</code></td>
<td>
<p>an object e.g. <code><a href="#topic+umxRAM">umxRAM()</a></code> <code><a href="#topic+umxMatrix">umxMatrix()</a></code> from which to show parameters.</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_what">what</code></td>
<td>
<p>legal options are &quot;values&quot; (default), &quot;free&quot;, or &quot;labels&quot;).</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_show">show</code></td>
<td>
<p>filter on what to show c(&quot;all&quot;, &quot;free&quot;, &quot;fixed&quot;).</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_matrices">matrices</code></td>
<td>
<p>to show  (default is c(&quot;S&quot;, &quot;A&quot;)). &quot;thresholds&quot; in beta.</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_digits">digits</code></td>
<td>
<p>precision to report. Default = round to 2 decimal places.</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_report">report</code></td>
<td>
<p>How to report the results. &quot;html&quot; = open in browser.</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_na.print">na.print</code></td>
<td>
<p>How to display NAs (default = &quot;&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_zero.print">zero.print</code></td>
<td>
<p>How to display 0 values (default = &quot;.&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_html_font">html_font</code></td>
<td>
<p>Default is null. Set (e.g. &quot;Optima&quot;) to override the style's default font.</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_style">style</code></td>
<td>
<p>The style for the table (Defaults to &quot;paper&quot;. Other options are &quot;material_dark&quot;, &quot;classic&quot;, &quot;classic_2&quot;, &quot;minimal&quot;, &quot;material&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_bootstrap_options">bootstrap_options</code></td>
<td>
<p>border etc. Defaults to c(&quot;hover&quot;, &quot;bordered&quot;, &quot;condensed&quot;, &quot;responsive&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxMatrix_+3A_lightable_options">lightable_options</code></td>
<td>
<p>Default is &quot;striped&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>

<hr>
<h2 id='tmx_show.MxModel'>Show matrices of RAM models in a easy-to-learn-from format.</h2><span id='topic+tmx_show.MxModel'></span>

<h3>Description</h3>

<p>A great way to learn about models is to look at the matrix contents. <code>tmx_show</code> is designed to
do this in a way that makes it easy to process for users: The matrix contents are formatted as
tables, and can even be displayed as tables in a web browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
tmx_show(
  x,
  what = c("values", "free", "labels", "nonzero_or_free"),
  show = c("free", "fixed", "all"),
  matrices = c("S", "A", "M"),
  digits = 2,
  report = c("html", "markdown"),
  na.print = "",
  zero.print = ".",
  html_font = NULL,
  style = c("paper", "material_dark", "classic", "classic_2", "minimal", "material"),
  bootstrap_options = c("hover", "bordered", "condensed", "responsive"),
  lightable_options = "striped"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tmx_show.MxModel_+3A_x">x</code></td>
<td>
<p>an object e.g. <code><a href="#topic+umxRAM">umxRAM()</a></code> <code><a href="#topic+umxMatrix">umxMatrix()</a></code> from which to show parameters.</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_what">what</code></td>
<td>
<p>legal options are &quot;values&quot; (default), &quot;free&quot;, or &quot;labels&quot;).</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_show">show</code></td>
<td>
<p>filter on what to show c(&quot;all&quot;, &quot;free&quot;, &quot;fixed&quot;).</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_matrices">matrices</code></td>
<td>
<p>to show  (default is c(&quot;S&quot;, &quot;A&quot;)). &quot;thresholds&quot; in beta.</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_digits">digits</code></td>
<td>
<p>precision to report. Default = round to 2 decimal places.</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_report">report</code></td>
<td>
<p>How to report the results. &quot;html&quot; = open in browser.</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_na.print">na.print</code></td>
<td>
<p>How to display NAs (default = &quot;&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_zero.print">zero.print</code></td>
<td>
<p>How to display 0 values (default = &quot;.&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_html_font">html_font</code></td>
<td>
<p>Default is null. Set (e.g. &quot;Optima&quot;) to override the style's default font.</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_style">style</code></td>
<td>
<p>The style for the table (Defaults to &quot;paper&quot;. Other options are &quot;material_dark&quot;, &quot;classic&quot;, &quot;classic_2&quot;, &quot;minimal&quot;, &quot;material&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_bootstrap_options">bootstrap_options</code></td>
<td>
<p>border etc. Defaults to c(&quot;hover&quot;, &quot;bordered&quot;, &quot;condensed&quot;, &quot;responsive&quot;)</p>
</td></tr>
<tr><td><code id="tmx_show.MxModel_+3A_lightable_options">lightable_options</code></td>
<td>
<p>Default is &quot;striped&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user can select which matrices to view, whether to show values, free, and/or labels, and the precision of rounding.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Teaching and Testing functions: 
<code><a href="#topic+umxDiagnose">umxDiagnose</a>()</code>,
<code><a href="#topic+umxPower">umxPower</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("tmx_sh", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

# =============================================
# = Show smart table on the web (the default) =
# =============================================
tmx_show(m1, report = "html")
tmx_show(m1, what = "free", matrices = "thresholds")
tmx_show(m1, zero.print = "-")

tmx_show(m1, report = "markdown")
tmx_show(m1, digits = 3, report = "markdown")
tmx_show(m1, matrices = "S", report = "markdown")
tmx_show(m1, what = "free"  , report = "markdown")
tmx_show(m1, what = "labels", report = "markdown")
tmx_show(m1, what = "free", matrices = "A", report= "markdown")

## End(Not run)

</code></pre>

<hr>
<h2 id='umx_aggregate'>Convenient formula-based cross-tabs &amp; built-in summary functions</h2><span id='topic+umx_aggregate'></span>

<h3>Description</h3>

<p>A common task is preparing summary tables, aggregating over some grouping factor.
Like mean and sd of age, by sex. R's <code><a href="stats.html#topic+aggregate">aggregate()</a></code> function is useful and powerful, allowing
xtabs based on a formula.
</p>
<p>umx_aggregate makes using it a bit easier. In particular, it has some common functions
for summarizing data built-in, like &quot;mean (sd)&quot; (the default).
</p>
<p><code>umx_aggregate(mpg ~ cyl, data = mtcars, what = "mean_sd")</code>
</p>

<table>
<tr>
 <td style="text-align: left;">
cyl        </td><td style="text-align: left;"> mpg</td>
</tr>
<tr>
 <td style="text-align: left;">
4 (n = 11) </td><td style="text-align: left;"> 26.66 (4.51)</td>
</tr>
<tr>
 <td style="text-align: left;">
6 (n = 7)  </td><td style="text-align: left;"> 19.74 (1.45)</td>
</tr>
<tr>
 <td style="text-align: left;">
8 (n = 14) </td><td style="text-align: left;"> 15.1 (2.56)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Usage</h3>

<pre><code class='language-R'>umx_aggregate(
  formula = DV ~ condition,
  data = df,
  what = c("mean_sd", "n"),
  digits = 2,
  report = c("markdown", "html", "txt")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_aggregate_+3A_formula">formula</code></td>
<td>
<p>The aggregation formula. e.g., DV ~ condition.</p>
</td></tr>
<tr><td><code id="umx_aggregate_+3A_data">data</code></td>
<td>
<p>frame to aggregate (defaults to df for common case)</p>
</td></tr>
<tr><td><code id="umx_aggregate_+3A_what">what</code></td>
<td>
<p>function to use. Default reports &quot;mean (sd)&quot;.</p>
</td></tr>
<tr><td><code id="umx_aggregate_+3A_digits">digits</code></td>
<td>
<p>to round results to.</p>
</td></tr>
<tr><td><code id="umx_aggregate_+3A_report">report</code></td>
<td>
<p>Format for the table: Default is markdown.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> table
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_apply">umx_apply()</a></code>, <code><a href="stats.html#topic+aggregate">aggregate()</a></code>
</p>
</li></ul>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># =====================================
# = Basic use, compare with aggregate =
# =====================================
aggregate(mpg ~ cyl, FUN = mean, na.rm = TRUE, data = mtcars)
umx_aggregate(mpg ~ cyl, data = mtcars)

# =============================================
# = Use different (or user-defined) functions =
# =============================================
umx_aggregate(mpg ~ cyl, data = mtcars, what = "n")
umx_aggregate(mpg ~ cyl, data = mtcars, what = function(x){sum(!is.na(x))})

# turn off markdown
umx_aggregate(mpg ~ cyl, data = mtcars, report = "txt")

# ============================================
# = More than one item on the left hand side =
# ============================================
umx_aggregate(cbind(mpg, qsec) ~ cyl, data = mtcars, digits = 3)
# Transpose table
t(umx_aggregate(cbind(mpg, qsec) ~ cyl, data = mtcars))

## Not run: 
umx_aggregate(cbind(moodAvg, mood) ~ condition, data = study1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_APA_pval'>Round p-values according to APA guidelines</h2><span id='topic+umx_APA_pval'></span>

<h3>Description</h3>

<p><code>umx_APA_pval</code> formats p-values, rounded in APA style. So you get '&lt; .001' instead of .000000002 or 1.00E-09.
</p>
<p>You probably would be better off using <code><a href="#topic+umxAPA">umxAPA()</a></code>, which handles many more object types.
</p>
<p>You set the precision with digits. Optionally, you can add '=' '&lt;' etc. The default for addComparison (NA) adds these when needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_APA_pval(p, min = 0.001, digits = 3, addComparison = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_APA_pval_+3A_p">p</code></td>
<td>
<p>The p-value to round</p>
</td></tr>
<tr><td><code id="umx_APA_pval_+3A_min">min</code></td>
<td>
<p>Values below min will be reported as &quot;&lt; min&quot;</p>
</td></tr>
<tr><td><code id="umx_APA_pval_+3A_digits">digits</code></td>
<td>
<p>Number of decimals to which to round (default = 3)</p>
</td></tr>
<tr><td><code id="umx_APA_pval_+3A_addcomparison">addComparison</code></td>
<td>
<p>Whether to add '=' '&lt;' etc. (NA adds when needed)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> p-value formatted in APA style
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxAPA">umxAPA()</a></code>, <code><a href="base.html#topic+round">round()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_APA_pval(.052347)
umx_APA_pval(1.23E-3)
umx_APA_pval(1.23E-4)
umx_APA_pval(c(1.23E-3, .5))
umx_APA_pval(c(1.23E-3, .5), addComparison = TRUE)
</code></pre>

<hr>
<h2 id='umx_apply'>umx_apply</h2><span id='topic+umx_apply'></span>

<h3>Description</h3>

<p>Tries to make apply more readable. so &quot;mean of x by columns&quot;, instead of &quot;of x, by 2, mean&quot;
Other functions to think of include:
<code><a href="base.html#topic+cumsum">cumsum()</a></code>, <code><a href="base.html#topic+rowSums">rowSums()</a></code>, <code><a href="base.html#topic+colMeans">colMeans()</a></code>, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_apply(FUN, of, by = c("columns", "rows"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_apply_+3A_fun">FUN</code></td>
<td>
<p>The function to apply.</p>
</td></tr>
<tr><td><code id="umx_apply_+3A_of">of</code></td>
<td>
<p>The dataframe to work with.</p>
</td></tr>
<tr><td><code id="umx_apply_+3A_by">by</code></td>
<td>
<p>Apply the function to columns or to rows (default = &quot;columns&quot;)</p>
</td></tr>
<tr><td><code id="umx_apply_+3A_...">...</code></td>
<td>
<p>optional arguments to FUN, e.g., na.rm = TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> object
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_aggregate">umx_aggregate()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_apply(mean, mtcars, by = "columns")
umx_apply("mean", of = mtcars, by = "columns")
tmp = mtcars[1:3,]; tmp[1,1] = NA
umx_apply("mean", by = "rows", of = tmp)
umx_apply("mean", by = "rows", of = tmp, na.rm = TRUE)
</code></pre>

<hr>
<h2 id='umx_array_shift'>Like the php array_shift function: shifts an item off the beginning of a list</h2><span id='topic+umx_array_shift'></span>

<h3>Description</h3>

<p>Returns x[1]. Has the SIDE EFFECT of assigning x to x[2:end] in the container environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_array_shift(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_array_shift_+3A_x">x</code></td>
<td>
<p>the vector to shift</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> first item of x
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = c("Alice", "Bob", "Carol")
umx_array_shift(x) # returns "Alice"
x # now only 2 items (altered in containing environment)
</code></pre>

<hr>
<h2 id='umx_as_numeric'>umx_as_numeric</h2><span id='topic+umx_as_numeric'></span>

<h3>Description</h3>

<p>Convert each column of a dataframe to numeric
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_as_numeric(df, which = NULL, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_as_numeric_+3A_df">df</code></td>
<td>
<p>A [data.frame()] to convert</p>
</td></tr>
<tr><td><code id="umx_as_numeric_+3A_which">which</code></td>
<td>
<p>which columns to convert (default (null) selects all)</p>
</td></tr>
<tr><td><code id="umx_as_numeric_+3A_force">force</code></td>
<td>
<p>Whether to force conversion to numeric for non-numeric columns (defaults to FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- data.frame
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># make mpg into string, and cyl into a factor
df = mtcars
df$mpg = as.character(df$mpg)
df$cyl = factor(df$cyl)
df$am = df$am==1
df = umx_as_numeric(df); str(df) # mpg not touched
df = umx_as_numeric(df, force=TRUE); str(df) # mpg coerced back to numeric
## Not run: 
# coercing a real string will cause NAs
df$mpg = c(letters[1:16]); str(df) # replace mpg with letters.
df = umx_as_numeric(df, force=TRUE); str(df)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_check'>umx_check</h2><span id='topic+umx_check'></span>

<h3>Description</h3>

<p>Check that a test evaluates to TRUE. If not, stop, warn, or message the user
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_check(
  boolean.test,
  action = c("stop", "warning", "message"),
  message = "check failed",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_check_+3A_boolean.test">boolean.test</code></td>
<td>
<p>test evaluating to TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="umx_check_+3A_action">action</code></td>
<td>
<p>One of &quot;stop&quot; (the default), &quot;warning&quot;, or &quot;message&quot;.</p>
</td></tr>
<tr><td><code id="umx_check_+3A_message">message</code></td>
<td>
<p>what to tell the user when boolean.test is FALSE.</p>
</td></tr>
<tr><td><code id="umx_check_+3A_...">...</code></td>
<td>
<p>extra text will be pasted after the messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- boolean
</p>


<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_check(length(1:3)==3, "message", "item must have length == 3", "another comment", "and another")
umx_check(1==2, "message", "one must be 2", ". Another comment", "and another")
</code></pre>

<hr>
<h2 id='umx_check_model'>Check for required features in an OpenMx.</h2><span id='topic+umx_check_model'></span>

<h3>Description</h3>

<p>Allows the user to straight-forwardly require a specific model type (i.e., 
&quot;RAM&quot;, &quot;LISREL&quot;, etc.), whether or not the model has data, if it has been run or not. 
You can also test whether is has a means model or not and (in future) test if it has submodels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_check_model(
  obj,
  type = NULL,
  hasData = NULL,
  beenRun = NULL,
  hasMeans = NULL,
  checkSubmodels = FALSE,
  callingFn = "a function"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_check_model_+3A_obj">obj</code></td>
<td>
<p>an object to check</p>
</td></tr>
<tr><td><code id="umx_check_model_+3A_type">type</code></td>
<td>
<p>what type the model must be, i.e., &quot;RAM&quot;, &quot;LISREL&quot;, etc. (defaults to not checking NULL)</p>
</td></tr>
<tr><td><code id="umx_check_model_+3A_hasdata">hasData</code></td>
<td>
<p>whether the model should have data or not (defaults to not checking NULL)</p>
</td></tr>
<tr><td><code id="umx_check_model_+3A_beenrun">beenRun</code></td>
<td>
<p>whether the model has been run or not (defaults to not checking NULL)</p>
</td></tr>
<tr><td><code id="umx_check_model_+3A_hasmeans">hasMeans</code></td>
<td>
<p>whether the model should have a means model or not (defaults to not checking NULL)</p>
</td></tr>
<tr><td><code id="umx_check_model_+3A_checksubmodels">checkSubmodels</code></td>
<td>
<p>whether to check submodels (not implemented yet) (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_check_model_+3A_callingfn">callingFn</code></td>
<td>
<p>= Name of the calling function to help the user locate the error.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- boolean
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("check_model_ex", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
umx_check_model(m1) # TRUE, this is a model
umx_check_model(m1, type = "RAM") # equivalent to umx_is_RAM()
umx_check_model(m1, hasData = TRUE)


umx_check_model(m1, hasMeans = TRUE)
umx_check_model(m1, beenRun = FALSE)
# Model with no data
m1 = umxRAM("x ~~ .3*y", autoRun = FALSE)
umx_check_model(m1, hasData = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_check_names'>Check if a request name exists in a dataframe or related object</h2><span id='topic+umx_check_names'></span>

<h3>Description</h3>

<p>Check if a list of names are in the [namez()] of a dataframe (or the [dimnames()] of a matrix), or the names of 
the observed data of an [mzData()]
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_check_names(
  namesNeeded,
  data = NA,
  die = TRUE,
  illegal = NULL,
  no_others = FALSE,
  intersection = FALSE,
  message = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_check_names_+3A_namesneeded">namesNeeded</code></td>
<td>
<p>Variable names to find (a dataframe is also allowed)</p>
</td></tr>
<tr><td><code id="umx_check_names_+3A_data">data</code></td>
<td>
<p>data.frame, matrix, or mxData to search in for names (default NA)</p>
</td></tr>
<tr><td><code id="umx_check_names_+3A_die">die</code></td>
<td>
<p>Whether to die if the check fails (default TRUE).</p>
</td></tr>
<tr><td><code id="umx_check_names_+3A_illegal">illegal</code></td>
<td>
<p>Optional list of names which must NOT be present.</p>
</td></tr>
<tr><td><code id="umx_check_names_+3A_no_others">no_others</code></td>
<td>
<p>Whether to test that the data contain no columns in addition to those in namesNeeded (default FALSE)</p>
</td></tr>
<tr><td><code id="umx_check_names_+3A_intersection">intersection</code></td>
<td>
<p>Show the intersection of names</p>
</td></tr>
<tr><td><code id="umx_check_names_+3A_message">message</code></td>
<td>
<p>Some helpful text to append when dieing.</p>
</td></tr>
</table>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>
<p>Other Check or test: 
<code><a href="#topic+umx_is_class">umx_is_class</a>()</code>,
<code><a href="#topic+umx_is_endogenous">umx_is_endogenous</a>()</code>,
<code><a href="#topic+umx_is_exogenous">umx_is_exogenous</a>()</code>,
<code><a href="#topic+umx_is_numeric">umx_is_numeric</a>()</code>,
<code><a href="#topic+umx_is_ordered">umx_is_ordered</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(demoOneFactor) # "x1" "x2" "x3" "x4" "x5"
umx_check_names(c("x1", "x2"), demoOneFactor)
umx_check_names(c("x1", "x2"), as.matrix(demoOneFactor))
umx_check_names(c("x1", "x2"), cov(demoOneFactor[, c("x1","x2")]))
umx_check_names(c("x1", "x2"), mxData(demoOneFactor, type="raw"))
umx_check_names(c("z1", "x2"), data = demoOneFactor, die = FALSE)
umx_check_names(c("x1", "x2"), data = demoOneFactor, die = FALSE, no_others = TRUE)
umx_check_names(c("x1","x2","x3","x4","x5"), data = demoOneFactor, die = FALSE, no_others = TRUE)
# no request
umx_check_names(c(), data = demoOneFactor, die = FALSE, no_others = TRUE)

## Not run: 
# An example error from vars that don't exist in the data
umx_check_names(c("bad_var_name", "x2"), data = demoOneFactor, die = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_check_OS'>umx_check_OS</h2><span id='topic+umx_check_OS'></span>

<h3>Description</h3>

<p>Check what OS we are running on (current default is OS X). Returns a boolean.
Optionally warn or die on failure of the test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_check_OS(
  target = c("OSX", "SunOS", "Linux", "Windows"),
  action = c("ignore", "warn", "die")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_check_OS_+3A_target">target</code></td>
<td>
<p>Which OS(s) you wish to check for (default = &quot;OSX&quot;)</p>
</td></tr>
<tr><td><code id="umx_check_OS_+3A_action">action</code></td>
<td>
<p>What to do on failure of the test: nothing (default), warn or die</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> TRUE if on the specified OS (else FALSE)
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_check_OS()
</code></pre>

<hr>
<h2 id='umx_check_parallel'>Check if OpenMx is using OpenMP, test cores, and get timings</h2><span id='topic+umx_check_parallel'></span>

<h3>Description</h3>

<p>Shows how many cores you are using, and runs a test script so user can check CPU usage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_check_parallel(
  nCores = c(1, omxDetectCores()),
  testScript = NULL,
  rowwiseParallel = TRUE,
  nSubjects = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_check_parallel_+3A_ncores">nCores</code></td>
<td>
<p>How many cores to run (defaults to c(1, max). -1 = all available.</p>
</td></tr>
<tr><td><code id="umx_check_parallel_+3A_testscript">testScript</code></td>
<td>
<p>A user-provided script to run (NULL)</p>
</td></tr>
<tr><td><code id="umx_check_parallel_+3A_rowwiseparallel">rowwiseParallel</code></td>
<td>
<p>Whether to parallel-ize rows (default) or gradient computation</p>
</td></tr>
<tr><td><code id="umx_check_parallel_+3A_nsubjects">nSubjects</code></td>
<td>
<p>Number of rows to model (Default = 1000) Reduce for quicker runs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some historical (starting 2017-09-06) speeds on my late 2015 iMac, 3.3 GHz
Quad-core i7 desktop and then a quad i7 2018 MacBook Pro
</p>

<table>
<tr>
 <td style="text-align: right;">
Date       </td><td style="text-align: left;"> Version         </td><td style="text-align: left;"> Cores </td><td style="text-align: left;"> Time              </td><td style="text-align: left;">     Notes               </td>
</tr>
<tr>
 <td style="text-align: right;">
2021-07-28 </td><td style="text-align: left;"> 2.19.6.19 (git) </td><td style="text-align: left;"> 8     </td><td style="text-align: left;"> 00 min 42.98 sec  </td><td style="text-align: left;"> <code class="reqn">\Delta</code>:-80 (SLSQP laptop (55 sec under NPSOL))  </td>
</tr>
<tr>
 <td style="text-align: right;">
2021-07-28 </td><td style="text-align: left;"> 2.19.6.19 (git) </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 02 min 03 sec     </td><td style="text-align: left;"> (SLSQP on laptop)       </td>
</tr>
<tr>
 <td style="text-align: right;">
2020-08-09 </td><td style="text-align: left;"> 2.17.3  (git)   </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min 52 sec     </td><td style="text-align: left;"> (CSOLNP on laptop)      </td>
</tr>
<tr>
 <td style="text-align: right;">
2020-08-09 </td><td style="text-align: left;"> 2.17.3  (git)   </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min 40.18 sec  </td><td style="text-align: left;"> (CSOLNP on laptop)      </td>
</tr>
<tr>
 <td style="text-align: right;">
2019-06-13 </td><td style="text-align: left;"> v2.13.2 (git)   </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 11 sec    </td><td style="text-align: left;"> (NPSOL)                 </td>
</tr>
<tr>
 <td style="text-align: right;">
2019-06-13 </td><td style="text-align: left;"> v2.13.2 (git)   </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 22 sec    </td><td style="text-align: left;"> (NPSOL)                 </td>
</tr>
<tr>
 <td style="text-align: right;">
2019-06-13 </td><td style="text-align: left;"> v2.13.2 (git)   </td><td style="text-align: left;"> 6     </td><td style="text-align: left;"> 00 min, 21 sec    </td><td style="text-align: left;"> (NPSOL)                 </td>
</tr>
<tr>
 <td style="text-align: right;">
2018-10-14 </td><td style="text-align: left;"> v2.11.5 (CRAN)  </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 36 sec    </td><td style="text-align: left;"> <code class="reqn">\Delta</code>:-39.598)   </td>
</tr>
<tr>
 <td style="text-align: right;">
2018-09-17 </td><td style="text-align: left;"> v2.11.3         </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 31 sec    </td><td style="text-align: left;">                         </td>
</tr>
<tr>
 <td style="text-align: right;">
2018-09-17 </td><td style="text-align: left;"> v2.11.3         </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 30.6 sec  </td><td style="text-align: left;"> <code class="reqn">\Delta</code>: -61.49)   </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-10-16 </td><td style="text-align: left;"> v2.7.18-9       </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 07.30 sec </td><td style="text-align: left;">                         </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-10-16 </td><td style="text-align: left;"> v2.7.18-9       </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 22.63 sec </td><td style="text-align: left;"> <code class="reqn">\Delta</code>: -44.68)   </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-10-16 </td><td style="text-align: left;"> Clang OpenMP    </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 08.38 sec </td><td style="text-align: left;">                         </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-10-16 </td><td style="text-align: left;"> Clang OpenMP    </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 24.89 sec </td><td style="text-align: left;"> <code class="reqn">\Delta</code>: -43.49)   </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-09-07 </td><td style="text-align: left;"> Clang OpenMP    </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 12.90 sec </td><td style="text-align: left;">                         </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-09-07 </td><td style="text-align: left;"> Clang OpenMP    </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 32.20 sec </td><td style="text-align: left;"> <code class="reqn">\Delta</code>: -40.70    </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-09-07 </td><td style="text-align: left;"> Clang notOpenMP </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 09.90 sec </td><td style="text-align: left;">                         </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-09-07 </td><td style="text-align: left;"> TRAVIS          </td><td style="text-align: left;"> 1     </td><td style="text-align: left;"> 01 min, 06.20 sec </td><td style="text-align: left;">                         </td>
</tr>
<tr>
 <td style="text-align: right;">
2017-09-07 </td><td style="text-align: left;"> TRAVIS          </td><td style="text-align: left;"> 4     </td><td style="text-align: left;"> 00 min, 21.10 sec </td><td style="text-align: left;"> <code class="reqn">\Delta</code>: -45.00    </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# On a fast machine, takes a minute with 1 core
umx_check_parallel()

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_cont_2_quantiles'>umx_cont_2_quantiles</h2><span id='topic+umx_cont_2_quantiles'></span><span id='topic+umx2ord'></span>

<h3>Description</h3>

<p>Recode a continuous variable into n-quantiles (default = deciles (10 levels)).
It returns an <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>, with the levels labeled with the max value
in each quantile (i.e., open on the left-side). quantiles are labeled &quot;quantile1&quot;
&quot;quantile2&quot; etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_cont_2_quantiles(
  x,
  nlevels = NULL,
  type = c("mxFactor", "ordered", "unordered"),
  verbose = FALSE,
  returnCutpoints = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_cont_2_quantiles_+3A_x">x</code></td>
<td>
<p>a variable to recode as ordinal (email maintainer(&quot;umx&quot;) if you'd like this upgraded to handle df input)</p>
</td></tr>
<tr><td><code id="umx_cont_2_quantiles_+3A_nlevels">nlevels</code></td>
<td>
<p>How many bins or levels (at most) to use (i.e., 10 = deciles)</p>
</td></tr>
<tr><td><code id="umx_cont_2_quantiles_+3A_type">type</code></td>
<td>
<p>what to return (Default is &quot;mxFactor&quot;) options: &quot;ordered&quot; and &quot;unordered&quot;)</p>
</td></tr>
<tr><td><code id="umx_cont_2_quantiles_+3A_verbose">verbose</code></td>
<td>
<p>report the min, max, and decile cuts used (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_cont_2_quantiles_+3A_returncutpoints">returnCutpoints</code></td>
<td>
<p>just return the cutpoints, for use directly</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Note</strong>: Redundant quantiles are merged. i.e., if the same score identifies
all deciles up to the fourth, then these will be merged into one bin, labeled &quot;quantile4&quot;.
</p>


<h3>Value</h3>


<ul>
<li><p> recoded variable as an <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = umx_cont_2_quantiles(rnorm(1000), nlevels = 10, verbose = TRUE)
x = data.frame(x)
str(x); levels(x)
table(x)
## Not run: 
ggplot2::qplot(x$x)
y = mxDataWLS(x, type = "WLS")

## End(Not run)

# ===========================
# = Use with twin variables =
# ===========================

data(twinData)
x = twinData
cuts  = umx_cont_2_quantiles(rbind(x$wt1, x$wt2) , nlevels = 10, returnCutpoints = TRUE)
x$wt1 = umx_cont_2_quantiles(x$wt1, nlevels = cuts) # use same for both...
x$wt2 = umx_cont_2_quantiles(x$wt2, nlevels = cuts) # use same for both...
str(x[, c("wt1", "wt2")])

# More examples

x = umx_cont_2_quantiles(mtcars[, "mpg"], nlevels = 5) # quintiles
x = umx2ord(mtcars[, "mpg"], nlevels = 5) # using shorter alias
x = umx_cont_2_quantiles(mtcars[, "cyl"], nlevels = 10) # more levels than integers exist
x = umx_cont_2_quantiles(rbinom(10000, 1, .5), nlevels = 2)
</code></pre>

<hr>
<h2 id='umx_cor'>Report correlations and their p-values</h2><span id='topic+umx_cor'></span>

<h3>Description</h3>

<p>For reporting correlations and their p-values in a compact table. Handles rounding, and skipping non-numeric columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_cor(
  X,
  df = nrow(X) - 2,
  use = c("pairwise.complete.obs", "complete.obs", "everything", "all.obs",
    "na.or.complete"),
  digits = 2,
  type = c("r and p-value", "smart")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_cor_+3A_x">X</code></td>
<td>
<p>a matrix or dataframe</p>
</td></tr>
<tr><td><code id="umx_cor_+3A_df">df</code></td>
<td>
<p>the degrees of freedom for the test</p>
</td></tr>
<tr><td><code id="umx_cor_+3A_use">use</code></td>
<td>
<p>how to handle missing data (defaults to pairwise complete)</p>
</td></tr>
<tr><td><code id="umx_cor_+3A_digits">digits</code></td>
<td>
<p>rounding of answers</p>
</td></tr>
<tr><td><code id="umx_cor_+3A_type">type</code></td>
<td>
<p>Unused argument for future directions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To compute heterochoric correlations, see <code><a href="#topic+umxHetCor">umxHetCor()</a></code>.
</p>
<p><em>note</em>: The Hmisc package has a more robust function called <code>rcorr</code>.
</p>


<h3>Value</h3>


<ul>
<li><p> Matrix of correlations and p-values
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>umxHetCor
</p>
<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = myFADataRaw[1:8,1:8]
umx_cor(tmp)
tmp$x1 = letters[1:8] # make one column non-numeric
umx_cor(tmp)
</code></pre>

<hr>
<h2 id='umx_explode'>Explode a string (Like the php function <code>explode</code>)</h2><span id='topic+umx_explode'></span>

<h3>Description</h3>

<p>Takes a string and returns an array of delimited strings (by default, each single character)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_explode(delimiter = character(), string)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_explode_+3A_delimiter">delimiter</code></td>
<td>
<p>what to break the string on. Default is empty string &quot;&quot;</p>
</td></tr>
<tr><td><code id="umx_explode_+3A_string">string</code></td>
<td>
<p>an character string, e.g. &quot;dog&quot;</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> a vector of strings, e.g. c(&quot;d&quot;, &quot;o&quot;, &quot;g&quot;)
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://www.php.net/manual/en/function.explode.php">https://www.php.net/manual/en/function.explode.php</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_explode("", "dog") # "d" "o" "g"
umx_explode(" ", "cats and dogs") # [1] "cats" "and"  "dogs"
</code></pre>

<hr>
<h2 id='umx_explode_twin_names'>Break twin variable names (BMI_T1, BMI_T2) into base variable names (BMI, &quot;_T&quot;, 1:2)</h2><span id='topic+umx_explode_twin_names'></span>

<h3>Description</h3>

<p>Break names like Dep_T1 into a list of base names, a separator, and a 
vector of twin indexes. e.g.: c(&quot;Dep_T1&quot;, &quot;Dep_T2&quot;, &quot;Anx_T1&quot;, &quot;Anx_T2&quot;) will become:
</p>
<p>list(baseNames = c(&quot;Dep&quot;, &quot;Anx&quot;), sep = &quot;_T&quot;, twinIndexes = c(1,2))
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_explode_twin_names(df, sep = "_T")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_explode_twin_names_+3A_df">df</code></td>
<td>
<p>vector of names or data.frame containing the data</p>
</td></tr>
<tr><td><code id="umx_explode_twin_names_+3A_sep">sep</code></td>
<td>
<p>text constant separating name from numeric 1:2 twin index.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- list(baseNames, sep, twinIndexes)
</p>


<h3>See Also</h3>

<p>[umx_paste_names()]
</p>
<p>Other String Functions: 
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data("twinData")
umx_explode_twin_names(twinData, sep = "")
umx_explode_twin_names(twinData, sep = NULL)

# Ignore this: just a single-character/single variable test case
x = round(10 * rnorm(1000, mean = -.2))
y = round(5 * rnorm(1000))
x[x &lt; 0] = 0; y[y &lt; 0] = 0
umx_explode_twin_names(data.frame(x_T1 = x, x_T2 = y), sep = "_T")
umx_explode_twin_names(data.frame(x_T11 = x, x_T22 = y), sep = "_T")
umx_explode_twin_names(c("x_T11", "x_T22"), sep = "_T")

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_file_load_pseudo'>Read in files from pseudocons.</h2><span id='topic+umx_file_load_pseudo'></span>

<h3>Description</h3>

<p>Read in PRS scored files from <a href="https://www.staff.ncl.ac.uk/richard.howey/pseudocons/example.html">pseudocons</a>.
</p>

<ol>
<li><p> Read the file
</p>
</li>
<li><p> Break it into pseudo and real rows
</p>
</li>
<li><p> Clean-up by deleting the pseudo suffix
</p>
</li>
<li><p> Rename NT vars with a suffix
</p>
</li>
<li><p> Merge files on ID and return
</p>
</li></ol>

<table>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;"> ID </td><td style="text-align: right;"> FID </td><td style="text-align: right;"> BMIS1 </td><td style="text-align: right;"> BMIS2 </td><td style="text-align: right;"> BMIS3 </td><td style="text-align: right;"> BMIS4 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
   1 </td><td style="text-align: left;"> 1234501 </td><td style="text-align: right;"> 12345 </td><td style="text-align: right;"> -0.032 </td><td style="text-align: right;"> -0.77 </td><td style="text-align: right;"> -0.40 </td><td style="text-align: right;"> -3.87 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
   2 </td><td style="text-align: left;"> 1234501-pseudo-1 </td><td style="text-align: right;"> 12345 </td><td style="text-align: right;"> 0.117 </td><td style="text-align: right;"> -0.66 </td><td style="text-align: right;"> -0.33 </td><td style="text-align: right;"> -4.08 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Usage</h3>

<pre><code class='language-R'>umx_file_load_pseudo(fn, bp, suffix = "_NT", chosenp = "S5")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_file_load_pseudo_+3A_fn">fn</code></td>
<td>
<p>The filename</p>
</td></tr>
<tr><td><code id="umx_file_load_pseudo_+3A_bp">bp</code></td>
<td>
<p>The path to the folder containing the file</p>
</td></tr>
<tr><td><code id="umx_file_load_pseudo_+3A_suffix">suffix</code></td>
<td>
<p>to add to the NT columns (Default = &quot;_NT&quot;)</p>
</td></tr>
<tr><td><code id="umx_file_load_pseudo_+3A_chosenp">chosenp</code></td>
<td>
<p>The suffix (pvalue) we desire to use (Default = &quot;S5&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> dataframe of real and pseudo PRS columns
</p>
</li></ul>



<h3>See Also</h3>

<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
basepath = "~/Dropbox/2016 (1). project EA/2018/EA3/"
tmp = umx_file_load_pseudo("PRS_EA3_R9_autosomes_HRC1.1_pseudo.txt", bp = bp)
str(tmp)
head(tmp[, c("BMIS4", "BMIS4_NT")]

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_find_object'>umx_find_object</h2><span id='topic+umx_find_object'></span>

<h3>Description</h3>

<p>Find objects of a given class, whose name matches a search string.
The string (pattern) is grep-enabled, so you can match wild-cards
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_find_object(pattern = ".*", requiredClass = "MxModel")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_find_object_+3A_pattern">pattern</code></td>
<td>
<p>the pattern that matching objects must contain</p>
</td></tr>
<tr><td><code id="umx_find_object_+3A_requiredclass">requiredClass</code></td>
<td>
<p>the class of object that will be matched</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- a list of objects matching the class and name
</p>


<h3>References</h3>

<p>-
</p>


<h3>See Also</h3>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umx_find_object("^m[0-9]") # mxModels beginning "m1" etc.
umx_find_object("", "MxModel") # all MxModels

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_fun_mean_sd'>Summarizing functions used in umx_aggregate and for umxAPA</h2><span id='topic+umx_fun_mean_sd'></span>

<h3>Description</h3>

<p>Miscellaneous functions that are handy in summary and other tasks where you might otherwise have
to craft a custom nameless functions. e.g.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_fun_mean_sd(x, na.rm = TRUE, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_fun_mean_sd_+3A_x">x</code></td>
<td>
<p>input</p>
</td></tr>
<tr><td><code id="umx_fun_mean_sd_+3A_na.rm">na.rm</code></td>
<td>
<p>How to handle missing (default = TRUE = remove)</p>
</td></tr>
<tr><td><code id="umx_fun_mean_sd_+3A_digits">digits</code></td>
<td>
<p>Rounding (default = 2)</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li> <p><code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd()</a></code>: returns &quot;mean (SD)&quot; of x.
</p>
</li></ul>

<p><em>note</em>: if a factor is given, then the mode is returned instead of the mean and SD.
</p>


<h3>Value</h3>


<ul>
<li><p> function result
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umxAPA(mtcars[,1:3]) # uses umx_fun_mean_sd
</code></pre>

<hr>
<h2 id='umx_get_bracket_addresses'>Get bracket-style addresses from an mxMatrix</h2><span id='topic+umx_get_bracket_addresses'></span>

<h3>Description</h3>

<p>Sometimes you want these :-) This also allows you to change the matrix name: useful for using mxMatrix addresses in an mxAlgebra.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_get_bracket_addresses(mat, free = NA, newName = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_get_bracket_addresses_+3A_mat">mat</code></td>
<td>
<p>an mxMatrix to get address labels from</p>
</td></tr>
<tr><td><code id="umx_get_bracket_addresses_+3A_free">free</code></td>
<td>
<p>how to filter on free (default = NA: take all)</p>
</td></tr>
<tr><td><code id="umx_get_bracket_addresses_+3A_newname">newName</code></td>
<td>
<p>= NA</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> a list of bracket style labels
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("get_add_ex", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)#'
umx_get_bracket_addresses(m1$matrices$A, free= TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_get_checkpoint'>Get or set checkpointing for a model</h2><span id='topic+umx_get_checkpoint'></span>

<h3>Description</h3>

<p>Get the checkpoint status for a model or global options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_get_checkpoint(model = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_get_checkpoint_+3A_model">model</code></td>
<td>
<p>an optional model to get options from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umx_get_checkpoint() # current global default
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
umx_get_checkpoint(model = m1)

## End(Not run)

</code></pre>

<hr>
<h2 id='umx_get_options'>Display umx options</h2><span id='topic+umx_get_options'></span>

<h3>Description</h3>

<p>Show the umx options. Useful for beginners to discover, or people like me to remember :-)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_get_options()
</code></pre>


<h3>Value</h3>

<p>- message
</p>


<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_get_options()
</code></pre>

<hr>
<h2 id='umx_grep'>Search for text</h2><span id='topic+umx_grep'></span>

<h3>Description</h3>

<p>Search names if given a data.frame, or strings if given a vector of strings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_grep(
  df,
  grepString,
  output = c("both", "label", "name"),
  ignore.case = TRUE,
  useNames = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_grep_+3A_df">df</code></td>
<td>
<p>The <code><a href="base.html#topic+data.frame">data.frame()</a></code> or string to search.</p>
</td></tr>
<tr><td><code id="umx_grep_+3A_grepstring">grepString</code></td>
<td>
<p>the search string.</p>
</td></tr>
<tr><td><code id="umx_grep_+3A_output">output</code></td>
<td>
<p>the column name, the label, or both (default).</p>
</td></tr>
<tr><td><code id="umx_grep_+3A_ignore.case">ignore.case</code></td>
<td>
<p>whether to be case sensitive or not (default TRUE = ignore case).</p>
</td></tr>
<tr><td><code id="umx_grep_+3A_usenames">useNames</code></td>
<td>
<p>whether to search the names as well as the labels (for SPSS files with label metadata).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>namez</code> function is more flexible. A handy feature of <code>umx_grep</code> is that it can
search the labels of data imported from SPSS.
</p>
<p><em>nb:</em> To simply grep for a pattern in a string use R's built-in <code><a href="base.html#topic+grep">grep()</a></code> functions, e.g.:
<code style="white-space: pre;">&#8288;grepl("^NA\\[0-9]", "NA.3")&#8288;</code>
</p>


<h3>Value</h3>


<ul>
<li><p> list of matched column names and/or labels.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+namez">namez()</a></code>, <code><a href="#topic+umx_aggregate">umx_aggregate()</a></code>, <code><a href="base.html#topic+grep">grep()</a></code>
</p>
</li></ul>

<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_grep(mtcars, "hp", output="both", ignore.case= TRUE)
umx_grep(c("hp", "ph"), "hp")
umx_grep(mtcars, "^h.*", output="both", ignore.case= TRUE)
## Not run: 
umx_grep(spss_df, "labeltext", output = "label") 
umx_grep(spss_df, "labeltext", output = "name") 

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_has_been_run'>umx_has_been_run</h2><span id='topic+umx_has_been_run'></span>

<h3>Description</h3>

<p>check if an mxModel has been run or not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_has_been_run(model, stop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_has_been_run_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> you want to check has been run</p>
</td></tr>
<tr><td><code id="umx_has_been_run_+3A_stop">stop</code></td>
<td>
<p>Whether to stop if the model has not been run (defaults to FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> boolean
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("has_been_run_example", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
umx_has_been_run(m1)

## End(Not run)

</code></pre>

<hr>
<h2 id='umx_has_CIs'>umx_has_CIs</h2><span id='topic+umx_has_CIs'></span>

<h3>Description</h3>

<p>A utility function to return a binary answer to the question &quot;does this <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> have confidence intervals?&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_has_CIs(model, check = c("both", "intervals", "output"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_has_CIs_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to check for presence of CIs</p>
</td></tr>
<tr><td><code id="umx_has_CIs_+3A_check">check</code></td>
<td>
<p>What to check for: &quot;intervals&quot; requested, &quot;output&quot; present, or &quot;both&quot;. Defaults to &quot;both&quot;</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> TRUE or FALSE
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("_has_CI_ex", data = demoOneFactor, type = "cov",
	umxPath("g", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "g", fixedAt = 1.0)
)

umx_has_CIs(m1) # FALSE: no CIs and no output
m1 = mxModel(m1, mxCI("g_to_x1"))
umx_has_CIs(m1, check = "intervals") # TRUE intervals set
umx_has_CIs(m1, check = "output")  # FALSE not yet run
m1 = mxRun(m1)
umx_has_CIs(m1, check = "output")  # Still FALSE: Set and Run
m1 = mxRun(m1, intervals = TRUE)
umx_has_CIs(m1, check = "output")  # TRUE: Set, and Run with intervals = T
umxSummary(m1)

## End(Not run)

</code></pre>

<hr>
<h2 id='umx_has_means'>umx_has_means</h2><span id='topic+umx_has_means'></span>

<h3>Description</h3>

<p>A utility function to return a binary answer to the question &quot;does this <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> have a means model?&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_has_means(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_has_means_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to check for presence of means</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> TRUE or FALSE
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("has_means_ex", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

umx_has_means(m1)
m1 = mxModel(m1,
	mxPath(from = "one", to = manifests),
	mxData(demoOneFactor[1:100,], type = "raw")
)
umx_has_means(m1)
m1 = mxRun(m1)
umx_has_means(m1)


## End(Not run)
</code></pre>

<hr>
<h2 id='umx_has_square_brackets'>Check if a label contains square brackets</h2><span id='topic+umx_has_square_brackets'></span>

<h3>Description</h3>

<p>Helper function to check if a label has square brackets, e.g. &quot;A[1,1]&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_has_square_brackets(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_has_square_brackets_+3A_input">input</code></td>
<td>
<p>The label to check for square brackets (string input)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> boolean
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_has_square_brackets("[hello]")
umx_has_square_brackets("goodbye")
</code></pre>

<hr>
<h2 id='umx_is_class'>Check if variables in a dataframe are in a list of classes.</h2><span id='topic+umx_is_class'></span>

<h3>Description</h3>

<p>Checks the class of each column in a dataframe, seeing if they are %in% a list of classes.
Returns a vector of TRUE and FALSE, or, if all ==TRUE, a single binary (the default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_class(df, classes = NULL, all = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_class_+3A_df">df</code></td>
<td>
<p>A dataframe to check</p>
</td></tr>
<tr><td><code id="umx_is_class_+3A_classes">classes</code></td>
<td>
<p>vector of valid classes, e.g. numeric</p>
</td></tr>
<tr><td><code id="umx_is_class_+3A_all">all</code></td>
<td>
<p>Whether to return a single all() Boolean or each column individually.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Boolean or Boolean vector
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_is_numeric">umx_is_numeric()</a></code>
</p>
</li></ul>

<p>Other Check or test: 
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_is_endogenous">umx_is_endogenous</a>()</code>,
<code><a href="#topic+umx_is_exogenous">umx_is_exogenous</a>()</code>,
<code><a href="#topic+umx_is_numeric">umx_is_numeric</a>()</code>,
<code><a href="#topic+umx_is_ordered">umx_is_ordered</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_is_class(mtcars) # report class list
# Are the variables in mtcars type character?
umx_is_class(mtcars, "character") # FALSE
# They're all numeric data
umx_is_class(mtcars, "numeric") # TRUE
# Show the test-result for each variable in mtcars
umx_is_class(mtcars, "numeric") # TRUE
# Are they _either_ a char OR a num?
umx_is_class(mtcars, c("character", "numeric"))
# Is zygosity a factor (note we don't drop = F to keep as dataframe)
umx_is_class(twinData[,"zygosity", drop=FALSE], classes = "factor")
umx_is_class(mtcars$mpg) # report class of this column (same as class(mpg))
</code></pre>

<hr>
<h2 id='umx_is_cov'>umx_is_cov</h2><span id='topic+umx_is_cov'></span>

<h3>Description</h3>

<p>test if a data frame, matrix or mxData is type cov or cor, or is likely to be raw...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_cov(data = NULL, boolean = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_cov_+3A_data">data</code></td>
<td>
<p>dataframe to test</p>
</td></tr>
<tr><td><code id="umx_is_cov_+3A_boolean">boolean</code></td>
<td>
<p>whether to return the type (&quot;cov&quot;) or a boolean (default = string)</p>
</td></tr>
<tr><td><code id="umx_is_cov_+3A_verbose">verbose</code></td>
<td>
<p>How much feedback to give (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- &quot;raw&quot;, &quot;cor&quot;, or &quot;cov&quot;, (or if boolean, then T | F)
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df = cov(mtcars)
umx_is_cov(df)
df = cor(mtcars)
umx_is_cov(df)
umx_is_cov(mxData(df[1:3,1:3], type= "cov", numObs = 200))
umx_is_cov(df, boolean = TRUE)
umx_is_cov(mtcars, boolean = TRUE)
</code></pre>

<hr>
<h2 id='umx_is_endogenous'>List endogenous variables in a model</h2><span id='topic+umx_is_endogenous'></span>

<h3>Description</h3>

<p>Return a list of all the endogenous variables (variables with at least one incoming single-arrow path) in a model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_endogenous(model, manifests_only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_endogenous_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to get endogenous variables</p>
</td></tr>
<tr><td><code id="umx_is_endogenous_+3A_manifests_only">manifests_only</code></td>
<td>
<p>Whether to check only manifests (default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> list of endogenous variables
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Check or test: 
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_is_class">umx_is_class</a>()</code>,
<code><a href="#topic+umx_is_exogenous">umx_is_exogenous</a>()</code>,
<code><a href="#topic+umx_is_numeric">umx_is_numeric</a>()</code>,
<code><a href="#topic+umx_is_ordered">umx_is_ordered</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
m1 = umxRAM("umx_is_endogenous", data = demoOneFactor, type = "cov",
	umxPath("g", to = names(demoOneFactor)),
	umxPath(var = "g", fixedAt = 1),
	umxPath(var = names(demoOneFactor))
)
umx_is_endogenous(m1, manifests_only = TRUE)
umx_is_endogenous(m1, manifests_only = FALSE)


## End(Not run)
</code></pre>

<hr>
<h2 id='umx_is_exogenous'>umx_is_exogenous</h2><span id='topic+umx_is_exogenous'></span>

<h3>Description</h3>

<p>Return a list of all the exogenous variables (variables with no incoming single-arrow path) in a model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_exogenous(model, manifests_only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_exogenous_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> from which to get exogenous variables</p>
</td></tr>
<tr><td><code id="umx_is_exogenous_+3A_manifests_only">manifests_only</code></td>
<td>
<p>Whether to check only manifests (default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> list of exogenous variables
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Check or test: 
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_is_class">umx_is_class</a>()</code>,
<code><a href="#topic+umx_is_endogenous">umx_is_endogenous</a>()</code>,
<code><a href="#topic+umx_is_numeric">umx_is_numeric</a>()</code>,
<code><a href="#topic+umx_is_ordered">umx_is_ordered</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("g", to = names(demoOneFactor)),
	umxPath(var = "g", fixedAt = 1),
	umxPath(var = names(demoOneFactor))
)
umx_is_exogenous(m1, manifests_only = TRUE)
umx_is_exogenous(m1, manifests_only = FALSE)


## End(Not run)
</code></pre>

<hr>
<h2 id='umx_is_MxData'>Check if an object is an mxData object</h2><span id='topic+umx_is_MxData'></span>

<h3>Description</h3>

<p>Is the input an MxData?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_MxData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_MxData_+3A_x">x</code></td>
<td>
<p>An object to test for being an MxData object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- Boolean
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_is_MxData(mtcars)
umx_is_MxData(mxData(mtcars, type= "raw"))
umx_is_MxData(mxData(cov(mtcars), type= "cov", numObs = 73))
umx_is_MxData(mxDataWLS(na.omit(twinData[, c("wt1", "wt2")]), type= "WLS"))
</code></pre>

<hr>
<h2 id='umx_is_MxMatrix'>umx_is_MxMatrix</h2><span id='topic+umx_is_MxMatrix'></span>

<h3>Description</h3>

<p>Utility function returning a binary answer to the question &quot;Is this an OpenMx mxMatrix?&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_MxMatrix(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_MxMatrix_+3A_obj">obj</code></td>
<td>
<p>an object to be tested to see if it is an OpenMx <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Boolean
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = mxMatrix(name = "eg", type = "Full", nrow = 3, ncol = 3, values = .3)
if(umx_is_MxMatrix(x)){
	message("nice OpenMx matrix!")
}
</code></pre>

<hr>
<h2 id='umx_is_MxModel'>umx_is_MxModel</h2><span id='topic+umx_is_MxModel'></span>

<h3>Description</h3>

<p>Utility function returning a binary answer to the question &quot;Is this an OpenMx model?&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_MxModel(obj, listOK = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_MxModel_+3A_obj">obj</code></td>
<td>
<p>An object to be tested to see if it is an OpenMx <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code></p>
</td></tr>
<tr><td><code id="umx_is_MxModel_+3A_listok">listOK</code></td>
<td>
<p>Is it acceptable to pass in a list of models? (Default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Boolean
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_RAM">umx_is_RAM</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m1 = mxModel("test")
if(umx_is_MxModel(m1)){
	message("nice OpenMx model!")
}
if(umx_is_MxModel(list(m1,m1), listOK = TRUE)){
	message("nice list of OpenMx models!")
}
</code></pre>

<hr>
<h2 id='umx_is_numeric'>Check if variables in a dataframe are numeric</h2><span id='topic+umx_is_numeric'></span>

<h3>Description</h3>

<p>Checks across columns of a dataframe, return a vector of TRUE and FALSE,
or, if all ==TRUE, a single binary (the default).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_numeric(df, all = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_numeric_+3A_df">df</code></td>
<td>
<p>A dataframe to check</p>
</td></tr>
<tr><td><code id="umx_is_numeric_+3A_all">all</code></td>
<td>
<p>Whether to return a single all() Boolean or each column individually.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Boolean or Boolean vector
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_is_class">umx_is_class()</a></code>
</p>
</li></ul>

<p>Other Check or test: 
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_is_class">umx_is_class</a>()</code>,
<code><a href="#topic+umx_is_endogenous">umx_is_endogenous</a>()</code>,
<code><a href="#topic+umx_is_exogenous">umx_is_exogenous</a>()</code>,
<code><a href="#topic+umx_is_ordered">umx_is_ordered</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_is_numeric(mtcars) # TRUE
umx_is_numeric(mtcars, all=FALSE) # vector of TRUE
</code></pre>

<hr>
<h2 id='umx_is_ordered'>Test if one or more variables in a dataframe are ordered</h2><span id='topic+umx_is_ordered'></span>

<h3>Description</h3>

<p>Return the names of any ordinal variables in a dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_ordered(
  df,
  names = FALSE,
  strict = TRUE,
  binary.only = FALSE,
  ordinal.only = FALSE,
  continuous.only = FALSE,
  summaryObject = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_ordered_+3A_df">df</code></td>
<td>
<p>A <code><a href="base.html#topic+data.frame">data.frame()</a></code> or <code><a href="OpenMx.html#topic+mxData">mxData()</a></code> to look in for ordinal variables (if you offer a
matrix or vector, it will be upgraded to a dataframe)</p>
</td></tr>
<tr><td><code id="umx_is_ordered_+3A_names">names</code></td>
<td>
<p>whether to return the names of ordinal variables, or a binary (T,F) list (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_is_ordered_+3A_strict">strict</code></td>
<td>
<p>whether to stop when unordered factors are found (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_is_ordered_+3A_binary.only">binary.only</code></td>
<td>
<p>only count binary factors (2-levels) (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_is_ordered_+3A_ordinal.only">ordinal.only</code></td>
<td>
<p>only count ordinal factors (3 or more levels) (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_is_ordered_+3A_continuous.only">continuous.only</code></td>
<td>
<p>use with names = TRUE to get the names of the continuous variables</p>
</td></tr>
<tr><td><code id="umx_is_ordered_+3A_summaryobject">summaryObject</code></td>
<td>
<p>whether to return a nice summary object. Overrides other settings (FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> vector of variable names or Booleans
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Check or test: 
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_is_class">umx_is_class</a>()</code>,
<code><a href="#topic+umx_is_endogenous">umx_is_endogenous</a>()</code>,
<code><a href="#topic+umx_is_exogenous">umx_is_exogenous</a>()</code>,
<code><a href="#topic+umx_is_numeric">umx_is_numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = data.frame(ordered(rbinom(100,1,.5))); names(x) = c("x")
umx_is_ordered(x, summaryObject= TRUE) # all ordered factors including binary
tmp = mtcars

tmp$cyl = ordered(mtcars$cyl) # ordered factor
tmp$vs = ordered(mtcars$vs) # binary factor
umx_is_ordered(tmp) # true/false
umx_is_ordered(tmp, strict=FALSE)
umx_is_ordered(tmp, names = TRUE)
umx_is_ordered(tmp, names = TRUE, binary.only = TRUE)
umx_is_ordered(tmp, names = TRUE, ordinal.only = TRUE)
umx_is_ordered(tmp, names = TRUE, continuous.only = TRUE)
umx_is_ordered(tmp, continuous.only = TRUE)

x = umx_is_ordered(tmp, summaryObject= TRUE)

isContinuous = !umx_is_ordered(tmp)
## Not run: 
# nb: By default, unordered factors cause a message...
tmp$gear = factor(mtcars$gear) # Unordered factor
umx_is_ordered(tmp)
umx_is_ordered(tmp, strict = FALSE) # compare: no warning

# also: not designed to work on single variables...
umx_is_ordered(tmp$cyl)
# Do this instead...
umx_is_ordered(tmp[, "cyl", drop= FALSE])

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_is_RAM'>umx_is_RAM</h2><span id='topic+umx_is_RAM'></span>

<h3>Description</h3>

<p>Utility function returning a binary answer to the question &quot;Is this a RAM model?&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_is_RAM(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_is_RAM_+3A_obj">obj</code></td>
<td>
<p>an object to be tested to see if it is an OpenMx RAM <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Boolean
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Test: 
<code><a href="#topic+umx_check_OS">umx_check_OS</a>()</code>,
<code><a href="#topic+umx_check_model">umx_check_model</a>()</code>,
<code><a href="#topic+umx_check_names">umx_check_names</a>()</code>,
<code><a href="#topic+umx_check_parallel">umx_check_parallel</a>()</code>,
<code><a href="#topic+umx_check">umx_check</a>()</code>,
<code><a href="#topic+umx_has_CIs">umx_has_CIs</a>()</code>,
<code><a href="#topic+umx_has_been_run">umx_has_been_run</a>()</code>,
<code><a href="#topic+umx_has_means">umx_has_means</a>()</code>,
<code><a href="#topic+umx_has_square_brackets">umx_has_square_brackets</a>()</code>,
<code><a href="#topic+umx_is_MxData">umx_is_MxData</a>()</code>,
<code><a href="#topic+umx_is_MxMatrix">umx_is_MxMatrix</a>()</code>,
<code><a href="#topic+umx_is_MxModel">umx_is_MxModel</a>()</code>,
<code><a href="#topic+umx_is_cov">umx_is_cov</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("is_RAM_ex", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

if(umx_is_RAM(m1)){
	message("nice RAM model!")
}
if(!umx_is_RAM(m1)){
	message("model needs to be a RAM model")
}

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_long2wide'>Take a long twin-data file and make it wide (one family per row)</h2><span id='topic+umx_long2wide'></span>

<h3>Description</h3>

<p><code>umx_long2wide</code> merges on <code>famID</code>. Family members are ordered by <code>twinID</code>.
</p>
<p>twinID is equivalent to birth order. Up to 10 twinIDs are allowed (family order).
</p>
<p><em>Note</em>: Not all data sets have an order column, but it is essential to rank subjects correctly.
</p>
<p>You might start off with a TWID which is a concatenation of a familyID and a 2 digit twinID
</p>
<p><strong>Generating famID and twinID as used by this function</strong>
</p>
<p>You can capture the last 2 digits with the <code>mod</code> function: <code>twinID = df$TWID %% 100</code>
</p>
<p>You can <em>drop</em> the last 2 digits with integer div: <code>famID = df$TWID %/% 100</code>
</p>
<p><em>Note</em>: The functions assumes that if zygosity or any passalong variables are NA in the first
family member, they are NA everywhere. i.e., it does not hunt for values that
are present elsewhere to try and self-heal missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_long2wide(
  data,
  famID = NA,
  twinID = NA,
  zygosity = NA,
  vars2keep = NA,
  passalong = NA,
  twinIDs2keep = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_long2wide_+3A_data">data</code></td>
<td>
<p>The original (long-format) data file</p>
</td></tr>
<tr><td><code id="umx_long2wide_+3A_famid">famID</code></td>
<td>
<p>The unique identifier for members of a family</p>
</td></tr>
<tr><td><code id="umx_long2wide_+3A_twinid">twinID</code></td>
<td>
<p>The twinID. Typically 1, 2, 50 51, etc...</p>
</td></tr>
<tr><td><code id="umx_long2wide_+3A_zygosity">zygosity</code></td>
<td>
<p>Typically MZFF, DZFF MZMM, DZMM DZOS</p>
</td></tr>
<tr><td><code id="umx_long2wide_+3A_vars2keep">vars2keep</code></td>
<td>
<p>= The variables you wish to analyse (these will be renamed with paste0(&quot;_T&quot;, twinID)</p>
</td></tr>
<tr><td><code id="umx_long2wide_+3A_passalong">passalong</code></td>
<td>
<p>= Variables you wish to pass-through (keep, even though not twin vars)</p>
</td></tr>
<tr><td><code id="umx_long2wide_+3A_twinids2keep">twinIDs2keep</code></td>
<td>
<p>= If NA (the default) all twinIDs are kept, else only those listed here. Useful to drop sibs.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> dataframe in wide format
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="base.html#topic+merge">merge()</a></code>
</p>
</li></ul>

<p>Other Twin Data functions: 
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_twin_data_nice">umx_make_twin_data_nice</a>()</code>,
<code><a href="#topic+umx_residualize">umx_residualize</a>()</code>,
<code><a href="#topic+umx_scale_wide_twin_data">umx_scale_wide_twin_data</a>()</code>,
<code><a href="#topic+umx_wide2long">umx_wide2long</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ==============================================
# = First make a long format file for the demo =
# ==============================================
data(twinData)
tmp = twinData[, -2]
tmp$twinID1 = 1; tmp$twinID2 = 2
long = umx_wide2long(data = tmp, sep = "")
str(long)
# 'data.frame':	7616 obs. of  11 variables:
#  $ fam     : int  1 2 3 4 5 6 7 8 9 10 ...
#  $ zyg     : int  1 1 1 1 1 1 1 1 1 1 ...
#  $ part    : int  2 2 2 2 2 2 2 2 2 2 ...
#  $ cohort  : chr  "younger" "younger" "younger" "younger" ...
#  $ zygosity: Factor w/ 5 levels "MZFF","MZMM",..: 1 1 1 1 1 1 1 1 1 1 ...
#  $ wt      : int  58 54 55 66 50 60 65 40 60 76 ...
#  $ ht      : num  1.7 1.63 1.65 1.57 1.61 ...
#  $ htwt    : num  20.1 20.3 20.2 26.8 19.3 ...
#  $ bmi     : num  21 21.1 21 23 20.7 ...
#  $ age     : int  21 24 21 21 19 26 23 29 24 28 ...
#  $ twinID  : num  1 1 1 1 1 1 1 1 1 1 ...

# OK. Now to demo long2wide...

# Keeping all columns
wide = umx_long2wide(data= long, famID= "fam", twinID= "twinID", zygosity= "zygosity")
namez(wide) # some vars, like part, should have been passed along instead of made into "part_T1"

# ======================================
# = Demo requesting specific vars2keep =
# ======================================

# Just keep bmi and wt
wide = umx_long2wide(data= long, famID= "fam", twinID= "twinID", 
    zygosity = "zygosity", vars2keep = c("bmi", "wt")
)

namez(wide)
# "fam" "twinID" "zygosity" "bmi_T1" "wt_T1" "bmi_T2" "wt_T2"

# ==================
# = Demo passalong =
# ==================
# Keep bmi and wt, and pass through 'cohort'
wide = umx_long2wide(data= long, famID= "fam", twinID= "twinID", zygosity= "zygosity", 
	vars2keep = c("bmi", "wt"), passalong = "cohort"
)
namez(wide)


## End(Not run)

</code></pre>

<hr>
<h2 id='umx_lower.tri'>Get values from lower triangle of a matrix</h2><span id='topic+umx_lower.tri'></span>

<h3>Description</h3>

<p><code>umx_lower.tri</code> is a wrapper for <code><a href="base.html#topic+lower.tri">lower.tri()</a></code> and a selection to return
values from a lower matrix in one step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_lower.tri(x, diag = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_lower.tri_+3A_x">x</code></td>
<td>
<p>a <code><a href="base.html#topic+matrix">matrix()</a></code> from which to extract values.</p>
</td></tr>
<tr><td><code id="umx_lower.tri_+3A_diag">diag</code></td>
<td>
<p>whether to include the diagonal (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> values of cells of the lower triangle.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="base.html#topic+lower.tri">lower.tri()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = qm(1,2,3|4,5,6|7,8,9)
umx_lower.tri(x)
# 4,7,8
umx_lower.tri(x, diag=TRUE) # 1 4 7 5 8 9

</code></pre>

<hr>
<h2 id='umx_lower2full'>Convert lower-only matrix data to full (or enforce symmetry on a full matrix)</h2><span id='topic+umx_lower2full'></span>

<h3>Description</h3>

<p>Takes a vector of the lower-triangle of cells in a matrix as you might read-in
from a journal article), OR a matrix (for instance from a &quot;lower&quot; [mxMatrix()], 
and returns a full matrix, copying the lower triangle into the upper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_lower2full(lower.data, diag = NULL, byrow = TRUE, dimnames = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_lower2full_+3A_lower.data">lower.data</code></td>
<td>
<p>An [mxMatrix()]</p>
</td></tr>
<tr><td><code id="umx_lower2full_+3A_diag">diag</code></td>
<td>
<p>A boolean specifying whether the lower.data includes the diagonal</p>
</td></tr>
<tr><td><code id="umx_lower2full_+3A_byrow">byrow</code></td>
<td>
<p>Whether the matrix is to be filled by row or by column (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_lower2full_+3A_dimnames">dimnames</code></td>
<td>
<p>Optional dimnames for the matrix (defaults to NULL)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>*note*: Can also take lower data presented in the form of a data.frame. Note also, if 
presented with a full matrix, the function will return a matrix with  symmetry enforced. Can be
handy when you have a &quot;nearly-symmetrical&quot; matrix (with differences in the tenth decimal place).
</p>


<h3>Value</h3>

<p>- [mxMatrix()]
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# 1. Test with a vector in byrow = TRUE order) 
tmp = c(
	1.0000, 
	0.6247, 1.0000,
	0.3269, 0.3669, 1.0000,
	0.4216, 0.3275, 0.6404, 1.0000,
	0.2137, 0.2742, 0.1124, 0.0839, 1.0000,
	0.4105, 0.4043, 0.2903, 0.2598, 0.1839, 1.0000,
	0.3240, 0.4047, 0.3054, 0.2786, 0.0489, 0.2220, 1.0000,
	0.2930, 0.2407, 0.4105, 0.3607, 0.0186, 0.1861, 0.2707,  1.0000,
	0.2995, 0.2863, 0.5191, 0.5007, 0.0782, 0.3355, 0.2302,  0.2950, 1.0000,
	0.0760, 0.0702, 0.2784, 0.1988, 0.1147, 0.1021, 0.0931, -0.0438, 0.2087, 1.000
)
x = umx_lower2full(tmp, diag = TRUE)
# check
isSymmetric(x)

# 2. Test with matrix input
tmpn = c("ROccAsp", "REdAsp", "FOccAsp", "FEdAsp", "RParAsp", 
         "RIQ", "RSES", "FSES", "FIQ", "FParAsp")
tmp = matrix(nrow = 10, ncol = 10, byrow = TRUE, dimnames = list(tmpn,tmpn), data = 
	c(1.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,  0.0000, 0.0000, 0,
	0.6247, 1.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000,  0.0000, 0.0000, 0,
	0.3269, 0.3669, 1.0000, 0.0000, 0.0000, 0.0000, 0.0000,  0.0000, 0.0000, 0,
	0.4216, 0.3275, 0.6404, 1.0000, 0.0000, 0.0000, 0.0000,  0.0000, 0.0000, 0,
	0.2137, 0.2742, 0.1124, 0.0839, 1.0000, 0.0000, 0.0000,  0.0000, 0.0000, 0,
	0.4105, 0.4043, 0.2903, 0.2598, 0.1839, 1.0000, 0.0000,  0.0000, 0.0000, 0,
	0.3240, 0.4047, 0.3054, 0.2786, 0.0489, 0.2220, 1.0000,  0.0000, 0.0000, 0,
	0.2930, 0.2407, 0.4105, 0.3607, 0.0186, 0.1861, 0.2707,  1.0000, 0.0000, 0,
	0.2995, 0.2863, 0.5191, 0.5007, 0.0782, 0.3355, 0.2302,  0.2950, 1.0000, 0,
	0.0760, 0.0702, 0.2784, 0.1988, 0.1147, 0.1021, 0.0931, -0.0438, 0.2087, 1)
)
x = umx_lower2full(tmp, diag= TRUE)
isSymmetric(x)

# 3. Test with lower-vector, no diagonal.
tmp = c(
	0.6247,
	0.3269, 0.3669,
	0.4216, 0.3275, 0.6404,
	0.2137, 0.2742, 0.1124, 0.0839,
	0.4105, 0.4043, 0.2903, 0.2598, 0.1839,
	0.3240, 0.4047, 0.3054, 0.2786, 0.0489, 0.2220,
	0.2930, 0.2407, 0.4105, 0.3607, 0.0186, 0.1861, 0.2707, 
	0.2995, 0.2863, 0.5191, 0.5007, 0.0782, 0.3355, 0.2302,  0.2950,
	0.0760, 0.0702, 0.2784, 0.1988, 0.1147, 0.1021, 0.0931, -0.0438, 0.2087
)
umx_lower2full(tmp, diag = FALSE)
	
	# An example with byrow = FALSE
	
	ldiag = c(
	1, -.17, -.22, -.19, -.12, .81, -.02, -.26, -.2, -.15,
	1, .11, .2, .21, -.01, .7, .1, .7, .1, .17, .22,
	1, .52, .68, -.12, .09, .49, .27, .46,
	1, .5, -.06, .17, .26, .80, .31,
	1, -.1, .19, .36, .23, .42,
	1, .02, -19, -.06, -.06,
	1, .1, .18, .27,
	1, .51, .7,
	1, .55, 
	1)
umx_lower2full(tmp, byrow = FALSE, diag = TRUE)

</code></pre>

<hr>
<h2 id='umx_make'>&quot;make&quot; the umx package using devtools: release to CRAN etc.</h2><span id='topic+umx_make'></span>

<h3>Description</h3>

<p>Easily  run devtools &quot;install&quot;, &quot;release&quot;, &quot;win&quot;, &quot;examples&quot; etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_make(
  what = c("load", "quickInst", "install", "spell", "sitrep", "deps_install",
    "checkCRAN", "testthat", "examples", "win", "rhub", "lastRhub", "release"),
  pkg = "~/bin/umx",
  check = TRUE,
  run = FALSE,
  start = NULL,
  spelling = "en_US",
  which = c("win", "mac", "linux", "solaris"),
  run_dont_test = FALSE,
  spell = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_+3A_what">what</code></td>
<td>
<p>whether to &quot;install&quot;, &quot;release&quot; to CRAN, &quot;test&quot;, &quot;check&quot; test on &quot;win&quot; or &quot;rhub&quot;, &quot;spell&quot;, or &quot;examples&quot;)).</p>
</td></tr>
<tr><td><code id="umx_make_+3A_pkg">pkg</code></td>
<td>
<p>the local path to your package. Defaults to my path to umx.</p>
</td></tr>
<tr><td><code id="umx_make_+3A_check">check</code></td>
<td>
<p>Whether to run check on the package before release (default = TRUE).</p>
</td></tr>
<tr><td><code id="umx_make_+3A_run">run</code></td>
<td>
<p>If what is &quot;examples&quot;, whether to also run examples marked don't run. (default FALSE).</p>
</td></tr>
<tr><td><code id="umx_make_+3A_start">start</code></td>
<td>
<p>If what is &quot;examples&quot;, which function to start from (default (NULL) = beginning).</p>
</td></tr>
<tr><td><code id="umx_make_+3A_spelling">spelling</code></td>
<td>
<p>Whether to check spelling before release (default = &quot;en_US&quot;: set NULL to not check).</p>
</td></tr>
<tr><td><code id="umx_make_+3A_which">which</code></td>
<td>
<p>What rhub platform to use? c(&quot;mac&quot;, &quot;linux&quot;, &quot;win&quot;).</p>
</td></tr>
<tr><td><code id="umx_make_+3A_run_dont_test">run_dont_test</code></td>
<td>
<p>When checking.</p>
</td></tr>
<tr><td><code id="umx_make_+3A_spell">spell</code></td>
<td>
<p>for rhub, check spelling? TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://devtools.r-lib.org">https://devtools.r-lib.org</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# umx_make()  # Just load new code (don't rebuild help etc)
# umx_make(what = "quickInst") # Quick install
# umx_make(what = "install")   # Full package rebuild and install
# umx_make(what = "spell")     # Spellcheck Rd documents
# umx_make(what = "sitrep")    # Are needed packages up to date?
# umx_make(what = "deps_install") # Update needed packages
# umx_make(what = "examples")  # Run the examples
# umx_make(what = "checkCRAN") # Run R CMD check
# umx_make(what = "rhub")      # Check on rhub
# umx_make(what = "win")       # Check on win-builder
# umx_make(what = "release")   # Release to CRAN
# tmp = umx_make(what = "lastRhub") # View rhub result

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_make_fake_data'>umx_make_fake_data</h2><span id='topic+umx_make_fake_data'></span>

<h3>Description</h3>

<p>This function takes as argument an existing dataset, which 
must be either a matrix or a data frame. Each column of the 
dataset must consist either of numeric variables or ordered 
factors. When one or more ordered factors are included, 
then a heterogeneous correlation matrix is computed using 
John Fox's polycor package. Pairwise complete observations 
are used for all covariances, and the exact pattern of 
missing data present in the input is placed in the output,
provided a new sample size is not requested. Warnings from
the polycor::hetcor function are suppressed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_make_fake_data(
  dataset,
  digits = 2,
  n = NA,
  use.names = TRUE,
  use.levels = TRUE,
  use.miss = TRUE,
  mvt.method = "eigen",
  het.ML = FALSE,
  het.suppress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_fake_data_+3A_dataset">dataset</code></td>
<td>
<p>The original dataset of which to make a simulacrum</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_digits">digits</code></td>
<td>
<p>= Round the data to the requested digits (default = 2)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_n">n</code></td>
<td>
<p>Number of rows to generate (NA = all rows in dataset)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_use.names">use.names</code></td>
<td>
<p>Whether to name the variables (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_use.levels">use.levels</code></td>
<td>
<p>= Whether to use existing levels (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_use.miss">use.miss</code></td>
<td>
<p>Whether to have data missing as in original (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_mvt.method">mvt.method</code></td>
<td>
<p>= Passed to hetcor (default = &quot;eigen&quot;)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_het.ml">het.ML</code></td>
<td>
<p>= Passed to hetcor (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_make_fake_data_+3A_het.suppress">het.suppress</code></td>
<td>
<p>Passed to hetcor (default = TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- new dataframe
</p>


<h3>See Also</h3>

<p>[OpenMx::mxGenerateData()]
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fakeCars = umx_make_fake_data(mtcars)
</code></pre>

<hr>
<h2 id='umx_make_MR_data'>Simulate Mendelian Randomization data</h2><span id='topic+umx_make_MR_data'></span>

<h3>Description</h3>

<p>umx_make_MR_data returns a dataset containing 4 variables: A variable of interest (Y), a putative cause (X),
a qtl (quantitative trait locus) influencing X, and a confounding variable (U) affecting both X and Y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_make_MR_data(
  nSubjects = 1000,
  Vqtl = 0.02,
  bXY = 0.1,
  bUX = 0.5,
  bUY = 0.5,
  pQTL = 0.5,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_MR_data_+3A_nsubjects">nSubjects</code></td>
<td>
<p>Number of subjects in sample</p>
</td></tr>
<tr><td><code id="umx_make_MR_data_+3A_vqtl">Vqtl</code></td>
<td>
<p>Variance of QTL affecting causal variable X (Default 0.02)</p>
</td></tr>
<tr><td><code id="umx_make_MR_data_+3A_bxy">bXY</code></td>
<td>
<p>Causal effect of X on Y (Default 0.1)</p>
</td></tr>
<tr><td><code id="umx_make_MR_data_+3A_bux">bUX</code></td>
<td>
<p>Confounding effect of confounder 'U' on X (Default 0.5)</p>
</td></tr>
<tr><td><code id="umx_make_MR_data_+3A_buy">bUY</code></td>
<td>
<p>Confounding effect of confounder 'U' on Y (Default 0.5)</p>
</td></tr>
<tr><td><code id="umx_make_MR_data_+3A_pqtl">pQTL</code></td>
<td>
<p>Decreaser allele frequency (Default 0.5)</p>
</td></tr>
<tr><td><code id="umx_make_MR_data_+3A_seed">seed</code></td>
<td>
<p>value for the random number generator (Default 123)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code to make these Data. Modified from Dave Evans 2016 Boulder workshop talk.
</p>


<h3>Value</h3>

<p>- data.frame
</p>


<h3>See Also</h3>

<p>umx_make_TwinData
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df = umx_make_MR_data(10000)
str(df)
## Not run: 
m1 = umxTwoStage(Y ~ X, ~qtl, data = df)
plot(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_make_raw_from_cov'>Turn a cov matrix into raw data</h2><span id='topic+umx_make_raw_from_cov'></span>

<h3>Description</h3>

<p>A wrapper for <code><a href="MASS.html#topic+mvrnorm">MASS::mvrnorm()</a></code> to simplify turning a covariance matrix into matching raw data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_make_raw_from_cov(covMat, n, means = 0, varNames = NULL, empirical = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_raw_from_cov_+3A_covmat">covMat</code></td>
<td>
<p>A covariance matrix</p>
</td></tr>
<tr><td><code id="umx_make_raw_from_cov_+3A_n">n</code></td>
<td>
<p>How many rows of data to return</p>
</td></tr>
<tr><td><code id="umx_make_raw_from_cov_+3A_means">means</code></td>
<td>
<p>the means of the raw data (default = 0)</p>
</td></tr>
<tr><td><code id="umx_make_raw_from_cov_+3A_varnames">varNames</code></td>
<td>
<p>default uses &quot;var1&quot;, &quot;var2&quot;</p>
</td></tr>
<tr><td><code id="umx_make_raw_from_cov_+3A_empirical">empirical</code></td>
<td>
<p>(passed to mvrnorm) Default = FALSE</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> data.frame
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+cov2cor">cov2cor()</a></code>, <code><a href="MASS.html#topic+mvrnorm">MASS::mvrnorm()</a></code>
</p>
</li></ul>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>covData &lt;- matrix(nrow=6, ncol=6, byrow=TRUE, dimnames=list(paste0("v", 1:6), paste0("v", 1:6)),
  data = c(0.9223099, 0.1862938, 0.4374359, 0.8959973, 0.9928430, 0.5320662,
           0.1862938, 0.2889364, 0.3927790, 0.3321639, 0.3371594, 0.4476898,
           0.4374359, 0.3927790, 1.0069552, 0.6918755, 0.7482155, 0.9013952,
           0.8959973, 0.3321639, 0.6918755, 1.8059956, 1.6142005, 0.8040448,
           0.9928430, 0.3371594, 0.7482155, 1.6142005, 1.9223567, 0.8777786,
           0.5320662, 0.4476898, 0.9013952, 0.8040448, 0.8777786, 1.3997558)
)

myData = umx_make_raw_from_cov(covData, n = 100, means = 1:6)
umxAPA(myData)
covMat = matrix(c(1, .3, .3, 1), nrow=2)
tmp= umx_make_raw_from_cov(covMat, n=10, varNames= c("x", "y"))
cov(tmp)
tmp= umx_make_raw_from_cov(covMat, n=10, varNames= c("x", "y"), empirical= TRUE)
cov(tmp)
tmp= umx_make_raw_from_cov(qm(1, .3| .3, 1), n=10, varNames= c("x", "y"))
cov(tmp)
</code></pre>

<hr>
<h2 id='umx_make_sql_from_excel'>Convert an excel spreadsheet in a text file on sql statements.</h2><span id='topic+umx_make_sql_from_excel'></span>

<h3>Description</h3>

<p>Unlikely to be of use to anyone but the package author :-)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_make_sql_from_excel(theFile = "Finder")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_sql_from_excel_+3A_thefile">theFile</code></td>
<td>
<p>The xlsx file to read. Default = &quot;Finder&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>On OS X, by default, the file selected in the front-most Finder window will be chosen.
If it is blank, a choose file dialog will be thrown.
</p>
<p>Read an xlsx file and convert into SQL insert statements (placed on the clipboard)
On MacOS, the function can access the current front-most Finder window.
</p>
<p>The file name should be the name of the test.
Columns should be headed:
itemText	direction	scale	type	[optional	response	options]
</p>
<p>The SQL fields generated are:
itemID, test, native_item_number, item_text, direction, scale, format, author
</p>
<p>tabbedPlus: list scored from 0 to n-1
</p>
<p>tabbedVertPlus: tabbed, but vertical lay-out
</p>
<p>number	2+2\&lt;itemBreak\&gt;min='0' max='7' step='1'
</p>
<p>5fm Scored 1-5, anchored: Strongly Disagree | Disagree  | Neutral | Agree | Strongly Agree
</p>
<p>intro (not) scored, and sequenced as item 0
</p>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# An example Excel spreadsheet
# local uncompiled path
fp = system.file("inst/extdata", "GQ6.sql.xlsx", package = "umx")
# installed path
fp = system.file("extdata", "GQ6.sql.xlsx", package = "umx")
umx_open(fp)
umx_make_sql_from_excel() # Using file selected in front-most Finder window
umx_make_sql_from_excel("~/Desktop/test.xlsx") # provide a path

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_make_twin_data_nice'>Convert a twin dataset into umx standard format.</h2><span id='topic+umx_make_twin_data_nice'></span>

<h3>Description</h3>

<p><code>umx_make_twin_data_nice</code> is a function to convert your twin data into a format used across <code>umx</code>. Specifically:
</p>

<ol>
<li><p> Existing column for zygosity is renamed to &quot;zygosity&quot;.
</p>
</li>
<li> <p><code>sep</code> is set to &quot;_T&quot;
</p>
</li>
<li><p> The twinID is is set to sequential digits, i.e. 1,2...
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>umx_make_twin_data_nice(
  data,
  sep = "",
  zygosity = "zygosity",
  numbering,
  labelNumericZygosity = FALSE,
  levels = 1:5,
  labels = c("MZFF", "MZMM", "DZFF", "DZMM", "DZOS")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_twin_data_nice_+3A_data">data</code></td>
<td>
<p>a <code><a href="base.html#topic+data.frame">data.frame()</a></code> to check/convert.</p>
</td></tr>
<tr><td><code id="umx_make_twin_data_nice_+3A_sep">sep</code></td>
<td>
<p>existing separator string (will be updated to &quot;_T&quot;).</p>
</td></tr>
<tr><td><code id="umx_make_twin_data_nice_+3A_zygosity">zygosity</code></td>
<td>
<p>existing zygosity column name (will be renamed <code>zygosity</code>).</p>
</td></tr>
<tr><td><code id="umx_make_twin_data_nice_+3A_numbering">numbering</code></td>
<td>
<p>existing twin sequence string (will be updated to _T1, _T2, _T3).</p>
</td></tr>
<tr><td><code id="umx_make_twin_data_nice_+3A_labelnumericzygosity">labelNumericZygosity</code></td>
<td>
<p>If TRUE numeric zygosity levels will be set to labels.</p>
</td></tr>
<tr><td><code id="umx_make_twin_data_nice_+3A_levels">levels</code></td>
<td>
<p>legal levels of zygosity (ignored if labelNumericZygosity = FALSE (default 1:5)</p>
</td></tr>
<tr><td><code id="umx_make_twin_data_nice_+3A_labels">labels</code></td>
<td>
<p>labels for each zyg level c(&quot;MZFF&quot;, &quot;MZMM&quot;, &quot;DZFF&quot;, &quot;DZMM&quot;, &quot;DZOS&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="base.html#topic+data.frame">data.frame()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">tutorials</a>, <a href="https://github.com/tbates/umx">tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_wide2long">umx_wide2long()</a></code>, <code><a href="#topic+umx_long2wide">umx_long2wide()</a></code>,
</p>
</li></ul>

<p>Other Twin Data functions: 
<code><a href="#topic+umx_long2wide">umx_long2wide</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_residualize">umx_residualize</a>()</code>,
<code><a href="#topic+umx_scale_wide_twin_data">umx_scale_wide_twin_data</a>()</code>,
<code><a href="#topic+umx_wide2long">umx_wide2long</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
tmp = twinData
tmp2 = umx_make_twin_data_nice(twinData, sep="", numbering = 1:5, zygosity="zygosity")
tmp$zygosity=NULL
tmp = umx_make_twin_data_nice(twinData, sep="", numbering = 1:5, zygosity="zygosity")
namez(tmp, "zyg")
levels(tmp$zygosity)

</code></pre>

<hr>
<h2 id='umx_make_TwinData'>Simulate twin data with control over A, C, and E parameters, as well as moderation of A.</h2><span id='topic+umx_make_TwinData'></span>

<h3>Description</h3>

<p>Makes MZ and DZ twin data, optionally with moderated A. By default, the three variance components must sum to 1.
</p>
<p>See examples for how to use this: it is pretty flexible.
</p>
<p>If you provide 2 varNames, they will be used for twin 1 and twin 2. If you provide one, it will be expanded to var_T1 and var_T2.
<strong>note</strong>: the function was designed around nSib = 2 and var names = var_T1. It isn't yet smart enough to do, for instance
scaling or shifting to make the min value 0 (normal for most traits we analyse) for nonstandard <code>varNames</code> and 'nSib&ldquo;.
</p>
<p><em>Note</em>, if you want a power calculator, see <code><a href="#topic+power.ACE.test">power.ACE.test()</a></code> and <code><a href="#topic+umxPower">umxPower()</a></code>.
</p>
<p><strong>Usage</strong>
</p>
<p>You must supply <code>nMZpairs</code> (you can omit <code>nDZpairs</code>).
You can give any two of A, C, or E and the function deduces the missing parameter so <code>A+C+E == 1</code>.
</p>
<p><strong>Moderation</strong>
</p>
<p><strong>Univariate GxE Data</strong>
To simulate data for <code>umxGxE</code>, offer up a list of the average, min and max values for <code>AA</code>, i.e., c(avg = .5, min = 0, max = 1).
</p>
<p><code>umx_make_TwinData</code> will return moderated data, with average value = avg, swinging
down to min and up to max across 3-SDs of the moderator.
</p>
<p><strong>Bivariate GxE Data</strong>
</p>
<p>To simulate data with a moderator that is not shared by both twins.
Moderated heritability is specified via the bivariate relationship (AA, CC, EE) and two moderators in each component.
AA   = list(a11 = .4, a12 = .1, a22 = .15)
CC   = list(c11 = .2, c12 = .1, c22 = .10)
EE   = list(e11 = .4, e12 = .3, e22 = .25)
Amod = list(Beta_a1 = .025, Beta_a2 = .025)
Cmod = list(Beta_c1 = .025, Beta_c2 = .025)
Emod = list(Beta_e1 = .025, Beta_e2 = .025)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_make_TwinData(
  nMZpairs,
  nDZpairs = nMZpairs,
  AA = NULL,
  CC = NULL,
  EE = NULL,
  DD = NULL,
  varNames = "var",
  MZr = NULL,
  DZr = MZr,
  nSib = 2,
  dzAr = 0.5,
  scale = FALSE,
  mean = 0,
  sd = 1,
  nThresh = NULL,
  sum2one = TRUE,
  bivAmod = NULL,
  bivCmod = NULL,
  bivEmod = NULL,
  seed = NULL,
  empirical = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_make_TwinData_+3A_nmzpairs">nMZpairs</code></td>
<td>
<p>Number of MZ pairs to simulate</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_ndzpairs">nDZpairs</code></td>
<td>
<p>Number of DZ pairs to simulate (defaults to nMZpairs)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_aa">AA</code></td>
<td>
<p>value for A variance. NOTE: See options for use in GxE and Bivariate GxE</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_cc">CC</code></td>
<td>
<p>value for C variance.</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_ee">EE</code></td>
<td>
<p>value for E variance.</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_dd">DD</code></td>
<td>
<p>value for E variance.</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_varnames">varNames</code></td>
<td>
<p>name for variables (defaults to 'var')</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_mzr">MZr</code></td>
<td>
<p>If MZr and DZr are set (default = NULL), the function returns dataframes of the request n and correlation.</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_dzr">DZr</code></td>
<td>
<p>Set to return dataframe using MZr and Dzr (Default NULL)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_nsib">nSib</code></td>
<td>
<p>Number of siblings in a family (default = 2). &quot;3&quot; = extra sib.</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_dzar">dzAr</code></td>
<td>
<p>DZ Ar (default .5)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_scale">scale</code></td>
<td>
<p>Whether to scale output to var=1 mean=0 (Default FALSE)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_mean">mean</code></td>
<td>
<p>mean for traits (default = 0) (not applied to moderated cases)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_sd">sd</code></td>
<td>
<p>sd of traits (default = 1) (not applied to moderated cases)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_nthresh">nThresh</code></td>
<td>
<p>If supplied, use as thresholds and return mxFactor output? (default is not to)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_sum2one">sum2one</code></td>
<td>
<p>Whether to enforce AA + CC + EE summing the one (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_bivamod">bivAmod</code></td>
<td>
<p>Used for Bivariate GxE data: list(Beta_a1 = .025, Beta_a2 = .025)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_bivcmod">bivCmod</code></td>
<td>
<p>Used for Bivariate GxE data: list(Beta_c1 = .025, Beta_c2 = .025)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_bivemod">bivEmod</code></td>
<td>
<p>Used for Bivariate GxE data: list(Beta_e1 = .025, Beta_e2 = .025)</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_seed">seed</code></td>
<td>
<p>Allows user to set.seed() if wanting reproducible dataset</p>
</td></tr>
<tr><td><code id="umx_make_TwinData_+3A_empirical">empirical</code></td>
<td>
<p>Passed to mvrnorm</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> list of mzData and dzData dataframes containing T1 and T2 plus, if needed M1 and M2 (moderator values)
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxACE">umxACE()</a></code>, <code><a href="#topic+umxGxE">umxGxE()</a></code>, <code><a href="#topic+umxGxEbiv">umxGxEbiv()</a></code>
</p>
</li></ul>

<p>Other Twin Data functions: 
<code><a href="#topic+umx_long2wide">umx_long2wide</a>()</code>,
<code><a href="#topic+umx_make_twin_data_nice">umx_make_twin_data_nice</a>()</code>,
<code><a href="#topic+umx_residualize">umx_residualize</a>()</code>,
<code><a href="#topic+umx_scale_wide_twin_data">umx_scale_wide_twin_data</a>()</code>,
<code><a href="#topic+umx_wide2long">umx_wide2long</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># =====================================================================
# = Basic Example, with all elements of std univariate data specified =
# =====================================================================
tmp = umx_make_TwinData(nMZpairs = 10000, AA = .30, CC = .00, EE = .70)
# Show dataframe with 20,000 rows and 3 variables: var_T1, var_T2, and zygosity
str(tmp)

# ===============================
# = How to consume the datasets =
# ===============================

mzData = tmp[tmp$zygosity == "MZ", ]
dzData = tmp[tmp$zygosity == "DZ", ]
str(mzData); str(dzData); 
cov(mzData[, c("var_T1", "var_T2")])
cov(dzData[, c("var_T1", "var_T2")])
umxAPA(mzData[, c("var_T1", "var_T2")])

# Prefer to work in path coefficient values? (little a?)
tmp    = umx_make_TwinData(2000, AA = .7^2, CC = .0)
mzData = tmp[tmp$zygosity == "MZ", ]
dzData = tmp[tmp$zygosity == "DZ", ]
m1 = umxACE(selDVs="var", sep="_T", mzData= mzData, dzData= dzData)

# Examine correlations
cor(mzData[,c("var_T1","var_T2")])
cor(dzData[,c("var_T1","var_T2")])

# Example with D (left un-modeled in ACE)
tmp = umx_make_TwinData(nMZpairs = 500, AA = .4, DD = .2, CC = .2)
m1 = umxACE(selDVs="var", data = tmp, mzData= "MZ", dzData= "DZ")
# |    |   a1|   c1|   e1|
# |:---|----:|----:|----:|
# |var | 0.86| 0.24| 0.45|

m1 = umxACE(selDVs="var", data = tmp, mzData= "MZ", dzData= "DZ", dzCr=.25)
# |    |  a1|d1 |   e1|
# |:---|---:|:--|----:|
# |var | 0.9|.  | 0.44|


# =============
# = Shortcuts =
# =============

# Omit nDZpairs (equal numbers of both by default)
tmp = umx_make_TwinData(100, AA = 0.5, CC = 0.3) # omit any one of A, C, or E (sums to 1)
cov(tmp[tmp$zygosity == "DZ", c("var_T1","var_T2")])

# Not limited to unit variance
tmp = umx_make_TwinData(100, AA = 3, CC = 2, EE = 3, sum2one = FALSE) 
cov(tmp[tmp$zygosity == "MZ", c("var_T1","var_T2")])

# Output can be scaled (mean=0, std=1)
tmp = umx_make_TwinData(100, AA = .7, CC = .1, scale = TRUE) 
cov(tmp[tmp$zygosity == "MZ", c("var_T1","var_T2")])

## Not run: 

# ===============
# = GxE Example =
# ===============

AA = c(avg = .5, min = .1, max = .8)
tmp = umx_make_TwinData(nMZpairs = 140, nDZpairs = 240, AA = AA, CC = .35, EE = .65, scale= TRUE)
mzData = tmp[tmp$zygosity == "MZ", ]
dzData = tmp[tmp$zygosity == "DZ", ]
m1 = umxGxE(selDVs = "var", selDefs = "M", sep = "_T", mzData = mzData, dzData = dzData)

# =====================
# = Threshold Example =
# =====================
tmp = umx_make_TwinData(100, AA = .6, CC = .2, nThresh = 3)
str(tmp)
umx_polychoric(subset(tmp, zygosity=="MZ", c("var_T1", "var_T2")))$polychorics
# Running model with 7 parameters
#           var_T1    var_T2
# var_T1 1.0000000 0.7435457
# var_T2 0.7435457 1.0000000


# =================================================
# = Just use MZr and DZr (also works with nSib&gt;2) =
# =================================================
tmp = umx_make_TwinData(100, MZr = .86, DZr = .60, nSib= 3, varNames = "IQ")
umxAPA(subset(tmp, zygosity == "MZ", paste0("IQ_T", 1:2)))
umxAPA(subset(tmp, zygosity == "DZ", paste0("IQ_T", 1:2)))
m1 = umxACE(selDVs= "IQ", data = tmp)
m1 = umxACE(selDVs= "IQ", data = tmp, nSib=3)
# TODO tmx_ examples of unmodeled D etc.

# Bivariate GxSES example (see umxGxEbiv)

AA   = list(a11 = .4, a12 = .1, a22 = .15)
CC   = list(c11 = .2, c12 = .1, c22 = .10)
EE   = list(e11 = .4, e12 = .3, e22 = .25)
Amod = list(Beta_a1 = .025, Beta_a2 = .025)
Cmod = list(Beta_c1 = .025, Beta_c2 = .025)
Emod = list(Beta_e1 = .025, Beta_e2 = .025)
tmp = umx_make_TwinData(5000, AA =AA, CC = CC, EE = EE, 
			bivAmod = Amod, bivCmod =Cmod, bivEmod =Emod)
str(tmp)
# 'data.frame':	10000 obs. of  7 variables:
#  $ defM_T1 : num  0.171 0.293 -0.173 0.238 -0.73 ...
#  $ defM_T2 : num  0.492 -0.405 -0.696 -0.829 -0.858 ...
#  $ M_T1    : num  0.171 0.293 -0.173 0.238 -0.73 ...
#  $ var_T1  : num  0.011 0.1045 0.5861 0.0583 1.0225 ...
#  $ M_T2    : num  0.492 -0.405 -0.696 -0.829 -0.858 ...
#  $ var_T2  : num  -0.502 -0.856 -0.154 0.065 -0.268 ...
#  $ zygosity: Factor w/ 2 levels "MZ","DZ": 1 1 1 1 1 1 1 1 1 1 ...

# TODO tmx example showing how moderation of A introduces heteroscedasticity in a regression model:
# More residual variance at one extreme of the x axis (moderator) 
# m1 = lm(var_T1~ M_T1, data = x); 
# x = rbind(tmp[[1]], tmp[[2]])
# plot(residuals(m1)~ x$M_T1, data=x)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_means'>umx_means</h2><span id='topic+umx_means'></span>

<h3>Description</h3>

<p>Helper to get means from a df that might contain ordered or string data.
Factor means are set to &quot;ordVar&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_means(df, ordVar = 0, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_means_+3A_df">df</code></td>
<td>
<p>a dataframe of raw data from which to get variances.</p>
</td></tr>
<tr><td><code id="umx_means_+3A_ordvar">ordVar</code></td>
<td>
<p>value to return for the means of factor data = 0</p>
</td></tr>
<tr><td><code id="umx_means_+3A_na.rm">na.rm</code></td>
<td>
<p>passed to mean - defaults to &quot;na.rm&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- frame of means
</p>


<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = mtcars[,1:4]
tmp$cyl = ordered(mtcars$cyl) # ordered factor
tmp$hp  = ordered(mtcars$hp)  # binary factor
umx_means(tmp, ordVar = 0, na.rm = TRUE)
</code></pre>

<hr>
<h2 id='umx_merge_randomized_columns'>umx_merge_randomized_columns</h2><span id='topic+umx_merge_randomized_columns'></span>

<h3>Description</h3>

<p><code>umx_merge_randomized_columns</code> is designed to merge data where subjects have been randomized
to conditions, so they have a value in one column, and NA in the other condition columns.
</p>
<p>It returns a new column of merged scores, and a new column of associated conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_merge_randomized_columns(
  colNames,
  df,
  levels = colNames,
  newVarName = "score",
  newCondName = "condition",
  as.factor = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_merge_randomized_columns_+3A_colnames">colNames</code></td>
<td>
<p>Names of the columns containing the condition data.</p>
</td></tr>
<tr><td><code id="umx_merge_randomized_columns_+3A_df">df</code></td>
<td>
<p>The data frame</p>
</td></tr>
<tr><td><code id="umx_merge_randomized_columns_+3A_levels">levels</code></td>
<td>
<p>optional names for the levels of condition (default = <code>colNames</code>).</p>
</td></tr>
<tr><td><code id="umx_merge_randomized_columns_+3A_newvarname">newVarName</code></td>
<td>
<p>Name for the new column holding the newVarName (default &quot;score&quot;).</p>
</td></tr>
<tr><td><code id="umx_merge_randomized_columns_+3A_newcondname">newCondName</code></td>
<td>
<p>Name for the new column holding the condition (default &quot;condition&quot;).</p>
</td></tr>
<tr><td><code id="umx_merge_randomized_columns_+3A_as.factor">as.factor</code></td>
<td>
<p>Turn condition into a factor? (FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> df with new cols
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_long2wide">umx_long2wide()</a></code>, <code><a href="#topic+prolific_check_ID">prolific_check_ID()</a></code>, <code><a href="#topic+prolific_read_demog">prolific_read_demog()</a></code>, <code><a href="#topic+prolific_anonymize">prolific_anonymize()</a></code>
</p>
</li></ul>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fp = "~/Desktop/Political Ideology_September 13, 2022_10.47.xlsx"
df = readxl::read_excel(fp)
df = df[c(-1,-2), ] # delete temp data and question text
df = data.frame(df)
namez(df, "ris", coll = "vec") # c('RiskAversionNoLotter', 'RiskAversionLottery')
colNames= c('RiskAversionNoLotter', 'RiskAversionLottery')
df = umx_as_numeric(df, colNames, force=TRUE)
tmp = umx_merge_randomized_columns(colNames, df); table(tmp$condition)
tmp = umx_merge_randomized_columns(colNames, df, 
	levels = c("treatment", "control")); table(tmp$condition)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_move_file'>Move files</h2><span id='topic+umx_move_file'></span>

<h3>Description</h3>

<p>On OS X, <code>umx_move_file</code> can access the current front-most Finder window.
The file moves are fast and, because you can use regular expressions, powerful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_move_file(
  baseFolder = NA,
  regex = NULL,
  fileNameList = NA,
  destFolder = NA,
  test = TRUE,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_move_file_+3A_basefolder">baseFolder</code></td>
<td>
<p>The folder to search in. If set to &quot;Finder&quot; (and you are on OS X) it will use the current
front-most Finder window. If it is blank, a choose folder dialog will be thrown.</p>
</td></tr>
<tr><td><code id="umx_move_file_+3A_regex">regex</code></td>
<td>
<p>string to select files to process within the selected folder.</p>
</td></tr>
<tr><td><code id="umx_move_file_+3A_filenamelist">fileNameList</code></td>
<td>
<p>List of files to move.</p>
</td></tr>
<tr><td><code id="umx_move_file_+3A_destfolder">destFolder</code></td>
<td>
<p>Folder to move files to.</p>
</td></tr>
<tr><td><code id="umx_move_file_+3A_test">test</code></td>
<td>
<p>Boolean determining whether to change the names, or just report a dry run.</p>
</td></tr>
<tr><td><code id="umx_move_file_+3A_overwrite">overwrite</code></td>
<td>
<p>Boolean determining whether to overwrite files or not (default = FALSE (safe)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+file.rename">file.rename()</a></code>, <code><a href="base.html#topic+regex">regex()</a></code>
</p>
<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
base = "~/Desktop/"
dest = "~/Music/iTunes/iTunes Music/Music/"
umx_move_file(baseFolder = base, fileNameList = toMove, destFolder = dest, test= TRUE)

# ============================================================
# = Move all files in downloads ending in ".jpeg" to Desktop =
# ============================================================
umx_move_file(baseFolder = "~/Downloads/", regex=".jpeg", 
	destFolder = "~/Desktop/", test= TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='umx_msg'>Print the name and compact contents of variable.</h2><span id='topic+umx_msg'></span>

<h3>Description</h3>

<p>Helper function to ease debugging with console notes like:  &quot;ObjectName = \&lt;Object Value\&gt;&quot;.
This is primarily useful for inline debugging, where seeing, e.g., &quot;nVar = 3&quot; can be useful.
The ability to say <code>umx_msg(nVar)</code> makes this easy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_msg(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_msg_+3A_x">x</code></td>
<td>
<p>the thing you want to pretty-print</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> NULL
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a = "brian"
umx_msg(a)
b = c("brian", "sally", "jane")
umx_msg(b)
umx_msg(mtcars)
</code></pre>

<hr>
<h2 id='umx_names'>umx_names</h2><span id='topic+umx_names'></span><span id='topic+namez'></span>

<h3>Description</h3>

<p>Convenient equivalent of running <a href="base.html#topic+grep">grep</a> on <a href="base.html#topic+names">names</a>, with value = TRUE and ignore.case = TRUE.
</p>
<p><strong>Plus</strong>:<code>umx_names</code> can handle dataframes, a model, list of models, model summary, or a vector of strings as input.
</p>
<p>In these cases, it will search column names, parameter or summary output names, or
the literal string values themselves respectively.
</p>
<p>In addition, <code>umx_names</code> can do <a href="base.html#topic+grep">replacement</a> of a found string (see examples). It can also collapse the result (using <a href="base.html#topic+paste0">paste0</a>)
</p>
<p><em>Note</em>: <code>namez</code> (with a z) is a shortcut for <code>umx_names</code>, which makes it easy to replace where you would otherwise use <a href="base.html#topic+names">names</a>.
</p>
<p>You can learn more about the matching options (like inverting the selection etc.) in the help for base-R <a href="base.html#topic+grep">grep</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_names(
  df,
  pattern = ".*",
  replacement = NULL,
  ignore.case = TRUE,
  perl = FALSE,
  value = TRUE,
  fixed = FALSE,
  useBytes = FALSE,
  invert = FALSE,
  global = FALSE,
  collapse = c("as.is", "vector", "formula")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_names_+3A_df">df</code></td>
<td>
<p>dataframe (or other objects, or a list of models) from which to get names.</p>
</td></tr>
<tr><td><code id="umx_names_+3A_pattern">pattern</code></td>
<td>
<p>Used to find only matching names (supports grep/regular expressions)</p>
</td></tr>
<tr><td><code id="umx_names_+3A_replacement">replacement</code></td>
<td>
<p>If not NULL, replaces the found string. Use backreferences (&quot;\1&quot; to &quot;\9&quot;) to refer to (subexpressions).</p>
</td></tr>
<tr><td><code id="umx_names_+3A_ignore.case">ignore.case</code></td>
<td>
<p>default = TRUE (opposite default to grep)</p>
</td></tr>
<tr><td><code id="umx_names_+3A_perl">perl</code></td>
<td>
<p>Should Perl-compatible regexps be used? Default = FALSE</p>
</td></tr>
<tr><td><code id="umx_names_+3A_value">value</code></td>
<td>
<p>Return matching elements themselves (TRUE) or their indices (FALSE) default = TRUE (opposite default to grep)</p>
</td></tr>
<tr><td><code id="umx_names_+3A_fixed">fixed</code></td>
<td>
<p>= FALSE (grep option If TRUE, pattern is a string to be matched as is. Overrides all conflicting arguments.)</p>
</td></tr>
<tr><td><code id="umx_names_+3A_usebytes">useBytes</code></td>
<td>
<p>= FALSE logical. grep option. If TRUE, matching is by byte rather than by character.</p>
</td></tr>
<tr><td><code id="umx_names_+3A_invert">invert</code></td>
<td>
<p>Return indices or values for elements that do not match (default = FALSE).</p>
</td></tr>
<tr><td><code id="umx_names_+3A_global">global</code></td>
<td>
<p>replace all instances in each strong, or just the first (Default).</p>
</td></tr>
<tr><td><code id="umx_names_+3A_collapse">collapse</code></td>
<td>
<p>&quot;as.is&quot; leaves alone. as.vector formats as pasteable code, i.e., &quot;c('a', 'b')&quot;, not &quot;a&quot;  &quot;b&quot; (default NULL), etc.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> vector of matches
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li><p> Base-R pattern matching functions: <code><a href="base.html#topic+grep">grep()</a></code>.
And <code><a href="#topic+umx_check_names">umx_check_names()</a></code> to check for existence of names in a dataframe.
</p>
</li></ul>

<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Names from a dataframe, with character matching
umx_names(mtcars, "mpg") # only "mpg" matches this

# Easy-to-type alias "namez"
namez(mtcars, "mpg")

# Use a regular expression to match a pattern
namez(mtcars, "r[ab]") # "drat", "carb"
namez(mtcars, "^d") # vars beginning with 'd' = "disp", drat

# Use this function to replace text in names!
umx_names(mtcars, "mpg", replacement = "hello") # "mpg" replaced with "hello"


# ========================================================================
# = Using the custom collapse option to quote each item, and wrap in c() =
# ========================================================================
namez(mtcars, "m", collapse = "vector") # Paste-able R-code for a vector

# Other options passed to R's grep command
umx_names(mtcars, "mpg" , invert = TRUE)  # Non-matches (instead of matches)
umx_names(mtcars, "disp", value  = FALSE) # Return indices of matches 
umx_names(mtcars, "disp", value  = "grepl")  # which var matches disp
umx_names(mtcars, "^d"  , fixed  = TRUE)  # Vars containing literal '^d' (none...)

# =======================================
# = Examples using built-in GFF dataset =
# =======================================

# Just show phenotypes for Twin 1
umx_names(GFF, "_T1$") # twin 1
# "zyg" "sex1" "age_T1" "gff_T1" "fc_T1" "qol_T1" "hap_T1"...

umx_names(GFF, "2$") # names ending in 2
umx_names(GFF, "[^12bs]$") # doesn't end in `1`, `2`, `b`, or `s`
# "zyg_6grp" "zyg_2grp" "divorce"
umx_names(mxData(twinData[, c("wt1", "wt2")], type= "raw"))
umx_names(mxData(cov(twinData[, c("wt1", "wt2")], use="comp"), type= "cov", numObs= 1000))
umx_names(mxDataWLS(na.omit(twinData[, c("wt1", "wt2")]), type= "WLS"))

namez(umxMatrix("bob", "Full", 3,3)$labels)

</code></pre>

<hr>
<h2 id='umx_open'>Open a file or folder</h2><span id='topic+umx_open'></span>

<h3>Description</h3>

<p>Open a file or folder. Works on OS X, mostly on windows, and hopefully on unix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_open(filepath = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_open_+3A_filepath">filepath</code></td>
<td>
<p>The file to open</p>
</td></tr>
</table>


<h3>Details</h3>

<p>NOTE: Your filepath is <code><a href="base.html#topic+shQuote">shQuote()</a></code>'d by this function.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umx_open() # Default is to open working directory getwd()
umx_open("~/bin/umx/R/misc_and_utility copy.r")

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_open_CRAN_page'>Open the CRAN page for a package</h2><span id='topic+umx_open_CRAN_page'></span>

<h3>Description</h3>

<p>On MacOS, this function opens the CRAN page for a package.
Useful for looking up documentation, checking you have an
up-to-date version, showing the package to people etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_open_CRAN_page(package = "umx", inst = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_open_CRAN_page_+3A_package">package</code></td>
<td>
<p>An <span class="rlang"><b>R</b></span> package name.</p>
</td></tr>
<tr><td><code id="umx_open_CRAN_page_+3A_inst">inst</code></td>
<td>
<p>Install and load if not already installed?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umx_open_CRAN_page("umx")

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_pad'>Pad an Object with NAs</h2><span id='topic+umx_pad'></span>

<h3>Description</h3>

<p>This function pads an R object (list, data.frame, matrix, atomic vector)
with <code>NA</code>s. For matrices, lists and data.frames, this occurs by extending
each (column) vector in the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_pad(x, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_pad_+3A_x">x</code></td>
<td>
<p>An <span class="rlang"><b>R</b></span> object (list, data.frame, matrix, atomic vector).</p>
</td></tr>
<tr><td><code id="umx_pad_+3A_n">n</code></td>
<td>
<p>The final length of each object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- padded object
</p>


<h3>References</h3>

<p>- <a href="https://github.com/kevinushey/Kmisc/tree/master/man">https://github.com/kevinushey/Kmisc/tree/master/man</a>
</p>


<h3>See Also</h3>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_pad(1:3, 4)
umx_pad(1:3, 3)
</code></pre>

<hr>
<h2 id='umx_paste_names'>Concatenate base variable names with suffixes to create wide-format variable names (i.e twin-format)</h2><span id='topic+umx_paste_names'></span><span id='topic+tvars'></span>

<h3>Description</h3>

<p>It's easier to work with base names, rather than the twice-as-long hard-to-typo list of column names.
<code>umx_paste_names</code> adds suffixes to names so you can work with that nice short list.
So, you provide <code>bmi</code>, and you get back fully specified family-wise names: <code>c("bmi_T1", "bmi_T2")</code>
</p>
<p><em>note</em>: <code>tvars</code> is a shortcut for <code>umx_paste_names</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_paste_names(
  varNames,
  sep = "",
  suffixes = 1:2,
  covNames = NULL,
  prefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_paste_names_+3A_varnames">varNames</code></td>
<td>
<p>a list of <em>base</em> names, e.g c(&quot;bmi&quot;, &quot;IQ&quot;)</p>
</td></tr>
<tr><td><code id="umx_paste_names_+3A_sep">sep</code></td>
<td>
<p>A string separating the name and the twin suffix, e.g. &quot;_T&quot; (default is &quot;&quot;)</p>
</td></tr>
<tr><td><code id="umx_paste_names_+3A_suffixes">suffixes</code></td>
<td>
<p>a list of terminal suffixes differentiating the twins default = 1:2)</p>
</td></tr>
<tr><td><code id="umx_paste_names_+3A_covnames">covNames</code></td>
<td>
<p>a list of <em>base</em> names for covariates (to be sorted last in list), e.g c(&quot;age&quot;, &quot;sex&quot;)</p>
</td></tr>
<tr><td><code id="umx_paste_names_+3A_prefix">prefix</code></td>
<td>
<p>a string to prepend to each label, e.g &quot;mean&quot; -&gt; &quot;mean_age&quot; &quot;mean_sex&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Method 1</strong>: <em>Use complete suffixes</em>
</p>
<p>You can provide complete suffixes like &quot;_T1&quot; and &quot;_T2&quot;. This has the benefit of being explicit
and very general:
</p>
<div class="sourceCode"><pre>umx_paste_names(c("var1", "var2"), suffixes = c("_T1", "_T2"))
</pre></div>
<p><em>Note</em>: for quick typing, <code>tvars</code> is an alias for <code>umx_paste_names</code>
</p>
<p><strong>Method 2</strong>: <em>Use sep and a suffix vector.</em>
</p>
<p>Alternatively, you can use <code>sep</code> to add a constant like &quot;_T&quot; after each basename, along
with a vector of suffixes. This has the benefit of showing what is varying:
This is then suffixed with e.g. &quot;1&quot;, &quot;2&quot;.
</p>
<div class="sourceCode"><pre>umx_paste_names(c("var1", "var2"), sep = "_T", suffixes = 1:2)
</pre></div>
<p><em>Working with covariates</em>
</p>
<p>If you are using <code><a href="#topic+umxACEcov">umxACEcov()</a></code>, you <strong>need</strong> to keep all the covariates at the end of the list.
Here's how:
</p>
<div class="sourceCode"><pre>umx_paste_names(c("var1", "var2"), cov = c("cov1"), sep = "_T", suffixes = 1:2)
</pre></div>
<p><em>note</em>: in conventional twin models, the expCov matrix is T1 vars, followed by T2 vars. For covariates, you want
T1vars, T2 vars, T1 covs, T2 covs. This is what <code>covNames</code> accomplishes.
</p>


<h3>Value</h3>


<ul>
<li><p> vector of suffixed var names, i.e., c(&quot;v1_T1&quot;, &quot;v2_T1&quot;, &quot;v1_T2&quot;, &quot;v2_T2&quot;, &quot;cov_T1&quot;, &quot;cov_T2&quot;)
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+namez">namez()</a></code> <code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names()</a></code>
</p>
<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># two styles doing the same thing: first is more general
umx_paste_names("bmi", suffixes = c("_T1", "_T2"))
umx_paste_names("bmi", sep = "_T", suffixes = 1:2)
varNames = umx_paste_names(c("N", "E", "O", "A", "C"), "_T", 1:2)
umx_paste_names(c("IQ", "C"), cov = c("age"), sep = "_T", suffixes = 1:2)
umx_paste_names(c("IQ", "C"), cov = c("age"), sep = "_T", prefix= "mean_")
# For quick-typing, tvars is an alias for umx_paste_names
tvars(c("IQ", "C"), cov = "age", sep = "_T", prefix= "mean_")
tvars("IQ")
</code></pre>

<hr>
<h2 id='umx_polychoric'>FIML-based polychoric, polyserial, and Pearson correlations</h2><span id='topic+umx_polychoric'></span>

<h3>Description</h3>

<p>Compute polychoric/polyserial/Pearson correlations with FIML.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_polychoric(
  data,
  useDeviations = TRUE,
  tryHard = c("no", "yes", "ordinal", "search")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_polychoric_+3A_data">data</code></td>
<td>
<p>Dataframe</p>
</td></tr>
<tr><td><code id="umx_polychoric_+3A_usedeviations">useDeviations</code></td>
<td>
<p>Whether to code the mode using deviation thresholds (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_polychoric_+3A_tryhard">tryHard</code></td>
<td>
<p>'no' uses normal mxRun (default), &quot;yes&quot; uses mxTryHard, and others used named versions: &quot;mxTryHardOrdinal&quot;, &quot;mxTryHardWideSearch&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- list of output and diagnostics. matrix of correlations = $polychorics
</p>


<h3>References</h3>

<p>- Barendse, M. T., Ligtvoet, R., Timmerman, M. E., &amp; Oort, F. J. (2016). Model Fit after Pairwise Maximum Likelihood. *Frontiers in psychology*, **7**, 528. <a href="https://doi.org/10.3389/fpsyg.2016.00528">doi:10.3389/fpsyg.2016.00528</a>.
</p>


<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tmp = mtcars
tmp$am = umxFactor(mtcars$am)
tmp$vs = umxFactor(mtcars$vs)
tmp = umx_scale(tmp)
x = umx_polychoric(tmp[, c("am", "vs")], tryHard = "yes")
x$polychorics
cor(mtcars[, c("am", "vs")])


## End(Not run)

</code></pre>

<hr>
<h2 id='umx_polypairwise'>FIML-based Pairwise polychoric, polyserial, and Pearson correlations</h2><span id='topic+umx_polypairwise'></span>

<h3>Description</h3>

<p>Compute polychoric/polyserial/Pearson correlations with FIML in OpenMx
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_polypairwise(
  data,
  useDeviations = TRUE,
  printFit = FALSE,
  use = "any",
  tryHard = c("no", "yes", "ordinal", "search")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_polypairwise_+3A_data">data</code></td>
<td>
<p>Dataframe</p>
</td></tr>
<tr><td><code id="umx_polypairwise_+3A_usedeviations">useDeviations</code></td>
<td>
<p>Whether to code the mode using deviation thresholds (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_polypairwise_+3A_printfit">printFit</code></td>
<td>
<p>Whether to print information about the fit achieved (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_polypairwise_+3A_use">use</code></td>
<td>
<p>parameter (default = &quot;any&quot;)</p>
</td></tr>
<tr><td><code id="umx_polypairwise_+3A_tryhard">tryHard</code></td>
<td>
<p>'no' uses normal mxRun (default), &quot;yes&quot; uses mxTryHard, and others used named versions: &quot;mxTryHardOrdinal&quot;, &quot;mxTryHardWideSearch&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- matrix of correlations
</p>


<h3>References</h3>

<p>- Barendse, M. T., Ligtvoet, R., Timmerman, M. E., &amp; Oort, F. J. (2016). Model Fit after Pairwise Maximum Likelihood. *Frontiers in psychology*, **7**, 528. <a href="https://doi.org/10.3389/fpsyg.2016.00528">doi:10.3389/fpsyg.2016.00528</a>.
</p>


<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_set_optimizer("SLSQP")
tmp = mtcars
tmp$am = umxFactor(mtcars$am)
tmp$vs = umxFactor(mtcars$vs)
tmp = umx_scale(tmp)
x = umx_polypairwise(tmp[, c("hp", "mpg", "am", "vs")], tryHard = "yes")
x$R
cov2cor(x$R)
cor(mtcars[, c("hp", "mpg", "am", "vs")])
</code></pre>

<hr>
<h2 id='umx_polytriowise'>FIML-based trio-based polychoric, polyserial, and Pearson correlations</h2><span id='topic+umx_polytriowise'></span>

<h3>Description</h3>

<p>Compute polychoric/polyserial/Pearson correlations with FIML in OpenMx.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_polytriowise(
  data,
  useDeviations = TRUE,
  printFit = FALSE,
  use = "any",
  tryHard = c("no", "yes", "ordinal", "search")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_polytriowise_+3A_data">data</code></td>
<td>
<p>Dataframe</p>
</td></tr>
<tr><td><code id="umx_polytriowise_+3A_usedeviations">useDeviations</code></td>
<td>
<p>Whether to code the mode using deviation thresholds (default = TRUE)</p>
</td></tr>
<tr><td><code id="umx_polytriowise_+3A_printfit">printFit</code></td>
<td>
<p>Whether to print information about the fit achieved (default = FALSE)</p>
</td></tr>
<tr><td><code id="umx_polytriowise_+3A_use">use</code></td>
<td>
<p>parameter (default = &quot;any&quot;)</p>
</td></tr>
<tr><td><code id="umx_polytriowise_+3A_tryhard">tryHard</code></td>
<td>
<p>'no' uses normal mxRun (default), &quot;yes&quot; uses mxTryHard, and others used named versions: &quot;mxTryHardOrdinal&quot;, &quot;mxTryHardWideSearch&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- matrix of correlations
</p>


<h3>References</h3>

<p>- <a href="https://doi.org/10.3389/fpsyg.2016.00528">doi:10.3389/fpsyg.2016.00528</a>
</p>


<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = mtcars
tmp$am = umxFactor(mtcars$am)
tmp$vs = umxFactor(mtcars$vs)
tmp = umx_scale(tmp)
x = umx_polytriowise(tmp[, c("hp", "mpg", "am", "vs")], tryHard = "yes")
x$R
cor(mtcars[, c("hp", "mpg", "am", "vs")])

</code></pre>

<hr>
<h2 id='umx_print'>Print tables in a range of formats (markdown default, see <code><a href="#topic+umx_set_table_format">umx_set_table_format()</a></code> for other formats)
or as a web browser table.</h2><span id='topic+umx_print'></span>

<h3>Description</h3>

<p>To aid interpretability of printed tables from OpenMx (and elsewhere)
you can change how NA and zero appear, and suppressing values below a certain cut-off.
By default, Zeros have the decimals suppressed, and NAs are suppressed altogether.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_print(
  x,
  digits = getOption("digits"),
  caption = NULL,
  report = c("markdown", "html"),
  file = c(NA, "tmp.html"),
  na.print = "",
  zero.print = "0",
  justify = "none",
  quote = FALSE,
  suppress = NULL,
  kableExtra = TRUE,
  append = FALSE,
  sortableDF = TRUE,
  html_font = NULL,
  style = c("paper", "material_dark", "classic", "classic_2", "minimal", "material"),
  bootstrap_options = c("hover", "bordered", "condensed", "responsive"),
  lightable_options = "striped",
  both = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_print_+3A_x">x</code></td>
<td>
<p>A data.frame to print (matrices will be coerced to data.frame)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_digits">digits</code></td>
<td>
<p>The number of decimal places to print (getOption(&quot;digits&quot;))</p>
</td></tr>
<tr><td><code id="umx_print_+3A_caption">caption</code></td>
<td>
<p>Optional caption.</p>
</td></tr>
<tr><td><code id="umx_print_+3A_report">report</code></td>
<td>
<p>How to report the results. &quot;html&quot; = open in browser.</p>
</td></tr>
<tr><td><code id="umx_print_+3A_file">file</code></td>
<td>
<p>Whether to write to a file (defaults to NA (no file). Use &quot;html&quot; to open table in browser.</p>
</td></tr>
<tr><td><code id="umx_print_+3A_na.print">na.print</code></td>
<td>
<p>How to display NAs (default = &quot;&quot;)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_zero.print">zero.print</code></td>
<td>
<p>How to display 0 values (default = &quot;0&quot;) for sparse tables, using &quot;.&quot; can produce more readable results.</p>
</td></tr>
<tr><td><code id="umx_print_+3A_justify">justify</code></td>
<td>
<p>Parameter passed to print (defaults to &quot;none&quot;)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_quote">quote</code></td>
<td>
<p>Whether or not to quote strings (FALSE)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_suppress">suppress</code></td>
<td>
<p>Minimum numeric value to print (NULL = print all values, no matter how small)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_kableextra">kableExtra</code></td>
<td>
<p>Whether to print the table using kableExtra (if report=&quot;html&quot;)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_append">append</code></td>
<td>
<p>If html, is this appended to file? (FALSE)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_sortabledf">sortableDF</code></td>
<td>
<p>If html, is table sortable? (TRUE)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_html_font">html_font</code></td>
<td>
<p>Override default font. e.g. &quot;Times&quot; or '&quot;Arial Narrow&quot;, arial, helvetica, sans-s'</p>
</td></tr>
<tr><td><code id="umx_print_+3A_style">style</code></td>
<td>
<p>The style for the table &quot;paper&quot;,&quot;material_dark&quot; etc.</p>
</td></tr>
<tr><td><code id="umx_print_+3A_bootstrap_options">bootstrap_options</code></td>
<td>
<p>e.g.  border etc.</p>
</td></tr>
<tr><td><code id="umx_print_+3A_lightable_options">lightable_options</code></td>
<td>
<p>e.g. striped</p>
</td></tr>
<tr><td><code id="umx_print_+3A_both">both</code></td>
<td>
<p>If html, is table also printed as markdown? (TRUE)</p>
</td></tr>
<tr><td><code id="umx_print_+3A_...">...</code></td>
<td>
<p>Optional parameters for print</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A dataframe of text
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umx_msg">umx_msg()</a></code>, <code><a href="#topic+umx_set_table_format">umx_set_table_format()</a></code>
</p>
<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mtcars)
umx_print(mtcars[1:10,], digits = 2, zero.print = ".", justify = "left")
umx_print(mtcars[1,1:2], digits = 2, zero.print = "")
umx_print(mtcars[1,1:2], digits = 2, caption = "Hi: I'm the caption!")
## Not run: 
umx_print(mtcars[1:10,], report = "html")

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_r_test'>Test the difference between correlations for significance.</h2><span id='topic+umx_r_test'></span>

<h3>Description</h3>

<p><code>umx_r_test</code> is a wrapper around the cocor test of difference between correlations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_r_test(
  data = NULL,
  vars = vars,
  alternative = c("two.sided", "greater", "less")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_r_test_+3A_data">data</code></td>
<td>
<p>The dataset.</p>
</td></tr>
<tr><td><code id="umx_r_test_+3A_vars">vars</code></td>
<td>
<p>Three or 4 variables forming the two pairs of columns.</p>
</td></tr>
<tr><td><code id="umx_r_test_+3A_alternative">alternative</code></td>
<td>
<p>A two (default) or one-sided (greater less) test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Non-overlapping (no variable in common) correlations in the same dataset.</strong>
If 4 variables are provided in <code>vars</code>, <code>umx_r_test</code> conducts a test of
the correlation of var 1 &amp; 2 differs in magnitude from the correlation of var 3 with var 4.
(<code>r.jk</code> and <code>r.hm</code> in cocor speak).
</p>
<p><strong>Overlapping (1 variable in common) correlations in the same dataset.</strong>
If 3 variables are provided in <code>vars</code>, <code>umx_r_test</code> conducts a test of whether
the correlation of var 1 &amp; 2 differs in magnitude from the correlation of var 1 with var 3.
(<code>r.jk</code> and <code>r.jh</code> in cocor speak).
</p>
<p>In the future it will be expanded to handle other correlations, and to take correlations as input.
</p>


<h3>Value</h3>

<p>cocor result.
</p>


<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Is the correlation of mpg with cylinder count different from that 
# obtaining between disp and hp?
vars = c("mpg", "cyl", "disp", "hp")
umx_r_test(mtcars, vars)
umx_r_test(mtcars, c("mpg", "disp", "hp"))
</code></pre>

<hr>
<h2 id='umx_read_lower'>Read lower-triangle of data matrix from console or file</h2><span id='topic+umx_read_lower'></span>

<h3>Description</h3>

<p>umx_read_lower will read a lower triangle of data, either from the
console, or from file, and return a full matrix, optionally coerced to
positive definite. This is useful, especially when copying data from a paper
that includes just the lower triangle of a correlation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_read_lower(file = "", diag = TRUE, names = NULL, ensurePD = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_read_lower_+3A_file">file</code></td>
<td>
<p>Path to file (Default &quot;&quot; will read from user input)</p>
</td></tr>
<tr><td><code id="umx_read_lower_+3A_diag">diag</code></td>
<td>
<p>Whether data include diagonal (Default TRUE)</p>
</td></tr>
<tr><td><code id="umx_read_lower_+3A_names">names</code></td>
<td>
<p>Variable names. (Default as.character(paste0(&quot;X&quot;, 1:n)) )</p>
</td></tr>
<tr><td><code id="umx_read_lower_+3A_ensurepd">ensurePD</code></td>
<td>
<p>Whether to coerce the resultant matrix to positive definite (Default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="base.html#topic+matrix">matrix()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx) # for umxRAM
IQtests = c("brainstorm", "matrix", "moral", "shopping", "typing")
allCols = c("C", IQtests, "avgIQ", "maxIQ", "video")

df = umx_read_lower(diag = FALSE, names = allCols)
0.38
0.86	0.30
0.42	0.12	0.27
0.66	0.21	0.38	0.18
0.80	0.13	0.50	0.25	0.43
0.19	0.11	0.19	0.12	-0.06	0.22
0.27	0.09	0.33	0.05	-0.04	0.28	.73
0.52	0.17	0.38	0.37	0.39	0.44	0.18	0.13

dimnames(df) = list(allCols, allCols) # manually add

df = umx_read_lower(file = "", diag = FALSE, names = allCols, ensurePD= TRUE)
0.38
0.86	0.30
0.42	0.12	0.27
0.66	0.21	0.38	0.18
0.80	0.13	0.50	0.25	0.43
0.19	0.11	0.19	0.12	-0.06	0.22
0.27	0.09	0.33	0.05	-0.04	0.28	.73
0.52	0.17	0.38	0.37	0.39	0.44	0.18	0.13


round(df, 2) 

m1 = umxRAM("wooley", data = mxData(df, type="cov", numObs = 90),
	umxPath("g", to = IQtests),
	umxPath(var = "g", fixedAt= 1),
	umxPath(var = IQtests)
)
summary(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_rename'>umx_rename</h2><span id='topic+umx_rename'></span>

<h3>Description</h3>

<p>Returns a dataframe with variables renamed as desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_rename(
  data,
  from = NULL,
  to = NULL,
  regex = NULL,
  test = FALSE,
  old = "deprecated_from",
  replace = "deprecated_to"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_rename_+3A_data">data</code></td>
<td>
<p>The dataframe in which to rename variables</p>
</td></tr>
<tr><td><code id="umx_rename_+3A_from">from</code></td>
<td>
<p>List of existing names that will be found and replaced by the contents of replace. (optional: Defaults to NULL).</p>
</td></tr>
<tr><td><code id="umx_rename_+3A_to">to</code></td>
<td>
<p>If used alone, a named collection of c(oldName = &quot;newName&quot;) pairs.
OR, if &quot;from&quot; is a list of existing names, the list of new names)
OR, if &quot;regex&quot; is a regular expression, the replace string)</p>
</td></tr>
<tr><td><code id="umx_rename_+3A_regex">regex</code></td>
<td>
<p>Regular expression with matches will be replaced using replace as the replace string. (Optional: Defaults to NULL).</p>
</td></tr>
<tr><td><code id="umx_rename_+3A_test">test</code></td>
<td>
<p>Whether to report a &quot;dry run&quot;, not changing anything. (Default = FALSE).</p>
</td></tr>
<tr><td><code id="umx_rename_+3A_old">old</code></td>
<td>
<p>deprecated: use from</p>
</td></tr>
<tr><td><code id="umx_rename_+3A_replace">replace</code></td>
<td>
<p>deprecated: use to</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Unlike similar functions in other packages, it checks that the variables exist, and that the new names do not.
</p>
<p>Importantly, it also supports <a href="base.html#topic+regex">regular expressions</a>. This allows you to find and replace
text based on patterns and replacements. so to change &quot;replacement&quot; to &quot;in place&quot;,
<code style="white-space: pre;">&#8288;grep=re(place)ment&#8288;</code>, <code style="white-space: pre;">&#8288;replace= in \\1&#8288;</code>.
</p>
<p><em>note</em>:To use replace list, you must say c(old = &quot;new&quot;), not c(old -&gt; &quot;new&quot;)
</p>


<h3>Value</h3>


<ul>
<li><p> dataframe with columns renamed.
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+namez">namez</a> to filter (and replace) names, Also <a href="#topic+umx_check_names">umx_check_names</a> to check for existence of names in a dataframe.
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = mtcars

tmp = umx_rename(tmp, to = c(cyl = "cylinder"))
# let's check cyl has been changed to cylinder...
namez(tmp, "c")

# Alternate style: from-&gt;to, first with a test-run
# Dry run
tmp = umx_rename(tmp, from = "disp", to = "displacement", test= TRUE)
# Actually do it
tmp = umx_rename(tmp, from = c("disp"), to = c("displacement"))
umx_check_names("displacement", data = tmp, die = TRUE)
namez(tmp, "disp")

# This will warn that "disp" does not exist (anymore)
new = c("auto", "displacement", "rear_axle_ratio")
tmp = umx_rename(tmp, from = c("am", "disp", "drat"), to = new)
namez(tmp, "a") # still updated am to auto (and rear_axle_ratio)

# Test using regex (in this case to revert "displacement" to "disp")
tmp = umx_rename(tmp, regex = "lacement", to = "", test= TRUE) 
tmp = umx_rename(tmp, regex = "lacement", to = "") # revert to disp
umx_names(tmp, "^d") # all names beginning with a d

# dev: check deprecated format handled...
tmp = umx_rename(tmp, old = c("am", "disp", "drat"), replace = new)
</code></pre>

<hr>
<h2 id='umx_rename_file'>Rename files</h2><span id='topic+umx_rename_file'></span>

<h3>Description</h3>

<p>Rename files. On OS X, the function can access the current front-most Finder window.
The file renaming is fast and, because you can use regular expressions too change names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_rename_file(
  findStr = "old",
  replaceStr = NA,
  baseFolder = "Finder",
  test = TRUE,
  ignoreSuffix = TRUE,
  listPattern = NULL,
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_rename_file_+3A_findstr">findStr</code></td>
<td>
<p>The pattern to find, i.e., &quot;cats&quot;</p>
</td></tr>
<tr><td><code id="umx_rename_file_+3A_replacestr">replaceStr</code></td>
<td>
<p>The replacement pattern &quot;\1 are not dogs&quot;</p>
</td></tr>
<tr><td><code id="umx_rename_file_+3A_basefolder">baseFolder</code></td>
<td>
<p>Folder to search in. Default (&quot;Finder&quot;) will use the current front-most Finder window (on MacOS).
Set to NA for a &quot;choose folder&quot; dialog.</p>
</td></tr>
<tr><td><code id="umx_rename_file_+3A_test">test</code></td>
<td>
<p>Boolean determining whether to change files on disk, or just report on what would have happened (Defaults to test = TRUE)</p>
</td></tr>
<tr><td><code id="umx_rename_file_+3A_ignoresuffix">ignoreSuffix</code></td>
<td>
<p>Whether to ignore (don't search in) the suffix (file-type like .mpg) TRUE.</p>
</td></tr>
<tr><td><code id="umx_rename_file_+3A_listpattern">listPattern</code></td>
<td>
<p>A pre-filter for files</p>
</td></tr>
<tr><td><code id="umx_rename_file_+3A_overwrite">overwrite</code></td>
<td>
<p>Boolean determining if an existing file will be overwritten (Defaults to the safe FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_write_to_clipboard">umx_write_to_clipboard</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# "Season 01" --&gt; "S01" in current folder in MacOS Finder
umx_rename_file("[Ss]eason +([0-9]+)", replaceStr="S\\1", test = TRUE)

# move date to end of file name
umx_rename_file("^(.*) *([0-9]{2}\\.[0-9]{2}\\.[0-9]+) *(.*)", replaceStr="\\1 \\3 \\2")


## End(Not run)
</code></pre>

<hr>
<h2 id='umx_reorder'>Reorder or drop variables from a correlation/covariance matrix.</h2><span id='topic+umx_reorder'></span>

<h3>Description</h3>

<p>Reorder the variables in a correlation matrix. Can also remove one or more variables from a matrix using this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_reorder(old, newOrder, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_reorder_+3A_old">old</code></td>
<td>
<p>a square matrix of correlation or covariances to reorder</p>
</td></tr>
<tr><td><code id="umx_reorder_+3A_neworder">newOrder</code></td>
<td>
<p>Variables you want in the order you wish to have</p>
</td></tr>
<tr><td><code id="umx_reorder_+3A_force">force</code></td>
<td>
<p>Just assume input is value (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- the re-ordered/resized matrix
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oldMatrix = cov(mtcars)
umx_reorder(oldMatrix, newOrder = c("mpg", "cyl", "disp")) # first 3
umx_reorder(oldMatrix, newOrder = c("hp", "disp", "cyl")) # subset and reordered
umx_reorder(oldMatrix, "hp") # edge-case of just 1-var
</code></pre>

<hr>
<h2 id='umx_residualize'>Easily residualize variables in long or wide dataframes, returning them changed in-place.</h2><span id='topic+umx_residualize'></span>

<h3>Description</h3>

<p>Residualize one or more variables residualized against covariates, and return a
complete dataframe with residualized variable in place.
Optionally, this also works on wide (i.e., twin) data. Just supply suffixes to identify
the paired-wide columns (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_residualize(var, covs = NULL, suffixes = NULL, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_residualize_+3A_var">var</code></td>
<td>
<p>The base name of the variable you want to residualize. Alternatively, a
regression <code><a href="stats.html#topic+formula">formula()</a></code> containing var on the lhs, and covs on the rhs</p>
</td></tr>
<tr><td><code id="umx_residualize_+3A_covs">covs</code></td>
<td>
<p>Covariates to residualize on.</p>
</td></tr>
<tr><td><code id="umx_residualize_+3A_suffixes">suffixes</code></td>
<td>
<p>Suffixes that identify the variable for each twin, i.e. c(&quot;_T1&quot;, &quot;_T2&quot;)
Up to you to check all variables are present!</p>
</td></tr>
<tr><td><code id="umx_residualize_+3A_data">data</code></td>
<td>
<p>The dataframe containing all the variables</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In R, residuals for a variable can be found with the following statement:
</p>
<p><code>tmp = residuals(lm(var ~ cov1 + cov2, data = data, na.action = na.exclude))</code>
</p>
<p>This tmp variable could then be written over the old data:
</p>
<p>umx_residualize obviates the user having to build the lm, set na.action, or replace the data.
In addition, it has the powerful feature of operating on a list of variables, and of operating on
wide data, expanding the var name using a set of variable-name suffixes.
</p>


<h3>Value</h3>


<ul>
<li><p> dataframe with var residualized in place (i.e under its original column name)
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Twin Data functions: 
<code><a href="#topic+umx_long2wide">umx_long2wide</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_twin_data_nice">umx_make_twin_data_nice</a>()</code>,
<code><a href="#topic+umx_scale_wide_twin_data">umx_scale_wide_twin_data</a>()</code>,
<code><a href="#topic+umx_wide2long">umx_wide2long</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Residualize mpg on cylinders and displacement
r1 = umx_residualize("mpg", c("cyl", "disp"), data = mtcars)
r2 = residuals(lm(mpg ~ cyl + disp, data = mtcars, na.action = na.exclude))
all(r1$mpg == r2)

# =============================
# = Use the formula interface =
# =============================
r1 = umx_residualize(mpg ~ cyl + I(cyl^2) + disp, data = mtcars)

# validate against using lm
r2 = residuals(lm(mpg ~ cyl + I(cyl^2) + disp, data = mtcars, na.action = na.exclude))
all(r1$mpg == r2)

# ===========================================================
# = Residualize twin data (i.e. wide or "1 family per row") =
# ===========================================================
# Make some toy "twin" data to demonstrate with
tmp = mtcars
tmp$mpg_T1  = tmp$mpg_T2  = tmp$mpg
tmp$cyl_T1  = tmp$cyl_T2  = tmp$cyl
tmp$disp_T1 = tmp$disp_T2 = tmp$disp

covs = c("cyl", "disp")
tmp= umx_residualize(var="mpg", covs=covs, suffixes=c("_T1","_T2"), data = tmp)
str(tmp[1:5, 12:17])

# ===================================
# = Residualize several DVs at once =
# ===================================
df1 = umx_residualize(c("mpg", "hp"), cov = c("cyl", "disp"), data = tmp)
df2 = residuals(lm(hp ~ cyl + disp, data = tmp, na.action = na.exclude))
all(df1$hp == df2)
</code></pre>

<hr>
<h2 id='umx_rot'>Rotate a vector</h2><span id='topic+umx_rot'></span>

<h3>Description</h3>

<p><code>umx_rot</code> rotates the items of a vector (1 place, by default). So: c(1,2,3) -&gt; c(2,3,1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_rot(vec, na.last = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_rot_+3A_vec">vec</code></td>
<td>
<p>vector to rotate</p>
</td></tr>
<tr><td><code id="umx_rot_+3A_na.last">na.last</code></td>
<td>
<p>Whether to set the last value to NA (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_rot(1:10)
umx_rot(c(3,4,5,6,7))
# [1] 4 5 6 7 3
</code></pre>

<hr>
<h2 id='umx_round'>umx_round</h2><span id='topic+umx_round'></span>

<h3>Description</h3>

<p>A version of round() which works on dataframes that contain non-numeric data (or data that cannot be coerced to numeric)
Helpful for dealing with table output that mixes numeric and string types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_round(df, digits = getOption("digits"), coerce = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_round_+3A_df">df</code></td>
<td>
<p>a dataframe to round in</p>
</td></tr>
<tr><td><code id="umx_round_+3A_digits">digits</code></td>
<td>
<p>how many digits to round to (defaults to getOption(&quot;digits&quot;))</p>
</td></tr>
<tr><td><code id="umx_round_+3A_coerce">coerce</code></td>
<td>
<p>whether to make the column numeric if it is not (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(umx_round(mtcars, coerce = FALSE))
head(umx_round(mtcars, coerce = TRUE))

</code></pre>

<hr>
<h2 id='umx_scale'>Scale data columns, skipping non-scalable columns</h2><span id='topic+umx_scale'></span>

<h3>Description</h3>

<p><code>umx_scale</code> applies scale to the columns of a data.frame. By default it scales all numeric columns,
and is smart enough to skip non-scalable columns (strings, factors, etc.).
</p>
<p>You can also select which columns to convert.
This is useful when you want to avoid numeric columns which are actually factors.
</p>
<p><em>note</em>: By default, the attributes which scale adds (&quot;scaled:center&quot; and
&quot;scaled:scale&quot; removed to leave nice numeric columns. Set <code>attr= TRUE</code> to preserve these.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_scale(
  df,
  varsToScale = NULL,
  coerce = FALSE,
  attr = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_scale_+3A_df">df</code></td>
<td>
<p>A dataframe to scale (or a numeric vector)</p>
</td></tr>
<tr><td><code id="umx_scale_+3A_varstoscale">varsToScale</code></td>
<td>
<p>(leave blank to scale all)</p>
</td></tr>
<tr><td><code id="umx_scale_+3A_coerce">coerce</code></td>
<td>
<p>Whether to coerce non-numerics to numeric (Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="umx_scale_+3A_attr">attr</code></td>
<td>
<p>to strip off the attributes scale creates (FALSE by default)</p>
</td></tr>
<tr><td><code id="umx_scale_+3A_verbose">verbose</code></td>
<td>
<p>Whether to report which columns were scaled (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> new dataframe with scaled variables
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>umx_scale_wide_twin_data scale
</p>
<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
# note: this example is here to remind us why scaling independently for each 
# twin would be very bad! Use  umx_scale_wide_twin_data() instead!
df = umx_scale(twinData, varsToScale = c("wt1", "wt2"))
df = umx_scale(twinData,  attr= TRUE)
plot(wt1 ~ wt2, data = df)
</code></pre>

<hr>
<h2 id='umx_scale_wide_twin_data'>Scale wide twin data</h2><span id='topic+umx_scale_wide_twin_data'></span>

<h3>Description</h3>

<p>Scale wide data across all twins. You offer up a list of variables to scale, e.g. c(&quot;DEP&quot;, &quot;bmi&quot;)
and the separator (e.g. <code>sep = "_T"</code>) and twin suffixes e.g. 1:2 that paste together to make
complete variable names: e.g. &quot;DEP_T1&quot; and &quot;DEP_T2&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_scale_wide_twin_data(varsToScale, sep, data, twins = 1:2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_scale_wide_twin_data_+3A_varstoscale">varsToScale</code></td>
<td>
<p>The base names of the variables (&quot;weight&quot; etc.)</p>
</td></tr>
<tr><td><code id="umx_scale_wide_twin_data_+3A_sep">sep</code></td>
<td>
<p>The suffix that distinguishes each case, e.g. &quot;_T&quot;)</p>
</td></tr>
<tr><td><code id="umx_scale_wide_twin_data_+3A_data">data</code></td>
<td>
<p>A wide dataframe</p>
</td></tr>
<tr><td><code id="umx_scale_wide_twin_data_+3A_twins">twins</code></td>
<td>
<p>Legal digits following sep (default 1:2)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> dataframe with varsToScale standardized
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>umx_scale
</p>
<p>Other Twin Data functions: 
<code><a href="#topic+umx_long2wide">umx_long2wide</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_twin_data_nice">umx_make_twin_data_nice</a>()</code>,
<code><a href="#topic+umx_residualize">umx_residualize</a>()</code>,
<code><a href="#topic+umx_wide2long">umx_wide2long</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData) 
df = umx_scale_wide_twin_data(data = twinData, varsToScale = c("ht", "wt"), sep = "")
plot(wt1 ~ wt2, data = df)
</code></pre>

<hr>
<h2 id='umx_score_scale'>Score a psychometric scale by summing normal and reversed items.</h2><span id='topic+umx_score_scale'></span>

<h3>Description</h3>

<p>Use this function to generate scores as the appropriate sum of responses to the normal and reversed items in a scale.
</p>
<p>Items must be named on the pattern <code>basename + N + suffix</code>, where <code>base</code> is the prefix common to all item (column) names, N is item number in the scale, and suffix an optional trail (like &quot;_T1&quot;).
</p>
<p><code>pos</code> and <code>rev</code> are vectors of the item numbers for the normal and reverse-scored item numbers.
</p>
<p>To reverse items, the function uses <code>max</code> and <code>min</code> as the lowest and highest possible response scores to compute how to reverse items.
</p>
<p><em>note</em>: <code>min</code> defaults to 1.
<strong>TIP</strong>: If you have strings, <code>umx_score_scale</code> will work (use <code style="white-space: pre;">&#8288;mapStrings = &#8288;</code>). <em>BUT</em> if you want to make a numeric copy, use <code>umx_strings2numeric</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_score_scale(
  base = NULL,
  pos = NULL,
  rev = NULL,
  min = 1,
  max = NULL,
  data = NULL,
  score = c("total", "proportionCorrect", "errors", "mean", "max", "factor"),
  name = NULL,
  na.rm = TRUE,
  minManifests = NA,
  alpha = FALSE,
  mapStrings = NULL,
  correctAnswer = NULL,
  omegaNfactors = 1,
  digits = 2,
  verbose = FALSE,
  suffix = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_score_scale_+3A_base">base</code></td>
<td>
<p>String common to all item names.</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_pos">pos</code></td>
<td>
<p>The positive-scored item numbers.</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_rev">rev</code></td>
<td>
<p>The reverse-scored item numbers.</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_min">min</code></td>
<td>
<p>Minimum legal response value (default = 1). Not implemented for values other than 1 so far...</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_max">max</code></td>
<td>
<p>Maximum legal response value (also used to compute reversed item values).</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_data">data</code></td>
<td>
<p>The data frame</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_score">score</code></td>
<td>
<p>Score total (default), proportionCorrect, errors, mean, max, or factor scores</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_name">name</code></td>
<td>
<p>The name of the scale to be returned. Defaults to &quot;<code>base</code>_score&quot;</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_na.rm">na.rm</code></td>
<td>
<p>Whether to delete NAs when computing scores (Default = TRUE) Note: Choice affects mean!</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_minmanifests">minManifests</code></td>
<td>
<p>How many missing items to tolerate for an individual (when score = factor)</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_alpha">alpha</code></td>
<td>
<p>print Reliability (omega and Cronbach's alpha) (TRUE)</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_mapstrings">mapStrings</code></td>
<td>
<p>Recoding input like &quot;No&quot;/&quot;Maybe&quot;/&quot;Yes&quot; into numeric values (0,1,2)</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_correctanswer">correctAnswer</code></td>
<td>
<p>Use when scoring items with one correct response (1/0).</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_omeganfactors">omegaNfactors</code></td>
<td>
<p>Number of factors for the omega reliability (default = 1)</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_digits">digits</code></td>
<td>
<p>Rounding for omega etc. (default 2)</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the whole omega output (FALSE)</p>
</td></tr>
<tr><td><code id="umx_score_scale_+3A_suffix">suffix</code></td>
<td>
<p>(if dealing with, e.g. &quot;_T1&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the presence of NAs, <code>score= "mean"</code> and <code>score = "totals"</code> both return NA unless na.rm = TRUE.
<code>score = "max"</code>, ignores NAs no matter what.
</p>


<h3>Value</h3>


<ul>
<li><p> scores
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Revelle, W. (2022) psych: Procedures for Personality and Psychological Research, Northwestern University, Evanston, Illinois, USA, <a href="https://CRAN.R-project.org/package=psych">https://CRAN.R-project.org/package=psych</a> Version = 2.2.9.
</p>
</li></ul>


<ul>
<li><p> McNeish, D. (2018). Thanks coefficient alpha, we’ll take it from here. <em>Psychological Methods</em>, <strong>23</strong>, 412-433. <a href="https://doi.org/10.1037/met0000144">doi:10.1037/met0000144</a>.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(psych)
library(psychTools)
data(bfi)

# ==============================
# = Score Agreeableness totals =
# ==============================

# Handscore subject 1
# A1(R)+A2+A3+A4+A5 = (6+1)-2 +4+3+4+4 = 20

tmp = umx_score_scale(base = "A", pos = 2:5, rev = 1, max = 6, data= bfi, name = "A")
tmp[1, namez(tmp, "A",ignore.case = FALSE)]
#  A1 A2 A3 A4 A5  A
#  2  4  3  4  4  20

# ====================
# = Request the mean =
# ====================
tmp = umx_score_scale(name = "A", base = "A", 
   pos = 2:5, rev = 1, max = 6, data= bfi, score="mean")
tmp$A[1] # = 4

# ========================
# = Request factor score =
# ========================
## Not run: 
tmp = umx_score_scale(name = "A", base = "A", pos = 2:5, rev = 1,
   max = 6, score = "factor", minManifests = 4, data= bfi)
#            g
# A2 0.6574826
# A3 0.7581274
# A4 0.4814788
# A5 0.6272332
# A1 0.3736021

# ==================
# = Request alpha  =
# ==================

tmp=umx_score_scale(base="A", pos=2:5, rev=1, max=6, data=bfi, alpha=TRUE)
# omega t = 0.72

## End(Not run)

# ==================
# = na.rm = TRUE ! =
# ==================
tmpDF = bfi
tmpDF[1, "A1"] = NA
tmp = umx_score_scale("A", pos = 2:5, rev = 1, max = 6, data= tmpDF, score="mean")
tmp$A_score[1] # 3.75

tmp= umx_score_scale("A", pos= 2:5, rev= 1, max = 6, data = tmpDF,
   score="mean", na.rm=FALSE)
tmp$A_score[1] # NA (reject cases with missing items)

# ===============
# = Score = max =
# ===============
tmp = umx_score_scale("A", pos = 2:5, rev = 1, max = 6,
  data = bfi, name = "A", score = "max")
tmp$A[1] # Subject 1 max = 5 (reversed) item 1

# Default scale name
tmp = umx_score_scale("E", pos = 3:5, rev = 1:2, max = 6, 
   data= tmp, score = "mean", na.rm = FALSE)
tmp$E_score[1]

# Using @BillRevelle's psych package: More diagnostics, including alpha
scores= psych::scoreItems(items = bfi, min = 1, max = 6, keys = list(
	E = c("-E1","-E2", "E3", "E4", "E5"),
	A = c("-A1", "A2", "A3", "A4", "A5")
))
summary(scores)
scores$scores[1, ]
#  E   A 
# 3.8 4.0 

# Compare output
# (note, by default psych::scoreItems replaces NAs with the sample median...)
RevelleE = as.numeric(scores$scores[,"E"])
RevelleE == tmp[,"E_score"]

# =======================
# = MapStrings examples =
# =======================
mapStrings = c(
   "Very Inaccurate", "Moderately Inaccurate", 
   "Slightly Inaccurate", "Slightly Accurate",
   "Moderately Accurate", "Very Accurate")
bfi$As1 = factor(bfi$A1, levels = 1:6, labels = mapStrings)
bfi$As2 = factor(bfi$A2, levels = 1:6, labels = mapStrings)
bfi$As3 = factor(bfi$A3, levels = 1:6, labels = mapStrings)
bfi$As4 = factor(bfi$A4, levels = 1:6, labels = mapStrings)
bfi$As5 = factor(bfi$A5, levels = 1:6, labels = mapStrings)
bfi= umx_score_scale(name="A" , base="A", pos=2:5, rev=1, max=6, data=bfi)
bfi= umx_score_scale(name="As", base="As", pos=2:5, rev=1, mapStrings = mapStrings, data= bfi)
</code></pre>

<hr>
<h2 id='umx_select_valid'>Update NA values in one column with valid entries from another</h2><span id='topic+umx_select_valid'></span>

<h3>Description</h3>

<p>Merge valid entries from two columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_select_valid(col1, col2, bothways = FALSE, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_select_valid_+3A_col1">col1</code></td>
<td>
<p>name of the first column</p>
</td></tr>
<tr><td><code id="umx_select_valid_+3A_col2">col2</code></td>
<td>
<p>name of the second column</p>
</td></tr>
<tr><td><code id="umx_select_valid_+3A_bothways">bothways</code></td>
<td>
<p>Whether to replace from 1 to 2 as well as from 2 to 1</p>
</td></tr>
<tr><td><code id="umx_select_valid_+3A_data">data</code></td>
<td>
<p>The dataframe containing the two columns.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Updated dataframe
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="base.html#topic+within">within()</a></code>
</p>
</li></ul>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = mtcars
tmp$newDisp = tmp$disp
tmp$disp[c(1,3,6)] = NA
anyNA(tmp$disp) # column has NAs
tmp = umx_select_valid("disp", "newDisp", data = tmp)
anyNA(tmp$disp) # column repaired
</code></pre>

<hr>
<h2 id='umx_set_auto_plot'>umx_set_auto_plot</h2><span id='topic+umx_set_auto_plot'></span>

<h3>Description</h3>

<p>Set autoPlot default for models like umxACE umxGxE etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_auto_plot(autoPlot = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_auto_plot_+3A_autoplot">autoPlot</code></td>
<td>
<p>If TRUE, sets the umx_auto_plot option. Else returns the current value of umx_auto_plot</p>
</td></tr>
<tr><td><code id="umx_set_auto_plot_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Current umx_auto_plot setting
</p>
</li></ul>


<ul>
<li><p> existing value
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_auto_plot() # print current state
old = umx_set_auto_plot(silent = TRUE) # store existing value
old
umx_set_auto_plot(TRUE)   # set to on (internally stored as "name")
umx_set_auto_plot(FALSE)  # set to off (internally stored as NA)
umx_set_auto_plot(old)    # reinstate
</code></pre>

<hr>
<h2 id='umx_set_auto_run'>Automatically run models?</h2><span id='topic+umx_set_auto_run'></span>

<h3>Description</h3>

<p>Set <code>autoRun</code> default for models like <code><a href="#topic+umxRAM">umxRAM()</a></code>, <code><a href="#topic+umxACE">umxACE()</a></code> etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_auto_run(autoRun = NA, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_auto_run_+3A_autorun">autoRun</code></td>
<td>
<p>If TRUE or FALSE, sets the umx_auto_run option. Else returns the current value of umx_auto_run</p>
</td></tr>
<tr><td><code id="umx_set_auto_run_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Current umx_auto_run setting
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_auto_run() # print existing value
old = umx_set_auto_run(silent = TRUE) # store existing value
umx_set_auto_run(FALSE)  # set to FALSE
umx_set_auto_run(old)    # reinstate
</code></pre>

<hr>
<h2 id='umx_set_checkpoint'>umx_set_checkpoint</h2><span id='topic+umx_set_checkpoint'></span><span id='topic+umx_checkpoint'></span>

<h3>Description</h3>

<p>Set the checkpoint status for a model or global options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_checkpoint(
  interval = 1,
  units = c("evaluations", "iterations", "minutes"),
  prefix = "",
  directory = getwd(),
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_checkpoint_+3A_interval">interval</code></td>
<td>
<p>How many units between checkpoints: Default =  1.
A value of zero sets always to 'No' (i.e., do not checkpoint all models during optimization)</p>
</td></tr>
<tr><td><code id="umx_set_checkpoint_+3A_units">units</code></td>
<td>
<p>units to count in: Default unit is 'evaluations' ('minutes' is also legal)</p>
</td></tr>
<tr><td><code id="umx_set_checkpoint_+3A_prefix">prefix</code></td>
<td>
<p>string prefix to add to all checkpoint filenames (default = &quot;&quot;)</p>
</td></tr>
<tr><td><code id="umx_set_checkpoint_+3A_directory">directory</code></td>
<td>
<p>a directory, i.e &quot;~/Desktop&quot; (defaults to getwd())</p>
</td></tr>
<tr><td><code id="umx_set_checkpoint_+3A_model">model</code></td>
<td>
<p>(optional) model to set options in (default = NULL)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> mxModel if provided
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
umx_set_checkpoint(interval = 1, "evaluations", dir = "~/Desktop/")
# Turn off checkpointing with interval = 0
umx_set_checkpoint(interval = 0)
umx_set_checkpoint(2, "evaluations", prefix="SNP_1")
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
m1 = umx_set_checkpoint(model = m1)
m1 = mxRun(m1)
umx_checkpoint(0)

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_set_condensed_slots'>umx_set_condensed_slots</h2><span id='topic+umx_set_condensed_slots'></span>

<h3>Description</h3>

<p>Sets whether newly-created mxMatrices are to be condensed (set to NULL if not being used) or not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_condensed_slots(state = NA, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_condensed_slots_+3A_state">state</code></td>
<td>
<p>what state (TRUE or FALSE) to set condensed slots (default NA returns current value).</p>
</td></tr>
<tr><td><code id="umx_set_condensed_slots_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> current value of condensed slots
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_condensed_slots() # print
old = umx_set_condensed_slots(silent = TRUE) # store the existing state
umx_set_condensed_slots(TRUE) # update globally
umx_set_condensed_slots(old) # set back
</code></pre>

<hr>
<h2 id='umx_set_cores'>umx_set_cores</h2><span id='topic+umx_set_cores'></span>

<h3>Description</h3>

<p>set the number of cores (threads) used by OpenMx
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_cores(cores = NA, model = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_cores_+3A_cores">cores</code></td>
<td>
<p>number of cores to use. NA (the default) returns current value. &quot;-1&quot; will set to <code>omxDetectCores()</code>.</p>
</td></tr>
<tr><td><code id="umx_set_cores_+3A_model">model</code></td>
<td>
<p>an (optional) model to set. If left NULL, the global option is updated.</p>
</td></tr>
<tr><td><code id="umx_set_cores_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> number of cores
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
manifests = c("mpg", "disp", "gear")
m1 = mxModel("ind", type = "RAM",
	manifestVars = manifests,
	mxPath(from = manifests, arrows = 2),
	mxPath(from = "one", to = manifests),
	mxData(mtcars[, manifests], type = "raw")
)
umx_set_cores() # print current value
oldCores = umx_set_cores(silent = TRUE)  # store existing value
umx_set_cores(omxDetectCores()) # set to max
umx_set_cores(-1); umx_set_cores() # set to max
m1 = umx_set_cores(1, m1)  # set m1 usage to 1 core
umx_set_cores(model = m1)  # show new value for m1
umx_set_cores(oldCores)    # reinstate old global value
</code></pre>

<hr>
<h2 id='umx_set_data_variance_check'>umx_set_data_variance_check</h2><span id='topic+umx_set_data_variance_check'></span>

<h3>Description</h3>

<p>Set default for data checking in models like umxACE umxGxE etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_data_variance_check(minVar = NULL, maxVarRatio = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_data_variance_check_+3A_minvar">minVar</code></td>
<td>
<p>Set the threshold at which to warn user about variables with too-small variance. Else returns the current value of umx_minVar</p>
</td></tr>
<tr><td><code id="umx_set_data_variance_check_+3A_maxvarratio">maxVarRatio</code></td>
<td>
<p>Set the option for threshold at which to warn user variances differ too much. Else returns the current value of umx_maxVarRatio</p>
</td></tr>
<tr><td><code id="umx_set_data_variance_check_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- list of umx_minVar and umx_maxVarRatio settings
</p>


<h3>See Also</h3>

<p>xmu_check_variance which uses these to check sanity in the variances of a data frame.
</p>
<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_data_variance_check() # print current state
old = umx_set_data_variance_check(silent = TRUE) # store existing value
umx_set_data_variance_check(minVar = .01)
umx_set_data_variance_check(maxVarRatio = 500)
umx_set_data_variance_check(minVar = old$minVar, maxVarRatio = old$maxVarRatio) # reinstate
</code></pre>

<hr>
<h2 id='umx_set_dollar_symbol'>Set the symbol for money</h2><span id='topic+umx_set_dollar_symbol'></span>

<h3>Description</h3>

<p>Set umx_set_dollar_symbol (used in e.g. [fin_interest()]
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_dollar_symbol(umx.dollar.symbol = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_dollar_symbol_+3A_umx.dollar.symbol">umx.dollar.symbol</code></td>
<td>
<p>symbol for money calculations.</p>
</td></tr>
<tr><td><code id="umx_set_dollar_symbol_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- Current umx.dollar.symbol
</p>


<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_dollar_symbol() # show current state
old = umx_set_dollar_symbol(silent=TRUE) # store existing value
fin_interest(100)
umx_set_dollar_symbol(old)    # reinstate
</code></pre>

<hr>
<h2 id='umx_set_optimization_options'>Set options that affect optimization in OpenMx</h2><span id='topic+umx_set_optimization_options'></span>

<h3>Description</h3>

<p><code>umx_set_optimization_options</code> provides access to get and set options affecting optimization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_optimization_options(
  opt = c("mvnRelEps", "mvnMaxPointsA", "Parallel diagnostics"),
  value = NULL,
  model = NULL,
  silent = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_optimization_options_+3A_opt">opt</code></td>
<td>
<p>default returns current values of the options listed. Currently
&quot;mvnRelEps&quot;, &quot;mvnMaxPointsA&quot;, and &quot;Parallel diagnostics&quot;.</p>
</td></tr>
<tr><td><code id="umx_set_optimization_options_+3A_value">value</code></td>
<td>
<p>If not NULL, the value to set the opt to (can be a list of length(opt))</p>
</td></tr>
<tr><td><code id="umx_set_optimization_options_+3A_model">model</code></td>
<td>
<p>A model for which to set the optimizer. Default (NULL) sets the optimizer globally.</p>
</td></tr>
<tr><td><code id="umx_set_optimization_options_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>note</em>: For <code>mvnRelEps</code>, values between .0001 to .01 are conventional. Smaller values slow optimization.
</p>


<h3>Value</h3>


<ul>
<li><p> current values if no value set.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># show current value for selected or all options
umx_set_optimization_options() # print the existing state(s)
umx_set_optimization_options("mvnRelEps") 
## Not run: 
umx_set_optimization_options("mvnRelEps", .01) # update globally
umx_set_optimization_options("Parallel diagnostics", value = "Yes") 

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_set_optimizer'>Set the optimizer in OpenMx</h2><span id='topic+umx_set_optimizer'></span>

<h3>Description</h3>

<p><code>umx_set_optimizer</code> provides an easy way to get and set the default optimizer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_optimizer(opt = NA, model = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_optimizer_+3A_opt">opt</code></td>
<td>
<p>default (NA) returns current value. Current alternatives are
&quot;NPSOL&quot; &quot;SLSQP&quot; and &quot;CSOLNP&quot;.</p>
</td></tr>
<tr><td><code id="umx_set_optimizer_+3A_model">model</code></td>
<td>
<p>A model for which to set the optimizer. Default (NULL) sets the optimizer globally.</p>
</td></tr>
<tr><td><code id="umx_set_optimizer_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> current optimizer if nothing requested to be set.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_optimizer() # print the existing state
old = umx_set_optimizer(silent = TRUE) # store the existing state
umx_set_optimizer("SLSQP") # update globally
umx_set_optimizer(old) # set back
</code></pre>

<hr>
<h2 id='umx_set_plot_file_suffix'>Set output suffix used in umx SEM diagram files saved to disk.</h2><span id='topic+umx_set_plot_file_suffix'></span>

<h3>Description</h3>

<p><code>umx</code> SEM diagram files can have a suffix of &quot;gv&quot; (default) or  &quot;dot&quot;.
Interrogate the setting by calling with no value: it will return the current setting.
To change the setting call with &quot;gv&quot; or &quot;dot&quot;. Or use TRUE to toggle the setting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_plot_file_suffix(umx.plot.suffix = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_plot_file_suffix_+3A_umx.plot.suffix">umx.plot.suffix</code></td>
<td>
<p>The suffix for plot files (if empty current value is returned). &quot;TRUE&quot;, toggles setting.</p>
</td></tr>
<tr><td><code id="umx_set_plot_file_suffix_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Current setting
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_set_plot_file_suffix() # print current state
old = umx_set_plot_file_suffix(silent = TRUE) # store current value
umx_set_plot_file_suffix("dot")
umx_set_plot_file_suffix("gv")
umx_set_plot_file_suffix(old) # reinstate
</code></pre>

<hr>
<h2 id='umx_set_plot_format'>Set output format of plots (structural diagrams) in umx</h2><span id='topic+umx_set_plot_format'></span>

<h3>Description</h3>

<p>Set output format of plots (default = &quot;DiagrammeR&quot;, alternative is &quot;graphviz&quot;). If you call this with no
value, it will return the current setting. If you call it with TRUE, it toggles the setting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_plot_format(umx.plot.format = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_plot_format_+3A_umx.plot.format">umx.plot.format</code></td>
<td>
<p>format for plots (if empty, returns the current value of umx.plot.format). If &quot;TRUE&quot;, then toggles</p>
</td></tr>
<tr><td><code id="umx_set_plot_format_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Current umx.plot.format setting
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_plot_format() # print current state
old = umx_set_plot_format(silent = TRUE) # store current value
umx_set_plot_format("graphviz")
umx_set_plot_format("DiagrammeR")
umx_set_plot_format("png")
umx_set_plot_format("pdf")
umx_set_plot_format(old) # reinstate
</code></pre>

<hr>
<h2 id='umx_set_plot_use_hrbrthemes'>Set theme system to use for plots.</h2><span id='topic+umx_set_plot_use_hrbrthemes'></span>

<h3>Description</h3>

<p>Set output file suffix (default = &quot;gv&quot;, alternative is &quot;dot&quot;). If you call this with no
value, it will return the current setting. If you call it with TRUE, it toggles the setting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_plot_use_hrbrthemes(umx.plot.use_hrbrthemes = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_plot_use_hrbrthemes_+3A_umx.plot.use_hrbrthemes">umx.plot.use_hrbrthemes</code></td>
<td>
<p>whether to them plots with hrbrthemes (if empty returns the current value)</p>
</td></tr>
<tr><td><code id="umx_set_plot_use_hrbrthemes_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Current setting
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_set_plot_use_hrbrthemes() # print current state
old = umx_set_plot_use_hrbrthemes(silent = TRUE) # store current value
umx_set_plot_use_hrbrthemes(TRUE)
umx_set_plot_use_hrbrthemes(old) # reinstate
</code></pre>

<hr>
<h2 id='umx_set_separator'>Set the separator</h2><span id='topic+umx_set_separator'></span>

<h3>Description</h3>

<p>Set umx_default_separator (used in CI\[low sep high\] ). Default = &quot;,&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_separator(umx_default_separator = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_separator_+3A_umx_default_separator">umx_default_separator</code></td>
<td>
<p>separator for CIs etc. (if empty, returns the current value)</p>
</td></tr>
<tr><td><code id="umx_set_separator_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- Current umx_default_separator
</p>


<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_separator() # show current state
old = umx_set_separator(silent=TRUE) # store existing value
umx_set_separator("|")
umxAPA(.3, .2)
umx_set_separator(old)    # reinstate
</code></pre>

<hr>
<h2 id='umx_set_silent'>Turn off most console and summary output from umx</h2><span id='topic+umx_set_silent'></span>

<h3>Description</h3>

<p>Running multiple analyses or simulations, it can be handy to turn off the automatic summary,
graphing, and printing that umx does to help interactive sessions. <code>umx_set_silent</code> does this.
Summary and graph output, as well as progress and durable console output will be suppressed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_silent(value = NA, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_silent_+3A_value">value</code></td>
<td>
<p>Boolean stating if umx Models should run silently (TRUE).</p>
</td></tr>
<tr><td><code id="umx_set_silent_+3A_silent">silent</code></td>
<td>
<p>If TRUE, this function itself will just return the state of the option, with no user message.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Not every function knows about silent, but most, like <code><a href="#topic+umxRAM">umxRAM()</a></code> etc do.
</p>
<p>Under the hood, <code>umx_set_silent</code> sets options(&quot;umx_silent&quot;). This can be set to either <code>TRUE</code> or <code>FALSE</code>.
If TRUE, then the progress messages from model runs are suppressed. Useful for power simulations etc.
</p>


<h3>Value</h3>


<ul>
<li><p> Current silent value
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_table_format">umx_set_table_format</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
old = umx_set_silent() # print &amp; store existing value
umx_set_silent(FALSE, silent = TRUE) # set to FALSE
umx_set_silent(old)   # reinstate
umx_set_silent() # print existing value
</code></pre>

<hr>
<h2 id='umx_set_table_format'>umx_set_table_format</h2><span id='topic+umx_set_table_format'></span>

<h3>Description</h3>

<p>Set knitr.table.format default (output style for tables). Legal values are
&quot;latex&quot;, &quot;html&quot;, &quot;markdown&quot;, &quot;pandoc&quot;, and &quot;rst&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_set_table_format(knitr.table.format = NULL, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_set_table_format_+3A_knitr.table.format">knitr.table.format</code></td>
<td>
<p>format for tables (if empty, returns the current value of knitr.table.format)</p>
</td></tr>
<tr><td><code id="umx_set_table_format_+3A_silent">silent</code></td>
<td>
<p>If TRUE, no message will be printed.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Current knitr.table.format setting
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Get and set: 
<code><a href="#topic+umx_get_checkpoint">umx_get_checkpoint</a>()</code>,
<code><a href="#topic+umx_get_options">umx_get_options</a>()</code>,
<code><a href="#topic+umx_set_auto_plot">umx_set_auto_plot</a>()</code>,
<code><a href="#topic+umx_set_auto_run">umx_set_auto_run</a>()</code>,
<code><a href="#topic+umx_set_checkpoint">umx_set_checkpoint</a>()</code>,
<code><a href="#topic+umx_set_condensed_slots">umx_set_condensed_slots</a>()</code>,
<code><a href="#topic+umx_set_cores">umx_set_cores</a>()</code>,
<code><a href="#topic+umx_set_data_variance_check">umx_set_data_variance_check</a>()</code>,
<code><a href="#topic+umx_set_dollar_symbol">umx_set_dollar_symbol</a>()</code>,
<code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options</a>()</code>,
<code><a href="#topic+umx_set_optimizer">umx_set_optimizer</a>()</code>,
<code><a href="#topic+umx_set_plot_file_suffix">umx_set_plot_file_suffix</a>()</code>,
<code><a href="#topic+umx_set_plot_format">umx_set_plot_format</a>()</code>,
<code><a href="#topic+umx_set_plot_use_hrbrthemes">umx_set_plot_use_hrbrthemes</a>()</code>,
<code><a href="#topic+umx_set_separator">umx_set_separator</a>()</code>,
<code><a href="#topic+umx_set_silent">umx_set_silent</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
umx_set_table_format() # show current state
old = umx_set_table_format() # store existing value
umx_set_table_format("latex")
umx_set_table_format("html")
umx_set_table_format("markdown")
umx_set_table_format("") # get available options
umx_set_table_format(old)    # reinstate
</code></pre>

<hr>
<h2 id='umx_stack'>Stack data like stack() does, with more control.</h2><span id='topic+umx_stack'></span>

<h3>Description</h3>

<p>Operates like <code><a href="utils.html#topic+stack">stack()</a></code>, but can preserve (&quot;passalong&quot;) other variables on each row,
and allows the user control over the values and group column names for ease of use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_stack(x, select, passalong, valuesName = "values", groupName = "ind")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_stack_+3A_x">x</code></td>
<td>
<p>a dataframe containing twin data.</p>
</td></tr>
<tr><td><code id="umx_stack_+3A_select">select</code></td>
<td>
<p>The variables to stack (wide 2 long)</p>
</td></tr>
<tr><td><code id="umx_stack_+3A_passalong">passalong</code></td>
<td>
<p>Variables to preserve on each row (e.g. age)</p>
</td></tr>
<tr><td><code id="umx_stack_+3A_valuesname">valuesName</code></td>
<td>
<p>The name for the new stacked column (default = &quot;values&quot;)</p>
</td></tr>
<tr><td><code id="umx_stack_+3A_groupname">groupName</code></td>
<td>
<p>The name for the column containing the grouping variable (default = &quot;ind&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> long-format dataframe
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umx_wide2long">umx_wide2long()</a></code>
</p>
<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Base-R stack function
df = stack(mtcars, select = c("disp", "hp"), drop=FALSE)

# umx_stack, with additional variables passed along 
df= umx_stack(mtcars, select= c("disp", "hp"), passalong= "mpg")
str(df) # ind is a factor, with levels select
ggplot2::qplot(x = mpg, y= values, color=ind, data = df)
</code></pre>

<hr>
<h2 id='umx_standardize'>Return a standardized version of a Structural Model</h2><span id='topic+umx_standardize'></span>

<h3>Description</h3>

<p>Return the standardized version of a model (such as ACE, CP etc.)
</p>
<p>Versions exist for RAM, ACE, ACEv, ACEcov, IP, CP and GxE models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_standardize(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_standardize_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> whose fit will be reported.</p>
</td></tr>
<tr><td><code id="umx_standardize_+3A_...">...</code></td>
<td>
<p>Other parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>umx_standardize takes umx models, including RAM and twin models, and returns a standardized version.
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='umx_str_chars'>Select desired characters from a string</h2><span id='topic+umx_str_chars'></span>

<h3>Description</h3>

<p><code>umx_str_chars</code> returns desired characters of a string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_str_chars(what, which)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_str_chars_+3A_what">what</code></td>
<td>
<p>A string</p>
</td></tr>
<tr><td><code id="umx_str_chars_+3A_which">which</code></td>
<td>
<p>Chars to select out.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Array of selected characters
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">tutorials</a>, <a href="https://github.com/tbates/umx">github</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_explode">umx_explode()</a></code>
</p>
</li></ul>

<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_str_chars("myFpassUword", c(3,8))
</code></pre>

<hr>
<h2 id='umx_str_from_object'>Return variable name as a string</h2><span id='topic+umx_str_from_object'></span>

<h3>Description</h3>

<p>Utility to return an object's name as a string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_str_from_object(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_str_from_object_+3A_x">x</code></td>
<td>
<p>an object</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> name as string
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_trim">umx_trim</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_str_from_object(mtcars)
# "mtcars"
</code></pre>

<hr>
<h2 id='umx_string_to_algebra'>Convert a string to an OpenMx algebra</h2><span id='topic+umx_string_to_algebra'></span>

<h3>Description</h3>

<p>This is useful use to quickly and easily insert values from R variables into the string (using paste() and rep() etc.),
then parse the string as an mxAlgebra argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_string_to_algebra(algString, name = NA, dimnames = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_string_to_algebra_+3A_algstring">algString</code></td>
<td>
<p>a string to turn into an algebra</p>
</td></tr>
<tr><td><code id="umx_string_to_algebra_+3A_name">name</code></td>
<td>
<p>of the returned algebra</p>
</td></tr>
<tr><td><code id="umx_string_to_algebra_+3A_dimnames">dimnames</code></td>
<td>
<p>of the returned algebra</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A use case is including a matrix exponent (that is A %<em>% A %</em>% A %*% A...) with a variable exponent.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxAlgebra">mxAlgebra()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
alg = umx_string_to_algebra(paste(rep("A", nReps), collapse = " %*% "), name = "test_case")

## End(Not run)
</code></pre>

<hr>
<h2 id='umx_strings2numeric'>A wrapper to map columns of strings to numeric.</h2><span id='topic+umx_strings2numeric'></span>

<h3>Description</h3>

<p>If you give one column name, this is changed to numeric, and returned as a <strong>vector</strong>.
If you give multiple column names, or don't set cols, each is changed to numeric, and the updated <strong>data.frame</strong> is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_strings2numeric(df, cols = NA, mapStrings = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_strings2numeric_+3A_df">df</code></td>
<td>
<p>The df</p>
</td></tr>
<tr><td><code id="umx_strings2numeric_+3A_cols">cols</code></td>
<td>
<p>(optional) list of columns (default = use all)</p>
</td></tr>
<tr><td><code id="umx_strings2numeric_+3A_mapstrings">mapStrings</code></td>
<td>
<p>legal strings which will be mapped in order to numbers.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> df
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = data.frame(x=letters)
umx_strings2numeric(tmp, mapStrings = letters)
umx_strings2numeric(tmp, cols= "x", mapStrings = letters)
</code></pre>

<hr>
<h2 id='umx_time'>umx_time</h2><span id='topic+umx_time'></span>

<h3>Description</h3>

<p>A function to compactly report how long a model took to execute. Comes with some preset styles
User can set the format with C-style string formatting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_time(
  x = NA,
  formatStr = c("simple", "std", "custom %H %M %OS3"),
  tz = "GMT",
  autoRun = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_time_+3A_x">x</code></td>
<td>
<p>A <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> or list of models for which to display elapsed time, or 'start' or 'stop'</p>
</td></tr>
<tr><td><code id="umx_time_+3A_formatstr">formatStr</code></td>
<td>
<p>A format string, defining how to show the time (defaults to human readable)</p>
</td></tr>
<tr><td><code id="umx_time_+3A_tz">tz</code></td>
<td>
<p>time zone in which the model was executed (defaults to &quot;GMT&quot;)</p>
</td></tr>
<tr><td><code id="umx_time_+3A_autorun">autoRun</code></td>
<td>
<p>If TRUE (default), run the model if it appears not to have been.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default time format is &quot;simple&quot;, which gives only the biggest unit used. i.e., &quot;x seconds&quot; for times under 1 minute.
&quot;std&quot; shows time in the format adopted in OpenMx 2.0 e.g. &quot;Wall clock time (HH:MM:SS.hh): 00:00:01.16&quot;
</p>
<p>If a list of models is provided, time deltas will also be reported.
</p>
<p>If instead of a model the key word &quot;start&quot; is given in x, a start time will be recorded. &quot;stop&quot; gives the
time since &quot;start&quot; was called (and clears the timer)
</p>
<p>If a model has not been run, umx_time will run it for you.
</p>


<h3>Value</h3>


<ul>
<li><p> invisible time string
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
umx_time('stop') # alert user stop called when not yet started... 
umx_time('stop')
umx_time('start')
data(demoOneFactor)
latents  = c("G")
manifests = names(demoOneFactor)
myData = mxData(cov(demoOneFactor), type = "cov", numObs=500)
m1 = umxRAM("umx_time_example", data = myData,
	umxPath(from = latents, to = manifests),
	umxPath(var = manifests),
	umxPath(var = latents, fixedAt = 1)
)
umx_time(m1) # report time from mxModel
m2 = umxRun(m1)
umx_time(c(m1, m2)) # print comparison table
umx_time('stop') # report the time since timer last started, and restart
umx_time('stop') # report the time since timer was restarted.

## End(Not run)

</code></pre>

<hr>
<h2 id='umx_trim'>Trim whitespace surrounding a string.</h2><span id='topic+umx_trim'></span>

<h3>Description</h3>

<p>Returns string without leading or trailing whitespace, like the php function.
See also built-in <code><a href="base.html#topic+trimws">base::trimws()</a></code> does the same.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_trim(string, removeThis = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_trim_+3A_string">string</code></td>
<td>
<p>to trim</p>
</td></tr>
<tr><td><code id="umx_trim_+3A_removethis">removeThis</code></td>
<td>
<p>if not NULL then this regular expression is removed wherever found in 'string'</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> string
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="base.html#topic+trimws">base::trimws()</a></code>
</p>
<p>Other String Functions: 
<code><a href="#topic+umx_explode_twin_names">umx_explode_twin_names</a>()</code>,
<code><a href="#topic+umx_explode">umx_explode</a>()</code>,
<code><a href="#topic+umx_grep">umx_grep</a>()</code>,
<code><a href="#topic+umx_names">umx_names</a>()</code>,
<code><a href="#topic+umx_paste_names">umx_paste_names</a>()</code>,
<code><a href="#topic+umx_rot">umx_rot</a>()</code>,
<code><a href="#topic+umx_str_chars">umx_str_chars</a>()</code>,
<code><a href="#topic+umx_str_from_object">umx_str_from_object</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umx_trim(" dog") # "dog"
trimws(" dog ", "l") # added by R in v 3.3.0
umx_trim("dog ") # "dog"
umx_trim("\t dog \n") # "dog"
umx_trim("xlsx dog.xlsx", "\\.?xlsx ?") # "dog"
</code></pre>

<hr>
<h2 id='umx_var'>Get variances from a df that might contain some non-numeric columns</h2><span id='topic+umx_var'></span>

<h3>Description</h3>

<p>Pass in any dataframe and get variances despite some non-numeric columns.
Cells involving these non-numeric columns are set to ordVar (default = 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_var(
  df,
  format = c("full", "diag", "lower"),
  use = c("complete.obs", "pairwise.complete.obs", "everything", "all.obs",
    "na.or.complete"),
  ordVar = 1,
  digits = NULL,
  strict = TRUE,
  allowCorForFactorCovs = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_var_+3A_df">df</code></td>
<td>
<p>A dataframe of raw data from which to get variances.</p>
</td></tr>
<tr><td><code id="umx_var_+3A_format">format</code></td>
<td>
<p>to return: options are c(&quot;full&quot;, &quot;diag&quot;, &quot;lower&quot;). Defaults to full, but this is not implemented yet.</p>
</td></tr>
<tr><td><code id="umx_var_+3A_use">use</code></td>
<td>
<p>Passed to <code><a href="stats.html#topic+cov">cov()</a></code> - defaults to &quot;complete.obs&quot; (see param default for other options).</p>
</td></tr>
<tr><td><code id="umx_var_+3A_ordvar">ordVar</code></td>
<td>
<p>The value to return at any ordinal columns (defaults to 1).</p>
</td></tr>
<tr><td><code id="umx_var_+3A_digits">digits</code></td>
<td>
<p>digits to round output to (Ignored if NULL). Set for easy printing.</p>
</td></tr>
<tr><td><code id="umx_var_+3A_strict">strict</code></td>
<td>
<p>Whether to allow non-ordered factors to be processed (default = FALSE (no)).</p>
</td></tr>
<tr><td><code id="umx_var_+3A_allowcorforfactorcovs">allowCorForFactorCovs</code></td>
<td>
<p>When ordinal data are present, use heterochoric correlations in affected cells, in place of covariances.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp     = mtcars[,1:4]
tmp$cyl = ordered(mtcars$cyl) # ordered factor
tmp$hp  = ordered(mtcars$hp)  # binary factor
umx_var(tmp, format = "diag", ordVar = 1, use = "pair")
tmp2 = tmp[, c(1, 3)]
umx_var(tmp2, format = "diag")
umx_var(tmp2, format = "full")

data(myFADataRaw)
df = myFADataRaw[,c("z1", "z2", "z3")]
df$z1 = mxFactor(df$z1, levels = c(0, 1))
df$z2 = mxFactor(df$z2, levels = c(0, 1))
df$z3 = mxFactor(df$z3, levels = c(0, 1, 2))    
umx_var(df, format = "diag")
umx_var(df, format = "full", allowCorForFactorCovs=TRUE)

# Ordinal/continuous mix
data(twinData)
twinData= umx_scale_wide_twin_data(data=twinData,varsToScale="wt",sep= "")
# Cut BMI column to form ordinal obesity variables
obLevels   = c('normal', 'overweight', 'obese')
cuts       = quantile(twinData[, "bmi1"], probs = c(.5, .8), na.rm = TRUE)
twinData$obese1=cut(twinData$bmi1,breaks=c(-Inf,cuts,Inf),labels=obLevels)
twinData$obese2=cut(twinData$bmi2,breaks=c(-Inf,cuts,Inf),labels=obLevels)
# Make the ordinal variables into mxFactors
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])
varStarts = umx_var(twinData[, c(ordDVs, "wt1", "wt2")], 
		format= "diag", ordVar = 1, use = "pairwise.complete.obs")

</code></pre>

<hr>
<h2 id='umx_wide2long'>Change data family data from wide (2 twins per row) to long format.</h2><span id='topic+umx_wide2long'></span>

<h3>Description</h3>

<p>Just detects the data columns for twin 1, and twin 2, then returns them stacked
on top of each other (rbind) with the non-twin specific columns copied for each as well.
</p>
<p>*Note*, zygosity codings differ among labs. One scheme uses 1 = MZFF, 2 = MZMM, 3 = DZFF, 4 = DZMM, 
5 = DZOS or DZFM, 6 = DZMF, with 9 = unknown, and then 50, 51,... for siblings.
</p>
<p>Typically, OS twins are ordered Female/Male.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_wide2long(data, sep = "_T", verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_wide2long_+3A_data">data</code></td>
<td>
<p>a dataframe containing twin data.</p>
</td></tr>
<tr><td><code id="umx_wide2long_+3A_sep">sep</code></td>
<td>
<p>the string between the var name and twin suffix, i.e., var_T1 = _T</p>
</td></tr>
<tr><td><code id="umx_wide2long_+3A_verbose">verbose</code></td>
<td>
<p>Report the non-twin and twin columns (default = FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- long-format dataframe
</p>


<h3>See Also</h3>

<p>[reshape()], [umx_merge_randomized_columns()], [umx_select_valid()]
</p>
<p>Other Twin Data functions: 
<code><a href="#topic+umx_long2wide">umx_long2wide</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_twin_data_nice">umx_make_twin_data_nice</a>()</code>,
<code><a href="#topic+umx_residualize">umx_residualize</a>()</code>,
<code><a href="#topic+umx_scale_wide_twin_data">umx_scale_wide_twin_data</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>long = umx_wide2long(data = twinData, sep = "")
long = umx_wide2long(data = twinData, sep = "", verbose = TRUE)
str(long)
str(twinData)
</code></pre>

<hr>
<h2 id='umx_write_to_clipboard'>umx_write_to_clipboard</h2><span id='topic+umx_write_to_clipboard'></span>

<h3>Description</h3>

<p>umx_write_to_clipboard writes data to the clipboard
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umx_write_to_clipboard(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx_write_to_clipboard_+3A_x">x</code></td>
<td>
<p>something to paste to the clipboard</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Works on Mac. Let me know if it fails on windows or Unix.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other File Functions: 
<code><a href="#topic+dl_from_dropbox">dl_from_dropbox</a>()</code>,
<code><a href="#topic+umx_file_load_pseudo">umx_file_load_pseudo</a>()</code>,
<code><a href="#topic+umx_make_sql_from_excel">umx_make_sql_from_excel</a>()</code>,
<code><a href="#topic+umx_move_file">umx_move_file</a>()</code>,
<code><a href="#topic+umx_open">umx_open</a>()</code>,
<code><a href="#topic+umx_rename_file">umx_rename_file</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umx_write_to_clipboard("hello")

## End(Not run)
</code></pre>

<hr>
<h2 id='umx-deprecated'>Deprecated. May already stop() code and ask to be updated. May be dropped entirely in future.</h2><span id='topic+umx-deprecated'></span>

<h3>Description</h3>

<p><code>xmuMakeThresholdsMatrices</code> should be replaced with <code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix()</a></code>
</p>
<p><code>umxTryHard</code> is deprecated: use <code><a href="#topic+umxRun">umxRun()</a></code> instead
</p>
<p><code>stringToMxAlgebra</code> is deprecated: please use <code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra()</a></code> instead
</p>
<p><code>genEpi_EvalQuote</code> is deprecated: please use <code><a href="OpenMx.html#topic+mxEvalByName">mxEvalByName()</a></code> instead
</p>
<p><code>umxReportCIs</code> is deprecated: please use <code><a href="#topic+umxCI">umxCI()</a></code> instead
</p>
<p>replace <code>umxReportFit</code> with <code><a href="#topic+umxSummary">umxSummary()</a></code>
</p>
<p>Replace <code>umxGraph_RAM</code> with <code><a href="#topic+plot">plot()</a></code>
</p>
<p>Replace <code>tryHard</code> with <code><a href="OpenMx.html#topic+mxTryHard">mxTryHard()</a></code>
</p>
<p>Replace <code>standardizeRAM</code> with <code><a href="#topic+umx_standardize">umx_standardize()</a></code>
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="umx-deprecated_+3A_...">...</code></td>
<td>
<p>the old function's parameters (now stripped out to avoid telling people how to do it the wrong way :-)</p>
</td></tr>
</table>


<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>


<hr>
<h2 id='umxACE'>Build and run a 2-group Cholesky ACE twin model (univariate or multivariate)</h2><span id='topic+umxACE'></span>

<h3>Description</h3>

<p>Implementing a core task in twin modeling, umxACE models the genetic and environmental
structure of one or more phenotypes (measured variables) using the Cholesky ACE model
(Neale and Cardon, 1996).
</p>
<p>Classical twin modeling uses the genetic and environmental differences
among pairs of mono-zygotic (MZ) and di-zygotic (DZ) twins reared together.
</p>
<p><code>umxACE</code> implements a 2-group model to capture these data and represent the phenotypic variance as a sum of Additive genetic,
unique environmental (E) and, optionally, either common or shared-environment (C) or
non-additive genetic effects (D).
</p>
<p>The following figure shows the ACE model for one variable &quot;x&quot; as a path diagram:
</p>
<p><img src="../help/figures/ACEunivariate.png" width=50% alt="Figure: ACE univariate.png" />

</p>
<p><code>umxACE</code> allows multivariate analyses, and this brings us to the Cholesky part of the model.
</p>
<p>The Cholesky decomposition creates as many latent A (and C and E) latent variables as there are phenotypes, and, moving
from left to right, decomposes the variance in each phenotype into successively restricted
factors. The following figure shows the multivariate ACE model for three variables:
</p>
<p><img src="../help/figures/ACEmatrix.png" width=50% alt="Figure: ACE matrix.png" />

</p>
<p>In this ACE model of three phenotypes, the expected variance-covariance matrix of the original data
is the product of each lower Cholesky and its transform (i.e., <code>A = a %*% t(a)</code> summed for <code>A+C+E</code>.
</p>
<p>This lower-triangle decomposition feature of the Cholesky yields a model which is certain to both
account for all the variance (with some restrictions) in the data and be solvable.
</p>
<p>This figure also contains the key to understanding how to modify models that <code>umxACE</code> produces
using <code>umxModify()</code> to drop paths  by label like <code>"a_r1c1"</code>. <strong>nb</strong>: Read the &quot;Matrices and Labels in ACE model&quot; section in details below...
</p>
<p><strong>NOTE</strong>: Scroll down to details for how to use the function, a figure and multiple examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxACE(
  name = "ACE",
  selDVs,
  selCovs = NULL,
  dzData = NULL,
  mzData = NULL,
  sep = NULL,
  data = NULL,
  zyg = "zygosity",
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  numObsDZ = NULL,
  numObsMZ = NULL,
  boundDiag = 0,
  allContinuousMethod = c("cumulants", "marginals"),
  autoRun = getOption("umx_auto_run"),
  intervals = FALSE,
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL,
  residualizeContinuousVars = FALSE,
  nSib = 2,
  dzAr = 0.5,
  dzCr = 1,
  weightVar = NULL,
  equateMeans = TRUE,
  addStd = TRUE,
  addCI = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxACE_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to&quot;ACE&quot;).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_seldvs">selDVs</code></td>
<td>
<p>The variables to include from the data: preferably, just &quot;dep&quot; not c(&quot;dep_T1&quot;, &quot;dep_T2&quot;).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_selcovs">selCovs</code></td>
<td>
<p>(optional) covariates to include from the data (do not include sep in names)</p>
</td></tr>
<tr><td><code id="umxACE_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_sep">sep</code></td>
<td>
<p>The separator in twin variable names, often &quot;_T&quot;, e.g. &quot;dep_T1&quot;. Simplifies selDVs.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_data">data</code></td>
<td>
<p>If provided, dzData and mzData are treated as levels of zyg to select() MZ and DZ data sets (default = NULL)</p>
</td></tr>
<tr><td><code id="umxACE_+3A_zyg">zyg</code></td>
<td>
<p>If data provided, this column is used to select rows by zygosity (Default = &quot;zygosity&quot;)</p>
</td></tr>
<tr><td><code id="umxACE_+3A_type">type</code></td>
<td>
<p>Analysis method one of c(&quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;)</p>
</td></tr>
<tr><td><code id="umxACE_+3A_numobsdz">numObsDZ</code></td>
<td>
<p>Number of DZ twins: Set this if you input covariance data.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_numobsmz">numObsMZ</code></td>
<td>
<p>Number of MZ twins: Set this if you input covariance data.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_bounddiag">boundDiag</code></td>
<td>
<p>Numeric lbound for diagonal of the a, c, and e matrices. Defaults to 0 since umx version 1.8</p>
</td></tr>
<tr><td><code id="umxACE_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_intervals">intervals</code></td>
<td>
<p>Whether to run mxCI confidence intervals (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxACE_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxACE_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_residualizecontinuousvars">residualizeContinuousVars</code></td>
<td>
<p>Not yet implemented.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_nsib">nSib</code></td>
<td>
<p>Number of siblings in a family (default - 2). &quot;3&quot; = extra sib.</p>
</td></tr>
<tr><td><code id="umxACE_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_weightvar">weightVar</code></td>
<td>
<p>If provided, a vector objective will be used to weight the data. (default = NULL).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_addstd">addStd</code></td>
<td>
<p>Whether to add the algebras to compute a std model (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACE_+3A_addci">addCI</code></td>
<td>
<p>Whether to add intervals to compute CIs (defaults to TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Covariates</strong>
umxACE handles covariates by modelling them in the means.
On the plus side, there is no distributional assumption for this method. A downside of this approach is that all
covariates must be non-NA, thus dropping any rows where one or more covariates are missing.
This can waste data. See also <code><a href="#topic+umx_residualize">umx_residualize()</a></code>).
</p>
<p><strong>Data Input</strong>
The function flexibly accepts raw data, and also summary covariance data
(in which case the user must also supple numbers of observations for the two input data sets).
</p>
<p>The <code>type</code> parameter can select how you want the model data treated.
&quot;FIML&quot; is the normal treatment. &quot;cov&quot; and &quot;cor&quot; will turn raw data into cor data for analysis, or
check that you've provided cor data as input.
</p>
<p>Types &quot;WLS&quot;, &quot;DWLS&quot;, and &quot;ULS&quot; will process raw data into WLS data of these types.
</p>
<p>The default, &quot;Auto&quot; will treat data as the type they are provided as.
</p>
<p><strong>Ordinal Data</strong>
In an important capability, the model transparently handles ordinal (binary or multi-level
ordered factor data) inputs, and can handle mixtures of continuous, binary, and ordinal
data in any combination. An experimental feature is under development to allow Tobit modeling.
</p>
<p>The function also supports weighting of individual data rows. In this case,
the model is estimated for each row individually, then each row likelihood
is multiplied by its weight, and these weighted likelihoods summed to form
the model-likelihood, which is to be minimized.
This feature is used in the non-linear GxE model functions.
</p>
<p><strong>Additional features</strong>
The umxACE function supports varying the DZ genetic association (defaulting to .5)
to allow exploring assortative mating effects, as well as varying the DZ &ldquo;C&rdquo; factor
from 1 (the default for modeling family-level effects shared 100% by twins in a pair),
to .25 to model dominance effects.
</p>
<p><strong>Matrices and Labels in ACE model</strong>
</p>
<p>Matrices 'a', 'c', and 'e' contain the path loadings of the Cholesky ACE factor model.
</p>
<p>So, labels relevant to modifying the model are of the form <code style="white-space: pre;">&#8288;"a_r1c1", "c_r1c1"&#8288;</code> etc.
</p>
<p>Variables are in rows, and factors are in columns. So to drop the influence of factor 2 on variable 3, you would say:
</p>
<p><code>m2 = umxModify(m1, update = "c_r3c2")</code>
</p>
<p>Less commonly-modified matrices are the mean matrix <code>expMean</code>. This has 1 row, and the columns are laid out for
each variable for twin 1, followed by each variable for twin 2.
</p>
<p>So, in a model where the means for twin 1 and twin 2 had been equated (set = to T1), you
could make them independent again with this script:
</p>
<p><code>m1$top$expMean$labels[1, 4:6] = c("expMean_r1c4", "expMean_r1c5", "expMean_r1c6")</code>
</p>
<p><em>note</em>: Only one of C or D may be estimated simultaneously. This restriction reflects the lack
of degrees of freedom to simultaneously model C and D with only MZ and DZ twin pairs (Eaves et al. 1978, p267).
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> of subclass mxModel.ACE
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Eaves, L. J., Last, K. A., Young, P. A., &amp; Martin, N. G. (1978). Model-fitting approaches
to the analysis of human behaviour. <em>Heredity</em>, <strong>41</strong>, 249-320. <a href="https://www.nature.com/articles/hdy1978101.pdf">https://www.nature.com/articles/hdy1978101.pdf</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxPlotACE">umxPlotACE()</a></code>, <code><a href="#topic+umxSummaryACE">umxSummaryACE()</a></code>, <code><a href="#topic+power.ACE.test">power.ACE.test()</a></code>, <code><a href="#topic+umxModify">umxModify()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
require(umx)
# ============================
# = How heritable is height? =
# ============================

# 1. Height in meters has a tiny variance, and this makes optimising hard.
#    We'll scale it by 10x to make the Optimizer's task easier.
data(twinData) # ?twinData from Australian twins.
twinData[, c("ht1", "ht2")] = twinData[, c("ht1", "ht2")] * 10

# 2. Make mz &amp; dz data.frames (no need to select variables: umx will do this)
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]

# 3. Built &amp; run the model, controlling for age in the means model
m1 = umxACE(selDVs = "ht", selCovs = "age", sep = "", dzData = dzData, mzData = mzData)

# sidebar: umxACE figures out variable names using sep: 
#    e.g. selVars = "wt" + sep= "_T" -&gt; "wt_T1" "wt_T2"

umxSummary(m1, std = FALSE) # un-standardized

# tip 1: set report = "html" and umxSummary prints the table to your browser!
# tip 2: plot works for umx: Get a figure of the model and parameters
# plot(m1) # Also, look at the options for ?plot.MxModel.

# ===========================================
# = Test ADE, AE, CE, E, and generate table =
# ===========================================

umxReduce(m1, report="html", silent= TRUE)

# ============================
# = Model, with 2 covariates =
# ============================

# Create another covariate: cohort
twinData$cohort1 = twinData$cohort2 =twinData$part
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]

# 1. def var approach
m2 = umxACE(selDVs = "ht", selCovs = c("age", "cohort"), sep = "", dzData = dzData, mzData = mzData)

# 2. Residualized approach: remove height variance accounted-for by age.
FFdata = twinData[twinData$zygosity %in% c("MZFF", "DZFF"), ]
FFdata = umx_residualize("ht", "age", suffixes = 1:2, data = FFdata)
mzData = FFdata[FFdata$zygosity %in% "MZFF", ]
dzData = FFdata[FFdata$zygosity %in% "DZFF", ]
m3 = umxACE(selDVs = "ht", sep = "", dzData = dzData, mzData = mzData)

# =============================================================
# = ADE: Evidence for dominance ? (DZ correlation set to .25) =
# =============================================================
m2 = umxACE(selDVs = "ht", sep = "", dzData = dzData, mzData = mzData, dzCr = .25)
umxCompare(m2, m1) # ADE is better
umxSummary(m2, comparison = m1) 
# nb: Although summary is smart enough to print d, the underlying 
#     matrices are still called a, c &amp; e.

# tip: try umxReduce(m1) to automatically build and compare ACE, ADE, AE, CE
# including conditional probabilities!

# ===================================================
# = WLS example using diagonal weight least squares =
# ===================================================

m3 = umxACE(selDVs = "ht", sep = "", dzData = dzData, mzData = mzData, 
	type = "DWLS", allContinuousMethod='marginals'
)


# ==============================
# = Univariate model of weight =
# ==============================

# Things to note:

# 1. Weight has a large variance, and this makes solution finding very hard.
# Here, we residualize the data for age, which also scales weight and height.

data(twinData)
tmp = umx_residualize(c("wt", "ht"), cov = "age", suffixes= c(1, 2), data = twinData)
mzData = tmp[tmp$zygosity %in% "MZFF", ]
dzData = tmp[tmp$zygosity %in% "DZFF", ]

# tip: You might also want transform variables
# tmp = twinData$wt1[!is.na(twinData$wt1)]
# car::powerTransform(tmp, family="bcPower"); hist(tmp^-0.6848438)
# twinData$wt1 = twinData$wt1^-0.6848438
# twinData$wt2 = twinData$wt2^-0.6848438

# 4. note: the default boundDiag = 0 lower-bounds a, c, and e at 0.
#    Prevents mirror-solutions. If not desired: set boundDiag = NULL.

m2 = umxACE(selDVs = "wt", dzData = dzData, mzData = mzData, sep = "", boundDiag = NULL)

# A short cut (which is even shorter for "_T" twin data with "MZ"/"DZ" data in zygosity column is:
m1 = umxACE(selDVs = "wt", sep = "", data = twinData,
	dzData = c("DZMM", "DZFF", "DZOS"), mzData = c("MZMM", "MZFF"))
# |   |   a1|c1 |   e1|
# |:--|----:|:--|----:|
# |wt | 0.93|.  | 0.38|

# tip: umx_make_twin_data_nice() will make data into this nice format for you!

# ======================
# = MODEL MODIFICATION =
# ======================
# We can modify this model, e.g. test shared environment. 
# Set comparison to modify, and show effect in one step.

m2 = umxModify(m1, update = "c_r1c1", name = "no_C", comparison = TRUE)
#*tip* call umxModify(m1) with no parameters, and it will print the labels available to fix!
# nb: You can see parameters of any model with parameters(m1)

# =========================================================
# = Well done! Now you can make modify twin models in umx =
# =========================================================

# =====================================
# = Bivariate height and weight model =
# =====================================
data(twinData)
# We'll scale height (ht1 and ht2) and weight
twinData = umx_scale_wide_twin_data(data = twinData, varsToScale = c("ht", "wt"), sep = "")
mzData = twinData[twinData$zygosity %in% c("MZFF", "MZMM"),]
dzData = twinData[twinData$zygosity %in% c("DZFF", "DZMM", "DZOS"), ]
m1 = umxACE(selDVs = c("ht", "wt"), sep = '', dzData = dzData, mzData = mzData)
umxSummary(m1)

# ===================
# = Ordinal example =
# ===================

# Prep data
require(umx)
data(twinData)
# Cut BMI column to form ordinal obesity variables
obLevels = c('normal', 'overweight', 'obese')
cuts = quantile(twinData[, "bmi1"], probs = c(.5, .2), na.rm = TRUE)
twinData$obese1=cut(twinData$bmi1, breaks=c(-Inf,cuts,Inf), labels=obLevels)
twinData$obese2=cut(twinData$bmi2, breaks=c(-Inf,cuts,Inf), labels=obLevels)

# Make the ordinal variables into umxFactors
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])

mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]

# Model and summary!
m1 = umxACE(selDVs = "obese", dzData = dzData, mzData = mzData, sep = '')

# And controlling age (otherwise manifests appearance as latent C)
m1 = umxACE(selDVs = "obese", selCov= "age", dzData = dzData, mzData = mzData, sep = '')
# umxSummary(m1)

# ============================================
# = Bivariate continuous and ordinal example =
# ============================================
data(twinData)
twinData= umx_scale_wide_twin_data(data=twinData,varsToScale="wt",sep= "")
# Cut BMI column to form ordinal obesity variables
obLevels   = c('normal', 'overweight', 'obese')
cuts       = quantile(twinData[, "bmi1"], probs = c(.5, .2), na.rm = TRUE)
twinData$obese1=cut(twinData$bmi1,breaks=c(-Inf,cuts,Inf),labels=obLevels)
twinData$obese2=cut(twinData$bmi2,breaks=c(-Inf,cuts,Inf),labels=obLevels)
# Make the ordinal variables into mxFactors
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])
mzData = twinData[twinData$zygosity %in% "MZFF",] 
dzData = twinData[twinData$zygosity %in% "DZFF",]
mzData = mzData[1:80,] # just top 80 so example runs in a couple of secs
dzData = dzData[1:80,]
m1 = umxACE(selDVs= c("wt","obese"), dzData= dzData, mzData= mzData, sep='')

# And controlling age
m1 = umxACE(selDVs = c("wt","obese"), selCov= "age", dzData = dzData, mzData = mzData, sep = '')

# =======================================
# = Mixed continuous and binary example =
# =======================================
require(umx)
data(twinData)
twinData= umx_scale_wide_twin_data(data= twinData,varsToScale= "wt", sep="")
# Cut to form category of 20% obese subjects
# and make into mxFactors (ensure ordered is TRUE, and require levels)
obLevels   = c('normal', 'obese')
cuts       = quantile(twinData[, "bmi1"], probs = .2, na.rm = TRUE)
twinData$obese1= cut(twinData$bmi1, breaks=c(-Inf,cuts,Inf), labels=obLevels) 
twinData$obese2= cut(twinData$bmi2, breaks=c(-Inf,cuts,Inf), labels=obLevels) 
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])

selDVs = c("wt", "obese")
mzData = twinData[twinData$zygosity %in% "MZFF",]
dzData = twinData[twinData$zygosity %in% "DZFF",]
m1 = umxACE(selDVs = selDVs, dzData = dzData, mzData = mzData, sep = '')
umxSummary(m1)

# ==============
# = Two binary =
# ==============
require(umx)
data(twinData)
htLevels   = c('short', 'tall')
obLevels   = c('normal', 'obese')
cuts       = quantile(twinData[, "bmi1"], probs = .2, na.rm = TRUE)
twinData$obese1= cut(twinData$bmi1, breaks=c(-Inf,cuts,Inf), labels=obLevels) 
twinData$obese2= cut(twinData$bmi2, breaks=c(-Inf,cuts,Inf), labels=obLevels) 
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])

twinData$short1 = cut(twinData$ht1, breaks=c(-Inf,1.6,Inf), labels=htLevels) 
twinData$short2 = cut(twinData$ht2, breaks=c(-Inf,1.6,Inf), labels=htLevels) 
ordDVs = c("short1", "short2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])

mzData = twinData[twinData$zygosity %in% "MZFF",]
dzData = twinData[twinData$zygosity %in% "DZFF",]
m1 = umxACE(selDVs = c("short", "obese"), dzData = dzData, mzData = mzData, sep = '')

# ===================================
# Example with covariance data only =
# ===================================

require(umx)
data(twinData)
twinData= umx_scale_wide_twin_data(data=twinData, varsToScale= "wt", sep="")
selDVs = c("wt1", "wt2")
mz = cov(twinData[twinData$zygosity %in%  "MZFF", selDVs], use = "complete")
dz = cov(twinData[twinData$zygosity %in%  "DZFF", selDVs], use = "complete")
m1 = umxACE(selDVs=selDVs, dzData=dz, mzData=mz, numObsDZ=569, numObsMZ=351)
umxSummary(m1)
plot(m1)


</code></pre>

<hr>
<h2 id='umxACEcov'>Run a Cholesky with covariates that are random (in the expected covariance matrix)</h2><span id='topic+umxACEcov'></span>

<h3>Description</h3>

<p>Often, researchers include covariates in 2-group Cholesky <code><a href="#topic+umxACE">umxACE()</a></code> twin models.
The umxACEcov 'random' option models the covariates in the expected covariance matrix, thus allowing
all data to be preserved. The downside is that this method has a strong assumption
of multivariate normality. Covariates like age, which are perfectly correlated in twins cannot be used.
Covariates like sex, which are ordinal, violate the normality assumption.
Binary and ordinal covariates like sex also violate the normality assumption. Which is most of the use cases :-(.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxACEcov(
  name = "ACEcov",
  selDVs,
  selCovs,
  dzData,
  mzData,
  sep = NULL,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  allContinuousMethod = c("cumulants", "marginals"),
  dzAr = 0.5,
  dzCr = 1,
  addStd = TRUE,
  addCI = TRUE,
  boundDiag = 0,
  equateMeans = TRUE,
  bVector = FALSE,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxACEcov_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to&quot;ACE&quot;).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_seldvs">selDVs</code></td>
<td>
<p>The variables to include from the data (do not include sep).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_selcovs">selCovs</code></td>
<td>
<p>The covariates to include from the data (do not include sep).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe.</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe.</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_sep">sep</code></td>
<td>
<p>Separator text between basename for twin variable names. Often &quot;_T&quot;.
Used to expand selDVs into full column names, i.e., &quot;dep&quot; &ndash;&gt; c(&quot;dep_T1&quot;, &quot;dep_T2&quot;).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_type">type</code></td>
<td>
<p>Analysis method one of c(&quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;)</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_addstd">addStd</code></td>
<td>
<p>Whether to add the algebras to compute a std model (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_addci">addCI</code></td>
<td>
<p>Whether to add intervals to compute CIs (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_bounddiag">boundDiag</code></td>
<td>
<p>= Whether to bound the diagonal of the a, c, and e matrices.</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_bvector">bVector</code></td>
<td>
<p>Whether to compute row-wise likelihoods (defaults to FALSE).</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxACEcov_+3A_optimizer">optimizer</code></td>
<td>
<p>optionally set the optimizer. Default (NULL) does nothing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following figure shows how the ACE model with random covariates appears as a path diagram:
</p>
<p><img src="../help/figures/ACEcovVarianceModel.png" width=50% alt="Figure: ACEcovVarianceModel.png" />

</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> of subclass mxModel.ACEcov
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Neale, M. C., &amp; Martin, N. G. (1989). The effects of age, sex,
and genotype on self-report drunkenness following a challenge dose of alcohol.
<em>Behavior Genetics</em>, <strong>19</strong>, 63-78. <a href="https://doi.org/10.1007/BF01065884">doi:10.1007/BF01065884</a>.
</p>
</li></ul>


<ul>
<li><p> Schwabe, I., Boomsma, D. I., Zeeuw, E. L., &amp; Berg, S. M. (2015). A New Approach
to Handle Missing Covariate Data in Twin Research : With an Application to
Educational Achievement Data. <em>Behavior Genetics</em>, <strong>46</strong>, 583-95. <a href="https://doi.org/10.1007/s10519-015-9771-1">doi:10.1007/s10519-015-9771-1</a>.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ============================================
# = BMI, can't use Age as a random covariate =
# ============================================
require(umx)
data(twinData)
# Replicate age to age1 &amp; age2
twinData$age1 = twinData$age2 = twinData$age
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")

# =====================================================================
# = Trying to use identical var (like age) as a random cov is ILLEGAL =
# =====================================================================
m1 = umxACEcov(selDVs = "bmi", selCovs = "age", dzData = dzData, mzData = mzData, sep = "")

# ========================================================
# = Use an lm-based age-residualisation approach instead =
# ========================================================

resid_data = umx_residualize("bmi", "age", suffixes = 1:2, twinData)
mzData = subset(resid_data, zygosity == "MZFF")
dzData = subset(resid_data, zygosity == "DZFF")
m2     = umxACE("resid", selDVs = "bmi", dzData = dzData, mzData = mzData, sep = "")

# Univariate BMI without covariate of age for comparison
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m3 = umxACE("raw_bmi", selDVs = "bmi", dzData = dzData, mzData = mzData, sep = "")

# ===========================================================================
# = A bivariate example (need a dataset with a VIABLE COVARIATE to do this) =
# ===========================================================================
selDVs  = "wt" # Set the DVs
selCovs = "ht" # Set the COV
selVars = umx_paste_names(selDVs, covNames = selCovs, sep = "", sep = 1:2)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACEcov(selDVs = selDVs, selCovs = selCovs,
   dzData = dzData, mzData = mzData, sep = "", autoRun = TRUE
)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxACEv'>Build and run 2-group uni- or multi-variate ACE models based on VARIANCE (not paths).</h2><span id='topic+umxACEv'></span>

<h3>Description</h3>

<p>A common task in twin modeling involves using the genetic and environmental differences
between large numbers of pairs of mono-zygotic (MZ) and di-zygotic (DZ) twins reared together
to model the genetic and environmental structure of one, or, typically, several phenotypes.
<code>umxACEv</code> directly estimates variance components (rather than paths, which
are then squared to produce variance and therefore cannot be negative). It offers better power,
correct Type I error and un-biased estimates (with no zero-bound for the variances) as a saturated model.
(Verhulst et al, 2019).
</p>
<p>The ACE variance-based model decomposes phenotypic variance into additive genetic (A),
unique environmental (E) and, optionally, either common environment (shared-environment, C) or
non-additive genetic effects (D). Scroll down to details for how to use the function, a figure
and multiple examples.
</p>
<p>The following figure shows the A components of a trivariate ACEv model:
</p>
<p><img src="../help/figures/ACEv.png" width=50% alt="Figure: ACEv.png" />

</p>
<p><em>NOTE</em>: This function does not use the Cholesky decomposition. Instead it directly models variance.
This ensures unbiased type-I error rates. It means that occasionally
estimates of variance may be negative. This should be used as an occasion to inspect you model
choices and data. <code>umxACEv</code> can be used as a base model to validate the ACE Cholesky model,
a core model in behavior genetics (Neale and Cardon, 1992).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxACEv(
  name = "ACEv",
  selDVs,
  selCovs = NULL,
  sep = NULL,
  dzData,
  mzData,
  dzAr = 0.5,
  dzCr = 1,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  allContinuousMethod = c("cumulants", "marginals"),
  data = NULL,
  zyg = "zygosity",
  weightVar = NULL,
  numObsDZ = NULL,
  numObsMZ = NULL,
  addStd = TRUE,
  addCI = TRUE,
  boundDiag = NULL,
  equateMeans = TRUE,
  bVector = FALSE,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL,
  nSib = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxACEv_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to&quot;ACE&quot;).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_seldvs">selDVs</code></td>
<td>
<p>The variables to include from the data: preferably, just &quot;dep&quot; not c(&quot;dep_T1&quot;, &quot;dep_T2&quot;).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_selcovs">selCovs</code></td>
<td>
<p>(optional) covariates to include from the data (do not include sep in names)</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_sep">sep</code></td>
<td>
<p>The separator in twin var names, often &quot;_T&quot; in vars like &quot;dep_T1&quot;. Simplifies selDVs.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_type">type</code></td>
<td>
<p>Analysis method one of c(&quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_data">data</code></td>
<td>
<p>If provided, dzData and mzData are treated as valid levels of zyg to select() data sets (default = NULL)</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_zyg">zyg</code></td>
<td>
<p>If data provided, this column is used to select rows by zygosity (Default = &quot;zygosity&quot;)</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_weightvar">weightVar</code></td>
<td>
<p>= If provided, a vector objective will be used to weight the data. (default = NULL).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_numobsdz">numObsDZ</code></td>
<td>
<p>= Number of DZ twins: Set this if you input covariance data.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_numobsmz">numObsMZ</code></td>
<td>
<p>= Number of MZ twins: Set this if you input covariance data.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_addstd">addStd</code></td>
<td>
<p>Whether to add the algebras to compute a std model (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_addci">addCI</code></td>
<td>
<p>Whether to add intervals to compute CIs (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_bounddiag">boundDiag</code></td>
<td>
<p>= Numeric lbound for diagonal of the a, c, and e matrices. Default = NULL (no bound)</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_bvector">bVector</code></td>
<td>
<p>Whether to compute row-wise likelihoods (defaults to FALSE).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
<tr><td><code id="umxACEv_+3A_nsib">nSib</code></td>
<td>
<p>Number of sibs, default is 2. Working on 3 :-)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Data Input</strong>
The function flexibly accepts raw data, and also summary covariance data
(in which case the user must also supple numbers of observations for the two input data sets).
</p>
<p><strong>Ordinal Data</strong>
In an important capability, the model transparently handles ordinal (binary or multi-level
ordered factor data) inputs, and can handle mixtures of continuous, binary, and ordinal
data in any combination.
</p>
<p>The function also supports weighting of individual data rows. In this case,
the model is estimated for each row individually, then each row likelihood
is multiplied by its weight, and these weighted likelihoods summed to form
the model-likelihood, which is to be minimized.
This feature is used in the non-linear GxE model functions.
</p>
<p><strong>Additional features</strong>
The umxACEv function supports varying the DZ genetic association (defaulting to .5)
to allow exploring assortative mating effects, as well as varying the DZ &ldquo;C&rdquo; factor
from 1 (the default for modeling family-level effects shared 100% by twins in a pair),
to .25 to model dominance effects.
</p>
<p><em>note</em>: Only one of C or D may be estimated simultaneously. This restriction reflects the lack
of degrees of freedom to simultaneously model C and D with only MZ and DZ twin pairs (Eaves et al. 1978 p267).
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> subclass <code>mxModelACEv</code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Verhulst, B., Prom-Wormley, E., Keller, M., Medland, S., &amp; Neale, M. C. (2019).
Type I Error Rates and Parameter Bias in Multivariate Behavioral Genetic Models. <em>Behav Genet</em>,
<strong>49</strong>, 99-111. <a href="https://doi.org/10.1007/s10519-018-9942-y">doi:10.1007/s10519-018-9942-y</a>
Eaves, L. J., Last, K. A., Young, P. A., &amp; Martin, N. G. (1978). Model-fitting approaches
to the analysis of human behaviour. <em>Heredity</em>, <strong>41</strong>, 249-320. <a href="https://www.nature.com/articles/hdy1978101.pdf">https://www.nature.com/articles/hdy1978101.pdf</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# ==============================
# = Univariate model of weight =
# ==============================
require(umx)
data(twinData) # ?twinData from Australian twins.

# Things to note: ACE model of weight will return a NEGATIVE variance in C.
#  This is exactly why we have ACEv! It suggests we need a different model
#  In this case: ADE.
# Other things to note:
# 1. umxACEv can figure out variable names: provide "sep", and selVars. 
#    Function generates: "wt" -&gt; "wt1" "wt2"
# 2. umxACEv picks the variables it needs from the data.

mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m1 = umxACEv(selDVs = "wt", sep = "", dzData = dzData, mzData = mzData)

# A short cut (which is even shorter for "_T" twin data with "MZ"/"DZ" data in zygosity column is:
m1 = umxACEv(selDVs = "wt", sep = "", dzData = "MZFF", mzData = "DZFF", data = twinData)
# ========================================================
# = Evidence for dominance ? (DZ correlation set to .25) =
# ========================================================
m2 = umxACEv("ADE", selDVs = "wt", sep = "", dzData = dzData, mzData = mzData, dzCr = .25)
# note: the underlying matrices are still called A, C, and E.
# I catch this in the summary table, so columns are labeled A, D, and E.
# However, currently, the plot will say A, C, E.

# We can modify this model, dropping dominance component (still called C), 
# and see a comparison:
m3 = umxModify(m2, update = "C_r1c1", comparison = TRUE, name="AE")
# =========================================================
# = Well done! Now you can make modify twin models in umx =
# =========================================================

# ============================
# = How heritable is height? =
# ============================
# 
# Note: Height has a small variance. umx can typically picks good starts,
#    but scaling is advisable.
# 
require(umx)
# Load data and rescale height to cm (var in m too small)
data(twinData) # ?twinData from Australian twins.
twinData[,c("ht1", "ht2")]= twinData[,c("ht1", "ht2")]*100

mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m1 = umxACEv(selDVs = "ht", sep = "", dzData = dzData, mzData = mzData)

umxSummary(m1, std = FALSE) # unstandardized
plot(m1)

# tip: with report = "html", umxSummary can print the table to your browser!
# tip: You can turn off auto-plot with umx_set_auto_plot(FALSE)

# ========================================================
# = Evidence for dominance ? (DZ correlation set to .25) =
# ========================================================
m2 = umxACEv("ADE", selDVs = "ht", dzCr = .25, sep="", dzData = dzData, mzData = mzData)
umxCompare(m2, m1) # Is ADE better?
umxSummary(m2, comparison = m1) # nb: though this is ADE, matrices are still called A,C,E

# We can modify this model, dropping shared environment, and see a comparison:
m3 = umxModify(m2, update = "C_r1c1", comparison = TRUE, name = "AE")

# =====================================
# = Bivariate height and weight model =
# =====================================

data(twinData)
twinData[,c("ht1", "ht2")]= twinData[,c("ht1", "ht2")]*100
mzData = twinData[twinData$zygosity %in% c("MZFF", "MZMM"), ]
dzData = twinData[twinData$zygosity %in% c("DZFF", "DZMM", "DZOS"), ]
m1 = umxACEv(selDVs = c("ht", "wt"), sep = '', dzData = dzData, mzData = mzData)

# ===================
# = Ordinal example =
# ===================
require(umx)
data(twinData)

# Cut bmi column to form ordinal obesity variables
cutPoints = quantile(twinData[, "bmi1"], probs = c(.5, .2), na.rm = TRUE)
obesityLevels = c('normal', 'overweight', 'obese')
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 

# Make the ordinal variables into mxFactors (ensure ordered is TRUE, and require levels)
twinData[, c("obese1", "obese2")] = umxFactor(twinData[, c("obese1", "obese2")])
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m2 = umxACEv(selDVs = "obese", dzData = dzData, mzData = mzData, sep = '')

# FYI: Show mz, dz, and t1 and t2 have the same levels!
str(mzData)

# ============================================
# = Bivariate continuous and ordinal example =
# ============================================
data(twinData)
# Cut bmi column to form ordinal obesity variables
ordDVs = c("obese1", "obese2")
obesityLevels = c('normal', 'overweight', 'obese')
cutPoints = quantile(twinData[, "bmi1"], probs = c(.5, .2), na.rm = TRUE)
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 

# Make the ordinal variables into ordered mxFactors
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])

# umxACEv can trim out unused variables on its own
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]

m1 = umxACEv(selDVs = c("wt", "obese"), dzData = dzData, mzData = mzData, sep = '')
plot(m1)

# =======================================
# = Mixed continuous and binary example =
# =======================================
require(umx)
data(twinData)
# Cut to form category of 20% obese subjects
# and make into mxFactors (ensure ordered is TRUE, and require levels)
cutPoints = quantile(twinData[, "bmi1"], probs = .2, na.rm = TRUE)
obesityLevels = c('normal', 'obese')
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])

selDVs = c("wt", "obese")
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m1 = umxACEv(selDVs = selDVs, dzData = dzData, mzData = mzData, sep = '')
umxSummary(m1)

# ===================================
# Example with covariance data only =
# ===================================

require(umx)
data(twinData)
selDVs = c("wt")
mz = cov(twinData[twinData$zygosity %in% "MZFF", tvars(selDVs, "")], use = "complete")
dz = cov(twinData[twinData$zygosity %in% "DZFF", tvars(selDVs, "")], use = "complete")
m1 = umxACEv(selDVs = selDVs, sep= "", dzData = dz, mzData= mz, numObsDZ= 569, numObsMZ= 351)
umxSummary(m1, std = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='umxAlgebra'>A simple wrapper for mxAlgebra with name as the first parameter for more readable compact code.</h2><span id='topic+umxAlgebra'></span>

<h3>Description</h3>

<p>umxAlgebra is a wrapper for mxAlgebra which has the name parameter first in order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxAlgebra(
  name = NA,
  expression,
  dimnames = NA,
  ...,
  joinKey = as.character(NA),
  joinModel = as.character(NA),
  verbose = 0L,
  initial = matrix(as.numeric(NA), 1, 1),
  recompute = c("always", "onDemand"),
  fixed = "deprecated_use_recompute"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxAlgebra_+3A_name">name</code></td>
<td>
<p>The name of the algebra (Default = NA). Note the different order compared to mxAlgebra!</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_expression">expression</code></td>
<td>
<p>The algebra</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_dimnames">dimnames</code></td>
<td>
<p>Dimnames of the algebra</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_...">...</code></td>
<td>
<p>Other parameters</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_joinkey">joinKey</code></td>
<td>
<p>See mxAlgebra documentation</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_joinmodel">joinModel</code></td>
<td>
<p>See mxAlgebra documentation</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_verbose">verbose</code></td>
<td>
<p>Quiet or informative</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_initial">initial</code></td>
<td>
<p>See mxAlgebra documentation</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_recompute">recompute</code></td>
<td>
<p>See mxAlgebra documentation</p>
</td></tr>
<tr><td><code id="umxAlgebra_+3A_fixed">fixed</code></td>
<td>
<p>= See mxAlgebra documentation</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxAlgebra">mxAlgebra()</a></code>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxMatrix">umxMatrix()</a></code>
</p>
</li></ul>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
A = umxMatrix("A", "Full", nrow = 3, ncol = 3, values=2)
B = umxAlgebra("B", A)
C = umxAlgebra(A + B, name = "C")
D = umxAlgebra(sin(C), name = "D")
m1 = mxRun(mxModel("AlgebraExample", A, B, C, D ))
mxEval(D, m1)
	
x = umxAlgebra("circ", expression = 2 * pi)
class(x$formula)
x = mxAlgebra(name = "circ", 2 * pi)
class(x$formula) # "call"

## End(Not run)
</code></pre>

<hr>
<h2 id='umxAPA'>Creates nicely formatted journal-style summaries of models, p-values, data-frames and much more.</h2><span id='topic+umxAPA'></span><span id='topic+summaryAPA'></span>

<h3>Description</h3>

<p><code>umxAPA</code> creates APA-style reports from a range of statistical models, or to summarize data. I wrote it to suit me.
</p>
<p>Nice alternatives include <code>jtools::summ</code>.
</p>
<p>Example functionality includes:
</p>

<ol>
<li><p> Given an <code><a href="stats.html#topic+lm">stats::lm()</a></code> model, <code>umxAPA</code> will return a formatted effect, including 95% CI.
e.g.: <code>umxAPA(lm(mpg~wt, data=mtcars), "wt")</code> yields: <code class="reqn">\beta</code> = -5.34 [-6.48, -4.20], p &lt; 0.001. here &quot;wt&quot;
restricts the output to just the named effect.
</p>
</li>
<li> <p><code>umxAPA</code> also supports <code><a href="stats.html#topic+t.test">t.test()</a></code>, <code><a href="stats.html#topic+glm">stats::glm()</a></code>, <code><a href="stats.html#topic+cor.test">cor.test()</a></code>, and others as I need them.
</p>
</li>
<li><p> Get a CI from <code>obj=beta</code> and se=se : <code>umxAPA(-0.30, .03)</code> returns <code class="reqn">\beta</code> = -0.3 [-0.36, -0.24]
</p>
</li>
<li><p> Back out an SE from <code class="reqn">\beta</code> and CI: <code>umxAPA(-0.030, c(-0.073, 0.013))</code> returns <code class="reqn">\beta</code> = -0.03, se = 0.02
</p>
</li>
<li><p> Given only a number as obj, will be treated as a p-value, and returned in APA format.
</p>
</li>
<li><p> Given a dataframe, <code>umxAPA</code> will return a table of correlations with means and SDs in the last row. e.g.:
<code style="white-space: pre;">&#8288;umxAPA(mtcars[,c("cyl", "wt", "mpg", )]&#8288;</code> yields:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: center;"> cyl         </td><td style="text-align: center;">  wt          </td><td style="text-align: center;"> mpg          </td>
</tr>
<tr>
 <td style="text-align: left;">
cyl     </td><td style="text-align: center;"> 1           </td><td style="text-align: center;">  0.78        </td><td style="text-align: center;"> -0.85        </td>
</tr>
<tr>
 <td style="text-align: left;">
wt      </td><td style="text-align: center;"> 0.78        </td><td style="text-align: center;">  1           </td><td style="text-align: center;"> -0.87        </td>
</tr>
<tr>
 <td style="text-align: left;">
mpg     </td><td style="text-align: center;"> -0.85       </td><td style="text-align: center;">  -0.87       </td><td style="text-align: center;"> 1            </td>
</tr>
<tr>
 <td style="text-align: left;">
mean_sd </td><td style="text-align: center;"> 6.19 (1.79) </td><td style="text-align: center;">  3.22 (0.98) </td><td style="text-align: center;"> 20.09 (6.03)
</td>
</tr>

</table>

</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>umxAPA(
  obj = .Last.value,
  se = NULL,
  p = NULL,
  std = FALSE,
  digits = 2,
  use = "complete",
  min = 0.001,
  addComparison = NA,
  report = c("markdown", "html", "none", "expression"),
  lower = TRUE,
  test = c("Chisq", "LRT", "Rao", "F", "Cp"),
  SEs = TRUE,
  means = TRUE,
  suffix = "",
  cols = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxAPA_+3A_obj">obj</code></td>
<td>
<p>A model (e.g. <code><a href="stats.html#topic+lm">lm()</a></code>, <code><a href="nlme.html#topic+lme">lme()</a></code>, <code><a href="stats.html#topic+glm">glm()</a></code>, <code><a href="stats.html#topic+t.test">t.test()</a></code>), beta-value, or <a href="base.html#topic+data.frame">data.frame</a></p>
</td></tr>
<tr><td><code id="umxAPA_+3A_se">se</code></td>
<td>
<p>If obj is a beta, se treated as standard-error (returning a CI).
If obj is a model, used to select effect of interest (blank for all effects).
Finally, set se to the CI c(lower, upper), to back out the SE.</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_p">p</code></td>
<td>
<p>If obj is a beta, use p-value to compute SE (returning a CI).</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_std">std</code></td>
<td>
<p>Whether to report std betas (re-runs model on standardized data).</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_digits">digits</code></td>
<td>
<p>How many digits to round output.</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_use">use</code></td>
<td>
<p>If obj is a data.frame, how to handle NAs (default = &quot;complete&quot;)</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_min">min</code></td>
<td>
<p>For a p-value, the smallest value to report numerically (default .001)</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_addcomparison">addComparison</code></td>
<td>
<p>For a p-value, whether to add &quot;&lt;/=&quot; default (NA) adds &quot;&lt;&quot; if necessary</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_report">report</code></td>
<td>
<p>What to return (default = 'markdown'). Use 'html' to open a web table. none doesn't print. expression can contain <code><a href="grDevices.html#topic+plotmath">plotmath()</a></code></p>
</td></tr>
<tr><td><code id="umxAPA_+3A_lower">lower</code></td>
<td>
<p>Whether to not show the lower triangle of correlations for a data.frame (Default TRUE)</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_test">test</code></td>
<td>
<p>If obj is a glm, which test to use to generate p-values options = &quot;Chisq&quot;, &quot;LRT&quot;, &quot;Rao&quot;, &quot;F&quot;, &quot;Cp&quot;</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_ses">SEs</code></td>
<td>
<p>Whether or not to show correlations with their SE (Default TRUE)</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_means">means</code></td>
<td>
<p>Whether or not to show means in a correlation table (Default TRUE)</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_suffix">suffix</code></td>
<td>
<p>A string to append to the result. Mostly used with report = &quot;expression&quot;</p>
</td></tr>
<tr><td><code id="umxAPA_+3A_cols">cols</code></td>
<td>
<p>Optional, pass in a list of column names when using umxAPA with a dataframe input.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> string
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://stats.oarc.ucla.edu/r/dae/logit-regression/">https://stats.oarc.ucla.edu/r/dae/logit-regression/</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+SE_from_p">SE_from_p()</a></code>
</p>
<p>Other Reporting Functions: 
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# ========================================
# = Report lm (regression/anova) results =
# ========================================
umxAPA(lm(mpg ~ wt + disp, mtcars)) # Report all parameters
umxAPA(lm(mpg ~ wt + disp, mtcars), "wt") # Just effect of weight
umxAPA(lm(mpg ~ wt + disp, mtcars), std = TRUE) # Standardize model!

###############
# GLM example #
###############

df = mtcars
df$mpg_thresh = 0
df$mpg_thresh[df$mpg &gt; 16] = 1
m1 = glm(mpg_thresh ~ wt + gear,data = df, family = binomial)
umxAPA(m1)

###############
# A t-Test    #
###############

umxAPA(t.test(x = 1:10, y = c(7:20)))
umxAPA(t.test(extra ~ group, data = sleep))

# ======================================================
# = Summarize DATA FRAME: Correlations + Means and SDs =
# ======================================================
umxAPA(mtcars[,1:3])
umxAPA(mtcars[,1:3], digits = 3)
umxAPA(mtcars[,1:3], lower = FALSE)
## Not run: 
umxAPA(mtcars[,1:3], report = "html")

## End(Not run)

# ==========================================
# = CONFIDENCE INTERVAL from effect and se =
# ==========================================
umxAPA(.4, .3) # parameter 2 interpreted as SE

# Input beta and CI, and back out the SE
umxAPA(-0.030, c(-0.073, 0.013), digits = 3)

# ====================
# = Format a p-value =
# ====================
umxAPA(.0182613)   #   0.02
umxAPA(.00018261) # &lt; 0.001
umxAPA(.00018261, addComparison = FALSE) # 0.001

# ========================
# = Report a correlation =
# ========================
data(twinData)
tmp = subset(twinData, zygosity %in% c("MZFF", "MZMM"))
m1 = cor.test(~ wt1 + wt2, data = tmp)
umxAPA(m1)

</code></pre>

<hr>
<h2 id='umxBrownie'>A recipe Easter-egg for umx</h2><span id='topic+umxBrownie'></span>

<h3>Description</h3>

<p>How to cook steak.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxBrownie()
</code></pre>


<h3>Details</h3>

<p>Equipment matters. You should buy a heavy cast-iron skillet, and a digital internal thermometer.
Preferably cook over a gas flame.
</p>
<p><em>note</em>: Cheaper cuts like blade steak can come out fine.
</p>
<p>A great reference is The Food Lab by Kenji Alt Lopez. https://www.amazon.co.uk/Food-Lab-Cooking-Through-Science/dp/0393081087.
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://www.amazon.co.uk/Food-Lab-Cooking-Through-Science/dp/0393081087">The Food Lab</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+omxBrownie">omxBrownie()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Functions: 
<code><a href="#topic+deg2rad">deg2rad</a>()</code>,
<code><a href="#topic+fin_JustifiedPE">fin_JustifiedPE</a>()</code>,
<code><a href="#topic+fin_NI">fin_NI</a>()</code>,
<code><a href="#topic+fin_interest">fin_interest</a>()</code>,
<code><a href="#topic+fin_percent">fin_percent</a>()</code>,
<code><a href="#topic+fin_ticker">fin_ticker</a>()</code>,
<code><a href="#topic+fin_valuation">fin_valuation</a>()</code>,
<code><a href="#topic+rad2deg">rad2deg</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umxBrownie()
</code></pre>

<hr>
<h2 id='umxCI'>Add (and, optionally, run) confidence intervals to a structural model.</h2><span id='topic+umxCI'></span>

<h3>Description</h3>

<p><code>umxCI</code> adds <code><a href="OpenMx.html#topic+mxCI">OpenMx::mxCI()</a></code> calls for requested (default all) parameters in a model,
runs these CIs if necessary, and reports them in a neat summary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxCI(
  model = NULL,
  which = c("ALL", NA, "list of your making"),
  remove = FALSE,
  run = c("no", "yes", "if necessary", "show"),
  interval = 0.95,
  type = c("both", "lower", "upper"),
  regex = NULL,
  showErrorCodes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxCI_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> you wish to report <code><a href="OpenMx.html#topic+mxCI">mxCI()</a></code>s on</p>
</td></tr>
<tr><td><code id="umxCI_+3A_which">which</code></td>
<td>
<p>What CIs to add: c(&quot;ALL&quot;, NA, &quot;list of your making&quot;)</p>
</td></tr>
<tr><td><code id="umxCI_+3A_remove">remove</code></td>
<td>
<p>= FALSE (if set, removes existing specified CIs from the model)</p>
</td></tr>
<tr><td><code id="umxCI_+3A_run">run</code></td>
<td>
<p>Whether or not to compute the CIs. Valid values = &quot;no&quot; (default), &quot;yes&quot;, &quot;if necessary&quot;.
'show' means print the intervals if computed, or list their names if not.</p>
</td></tr>
<tr><td><code id="umxCI_+3A_interval">interval</code></td>
<td>
<p>The interval for newly added CIs (defaults to 0.95)</p>
</td></tr>
<tr><td><code id="umxCI_+3A_type">type</code></td>
<td>
<p>The type of CI (defaults to &quot;both&quot;, options are &quot;lower&quot; and  &quot;upper&quot;)</p>
</td></tr>
<tr><td><code id="umxCI_+3A_regex">regex</code></td>
<td>
<p>Add CIs for labels matching this regular expression (over-rides which)</p>
</td></tr>
<tr><td><code id="umxCI_+3A_showerrorcodes">showErrorCodes</code></td>
<td>
<p>Whether to show errors (default == TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>umxCI</code> also reports if any problems were encountered. The codes are standard OpenMx errors and warnings
</p>

<ul>
<li><p> 1: The final iterate satisfies the optimality conditions to the accuracy requested, but the sequence of iterates has not yet converged. NPSOL was terminated because no further improvement could be made in the merit function (Mx status GREEN)
</p>
</li>
<li><p> 2: The linear constraints and bounds could not be satisfied. The problem has no feasible solution.
</p>
</li>
<li><p> 3: The nonlinear constraints and bounds could not be satisfied. The problem may have no feasible solution.
</p>
</li>
<li><p> 4: The major iteration limit was reached (Mx status BLUE).
</p>
</li>
<li><p> 6: The model does not satisfy the first-order optimality conditions to the required accuracy, and no improved point for the merit function could be found during the final linesearch (Mx status RED)
</p>
</li>
<li><p> 7: The function derivatives returned by funcon or funobj appear to be incorrect.
</p>
</li>
<li><p> 9: An input parameter was invalid.
</p>
</li></ul>

<p>If <code>run = "no"</code>, the function simply adds the CI requests, but returns the model without running them.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+confint">stats::confint()</a></code>, <code><a href="#topic+umxConfint">umxConfint()</a></code>, <code><a href="#topic+umxCI">umxCI()</a></code>, <code><a href="#topic+umxModify">umxModify()</a></code>
</p>
</li></ul>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
m1$intervals # none yet - empty list()
m1 = umxCI(m1)
m1$intervals # $G_to_x1...
m1 = umxCI(m1, remove = TRUE) # remove CIs from the model and return it
m1$intervals # none again

# Add CIs by name
parameters(m1, patt="_with_")
m1 = umxCI(m1, which = "x1_with_x1")
m1 = umxCI(m1, which = c("x1_with_x1", "x2_with_x2"))
m1 = umxCI(m1, regex = "x1_with_", run= "yes")
#           lbound estimate ubound lbound Code ubound Code
# x1_with_x1  0.036    0.041  0.047           0           0

# ========================
# = A twin model example =
# ========================
data(twinData) 
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACE(selDVs = c("bmi1","bmi2"), dzData = dzData, mzData = mzData)
umxCI(m1, run = "show") # show what will be requested
umxCI(m1, run = "yes") # actually compute the CIs
# Don't force update of CIs, but if they were just added, then calculate them
umxCI(m1, run = "if necessary")
m1 = umxCI(m1, remove = TRUE) # remove them all
m1$intervals # none!
# Show what parameters are available to get CIs on
umxParameters(m1) 
# Request a CI by label:
m1 = umxCI(m1, which = "a_r1c1", run = "yes")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxCI_boot'>umxCI_boot</h2><span id='topic+umxCI_boot'></span>

<h3>Description</h3>

<p>Compute boot-strapped Confidence Intervals for parameters in an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
The function creates a sampling distribution for parameters by repeatedly drawing samples
with replacement from your data and then computing the statistic for each redrawn sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxCI_boot(
  model,
  rawData = NULL,
  type = c("par.expected", "par.observed", "empirical"),
  std = TRUE,
  rep = 1000,
  conf = 95,
  dat = FALSE,
  digits = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxCI_boot_+3A_model">model</code></td>
<td>
<p>is an optimized mxModel</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_rawdata">rawData</code></td>
<td>
<p>is the raw data matrix used to estimate model</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_type">type</code></td>
<td>
<p>is the kind of bootstrap you want to run. &quot;par.expected&quot; and &quot;par.observed&quot;
use parametric Monte Carlo bootstrapping based on your expected and observed covariance matrices, respectively.
&quot;empirical&quot; uses empirical bootstrapping based on rawData.</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_std">std</code></td>
<td>
<p>specifies whether you want CIs for unstandardized or standardized parameters (default: std = TRUE)</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_rep">rep</code></td>
<td>
<p>is the number of bootstrap samples to compute (default = 1000).</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_conf">conf</code></td>
<td>
<p>is the confidence value (default = 95)</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_dat">dat</code></td>
<td>
<p>specifies whether you want to store the bootstrapped data in the output (useful for multiple analyses, such as mediation analysis)</p>
</td></tr>
<tr><td><code id="umxCI_boot_+3A_digits">digits</code></td>
<td>
<p>rounding precision</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> expected covariance matrix
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://openmx.ssri.psu.edu/thread/2598">https://openmx.ssri.psu.edu/thread/2598</a>
Original written by <a href="https://openmx.ssri.psu.edu/users/bwiernik">https://openmx.ssri.psu.edu/users/bwiernik</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxExpMeans">umxExpMeans()</a></code>, <code><a href="#topic+umxExpCov">umxExpCov()</a></code>
</p>
</li></ul>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1.0)
)

umxCI_boot(m1, type = "par.expected")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxCLPM'>Runs cross-lagged panel models</h2><span id='topic+umxCLPM'></span>

<h3>Description</h3>

<p>One way of assessing causal relationships is by introducing time into the analyses.
<code>umxCLPM</code> implements three cross-lagged panel models (CLPM) from the literature.
The first is the classic CLPM from Heise (1969), the second is the CLPM from Hamaker et al. (2015),
and the third is the CLPM from STARTS (1995).
You simply pass the number of waves and the data set and the model you want to run.
</p>
<p>Sketch mode is available, just do no pass data and you will be returned a model object to be manipulated later.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxCLPM(
  waves,
  name = NULL,
  model = c("Hamaker2015", "Heise1969", "STARTS1995"),
  data = NULL,
  summary = !umx_set_silent(silent = TRUE),
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxCLPM_+3A_waves">waves</code></td>
<td>
<p>Number of waves of data.</p>
</td></tr>
<tr><td><code id="umxCLPM_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to either &quot;Heise1969&quot;, &quot;Hamaker2015&quot;, or &quot;STARTS1995&quot;).</p>
</td></tr>
<tr><td><code id="umxCLPM_+3A_model">model</code></td>
<td>
<p>Model type (&quot;Hamaker2015&quot;, &quot;Heise1969&quot;, or &quot;STARTS1995&quot;)</p>
</td></tr>
<tr><td><code id="umxCLPM_+3A_data">data</code></td>
<td>
<p>Data frame for the analysis</p>
</td></tr>
<tr><td><code id="umxCLPM_+3A_summary">summary</code></td>
<td>
<p>Optionally show a summary.</p>
</td></tr>
<tr><td><code id="umxCLPM_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxCLPM_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Kenny, D.A., &amp; Zautra, A. (1995). The trait-state-error model for multiwave data. <em>Journal of Consulting and Clinical Psychology</em>, <strong>63</strong>, 52–59. <a href="https://doi.org/10.1037/0022-006X.63.1.52">doi:10.1037/0022-006X.63.1.52</a>
</p>
</li></ul>


<ul>
<li><p> Hamaker E.L., Kuiper R.M., &amp; Grasman R. (2015). A critique of the cross-lagged panel model. <em>Psychological Methods</em>, <strong>20</strong>, 102–116. <a href="https://doi.org/10.1037/a0038889">doi:10.1037/a0038889</a>
</p>
</li>
<li><p> Heise D. R. (1970). Causal inference from panel data. <em>Sociological Methodology</em>, 2, 3–27. <a href="https://doi.org/10.2307/270780">doi:10.2307/270780</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# ================
# = 1. Load Data =
# ================
data(docData)
dt &lt;- docData[2:9]

# ============================
# = 2. Make a CLPM model     =
# ============================
hamaker &lt;- umxCLPM(waves = 4, name = "mymodel", model = "Hamaker2015", data = dt)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxCompare'>Print a comparison table of one or more <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>s, formatted nicely.</h2><span id='topic+umxCompare'></span>

<h3>Description</h3>

<p>umxCompare compares two or more <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>s. It has several nice features:
</p>

<ol>
<li><p> It supports direct control of rounding, and reports p-values rounded to APA style.
</p>
</li>
<li><p> It reports the table in your preferred format (default is markdown, options include latex).
</p>
</li>
<li><p> Table columns are arranged to make for easy comparison for readers.
</p>
</li>
<li><p> report = 'inline', will provide an English sentence suitable for a paper.
</p>
</li>
<li><p> report = &quot;html&quot; opens a web table in your browser to paste into a word processor.
</p>
</li></ol>

<p><em>Note</em>: If you leave comparison blank, it will just give fit info for the base model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxCompare(
  base = NULL,
  comparison = NULL,
  all = TRUE,
  digits = 3,
  report = c("markdown", "html", "inline"),
  compareWeightedAIC = FALSE,
  silent = FALSE,
  file = "tmp.html"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxCompare_+3A_base">base</code></td>
<td>
<p>The base <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> for comparison</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_comparison">comparison</code></td>
<td>
<p>The model (or list of models) which will be compared for fit with the base model (can be empty)</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_all">all</code></td>
<td>
<p>Whether to make all possible comparisons if there is more than one base model (defaults to T)</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_digits">digits</code></td>
<td>
<p>rounding for p-values etc.</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_report">report</code></td>
<td>
<p>&quot;markdown&quot; (default), &quot;inline&quot; (a sentence suitable for inclusion in a paper), or &quot;html&quot;.
create a web table and open your default browser.
(handy for getting tables into Word, and other text systems!)</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_compareweightedaic">compareWeightedAIC</code></td>
<td>
<p>Show the Wagenmakers AIC weighted comparison (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_silent">silent</code></td>
<td>
<p>(don't print, just return the table as a dataframe (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxCompare_+3A_file">file</code></td>
<td>
<p>file to write html too if report = &quot;html&quot; (defaults to &quot;tmp.html&quot;)</p>
</td></tr>
</table>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSummary">umxSummary()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>,<code><a href="OpenMx.html#topic+mxCompare">mxCompare()</a></code>
</p>
</li></ul>

<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxEquate">umxEquate</a>()</code>,
<code><a href="#topic+umxMI">umxMI</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSetParameters">umxSetParameters</a>()</code>,
<code><a href="#topic+umxSummary">umxSummary</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

m2 = umxModify(m1, update = "G_to_x2", name = "drop_path_2_x2")
umxCompare(m1, m2)
umxCompare(m1, m2, report = "inline") # Add English-sentence descriptions
umxCompare(m1, m2, report = "html") # Open table in browser

# Two comparison models
m3 = umxModify(m2, update = "G_to_x3", name = "drop_path_2_x2_and_3")

umxCompare(m1, c(m2, m3))
umxCompare(m1, c(m2, m3), compareWeightedAIC = TRUE)
umxCompare(c(m1, m2), c(m2, m3), all = TRUE)

manifests = names(demoOneFactor)
m1 = umxRAM("WLS", data = demoOneFactor, type = "DWLS",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

m2 = umxModify(m1, update = "G_to_x2", name = "drop_path_2_x2")
umxCompare(m1, m2)
umxCompare(m1, m2, report = "inline") # Add English-sentence descriptions
umxCompare(m1, m2, report = "html") # Open table in browser

## End(Not run)
</code></pre>

<hr>
<h2 id='umxConfint'>Get confidence intervals from a umx model</h2><span id='topic+umxConfint'></span>

<h3>Description</h3>

<p>Implements confidence interval function for umx models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxConfint(
  object,
  parm = c("existing", "all", "or one or more labels", "smart"),
  wipeExistingRequests = TRUE,
  level = 0.95,
  run = FALSE,
  showErrorCodes = FALSE,
  optimizer = c("SLSQP", "NPSOL", "CSOLNP", "current")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxConfint_+3A_object">object</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>, possibly already containing <code><a href="OpenMx.html#topic+mxCI">mxCI()</a></code>s that have been <code><a href="OpenMx.html#topic+mxRun">mxRun()</a></code> with intervals = TRUE))</p>
</td></tr>
<tr><td><code id="umxConfint_+3A_parm">parm</code></td>
<td>
<p>Which parameters to get confidence intervals for. Can be &quot;existing&quot;, &quot;all&quot;, or one or more parameter names.</p>
</td></tr>
<tr><td><code id="umxConfint_+3A_wipeexistingrequests">wipeExistingRequests</code></td>
<td>
<p>Whether to remove existing CIs when adding new ones (ignored if parm = 'existing').</p>
</td></tr>
<tr><td><code id="umxConfint_+3A_level">level</code></td>
<td>
<p>The confidence level required (default = .95)</p>
</td></tr>
<tr><td><code id="umxConfint_+3A_run">run</code></td>
<td>
<p>Whether to run the model (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxConfint_+3A_showerrorcodes">showErrorCodes</code></td>
<td>
<p>(default = FALSE)</p>
</td></tr>
<tr><td><code id="umxConfint_+3A_optimizer">optimizer</code></td>
<td>
<p>For difficult CIs, trying other optimizers can help!</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: By default, requesting new CIs wipes the existing ones.
To keep these, set wipeExistingRequests = FALSE.
</p>
<p>Because CIs can take time to run, by default only already-computed CIs will be reported. To run new CIs, set run = TRUE .
</p>
<p><em>Note</em>: <code><a href="stats.html#topic+confint">confint()</a></code> is an OpenMx function which will return SE-based CIs.
</p>
<p>If <code>parm</code> is empty, and <code>run = FALSE</code>, a message will alert you to set <code>run = TRUE</code>.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+confint">stats::confint()</a></code>, <code><a href="OpenMx.html#topic+mxSE">OpenMx::mxSE()</a></code>, <code><a href="#topic+umxCI">umxCI()</a></code>, <code><a href="OpenMx.html#topic+mxCI">OpenMx::mxCI()</a></code>
</p>
</li></ul>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)

manifests = names(demoOneFactor)
m1 = umxRAM("OneFactor", data = demoOneFactor, type = "cov",
	umxPath(from = "G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

m1 = umxConfint(m1, run = TRUE) # There are no existing CI requests...

# Add a CI request for "G_to_x1", run, and report. Save with this CI computed
m2 = umxConfint(m1, parm = "G_to_x1", run = TRUE) 

# Just print out any existing CIs
umxConfint(m2)

# CI requests added for free matrix parameters. User prompted to set run = TRUE
m3 = umxConfint(m1, "all")

# Run the requested CIs
m3 = umxConfint(m3, run = TRUE) 

# Run CIs for free one-headed (asymmetric) paths in RAM model. 
#   note: Deletes other existing requests,
tmp = umxConfint(m1, parm = "A", run = TRUE)

# Wipe existing CIs, add G_to_x1
tmp = umxConfint(m1, parm = "G_to_x1", run = TRUE, wipeExistingRequests = TRUE) 

# For some twin models, a "smart" mode is implemented
# note: only implemented for umxCP so far
m2 =  umxConfint(m1, "smart")

## End(Not run)

</code></pre>

<hr>
<h2 id='umxCov2cor'>Convert a covariance matrix into a correlation matrix</h2><span id='topic+umxCov2cor'></span>

<h3>Description</h3>

<p>A version of <code><a href="stats.html#topic+cov2cor">cov2cor()</a></code> that forces upper and lower triangles to be <em>identical</em> (rather than nearly identical)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxCov2cor(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxCov2cor_+3A_x">x</code></td>
<td>
<p>something that cov2cor can work on (matrix, df, etc.)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A correlation matrix
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="stats.html#topic+cov2cor">cov2cor()</a></code>
</p>
<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umxCov2cor(cov(mtcars[,1:5]))
</code></pre>

<hr>
<h2 id='umxCP'>umxCP: Build and run a Common Pathway twin model</h2><span id='topic+umxCP'></span>

<h3>Description</h3>

<p>Make a 2-group Common Pathway twin model.
</p>
<p>The common-pathway model  (aka &quot;psychometric model&quot; (McArdle and Goldsmith, 1990) provides a powerful tool
for theory-based testing of genetic and environmental differences. It proposes that <code>A</code>, <code>C</code>, and <code>E</code> components
act on a latent substrate (organ, mental mechanism etc.) and this is manifested in the measured phenotypes.
</p>
<p><code>umxCP</code> supports this with pairs of mono-zygotic (MZ) and di-zygotic (DZ) twins reared together
to model the genetic and environmental structure of multiple phenotypes
(measured behaviors).
</p>
<p>Common-pathway path diagram:
</p>
<p><img src="../help/figures/CP.svg" width=50% alt="Figure: CP model" />

</p>
<p>As can be seen, each phenotype also by default has A, C, and E influences specific to that phenotype.
</p>
<p>Features include the ability to include more than one common pathway, to use ordinal data.
</p>
<p><strong>note</strong>: The function <code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options()</a></code> allows users to see and set <code>mvnRelEps</code> and <code>mvnMaxPointsA</code>
mvnRelEps defaults to .005. For ordinal models, you might find that '0.01' works better.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxCP(
  name = "CP",
  selDVs,
  selCovs = NULL,
  dzData = NULL,
  mzData = NULL,
  sep = NULL,
  nFac = 1,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  data = NULL,
  zyg = "zygosity",
  allContinuousMethod = c("cumulants", "marginals"),
  correlatedACE = FALSE,
  dzAr = 0.5,
  dzCr = 1,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("yes", "no", "ordinal", "search"),
  optimizer = NULL,
  equateMeans = TRUE,
  weightVar = NULL,
  bVector = FALSE,
  boundDiag = 0,
  addStd = TRUE,
  addCI = TRUE,
  numObsDZ = NULL,
  numObsMZ = NULL,
  freeLowerA = FALSE,
  freeLowerC = FALSE,
  freeLowerE = FALSE,
  correlatedA = "deprecated"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxCP_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to &quot;CP&quot;).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_seldvs">selDVs</code></td>
<td>
<p>The variables to include.
omit sep in selDVs, i.e., just &quot;dep&quot; not c(&quot;dep_T1&quot;, &quot;dep_T2&quot;).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_selcovs">selCovs</code></td>
<td>
<p>basenames for covariates</p>
</td></tr>
<tr><td><code id="umxCP_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_sep">sep</code></td>
<td>
<p>(required) The suffix for twin 1 and twin 2, often &quot;_T&quot;.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_nfac">nFac</code></td>
<td>
<p>How many common factors (default = 1)</p>
</td></tr>
<tr><td><code id="umxCP_+3A_type">type</code></td>
<td>
<p>One of &quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;</p>
</td></tr>
<tr><td><code id="umxCP_+3A_data">data</code></td>
<td>
<p>If provided, dzData and mzData are treated as valid levels of zyg to select() data sets (default = NULL)</p>
</td></tr>
<tr><td><code id="umxCP_+3A_zyg">zyg</code></td>
<td>
<p>If data provided, this column is used to select rows by zygosity (Default = &quot;zygosity&quot;)</p>
</td></tr>
<tr><td><code id="umxCP_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_correlatedace">correlatedACE</code></td>
<td>
<p>DON'T USE THIS! Allows correlations between the factors built by each of the a, c, and e matrices. Default = FALSE.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_tryhard">tryHard</code></td>
<td>
<p>Default (&quot;yes&quot;) uses mxTryHard, &quot;no&quot; uses normal mxRun. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxCP_+3A_optimizer">optimizer</code></td>
<td>
<p>optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_weightvar">weightVar</code></td>
<td>
<p>If provided, a vector objective will be used to weight the data. (default = NULL).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_bvector">bVector</code></td>
<td>
<p>Whether to compute row-wise likelihoods (defaults to FALSE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_bounddiag">boundDiag</code></td>
<td>
<p>= Numeric lbound for diagonal of the a_cp, c_cp, &amp; e_cp matrices. Set = NULL to ignore.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_addstd">addStd</code></td>
<td>
<p>Whether to add the algebras to compute a std model (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_addci">addCI</code></td>
<td>
<p>Whether to add the interval requests for CIs (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_numobsdz">numObsDZ</code></td>
<td>
<p>= not yet implemented: Ordinal Number of DZ twins: Set this if you input covariance data.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_numobsmz">numObsMZ</code></td>
<td>
<p>= not yet implemented: Ordinal Number of MZ twins: Set this if you input covariance data.</p>
</td></tr>
<tr><td><code id="umxCP_+3A_freelowera">freeLowerA</code></td>
<td>
<p>(ignore): Whether to leave the lower triangle of A free (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_freelowerc">freeLowerC</code></td>
<td>
<p>(ignore): Whether to leave the lower triangle of C free (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_freelowere">freeLowerE</code></td>
<td>
<p>(ignore): Whether to leave the lower triangle of E free (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxCP_+3A_correlateda">correlatedA</code></td>
<td>
<p>deprecated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Like the <code><a href="#topic+umxACE">umxACE()</a></code> model, the CP model decomposes phenotypic variance
into additive genetic (A), unique environmental (E) and, optionally, either
common or shared-environment (C) or non-additive genetic effects (D).
</p>
<p>Unlike the Cholesky, these factors do not act directly on the phenotype. Instead latent A,
C, and E influences impact on one or more latent factors which in turn account for variance in the phenotypes (see Figure).
</p>
<p><strong>Data Input</strong>
Currently, the umxCP function accepts only raw data. This may change in future versions.
</p>
<p><strong>Ordinal Data</strong>
</p>
<p>In an important capability, the model transparently handles ordinal (binary or multi-level
ordered factor data) inputs, and can handle mixtures of continuous, binary, and ordinal
data in any combination.
</p>
<p><strong>Additional features</strong>
</p>
<p>The umxCP function supports varying the DZ genetic association (defaulting to .5)
to allow exploring assortative mating effects, as well as varying the DZ &ldquo;C&rdquo; factor
from 1 (the default for modeling family-level effects shared 100% by twins in a pair),
to .25 to model dominance effects.
</p>
<p><strong>Matrices and Labels in CP model</strong>
</p>
<p>A good way to see which matrices are used in umxCP is to run an example model and plot it.
</p>
<p>All the shared matrices are in the model &quot;top&quot;.
</p>
<p>Matrices <code>top$as</code>, <code>top$cs</code>, and <code>top$es</code> contain the path loadings specific to each variable on their diagonals.
</p>
<p>So, to see the 'as' values, labels, or free states, you can say:
</p>
<p><code>m1$top$as$values</code>
</p>
<p><code>m1$top$as$free</code>
</p>
<p><code>m1$top$as$labels</code>
</p>
<p>Labels relevant to modifying the specific loadings take the form &quot;as_r1c1&quot;, &quot;as_r2c2&quot; etc.
</p>
<p>The common-pathway loadings on the factors are in matrices <code>top$a_cp</code>, <code>top$c_cp</code>, <code>top$e_cp</code>.
</p>
<p>The common factors themselves are in the matrix <code>top$cp_loadings</code> (an nVar * 1 matrix)
</p>
<p>Less commonly-modified matrices are the mean matrix <code>expMean</code>. This has 1 row, and the columns are laid out for each variable for twin 1, followed by each variable for twin 2.
So, in a model where the means for twin 1 and twin 2 had been equated (set = to T1), you could make them independent again with this line:
</p>
<p><code>m1$top$expMean$labels[1,4:6] = c("expMean_r1c4", "expMean_r1c5", "expMean_r1c6")</code>
</p>
<p>For a deep-dive, see <code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel()</a></code>
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Martin, N. G., &amp; Eaves, L. J. (1977). The Genetical Analysis of Covariance Structure. <em>Heredity</em>, <strong>38</strong>, 79-95.
</p>
</li>
<li><p> Kendler, K. S., Heath, A. C., Martin, N. G., &amp; Eaves, L. J. (1987). Symptoms of anxiety and symptoms of depression.
Same genes, different environments? <em>Archives of General Psychiatry</em>, <strong>44</strong>, 451-457. <a href="https://doi.org/10.1001/archpsyc.1987.01800170073010">doi:10.1001/archpsyc.1987.01800170073010</a>.
</p>
</li>
<li><p> McArdle, J. J., &amp; Goldsmith, H. H. (1990). Alternative common factor models for multivariate biometric analyses.
<em>Behavior Genetics</em>, <strong>20</strong>, 569-608. <a href="https://doi.org/10.1007/BF01065873">doi:10.1007/BF01065873</a>.
</p>
</li>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSummaryCP">umxSummaryCP()</a></code>, <code><a href="#topic+umxPlotCP">umxPlotCP()</a></code>. See <code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP()</a></code> to rotate the factor loadings of a <code><a href="#topic+umxCP">umxCP()</a></code> model. See <code><a href="#topic+umxACE">umxACE()</a></code> for more examples of twin modeling.
<code><a href="#topic+plot">plot()</a></code> and <code><a href="#topic+umxSummary">umxSummary()</a></code> work for all twin models, e.g., <code><a href="#topic+umxIP">umxIP()</a></code>, <code><a href="#topic+umxCP">umxCP()</a></code>, <code><a href="#topic+umxGxE">umxGxE()</a></code>, and <code><a href="#topic+umxACE">umxACE()</a></code>.
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ========================================================
# = Run a 3-factor Common pathway twin model of 6 traits =
# ========================================================
require(umx)
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff", "fc", "qol", "hap", "sat", "AD") 
m1 = umxCP(selDVs = selDVs, sep = "_T", nFac = 3, tryHard = "yes",
		dzData = dzData, mzData = mzData)

# Shortcut using "data ="
selDVs = c("gff", "fc", "qol", "hap", "sat", "AD") 
m1 = umxCP(selDVs= selDVs, nFac= 3, data=GFF, zyg="zyg_2grp")

# ===================
# = Do it using WLS =
# ===================
m2 = umxCP("new", selDVs = selDVs, sep = "_T", nFac = 3, optimizer = "SLSQP",
		dzData = dzData, mzData = mzData, tryHard = "ordinal", 
	type= "DWLS", allContinuousMethod='marginals'
)

# =================================================
# = Find and test dropping of shared environment  =
# =================================================
# Show all labels for C parameters 
umxParameters(m1, patt = "^c")
# Test dropping the 9 specific and common-factor C paths
m2 = umxModify(m1, regex = "(cs_.*$)|(c_cp_)", name = "dropC", comp = TRUE)
umxSummaryCP(m2, comparison = m1, file = NA)
umxCompare(m1, m2)

# =======================================
# = Mixed continuous and binary example =
# =======================================
data(GFF)
# Cut to form umxFactor 20% depressed  DEP
cutPoints = quantile(GFF[, "AD_T1"], probs = .2, na.rm = TRUE)
ADLevels  = c('normal', 'depressed')
GFF$DEP_T1 = cut(GFF$AD_T1, breaks = c(-Inf, cutPoints, Inf), labels = ADLevels) 
GFF$DEP_T2 = cut(GFF$AD_T2, breaks = c(-Inf, cutPoints, Inf), labels = ADLevels) 
ordDVs = c("DEP_T1", "DEP_T2")
GFF[, ordDVs] = umxFactor(GFF[, ordDVs])

selDVs = c("gff","fc","qol","hap","sat","DEP") 
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")

# umx_set_optimizer("NPSOL")
# umx_set_optimization_options("mvnRelEps", .01)
m1 = umxCP(selDVs = selDVs, sep = "_T", nFac = 3, dzData = dzData, mzData = mzData)
m2 = umxModify(m1, regex = "(cs_r[3-5]|c_cp_r[12])", name = "dropC", comp= TRUE)

# Do it using WLS
m3 = umxCP(selDVs = selDVs, sep = "_T", nFac = 3, dzData = dzData, mzData = mzData,
		tryHard = "ordinal", type= "DWLS")
# TODO umxCPL fix WLS here
# label at row 1 and column 1 of matrix 'top.binLabels'' in model 'CP3fac' : object 'Vtot'

# ==============================
# = Correlated factors example =
# ==============================
# ====================
# = DON'T USE THIS!!! =
# ====================
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff", "fc", "qol", "hap", "sat", "AD")
m1 = umxCP("base_model", selDVs = selDVs, sep = "_T", correlatedACE = TRUE, 
	 dzData = dzData, mzData = mzData, nFac = 3, tryHard = "yes")

# What are the ace covariance labels? (two ways to get)
umx_lower.tri(m1$top$a_cp$labels)
parameters(m1, patt = "[ace]_cp")

# 1. Now allow a1 and a2 to correlate
m2=umxModify(m1,regex="a_cp_r2c1",name="a2_a1_cov",free=TRUE,tryHard="yes")
umxCompare(m2, m1)

# 2. Drop all (a|c|e) correlations from a model
tmp= namez(umx_lower.tri(m2$top$a_cp$labels), "a_cp", replace= "[ace]_cp")
m3 = umxModify(m2, regex= tmp, comparison = TRUE)

## End(Not run) # end dontrun

</code></pre>

<hr>
<h2 id='umxDiagnose'>Diagnose problems in a model - this is a work in progress.</h2><span id='topic+umxDiagnose'></span>

<h3>Description</h3>

<p>The goal of this function <strong>WILL BE</strong> (not currently functional) to diagnose problems in
a model and return suggestions to the user.
It is a work in progress, and of no use as yet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxDiagnose(model, tryHard = FALSE, diagonalizeExpCov = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxDiagnose_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to diagnose</p>
</td></tr>
<tr><td><code id="umxDiagnose_+3A_tryhard">tryHard</code></td>
<td>
<p>whether I should try and fix it? (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxDiagnose_+3A_diagonalizeexpcov">diagonalizeExpCov</code></td>
<td>
<p>Whether to diagonalize the ExpCov</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Best diagnostics are:
</p>

<ol>
<li><p> Observed data variances and means
</p>
</li>
<li><p> Expected variances and means
</p>
</li>
<li><p> Difference of these?
</p>
</li></ol>

<p>Try
* diagonalizeExpCov diagonal
* <code><a href="#topic+umx_is_ordered">umx_is_ordered()</a></code>
</p>
<div class="sourceCode"><pre>more tricky - we should really report the variances and the standardized thresholds.
</pre></div>
<p>The guidance would be to try starting with unit variances and thresholds that are within
+/- 2 SD of the mean. <a href="https://openmx.ssri.psu.edu/thread/3899">bivariate outliers %p option</a>
</p>


<h3>Value</h3>


<ul>
<li><p> helpful messages and perhaps a modified model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Teaching and Testing functions: 
<code><a href="#topic+tmx_show.MxModel">tmx_show.MxModel</a>()</code>,
<code><a href="#topic+umxPower">umxPower</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("OneFactor", data = demoOneFactor, type= "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
m1 = mxRun(m1)
umxSummary(m1, std = TRUE)
umxDiagnose(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxDiffMZ'>MZ differences method for testing evidence for causality.</h2><span id='topic+umxDiffMZ'></span>

<h3>Description</h3>

<p><code>umxDiffMZ</code> implements the simple twin1-twin2 based correlation method, e.g. De Moor (2008), in which MZ differences
on a variable <code>x</code> asserted to be causal of an outcome variable <code>y</code> are tested for association with differences on y.
The logic of the design is shown below:
</p>
<p><img src="../help/figures/DiffMZRainMud.png" width=50% alt="Figure: MZ differences model" />

</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxDiffMZ(
  x,
  y,
  data,
  sep = "_T",
  mzZygs = c("MZFF", "MZMM"),
  zyg = "zygosity",
  labxy = c(-1.2, 1.8),
  xylim = c(NA, NA),
  digits = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxDiffMZ_+3A_x">x</code></td>
<td>
<p>Presumed causal variable, e.g. &quot;effort&quot;</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_y">y</code></td>
<td>
<p>Presumed caused outcome, e.g. &quot;score&quot;</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_data">data</code></td>
<td>
<p>Dataframe containing the twin data.</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_sep">sep</code></td>
<td>
<p>The separator &quot;_T&quot; used to make twin var names from x and y.</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_mzzygs">mzZygs</code></td>
<td>
<p>The MZ zygosity codes c(&quot;MZFF&quot;, &quot;MZMM&quot;)</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_zyg">zyg</code></td>
<td>
<p>The column containing &quot;zygosity&quot; data</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_labxy">labxy</code></td>
<td>
<p>Where to locate the R2 label (default = c(x=-2,y=3))</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_xylim">xylim</code></td>
<td>
<p>= clip x any axes to range, e.g c(-3,-3)</p>
</td></tr>
<tr><td><code id="umxDiffMZ_+3A_digits">digits</code></td>
<td>
<p>Rounding for beta (def2)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Example output is shown below, with the fitted line and fit inscribed. The plot is just a ggplot graph that is returned and can be edited and formatted.
</p>
<p><img src="../help/figures/DiffMZexample.png" width=50% alt="Figure: MZ differences model" />

</p>
<p>For a more sophisticated linear mixed model approach, see <code><a href="#topic+umxDiscTwin">umxDiscTwin()</a></code>.
</p>


<h3>Value</h3>


<ul>
<li><p> Graph for decorating
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> De Moor, M. H., Boomsma, D. I., Stubbe, J. H., Willemsen, G., &amp; de Geus, E. J. (2008). Testing causality in the association between regular exercise and symptoms of anxiety and depression. Archives of General Psychiatry, 65(8), 897-905. <a href="https://doi.org/10.1001/archpsyc.65.8.897">doi:10.1001/archpsyc.65.8.897</a>.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>, <code><a href="#topic+umxDiscTwin">umxDiscTwin()</a></code>, <code><a href="#topic+umxMR">umxMR()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
umxDiffMZ(x="ht", y="wt", labxy = c(-.5, 3), data = twinData, sep = "")
umxDiffMZ(x="ht", y="wt", xylim = c( -2, 2), data = twinData, sep = "")
</code></pre>

<hr>
<h2 id='umxDiscTwin'>Intra-pair association in MZ, DZ  twin models. (ALPHA quality!)</h2><span id='topic+umxDiscTwin'></span>

<h3>Description</h3>

<p>Testing causal claims is often difficult due to an inability to experimentally randomize traits and situations.
A combination of control data and data from twins discordant for the putative causal trait can falsify causal hypotheses.
</p>
<p><code>umxDiscTwin</code> uses <code><a href="nlme.html#topic+nlme">nlme::nlme()</a></code> to compute the beta for x in <code>y ~ x</code> in models either a) Only controlling non-independence,
and b) MZ and DZ subsample models in which the family level of the predictor y is also controlled.
</p>
<p>If <code>x</code> is causal, then the effect size of x on y is expected to be equally large in all three samples.
If the population association reflects confounded genes or shared environments,
then the association in MZ twins will reduce to zero/non-significance.
</p>
<p><img src="../help/figures/discordantCausalPatterns.png" width=50% alt="Figure: Types of confounding" />

</p>
<p>The function uses the <code><a href="nlme.html#topic+lme">nlme::lme()</a></code> function to compute the effect of the presumed causal variable on the outcome,
controlling, for mid-family score and with random means model using familyID. e.g.:
</p>
<p><code>mzModel  = lme(fixed = y ~ x + FamMeanX, random = ~ 1+FamMeanX|FAMID, data = umx_scale(MZ), na.action = "na.omit")</code>
</p>
<p><strong>Example output from <code>umxDiscTwin</code></strong>
</p>
<p><img src="../help/figures/DiscTwinsExample.png" width=50% alt="Figure: Causation in Discordant twins" />

</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxDiscTwin(
  x,
  y,
  data,
  mzZygs = c("MZFF", "MZMM"),
  dzZygs = c("DZFF", "DZMM", "DZOS"),
  FAMID = "FAMID",
  out = c("table", "plot", "model"),
  use = "complete.obs",
  sep = "_T"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxDiscTwin_+3A_x">x</code></td>
<td>
<p>Cause</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_y">y</code></td>
<td>
<p>Effect</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_data">data</code></td>
<td>
<p>dataframe containing MZ and DZ data</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_mzzygs">mzZygs</code></td>
<td>
<p>MZ zygosities c(&quot;MZFF&quot;, &quot;MZMM&quot;)</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_dzzygs">dzZygs</code></td>
<td>
<p>DZ zygosities c(&quot;DZFF&quot;, &quot;DZMM&quot;, &quot;DZOS&quot;)</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_famid">FAMID</code></td>
<td>
<p>The column containing family IDs (default = &quot;FAMID&quot;)</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_out">out</code></td>
<td>
<p>Whether to return the table or the ggplot (if you want to decorate it)</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_use">use</code></td>
<td>
<p>NA handling in corr.test (default= &quot;complete.obs&quot;)</p>
</td></tr>
<tr><td><code id="umxDiscTwin_+3A_sep">sep</code></td>
<td>
<p>The separator in twin variable names, default = &quot;_T&quot;, e.g. &quot;dep_T1&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> table of results
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Begg, M. D., &amp; Parides, M. K. (2003). Separation of individual-level and cluster-level covariate effects in regression analysis of correlated data. Stat Med, 22(16), 2591-2602. <a href="https://doi.org/10.1002/sim.1524">doi:10.1002/sim.1524</a>
</p>
</li></ul>


<ul>
<li><p> Bergen, S. E., Gardner, C. O., Aggen, S. H., &amp; Kendler, K. S. (2008). Socioeconomic status and social support following illicit drug use: causal pathways or common liability? <em>Twin Res Hum Genet</em>, <strong>11</strong>, 266-274. <a href="https://doi.org/10.1375/twin.11.3.266">doi:10.1375/twin.11.3.266</a>
</p>
</li>
<li><p> McGue, M., Osler, M., &amp; Christensen, K. (2010). Causal Inference and Observational Research: The Utility of Twins. <em>Perspectives on Psychological Science</em>, <strong>5</strong>, 546-556. <a href="https://doi.org/10.1177/1745691610383511">doi:10.1177/1745691610383511</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>, <code><a href="#topic+umxDiffMZ">umxDiffMZ()</a></code>, <code><a href="#topic+umxMR">umxMR()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData)
# add to test must set FAMID umxDiscTwin(x = "ht", y = "wt", data = twinData, sep="")
tmp = umxDiscTwin(x = "ht", y = "wt", data = twinData, sep="", FAMID = "fam")
print(tmp, digits = 3)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxDoC'>Build and run a 2-group Direction of Causation twin models.</h2><span id='topic+umxDoC'></span>

<h3>Description</h3>

<p>Testing causal claims is often difficult due to an inability to conduct experimental randomization of traits and
situations to people. When twins are available, even when measured on a single occasion, the pattern of cross-twin
cross-trait correlations can (given distinguishable modes of inheritance for the two traits) falsify causal hypotheses.
</p>
<p><code>umxDoC</code> implements a 2-group model to form latent variables for each of two traits, and allows testing whether
trait 1 causes trait 2, vice-versa, or even reciprocal causation.
</p>
<p>Using latent variables instead of a manifest measure for testing causation, avoids the bias created by differences in
measurement error in which the more reliable measure appears to &quot;cause&quot; the less reliable one (Gillespie and Martin, 2005).
</p>
<p>The following figure shows how the DoC model appears as a path diagram (for two latent variables X and Y,
each with three indicators). Note: For pedagogical reasons, only the model for 1 twin is shown, and only one DoC pathway drawn.
</p>
<p><img src="../help/figures/DoC.png" width=50% alt="Figure: Direction of Causation" />

</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxDoC(
  name = "DoC",
  var1Indicators,
  var2Indicators,
  mzData = NULL,
  dzData = NULL,
  sep = "_T",
  causal = TRUE,
  autoRun = getOption("umx_auto_run"),
  intervals = FALSE,
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL,
  data = NULL,
  zyg = "zygosity"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxDoC_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to &quot;DOC&quot;).</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_var1indicators">var1Indicators</code></td>
<td>
<p>variables defining latent trait 1</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_var2indicators">var2Indicators</code></td>
<td>
<p>variables defining latent trait 2</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_sep">sep</code></td>
<td>
<p>The separator in twin variable names, default = &quot;_T&quot;, e.g. &quot;dep_T1&quot;.</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_causal">causal</code></td>
<td>
<p>whether to add the causal paths (default TRUE)</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_intervals">intervals</code></td>
<td>
<p>Whether to run mxCI confidence intervals (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_data">data</code></td>
<td>
<p>= NULL If building the MZ and DZ datasets internally from a complete data set.</p>
</td></tr>
<tr><td><code id="umxDoC_+3A_zyg">zyg</code></td>
<td>
<p>= &quot;zygosity&quot; (for the data= method of using this function)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> of subclass MxModelDoC
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> N.A. Gillespie and N.G. Martin (2005). Direction of Causation Models. In <em>Encyclopedia of Statistics in Behavioral Science</em>, <strong>1</strong>. 496–499. Eds. Brian S. Everitt &amp; David C. Howell.
</p>
</li></ul>


<ul>
<li><p> McGue, M., Osler, M., &amp; Christensen, K. (2010). Causal Inference and Observational Research: The Utility of Twins. <em>Perspectives on Psychological Science</em>, <strong>5</strong>, 546-556. <a href="https://doi.org/10.1177/1745691610383511">doi:10.1177/1745691610383511</a>
</p>
</li>
<li><p> Rasmussen, S. H. R., Ludeke, S., &amp; Hjelmborg, J. V. B. (2019). A major limitation of the direction of causation model: non-shared environmental confounding. <em>Twin Res Hum Genet</em>, <strong>22</strong>, 1-13. <a href="https://doi.org/10.1017/thg.2018.67">doi:10.1017/thg.2018.67</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDiscTwin">umxDiscTwin()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# ========================
# = Does Rain cause Mud? =
# ========================

# ================
# = 1. Load Data =
# ================
data(docData)
docData = umx_scale_wide_twin_data(c(var1, var2), docData, sep= "_T")
mzData  = subset(docData, zygosity %in% c("MZFF", "MZMM"))
dzData  = subset(docData, zygosity %in% c("DZFF", "DZMM"))

# =======================================
# = 2. Define manifests for var 1 and 2 =
# =======================================
var1 = paste0("varA", 1:3)
var2 = paste0("varB", 1:3)

# =======================================================
# = 3. Make the non-causal (Cholesky) and causal models =
# =======================================================
Chol = umxDoC(var1= var1, var2= var2, mzData= mzData, dzData= dzData, causal= FALSE)
# nb: DoC initially has causal paths fixed @0
DoC  = umxDoC(var1= var1, var2= var2, mzData= mzData, dzData= dzData, causal= TRUE)
a2b   = umxModify(DoC, "a2b", free = TRUE, name = "a2b"); summary(a2b)
b2a   = umxModify(DoC, "b2a", free = TRUE, name = "b2a"); summary(b2a)
Recip = umxModify(DoC, c("a2b", "b2a"), free = TRUE, name = "Recip"); summary(Recip)

# Compare fits
umxCompare(Chol, c(a2b, b2a, Recip))

# ==========================================
# = Alternative call with data in one file =
# ==========================================
data(docData)
docData = umx_scale_wide_twin_data(c(var1, var2), docData, sep= "_T")
DoC = umxDoC(var1= paste0("varA", 1:3), var2= paste0("varB", 1:3),
	  mzData= c("MZFF", "MZMM"), dzData= c("DZFF", "DZMM"), data = docData
)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxDoCp'>Make a direction of causation model based on umxPath statements</h2><span id='topic+umxDoCp'></span>

<h3>Description</h3>

<p>Makes a direction of causation model with <code><a href="#topic+umxPath">umxPath()</a></code> statements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxDoCp(
  var1Indicators,
  var2Indicators,
  mzData = NULL,
  dzData = NULL,
  sep = "_T",
  causal = TRUE,
  name = "DoC",
  autoRun = getOption("umx_auto_run"),
  intervals = FALSE,
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxDoCp_+3A_var1indicators">var1Indicators</code></td>
<td>
<p>The indicators of trait 1</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_var2indicators">var2Indicators</code></td>
<td>
<p>The indicators of trait 2</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ twin dataframe</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ twin dataframe</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_sep">sep</code></td>
<td>
<p>(Default &quot;_T&quot;)</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_causal">causal</code></td>
<td>
<p>(Default TRUE)</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_name">name</code></td>
<td>
<p>= &quot;DoC&quot;</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_autorun">autoRun</code></td>
<td>
<p>Default: getOption(&quot;umx_auto_run&quot;)_</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_intervals">intervals</code></td>
<td>
<p>Whether to run intervals (Default FALSE)</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_tryhard">tryHard</code></td>
<td>
<p>Default &quot;no&quot; (valid = &quot;yes&quot;, &quot;ordinal&quot;, &quot;search&quot;)</p>
</td></tr>
<tr><td><code id="umxDoCp_+3A_optimizer">optimizer</code></td>
<td>
<p>Whether to set this for this run (Default no))</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See also <code><a href="#topic+umxDoC">umxDoC()</a></code>
</p>


<h3>Value</h3>


<ul>
<li><p> [A direction of causation model with <code><a href="#topic+umxPath">umxPath()</a></code> statements.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ================
# = 1. Load Data =
# ================
data(docData)
var1 = paste0("varA", 1:3)
var2 = paste0("varB", 1:3)
tmp = umx_scale_wide_twin_data(varsToScale= c(var1, var2), sep= "_T", data= docData)
mzData = subset(docData, zygosity %in% c("MZFF", "MZMM"))
dzData = subset(docData, zygosity %in% c("DZFF", "DZMM"))
m1 = umxDoCp(var1, var2, mzData= mzData, dzData= dzData, sep = "_T", causal= TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxEFA'>FIML-based Exploratory Factor Analysis (EFA)</h2><span id='topic+umxEFA'></span><span id='topic+umxFactanal'></span>

<h3>Description</h3>

<p>Perform full-information maximum-likelihood factor analysis on a data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxEFA(
  x = NULL,
  factors = NULL,
  data = NULL,
  scores = c("none", "ML", "WeightedML", "Regression"),
  minManifests = NA,
  rotation = c("varimax", "promax", "none"),
  return = c("model", "loadings"),
  report = c("markdown", "html"),
  summary = FALSE,
  name = "efa",
  digits = 2,
  tryHard = c("no", "yes", "ordinal", "search"),
  n.obs = NULL,
  covmat = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxEFA_+3A_x">x</code></td>
<td>
<p>Either 1: data, 2: Right-hand-side ~ formula , 3: Vector of variable names, or 4: Name for the model.</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_factors">factors</code></td>
<td>
<p>Either number of factors to request or a vector of factor names.</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_data">data</code></td>
<td>
<p>A dataframe you are modeling.</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_scores">scores</code></td>
<td>
<p>Type of scores to produce, if any. The default is none, &quot;Regression&quot; gives Thompson's scores. Other options are 'ML', 'WeightedML', Partial matching allows these names to be abbreviated.</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_minmanifests">minManifests</code></td>
<td>
<p>The least number of variables required to return a score for a participant (Default = NA).</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_rotation">rotation</code></td>
<td>
<p>A rotation to perform on the loadings (default  = &quot;varimax&quot; (orthogonal))</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_return">return</code></td>
<td>
<p>by default, the resulting MxModel is returned. Say &quot;loadings&quot; to get a fact.anal object.</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_report">report</code></td>
<td>
<p>Report as markdown to the console, or open a table in browser (&quot;html&quot;)</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_summary">summary</code></td>
<td>
<p>run <code><a href="#topic+umxSummary">umxSummary()</a></code> on the underlying umxRAM model? (Default = FALSE)</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_name">name</code></td>
<td>
<p>A name for your model (default = efa)</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_digits">digits</code></td>
<td>
<p>rounding (default = 2)</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_n.obs">n.obs</code></td>
<td>
<p>Number of observations in if covmat provided (default = NA)</p>
</td></tr>
<tr><td><code id="umxEFA_+3A_covmat">covmat</code></td>
<td>
<p>Covariance matrix of data you are modeling (not implemented)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As in <code><a href="stats.html#topic+factanal">factanal()</a></code>, you need only specify the number of factors and offer up
some manifest data, e.g:
</p>
<p><code>umxEFA(factors = 2, data = mtcars)</code>
</p>
<p>Equivalently, you can also give a list of factor names:
</p>
<p><code>umxEFA(factors = c("g", "v"), data = mtcars)</code>
</p>
<p>The factor model is implemented as a structural equation model, e.g.
</p>
<p><img src="../help/figures/umxEFA.png" width=50% alt="Figure: umxEFA.png" />

</p>
<p>You can request <code>scores</code> from the model. Unlike factanal, these can cope with missing data.
</p>
<p>You can also rotate the factors using any rotation function.
</p>
<p>In an EFA, all items may load on all factors.
</p>
<p>Should work with rotations provided in <code>library("GPArotation")</code> and <code>library("psych")</code>, e.g
</p>
<p><strong>Orthogonal</strong>: &quot;varimax&quot;, &quot;quartimax&quot;, &quot;bentlerT&quot;, &quot;equamax&quot;, &quot;varimin&quot;, &quot;geominT&quot; and &quot;bifactor&quot;
<strong>Oblique</strong>: &quot;Promax&quot;, &quot;promax&quot;, &quot;oblimin&quot;, &quot;simplimax&quot;, &quot;bentlerQ&quot;, &quot;geominQ&quot;, &quot;biquartimin&quot; and &quot;cluster&quot;
</p>
<p>For identification we need m<sup>2</sup> degrees of freedom.
We get m(m+1)/2 from fixing factor variances to 1 and covariances to 0.
We get another m(m-1)/2 degrees of freedom by fixing the upper-right hand corner of
the factor loadings component of the A matrix at 0.
</p>
<p>To aid optimization, manifest residual variances are <code>lbounded</code> at 0.
</p>
<p>EFA reports standardized loadings: to do this, we scale the data.
</p>
<p><em>note</em>: Bear in mind that factor scores are indeterminate (can be rotated to an infinity of equivalent solutions).
</p>
<p>Thanks to @ConorDolan for code implementing the rotation matrix and other suggestions!
</p>


<h3>Value</h3>


<ul>
<li><p> EFA <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>,
</p>
</li></ul>

<p>Hendrickson, A. E. and White, P. O. (1964). Promax: a quick method for rotation to orthogonal oblique structure. <em>British Journal of Statistical Psychology</em>, <strong>17</strong>, 65–70. <a href="https://doi.org/10.1111/j.2044-8317.1964.tb00244.x">doi:10.1111/j.2044-8317.1964.tb00244.x</a>.
</p>
<p>Kaiser, H. F. (1958). The varimax criterion for analytic rotation in factor analysis. <em>Psychometrika</em>, <strong>23</strong>, 187–200. <a href="https://doi.org/10.1007/BF02289233">doi:10.1007/BF02289233</a>.
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+factanal">factanal()</a></code>, <code><a href="OpenMx.html#topic+mxFactorScores">mxFactorScores()</a></code>
</p>
</li></ul>

<p>Other Super-easy helpers: 
<code><a href="#topic+umxTwoStage">umxTwoStage</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
myVars = c("mpg", "disp", "hp", "wt", "qsec")
m1 = umxEFA(mtcars[, myVars], factors =   2, rotation = "promax")
# By default, returns the model
umx_is_MxModel(m1) # TRUE
# The loadings are stashed in the model:
loadings(m1)

# Formula interface in umxEFA
m2 = umxEFA(~ mpg + disp + hp + wt + qsec, factors = 2, rotation = "promax", data = mtcars)
loadings(m2)

# base-R factanal Formula interface for comparison
m2 = factanal(~ mpg + disp + hp + wt + qsec, factors = 2, rotation = "promax", data = mtcars)
loadings(m2)

# Return the loadings object
x = umxEFA(mtcars[, myVars], factors = 2, return = "loadings")
names(x) # "loadings" "rotmat"

# scores requested, so these will be returned
x = umxEFA(name = "score", factors = "g", data = mtcars[, myVars], scores= "Regression")
head(x)
#       g
# 1  -0.48059346
# 2  -0.42354000
# 3  -0.87078110

m1 = umxEFA(myVars, factors = 2, data = mtcars, rotation = "promax")
m1 = umxEFA(name = "named", factors = "g", data = mtcars[, myVars])
m1 = umxEFA(name = "by_number", factors = 2, rotation = "promax", data = mtcars[, myVars])


## End(Not run)
</code></pre>

<hr>
<h2 id='umxEquate'>umxEquate: Equate two or more paths</h2><span id='topic+umxEquate'></span>

<h3>Description</h3>

<p>In addition to dropping or adding parameters, a second common task in modeling
is to equate parameters. umx provides a convenience function to equate parameters
by setting one or more parameters (the &quot;slave&quot; set) equal to one or more &quot;master&quot;
parameters. These parameters are picked out via their labels, and setting two or more
parameters to have the same value is accomplished by setting the slave(s) to have
the same label(s) as the master parameters, thus constraining them to take the same
value during model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxEquate(
  model,
  a,
  b,
  newlabels = NULL,
  free = c(TRUE, FALSE, NA),
  verbose = FALSE,
  name = NULL,
  autoRun = FALSE,
  tryHard = c("no", "yes", "ordinal", "search"),
  comparison = TRUE,
  master = NULL,
  slave = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxEquate_+3A_model">model</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> within which to equate parameters listed in &quot;a&quot; with those in &quot;b&quot;</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_a">a</code></td>
<td>
<p>one or more labels to equate with those in the &quot;b&quot; set.</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_b">b</code></td>
<td>
<p>one or more labels to equate with those in the 'a' set. (if 'newlabels' is NULL, labels will be set to 'a' list).</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_newlabels">newlabels</code></td>
<td>
<p>(optional) list of new labels for the equated parameters.</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_free">free</code></td>
<td>
<p>Must the parameter(s) initially be free? (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_verbose">verbose</code></td>
<td>
<p>Whether to give verbose feedback (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_name">name</code></td>
<td>
<p>name for the returned model (optional: Leave empty to leave name unchanged)</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_comparison">comparison</code></td>
<td>
<p>Compare the new model to the old (if updating an existing model: default = TRUE)</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_master">master</code></td>
<td>
<p>synonym for 'a'</p>
</td></tr>
<tr><td><code id="umxEquate_+3A_slave">slave</code></td>
<td>
<p>synonym for 'b'</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>note</em>: In addition to using this method to equating parameters, you can
also equate one parameter to another by setting its label to the
&quot;square bracket&quot; address of the master, e.g. &quot;a[r,c]&quot;.
</p>
<p><em>Tip</em>: To find labels of free parameters use <code><a href="#topic+umxGetParameters">umxGetParameters()</a></code>
with free = TRUE
</p>
<p><em>Tip</em>: To find labels by name, use the regex parameter of <code><a href="#topic+umxGetParameters">umxGetParameters()</a></code>
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxModify">umxModify()</a></code>, <code><a href="#topic+umxCompare">umxCompare()</a></code>
</p>
<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxCompare">umxCompare</a>()</code>,
<code><a href="#topic+umxMI">umxMI</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSetParameters">umxSetParameters</a>()</code>,
<code><a href="#topic+umxSummary">umxSummary</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
# By default, umxEquate just equates master and slave labels: doesn't run model
m2 = umxEquate(m1, a = "G_to_x1", b = "G_to_x2", name = "Eq x1 x2 loadings")

# Set autoRun = TRUE and comparison = TRUE to run and output a comparison
m2 = umxEquate(m1, autoRun = TRUE, comparison = TRUE, name = "Eq_x1_x2",
	     a = "G_to_x1", b = "G_to_x2"
)

# rename the equated paths
m2 = umxEquate(m1, autoRun = TRUE, comparison = TRUE, name = "Eq_x1_x2",
	     a = "G_to_x1", b = "G_to_x2", newlabels = c("equated")
)
parameters(m2)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxExamples'>Example code from Twin Research and Human Genetics Paper on umx</h2><span id='topic+umxExamples'></span><span id='topic+umxExampleCode_TRHGpaper'></span>

<h3>Description</h3>

<p>This is the example code used in our Twin Research and Human Genetics Paper on <code>umx</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxExamples()
</code></pre>


<h3>References</h3>


<ul>
<li><p> Bates, T. C., Neale, M. C., &amp; Maes, H. H. (2019).
umx: A library for Structural Equation and Twin Modelling in R.
<em>Twin Research and Human Genetics</em>, <strong>22</strong>, 27-41. <a href="https://doi.org/10.1017/thg.2019.2">doi:10.1017/thg.2019.2</a>.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx">umx()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# ==========================================================================
# = Example code from Twin Research and Human Genetics Paper on umx(model) =
# ==========================================================================

# Installing umx can be done using the R-code:
install.packages("umx")
# load as usual
library("umx")

# The current package version can be shown with:
umxVersion("umx")

# Get the latest NPSOL and multi-core build of OpenMx
install.OpenMx("NPSOL")

# Bleeding edge version of OpenMx for MacOS
install.OpenMx("travis")

# ============
# = CFA Code =
# ============

# Load the umx library (this is assumed in subsequent examples
library("umx")

# Load demo data consisting of 5 correlated variables, x1:x5 
data(demoOneFactor)

# Create a list of the manifest variables for use in specifying the model
manifests = paste0("x", 1:5) # 'x1', 'x2', ...'x5'

# Create model cfa1, with name 'CFA', data demoOneFactor, and the CFA paths.

cfa1 = umxRAM("CFA", data = demoOneFactor,
	# Create latent variable 'G', with fixed variance of 1 and mean of 0
	umxPath(v1m0 = "G"),
	# Create 5 manifest variables, x1:x5, with free variance and mean
	umxPath(v.m. = manifests),
	# Create 1-headed paths from G to each of the manifests
	umxPath("G", to = manifests)
)

# ====================
# = Parameter labels =
# ====================

x = xmuLabel(mxMatrix(name="means", "Full", ncol = 2, nrow = 2))
x$labels

# ========
# = Plot =
# ========

plot(cfa1, means = FALSE, fixed = TRUE)
plot(cfa1, std = TRUE, digits = 3, resid= 'line')


m1 = umxRAM("play", data = c("A", "B", "C"),
	umxPath(unique.pairs = c("A", "B", "C"))
)

# ==============================================
# = Inspecting model parameters and residuals. =
# ==============================================

# Show parameters, below .1, with label containing `x2' 
parameters(cfa1, "above", .5, pattern= "x2")

residuals(cfa1, suppress = .005)

# ==================================
# = Modifying and comparing models =
# ==================================

# Variable names in the Duncan data
dimnames = c("RespOccAsp", "RespEduAsp", "RespParAsp", "RespIQ", "RespSES",
             "FrndOccAsp", "FrndEduAsp", "FrndParAsp", "FrndIQ", "FrndSES")
# lower-triangle of correlations among these variables
tmp = c(
	0.6247,
	0.2137, 0.2742,
	0.4105, 0.4043, 0.1839,
	0.3240, 0.4047, 0.0489, 0.2220,
	0.3269, 0.3669, 0.1124, 0.2903, 0.3054,
	0.4216, 0.3275, 0.0839, 0.2598, 0.2786, 0.6404,
	0.0760, 0.0702, 0.1147, 0.1021, 0.0931, 0.2784, 0.1988, 
	0.2995, 0.2863, 0.0782, 0.3355, 0.2302, 0.5191, 0.5007,  0.2087,
	0.2930, 0.2407, 0.0186, 0.1861, 0.2707, 0.4105, 0.3607, -0.0438, 0.2950
)

# Use the umx_lower2full function to create a full correlation matrix
duncanCov = umx_lower2full(tmp, diag = FALSE, dimnames = dimnames)

# Turn the duncan data into an mxData object for the model
duncanCov = mxData(duncanCov, type = "cov", numObs = 300)

respondentFormants   = c("RespSES", "FrndSES", "RespIQ", "RespParAsp")
friendFormants       = c("FrndSES", "RespSES", "FrndIQ", "FrndParAsp")
latentAspiration     = c("RespLatentAsp", "FrndLatentAsp")
respondentOutcomeAsp = c("RespOccAsp", "RespEduAsp")
friendOutcomeAsp     = c("FrndOccAsp", "FrndEduAsp")


duncan1 = umxRAM("Duncan", data = duncanCov,
	# Working from the left of the model, as laid out in the figure, to right...

	# 1. Add all distinct paths between variables to allow the 
	# exogenous manifests to covary with each other.
	umxPath(unique.bivariate = c(friendFormants, respondentFormants)),

	# 2. Add variances for the exogenous manifests,
	# These are assumed to be error-free in this model,
	# and are fixed at their known value).
	umxPath(var = c(friendFormants, respondentFormants), fixedAt = 1),

	# 3. Paths from IQ, SES, and parental aspiration 
	# to latent aspiration for Respondents:
	umxPath(respondentFormants, to = "RespLatentAsp"),
	# And same for friends
	umxPath(friendFormants,     to = "FrndLatentAsp"),

	# 4. Add residual variance for the two aspiration latent traits.
	umxPath(var = latentAspiration),

	# 5. Allow the latent traits each influence the other.
	# This is done using fromEach, and the values are 
	# bounded to improve stability.
	# note: Using one-label would equate these 2 influences
	umxPath(fromEach = latentAspiration, lbound = 0, ubound = 1), 

	# 6. Allow latent aspiration to affect respondent's
	# occupational &amp; educational aspiration.
	# note: firstAt = 1 is used to provide scale to the latent variables.
	umxPath("RespLatentAsp", to = respondentOutcomeAsp, firstAt = 1),

	# And their friends
	umxPath("FrndLatentAsp", to = friendOutcomeAsp, firstAt = 1),
	
	# 7. Finally, on the right hand side of figure, we add
	# residual variance for the endogenous manifests.
	umxPath(var = c(respondentOutcomeAsp, friendOutcomeAsp))
)

# ====================
# = Modifying models =
# ====================

# Collect a list of paths to drop
pathList = c("RespLatentAsp_to_FrndLatentAsp", "FrndLatentAsp_to_RespLatentAsp")

# Modify the model duncan1, requesting a comparison table:
duncan2 = umxModify(duncan1, update = pathList, name = "No_influence", comparison = TRUE)

# An example using regex, to drop all paths beginning "G_to_"
cfa2 = umxModify(cfa1, regex = "^G_to.*")

# ====================
# = Comparing models =
# ====================

umxCompare(duncan1, duncan2, report = "inline")

# To open the output as an html table in a browser, say:
umxCompare(duncan1, duncan2, report = "html")

# =============================
# = Equating model parameters =
# =============================

parameters(duncan1, pattern = "IQ_to_")


duncan3 = umxModify(duncan1, name = "Equate IQ effect", comparison = TRUE,
	master = "RespIQ_to_RespLatentAsp", 
	update = "FrndIQ_to_FrndLatentAsp" 
)

# ================
# = ACE examples =
# ================

require(umx); 
# open the built in dataset of Australian height and weight twin data
data("twinData")
selDVs = c("wt")
dz = twinData[twinData$zygosity == "DZFF", ]
mz = twinData[twinData$zygosity == "MZFF", ]

ACE1 = umxACE(selDVs = selDVs, dzData = dz, mzData = mz, sep = "")
ACE2 = umxModify(ACE1, update = "c_r1c1", name = "dropC")
umxSummary(ACE1, std = FALSE, report = 'html', digits = 3, comparison = ACE2)
parameters(ACE1)

ACE2 = umxModify(ACE1, update = "c_r1c1", name = "dropC")

# ================================
# = Example Common Pathway model =
# ================================

# load twin data built into umx
data("twinData")

# Selecting the 'ht' and 'wt' variables
selDVs = c("ht", "wt")
mzData = subset(twinData, zygosity == "MZFF",)
dzData = subset(twinData, zygosity == "DZFF",)

# Run and report a common-pathway model
CP1 = umxCP(selDVs = selDVs, dzData = dzData, mzData = mzData, suffix = "")

paths = c("c_cp_r1c1", "cs_r1c1", "cs_r2c2")
CP2 = umxModify(CP1, update = paths, name = "dropC", comparison = TRUE)

CP2 = umxModify(CP1, regex = "(^cs_)|(^c_cp_)", name = "dropC")
umxSummary(CP2, comparison = CP1)

# ====================================
# = Example Gene x environment model =
# ====================================

data("twinData")
twinData$age1 = twinData$age2 = twinData$age
# Define the DV and definition variables
selDVs  = c("bmi1", "bmi2")
selDefs = c("age1", "age2")
selVars = c(selDVs, selDefs)

# Create datasets
mzData  = subset(twinData, zygosity == "MZFF")
dzData  = subset(twinData, zygosity == "DZFF")

# Build, run and report the GxE model using selected DV and moderator
# umxGxE will remove and report rows with missing data in definition variables.
GE1 = umxGxE(selDVs = selDVs, selDefs = selDefs, 
  dzData = dzData, mzData = mzData,  dropMissingDef = TRUE)

# Shift the legend to the top right
umxSummary(GE1, location = "topright")

# plot standardized and raw output in separate graphs
umxSummary(GE1, separateGraphs = TRUE)

GE2 = umxModify(GE1, update = "am_r1c1", comparison = TRUE)
umxReduce(GE1)

# =================================
# = Example GxE windowed analysis =
# =================================

require(umx);
data("twinData") 
mod     = "age"
selDVs  = c("bmi1", "bmi2")

# select the younger cohort of twins
tmpTwin = twinData[twinData$cohort == "younger", ]
# Drop twins with missing moderator
tmpTwin = tmpTwin[!is.na(tmpTwin[mod]), ]
mzData  = subset(tmpTwin, zygosity == "MZFF", c(selDVs, mod))
dzData  = subset(tmpTwin, zygosity == "DZFF", c(selDVs, mod))
# toggle autoplot off, so we don't plot every level of the moderator
umx_set_auto_plot(FALSE)
umxGxE_window(selDVs = selDVs, moderator = mod, mzData = mzData, dzData = dzData)
umx_set_auto_plot(TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxExpCov'>Get the expected vcov matrix</h2><span id='topic+umxExpCov'></span><span id='topic+vcov.MxModel'></span>

<h3>Description</h3>

<p>Extract the expected covariance matrix from an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxExpCov(object, latents = FALSE, manifests = TRUE, digits = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxExpCov_+3A_object">object</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to get the covariance matrix from</p>
</td></tr>
<tr><td><code id="umxExpCov_+3A_latents">latents</code></td>
<td>
<p>Whether to select the latent variables (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxExpCov_+3A_manifests">manifests</code></td>
<td>
<p>Whether to select the manifest variables (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxExpCov_+3A_digits">digits</code></td>
<td>
<p>precision of reporting. NULL (Default) = no rounding.</p>
</td></tr>
<tr><td><code id="umxExpCov_+3A_...">...</code></td>
<td>
<p>extra parameters (to match <code><a href="stats.html#topic+vcov">vcov()</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> expected covariance matrix
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://openmx.ssri.psu.edu/thread/2598">https://openmx.ssri.psu.edu/thread/2598</a>
Original written by <a href="https://openmx.ssri.psu.edu/users/bwiernik">https://openmx.ssri.psu.edu/users/bwiernik</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxRun">umxRun()</a></code>, <code><a href="#topic+umxCI_boot">umxCI_boot()</a></code>
</p>
</li></ul>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)#'
vcov(m1) # supplied by OpenMx
umxExpCov(m1, digits = 3)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxExpMeans'>Extract the expected means matrix from an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code></h2><span id='topic+umxExpMeans'></span>

<h3>Description</h3>

<p>Extract the expected means matrix from an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxExpMeans(model, manifests = TRUE, latents = NULL, digits = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxExpMeans_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to get the means from</p>
</td></tr>
<tr><td><code id="umxExpMeans_+3A_manifests">manifests</code></td>
<td>
<p>Whether to select the manifest variables (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxExpMeans_+3A_latents">latents</code></td>
<td>
<p>Whether to select the latent variables (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxExpMeans_+3A_digits">digits</code></td>
<td>
<p>precision of reporting. Default (NULL) will not round at all.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> expected means
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://openmx.ssri.psu.edu/thread/2598">https://openmx.ssri.psu.edu/thread/2598</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor,
	umxPath("G", to = manifests),
	umxPath(v.m. = manifests),
	umxPath(v1m0 = "G")
)

umxExpMeans(m1)
umxExpMeans(m1, digits = 3)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxFactor'>umxFactor</h2><span id='topic+umxFactor'></span><span id='topic+umx_factor'></span>

<h3>Description</h3>

<p>A convenient version of <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code> supporting the common
case in which the factor levels are those in the variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxFactor(
  x = character(),
  levels = NULL,
  labels = levels,
  exclude = NA,
  ordered = TRUE,
  collapse = FALSE,
  verbose = FALSE,
  sep = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxFactor_+3A_x">x</code></td>
<td>
<p>A variable to recode as an mxFactor (see <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>)</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_levels">levels</code></td>
<td>
<p>(default NULL). Like <code><a href="base.html#topic+factor">factor()</a></code> but UNLIKE <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>,
unique values will be used if levels not specified.</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_labels">labels</code></td>
<td>
<p>= levels (see <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>)</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_exclude">exclude</code></td>
<td>
<p>= NA (see <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>)</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_ordered">ordered</code></td>
<td>
<p>= TRUE By default return an ordered mxFactor</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_collapse">collapse</code></td>
<td>
<p>= FALSE (see <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>)</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_verbose">verbose</code></td>
<td>
<p>Whether to tell user about such things as coercing to factor</p>
</td></tr>
<tr><td><code id="umxFactor_+3A_sep">sep</code></td>
<td>
<p>If twin data are being used, the string that separates the base from twin index
will try and ensure factor levels same across all twins.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxFactanal">umxFactanal()</a></code>, <code><a href="OpenMx.html#topic+mxFactor">mxFactor()</a></code>
</p>
</li></ul>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umxFactor(letters)
umxFactor(letters, verbose = TRUE) # report coercions
umxFactor(letters, ordered = FALSE) # non-ordered factor like factor(x)
# Dataframe example:
x = umx_factor(mtcars[,c("cyl", "am")], ordered = FALSE); str(x)
# =================
# = Twin example: =
# =================
data(twinData)
tmp = twinData[, c("bmi1", "bmi2")]
tmp$bmi1[tmp$bmi1 &lt;= 22] = 22
tmp$bmi2[tmp$bmi2 &lt;= 22] = 22
# remember to factor _before_ breaking into MZ and DZ groups
x = umxFactor(tmp, sep = ""); str(x)
xmu_check_levels_identical(x, "bmi", sep="")

# Simple example to check behavior
x = round(10 * rnorm(1000, mean = -.2))
y = round(5 * rnorm(1000))
x[x &lt; 0] = 0; y[y &lt; 0] = 0
jnk = umxFactor(x); str(jnk)
df  = data.frame(x = x, y = y)
jnk = umxFactor(df); str(jnk)
</code></pre>

<hr>
<h2 id='umxFactorScores'>Return factor scores from a model as an easily consumable dataframe.</h2><span id='topic+umxFactorScores'></span>

<h3>Description</h3>

<p>umxFactorScores takes a model, and computes factors scores using the selected method (one
of 'ML', 'WeightedML', or 'Regression')
It is a simple wrapper around mxFactorScores. For missing data, you must specify the least number of
variables allowed for a score (subjects with fewer than minManifests will return a score of NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxFactorScores(
  model,
  type = c("ML", "WeightedML", "Regression"),
  minManifests = NA,
  return = c("Scores", "StandardErrors")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxFactorScores_+3A_model">model</code></td>
<td>
<p>The model from which to generate scores.</p>
</td></tr>
<tr><td><code id="umxFactorScores_+3A_type">type</code></td>
<td>
<p>Method of computing the score ('ML', 'WeightedML', or 'Regression').</p>
</td></tr>
<tr><td><code id="umxFactorScores_+3A_minmanifests">minManifests</code></td>
<td>
<p>The minimum number of variables not NA to return a score for a participant (Default = ask).</p>
</td></tr>
<tr><td><code id="umxFactorScores_+3A_return">return</code></td>
<td>
<p>What to return (defaults to &quot;Scores&quot;, which is what most users want, but can return &quot;StandardErrors&quot; on each score.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> dataframe of scores.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxFactorScores">mxFactorScores()</a></code>
</p>
</li></ul>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
m1 = umxEFA(mtcars, factors = 2)
x = umxFactorScores(m1, type = 'Regression', minManifests = 3)

# =========================================================================
# = histogram of F1 and plot of F1 against F2 showing they are orthogonal =
# =========================================================================
hist(x$F1)
plot(F1 ~ F2, data = x)

m1 = umxEFA(mtcars, factors = 1)
x = umxFactorScores(m1, type = 'Regression', minManifests = 3)
x

## End(Not run)
</code></pre>

<hr>
<h2 id='umxFitIndices'>Get additional fit-indices for a model with umxFitIndices</h2><span id='topic+umxFitIndices'></span>

<h3>Description</h3>

<p>Computes a variety of fit indices. Originated in this thread: <a href="http://openmx.ssri.psu.edu/thread/765">http://openmx.ssri.psu.edu/thread/765</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxFitIndices(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxFitIndices_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel</a></code> for which you want fit indices.</p>
</td></tr>
<tr><td><code id="umxFitIndices_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to <code><a href="OpenMx.html#topic+summary.MxModel">summary.MxModel</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: This function is currently not robust across multi-group designs or
definition variables. It is designed to provide residual-based fit indices
(SRMR, CRMR, SMAR, CMAR, etc.) and less-often reported fit indices where
Reviewer 2 wants something other than CFA/TLI/RMSEA.
</p>
<p>Fit information reported includes:
</p>
<p><strong>Model characteristics</strong>:
numObs, estimated parameters,
observed statistics, observed summary statistics,
-2*log(Likelihood), degrees of freedom
</p>
<p><strong>Chi-squared test</strong>:
Chi, ChiDoF, p (of Chi), ChiPerDoF,
</p>
<p><strong>Noncentrality-based indices</strong>:
RMSEA, RMSEACI, RMSEANull, RMSEAClose (p value), independenceRMSEA,
NCP, NCPCI, F0, F0CI, Mc (aka NCI, MFI)
</p>
<p><strong>Comparative fit indices</strong>:
TLI (aka NNFI), CFI, IFI, PRATIO, PCFI
</p>
<p><strong>Residual-based indices</strong>:
RMR, SRMR, SRMR_mplus, CRMR,
MAR, SMAR, SMAR_mplus, CMAR
</p>
<p><strong>Information-theory criteria</strong> (computed using chi-square or -2LL; df or parameters penalties)
AIC, AICc, BIC, SABIC, CAIC, BCC
ECVI, ECVICI, MECVI, MECVICI
</p>
<p><strong>LISREL and other early fit indices</strong> (we recommend not reporting these)
GFI, AGFI, PGFI, GH, NFI, PNFI, RFI
</p>
<p>Want more? <em>Open an Issue</em> at <a href="https://github.com/tbates/umx/issues">GitHub</a>.
</p>


<h3>Value</h3>

<p>List of fit statistics
</p>


<h3>Author(s)</h3>

<p>Brenton M. Wiernik, Athanassios Protopapas, Paolo Ghisletta, Markus Brauer
</p>


<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxRotate">umxRotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umx)
data(demoOneFactor)
latents = c("G")
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor",
	data = mxData(cov(demoOneFactor), type = "cov", numObs = 500),
	umxPath(latents, to = manifests),
	umxPath(var = manifests),
	umxPath(var = latents, fixedAt = 1)
)
umxFitIndices(m1)

# And with raw data
m2 = umxRAM("m1", data = demoOneFactor,
	umxPath(latents, to = manifests),
	umxPath(v.m. = manifests),
	umxPath(v1m0 = latents)
)
umxFitIndices(m1, refModels = mxRefModels(m2, run = TRUE))

## End(Not run)

</code></pre>

<hr>
<h2 id='umxFixAll'>umxFixAll: Fix all free parameters</h2><span id='topic+umxFixAll'></span>

<h3>Description</h3>

<p>Fix all free parameters in a model using omxGetParameters()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxFixAll(model, name = "_fixed", run = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxFixAll_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> within which to fix free parameters</p>
</td></tr>
<tr><td><code id="umxFixAll_+3A_name">name</code></td>
<td>
<p>optional new name for the model. if you begin with a _ it will be made a suffix</p>
</td></tr>
<tr><td><code id="umxFixAll_+3A_run">run</code></td>
<td>
<p>whether to fix and re-run the model, or just return it (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxFixAll_+3A_verbose">verbose</code></td>
<td>
<p>whether to mention how many paths were fixed (default is FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> the fixed <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("OneFactor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
m2 = umxFixAll(m1, run = TRUE, verbose = TRUE)
mxCompare(m1, m2)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxGetLatents'>Get the latentVars from a RAM model</h2><span id='topic+umxGetLatents'></span>

<h3>Description</h3>

<p>Get the latentVars from a RAM model, optionally targeting a submodel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGetLatents(model, targetModel = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGetLatents_+3A_model">model</code></td>
<td>
<p>a <code><a href="#topic+umxRAM">umxRAM()</a></code></p>
</td></tr>
<tr><td><code id="umxGetLatents_+3A_targetmodel">targetModel</code></td>
<td>
<p>name of the model to extract from</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> variables
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxGetManifests">umxGetManifests()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>, <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>
</p>
</li></ul>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umx)
# Create two sets of data in which X &amp; Y correlate ~ .4 in both datasets.
manifests = c("x", "y")
tmp = umx_make_TwinData(nMZpairs = 100, nDZpairs = 150, 
		AA = 0, CC = .4, EE = .6, varNames = manifests)

grp1   = tmp[tmp$zygosity == "MZ", manifests]
g1Data = mxData(cov(grp1), type = "cov", numObs = nrow(grp1), means=umx_means(grp1))

grp2   = tmp[tmp$zygosity == "DZ", manifests]
g2Data = mxData(cov(grp2), type = "cov", numObs = nrow(grp2), means=umx_means(grp2))
# Model 1 (could add autoRun = FALSE if you don't want to run this as it is being built)
m1 = umxRAM("m1", data = g1Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels = c("Var_x", "Resid_y_grp1")),
	umxPath(means = manifests, labels = c("Mean_x", "Mean_y"))
)

# Model 2
m2 = umxRAM("m2", data = g2Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels=c("Var_x", "Resid_y_grp2")),
	umxPath(means = manifests, labels=c("Mean_x", "Mean_y"))
)

m3 = umxSuperModel('top', m1, m2)
umxGetLatents(m3)
umxGetLatents(m3, targetModel = "m1")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxGetManifests'>Get the manifestVars from a RAM model</h2><span id='topic+umxGetManifests'></span>

<h3>Description</h3>

<p>Get the latentVars from a RAM model, optionally targeting a submodel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGetManifests(model, targetModel = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGetManifests_+3A_model">model</code></td>
<td>
<p>a <code><a href="#topic+umxRAM">umxRAM()</a></code></p>
</td></tr>
<tr><td><code id="umxGetManifests_+3A_targetmodel">targetModel</code></td>
<td>
<p>name of the model to extract from</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> variables
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxGetManifests">umxGetManifests()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>, <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>
</p>
</li></ul>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umx)
# Create two sets of data in which X &amp; Y correlate ~ .4 in both datasets.
manifests = c("x", "y")
tmp = umx_make_TwinData(nMZpairs = 100, nDZpairs = 150, 
		AA = 0, CC = .4, EE = .6, varNames = manifests)

grp1   = tmp[tmp$zygosity == "MZ", manifests]
g1Data = mxData(cov(grp1), type = "cov", numObs = nrow(grp1), means=umx_means(grp1))

grp2   = tmp[tmp$zygosity == "DZ", manifests]
g2Data = mxData(cov(grp2), type = "cov", numObs = nrow(grp2), means=umx_means(grp2))
# Model 1 (could add autoRun = FALSE if you don't want to run this as it is being built)
m1 = umxRAM("m1", data = g1Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels = c("Var_x", "Resid_y_grp1")),
	umxPath(means = manifests, labels = c("Mean_x", "Mean_y"))
)

# Model 2
m2 = umxRAM("m2", data = g2Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels=c("Var_x", "Resid_y_grp2")),
	umxPath(means = manifests, labels=c("Mean_x", "Mean_y"))
)

m3 = umxSuperModel('top', m1, m2)
umxGetManifests(m3)
umxGetManifests(m3, targetModel = "m1")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxGetModel'>Used to get a RAM submodel by name</h2><span id='topic+umxGetModel'></span>

<h3>Description</h3>

<p>Get any model from a RAM model, including submodels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGetModel(model, targetModel = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGetModel_+3A_model">model</code></td>
<td>
<p>a <code><a href="#topic+umxRAM">umxRAM()</a></code> model.</p>
</td></tr>
<tr><td><code id="umxGetModel_+3A_targetmodel">targetModel</code></td>
<td>
<p>name of the model to extract from</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> model
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxGetManifests">umxGetManifests()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>, <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>
</p>
</li></ul>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umx)
# Create two sets of data in which X &amp; Y correlate ~ .4 in both datasets.
manifests = c("x", "y")
tmp = umx_make_TwinData(nMZpairs = 100, nDZpairs = 150, 
		AA = 0, CC = .4, EE = .6, varNames = manifests)

grp1   = tmp[tmp$zygosity == "MZ", manifests]
g1Data = mxData(cov(grp1), type = "cov", numObs = nrow(grp1), means=umx_means(grp1))

grp2   = tmp[tmp$zygosity == "DZ", manifests]
g2Data = mxData(cov(grp2), type = "cov", numObs = nrow(grp2), means=umx_means(grp2))
# Model 1 (could add autoRun = FALSE if you don't want to run this as it is being built)
m1 = umxRAM("m1", data = g1Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels = c("Var_x", "Resid_y_grp1")),
	umxPath(means = manifests, labels = c("Mean_x", "Mean_y"))
)

# Model 2
m2 = umxRAM("m2", data = g2Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels = c("Var_x", "Resid_y_grp2")),
	umxPath(means = manifests, labels = c("Mean_x", "Mean_y"))
)

m3 = umxSuperModel('top', m1, m2)
umxGetModel(m3)
umxGetModel(m3, targetModel = "m1")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxGetParameters'>Get parameters from a model, with support for pattern matching!</h2><span id='topic+umxGetParameters'></span>

<h3>Description</h3>

<p>umxGetParameters retrieves parameter labels from a model, like <code><a href="OpenMx.html#topic+omxGetParameters">omxGetParameters()</a></code>.
However, it is supercharged with regular expressions, so you can get labels that match a pattern.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGetParameters(
  inputTarget,
  regex = NA,
  free = NA,
  fetch = c("labels", "values", "free", "lbound", "ubound", "all"),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGetParameters_+3A_inputtarget">inputTarget</code></td>
<td>
<p>An object to get parameters from: could be a RAM <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code></p>
</td></tr>
<tr><td><code id="umxGetParameters_+3A_regex">regex</code></td>
<td>
<p>A regular expression to filter the labels. Default (NA) returns all labels. If vector, treated as raw labels to find.</p>
</td></tr>
<tr><td><code id="umxGetParameters_+3A_free">free</code></td>
<td>
<p>A Boolean determining whether to return only free parameters.</p>
</td></tr>
<tr><td><code id="umxGetParameters_+3A_fetch">fetch</code></td>
<td>
<p>What to return: &quot;labels&quot; (default) or &quot;values&quot;, &quot;free&quot;, &quot;lbound&quot;, &quot;ubound&quot;, or &quot;all&quot;</p>
</td></tr>
<tr><td><code id="umxGetParameters_+3A_verbose">verbose</code></td>
<td>
<p>How much feedback to give</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In addition, if regex contains a vector, this is treated as a list of raw labels to search for,
and return if all are found.
<em>note</em>: To return all labels, just leave regex as is.
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="OpenMx.html#topic+omxGetParameters">omxGetParameters()</a></code>, <code><a href="#topic+parameters">parameters()</a></code>
</p>
<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxParameters">umxParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

# Show all parameters
umxGetParameters(m1)
umxGetParameters(m1, free = TRUE)  # Only free parameters
umxGetParameters(m1, free = FALSE) # Only fixed parameters
# Complex regex pattern
umxGetParameters(m1, regex = "x[1-3]_with_x[2-5]", free = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='umxGxE'>umxGxE: Implements ACE models with moderation of paths, e.g. by SES.</h2><span id='topic+umxGxE'></span>

<h3>Description</h3>

<p>Make a 2-group GxE (moderated ACE) model (Purcell, 2002). GxE interaction studies test the hypothesis that the strength
of genetic (or environmental) influence varies parametrically (usually linear effects on path estimates)
across levels of environment. umxGxE allows detecting,
testing, and visualizing  G xE (or C or E x E) interaction forms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGxE(
  name = "G_by_E",
  selDVs,
  selDefs,
  dzData,
  mzData,
  sep = NULL,
  data = NULL,
  zyg = "zygosity",
  digits = 3,
  lboundACE = NA,
  lboundM = NA,
  dropMissingDef = TRUE,
  dzAr = 0.5,
  dzCr = 1,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGxE_+3A_name">name</code></td>
<td>
<p>The name of the model (default= &quot;G_by_E&quot;)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_seldvs">selDVs</code></td>
<td>
<p>The dependent variable (e.g. &quot;IQ&quot;)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_seldefs">selDefs</code></td>
<td>
<p>The definition variable (e.g. &quot;SES&quot;)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe containing the Twin 1 and Twin 2 DV and moderator (4 columns)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe containing the Twin 1 and Twin 2 DV and moderator (4 columns)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_sep">sep</code></td>
<td>
<p>How to expand selDVs into full names, i.e., &quot;_T&quot; makes &quot;var&quot; -&gt; &quot;var_T1&quot; and &quot;var_T2&quot;</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_data">data</code></td>
<td>
<p>If provided, dzData and mzData are treated as valid levels of zyg to select() data sets (default = NULL)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_zyg">zyg</code></td>
<td>
<p>If data provided, this column is used to select rows by zygosity (Default = &quot;zygosity&quot;)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_digits">digits</code></td>
<td>
<p>Rounding precision for tables (default 3)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_lboundace">lboundACE</code></td>
<td>
<p>If not NA, then lbound the main effects at this value (default = NA, can help to set this to 0)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_lboundm">lboundM</code></td>
<td>
<p>If not NA, then lbound the moderator effects at this value (default = NA, can help to set this to 0)</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_dropmissingdef">dropMissingDef</code></td>
<td>
<p>Whether to automatically drop missing def var rows for the user (default = TRUE). You get a polite note.</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_autorun">autoRun</code></td>
<td>
<p>Optionally run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_tryhard">tryHard</code></td>
<td>
<p>Optionally tryHard to get the model to converge (Default = 'no'). &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;.</p>
</td></tr>
<tr><td><code id="umxGxE_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following figure the GxE model as a path diagram:
</p>
<p><img src="../help/figures/GxE.png" width=50% alt="Figure: GxE.png" />

</p>


<h3>Value</h3>


<ul>
<li><p> GxE <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Purcell, S. (2002). Variance components models for gene-environment interaction in twin analysis. <em>Twin Research</em>,
<strong>6</strong>, 554-571. <a href="https://doi.org/10.1375/twin.5.6.554">doi:10.1375/twin.5.6.554</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxGxE_window">umxGxE_window()</a></code>, <code><a href="#topic+umxReduce">umxReduce()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData) 
twinData$age1 = twinData$age2 = twinData$age
selDVs  = "bmi"
selDefs = "age"
mzData  = subset(twinData, zygosity == "MZFF")[1:100,]
dzData  = subset(twinData, zygosity == "DZFF")[1:100,]
m1 = umxGxE(selDVs= "bmi", selDefs= "age", sep= "", dzData= dzData, mzData= mzData, tryHard= "yes")

# Select the data on the fly with data= and zygosity levels
m1 = umxGxE(selDVs= "bmi", selDefs= "age", sep="", dzData= "DZFF", mzData= "MZFF", data= twinData)

# ===============================================================
# = example with Twins having different values of the moderator =
# ===============================================================

twinData$age1 = twinData$age2 = twinData$age
tmp = twinData
tmp$age2 = tmp$age2 +rnorm(n=length(tmp$age2))
selDVs  = "bmi"
selDefs = "age"
mzData = subset(tmp, zygosity == "MZFF")
dzData = subset(tmp, zygosity == "DZFF")
m1 = umxGxE(selDVs= "bmi", selDefs= "age", sep= "", dzData= dzData, mzData= mzData, tryHard= "yes")

# ====================================
# = Controlling output of umxSummary =
# ====================================
umxSummaryGxE(m1)
umxSummary(m1, location = "topright")
umxSummary(m1, separateGraphs = TRUE)

m2 = umxModify(m1, regex = "am_.*", comparison = TRUE, tryHard = "yes")

# umxReduce knows how to test all relevant hypotheses for GxE models,
# reporting these in a nice table.
umxReduce(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxGxE_window'>Implement the moving-window form of GxE analysis.</h2><span id='topic+umxGxE_window'></span>

<h3>Description</h3>

<p>Make a 2-group GxE (moderated ACE) model using LOSEM. In GxE interaction studies, typically,
the hypothesis that the strength of genetic influence varies parametrically (usually linear effects
on path estimates) across levels of environment. Of course, the function linking genetic influence
and context is not necessarily linear, but may react more steeply at the extremes, or take other, unknown forms.
To avoid obscuring the underlying shape of the interaction effect, local structural equation
modeling (LOSEM) may be used, and GxE_window implements this. LOSEM is a non-parametric,
estimating latent interaction effects across the range of a measured moderator using a
windowing function which is walked along the context dimension, and which weights subjects
near the center of the window highly relative to subjects far above or below the window center.
This allows detecting and visualizing arbitrary GxE (or CxE or ExE) interaction forms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGxE_window(
  selDVs = NULL,
  moderator = NULL,
  mzData = mzData,
  dzData = dzData,
  sep = NULL,
  weightCov = FALSE,
  target = NULL,
  width = 1,
  plotWindow = FALSE,
  tryHard = c("no", "yes", "ordinal", "search"),
  return = c("estimates", "last_model")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGxE_window_+3A_seldvs">selDVs</code></td>
<td>
<p>The dependent variables for T1 and T2, e.g. c(&quot;bmi_T1&quot;, &quot;bmi_T2&quot;)</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_moderator">moderator</code></td>
<td>
<p>The name of the moderator variable in the dataset e.g. &quot;age&quot;, &quot;SES&quot; etc.</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_mzdata">mzData</code></td>
<td>
<p>Dataframe containing the DV and moderator for MZ twins</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_dzdata">dzData</code></td>
<td>
<p>Dataframe containing the DV and moderator for DZ twins</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_sep">sep</code></td>
<td>
<p>(optional) separator, e.g. &quot;_T&quot; which will be used expand base names into full variable names:
e.g.: 'bmi' &ndash;&gt; c(&quot;bmi_T1&quot;, &quot;bmi_T2&quot;)</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_weightcov">weightCov</code></td>
<td>
<p>Whether to use cov.wt matrices or FIML default = FALSE, i.e., FIML</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_target">target</code></td>
<td>
<p>A user-selected list of moderator values to test (default = NULL = explore the full range)</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_width">width</code></td>
<td>
<p>An option to widen or narrow the window from its default (of 1)</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_plotwindow">plotWindow</code></td>
<td>
<p>whether to plot the data window.</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxGxE_window_+3A_return">return</code></td>
<td>
<p>whether to return the last model (useful for specifiedTargets) or the list of estimates (default = &quot;estimates&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Table of estimates of ACE along the moderator
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Hildebrandt, A., Wilhelm, O, &amp; Robitzsch, A. (2009)
Complementary and competing factor analytic approaches for the investigation
of measurement invariance. <em>Review of Psychology</em>, <strong>16</strong>, 87&ndash;107.
</p>
</li></ul>

<p>Briley, D.A., Harden, K.P., Bates, T.C., Tucker-Drob, E.M. (2015).
Nonparametric Estimates of Gene x Environment Interaction Using Local Structural Equation Modeling.
<em>Behavior Genetics</em>, <strong>45</strong>, 581-96. <a href="https://doi.org/10.1007/s10519-015-9732-8">doi:10.1007/s10519-015-9732-8</a> <a href="https://link.springer.com/article/10.1007/s10519-015-9732-8">https://link.springer.com/article/10.1007/s10519-015-9732-8</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+umxGxE">umxGxE()</a></code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umx);

# ==============================
# = 1. Open and clean the data =
# ==============================
# umxGxE_window takes a data.frame consisting of a moderator and two DV columns: one for each twin.
# The model assumes two groups (MZ and DZ). Moderator can't be missing
mod = "age" # The full name of the moderator column in the dataset
selDVs = c("bmi1", "bmi2") # The DV for twin 1 and twin 2
data(twinData) # Dataset of Australian twins, built into OpenMx
# The twinData consist of two cohorts: "younger" and "older".
# zygosity is a factor. levels =  MZFF, MZMM, DZFF, DZMM, DZOS.

# Delete missing moderator rows
twinData = twinData[!is.na(twinData[mod]), ]
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")

# ========================
# = 2. Run the analyses! =
# ========================
# Run and plot for specified windows (in this case just 1927)
umxGxE_window(selDVs = selDVs, moderator = mod, mzData = mzData, dzData = dzData, 
		target = 40, plotWindow = TRUE)

umxGxE_window(selDVs = "bmi", sep="", moderator = mod, mzData = mzData, dzData = dzData, 
		target = 40, plotWindow = TRUE, tryHard = "yes")

# Run with tryHard
umxGxE_window(selDVs = "bmi", sep="", moderator = "age", mzData = mzData, dzData = dzData)
umxGxE_window(selDVs="bmi", sep="", moderator="age", mzData=mzData, dzData=dzData, tryHard="yes")

# Run creating weighted covariance matrices (excludes missing data)
umxGxE_window(selDVs = "bmi", sep="", moderator= "age", mzData = mzData, dzData = dzData, 
		weightCov = TRUE)
# This example runs multiple target moderator values
mxGxE_window(selDVs = selDVs, moderator = mod, mzData = mzData, dzData = dzData, 
	target = c(39,40,50), plotWindow = TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxGxEbiv'>Purcell (2002) Bivariate GxE model: Suitable when twins differ on the moderator.</h2><span id='topic+umxGxEbiv'></span>

<h3>Description</h3>

<p>GxE interaction models test the hypothesis that the strength of genetic
and environmental influences vary parametrically across levels of a measured environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxGxEbiv(
  name = "GxEbiv",
  selDVs,
  selDefs,
  dzData,
  mzData,
  sep = NULL,
  lboundACE = 0,
  lboundM = NA,
  dropMissingDef = FALSE,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxGxEbiv_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to &quot;GxEbiv&quot;)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_seldvs">selDVs</code></td>
<td>
<p>The dependent variable (e.g. IQ)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_seldefs">selDefs</code></td>
<td>
<p>The definition variable (e.g. socioeconomic status)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe containing the Twin 1 and Twin 2 DV and moderator (4 columns)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe containing the Twin 1 and Twin 2 DV and moderator (4 columns)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_sep">sep</code></td>
<td>
<p>Expand variable base names, i.e., &quot;_T&quot; makes var -&gt; var_T1 and var_T2</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_lboundace">lboundACE</code></td>
<td>
<p>If !NA, then lbound the main effects at this value (default = NA)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_lboundm">lboundM</code></td>
<td>
<p>If !NA, then lbound the moderators at this value (default = NA)</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_dropmissingdef">dropMissingDef</code></td>
<td>
<p>Whether to automatically drop missing def var rows for the user (gives a warning) default = FALSE</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxGxEbiv_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Whereas univariate <code><a href="#topic+umxGxE">umxGxE()</a></code> models assume the twins share the moderator,
or have zero correlation on the moderator, <code><a href="#topic+umxGxEbiv">umxGxEbiv()</a></code> allows testing moderation in
cases where members of a twin pair differ on the moderator, (Purcell, 2002; van der Sluis et al., 2012).
</p>
<p>This is the same model we teach at Boulder.
</p>
<p>The following figure shows this bivariate GxE model as a path diagram (Twin 1 shown). Whereas
the univariate model incorporates the moderator in the means model, the bivariate model incorporates
the moderator as a first class variable, with its own ACE structure, shared pathways to the trait of interest,
and the ability to moderate both specific and shared A, C, and E, influences on the trait of interest.
</p>
<p><img src="../help/figures/GxEbiv.png" alt="GxEbiv.png" />
</p>
<p>Twin 1 and twin 2 A, C, and E latent traits are connected in the standard fashion, with the
covariance of the T1 and T2 latent genetic traits set to .5 for DZ and 1.0 for MZ pairs.
For the sake of clarity, C, and E paths are omitted here. These mirror those for A.
</p>


<h3>Value</h3>


<ul>
<li><p> GxEbiv <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Purcell, S. (2002). Variance components models for gene-environment interaction in twin analysis. <em>Twin Research</em>,
<strong>6</strong>, 554-571. <a href="https://doi.org/10.1375/twin.5.6.554">doi:10.1375/twin.5.6.554</a>.
</p>
</li>
<li><p> van der Sluis, S., Posthuma, D., &amp; Dolan, C. V. (2012). A note on false positives and
power in G x E modelling of twin data. <em>Behavior Genetics</em>,
<strong>42</strong>, 170-186. <a href="https://doi.org/10.1007/s10519-011-9480-3">doi:10.1007/s10519-011-9480-3</a>.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code>, <code><a href="#topic+umxReduce">umxReduce()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(twinData)
selDVs  = "wt"
selDefs = "ht"
df = umx_scale_wide_twin_data(twinData, varsToScale = c("ht", "wt"), sep = "")
mzData  = subset(df, zygosity %in%  c("MZFF", "MZMM"))
dzData  = subset(df, zygosity %in%  c("DZFF", "DZMM", "DZOS"))

## Not run: 
m1 = umxGxEbiv(selDVs = selDVs, selDefs = selDefs, 
	dzData = dzData, mzData = mzData, sep = "", dropMissingDef = TRUE)

# Plot Moderation
umxSummaryGxEbiv(m1)
umxSummary(m1, location = "topright")
umxSummary(m1, separateGraphs = FALSE)
m2 = umxModify(m1, update = c("cBeta2_r1c1", "eBeta1_r1c1", "eBeta2_r1c1"), comparison = TRUE)

# TODO: teach umxReduce to test all relevant hypotheses for umxGxEbiv
umxReduce(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxHetCor'>Create a matrix of correlations for variables of diverse types (binary, ordinal, continuous)</h2><span id='topic+umxHetCor'></span>

<h3>Description</h3>

<p><code>umxHetCor</code> is a helper to:
</p>

<ol>
<li><p> return just the correlations from John Fox's polycor::hetcor function
</p>
</li>
<li><p> If you give it a covariance matrix, return the nearest positive-definite correlation matrix.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>umxHetCor(
  data,
  ML = FALSE,
  use = c("pairwise.complete.obs", "complete.obs"),
  treatAllAsFactor = FALSE,
  verbose = FALSE,
  return = c("correlations", "hetcor object"),
  std.err = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxHetCor_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+data.frame">data.frame()</a></code> of columns for which to compute heterochoric correlations. OR an existing covariance matrix.</p>
</td></tr>
<tr><td><code id="umxHetCor_+3A_ml">ML</code></td>
<td>
<p>Whether to use Maximum likelihood computation of correlations (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxHetCor_+3A_use">use</code></td>
<td>
<p>How to handle missing data: Default= &quot;pairwise.complete.obs&quot;. Alternative =&quot;complete.obs&quot;.</p>
</td></tr>
<tr><td><code id="umxHetCor_+3A_treatallasfactor">treatAllAsFactor</code></td>
<td>
<p>Whether to treat all columns as factors, whether they are or not (Default = FALSE)</p>
</td></tr>
<tr><td><code id="umxHetCor_+3A_verbose">verbose</code></td>
<td>
<p>How much to tell the user about what was done.</p>
</td></tr>
<tr><td><code id="umxHetCor_+3A_return">return</code></td>
<td>
<p>Return just the correlations (default) or the hetcor object (contains, method, SEs etc.)</p>
</td></tr>
<tr><td><code id="umxHetCor_+3A_std.err">std.err</code></td>
<td>
<p>Compute the SEs? (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A matrix of correlations
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Data Functions: 
<code><a href="#topic+noNAs">noNAs</a>()</code>,
<code><a href="#topic+prolific_anonymize">prolific_anonymize</a>()</code>,
<code><a href="#topic+prolific_check_ID">prolific_check_ID</a>()</code>,
<code><a href="#topic+prolific_read_demog">prolific_read_demog</a>()</code>,
<code><a href="#topic+umxFactor">umxFactor</a>()</code>,
<code><a href="#topic+umx_as_numeric">umx_as_numeric</a>()</code>,
<code><a href="#topic+umx_cont_2_quantiles">umx_cont_2_quantiles</a>()</code>,
<code><a href="#topic+umx_lower2full">umx_lower2full</a>()</code>,
<code><a href="#topic+umx_make_MR_data">umx_make_MR_data</a>()</code>,
<code><a href="#topic+umx_make_TwinData">umx_make_TwinData</a>()</code>,
<code><a href="#topic+umx_make_fake_data">umx_make_fake_data</a>()</code>,
<code><a href="#topic+umx_make_raw_from_cov">umx_make_raw_from_cov</a>()</code>,
<code><a href="#topic+umx_merge_randomized_columns">umx_merge_randomized_columns</a>()</code>,
<code><a href="#topic+umx_polychoric">umx_polychoric</a>()</code>,
<code><a href="#topic+umx_polypairwise">umx_polypairwise</a>()</code>,
<code><a href="#topic+umx_polytriowise">umx_polytriowise</a>()</code>,
<code><a href="#topic+umx_read_lower">umx_read_lower</a>()</code>,
<code><a href="#topic+umx_rename">umx_rename</a>()</code>,
<code><a href="#topic+umx_reorder">umx_reorder</a>()</code>,
<code><a href="#topic+umx_score_scale">umx_score_scale</a>()</code>,
<code><a href="#topic+umx_select_valid">umx_select_valid</a>()</code>,
<code><a href="#topic+umx_stack">umx_stack</a>()</code>,
<code><a href="#topic+umx_strings2numeric">umx_strings2numeric</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>
<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>umxHetCor(mtcars[,c("mpg", "am")])
umxHetCor(mtcars[,c("mpg", "am")], treatAllAsFactor = TRUE, verbose = TRUE)
</code></pre>

<hr>
<h2 id='umxIP'>umxIP: Build and run an Independent Pathway twin model</h2><span id='topic+umxIP'></span>

<h3>Description</h3>

<p>Make a 2-group Independent Pathway twin model.
</p>
<p>The independent-pathway model  (aka &quot;biometric model&quot; (McArdle and Goldsmith, 1990) proposes that <code>A</code>,
<code>C</code>, and <code>E</code> components act directly on the manifest or measured phenotypes. This contrasts with
the <code><a href="#topic+umxCP">umxCP()</a></code> model, in which these influences are collected on a hypothesized or latent causal
variable, which is manifested in the measured phenotypes.
</p>
<p>The following figure shows the IP model diagrammatically:
</p>
<p><img src="../help/figures/IP.svg" width=50% alt="Figure: IP model" />

</p>
<p>As can be seen, each phenotype also by default has A, C, and E influences specific to that phenotype.
</p>
<p>Features of the model include the ability to include add more one set of independent pathways, different numbers
of pathways for a, c, and e, as well the ability to use ordinal data, and different fit functions, e.g. WLS.
</p>
<p><strong>note</strong>: The function <code><a href="#topic+umx_set_optimization_options">umx_set_optimization_options()</a></code> allows users to see and set <code>mvnRelEps</code> and <code>mvnMaxPointsA</code>
mvnRelEps defaults to .005. For ordinal models, you might find that '0.01' works better.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxIP(
  name = "IP",
  selDVs,
  dzData,
  mzData,
  sep = NULL,
  nFac = c(a = 1, c = 1, e = 1),
  data = NULL,
  zyg = "zygosity",
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  allContinuousMethod = c("cumulants", "marginals"),
  dzAr = 0.5,
  dzCr = 1,
  correlatedA = FALSE,
  numObsDZ = NULL,
  numObsMZ = NULL,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL,
  equateMeans = TRUE,
  weightVar = NULL,
  addStd = TRUE,
  addCI = TRUE,
  freeLowerA = FALSE,
  freeLowerC = FALSE,
  freeLowerE = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxIP_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to &quot;IP&quot;).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_seldvs">selDVs</code></td>
<td>
<p>The base names of the variables to model. note: Omit suffixes - just &quot;dep&quot; not c(&quot;dep_T1&quot;, &quot;dep_T2&quot;)</p>
</td></tr>
<tr><td><code id="umxIP_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_sep">sep</code></td>
<td>
<p>The suffix for twin 1 and twin 2. e.g. selDVs= &quot;dep&quot;, sep= &quot;_T&quot; -&gt; c(&quot;dep_T1&quot;, &quot;dep_T2&quot;)</p>
</td></tr>
<tr><td><code id="umxIP_+3A_nfac">nFac</code></td>
<td>
<p>How many common factors for a, c, and e. If one number is given, applies to all three.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_data">data</code></td>
<td>
<p>If provided, dzData and mzData are treated as levels of zyg to select() MZ and DZ data sets (default = NULL)</p>
</td></tr>
<tr><td><code id="umxIP_+3A_zyg">zyg</code></td>
<td>
<p>If data provided, this column is used to select rows by zygosity (Default = &quot;zygosity&quot;)</p>
</td></tr>
<tr><td><code id="umxIP_+3A_type">type</code></td>
<td>
<p>Analysis method one of c(&quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;)</p>
</td></tr>
<tr><td><code id="umxIP_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_correlateda">correlatedA</code></td>
<td>
<p>Whether factors are allowed to correlate (not implemented yet: FALSE).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_numobsdz">numObsDZ</code></td>
<td>
<p>= For cov data, the number of DZ pairs.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_numobsmz">numObsMZ</code></td>
<td>
<p>= For cov data, the number of MZ pairs.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run and return the model (default), or just to create and return without running.</p>
</td></tr>
<tr><td><code id="umxIP_+3A_tryhard">tryHard</code></td>
<td>
<p>Whether to tryHard (default 'no' uses normal mxRun). options: &quot;mxTryHard&quot;, &quot;mxTryHardOrdinal&quot;, or &quot;mxTryHardWideSearch&quot;</p>
</td></tr>
<tr><td><code id="umxIP_+3A_optimizer">optimizer</code></td>
<td>
<p>optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_weightvar">weightVar</code></td>
<td>
<p>If a weighting variable is provided, a vector objective will be used to weight the data. (default = NULL).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_addstd">addStd</code></td>
<td>
<p>Whether to add algebras for a standardized model (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_addci">addCI</code></td>
<td>
<p>Whether to add CIs (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_freelowera">freeLowerA</code></td>
<td>
<p>ignore: Whether to leave the lower triangle of A free (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_freelowerc">freeLowerC</code></td>
<td>
<p>ignore: Whether to leave the lower triangle of C free (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxIP_+3A_freelowere">freeLowerE</code></td>
<td>
<p>ignore: Whether to leave the lower triangle of E free (default = FALSE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Like the <code><a href="#topic+umxACE">umxACE()</a></code> model, the IP model decomposes phenotypic variance
into additive genetic (A), unique environmental (E) and, optionally, either
common or shared-environment (C) or
non-additive genetic effects (D).
</p>
<p>Unlike the Cholesky, these factors do not act directly on the phenotype. Instead latent A,
C, and E influences impact on one or more latent common factors which, in turn, account for
variance in the phenotypes (see Figure).
</p>
<p><strong>Data Input</strong>
Currently, <code>umxIP</code> accepts only raw data. This may change in future versions. You can
choose other fit functions, e.g. WLS.
</p>
<p><strong>Ordinal Data</strong>
</p>
<p>In an important capability, the model transparently handles ordinal (binary or multi-level
ordered factor data) inputs, and can handle mixtures of continuous, binary, and ordinal
data in any combination.
</p>
<p><strong>Additional features</strong>
</p>
<p><code>umxIP</code> supports varying the DZ genetic association (defaulting to .5)
to allow exploring assortative mating effects, as well as varying the DZ &ldquo;C&rdquo; factor
from 1 (the default for modeling family-level effects shared 100% by twins in a pair),
to .25 to model dominance effects.
</p>
<p><strong>Matrices and Labels in IP model</strong>
</p>
<p>A good way to see which matrices are used in umxIP is to run an example model and plot it.
</p>
<p>All the shared matrices are in the model &quot;top&quot;.
</p>
<p>Matrices <code>as</code>, <code>cs</code>, and <code>es</code> contain the path loadings specific to each variable on their diagonals.
</p>
<p>To see the 'as' values, you can simply execute:
</p>
<p>m1$top#as$values
</p>
<p>m1$top#as$labels
</p>
<p>m1$top#as$free
</p>
<p>Labels relevant to modifying the specific loadings take the form &quot;as_r1c1&quot;, &quot;as_r2c2&quot; etc.
</p>
<p>The independent-pathway loadings on the manifests are in matrices <code>a_ip</code>, <code>c_ip</code>, <code>e_ip</code>.
</p>
<p>Less commonly-modified matrices are the mean matrix <code>expMean</code>.
This has 1 row, and the columns are laid out for each variable
for twin 1, followed by each variable for twin 2.
</p>
<p>So, in a model where the means for twin 1 and twin 2 had been equated
(set = to T1), you could make them independent again with this line:
</p>
<p><code>m1$top$expMean$labels[1,4:6] = c("expMean_r1c4", "expMean_r1c5", "expMean_r1c6")</code>
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Kendler, K. S., Heath, A. C., Martin, N. G., &amp; Eaves, L. J. (1987). Symptoms of anxiety and symptoms of depression.
Same genes, different environments? <em>Archives of General Psychiatry</em>, <strong>44</strong>, 451-457. <a href="https://doi.org/10.1001/archpsyc.1987.01800170073010">doi:10.1001/archpsyc.1987.01800170073010</a>.
</p>
</li>
<li><p> McArdle, J. J., &amp; Goldsmith, H. H. (1990). Alternative common factor models for multivariate biometric analyses.
<em>Behavior Genetics</em>, <strong>20</strong>, 569-608. <a href="https://doi.org/10.1007/BF01065873">doi:10.1007/BF01065873</a>.
</p>
</li>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code>, <code><a href="#topic+umxCP">umxCP()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff","fc","qol","hap","sat","AD") # These will be expanded into "gff_T1" "gff_T2" etc.
m1 =    umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData)

# WLS example: Use "marginals" method to enable all continuous data with missingness.
m3 = umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData, 
	type = "DWLS", allContinuousMethod='marginals')
# omit missing to enable default WLS method to work on all continuous data
dzD = na.omit(dzData[, tvars(selDVs, "_T")])
mzD = na.omit(dzData[, tvars(selDVs, "_T")])
m4 = umxIP(selDVs = selDVs, sep = "_T", dzData = dzD, mzData = mzD, type = "DWLS")

# ====================================================================
# = Try with a non-default number of a, c, and e independent factors =
# ====================================================================
nFac = c(a = 2, c = 1, e = 1)
m2 = umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData, nFac = nFac, 
	tryHard = "yes")
umxCompare(m1, m2)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxJiggle'>umxJiggle</h2><span id='topic+umxJiggle'></span>

<h3>Description</h3>

<p>umxJiggle takes values in a matrix and jiggles them
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxJiggle(matrixIn, mean = 0, sd = 0.1, dontTouch = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxJiggle_+3A_matrixin">matrixIn</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code> to jiggle the values of</p>
</td></tr>
<tr><td><code id="umxJiggle_+3A_mean">mean</code></td>
<td>
<p>the mean value to add to each value</p>
</td></tr>
<tr><td><code id="umxJiggle_+3A_sd">sd</code></td>
<td>
<p>the sd of the jiggle noise</p>
</td></tr>
<tr><td><code id="umxJiggle_+3A_donttouch">dontTouch</code></td>
<td>
<p>A value, which, if found, will be left as-is (defaults to 0)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
mat1 = umxJiggle(mat1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxLav2RAM'>Convert lavaan string to a umxRAM model</h2><span id='topic+umxLav2RAM'></span>

<h3>Description</h3>

<p>Takes a lavaan syntax string and creates the matching one or more <code><a href="#topic+umxRAM">umxRAM()</a></code> models.
</p>
<p>If data are provided, a <code><a href="#topic+umxRAM">umxRAM()</a></code> model is returned.
</p>
<p>If more than one group is found, a <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code> is returned.
</p>
<p>This function is at the alpha quality stage, and <strong>should be expected to have bugs</strong>.
Several features are not yet supported. Let me know if you would like them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxLav2RAM(
  model = NA,
  data = "auto",
  group = NULL,
  group.equal = NULL,
  name = NA,
  lavaanMode = c("sem", "lavaan"),
  std.lv = FALSE,
  suffix = "",
  comparison = TRUE,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  allContinuousMethod = c("cumulants", "marginals"),
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  verbose = FALSE,
  optimizer = NULL,
  std = FALSE,
  printTab = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxLav2RAM_+3A_model">model</code></td>
<td>
<p>A lavaan syntax string, e.g. &quot;A~~B&quot;</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_data">data</code></td>
<td>
<p>Data to add to model (defaults to auto, which is just sketch mode)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_group">group</code></td>
<td>
<p>= Column to use for multi-group (default = NULL)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_group.equal">group.equal</code></td>
<td>
<p>= what to equate across groups. Default (NULL) means no equates. See details for what we might implement in future.</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_name">name</code></td>
<td>
<p>Model name (can also add name in # commented first line)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_lavaanmode">lavaanMode</code></td>
<td>
<p>Auto-magical path settings for cfa/sem (default) or no-defaults (&quot;lavaan&quot;)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_std.lv">std.lv</code></td>
<td>
<p>= FALSE Whether to set var of latents to 1 (default FALSE). nb. Toggles fix first.</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_suffix">suffix</code></td>
<td>
<p>String to append to each label (useful if model will be used in a multi-group model)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_comparison">comparison</code></td>
<td>
<p>Compare the new model to the old (if updating an existing model: default = TRUE)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_type">type</code></td>
<td>
<p>One of &quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_verbose">verbose</code></td>
<td>
<p>Whether to tell the user what latents and manifests were created etc. (Default = FALSE)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_optimizer">optimizer</code></td>
<td>
<p>optionally set the optimizer (default NULL does nothing)</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_std">std</code></td>
<td>
<p>Whether to print estimates. Defaults to FALSE (&quot;raw&quot;), TRUE = &quot;std&quot;, for no parameter table use NULL.</p>
</td></tr>
<tr><td><code id="umxLav2RAM_+3A_printtab">printTab</code></td>
<td>
<p>= TRUE (more for debugging)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses the defaults of <code>lavaan::sem</code>
</p>

<ul>
<li><p> int.ov.free     = TRUE
</p>
</li>
<li><p> int.lv.free     = FALSE
</p>
</li>
<li><p> auto.fix.first  = TRUE (unless std.lv = TRUE)
</p>
</li>
<li><p> auto.fix.single = TRUE
</p>
</li>
<li><p> auto.var        = TRUE
</p>
</li>
<li><p> auto.cov.lv.x   = TRUE
</p>
</li>
<li><p> auto.th         = TRUE
</p>
</li>
<li><p> auto.delta      = TRUE
</p>
</li>
<li><p> auto.cov.y      = TRUE
</p>
</li>
<li><p> fixed.x         = FALSE (not standard in <code>lavaan::sem</code>, but needed for RAM)
</p>
</li></ul>

<p>Lavaan is well documented. For quick reference, some common symbols in lavaan strings are:
</p>

<table>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> lav     </td><td style="text-align: left;"> Mplus   </td><td style="text-align: left;"> sem   </td><td style="text-align: left;"> Action                          	                      </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;">  A =~ B </td><td style="text-align: left;"> A by B  </td><td style="text-align: left;">       </td><td style="text-align: left;"> A (Latent) is measured by B	                          </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;">  A ~  B </td><td style="text-align: left;"> A on B  </td><td style="text-align: left;"> A&lt;- B </td><td style="text-align: left;"> A "is regressed on" (&lt;- ) B   	                      </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;">  A ~~ B </td><td style="text-align: left;"> A with B</td><td style="text-align: left;"> A&lt;-&gt;B </td><td style="text-align: left;"> A covaries with B	                                      </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;">  A ~  1 </td><td style="text-align: left;"> [A]    </td><td style="text-align: left;">       </td><td style="text-align: left;"> A has mean	                                          </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;">  A := B </td><td style="text-align: left;">         </td><td style="text-align: left;">       </td><td style="text-align: left;"> A is defined by B (see <code><a href="OpenMx.html#topic+mxAlgebra">OpenMx::mxAlgebra()</a></code>)	          </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;">  A == B </td><td style="text-align: left;">         </td><td style="text-align: left;">       </td><td style="text-align: left;"> A is constrained == to B (see <code><a href="OpenMx.html#topic+mxConstraint">OpenMx::mxConstraint()</a></code> )
</td>
</tr>

</table>


<table>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> <code style="white-space: pre;">&#8288;=~&#8288;</code>   </td><td style="text-align: left;"> lhs (Latent) is manifested by rhs                   </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> <code>~</code>    </td><td style="text-align: left;"> lhs "is regressed on" (&lt;- ) rhs                     </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> <code style="white-space: pre;">&#8288;~~&#8288;</code>   </td><td style="text-align: left;"> lhs covaries with rhs                               </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> <code>~ 1</code>  </td><td style="text-align: left;"> lhs has mean                                        </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> <code style="white-space: pre;">&#8288;:=&#8288;</code>   </td><td style="text-align: left;"> lhs is defined by rhs (see <code><a href="OpenMx.html#topic+mxAlgebra">OpenMx::mxAlgebra()</a></code>)   </td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: left;"> <code>==</code>   </td><td style="text-align: left;"> lhs is constrained == to rhs (see <code><a href="OpenMx.html#topic+mxConstraint">OpenMx::mxConstraint()</a></code> )
</td>
</tr>

</table>

<p><strong>Naming of multiple groups</strong>
</p>
<p>When multiple groups are found the groups are named <code>name_grouplevel</code>
White space is replaced with &quot;_&quot; and illegal characters are replaced with &quot;x&quot;
</p>
<p><em>note</em>: Options for  group.equal. In future, we might implement (but have not as yet):
</p>

<ol>
<li><p> c(&quot;loadings&quot;
</p>
</li>
<li><p> &quot;intercepts&quot;
</p>
</li>
<li><p> &quot;means&quot;
</p>
</li>
<li><p> &quot;regressions&quot;
</p>
</li>
<li><p> &quot;residuals&quot;
</p>
</li>
<li><p> &quot;covariances&quot;
</p>
</li></ol>



<h3>Value</h3>


<ul>
<li><p> list of <code><a href="#topic+umxPath">umxPath()</a></code>s
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxRAM2Lav">umxRAM2Lav()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# auto-data, print table, return umxRAM model
m1 = umxLav2RAM("y ~ x", printTab= TRUE)

lav = "y ~ x1 + 2.4*x2 + x3"
tmp = umxLav2RAM(lav, data = "auto", printTab= FALSE)

# Add labels to parameters, e.g. "x3_loading" as a loading for x3-&gt;x1
tmp = umxLav2RAM("x1 ~ x3_loading*x3")
umx_print(tmp$A$labels)
# |   |x1       |x3         |
# |:--|:--------|:----------|
# |x1 |x1_to_x1 |x3_loading |
# |x3 |x1_to_x3 |x3_to_x3   |

# Fix values, e.g. x2 -&gt; y fixed at 2.4
tmp = umxLav2RAM("y ~ x1 + 2.4*x2; s =~ 0*y11 + 1*y12 + 2*y13 + 3*y14")

tmp = umxLav2RAM("L =~ X1 + X2; L ~ Y")
plot(tmp, min=c("L", "Y"))

# Factor model showing auto-addition of correlations among exogenous latents
# and auto-residuals on manifests
data("HS.ability.data", package = "OpenMx")

cov(HS.ability.data[, c("visual"  , "cubes"   , "flags")])
cov(HS.ability.data[, c("paragrap", "sentence", "wordm")])
cov(HS.ability.data[, c("addition", "counting", "straight")])

HS = "spatial =~ visual   + cubes    + flags
      verbal  =~ paragrap + sentence + wordm
      speed   =~ addition + counting + straight"

m1 = umxRAM(HS, data = umx_scale(HS.ability.data))

# Multiple groups
m1 = umxRAM(HS, data = umx_scale(HS.ability.data), group = "school")

# More examples

lav = " # Moderated mediation
gnt ~ a*cb
INT ~ b1*gnt + b2*cn + b3*cngn + c*cb

indirect := a*b1
direct := c

ab3 := a * b3
loCN := a * b1 + ab3 * -0.5
hiCN := a * b1 + ab3 * 0.5
"
tmp = umxRAM(lav)
# plot showing ability to influence layout with max min same groupings
plot(tmp, max = c("cb", "cn", "cngn"), same = "gnt", min= "INT")

# Algebra: e.g. b1^2
m1 = umxRAM("x1~b1*x2; B1_sq := b1^2", data = demoOneFactor)
m1$B1_sq$result # = 0.47

# Model with constraints and labeled parameters
lav = "
y ~ b1*x1 + b2*x2 + b3*x3
# constraints
b1 == (b2 + b3)^2
b1 &gt; exp(b2 + b3)"

tmp = umxLav2RAM(lav)

namedModel = " 	# my name
	y ~x"
m1 = umxRAM(namedModel) 

# Formative factor
# lavaanify("f5 &lt;~ z1 + z2 + z3 + z4")

## End(Not run)

</code></pre>

<hr>
<h2 id='umxMatrix'>Make a mxMatrix with automatic labels. Also takes name as the first parameter for more readable code.</h2><span id='topic+umxMatrix'></span>

<h3>Description</h3>

<p>umxMatrix is a wrapper for mxMatrix which labels cells buy default, and has the name parameter first in order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxMatrix(
  name = NA,
  type = "Full",
  nrow = NA,
  ncol = NA,
  free = FALSE,
  values = NA,
  labels = TRUE,
  lbound = NA,
  ubound = NA,
  byrow = getOption("mxByrow"),
  baseName = NA,
  dimnames = NA,
  condenseSlots = getOption("mxCondenseMatrixSlots"),
  ...,
  joinKey = as.character(NA),
  joinModel = as.character(NA),
  jiggle = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxMatrix_+3A_name">name</code></td>
<td>
<p>The name of the matrix (Default = NA). Note the different order compared to mxMatrix!</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_type">type</code></td>
<td>
<p>The type of the matrix (Default = &quot;Full&quot;)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_nrow">nrow</code></td>
<td>
<p>Number of rows in the matrix: Must be set</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns in the matrix: Must be set</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_free">free</code></td>
<td>
<p>Whether cells are free (Default FALSE)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_values">values</code></td>
<td>
<p>The values of the matrix (Default NA)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_labels">labels</code></td>
<td>
<p>Either whether to label the matrix (default TRUE), OR a vector of labels to apply.</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_lbound">lbound</code></td>
<td>
<p>Lower bounds on cells (Defaults to NA)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_ubound">ubound</code></td>
<td>
<p>Upper bounds on cells (Defaults to NA)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_byrow">byrow</code></td>
<td>
<p>Whether to fill the matrix down columns or across rows first (Default = getOption('mxByrow')</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_basename">baseName</code></td>
<td>
<p>Set to override the default (which is to use the matrix name as the prefix).</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_dimnames">dimnames</code></td>
<td>
<p>NA</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_condenseslots">condenseSlots</code></td>
<td>
<p>Whether to save memory by NULLing out unused matrix elements, like labels, ubound etc. Default = getOption('mxCondenseMatrixSlots')</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_...">...</code></td>
<td>
<p>Additional parameters (!! not currently supported by umxMatrix)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_joinkey">joinKey</code></td>
<td>
<p>See mxMatrix documentation: Defaults to as.character(NA)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_joinmodel">joinModel</code></td>
<td>
<p>See mxMatrix documentation: Defaults to as.character(NA)</p>
</td></tr>
<tr><td><code id="umxMatrix_+3A_jiggle">jiggle</code></td>
<td>
<p>= NA passed to xmuLabel to jiggle start values (default does nothing)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner()</a></code>, <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>, <code><a href="#topic+xmuLabel">xmuLabel()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other Core Model Building Functions: 
<code><a href="#topic+umxModify">umxModify</a>()</code>,
<code><a href="#topic+umxPath">umxPath</a>()</code>,
<code><a href="#topic+umxRAM">umxRAM</a>()</code>,
<code><a href="#topic+umxSuperModel">umxSuperModel</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ==================================================================================
# = 1. Showing how name is first parameter, and how cells are labelled by default. =
# ==================================================================================
umxMatrix("test", "Full", 2, 2)$labels
#      [,1]        [,2]
# [1,] "test_r1c1" "test_r1c2"
# [2,] "test_r2c1" "test_r2c2"

# ===========================================================
# = 2. Over-ride default (matrix name) as prefix for labels =
# ===========================================================
umxMatrix("test", "Full", 2, 2, baseName = "bob")$labels # bob_r1c1


# ==========================================
# = 3. User-provided labels are left as-is =
# ==========================================
umxMatrix("foo", "Lower", nrow=2, ncol=2, labels= c(NA, "beta1", NA))
#      [,1]    [,2]
# [1,] NA      NA  
# [2,] "beta1" NA  


## End(Not run)

</code></pre>

<hr>
<h2 id='umxMatrixFree'>Sets labeled matrix cells to free</h2><span id='topic+umxMatrixFree'></span>

<h3>Description</h3>

<p>In simulation studies, it is often necessary to rewrite the matrices while testing
alternative specifications. This can become very tedious with increasing
number of distinct specifications. This tool injects changes into umxMatrix so
that this tasks gets more manageable. First, it sets byRow by default. Second,
it infers the number of rows automatically. The user needs only passing ncol.
Finally and most importantly this function disables auto-labeling, and whenever
a label is set, that cell position will be freed. It is required to pass
a matrix of labels, as well as a label name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxMatrixFree(
  name = name,
  nrow = NULL,
  ncol = NA,
  free = FALSE,
  values = NA,
  labels = labels,
  byrow = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxMatrixFree_+3A_name">name</code></td>
<td>
<p>The name of the matrix: Must be set</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_nrow">nrow</code></td>
<td>
<p>Number of rows in the matrix (Optional)</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns in the matrix (Required)</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_free">free</code></td>
<td>
<p>Whether cells are free (Default FALSE)</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_values">values</code></td>
<td>
<p>The values of the matrix (Default NA)</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_labels">labels</code></td>
<td>
<p>The labels of the matrix (Default NA)</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_byrow">byrow</code></td>
<td>
<p>Default for byrow (TRUE)</p>
</td></tr>
<tr><td><code id="umxMatrixFree_+3A_...">...</code></td>
<td>
<p>Accepts all other arguments from <code><a href="#topic+umxMatrix">umxMatrix()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxMatrix">umxMatrix()</a></code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

umxMatrixFree('E', type='Symm',  ncol = 3,
  labels =c("eb2",NA,NA,
             NA,"es2",NA,
            NA,NA,NA),
  values=c(.2,0,0,
           0,.2,0,
           0,0,0))
           
 # Will return a umxMatrix free at the eb2 and es2 positions.

## End(Not run)
</code></pre>

<hr>
<h2 id='umxMI'>Report modifications which would improve fit.</h2><span id='topic+umxMI'></span>

<h3>Description</h3>

<p>This function uses the mechanical modification-indices approach to detect single paths which, if added
or dropped, would improve fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxMI(
  model = NA,
  matrices = NA,
  full = TRUE,
  numInd = NA,
  typeToShow = "both",
  decreasing = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxMI_+3A_model">model</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> for which to report modification indices</p>
</td></tr>
<tr><td><code id="umxMI_+3A_matrices">matrices</code></td>
<td>
<p>which matrices to test. The default (NA) will test A &amp; S for RAM models</p>
</td></tr>
<tr><td><code id="umxMI_+3A_full">full</code></td>
<td>
<p>Change in fit allowing all parameters to move. If FALSE only the parameter under test can move.</p>
</td></tr>
<tr><td><code id="umxMI_+3A_numind">numInd</code></td>
<td>
<p>How many modifications to report. Use -1 for all. Default (NA) will report all over 6.63 (p = .01)</p>
</td></tr>
<tr><td><code id="umxMI_+3A_typetoshow">typeToShow</code></td>
<td>
<p>Whether to shown additions or deletions (default = &quot;both&quot;)</p>
</td></tr>
<tr><td><code id="umxMI_+3A_decreasing">decreasing</code></td>
<td>
<p>How to sort (default = TRUE, decreasing)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Notes:
</p>

<ol>
<li><p> Runs much faster with full = FALSE (but this does not allow the model to re-fit around the newly-
freed parameter).
</p>
</li>
<li><p> Compared to mxMI, this function returns top changes, and also suppresses the run message.
</p>
</li>
<li><p> Finally, of course: see the requirements for (legitimate) post-hoc modeling in <code><a href="OpenMx.html#topic+mxMI">mxMI()</a></code>
You are almost certainly doing better science when testing competing models rather than modifying a model to fit.
</p>
</li></ol>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxMI">mxMI()</a></code>
</p>
</li></ul>

<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxCompare">umxCompare</a>()</code>,
<code><a href="#topic+umxEquate">umxEquate</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSetParameters">umxSetParameters</a>()</code>,
<code><a href="#topic+umxSummary">umxSummary</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
umxMI(m1, full = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxModel'>Catches users typing umxModel instead of umxRAM.</h2><span id='topic+umxModel'></span>

<h3>Description</h3>

<p>Catches a common typo, moving from mxModel to umx.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxModel(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxModel_+3A_...">...</code></td>
<td>
<p>Anything. We're just going to throw an error.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxRAM">umxRAM()</a></code>, <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umxModel()

## End(Not run)
</code></pre>

<hr>
<h2 id='umxModelNames'>Return names of models found within a model</h2><span id='topic+umxModelNames'></span>

<h3>Description</h3>

<p><code>umxModelNames</code> returns the names of each model contained in the model provided to it
(optionally excluding the out model itself).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxModelNames(model, includeOuterModelName = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxModelNames_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to search for model names.</p>
</td></tr>
<tr><td><code id="umxModelNames_+3A_includeoutermodelname">includeOuterModelName</code></td>
<td>
<p>FALSE</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> All models names
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxRename">mxRename()</a></code>, <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff", "fc", "qol")
m1 = umxCP(selDVs= selDVs, nFac= 1, dzData= dzData, mzData= mzData, sep= "_T", autoRun= TRUE)
m2 = mxRename(m1, "model2")
umxModelNames(m1) # "top" "MZ" "DZ"
umxModelNames(m2) # "top" "MZ" "DZ"

super = umxSuperModel("myModel", m1, m2, autoRun = TRUE)
umxModelNames(super)

plot(super$CP1fac)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxModify'>umxModify: Add, set, or drop model paths by label.</h2><span id='topic+umxModify'></span>

<h3>Description</h3>

<p>umxModify allows you to modify, re-run and summarize an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>, all in one line of script.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxModify(
  lastFit,
  update = NULL,
  regex = FALSE,
  free = FALSE,
  value = 0,
  newlabels = NULL,
  freeToStart = NA,
  name = NULL,
  comparison = FALSE,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  master = NULL,
  intervals = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxModify_+3A_lastfit">lastFit</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> you wish to update and run.</p>
</td></tr>
<tr><td><code id="umxModify_+3A_update">update</code></td>
<td>
<p>What to update before re-running. Can be a list of labels, a regular expression (set regex = TRUE) or an object such as mxCI etc.</p>
</td></tr>
<tr><td><code id="umxModify_+3A_regex">regex</code></td>
<td>
<p>Whether or not update is a regular expression (default FALSE). If you provide a string, it overrides the contents of update, and sets regex to TRUE.</p>
</td></tr>
<tr><td><code id="umxModify_+3A_free">free</code></td>
<td>
<p>The state to set &quot;free&quot; to for the parameters whose labels you specify (defaults to free = FALSE, i.e., fixed)</p>
</td></tr>
<tr><td><code id="umxModify_+3A_value">value</code></td>
<td>
<p>The value to set the parameters whose labels you specify too (defaults to 0)</p>
</td></tr>
<tr><td><code id="umxModify_+3A_newlabels">newlabels</code></td>
<td>
<p>If not NULL, used as a replacement set of labels (can be regular expression). value and free are ignored!</p>
</td></tr>
<tr><td><code id="umxModify_+3A_freetostart">freeToStart</code></td>
<td>
<p>Whether to update parameters based on their current free-state. free = c(TRUE, FALSE, NA), (defaults to NA - i.e, not checked)</p>
</td></tr>
<tr><td><code id="umxModify_+3A_name">name</code></td>
<td>
<p>The name for the new model</p>
</td></tr>
<tr><td><code id="umxModify_+3A_comparison">comparison</code></td>
<td>
<p>Whether to run umxCompare() on the new and old models.</p>
</td></tr>
<tr><td><code id="umxModify_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxModify_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxModify_+3A_master">master</code></td>
<td>
<p>If you set master, then the update labels will be equated to these (i.e. replaced by them).</p>
</td></tr>
<tr><td><code id="umxModify_+3A_intervals">intervals</code></td>
<td>
<p>Whether to run confidence intervals (see <code><a href="OpenMx.html#topic+mxRun">mxRun()</a></code>)</p>
</td></tr>
<tr><td><code id="umxModify_+3A_verbose">verbose</code></td>
<td>
<p>How much feedback to give</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can add paths, or other model elements, set path values (default is 0), or replace labels.
As an example, this one-liner drops a path labelled &quot;Cs&quot;, and returns the updated model:
</p>
<p><code>fit2 = umxModify(fit1, update = "Cs", name = "newModelName", comparison = TRUE)</code>
</p>
<p>Regular expressions are a powerful feature: they let you drop collections of paths by matching patterns
for instance, this would match labels containing either &quot;Cs&quot; or &quot;Cr&quot;:
</p>
<div class="sourceCode R"><pre>fit2 = umxModify(fit1, regex = "C\[sr\]", name = "drop_Cs_and_Cr", comparison = TRUE)
</pre></div>
<p>You may find it easier to be more explicit. Like this:
</p>
<div class="sourceCode R"><pre>fit2 = umxSetParameters(fit1, labels = c("Cs", "Cr"), values = 0, free = FALSE, name = "newName")
fit2 = mxRun(fit2)
summary(fit2)
</pre></div>
<p><em>Note</em>: A (minor) limitation is that you cannot simultaneously set value to 0
AND relabel cells (because the default value is 0, so it is ignored when using newlabels).
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Core Model Building Functions: 
<code><a href="#topic+umxMatrix">umxMatrix</a>()</code>,
<code><a href="#topic+umxPath">umxPath</a>()</code>,
<code><a href="#topic+umxRAM">umxRAM</a>()</code>,
<code><a href="#topic+umxSuperModel">umxSuperModel</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
# First we'll just build a 1-factor model
umx_set_optimizer("SLSQP")
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)

# 1. Drop the path to x1 (also updating the name so it's
#    self-explanatory, and get a fit comparison
m2 = umxModify(m1, update = "G_to_x1", name = "drop_X1", comparison = TRUE)

# 2. Add the path back (setting free = TRUE)
m2 = umxModify(m1, update = "G_to_x1", free= TRUE, name = "addback_X1", comparison = TRUE)
# 3. Fix a value at a non-zero value
m3 = umxModify(m1, update = "G_to_x1", value = .35, name = "fix_G_x1_at_35", comp = TRUE)
# You can add objects to models. For instance this would add a path (overwriting the existing one)
# (thanks Johannes!)
m3 = umxModify(m1, umxPath("G", with = "x1"), name= "addedPath")

# Use regular expression to drop multiple paths: e.g. G to x3, x4, x5
m3 = umxModify(m1, regex = "^G_to_x[3-5]", name = "tried_hard", comp = TRUE, tryHard="yes")

# Same, but don't autoRun
m2 = umxModify(m1, regex  = "^G_to_x[3-5]", name = "no_G_to_x3_5", autoRun = FALSE) 

# Re-write a label
newLabel = "A_rose_by_any_other_name"
newModelName = "model_doth_smell_as_sweet"
m2 = umxModify(m1, update = "G_to_x1", newlabels= newLabel, name = newModelName, comparison = TRUE)
# Change labels in 2 places
labsToUpdate = c("G_to_x1", "G_to_x2")
newLabel = "G_to_1_or_2"
m2 = umxModify(m1, update = labsToUpdate, newlabels= newLabel, name = "equated", comparison = TRUE)

# Advanced!
# Regular expressions let you use pieces of the old names in creating new ones!
searchString = "G_to_x([0-9])"
newLabel = "loading_for_path\\1" # use value in regex group 1
m2 = umxModify(m1, regex = searchString, newlabels= newLabel, name = "grep", comparison = TRUE)

## End(Not run) # end dontrun

</code></pre>

<hr>
<h2 id='umxMRDoC'>Extends Mendelian randomization with the twin design to test evidence of causality</h2><span id='topic+umxMRDoC'></span>

<h3>Description</h3>

<p>Testing causal claims is often difficult due to an inability to conduct experimental randomization of traits and
situations to people. When twins are available, even when measured on a single occasion, the pattern of cross-twin
cross-trait correlations can (given distinguishable modes of inheritance for the two traits) falsify causal hypotheses.
</p>
<p><code>umxMRDoC</code> implements a 2-group model to form latent variables for each of two traits, and allows testing whether
trait 1 causes trait 2, vice-versa, or even reciprocal causation. This is robust to several types of confounding
due to the instrumental variable approach included in the model.
</p>
<p>This function applies both the MRDoC model and the MRDoC2 model depending on how many PRSs are passed as arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxMRDoC(
  pheno,
  prss,
  mzData = NULL,
  dzData = NULL,
  data = NULL,
  zyg = NULL,
  sep = "_T",
  summary = !umx_set_silent(silent = TRUE),
  name = NULL,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxMRDoC_+3A_pheno">pheno</code></td>
<td>
<p>Phenotypes of interest, order matters (&quot;exposure&quot;, &quot;outcome&quot;)</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_prss">prss</code></td>
<td>
<p>Polygenic score(s). If a single one is passed MRDoC is run, MRDoC2 otherwise.</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_data">data</code></td>
<td>
<p>= NULL If building the MZ and DZ datasets internally from a complete data set.</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_zyg">zyg</code></td>
<td>
<p>= &quot;zygosity&quot; (for the data= method of using this function)</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_sep">sep</code></td>
<td>
<p>The separator in twin variable names, default = &quot;_T&quot;, e.g. &quot;dep_T1&quot;.</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_summary">summary</code></td>
<td>
<p>Optionally show a summary.</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to either &quot;MRDoC&quot; or &quot;MRDoC2).</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxMRDoC_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> of subclass MxModelMRDoC
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Minica CC, Dolan CV, Boomsma DI, et al. (2018) Extending Causality Tests with Genetic Instruments: An Integration of Mendelian Randomization with the Classical Twin Design. Behavior Genetics 48(4): 337–349. <a href="https://doi.org/10.1007/s10519-018-9904-4">doi:10.1007/s10519-018-9904-4</a>
</p>
</li></ul>


<ul>
<li><p> McGue, M., Osler, M., &amp; Christensen, K. (2010). Causal Inference and Observational Research: The Utility of Twins. <em>Perspectives on Psychological Science</em>, <strong>5</strong>, 546-556. <a href="https://doi.org/10.1177/1745691610383511">doi:10.1177/1745691610383511</a>
</p>
</li>
<li><p> Castro-de-Araujo LFS, Singh M, Zhou Y, et al. (2022) MR-DoC2: Bidirectional Causal Modeling with Instrumental Variables and Data from Relatives. Behavior Genetics. <a href="https://doi.org/10.1007/s10519-022-10122-x">doi:10.1007/s10519-022-10122-x</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# ================
# = 1. Load Data =
# ================
data(docData)
mzData  = subset(docData, zygosity %in% c("MZFF", "MZMM"))
dzData  = subset(docData, zygosity %in% c("DZFF", "DZMM"))

# ============================
# = 2. Make a MRDoC2 model   =
# ============================
out = umxMRDoC(mzData = mzData, dzData = dzData,  
pheno = c("varA1", "varA2"), prss = c("varB1", "varA3") )

## End(Not run)
</code></pre>

<hr>
<h2 id='umxParameters'>Display path estimates from a model, filtering by name and value.</h2><span id='topic+umxParameters'></span><span id='topic+parameters'></span>

<h3>Description</h3>

<p>Often you want to see the estimates from a model, and often you don't want all of them.
<code><a href="#topic+umxParameters">umxParameters()</a></code> helps in this case, allowing you to select parameters matching a name filter,
and also to only show parameters above or below a certain value.
</p>
<p>If pattern is a vector, each regular expression is matched, and all unique matches to the whole vector are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxParameters(
  x,
  thresh = c("all", "above", "below", "&gt;", "&lt;", "NS", "sig"),
  b = NULL,
  pattern = ".*",
  std = FALSE,
  digits = 2
)

parameters(
  x,
  thresh = c("all", "above", "below", "&gt;", "&lt;", "NS", "sig"),
  b = NULL,
  pattern = ".*",
  std = FALSE,
  digits = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxParameters_+3A_x">x</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> or model summary from which to report parameter estimates.</p>
</td></tr>
<tr><td><code id="umxParameters_+3A_thresh">thresh</code></td>
<td>
<p>optional: Filter out estimates 'below' or 'above' a certain value (default = &quot;all&quot;).</p>
</td></tr>
<tr><td><code id="umxParameters_+3A_b">b</code></td>
<td>
<p>Combine with thresh to set a minimum or maximum for which estimates to show.</p>
</td></tr>
<tr><td><code id="umxParameters_+3A_pattern">pattern</code></td>
<td>
<p>Optional string to match in the parameter names. Default '.*' matches all. <code><a href="base.html#topic+regex">regex()</a></code> allowed!</p>
</td></tr>
<tr><td><code id="umxParameters_+3A_std">std</code></td>
<td>
<p>Standardize output: NOT IMPLEMENTED YET</p>
</td></tr>
<tr><td><code id="umxParameters_+3A_digits">digits</code></td>
<td>
<p>Round to how many digits (2 = default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is on my TODO list to implement filtering by significance, and to add standardizing.
</p>


<h3>Value</h3>


<ul>
<li><p> list of matching parameters, filtered by name and value
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxGetParameters">umxGetParameters()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code>, <code><a href="#topic+namez">namez()</a></code>
</p>
</li></ul>

<p>Other Reporting Functions: 
<code><a href="#topic+umxAPA">umxAPA</a>()</code>,
<code><a href="#topic+umxFactorScores">umxFactorScores</a>()</code>,
<code><a href="#topic+umxGetLatents">umxGetLatents</a>()</code>,
<code><a href="#topic+umxGetManifests">umxGetManifests</a>()</code>,
<code><a href="#topic+umxGetModel">umxGetModel</a>()</code>,
<code><a href="#topic+umxGetParameters">umxGetParameters</a>()</code>,
<code><a href="#topic+umx_aggregate">umx_aggregate</a>()</code>,
<code><a href="#topic+umx_time">umx_time</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("OneFactor", data = demoOneFactor,
	umxPath(from = "G", to = manifests), # factor loadings
	umxPath(v.m. = manifests),           # residual variance
	umxPath(v1m0 = "G")                  # standardized latent
)
# Parameters with values below .1
umxParameters(m1, "below", .1)
# Parameters with values above .5
umxParameters(m1, "above", .5)
# Parameters with values below .1 and containing "_to_" in their label
umxParameters(m1, "below", .1, "_to_")

## End(Not run)

</code></pre>

<hr>
<h2 id='umxParan'>A wrapper to make paran easier to use.
Just automates applying <code><a href="stats.html#topic+complete.cases">complete.cases()</a></code></h2><span id='topic+umxParan'></span>

<h3>Description</h3>

<p>A wrapper to make paran easier to use.
Just automates applying <code><a href="stats.html#topic+complete.cases">complete.cases()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxParan(df, cols = NA, graph = TRUE, mapStrings = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxParan_+3A_df">df</code></td>
<td>
<p>The df (just the relevant columns)</p>
</td></tr>
<tr><td><code id="umxParan_+3A_cols">cols</code></td>
<td>
<p>(optional) list of columns (default = use all)</p>
</td></tr>
<tr><td><code id="umxParan_+3A_graph">graph</code></td>
<td>
<p>Whether to graph.</p>
</td></tr>
<tr><td><code id="umxParan_+3A_mapstrings">mapStrings</code></td>
<td>
<p>optional mapping if cols are strings</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> nothing
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxWeightedAIC">umxWeightedAIC</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(psych)
library(psychTools)
data(bfi)
umxParan(bfi[, paste0("A", 1:5)])
umxParan(bfi, cols= paste0("A", 1:5))
# umxParan(bfi, paste0("AB", 1))
</code></pre>

<hr>
<h2 id='umxPath'>Easier (and powerful) specification of paths in SEM.</h2><span id='topic+umxPath'></span>

<h3>Description</h3>

<p>This function is used to easily and compactly specify paths in models. In addition to
<code>from</code> and <code>to</code>, it adds specialised parameters for variances (var), two headed paths (with) and means (mean).
There are also new terms to describe fixing values: <code>fixedAt</code> and <code>fixFirst</code>.
To give a couple of the most common, time-saving examples:
</p>

<ul>
<li> <p><code>umxPath("A", with = "B",  fixedAt = 1)</code>
</p>
</li>
<li> <p><code>umxPath(var = c("A", "B"),  fixedAt = 1)</code>
</p>
</li>
<li> <p><code>umxPath(v.m. = manifests)</code>
</p>
</li>
<li> <p><code>umxPath(v1m0 = latents)</code>
</p>
</li>
<li> <p><code>umxPath(v1m0 = latents)</code>
</p>
</li>
<li> <p><code>umxPath(means = manifests)</code>
</p>
</li>
<li> <p><code>umxPath(fromEach = c('A',"B","C"), to = c("y1","y2"))</code>
</p>
</li>
<li> <p><code>umxPath(unique.bivariate = c('A',"B","C"))</code>
</p>
</li>
<li> <p><code>umxPath("A", to = c("B","C","D"),  firstAt = 1)</code>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>umxPath(
  from = NULL,
  to = NULL,
  with = NULL,
  var = NULL,
  cov = NULL,
  means = NULL,
  v1m0 = NULL,
  v.m. = NULL,
  v0m0 = NULL,
  v.m0 = NULL,
  v0m. = NULL,
  fixedAt = NULL,
  freeAt = NULL,
  firstAt = NULL,
  unique.bivariate = NULL,
  unique.pairs = NULL,
  fromEach = NULL,
  forms = NULL,
  Cholesky = NULL,
  defn = NULL,
  connect = c("single", "all.pairs", "all.bivariate", "unique.pairs", "unique.bivariate"),
  arrows = 1,
  free = TRUE,
  values = NA,
  labels = NA,
  lbound = NA,
  ubound = NA,
  hasMeans = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPath_+3A_from">from</code></td>
<td>
<p>One or more source variables e.g &quot;A&quot; or c(&quot;A&quot;,&quot;B&quot;)</p>
</td></tr>
<tr><td><code id="umxPath_+3A_to">to</code></td>
<td>
<p>One or more target variables for one-headed paths, e.g &quot;A&quot; or c(&quot;A&quot;,&quot;B&quot;).</p>
</td></tr>
<tr><td><code id="umxPath_+3A_with">with</code></td>
<td>
<p>2-headed path &lt;&ndash;&gt; from 'from' to 'with'.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_var">var</code></td>
<td>
<p>Equivalent to setting 'from' and 'arrows' = 2. nb: from, to, and with must be left empty.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_cov">cov</code></td>
<td>
<p>Convenience to allow 2 variables to covary (equivalent to 'from' and 'with'). nb: leave from, to, etc. empty</p>
</td></tr>
<tr><td><code id="umxPath_+3A_means">means</code></td>
<td>
<p>equivalent to &quot;from = 'one', to = x. nb: from, to, with and var must be left empty (their default).</p>
</td></tr>
<tr><td><code id="umxPath_+3A_v1m0">v1m0</code></td>
<td>
<p>variance of 1 and mean of zero in one call.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_v.m.">v.m.</code></td>
<td>
<p>variance and mean, both free.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_v0m0">v0m0</code></td>
<td>
<p>variance and mean, both fixed at zero.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_v.m0">v.m0</code></td>
<td>
<p>variance free, mean fixed at zero.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_v0m.">v0m.</code></td>
<td>
<p>variance fixed at 0, mean free.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_fixedat">fixedAt</code></td>
<td>
<p>Equivalent to setting &quot;free = FALSE, values = fixedAt&quot;</p>
</td></tr>
<tr><td><code id="umxPath_+3A_freeat">freeAt</code></td>
<td>
<p>Equivalent to setting &quot;free = TRUE, values = freeAt&quot;</p>
</td></tr>
<tr><td><code id="umxPath_+3A_firstat">firstAt</code></td>
<td>
<p>First path is fixed at this value (free is ignored: warning if other than a single TRUE)</p>
</td></tr>
<tr><td><code id="umxPath_+3A_unique.bivariate">unique.bivariate</code></td>
<td>
<p>equivalent to setting from, and &quot;connect = &quot;unique.bivariate&quot;, arrows = 2&quot;.
nb: from, to, and with must be left empty (their default)</p>
</td></tr>
<tr><td><code id="umxPath_+3A_unique.pairs">unique.pairs</code></td>
<td>
<p>equivalent to setting &quot;connect = &quot;unique.pairs&quot;, arrows = 2&quot; (don't use from, to, or with)</p>
</td></tr>
<tr><td><code id="umxPath_+3A_fromeach">fromEach</code></td>
<td>
<p>Like all.bivariate, but with one head arrows. 'to' can be set.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_forms">forms</code></td>
<td>
<p>Build a formative variable. 'from' variables form the latent.
Latent variance is fixed at 0. Loading of path 1 is fixed at 1. unique.bivariate between 'from' variables.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_cholesky">Cholesky</code></td>
<td>
<p>Treat <strong>Cholesky</strong> variables as latent and <strong>to</strong> as measured, and connect as in an ACE model.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_defn">defn</code></td>
<td>
<p>Implements a definition variable as a latent with zero variance &amp; mean and labeled 'data.defVar'</p>
</td></tr>
<tr><td><code id="umxPath_+3A_connect">connect</code></td>
<td>
<p>as in mxPath - nb: from and to must also be set.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_arrows">arrows</code></td>
<td>
<p>as in mxPath - nb: from and to must also be set.</p>
</td></tr>
<tr><td><code id="umxPath_+3A_free">free</code></td>
<td>
<p>whether the value is free to be optimised</p>
</td></tr>
<tr><td><code id="umxPath_+3A_values">values</code></td>
<td>
<p>default value list</p>
</td></tr>
<tr><td><code id="umxPath_+3A_labels">labels</code></td>
<td>
<p>labels for each path</p>
</td></tr>
<tr><td><code id="umxPath_+3A_lbound">lbound</code></td>
<td>
<p>lower bounds for each path value</p>
</td></tr>
<tr><td><code id="umxPath_+3A_ubound">ubound</code></td>
<td>
<p>upper bounds for each path value</p>
</td></tr>
<tr><td><code id="umxPath_+3A_hasmeans">hasMeans</code></td>
<td>
<p>Used in 'forms' case to know whether the data have means or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>umxPath</code> introduces the following new words to your path-defining vocabulary: <code>with</code>, <code>var</code>, <code>cov</code>, <code>means</code>, <code>v1m0</code>,
<code>v0m0</code>, <code>v.m0</code>, <code>v.m</code>, <code>fixedAt</code>, <code>freeAt</code>, <code>firstAt</code>, <code>unique.bivariate</code>, <code>unique.pairs</code>, <code>fromEach</code>, <code>Cholesky</code>, <code>defn</code>, <code>forms</code>.
</p>
<p><code>with</code> creates covariances (2-headed paths):
<code>umxPath(A, with = B)</code>
</p>
<p>Specify a variance for A with
<code>umxPath(var = "A")</code>.
</p>
<p>Of course you can use vectors anywhere:
<code>umxPath(var = c('N','E', 'O'))</code>
</p>
<p>To specify a mean, you just say:
<code>umxPath(mean = "A")</code>, which is equivalent to <code>mxPath(from = "one", to = "A")</code>.
</p>
<p>To fix a path at a value, you can say:
<code>umxPath(var = "A", fixedAt = 1)</code>
</p>
<p>The common task of creating a variable with variance fixed at 1 and mean at 0 is done thus:
<code>umxPath(v1m0 = "A")</code>
</p>
<p>For free variance and means use:
<code>umxPath(v.m. = "A")</code>
</p>
<p><code>umxPath</code> exposes <code>unique.bivariate</code> and <code>unique.pairs</code>, So to create paths A&lt;-&gt;A, B&lt;-&gt;B,
and A-&gt;B, you would say:
<code>umxPath(unique.pairs = c('A',"B"))</code>
</p>
<p>To create paths A&lt;-&gt;B, B&lt;-&gt;C, and A&lt;-&gt;C, you would say:
<code>umxPath(unique.bivariate = c('A',"B","C"))</code>
</p>
<p>Creates one-headed arrows on the all.bivariate pattern
<code>umxPath(fromEach = c('A',"B","C"))</code>
</p>
<p>Setting up a latent trait, you can scale with a fixed first path thus:
</p>
<p><code>umxPath("A", to = c("B","C","D"),  firstAt = 1)</code>
</p>
<p>To create Cholesky-pattern connections:
</p>
<p><code style="white-space: pre;">&#8288;umxPath(Cholesky = c("A1", "A2"), to c("var1", "var2"))&#8288;</code>
</p>


<h3>Value</h3>


<ul>
<li><p> 1 or more <code><a href="OpenMx.html#topic+mxPath">mxPath()</a></code>s
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxPath">mxPath()</a></code>
</p>
</li></ul>

<p>Other Core Model Building Functions: 
<code><a href="#topic+umxMatrix">umxMatrix</a>()</code>,
<code><a href="#topic+umxModify">umxModify</a>()</code>,
<code><a href="#topic+umxRAM">umxRAM</a>()</code>,
<code><a href="#topic+umxSuperModel">umxSuperModel</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# ==========================================
# = Examples of each path type, and option =
# ==========================================

umxPath("A", to = "B") # One-headed path from A to B
umxPath("A", to = "B", fixedAt = 1) # same, with value fixed @1
umxPath("A", to = c("B", "C"), fixedAt = 1:2) # same, with more than 1 value
umxPath("A", to = c("B","C"), firstAt = 1) # Fix only the first path, others free
umxPath(var = "A") # Give a variance to A
umxPath(var = "A", fixedAt = 1) # Give A variance, fixed at 1
umxPath(means = c("A","B")) # Create a means model for A: from = "one", to = "A"
umxPath(v1m0 = "A") # Give "A" variance and a mean, fixed at 1 and 0 respectively
umxPath(v.m. = "A") # Give "A" variance and a mean, leaving both free.
umxPath(v0m0 = "W", label = c(NA, "data.W"))
umxPath("A", with = "B") # using with: same as "to = B, arrows = 2"
umxPath("A", with = "B", fixedAt = .5) # 2-head path fixed at .5
umxPath("A", with = c("B", "C"), firstAt = 1) # first covariance fixed at 1
umxPath(cov = c("A", "B"))  # Covariance A &lt;-&gt; B
umxPath(defn = "mpg") # create latent called def_mpg, with var = 1 and label = "data.mpg"
umxPath(fromEach = c('a','b'), to = c('c','d')) # a-&gt;c, a&lt;-&gt;d, b&lt;-&gt;c, b&lt;-&gt;d
umxPath(unique.bivariate = c('a','b','c')) # bivariate paths a&lt;-&gt;b, a&lt;-&gt;c, b&lt;-&gt;c etc.
umxPath(unique.pairs = letters[1:3]) # all distinct pairs: a&lt;-&gt;a, a&lt;-&gt;b, a&lt;-&gt;c, b&lt;-&gt;b, etc.
umxPath(Cholesky = c("A1","A2"), to = c("m1", "m2")) # Cholesky

## Not run: 
# A worked example
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type= "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1.0)
)
umxSummary(m1, std = TRUE)
require(umx)


# ====================
# = Cholesky example =
# ====================
# ======================================================================
# = 3-factor Cholesky (A component of a 5-variable 3-factor ACE model) =
# ======================================================================
latents   = paste0("A", 1:3)
manifests = names(demoOneFactor)
m1 = umxRAM("Chol", data = demoOneFactor, type = "cov",
	umxPath(Cholesky = latents, to = manifests),
	umxPath(var = manifests),
	umxPath(var = latents, fixedAt = 1)
)
plot(m1, splines= FALSE)

# ======================================================================
# = Definition variable example. for a RAM model                       =
# = def vars are instantiated as dummy latents with data on the "mean" = 
# ======================================================================
library(umx); libs("MASS") # for mvrnorm()
# 1. Create Data
N = 500 # size of each group
Sigma  = matrix(c(1,.5,.5,1),2,2) # cov (.5)
group1 = MASS::mvrnorm(N, c(1,2), Sigma)
group2 = MASS::mvrnorm(N, c(0,0), Sigma)
# rbind groups and name cols "x" and "y"
xy = rbind(group1, group2)
dimnames(xy)[2]= list(c("x", "y"))

# Create a definition variable for group status
groupID = rep(c(1,0), each = N) 
df = data.frame(xy, groupID = groupID)

# Make the model with a definition variable on means
m1 = umxRAM("Def Means", data = df,
	umxPath(v.m. = c("x","y")),
	umxPath("x", with = "y"),
 # create a unit latent called "def_groupID" with data "data.groupID"
	umxPath(defn = "groupID"),
 # Add it to the x and y means
	umxPath("def_groupID", to = c("x", "y"))
)
plot(m1)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlot'>Quickly plot y ~ x with a regression line and R^2, and nice labels.</h2><span id='topic+umxPlot'></span>

<h3>Description</h3>

<p>Want a figure for your paper or presentation but not the work of combining <code><a href="ggplot2.html#topic+ggplot">ggplot2::ggplot()</a></code>,
<code><a href="ggplot2.html#topic+geom_smooth">ggplot2::geom_smooth()</a></code> and <code>method</code> options, plus <code><a href="ggplot2.html#topic+geom_point">ggplot2::geom_point()</a></code>. Organizing <code><a href="ggplot2.html#topic+labs">ggplot2::labs()</a></code> and its
<code>x</code>, <code>y</code>, and <code>title</code> components. Adding your preferred theme like <code><a href="ggplot2.html#topic+ggtheme">ggplot2::theme_gray()</a></code>, plus recalling
for <code><a href="cowplot.html#topic+draw_label">cowplot::draw_label()</a></code>, and/or <code><a href="ggplot2.html#topic+annotate">ggplot2::annotate()</a></code> to draw math-enabled labels on the plot, as well
as the required <code><a href="base.html#topic+bquote">bquote()</a></code>, and extracting the relevant fit statistics from <code><a href="stats.html#topic+lm">lm()</a></code> and the subsidiary tasks
of <code><a href="stats.html#topic+reformulate">reformulate()</a></code> programmatic variables?
</p>
<p><code>umxPlot</code> just takes <code>y ~ x</code> (or &quot;x&quot; and &quot;y&quot; as strings), and gives you a nicely labelled plot, with a fitted line, the
<em>R</em><sup>2</sup> so readers can see how well this fitted. It knows how to put Greek symbols like <code class="reqn">beta</code> into axes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlot(
  x,
  y = NULL,
  data,
  xlab = x,
  ylab = y,
  title = paste0(y, " as a function of ", x),
  fitx = NA,
  fity = NA,
  geom_point = TRUE,
  method = c("lm", "auto", "loess", "glm", "gam"),
  family = c("gaussian", "binomial", "Gamma", "inverse", "poisson", "quasi",
    "quasibinomial", "quasipoisson")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlot_+3A_x">x</code></td>
<td>
<p>formula or (alternatively) x as string</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_y">y</code></td>
<td>
<p>variable as string.</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_data">data</code></td>
<td>
<p>The data for the graph.</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_xlab">xlab</code></td>
<td>
<p>X-axis label (default y).</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_ylab">ylab</code></td>
<td>
<p>Y-axis label (default y).</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_title">title</code></td>
<td>
<p>Graph title. Default =  paste0(y, &quot; as a function of &quot;, x)</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_fitx">fitx</code></td>
<td>
<p>x location for the fit summary (default 1).</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_fity">fity</code></td>
<td>
<p>y location for the fit summary (default 2).</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_geom_point">geom_point</code></td>
<td>
<p>show points? (TRUE)</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_method">method</code></td>
<td>
<p>Method for fitting curve (default = lm)</p>
</td></tr>
<tr><td><code id="umxPlot_+3A_family">family</code></td>
<td>
<p>for glm default = &quot;gaussian&quot;</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> plot you can edit.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="ggplot2.html#topic+qplot">ggplot2::qplot()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mtcars)
umxPlot(mpg ~ wt, data = mtcars, fitx = 2, fity = 10)
umxPlot(x = "wt", y = "mpg", mtcars, fitx = 2, fity = 10)
</code></pre>

<hr>
<h2 id='umxPlotACE'>Make a graphical display of an ACE model</h2><span id='topic+umxPlotACE'></span><span id='topic+plot.MxModelACE'></span>

<h3>Description</h3>

<p>plot method for <code><a href="#topic+umxACE">umxACE()</a></code> models. Make a graphical display of an ACE model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotACE(
  x = NA,
  file = "name",
  digits = 2,
  means = FALSE,
  std = TRUE,
  strip_zero = TRUE,
  showFixed = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotACE_+3A_x">x</code></td>
<td>
<p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to plot (created by umxACE in order to inherit the MxModelACE class)</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (default is 2)</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (default is FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (default is TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_showfixed">showFixed</code></td>
<td>
<p>Whether too draw fixed parameters.</p>
</td></tr>
<tr><td><code id="umxPlotACE_+3A_...">...</code></td>
<td>
<p>Additional (optional) parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxACE">umxACE()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACE("plotACE example", selDVs = "bmi", dzData = dzData, mzData = mzData, sep = "")
plot(m1, std = FALSE) # don't standardize

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotACEcov'>Make a graphical display of an ACE model with covariates.</h2><span id='topic+umxPlotACEcov'></span><span id='topic+plot.MxModelACEcov'></span>

<h3>Description</h3>

<p>Make a graphical display of an ACE model with covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotACEcov(
  x = NA,
  file = "name",
  digits = 2,
  means = FALSE,
  std = TRUE,
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotACEcov_+3A_x">x</code></td>
<td>
<p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to plot (created by umxACE in order to inherit the MxModelACE class)</p>
</td></tr>
<tr><td><code id="umxPlotACEcov_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotACEcov_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (default is 2)</p>
</td></tr>
<tr><td><code id="umxPlotACEcov_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (default is FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotACEcov_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (default is TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotACEcov_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotACEcov_+3A_...">...</code></td>
<td>
<p>Additional (optional) parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxACE">umxACE()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
# BMI ?twinData from Australian twins. 
# Cohort 1 Zygosity 1 == MZ females 3 == DZ females
data(twinData)

# Pick the variables. We will use base names (i.e., "bmi") and set suffix.
selDVs  = c("bmi")
selCovs = c("ht")
selVars = umx_paste_names(c(selDVs, selCovs), sep = "", suffixes= 1:2)
# Just top few pairs so example runs quickly
mzData = subset(twinData, zygosity == "MZFF", selVars)[1:100, ]
dzData = subset(twinData, zygosity == "DZFF", selVars)[1:100, ]
m1 = umxACEcov(selDVs= selDVs, selCovs= selCovs, dzData= dzData, mzData= mzData, sep= "")
plot(m1)
plot(m1, std = FALSE) # don't standardize

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotACEv'>Produce a graphical display of an ACE variance-components twin model</h2><span id='topic+umxPlotACEv'></span><span id='topic+plot.MxModelACEv'></span>

<h3>Description</h3>

<p>Plots an ACE model graphically, opening the result in the browser (or a graphviz application).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotACEv(
  x = NA,
  file = "name",
  digits = 2,
  means = FALSE,
  std = TRUE,
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotACEv_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+umxACEv">umxACEv()</a></code> model to plot.</p>
</td></tr>
<tr><td><code id="umxPlotACEv_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: Default (&quot;name&quot;) = use the name of the model. NA = don't plot.</p>
</td></tr>
<tr><td><code id="umxPlotACEv_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (default = 2)</p>
</td></tr>
<tr><td><code id="umxPlotACEv_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotACEv_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotACEv_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotACEv_+3A_...">...</code></td>
<td>
<p>Additional (optional) parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(umx)
data(twinData)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACEv(selDVs = "bmi", dzData = dzData, mzData = mzData, sep = "")
umxSummary(m1)
umxPlotACEv(m1, std = FALSE) # Don't standardize
plot(m1, std = FALSE) # don't standardize

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotCP'>Draw and display a graphical figure of Common Pathway model</h2><span id='topic+umxPlotCP'></span><span id='topic+plot.MxModelCP'></span>

<h3>Description</h3>

<p>Options include digits (rounding), showing means or not, and which output format is desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotCP(
  x = NA,
  means = FALSE,
  std = TRUE,
  digits = 2,
  showFixed = TRUE,
  file = "name",
  format = c("current", "graphviz", "DiagrammeR"),
  SEstyle = FALSE,
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotCP_+3A_x">x</code></td>
<td>
<p>The Common Pathway <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to display graphically</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (defaults to 2)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_showfixed">showFixed</code></td>
<td>
<p>Whether to graph paths that are fixed but != 0 (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_format">format</code></td>
<td>
<p>= c(&quot;current&quot;, &quot;graphviz&quot;, &quot;DiagrammeR&quot;)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_sestyle">SEstyle</code></td>
<td>
<p>report &quot;b (se)&quot; instead of &quot;b [lower, upper]&quot; when CIs are found (Default FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotCP_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxCP">umxCP()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
umx_set_optimizer("SLSQP")
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff", "fc", "qol", "hap", "sat", "AD") 
m1 = umxCP("new", selDVs = selDVs, sep = "_T", 
	dzData = dzData, mzData = mzData, nFac = 3
)
# m1 = mxTryHardOrdinal(m1)
umxPlotCP(m1)
plot(m1) # No need to remember a special name: plot works fine!

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotDoC'>Plot a Direction of Causation Model.</h2><span id='topic+umxPlotDoC'></span><span id='topic+plot.MxModelDoC'></span>

<h3>Description</h3>

<p>Summarize a fitted model returned by <code><a href="#topic+umxDoC">umxDoC()</a></code>. Can control digits, report comparison model fits,
optionally show the <em>Rg</em> (genetic and environmental correlations), and show confidence intervals.
<em>note</em>: <code>std</code> is not implemented as yet.
See documentation for other umx models here: <code><a href="#topic+umxSummary">umxSummary()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotDoC(
  x = NA,
  means = FALSE,
  std = FALSE,
  digits = 2,
  showFixed = TRUE,
  file = "name",
  format = c("current", "graphviz", "DiagrammeR"),
  SEstyle = FALSE,
  strip_zero = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotDoC_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+umxDoC">umxDoC()</a></code> model to display graphically</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (defaults to 2)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_showfixed">showFixed</code></td>
<td>
<p>Whether to graph paths that are fixed but != 0 (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_format">format</code></td>
<td>
<p>= c(&quot;current&quot;, &quot;graphviz&quot;, &quot;DiagrammeR&quot;)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_sestyle">SEstyle</code></td>
<td>
<p>report &quot;b (se)&quot; instead of &quot;b [lower, upper]&quot; when CIs are found (Default FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotDoC_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>, <code><a href="#topic+umxSummary.MxModelDoC">umxSummary.MxModelDoC()</a></code>, <code><a href="#topic+umxModify">umxModify()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# ================
# = 1. Load Data =
# ================
data(docData)
mzData = subset(docData, zygosity %in% c("MZFF", "MZMM"))
dzData = subset(docData, zygosity %in% c("DZFF", "DZMM"))

# =======================================
# = 2. Define manifests for var 1 and 2 =
# =======================================
var1 = paste0("varA", 1:3)
var2 = paste0("varB", 1:3)

# =======================================================
# = 2. Make the non-causal (Cholesky) and causal models =
# =======================================================
Chol= umxDoC(var1= var1, var2= var2, mzData= mzData, dzData= dzData, causal= FALSE)
DoC = umxDoC(var1= var1, var2= var2, mzData= mzData, dzData= dzData, causal= TRUE)

# ================================================
# = Make the directional models by modifying DoC =
# ================================================
a2b = umxModify(DoC, "a2b", free = TRUE, name = "A2B")
plot(a2b)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotFun'>Easily plot functions in R</h2><span id='topic+umxPlotFun'></span>

<h3>Description</h3>

<p>A wrapper for <code><a href="ggplot2.html#topic+geom_function">ggplot2::stat_function()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotFun(
  fun = c(dnorm, "sin(x) + sqrt(1/x)"),
  min = -1,
  max = 5,
  xlab = NULL,
  ylab = NULL,
  title = NULL,
  logY = c("no", "log", "log10"),
  p = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotFun_+3A_fun">fun</code></td>
<td>
<p>Function to plot. Also takes strings like &quot;sin(x) + sqrt(1/x)&quot;.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_min">min</code></td>
<td>
<p>x-range min.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_max">max</code></td>
<td>
<p>x-range max.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_xlab">xlab</code></td>
<td>
<p>= Optional x axis label.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_ylab">ylab</code></td>
<td>
<p>= Optional y axis label.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_title">title</code></td>
<td>
<p>Optional title for the plot.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_logy">logY</code></td>
<td>
<p>Set to, e.g. &quot;log&quot; to set COORDINATE of y to log.</p>
</td></tr>
<tr><td><code id="umxPlotFun_+3A_p">p</code></td>
<td>
<p>Optional plot onto which to draw the function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Easily plot a function - like sin, using ggplot.
</p>


<h3>Value</h3>


<ul>
<li><p> A ggplot graph
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="ggplot2.html#topic+geom_function">ggplot2::stat_function()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Uses fonts not available on CRAN
umxPlotFun(sin, max= 2*pi)
umxPlotFun("sqrt(1/x)", max= 2*pi)
umxPlotFun(sin, max= 2*pi, ylab="Output of sin", title="My Big Graph")
p = umxPlotFun(function(x){x^2}, max= 100, title="Supply and demand")
umxPlotFun(function(x){100^2-x^2}, p = p)

# Controlling other plot features
umxPlotFun(c("sin(x)", "x^3")) + ylim(c(-1,5)) 

## End(Not run)

</code></pre>

<hr>
<h2 id='umxPlotGxE'>Plot the results of a GxE univariate test for moderation of ACE components.</h2><span id='topic+umxPlotGxE'></span><span id='topic+plot.MxModelGxE'></span>

<h3>Description</h3>

<p>Plot GxE results (univariate environmental moderation of ACE components).
Options include plotting the raw and standardized graphs separately, or in a combined panel.
You can also set the label for the x axis (xlab), and choose the location of the legend.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotGxE(
  x,
  xlab = NA,
  location = "topleft",
  separateGraphs = FALSE,
  acergb = c("red", "green", "blue", "black"),
  gg = TRUE,
  moderatorValues = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotGxE_+3A_x">x</code></td>
<td>
<p>A fitted <code><a href="#topic+umxGxE">umxGxE()</a></code> model to plot</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_xlab">xlab</code></td>
<td>
<p>String to use for the x label (default = NA, which will use the variable name)</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_location">location</code></td>
<td>
<p>Where to plot the legend (default = &quot;topleft&quot;)
see ?legend for alternatives like bottomright</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_separategraphs">separateGraphs</code></td>
<td>
<p>(default = FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_acergb">acergb</code></td>
<td>
<p>Colors to use for plot c(a = &quot;red&quot;, c = &quot;green&quot;, e = &quot;blue&quot;, tot = &quot;black&quot;)</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_gg">gg</code></td>
<td>
<p>Use ggplot2 (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_moderatorvalues">moderatorValues</code></td>
<td>
<p>If you want to pass in your own list of moderator values instead of the real ones in the data (Default = NULL)</p>
</td></tr>
<tr><td><code id="umxPlotGxE_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p><em>note</em>: If <code>gg=TRUE</code>, the plots are drawn in ggplot, and also returned as a <code>list(raw, std)</code> so you can edit them.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxGxE">umxGxE()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData)
twinData$age1 = twinData$age2 = twinData$age
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1= umxGxE(selDVs= "bmi", selDefs= "age", dzData= dzData, mzData= mzData, sep="", tryHard="yes")
plot(m1)
# Directly call umxPlotGxE
umxPlotGxE(m1, xlab = "Age", separateGraphs = TRUE, gg = FALSE)
umxPlotGxE(m1, moderatorValues=18:67)

## End(Not run)

</code></pre>

<hr>
<h2 id='umxPlotGxEbiv'>Plot the results of a GxE univariate test for moderation of ACE components.</h2><span id='topic+umxPlotGxEbiv'></span><span id='topic+plot.MxModelGxEbiv'></span>

<h3>Description</h3>

<p>Plot GxE results (univariate environmental moderation of ACE components).
Options include plotting the raw and standardized graphs separately, or in a combined panel.
You can also set the label for the x axis (xlab), and choose the location of the legend.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotGxEbiv(x, xlab = NA, location = "topleft", separateGraphs = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotGxEbiv_+3A_x">x</code></td>
<td>
<p>A fitted <code><a href="#topic+umxGxEbiv">umxGxEbiv()</a></code> model to plot</p>
</td></tr>
<tr><td><code id="umxPlotGxEbiv_+3A_xlab">xlab</code></td>
<td>
<p>String to use for the x label (default = NA, which will use the variable name)</p>
</td></tr>
<tr><td><code id="umxPlotGxEbiv_+3A_location">location</code></td>
<td>
<p>Where to plot the legend (default = &quot;topleft&quot;)
see ?legend for alternatives like bottomright</p>
</td></tr>
<tr><td><code id="umxPlotGxEbiv_+3A_separategraphs">separateGraphs</code></td>
<td>
<p>(default = FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotGxEbiv_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxGxEbiv">umxGxEbiv()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(twinData)
## Not run: 
selDVs  = "wt"; selDefs = "ht"
df = umx_scale_wide_twin_data(twinData, varsToScale = c("ht", "wt"), suffix = "")
mzData  = subset(df, zygosity %in%  c("MZFF", "MZMM"))
dzData  = subset(df, zygosity %in%  c("DZFF", "DZMM", "DZOS"))

m1 = umxGxEbiv(selDVs = selDVs, selDefs = selDefs, 
	dzData = dzData, mzData = mzData, sep = "", dropMissingDef = TRUE)
# Plot Moderation
plot(m1)
umxPlotGxEbiv(m1, xlab = "wt", separateGraphs = TRUE, location = "topleft")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotIP'>Draw a graphical figure for a Independent Pathway model</h2><span id='topic+umxPlotIP'></span><span id='topic+plot.MxModelIP'></span>

<h3>Description</h3>

<p>Options include digits (rounding), showing means or not, standardization, and which output format is desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotIP(
  x = NA,
  file = "name",
  digits = 2,
  means = FALSE,
  std = TRUE,
  showFixed = TRUE,
  format = c("current", "graphviz", "DiagrammeR"),
  SEstyle = FALSE,
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotIP_+3A_x">x</code></td>
<td>
<p>The <code><a href="#topic+umxIP">umxIP()</a></code> model to plot</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (defaults to 2)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_showfixed">showFixed</code></td>
<td>
<p>Whether to graph paths that are fixed but != 0 (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_format">format</code></td>
<td>
<p>= c(&quot;current&quot;, &quot;graphviz&quot;, &quot;DiagrammeR&quot;)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_sestyle">SEstyle</code></td>
<td>
<p>Report &quot;b (se)&quot; instead of &quot;b [lower, upper]&quot; (Default)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotIP_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxIP">umxIP()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff","fc","qol","hap","sat","AD") # These will be expanded into "gff_T1" "gff_T2" etc.
m1 =    umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData)
plot(model)
umxPlotIP(model, file = NA)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotSexLim'>Draw and display a graphical figure of a Sex limitation model</h2><span id='topic+umxPlotSexLim'></span><span id='topic+plot.MxModelSexLim'></span>

<h3>Description</h3>

<p>Will plot a graphical figure for a sex limitation model.
Options include <code>digits</code> (rounding), showing <code>means</code> or not, and which output <code>format</code> is desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotSexLim(
  x = NA,
  file = "name",
  digits = 2,
  means = FALSE,
  std = TRUE,
  format = c("current", "graphviz", "DiagrammeR"),
  SEstyle = FALSE,
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotSexLim_+3A_x">x</code></td>
<td>
<p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to display graphically</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (defaults to 2)</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_format">format</code></td>
<td>
<p>= c(&quot;current&quot;, &quot;graphviz&quot;, &quot;DiagrammeR&quot;)</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_sestyle">SEstyle</code></td>
<td>
<p>report &quot;b (se)&quot; instead of &quot;b [lower, upper]&quot; (Default)</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotSexLim_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Optionally return the dot code
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSexLim">umxSexLim()</a></code>, <code><a href="#topic+umxSummarySexLim">umxSummarySexLim()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSimplex">umxPlotSimplex</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
umx_set_optimizer("SLSQP")
data("us_skinfold_data")
# Rescale vars
us_skinfold_data[, c('bic_T1', 'bic_T2')] = us_skinfold_data[, c('bic_T1', 'bic_T2')]/3.4
us_skinfold_data[, c('tri_T1', 'tri_T2')] = us_skinfold_data[, c('tri_T1', 'tri_T2')]/3
us_skinfold_data[, c('caf_T1', 'caf_T2')] = us_skinfold_data[, c('caf_T1', 'caf_T2')]/3
us_skinfold_data[, c('ssc_T1', 'ssc_T2')] = us_skinfold_data[, c('ssc_T1', 'ssc_T2')]/5
us_skinfold_data[, c('sil_T1', 'sil_T2')] = us_skinfold_data[, c('sil_T1', 'sil_T2')]/5

# Data for each of the 5 twin-type groups
mzmData = subset(us_skinfold_data, zyg == 1)
mzfData = subset(us_skinfold_data, zyg == 2)
dzmData = subset(us_skinfold_data, zyg == 3)
dzfData = subset(us_skinfold_data, zyg == 4)
dzoData = subset(us_skinfold_data, zyg == 5)

# ==========================
# = Run univariate example =
# ==========================
m1 = umxSexLim(selDVs = "bic", sep = "_T", A_or_C = "A", autoRun= FALSE,
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)
m1 = mxTryHard(m1)
umxPlotSexLim(m1)
plot(m1) # no need to remember a special name: plot works fine!

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPlotSimplex'>Draw and display a graphical figure of a simplex model</h2><span id='topic+umxPlotSimplex'></span><span id='topic+plot.MxModelSimplex'></span>

<h3>Description</h3>

<p>Options include digits (rounding), showing means or not, and which output format is desired.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPlotSimplex(
  x = NA,
  file = "name",
  digits = 2,
  means = FALSE,
  std = TRUE,
  format = c("current", "graphviz", "DiagrammeR"),
  strip_zero = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPlotSimplex_+3A_x">x</code></td>
<td>
<p>The <code><a href="#topic+umxSimplex">umxSimplex()</a></code> model to display graphically</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_digits">digits</code></td>
<td>
<p>How many decimals to include in path loadings (defaults to 2)</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_means">means</code></td>
<td>
<p>Whether to show means paths (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_std">std</code></td>
<td>
<p>Whether to standardize the model (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_format">format</code></td>
<td>
<p>= c(&quot;current&quot;, &quot;graphviz&quot;, &quot;DiagrammeR&quot;)</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to strip the leading &quot;0&quot; and decimal point from parameter estimates (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxPlotSimplex_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Optionally return the dot code
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, simplex, ACEv, or ACE model.
</p>
</li></ul>


<ul>
<li> <p><code><a href="#topic+umxSimplex">umxSimplex()</a></code>
</p>
</li></ul>

<p>Other Plotting functions: 
<code><a href="#topic+ggAddR">ggAddR</a>()</code>,
<code><a href="#topic+plot.MxLISRELModel">plot.MxLISRELModel</a>()</code>,
<code><a href="#topic+plot.MxModelTwinMaker">plot.MxModelTwinMaker</a>()</code>,
<code><a href="#topic+plot.MxModel">plot.MxModel</a>()</code>,
<code><a href="#topic+umxPlotACEcov">umxPlotACEcov</a>()</code>,
<code><a href="#topic+umxPlotACEv">umxPlotACEv</a>()</code>,
<code><a href="#topic+umxPlotACE">umxPlotACE</a>()</code>,
<code><a href="#topic+umxPlotCP">umxPlotCP</a>()</code>,
<code><a href="#topic+umxPlotDoC">umxPlotDoC</a>()</code>,
<code><a href="#topic+umxPlotFun">umxPlotFun</a>()</code>,
<code><a href="#topic+umxPlotGxEbiv">umxPlotGxEbiv</a>()</code>,
<code><a href="#topic+umxPlotGxE">umxPlotGxE</a>()</code>,
<code><a href="#topic+umxPlotIP">umxPlotIP</a>()</code>,
<code><a href="#topic+umxPlotSexLim">umxPlotSexLim</a>()</code>,
<code><a href="#topic+umxPlot">umxPlot</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(iqdat)
mzData = subset(iqdat, zygosity == "MZ")
dzData = subset(iqdat, zygosity == "DZ")
selDVs = c("IQ_age1", "IQ_age2", "IQ_age3", "IQ_age4")
m1 = umxSimplex(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData)
# plot(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxPower'>Test power to detect specified path values in a model.</h2><span id='topic+umxPower'></span>

<h3>Description</h3>

<p><code>umxPower</code> takes an input model (the model of the true data), and tests power (or determines n)
to detect dropping (or changing the value) a path in this true model.
</p>
<p>A typical target for power is 80%. Much as the accepted critical p-value is .05, this has emerged as a trade off, in this case
of resources required for more powerful studies against the cost of missing a true effect.  People interested in truth
discourage running studies with low power: A study with 20 percent power will fail to detect real effects 80% of the time.
But even with zero power, the Type-I error rate remains a nominal 5% (and with any researcher degrees of freedom, perhaps much more than that).
Low powered research, then, fails to detect true effects, and generates support for random false theories about as often.
This sounds silly, but empirical rates are often as low as 20% (Button, et al., 2013).
</p>
<p>Illustration of  <code class="reqn">\alpha</code>, <code class="reqn">\beta</code>, and power (1-<code class="reqn">\beta</code>):
</p>
<p><img src="../help/figures/power.png" width=50% alt="Figure: power.png" />

</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxPower(
  trueModel,
  update = NULL,
  n = NULL,
  power = NULL,
  sig.level = 0.05,
  value = 0,
  method = c("ncp", "empirical"),
  explore = FALSE,
  digits = 2,
  plot = TRUE,
  silent = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxPower_+3A_truemodel">trueModel</code></td>
<td>
<p>The model with the parameters at values you expect in the population.</p>
</td></tr>
<tr><td><code id="umxPower_+3A_update">update</code></td>
<td>
<p>The parameter(s) to drop</p>
</td></tr>
<tr><td><code id="umxPower_+3A_n">n</code></td>
<td>
<p>How many subjects? (Default = NULL)</p>
</td></tr>
<tr><td><code id="umxPower_+3A_power">power</code></td>
<td>
<p>Default = NULL (conventional level = .8)</p>
</td></tr>
<tr><td><code id="umxPower_+3A_sig.level">sig.level</code></td>
<td>
<p>Default = .05</p>
</td></tr>
<tr><td><code id="umxPower_+3A_value">value</code></td>
<td>
<p>Value of dropped parameter (default = 0)</p>
</td></tr>
<tr><td><code id="umxPower_+3A_method">method</code></td>
<td>
<p>&quot;ncp&quot; (default) or &quot;empirical&quot;</p>
</td></tr>
<tr><td><code id="umxPower_+3A_explore">explore</code></td>
<td>
<p>Whether to tabulate the range of n or effect size (if n specified). Default = FALSE.</p>
</td></tr>
<tr><td><code id="umxPower_+3A_digits">digits</code></td>
<td>
<p>Rounding precision for reporting result.</p>
</td></tr>
<tr><td><code id="umxPower_+3A_plot">plot</code></td>
<td>
<p>whether to plot the power.</p>
</td></tr>
<tr><td><code id="umxPower_+3A_silent">silent</code></td>
<td>
<p>Suppress model runs printouts to console (TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>power table
</p>


<h3>References</h3>


<ul>
<li><p> Miles, J. (2003). A framework for power analysis using a structural equation modelling procedure. <em>BMC Medical Research Methodology</em>, <strong>3</strong>, 27. <a href="https://doi.org/10.1186/1471-2288-3-27">doi:10.1186/1471-2288-3-27</a>
</p>
</li></ul>


<ul>
<li> <p><a href="https://CRAN.R-project.org/package=Superpower">Superpower package</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+power.ACE.test">power.ACE.test()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other Teaching and Testing functions: 
<code><a href="#topic+tmx_show.MxModel">tmx_show.MxModel</a>()</code>,
<code><a href="#topic+umxDiagnose">umxDiagnose</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ===================================================
# = Power to detect correlation of .3 in 200 people =
# ===================================================

# 1 Make some data
tmp = umx_make_raw_from_cov(qm(1, .3| .3, 1), n=2000, varNames= c("X", "Y"), empirical= TRUE)

# 2. Make model of true XY correlation of .3
m1 = umxRAM("corXY", data = tmp,
   umxPath("X", with = "Y"),
   umxPath(var = c("X", "Y"))
)
# 3. Test power to detect .3 versus 0, with n= 90 subjects
umxPower(m1, "X_with_Y", n= 90)

# ####################
# # Estimating power #
# ####################
# 
#    method = ncp
#         n = 90
#     power = 0.83
# sig.level = 0.05
# statistic = LRT

# =================================================
# = Tabulate Power across a range of values of  n =
# =================================================
umxPower(m1, "X_with_Y", explore = TRUE)

# =====================================
# = Examples with method = empirical  =
# =====================================

# Power to detect r = .3 given n=90
umxPower(m1, "X_with_Y", n = 90, method = "empirical")
# power is .823
# Test using cor.test doing the same thing.
pwr::pwr.r.test(r = .3, n = 90)
#           n = 90
#           r = 0.3
#   sig.level = 0.05
#       power = 0.827
# alternative = two.sided

# Power search for detectable effect size, given n = 90
umxPower(m1, "X_with_Y", explore = TRUE)
umxPower(m1, "X_with_Y", n= 90, explore = TRUE)
umxPower(m1, "X_with_Y", n= 90, method = "empirical", explore = TRUE)


data(twinData) # ?twinData from Australian twins.
twinData[, c("ht1", "ht2")] = twinData[, c("ht1", "ht2")] * 10
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m1 = umxACE(selDVs = "ht", selCovs = "age", sep = "", dzData = dzData, mzData = mzData)

# drop more than 1 path
umxPower(m1, update = c("c_r1c1", "age_b_Var1"), method = 'ncp', n=90, explore = TRUE)

# Specify only 1 parameter (not 'age_b_Var1' and 'c_r1c1' ) to search a parameter:power relationship
# note: Can't use method = "ncp" with search)
umxPower(m1, update = c("c_r1c1", "age_b_Var1"), method = 'empirical', n=90, explore = TRUE)
umxPower(m1, update = c("c_r1c1"), method = 'empirical', n=90, explore = TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxRAM'>Build and run path-based SEM models</h2><span id='topic+umxRAM'></span>

<h3>Description</h3>

<p><code>umxRAM</code> expedites creation of structural equation models, still without doing invisible things to the model. It
supports <code><a href="#topic+umxPath">umxPath()</a></code>. To support cross-language sharing and science learning, <code>umxRAM</code> also supports lavaan model strings.
</p>
<p>Here's a path example that models miles per gallon (mpg) as a function of weight (wt) and engine displacement (disp)
using the widely used <code>mtcars</code> data set.
</p>
<div class="sourceCode"><pre>m1 = umxRAM("tim", data = mtcars,
	umxPath(c("wt", "disp"), to = "mpg"),
	umxPath("wt", with = "disp"),
	umxPath(v.m. = c("wt", "disp", "mpg"))
)
</pre></div>
<p>As you can see, most of the work is done by <code><a href="#topic+umxPath">umxPath()</a></code>. <code>umxRAM</code> wraps these paths up, takes the <code style="white-space: pre;">&#8288;data =&#8288;</code> input, and
then internally sets up all the labels and start values for the model, runs it, and calls <code><a href="#topic+umxSummary">umxSummary()</a></code>, and <code><a href="#topic+plot.MxModel">plot.MxModel()</a></code>.
</p>
<p>Try it, or one of the several models in the examples at the bottom of this page.
</p>
<p>A common error is to include data in the main list, a bit like
saying <code>lm(y ~ x + df)</code> instead of <code>lm(y ~ x, data = df)</code>.
</p>
<p><strong>nb</strong>: Because it uses the presence of a variable in the data to detect if a variable is latent or not, <code>umxRAM</code> needs data at build time.
</p>
<p><strong>String Syntax</strong>
</p>
<p>Here is an example using lavaan syntax (for more, see <code><a href="#topic+umxLav2RAM">umxLav2RAM()</a></code>)
</p>
<div class="sourceCode R"><pre>m1 = umxRAM("mpg ~ wt + disp", data = mtcars)
</pre></div>
<p><strong>Sketch mode</strong>
</p>
<p>If you are at the &quot;sketching&quot; stage of theory consideration, <code>umxRAM</code> supports
setting data to a simple vector of manifest names.
As usual in <code>umxRAM</code>, any variables you refer to that are not in data are treated as latents.
</p>
<div class="sourceCode R"><pre>m1 = umxRAM("sketch", data = c("A", "B"),
	umxPath("C", to = c("A", "B"), values=.3),
	umxPath("A", with = "B", values=.45),
	umxPath(v.m. = c("A", "B")),
	umxPath(v1m0 = "C")
)
plot(m1, means = FALSE)
</pre></div>
<p>Will create this figure:
</p>
<p><img src="../help/figures/sketch.png" alt="Figure: sketch.png" />

</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxRAM(
  model = NA,
  ...,
  data = NULL,
  name = NA,
  group = NULL,
  group.equal = NULL,
  suffix = "",
  comparison = TRUE,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  weight = NULL,
  allContinuousMethod = c("cumulants", "marginals"),
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  std = FALSE,
  refModels = NULL,
  remove_unused_manifests = TRUE,
  independent = NA,
  setValues = TRUE,
  optimizer = NULL,
  verbose = FALSE,
  std.lv = FALSE,
  lavaanMode = c("sem", "lavaan"),
  printTab = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxRAM_+3A_model">model</code></td>
<td>
<p>A model to update (or set to string to use as name for new model)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_...">...</code></td>
<td>
<p>umxPaths, mxThreshold objects, etc.</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_data">data</code></td>
<td>
<p>data for the model. Can be an <code><a href="OpenMx.html#topic+mxData">mxData()</a></code> or a data.frame</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_name">name</code></td>
<td>
<p>A friendly name for the model</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_group">group</code></td>
<td>
<p>(optional) Column name to use for a multi-group model (default = NULL)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_group.equal">group.equal</code></td>
<td>
<p>In multi-group models, what to equate across groups (default = NULL: all free)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_suffix">suffix</code></td>
<td>
<p>String to append to each label (useful if model will be used in a multi-group model)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_comparison">comparison</code></td>
<td>
<p>Compare the new model to the old (if updating an existing model: default = TRUE)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_type">type</code></td>
<td>
<p>One of &quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, &quot;WLS&quot;, &quot;DWLS&quot;, &quot;ULS&quot;</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_weight">weight</code></td>
<td>
<p>Passes weight values to mxData</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_std">std</code></td>
<td>
<p>Whether to show standardized estimates, raw (NULL print fit only)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_refmodels">refModels</code></td>
<td>
<p>pass in reference models if available. Use FALSE to suppress computing these if not provided.</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_remove_unused_manifests">remove_unused_manifests</code></td>
<td>
<p>Whether to remove variables in the data to which no path makes reference (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_independent">independent</code></td>
<td>
<p>Whether the model is independent (default = NA)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_setvalues">setValues</code></td>
<td>
<p>Whether to generate likely good start values (Defaults to TRUE)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_optimizer">optimizer</code></td>
<td>
<p>optionally set the optimizer (default NULL does nothing)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_verbose">verbose</code></td>
<td>
<p>Whether to tell the user what latents and manifests were created etc. (Default = FALSE)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_std.lv">std.lv</code></td>
<td>
<p>Whether to auto standardize latent variables when using string syntax (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_lavaanmode">lavaanMode</code></td>
<td>
<p>Defaults when building out string syntax default = &quot;sem&quot; (alternative is &quot;lavaan&quot;, with very few defaults)</p>
</td></tr>
<tr><td><code id="umxRAM_+3A_printtab">printTab</code></td>
<td>
<p>(for string input, whether to output a table of paths (FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Comparison for OpenMx users</strong>
</p>
<p><code>umxRAM</code> differs from <code><a href="OpenMx.html#topic+mxModel">OpenMx::mxModel()</a></code> in the following ways:
</p>

<ol>
<li><p> You don't need to set type = &quot;RAM&quot;.
</p>
</li>
<li><p> You don't need to list manifestVars (they are detected from path usage).
</p>
</li>
<li><p> You don't need to list latentVars (detected as anything in paths but not in <code>mxData</code>).
</p>
</li>
<li><p> You don't need to create mxData when you already have a data.frame.
</p>
</li>
<li><p> You add data with <code style="white-space: pre;">&#8288;data = &#8288;</code> (as elsewhere in R, e.g. <code><a href="stats.html#topic+lm">lm()</a></code>).
</p>
</li>
<li><p> You don't need to add labels: paths are automatically labelled &quot;a_to_b&quot; etc.
</p>
</li>
<li><p> You don't need to set start values, they will be done for you.
</p>
</li>
<li><p> You don't need to <code>mxRun</code> the model: it will run automatically, and print a summary.
</p>
</li>
<li><p> You don't need to run <code>summary</code>: with <code>autoRun=TRUE</code>, it will print a summary.
</p>
</li>
<li><p> You get a plot of the model with estimates on the paths, including multiple groups.
</p>
</li>
<li><p> Less typing: <code><a href="#topic+umxPath">umxPath()</a></code> offers powerful verbs to describe paths.
</p>
</li>
<li><p> Supports a subset of lavaan string input.
</p>
</li></ol>

<p><strong>Start values</strong>. Currently, manifest variable means are set to the observed means,
residual variances are set to 80%  of the observed variance of each variable,
and single-headed paths are set to a positive starting value (currently .9).
<em>note</em>: The start-value strategy is subject to improvement, and will be documented in the help for <code><a href="#topic+umxRAM">umxRAM()</a></code>.
</p>
<p><strong>Comparison with other software</strong>
</p>
<p>Some SEM software does a lot of behind-the-scenes defaulting and path addition.
If you want this, I'd say use <code>umxRAM</code> with lavaan string input.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxPath">umxPath()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code>, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+parameters">parameters()</a></code>, <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>, <code><a href="#topic+umxLav2RAM">umxLav2RAM()</a></code>
</p>
<p>Other Core Model Building Functions: 
<code><a href="#topic+umxMatrix">umxMatrix</a>()</code>,
<code><a href="#topic+umxModify">umxModify</a>()</code>,
<code><a href="#topic+umxPath">umxPath</a>()</code>,
<code><a href="#topic+umxSuperModel">umxSuperModel</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# ============================================
# = 1. Here's a simple example with raw data =
# ============================================
mtcars$litres = mtcars$disp/61.02
m1 = umxRAM("tim", data = mtcars,
	umxPath(c("wt", "litres"), to = "mpg"),
	umxPath("wt", with = "litres"),
	umxPath(v.m. = c("wt", "litres", "mpg"))
)

# 2. Use parameters to see the parameter estimates and labels
parameters(m1)

# And umxSummary to get standardized parameters, CIs etc from the run model.
umxSummary(m1, std=TRUE)
# |name           | Std.Estimate| Std.SE|CI                   |
# |:--------------|------------:|------:|:--------------------|
# |wt_to_mpg      |        -0.54|   0.17|-0.54 [-0.89, -0.2]  |
# |disp_to_mpg    |        -0.36|   0.18|-0.36 [-0.71, -0.02] |
# |mpg_with_mpg   |         0.22|   0.07|0.22 [0.08, 0.35]    |
# |wt_with_wt     |         1.00|   0.00|1 [1, 1]             |
# |b1             |         0.89|   0.04|0.89 [0.81, 0.96]    |
# |disp_with_disp |         1.00|   0.00|1 [1, 1]             |

# 3. Of course you can plot the model
plot(m1)
plot(m1, std=TRUE, means=FALSE)
plot(m1, std = TRUE, means=FALSE, strip= TRUE, resid = "line")

# ===============================================
# = lavaan string example (more at ?umxLav2RAM) =
# ===============================================
m1 = umxRAM(data = mtcars, "#modelName
 mpg ~ wt + disp")


# =======================
# = A multi-group model =
# =======================

mtcars$litres = mtcars$disp/61.02
m1 = umxRAM("tim", data = mtcars, group = "am",
	umxPath(c("wt", "litres"), to = "mpg"),
	umxPath("wt", with = "litres"),
	umxPath(v.m. = c("wt", "litres", "mpg"))
)
# In this model, all parameters are free across the two groups.

# ====================================
# = A cov model, with steps laid out =
# ====================================

# *note*: The variance of displacement is in cubic inches and is very large.
# to help the optimizer, one might, say, multiply disp *.016 to work in litres
tmp = mtcars; tmp$disp= tmp$disp *.016

# We can just give the raw data and ask for it to be made into type cov:
m1 = umxRAM("tim", data = tmp, type="cov",
	umxPath(c("wt", "disp"), to = "mpg"),
	umxPath("wt", with = "disp"),
	umxPath(var = c("mpg", "wt", "disp"))
)

# (see ?umxPath for more nifty options making paths...)

# =========================================
# = umxRAM can also accept mxData as data =
# =========================================
# For convenience, list up the manifests you will be using

selVars = c("mpg", "wt", "disp")
tmp = mtcars; tmp$disp= tmp$disp *.016
myCov = mxData(cov(tmp[, selVars]), type = "cov", numObs = nrow(mtcars) )

m1 = umxRAM("tim", data = myCov,
	umxPath(c("wt", "disp"), to = "mpg"),
	umxPath("wt", with = "disp"),
	umxPath(var = selVars)
)


# =======================
# = umxRAM supports WLS =
# =======================

# 1. Run an all-continuous WLS model
 mw = umxRAM("raw", data = mtcars[, c("mpg", "wt", "disp")], 
	type = "WLS", allContinuousMethod = "cumulants",
 	umxPath(var = c("wt", "disp", "mpg")),
 	umxPath(c("wt", "disp"), to = "mpg"),
 	umxPath("wt", with = "disp"),
     umxPath(var = c("wt", "disp", "mpg"))
 )
# 2. Switch to marginals to support means
 mw = umxRAM("raw", data = mtcars[, c("mpg", "wt", "disp")], 
	type = "WLS", allContinuousMethod= "marginals",
 	umxPath(var = c("wt", "disp", "mpg")),
 	umxPath(c("wt", "disp"), to = "mpg"),
 	umxPath("wt", with = "disp"),
     umxPath(var = c("wt", "disp", "mpg"))
 )


# ===============================
# = Using umxRAM in Sketch mode =
# ===============================
# No data needed: just list variable names!
# Resulting model will be plotted automatically
m1 = umxRAM("what does unique pairs do, I wonder", data = c("A", "B", "C"),
   umxPath(unique.pairs = c("A", "B", "C"))
)

m1 = umxRAM("ring around the rosey", data = c("B", "C"),
  umxPath(fromEach = c("A", "B", "C"))
)

m1 = umxRAM("fromEach with to", data = c("B", "C"),
   umxPath(fromEach = c("B", "C"), to= "D")
)

m1 = umxRAM("CFA_sketch", data = paste0("x", 1:4),
	umxPath("g", to = paste0("x", 1:4)),
	umxPath(var = paste0("x", 1:4)),
	umxPath(v1m0 = "g")
)

# =================================================
# = This is an example of using your own labels:  =
#   umxRAM will not over-ride them                =
# =================================================
m1 = umxRAM("tim", data = mtcars, type="cov",
	umxPath(c("wt", "disp"), to = "mpg"),
	umxPath(cov = c("wt", "disp"), labels = "b1"),
	umxPath(var = c("wt", "disp", "mpg"))
)
omxCheckEquals(m1$S$labels["disp", "wt"], "b1") # label preserved
m1$S$labels
#      mpg             wt            disp
# mpg  "mpg_with_mpg"  "mpg_with_wt" "disp_with_mpg"
# wt   "mpg_with_wt"   "wt_with_wt"  "b1"
# disp "disp_with_mpg" "b1"          "disp_with_disp"
parameters(m1)

# ===========
# = Weights =
# ===========
# !!! Not tested !!!
mtcars$litres = mtcars$disp/61.02
m1 = umxRAM("tim", data = mtcars, weight= "cyl",
	umxPath(c("wt", "litres"), to = "mpg"),
	umxPath("wt", with = "litres"),
	umxPath(v.m. = c("wt", "litres", "mpg"))
)


## End(Not run)

</code></pre>

<hr>
<h2 id='umxRAM2Lav'>Convert a RAM model to a lavaan string</h2><span id='topic+umxRAM2Lav'></span>

<h3>Description</h3>

<p>Takes an OpenMx RAM model and creates the corresponding lavaan syntax string.
</p>
<p>This function is at the alpha quality stage, and **should be expected to have bugs**.
Also likely to change functionality and even parameters as new features are supported (e.g. groups)
and lavaan-style strings exported. Several features are not yet supported. Let me know if you would like them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxRAM2Lav(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxRAM2Lav_+3A_model">model</code></td>
<td>
<p>an OpenMx RAM model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A lavaan syntax string, e.g. &quot;A~~B&quot;
</p>


<h3>See Also</h3>

<p>- [umxLav2RAM()], [umxRAM()]
</p>
<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxVersion">umxVersion</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umxRAM2Lav(umxLav2RAM("x ~ y", autoRun = FALSE, printTab = FALSE, lavaanMode = "lavaan"))

## End(Not run)
</code></pre>

<hr>
<h2 id='umxReduce'>Reduce models, and report the results.</h2><span id='topic+umxReduce'></span>

<h3>Description</h3>

<p>Given a <code>umx</code> model (currently <code>umxACE</code> and <code>umxGxE</code> are supported - ask for more!)
<code>umxReduce</code> will conduct a formalised reduction process. It will also report
Akaike weights are also reported showing relative support across models.
</p>
<p>Specialized functions are called for different type of input:
</p>

<ol>
<li> <p><strong>GxE model reduction</strong> For <code><a href="#topic+umxGxE">umxGxE()</a></code> models <code><a href="#topic+umxReduceGxE">umxReduceGxE()</a></code> is called.
</p>
</li>
<li> <p><strong>ACE model reduction</strong> For <code><a href="#topic+umxACE">umxACE()</a></code> models,<code><a href="#topic+umxReduceACE">umxReduceACE()</a></code> is called.
</p>
</li></ol>

<p><code>umxReduce</code> reports the results in a table. Set the format of the table with
<code><a href="#topic+umx_set_table_format">umx_set_table_format()</a></code>, or set <code>report= "html"</code> to open a
table for pasting into a word processor.
</p>
<p><code>umxReduce</code> is a work in progress, with more automatic reductions coming as demand emerges.
I am thinking for RAM models to drop NS paths, and report that test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxReduce(
  model,
  report = c("markdown", "inline", "html"),
  intervals = TRUE,
  baseFileName = "tmp",
  tryHard = "yes",
  silent = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxReduce_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> which will be reduced.</p>
</td></tr>
<tr><td><code id="umxReduce_+3A_report">report</code></td>
<td>
<p>How to report the results. &quot;html&quot; = open in browser</p>
</td></tr>
<tr><td><code id="umxReduce_+3A_intervals">intervals</code></td>
<td>
<p>Recompute CIs (if any included) on the best model (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxReduce_+3A_basefilename">baseFileName</code></td>
<td>
<p>(optional) custom filename for html output (defaults to &quot;tmp&quot;)</p>
</td></tr>
<tr><td><code id="umxReduce_+3A_tryhard">tryHard</code></td>
<td>
<p>Default = &quot;yes&quot;</p>
</td></tr>
<tr><td><code id="umxReduce_+3A_silent">silent</code></td>
<td>
<p>Default = FALSE</p>
</td></tr>
<tr><td><code id="umxReduce_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>References</h3>


<ul>
<li><p> Wagenmakers, E.J., &amp; Farrell, S. (2004). AIC model selection using Akaike weights.
<em>Psychonomic Bulletin and Review</em>, <strong>11</strong>, 192-196. <a href="https://doi.org/10.3758/BF03206482">doi:10.3758/BF03206482</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxReduceGxE">umxReduceGxE()</a></code>, <code><a href="#topic+umxReduceACE">umxReduceACE()</a></code>
</p>
<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxCompare">umxCompare</a>()</code>,
<code><a href="#topic+umxEquate">umxEquate</a>()</code>,
<code><a href="#topic+umxMI">umxMI</a>()</code>,
<code><a href="#topic+umxSetParameters">umxSetParameters</a>()</code>,
<code><a href="#topic+umxSummary">umxSummary</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>

<hr>
<h2 id='umxReduceACE'>Reduce an ACE model.</h2><span id='topic+umxReduceACE'></span>

<h3>Description</h3>

<p>This function can perform model reduction on <code><a href="#topic+umxACE">umxACE()</a></code> models,
testing dropping A and C, as well as an ADE or ACE model, displaying the results
in a table, and returning the best model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxReduceACE(
  model,
  report = c("markdown", "inline", "html", "report"),
  intervals = TRUE,
  baseFileName = "tmp",
  tryHard = c("yes", "no", "ordinal", "search"),
  silent = FALSE,
  digits = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxReduceACE_+3A_model">model</code></td>
<td>
<p>an ACE or ADE <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to reduce</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_report">report</code></td>
<td>
<p>How to report the results. &quot;html&quot; = open in browser</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_intervals">intervals</code></td>
<td>
<p>Recompute CIs (if any included) on the best model (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_basefilename">baseFileName</code></td>
<td>
<p>(optional) custom filename for html output (defaults to &quot;tmp&quot;)</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_tryhard">tryHard</code></td>
<td>
<p>(default = &quot;yes&quot;)</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_silent">silent</code></td>
<td>
<p>Don't print the ACE models (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_digits">digits</code></td>
<td>
<p>rounding in printout (default = 2)</p>
</td></tr>
<tr><td><code id="umxReduceACE_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is designed for testing univariate models. You can offer up either the ACE or ADE base model.
</p>
<p>Suggestions for more sophisticated automation welcomed!
</p>


<h3>Value</h3>

<p>Best fitting model
</p>


<h3>References</h3>


<ul>
<li><p> Wagenmakers, E.J., &amp; Farrell, S. (2004). AIC model selection using Akaike weights. <em>Psychonomic Bulletin and Review</em>, <strong>11</strong>, 192-196. <a href="https://doi.org/10.3758/BF03206482">doi:10.3758/BF03206482</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxReduceGxE">umxReduceGxE()</a></code>, <code><a href="#topic+umxReduce">umxReduce()</a></code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACE(selDVs = "bmi", dzData = dzData, mzData = mzData, sep = "")

# ===========================================================================
# = Table of parameters + fit comparisons, ready too copy to word processor =
# ===========================================================================
umxReduce(m1, silent=TRUE, digits=2, repo="h")

# ==========================================
# = Function captures the preferred model =
# ==========================================
m2 = umxReduce(m1)
umxSummary(m2)

# works for ADE input also
m1 = umxACE(selDVs = "bmi", dzData = dzData, mzData = mzData, sep = "", dzCr = .25)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxReduceGxE'>Reduce a GxE model.</h2><span id='topic+umxReduceGxE'></span>

<h3>Description</h3>

<p>This function can perform model reduction for <code><a href="#topic+umxGxE">umxGxE()</a></code> models,
testing dropping a<code style="white-space: pre;">&#8288;,c&#8288;</code> &amp; e<code style="white-space: pre;">&#8288;, as well as c &amp; c&#8288;</code>, a &amp; a' etc.
</p>
<p>It reports the results in a table. Set the format of the table with
<code><a href="#topic+umx_set_table_format">umx_set_table_format()</a></code>. Or set <code>report = "html"</code> to open a
table for pasting into a word processor.
</p>
<p>In addition to printing a table, the function returns the preferred model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxReduceGxE(
  model,
  report = c("markdown", "inline", "html", "report"),
  intervals = TRUE,
  baseFileName = "tmp_gxe",
  tryHard = c("yes", "no", "ordinal", "search"),
  silent = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxReduceGxE_+3A_model">model</code></td>
<td>
<p>A <code><a href="#topic+umxGxE">umxGxE()</a></code> to reduce.</p>
</td></tr>
<tr><td><code id="umxReduceGxE_+3A_report">report</code></td>
<td>
<p>How to report the results. default = &quot;markdown&quot;. &quot;html&quot; = open in browser.</p>
</td></tr>
<tr><td><code id="umxReduceGxE_+3A_intervals">intervals</code></td>
<td>
<p>Recompute CIs (if any included) on the best model (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxReduceGxE_+3A_basefilename">baseFileName</code></td>
<td>
<p>(optional) custom filename for html output (default = &quot;tmp&quot;).</p>
</td></tr>
<tr><td><code id="umxReduceGxE_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxReduceGxE_+3A_silent">silent</code></td>
<td>
<p>Default (FALSE)</p>
</td></tr>
<tr><td><code id="umxReduceGxE_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>best model
</p>


<h3>References</h3>


<ul>
<li><p> Wagenmakers, E.J., &amp; Farrell, S. (2004). AIC model selection using Akaike weights.
<em>Psychonomic Bulletin and Review</em>, <strong>11</strong>, 192-196. <a href="https://doi.org/10.3758/BF03206482">doi:10.3758/BF03206482</a>.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxReduce">umxReduce()</a></code>, <code><a href="#topic+umxReduceACE">umxReduceACE()</a></code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
model = umxReduce(model)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxRenameMatrix'>Rename a umxMatrix (even in a model)</h2><span id='topic+umxRenameMatrix'></span>

<h3>Description</h3>

<p>Rename a <code><a href="#topic+umxMatrix">umxMatrix()</a></code>, including updating its labels to match the new name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxRenameMatrix(x, matrixName, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxRenameMatrix_+3A_x">x</code></td>
<td>
<p>A model or matrix</p>
</td></tr>
<tr><td><code id="umxRenameMatrix_+3A_matrixname">matrixName</code></td>
<td>
<p>Name of the matrix</p>
</td></tr>
<tr><td><code id="umxRenameMatrix_+3A_name">name</code></td>
<td>
<p>The new name</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> updated matrix or model with updated matrix in it.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData) # ?twinData from Australian twins.
twinData[, c("ht1", "ht2")] = twinData[, c("ht1", "ht2")] * 10
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m1  = umxACE(selDVs= "ht", sep= "", dzData= dzData, mzData= mzData, autoRun= FALSE)
tmp = umxRenameMatrix(m1$top, matrixName = "a", name="hello")
umx_check(tmp$hello$labels == "hello_r1c1") # new is there
umx_check(is.null(tmp$a))                   # old is gone

## End(Not run)

</code></pre>

<hr>
<h2 id='umxRotate'>Generic SEM factor model loading rotation function</h2><span id='topic+umxRotate'></span>

<h3>Description</h3>

<p>See <code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP()</a></code> to rotate the factor loadings of a <code><a href="#topic+umxCP">umxCP()</a></code> model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxRotate(
  model,
  rotation = c("varimax", "promax"),
  tryHard = "yes",
  freeLoadingsAfter = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxRotate_+3A_model">model</code></td>
<td>
<p>a model to rotate</p>
</td></tr>
<tr><td><code id="umxRotate_+3A_rotation">rotation</code></td>
<td>
<p>name of the rotation.</p>
</td></tr>
<tr><td><code id="umxRotate_+3A_tryhard">tryHard</code></td>
<td>
<p>Default (&quot;yes&quot;) is to tryHard</p>
</td></tr>
<tr><td><code id="umxRotate_+3A_freeloadingsafter">freeLoadingsAfter</code></td>
<td>
<p>Whether to keep the rotated loadings fixed (Default, free them again)</p>
</td></tr>
<tr><td><code id="umxRotate_+3A_verbose">verbose</code></td>
<td>
<p>print detail about the rotation</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Rotated solution
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Reporting functions: 
<code><a href="#topic+RMSEA.MxModel">RMSEA.MxModel</a>()</code>,
<code><a href="#topic+RMSEA.summary.mxmodel">RMSEA.summary.mxmodel</a>()</code>,
<code><a href="#topic+RMSEA">RMSEA</a>()</code>,
<code><a href="#topic+extractAIC.MxModel">extractAIC.MxModel</a>()</code>,
<code><a href="#topic+loadings.MxModel">loadings.MxModel</a>()</code>,
<code><a href="#topic+loadings">loadings</a>()</code>,
<code><a href="#topic+residuals.MxModel">residuals.MxModel</a>()</code>,
<code><a href="#topic+tmx_show.MxMatrix">tmx_show.MxMatrix</a>()</code>,
<code><a href="#topic+tmx_show">tmx_show</a>()</code>,
<code><a href="#topic+umxCI_boot">umxCI_boot</a>()</code>,
<code><a href="#topic+umxCI">umxCI</a>()</code>,
<code><a href="#topic+umxConfint">umxConfint</a>()</code>,
<code><a href="#topic+umxExpCov">umxExpCov</a>()</code>,
<code><a href="#topic+umxExpMeans">umxExpMeans</a>()</code>,
<code><a href="#topic+umxFitIndices">umxFitIndices</a>()</code>
</p>

<hr>
<h2 id='umxRotate.MxModelCP'>Rotate a CP solution</h2><span id='topic+umxRotate.MxModelCP'></span>

<h3>Description</h3>

<p>Rotate a CP solution.
Should work with rotations provided in <code>library("GPArotation")</code> and <code>library("psych")</code>, e.g
</p>
<p><strong>Orthogonal</strong>: &quot;varimax&quot;, &quot;quartimax&quot;, &quot;bentlerT&quot;, &quot;equamax&quot;, &quot;varimin&quot;, &quot;geominT&quot; and &quot;bifactor&quot;
</p>
<p><strong>Oblique</strong>: &quot;Promax&quot;, &quot;promax&quot;, &quot;oblimin&quot;, &quot;simplimax&quot;, &quot;bentlerQ&quot;, &quot;geominQ&quot;, &quot;biquartimin&quot; and &quot;cluster&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModelCP'
umxRotate(
  model,
  rotation = c("varimax", "promax"),
  tryHard = "yes",
  freeLoadingsAfter = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxRotate.MxModelCP_+3A_model">model</code></td>
<td>
<p>a <code><a href="#topic+umxCP">umxCP()</a></code> model to rotate.</p>
</td></tr>
<tr><td><code id="umxRotate.MxModelCP_+3A_rotation">rotation</code></td>
<td>
<p>name of the rotation.</p>
</td></tr>
<tr><td><code id="umxRotate.MxModelCP_+3A_tryhard">tryHard</code></td>
<td>
<p>Default (&quot;yes&quot;) is to tryHard.</p>
</td></tr>
<tr><td><code id="umxRotate.MxModelCP_+3A_freeloadingsafter">freeLoadingsAfter</code></td>
<td>
<p>return the model with factor loadings free (default) or fixed in the new locations.</p>
</td></tr>
<tr><td><code id="umxRotate.MxModelCP_+3A_verbose">verbose</code></td>
<td>
<p>print detail about the rotation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This works by taking the common-pathways loadings matrix from a solved <code><a href="#topic+umxCP">umxCP()</a></code> model, rotating these, placing
them back into the loadings matrix, re-estimating the model with the parameters fixed at this rotation, then return the new model.
</p>


<h3>Value</h3>


<ul>
<li><p> Rotated solution.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxCP">umxCP()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Rotate a CP solution(param)
# Common pathway model rotation
library(umx)
# Fit 3 factor CPM
data(GFF)
selDVs = c("gff", "fc", "qol", "hap", "sat", "AD") 
m1 = umxCP(selDVs = selDVs, nFac = 2, data = data, tryHard = "yes")
m2 = umxRotate(m1, rotation = "varimax",  tryHard = "yes")


## End(Not run)
</code></pre>

<hr>
<h2 id='umxRun'>umxRun: Run an mxModel</h2><span id='topic+umxRun'></span>

<h3>Description</h3>

<p><code>umxRun</code> is a version of <code><a href="OpenMx.html#topic+mxRun">mxRun()</a></code> which can run also set start values, labels, and run multiple times
It can also calculate the saturated and independence likelihoods necessary for most fit indices.
<strong>Note</strong> this is not needed for umxRAM models or twin models - it is just a convenience to get base OpenMx models to run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxRun(
  model,
  tryHard = c("yes", "no", "ordinal", "search"),
  calc_sat = TRUE,
  setValues = FALSE,
  setLabels = FALSE,
  summary = !umx_set_silent(silent = TRUE),
  intervals = FALSE,
  optimizer = NULL,
  comparison = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxRun_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> you wish to run.</p>
</td></tr>
<tr><td><code id="umxRun_+3A_tryhard">tryHard</code></td>
<td>
<p>How to tryHard. Default = &quot;yes&quot;. Alternatives &quot;no&quot;, &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxRun_+3A_calc_sat">calc_sat</code></td>
<td>
<p>Whether to calculate the saturated and independence models (for raw <code><a href="OpenMx.html#topic+mxData">mxData()</a></code> <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>s)</p>
</td></tr>
<tr><td><code id="umxRun_+3A_setvalues">setValues</code></td>
<td>
<p>Whether to set the starting values of free parameters (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxRun_+3A_setlabels">setLabels</code></td>
<td>
<p>Whether to set the labels (default =  FALSE)</p>
</td></tr>
<tr><td><code id="umxRun_+3A_summary">summary</code></td>
<td>
<p>Whether to print summary or not (default = !umx_set_silent() )</p>
</td></tr>
<tr><td><code id="umxRun_+3A_intervals">intervals</code></td>
<td>
<p>Whether to run mxCI confidence intervals (default = FALSE) intervals = FALSE</p>
</td></tr>
<tr><td><code id="umxRun_+3A_optimizer">optimizer</code></td>
<td>
<p>optional to set the optimizer.</p>
</td></tr>
<tr><td><code id="umxRun_+3A_comparison">comparison</code></td>
<td>
<p>Comparison model (will be used to drive umxCompare() after umxRun</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
latents  = c("G")
manifests = names(demoOneFactor)
m1 = mxModel("fact", type="RAM", manifestVars=manifests, latentVars=latents,
	mxPath(latents  , to = manifests),
	mxPath(manifests, arrows = 2),
	mxPath(latents  , arrows = 2, free = FALSE, values = 1),
	mxData(cov(demoOneFactor), type = "cov", numObs=500)
)

m1 = umxRun(m1) # just run: will create saturated model if needed
m1 = umxRun(m1, setValues = TRUE, setLabels = TRUE) # set start values and label all parameters
umxSummary(m1, std = TRUE)
m1 = mxModel(m1, mxCI("G_to_x1")) # add one CI
m1 = mxRun(m1, intervals = TRUE)
residuals(m1, run = TRUE) # get CIs on all free parameters
confint(m1) # OpenMx's SE-based CIs
umxConfint(m1, run = TRUE) # get likelihood-based CIs on all free parameters
m1 = umxRun(m1, tryHard = "yes")

## End(Not run)

</code></pre>

<hr>
<h2 id='umxSetParameters'>Change or fix parameters (e.g. their values, labels, bounds, ..) in a model.</h2><span id='topic+umxSetParameters'></span>

<h3>Description</h3>

<p><code>umxSetParameters</code> is used to alter values, and other parameter properties in an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>.
A common use is setting new values and changing parameters from free to false.
<em>Note</em>: If you just want to modify and re-run a model, you probably want <code><a href="#topic+umxModify">umxModify()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSetParameters(
  model,
  labels,
  free = NULL,
  values = NULL,
  newlabels = NULL,
  lbound = NULL,
  ubound = NULL,
  indep = FALSE,
  strict = TRUE,
  name = NULL,
  regex = FALSE,
  test = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSetParameters_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to set parameters in.</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_labels">labels</code></td>
<td>
<p>= labels to find</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_free">free</code></td>
<td>
<p>= new value for free</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_values">values</code></td>
<td>
<p>= new values</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_newlabels">newlabels</code></td>
<td>
<p>= newlabels</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_lbound">lbound</code></td>
<td>
<p>= value for lbound</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_ubound">ubound</code></td>
<td>
<p>= value for ubound</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_indep">indep</code></td>
<td>
<p>= whether to look in indep models</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_strict">strict</code></td>
<td>
<p>whether to complain if labels not found</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_name">name</code></td>
<td>
<p>= new name for the returned model</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_regex">regex</code></td>
<td>
<p>patterns to match for labels (or if TRUE, use labels as regular expressions)</p>
</td></tr>
<tr><td><code id="umxSetParameters_+3A_test">test</code></td>
<td>
<p>Just show what you would do? (defaults to FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Using <code>umxSetParameters</code>, you use <code style="white-space: pre;">&#8288;labels=&#8288;</code> to select the parameters you want to update.
You can set their free/fixed state with <code style="white-space: pre;">&#8288;free=&#8288;</code>, and set new values with <code style="white-space: pre;">&#8288;values = &#8288;</code>. Likewise
for bounds.
</p>
<p><code>umxSetParameters</code> supports pattern matching (regular expressions) to select labels. Set <code style="white-space: pre;">&#8288;regex=&#8288;</code>
to a regular expression matching the labels you want to select. e.g. &quot;G_to_.*&quot; would match
&quot;G_to_anything&quot;.
</p>
<p><strong>Details</strong>
Internally, <code>umxSetParameters</code> is equivalent to a call to <code>omxSetParameters</code> where you
have the ability to generate a pattern-based label list,
and, because this can create duplicate labels, we also call <code><a href="OpenMx.html#topic+omxAssignFirstParameters">omxAssignFirstParameters()</a></code>
to equate the start values for parameters which now have identical labels.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxModify">umxModify()</a></code>, <code><a href="#topic+xmuLabel">xmuLabel()</a></code>
</p>
</li></ul>

<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxCompare">umxCompare</a>()</code>,
<code><a href="#topic+umxEquate">umxEquate</a>()</code>,
<code><a href="#topic+umxMI">umxMI</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSummary">umxSummary</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
latents  = c("G")
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = mxData(demoOneFactor[1:80,], type = "raw"),
	umxPath(from = latents, to = manifests),
	umxPath(v.m. = manifests),
	umxPath(v1m0 = latents)
)
parameters(m1)
# Match all labels
umxSetParameters(m1, regex = "^", newlabels= "m1_", test = TRUE)
# Change path to x1 to x2, equating these two paths
m2 = umxSetParameters(m1, "G_to_x1", newlabels= "G_to_x2", test = FALSE)
parameters(m2) 


## End(Not run)
</code></pre>

<hr>
<h2 id='umxSexLim'>Multivariate sex limitation twin model</h2><span id='topic+umxSexLim'></span>

<h3>Description</h3>

<p>Multivariate twin analysis allowing for sex limitation (factors operate differently in males
vs. females) based on a correlated factors model. With 5-groups of twins, this model allows
for both Quantitative and Qualitative Sex-Limitation.
</p>
<p><em>Quantitative</em> differences refer to different amounts of phenotypic variance produced by
the same A, C, or E components when operating in one sex compared to the other sex.
</p>
<p><em>Qualitative</em> differences refer to phenotypic variance attributable to an A, C, or E
component which operates in one sex one but not in the other.
</p>
<p>The correlation approach ensures that variable order does not affect the ability
of the model to account for DZOS data.
</p>
<p><strong>1. Nonscalar Sex Limitation</strong>
</p>
<p>Allow quantitative (distinct male and female paths) and qualitative sex differences
on A or C. Allows distinct between variable correlations (<code>Ra</code>, <code>Rc</code> and <code>Re</code>)
for males and for females. Male-Female correlations also free (<code>Rao</code> or <code>Rco</code> free in DZO group).
</p>
<p><strong>2. Scalar Sex Limitation</strong>
</p>
<p>Quantitative sex differences only (distinct Male and female paths).
Just one set of Ra, Rc and Re between variables (same for males and females)
</p>
<p><strong>3. Homogeneity</strong>
</p>
<p>This is the model assumed by the basic ACE model: equal variance components in both sexes.
Different means may be allowed for males and females.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSexLim(
  name = "sexlim",
  selDVs,
  mzmData,
  dzmData,
  mzfData,
  dzfData,
  dzoData,
  sep = NA,
  A_or_C = c("A", "C"),
  sexlim = c("Nonscalar", "Scalar", "Homogeneity"),
  dzAr = 0.5,
  dzCr = 1,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSexLim_+3A_name">name</code></td>
<td>
<p>The name of the model (Default = &quot;sexlim&quot;)</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_seldvs">selDVs</code></td>
<td>
<p>BASE NAMES of the variables in the analysis. You MUST provide sep.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_mzmdata">mzmData</code></td>
<td>
<p>Dataframe containing the MZ male data.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_dzmdata">dzmData</code></td>
<td>
<p>Dataframe containing the DZ male data.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_mzfdata">mzfData</code></td>
<td>
<p>Dataframe containing the MZ female data.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_dzfdata">dzfData</code></td>
<td>
<p>Dataframe containing the DZ female data.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_dzodata">dzoData</code></td>
<td>
<p>Dataframe containing the DZ opposite-sex data (be sure and get in right order).</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_sep">sep</code></td>
<td>
<p>Suffix used for twin variable naming. Allows using just the base names in selVars.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_a_or_c">A_or_C</code></td>
<td>
<p>Whether to model sex-limitation on A or on C. (Defaults to &quot;A&quot;).</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_sexlim">sexlim</code></td>
<td>
<p>Which model type: &quot;Nonscalar&quot; (default), &quot;Scalar&quot;, or &quot;Homogeneity&quot;.</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (defaults to .5, vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults to 1: set to .25 to make an ADE model).</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to mxRun the model (default TRUE: the estimated model will be returned).</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxSexLim_+3A_optimizer">optimizer</code></td>
<td>
<p>optionally set the optimizer. Default (NULL) does nothing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>A or C</strong>
</p>
<p>Due to limitations on the degrees of freedom allowed by the twin model, we can model
qualitative sex differences for only one of A or C at a time.
</p>
<p><em>notes</em>: There is a half-way house model of heterogeneity in which a, c, and e components are scaled by a
scalar constant in one sex.
</p>
<p><em>General restrictions</em>: Assumes means and variances can be equated across birth order within zygosity groups.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> of subclass mxModel.CFSexLim
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Neale et al. (2006).
Multivariate genetic analysis of sex-lim and GxE interaction. <em>Twin Research &amp; Human Genetics</em>, <strong>9</strong>, pp. 481&ndash;489.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+umxSummarySexLim">umxSummarySexLim()</a></code>, <code><a href="#topic+umxPlotSexLim">umxPlotSexLim()</a></code>
</p>
<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># =========================
# = Load and Process Data =
# =========================
## Not run: 
require(umx)
data("us_skinfold_data")
# Rescale vars
us_skinfold_data[, c('bic_T1', 'bic_T2')] = us_skinfold_data[, c('bic_T1', 'bic_T2')]/3.4
us_skinfold_data[, c('tri_T1', 'tri_T2')] = us_skinfold_data[, c('tri_T1', 'tri_T2')]/3
us_skinfold_data[, c('caf_T1', 'caf_T2')] = us_skinfold_data[, c('caf_T1', 'caf_T2')]/3
us_skinfold_data[, c('ssc_T1', 'ssc_T2')] = us_skinfold_data[, c('ssc_T1', 'ssc_T2')]/5
us_skinfold_data[, c('sil_T1', 'sil_T2')] = us_skinfold_data[, c('sil_T1', 'sil_T2')]/5

# Data for each of the 5 twin-type groups
mzmData = subset(us_skinfold_data, zyg == 1)
mzfData = subset(us_skinfold_data, zyg == 2)
dzmData = subset(us_skinfold_data, zyg == 3)
dzfData = subset(us_skinfold_data, zyg == 4)
dzoData = subset(us_skinfold_data, zyg == 5)

umxSummarizeTwinData(us_skinfold_data, selVars="bic",zyg="zyg", sep="_T",
		MZFF=2, DZFF=4, MZMM=1, DZMM=3, DZOS=5
)

# ==========================
# = Run univariate example =
# ==========================

m1 = umxSexLim(selDVs = "bic", sep = "_T", A_or_C = "A", tryHard = "yes",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)

# Drop qualitative sex limitation
m1a = umxModify(m1, regex = "^Rao_", value=1, name = "no_qual", comparison = TRUE)


# Equate a, ac, and try ace across m &amp; f in scalar model
m1b = umxModify(m1a, regex = "^a[fm]_", newlabels="a_", name = "eq_a_no_qual", comparison = TRUE)
m1c = umxModify(m1b, regex = "^c[fm]_", newlabels="c_", name = "eq_ac_no_qual", comparison = TRUE)
m1d = umxModify(m1c, regex = "^e[fm]_", newlabels="e_", name = "eq_ace_no_qual", comparison = TRUE)
umxCompare(m1, c(m1a, m1b, m1c, m1d))

# ============================
# = Scalar Sex Limitation =
# ============================

m2 = umxSexLim(selDVs = "bic", sep = "_T", sexlim = "Scalar", tryHard = "yes",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
) 

# Show our manual drop of qualitative is the same as umxSexLim with sexlim= "scalar"s
umxCompare(m1a, m2)

# ===============
# = Homogeneity =
# ===============

m3 = umxSexLim(selDVs = "bic", sep = "_T", sexlim = "Homogeneity", tryHard = "yes",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)
umxCompare(m1, c(m2, m3))

# ===========================================
# = Bivariate example with manual reduction =
# ===========================================
m1 = umxSexLim(selDVs = c("bic", "tri"), sep = "_T", A_or_C = "A", tryHard="yes",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)

# Scalar sex limitation (same correlation among components for m and f)
m2 = umxSexLim(selDVs = c("bic", "tri"), sep = "_T", 
	A_or_C = "A", tryHard="yes", sexlim="Scalar",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)
# Drop qualitative sex limitation
#  Distinct af and am (&amp; c &amp; e), but shared Ra (&amp; Rc &amp; Re) between variables
#  	i.e., same correlations for males and females.
m1a = umxModify(m1 , regex = "^Ra[mfo]_", newlabels="^Ra_", name = "no_qual_a", comparison = TRUE)
m1b = umxModify(m1a, regex = "^Rc[mfo]_", newlabels="^Rc_", name = "no_qual_ac", comparison = TRUE)
m1c = umxModify(m1b, regex = "^Re[mfo]_", newlabels="^Re_", name = "no_qual_ace", comparison = TRUE)
umxCompare(m1, c(m1a, m1b, m1c, m2))

# In one smart regular expression
m2 = umxModify(m1, regex = "^R([ace])[fmo]_", newlabels = "R\\1_", 
  name = "scalar", comparison = TRUE)

# Equate a, ac, and try ace across m &amp; f in scalar model
m2a = umxModify(m2 , regex = "^a[fm]_", newlabels="a_", name = "eq_a_no_qual"  , comparison = TRUE)
m2b = umxModify(m2a, regex = "^c[fm]_", newlabels="c_", name = "eq_ac_no_qual" , comparison = TRUE)
m2c = umxModify(m2b, regex = "^e[fm]_", newlabels="e_", name = "eq_ace_no_qual", comparison = TRUE)
umxCompare(m1, c(m1a, m1b, m1c, m1d))

# =============================
# = Run multi-variate example =
# =============================
# Variables for Analysis
selDVs = c('ssc','sil','caf','tri','bic')
selDVs = c('ssc','tri','bic')
m1 = umxSexLim(selDVs = selDVs, sep = "_T", A_or_C = "A", tryHard = "yes",
mzmData = mzmData, dzmData = dzmData, 
 mzfData = mzfData, dzfData = dzfData, dzoData = dzoData
)

m2 = umxSexLim(selDVs = selDVs, sep = "_T", A_or_C = "A", sexlim = "Nonscalar",
	tryHard = "yes",
mzmData = mzmData, dzmData = dzmData, 
 mzfData = mzfData, dzfData = dzfData, dzoData = dzoData
)

# umxSummary(m1)
# summary(m1)
# summary(m1)$Mi

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSimplex'>Build and run a simplex twin model (not ready for use!)</h2><span id='topic+umxSimplex'></span>

<h3>Description</h3>

<p>The simplex model provides a powerful tool for theory-based decomposition of genetic
and environmental differences. <code>umxSimplex</code> makes a 2-group simplex twin model.
</p>
<p><strong>This code is beta</strong> quality: <strong>not</strong> for publication use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSimplex(
  name = "simplex",
  selDVs,
  dzData,
  mzData,
  sep = "_T",
  equateMeans = TRUE,
  dzAr = 0.5,
  dzCr = 1,
  addStd = TRUE,
  addCI = TRUE,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSimplex_+3A_name">name</code></td>
<td>
<p>The name of the model (defaults to &quot;simplex&quot;)</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_seldvs">selDVs</code></td>
<td>
<p>The BASENAMES of the variables i.e., c(<code>obese</code>), not c(<code>obese_T1</code>, <code>obese_T2</code>)</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ dataframe</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ dataframe</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_sep">sep</code></td>
<td>
<p>The string preceding the final numeric twin identifier (often &quot;_T&quot;)
Combined with selDVs to form the full var names, i.e., just &quot;dep&quot; &ndash;&gt; c(&quot;dep_T1&quot;, &quot;dep_T2&quot;)</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (defaults to TRUE).</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_dzar">dzAr</code></td>
<td>
<p>The DZ genetic correlation (default = .5. Vary to examine assortative mating).</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_dzcr">dzCr</code></td>
<td>
<p>The DZ &quot;C&quot; correlation (defaults = 1. To make an ADE model, set = .25).</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_addstd">addStd</code></td>
<td>
<p>Whether to add the algebras to compute a std model (default = TRUE).</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_addci">addCI</code></td>
<td>
<p>Whether to add the interval requests for CIs (default = TRUE).</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxSimplex_+3A_optimizer">optimizer</code></td>
<td>
<p>Optionally set the optimizer (default NULL does nothing).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The simplex model decomposes phenotypic variance
into Additive genetic, unique environmental (E) and, optionally, either
common or shared-environment (C) or non-additive genetic effects (D).
</p>
<p>In the simplex model, these influences are modeled as a combination of:
</p>

<ul>
<li><p> Innovations at a given time (<code>ai</code> <code>ci</code> and <code>ei</code> matrices).
</p>
</li>
<li><p> Influences transmitted from previous time (<code>at</code>, <code>ct</code>, and <code>et</code> matrices).
</p>
</li>
<li><p> Influences specific to a single time (<code>as</code>, <code>cs</code>, <code>es</code>).
</p>
</li></ul>

<p>These combine to explain the causes of variance in the phenotype (see Figure).
</p>
<p><strong>Simplex path diagram</strong>:
</p>
<p><img src="../help/figures/simplex.png" width=50% alt="Figure: simplex.png" />

</p>
<p><strong>Data Input</strong>
Currently, the umxSimplex function accepts only raw data.
</p>
<p><strong>Ordinal Data</strong>
In an important capability, the model transparently handles ordinal (binary or multi-level
ordered factor data) inputs, and can handle mixtures of continuous, binary, and ordinal
data in any combination.
</p>
<p><strong>Additional features</strong>
The <code>umxSimplex</code> function supports varying the DZ genetic association (defaulting to .5)
to allow exploring assortative mating effects, as well as varying the DZ &ldquo;C&rdquo; factor
from 1 (the default for modeling family-level effects shared 100% by twins in a pair),
to .25 to model dominance effects.
</p>
<p><strong>Matrices and Labels in the simplex model</strong>
A good way to see which matrices are used in umxSummary is to run an example model and plot it.
</p>
<p>The loadings specific to each time point are contained on the diagonals of matrices
<code>as</code>, <code>cs</code>, and <code>es</code>. So labels relevant to modifying these are of the form &quot;as_r1c1&quot;, &quot;as_r2c2&quot; etc.
</p>
<p>All the shared matrices are in the model &quot;top&quot;. So to see the 'as' values, you can simply execute:
</p>
<p><code>m1$top$as$values</code>
</p>
<p>The transmitted loadings are in matrices <code>at</code>, <code>ct</code>, <code>et</code>.
</p>
<p>The innovations are in the matrix <code>ai</code>, <code>ci</code>, and <code>ei</code>.
</p>
<p>Less commonly-modified matrices are the mean matrix <code>expMean</code>.
This has 1 row, and the columns are laid out for each variable for
twin 1, followed by each variable for twin 2.
</p>
<p>Thus, in a model where the means for twin 1 and twin 2 had been equated
(set = to T1), you could make them independent again with this script:
</p>
<p><code>m1$top$expMean$labels[1,4:6] = c("expMean_r1c4", "expMean_r1c5", "expMean_r1c6")</code>
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxACE">umxACE()</a></code> for more examples of twin modeling, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(iqdat)
mzData = subset(iqdat, zygosity == "MZ")
dzData = subset(iqdat, zygosity == "DZ")
baseVars = c("IQ_age1", "IQ_age2", "IQ_age3", "IQ_age4")
m1= umxSimplex(selDVs= baseVars, dzData= dzData, mzData= mzData, sep= "_T", tryHard= "yes")

umxSummary(m1)
parameters(m1, patt = "^s")
m2 = umxModify(m1, regex = "as_r1c1", name = "no_as", comp = TRUE)
umxCompare(m1, m2)

# =============================
# = Test a 3 time-point model =
# =============================
m1 = umxSimplex(selDVs = paste0("IQ_age", 1:3), 
dzData = dzData, mzData = mzData, tryHard = "yes")

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummarizeTwinData'>Summarize twin data</h2><span id='topic+umxSummarizeTwinData'></span>

<h3>Description</h3>

<p>Produce a summary of wide-format twin data, showing the number of individuals, the mean and SD for each trait, and the correlation for each twin-type.
</p>
<p>Set MZ and DZ to summarize the two-group case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummarizeTwinData(
  data = NULL,
  selVars = NULL,
  sep = "_T",
  zyg = "zygosity",
  age = "age",
  MZ = NULL,
  DZ = NULL,
  MZFF = "MZFF",
  DZFF = "DZFF",
  MZMM = "MZMM",
  DZMM = "DZMM",
  DZOS = "DZOS",
  digits = 2,
  report = c("markdown", "html")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummarizeTwinData_+3A_data">data</code></td>
<td>
<p>The twin data.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_selvars">selVars</code></td>
<td>
<p>Collection of variables to report on, e.g. c(&quot;wt&quot;, &quot;ht&quot;).</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_sep">sep</code></td>
<td>
<p>The separator string that will turn a variable name into a twin variable name, default= &quot;_T&quot; for wt_T1 and wt_T2.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_zyg">zyg</code></td>
<td>
<p>The zygosity column in the dataset (default &quot;zygosity&quot;).</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_age">age</code></td>
<td>
<p>The age column in the dataset (default &quot;age&quot;)</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_mz">MZ</code></td>
<td>
<p>Set level in zyg corresponding to MZ for two group case (defaults to using 5-group case).</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_dz">DZ</code></td>
<td>
<p>Set level in zyg corresponding to DZ for two group case (defaults to using 5-group case).</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_mzff">MZFF</code></td>
<td>
<p>The level of zyg corresponding to MZ FF pairs: default= &quot;MZFF&quot;.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_dzff">DZFF</code></td>
<td>
<p>The level of zyg corresponding to DZ FF pairs: default= &quot;DZFF&quot;.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_mzmm">MZMM</code></td>
<td>
<p>The level of zyg corresponding to MZ MM pairs: default= &quot;MZMM&quot;.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_dzmm">DZMM</code></td>
<td>
<p>The level of zyg corresponding to DZ MM pairs: default= &quot;DZMM&quot;.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_dzos">DZOS</code></td>
<td>
<p>The level of zyg corresponding to DZ OS pairs: default= &quot;DZOS&quot;.</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_digits">digits</code></td>
<td>
<p>Rounding precision of the report (default 2).</p>
</td></tr>
<tr><td><code id="umxSummarizeTwinData_+3A_report">report</code></td>
<td>
<p>What to return (default = 'markdown'). Use 'html' to open a web table.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> formatted table, e.g. in markdown.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxAPA">umxAPA()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
umxSummarizeTwinData(twinData, sep = "", selVars = c("wt", "ht"))
MZs = c("MZMM", "MZFF"); DZs = c("DZFF","DZMM", "DZOS")
umxSummarizeTwinData(twinData, sep = "", selVars = c("wt", "ht"), MZ = MZs, DZ = DZs)
</code></pre>

<hr>
<h2 id='umxSummary'>Shows a compact, publication-style, summary of umx models</h2><span id='topic+umxSummary'></span>

<h3>Description</h3>

<p>Report the fit of a OpenMx model or specialized model class (such as ACE, CP etc.)
in a compact form suitable for reporting in a journal.
</p>
<p>See documentation for RAM models summary here: <code><a href="#topic+umxSummary.MxModel">umxSummary.MxModel()</a></code>.
</p>
<p>View documentation on the ACE model subclass here: <code><a href="#topic+umxSummaryACE">umxSummaryACE()</a></code>.
</p>
<p>View documentation on the ACEv model subclass here: <code><a href="#topic+umxSummaryACEv">umxSummaryACEv()</a></code>.
</p>
<p>View documentation on the IP model subclass here: <code><a href="#topic+umxSummaryIP">umxSummaryIP()</a></code>.
</p>
<p>View documentation on the CP model subclass here: <code><a href="#topic+umxSummaryCP">umxSummaryCP()</a></code>.
</p>
<p>View documentation on the GxE model subclass here: <code><a href="#topic+umxSummaryGxE">umxSummaryGxE()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummary(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummary_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> whose fit will be reported</p>
</td></tr>
<tr><td><code id="umxSummary_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other Model Summary and Comparison: 
<code><a href="#topic+umxCompare">umxCompare</a>()</code>,
<code><a href="#topic+umxEquate">umxEquate</a>()</code>,
<code><a href="#topic+umxMI">umxMI</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxSetParameters">umxSetParameters</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>

<hr>
<h2 id='umxSummary.MxModel'>Shows a compact, publication-style, summary of a RAM model</h2><span id='topic+umxSummary.MxModel'></span><span id='topic+umxSummary.MxRAMModel'></span>

<h3>Description</h3>

<p>Report the fit of a model in a compact form suitable for a journal.
It reports parameters in a markdown or html table (optionally standardized), and fit indices
RMSEA (an absolute fit index, comparing the model to a perfect model) and CFI and TLI (incremental fit indices comparing model a model with the worst fit).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MxModel'
umxSummary(
  model,
  refModels = NULL,
  std = FALSE,
  digits = 2,
  report = c("markdown", "html"),
  means = TRUE,
  residuals = TRUE,
  SE = TRUE,
  filter = c("ALL", "NS", "SIG"),
  RMSEA_CI = FALSE,
  ...,
  matrixAddresses = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummary.MxModel_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> whose fit will be reported</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_refmodels">refModels</code></td>
<td>
<p>Saturated models if needed for fit indices (see example below:
If NULL will be computed on demand. If FALSE will not be computed.</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_std">std</code></td>
<td>
<p>If TRUE, model is standardized (Default FALSE, NULL means &quot;don't show&quot;).</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_digits">digits</code></td>
<td>
<p>How many decimal places to report (Default 2)</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_report">report</code></td>
<td>
<p>If &quot;html&quot;, then show results in browser (default = &quot;markdown&quot;)</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_means">means</code></td>
<td>
<p>Whether to include means in the summary (TRUE)</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_residuals">residuals</code></td>
<td>
<p>Whether to include residuals in the summary (TRUE)</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_se">SE</code></td>
<td>
<p>Whether to compute SEs... defaults to TRUE. In rare cases, you might need to turn off to avoid errors.</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_filter">filter</code></td>
<td>
<p>whether to show significant paths (SIG) or NS paths (NS) or all paths (ALL)</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_rmsea_ci">RMSEA_CI</code></td>
<td>
<p>Whether to compute the CI on RMSEA (Defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
<tr><td><code id="umxSummary.MxModel_+3A_matrixaddresses">matrixAddresses</code></td>
<td>
<p>Whether to show &quot;matrix address&quot; columns (Default = FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>umxSummary</code> alerts you when model fit is worse than accepted criterion (TLI &gt;= .95 and RMSEA &lt;= .06; (Hu &amp; Bentler, 1999; Yu, 2002).
</p>
<p>Note: For some (multi-group) models, you will need to fall back on <code><a href="base.html#topic+summary">summary()</a></code>
</p>
<p><strong>CIs and Identification</strong>
This function uses the standard errors reported by OpenMx to produce the CIs you see in umxSummary
These are used to derive confidence intervals based on the formula 95%CI = estimate +/- 1.96*SE)
</p>
<p>Sometimes SEs appear NA. This may reflect a model which is not identified (see <a href="http://davidakenny.net/cm/identify.htm">http://davidakenny.net/cm/identify.htm</a>).
This can include empirical under-identification - for instance two factors
that are essentially identical in structure. use <code><a href="OpenMx.html#topic+mxCheckIdentification">mxCheckIdentification()</a></code> to check identification.
</p>
<p>Solutions: If there are paths estimated at or close to zero suggests that fixing one or two of
these to zero may fix the standard error calculation.
</p>
<p>If factor loadings can flip sign and provide identical fit, this creates another form of
under-identification and can break confidence interval estimation.
<em>Solution</em>: Fixing a factor loading to 1 and estimating factor variances can help here.
</p>


<h3>Value</h3>


<ul>
<li><p> parameterTable returned invisibly, if estimates requested
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Hu, L., &amp; Bentler, P. M. (1999). Cutoff criteria for fit indexes in covariance
structure analysis: Conventional criteria versus new alternatives. <em>Structural Equation Modeling</em>, <strong>6</strong>, 1-55.
</p>
</li>
<li><p> Yu, C.Y. (2002). Evaluating cutoff criteria of model fit indices for latent variable models
with binary and continuous outcomes. University of California, Los Angeles, Los Angeles.
Retrieved from <a href="https://www.statmodel.com/download/Yudissertation.pdf">https://www.statmodel.com/download/Yudissertation.pdf</a>
</p>
</li></ul>

<p><a href="https://tbates.github.io">https://tbates.github.io</a>
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other Summary functions: 
<code><a href="#topic+umxSummaryACEcov">umxSummaryACEcov</a>()</code>,
<code><a href="#topic+umxSummaryCP">umxSummaryCP</a>()</code>,
<code><a href="#topic+umxSummaryGxE">umxSummaryGxE</a>()</code>,
<code><a href="#topic+umxSummaryIP">umxSummaryIP</a>()</code>,
<code><a href="#topic+umxSummaryMRDoC">umxSummaryMRDoC</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
umxSummary(m1, std = TRUE)
# output as latex
umx_set_table_format("latex")
umxSummary(m1, std = TRUE)
umx_set_table_format("markdown")
# output as raw
umxSummary(m1, std = FALSE)

# switch to a raw data model
m1 = umxRAM("One Factor", data = demoOneFactor[1:100, ],
	umxPath("G", to = manifests),
	umxPath(v.m. = manifests),
	umxPath(v1m0 = "G")
)
umxSummary(m1, std = TRUE, filter = "NS")

## End(Not run)

</code></pre>

<hr>
<h2 id='umxSummaryACE'>Shows a compact, publication-style, summary of a umx Cholesky ACE model</h2><span id='topic+umxSummaryACE'></span><span id='topic+umxSummary.MxModelACE'></span>

<h3>Description</h3>

<p>Summarize a fitted Cholesky model returned by <code><a href="#topic+umxACE">umxACE()</a></code>. Can control digits, report comparison model fits,
optionally show the Rg (genetic and environmental correlations), and show confidence intervals. the report parameter allows
drawing the tables to a web browser where they may readily be copied into non-markdown programs like Word.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryACE(
  model,
  digits = 2,
  comparison = NULL,
  std = TRUE,
  showRg = FALSE,
  CIs = TRUE,
  report = c("markdown", "html"),
  file = getOption("umx_auto_plot"),
  returnStd = FALSE,
  extended = FALSE,
  zero.print = ".",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryACE_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to summarize.</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2).
Defaults to NA = do not create plot output.</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_comparison">comparison</code></td>
<td>
<p>you can run mxCompare on a comparison model (NULL).</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_std">std</code></td>
<td>
<p>Whether to standardize the output (default = TRUE).</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_showrg">showRg</code></td>
<td>
<p>= whether to show the genetic correlations (FALSE).</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_cis">CIs</code></td>
<td>
<p>Whether to show Confidence intervals if they exist (TRUE).</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_report">report</code></td>
<td>
<p>If &quot;html&quot;, then open an html table of the results.</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_file">file</code></td>
<td>
<p>The name of the dot file for figure: &quot;name&quot; = use the name of the model.</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_extended">extended</code></td>
<td>
<p>how much to report (FALSE).</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_zero.print">zero.print</code></td>
<td>
<p>How to show zeros (&quot;.&quot;)</p>
</td></tr>
<tr><td><code id="umxSummaryACE_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See documentation for other umx models here: <code><a href="#topic+umxSummary">umxSummary()</a></code>.
</p>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxACE">umxACE()</a></code>, <code><a href="#topic+plot.MxModelACE">plot.MxModelACE()</a></code>, <code><a href="#topic+umxModify">umxModify()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData)
selDVs = c("bmi1", "bmi2")
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACE(selDVs = selDVs, dzData = dzData, mzData = mzData)
umxSummary(m1)
umxSummaryACE(m1, file = NA);
umxSummaryACE(m1, file = "name", std = TRUE)
stdFit = umxSummaryACE(m1, returnStd = TRUE);

## End(Not run)

</code></pre>

<hr>
<h2 id='umxSummaryACEcov'>Present results of a twin ACE-model with covariates in table and graphical forms.</h2><span id='topic+umxSummaryACEcov'></span><span id='topic+umxSummary.MxModelACEcov'></span>

<h3>Description</h3>

<p>Summarize a Cholesky model with random-effects covariates, as returned by <code><a href="#topic+umxACEcov">umxACEcov()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryACEcov(
  model,
  digits = 2,
  showRg = FALSE,
  std = TRUE,
  comparison = NULL,
  CIs = TRUE,
  zero.print = ".",
  report = c("markdown", "html"),
  file = getOption("umx_auto_plot"),
  returnStd = FALSE,
  extended = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryACEcov_+3A_model">model</code></td>
<td>
<p>A <code><a href="#topic+umxACEcov">umxACEcov()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_digits">digits</code></td>
<td>
<p>Round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_showrg">showRg</code></td>
<td>
<p>= Whether to show the genetic correlations (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_std">std</code></td>
<td>
<p>= Whether to show the standardized model (TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_comparison">comparison</code></td>
<td>
<p>You can run mxCompare on a comparison model (NULL)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_cis">CIs</code></td>
<td>
<p>Whether to show Confidence intervals if they exist (TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_zero.print">zero.print</code></td>
<td>
<p>How to show zeros (&quot;.&quot;)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_report">report</code></td>
<td>
<p>If &quot;html&quot;, then open an html table of the results.</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_extended">extended</code></td>
<td>
<p>How much to report (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEcov_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxACEcov">umxACEcov()</a></code>
</p>
</li></ul>

<p>Other Summary functions: 
<code><a href="#topic+umxSummary.MxModel">umxSummary.MxModel</a>()</code>,
<code><a href="#topic+umxSummaryCP">umxSummaryCP</a>()</code>,
<code><a href="#topic+umxSummaryGxE">umxSummaryGxE</a>()</code>,
<code><a href="#topic+umxSummaryIP">umxSummaryIP</a>()</code>,
<code><a href="#topic+umxSummaryMRDoC">umxSummaryMRDoC</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACEcov(selDVs = c("bmi", "wt"), selCovs = "ht", dzData = dzData, mzData = mzData, sep="")
umxSummaryACEcov(m1, file = NA)
umxSummaryACEcov(m1, file = "name", std = TRUE)
stdFit = umxSummary(m1, returnStd = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummaryACEv'>Shows a compact, publication-style, summary of a variance-based Cholesky ACE model.</h2><span id='topic+umxSummaryACEv'></span><span id='topic+umxSummary.MxModelACEv'></span>

<h3>Description</h3>

<p>Summarize a fitted Cholesky model returned by <code><a href="#topic+umxACEv">umxACEv()</a></code>. Can control digits, report comparison model fits,
optionally show the Rg (genetic and environmental correlations), and show confidence intervals. the report parameter allows
drawing the tables to a web browser where they may readily be copied into non-markdown programs like Word.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryACEv(
  model,
  digits = 2,
  comparison = NULL,
  std = TRUE,
  showRg = FALSE,
  CIs = TRUE,
  report = c("markdown", "html"),
  file = getOption("umx_auto_plot"),
  returnStd = FALSE,
  extended = FALSE,
  zero.print = ".",
  show = c("std", "raw"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryACEv_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_comparison">comparison</code></td>
<td>
<p>you can run mxCompare on a comparison model (NULL)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_std">std</code></td>
<td>
<p>Whether to standardize the output (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_showrg">showRg</code></td>
<td>
<p>= whether to show the genetic correlations (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_cis">CIs</code></td>
<td>
<p>Whether to show Confidence intervals if they exist (TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_report">report</code></td>
<td>
<p>If &quot;html&quot;, then open an html table of the results</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: &quot;name&quot; = use the name of the model.
Defaults to getOption(&quot;umx_auto_plot&quot;), which is likely &quot;name&quot;.</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_extended">extended</code></td>
<td>
<p>how much to report (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_zero.print">zero.print</code></td>
<td>
<p>How to show zeros (&quot;.&quot;)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_show">show</code></td>
<td>
<p>Here to support being called from generic xmu_safe_run_summary. User should ignore: can be c(&quot;std&quot;, &quot;raw&quot;)</p>
</td></tr>
<tr><td><code id="umxSummaryACEv_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See documentation for other umx models here: <code><a href="#topic+umxSummary">umxSummary()</a></code>.
</p>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxACEv">umxACEv()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(twinData)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")
m1 = umxACEv(selDVs = "bmi", sep = "", dzData = dzData, mzData = mzData)
umxSummary(m1, std = FALSE)
## Not run: 
umxSummary(m1, file = NA);
umxSummary(m1, file = "name", std = TRUE)
stdFit = umxSummary(m1, returnStd = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummaryCP'>Present the results of a Common-pathway twin model in table and graphical form</h2><span id='topic+umxSummaryCP'></span><span id='topic+umxSummary.MxModelCP'></span>

<h3>Description</h3>

<p>Summarizes a Common-Pathway model, as returned by <code><a href="#topic+umxCP">umxCP()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryCP(
  model,
  digits = 2,
  std = TRUE,
  CIs = FALSE,
  showRg = FALSE,
  comparison = NULL,
  report = c("markdown", "html"),
  file = getOption("umx_auto_plot"),
  returnStd = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryCP_+3A_model">model</code></td>
<td>
<p>A fitted <code><a href="#topic+umxCP">umxCP()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_digits">digits</code></td>
<td>
<p>Round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_std">std</code></td>
<td>
<p>Whether to show the standardized model (TRUE) (ignored: used extended = TRUE to get unstandardized)</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_cis">CIs</code></td>
<td>
<p>Confidence intervals (default FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_showrg">showRg</code></td>
<td>
<p>Whether to show the genetic correlations (default FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_comparison">comparison</code></td>
<td>
<p>Run mxCompare on a comparison model (default NULL)</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_report">report</code></td>
<td>
<p>Print tables to the console (as 'markdown'), or open in browser ('html')</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryCP_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxCP">umxCP</a>()</code>, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>

<p>Other Summary functions: 
<code><a href="#topic+umxSummary.MxModel">umxSummary.MxModel</a>()</code>,
<code><a href="#topic+umxSummaryACEcov">umxSummaryACEcov</a>()</code>,
<code><a href="#topic+umxSummaryGxE">umxSummaryGxE</a>()</code>,
<code><a href="#topic+umxSummaryIP">umxSummaryIP</a>()</code>,
<code><a href="#topic+umxSummaryMRDoC">umxSummaryMRDoC</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData)

twinData$wt1 = twinData$wt1/10
twinData$wt2 = twinData$wt2/10
selDVs = c("ht", "wt")
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "DZFF")

m1 = umxCP(selDVs = selDVs, dzData = dzData, mzData = mzData, sep = "", optimizer = "SLSQP")
umxSummaryCP(m1, file = NA) # Suppress plot creation with file
umxSummary(m1, file = NA)   # Generic summary is the same
stdFit = umxSummaryCP(m1, digits = 2, std = TRUE, file = NA, returnStd = TRUE);

umxSummary(m1, std = FALSE, showRg = TRUE, file = NA);
umxSummary(m1, std = FALSE, file = NA)

# =================
# = Print example =
# =================
umxSummary(m1, file = "Figure 3", std = TRUE)

# =================
# = Confint example =
# =================
m1 = umxConfint(m1, "smart", run = FALSE);
m1 = umxConfint(m1, "smart", run = TRUE);
umxSummary(m1, CIs = TRUE, file = NA);

## End(Not run)

</code></pre>

<hr>
<h2 id='umxSummaryDoC'>Shows a compact, publication-style, summary of a umx Direction of Causation model</h2><span id='topic+umxSummaryDoC'></span><span id='topic+umxSummary.MxModelDoC'></span>

<h3>Description</h3>

<p>Summarize a fitted model returned by <code><a href="#topic+umxDoC">umxDoC()</a></code>. Can control digits, report comparison model fits,
optionally show the Rg (genetic and environmental correlations), and show confidence intervals. the report parameter allows
drawing the tables to a web browser where they may readily be copied into non-markdown programs like Word.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryDoC(
  model,
  digits = 2,
  comparison = NULL,
  std = TRUE,
  showRg = FALSE,
  CIs = TRUE,
  report = c("markdown", "html"),
  file = getOption("umx_auto_plot"),
  returnStd = FALSE,
  zero.print = ".",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryDoC_+3A_model">model</code></td>
<td>
<p>a fitted <code><a href="#topic+umxDoC">umxDoC()</a></code> model to summarize.</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2).</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_comparison">comparison</code></td>
<td>
<p>Run mxCompare on a comparison model (default NULL)</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_std">std</code></td>
<td>
<p>Whether to standardize the output (default = TRUE).</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_showrg">showRg</code></td>
<td>
<p>= whether to show the genetic correlations (FALSE).</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_cis">CIs</code></td>
<td>
<p>Whether to show Confidence intervals if they exist (TRUE).</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_report">report</code></td>
<td>
<p>Print tables to the console (as 'markdown'), or open in browser ('html')</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: &quot;name&quot; = use the name of the model.
Defaults to NA = do not create plot output.</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE).</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_zero.print">zero.print</code></td>
<td>
<p>How to show zeros (&quot;.&quot;)</p>
</td></tr>
<tr><td><code id="umxSummaryDoC_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See documentation for other umx models here: <code><a href="#topic+umxSummary">umxSummary()</a></code>.
</p>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC()</a></code>, <code><a href="#topic+plot.MxModelDoC">plot.MxModelDoC()</a></code>, <code><a href="#topic+umxModify">umxModify()</a></code>, <code><a href="#topic+umxCP">umxCP()</a></code>, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ================
# = 1. Load Data =
# ================
data(docData)
mzData = subset(docData, zygosity %in% c("MZFF", "MZMM"))
dzData = subset(docData, zygosity %in% c("DZFF", "DZMM"))

# =======================================
# = 2. Define manifests for var 1 and 2 =
# =======================================
var1 = paste0("varA", 1:3)
var2 = paste0("varB", 1:3)

# =======================================================
# = 2. Make the non-causal (Cholesky) and causal models =
# =======================================================
Chol= umxDoC(var1= var1, var2= var2, mzData= mzData, dzData= dzData, causal= FALSE)
DoC = umxDoC(var1= var1, var2= var2, mzData= mzData, dzData= dzData, causal= TRUE)

# ================================================
# = Make the directional models by modifying DoC =
# ================================================
A2B = umxModify(DoC, "a2b", free = TRUE, name = "A2B")
A2B = umxModify(DoC, "a2b", free = TRUE, name = "A2B", comp=TRUE)
B2A = umxModify(DoC, "b2a", free = TRUE, name = "B2A", comp=TRUE)
umxCompare(B2A, A2B)


## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummaryGxE'>Summarize a GxE model</h2><span id='topic+umxSummaryGxE'></span><span id='topic+umxSummary.MxModelGxE'></span>

<h3>Description</h3>

<p>Summarize a genetic moderation model, as returned by <code><a href="#topic+umxGxE">umxGxE()</a></code>. Prints graphs of A, C, and E, standardized and raw.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryGxE(
  model = NULL,
  digits = 2,
  xlab = NA,
  location = "topleft",
  separateGraphs = FALSE,
  gg = TRUE,
  file = getOption("umx_auto_plot"),
  returnStd = NULL,
  std = NULL,
  reduce = FALSE,
  CIs = NULL,
  report = c("markdown", "html"),
  show = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryGxE_+3A_model">model</code></td>
<td>
<p>A fitted <code><a href="#topic+umxGxE">umxGxE()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_xlab">xlab</code></td>
<td>
<p>label for the x-axis of plot</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_location">location</code></td>
<td>
<p>default = &quot;topleft&quot;</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_separategraphs">separateGraphs</code></td>
<td>
<p>If TRUE, both std and raw plots in one figure (default FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_gg">gg</code></td>
<td>
<p>Whether to use ggplot to create the graphs (default TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_std">std</code></td>
<td>
<p>Whether to show the standardized model (not implemented! TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_reduce">reduce</code></td>
<td>
<p>Whether run and tabulate a complete model reduction...(Defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_cis">CIs</code></td>
<td>
<p>Confidence intervals (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_report">report</code></td>
<td>
<p>&quot;markdown&quot; or &quot;html&quot; = open a browser for copyable tables</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_show">show</code></td>
<td>
<p>not doing anything yet (required for all summary functions)</p>
</td></tr>
<tr><td><code id="umxSummaryGxE_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: see also <code><a href="#topic+umxReduce">umxReduce()</a></code> which knows how to reduce a GxE model.
</p>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxGxE">umxGxE()</a></code>, <code><a href="#topic+umxReduce">umxReduce()</a></code>, <code><a href="#topic+plot">plot()</a></code>, [umxSummary)] all work for IP, CP, GxE, and ACE models.
</p>
</li></ul>

<p>[umxSummary)]: R:umxSummary)
</p>
<p>Other Summary functions: 
<code><a href="#topic+umxSummary.MxModel">umxSummary.MxModel</a>()</code>,
<code><a href="#topic+umxSummaryACEcov">umxSummaryACEcov</a>()</code>,
<code><a href="#topic+umxSummaryCP">umxSummaryCP</a>()</code>,
<code><a href="#topic+umxSummaryIP">umxSummaryIP</a>()</code>,
<code><a href="#topic+umxSummaryMRDoC">umxSummaryMRDoC</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# The total sample has been subdivided into a young cohort, 
# aged 18-30 years, and an older cohort aged 31 and above.
# Cohort 1 Zygosity is coded as follows 1 == MZ females 2 == MZ males 
# 3 == DZ females 4 == DZ males 5 == DZ opposite sex pairs
require(umx)
data(twinData) 
twinData$age1 = twinData$age2 = twinData$age
selDVs  = c("bmi1", "bmi2")
selDefs = c("age1", "age2")
selVars = c(selDVs, selDefs)
mzData  = subset(twinData, zygosity == "MZFF", selVars)
dzData  = subset(twinData, zygosity == "DZMM", selVars)
# Exclude cases with missing Def
mzData = mzData[!is.na(mzData[selDefs[1]]) &amp; !is.na(mzData[selDefs[2]]),]
dzData = dzData[!is.na(dzData[selDefs[1]]) &amp; !is.na(dzData[selDefs[2]]),]
m1 = umxGxE(selDVs = "bmi", selDefs = "age", sep="", dzData = dzData, mzData = mzData)
# Plot Moderation
umxSummaryGxE(m1)
umxSummaryGxE(m1, location = "topright")
umxSummaryGxE(m1, separateGraphs = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummaryGxEbiv'>Summarize a bivariate GxE twin model</h2><span id='topic+umxSummaryGxEbiv'></span><span id='topic+umxSummary.MxModelGxEbiv'></span>

<h3>Description</h3>

<p><code>umxSummaryGxEbiv</code> summarizes a bivariate moderation model, as returned by <code><a href="#topic+umxGxEbiv">umxGxEbiv()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryGxEbiv(
  model = NULL,
  digits = 2,
  xlab = NA,
  location = "topleft",
  separateGraphs = FALSE,
  file = getOption("umx_auto_plot"),
  comparison = NULL,
  std = NULL,
  reduce = FALSE,
  CIs = NULL,
  report = c("markdown", "html"),
  returnStd = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryGxEbiv_+3A_model">model</code></td>
<td>
<p>A fitted <code><a href="#topic+umxGxEbiv">umxGxEbiv()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_xlab">xlab</code></td>
<td>
<p>label for the x-axis of plot</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_location">location</code></td>
<td>
<p>default = &quot;topleft&quot;</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_separategraphs">separateGraphs</code></td>
<td>
<p>Std and raw plots in separate graphs? (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_comparison">comparison</code></td>
<td>
<p>mxCompare model with this model if offered up (default = NULL).</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_std">std</code></td>
<td>
<p>Whether to show the standardized model (not implemented! TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_reduce">reduce</code></td>
<td>
<p>Whether to run and tabulate a complete model reduction...(Defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_cis">CIs</code></td>
<td>
<p>Confidence intervals (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_report">report</code></td>
<td>
<p>markdown or html (html opens in browser)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryGxEbiv_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, and ACE models.
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
df = umx_scale_wide_twin_data(twinData, varsToScale = c("ht", "wt"), sep = "")
mzData  = subset(df, zygosity %in% c("MZFF", "MZMM"))
dzData  = subset(df, zygosity %in% c("DZFF", "DZMM", "DZOS"))

## Not run: 
m1 = umxGxEbiv(selDVs = "wt", selDefs = "ht", 
	dzData = dzData, mzData = mzData, sep = "", dropMissingDef = TRUE)
# Plot Moderation
umxSummary(m1)
umxSummary(m1, location = "topright")
umxSummary(m1, separateGraphs = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummaryIP'>Present the results of an independent-pathway twin model in table and graphical form</h2><span id='topic+umxSummaryIP'></span><span id='topic+umxSummary.MxModelIP'></span>

<h3>Description</h3>

<p>Summarize a Independent Pathway model, as returned by <code><a href="#topic+umxIP">umxIP()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryIP(
  model,
  digits = 2,
  file = getOption("umx_auto_plot"),
  std = TRUE,
  showRg = FALSE,
  comparison = NULL,
  CIs = FALSE,
  returnStd = FALSE,
  report = c("markdown", "html"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryIP_+3A_model">model</code></td>
<td>
<p>A fitted <code><a href="#topic+umxIP">umxIP()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_std">std</code></td>
<td>
<p>= Whether to show the standardized model (TRUE)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_showrg">showRg</code></td>
<td>
<p>= whether to show the genetic correlations (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_comparison">comparison</code></td>
<td>
<p>Whether to run mxCompare on a comparison model (NULL)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_cis">CIs</code></td>
<td>
<p>Confidence intervals (F)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_report">report</code></td>
<td>
<p>how to display the results (&quot;html&quot; will open in browser as table)</p>
</td></tr>
<tr><td><code id="umxSummaryIP_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxIP">umxIP</a>()</code>, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for IP, CP, GxE, SAT, and ACE models.
</p>
</li></ul>

<p>Other Summary functions: 
<code><a href="#topic+umxSummary.MxModel">umxSummary.MxModel</a>()</code>,
<code><a href="#topic+umxSummaryACEcov">umxSummaryACEcov</a>()</code>,
<code><a href="#topic+umxSummaryCP">umxSummaryCP</a>()</code>,
<code><a href="#topic+umxSummaryGxE">umxSummaryGxE</a>()</code>,
<code><a href="#topic+umxSummaryMRDoC">umxSummaryMRDoC</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(GFF) # family function and well-being data
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("hap", "sat", "AD") # These will be expanded into "hap_T1" "hap_T2" etc.
m1 = umxIP(selDVs = selDVs, sep = "_T", dzData = dzData, mzData = mzData)
umxSummaryIP(m1)
plot(m1)
umxSummaryIP(m1, digits = 2, file = "Figure3", showRg = FALSE, CIs = TRUE);

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummaryMRDoC'>Present the results of a Mendelian Randomization Direction of Causation Model in a table</h2><span id='topic+umxSummaryMRDoC'></span><span id='topic+umxSummary.MxModelMRDoC'></span>

<h3>Description</h3>

<p>Summarizes a MR Direction of Causation model, as returned by <code><a href="#topic+umxMRDoC">umxMRDoC()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummaryMRDoC(
  model,
  digits = 2,
  std = TRUE,
  CIs = FALSE,
  comparison = NULL,
  RMSEA_CI = FALSE,
  report = c("markdown", "html"),
  file = getOption("umx_auto_plot"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummaryMRDoC_+3A_model">model</code></td>
<td>
<p>A fitted <code><a href="#topic+umxDoC">umxDoC()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_digits">digits</code></td>
<td>
<p>Round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_std">std</code></td>
<td>
<p>Whether to show the standardized model (TRUE) (ignored: used extended = TRUE to get unstandardized)</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_cis">CIs</code></td>
<td>
<p>Confidence intervals (default FALSE)</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_comparison">comparison</code></td>
<td>
<p>Run mxCompare on a comparison model (default NULL)</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_rmsea_ci">RMSEA_CI</code></td>
<td>
<p>Optionally compute CI on RMSEA.</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_report">report</code></td>
<td>
<p>Print tables to the console (as 'markdown'), or open in browser ('html')</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: NA = none; &quot;name&quot; = use the name of the model</p>
</td></tr>
<tr><td><code id="umxSummaryMRDoC_+3A_...">...</code></td>
<td>
<p>Optional additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> nothing
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxDoC">umxDoC</a>()</code>, <code><a href="#topic+plot">plot()</a></code>, <code><a href="#topic+umxSummary">umxSummary()</a></code> work for DoC models.
</p>
</li></ul>

<p>Other Summary functions: 
<code><a href="#topic+umxSummary.MxModel">umxSummary.MxModel</a>()</code>,
<code><a href="#topic+umxSummaryACEcov">umxSummaryACEcov</a>()</code>,
<code><a href="#topic+umxSummaryCP">umxSummaryCP</a>()</code>,
<code><a href="#topic+umxSummaryGxE">umxSummaryGxE</a>()</code>,
<code><a href="#topic+umxSummaryIP">umxSummaryIP</a>()</code>
</p>

<hr>
<h2 id='umxSummarySexLim'>Shows a compact, publication-style, summary of a umx Sex Limitation model</h2><span id='topic+umxSummarySexLim'></span><span id='topic+umxSummary.MxModelSexLim'></span>

<h3>Description</h3>

<p>Summarize a fitted Cholesky model returned by <code><a href="#topic+umxSexLim">umxSexLim()</a></code>. Can control digits, report comparison model fits,
optionally show the Rg (genetic and environmental correlations), and show confidence intervals. The report parameter allows
drawing the tables to a web browser where they may readily be copied into non-markdown programs like Word.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummarySexLim(
  model,
  digits = 2,
  file = getOption("umx_auto_plot"),
  comparison = NULL,
  std = TRUE,
  showRg = FALSE,
  CIs = TRUE,
  report = c("markdown", "html"),
  extended = FALSE,
  zero.print = ".",
  show = c("std", "raw"),
  returnStd = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummarySexLim_+3A_model">model</code></td>
<td>
<p>a <code><a href="#topic+umxSexLim">umxSexLim()</a></code> model to summarize</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: &quot;name&quot; = use the name of the model.
Defaults to NA = do not create plot output</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_comparison">comparison</code></td>
<td>
<p>you can run mxCompare on a comparison model (NULL)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_std">std</code></td>
<td>
<p>Whether to standardize the output (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_showrg">showRg</code></td>
<td>
<p>= whether to show the genetic correlations (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_cis">CIs</code></td>
<td>
<p>Whether to show Confidence intervals if they exist (T)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_report">report</code></td>
<td>
<p>If &quot;html&quot;, then open an html table of the results</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_extended">extended</code></td>
<td>
<p>how much to report (FALSE)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_zero.print">zero.print</code></td>
<td>
<p>How to show zeros (&quot;.&quot;)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_show">show</code></td>
<td>
<p>Here to support being called from generic xmu_safe_run_summary. User should ignore: can be c(&quot;std&quot;, &quot;raw&quot;)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummarySexLim_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See documentation for summary functions for other types of umx model here: <code><a href="#topic+umxSummary">umxSummary()</a></code>.
</p>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSexLim">umxSexLim()</a></code>, <code><a href="#topic+umxPlotSexLim">umxPlotSexLim()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ======================================================
# = Beta: Should be good to use for Boulder/March 2020 =
# ======================================================

# =============================================
# = Run Qualitative Sex Differences ACE model =
# =============================================

# =========================
# = Load and Process Data =
# =========================
require(umx)
umx_set_optimizer("SLSQP")
data("us_skinfold_data")
# rescale vars
us_skinfold_data[, c('bic_T1', 'bic_T2')] = us_skinfold_data[, c('bic_T1', 'bic_T2')]/3.4
us_skinfold_data[, c('tri_T1', 'tri_T2')] = us_skinfold_data[, c('tri_T1', 'tri_T2')]/3
us_skinfold_data[, c('caf_T1', 'caf_T2')] = us_skinfold_data[, c('caf_T1', 'caf_T2')]/3
us_skinfold_data[, c('ssc_T1', 'ssc_T2')] = us_skinfold_data[, c('ssc_T1', 'ssc_T2')]/5
us_skinfold_data[, c('sil_T1', 'sil_T2')] = us_skinfold_data[, c('sil_T1', 'sil_T2')]/5

# Variables for Analysis
selDVs = c('ssc','sil','caf','tri','bic')
# Data for each of the 5 twin-type groups
mzmData = subset(us_skinfold_data, zyg == 1)
mzfData = subset(us_skinfold_data, zyg == 2)
dzmData = subset(us_skinfold_data, zyg == 3)
dzfData = subset(us_skinfold_data, zyg == 4)
dzoData = subset(us_skinfold_data, zyg == 5)

# ======================
# = Bivariate example =
# ======================

selDVs = c('tri','bic')
m1 = umxSexLim(selDVs = selDVs, sep = "_T", A_or_C = "A", tryHard = "yes",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)
umxSummary(m1, file = NA);

# ===============
# = Switch to C =
# ===============
m1 = umxSexLim(selDVs = selDVs, sep = "_T", A_or_C = "C", tryHard = "yes",
	mzmData = mzmData, dzmData = dzmData, 
	mzfData = mzfData, dzfData = dzfData, 
	dzoData = dzoData
)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSummarySimplex'>Shows a compact, publication-style, summary of a Simplex model.</h2><span id='topic+umxSummarySimplex'></span><span id='topic+umxSummary.MxModelSimplex'></span>

<h3>Description</h3>

<p>Summarize a fitted Simplex model returned by <code><a href="#topic+umxSimplex">umxSimplex()</a></code>. Can control digits, report comparison model fits,
optionally show the Rg (genetic and environmental correlations), and show confidence intervals. the report parameter allows
drawing the tables to a web browser where they may readily be copied into non-markdown programs like Word.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSummarySimplex(
  model,
  digits = 2,
  file = getOption("umx_auto_plot"),
  comparison = NULL,
  std = TRUE,
  showRg = FALSE,
  CIs = TRUE,
  report = c("markdown", "html"),
  returnStd = FALSE,
  extended = FALSE,
  zero.print = ".",
  show = c("std", "raw"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSummarySimplex_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to summarize</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_digits">digits</code></td>
<td>
<p>round to how many digits (default = 2)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_file">file</code></td>
<td>
<p>The name of the dot file to write: &quot;name&quot; = use the name of the model.
Defaults to NA = no plot.</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_comparison">comparison</code></td>
<td>
<p>you can run mxCompare on a comparison model (default = NULL)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_std">std</code></td>
<td>
<p>Whether to standardize the output (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_showrg">showRg</code></td>
<td>
<p>(T/F) Whether to show the genetic correlations (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_cis">CIs</code></td>
<td>
<p>Whether to show Confidence intervals if they exist (default = TRUE)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_report">report</code></td>
<td>
<p>If &quot;html&quot;, then open an html table of the results (default = 'markdown')</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_returnstd">returnStd</code></td>
<td>
<p>Whether to return the standardized form of the model (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_extended">extended</code></td>
<td>
<p>how much to report (default = FALSE)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_zero.print">zero.print</code></td>
<td>
<p>How to show zeros (default = &quot;.&quot;)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_show">show</code></td>
<td>
<p>Here to support being called from generic xmu_safe_run_summary. User should ignore: can be c(&quot;std&quot;, &quot;raw&quot;)</p>
</td></tr>
<tr><td><code id="umxSummarySimplex_+3A_...">...</code></td>
<td>
<p>Other parameters to control model summary</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See documentation for other umx models here: <code><a href="#topic+umxSummary">umxSummary()</a></code>.
</p>


<h3>Value</h3>


<ul>
<li><p> optional <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSimplex">umxSimplex()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxTwinMaker">umxTwinMaker</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# 4 time model
# Select Data
data(iqdat)
mzData &lt;- subset(iqdat, zygosity == "MZ")
dzData &lt;- subset(iqdat, zygosity == "DZ")
vars = c("IQ_age1", "IQ_age2", "IQ_age3", "IQ_age4")
m1= umxSimplex(selDVs= vars, sep= "_T", dzData= dzData, mzData= mzData, tryHard= "yes")
umxSummary(m1, file = NA);

## End(Not run)
</code></pre>

<hr>
<h2 id='umxSuperModel'>Make a multi-group model</h2><span id='topic+umxSuperModel'></span>

<h3>Description</h3>

<p><code>umxSuperModel</code> takes 1 or more models and wraps them in a supermodel with a
<code><a href="OpenMx.html#topic+mxFitFunctionMultigroup">mxFitFunctionMultigroup()</a></code> fit function that minimizes the sum of the
fits of the sub-models.
</p>
<p><em>note</em>: Any duplicate model-names are renamed to be unique by suffixing <code style="white-space: pre;">&#8288;_1&#8288;</code> etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxSuperModel(
  name = "super",
  ...,
  autoRun = getOption("umx_auto_run"),
  tryHard = c("no", "yes", "ordinal", "search"),
  std = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxSuperModel_+3A_name">name</code></td>
<td>
<p>The name for the container model (default = 'super')</p>
</td></tr>
<tr><td><code id="umxSuperModel_+3A_...">...</code></td>
<td>
<p>Models forming the multiple groups contained in the supermodel.</p>
</td></tr>
<tr><td><code id="umxSuperModel_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the model (default), or just to create it and return without running.</p>
</td></tr>
<tr><td><code id="umxSuperModel_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxSuperModel_+3A_std">std</code></td>
<td>
<p>Show standardized parameters, raw (default), or just the fit indices (null)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxFitFunctionMultigroup">mxFitFunctionMultigroup()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other Core Model Building Functions: 
<code><a href="#topic+umxMatrix">umxMatrix</a>()</code>,
<code><a href="#topic+umxModify">umxModify</a>()</code>,
<code><a href="#topic+umxPath">umxPath</a>()</code>,
<code><a href="#topic+umxRAM">umxRAM</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(umx)
# Create two sets of data in which X &amp; Y correlate ~ .4 in both datasets.
manifests = c("x", "y")
tmp = umx_make_TwinData(nMZpairs = 100, nDZpairs = 150, 
		AA = 0, CC = .4, EE = .6, varNames = manifests)

# Group 1
grp1   = tmp[tmp$zygosity == "MZ", manifests]
g1Data = mxData(cov(grp1), type = "cov", numObs = nrow(grp1), means=umx_means(grp1))

# Group 2
grp2   = tmp[tmp$zygosity == "DZ", manifests]
g2Data = mxData(cov(grp2), type = "cov", numObs = nrow(grp2), means=umx_means(grp2))


# Model 1 (could add autoRun = FALSE if you don't want to run this as it is being built)
m1 = umxRAM("m1", data = g1Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels = c("Var_x", "Resid_y_grp1")),
	umxPath(means = manifests, labels = c("Mean_x", "Mean_y"))
)

# Model 2
m2 = umxRAM("m2", data = g2Data,
	umxPath("x", to = "y", labels = "beta"),
	umxPath(var = manifests, labels=c("Var_x", "Resid_y_grp2")),
	umxPath(means = manifests, labels=c("Mean_x", "Mean_y"))
)

# Place m1 and m2 into a supermodel, and autoRun it
# NOTE: umxSummary is only semi-smart/certain enough to compute saturated models etc
# and report multiple groups correctly.

m3 = umxSuperModel('top', m1, m2)

umxSummary(m3, std= TRUE)

# |name         | Std.Estimate| Std.SE|CI                |
# |:------------|------------:|------:|:-----------------|
# |beta         |         0.51|   0.05|0.51 [0.41, 0.61] |
# |Var_x        |         1.00|   0.00|1 [1, 1]          |
# |Resid_y_grp1 |         0.74|   0.05|0.74 [0.64, 0.84] |
# |beta         |         0.50|   0.05|0.5 [0.41, 0.6]   |
# |Var_x        |         1.00|   0.00|1 [1, 1]          |
# |Resid_y_grp2 |         0.75|   0.05|0.75 [0.65, 0.84] |

summary(m3)

# ====================================
# = Test models with duplicate names =
# ====================================
data(GFF)
mzData = subset(GFF, zyg_2grp == "MZ")
dzData = subset(GFF, zyg_2grp == "DZ")
selDVs = c("gff", "fc", "qol")
m1 = umxCP(selDVs= selDVs, nFac= 1, dzData= dzData, mzData= mzData, sep= "_T", autoRun= TRUE)
m2 = mxRename(m1, "CP2")
umxModelNames(m1) # "top" "MZ" "DZ"
umxModelNames(m2) # "top" "MZ" "DZ"
super = umxSuperModel("myModel", m1, m2, autoRun = TRUE)
umxModelNames(super)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxThresholdMatrix'>Create the threshold matrix needed for modeling ordinal data.</h2><span id='topic+umxThresholdMatrix'></span>

<h3>Description</h3>

<p>High-level helper for ordinal modeling. Creates, labels, and sets smart-starts for this
complex set set of an algebra and matrices. Big time saver!
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxThresholdMatrix(
  df,
  fullVarNames = NULL,
  sep = NULL,
  method = c("Mehta", "allFree"),
  threshMatName = "threshMat",
  l_u_bound = c(NA, NA),
  droplevels = FALSE,
  verbose = FALSE,
  selDVs = "deprecated"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxThresholdMatrix_+3A_df">df</code></td>
<td>
<p>The data being modeled (to allow access to the factor levels and quantiles within these for each variable)</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_fullvarnames">fullVarNames</code></td>
<td>
<p>The variable names. Note for twin data, just the base names, which sep will be used to fill out.</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_sep">sep</code></td>
<td>
<p>(e.g. &quot;_T&quot;) Required for wide (twin) data. It is used to break the base names our from their numeric suffixes.</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_method">method</code></td>
<td>
<p>How to implement the thresholds: Mehta, (1 free thresh for binary, first two fixed for ordinal) or &quot;allFree&quot;</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_threshmatname">threshMatName</code></td>
<td>
<p>name of the matrix which is returned. Defaults to &quot;threshMat&quot; - best not to change it.</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_l_u_bound">l_u_bound</code></td>
<td>
<p>c(NA, NA) by default, you can use this to bound the first (base) threshold.</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_droplevels">droplevels</code></td>
<td>
<p>Whether to drop levels with no observed data (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_verbose">verbose</code></td>
<td>
<p>How much to say about what was done. (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="umxThresholdMatrix_+3A_seldvs">selDVs</code></td>
<td>
<p>deprecated. Use &quot;fullVarNames&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We often need to model ordinal data: sex, low-med-hi, depressed/normal, etc.,
A useful conceptual strategy to handle these data is to build a standard model for normally-varying data
and then to threshold this normal distribution to generate the observed data. Thus an observation of &quot;depressed&quot;
is modeled as a high score on the latent normally distributed trait, with thresholds set so that only scores above
this threshold (1-minus the number of categories) reach the criteria for the diagnosis.
</p>
<p>Making this work can require fixing the first 2 thresholds of ordinal data, or fixing both the mean and variance of
a latent variable driving binary data, in order to estimate its one-free parameter: where to place the single threshold
separating low from high cases.
</p>
<p>The function returns a 3-item list consisting of:
</p>

<ol>
<li><p> A thresholdsAlgebra (named <code>threshMatName</code>)
</p>
</li>
<li><p> A matrix of deviations for the thresholds (<code>deviations_for_thresh</code>)
</p>
</li>
<li><p> A lower matrix of ones (<code>lowerOnes_for_thresh</code>)
</p>
</li></ol>

<p><em>Twin Data</em>
</p>
<p>With twin data, make sure to provide the <strong>full names</strong> for twin data... this is not standard I know...
</p>
<p>For twins (the function currently handles only pairs), the thresholds are equated for both twins using labels:
</p>
<p>$labels
</p>
<div class="sourceCode"><pre>  obese_T1         obese_T2
</pre></div>
<p>dev_1 &quot;obese_dev1&quot;   &quot;obese_dev1&quot;
</p>


<h3>Value</h3>


<ul>
<li><p> list of thresholds matrix, deviations, lowerOnes
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="OpenMx.html#topic+mxThreshold">OpenMx::mxThreshold()</a></code>
</p>
<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# ============================
# = Simple non-twin examples =
# ============================

# data: 1 2-level ordered factor
x = data.frame(ordered(rbinom(100,1,.5))); names(x) = c("x")

tmp = umxThresholdMatrix(x, fullVarNames = "x")
# The lower ones matrix (all fixed)
tmp[[1]]$values
tmp[[1]]$free

# The deviations matrix
tmp[[2]]$values
tmp[[2]]$labels # note: for twins, labels will be equated across twins

# The algebra that adds the deviations to create thresholds:
tmp[[3]]$formula

# Example of a warning to not omit the variable names
# tmp = umxThresholdMatrix(x)
# Polite message: For coding safety, when calling umxThresholdMatrix, set fullVarNames...

# One ordered factor with 5-levels
x = cut(rnorm(100), breaks = c(-Inf,.2,.5, .7, Inf)); levels(x) = 1:5
x = data.frame(ordered(x)); names(x) &lt;- c("x")
tmp = umxThresholdMatrix(x, fullVarNames = "x")
tmp[[2]]$name
tmp[[2]]$free # last one is free.. (method = Mehta)

tmp = umxThresholdMatrix(x, fullVarNames = "x", l_u_bound= c(-1,1))
tmp[[2]]$lbound # bounds applied to base threshold

# =================================
# = Binary example with twin data =
# =================================
# ===============================================================
# = Create a series of binary and ordinal columns to work with =
# ===============================================================
data(twinData)

# Make "obese" variable with ~20% subjects categorised as obese
obesityLevels   = c('normal', 'obese')
cutPoints       = quantile(twinData[, "bmi1"], probs = .2, na.rm = TRUE)
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
# Step 2: Make the ordinal variables into umxFactors (ordered, with the levels found in the data)
selVars = c("obese1", "obese2")
twinData[, selVars] = umxFactor(twinData[, selVars])

# Example 1
# use verbose = TRUE to see informative messages
tmp = umxThresholdMatrix(twinData, fullVarNames = selVars, sep = "", verbose = TRUE) 


# ======================================
# = Ordinal (n categories &gt; 2) example =
# ======================================
# Repeat for three-level weight variable
obesityLevels = c('normal', 'overweight', 'obese')
cutPoints = quantile(twinData[, "bmi1"], probs = c(.4, .7), na.rm = TRUE)
twinData$obeseTri1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obeseTri2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
selDVs = "obeseTri"; selVars = tvars(selDVs, sep = "", suffixes = 1:2)
twinData[, selVars] = umxFactor(twinData[, selVars])
tmp = umxThresholdMatrix(twinData, fullVarNames = selVars, sep = "", verbose = TRUE)


# ========================================================
# = Mix of all three kinds example (and a 4-level trait) =
# ========================================================
obesityLevels = c('underWeight', 'normal', 'overweight', 'obese')
cutPoints = quantile(twinData[, "bmi1"], probs = c(.25, .4, .7), na.rm = TRUE)
twinData$obeseQuad1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obeseQuad2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
selVars = c("obeseQuad1", "obeseQuad2")
twinData[, selVars] = umxFactor(twinData[, selVars])

selDVs =c("bmi", "obese", "obeseTri", "obeseQuad")
tmp = umxThresholdMatrix(twinData, fullVarNames = tvars(selDVs, sep= ""), sep = "", verbose = TRUE)
# The lower ones matrix (all fixed)
tmp[[1]]$values
# The deviations matrix
tmp[[2]]$values
tmp[[2]]$labels # note labels are equated across twins
# Check to be sure twin-1 column labels same as twin-2
tmp[[2]]$labels[,2]==tmp[[2]]$labels[,4]

# The algebra that assembles these into thresholds:
tmp[[3]]$formula
# =================================
# = Example with method = allFree =
# =================================

tmp = umxThresholdMatrix(twinData, fullVarNames = tvars(selDVs, sep= ""), sep = "", 
	method = "allFree")
all(tmp[[2]]$free)

</code></pre>

<hr>
<h2 id='umxTwinMaker'>Make a twin model from the model describing just one person</h2><span id='topic+umxTwinMaker'></span>

<h3>Description</h3>

<p><code>xmu_path2twin</code> takes a collection of paths describing the model for 1 person
and returns a completed twin model. This consists of a <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code> containing
<code>MZ</code> and <code>DZ</code> <code><a href="#topic+umxRAM">umxRAM()</a></code> models.
</p>
<p>Pass into <code>umxTwinMaker</code>:
</p>

<ol>
<li><p> A list of <code>paths</code> making up the twin 1 model
</p>
</li>
<li><p> In <code>t1_t2links</code>, a vector describing the component relationships connecting twin 1 to twin 2
(The default here is 1 and .5 for the a, and, for c and e are 1 and 0 in both groups, respectively.
</p>
</li></ol>

<p><em>Details</em>
</p>
<p>Some rules. All labels are expanded with a twin suffix: so &quot;var1&quot; -&gt; &quot;var1_T1&quot; etc. so you
provide the person-model using just the base name (and tell <code><a href="#topic+umxTwinMaker">umxTwinMaker()</a></code> how to expand it by providing a separator string).
</p>
<p>Rule 2: The latent a, c, and e latent variables must be labelled to match the base name given in t1_t2links.
To avoid clashes, variables must not match the numbered variables in <code>t1_t2links</code>  - by default names like &quot;a1&quot; are reserved for ace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxTwinMaker(
  name = "m1",
  paths,
  t1_t2links = list(a = c(1, 0.5), c = c(1, 1), e = c(0, 0)),
  mzData = NULL,
  dzData = NULL,
  sep = "_T",
  autoRun = getOption("umx_auto_run")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxTwinMaker_+3A_name">name</code></td>
<td>
<p>The name for the resulting <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code> (Default &quot;m1&quot;).</p>
</td></tr>
<tr><td><code id="umxTwinMaker_+3A_paths">paths</code></td>
<td>
<p>A vector of <code><a href="#topic+umxPath">umxPath()</a></code>s describing one person.</p>
</td></tr>
<tr><td><code id="umxTwinMaker_+3A_t1_t2links">t1_t2links</code></td>
<td>
<p>base name (and values) of paths that covary between T1 and T2. Default: c('a'=c(1,.5), 'c'=c(1,1), 'e'=c(0,0))</p>
</td></tr>
<tr><td><code id="umxTwinMaker_+3A_mzdata">mzData</code></td>
<td>
<p>Data for MZ twins.</p>
</td></tr>
<tr><td><code id="umxTwinMaker_+3A_dzdata">dzData</code></td>
<td>
<p>Data for DZ twins.</p>
</td></tr>
<tr><td><code id="umxTwinMaker_+3A_sep">sep</code></td>
<td>
<p>The separator used to create twin 1 and 2 names (Default &quot;_T&quot;)</p>
</td></tr>
<tr><td><code id="umxTwinMaker_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run the supermodel before returning it.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">tutorials</a>, <a href="https://github.com/tbates/umx">github</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxRAM">umxRAM()</a></code>, <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code>, <code><a href="#topic+umxPath">umxPath()</a></code>
</p>
</li></ul>

<p>Other Twin Modeling Functions: 
<code><a href="#topic+power.ACE.test">power.ACE.test</a>()</code>,
<code><a href="#topic+umxACEcov">umxACEcov</a>()</code>,
<code><a href="#topic+umxACEv">umxACEv</a>()</code>,
<code><a href="#topic+umxACE">umxACE</a>()</code>,
<code><a href="#topic+umxCP">umxCP</a>()</code>,
<code><a href="#topic+umxDiffMZ">umxDiffMZ</a>()</code>,
<code><a href="#topic+umxDiscTwin">umxDiscTwin</a>()</code>,
<code><a href="#topic+umxDoCp">umxDoCp</a>()</code>,
<code><a href="#topic+umxDoC">umxDoC</a>()</code>,
<code><a href="#topic+umxGxE_window">umxGxE_window</a>()</code>,
<code><a href="#topic+umxGxEbiv">umxGxEbiv</a>()</code>,
<code><a href="#topic+umxGxE">umxGxE</a>()</code>,
<code><a href="#topic+umxIP">umxIP</a>()</code>,
<code><a href="#topic+umxMRDoC">umxMRDoC</a>()</code>,
<code><a href="#topic+umxReduceACE">umxReduceACE</a>()</code>,
<code><a href="#topic+umxReduceGxE">umxReduceGxE</a>()</code>,
<code><a href="#topic+umxReduce">umxReduce</a>()</code>,
<code><a href="#topic+umxRotate.MxModelCP">umxRotate.MxModelCP</a>()</code>,
<code><a href="#topic+umxSexLim">umxSexLim</a>()</code>,
<code><a href="#topic+umxSimplex">umxSimplex</a>()</code>,
<code><a href="#topic+umxSummarizeTwinData">umxSummarizeTwinData</a>()</code>,
<code><a href="#topic+umxSummaryACEv">umxSummaryACEv</a>()</code>,
<code><a href="#topic+umxSummaryACE">umxSummaryACE</a>()</code>,
<code><a href="#topic+umxSummaryDoC">umxSummaryDoC</a>()</code>,
<code><a href="#topic+umxSummaryGxEbiv">umxSummaryGxEbiv</a>()</code>,
<code><a href="#topic+umxSummarySexLim">umxSummarySexLim</a>()</code>,
<code><a href="#topic+umxSummarySimplex">umxSummarySimplex</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# We'll make some ACE models, but first, let's clean up the twinData 
# set for analysis
# 1. Add a separator to the twin variable names (with sep = "_T")
# 2. Scale the data so it's easier for the optimizer.
data(twinData)
tmp = umx_make_twin_data_nice(data=twinData, sep="", zygosity="zygosity", numbering=1:2)
tmp = umx_scale_wide_twin_data(varsToScale= c("wt", "ht"), sep= "_T", data= tmp)
mzData = subset(tmp, zygosity %in%  c("MZFF", "MZMM"))
dzData = subset(tmp, zygosity %in%  c("DZFF", "DZMM"))

# ==========================
# = Make an ACE twin model =
# ==========================
# 1. Define paths for *one* person:
paths = c(
   umxPath(v1m0 = c("a1", 'c1', "e1")),
   umxPath(means = c("wt")),
   umxPath(c("a1", 'c1', "e1"), to = "wt", values=.2)
)
# 2. Make a twin model from the paths for one person
m1 = umxTwinMaker("test", paths, mzData = mzData, dzData= dzData)
plot(m1, std= TRUE, means= FALSE)

# 3. comparison with umxACE...
m2 = umxACE(selDVs="wt", mzData = mzData, dzData=dzData, sep="_T")

# =====================
# = Bivariate example =
# =====================
latents = paste0(rep(c("a", "c", "e"), each = 2), 1:2)
biv = c(
umxPath(v1m0 = latents),
umxPath(mean = c("wt", "ht")),
umxPath(fromEach = c("a1", 'c1', "e1"), to = c("ht", "wt")),
umxPath(c("a2", 'c2', "e2"), to = "wt")
)
tmp= umxTwinMaker(paths= biv, mzData = mzData, dzData= dzData)
plot(tmp, means=FALSE)

# How to use latents other than a, c, and e: define in t1_t2links
paths = c(
umxPath(v1m0 = c("as1", 'c1', "e1")),
umxPath(means = c("wt")),
umxPath(c("as1", 'c1', "e1"), to = "wt", values=.2)
)
m1 = umxTwinMaker("test", paths, mzData = mzData, dzData= dzData, 
	t1_t2links = list('as'=c(1, .5), 'c'=c(1, 1), 'e'=c(0, 0))
)


## End(Not run)

</code></pre>

<hr>
<h2 id='umxTwoStage'>Build a SEM implementing the instrumental variable design</h2><span id='topic+umxTwoStage'></span><span id='topic+umxMR'></span>

<h3>Description</h3>

<p><code>umxMR</code> (<code>umxTwoStage</code>) implements a Mendelian randomization or instrumental variable Structural Equation Model.
For ease of learning, the parameters follow the <code>tsls()</code> function in the sem package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxTwoStage(
  formula = Y ~ X,
  instruments = ~qtl,
  data,
  std = FALSE,
  subset,
  contrasts = NULL,
  name = "IV_model",
  tryHard = c("no", "yes", "ordinal", "search"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxTwoStage_+3A_formula">formula</code></td>
<td>
<p>The structural equation to be estimated (default = Y ~ X). A constant is implied if not explicitly deleted.</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_instruments">instruments</code></td>
<td>
<p>A one-sided formula specifying instrumental variables (default = qtl).</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_data">data</code></td>
<td>
<p>Frame containing the variables in the model.</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_std">std</code></td>
<td>
<p>Standardize the manifests before running model (default is FALSE)</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_subset">subset</code></td>
<td>
<p>(optional) vector specifying a subset of observations to be used in fitting the model.</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_contrasts">contrasts</code></td>
<td>
<p>An optional list (not supported)</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_name">name</code></td>
<td>
<p>The model name (default is &quot;IVmodel&quot;)</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="umxTwoStage_+3A_...">...</code></td>
<td>
<p>arguments to be passed along. (not supported)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The example is a <a href="https://en.wikipedia.org/wiki/Mendelian_randomization">Mendelian Randomization</a>
analysis showing the utility of SEM over two-stage regression.
</p>
<p>The following figure shows how the MR model appears as a path diagram:
</p>
<p><img src="../help/figures/TSLS.png" width=50% alt="Figure: Mendelian Randomization analysis.png" />

</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Fox, J. (1979) Simultaneous equation models and two-stage least-squares. In Schuessler, K. F. (ed.) <em>Sociological Methodology</em>, Jossey-Bass.
</p>
</li></ul>


<ul>
<li><p> Greene, W. H. (1993) <em>Econometric Analysis</em>, Second Edition, Macmillan.
</p>
</li>
<li><p> Sekula, P., Del Greco, M. F., Pattaro, C., &amp; Kottgen, A. (2016). Mendelian Randomization as an Approach to
Assess Causality Using Observational Data. <em>Journal of the American Society of Nephrology</em>, <strong>27</strong>), 3253-3265. <a href="https://doi.org/10.1681/ASN.2016010098">doi:10.1681/ASN.2016010098</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umx_make_MR_data">umx_make_MR_data()</a></code>, <code><a href="#topic+umxDiffMZ">umxDiffMZ()</a></code>, <code><a href="#topic+umxDoC">umxDoC()</a></code>, <code><a href="#topic+umxDiscTwin">umxDiscTwin()</a></code>
</p>
</li></ul>

<p>Other Super-easy helpers: 
<code><a href="#topic+umxEFA">umxEFA</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ====================================
# = Mendelian Randomization analysis =
# ====================================

library(umx)
df = umx_make_MR_data(10e4)
m1 = umxMR(Y ~ X, instruments = ~ qtl, data = df)
parameters(m1)
plot(m1, means = FALSE, min="") # help DiagrammR layout the plot.
m2 = umxModify(m1, "qtl_to_X", comparison=TRUE, tryHard="yes", name="QTL_affects_X") # yip
m3 = umxModify(m1, "X_to_Y"  , comparison=TRUE, tryHard="yes", name="X_affects_Y") # nope
plot(m3, means = FALSE)

# Errant analysis using ordinary least squares regression (WARNING this result is CONFOUNDED!!)
m1 = lm(Y ~ X    , data = df); coef(m1) # incorrect .35 effect of X on Y
m1 = lm(Y ~ X + U, data = df); coef(m1) # Controlling U reveals the true 0.1 beta weight

df = umx_make_MR_data(10e4) 
m1 = umxMR(Y ~ X, instruments = ~ qtl, data = df)
coef(m1)

# ======================
# = Now with sem::tsls =
# ======================
# library(sem) # may require you to install X11
m2 = sem::tsls(formula = Y ~ X, instruments = ~ qtl, data = df)
coef(m2)

# Try with missing value for one subject: A benefit of the FIML approach in OpenMx.
m3 = tsls(formula = Y ~ X, instruments = ~ qtl, data = (df[1, "qtl"] = NA))

## End(Not run)
</code></pre>

<hr>
<h2 id='umxUnexplainedCausalNexus'>umxUnexplainedCausalNexus</h2><span id='topic+umxUnexplainedCausalNexus'></span>

<h3>Description</h3>

<p>umxUnexplainedCausalNexus report the effect of a change (delta) in a variable (from) on an output (to)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxUnexplainedCausalNexus(from, delta, to, model = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxUnexplainedCausalNexus_+3A_from">from</code></td>
<td>
<p>A variable in the model for which you want to compute the effect of a change.</p>
</td></tr>
<tr><td><code id="umxUnexplainedCausalNexus_+3A_delta">delta</code></td>
<td>
<p>A the amount to simulate changing &lsquo;from&rsquo; by.</p>
</td></tr>
<tr><td><code id="umxUnexplainedCausalNexus_+3A_to">to</code></td>
<td>
<p>The dependent variable that you want to watch changing.</p>
</td></tr>
<tr><td><code id="umxUnexplainedCausalNexus_+3A_model">model</code></td>
<td>
<p>The model containing variables from and to.</p>
</td></tr>
</table>


<h3>References</h3>


<ul>
<li><p> https://github.com/tbates/umx/
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxCheckIdentification">mxCheckIdentification()</a></code>, <code><a href="OpenMx.html#topic+mxCompare">mxCompare()</a></code>
</p>
</li></ul>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
umxUnexplainedCausalNexus(from="yrsEd", delta = .5, to = "income35", model)

## End(Not run)
</code></pre>

<hr>
<h2 id='umxVersion'>Get or print the version of umx, along with detail from OpenMx and general system info.</h2><span id='topic+umxVersion'></span>

<h3>Description</h3>

<p>umxVersion returns the version information for umx, and for OpenMx and R.
Essential for bug-reports! This function can also test for a minimum version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxVersion(
  model = NULL,
  min = NULL,
  verbose = TRUE,
  return = c("umx_vers", "OpenMx_vers")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxVersion_+3A_model">model</code></td>
<td>
<p>Optional to show optimizer in this model</p>
</td></tr>
<tr><td><code id="umxVersion_+3A_min">min</code></td>
<td>
<p>Optional minimum version string to test for, e.g. '2.7.0' (Default = NULL).</p>
</td></tr>
<tr><td><code id="umxVersion_+3A_verbose">verbose</code></td>
<td>
<p>= TRUE</p>
</td></tr>
<tr><td><code id="umxVersion_+3A_return">return</code></td>
<td>
<p>Which package (umx or OpenMx) to 'return' version info for (Default = umx).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="utils.html#topic+packageVersion">packageVersion()</a></code>, <code><a href="#topic+install.OpenMx">install.OpenMx()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Utility Functions: 
<code><a href="#topic+install.OpenMx">install.OpenMx</a>()</code>,
<code><a href="#topic+libs">libs</a>()</code>,
<code><a href="#topic+qm">qm</a>()</code>,
<code><a href="#topic+umxLav2RAM">umxLav2RAM</a>()</code>,
<code><a href="#topic+umxModelNames">umxModelNames</a>()</code>,
<code><a href="#topic+umxRAM2Lav">umxRAM2Lav</a>()</code>,
<code><a href="#topic+umx_array_shift">umx_array_shift</a>()</code>,
<code><a href="#topic+umx_find_object">umx_find_object</a>()</code>,
<code><a href="#topic+umx_lower.tri">umx_lower.tri</a>()</code>,
<code><a href="#topic+umx_msg">umx_msg</a>()</code>,
<code><a href="#topic+umx_open_CRAN_page">umx_open_CRAN_page</a>()</code>,
<code><a href="#topic+umx_pad">umx_pad</a>()</code>,
<code><a href="#topic+umx_print">umx_print</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = umxVersion(); x
</code></pre>

<hr>
<h2 id='umxWeightedAIC'>AIC weight-based conditional probabilities.</h2><span id='topic+umxWeightedAIC'></span>

<h3>Description</h3>

<p>Returns the best model by AIC, and computes the probabilities
according to AIC weight-based conditional probabilities (Wagenmakers &amp; Farrell, 2004).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>umxWeightedAIC(models, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="umxWeightedAIC_+3A_models">models</code></td>
<td>
<p>a list of models to compare.</p>
</td></tr>
<tr><td><code id="umxWeightedAIC_+3A_digits">digits</code></td>
<td>
<p>(default 2)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Best model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li><p> Wagenmakers E.J., Farrell S. (2004), 192-196. AIC model selection using Akaike weights. <em>Psychonomic Bulletin and Review</em>. <strong>11</strong>, 192-196. <a href="https://pubmed.ncbi.nlm.nih.gov/15117008/">https://pubmed.ncbi.nlm.nih.gov/15117008/</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+AIC">AIC()</a></code>
</p>
</li></ul>

<p>Other Miscellaneous Stats Functions: 
<code><a href="#topic+FishersMethod">FishersMethod</a>()</code>,
<code><a href="#topic+SE_from_p">SE_from_p</a>()</code>,
<code><a href="#topic+geometric_mean">geometric_mean</a>()</code>,
<code><a href="#topic+harmonic_mean">harmonic_mean</a>()</code>,
<code><a href="#topic+oddsratio">oddsratio</a>()</code>,
<code><a href="#topic+reliability">reliability</a>()</code>,
<code><a href="#topic+umxCov2cor">umxCov2cor</a>()</code>,
<code><a href="#topic+umxHetCor">umxHetCor</a>()</code>,
<code><a href="#topic+umxParan">umxParan</a>()</code>,
<code><a href="#topic+umx_apply">umx_apply</a>()</code>,
<code><a href="#topic+umx_cor">umx_cor</a>()</code>,
<code><a href="#topic+umx_means">umx_means</a>()</code>,
<code><a href="#topic+umx_r_test">umx_r_test</a>()</code>,
<code><a href="#topic+umx_round">umx_round</a>()</code>,
<code><a href="#topic+umx_scale">umx_scale</a>()</code>,
<code><a href="#topic+umx_var">umx_var</a>()</code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>l1 = lm(mpg~ wt + disp, data=mtcars)
l2 = lm(mpg~ wt, data=mtcars)
umxWeightedAIC(models = list(l1, l2))
</code></pre>

<hr>
<h2 id='us_skinfold_data'>Anthropometric data on twins</h2><span id='topic+us_skinfold_data'></span>

<h3>Description</h3>

<p>A dataset containing height, weight, BMI, and skin-fold fat measures in several
hundred US twin families participating in the MCV Cardiovascular Twin Study (PI Schieken).
Biceps and Triceps are folds above and below the upper arm (holding arm palm upward),
Calf (fold on the calf muscle), Subscapular (fold over the shoulder blade),
Suprailiacal (fold between the hip and ribs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(us_skinfold_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 53940 twin families (1 per row) each twin measured on 10 variables.
</p>


<h3>Details</h3>


<ul>
<li> <p><em>fan</em> FamilyID (t1=male,t2=female)
</p>
</li>
<li> <p><em>zyg</em> Zygosity  1:mzm, 2:mzf, 3:dzm, 4:dzf, 5:dzo
</p>
</li>
<li> <p><em>ht_T1</em>  Height of twin 1 (cm)
</p>
</li>
<li> <p><em>wt_T1</em>  Weight of twin 1 (kg)
</p>
</li>
<li> <p><em>bmi_T1</em> BMI of twin 1
</p>
</li>
<li> <p><em>bml_T1</em> log BMI of twin 1
</p>
</li>
<li> <p><em>bic_T1</em> Biceps Skinfold of twin 1
</p>
</li>
<li> <p><em>caf_T1</em> Calf Skinfold of twin 1
</p>
</li>
<li> <p><em>ssc_T1</em> Subscapular Skinfold of twin 1
</p>
</li>
<li> <p><em>sil_T1</em> Suprailiacal Skinfold of twin 1
</p>
</li>
<li> <p><em>tri_T1</em> Triceps Skinfold of twin 1
</p>
</li>
<li> <p><em>ht_T2</em>  Height of twin 2
</p>
</li>
<li> <p><em>wt_T2</em>  Weight of twin 2
</p>
</li>
<li> <p><em>bmi_T2</em> BMI of twin 2
</p>
</li>
<li> <p><em>bml_T2</em> log BMI of twin 2
</p>
</li>
<li> <p><em>bic_T2</em> Biceps Skinfold of twin 2
</p>
</li>
<li> <p><em>caf_T2</em> Calf Skinfold of twin 2
</p>
</li>
<li> <p><em>ssc_T2</em> Subscapular Skinfold of twin 2
</p>
</li>
<li> <p><em>sil_T2</em> Suprailiacal Skinfold of twin 2
</p>
</li>
<li> <p><em>tri_T2</em> Triceps Skinfold of twin 2
</p>
</li></ul>



<h3>References</h3>

<p>Moskowitz, W. B., Schwartz, P. F., &amp; Schieken, R. M. (1999).
Childhood passive smoking, race, and coronary artery disease risk:
the MCV Twin Study. Medical College of Virginia.
<em>Archives of Pediatrics and Adolescent Medicine</em>, <strong>153</strong>, 446-453.
<a href="https://pubmed.ncbi.nlm.nih.gov/10323623/">https://pubmed.ncbi.nlm.nih.gov/10323623/</a>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+Fischbein_wt">Fischbein_wt</a></code>,
<code><a href="#topic+GFF">GFF</a></code>,
<code><a href="#topic+docData">docData</a></code>,
<code><a href="#topic+iqdat">iqdat</a></code>,
<code><a href="#topic+umx">umx</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(us_skinfold_data)
str(us_skinfold_data)
par(mfrow = c(1, 2))  # 1 rows and 3 columns
plot(ht_T1 ~ht_T2, ylim = c(130, 165), data = subset(us_skinfold_data, zyg == 1))
plot(ht_T1 ~ht_T2, ylim = c(130, 165), data = subset(us_skinfold_data, zyg == 3))
par(mfrow = c(1, 1))  # back to as it was

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_bracket_address2rclabel'>Convert a bracket address into an A_rXcX-style label.</h2><span id='topic+xmu_bracket_address2rclabel'></span>

<h3>Description</h3>

<p>Takes a label like A[1,1] and returns &quot;A_r1c1&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_bracket_address2rclabel(label, keepPrefix = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_bracket_address2rclabel_+3A_label">label</code></td>
<td>
<p>A bracket label</p>
</td></tr>
<tr><td><code id="xmu_bracket_address2rclabel_+3A_keepprefix">keepPrefix</code></td>
<td>
<p>Keep any prefix found e.g. &quot;model.top&quot;</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> label e.g. &quot;ai_r1c1&quot;
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_bracket_address2rclabel(label = "A[1,1]")
xmu_bracket_address2rclabel(label = "top.A[1,1]")
xmu_bracket_address2rclabel(label = "A_std[1,1]")

</code></pre>

<hr>
<h2 id='xmu_cell_is_on'>Return whether a cell is in a set location of a matrix</h2><span id='topic+xmu_cell_is_on'></span>

<h3>Description</h3>

<p>Helper to determine is a cell is in a set location of a matrix or not.
Left is useful for, e.g. twin means matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_cell_is_on(
  r,
  c,
  where = c("diag", "lower", "lower_inc", "upper", "upper_inc", "any", "left"),
  mat = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_cell_is_on_+3A_r">r</code></td>
<td>
<p>which row the cell is on.</p>
</td></tr>
<tr><td><code id="xmu_cell_is_on_+3A_c">c</code></td>
<td>
<p>which column the cell is in.</p>
</td></tr>
<tr><td><code id="xmu_cell_is_on_+3A_where">where</code></td>
<td>
<p>the location (any, diag, lower or upper (or _inc) or left).</p>
</td></tr>
<tr><td><code id="xmu_cell_is_on_+3A_mat">mat</code></td>
<td>
<p>(optionally) provide matrix to check dimensions against r and c.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmuLabel">xmuLabel()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_cell_is_on(r = 3, c = 3, "lower")
xmu_cell_is_on(r = 3, c = 3, "lower_inc")
xmu_cell_is_on(r = 3, c = 3, "upper")
xmu_cell_is_on(r = 3, c = 3, "upper_inc")
xmu_cell_is_on(r = 3, c = 3, "diag")
xmu_cell_is_on(r = 2, c = 3, "diag")
xmu_cell_is_on(r = 3, c = 3, "any")
a_cp = umxMatrix("a_cp", "Lower", 3, 3, free = TRUE, values = 1:6)
xmu_cell_is_on(r = 3, c = 3, "left", mat = a_cp)
</code></pre>

<hr>
<h2 id='xmu_check_levels_identical'>xmu_check_levels_identical</h2><span id='topic+xmu_check_levels_identical'></span>

<h3>Description</h3>

<p>Just checks that the factor levels for twins 1 and 2 are the same
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_check_levels_identical(df, selDVs, sep, action = c("stop", "ignore"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_check_levels_identical_+3A_df">df</code></td>
<td>
<p>data.frame containing the data</p>
</td></tr>
<tr><td><code id="xmu_check_levels_identical_+3A_seldvs">selDVs</code></td>
<td>
<p>base names of variables (without suffixes)</p>
</td></tr>
<tr><td><code id="xmu_check_levels_identical_+3A_sep">sep</code></td>
<td>
<p>text-constant separating base variable names the twin index (1:2)</p>
</td></tr>
<tr><td><code id="xmu_check_levels_identical_+3A_action">action</code></td>
<td>
<p>if unequal levels found:  c(&quot;stop&quot;, &quot;ignore&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(twinData)
baseNames = c("bmi")
selDVs = umx_paste_names(baseNames, "", 1:2)
tmp = twinData[, selDVs]
tmp$bmi1[tmp$bmi1 &lt;= 22] = 22
tmp$bmi2[tmp$bmi2 &lt;= 22] = 22
xmu_check_levels_identical(umxFactor(tmp, sep = ""), selDVs = baseNames, sep = "")
## Not run: 
xmu_check_levels_identical(umxFactor(tmp), selDVs = baseNames, sep = "")

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_check_needs_means'>Check data to see if model needs means.</h2><span id='topic+xmu_check_needs_means'></span>

<h3>Description</h3>

<p>Check data to see if model needs means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_check_needs_means(
  data,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  allContinuousMethod = c("cumulants", "marginals")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_check_needs_means_+3A_data">data</code></td>
<td>
<p><code><a href="OpenMx.html#topic+mxData">mxData()</a></code> to check.</p>
</td></tr>
<tr><td><code id="xmu_check_needs_means_+3A_type">type</code></td>
<td>
<p>of the data requested by the model.</p>
</td></tr>
<tr><td><code id="xmu_check_needs_means_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>How data will be processed if used for WLS.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> T/F
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmu_make_mxData">xmu_make_mxData()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_check_needs_means(mtcars, type = "Auto")
xmu_check_needs_means(mtcars, type = "FIML")
# xmu_check_needs_means(mtcars, type = "cov")
# xmu_check_needs_means(mtcars, type = "cor")

# TRUE - marginals means means
xmu_check_needs_means(mtcars, type = "WLS", allContinuousMethod= "marginals")
xmu_check_needs_means(mtcars, type = "ULS", allContinuousMethod= "marginals")
xmu_check_needs_means(mtcars, type = "DWLS", allContinuousMethod= "marginals")

# ================================
# = Provided as an mxData object =
# ================================
tmp = mxData(mtcars, type="raw")
xmu_check_needs_means(tmp, type = "FIML") # TRUE
xmu_check_needs_means(tmp, type = "ULS", allContinuousMethod= "cumulants") #FALSE
# TRUE - means with marginals
xmu_check_needs_means(tmp, type = "WLS", allContinuousMethod= "marginals")
tmp = mxData(cov(mtcars), type="cov", numObs= 100)
# Should catch this can't be type FIML
xmu_check_needs_means(tmp) # FALSE
tmp = mxData(cov(mtcars), means = umx_means(mtcars), type="cov", numObs= 100)
xmu_check_needs_means(tmp) # TRUE

# =======================
# = One var is a factor =
# =======================
tmp = mtcars
tmp$cyl = factor(tmp$cyl)
xmu_check_needs_means(tmp, allContinuousMethod= "cumulants") # TRUE
xmu_check_needs_means(tmp, allContinuousMethod= "marginals") # TRUE - always has means
</code></pre>

<hr>
<h2 id='xmu_check_variance'>Check the minimum variance in data frame</h2><span id='topic+xmu_check_variance'></span>

<h3>Description</h3>

<p>Check that each variable exceeds a minimum variance and all are on compatible scales. Let the user know what to do if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_check_variance(
  data,
  minVar = umx_set_data_variance_check(silent = T)$minVar,
  maxVarRatio = umx_set_data_variance_check(silent = T)$maxVarRatio
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_check_variance_+3A_data">data</code></td>
<td>
<p>the data frame to check</p>
</td></tr>
<tr><td><code id="xmu_check_variance_+3A_minvar">minVar</code></td>
<td>
<p>Minimum allowed variance in variables before warning user variances differ too much.</p>
</td></tr>
<tr><td><code id="xmu_check_variance_+3A_maxvarratio">maxVarRatio</code></td>
<td>
<p>Maximum allowed ratio of variance in data before warning user variances differ too much.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
xmu_check_variance(twinData[, c("wt1", "ht1", "wt2", "ht2")])
twinData[,c("ht1", "ht2")]= twinData[,c("ht1", "ht2")] * 100
xmu_check_variance(twinData[, c("wt1", "ht1", "wt2", "ht2")])
</code></pre>

<hr>
<h2 id='xmu_CI_merge'>xmu_CI_merge</h2><span id='topic+xmu_CI_merge'></span>

<h3>Description</h3>

<p>if you compute some CIs in one model and some in another (copy of the same model, perhaps to get some parallelism),
this is a simple helper to kludge them together.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_CI_merge(m1, m2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_CI_merge_+3A_m1">m1</code></td>
<td>
<p>first copy of the model</p>
</td></tr>
<tr><td><code id="xmu_CI_merge_+3A_m2">m2</code></td>
<td>
<p>second copy of the model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- [mxModel()]
</p>


<h3>References</h3>

<p>- &lt;https://github.com/tbates/umx&gt;
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xmu_CI_merge(m1, m2)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_CI_stash'>Stash the CI values of a model as strings in the values of the model</h2><span id='topic+xmu_CI_stash'></span>

<h3>Description</h3>

<p>Stash formatted CIs (e.g. &quot;.1 [-.1, .3]&quot;) as strings, <em>overwriting</em> the parameter values of the model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_CI_stash(model, digits = 3, dropZeros = FALSE, stdAlg2mat = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_CI_stash_+3A_model">model</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to get CIs from.</p>
</td></tr>
<tr><td><code id="xmu_CI_stash_+3A_digits">digits</code></td>
<td>
<p>rounding.</p>
</td></tr>
<tr><td><code id="xmu_CI_stash_+3A_dropzeros">dropZeros</code></td>
<td>
<p>makes strings for failed CIs?</p>
</td></tr>
<tr><td><code id="xmu_CI_stash_+3A_stdalg2mat">stdAlg2mat</code></td>
<td>
<p>treat std as algebra: stash in non std matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>I might change this to a lookup-function that gets a CI string if one exists.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxConfint">umxConfint()</a></code>, <code><a href="#topic+xmu_get_CI">xmu_get_CI()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmu_clean_label'>Remove illegal characters from labels</h2><span id='topic+xmu_clean_label'></span>

<h3>Description</h3>

<p>Replaces . with _ in labels - e.g. from lavaan where . is common.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_clean_label(label, replace = "_")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_clean_label_+3A_label">label</code></td>
<td>
<p>A label to clean.</p>
</td></tr>
<tr><td><code id="xmu_clean_label_+3A_replace">replace</code></td>
<td>
<p>character to replace . with (default = _)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> legal label string
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmuLabel">xmuLabel()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_clean_label("data.var", replace = "_")
xmu_clean_label("my.var.lab", replace = "_")
</code></pre>

<hr>
<h2 id='xmu_data_missing'>Drop rows with missing definition variables</h2><span id='topic+xmu_data_missing'></span>

<h3>Description</h3>

<p>Definition variables can't be missing. This function helps with that.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_data_missing(
  data,
  selVars,
  sep = NULL,
  dropMissingDef = TRUE,
  hint = "data"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_data_missing_+3A_data">data</code></td>
<td>
<p>The dataframe to check for missing variables</p>
</td></tr>
<tr><td><code id="xmu_data_missing_+3A_selvars">selVars</code></td>
<td>
<p>The variables to check for missingness</p>
</td></tr>
<tr><td><code id="xmu_data_missing_+3A_sep">sep</code></td>
<td>
<p>A sep if this is twin data and selVars are baseNames (default NULL)</p>
</td></tr>
<tr><td><code id="xmu_data_missing_+3A_dropmissingdef">dropMissingDef</code></td>
<td>
<p>Whether to drop the rows, or just stop (TRUE)</p>
</td></tr>
<tr><td><code id="xmu_data_missing_+3A_hint">hint</code></td>
<td>
<p>info for message to user (&quot;data&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> data with missing rows dropped
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="stats.html#topic+complete.cases">complete.cases()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = mtcars; 
tmp[1,]; tmp[1, "wt"] = NA
tmp = xmu_data_missing(tmp, selVars = "wt", sep= NULL, dropMissingDef= TRUE, hint= "mtcars")
dim(mtcars)
dim(tmp)

## Not run: 
tmp = xmu_data_missing(tmp, selVars = "wt", sep= NULL, dropMissingDef= FALSE, hint= "mtcars")

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_data_swap_a_block'>Data helper function to swap blocks of data from one set of columns to another.</h2><span id='topic+xmu_data_swap_a_block'></span>

<h3>Description</h3>

<p>Swap a block of rows of a dataset between two sets of variables (typically twin 1 and twin 2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_data_swap_a_block(theData, rowSelector, T1Names, T2Names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_data_swap_a_block_+3A_thedata">theData</code></td>
<td>
<p>A data frame to swap within.</p>
</td></tr>
<tr><td><code id="xmu_data_swap_a_block_+3A_rowselector">rowSelector</code></td>
<td>
<p>Rows to swap between first and second set of columns.</p>
</td></tr>
<tr><td><code id="xmu_data_swap_a_block_+3A_t1names">T1Names</code></td>
<td>
<p>The first set of columns.</p>
</td></tr>
<tr><td><code id="xmu_data_swap_a_block_+3A_t2names">T2Names</code></td>
<td>
<p>The second set of columns.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> dataframe
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="base.html#topic+subset">subset()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test = data.frame(
a = paste0("a", 1:10),
b = paste0("b", 1:10),
c = paste0("c", 1:10),
d = paste0("d", 1:10), stringsAsFactors = FALSE)
xmu_data_swap_a_block(test, rowSelector = c(1,2,3,6), T1Names = "b", T2Names = "c")
xmu_data_swap_a_block(test, rowSelector = c(1,2,3,6), T1Names = c("a","c"), T2Names = c("b","d"))

</code></pre>

<hr>
<h2 id='xmu_describe_data_WLS'>Determine if a dataset will need statistics for the means if used in a WLS model.</h2><span id='topic+xmu_describe_data_WLS'></span>

<h3>Description</h3>

<p>Given either a data.frame or raw <code>mxData</code>, this function determines whether <code><a href="OpenMx.html#topic+mxFitFunctionWLS">mxFitFunctionWLS()</a></code>
will generate expectations for means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_describe_data_WLS(
  data,
  allContinuousMethod = c("cumulants", "marginals"),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_describe_data_WLS_+3A_data">data</code></td>
<td>
<p>The raw data being used in a <code><a href="OpenMx.html#topic+mxFitFunctionWLS">mxFitFunctionWLS()</a></code> model.</p>
</td></tr>
<tr><td><code id="xmu_describe_data_WLS_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>the method used to process data when all columns are continuous (default = &quot;cumulants&quot;)</p>
</td></tr>
<tr><td><code id="xmu_describe_data_WLS_+3A_verbose">verbose</code></td>
<td>
<p>Whether or not to report diagnostics.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All-continuous models processed using the &quot;cumulants&quot; method LACK means, while
all continuous processed with allContinuousMethod = &quot;marginals&quot; will HAVE means.
</p>
<p>When data are not all continuous, means are modeled and <code>allContinuousMethod</code> is ignored.
</p>


<h3>Value</h3>


<ul>
<li><p> list describing the data.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxFitFunctionWLS">mxFitFunctionWLS()</a></code>, <code><a href="OpenMx.html#topic+omxAugmentDataWithWLSSummary">omxAugmentDataWithWLSSummary()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# ====================================
# = All continuous, data.frame input =
# ====================================

tmp =xmu_describe_data_WLS(mtcars, allContinuousMethod= "cumulants", verbose = TRUE)
tmp$hasMeans # FALSE - no means with cumulants
tmp =xmu_describe_data_WLS(mtcars, allContinuousMethod= "marginals") 
tmp$hasMeans # TRUE we get means with marginals

# ==========================
# = mxData object as input =
# ==========================
tmp = mxData(mtcars, type="raw")
xmu_describe_data_WLS(tmp, allContinuousMethod= "cumulants", verbose = TRUE)$hasMeans # FALSE
xmu_describe_data_WLS(tmp, allContinuousMethod= "marginals")$hasMeans  # TRUE

# =======================================
# = One var is a factor: Means modeled =
# =======================================
tmp = mtcars
tmp$cyl = factor(tmp$cyl)
xmu_describe_data_WLS(tmp, allContinuousMethod= "cumulants")$hasMeans # TRUE - always has means
xmu_describe_data_WLS(tmp, allContinuousMethod= "marginals")$hasMeans # TRUE

</code></pre>

<hr>
<h2 id='xmu_DF_to_mxData_TypeCov'>Convert a dataframe into a cov mxData object</h2><span id='topic+xmu_DF_to_mxData_TypeCov'></span>

<h3>Description</h3>

<p><code>xmu_DF_to_mxData_TypeCov</code> converts a dataframe into <code><a href="OpenMx.html#topic+mxData">mxData()</a></code> with <code>type="cov"</code> and <code>nrow = numObs</code>
and optionally adding means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_DF_to_mxData_TypeCov(
  df,
  columns = NA,
  use = c("complete.obs", "everything", "all.obs", "na.or.complete",
    "pairwise.complete.obs")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_DF_to_mxData_TypeCov_+3A_df">df</code></td>
<td>
<p>the dataframe to covert to an mxData type cov object.</p>
</td></tr>
<tr><td><code id="xmu_DF_to_mxData_TypeCov_+3A_columns">columns</code></td>
<td>
<p>= Which columns to keep (default is all).</p>
</td></tr>
<tr><td><code id="xmu_DF_to_mxData_TypeCov_+3A_use">use</code></td>
<td>
<p>= Default is &quot;complete.obs&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxData">mxData()</a></code> of type = cov
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_DF_to_mxData_TypeCov(mtcars, c("mpg", "hp"))
</code></pre>

<hr>
<h2 id='xmu_dot_define_shapes'>Helper to make the list of vars and their shapes for a graphviz string</h2><span id='topic+xmu_dot_define_shapes'></span>

<h3>Description</h3>

<p>Helper to make a graphviz rank string defining the latent, manifest, and means and their shapes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_define_shapes(latents, manifests, preOut = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_define_shapes_+3A_latents">latents</code></td>
<td>
<p>list of latent variables (including &quot;one&quot;)</p>
</td></tr>
<tr><td><code id="xmu_dot_define_shapes_+3A_manifests">manifests</code></td>
<td>
<p>list of manifest variables</p>
</td></tr>
<tr><td><code id="xmu_dot_define_shapes_+3A_preout">preOut</code></td>
<td>
<p>existing output string (pasted in front of this: &quot;&quot; by default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>string
</p>


<h3>See Also</h3>

<p>- [xmu_dot_rank()]
</p>
<p>Other Graphviz: 
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_mat2dot">xmu_dot_mat2dot</a>()</code>,
<code><a href="#topic+xmu_dot_rank">xmu_dot_rank</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_dot_define_shapes(c("as1"), c("E", "N"))
</code></pre>

<hr>
<h2 id='xmu_dot_make_paths'>xmu_dot_make_paths (not for end users)</h2><span id='topic+xmu_dot_make_paths'></span>

<h3>Description</h3>

<p>Makes graphviz paths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_make_paths(
  mxMat,
  stringIn,
  heads = NULL,
  fixed = TRUE,
  comment = "More paths",
  showResiduals = TRUE,
  labels = "labels",
  digits = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_make_paths_+3A_mxmat">mxMat</code></td>
<td>
<p>An mxMatrix</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_stringin">stringIn</code></td>
<td>
<p>Input string</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_heads">heads</code></td>
<td>
<p>1 or 2 arrows (default NULL - you must set this)</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_fixed">fixed</code></td>
<td>
<p>Whether show fixed values or not (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_comment">comment</code></td>
<td>
<p>A comment to include</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_showresiduals">showResiduals</code></td>
<td>
<p>Whether to show residuals</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_labels">labels</code></td>
<td>
<p>show labels on the path? (&quot;none&quot;, &quot;labels&quot;, &quot;both&quot;)</p>
</td></tr>
<tr><td><code id="xmu_dot_make_paths_+3A_digits">digits</code></td>
<td>
<p>how many digits to report</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- string
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>
<p>Other Graphviz: 
<code><a href="#topic+xmu_dot_define_shapes">xmu_dot_define_shapes</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_mat2dot">xmu_dot_mat2dot</a>()</code>,
<code><a href="#topic+xmu_dot_rank">xmu_dot_rank</a>()</code>
</p>

<hr>
<h2 id='xmu_dot_make_residuals'>xmu_dot_make_residuals (not for end users)</h2><span id='topic+xmu_dot_make_residuals'></span>

<h3>Description</h3>

<p>xmu_dot_make_residuals (not for end users)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_make_residuals(
  mxMat,
  latents = NULL,
  fixed = TRUE,
  digits = 2,
  resid = c("circle", "line")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_make_residuals_+3A_mxmat">mxMat</code></td>
<td>
<p>An A or S mxMatrix</p>
</td></tr>
<tr><td><code id="xmu_dot_make_residuals_+3A_latents">latents</code></td>
<td>
<p>Optional list of latents to alter location of circles (defaults to NULL)</p>
</td></tr>
<tr><td><code id="xmu_dot_make_residuals_+3A_fixed">fixed</code></td>
<td>
<p>Whether to show fixed values or not</p>
</td></tr>
<tr><td><code id="xmu_dot_make_residuals_+3A_digits">digits</code></td>
<td>
<p>How many digits to report</p>
</td></tr>
<tr><td><code id="xmu_dot_make_residuals_+3A_resid">resid</code></td>
<td>
<p>How to show residuals and variances default is &quot;circle&quot;. Other option is &quot;line&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- list of variance names and variances
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>
<p>Other Graphviz: 
<code><a href="#topic+xmu_dot_define_shapes">xmu_dot_define_shapes</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_mat2dot">xmu_dot_mat2dot</a>()</code>,
<code><a href="#topic+xmu_dot_rank">xmu_dot_rank</a>()</code>
</p>

<hr>
<h2 id='xmu_dot_maker'>Internal umx function to help plotting graphviz</h2><span id='topic+xmu_dot_maker'></span>

<h3>Description</h3>

<p>Helper to print a digraph to file and open it
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_maker(model, file, digraph, strip_zero = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_maker_+3A_model">model</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to get the name from</p>
</td></tr>
<tr><td><code id="xmu_dot_maker_+3A_file">file</code></td>
<td>
<p>Either &quot;name&quot; (use model name) or a file name</p>
</td></tr>
<tr><td><code id="xmu_dot_maker_+3A_digraph">digraph</code></td>
<td>
<p>Graphviz code for a model</p>
</td></tr>
<tr><td><code id="xmu_dot_maker_+3A_strip_zero">strip_zero</code></td>
<td>
<p>Whether to remove the leading &quot;0.&quot; in digits in the diagram</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> optionally returns the digraph text.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>
<p>Other Graphviz: 
<code><a href="#topic+xmu_dot_define_shapes">xmu_dot_define_shapes</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_mat2dot">xmu_dot_mat2dot</a>()</code>,
<code><a href="#topic+xmu_dot_rank">xmu_dot_rank</a>()</code>
</p>

<hr>
<h2 id='xmu_dot_mat2dot'>Return dot code for paths in a matrix</h2><span id='topic+xmu_dot_mat2dot'></span>

<h3>Description</h3>

<p>Return dot code for paths in a matrix is a function which walks the rows and cols of a matrix.
At each free cell, it creates a dot-string specifying the relevant path, e.g.:
</p>
<p><code>ai1 -&gt; var1 [label=".35"]</code>
</p>
<p>Its main use is to correctly generate paths (and their sources and sink objects)
without depending on the label of the parameter.
</p>
<p>It is highly customizable:
</p>

<ol>
<li><p> You can specify which cells to inspect, e.g. &quot;lower&quot;.
</p>
</li>
<li><p> You can choose how to interpret path direction, from = &quot;cols&quot;.
</p>
</li>
<li><p> You can choose the label for the from to ends of the path (by default, the matrix name is used).
</p>
</li>
<li><p> Offer up a list of from and toLabel which will be indexed into for source and sink
</p>
</li>
<li><p> You can set the number of arrows on a path (e.g. both).
</p>
</li>
<li><p> If <code>type</code> is set, then sources and sinks added manifests and/or latents output (p)
</p>
</li></ol>

<p>Finally, you can pass in previous output and new paths will be concatenated to these.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_mat2dot(
  x,
  cells = c("diag", "lower", "lower_inc", "upper", "upper_inc", "any", "left"),
  from = c("rows", "cols"),
  fromLabel = NULL,
  toLabel = NULL,
  showFixed = FALSE,
  arrows = c("forward", "both", "back"),
  fromType = NULL,
  toType = NULL,
  digits = 2,
  model = NULL,
  SEstyle = FALSE,
  p = list(str = "", latents = c(), manifests = c())
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_mat2dot_+3A_x">x</code></td>
<td>
<p>a <code><a href="#topic+umxMatrix">umxMatrix()</a></code> to make paths from.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_cells">cells</code></td>
<td>
<p>which cells to process: &quot;any&quot; (default), &quot;diag&quot;, &quot;lower&quot;, &quot;upper&quot;. &quot;left&quot; is the left half (e.g. in a twin means matrix)</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_from">from</code></td>
<td>
<p>one of &quot;rows&quot;, &quot;columns&quot;</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_fromlabel">fromLabel</code></td>
<td>
<p>= NULL. NULL = use matrix name (default). If one, if suffixed with index, length() &gt; 1, index into list. &quot;one&quot; is special.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_tolabel">toLabel</code></td>
<td>
<p>= NULL. NULL = use matrix name (default). If one, if suffixed with index, length() &gt; 1, index into list.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_showfixed">showFixed</code></td>
<td>
<p>= FALSE.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_arrows">arrows</code></td>
<td>
<p>&quot;forward&quot; &quot;both&quot; or &quot;back&quot;</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_fromtype">fromType</code></td>
<td>
<p>one of &quot;latent&quot; or &quot;manifest&quot; NULL (default) = don't accumulate new names.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_totype">toType</code></td>
<td>
<p>one of &quot;latent&quot; or &quot;manifest&quot; NULL (default) = don't accumulate new names.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_digits">digits</code></td>
<td>
<p>to round values to (default = 2).</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_model">model</code></td>
<td>
<p>If you want to get CIs, you can pass in the model (default = NULL).</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_sestyle">SEstyle</code></td>
<td>
<p>If TRUE, CIs shown as &quot;b(SE)&quot; (&quot;b [l,h]&quot; if FALSE (default)). Ignored if model NULL.</p>
</td></tr>
<tr><td><code id="xmu_dot_mat2dot_+3A_p">p</code></td>
<td>
<p>input to build on. list(str = &quot;&quot;, latents = c(), manifests = c())</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> list(str = &quot;&quot;, latents = c(), manifests = c())
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+plot">plot()</a></code>
</p>
</li></ul>

<p>Other Graphviz: 
<code><a href="#topic+xmu_dot_define_shapes">xmu_dot_define_shapes</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_rank">xmu_dot_rank</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# test with a 1 * 1
a_cp = umxMatrix("a_cp", "Lower", 1, 1, free = TRUE, values = pi)
out = xmu_dot_mat2dot(a_cp, cells = "lower_inc", from = "cols", arrows = "both")
cat(out$str) # a_cp -&gt; a_cp [dir = both label="2"];
out = xmu_dot_mat2dot(a_cp, cells = "lower_inc", from = "cols", arrows = "forward",
	fromLabel = "fromMe", toLabel = "toYou", 
	fromType  = "latent", toType  = "manifest", digits = 3, SEstyle = TRUE
	)
cat(out$str) # fromMe -&gt; toYou [dir = forward label="3.142"];
cat(out$latent) # fromMe
cat(out$manifest) # toYou

# Make a lower 3 * 3 value= 1:6 (1, 4, 6 on the diag)
a_cp = umxMatrix("a_cp", "Lower", 3, 3, free = TRUE, values = 1:6)

# Get dot strings for lower triangle (default from and to based on row and column number)
out = xmu_dot_mat2dot(a_cp, cells = "lower", from = "cols", arrows = "both")
cat(out$str) # a_cp1 -&gt; a_cp2 [dir = both label="2"];

# one arrow (the default = "forward")
out = xmu_dot_mat2dot(a_cp, cells = "lower", from = "cols")
cat(out$str) # a_cp1 -&gt; a_cp2 [dir = forward label="2"];

# label to (rows) using var names

out = xmu_dot_mat2dot(a_cp, toLabel= paste0("v", 1:3), cells = "lower", from = "cols")
umx_msg(out$str) # a_cp1 -&gt; v2 [dir = forward label="2"] ...

# First call also inits the plot struct
out = xmu_dot_mat2dot(a_cp, from = "rows", cells = "lower", arrows = "both", fromType = "latent")
out = xmu_dot_mat2dot(a_cp, from = "rows", cells = "diag", 
		toLabel= "common", toType = "manifest", p = out)
umx_msg(out$str); umx_msg(out$manifests); umx_msg(out$latents)

# ================================
# = Add found sinks to manifests =
# ================================
out = xmu_dot_mat2dot(a_cp, from= "rows", cells= "diag", 
		toLabel= c('a','b','c'), toType= "manifest");
umx_msg(out$manifests)

# ================================
# = Add found sources to latents =
# ================================
out = xmu_dot_mat2dot(a_cp, from= "rows", cells= "diag", 
		toLabel= c('a','b','c'), fromType= "latent");
umx_msg(out$latents)


# ========================
# = Label a means matrix =
# ========================

tmp = umxMatrix("expMean", "Full", 1, 4, free = TRUE, values = 1:4)
out = xmu_dot_mat2dot(tmp, cells = "left", from = "rows",
	fromLabel= "one", toLabel= c("v1", "v2")
)
cat(out$str)

## Not run: 
# ==============================================
# = Get a string which includes CI information =
# ==============================================
data(demoOneFactor)
latents = c("g"); manifests = names(demoOneFactor)
m1 = umxRAM("xmu_dot", data = demoOneFactor, type = "cov",
	umxPath(latents, to = manifests),
	umxPath(var = manifests),
	umxPath(var = latents, fixedAt = 1.0)
)
m1 = umxCI(m1, run= "yes")
out = xmu_dot_mat2dot(m1$A, from = "cols", cells = "any", 
      toLabel= paste0("x", 1:5), fromType = "latent", model= m1);
umx_msg(out$str); umx_msg(out$latents)


## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_dot_move_ranks'>xmu_dot_move_ranks (not for end users)</h2><span id='topic+xmu_dot_move_ranks'></span>

<h3>Description</h3>

<p>Variables will be moved from any existing rank to the new one. Setting a rank to &quot;&quot; will clear it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_move_ranks(
  min = NULL,
  same = NULL,
  max = NULL,
  old_min,
  old_same,
  old_max
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_move_ranks_+3A_min">min</code></td>
<td>
<p>vars to group at top of plot</p>
</td></tr>
<tr><td><code id="xmu_dot_move_ranks_+3A_same">same</code></td>
<td>
<p>vars to group at the same level</p>
</td></tr>
<tr><td><code id="xmu_dot_move_ranks_+3A_max">max</code></td>
<td>
<p>vars to group at bottom of plot</p>
</td></tr>
<tr><td><code id="xmu_dot_move_ranks_+3A_old_min">old_min</code></td>
<td>
<p>vars to group at top of plot</p>
</td></tr>
<tr><td><code id="xmu_dot_move_ranks_+3A_old_same">old_same</code></td>
<td>
<p>vars to group at the same level</p>
</td></tr>
<tr><td><code id="xmu_dot_move_ranks_+3A_old_max">old_max</code></td>
<td>
<p>vars to group at bottom of plot</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> list(min=min, same=same, max=max)
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>old_min = c("min1", "min2")
old_same = c("s1", "s2")
old_max = paste0("x", 1:3)

# Add L1 to min
xmu_dot_move_ranks(min = "L1", old_min= old_min, old_same= old_same, old_max= old_max)

# Move min1 to max
xmu_dot_move_ranks(max = "min1", old_min= old_min, old_same= old_same, old_max= old_max)

# Clear min
xmu_dot_move_ranks(min = "", old_min= old_min, old_same= old_same, old_max= old_max)
</code></pre>

<hr>
<h2 id='xmu_dot_rank'>Helper to make a graphviz rank string</h2><span id='topic+xmu_dot_rank'></span>

<h3>Description</h3>

<p>Given a list of names, this filters the list, and returns a graphviz string to force them into the given rank.
e.g. <code>"{rank=same; as1};"</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_rank(vars, pattern, rank)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_rank_+3A_vars">vars</code></td>
<td>
<p>a list of strings</p>
</td></tr>
<tr><td><code id="xmu_dot_rank_+3A_pattern">pattern</code></td>
<td>
<p>regular expression to filter vars</p>
</td></tr>
<tr><td><code id="xmu_dot_rank_+3A_rank">rank</code></td>
<td>
<p>&quot;same&quot;, &quot;max&quot;, &quot;min&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>string
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmu_dot_define_shapes">xmu_dot_define_shapes()</a></code>
</p>
</li></ul>

<p>Other Graphviz: 
<code><a href="#topic+xmu_dot_define_shapes">xmu_dot_define_shapes</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_mat2dot">xmu_dot_mat2dot</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_dot_rank(c("as1"), "^[ace]s[0-9]+$", "same")
</code></pre>

<hr>
<h2 id='xmu_dot_rank_str'>xmu_dot_rank_str (not for end users)</h2><span id='topic+xmu_dot_rank_str'></span>

<h3>Description</h3>

<p>xmu_dot_rank_str (not for end users)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_dot_rank_str(min = NULL, same = NULL, max = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_dot_rank_str_+3A_min">min</code></td>
<td>
<p>vars to group at top of plot</p>
</td></tr>
<tr><td><code id="xmu_dot_rank_str_+3A_same">same</code></td>
<td>
<p>vars to group at the same level</p>
</td></tr>
<tr><td><code id="xmu_dot_rank_str_+3A_max">max</code></td>
<td>
<p>vars to group at bottom of plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- GraphViz rank string
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_dot_rank_str(min = "L1", same = c("x1", "x2"), max = paste0("e", 1:3))

</code></pre>

<hr>
<h2 id='xmu_extract_column'>Get one or more columns from mzData or regular data.frame</h2><span id='topic+xmu_extract_column'></span>

<h3>Description</h3>

<p>same effect as <code>df[, col]</code> but works for <code><a href="OpenMx.html#topic+mxData">mxData()</a></code> and check the names are present
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_extract_column(data, col, drop = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_extract_column_+3A_data">data</code></td>
<td>
<p>mxData or data.frame</p>
</td></tr>
<tr><td><code id="xmu_extract_column_+3A_col">col</code></td>
<td>
<p>the name(s) of the column(s) to extract</p>
</td></tr>
<tr><td><code id="xmu_extract_column_+3A_drop">drop</code></td>
<td>
<p>whether to drop the structure of the data.frame when extracting one column</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> column of data
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_extract_column(mtcars, "wt")
xmu_extract_column(mxData(mtcars, type = "raw"), "wt")
xmu_extract_column(mxData(mtcars, type = "raw"), "wt", drop=TRUE)
xmu_extract_column(mxData(mtcars, type = "raw"), c("wt", "mpg"))
</code></pre>

<hr>
<h2 id='xmu_get_CI'>Look up and report CIs for free parameters</h2><span id='topic+xmu_get_CI'></span>

<h3>Description</h3>

<p>Look up CIs for free parameters in a model, and return as APA-formatted text string.
If std are available, then these are reported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_get_CI(
  model,
  label,
  prefix = "top.",
  suffix = "_std",
  digits = 2,
  SEstyle = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_get_CI_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to get CIs from</p>
</td></tr>
<tr><td><code id="xmu_get_CI_+3A_label">label</code></td>
<td>
<p>the label of the cell to interrogate for a CI, e.g. &quot;ai_r1c1&quot;</p>
</td></tr>
<tr><td><code id="xmu_get_CI_+3A_prefix">prefix</code></td>
<td>
<p>The submodel to look in (default = &quot;top.&quot;)</p>
</td></tr>
<tr><td><code id="xmu_get_CI_+3A_suffix">suffix</code></td>
<td>
<p>The suffix for algebras when standardized (default = &quot;_std&quot;)</p>
</td></tr>
<tr><td><code id="xmu_get_CI_+3A_digits">digits</code></td>
<td>
<p>Rounding digits.</p>
</td></tr>
<tr><td><code id="xmu_get_CI_+3A_sestyle">SEstyle</code></td>
<td>
<p>If TRUE, report &quot;b(se)&quot; instead of b CI95[l,u] (default = FALSE)</p>
</td></tr>
<tr><td><code id="xmu_get_CI_+3A_verbose">verbose</code></td>
<td>
<p>= FALSE</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> the CI string, e.g. &quot;.73[-.20, .98]&quot; or .73(.10)
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx); data(demoOneFactor)
manifests = names(demoOneFactor)

tmp = umxRAM("get_CI_example", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1)
)
tmp = umxCI(tmp, run= "yes")

# Get CI by parameter label
xmu_get_CI(model= tmp, "x1_with_x1")
xmu_get_CI(model= tmp, "x1_with_x1", SEstyle = TRUE, digits = 3)

# prefix (submodel) and suffix (e.g. std) are ignored if not needed
xmu_get_CI(model= tmp, "x1_with_x1", prefix = "top.", suffix = "_std")

xmu_get_CI(fit_IP, label = "ai_r1c1", prefix = "top.", suffix = "_std")
xmu_get_CI(fit_IP, label = "ai_r1c1", prefix = "top.", SEstyle = TRUE, suffix = "_std")

## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_lavaan_process_group'>Process table of paths to model</h2><span id='topic+xmu_lavaan_process_group'></span>

<h3>Description</h3>

<p>Process a set of lavaan tables rows forming a group (Model).
Returns empty arrays if no rows matching the requested group are found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_lavaan_process_group(tab, groupNum)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_lavaan_process_group_+3A_tab">tab</code></td>
<td>
<p>a parameter table</p>
</td></tr>
<tr><td><code id="xmu_lavaan_process_group_+3A_groupnum">groupNum</code></td>
<td>
<p>group number to filter table on</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> list(plist=plist, latents = latents, manifests = manifests)
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxLav2RAM">umxLav2RAM()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tab = lavaan::lavaanify("y~x")
xmu_lavaan_process_group(tab, groupNum = 1)
xmu_lavaan_process_group(tab, groupNum = 0)


## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_make_bin_cont_pair_data'>Make pairs of  bin &amp; continuous columns to represent censored data</h2><span id='topic+xmu_make_bin_cont_pair_data'></span>

<h3>Description</h3>

<p>Takes a dataframe of left-censored variables (vars with a floor effect) and does two things to it:
1. It creates new binary (1/0) copies of each column (with the suffix &quot;bin&quot;). These contain 0 where
the variable is below the minimum and NA otherwise.
2. In each existing variable, it sets all instances of min for that var to NA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_make_bin_cont_pair_data(data, vars = NULL, suffixes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_make_bin_cont_pair_data_+3A_data">data</code></td>
<td>
<p>A [data.frame()] to convert</p>
</td></tr>
<tr><td><code id="xmu_make_bin_cont_pair_data_+3A_vars">vars</code></td>
<td>
<p>The variables to process</p>
</td></tr>
<tr><td><code id="xmu_make_bin_cont_pair_data_+3A_suffixes">suffixes</code></td>
<td>
<p>Suffixes if the data are family (wide, more than one persona on a row)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- copy of the dataframe with new binary variables and censoring
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df = xmu_make_bin_cont_pair_data(mtcars, vars = c("mpg"))
str(df)
df[order(df$mpg), c(1,12)]
# Introduce a floor effect
tmp = mtcars; tmp$mpg[tmp$mpg&lt;=15]=15
tmp$mpg_T1 = tmp$mpg_T2 = tmp$mpg
df = xmu_make_bin_cont_pair_data(tmp, vars = c("mpg"), suffixes = c("_T1", "_T2"))
df[order(df$mpg), 12:15]
</code></pre>

<hr>
<h2 id='xmu_make_mxData'>Upgrade a dataframe to an mxData type.</h2><span id='topic+xmu_make_mxData'></span>

<h3>Description</h3>

<p><code>xmu_make_mxData</code> is an internal function to upgrade a dataframe to <code>mxData</code>. It can also drop variables and rows
from the dataframe.
The most common use will be to give it a dataframe, and get back an <code>mxData</code> object of type raw, cov, cor (WLS is just raw).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_make_mxData(
  data = NULL,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  manifests = NULL,
  numObs = NULL,
  weight = NULL,
  fullCovs = NULL,
  dropMissingDef = TRUE,
  verbose = FALSE,
  use = "pairwise.complete.obs"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_make_mxData_+3A_data">data</code></td>
<td>
<p>A <code><a href="base.html#topic+data.frame">data.frame()</a></code> or <code><a href="OpenMx.html#topic+mxData">mxData()</a></code></p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_type">type</code></td>
<td>
<p>What data type is wanted out c(&quot;Auto&quot;, &quot;FIML&quot;, &quot;cov&quot;, &quot;cor&quot;, 'WLS', 'DWLS', 'ULS')</p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_manifests">manifests</code></td>
<td>
<p>If set, only these variables will be retained.</p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_numobs">numObs</code></td>
<td>
<p>Only needed if you pass in a cov/cor matrix wanting this to be upgraded to mxData</p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_weight">weight</code></td>
<td>
<p>Passes weight values to mxData</p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_fullcovs">fullCovs</code></td>
<td>
<p>Covariate names if any (NULL = none) These are checked by <code>dropMissingDef</code></p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_dropmissingdef">dropMissingDef</code></td>
<td>
<p>Whether to automatically drop missing def var rows for the user (default = TRUE). You get a polite note.</p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_verbose">verbose</code></td>
<td>
<p>If verbose, report on columns kept and dropped (default FALSE)</p>
</td></tr>
<tr><td><code id="xmu_make_mxData_+3A_use">use</code></td>
<td>
<p>When type = cov or cor, should this drop NAs? (use = &quot;pairwise.complete.obs&quot; by default, with a polite note)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxData">mxData()</a></code>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># =========================
# = Continuous ML example =
# =========================
data(mtcars)
tmp = xmu_make_mxData(data= mtcars, type = "Auto"); # class(tmp); # "MxDataStatic"
# names(tmp$observed) # "mpg"  "cyl"  "disp"
manVars = c("mpg", "cyl", "disp")
tmp = xmu_make_mxData(data= mtcars, type = "Auto", manifests = manVars); 
tmp$type == "raw" # TRUE

# ==============================
# = All continuous WLS example =
# ==============================
tmp = xmu_make_mxData(data= mtcars, type = "WLS" , manifests = manVars, verbose= TRUE)
tmp$type == "raw" # TRUE (WLS is triggered by the fit function, not the data type)

# ============================
# = Missing data WLS example =
# ============================
tmp = mtcars; tmp[1, "mpg"] = NA # add NA
tmp = xmu_make_mxData(data= tmp, type = "WLS", manifests = manVars, verbose= TRUE)

## Not run: 
# ==========================
# = already mxData example =
# ==========================
m1 = umxRAM("auto", data = mxData(mtcars, type = "raw"),
umxPath(var= "wt"),
umxPath(mean=  "wt")
)


## End(Not run)

# ========================
# = Cov and cor examples =
# ========================
tmp = xmu_make_mxData(data= mtcars, type = "cov", manifests = c("mpg", "cyl"))
tmp = xmu_make_mxData(data= mtcars, type = "cor", manifests = c("mpg", "cyl"))
tmp = xmu_make_mxData(data= cov(mtcars[, c("mpg", "cyl")]), 
        type = "cov", manifests = c("mpg", "cyl"), numObs=200)

# mxData input examples
tmp = mxData(cov(mtcars[, c("mpg", "cyl")]), type = "cov", numObs= 100)
xmu_make_mxData(data= tmp, type = "cor", manifests = c("mpg", "cyl")) # consume mxData
xmu_make_mxData(data= tmp, type = "cor", manifests = c("mpg"))        # trim existing mxData
xmu_make_mxData(data= tmp, type = "cor") # no manifests specified (use all)
xmu_make_mxData(data= tmp, manifests = c("mpg", "cyl")) # auto

# =======================
# = Pass string through =
# =======================
xmu_make_mxData(data= c("a", "b", "c"), type = "Auto")

</code></pre>

<hr>
<h2 id='xmu_make_TwinSuperModel'>Helper to make a basic top, MZ, and DZ model.</h2><span id='topic+xmu_make_TwinSuperModel'></span>

<h3>Description</h3>

<p><code>xmu_make_TwinSuperModel</code> makes basic twin model containing <code>top</code>, <code>MZ</code>, and <code>DZ</code> models. It intelligently handles thresholds for
ordinal data, and means model for covariates matrices in the twin models if needed.
</p>
<p>It's the replacement for <code>xmu_assemble_twin_supermodel</code> approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_make_TwinSuperModel(
  name = "twin_super",
  mzData,
  dzData,
  selDVs,
  selCovs = NULL,
  sep = NULL,
  type = c("Auto", "FIML", "cov", "cor", "WLS", "DWLS", "ULS"),
  allContinuousMethod = c("cumulants", "marginals"),
  numObsMZ = NULL,
  numObsDZ = NULL,
  nSib = 2,
  equateMeans = TRUE,
  weightVar = NULL,
  bVector = FALSE,
  dropMissingDef = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_name">name</code></td>
<td>
<p>for the supermodel</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_mzdata">mzData</code></td>
<td>
<p>Dataframe containing the MZ data</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_dzdata">dzData</code></td>
<td>
<p>Dataframe containing the DZ data</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_seldvs">selDVs</code></td>
<td>
<p>List of manifest base names (e.g. BMI, NOT 'BMI_T1') (OR, you don't set &quot;sep&quot;, the full variable names)</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_selcovs">selCovs</code></td>
<td>
<p>List of covariate base names (e.g. age, NOT 'age_T1') (OR, you don't set &quot;sep&quot;, the full variable names)</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_sep">sep</code></td>
<td>
<p>string used to expand selDVs into selVars, i.e., &quot;_T&quot; to expand BMI into BMI_T1 and BMI_T2 (optional but STRONGLY encouraged)</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_type">type</code></td>
<td>
<p>One of 'Auto','FIML','cov', 'cor', 'WLS','DWLS', or 'ULS'. Auto tries to react to the incoming mxData type (raw/cov).</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>&quot;cumulants&quot; or &quot;marginals&quot;. Used in all-continuous WLS data to determine if a means model needed.</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_numobsmz">numObsMZ</code></td>
<td>
<p>Number of MZ observations contributing (for summary data only)</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_numobsdz">numObsDZ</code></td>
<td>
<p>Number of DZ observations contributing (for summary data only)</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_nsib">nSib</code></td>
<td>
<p>Number of members per family (default = 2)</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate T1 and T2 means (default = TRUE).</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_weightvar">weightVar</code></td>
<td>
<p>If provided, a vector objective will be used to weight the data. (default = NULL).</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_bvector">bVector</code></td>
<td>
<p>Whether to compute row-wise likelihoods (defaults to FALSE).</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_dropmissingdef">dropMissingDef</code></td>
<td>
<p>Whether to automatically drop missing def var rows for the user (default = TRUE). You get a polite note.</p>
</td></tr>
<tr><td><code id="xmu_make_TwinSuperModel_+3A_verbose">verbose</code></td>
<td>
<p>(default = FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>xmu_make_TwinSuperModel</code> is used in twin models (e.g.<code><a href="#topic+umxCP">umxCP()</a></code>, <code><a href="#topic+umxACE">umxACE()</a></code> and <code><a href="#topic+umxACEv">umxACEv()</a></code> and will be added to the other models: <code><a href="#topic+umxGxE">umxGxE()</a></code>, <code><a href="#topic+umxIP">umxIP()</a></code>,
simplifying code maintenance.
</p>
<p>It takes <code>mzData</code> and <code>dzData</code>, a list of the <code>selDVs</code> to analyse and optional <code>selCovs</code> (as well as <code>sep</code> and <code>nSib</code>), along with other
relevant information such as whether the user wants to <code>equateMeans</code>.
It can also handle a <code>weightVar</code>.
</p>
<p>If covariates are passed in these are included in the means model (via a call to <code>xmuTwinUpgradeMeansToCovariateModel</code>.
</p>
<p><strong>Modeling</strong>
</p>
<p><strong>Matrices created</strong>
</p>
<p><em>top model</em>
</p>
<p>For raw and WLS data, <code>top</code> contains a <code>expMeans</code> matrix (if needed). For summary data, the top model contains only a name.
</p>
<p>For ordinal data, <code>top</code> gains <code>top.threshMat</code> (from a call to <code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix()</a></code>).
</p>
<p>For covariates, top stores the <code>intercepts</code> matrix and a <code>betaDef</code> matrix. These are then used to make expMeans in <code>MZ</code> and <code>DZ</code>.
</p>
<p><em>MZ and DZ models</em>
</p>
<p><code>MZ</code> and <code>DZ</code> contain the data, and an expectation referencing <code>top.expCovMZ</code> and <code>top.expMean</code>, and, <code>vector = bVector</code>.
For continuous raw data, MZ and DZ contain <code><a href="OpenMx.html#topic+mxExpectationNormal">OpenMx::mxExpectationNormal()</a></code> and <code><a href="OpenMx.html#topic+mxFitFunctionML">OpenMx::mxFitFunctionML()</a></code>.
For WLS these the fit function is switched to <code><a href="OpenMx.html#topic+mxFitFunctionWLS">OpenMx::mxFitFunctionWLS()</a></code> with appropriate <code>type</code> and <code>allContinuousMethod</code>.
</p>
<p>For binary, a constraint and algebras are included to constrain <code>Vtot</code> (A+C+E) to 1.
</p>
<p>If a <code>weightVar</code> is detected, these columns are used to create a row-weighted MZ and DZ models.
</p>
<p>If <code>equateMeans</code> is <code>TRUE</code>, then the Twin-2 vars in the mean matrix are equated by label with Twin-1.
</p>
<p>Decent starts are guessed from the data.
<code>varStarts</code> is computed as <code>sqrt(variance)/3</code> of the DVs and <code>meanStarts</code> as the variable means.
For raw data, a check is made for ordered variables. For Binary variables, means are fixed at 0 and
total variance (A+C+E) is fixed at 1. For ordinal variables, the first 2 thresholds are fixed.
</p>
<p>Where needed, e.g. continuous raw data, top adds a means matrix &quot;expMean&quot;.
For ordinal data, top adds a <code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix()</a></code>.
</p>
<p>If binary variables are present, matrices and a constraint to hold <code>A+C+E == 1</code> are added to top.
</p>
<p>If a weight variable is offered up, an <code>mzWeightMatrix</code> will be added.
</p>
<p><strong>Data handling</strong>
</p>
<p>In terms of data handling, <code>xmu_make_TwinSuperModel</code> was primarily designed to take
data.frames and process these into mxData.
It can also, however, handle cov and mxData input.
</p>
<p>It can process data into all the types supported by <code>mxData</code>.
</p>
<p>Raw data input with a target of <code>cov</code> or <code>cor</code> type requires the <code>numObsMZ</code> and <code>numObsDZ</code> to be set.
</p>
<p>Type &quot;WLS&quot;, &quot;DWLS&quot;, or &quot;ULS&quot;, data remain raw, but are handled as WLS in the <code><a href="OpenMx.html#topic+mxFitFunctionWLS">OpenMx::mxFitFunctionWLS()</a></code>.
</p>
<p>Unused columns are dropped.
</p>
<p>If you pass in raw data, you can't request type cov/cor yet. Will work on this if desired.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>s for top, MZ and DZ.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ==============
# = Continuous =
# ==============
library(umx)
data(twinData)
twinData = umx_scale(twinData, varsToScale= c('ht1','ht2'))
mzData = twinData[twinData$zygosity %in%  "MZFF",] 
dzData = twinData[twinData$zygosity %in%  "DZFF",]
m1= xmu_make_TwinSuperModel(mzData=mzData, dzData=dzData, selDVs=c("wt","ht"), sep="", nSib=2)
names(m1) # "top" "MZ"  "DZ"
class(m1$MZ$fitfunction)[[1]] == "MxFitFunctionML"

# ====================
# = With a covariate =
# ====================

m1= xmu_make_TwinSuperModel(mzData=mzData, dzData=dzData, 
		selDVs= "wt", selCovs= "age", sep="", nSib=2)
m1$top$intercept$labels
m1$MZ$expMean

# ===============
# = WLS example =
# ===============
m1=xmu_make_TwinSuperModel(mzData=mzData, dzData=dzData,selDVs=c("wt","ht"),sep="",type="WLS")
class(m1$MZ$fitfunction)[[1]] == "MxFitFunctionWLS"
m1$MZ$fitfunction$type =="WLS"
# Check default all-continuous method
m1$MZ$fitfunction$continuousType == "cumulants"

# Choose non-default type (DWLS)
m1= xmu_make_TwinSuperModel(mzData= mzData, dzData= dzData,
	selDVs= c("wt","ht"), sep="", type="DWLS")
m1$MZ$fitfunction$type =="DWLS"
class(m1$MZ$fitfunction)[[1]] == "MxFitFunctionWLS"

# Switch WLS method
m1 = xmu_make_TwinSuperModel(mzData= mzData, dzData= dzData, selDVs= c("wt","ht"), sep= "",
  type = "WLS", allContinuousMethod = "marginals")
m1$MZ$fitfunction$continuousType == "marginals"
class(m1$MZ$fitfunction)[[1]] == "MxFitFunctionWLS"


# ============================================
# = Bivariate continuous and ordinal example =
# ============================================
data(twinData)
selDVs = c("wt", "obese")
# Cut BMI column to form ordinal obesity variables
ordDVs          = c("obese1", "obese2")
obesityLevels   = c('normal', 'overweight', 'obese')
cutPoints       = quantile(twinData[, "bmi1"], probs = c(.5, .2), na.rm = TRUE)
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
# Make the ordinal variables into mxFactors (ensure ordered is TRUE, and require levels)
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])
mzData = twinData[twinData$zygosity %in%  "MZFF",] 
dzData = twinData[twinData$zygosity %in%  "DZFF",]
m1 = xmu_make_TwinSuperModel(mzData= mzData, dzData= dzData, selDVs= selDVs, sep="", nSib= 2)
names(m1) # "top" "MZ"  "DZ" 

# ==============
# = One binary =
# ==============
data(twinData)
cutPoints       = quantile(twinData[, "bmi1"], probs = .2, na.rm = TRUE)
obesityLevels   = c('normal', 'obese')
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])
selDVs = c("wt", "obese")
mzData = twinData[twinData$zygosity %in% "MZFF",]
dzData = twinData[twinData$zygosity %in% "DZFF",]
m1 = xmu_make_TwinSuperModel(mzData= mzData, dzData= dzData, selDVs= selDVs, sep= "", nSib= 2)

# ========================================
# = Cov data (calls xmuTwinSuper_CovCor) =
# ========================================

data(twinData)
mzData =cov(twinData[twinData$zygosity %in% "MZFF", tvars(c("wt","ht"), sep="")], use="complete")
dzData =cov(twinData[twinData$zygosity %in% "DZFF", tvars(c("wt","ht"), sep="")], use="complete")
m1 = xmu_make_TwinSuperModel(mzData= mzData, dzData= dzData, selDVs= "wt", sep= "", 
	nSib= 2, numObsMZ = 100, numObsDZ = 100, verbose=TRUE)
class(m1$MZ$fitfunction)[[1]] =="MxFitFunctionML"
dimnames(m1$MZ$data$observed)[[1]]==c("wt1", "wt2")

</code></pre>

<hr>
<h2 id='xmu_match.arg'>Select first item in list of options, while being flexible about choices.</h2><span id='topic+xmu_match.arg'></span>

<h3>Description</h3>

<p>Like a smart version of <code><a href="base.html#topic+match.arg">match.arg()</a></code>: Handles selecting parameter options when default is a list.
Unlike  <code><a href="base.html#topic+match.arg">match.arg()</a></code> <code>xmu_match.arg</code> allows items not in the list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_match.arg(x, option_list, check = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_match.arg_+3A_x">x</code></td>
<td>
<p>the value chosen (may be the default option list)</p>
</td></tr>
<tr><td><code id="xmu_match.arg_+3A_option_list">option_list</code></td>
<td>
<p>A vector of valid options</p>
</td></tr>
<tr><td><code id="xmu_match.arg_+3A_check">check</code></td>
<td>
<p>Whether to check that single items are in the list. Set false to accept abbreviations (defaults to TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> one validated option
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="base.html#topic+match.arg">match.arg()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>option_list = c("default", "par.observed", "empirical")

xmu_match.arg("par.observed", option_list)
xmu_match.arg("allow me", option_list, check = FALSE)
xmu_match.arg(option_list, option_list)
option_list = c(NULL, "par.observed", "empirical")
 # fails with NULL!!!!!
xmu_match.arg(option_list, option_list)
option_list = c(NA, "par.observed", "empirical")
xmu_match.arg(option_list, option_list) # use NA instead
option_list = c(TRUE, FALSE, NA)
xmu_match.arg(option_list, option_list) # works with non character
# An example of checking a bad item and stopping
## Not run: 
	tmp &lt;- function(x= c("one", "two", "three")) {
		xmu_match.arg(x, option_list = c("one", "two", "three"))
	}
testthat::expect_true(tmp() == "one")
testthat::expect_error(tmp("bad"))
	tmp &lt;- function(x= c("one", "two", "three")) {
		xmu_match.arg(x, option_list = c("one", "two", "three"), check = FALSE)
	}
testthat::expect_true(tmp("OK") == "OK")
testthat::expect_error(tmp(), NA)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_name_from_lavaan_str'>Find name for model</h2><span id='topic+xmu_name_from_lavaan_str'></span>

<h3>Description</h3>

<p>Use name if provided. If first line contains a #, uses this line as name. Else use default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_name_from_lavaan_str(lavaanString = NULL, name = NA, default = "m1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_name_from_lavaan_str_+3A_lavaanstring">lavaanString</code></td>
<td>
<p>A model string, possibly with # model name on line 1.</p>
</td></tr>
<tr><td><code id="xmu_name_from_lavaan_str_+3A_name">name</code></td>
<td>
<p>A desired model name (optional).</p>
</td></tr>
<tr><td><code id="xmu_name_from_lavaan_str_+3A_default">default</code></td>
<td>
<p>A default name if nothing else found.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A name string
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>"m1" == xmu_name_from_lavaan_str("x~~x")
"bob" == xmu_name_from_lavaan_str(name = "bob")
"my_model" == xmu_name_from_lavaan_str("# my model")

</code></pre>

<hr>
<h2 id='xmu_PadAndPruneForDefVars'>Where all data are missing for a twin, add default values for definition variables, allowing the row to be kept</h2><span id='topic+xmu_PadAndPruneForDefVars'></span>

<h3>Description</h3>

<p>Replaces NAs in definition slots with the mean for that variable ONLY where all data are missing for that twin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_PadAndPruneForDefVars(
  df,
  varNames,
  defNames,
  suffixes,
  highDefValue = 99,
  rm = c("drop_missing_def", "pad_with_mean")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_PadAndPruneForDefVars_+3A_df">df</code></td>
<td>
<p>The dataframe to process</p>
</td></tr>
<tr><td><code id="xmu_PadAndPruneForDefVars_+3A_varnames">varNames</code></td>
<td>
<p>list of names of the variables being analysed</p>
</td></tr>
<tr><td><code id="xmu_PadAndPruneForDefVars_+3A_defnames">defNames</code></td>
<td>
<p>list of covariates</p>
</td></tr>
<tr><td><code id="xmu_PadAndPruneForDefVars_+3A_suffixes">suffixes</code></td>
<td>
<p>that map names on columns in df (i.e., c(&quot;T1&quot;, &quot;T2&quot;))</p>
</td></tr>
<tr><td><code id="xmu_PadAndPruneForDefVars_+3A_highdefvalue">highDefValue</code></td>
<td>
<p>What to replace missing definition variables (covariates) with. Default = 99</p>
</td></tr>
<tr><td><code id="xmu_PadAndPruneForDefVars_+3A_rm">rm</code></td>
<td>
<p>= how to handle missing values in the varNames. Default is &quot;drop_missing_def&quot;, &quot;pad_with_mean&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> dataframe
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData)
sum(is.na(twinData$ht1))
df = xmu_PadAndPruneForDefVars(twinData, varNames = "ht", defNames = "wt", c("1", "2"))

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_path_regex'>Re-name variables umxPaths to twin versions</h2><span id='topic+xmu_path_regex'></span>

<h3>Description</h3>

<p><code>xmu_path2twin</code> takes a collection of <code><a href="#topic+umxPath">umxPath()</a></code>s (use base variable names),
and returns a model for both twins (and using the expanded variable names).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_path_regex(input, pattern = NA, replacement = NA, ignore = "one")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_path_regex_+3A_input">input</code></td>
<td>
<p>vector of path labels</p>
</td></tr>
<tr><td><code id="xmu_path_regex_+3A_pattern">pattern</code></td>
<td>
<p>= pattern to match and replace</p>
</td></tr>
<tr><td><code id="xmu_path_regex_+3A_replacement">replacement</code></td>
<td>
<p>= replacement string</p>
</td></tr>
<tr><td><code id="xmu_path_regex_+3A_ignore">ignore</code></td>
<td>
<p>Labels to ignore (reserved words like &quot;one&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A path like <code style="white-space: pre;">&#8288;a to b&#8288;</code> will be returned as <code style="white-space: pre;">&#8288;a_T1 to b_T1&#8288;</code>.
</p>


<h3>Value</h3>


<ul>
<li><p> renamed paths
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">tutorials</a>, <a href="https://github.com/tbates/umx">github</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmu_path2twin">xmu_path2twin()</a></code>, <code><a href="#topic+umxTwinMaker">umxTwinMaker()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_path_regex(c("a", "one", "b"), pattern = "$", replacement = "_T1")
# "a_T1" "one"  "b_T1"
</code></pre>

<hr>
<h2 id='xmu_path2twin'>Re-name variables in umxPaths to twin versions</h2><span id='topic+xmu_path2twin'></span>

<h3>Description</h3>

<p><code>xmu_path2twin</code> takes a collection of paths that use base variable names,
and returns a model with twin names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_path2twin(paths, thisTwin = 1, sep = "_T")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_path2twin_+3A_paths">paths</code></td>
<td>
<p>A collection of paths using base variable names.</p>
</td></tr>
<tr><td><code id="xmu_path2twin_+3A_thistwin">thisTwin</code></td>
<td>
<p>The twin we are making (i.e., &quot;_T1&quot;, or &quot;_T2&quot;)</p>
</td></tr>
<tr><td><code id="xmu_path2twin_+3A_sep">sep</code></td>
<td>
<p>The separator (default &quot;_T&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A path like <code style="white-space: pre;">&#8288;a to b&#8288;</code> will be returned as <code style="white-space: pre;">&#8288;a_T1 to b_T1&#8288;</code>.
</p>


<h3>Value</h3>


<ul>
<li><p> list of relabeled paths
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxTwinMaker">umxTwinMaker()</a></code>, <code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>twin1PathList = c(
umxPath(v1m0 = c("a1", 'c1', "e1")),
umxPath(fromEach = c("a1", 'c1', "e1"), to = "NFC3", values=.2)
)
xmu_path2twin(twin1PathList, thisTwin = 2)

</code></pre>

<hr>
<h2 id='xmu_print_algebras'>Print algebras from a umx model</h2><span id='topic+xmu_print_algebras'></span>

<h3>Description</h3>

<p><code>xmu_print_algebras</code> adds the results of algebras to a summary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_print_algebras(model, digits = 3, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_print_algebras_+3A_model">model</code></td>
<td>
<p>A umx model from which to print algebras.</p>
</td></tr>
<tr><td><code id="xmu_print_algebras_+3A_digits">digits</code></td>
<td>
<p>rounding (default = 3)</p>
</td></tr>
<tr><td><code id="xmu_print_algebras_+3A_verbose">verbose</code></td>
<td>
<p>tell user if no algebras found</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Non-user function called by <code><a href="#topic+umxSummary">umxSummary()</a></code>
</p>


<h3>Value</h3>


<ul>
<li><p> nothing
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSummary">umxSummary()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(mlbench)
data(BostonHousing2)
BostonHousing2$log_crim = log2(BostonHousing2$crim)
BostonHousing2$nox      = BostonHousing2$nox*100
m2 = umxRAM(data = BostonHousing2, "#crime_model
	cmedv ~ log_crim + b1*nox; 
	nox   ~ a1*rad + a2*log_crim
i_1 := a1*b1
i_2 := a2*b1"
)
m3 = mxRun(mxModel(m1, mxAlgebra(name= "rtwo", rbind(i_1, i_2))))
m3 = mxRun(mxModel(m3, mxAlgebra(name= "ctwo", cbind(i_1, i_2))))
xmu_print_algebras(m3)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_rclabel_2_bracket_address'>Convert an &quot;A_r1c1&quot;-style label to a bracket address.</h2><span id='topic+xmu_rclabel_2_bracket_address'></span>

<h3>Description</h3>

<p>Takes a label like &quot;A_r1c1&quot; and returns &quot;A[1,1]&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_rclabel_2_bracket_address(label, dotprefix = "", suffix = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_rclabel_2_bracket_address_+3A_label">label</code></td>
<td>
<p>A umx style row col label</p>
</td></tr>
<tr><td><code id="xmu_rclabel_2_bracket_address_+3A_dotprefix">dotprefix</code></td>
<td>
<p>Dot address prefix for label (e.g., &quot;ai&quot;</p>
</td></tr>
<tr><td><code id="xmu_rclabel_2_bracket_address_+3A_suffix">suffix</code></td>
<td>
<p>e.g. &quot;_std&quot; default = &quot;&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> label e.g. &quot;ai[1,1]&quot;
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>, <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_rclabel_2_bracket_address(label = "A_r1c1") #A[1,1]
xmu_rclabel_2_bracket_address(label = "A_r10c1")
xmu_rclabel_2_bracket_address(label = "A_r1c1", dotprefix = "model.top")
xmu_rclabel_2_bracket_address("A_r1c1", suffix= "_std")
xmu_rclabel_2_bracket_address("A_r1c1", dotprefix="myModel", suffix="_std")

</code></pre>

<hr>
<h2 id='xmu_safe_run_summary'>Safely run and summarize a model</h2><span id='topic+xmu_safe_run_summary'></span>

<h3>Description</h3>

<p>The main benefit is that it returns the model, even if it can't be run.
</p>
<p>The function will run the model if requested, wrapped in <code><a href="base.html#topic+tryCatch">tryCatch()</a></code> to avoid throwing an error.
If summary = TRUE then <code><a href="#topic+umxSummary">umxSummary()</a></code> is requested (again, wrapped in try).
</p>
<p><em>note</em>: If <code>autoRun</code> is logical, then it over-rides <code>summary</code> to match <code>autoRun</code>. This is useful for easy use <code><a href="#topic+umxRAM">umxRAM()</a></code> and twin models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_safe_run_summary(
  model1,
  model2 = NULL,
  autoRun = TRUE,
  tryHard = c("no", "yes", "ordinal", "search"),
  summary = !umx_set_silent(silent = TRUE),
  std = "default",
  comparison = TRUE,
  digits = 3,
  intervals = FALSE,
  returning = c("model", "summary"),
  refModels = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_safe_run_summary_+3A_model1">model1</code></td>
<td>
<p>The model to attempt to run and summarize.</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_model2">model2</code></td>
<td>
<p>Optional second model to compare with model1.</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_autorun">autoRun</code></td>
<td>
<p>Whether to run or not (default = TRUE) Options are FALSE and &quot;if needed&quot;.</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_tryhard">tryHard</code></td>
<td>
<p>Default ('no') uses normal mxRun. &quot;yes&quot; uses mxTryHard. Other options: &quot;ordinal&quot;, &quot;search&quot;</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_summary">summary</code></td>
<td>
<p>Whether to print model summary (default = autoRun).</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_std">std</code></td>
<td>
<p>What to print in summary. &quot;default&quot; = the object's summary default. FALSE = raw, TRUE = standardize, NULL = omit parameter table.</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_comparison">comparison</code></td>
<td>
<p>Toggle to allow not making comparison, even if second model is provided (more flexible in programming).</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_digits">digits</code></td>
<td>
<p>Rounding precision in tables and plots</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_intervals">intervals</code></td>
<td>
<p>whether to run intervals or not (default FALSE)</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_returning">returning</code></td>
<td>
<p>What to return (default, the run model)</p>
</td></tr>
<tr><td><code id="xmu_safe_run_summary_+3A_refmodels">refModels</code></td>
<td>
<p>whether to run refModels or not (default NULL)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxTryHard">mxTryHard()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
tmp = mtcars
tmp$disp = tmp$disp/100
m1 = umxRAM("tim", data = tmp,
	umxPath(c("wt", "disp"), to = "mpg"),
	umxPath("wt", with = "disp"),
	umxPath(v.m. = c("wt", "disp", "mpg"))
)
m2 = umxModify(m1, "wt_to_mpg")

# Summary ignored if run is false
xmu_safe_run_summary(m1, autoRun = FALSE, summary = TRUE)
# Run, no summary
xmu_safe_run_summary(m1, autoRun = TRUE, summary = FALSE)
# Default summary is just fit string
xmu_safe_run_summary(m1, autoRun = TRUE, summary = TRUE)
# Show std parameters
xmu_safe_run_summary(m1, autoRun = TRUE, summary = TRUE, std = TRUE)
# Run + Summary + comparison
xmu_safe_run_summary(m1, m2, autoRun = TRUE, summary = TRUE, intervals = TRUE)
# Run + Summary + no comparison
xmu_safe_run_summary(m1, m2, autoRun = TRUE, summary = TRUE, std = TRUE, comparison= FALSE)


## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_set_sep_from_suffix'>Just a helper to cope with deprecated suffix lying around.</h2><span id='topic+xmu_set_sep_from_suffix'></span>

<h3>Description</h3>

<p>Returns either suffix or sep, with a deprecation warning if suffix is set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_set_sep_from_suffix(sep, suffix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_set_sep_from_suffix_+3A_sep">sep</code></td>
<td>
<p>The separator (if suffix != 'deprecated', then this is returned).</p>
</td></tr>
<tr><td><code id="xmu_set_sep_from_suffix_+3A_suffix">suffix</code></td>
<td>
<p>The suffix, defaults to 'deprecated'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- sep
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_set_sep_from_suffix(sep = "_T", suffix = "deprecated")
</code></pre>

<hr>
<h2 id='xmu_show_fit_or_comparison'>Show model logLik of model or print comparison table</h2><span id='topic+xmu_show_fit_or_comparison'></span>

<h3>Description</h3>

<p>Just a helper to show the logLik of a model or print a comparison table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_show_fit_or_comparison(model, comparison = NULL, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_show_fit_or_comparison_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to report on</p>
</td></tr>
<tr><td><code id="xmu_show_fit_or_comparison_+3A_comparison">comparison</code></td>
<td>
<p>If not NULL, used as comparison model</p>
</td></tr>
<tr><td><code id="xmu_show_fit_or_comparison_+3A_digits">digits</code></td>
<td>
<p>(default = 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSummary">umxSummary()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
xmu_show_fit_or_comparison(model, comparison, digits=3)

## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_simplex_corner'>Internal function to help building simplex models</h2><span id='topic+xmu_simplex_corner'></span>

<h3>Description</h3>

<p>internal function to help building simplex models is a function which
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_simplex_corner(x, start = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_simplex_corner_+3A_x">x</code></td>
<td>
<p>size of matrix, or an <code><a href="#topic+umxMatrix">umxMatrix()</a></code> of which to free the bottom triangle.</p>
</td></tr>
<tr><td><code id="xmu_simplex_corner_+3A_start">start</code></td>
<td>
<p>a default start value for the freed items.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="#topic+umxMatrix">umxMatrix()</a></code>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxMatrix">umxMatrix()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = umxMatrix('test', 'Full', nrow = 4, ncol = 4)
xmu_simplex_corner(x, start = .9)
# See how we have a diag free, but offset 1-down?
umx_print( xmu_simplex_corner(x, start = .9)$values, zero=".")
</code></pre>

<hr>
<h2 id='xmu_standardize_ACE'>xmu_standardize_ACE</h2><span id='topic+xmu_standardize_ACE'></span>

<h3>Description</h3>

<p>Standardize an ACE model <em>BUT</em> you probably want <code><a href="#topic+umx_standardize">umx_standardize()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_ACE(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_ACE_+3A_model">model</code></td>
<td>
<p>an <code><a href="#topic+umxACE">umxACE()</a></code> model to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_ACE_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Standardized ACE <code><a href="#topic+umxACE">umxACE()</a></code> model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(umx)
data(twinData)
selDVs = c("bmi1", "bmi2")
mzData = twinData[twinData$zygosity %in% "MZFF", selDVs]
dzData = twinData[twinData$zygosity %in% "DZFF", selDVs]
m1     = umxACE(selDVs = selDVs, dzData = dzData, mzData = mzData)
std    = xmu_standardize_ACE(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_ACEcov'>xmu_standardize_ACEcov</h2><span id='topic+xmu_standardize_ACEcov'></span>

<h3>Description</h3>

<p>Standardize an ACE model with covariates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_ACEcov(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_ACEcov_+3A_model">model</code></td>
<td>
<p>an <code><a href="#topic+umxACEcov">umxACEcov()</a></code> model to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_ACEcov_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Standardized <code><a href="#topic+umxACEcov">umxACEcov()</a></code> model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(twinData)
twinData$age1 = twinData$age2 = twinData$age
selDVs  = c("bmi")
selCovs = c("ht") # silly example
selVars = umx_paste_names(c(selDVs, selCovs), sep = "", suffixes= 1:2)
mzData = subset(twinData, zyg == 1, selVars)[1:80, ]
dzData = subset(twinData, zyg == 3, selVars)[1:80, ]
m1 = umxACEcov(selDVs = selDVs, selCovs = selCovs, dzData = dzData, mzData = mzData, 
	 sep = "", autoRun = TRUE)
fit = xmu_standardize_ACEcov(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_ACEv'>Standardize an ACE variance components model (ACEv)</h2><span id='topic+xmu_standardize_ACEv'></span>

<h3>Description</h3>

<p>xmu_standardize_ACE allows umx_standardize to standardize an ACE variance components model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_ACEv(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_ACEv_+3A_model">model</code></td>
<td>
<p>An <code><a href="#topic+umxACEv">umxACEv()</a></code> model to standardize.</p>
</td></tr>
<tr><td><code id="xmu_standardize_ACEv_+3A_...">...</code></td>
<td>
<p>Other parameters.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A standardized <code><a href="#topic+umxACEv">umxACEv()</a></code> model.
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
require(umx)
data(twinData)
mzData = twinData[twinData$zygosity %in% "MZFF",]
dzData = twinData[twinData$zygosity %in% "DZFF",]
m1  = umxACEv(selDVs = "bmi", sep="", dzData = dzData, mzData = mzData)
std = umx_standardize(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_CP'>Function to standardize a common pathway model</h2><span id='topic+xmu_standardize_CP'></span>

<h3>Description</h3>

<p>You probably want <code><a href="#topic+umx_standardize">umx_standardize()</a></code>. This function simply inserts the standardized CP components into the ai ci ei and as cs es matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_CP(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_CP_+3A_model">model</code></td>
<td>
<p>an <code><a href="#topic+umxCP">umxCP()</a></code> model to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_CP_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> standardized <code><a href="#topic+umxCP">umxCP()</a></code> model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
selDVs = c("gff", "fc", "qol", "hap", "sat", "AD") 
m1 = umxCP(selDVs = selDVs, nFac = 3, data=GFF, zyg="zyg_2grp")
m2 = xmu_standardize_CP(m1)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_IP'>non-user: Standardize an IP model</h2><span id='topic+xmu_standardize_IP'></span>

<h3>Description</h3>

<p>You probably want <code><a href="#topic+umx_standardize">umx_standardize()</a></code>. This function simply copies the standardized IP components into the ai ci ei and as cs es matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_IP(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_IP_+3A_model">model</code></td>
<td>
<p>an <code><a href="#topic+umxIP">umxIP()</a></code> model to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_IP_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> standardized IP <code><a href="#topic+umxIP">umxIP()</a></code> model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
model = xmu_standardize_IP(model)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_RAM'>Standardize a Structural Model (not for end users)</h2><span id='topic+xmu_standardize_RAM'></span>

<h3>Description</h3>

<p>You probably want <code><a href="#topic+umx_standardize">umx_standardize()</a></code>, not this.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_RAM(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_RAM_+3A_model">model</code></td>
<td>
<p>The <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> you wish to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_RAM_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>xmu_standardize_RAM</code> takes a RAM-style model, and returns standardized version.
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
manifests = names(demoOneFactor)

m1 = umxRAM("std_ex", data = demoOneFactor, type = "cov",
	umxPath("G", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "G", fixedAt = 1.0)
)

m1 = xmu_standardize_RAM(m1)
m1 = umx_standardize(m1)
umxSummary(m1)


## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_SexLim'>Standardize a SexLim model</h2><span id='topic+xmu_standardize_SexLim'></span>

<h3>Description</h3>

<p><code>xmu_standardize_SexLim</code> would move standardized Sexlim values into raw cells, but can't as these are algebras.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_SexLim(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_SexLim_+3A_model">model</code></td>
<td>
<p>an <code><a href="#topic+umxSexLim">umxSexLim()</a></code> model to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_SexLim_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> standardized <code><a href="#topic+umxSexLim">umxSexLim()</a></code> model
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
model = xmu_standardize_SexLim(model)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_standardize_Simplex'>Standardize a Simplex twin model</h2><span id='topic+xmu_standardize_Simplex'></span>

<h3>Description</h3>

<p>xmu_standardize_Simplex
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_standardize_Simplex(model, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_standardize_Simplex_+3A_model">model</code></td>
<td>
<p>an <code><a href="#topic+umxSimplex">umxSimplex()</a></code> model to standardize</p>
</td></tr>
<tr><td><code id="xmu_standardize_Simplex_+3A_...">...</code></td>
<td>
<p>Other options</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Standardized Simplex <code><a href="#topic+umxSimplex">umxSimplex()</a></code> model
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://tbates.github.io">https://tbates.github.io</a>,  <a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(iqdat)
mzData = subset(iqdat, zygosity == "MZ")
dzData = subset(iqdat, zygosity == "DZ")
m1  = umxSimplex(selDVs = paste0("IQ_age", 1:4), sep = "_T", 
			dzData = dzData, mzData = mzData, tryHard = "yes")
std = xmu_standardize_Simplex(m1)

## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_start_value_list'>Make start values</h2><span id='topic+xmu_start_value_list'></span>

<h3>Description</h3>

<p>Purpose: Create startvalues for OpenMx paths
use cases
umx:::xmuStart_value_list(1)
xmuValues(1) # 1 value, varying around 1, with sd of .1
xmuValues(1, n=letters) # length(letters) start values, with mean 1 and sd .1
xmuValues(100, 15)  # 1 start, with mean 100 and sd 15
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_start_value_list(mean = 1, sd = NA, n = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_start_value_list_+3A_mean">mean</code></td>
<td>
<p>the mean start value</p>
</td></tr>
<tr><td><code id="xmu_start_value_list_+3A_sd">sd</code></td>
<td>
<p>the sd of values</p>
</td></tr>
<tr><td><code id="xmu_start_value_list_+3A_n">n</code></td>
<td>
<p>how many to generate</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> start value list
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmu_starts'>Helper providing boilerplate start values for means and variance in twin models</h2><span id='topic+xmu_starts'></span>

<h3>Description</h3>

<p><code>xmu_starts</code> can handle several common/boilerplate situations in which means and variance start values
are used in twin models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_starts(
  mzData,
  dzData,
  selVars = selVars,
  sep = NULL,
  equateMeans = NULL,
  nSib,
  varForm = c("Cholesky"),
  SD = TRUE,
  divideBy = 3
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_starts_+3A_mzdata">mzData</code></td>
<td>
<p>Data for MZ pairs.</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_dzdata">dzData</code></td>
<td>
<p>Data for DZ pairs.</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_selvars">selVars</code></td>
<td>
<p>Variable names: If sep = NULL, then treated as full names for both sibs.</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_sep">sep</code></td>
<td>
<p>All the variables full names.</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_equatemeans">equateMeans</code></td>
<td>
<p>(NULL)</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_nsib">nSib</code></td>
<td>
<p>How many subjects in a family.</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_varform">varForm</code></td>
<td>
<p>currently just &quot;Cholesky&quot; style.</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_sd">SD</code></td>
<td>
<p>= TRUE (FALSE = variance, not SD).</p>
</td></tr>
<tr><td><code id="xmu_starts_+3A_divideby">divideBy</code></td>
<td>
<p>= 3 (A,C,E) 1/3rd each. Use 1 to do this yourself post-hoc.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> varStarts and meanStarts
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(twinData)
selDVs = c("wt", "ht")
mzData = twinData[twinData$zygosity %in%  "MZFF", ] 
dzData = twinData[twinData$zygosity %in%  "DZFF", ]

round(sqrt(var(dzData[,tvars(selDVs, "")], na.rm=TRUE)/3),3)
xmu_starts(mzData, dzData, selVars=selDVs, nSib= 2, sep="", equateMeans=TRUE, varForm="Cholesky")

# Variance instead of SD
round(var(dzData[,tvars(selDVs, "")], na.rm=TRUE)/3,3)
xmu_starts(mzData, dzData, selVars = selDVs, nSib = 2, sep= "", 
	equateMeans= TRUE, varForm= "Cholesky", SD= FALSE)

# one variable
xmu_starts(mzData, dzData, selVars= "wt", nSib = 2, sep="", equateMeans = TRUE)

# Ordinal/continuous mix
data(twinData)
twinData= umx_scale_wide_twin_data(data=twinData,varsToScale="wt",sep= "")
# Cut BMI column to form ordinal obesity variables
cuts     = quantile(twinData[, "bmi1"], probs = c(.5, .8), na.rm = TRUE)
obLevels = c('normal', 'overweight', 'obese')
twinData$obese1= cut(twinData$bmi1,breaks=c(-Inf,cuts,Inf),labels=obLevels)
twinData$obese2= cut(twinData$bmi2,breaks=c(-Inf,cuts,Inf),labels=obLevels)
# Make the ordinal variables into mxFactors
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])
mzData = twinData[twinData$zygosity %in% "MZFF",] 
dzData = twinData[twinData$zygosity %in% "DZFF",]
xmu_starts(mzData, dzData, selVars = c("wt","obese"), sep= "", 
 nSib= 2, equateMeans = TRUE, SD= FALSE)

xmu_starts(mxData(mzData, type="raw"), mxData(mzData, type="raw"), 
   selVars = c("wt","obese"), sep= "", nSib= 2, equateMeans = TRUE, SD= FALSE)

# ==============
# = Three sibs =
# ==============
data(twinData)
twinData$wt3 = twinData$wt2
twinData$ht3 = twinData$ht2
selDVs = c("wt", "ht")
mzData = twinData[twinData$zygosity %in%  "MZFF", ] 
dzData = twinData[twinData$zygosity %in%  "DZFF", ]

xmu_starts(mzData, dzData, selVars=selDVs, sep="", nSib=3, equateMeans=TRUE)
xmu_starts(mzData, dzData, selVars=selDVs, sep="", nSib=3, equateMeans=FALSE)
</code></pre>

<hr>
<h2 id='xmu_summary_RAM_group_parameters'>Order and group the parameters in a RAM summary</h2><span id='topic+xmu_summary_RAM_group_parameters'></span>

<h3>Description</h3>

<p>Makes understanding complex model output easier by grouping parameters are type: residuals, latent variance, factor loading etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_summary_RAM_group_parameters(
  model,
  paramTable,
  means = FALSE,
  residuals = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_summary_RAM_group_parameters_+3A_model">model</code></td>
<td>
<p>the model containing the parameters.</p>
</td></tr>
<tr><td><code id="xmu_summary_RAM_group_parameters_+3A_paramtable">paramTable</code></td>
<td>
<p>The parameter table.</p>
</td></tr>
<tr><td><code id="xmu_summary_RAM_group_parameters_+3A_means">means</code></td>
<td>
<p>Whether to show the means (FALSE)</p>
</td></tr>
<tr><td><code id="xmu_summary_RAM_group_parameters_+3A_residuals">residuals</code></td>
<td>
<p>Whether to show the residuals (FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> Sorted parameter table
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxSummary">umxSummary()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(demoOneFactor)
manifests = names(demoOneFactor)
m1 = umxRAM("One Factor", data = demoOneFactor,
	umxPath("G", to = manifests),
	umxPath(v.m. = manifests),
	umxPath(v1m0 = "G")
)
tmp = umxSummary(m1, means=FALSE, residuals = FALSE)
xmu_summary_RAM_group_parameters(m1, paramTable = tmp,  means= FALSE, residuals= FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_twin_add_WeightMatrices'>Add weight matrices to twin models.</h2><span id='topic+xmu_twin_add_WeightMatrices'></span>

<h3>Description</h3>

<p>Add weight models (MZw, DZw) with matrices (e.g. mzWeightMatrix) to a twin model, and
update <code>mxFitFunctionMultigroup</code>. This yields a weighted model with vector objective.
</p>
<p>To weight objective functions in OpenMx, you specify a container model that applies the weights
m1 is the model with no weights, but with &quot;vector = TRUE&quot; option added to the FIML objective.
This option makes FIML return individual likelihoods for each row of the data (rather than a single
-2LL value for the model)
You then optimize weighted versions of these likelihoods by building additional models containing
weight data and an algebra that multiplies the likelihoods from the first model by the weight vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_twin_add_WeightMatrices(model, mzWeights = NULL, dzWeights = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_twin_add_WeightMatrices_+3A_model">model</code></td>
<td>
<p>umx-style twin model</p>
</td></tr>
<tr><td><code id="xmu_twin_add_WeightMatrices_+3A_mzweights">mzWeights</code></td>
<td>
<p>data for MZ weights matrix</p>
</td></tr>
<tr><td><code id="xmu_twin_add_WeightMatrices_+3A_dzweights">dzWeights</code></td>
<td>
<p>data for DZ weights matrix</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> model
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tmp = umx_make_twin_data_nice(data=twinData, sep="", zygosity="zygosity", numbering= 1:2)
m1  = umxACE(selDVs = "wt", data = tmp, dzData = "DZFF", mzData = "MZFF", autoRun= FALSE)
m1$MZ$fitfunction$vector= TRUE

tmp = xmu_twin_add_WeightMatrices(m1,
	mzWeights= rnorm(nrow(m1$MZ$data$observed)), 
	dzWeights= rnorm(nrow(m1$DZ$data$observed))
)

</code></pre>

<hr>
<h2 id='xmu_twin_check'>Check basic aspects of input for twin models.</h2><span id='topic+xmu_twin_check'></span>

<h3>Description</h3>

<p>Check that DVs are in the data, that the data have rows, set the optimizer if requested.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_twin_check(
  selDVs,
  dzData = dzData,
  mzData = mzData,
  sep = NULL,
  enforceSep = TRUE,
  nSib = 2,
  numObsMZ = NULL,
  numObsDZ = NULL,
  optimizer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_twin_check_+3A_seldvs">selDVs</code></td>
<td>
<p>Variables used in the data.</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_dzdata">dzData</code></td>
<td>
<p>The DZ twin data.</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_mzdata">mzData</code></td>
<td>
<p>The MZ twin data.</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_sep">sep</code></td>
<td>
<p>Separator between base-name and numeric suffix when creating variable names, e.g. &quot;_T&quot;</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_enforcesep">enforceSep</code></td>
<td>
<p>Whether to require sep to be set, or just warn if it is not (Default = TRUE: enforce).</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_nsib">nSib</code></td>
<td>
<p>How many people per family? (Default = 2).</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_numobsmz">numObsMZ</code></td>
<td>
<p>set if data are not raw.</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_numobsdz">numObsDZ</code></td>
<td>
<p>set if data are not raw.</p>
</td></tr>
<tr><td><code id="xmu_twin_check_+3A_optimizer">optimizer</code></td>
<td>
<p>Set by name (if you want to change it).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(umx)
data(twinData)
mzData = subset(twinData, zygosity == "MZFF")
dzData = subset(twinData, zygosity == "MZFF")
xmu_twin_check(selDVs = c("wt", "ht"), dzData = dzData, mzData = mzData, 
	sep = "", enforceSep = TRUE)
xmu_twin_check(selDVs = c("wt", "ht"), dzData = dzData, mzData = mzData, 
	sep = "", enforceSep = FALSE)
xmu_twin_check(selDVs = c("wt", "ht"), dzData = dzData, mzData = mzData, 
	sep = "", enforceSep = TRUE, nSib = 2, optimizer = NULL)

## Not run: 
# TODO xmu_twin_check: move to a test file:
# 1. stop on no rows
xmu_twin_check("Generativity", twinData[NULL,], twinData[NULL,], sep="_T")
# Error in xmu_twin_check("Generativity", twinData[NULL, ], twinData[NULL,  : 
#   Your DZ dataset has no rows!

# 2. Stop on a NULL sep  = NULL IFF enforceSep = TRUE
xmu_twin_check(selDVs = c("wt", "ht"), dzData = dzData, mzData = mzData, enforceSep = TRUE)
# 3. stop on a factor with sep = NULL

## End(Not run)
</code></pre>

<hr>
<h2 id='xmu_twin_get_var_names'>Not for user: pull variable names from a twin model</h2><span id='topic+xmu_twin_get_var_names'></span>

<h3>Description</h3>

<p>Barely useful, but justified perhaps by centralizing trimming the &quot;_T1&quot; off, and returning just twin 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_twin_get_var_names(
  model,
  source = c("expCovMZ", "observed"),
  trim = TRUE,
  twinOneOnly = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_twin_get_var_names_+3A_model">model</code></td>
<td>
<p>A model to get the variables from</p>
</td></tr>
<tr><td><code id="xmu_twin_get_var_names_+3A_source">source</code></td>
<td>
<p>Whether to access the dimnames of the &quot;expCovMZ&quot; or the names of the &quot;observed&quot; data (will include covariates)</p>
</td></tr>
<tr><td><code id="xmu_twin_get_var_names_+3A_trim">trim</code></td>
<td>
<p>Whether to trim the suffix (TRUE)</p>
</td></tr>
<tr><td><code id="xmu_twin_get_var_names_+3A_twinoneonly">twinOneOnly</code></td>
<td>
<p>Whether to return on the names for twin 1 (i.e., unique names)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> variable names from twin model
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData) # ?twinData from Australian twins.
twinData[, c("ht1", "ht2")] = twinData[, c("ht1", "ht2")] * 10
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
m1 = umxACE(selDVs= "ht", sep= "", dzData= dzData, mzData= mzData, autoRun= FALSE)
selVars = xmu_twin_get_var_names(m1, source = "expCovMZ", trim = TRUE, twinOneOnly = TRUE) # "ht"
umx_check(selVars == "ht")
xmu_twin_get_var_names(m1, source= "expCovMZ", trim= FALSE, twinOneOnly= FALSE) # "ht1" "ht2"
selVars = xmu_twin_get_var_names(m1, source= "observed", trim= TRUE, twinOneOnly= TRUE)# "ht"
nVar = length(selVars)
umx_check(nVar == 1)

## End(Not run)

</code></pre>

<hr>
<h2 id='xmu_twin_make_def_means_mats_and_alg'>Make the matrices and algebras for definition-based means models</h2><span id='topic+xmu_twin_make_def_means_mats_and_alg'></span>

<h3>Description</h3>

<p>not-for-end-user helper for means in twin models. Returns matrices for
each definition variable, and an algebra to compute means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_twin_make_def_means_mats_and_alg(baseCovs, fullVars, nSib, sep)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_twin_make_def_means_mats_and_alg_+3A_basecovs">baseCovs</code></td>
<td>
<p>base names of the DVs, e.g. &quot;age&quot;</p>
</td></tr>
<tr><td><code id="xmu_twin_make_def_means_mats_and_alg_+3A_fullvars">fullVars</code></td>
<td>
<p>full names of the DVs, e.g. &quot;E_T1&quot;</p>
</td></tr>
<tr><td><code id="xmu_twin_make_def_means_mats_and_alg_+3A_nsib">nSib</code></td>
<td>
<p>how many siblings - typically 2</p>
</td></tr>
<tr><td><code id="xmu_twin_make_def_means_mats_and_alg_+3A_sep">sep</code></td>
<td>
<p>in twin variable, i.e., &quot;_T&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrices and an algebra
</p>


<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># xmu_twin_make_def_means_mats_and_alg(baseCovs= baseCovs, 
#   fullVars = fullVars, nSib = nSib, sep= sep)
</code></pre>

<hr>
<h2 id='xmu_twin_upgrade_selDvs2SelVars'>Upgrade selDVs to selVars</h2><span id='topic+xmu_twin_upgrade_selDvs2SelVars'></span>

<h3>Description</h3>

<p>Just a helper to go from &quot;wt&quot; to &quot;wt_T1&quot; contingent on sep not being null
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmu_twin_upgrade_selDvs2SelVars(selDVs, sep, nSib)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmu_twin_upgrade_selDvs2SelVars_+3A_seldvs">selDVs</code></td>
<td>
<p>with wt or wt_T1</p>
</td></tr>
<tr><td><code id="xmu_twin_upgrade_selDvs2SelVars_+3A_sep">sep</code></td>
<td>
<p>either &quot;&quot; etc., or NULL</p>
</td></tr>
<tr><td><code id="xmu_twin_upgrade_selDvs2SelVars_+3A_nsib">nSib</code></td>
<td>
<p>wideness of data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of wt_T1 wt_T2 etc.
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="#topic+umx">umx</a>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmu_twin_upgrade_selDvs2SelVars("wt", NULL, 2)

</code></pre>

<hr>
<h2 id='xmuHasSquareBrackets'>xmuHasSquareBrackets</h2><span id='topic+xmuHasSquareBrackets'></span>

<h3>Description</h3>

<p>Tests if an input has square brackets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuHasSquareBrackets(input)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuHasSquareBrackets_+3A_input">input</code></td>
<td>
<p>an input to test</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> TRUE/FALSE
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmuHasSquareBrackets("A[1,2]")
</code></pre>

<hr>
<h2 id='xmuLabel'>xmuLabel: Add labels to a RAM model, matrix, or path</h2><span id='topic+xmuLabel'></span>

<h3>Description</h3>

<p>xmuLabel adds labels to things, be it an: <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> (RAM or matrix based), an <code><a href="OpenMx.html#topic+mxPath">mxPath()</a></code>, or an <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>
This is a core function in umx: Adding labels to paths opens the door to <code><a href="#topic+umxEquate">umxEquate()</a></code>, as well as <code><a href="OpenMx.html#topic+omxSetParameters">omxSetParameters()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuLabel(
  obj,
  suffix = "",
  baseName = NA,
  setfree = FALSE,
  drop = 0,
  labelFixedCells = TRUE,
  jiggle = NA,
  boundDiag = NA,
  verbose = FALSE,
  overRideExisting = FALSE,
  name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuLabel_+3A_obj">obj</code></td>
<td>
<p>An <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> (RAM or matrix based), <code><a href="OpenMx.html#topic+mxPath">mxPath()</a></code>, or <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code></p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_suffix">suffix</code></td>
<td>
<p>String to append to each label (might be used to distinguish, say male and female submodels in a model)</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_basename">baseName</code></td>
<td>
<p>String to prepend to labels. Defaults to NA (&quot;&quot;)</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_setfree">setfree</code></td>
<td>
<p>Whether to label only the free paths (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_drop">drop</code></td>
<td>
<p>The value to fix &quot;drop&quot; paths to (defaults to 0)</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_labelfixedcells">labelFixedCells</code></td>
<td>
<p>= TRUE</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_jiggle">jiggle</code></td>
<td>
<p>How much to jiggle values in a matrix or list of path values</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_bounddiag">boundDiag</code></td>
<td>
<p>Whether to bound the diagonal of a matrix</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_verbose">verbose</code></td>
<td>
<p>How much feedback to give the user (default = FALSE)</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_overrideexisting">overRideExisting</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="xmuLabel_+3A_name">name</code></td>
<td>
<p>Optional new name if given a model. Default (NULL) does not rename model.</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuValues">xmuValues</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# ==============================================================
# = Show how OpenMx models are not labeled, and then add labels =
# ==============================================================
require(umx)
data(demoOneFactor)
latents  = c("G")
manifests = names(demoOneFactor)
m1 = mxModel("One Factor", type = "RAM", 
	manifestVars = manifests, latentVars = latents, 
	mxPath(from = latents  , to = manifests),
	mxPath(from = manifests, arrows = 2),
	mxPath(from = latents  , arrows = 2, free = FALSE, values = 1.0),
	mxData(cov(demoOneFactor), type = "cov", numObs=500)
)

umxGetParameters(m1) # Default "matrix address" labels, i.e "One Factor.S[2,2]"
m1 = xmuLabel(m1)
umxGetParameters(m1, free = TRUE) # Informative labels: "G_to_x1", "x4_with_x4", etc.

# =======================================================================
# = Create a new model, with suffixes added to paths, and model renamed =
# =======================================================================
m2 = xmuLabel(m1, suffix= "_male", overRideExisting= TRUE, name = "male")
umxGetParameters(m2, free = TRUE) # suffixes added

# =============================
# = Example Labeling a matrix =
# =============================
a = xmuLabel(mxMatrix(name = "a", "Full", 3, 3, values = 1:9))
a$labels
a = xmuLabel(mxMatrix(name = "a", "Full", 3, 3, values = 1:9), baseName="bob")
a$labels
# note: labels with "data." in the name are left untouched!
a = mxMatrix(name = "a", "Full", 1,3, labels = c("data.a", "test", NA))
a$labels
xmuLabel(a, verbose = TRUE)
xmuLabel(a, verbose = TRUE, overRideExisting = FALSE)
xmuLabel(a, verbose = TRUE, overRideExisting = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmuLabel_Matrix'>xmuLabel_Matrix (not a user function)</h2><span id='topic+xmuLabel_Matrix'></span>

<h3>Description</h3>

<p>This function will label all the free parameters in an <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuLabel_Matrix(
  mx_matrix = NA,
  baseName = NA,
  setfree = FALSE,
  drop = 0,
  jiggle = NA,
  boundDiag = NA,
  suffix = "",
  verbose = TRUE,
  labelFixedCells = FALSE,
  overRideExisting = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuLabel_Matrix_+3A_mx_matrix">mx_matrix</code></td>
<td>
<p>an mxMatrix</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_basename">baseName</code></td>
<td>
<p>A base name for the labels NA</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_setfree">setfree</code></td>
<td>
<p>Whether to set free cells FALSE</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_drop">drop</code></td>
<td>
<p>What values to drop 0</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_jiggle">jiggle</code></td>
<td>
<p>= whether to jiggle start values</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_bounddiag">boundDiag</code></td>
<td>
<p>set diagonal element lbounds to this numeric value (default = NA = ignore)</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_suffix">suffix</code></td>
<td>
<p>a string to append to each label</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_verbose">verbose</code></td>
<td>
<p>how much feedback to give</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_labelfixedcells">labelFixedCells</code></td>
<td>
<p>= FALSE</p>
</td></tr>
<tr><td><code id="xmuLabel_Matrix_+3A_overrideexisting">overRideExisting</code></td>
<td>
<p>Whether to overRideExisting (Default FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Model developers should just call <code><a href="#topic+xmuLabel">xmuLabel()</a></code>
</p>
<p>Purpose: label the cells of an mxMatrix
Detail: Defaults to the handy &quot;name_r1c1&quot; where name is the matrix name, and r1c1 = row 1 col 1.
Use case: You should not use this: call xmuLabel
umx:::xmuLabel_Matrix(mxMatrix(&quot;Lower&quot;, 3, 3, values = 1, name = &quot;a&quot;, byrow = TRUE), jiggle = .05, boundDiag = NA);
umx:::xmuLabel_Matrix(mxMatrix(&quot;Full&quot; , 3, 3, values = 1, name = &quot;a&quot;, byrow = TRUE));
umx:::xmuLabel_Matrix(mxMatrix(&quot;Symm&quot; , 3, 3, values = 1, name = &quot;a&quot;, byrow = TRUE), jiggle = .05, boundDiag = NA);
umx:::xmuLabel_Matrix(mxMatrix(&quot;Full&quot; , 1, 1, values = 1, name = &quot;a&quot;, labels= &quot;data.a&quot;));
umx:::xmuLabel_Matrix(mxMatrix(&quot;Full&quot; , 1, 1, values = 1, name = &quot;a&quot;, labels= &quot;data.a&quot;), overRideExisting = TRUE);
umx:::xmuLabel_Matrix(mxMatrix(&quot;Full&quot; , 1, 1, values = 1, name = &quot;a&quot;, labels= &quot;test&quot;), overRideExisting = TRUE);
See also: fit2 = omxSetParameters(fit1, labels = &quot;a_r1c1&quot;, free = FALSE, value = 0, name = &quot;drop_a_row1_c1&quot;)
</p>


<h3>Value</h3>


<ul>
<li><p> The labeled <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmuLabel_MATRIX_Model'>xmuLabel_MATRIX_Model (not a user function)</h2><span id='topic+xmuLabel_MATRIX_Model'></span>

<h3>Description</h3>

<p>This function will label all the free parameters in a (non-RAM) OpenMx <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
nb: We don't assume what each matrix is for. Instead, the function just sticks labels like &quot;a_r1c1&quot; into each cell
i.e., matrix-name + _ + r + rowNumber + c + colNumber
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuLabel_MATRIX_Model(model, suffix = "", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuLabel_MATRIX_Model_+3A_model">model</code></td>
<td>
<p>a matrix-style mxModel to label</p>
</td></tr>
<tr><td><code id="xmuLabel_MATRIX_Model_+3A_suffix">suffix</code></td>
<td>
<p>a string to append to each label</p>
</td></tr>
<tr><td><code id="xmuLabel_MATRIX_Model_+3A_verbose">verbose</code></td>
<td>
<p>how much feedback to give</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Model developers should just call <code><a href="#topic+xmuLabel">xmuLabel()</a></code>
</p>


<h3>Value</h3>


<ul>
<li><p> The labeled <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(demoOneFactor)
m2 &lt;- mxModel("label_ex",
	mxMatrix("Full", 5, 1, values = 0.2, free = TRUE, name = "A"), 
	mxMatrix("Symm", 1, 1, values = 1.0, free = FALSE, name = "L"), 
	mxMatrix("Diag", 5, 5, values = 1.0, free = TRUE, name = "U"), 
	mxAlgebra(A %*% L %*% t(A) + U, name = "R"), 
	mxExpectationNormal("R", dimnames = names(demoOneFactor)),
	mxFitFunctionML(),
	mxData(cov(demoOneFactor), type = "cov", numObs=500)
)
m3 = umx:::xmuLabel_MATRIX_Model(m2)
m4 = umx:::xmuLabel_MATRIX_Model(m2, suffix = "male")
# explore these with omxGetParameters(m4)
</code></pre>

<hr>
<h2 id='xmuLabel_RAM_Model'>xmuLabel_RAM_Model (not a user function)</h2><span id='topic+xmuLabel_RAM_Model'></span>

<h3>Description</h3>

<p>This function will label all the free parameters in a RAM <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuLabel_RAM_Model(
  model,
  suffix = "",
  labelFixedCells = TRUE,
  overRideExisting = FALSE,
  verbose = FALSE,
  name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuLabel_RAM_Model_+3A_model">model</code></td>
<td>
<p>a RAM mxModel to label</p>
</td></tr>
<tr><td><code id="xmuLabel_RAM_Model_+3A_suffix">suffix</code></td>
<td>
<p>a string to append to each label</p>
</td></tr>
<tr><td><code id="xmuLabel_RAM_Model_+3A_labelfixedcells">labelFixedCells</code></td>
<td>
<p>Whether to labelFixedCells (Default TRUE)</p>
</td></tr>
<tr><td><code id="xmuLabel_RAM_Model_+3A_overrideexisting">overRideExisting</code></td>
<td>
<p>Whether to overRideExisting (Default FALSE)</p>
</td></tr>
<tr><td><code id="xmuLabel_RAM_Model_+3A_verbose">verbose</code></td>
<td>
<p>how much feedback to give</p>
</td></tr>
<tr><td><code id="xmuLabel_RAM_Model_+3A_name">name</code></td>
<td>
<p>Add optional name parameter to rename returned model (default = leave it along)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Model developers should just call <code><a href="#topic+xmuLabel">xmuLabel()</a></code>
</p>


<h3>Value</h3>


<ul>
<li><p> The labeled <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx); data(demoOneFactor)
# raw but no means
m1 &lt;- mxModel("label_ex", mxData(demoOneFactor, type = "raw"), type="RAM",
	manifestVars = "x1", latentVars= "G",
	umxPath("G", to = "x1"),
	umxPath(var = "x1"),
	umxPath(var = "G", fixedAt = 1)
)
xmuLabel_RAM_Model(m1)

</code></pre>

<hr>
<h2 id='xmuMakeDeviationThresholdsMatrices'>Make a deviation-based mxRAMObjective for ordinal models.</h2><span id='topic+xmuMakeDeviationThresholdsMatrices'></span>

<h3>Description</h3>

<p>Purpose: return a mxRAMObjective(A = &quot;A&quot;, S = &quot;S&quot;, F = &quot;F&quot;, M = &quot;M&quot;, thresholds = &quot;thresh&quot;), mxData(df, type = &quot;raw&quot;)
use-case see: umxMakeThresholdMatrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuMakeDeviationThresholdsMatrices(df, droplevels, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuMakeDeviationThresholdsMatrices_+3A_df">df</code></td>
<td>
<p>a dataframe</p>
</td></tr>
<tr><td><code id="xmuMakeDeviationThresholdsMatrices_+3A_droplevels">droplevels</code></td>
<td>
<p>whether to droplevels or not</p>
</td></tr>
<tr><td><code id="xmuMakeDeviationThresholdsMatrices_+3A_verbose">verbose</code></td>
<td>
<p>how verbose to be</p>
</td></tr>
</table>


<h3>Value</h3>

<p>- list of matrices
</p>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmuMakeOneHeadedPathsFromPathList'>xmuMakeOneHeadedPathsFromPathList</h2><span id='topic+xmuMakeOneHeadedPathsFromPathList'></span>

<h3>Description</h3>

<p>Make one-headed paths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuMakeOneHeadedPathsFromPathList(sourceList, destinationList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuMakeOneHeadedPathsFromPathList_+3A_sourcelist">sourceList</code></td>
<td>
<p>A sourceList</p>
</td></tr>
<tr><td><code id="xmuMakeOneHeadedPathsFromPathList_+3A_destinationlist">destinationList</code></td>
<td>
<p>A destinationList</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> added items
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmuMakeTwoHeadedPathsFromPathList'>xmuMakeTwoHeadedPathsFromPathList</h2><span id='topic+xmuMakeTwoHeadedPathsFromPathList'></span>

<h3>Description</h3>

<p>Make two-headed paths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuMakeTwoHeadedPathsFromPathList(pathList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuMakeTwoHeadedPathsFromPathList_+3A_pathlist">pathList</code></td>
<td>
<p>A list of paths</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> added items
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmuMaxLevels'>xmuMaxLevels</h2><span id='topic+xmuMaxLevels'></span>

<h3>Description</h3>

<p>Get the max levels from df
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuMaxLevels(df, what = c("value", "name"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuMaxLevels_+3A_df">df</code></td>
<td>
<p>Dataframe to search through</p>
</td></tr>
<tr><td><code id="xmuMaxLevels_+3A_what">what</code></td>
<td>
<p>Either &quot;value&quot; or &quot;name&quot; ( of the max-level column)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> max number of levels in frame
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmuMaxLevels(mtcars) # NA = no ordinal vars
xmuMaxLevels(umxFactor(mtcars))
xmuMaxLevels(umxFactor(mtcars), what = "name")
</code></pre>

<hr>
<h2 id='xmuMI'>xmuMI (not for end users)</h2><span id='topic+xmuMI'></span>

<h3>Description</h3>

<p>A function to compute and report modifications which would improve fit.
You will probably use <code><a href="#topic+umxMI">umxMI()</a></code> instead
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuMI(model, vector = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuMI_+3A_model">model</code></td>
<td>
<p>an <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> to derive modification indices for</p>
</td></tr>
<tr><td><code id="xmuMI_+3A_vector">vector</code></td>
<td>
<p>= Whether to report the results as a vector default = TRUE</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>

<hr>
<h2 id='xmuMinLevels'>xmuMinLevels</h2><span id='topic+xmuMinLevels'></span>

<h3>Description</h3>

<p>Get the min levels from df
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuMinLevels(df, what = c("value", "name"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuMinLevels_+3A_df">df</code></td>
<td>
<p>Dataframe to search through</p>
</td></tr>
<tr><td><code id="xmuMinLevels_+3A_what">what</code></td>
<td>
<p>Either &quot;value&quot; or &quot;name&quot; (of the min-level column)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> min number of levels in frame
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xmuMinLevels(mtcars) # NA = no ordinal vars
xmuMinLevels(umxFactor(mtcars))
xmuMinLevels(umxFactor(mtcars), what = "name")
</code></pre>

<hr>
<h2 id='xmuPropagateLabels'>xmuPropagateLabels (not a user function)</h2><span id='topic+xmuPropagateLabels'></span>

<h3>Description</h3>

<p>You should be calling <code><a href="#topic+xmuLabel">xmuLabel()</a></code>.
This function is called by xmuLabel_MATRIX_Model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuPropagateLabels(model, suffix = "", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuPropagateLabels_+3A_model">model</code></td>
<td>
<p>a model to label</p>
</td></tr>
<tr><td><code id="xmuPropagateLabels_+3A_suffix">suffix</code></td>
<td>
<p>a string to append to each label</p>
</td></tr>
<tr><td><code id="xmuPropagateLabels_+3A_verbose">verbose</code></td>
<td>
<p>whether to say what is being done</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(umx)
data(demoOneFactor)
latents  = c("G")
manifests = names(demoOneFactor)
m1 = mxModel("propage_example", type = "RAM", 
	manifestVars = manifests, latentVars = latents, 
	mxPath(from = latents  , to = manifests),
	mxPath(from = manifests, arrows = 2),
	mxPath(from = latents  , arrows = 2, free = FALSE, values = 1.0),
	mxData(cov(demoOneFactor), type = "cov", numObs=500)
)

m1 = umx:::xmuPropagateLabels(m1, suffix = "MZ")
</code></pre>

<hr>
<h2 id='xmuRAM2Ordinal'>xmuRAM2Ordinal</h2><span id='topic+xmuRAM2Ordinal'></span>

<h3>Description</h3>

<p>xmuRAM2Ordinal: Convert a RAM model whose data contain ordinal variables to a threshold-based model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuRAM2Ordinal(model, verbose = TRUE, name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuRAM2Ordinal_+3A_model">model</code></td>
<td>
<p>An RAM model to add thresholds too.</p>
</td></tr>
<tr><td><code id="xmuRAM2Ordinal_+3A_verbose">verbose</code></td>
<td>
<p>Tell the user what was added and why (Default = TRUE).</p>
</td></tr>
<tr><td><code id="xmuRAM2Ordinal_+3A_name">name</code></td>
<td>
<p>= A new name for the modified model. Default (NULL) = leave it as is).</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+umxRAM">umxRAM()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData)
# Cut to form category of 20% obese subjects
obesityLevels   = c('normal', 'obese')
cutPoints       = quantile(twinData[, "bmi1"], probs = .2, na.rm = TRUE)
twinData$obese1 = cut(twinData$bmi1, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
twinData$obese2 = cut(twinData$bmi2, breaks = c(-Inf, cutPoints, Inf), labels = obesityLevels) 
ordDVs = c("obese1", "obese2")
twinData[, ordDVs] = umxFactor(twinData[, ordDVs])
mzData = twinData[twinData$zygosity %in% "MZFF",]
m1 = umxRAM("tim", data = mzData,
	umxPath("bmi1", with = "bmi2"),
	umxPath(v.m.= c("bmi1", "bmi2"))
)

m1 = umxRAM("tim", data = mzData,
	umxPath("obese1", with = "obese2"),
	umxPath(v.m.= c("obese1", "obese2"))
)

## End(Not run)
</code></pre>

<hr>
<h2 id='xmuTwinSuper_Continuous'>Create core of twin model for all-continuous data.</h2><span id='topic+xmuTwinSuper_Continuous'></span>

<h3>Description</h3>

<p>Sets up top, MZ and DZ submodels with a means model, data, and expectation for all-continuous data.
called by <code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuTwinSuper_Continuous(
  name = NULL,
  fullVars,
  fullCovs = NULL,
  sep,
  mzData,
  dzData,
  equateMeans,
  type,
  allContinuousMethod,
  nSib
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_name">name</code></td>
<td>
<p>The name of the supermodel</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_fullvars">fullVars</code></td>
<td>
<p>Full Variable names (wt_T1)</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_fullcovs">fullCovs</code></td>
<td>
<p>Full Covariate names (age_T1)</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_sep">sep</code></td>
<td>
<p>default &quot;_T&quot;</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_mzdata">mzData</code></td>
<td>
<p>An mxData object containing the MZ data</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_dzdata">dzData</code></td>
<td>
<p>An mxData object containing the DZ data</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_equatemeans">equateMeans</code></td>
<td>
<p>Whether to equate the means across twins (default TRUE)</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_type">type</code></td>
<td>
<p>type</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_allcontinuousmethod">allContinuousMethod</code></td>
<td>
<p>allContinuousMethod</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_Continuous_+3A_nsib">nSib</code></td>
<td>
<p>nSib</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> A twin model
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 xmuTwinSuper_Continuous(name="twin_super", selVars = selVars, selCovs = selCovs, 
    mzData = mzData, dzData = dzData, equateMeans = TRUE, type = type, 
    allContinuousMethod = allContinuousMethod, nSib= nSib, sep = "_T" )

## End(Not run)
</code></pre>

<hr>
<h2 id='xmuTwinSuper_NoBinary'>xmuTwinSuper_NoBinary</h2><span id='topic+xmuTwinSuper_NoBinary'></span>

<h3>Description</h3>

<p>xmuTwinSuper_NoBinary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuTwinSuper_NoBinary(
  name = NULL,
  fullVars,
  fullCovs = NULL,
  mzData,
  dzData,
  sep,
  nSib,
  equateMeans = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_name">name</code></td>
<td>
<p>= NULL</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_fullvars">fullVars</code></td>
<td>
<p>full names of variables</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_fullcovs">fullCovs</code></td>
<td>
<p>full names of covariates</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_mzdata">mzData</code></td>
<td>
<p>mzData</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_dzdata">dzData</code></td>
<td>
<p>dzData</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_sep">sep</code></td>
<td>
<p>sep</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_nsib">nSib</code></td>
<td>
<p>nSib</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_equatemeans">equateMeans</code></td>
<td>
<p>T/F</p>
</td></tr>
<tr><td><code id="xmuTwinSuper_NoBinary_+3A_verbose">verbose</code></td>
<td>
<p>(Default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>


<ul>
<li><p> twin model
</p>
</li></ul>



<h3>Handle 1 or more ordinal variables (no binary)</h3>

<p>Means ordinal, but no binary
Means: all free, start cont at the measured value, ordinals @0
</p>
<p>Notes: Ordinal requires:
</p>

<ol>
<li><p> Variable set to mxFactor
</p>
</li>
<li><p> For Binary variables:
</p>
</li>
<li><p> Latent means of binary variables fixedAt 0 (or by data.def?)
</p>
</li>
<li><p> Latent variance (A + C + E) constrained == 1
</p>
</li>
<li><p> For Ordinal variables, first 2 thresholds fixed
</p>
</li></ol>



<h3>See Also</h3>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinUpgradeMeansToCovariateModel">xmuTwinUpgradeMeansToCovariateModel</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
</code></pre>

<hr>
<h2 id='xmuTwinUpgradeMeansToCovariateModel'>Not for end-users: Add a means model with covariates to a twin model</h2><span id='topic+xmuTwinUpgradeMeansToCovariateModel'></span>

<h3>Description</h3>

<p>Does the following to <code>model</code> (i.e., a umx top/MZ/DZ supermodel):
</p>

<ol>
<li><p> Change <code>top.expMeans</code> to <code>top.intercept</code>.
</p>
</li>
<li><p> Create <code>top.meansBetas</code> for beta weights in rows (of covariates) and columns for each variable.
</p>
</li>
<li><p> Add matrices for each twin's data.cov vars (matrixes are called <code>T1DefVars</code>).
</p>
</li>
<li><p> Switch <code>mxExpectationNormal</code> in each data group to point to the local <code>expMean</code>.
</p>
</li>
<li><p> Add &quot;expMean&quot; algebra to each data group.
</p>
</li></ol>


<ul>
<li> <p><code>grp.expMean</code> sums <code>top.intercept</code>  and <code>grp.DefVars %*% top.meansBetas</code> for each twin.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>xmuTwinUpgradeMeansToCovariateModel(model, fullVars, fullCovs, nSib, sep)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuTwinUpgradeMeansToCovariateModel_+3A_model">model</code></td>
<td>
<p>The <code><a href="#topic+umxSuperModel">umxSuperModel()</a></code> we are modifying (must have <code>MZ</code> <code>DZ</code> and <code>top</code> submodels)</p>
</td></tr>
<tr><td><code id="xmuTwinUpgradeMeansToCovariateModel_+3A_fullvars">fullVars</code></td>
<td>
<p>the FULL names of manifest variables</p>
</td></tr>
<tr><td><code id="xmuTwinUpgradeMeansToCovariateModel_+3A_fullcovs">fullCovs</code></td>
<td>
<p>the FULL names of definition variables</p>
</td></tr>
<tr><td><code id="xmuTwinUpgradeMeansToCovariateModel_+3A_nsib">nSib</code></td>
<td>
<p>How many siblings</p>
</td></tr>
<tr><td><code id="xmuTwinUpgradeMeansToCovariateModel_+3A_sep">sep</code></td>
<td>
<p>How twin variable names have been expanded, e.g. &quot;_T&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In umx models with no covariates, means live in <code>top$expMean</code>
</p>


<h3>Value</h3>


<ul>
<li><p> model, now with means model extended to covariates.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li><p> called by <code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous()</a></code>
</p>
</li></ul>

<p>Other xmu internal not for end user: 
<code><a href="#topic+umxModel">umxModel</a>()</code>,
<code><a href="#topic+umxRenameMatrix">umxRenameMatrix</a>()</code>,
<code><a href="#topic+umx_APA_pval">umx_APA_pval</a>()</code>,
<code><a href="#topic+umx_fun_mean_sd">umx_fun_mean_sd</a>()</code>,
<code><a href="#topic+umx_get_bracket_addresses">umx_get_bracket_addresses</a>()</code>,
<code><a href="#topic+umx_make">umx_make</a>()</code>,
<code><a href="#topic+umx_standardize">umx_standardize</a>()</code>,
<code><a href="#topic+umx_string_to_algebra">umx_string_to_algebra</a>()</code>,
<code><a href="#topic+xmuHasSquareBrackets">xmuHasSquareBrackets</a>()</code>,
<code><a href="#topic+xmuLabel_MATRIX_Model">xmuLabel_MATRIX_Model</a>()</code>,
<code><a href="#topic+xmuLabel_Matrix">xmuLabel_Matrix</a>()</code>,
<code><a href="#topic+xmuLabel_RAM_Model">xmuLabel_RAM_Model</a>()</code>,
<code><a href="#topic+xmuMI">xmuMI</a>()</code>,
<code><a href="#topic+xmuMakeDeviationThresholdsMatrices">xmuMakeDeviationThresholdsMatrices</a>()</code>,
<code><a href="#topic+xmuMakeOneHeadedPathsFromPathList">xmuMakeOneHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMakeTwoHeadedPathsFromPathList">xmuMakeTwoHeadedPathsFromPathList</a>()</code>,
<code><a href="#topic+xmuMaxLevels">xmuMaxLevels</a>()</code>,
<code><a href="#topic+xmuMinLevels">xmuMinLevels</a>()</code>,
<code><a href="#topic+xmuPropagateLabels">xmuPropagateLabels</a>()</code>,
<code><a href="#topic+xmuRAM2Ordinal">xmuRAM2Ordinal</a>()</code>,
<code><a href="#topic+xmuTwinSuper_Continuous">xmuTwinSuper_Continuous</a>()</code>,
<code><a href="#topic+xmuTwinSuper_NoBinary">xmuTwinSuper_NoBinary</a>()</code>,
<code><a href="#topic+xmu_CI_merge">xmu_CI_merge</a>()</code>,
<code><a href="#topic+xmu_CI_stash">xmu_CI_stash</a>()</code>,
<code><a href="#topic+xmu_DF_to_mxData_TypeCov">xmu_DF_to_mxData_TypeCov</a>()</code>,
<code><a href="#topic+xmu_PadAndPruneForDefVars">xmu_PadAndPruneForDefVars</a>()</code>,
<code><a href="#topic+xmu_bracket_address2rclabel">xmu_bracket_address2rclabel</a>()</code>,
<code><a href="#topic+xmu_cell_is_on">xmu_cell_is_on</a>()</code>,
<code><a href="#topic+xmu_check_levels_identical">xmu_check_levels_identical</a>()</code>,
<code><a href="#topic+xmu_check_needs_means">xmu_check_needs_means</a>()</code>,
<code><a href="#topic+xmu_check_variance">xmu_check_variance</a>()</code>,
<code><a href="#topic+xmu_clean_label">xmu_clean_label</a>()</code>,
<code><a href="#topic+xmu_data_missing">xmu_data_missing</a>()</code>,
<code><a href="#topic+xmu_data_swap_a_block">xmu_data_swap_a_block</a>()</code>,
<code><a href="#topic+xmu_describe_data_WLS">xmu_describe_data_WLS</a>()</code>,
<code><a href="#topic+xmu_dot_make_paths">xmu_dot_make_paths</a>()</code>,
<code><a href="#topic+xmu_dot_make_residuals">xmu_dot_make_residuals</a>()</code>,
<code><a href="#topic+xmu_dot_maker">xmu_dot_maker</a>()</code>,
<code><a href="#topic+xmu_dot_move_ranks">xmu_dot_move_ranks</a>()</code>,
<code><a href="#topic+xmu_dot_rank_str">xmu_dot_rank_str</a>()</code>,
<code><a href="#topic+xmu_extract_column">xmu_extract_column</a>()</code>,
<code><a href="#topic+xmu_get_CI">xmu_get_CI</a>()</code>,
<code><a href="#topic+xmu_lavaan_process_group">xmu_lavaan_process_group</a>()</code>,
<code><a href="#topic+xmu_make_TwinSuperModel">xmu_make_TwinSuperModel</a>()</code>,
<code><a href="#topic+xmu_make_bin_cont_pair_data">xmu_make_bin_cont_pair_data</a>()</code>,
<code><a href="#topic+xmu_make_mxData">xmu_make_mxData</a>()</code>,
<code><a href="#topic+xmu_match.arg">xmu_match.arg</a>()</code>,
<code><a href="#topic+xmu_name_from_lavaan_str">xmu_name_from_lavaan_str</a>()</code>,
<code><a href="#topic+xmu_path2twin">xmu_path2twin</a>()</code>,
<code><a href="#topic+xmu_path_regex">xmu_path_regex</a>()</code>,
<code><a href="#topic+xmu_print_algebras">xmu_print_algebras</a>()</code>,
<code><a href="#topic+xmu_rclabel_2_bracket_address">xmu_rclabel_2_bracket_address</a>()</code>,
<code><a href="#topic+xmu_safe_run_summary">xmu_safe_run_summary</a>()</code>,
<code><a href="#topic+xmu_set_sep_from_suffix">xmu_set_sep_from_suffix</a>()</code>,
<code><a href="#topic+xmu_show_fit_or_comparison">xmu_show_fit_or_comparison</a>()</code>,
<code><a href="#topic+xmu_simplex_corner">xmu_simplex_corner</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEcov">xmu_standardize_ACEcov</a>()</code>,
<code><a href="#topic+xmu_standardize_ACEv">xmu_standardize_ACEv</a>()</code>,
<code><a href="#topic+xmu_standardize_ACE">xmu_standardize_ACE</a>()</code>,
<code><a href="#topic+xmu_standardize_CP">xmu_standardize_CP</a>()</code>,
<code><a href="#topic+xmu_standardize_IP">xmu_standardize_IP</a>()</code>,
<code><a href="#topic+xmu_standardize_RAM">xmu_standardize_RAM</a>()</code>,
<code><a href="#topic+xmu_standardize_SexLim">xmu_standardize_SexLim</a>()</code>,
<code><a href="#topic+xmu_standardize_Simplex">xmu_standardize_Simplex</a>()</code>,
<code><a href="#topic+xmu_start_value_list">xmu_start_value_list</a>()</code>,
<code><a href="#topic+xmu_starts">xmu_starts</a>()</code>,
<code><a href="#topic+xmu_summary_RAM_group_parameters">xmu_summary_RAM_group_parameters</a>()</code>,
<code><a href="#topic+xmu_twin_add_WeightMatrices">xmu_twin_add_WeightMatrices</a>()</code>,
<code><a href="#topic+xmu_twin_check">xmu_twin_check</a>()</code>,
<code><a href="#topic+xmu_twin_get_var_names">xmu_twin_get_var_names</a>()</code>,
<code><a href="#topic+xmu_twin_make_def_means_mats_and_alg">xmu_twin_make_def_means_mats_and_alg</a>()</code>,
<code><a href="#topic+xmu_twin_upgrade_selDvs2SelVars">xmu_twin_upgrade_selDvs2SelVars</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(twinData) # ?twinData from Australian twins.
twinData[, c("ht1", "ht2")] = twinData[, c("ht1", "ht2")] * 10
mzData = twinData[twinData$zygosity %in% "MZFF", ]
dzData = twinData[twinData$zygosity %in% "DZFF", ]
# m1 = umxACE(selDVs= "ht", sep= "", dzData= dzData, mzData= mzData, autoRun= FALSE)
# m2 = xmuTwinUpgradeMeansToCovariateModel(m1, fullVars = c("ht1", "ht2"),
# 	fullCovs = c("age1", "sex1", "age2", "sex2"), sep = "")


## End(Not run)

</code></pre>

<hr>
<h2 id='xmuValues'>xmuValues: Set values in RAM model, matrix, or path</h2><span id='topic+xmuValues'></span>

<h3>Description</h3>

<p>For models to be estimated, it is essential that path values start at credible values.
<code>xmuValues</code> takes on that task for you.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xmuValues(obj = NA, sd = NA, n = 1, onlyTouchZeros = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xmuValues_+3A_obj">obj</code></td>
<td>
<p>The RAM or matrix <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>, or <code><a href="OpenMx.html#topic+mxMatrix">mxMatrix()</a></code> that you want to set start values for.</p>
</td></tr>
<tr><td><code id="xmuValues_+3A_sd">sd</code></td>
<td>
<p>Optional Standard Deviation for start values</p>
</td></tr>
<tr><td><code id="xmuValues_+3A_n">n</code></td>
<td>
<p>Optional Mean for start values</p>
</td></tr>
<tr><td><code id="xmuValues_+3A_onlytouchzeros">onlyTouchZeros</code></td>
<td>
<p>Don't alter parameters that have starts (useful to speed <code><a href="#topic+umxModify">umxModify()</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>xmuValues can set start values for the free parameters in both RAM and Matrix <code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code>s.
It can also take an mxMatrix as input.
It tries to be smart in guessing starts from the values in your data and the model type.
</p>
<p><em>note</em>: If you give xmuValues a numeric input, it will use obj as the mean, and return a
list of length n, with sd = sd.
</p>


<h3>Value</h3>


<ul>
<li> <p><code><a href="OpenMx.html#topic+mxModel">mxModel()</a></code> with updated start values
</p>
</li></ul>



<h3>References</h3>


<ul>
<li> <p><a href="https://github.com/tbates/umx">https://github.com/tbates/umx</a>, <a href="https://tbates.github.io">https://tbates.github.io</a>
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li><p> Core functions:
</p>
</li></ul>

<p>Other Advanced Model Building Functions: 
<code><a href="#topic+umxAlgebra">umxAlgebra</a>()</code>,
<code><a href="#topic+umxFixAll">umxFixAll</a>()</code>,
<code><a href="#topic+umxJiggle">umxJiggle</a>()</code>,
<code><a href="#topic+umxRun">umxRun</a>()</code>,
<code><a href="#topic+umxThresholdMatrix">umxThresholdMatrix</a>()</code>,
<code><a href="#topic+umxUnexplainedCausalNexus">umxUnexplainedCausalNexus</a>()</code>,
<code><a href="#topic+umx">umx</a></code>,
<code><a href="#topic+xmuLabel">xmuLabel</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
require(umx)
data(demoOneFactor)
latents = c("G")
manifests = names(demoOneFactor)

# ====================================================================
# = Make an OpenMx model (which will lack start values and labels..) =
# ====================================================================
m1 = mxModel("One Factor", type = "RAM", 
	manifestVars = manifests, latentVars = latents, 
	mxPath(from = latents  , to = manifests),
	mxPath(from = manifests, arrows = 2),
	mxPath(from = latents  , arrows = 2, free = FALSE, values = 1.0),
	mxData(cov(demoOneFactor), type = "cov", numObs=500)
)
mxEval(S, m1) # default variances are jiggled away from near-zero
# Add start values to the model
m1 = xmuValues(m1)
mxEval(S, m1) # plausible variances
umx_print(mxEval(S,m1), 3, zero.print = ".") # plausible variances
xmuValues(14, sd = 1, n = 10) # Return vector of length 10, with mean 14 and sd 1


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
