<!DOCTYPE html><html><head><title>Help for package Unico</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Unico}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#association_asymptotic'><p>Performs asymptotic statistical testing under no distribution assumption</p></a></li>
<li><a href='#association_parametric'><p>Performs parametric statistical testing</p></a></li>
<li><a href='#simulate_data'><p>Simulate data under Unico model assumption</p></a></li>
<li><a href='#tensor'><p>Inferring the underlying source-specific 3D tensor</p></a></li>
<li><a href='#Unico'><p>Fitting the Unico model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Unified Cross-Omics Deconvolution</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>UNIfied Cross-Omics deconvolution (Unico) deconvolves
    standard 2-dimensional bulk matrices of samples by features into a
    3-dimensional tensors representing samples by features by cell types.
    Unico stands out as the first principled model-based deconvolution
    method that is theoretically justified for any heterogeneous genomic
    data. For more details see Chen and Rahmani et al. (2024)
    &lt;<a href="https://doi.org/10.1101%2F2024.01.27.577588">doi:10.1101/2024.01.27.577588</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://cozygene.github.io/Unico/">https://cozygene.github.io/Unico/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cozygene/Unico/issues">https://github.com/cozygene/Unico/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>compositions, config, data.table, futile.logger, MASS,
matrixcalc, matrixStats, mgcv, nloptr, parallel, pbapply,
pracma, stats, testit, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>egg, hexbin, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-23 18:25:07 UTC; johnsonchen</td>
</tr>
<tr>
<td>Author:</td>
<td>Zeyuan Chen [aut, cre],
  Elior Rahmani [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zeyuan Chen &lt;johnsonchen@cs.ucla.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-26 16:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='association_asymptotic'>Performs asymptotic statistical testing under no distribution assumption</h2><span id='topic+association_asymptotic'></span>

<h3>Description</h3>

<p>Performs asymptotic statistical testing on (1) the marginal effect of each covariate in <code>C1</code> at source-specific level (2) non-source-specific effect for each covariate in <code>C2</code>. In the context of bulk genomic data containing a mixture of cell types, these correspond to the marginal effect of each covariate in <code>C1</code> (potentially including the phenotype of interest) at each cell type and tissue-level effect for each covariate in <code>C2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association_asymptotic(
  X,
  Unico.mdl,
  slot_name = "asymptotic",
  diag_only = FALSE,
  intercept = TRUE,
  X_max_stds = 2,
  Q_max_stds = Inf,
  V_min_qlt = 0.05,
  parallel = TRUE,
  num_cores = NULL,
  log_file = "Unico.log",
  verbose = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association_asymptotic_+3A_x">X</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of measurements of <code>m</code> features for <code>n</code> observations. Each column in <code>X</code> is assumed to be a mixture of <code>k</code> sources. Note that <code>X</code> must include row names and column names and that NA values are currently not supported. <code>X</code> should not include features that are constant across all observations. Note that <code>X</code> must be the same <code>X</code> used to learn <code>Unico.mdl</code> (i.e. the original observed 2D mixture used to fit the model).</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_unico.mdl">Unico.mdl</code></td>
<td>
<p>The entire set of model parameters estimated by Unico on the 2D mixture matrix (i.e. the list returned by applying function <code>Unico</code> to <code>X</code>).</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_slot_name">slot_name</code></td>
<td>
<p>A string indicating the key for storing the results under <code>Unico.mdl</code></p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_diag_only">diag_only</code></td>
<td>
<p>A logical value indicating whether to only use the estimated source-level variances (and thus ignoring the estimate covariance) for controlling the heterogeneity in the observed mixture. if set to FALSE, Unico instead estimates the observation- and feature-specific variance in the mixture by leveraging the entire <code>k</code> by <code>k</code> variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_intercept">intercept</code></td>
<td>
<p>A logical value indicating whether to fit the intercept term when performing the statistical testing.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_x_max_stds">X_max_stds</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers due to the observed mixture value. Only samples whose observed mixture value fall within <code>X_max_stds</code> standard deviations from the mean will be used for the statistical testing of a given feature.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_q_max_stds">Q_max_stds</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers due to the estimated mixture variance. Only samples whose estimated mixture variance fall within <code>Q_max_stds</code> standard deviations from the mean will be used for the statistical testing of a given feature.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_v_min_qlt">V_min_qlt</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers due to the estimated moment condition variance. This value should be between 0 and 1. Only samples whose estimated moment condition variance fall outside the bottom <code>V_min_qlt</code> quantile will be used for the statistical testing of a given feature.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_parallel">parallel</code></td>
<td>
<p>A logical value indicating whether to use parallel computing (possible when using a multi-core machine).</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_num_cores">num_cores</code></td>
<td>
<p>A numeric value indicating the number of cores to use (activated only if <code>parallel == TRUE</code>). If <code>num_cores == NULL</code> then all available cores except for one will be used.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_log_file">log_file</code></td>
<td>
<p>A path to an output log file. Note that if the file <code>log_file</code> already exists then logs will be appended to the end of the file. Set <code>log_file</code> to <code>NULL</code> to prevent output from being saved into a file; note that if <code>verbose == FALSE</code> then no output file will be generated regardless of the value of <code>log_file</code>.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print logs.</p>
</td></tr>
<tr><td><code id="association_asymptotic_+3A_debug">debug</code></td>
<td>
<p>A logical value indicating whether to set the logger to a more detailed debug level; set <code>debug</code> to <code>TRUE</code> before reporting issues.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Under no distribution assumption, we can solve for the following weighted least square problem, which is similar to the heteroskedastic regression view described in <a href="#topic+association_parametric">association_parametric</a>.
</p>
<p style="text-align: center;"><code class="reqn">\hat{\phi_j}^{\text{asym}} = \text{argmin}_{\phi_j} (x_j - S\phi_j) ^T Q_j (x_j - S\phi_j)</code>
</p>

<p><code class="reqn">S</code> denotes the design matrix formed by stacking samples in the rows and dependent variables <code class="reqn">\{\{w_i\}, \{w_i c_i^{(1)}\}, \{c_i^{(2)}\}\}</code> on the columns.
<code class="reqn">\phi_j</code> denotes the corresponding effect sizes on the dependent variables.
<code class="reqn">Q_j</code> denotes the feature-specific weighting scheme. Similar to the parametric counterpart, <code class="reqn">Q_j=\text{diag}(q_{1j}^2,...,q_{nj}^2)</code>, where for each sample <code class="reqn">i</code>, its corresponding weight will be the inverse of the estimated variance in the mixture: <code class="reqn">q_{ij}^2 = \frac{1}{sum(w_i w_i^T \odot \hat{\Sigma}_j)}</code>.
Marginal testing can thus be carried out on each dependent variable via the asymptotic distribution of the estimator <code class="reqn">\hat{\phi_j}^{\text{asym}}</code>.
</p>


<h3>Value</h3>

<p>An updated <code>Unico.mdl</code> object with the the following list of effect size and p-value estimates stored in an additional key specified by <code>slot_name</code>
</p>
<table>
<tr><td><code>gammas_hat</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of the estimated effects of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code>, where the first <code>p1</code> columns are the source-specific effects of the <code>p1</code> covariates on the first source, the following <code>p1</code> columns are the source-specific effects on the second source and so on.</p>
</td></tr>
<tr><td><code>betas_hat</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of the estimated effects of the <code>p2</code> covariates in <code>C2</code> on the mixture values of each of the <code>m</code> features in <code>X</code>.</p>
</td></tr>
<tr><td><code>gammas_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of p-values for the estimates in <code>gammas_hat</code> (based on a T-test).</p>
</td></tr>
<tr><td><code>betas_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of p-values for the estimates in <code>betas_hat</code> (based on a T-test).</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of weights used for controlling the heterogeneity of each observation at each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>masks</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of logical values indicating whether observation participated in statistical testing at each feature  (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>fphi_hat</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix containing the entire estimated moment condition variance for each feature. Note that observations who are considered as outliers due to any of the criteria will be marked as -1 in the estimated moment condition variance (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>phi_hat</code></td>
<td>
<p>An <code>m</code> by <code>k+p1*k+p2</code> matrix containing the entire estimated effect sizes (including those on source weights) for each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>phi_se</code></td>
<td>
<p>An <code>m</code> by <code>k+p1*k+p2</code> matrix containing the estimated standard errors associated with <code>phi_hat</code> for each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>phi_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>k+p1*k+p2</code> matrix containing the p-values associated with <code>phi_hat</code> for each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data = simulate_data(n=100, m=2, k=3, p1=1, p2=1, taus_std=0, log_file=NULL)
res = list()
res$params.hat = Unico(data$X, data$W, data$C1, data$C2, parallel=FALSE, log_file=NULL)
res$params.hat = association_asymptotic(data$X, res$params.hat, parallel=FALSE, log_file=NULL)

</code></pre>

<hr>
<h2 id='association_parametric'>Performs parametric statistical testing</h2><span id='topic+association_parametric'></span>

<h3>Description</h3>

<p>Performs parametric statistical testing (T-test) on (1) the marginal effect of each covariate in <code>C1</code> at source-specific level (2) the joint effect across all sources for each covariate in <code>C1</code> (3) non-source-specific effect for each covariate in <code>C2</code>. In the context of bulk genomic data containing a mixture of cell types, these correspond to the marginal effect of each covariate in <code>C1</code> (potentially including the phenotype of interest) at each cell type, joint tissue-level effect for each covariate in <code>C1</code>, and tissue-level effect for each covariate in <code>C2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association_parametric(
  X,
  Unico.mdl,
  slot_name = "parametric",
  diag_only = FALSE,
  intercept = TRUE,
  X_max_stds = 2,
  Q_max_stds = Inf,
  XQ_max_stds = Inf,
  parallel = TRUE,
  num_cores = NULL,
  log_file = "Unico.log",
  verbose = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association_parametric_+3A_x">X</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of measurements of <code>m</code> features for <code>n</code> observations. Each column in <code>X</code> is assumed to be a mixture of <code>k</code> sources. Note that <code>X</code> must include row names and column names and that NA values are currently not supported. <code>X</code> should not include features that are constant across all observations. Note that <code>X</code> must be the same <code>X</code> used to learn <code>Unico.mdl</code> (i.e. the original observed 2D mixture used to fit the model).</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_unico.mdl">Unico.mdl</code></td>
<td>
<p>The entire set of model parameters estimated by Unico on the 2D mixture matrix (i.e. the list returned by applying function <code>Unico</code> to <code>X</code>).</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_slot_name">slot_name</code></td>
<td>
<p>A string indicating the key for storing the results under <code>Unico.mdl</code></p>
</td></tr>
<tr><td><code id="association_parametric_+3A_diag_only">diag_only</code></td>
<td>
<p>A logical value indicating whether to only use the estimated source-level variances (and thus ignoring the estimate covariance) for controlling the heterogeneity in the observed mixture. if set to FALSE, Unico instead estimates the observation- and feature-specific variance in the mixture by leveraging the entire <code>k</code> by <code>k</code> variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_intercept">intercept</code></td>
<td>
<p>A logical value indicating whether to fit the intercept term when performing the statistical testing.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_x_max_stds">X_max_stds</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers due to the observed mixture value. Only samples whose observed mixture value fall within <code>X_max_stds</code> standard deviations from the mean will be used for the statistical testing of a given feature.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_q_max_stds">Q_max_stds</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers due to the estimated mixture variance. Only samples whose estimated mixture variance fall within <code>Q_max_stds</code> standard deviations from the mean will be used for the statistical testing of a given feature.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_xq_max_stds">XQ_max_stds</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers due to the weighted mixture value. Only samples whose weighted mixture value fall within <code>XQ_max_stds</code> standard deviations from the mean will be used for the statistical testing of a given feature.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_parallel">parallel</code></td>
<td>
<p>A logical value indicating whether to use parallel computing (possible when using a multi-core machine).</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_num_cores">num_cores</code></td>
<td>
<p>A numeric value indicating the number of cores to use (activated only if <code>parallel == TRUE</code>). If <code>num_cores == NULL</code> then all available cores except for one will be used.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_log_file">log_file</code></td>
<td>
<p>A path to an output log file. Note that if the file <code>log_file</code> already exists then logs will be appended to the end of the file. Set <code>log_file</code> to <code>NULL</code> to prevent output from being saved into a file; note that if <code>verbose == FALSE</code> then no output file will be generated regardless of the value of <code>log_file</code>.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print logs.</p>
</td></tr>
<tr><td><code id="association_parametric_+3A_debug">debug</code></td>
<td>
<p>A logical value indicating whether to set the logger to a more detailed debug level; set <code>debug</code> to <code>TRUE</code> before reporting issues.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If we assume that source-specific values <code class="reqn">Z_{ijh}</code> are normally distributed, under the Unico model, we have the following:
</p>
<p style="text-align: center;"><code class="reqn">Z_{ij} \sim \mathcal{N}\left(\mu_{j} + (c_i^{(1)})^T \gamma_{jh}, \sigma_{jh}^2 \right)</code>
</p>

<p style="text-align: center;"><code class="reqn">X_{ij} \sim \mathcal{N}\left(w_{i}^T (\mu_{j} + (c_i^{(1)})^T \gamma_{jh}) + (c_i^{(2)})^T \beta_j,  \text{Sum}\left((w_i w_i^T ) \odot \Sigma_j\right) + \tau_j^2\right)</code>
</p>

<p>For a given feature <code class="reqn">j</code> under test, the above equation corresponds to a heteroskedastic regression problem with <code class="reqn">X_{ij}</code> as the dependent variable and <code class="reqn">\{\{w_i\}, \{w_i c_i^{(1)}\}, \{c_i^{(2)}\}\}</code> as the set of independent variables.
This view allows us to perform parametric statistical testing (T-test for marginal effects and partial F-test for joint effects) by solving a generalized least squares problem with sample <code class="reqn">i</code> scaled by the inverse of its estimated standard deviation.
</p>


<h3>Value</h3>

<p>An updated <code>Unico.mdl</code> object with the the following list of effect size and p-value estimates stored in an additional key specified by <code>slot_name</code>
</p>
<table>
<tr><td><code>gammas_hat</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of the estimated effects of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code>, where the first <code>p1</code> columns are the source-specific effects of the <code>p1</code> covariates on the first source, the following <code>p1</code> columns are the source-specific effects on the second source and so on.</p>
</td></tr>
<tr><td><code>betas_hat</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of the estimated effects of the <code>p2</code> covariates in <code>C2</code> on the mixture values of each of the <code>m</code> features in <code>X</code>.</p>
</td></tr>
<tr><td><code>gammas_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of p-values for the estimates in <code>gammas_hat</code> (based on a T-test).</p>
</td></tr>
<tr><td><code>betas_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of p-values for the estimates in <code>betas_hat</code> (based on a T-test).</p>
</td></tr>
<tr><td><code>gammas_hat_pvals.joint</code></td>
<td>
<p>An <code>m</code> by <code>p1</code> matrix of p-values for the joint effects (i.e. across all <code>k</code> sources) of each of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code> (based on a partial F-test). In other words, these are p-values for the combined statistical effects (across all sources) of each one of the <code>p1</code> covariates on each of the <code>m</code> features under the Unico model.</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of weights used for controlling the heterogeneity of each observation at each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>masks</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of logical values indicating whether observation participated in statistical testing at each feature  (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>phi_hat</code></td>
<td>
<p>An <code>m</code> by <code>k+p1*k+p2</code> matrix containing the entire estimated effect sizes (including those on source weights) for each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>phi_se</code></td>
<td>
<p>An <code>m</code> by <code>k+p1*k+p2</code> matrix containing the estimated standard errors associated with <code>phi_hat</code> for each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>phi_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>k+p1*k+p2</code> matrix containing the p-values associated with <code>phi_hat</code> for each feature (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data = simulate_data(n=100, m=2, k=3, p1=1, p2=1, taus_std=0, log_file=NULL)
res = list()
res$params.hat = Unico(data$X, data$W, data$C1, data$C2, parallel=FALSE, log_file=NULL)
res$params.hat = association_parametric(data$X, res$params.hat, parallel=FALSE, log_file=NULL)

</code></pre>

<hr>
<h2 id='simulate_data'>Simulate data under Unico model assumption</h2><span id='topic+simulate_data'></span>

<h3>Description</h3>

<p>Simulate all model parameters and sample source specific data from multivariate gaussian with full covariance structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_data(
  n,
  m,
  k,
  p1,
  p2,
  mus_mean = 10,
  mus_std = 2,
  gammas_mean = 1,
  gammas_std = 0.1,
  betas_mean = 1,
  betas_std = 0.1,
  sigmas_lb = 0,
  sigmas_ub = 1,
  taus_std = 0.1,
  log_file = "Unico.log",
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_data_+3A_n">n</code></td>
<td>
<p>A positive integer indicating the number of observations to simulate.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_m">m</code></td>
<td>
<p>A positive integer indicating the number of features to simulate.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_k">k</code></td>
<td>
<p>A positive integer indicating the number of sources to simulate.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_p1">p1</code></td>
<td>
<p>A non-negative integer indicating the number of source-specific covariates to simulate.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_p2">p2</code></td>
<td>
<p>A non-negative indicating the number of non-source-specific covariates to simulate.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_mus_mean">mus_mean</code></td>
<td>
<p>A numerical value indicating the average of the source specific means.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_mus_std">mus_std</code></td>
<td>
<p>A positive value indicating the variation of the source specific means across difference sources.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_gammas_mean">gammas_mean</code></td>
<td>
<p>A numerical value indicating the average effect sizes of the source-specific covariates.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_gammas_std">gammas_std</code></td>
<td>
<p>A non-negative numerical value indicating the variation of the effect sizes of the source-specific covariates.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_betas_mean">betas_mean</code></td>
<td>
<p>A numerical value indicating the average effect sizes of the non-source-specific covariates.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_betas_std">betas_std</code></td>
<td>
<p>A non-negative numerical value indicating the variation of the effect sizes of the non-source-specific covariates.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_sigmas_lb">sigmas_lb</code></td>
<td>
<p>A numerical value indicating the lower bound of a uniform distribution from which we sample entries of matrix <code>A</code> used to construct the feature specific <code>k</code> by <code>k</code> variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_sigmas_ub">sigmas_ub</code></td>
<td>
<p>A numerical value indicating the upper bound of a uniform distribution from which we sample entries of matrix <code>A</code> used to construct the feature specific <code>k</code> by <code>k</code> variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_taus_std">taus_std</code></td>
<td>
<p>non-negative numerical value indicating the variation of the measurement noise across difference features.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_log_file">log_file</code></td>
<td>
<p>A path to an output log file. Note that if the file <code>log_file</code> already exists then logs will be appended to the end of the file. Set <code>log_file</code> to <code>NULL</code> to prevent output from being saved into a file; note that if <code>verbose == FALSE</code> then no output file will be generated regardless of the value of <code>log_file</code>.</p>
</td></tr>
<tr><td><code id="simulate_data_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print logs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Simulate data based on the generative model described in function <a href="#topic+Unico">Unico</a>.
</p>


<h3>Value</h3>

<p>A list of simulated model parameters, covariates, observed mixture, and source-specific data.
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of the simulated mixture for <code>m</code> features and <code>n</code> observations.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>An <code>n</code> by <code>k</code> matrix of the weights/proportions of <code>k</code> source for each of the <code>n</code> observations.</p>
</td></tr>
<tr><td><code>C1</code></td>
<td>
<p>An <code>n</code> by <code>p1</code> matrix of the simulated covariates that affect the source-specific values.</p>
</td></tr>
<tr><td><code>C2</code></td>
<td>
<p>An <code>n</code> by <code>p2</code> matrix of the simulated covariates that affect the mixture values.</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>A <code>k</code> by <code>m</code> by <code>n</code> tensor of the source specific values for each of the <code>k</code> sources</p>
</td></tr>
<tr><td><code>mus</code></td>
<td>
<p>An <code>m</code> by <code>k</code> matrix of the mean of each of the <code>m</code> features for each of the <code>k</code> sources.</p>
</td></tr>
<tr><td><code>gammas</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of the effect sizes of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code>, where the first <code>p1</code> columns are the source-specific effects of the <code>p1</code> covariates on the first source, the following <code>p1</code> columns are the source-specific effects on the second source and so on.</p>
</td></tr>
<tr><td><code>betas</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of the effect sizes of the <code>p2</code> covariates in <code>C2</code> on the mixture values of each of the <code>m</code> features.</p>
</td></tr>
<tr><td><code>sigmas</code></td>
<td>
<p>An <code>m</code> by <code>k</code> by <code>k</code> tensor of the variance-covariance matrix of each of the <code>m</code> features.</p>
</td></tr>
<tr><td><code>taus</code></td>
<td>
<p>An <code>m</code> by <code>1</code> matrix of the feature specific variance of the measurement noise for all m features.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data = simulate_data(n=100, m=2, k=3, p1=1, p2=1, taus_std=0, log_file=NULL)

</code></pre>

<hr>
<h2 id='tensor'>Inferring the underlying source-specific 3D tensor</h2><span id='topic+tensor'></span>

<h3>Description</h3>

<p>Infers the underlying (sources by features by observations) 3D tensor from the observed (features by observations) 2D mixture, under the assumption of the Unico model that each observation is a mixture of unique source-specific values (in each feature in the data). In the context of bulk genomics containing a mixture of cell types (i.e. the input could be CpG sites by individuals for DNA methylation and genes by individuals for RNA expression), <code>tensor</code> allows to estimate the cell-type-specific levels for each individual in each CpG site/gene (i.e. a tensor of CpG sites/genes by individuals by cell types).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tensor(
  X,
  W,
  C1,
  C2,
  Unico.mdl,
  parallel = TRUE,
  num_cores = NULL,
  log_file = "Unico.log",
  verbose = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tensor_+3A_x">X</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of measurements of <code>m</code> features for <code>n</code> observations. Each column in <code>X</code> is assumed to be a mixture of <code>k</code> sources. Note that <code>X</code> must include row names and column names and that NA values are currently not supported. <code>X</code> should not include features that are constant across all observations. Note that <code>X</code> could potentially be different from the <code>X</code> used to learn <code>Unico.mdl</code> (i.e. the original observed 2D mixture used to fit the model).</p>
</td></tr>
<tr><td><code id="tensor_+3A_w">W</code></td>
<td>
<p>An <code>n</code> by <code>k</code> matrix of weights - the weights of <code>k</code> sources for each of the <code>n</code> mixtures (observations). All the weights must be positive and each row - corresponding to the weights of a single observation - must sum up to 1. Note that <code>W</code> must include row names and column names and that NA values are currently not supported.</p>
</td></tr>
<tr><td><code id="tensor_+3A_c1">C1</code></td>
<td>
<p>An <code>n</code> by <code>p1</code> design matrix of covariates that may affect the hidden source-specific values (possibly a different effect size in each source). Note that <code>C1</code> must include row names and column names and should not include an intercept term. NA values are currently not supported. Note that all covariates in <code>C1</code> must be present and match the order of the set of covariates in <code>C1</code> stored in <code>Unico.mdl</code> (i.e. the original set of source-specific covariates available when initially fitting the model).</p>
</td></tr>
<tr><td><code id="tensor_+3A_c2">C2</code></td>
<td>
<p>An <code>n</code> by <code>p2</code> design matrix of covariates that may affect the mixture (i.e. rather than directly the sources of the mixture; for example, variables that capture biases in the collection of the measurements). Note that <code>C2</code> must include row names and column names and should not include an intercept term. NA values are currently not supported. Note that all covariates in <code>C2</code> must be present and match the order of the set of covariates in <code>C2</code> stored in <code>Unico.mdl</code> (i.e. the original set of not source-specific covariates available when initially fitting the model).</p>
</td></tr>
<tr><td><code id="tensor_+3A_unico.mdl">Unico.mdl</code></td>
<td>
<p>The entire set of model parameters estimated by Unico on the 2D mixture matrix (i.e. the list returned by applying function <code>Unico</code> to <code>X</code>).</p>
</td></tr>
<tr><td><code id="tensor_+3A_parallel">parallel</code></td>
<td>
<p>A logical value indicating whether to use parallel computing (possible when using a multi-core machine).</p>
</td></tr>
<tr><td><code id="tensor_+3A_num_cores">num_cores</code></td>
<td>
<p>A numeric value indicating the number of cores to use (activated only if <code>parallel == TRUE</code>). If <code>num_cores == NULL</code> then all available cores except for one will be used.</p>
</td></tr>
<tr><td><code id="tensor_+3A_log_file">log_file</code></td>
<td>
<p>A path to an output log file. Note that if the file <code>log_file</code> already exists then logs will be appended to the end of the file. Set <code>log_file</code> to <code>NULL</code> to prevent output from being saved into a file; note that if <code>verbose == FALSE</code> then no output file will be generated regardless of the value of <code>log_file</code>.</p>
</td></tr>
<tr><td><code id="tensor_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print logs.</p>
</td></tr>
<tr><td><code id="tensor_+3A_debug">debug</code></td>
<td>
<p>A logical value indicating whether to set the logger to a more detailed debug level; set <code>debug</code> to <code>TRUE</code> before reporting issues.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After obtaining all the estimated parameters in the Unico model (by calling <a href="#topic+Unico">Unico</a>), <code>tensor</code> uses the conditional distribution <code class="reqn">Z_{jh}^i|X_{ij}=x_{ij}</code> for estimating the <code class="reqn">k</code> source-specific levels of each sample <code class="reqn">i</code> at each feature <code class="reqn">j</code>.
</p>


<h3>Value</h3>

<p>A <code>k</code> by <code>m</code> by <code>n</code> array with the estimated source-specific values. The first axis/dimension in the array corresponds to the different sources.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data = simulate_data(n=100, m=2, k=3, p1=1, p2=1, taus_std=0, log_file=NULL)
res = list()
res$params.hat = Unico(data$X, data$W, data$C1, data$C2, parallel=FALSE, log_file=NULL)
res$Z = tensor(data$X, data$W, data$C1, data$C2, res$params.hat, parallel=FALSE, log_file=NULL)

</code></pre>

<hr>
<h2 id='Unico'>Fitting the Unico model</h2><span id='topic+Unico'></span>

<h3>Description</h3>

<p>Fits the Unico model for an input matrix of features by observations that are coming from a mixture of <code>k</code> sources, under the assumption that each observation is a mixture of unique (unobserved) source-specific values (in each feature in the data). Specifically, for each feature, it standardizes the data and learns the source-specific mean and full <code>k</code> by <code>k</code> variance-covariance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Unico(
  X,
  W,
  C1,
  C2,
  fit_tau = FALSE,
  mean_penalty = 0,
  var_penalty = 0.01,
  covar_penalty = 0.01,
  mean_max_iterations = 2,
  var_max_iterations = 3,
  nloptr_opts_algorithm = "NLOPT_LN_COBYLA",
  max_stds = 2,
  init_weight = "default",
  max_u = 1,
  max_v = 1,
  parallel = TRUE,
  num_cores = NULL,
  log_file = "Unico.log",
  verbose = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Unico_+3A_x">X</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of measurements of <code>m</code> features for <code>n</code> observations. Each column in <code>X</code> is assumed to be a mixture of <code>k</code> sources. Note that <code>X</code> must include row names and column names and that NA values are currently not supported. <code>X</code> should not include features that are constant across all observations.</p>
</td></tr>
<tr><td><code id="Unico_+3A_w">W</code></td>
<td>
<p>An <code>n</code> by <code>k</code> matrix of weights - the weights of <code>k</code> sources for each of the <code>n</code> mixtures (observations). All the weights must be positive and each row - corresponding to the weights of a single observation - must sum up to 1. Note that <code>W</code> must include row names and column names and that NA values are currently not supported.</p>
</td></tr>
<tr><td><code id="Unico_+3A_c1">C1</code></td>
<td>
<p>An <code>n</code> by <code>p1</code> design matrix of covariates that may affect the hidden source-specific values (possibly a different effect size in each source). Note that <code>C1</code> must include row names and column names and should not include an intercept term. NA values are currently not supported. Note that each covariate in <code>C1</code> results in <code>k</code> additional parameters in the model of each feature, therefore, in order to alleviate the possibility of model overfitting, it is advised to be mindful of the balance between the size of <code>C1</code> and the sample size in <code>X</code>.</p>
</td></tr>
<tr><td><code id="Unico_+3A_c2">C2</code></td>
<td>
<p>An <code>n</code> by <code>p2</code> design matrix of covariates that may affect the mixture (i.e. rather than directly the sources of the mixture; for example, variables that capture biases in the collection of the measurements). Note that <code>C2</code> must include row names and column names and should not include an intercept term. NA values are currently not supported.</p>
</td></tr>
<tr><td><code id="Unico_+3A_fit_tau">fit_tau</code></td>
<td>
<p>A logical value indicating whether to fit the standard deviation of the measurement noise (i.e. the i.i.d. component of variation in the model denoted as <code class="reqn">\tau</code>).</p>
</td></tr>
<tr><td><code id="Unico_+3A_mean_penalty">mean_penalty</code></td>
<td>
<p>A non-negative numeric value indicating the regularization strength on the source-specific mean estimates.</p>
</td></tr>
<tr><td><code id="Unico_+3A_var_penalty">var_penalty</code></td>
<td>
<p>A non-negative numeric value indicating the regularization strength on the diagonal entries of the full <code>k</code> by <code>k</code> variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="Unico_+3A_covar_penalty">covar_penalty</code></td>
<td>
<p>A non-negative numeric value indicating the regularization strength on the off diagonal entries of the full <code>k</code> by <code>k</code> variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="Unico_+3A_mean_max_iterations">mean_max_iterations</code></td>
<td>
<p>A non-negative numeric value indicating the number of iterative updates performed on the mean estimates.</p>
</td></tr>
<tr><td><code id="Unico_+3A_var_max_iterations">var_max_iterations</code></td>
<td>
<p>A non-negative numeric value indicating the number of iterative updates performed on the variance-covariance matrix.</p>
</td></tr>
<tr><td><code id="Unico_+3A_nloptr_opts_algorithm">nloptr_opts_algorithm</code></td>
<td>
<p>A string indicating the optimization algorithm to use.</p>
</td></tr>
<tr><td><code id="Unico_+3A_max_stds">max_stds</code></td>
<td>
<p>A non-negative numeric value indicating, for each feature, the portions of data that are considered as outliers. Only samples within <code>max_stds</code> standard deviations from the mean will be used for the moments estimation of a given feature.</p>
</td></tr>
<tr><td><code id="Unico_+3A_init_weight">init_weight</code></td>
<td>
<p>A string indicating the initial weights on the samples to start the iterative optimization.</p>
</td></tr>
<tr><td><code id="Unico_+3A_max_u">max_u</code></td>
<td>
<p>A non-negative numeric value indicating the maximum weights/influence a sample can have on mean estimates.</p>
</td></tr>
<tr><td><code id="Unico_+3A_max_v">max_v</code></td>
<td>
<p>A non-negative numeric value indicating the maximum weights/influence a sample can have on variance-covariance estimates.</p>
</td></tr>
<tr><td><code id="Unico_+3A_parallel">parallel</code></td>
<td>
<p>A logical value indicating whether to use parallel computing (possible when using a multi-core machine).</p>
</td></tr>
<tr><td><code id="Unico_+3A_num_cores">num_cores</code></td>
<td>
<p>A numeric value indicating the number of cores to use (activated only if <code>parallel == TRUE</code>). If <code>num_cores == NULL</code> then all available cores except for one will be used.</p>
</td></tr>
<tr><td><code id="Unico_+3A_log_file">log_file</code></td>
<td>
<p>A path to an output log file. Note that if the file <code>log_file</code> already exists then logs will be appended to the end of the file. Set <code>log_file</code> to <code>NULL</code> to prevent output from being saved into a file; note that if <code>verbose == FALSE</code> then no output file will be generated regardless of the value of <code>log_file</code>.</p>
</td></tr>
<tr><td><code id="Unico_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print logs.</p>
</td></tr>
<tr><td><code id="Unico_+3A_debug">debug</code></td>
<td>
<p>A logical value indicating whether to set the logger to a more detailed debug level; set <code>debug</code> to <code>TRUE</code> before reporting issues.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Unico assumes the following model: </p>
<p style="text-align: center;"><code class="reqn">X_{ij} =  w_{i}^T Z_{ij} +(c_i^{(2)})^T \beta_j+ e_{ij}</code>
</p>

<p>The mixture value at sample <code class="reqn">i</code> feature <code class="reqn">j</code>: <code class="reqn">X_{ij}</code> is modeled as a weighted linear combination, specified by weights <code class="reqn">w_i = (w_{i1},...,w_{ik})</code>, of a total of <code class="reqn">k</code> source-specific levels, specified by <code class="reqn">Z_{ij}=(Z_{ij1},...,Z_{ijk})</code>.
In addition, we also consider global-level covariates <code class="reqn">c_i^{(2)}</code> that systematically affect the observed mixture values and their effect sizes <code class="reqn">\beta_j</code>. <code class="reqn">e_{ij}</code> denotes the i.i.d measurement noise with variance <code class="reqn">\tau</code> across all samples.
Weights have be to non-negative and sum up to <code class="reqn">1</code> across all sources for each sample. In practice, we assume that the weights are fixed and estimated by external methods.
</p>
<p>Source specific profiles are further modeled as: </p>
<p style="text-align: center;"><code class="reqn">Z_{ijh} = \mu_{jh} + (c_i^{(1)})^T \gamma_{jh} + \epsilon_{ijh}</code>
</p>

<p><code class="reqn">\mu_{jh}</code> denotes the population level mean of feature <code class="reqn">j</code> at source <code class="reqn">h</code>.
We also consider covariates <code class="reqn">c_i^{(1)}</code> that systematically affect the source-specific values and their effect sizes <code class="reqn">\gamma_{jh}</code> on each source.
Finally, we actively model the <code class="reqn">k</code> by <code class="reqn">k</code> covariance structure of a given feature <code class="reqn">j</code> across all <code class="reqn">k</code> sources <code class="reqn">Var[\vec{\epsilon_{ij}}] = \Sigma_{j} \in \mathbf{R}^{k \times k}</code>.
</p>


<h3>Value</h3>

<p>A list with the estimated parameters of the model. This list can be then used as the input to other functions such as <a href="#topic+tensor">tensor</a>.
</p>
<table>
<tr><td><code>W</code></td>
<td>
<p>An <code>n</code> by <code>k</code> matrix of weights. This is the same as <code>W</code> from input.</p>
</td></tr>
<tr><td><code>C1</code></td>
<td>
<p>An <code>n</code> by <code>p1</code> design matrix of source-specific covariates. This is the same as <code>C1</code> from input.</p>
</td></tr>
<tr><td><code>C2</code></td>
<td>
<p>An <code>n</code> by <code>p2</code> design matrix of not source-specific covariates. This is the same as <code>C2</code> from input.</p>
</td></tr>
<tr><td><code>mus_hat</code></td>
<td>
<p>An <code>m</code> by <code>k</code> matrix of estimates for the mean of each source in each feature.</p>
</td></tr>
<tr><td><code>gammas_hat</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of the estimated effects of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code>, where the first <code>p1</code> columns are the source-specific effects of the <code>p1</code> covariates on the first source, the following <code>p1</code> columns are the source-specific effects on the second source and so on.</p>
</td></tr>
<tr><td><code>betas_hat</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of the estimated effects of the <code>p2</code> covariates in <code>C2</code> on the mixture values of each of the <code>m</code> features in <code>X</code>.</p>
</td></tr>
<tr><td><code>sigmas_hat</code></td>
<td>
<p>An <code>m</code> by <code>k</code> by <code>k</code> tensor of estimates for the cross source <code>k</code> by <code>k</code> variance-covariance matrix in each feature.</p>
</td></tr>
<tr><td><code>taus_hat</code></td>
<td>
<p>An <code>m</code> by <code>1</code> matrix of estimates for the variance of the measurement noise.</p>
</td></tr>
<tr><td><code>scale.factor</code></td>
<td>
<p>An <code>m</code> by <code>1</code> matrix of scaling factors for standardizing each feature.</p>
</td></tr>
<tr><td><code>config</code></td>
<td>
<p>A list with hyper-parameters used for fitting the model and configurations for in the optimization algorithm.</p>
</td></tr>
<tr><td><code>Us_hat_list</code></td>
<td>
<p>A list tracking, for each feature, the sample weights used for each iteration of the mean optimization  (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>Vs_hat_list</code></td>
<td>
<p>A list tracking, for each feature, the sample weights used for each iteration of the variance-covariance optimization  (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>Ls_hat_list</code></td>
<td>
<p>A list tracking, for each feature, the computed estimates of the upper triangular cholesky decomposition of variance-covariance matrix at each iteration of the variance-covariance optimization (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
<tr><td><code>sigmas_hat_list</code></td>
<td>
<p>A list tracking, for each feature, the computed estimates of the variance-covariance matrix at each iteration of the variance-covariance optimization (activated only if <code>debug == TRUE</code>).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data = simulate_data(n=100, m=2, k=3, p1=1, p2=1, taus_std=0, log_file=NULL)
res = list()
res$params.hat = Unico(data$X, data$W, data$C1, data$C2, parallel=FALSE, log_file=NULL)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
