<!DOCTYPE html><html lang="en"><head><title>Help for package UniprotR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {UniprotR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ConstructGenesTree'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#ConstructLocTree'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#ConvertID'><p>Connect and parse UniProt database identifiers information.</p></a></li>
<li><a href='#Enrichment.BP'><p>Connect and parse UniProt information</p></a></li>
<li><a href='#Enrichment.CC'><p>Connect and parse UniProt information</p></a></li>
<li><a href='#Enrichment.KEGG'><p>Connect and parse UniProt information</p></a></li>
<li><a href='#Enrichment.MF'><p>Connect and parse UniProt information</p></a></li>
<li><a href='#Enrichment.REAC'><p>Connect and parse UniProt information</p></a></li>
<li><a href='#Get.diseases'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#GetAccessionList'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#GetExpression'><p>Connect and parse UniProt Expression information.</p></a></li>
<li><a href='#GetFamily_Domains'><p>Connect and parse UniProt Family Domains information.</p></a></li>
<li><a href='#GetGeneral_Information'><p>Connect and parse UniProt General Information.</p></a></li>
<li><a href='#GetMiscellaneous'><p>Connect and parse UniProt Miscellaneous information.</p></a></li>
<li><a href='#GetNamesTaxa'><p>Connect and parse UniProt Names Taxa information.</p></a></li>
<li><a href='#GetPathology_Biotech'><p>Connect and parse UniProt Pathology_Biotech information.</p></a></li>
<li><a href='#GetpdbStructure'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#GetProteinAnnontate'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#GetProteinFunction'><p>Connect and parse UniProt Protein Function information.</p></a></li>
<li><a href='#GetProteinGOInfo'><p>Connect and parse UniProt proteins gene ontology information.</p></a></li>
<li><a href='#GetProteinInteractions'><p>Connect and parse UniProt protein Interactions information.</p></a></li>
<li><a href='#GetproteinNetwork'><p>Connect and parse stringdb information.</p></a></li>
<li><a href='#GetproteinNetwork_all'><p>Connect and parse stringdb information.</p></a></li>
<li><a href='#GetProteomeFasta'><p>Connect and DOWNLOAD Proteome info.</p></a></li>
<li><a href='#GetProteomeInfo'><p>Connect and DOWNLOAD Proteome info.</p></a></li>
<li><a href='#GetPTM_Processing'><p>Connect and parse UniProt PTM_Processsing information.</p></a></li>
<li><a href='#GetPublication'><p>Connect and parse UniProt Publication about a protein information.</p></a></li>
<li><a href='#GETSeqFastaUniprot'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#GetSeqLength'><p>Connect and parse UniProt Sequences information.</p></a></li>
<li><a href='#GetSequenceIso'><p>Connect and parse UniProt Sequences information.</p></a></li>
<li><a href='#GetSequences'><p>Connect and parse UniProt Sequences information.</p></a></li>
<li><a href='#GetStructureInfo'><p>Connect and parse UniProt protein Structure information.</p></a></li>
<li><a href='#GetSubcellular_location'><p>Connect and parse UniProt protein Subcellular location information.</p></a></li>
<li><a href='#Goparse'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#HandleBadRequests'><p>Handle bad requests</p>
This Function is used to handle possible errors when trying to get url response.</a></li>
<li><a href='#Pathway.Enr'><p>Connect and parse UniProt information</p></a></li>
<li><a href='#Plot.GOMolecular'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#Plot.GOSubCellular'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotAcidity'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotCharge'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotChromosomeInfo'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotGenesNetwork'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotGOAll'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotGOBiological'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotGoInfo'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotGoterms'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotGravy'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotPhysicochemical'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotproteinExist'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotProteinGO_bio'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotProteinGO_cel'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotProteinGO_molc'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotproteinStatus'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#PlotProteinTaxa'><p>Connect and parse UniProt information.</p></a></li>
<li><a href='#ProcessAcc'><p>Connect and parse UniProt information.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Retrieving Information of Proteins from Uniprot</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Mohamed Soudy [aut, cre],
  Ali Mostafa [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mohamed Soudy &lt;MohmedSoudy2009@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Connect to Uniprot <a href="https://www.uniprot.org/">https://www.uniprot.org/</a> to retrieve information about proteins using their accession number such information could be name or taxonomy information, For detailed information kindly read the publication <a href="https://www.sciencedirect.com/science/article/pii/S1874391919303859">https://www.sciencedirect.com/science/article/pii/S1874391919303859</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils , grDevices , stats , grid , graphics , httr , plyr ,
dplyr , scales , magrittr , magick , data.tree , ggplot2 ,
tidyverse , gridExtra , ggpubr , curl, networkD3, stringr ,
qdapRegex , htmlwidgets , alakazam, gprofiler2, progress, ggsci</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Proteomicslab57357/UniprotR">https://github.com/Proteomicslab57357/UniprotR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Proteomicslab57357/UniprotR/issues">https://github.com/Proteomicslab57357/UniprotR/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-05 14:51:30 UTC; mohamed.soudy</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-05 15:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ConstructGenesTree'>Connect and parse UniProt information.</h2><span id='topic+ConstructGenesTree'></span>

<h3>Description</h3>

<p>This Function is used to plot Genes Tree in the data of the accession/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConstructGenesTree(ProteinDataObject,directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ConstructGenesTree_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>input a Dataframe returned from GetNamesTaxa function</p>
</td></tr>
<tr><td><code id="ConstructGenesTree_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save txt file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='ConstructLocTree'>Connect and parse UniProt information.</h2><span id='topic+ConstructLocTree'></span>

<h3>Description</h3>

<p>This Function is used to plot location's Tree in the data of the accession/s in the chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConstructLocTree(ProteinDataObject,directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ConstructLocTree_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>input a Dataframe returned from GetNamesTaxa function</p>
</td></tr>
<tr><td><code id="ConstructLocTree_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save txt file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='ConvertID'>Connect and parse UniProt database identifiers information.</h2><span id='topic+ConvertID'></span>

<h3>Description</h3>

<p>The function is work to convert the UniProtKB AC/ID to any database identifiers
available by the UniProtKB.
For more information about available database identifiers
see https://www.uniprot.org/help/id_mapping.
see https://raw.githubusercontent.com/MohmedSoudy/UniprotR/master/uniprot_ids.csv
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConvertID(ProteinAccList,ID_from="UniProtKB_AC-ID",ID_to=NULL,taxId=NULL,path=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ConvertID_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="ConvertID_+3A_id_from">ID_from</code></td>
<td>
<p>string of database identifier abbreviation, from which the Accession/ID will be converted</p>
</td></tr>
<tr><td><code id="ConvertID_+3A_id_to">ID_to</code></td>
<td>
<p>string of database identifier abbreviation, to which the Accession/ID will be converted.
default is all database identifier available in UniProtKB</p>
</td></tr>
<tr><td><code id="ConvertID_+3A_taxid">taxId</code></td>
<td>
<p>string Needed when the ID_to is 'UniProtKB' that could be '9606' for human.</p>
</td></tr>
<tr><td><code id="ConvertID_+3A_path">path</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where column one contains the Accession/ID before conversion
and other columns contains the Accession/ID after conversion
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Enrichment.BP'>Connect and parse UniProt information</h2><span id='topic+Enrichment.BP'></span>

<h3>Description</h3>

<p>This function is used for Enrichment analysis of biological process of given list of genes or proteins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Enrichment.BP(Accs,OS="hsapiens",p_value=0.05,directorypath=NULL,top=10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Enrichment.BP_+3A_accs">Accs</code></td>
<td>
<p>Vector of UniProt Accession/s or genes</p>
</td></tr>
<tr><td><code id="Enrichment.BP_+3A_os">OS</code></td>
<td>
<p>organism name Example: human - 'hsapiens', mouse - 'mmusculus'</p>
</td></tr>
<tr><td><code id="Enrichment.BP_+3A_p_value">p_value</code></td>
<td>
<p>custom p-value threshold for significance, default = 0.05</p>
</td></tr>
<tr><td><code id="Enrichment.BP_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save output plot</p>
</td></tr>
<tr><td><code id="Enrichment.BP_+3A_top">top</code></td>
<td>
<p>Top N terms to be visualized</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Enrichment.CC'>Connect and parse UniProt information</h2><span id='topic+Enrichment.CC'></span>

<h3>Description</h3>

<p>This function is used for Enrichment analysis of cellular component of given list of genes or proteins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Enrichment.CC(Accs,OS="hsapiens",p_value=0.05,directorypath=NULL,top=10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Enrichment.CC_+3A_accs">Accs</code></td>
<td>
<p>Vector of UniProt Accession/s or genes</p>
</td></tr>
<tr><td><code id="Enrichment.CC_+3A_os">OS</code></td>
<td>
<p>organism name Example: human - 'hsapiens', mouse - 'mmusculus'</p>
</td></tr>
<tr><td><code id="Enrichment.CC_+3A_p_value">p_value</code></td>
<td>
<p>custom p-value threshold for significance, default = 0.05</p>
</td></tr>
<tr><td><code id="Enrichment.CC_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save output plot</p>
</td></tr>
<tr><td><code id="Enrichment.CC_+3A_top">top</code></td>
<td>
<p>Top N terms to be visualized</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Enrichment.KEGG'>Connect and parse UniProt information</h2><span id='topic+Enrichment.KEGG'></span>

<h3>Description</h3>

<p>This function is used for Enrichment analysis of given list of genes or proteins from KEGG database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Enrichment.KEGG(Accs,OS="hsapiens",p_value=0.05,directorypath=NULL,top=10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Enrichment.KEGG_+3A_accs">Accs</code></td>
<td>
<p>Vector of UniProt Accession/s or genes</p>
</td></tr>
<tr><td><code id="Enrichment.KEGG_+3A_os">OS</code></td>
<td>
<p>organism name Example: human - 'hsapiens', mouse - 'mmusculus'</p>
</td></tr>
<tr><td><code id="Enrichment.KEGG_+3A_p_value">p_value</code></td>
<td>
<p>custom p-value threshold for significance, default = 0.05</p>
</td></tr>
<tr><td><code id="Enrichment.KEGG_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save output plot</p>
</td></tr>
<tr><td><code id="Enrichment.KEGG_+3A_top">top</code></td>
<td>
<p>Top N terms to be visualized</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Enrichment.MF'>Connect and parse UniProt information</h2><span id='topic+Enrichment.MF'></span>

<h3>Description</h3>

<p>This function is used for Enrichment analysis of Molecular function of given list of genes or proteins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Enrichment.MF(Accs,OS="hsapiens",p_value=0.05,directorypath=NULL,top=10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Enrichment.MF_+3A_accs">Accs</code></td>
<td>
<p>Vector of UniProt Accession/s or genes</p>
</td></tr>
<tr><td><code id="Enrichment.MF_+3A_os">OS</code></td>
<td>
<p>organism name Example: human - 'hsapiens', mouse - 'mmusculus'</p>
</td></tr>
<tr><td><code id="Enrichment.MF_+3A_p_value">p_value</code></td>
<td>
<p>custom p-value threshold for significance, default = 0.05</p>
</td></tr>
<tr><td><code id="Enrichment.MF_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save output plot</p>
</td></tr>
<tr><td><code id="Enrichment.MF_+3A_top">top</code></td>
<td>
<p>Top N terms to be visualized</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Enrichment.REAC'>Connect and parse UniProt information</h2><span id='topic+Enrichment.REAC'></span>

<h3>Description</h3>

<p>This function is used for Enrichment analysis of given list of genes or proteins from REACTOME
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Enrichment.REAC(Accs,OS="hsapiens",p_value=0.05,directorypath=NULL,top=10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Enrichment.REAC_+3A_accs">Accs</code></td>
<td>
<p>Vector of UniProt Accession/s or genes</p>
</td></tr>
<tr><td><code id="Enrichment.REAC_+3A_os">OS</code></td>
<td>
<p>organism name Example: human - 'hsapiens', mouse - 'mmusculus'</p>
</td></tr>
<tr><td><code id="Enrichment.REAC_+3A_p_value">p_value</code></td>
<td>
<p>custom p-value threshold for significance, default = 0.05</p>
</td></tr>
<tr><td><code id="Enrichment.REAC_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save output plot</p>
</td></tr>
<tr><td><code id="Enrichment.REAC_+3A_top">top</code></td>
<td>
<p>Top N terms to be visualized</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Get.diseases'>Connect and parse UniProt information.</h2><span id='topic+Get.diseases'></span>

<h3>Description</h3>

<p>This Function is used to get diseases associated with proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get.diseases(Pathology_object , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Get.diseases_+3A_pathology_object">Pathology_object</code></td>
<td>
<p>Dataframe retrieved from UniprotR Function &quot;GetPathology_Biotech&quot;</p>
</td></tr>
<tr><td><code id="Get.diseases_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output file</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetAccessionList'>Connect and parse UniProt information.</h2><span id='topic+GetAccessionList'></span>

<h3>Description</h3>

<p>This function can be used to get a list of UniProt Accession/s from a csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetAccessionList(DataObjPath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetAccessionList_+3A_dataobjpath">DataObjPath</code></td>
<td>
<p>input path of excel file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of UniProt Accession/s
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetExpression'>Connect and parse UniProt Expression information.</h2><span id='topic+GetExpression'></span>

<h3>Description</h3>

<p>The function is work to retrieve Expression data from UniProt for a
list of proteins accessions.For more information about what included
in the Expression data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetExpression(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetExpression_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetExpression_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetFamily_Domains'>Connect and parse UniProt Family Domains information.</h2><span id='topic+GetFamily_Domains'></span>

<h3>Description</h3>

<p>The function is work to retrieve Family Domains data from UniProt for a
list of proteins accessions.For more information about what included
in the Family Domains data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetFamily_Domains(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetFamily_Domains_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetFamily_Domains_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetGeneral_Information'>Connect and parse UniProt General Information.</h2><span id='topic+GetGeneral_Information'></span>

<h3>Description</h3>

<p>The function is work to retrieve General Information data from UniProt
for a list of proteins accessions.For more information about what included in the
General Information data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetGeneral_Information(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetGeneral_Information_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetGeneral_Information_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the General Information of protein from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetGeneral_Information("O14520")

</code></pre>

<hr>
<h2 id='GetMiscellaneous'>Connect and parse UniProt Miscellaneous information.</h2><span id='topic+GetMiscellaneous'></span>

<h3>Description</h3>

<p>The function is work to retrieve Miscellaneous data from UniProt for a list
of proteins accessions.For more information about what included in the
Miscellaneous data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetMiscellaneous(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetMiscellaneous_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetMiscellaneous_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetMiscellaneous("O14520")

</code></pre>

<hr>
<h2 id='GetNamesTaxa'>Connect and parse UniProt Names Taxa information.</h2><span id='topic+GetNamesTaxa'></span>

<h3>Description</h3>

<p>The function is work to retrieve Names Taxa data from UniProt for a list of proteins accessions.
For more information about what included in the NamesTaxa
data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetNamesTaxa(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetNamesTaxa_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetNamesTaxa_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information of protein name &amp; taxonomy from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetNamesTaxa("O14520")

</code></pre>

<hr>
<h2 id='GetPathology_Biotech'>Connect and parse UniProt Pathology_Biotech information.</h2><span id='topic+GetPathology_Biotech'></span>

<h3>Description</h3>

<p>The function is work to retrieve Pathology_Biotech data from UniProt for
a list of proteins accessions.For more information about what included in the
Pathology_Biotech data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetPathology_Biotech(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetPathology_Biotech_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetPathology_Biotech_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetPathology_Biotech("O14520")

</code></pre>

<hr>
<h2 id='GetpdbStructure'>Connect and parse UniProt information.</h2><span id='topic+GetpdbStructure'></span>

<h3>Description</h3>

<p>The function is work to retrieving GetpdbStructure and download it
to user directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetpdbStructure(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetpdbStructure_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>input a vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetpdbStructure_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetProteinAnnontate'>Connect and parse UniProt information.</h2><span id='topic+GetProteinAnnontate'></span>

<h3>Description</h3>

<p>The function is work to retrieve user-defined information data from UniProt
for a list of proteins accessions For more
information see https://www.uniprot.org/help/uniprotkb_column_names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetProteinAnnontate(ProteinAccList , columns)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetProteinAnnontate_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>a vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetProteinAnnontate_+3A_columns">columns</code></td>
<td>
<p>a vector of UniProtKB column names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetProteinInteractions("O14520")

</code></pre>

<hr>
<h2 id='GetProteinFunction'>Connect and parse UniProt Protein Function information.</h2><span id='topic+GetProteinFunction'></span>

<h3>Description</h3>

<p>The function is work to retrieve Protein Function data from UniProt for
a list of proteins accessions.For more information about what included in the
Protein Function data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetProteinFunction(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetProteinFunction_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetProteinFunction_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information of protein function roles from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetProteinFunction("O14520")

</code></pre>

<hr>
<h2 id='GetProteinGOInfo'>Connect and parse UniProt proteins gene ontology information.</h2><span id='topic+GetProteinGOInfo'></span>

<h3>Description</h3>

<p>The function is work to retrieve proteins gene ontology  data from UniProt
for a list of proteins accessions.For more information about what included in the
proteins gene ontology data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetProteinGOInfo(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetProteinGOInfo_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetProteinGOInfo_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information of Gene ontology of protein from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetProteinGOInfo("O14520")

</code></pre>

<hr>
<h2 id='GetProteinInteractions'>Connect and parse UniProt protein Interactions information.</h2><span id='topic+GetProteinInteractions'></span>

<h3>Description</h3>

<p>The function is work to retrieve protein Interactions data from UniProt for a
list of proteins accessions.For more information about what included in the
protein Interactions data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetProteinInteractions(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetProteinInteractions_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetProteinInteractions_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information of protein Interactions from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetProteinInteractions("ProteinAccession")

</code></pre>

<hr>
<h2 id='GetproteinNetwork'>Connect and parse stringdb information.</h2><span id='topic+GetproteinNetwork'></span>

<h3>Description</h3>

<p>This function is connecting to stringdb and retrieve all possible interactions
for the searched protein/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetproteinNetwork(ProteinAccList , directorypath = NULL)

GetproteinNetwork(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetproteinNetwork_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>input a vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetproteinNetwork_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetproteinNetwork_all'>Connect and parse stringdb information.</h2><span id='topic+GetproteinNetwork_all'></span>

<h3>Description</h3>

<p>This function is connecting to stringdb and retrieve PPI between input list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetproteinNetwork_all(ProteinAccList , directorypath = NULL, SpeciesID = 9606)

GetproteinNetwork_all(ProteinAccList , directorypath = NULL,SpeciesID = 9606)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetproteinNetwork_all_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>input a vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetproteinNetwork_all_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
<tr><td><code id="GetproteinNetwork_all_+3A_speciesid">SpeciesID</code></td>
<td>
<p>Taxonomic id of accession's species ex. homo sapines 9606</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetProteomeFasta'>Connect and DOWNLOAD Proteome info.</h2><span id='topic+GetProteomeFasta'></span>

<h3>Description</h3>

<p>The function is work to retrieve proteome information in FASTA format based on proteome id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetProteomeFasta(ProteomeID , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetProteomeFasta_+3A_proteomeid">ProteomeID</code></td>
<td>
<p>Proteome ID from UniProt</p>
</td></tr>
<tr><td><code id="GetProteomeFasta_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save FASTA file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The function Download fasta format of proteome.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetProteomeInfo'>Connect and DOWNLOAD Proteome info.</h2><span id='topic+GetProteomeInfo'></span>

<h3>Description</h3>

<p>The function is work to retrieve proteome information based on proteome id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetProteomeInfo(ProteomeID , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetProteomeInfo_+3A_proteomeid">ProteomeID</code></td>
<td>
<p>Proteome ID from UniProt</p>
</td></tr>
<tr><td><code id="GetProteomeInfo_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save CSV file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The function Download csv Info of proteome.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetPTM_Processing'>Connect and parse UniProt PTM_Processsing information.</h2><span id='topic+GetPTM_Processing'></span>

<h3>Description</h3>

<p>The function is work to retrieve PTM_Processsing data from UniProt for a list of proteins accessions.
For more information about what included in the PTM_Processsing data
see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetPTM_Processing(ProteinAccList, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetPTM_Processing_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetPTM_Processing_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetPTM_Processing("O14520" )

</code></pre>

<hr>
<h2 id='GetPublication'>Connect and parse UniProt Publication about a protein information.</h2><span id='topic+GetPublication'></span>

<h3>Description</h3>

<p>The function is work to retrieve Publication data from UniProt for a list of proteins accessions.
For more information about what included in the Publication
data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetPublication(ProteinAccList , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetPublication_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetPublication_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the Publication of protein from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GETSeqFastaUniprot'>Connect and parse UniProt information.</h2><span id='topic+GETSeqFastaUniprot'></span>

<h3>Description</h3>

<p>This Function is used to get Sequence information of accession/s from Uniprot as a Fasta file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GETSeqFastaUniprot(Accessions,FilePath = NULL, FileName = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GETSeqFastaUniprot_+3A_accessions">Accessions</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GETSeqFastaUniprot_+3A_filepath">FilePath</code></td>
<td>
<p>path of directory to save the output fasta.</p>
</td></tr>
<tr><td><code id="GETSeqFastaUniprot_+3A_filename">FileName</code></td>
<td>
<p>Name of the fasta file.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='GetSeqLength'>Connect and parse UniProt Sequences information.</h2><span id='topic+GetSeqLength'></span>

<h3>Description</h3>

<p>The function is work to retrieve Sequence's Length data from Uniparc for a list of proteins accessions.
This function was added to overcome the NAs returned when Uniprot database deleted the protein from the database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetSeqLength(ProteinAccList, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetSeqLength_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetSeqLength_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetSeqLength("O14520")

</code></pre>

<hr>
<h2 id='GetSequenceIso'>Connect and parse UniProt Sequences information.</h2><span id='topic+GetSequenceIso'></span>

<h3>Description</h3>

<p>The function is work to retrieve protein's Sequence data from Uniparc for a list of proteins accessions.
This function was added to handle isoformes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetSequenceIso(ProteinAccList, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetSequenceIso_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetSequenceIso_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetSequenceIso("O14520")

</code></pre>

<hr>
<h2 id='GetSequences'>Connect and parse UniProt Sequences information.</h2><span id='topic+GetSequences'></span>

<h3>Description</h3>

<p>The function is work to retrieve Sequences data from UniProt for a list of proteins accessions.
For more information about what included in the Sequences data
see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetSequences(ProteinAccList, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetSequences_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s</p>
</td></tr>
<tr><td><code id="GetSequences_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information retrieved from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Obj &lt;- GetSequences("O14520")

</code></pre>

<hr>
<h2 id='GetStructureInfo'>Connect and parse UniProt protein Structure information.</h2><span id='topic+GetStructureInfo'></span>

<h3>Description</h3>

<p>The function is work to retrieve Structral data from UniProt for a list of proteins accessions.
For more information about what included in the structral 
data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetStructureInfo(ProteinAccList, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetStructureInfo_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s.</p>
</td></tr>
<tr><td><code id="GetStructureInfo_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the Structural information of protein from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>

<hr>
<h2 id='GetSubcellular_location'>Connect and parse UniProt protein Subcellular location information.</h2><span id='topic+GetSubcellular_location'></span>

<h3>Description</h3>

<p>The function is work to retrieve protein Subcellular location data from UniProt for a list of proteins accessions.
For more information about what included in the protein Subcellular location
data see https://www.uniprot.org/help/return_fields.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetSubcellular_location(ProteinAccList, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetSubcellular_location_+3A_proteinacclist">ProteinAccList</code></td>
<td>
<p>Vector of UniProt Accession/s.</p>
</td></tr>
<tr><td><code id="GetSubcellular_location_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>DataFrame where rows names are the accession
and columns contains the information about Subcellular location of protein from the UniProt
</p>


<h3>Note</h3>

<p>The function also, Creates a csv file with the retrieved information.
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Goparse'>Connect and parse UniProt information.</h2><span id='topic+Goparse'></span>

<h3>Description</h3>

<p>This Function is used to parse data retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Goparse(GOObj , index = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Goparse_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe.</p>
</td></tr>
<tr><td><code id="Goparse_+3A_index">index</code></td>
<td>
<p>idex of Go term in GoObj</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='HandleBadRequests'>Handle bad requests
This Function is used to handle possible errors when trying to get url response.</h2><span id='topic+HandleBadRequests'></span>

<h3>Description</h3>

<p>Handle bad requests
This Function is used to handle possible errors when trying to get url response.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HandleBadRequests(RequestCode)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HandleBadRequests_+3A_requestcode">RequestCode</code></td>
<td>
<p>Response returned from url</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>HandleBadRequests(400)

</code></pre>

<hr>
<h2 id='Pathway.Enr'>Connect and parse UniProt information</h2><span id='topic+Pathway.Enr'></span>

<h3>Description</h3>

<p>This function is used for Enrichment analysis of given list of genes or proteins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pathway.Enr(Accs,OS="hsapiens",p_value=0.05,directorypath=NULL,top=10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Pathway.Enr_+3A_accs">Accs</code></td>
<td>
<p>Vector of UniProt Accession/s or genes</p>
</td></tr>
<tr><td><code id="Pathway.Enr_+3A_os">OS</code></td>
<td>
<p>organism Example: human - 'hsapiens' for more info https://biit.cs.ut.ee/gprofiler/page/organism-list</p>
</td></tr>
<tr><td><code id="Pathway.Enr_+3A_p_value">p_value</code></td>
<td>
<p>custom p-value threshold for significance, default = 0.05</p>
</td></tr>
<tr><td><code id="Pathway.Enr_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save output plot</p>
</td></tr>
<tr><td><code id="Pathway.Enr_+3A_top">top</code></td>
<td>
<p>Top N terms to be visualized</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Plot.GOMolecular'>Connect and parse UniProt information.</h2><span id='topic+Plot.GOMolecular'></span>

<h3>Description</h3>

<p>This Function is used to plot Molecular function of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot.GOMolecular(GOObj, Top = 10, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Plot.GOMolecular_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe returned from UniprotR Function &quot;GetProteinGOInfo&quot;</p>
</td></tr>
<tr><td><code id="Plot.GOMolecular_+3A_top">Top</code></td>
<td>
<p>Number of molecular functions to be visualized</p>
</td></tr>
<tr><td><code id="Plot.GOMolecular_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='Plot.GOSubCellular'>Connect and parse UniProt information.</h2><span id='topic+Plot.GOSubCellular'></span>

<h3>Description</h3>

<p>This Function is used to plot subcellular localization of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Plot.GOSubCellular(GOObj, Top = 10, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Plot.GOSubCellular_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe returned from UniprotR Function &quot;GetProteinGOInfo&quot;</p>
</td></tr>
<tr><td><code id="Plot.GOSubCellular_+3A_top">Top</code></td>
<td>
<p>Number of molecular functions to be visualized</p>
</td></tr>
<tr><td><code id="Plot.GOSubCellular_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotAcidity'>Connect and parse UniProt information.</h2><span id='topic+PlotAcidity'></span>

<h3>Description</h3>

<p>This Function is used to plot proteins acidity retrieved from &quot;GetSequences&quot; Function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotAcidity(SeqDataObjPath , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotAcidity_+3A_seqdataobjpath">SeqDataObjPath</code></td>
<td>
<p>Dataframe retrieved from UniprotR Function &quot;GetSequences&quot;</p>
</td></tr>
<tr><td><code id="PlotAcidity_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotCharge'>Connect and parse UniProt information.</h2><span id='topic+PlotCharge'></span>

<h3>Description</h3>

<p>This Function is used to plot proteins charge retrieved from &quot;GetSequences&quot; Function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCharge(SeqDataObjPath , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotCharge_+3A_seqdataobjpath">SeqDataObjPath</code></td>
<td>
<p>Dataframe retrieved from UniprotR Function &quot;GetSequences&quot;</p>
</td></tr>
<tr><td><code id="PlotCharge_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotChromosomeInfo'>Connect and parse UniProt information.</h2><span id='topic+PlotChromosomeInfo'></span>

<h3>Description</h3>

<p>This Function is used to plot location's frequency in the data of the accession/s in the chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotChromosomeInfo(ProteinDataObject,directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotChromosomeInfo_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>input a Dataframe returned from GetNamesTaxa function</p>
</td></tr>
<tr><td><code id="PlotChromosomeInfo_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save files returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotGenesNetwork'>Connect and parse UniProt information.</h2><span id='topic+PlotGenesNetwork'></span>

<h3>Description</h3>

<p>This Function is used to cluster proteins based on primary genes retrieved from &quot;GetNamesTaxa&quot; Function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGenesNetwork(ProteinDataObject , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotGenesNetwork_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>Dataframe retrieved from UniprotR Function &quot;GetNamesTaxa&quot;</p>
</td></tr>
<tr><td><code id="PlotGenesNetwork_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotGOAll'>Connect and parse UniProt information.</h2><span id='topic+PlotGOAll'></span>

<h3>Description</h3>

<p>This Function is used to plot the retrieved Gene Ontology from function 'GetProteinGOInfo'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGOAll(GOObj, Top = 10, directorypath = NULL, width = width, height = height)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotGOAll_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe returned from UniprotR Function &quot;GetProteinGOInfo&quot;</p>
</td></tr>
<tr><td><code id="PlotGOAll_+3A_top">Top</code></td>
<td>
<p>Number of molecular functions to be visualized</p>
</td></tr>
<tr><td><code id="PlotGOAll_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
<tr><td><code id="PlotGOAll_+3A_width">width</code></td>
<td>
<p>width of the generated plot</p>
</td></tr>
<tr><td><code id="PlotGOAll_+3A_height">height</code></td>
<td>
<p>height of the generated plot</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotGOBiological'>Connect and parse UniProt information.</h2><span id='topic+PlotGOBiological'></span>

<h3>Description</h3>

<p>This Function is used to plot Biological process of proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGOBiological(GOObj, Top = 10, directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotGOBiological_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe returned from UniprotR Function &quot;GetProteinGOInfo&quot;</p>
</td></tr>
<tr><td><code id="PlotGOBiological_+3A_top">Top</code></td>
<td>
<p>Number of molecular functions to be visualized</p>
</td></tr>
<tr><td><code id="PlotGOBiological_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotGoInfo'>Connect and parse UniProt information.</h2><span id='topic+PlotGoInfo'></span>

<h3>Description</h3>

<p>This Function is used to plot data retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGoInfo(GOObj , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotGoInfo_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.</p>
</td></tr>
<tr><td><code id="PlotGoInfo_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save excel file containig results returened by the function ( default = NA ).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotGoterms'>Connect and parse UniProt information.</h2><span id='topic+PlotGoterms'></span>

<h3>Description</h3>

<p>This Function is used to plot data retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGoterms(GOObj , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotGoterms_+3A_goobj">GOObj</code></td>
<td>
<p>Dataframe.</p>
</td></tr>
<tr><td><code id="PlotGoterms_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save plot returned by function ( default = NA ).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotGravy'>Connect and parse UniProt information.</h2><span id='topic+PlotGravy'></span>

<h3>Description</h3>

<p>This Function is used to plot proteins gravy index retrieved from &quot;GetSequences&quot; Function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGravy(SeqDataObjPath , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotGravy_+3A_seqdataobjpath">SeqDataObjPath</code></td>
<td>
<p>Dataframe retrieved from UniprotR Function &quot;GetSequences&quot;</p>
</td></tr>
<tr><td><code id="PlotGravy_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save Output plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotPhysicochemical'>Connect and parse UniProt information.</h2><span id='topic+PlotPhysicochemical'></span>

<h3>Description</h3>

<p>This function can be used to get a list of UniProt Accession/s from a csv file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPhysicochemical(SeqDataObjPath , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotPhysicochemical_+3A_seqdataobjpath">SeqDataObjPath</code></td>
<td>
<p>Dataframe returned from GetSequence function.</p>
</td></tr>
<tr><td><code id="PlotPhysicochemical_+3A_directorypath">directorypath</code></td>
<td>
<p>Path to save Physcochemical properties plot.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotproteinExist'>Connect and parse UniProt information.</h2><span id='topic+PlotproteinExist'></span>

<h3>Description</h3>

<p>This Function is used to plot protein status in the data of the accession/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotproteinExist(ProteinDataObject,directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotproteinExist_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>input a Dataframe returned from GetMiscellaneous function</p>
</td></tr>
<tr><td><code id="PlotproteinExist_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save files returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotProteinGO_bio'>Connect and parse UniProt information.</h2><span id='topic+PlotProteinGO_bio'></span>

<h3>Description</h3>

<p>This Function is used to plot biological process data retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotProteinGO_bio(GO_df , dir_path = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotProteinGO_bio_+3A_go_df">GO_df</code></td>
<td>
<p>Dataframe.</p>
</td></tr>
<tr><td><code id="PlotProteinGO_bio_+3A_dir_path">dir_path</code></td>
<td>
<p>path to save files returened by the function ( default = NA ).</p>
</td></tr>
</table>


<h3>Note</h3>

<p>if no dir_path was given ( default = NA ) the function will only view the plot and will not save it
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotProteinGO_cel'>Connect and parse UniProt information.</h2><span id='topic+PlotProteinGO_cel'></span>

<h3>Description</h3>

<p>This Function is used to plot cellular components data retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotProteinGO_cel(GO_df , dir_path = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotProteinGO_cel_+3A_go_df">GO_df</code></td>
<td>
<p>Dataframe.</p>
</td></tr>
<tr><td><code id="PlotProteinGO_cel_+3A_dir_path">dir_path</code></td>
<td>
<p>path to save files returened by the function ( default = NA ).</p>
</td></tr>
</table>


<h3>Note</h3>

<p>if no dir_path was given ( default = NA ) the function will only view the plot and Will not save it
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotProteinGO_molc'>Connect and parse UniProt information.</h2><span id='topic+PlotProteinGO_molc'></span>

<h3>Description</h3>

<p>This Function is used to plot molecular functions data retrieved from UniprotR Function &quot;GetProteinGOInfo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotProteinGO_molc(GO_df , dir_path = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotProteinGO_molc_+3A_go_df">GO_df</code></td>
<td>
<p>Dataframe.</p>
</td></tr>
<tr><td><code id="PlotProteinGO_molc_+3A_dir_path">dir_path</code></td>
<td>
<p>path to save files returened by the function ( default = NA ).</p>
</td></tr>
</table>


<h3>Note</h3>

<p>if no dir_path was given ( default = NA ) the function will only view the plot and will not save it
</p>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotproteinStatus'>Connect and parse UniProt information.</h2><span id='topic+PlotproteinStatus'></span>

<h3>Description</h3>

<p>This Function is used to plot protein status in the data of the accession/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotproteinStatus (ProteinDataObject,directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotproteinStatus_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>input a Dataframe returned from GetMiscellaneous function</p>
</td></tr>
<tr><td><code id="PlotproteinStatus_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save files returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='PlotProteinTaxa'>Connect and parse UniProt information.</h2><span id='topic+PlotProteinTaxa'></span>

<h3>Description</h3>

<p>This Function is used to plot different taxas found of the accessions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotProteinTaxa(ProteinDataObject , directorypath = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotProteinTaxa_+3A_proteindataobject">ProteinDataObject</code></td>
<td>
<p>input a Dataframe of proteins as rownames.</p>
</td></tr>
<tr><td><code id="PlotProteinTaxa_+3A_directorypath">directorypath</code></td>
<td>
<p>path to save files returened by the function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

<hr>
<h2 id='ProcessAcc'>Connect and parse UniProt information.</h2><span id='topic+ProcessAcc'></span>

<h3>Description</h3>

<p>This Function is used to check validty of input accessions the data of the accession/s.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProcessAcc(Accessions)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ProcessAcc_+3A_accessions">Accessions</code></td>
<td>
<p>acession list returened from GetAccession function</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mohmed Soudy <a href="mailto:Mohamed.soudy@57357.com">Mohamed.soudy@57357.com</a> and Ali Mostafa <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
