<!DOCTYPE html><html><head><title>Help for package VALERIE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {VALERIE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ComputePSI'><p>Percent spliced-in (PSI) computation</p></a></li>
<li><a href='#PlotPSI'><p>Percent spliced-in (PSI) visualization for alternative splicing events</p></a></li>
<li><a href='#PlotPSI.A3SS.MultiGroups'><p>Percent spliced in (PSI) visualization for alternative 3' splice sites (A3SS)</p></a></li>
<li><a href='#PlotPSI.A3SS.TwoGroups'><p>Percent spliced in (PSI) visualization for alternative 3' splice sites (A3SS)</p></a></li>
<li><a href='#PlotPSI.A5SS.MultiGroups'><p>Percent spliced-in (PSI) visualization for alternative 5' splice sites (A5SS)</p></a></li>
<li><a href='#PlotPSI.A5SS.TwoGroups'><p>Percent spliced-in (PSI) visualization for alternative 5' splice sites (A5SS)</p></a></li>
<li><a href='#PlotPSI.MXE.MultiGroups'><p>Percent spliced-in (PSI) visualization for mutually exclusive exons (MXE)</p></a></li>
<li><a href='#PlotPSI.MXE.TwoGroups'><p>Percent spliced-in (PSI) visualization for mutually exclusive exons (MXE)</p></a></li>
<li><a href='#PlotPSI.RI.MultiGroups'><p>Percent spliced-in (PSI) visualization for retained introns (RI)</p></a></li>
<li><a href='#PlotPSI.RI.TwoGroups'><p>Percent spliced-in (PSI) visualization for retained introns (RI)</p></a></li>
<li><a href='#PlotPSI.SE.MultiGroups'><p>Percent spliced-in (PSI) visualization for skipped exons (SE)</p></a></li>
<li><a href='#PlotPSI.SE.TwoGroups'><p>Percent spliced-in (PSI) visualization for skipped exons (SE)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Visualising Splicing at Single-Cell Resolution</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Alternative splicing produces a variety of different protein products from a given gene. 'VALERIE' enables visualisation of alternative splicing events from high-throughput single-cell RNA-sequencing experiments. 'VALERIE' computes percent spliced-in (PSI) values for user-specified genomic coordinates corresponding to alternative splicing events. PSI is the proportion of sequencing reads supporting the included exon/intron as defined by Shiozawa (2018) &lt;<a href="https://doi.org/10.1038%2Fs41467-018-06063-x">doi:10.1038/s41467-018-06063-x</a>&gt;. PSI are inferred from sequencing reads data based on specialised infrastructures for representing and computing annotated genomic ranges by Lawrence (2013) &lt;<a href="https://doi.org/10.1371%2Fjournal.pcbi.1003118">doi:10.1371/journal.pcbi.1003118</a>&gt;. Computed PSI for each single cell are subsequently presented in the form of a heatmap implemented using the 'pheatmap' package by Kolde (2010) <a href="https://CRAN.R-project.org/package=pheatmap">https://CRAN.R-project.org/package=pheatmap</a>. Board overview of the mean PSI difference and associated p-values across different user-defined groups of single cells are presented in the form of a line graph using the 'ggplot2' package by Wickham (2007) <a href="https://CRAN.R-project.org/package=ggplot2">https://CRAN.R-project.org/package=ggplot2</a>.</td>
</tr>
<tr>
<td>Imports:</td>
<td>GenomicAlignments (&ge; 1.16.0), GenomicRanges (&ge; 1.32.0),
IRanges (&ge; 2.14.0), Rsamtools (&ge; 1.34.0), plyr (&ge; 1.8.4),
ggplot2 (&ge; 3.0.0), pheatmap (&ge; 1.0.10), ggplotify (&ge; 0.0.3),
ggpubr (&ge; 0.2.4), scales (&ge; 1.0.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Sean Wen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sean Wen &lt;sean.wenwx@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-07-10 09:44:19 UTC; seanwen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-07-10 10:20:13 UTC</td>
</tr>
</table>
<hr>
<h2 id='ComputePSI'>Percent spliced-in (PSI) computation</h2><span id='topic+ComputePSI'></span>

<h3>Description</h3>

<p><code>ComputePSI</code> computes percent spliced-in (PSI) at each genomic coordinate for exon-level alternative splicing events, namely skipped exon (SE), mutually exclusive exons (MXE), retained intron (RI), alternative 5' splice site (A5SS), and alternative 3' splice site (A3SS)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI(SampleInfo, ExonInfo, BAM, MinCoverage)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First columns contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2018) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, and A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_bam">BAM</code></td>
<td>
<p>Folder containing the BAM files sorted by genomic coordinates.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_mincoverage">MinCoverage</code></td>
<td>
<p>numeric. Coverage (Total reads) threshold below which the PSI value of the genomic coordinate is annotate as missing value, i.e. no coverage.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons. Formula for computing PSI is number of reads with non-N CIGAR operation divided by the total number of reads. Total number of reads is the sum of reads with non-N CIGAR operation and reads with N-CIGAR operation
</p>


<h3>Value</h3>

<p>A data frame of class rehab where rows are the genomic coordinates and columns are the sample names.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- ComputePSI(SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_small.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info.txt", package="VALERIE"),
  BAM=system.file("extdata/BAM", "", package="VALERIE"),
  MinCoverage=10)
PSI[1:5,1:4]
</code></pre>

<hr>
<h2 id='PlotPSI'>Percent spliced-in (PSI) visualization for alternative splicing events</h2><span id='topic+PlotPSI'></span>

<h3>Description</h3>

<p><code>PlotPSI</code> visualizes percent spliced-in (PSI) for each genomic coordinate for alternative splicing events across two groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI(object, SampleInfo, ExonInfo, statistical.test = c("wilcox",
  "t.test", "KW", "ANOVA"), multiple.testing = c("holm", "hochberg",
  "hommel", "bonferroni", "BH", "BY", "fdr", "none"), Plots, plot.width,
  plot.height, EventType = c("SE", "MXE", "RI", "A5SS", "A3SS"), Groups)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1, Group2, etc. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, and A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between groups of single cells. Parametric methods include student t-test and analysis of variance. Non-parametric methods include wilcoxon rank sum test and Kruskal-Wallis test.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_eventtype">EventType</code></td>
<td>
<p>Indicates the type of alternative splicing event to plot.</p>
</td></tr>
<tr><td><code id="PlotPSI_+3A_groups">Groups</code></td>
<td>
<p>Indicate the number of groups of single cells.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric (student t-test/ANOVA) or non-parameteric (Wilcoxon rank-sum/Kruskal-Wallis) test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for the two groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between the groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Two_Groups_small.rds", package="VALERIE"))
PlotPSI(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Two_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="wilcox",  multiple.testing="fdr",
  Plots=tempdir(),
  plot.width=5, plot.height=8, EventType="SE", Groups=2)
</code></pre>

<hr>
<h2 id='PlotPSI.A3SS.MultiGroups'>Percent spliced in (PSI) visualization for alternative 3' splice sites (A3SS)</h2><span id='topic+PlotPSI.A3SS.MultiGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.A3SS</code> visualizes percent spliced-in (PSI) for each genomic coordinate for alternative 3' split site (A3SS) event across groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.A3SS.MultiGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("KW", "ANOVA"), multiple.testing = c("holm",
  "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none"), Plots,
  plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.MultiGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric ANOVA or non-parameteric Kruskal-Wallis test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Multi_Groups_small.rds", package="VALERIE"))
PlotPSI.A3SS.MultiGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Multi_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="KW",  multiple.testing="bonferroni",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.A3SS.TwoGroups'>Percent spliced in (PSI) visualization for alternative 3' splice sites (A3SS)</h2><span id='topic+PlotPSI.A3SS.TwoGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.A3SS</code> visualizes percent spliced-in (PSI) for each genomic coordinate for alternative 3' split site (A3SS) event across two groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.A3SS.TwoGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("wilcox", "t.test"),
  multiple.testing = c("holm", "hochberg", "hommel", "bonferroni", "BH",
  "BY", "fdr", "none"), Plots, plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between two groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A3SS.TwoGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line group to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric student t-test or non-parameteric Wilcoxon rank-sum test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for two groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between two groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Two_Groups_small.rds", package="VALERIE"))
PlotPSI.A3SS.TwoGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Two_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="wilcox",  multiple.testing="bonferroni",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.A5SS.MultiGroups'>Percent spliced-in (PSI) visualization for alternative 5' splice sites (A5SS)</h2><span id='topic+PlotPSI.A5SS.MultiGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.A5SS</code> visualizes percent spliced-in (PSI) for each genomic coordinate for alternative 5' splice site (A5SS) event across groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.A5SS.MultiGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("KW", "ANOVA"), multiple.testing = c("holm",
  "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none"), Plots,
  plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.MultiGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric ANOVA or non-parameteric Kruskal-Wallis test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Multi_Groups_small.rds", package="VALERIE"))
PlotPSI.A5SS.MultiGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Multi_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="KW",  multiple.testing="none",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.A5SS.TwoGroups'>Percent spliced-in (PSI) visualization for alternative 5' splice sites (A5SS)</h2><span id='topic+PlotPSI.A5SS.TwoGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.A5SS</code> visualizes percent spliced-in (PSI) for each genomic coordinate for alternative 5' splice site (A5SS) event across two groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.A5SS.TwoGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("wilcox", "t.test"),
  multiple.testing = c("holm", "hochberg", "hommel", "bonferroni", "BH",
  "BY", "fdr", "none"), Plots, plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between two groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.A5SS.TwoGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric student t-test or non-parameteric Wilcoxon rank-sum test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for two groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between two groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Two_Groups_small.rds", package="VALERIE"))
PlotPSI.A5SS.TwoGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Two_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="wilcox",  multiple.testing="fdr",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.MXE.MultiGroups'>Percent spliced-in (PSI) visualization for mutually exclusive exons (MXE)</h2><span id='topic+PlotPSI.MXE.MultiGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.MXE</code> visualizes percent spliced-in (PSI) for each genomic coordinate for mutually exclusive exons (MXE) event across groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.MXE.MultiGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("KW", "ANOVA"), multiple.testing = c("holm",
  "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none"), Plots,
  plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.MultiGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric ANOVA or non-parameteric Kruskal-Wallis test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Multi_Groups_small.rds", package="VALERIE"))
PlotPSI.MXE.MultiGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Multi_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="KW",  multiple.testing="bonferroni",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.MXE.TwoGroups'>Percent spliced-in (PSI) visualization for mutually exclusive exons (MXE)</h2><span id='topic+PlotPSI.MXE.TwoGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.MXE</code> visualizes percent spliced-in (PSI) for each genomic coordinate for mutually exclusive exons (MXE) event across two groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.MXE.TwoGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("wilcox", "t.test"),
  multiple.testing = c("holm", "hochberg", "hommel", "bonferroni", "BH",
  "BY", "fdr", "none"), Plots, plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between two groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.MXE.TwoGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric student t-test or non-parameteric Wilcoxon rank-sum test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for two groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between two groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Two_Groups_small.rds", package="VALERIE"))
PlotPSI.MXE.TwoGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Two_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="wilcox",  multiple.testing="fdr",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.RI.MultiGroups'>Percent spliced-in (PSI) visualization for retained introns (RI)</h2><span id='topic+PlotPSI.RI.MultiGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.RI</code> visualizes percent spliced-in (PSI) for each genomic coordinate for retained intron (RI) event across groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.RI.MultiGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("KW", "ANOVA"), multiple.testing = c("holm",
  "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none"), Plots,
  plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.MultiGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric ANOVA or non-parameteric Kruskal-Wallis test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Multi_Groups_small.rds", package="VALERIE"))
PlotPSI.RI.MultiGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Multi_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="KW",  multiple.testing="none",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.RI.TwoGroups'>Percent spliced-in (PSI) visualization for retained introns (RI)</h2><span id='topic+PlotPSI.RI.TwoGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.RI</code> visualizes percent spliced-in (PSI) for each genomic coordinate for retained intron (RI) event across two groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.RI.TwoGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("wilcox", "t.test"),
  multiple.testing = c("holm", "hochberg", "hommel", "bonferroni", "BH",
  "BY", "fdr", "none"), Plots, plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between two groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.RI.TwoGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric student t-test or non-parameteric Wilcoxon rank-sum test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for two groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between two groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Two_Groups_small.rds", package="VALERIE"))
PlotPSI.RI.TwoGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Two_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="t.test",  multiple.testing="none",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.SE.MultiGroups'>Percent spliced-in (PSI) visualization for skipped exons (SE)</h2><span id='topic+PlotPSI.SE.MultiGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.SE</code> visualizes percent spliced-in (PSI) for each genomic coordinate for skipped exon (SE) event across groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.SE.MultiGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("KW", "ANOVA"), multiple.testing = c("holm",
  "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none"), Plots,
  plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1, Group2, etc. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, and A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between more groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.MultiGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric ANOVA or non-parameteric Kruskal-Wallis test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for the groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Multi_Groups_small.rds", package="VALERIE"))
PlotPSI.SE.MultiGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Multi_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="KW",  multiple.testing="bonferroni",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

<hr>
<h2 id='PlotPSI.SE.TwoGroups'>Percent spliced-in (PSI) visualization for skipped exons (SE)</h2><span id='topic+PlotPSI.SE.TwoGroups'></span>

<h3>Description</h3>

<p><code>PlotPSI.SE</code> visualizes percent spliced-in (PSI) for each genomic coordinate for skipped exon (SE) event across two groups of single cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPSI.SE.TwoGroups(object, SampleInfo, ExonInfo,
  statistical.test = c("wilcox", "t.test"),
  multiple.testing = c("holm", "hochberg", "hommel", "bonferroni", "BH",
  "BY", "fdr", "none"), Plots, plot.width, plot.height)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_object">object</code></td>
<td>
<p>Object of class rehab generated using <code>ComputePSI</code>.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_sampleinfo">SampleInfo</code></td>
<td>
<p>Tab-delimited file describing the naming and grouping of the single cells. First column should contain the names of the binary alignment map (BAM) files. Second column indicates the grouping for each single cell, namely Group1 and Group2. Third column indicates the group names. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_exoninfo">ExonInfo</code></td>
<td>
<p>Tab-delimited file describing the alternative splicing events. First column contains the alternative splicing nomenclature as per BRIE (Huang et al, Genome Biology, 2019) or MISO (Katz et al, Nature Methods, 2010). Second column indicates the type of alternative splicing event, namely SE, MXE, RI, A5SS, and A3SS. Third column contains the gene name or any personal notation. Example file provided in extdata directory of the package.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_statistical.test">statistical.test</code></td>
<td>
<p>Method for comparising PSI values at each genomic coordinates between two groups of single cells.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_multiple.testing">multiple.testing</code></td>
<td>
<p>Method for adjusting p-values for multiple comparisons.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_plots">Plots</code></td>
<td>
<p>Folder to output PSI plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_plot.width">plot.width</code></td>
<td>
<p>Width of outplot plots.</p>
</td></tr>
<tr><td><code id="PlotPSI.SE.TwoGroups_+3A_plot.height">plot.height</code></td>
<td>
<p>Height of outplot plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function visualizes the percent spliced-in (PSI) at each genomic coordinate encompassing the alternative exon and its flanking constitutive exons for each single cell in the form of a heatmap. The PSI mean for the respective groups are also display in the form of a line graph to summarize the PSI distributions of the respective groups. Pair-wise comparison of PSI at each genomic coordinate is performed using either the parametric student t-test or non-parameteric Wilcoxon rank-sum test. The p-values can be adjusted for multiple testing using the <code>p.adjust</code> function.
</p>


<h3>Value</h3>

<p>For each alternative splicing event, a single plot consisting of three subplots arranged from top to bottom is returned. Bottom subplot is a line graph of PSI means at each genomic coordinate for two groups of single cells. Middle subplot is a line graph of p-values corresponding to the comparison of PSI values at each genomic coordinate between two groups of single cells. Top subplot is a heatmap of PSI values at each genomic coordinate across all single cells. Location of plots as per specified in the <code>Plots</code> argument.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PSI &lt;- readRDS(system.file("extdata/PSI", "PSI_RED_Two_Groups_small.rds", package="VALERIE"))
PlotPSI.SE.TwoGroups(PSI, SampleInfo=system.file("extdata/Sample_Info",
  "Sample_Info_RED_Two_Groups.txt", package="VALERIE"),
  ExonInfo=system.file("extdata/Exon_Info", "Exon_Info_RED_small.txt", package="VALERIE"),
  statistical.test="wilcox",  multiple.testing="fdr",
  Plots=tempdir(), plot.width=5, plot.height=8)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
