<!DOCTYPE html><html lang="en"><head><title>Help for package VAM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {VAM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#VAM-package'>
<p>Variance-Adjusted Mahalanobis</p></a></li>
<li><a href='#createGeneSetCollection'><p>Utility function to help create gene set collection list object</p></a></li>
<li><a href='#vam'><p>Variance-adjusted Mahalanobis (VAM) algorithm</p></a></li>
<li><a href='#vamForCollection'><p>VAM method for multiple gene sets</p></a></li>
<li><a href='#vamForSeurat'><p>VAM wrapper for scRNA-seq data processed using the Seurat framework</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Variance-Adjusted Mahalanobis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>H. Robert Frost</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>H. Robert Frost &lt;rob.frost@dartmouth.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains logic for cell-specific gene set scoring of single cell RNA sequencing data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0), MASS, Matrix</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Seurat (&ge; 4.0.0), SeuratObject (&ge; 4.0.0), sctransform (&ge;
0.3.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Copyright:</td>
<td>Dartmouth College</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-05 16:02:41 UTC; d37329b</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-05 16:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='VAM-package'>
Variance-Adjusted Mahalanobis
</h2><span id='topic+VAM-package'></span>

<h3>Description</h3>

<p>Implementation of Variance-adjusted Mahalanobis (VAM), a method for cell-specific gene set scoring of scRNA-seq data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> VAM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.0 </td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2021</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Note</h3>

<p>This work was supported by the National Institutes of Health grants K01LM012426, R21CA253408, P20GM130454 and P30CA023108.
</p>


<h3>Author(s)</h3>

<p>H. Robert Frost
</p>


<h3>References</h3>


<ul>
<li><p> Frost, H. R. (2020). Variance-adjusted Mahalanobis (VAM): a fast and accurate method for cell-specific gene set scoring. biorXiv e-prints. doi: https://doi.org/10.1101/2020.02.18.954321 
</p>
</li></ul>


<hr>
<h2 id='createGeneSetCollection'>Utility function to help create gene set collection list object</h2><span id='topic+createGeneSetCollection'></span>

<h3>Description</h3>

<p>Utility function that creates a gene set collection list in the format required by vamForCollection() given the gene IDs measured in the expression matrix and a list of gene sets as defined by the IDs of the member genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    createGeneSetCollection(gene.ids, gene.set.collection, min.size=1, max.size)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="createGeneSetCollection_+3A_gene.ids">gene.ids</code></td>
<td>
<p>Vector of gene IDs. This should correspond to the genes measured in the gene expression data.</p>
</td></tr>
<tr><td><code id="createGeneSetCollection_+3A_gene.set.collection">gene.set.collection</code></td>
<td>
<p>List of gene sets where each element in the list corresponds to a gene set and the list element is a vector of gene IDs. List names are gene set names. Must contain at least one gene set.</p>
</td></tr>
<tr><td><code id="createGeneSetCollection_+3A_min.size">min.size</code></td>
<td>
<p>Minimum gene set size after filtering out genes not in the gene.ids vector. Gene sets whose post-filtering size is below this are removed from the final collection list. Default is 1 and cannot be set to less than 1.</p>
</td></tr>
<tr><td><code id="createGeneSetCollection_+3A_max.size">max.size</code></td>
<td>
<p>Maximum gene set size after filtering out genes not in the gene.ids vector. Gene sets whose post-filtering size is above this are removed from the final collection list. If not specified, no filtering is performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Version of the input gene.set.collection list where gene IDs have been replaced by position indices, genes not present in the gene.ids vector have been removed and gene sets failing the min/max size constraints have been removed.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vam">vam</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Create a collection with two sets defined over 3 genes
    createGeneSetCollection(gene.ids=c("A", "B", "C"),
        gene.set.collection = list(set1=c("A", "B"), set2=c("B", "C")),
        min.size=2, max.size=3)                    
</code></pre>

<hr>
<h2 id='vam'>Variance-adjusted Mahalanobis (VAM) algorithm</h2><span id='topic+vam'></span>

<h3>Description</h3>

<p>Implementation of the Variance-adjusted Mahalanobis (VAM) method, which computes distance statistics and one-sided p-values 
for all cells in the specified single cell gene expression matrix. This matrix should reflect the subset of the full 
expression profile that corresponds to a single gene set. The p-values will be computed using either a 
chi-square distribution, a non-central chi-square distribution or gamma distribution as controlled by the
<code>center</code> and <code>gamma</code> arguments for the one-sided alternative hypothesis that the expression values in the 
cell are further from the mean (<code>center=T</code>) or origin (<code>center=F</code>) than expected under the null 
of uncorrelated technical noise, i.e., gene expression variance is purely technical and all genes are uncorrelated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    vam(gene.expr, tech.var.prop, gene.weights, center=FALSE, gamma=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vam_+3A_gene.expr">gene.expr</code></td>
<td>
<p>An n x p matrix of gene expression values for n cells and p genes.</p>
</td></tr> 
<tr><td><code id="vam_+3A_tech.var.prop">tech.var.prop</code></td>
<td>
<p>Vector of technical variance proportions for each of the p genes. 
If specified, the Mahalanobis distance will be computed using a diagonal covariance matrix generated using these proportions. 
If not specified, the Mahalanobis distances will be computed using a diagonal covariance matrix generated from the 
sample variances.</p>
</td></tr>
<tr><td><code id="vam_+3A_gene.weights">gene.weights</code></td>
<td>
<p>Optional vector of gene weights. If specified, weights must be &gt; 0. 
The weights are used to adjust the gene variance values included in the computation of the modified Mahalanobis distances.
Specifically, the gene variance is divided by the gene weight. This adjustment means that large 
weights will increase the influence of a given gene in the computation of the modified Mahalanobis distance.</p>
</td></tr>    
<tr><td><code id="vam_+3A_center">center</code></td>
<td>
<p>If true, will mean center the values in the computation of the Mahalanobis statistic.
If false, will compute the Mahalanobis distance from the origin. Default is F.</p>
</td></tr>
<tr><td><code id="vam_+3A_gamma">gamma</code></td>
<td>
<p>If true, will fit a gamma distribution to the non-zero squared Mahalanobis distances computed from 
a row-permuted version of <code>gene.expr</code>. The estimated gamma distribution will be used to compute
a one-sided p-value for each cell. If false, will compute the p-value using the standard 
chi-square approximation for the squared Mahalanobis distance (or non-central if <code>center=F</code>). Default is T.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> with the following elements (row names will match row names from gene.expr):
</p>

<ul>
<li><p> &quot;cdf.value&quot;: 1 minus the one-sided p-values computed from the squared adjusted Mahalanobis distances. 
</p>
</li>
<li><p> &quot;distance.sq&quot;: The squared adjusted Mahalanobis distances for the n cells. 
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+vamForCollection">vamForCollection</a></code>,<code><a href="#topic+vamForSeurat">vamForSeurat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Simulate Poisson expression data for 10 genes and 10 cells
    gene.expr=matrix(rpois(100, lambda=2), nrow=10)
    # Simulate technical variance proportions
    tech.var.prop=runif(10)
    # Execute VAM to compute scores for the 10 genes on each cell
    vam(gene.expr=gene.expr, tech.var.prop=tech.var.prop)
    # Create weights that prioritize the first 5 genes
    gene.weights = c(rep(2,5), rep(1,5))
    # Execute VAM using the weights
    vam(gene.expr=gene.expr, tech.var.prop=tech.var.prop, 
    	gene.weights=gene.weights)    
</code></pre>

<hr>
<h2 id='vamForCollection'>VAM method for multiple gene sets</h2><span id='topic+vamForCollection'></span>

<h3>Description</h3>

<p>Executes the Variance-adjusted Mahalanobis (VAM) method (<code><a href="#topic+vam">vam</a></code>) on multiple gene sets, i.e., a gene set collection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    vamForCollection(gene.expr, gene.set.collection, tech.var.prop, 
        gene.weights, center=FALSE, gamma=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vamForCollection_+3A_gene.expr">gene.expr</code></td>
<td>
<p>An n x p matrix of gene expression values for n cells and p genes.</p>
</td></tr>
<tr><td><code id="vamForCollection_+3A_gene.set.collection">gene.set.collection</code></td>
<td>
<p>List of m gene sets for which scores are computed.
Each element in the list corresponds to a gene set and the list element is a vector
of indices for the genes in the set. The index value is defined relative to the
order of genes in the <code>gene.expr</code> matrix. Gene set names should be specified as list names.</p>
</td></tr>
<tr><td><code id="vamForCollection_+3A_tech.var.prop">tech.var.prop</code></td>
<td>
<p>See description in <code><a href="#topic+vam">vam</a></code></p>
</td></tr> 
<tr><td><code id="vamForCollection_+3A_gene.weights">gene.weights</code></td>
<td>
<p>See description in <code><a href="#topic+vam">vam</a></code>. 
If specified as a single vector of weights, weights must be specified for all p genes and
the same weights are used for all gene sets.
To use different weights for each set, specify as a list of the same length as the 
<code>gene.set.collection</code> list. In this case, each list element should be a vector of 
gene weights of the same length as the size of the corresponding gene set.</p>
</td></tr>
<tr><td><code id="vamForCollection_+3A_center">center</code></td>
<td>
<p>See description in <code><a href="#topic+vam">vam</a></code></p>
</td></tr> 
<tr><td><code id="vamForCollection_+3A_gamma">gamma</code></td>
<td>
<p>See description in <code><a href="#topic+vam">vam</a></code></p>
</td></tr> 
</table>


<h3>Value</h3>

<p>A list containing two elements:
</p>

<ul>
<li><p> &quot;cdf.value&quot;: n x m matrix of 1 minus the one-sided p-values for the m gene sets and n cells. 
</p>
</li>
<li><p> &quot;distance.sq&quot;: n x m matrix of squared adjusted Mahalanobis distances for the m gene sets and n cells. 
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+vam">vam</a></code>,<code><a href="#topic+vamForSeurat">vamForSeurat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Simulate Poisson expression data for 10 genes and 10 cells
    gene.expr=matrix(rpois(100, lambda=2), nrow=10)
    # Simulate technical variance proportions
    tech.var.prop=runif(10)
    # Define a collection with two disjoint sets that span the 10 genes
    collection=list(set1=1:5, set2=6:10)    
    # Execute VAM on both sets using default values for center and gamma
    vamForCollection(gene.expr=gene.expr, gene.set.collection=collection,
        tech.var.prop=tech.var.prop)
    # Create weights that prioritize the first 2 genes for the first set 
    # and the last 2 genes for the second set
    gene.weights = list(c(2,2,1,1,1),c(1,1,1,2,2))
    # Execute VAM using the weights
    vamForCollection(gene.expr=gene.expr, gene.set.collection=collection,
        tech.var.prop=tech.var.prop, gene.weights=gene.weights)
</code></pre>

<hr>
<h2 id='vamForSeurat'>VAM wrapper for scRNA-seq data processed using the Seurat framework</h2><span id='topic+vamForSeurat'></span>

<h3>Description</h3>

<p>Executes the Variance-adjusted Mahalanobis (VAM) method (<code><a href="#topic+vamForCollection">vamForCollection</a></code>) on 
normalized scRNA-seq data stored in a Seurat object.
If the Seurat <code>NormalizeData</code> method was used for normalization, the technical variance of each gene is computed as
the proportion of technical variance (from <code>FindVariableFeatures</code>) multiplied by the variance of the  normalized counts. 
If <code>SCTransform</code> was used for normalization, the technical variance for each gene is set
to 1 (the normalized counts output by <code>SCTransform</code> should have variance 1 if there is only technical variation).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    vamForSeurat(seurat.data, gene.weights, gene.set.collection, 
    	center=FALSE, gamma=TRUE, sample.cov=FALSE, return.dist=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vamForSeurat_+3A_seurat.data">seurat.data</code></td>
<td>
<p>The Seurat object that holds the scRNA-seq data. Assumes normalization has already been performed.</p>
</td></tr>
<tr><td><code id="vamForSeurat_+3A_gene.weights">gene.weights</code></td>
<td>
<p>See description in <code><a href="#topic+vamForCollection">vamForCollection</a></code></p>
</td></tr> 
<tr><td><code id="vamForSeurat_+3A_gene.set.collection">gene.set.collection</code></td>
<td>
<p>List of m gene sets for which scores are computed.
Each element in the list corresponds to a gene set and the list element is a vector
of indices for the genes in the set. The index value is defined relative to the
order of genes in the relevant <code>seurat.data</code> Assay object. Gene set names should be specified as list names.</p>
</td></tr>
<tr><td><code id="vamForSeurat_+3A_center">center</code></td>
<td>
<p>See description in <code><a href="#topic+vam">vam</a></code></p>
</td></tr> 
<tr><td><code id="vamForSeurat_+3A_gamma">gamma</code></td>
<td>
<p>See description in <code><a href="#topic+vam">vam</a></code></p>
</td></tr> 
<tr><td><code id="vamForSeurat_+3A_sample.cov">sample.cov</code></td>
<td>
<p>If true, will use the a diagonal covariance matrix generated from the 
sample variances to compute the squared adjusted Mahalanobis distances (this is equivalent to not specifying
<code>tech.var</code> for the <code><a href="#topic+vam">vam</a></code> method). If false (default), will use the technical variances
as determined based on the type of Seurat normalization.</p>
</td></tr>
<tr><td><code id="vamForSeurat_+3A_return.dist">return.dist</code></td>
<td>
<p>If true, will return the squared adjusted Mahalanobis distances in a new Assay object called &quot;VAM.dist&quot;. 
Default is F.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Updated Seurat object that hold the VAM results in one or two new Assay objects:        
</p>

<ul>
<li><p> If <code>return.dist</code> is true, the matrix of squared adjusted Mahalanobis distances will be stored in new 
Assay object called &quot;VAM.dist&quot;. 
</p>
</li>
<li><p> The matrix of CDF values (1 minus the one-sided p-values) will be stored in new Assay object called &quot;VAM.cdf&quot;. 
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+vam">vam</a></code>,<code><a href="#topic+vamForCollection">vamForCollection</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>    # Only run example code if Seurat package is available
    if (requireNamespace("Seurat", quietly=TRUE) &amp; requireNamespace("SeuratObject", quietly=TRUE)) {
        # Define a collection with one gene set for the first 10 genes
        collection=list(set1=1:10)
        # Execute on the pbmc_small scRNA-seq data set included with SeuratObject
        # See vignettes for more detailed Seurat examples
        vamForSeurat(seurat.data=SeuratObject::pbmc_small,
            gene.set.collection=collection)
    }
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
