<!DOCTYPE html><html><head><title>Help for package vcfppR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {vcfppR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#vcfppR-package'><p>vcfppR: Rapid Manipulation of the Variant Call Format (VCF)</p></a></li>
<li><a href='#heterozygosity'><p>calculate the number of heterozygous SNPs for each sample</p></a></li>
<li><a href='#vcfcomp'><p>Compare two VCF/BCF files reporting various statistics</p></a></li>
<li><a href='#vcfpopgen'><p>count the heterozygous sites per sample in the VCF/BCF</p></a></li>
<li><a href='#vcfreader'><p>API for manipulating the VCF/BCF.</p></a></li>
<li><a href='#vcfsummary'><p>summarize the various variant types at both variant level and sample level.</p></a></li>
<li><a href='#vcftable'><p>read VCF/BCF contents into R data structure</p></a></li>
<li><a href='#vcfwriter'><p>API for writing the VCF/BCF.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Rapid Manipulation of the Variant Call Format (VCF)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.5</td>
</tr>
<tr>
<td>Copyright:</td>
<td>See the file COPYRIGHTS for various htslib copyright details</td>
</tr>
<tr>
<td>Description:</td>
<td>The 'vcfpp.h' (<a href="https://github.com/Zilong-Li/vcfpp">https://github.com/Zilong-Li/vcfpp</a>) provides an easy-to-use 'C++' 'API' of 'htslib', offering full functionality for manipulating Variant Call Format (VCF) files. The 'vcfppR' package serves as the R bindings of the 'vcfpp.h' library, enabling rapid processing of both compressed and uncompressed VCF files. Explore a range of powerful features for efficient VCF data manipulation.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>libcurl: libcurl-devel (rpm) or
libcurl4-openssl-dev (deb), GNU make.</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, methods, stats, utils</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Zilong-Li/vcfppR">https://github.com/Zilong-Li/vcfppR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Zilong-Li/vcfppR/issues">https://github.com/Zilong-Li/vcfppR/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-08 05:32:15 UTC; rlk420</td>
</tr>
<tr>
<td>Author:</td>
<td>Zilong Li <a href="https://orcid.org/0000-0001-5859-2078"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Bonfield, James K and Marshall, John and Danecek, Petr and Li, Heng and
    Ohan, Valeriu and Whitwham, Andrew and Keane, Thomas and Davies,
    Robert M [cph] (Authors of included htslib library)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zilong Li &lt;zilong.dk@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-08 07:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='vcfppR-package'>vcfppR: Rapid Manipulation of the Variant Call Format (VCF)</h2><span id='topic+vcfppR'></span><span id='topic+vcfppR-package'></span>

<h3>Description</h3>

<p>The 'vcfpp.h' (<a href="https://github.com/Zilong-Li/vcfpp">https://github.com/Zilong-Li/vcfpp</a>) provides an easy-to-use 'C++' 'API' of 'htslib', offering full functionality for manipulating Variant Call Format (VCF) files. The 'vcfppR' package serves as the R bindings of the 'vcfpp.h' library, enabling rapid processing of both compressed and uncompressed VCF files. Explore a range of powerful features for efficient VCF data manipulation.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Zilong Li <a href="mailto:zilong.dk@gmail.com">zilong.dk@gmail.com</a> (<a href="https://orcid.org/0000-0001-5859-2078">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Bonfield, James K and Marshall, John and Danecek, Petr and Li, Heng and Ohan, Valeriu and Whitwham, Andrew and Keane, Thomas and Davies, Robert M (Authors of included htslib library) [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/Zilong-Li/vcfppR">https://github.com/Zilong-Li/vcfppR</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Zilong-Li/vcfppR/issues">https://github.com/Zilong-Li/vcfppR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='heterozygosity'>calculate the number of heterozygous SNPs for each sample</h2><span id='topic+heterozygosity'></span>

<h3>Description</h3>

<p>calculate the number of heterozygous SNPs for each sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heterozygosity(vcffile, region = "", samples = "-", pass = FALSE, qual = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heterozygosity_+3A_vcffile">vcffile</code></td>
<td>
<p>path to the VCF file with index</p>
</td></tr>
<tr><td><code id="heterozygosity_+3A_region">region</code></td>
<td>
<p>region to extract, default &quot;&quot; for all</p>
</td></tr>
<tr><td><code id="heterozygosity_+3A_samples">samples</code></td>
<td>
<p>samples to extract, default &quot;-&quot; for all</p>
</td></tr>
<tr><td><code id="heterozygosity_+3A_pass">pass</code></td>
<td>
<p>restrict to variants with FILTER==PASS</p>
</td></tr>
<tr><td><code id="heterozygosity_+3A_qual">qual</code></td>
<td>
<p>restrict to variants with QUAL &gt; qual.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of heterozygosity couts for each sample along with its id in the vcf header
</p>

<hr>
<h2 id='vcfcomp'>Compare two VCF/BCF files reporting various statistics</h2><span id='topic+vcfcomp'></span>

<h3>Description</h3>

<p>Compare two VCF/BCF files reporting various statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcfcomp(
  test,
  truth,
  stats = "all",
  formats = c("DS", "GT"),
  by.sample = FALSE,
  by.variant = FALSE,
  flip = FALSE,
  names = NULL,
  bins = NULL,
  af = NULL,
  out = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcfcomp_+3A_test">test</code></td>
<td>
<p>path to the first VCF/BCF file referred as test.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_truth">truth</code></td>
<td>
<p>path to the second VCF/BCF file referred as truth, or saved RDS file.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_stats">stats</code></td>
<td>
<p>the statistics to be calculated. supports the following.
&quot;r2&quot;: pearson correlation coefficient ** 2.
&quot;f1&quot;: F1-score, good balance between sensitivity and precision.
&quot;nrc&quot;: Non-Reference Concordance rate</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_formats">formats</code></td>
<td>
<p>character vector. the FORMAT tags to extract for the test and truth respectively.
default c(&quot;DS&quot;, &quot;GT&quot;) extracts 'DS' of the target and 'GT' of the truth.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_by.sample">by.sample</code></td>
<td>
<p>logical. calculate concordance for each samples, then average by bins.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_by.variant">by.variant</code></td>
<td>
<p>logical. calculate concordance for each variant, then average by bins.
if both bysample and by variant are TRUE, then do average on all samples first.
if both bysample and by variant are FALSE, then do average on all samples and variants.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_flip">flip</code></td>
<td>
<p>logical. flip the ref and alt variants</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_names">names</code></td>
<td>
<p>character vector. reset samples' names in the test VCF.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_bins">bins</code></td>
<td>
<p>numeric vector. break statistics into allele frequency bins.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_af">af</code></td>
<td>
<p>file path to allele frequency text file or saved RDS file.</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_out">out</code></td>
<td>
<p>output prefix for saving objects into RDS file</p>
</td></tr>
<tr><td><code id="vcfcomp_+3A_...">...</code></td>
<td>
<p>options passed to <code>vcftable</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>vcfcomp</code> implements various statisitcs to compare two VCF/BCF files,
e.g. report genotype concocrdance, correlation stratified by allele frequency.
</p>


<h3>Value</h3>

<p>a list of various statistics
</p>


<h3>Author(s)</h3>

<p>Zilong Li <a href="mailto:zilong.dk@gmail.com">zilong.dk@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('vcfppR')
test &lt;- system.file("extdata", "imputed.gt.vcf.gz", package="vcfppR")
truth &lt;- system.file("extdata", "imputed.gt.vcf.gz", package="vcfppR")
samples &lt;- "HG00133,HG00143,HG00262"
res &lt;- vcfcomp(test, truth, stats="f1", format=c('GT','GT'), samples=samples)
str(res)
</code></pre>

<hr>
<h2 id='vcfpopgen'>count the heterozygous sites per sample in the VCF/BCF</h2><span id='topic+vcfpopgen'></span>

<h3>Description</h3>

<p>count the heterozygous sites per sample in the VCF/BCF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcfpopgen(
  vcffile,
  region = "",
  samples = "-",
  pass = FALSE,
  qual = 0,
  fun = "heterozygosity"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcfpopgen_+3A_vcffile">vcffile</code></td>
<td>
<p>path to the VCF/BCF file</p>
</td></tr>
<tr><td><code id="vcfpopgen_+3A_region">region</code></td>
<td>
<p>region to subset like bcftools</p>
</td></tr>
<tr><td><code id="vcfpopgen_+3A_samples">samples</code></td>
<td>
<p>samples to subset like bcftools</p>
</td></tr>
<tr><td><code id="vcfpopgen_+3A_pass">pass</code></td>
<td>
<p>restrict to variants with FILTER==PASS</p>
</td></tr>
<tr><td><code id="vcfpopgen_+3A_qual">qual</code></td>
<td>
<p>restrict to variants with QUAL &gt; qual.</p>
</td></tr>
<tr><td><code id="vcfpopgen_+3A_fun">fun</code></td>
<td>
<p>which popgen function to run. available functions are
&quot;heterozygosity&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>vcfpopgen</code> a list containing the following components:
</p>

<dl>
<dt>samples</dt><dd><p>: character vector; <br />
the samples ids in the VCF file after subsetting
</p>
</dd>
<dt>hets</dt><dd><p>: integer vector; <br />
the counts of heterozygous sites of each sample in the same order as <code>samples</code>
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Zilong Li <a href="mailto:zilong.dk@gmail.com">zilong.dk@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('vcfppR')
vcffile &lt;- system.file("extdata", "raw.gt.vcf.gz", package="vcfppR")
res &lt;- vcfpopgen(vcffile)
str(res)
</code></pre>

<hr>
<h2 id='vcfreader'>API for manipulating the VCF/BCF.</h2><span id='topic+vcfreader'></span>

<h3>Description</h3>

<p>Type the name of the class to see the details and methods
</p>


<h3>Value</h3>

<p>A C++ class with the following fields/methods for manipulating the VCF/BCF
</p>


<h3>Fields</h3>


<dl>
<dt><code>new</code></dt><dd><p>Constructor given a vcf file </p>

<ul>
<li><p> Parameter: vcffile - The path of a vcf file
</p>
</li></ul>
</dd>
<dt><code>new</code></dt><dd><p>Constructor given a vcf file and the region </p>

<ul>
<li><p> Parameter: vcffile - The path of a vcf file
</p>
</li>
<li><p> Parameter: region - The region to be constrained
</p>
</li></ul>
</dd>
<dt><code>new</code></dt><dd><p>Constructor given a vcf file, the region and the samples </p>

<ul>
<li><p> Parameter: vcffile - The path of a vcf file
</p>
</li>
<li><p> Parameter: region - The region to be constrained
</p>
</li>
<li><p> Parameter: samples - The samples to be constrained. Comma separated list of samples to include (or exclude with &quot;^&quot; prefix).
</p>
</li></ul>
</dd>
<dt><code>variant</code></dt><dd><p>Try to get next variant record. return FALSE if there are no more variants or hit the end of file, otherwise TRUE.</p>
</dd>
<dt><code>chr</code></dt><dd><p>Return the CHROM field of current variant</p>
</dd>
<dt><code>pos</code></dt><dd><p>Return the POS field of current variant</p>
</dd>
<dt><code>id</code></dt><dd><p>Return the CHROM field of current variant</p>
</dd>
<dt><code>ref</code></dt><dd><p>Return the REF field of current variant</p>
</dd>
<dt><code>alt</code></dt><dd><p>Return the ALT field of current variant</p>
</dd>
<dt><code>qual</code></dt><dd><p>Return the QUAL field of current variant</p>
</dd>
<dt><code>filter</code></dt><dd><p>Return the FILTER field of current variant</p>
</dd>
<dt><code>info</code></dt><dd><p>Return the INFO field of current variant</p>
</dd>
<dt><code>infoInt</code></dt><dd><p>Return the tag value of integer type in INFO field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in INFO</p>
</li></ul>
</dd>
<dt><code>infoFloat</code></dt><dd><p>Return the tag value of float type in INFO field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in INFO</p>
</li></ul>
</dd>
<dt><code>infoStr</code></dt><dd><p>Return the tag value of string type in INFO field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in INFO</p>
</li></ul>
</dd>
<dt><code>infoIntVec</code></dt><dd><p>Return the tag value in a vector of integer type in INFO field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in INFO</p>
</li></ul>
</dd>
<dt><code>infoFloatVec</code></dt><dd><p>Return the tag value in a vector of float type in INFO field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in INFO</p>
</li></ul>
</dd>
<dt><code>genotypes</code></dt><dd><p>Return the genotype values in a vector of integers  </p>
 <ul>
<li><p> Parameter: collapse - Boolean value indicates wheather to collapse the size of genotypes, eg, return diploid genotypes.</p>
</li></ul>
</dd>
<dt><code>formatInt</code></dt><dd><p>Return the tag value of integer type for each sample in FORAMT field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in FORAMT</p>
</li></ul>
</dd>
<dt><code>formatFloat</code></dt><dd><p>Return the tag value of float type for each sample in FORAMT field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in FORAMT</p>
</li></ul>
</dd>
<dt><code>formatStr</code></dt><dd><p>Return the tag value of string type for each sample in FORAMT field of current variant </p>
 <ul>
<li><p> Parameter: tag - The tag name to retrieve in FORAMT</p>
</li></ul>
</dd>
<dt><code>isSNP</code></dt><dd><p>Test if current variant is exculsively a SNP or not</p>
</dd>
<dt><code>isIndel</code></dt><dd><p>Test if current variant is exculsively a INDEL or not</p>
</dd>
<dt><code>isSV</code></dt><dd><p>Test if current variant is exculsively a SV or not</p>
</dd>
<dt><code>isMultiAllelics</code></dt><dd><p>Test if current variant is exculsively a Multi Allelics or not</p>
</dd>
<dt><code>isMultiAllelicSNP</code></dt><dd><p>Test if current variant is exculsively a Multi Biallelics (SNPs) or not</p>
</dd>
<dt><code>hasSNP</code></dt><dd><p>Test if current variant has a SNP or not</p>
</dd>
<dt><code>hasINDEL</code></dt><dd><p>Test if current variant has a INDEL or not</p>
</dd>
<dt><code>hasINS</code></dt><dd><p>Test if current variant has a INS or not</p>
</dd>
<dt><code>hasDEL</code></dt><dd><p>Test if current variant has a DEL or not</p>
</dd>
<dt><code>hasMNP</code></dt><dd><p>Test if current variant has a MNP or not</p>
</dd>
<dt><code>hasBND</code></dt><dd><p>Test if current variant has a BND or not</p>
</dd>
<dt><code>hasOTHER</code></dt><dd><p>Test if current variant has a OTHER or not</p>
</dd>
<dt><code>hasOVERLAP</code></dt><dd><p>Test if current variant has a OVERLAP or not</p>
</dd>
<dt><code>nsamples</code></dt><dd><p>Return the number of samples</p>
</dd>
<dt><code>samples</code></dt><dd><p>Return a vector of samples id</p>
</dd>
<dt><code>header</code></dt><dd><p>Return the raw string of the vcf header</p>
</dd>
<dt><code>string</code></dt><dd><p>Return the raw string of current variant including newline</p>
</dd>
<dt><code>line</code></dt><dd><p>Return the raw string of current variant without newline</p>
</dd>
<dt><code>output</code></dt><dd><p>Init an output object for streaming out the variants to another vcf</p>
</dd>
<dt><code>write</code></dt><dd><p>Streaming out current variant the output vcf</p>
</dd>
<dt><code>close</code></dt><dd><p>Close the connection to the output vcf</p>
</dd>
<dt><code>setCHR</code></dt><dd><p>Modify the CHR of current variant </p>
 <ul>
<li><p> Parameter: s - A string for CHR</p>
</li></ul>
</dd>
<dt><code>setID</code></dt><dd><p>Modify the ID of current variant </p>
 <ul>
<li><p> Parameter: s - A string for ID</p>
</li></ul>
</dd>
<dt><code>setPOS</code></dt><dd><p>Modify the POS of current variant </p>
 <ul>
<li><p> Parameter: pos - An integer for POS</p>
</li></ul>
</dd>
<dt><code>setRefAlt</code></dt><dd><p>Modify the REF and ALT of current variant </p>
 <ul>
<li><p> Parameter: s - A string reperated by comma</p>
</li></ul>
</dd>
<dt><code>setInfoInt</code></dt><dd><p>Modify the given tag of INT type in the INFO of current variant
</p>

<ul>
<li><p> Parameter: tag - A string for the tag name
</p>
</li>
<li><p> Parameter: v - An integer for the tag value</p>
</li></ul>
</dd>
<dt><code>setInfoFloat</code></dt><dd><p>Modify the given tag of FLOAT type in the INFO of current variant
</p>

<ul>
<li><p> Parameter: tag - A string for the tag name
</p>
</li>
<li><p> Parameter: v - A double for the tag value</p>
</li></ul>
</dd>
<dt><code>setInfoStr</code></dt><dd><p>Modify the given tag of STRING type in the INFO of current variant
</p>

<ul>
<li><p> Parameter: tag - A string for the tag name
</p>
</li>
<li><p> Parameter: s - A string for the tag value</p>
</li></ul>
</dd>
<dt><code>setPhasing</code></dt><dd><p>Modify the phasing status of each sample
</p>
<ul>
<li><p> Parameter: v - An integer vector with size of the number of samples. only 1s and 0s are valid.</p>
</li></ul>
</dd>
<dt><code>setGenotypes</code></dt><dd><p>Modify the genotypes of current variant
</p>
<ul>
<li><p> Parameter: v - An integer vector for genotypes. Use NA or -9 for missing value.</p>
</li></ul>
</dd>
<dt><code>setFormatInt</code></dt><dd><p>Modify the given tag of INT type in the FORMAT of current variant
</p>

<ul>
<li><p> Parameter: tag - A string for the tag name
</p>
</li>
<li><p> Parameter: v - An integer for the tag value</p>
</li></ul>
</dd>
<dt><code>setFormatFloat</code></dt><dd><p>Modify the given tag of FLOAT type in the FORMAT of current variant
</p>

<ul>
<li><p> Parameter: tag - A string for the tag name
</p>
</li>
<li><p> Parameter: v - A double for the tag value</p>
</li></ul>
</dd>
<dt><code>setFormatStr</code></dt><dd><p>Modify the given tag of STRING type in the FORMAT of current variant
</p>

<ul>
<li><p> Parameter: tag - A string for the tag name
</p>
</li>
<li><p> Parameter: s - A string for the tag value</p>
</li></ul>
</dd>
<dt><code>rmInfoTag</code></dt><dd><p>Remove the given tag from the INFO of current variant
</p>
<ul>
<li><p> Parameter: s - A string for the tag name</p>
</li></ul>
</dd>
<dt><code>rmFormatTag</code></dt><dd><p>Remove the given tag from the FORMAT of current variant
</p>
<ul>
<li><p> Parameter: s - A string for the tag name</p>
</li></ul>
</dd>
<dt><code>setVariant</code></dt><dd><p>Modify current variant by adding a vcf line
</p>
<ul>
<li><p> Parameter: s - A string for one line in the VCF</p>
</li></ul>
</dd>
<dt><code>addINFO</code></dt><dd><p>Add a INFO in the header of the vcf
</p>

<ul>
<li><p> Parameter: id - A string for the tag name
</p>
</li>
<li><p> Parameter: number - A string for the number
</p>
</li>
<li><p> Parameter: type - A string for the type
</p>
</li>
<li><p> Parameter: desc - A string for description of what it means</p>
</li></ul>
</dd>
<dt><code>addFORMAT</code></dt><dd><p>Add a FORMAT in the header of the vcf
</p>

<ul>
<li><p> Parameter: id - A string for the tag name
</p>
</li>
<li><p> Parameter: number - A string for the number
</p>
</li>
<li><p> Parameter: type - A string for the type
</p>
</li>
<li><p> Parameter: desc - A string for description of what it means</p>
</li></ul>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>vcffile &lt;- system.file("extdata", "raw.gt.vcf.gz", package="vcfppR")
br &lt;- vcfreader$new(vcffile)
res &lt;- rep(0L, br$nsamples())
while(br$variant()) {
  if(br$isSNP()) {
  gt &lt;- br$genotypes(TRUE) == 1
  gt[is.na(gt)] &lt;- FALSE
  res &lt;- res + gt
  }
}
</code></pre>

<hr>
<h2 id='vcfsummary'>summarize the various variant types at both variant level and sample level.</h2><span id='topic+vcfsummary'></span>

<h3>Description</h3>

<p>summarize the various variant types at both variant level and sample level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcfsummary(
  vcffile,
  region = "",
  samples = "-",
  pass = FALSE,
  qual = 0,
  svtype = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcfsummary_+3A_vcffile">vcffile</code></td>
<td>
<p>path to the VCF/BCF file</p>
</td></tr>
<tr><td><code id="vcfsummary_+3A_region">region</code></td>
<td>
<p>region to subset like bcftools</p>
</td></tr>
<tr><td><code id="vcfsummary_+3A_samples">samples</code></td>
<td>
<p>samples to subset like bcftools</p>
</td></tr>
<tr><td><code id="vcfsummary_+3A_pass">pass</code></td>
<td>
<p>restrict to variants with FILTER==PASS</p>
</td></tr>
<tr><td><code id="vcfsummary_+3A_qual">qual</code></td>
<td>
<p>restrict to variants with QUAL &gt; qual.</p>
</td></tr>
<tr><td><code id="vcfsummary_+3A_svtype">svtype</code></td>
<td>
<p>summarize the variants with SVTYPE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>bcftools view -s &quot;id01,id02&quot; input.bcf.gz chr1:100000-20000
</p>


<h3>Value</h3>

<p><code>vcfsummary</code> a list containing the following components:
</p>

<dl>
<dt>summary</dt><dd><p>: named integer vector; <br />
summarize the counts of each variant type
</p>
</dd>
<dt>samples</dt><dd><p>: character vector; <br />
the samples ids in the VCF file after subsetting
</p>
</dd>
<dt>vartype</dt><dd><p>: integer vector; <br />
the counts of the variant type at sample level in the same order as <code>samples</code>
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Zilong Li <a href="mailto:zilong.dk@gmail.com">zilong.dk@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('vcfppR')
svfile &lt;- system.file("extdata", "sv.vcf.gz", package="vcfppR")
res &lt;- vcfsummary(svfile, region = "chr21:1-10000000", svtype = TRUE)
str(res)
</code></pre>

<hr>
<h2 id='vcftable'>read VCF/BCF contents into R data structure</h2><span id='topic+vcftable'></span>

<h3>Description</h3>

<p>The swiss army knife for reading VCF/BCF into R data types rapidly and easily.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcftable(
  vcffile,
  region = "",
  samples = "-",
  vartype = "all",
  format = "GT",
  ids = NULL,
  qual = 0,
  pass = FALSE,
  info = TRUE,
  collapse = TRUE,
  setid = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcftable_+3A_vcffile">vcffile</code></td>
<td>
<p>path to the VCF/BCF file</p>
</td></tr>
<tr><td><code id="vcftable_+3A_region">region</code></td>
<td>
<p>region to subset in bcftools-like style: &quot;chr1&quot;, &quot;chr1:1-10000000&quot;</p>
</td></tr>
<tr><td><code id="vcftable_+3A_samples">samples</code></td>
<td>
<p>samples to subset in bcftools-like style.
comma separated list of samples to include (or exclude with &quot;^&quot; prefix).
e.g. &quot;id01,id02&quot;, &quot;^id01,id02&quot;.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_vartype">vartype</code></td>
<td>
<p>restrict to specific type of variants. supports &quot;snps&quot;,&quot;indels&quot;, &quot;sv&quot;, &quot;multisnps&quot;,&quot;multiallelics&quot;</p>
</td></tr>
<tr><td><code id="vcftable_+3A_format">format</code></td>
<td>
<p>the FORMAT tag to extract. default &quot;GT&quot; is extracted.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_ids">ids</code></td>
<td>
<p>character vector. restrict to sites with ID in the given vector. default NULL won't filter any sites.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_qual">qual</code></td>
<td>
<p>numeric. restrict to variants with QUAL &gt; qual.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_pass">pass</code></td>
<td>
<p>logical. restrict to variants with FILTER = &quot;PASS&quot;.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_info">info</code></td>
<td>
<p>logical. drop INFO column in the returned list.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_collapse">collapse</code></td>
<td>
<p>logical. It acts on the FORMAT. If the FORMAT to extract is &quot;GT&quot;, the dim of raw genotypes matrix of diploid is (M, 2 * N),
where M is #markers and N is #samples. default TRUE will collapse the genotypes for each sample such that the matrix is (M, N).
Set this to FALSE if one wants to maintain the phasing order, e.g. &quot;1|0&quot; is parsed as c(1, 0) with collapse=FALSE.
If the FORMAT to extract is not &quot;GT&quot;, then with collapse=TRUE it will try to turn a list of the extracted vector into a matrix.
However, this raises issues when one variant is mutliallelic resulting in more vaules than others.</p>
</td></tr>
<tr><td><code id="vcftable_+3A_setid">setid</code></td>
<td>
<p>logical. reset ID column as CHR_POS_REF_ALT.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>vcftable</code> uses the C++ API of vcfpp, which is a wrapper of htslib, to read VCF/BCF files.
Thus, it has the full functionalities of htslib, such as restrict to specific variant types,
samples and regions. For the memory efficiency reason, the <code>vcftable</code> is designed
to parse only one tag at a time in the FORMAT column of the VCF. In default, only the matrix of genotypes,
i.e. &quot;GT&quot; tag, are returned by <code>vcftable</code>, but there are many other tags supported by the <code>format</code> option.
</p>


<h3>Value</h3>

<p>Return a list containing the following components:
</p>

<dl>
<dt>samples</dt><dd><p>: character vector; <br />
the samples ids in the VCF file after subsetting
</p>
</dd>
<dt>chr</dt><dd><p>: character vector; <br />
the CHR column in the VCF file
</p>
</dd>
<dt>pos</dt><dd><p>: character vector; <br />
the POS column in the VCF file
</p>
</dd>
<dt>id</dt><dd><p>: character vector; <br />
the ID column in the VCF file
</p>
</dd>
<dt>ref</dt><dd><p>: character vector; <br />
the REF column in the VCF file
</p>
</dd>
<dt>alt</dt><dd><p>: character vector; <br />
the ALT column in the VCF file
</p>
</dd>
<dt>qual</dt><dd><p>: character vector; <br />
the QUAL column in the VCF file
</p>
</dd>
<dt>filter</dt><dd><p>: character vector; <br />
the FILTER column in the VCF file
</p>
</dd>
<dt>info</dt><dd><p>: character vector; <br />
the INFO column in the VCF file
</p>
</dd>
<dt>format</dt><dd><p>: matrix of either integer of numberic values depending on the tag to extract; <br />
a specifiy tag in the FORMAT column to be extracted
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Zilong Li <a href="mailto:zilong.dk@gmail.com">zilong.dk@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('vcfppR')
vcffile &lt;- system.file("extdata", "raw.gt.vcf.gz", package="vcfppR")
res &lt;- vcftable(vcffile, "chr21:1-5050000", vartype = "snps")
str(res)
</code></pre>

<hr>
<h2 id='vcfwriter'>API for writing the VCF/BCF.</h2><span id='topic+vcfwriter'></span>

<h3>Description</h3>

<p>Type the name of the class to see the details and methods
</p>


<h3>Value</h3>

<p>A C++ class with the following fields/methods for writing the VCF/BCF
</p>


<h3>Fields</h3>


<dl>
<dt><code>new</code></dt><dd><p>Constructor given a vcf file </p>

<ul>
<li><p> Parameter: vcffile - The path of a vcf file. don't start with &quot;~&quot;
</p>
</li>
<li><p> Parameter: version - The version of VCF specification
</p>
</li></ul>
</dd>
<dt><code>addContig</code></dt><dd><p>Add a Contig in the header of the vcf
</p>
 <ul>
<li><p> Parameter: str - A string for the CONTIG name </p>
</li></ul>
</dd>
<dt><code>addFILTER</code></dt><dd><p>Add a FILTER in the header of the vcf
</p>

<ul>
<li><p> Parameter: id - A string for the FILTER name
</p>
</li>
<li><p> Parameter: desc - A string for description of what it means</p>
</li></ul>
</dd>
<dt><code>addINFO</code></dt><dd><p>Add a INFO in the header of the vcf
</p>

<ul>
<li><p> Parameter: id - A string for the tag name
</p>
</li>
<li><p> Parameter: number - A string for the number
</p>
</li>
<li><p> Parameter: type - A string for the type
</p>
</li>
<li><p> Parameter: desc - A string for description of what it means</p>
</li></ul>
</dd>
<dt><code>addFORMAT</code></dt><dd><p>Add a FORMAT in the header of the vcf
</p>

<ul>
<li><p> Parameter: id - A string for the tag name
</p>
</li>
<li><p> Parameter: number - A string for the number
</p>
</li>
<li><p> Parameter: type - A string for the type
</p>
</li>
<li><p> Parameter: desc - A string for description of what it means</p>
</li></ul>
</dd>
<dt><code>addSample</code></dt><dd><p>Add a SAMPLE in the header of the vcf
</p>
 <ul>
<li><p> Parameter: str - A string for a SAMPLE name </p>
</li></ul>
</dd>
<dt><code>addLine</code></dt><dd><p>Add a line in the header of the vcf
</p>
 <ul>
<li><p> Parameter: str - A string for a line in the header of VCF </p>
</li></ul>
</dd>
<dt><code>writeline</code></dt><dd><p>Write a variant record given a line
</p>
 <ul>
<li><p> Parameter: line - A string for a line in the variant of VCF. Not ended with &quot;newline&quot; </p>
</li></ul>
</dd>
<dt><code>close</code></dt><dd><p>Close and save the vcf file</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>outvcf &lt;- file.path(paste0(tempfile(), ".vcf.gz"))
bw &lt;- vcfwriter$new(outvcf, "VCF4.1")
bw$addContig("chr20")
bw$addFORMAT("GT", "1", "String", "Genotype");
bw$addSample("NA12878")
s1 &lt;- "chr20\t2006060\t.\tG\tC\t100\tPASS\t.\tGT\t1|0"
bw$writeline(s1)
bw$close()
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
