<!DOCTYPE html><html><head><title>Help for package vDiveR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {vDiveR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#vDiveR-package'><p>vDiveR: Visualization of Viral Protein Sequence Diversity Dynamics</p></a></li>
<li><a href='#concat_conserved_kmer'><p>k-mer sequences concatenation</p></a></li>
<li><a href='#extract_from_GISAID'><p>Extract metadata via fasta file from GISAID</p></a></li>
<li><a href='#extract_from_NCBI'><p>Extract metadata via fasta file from ncbi</p></a></li>
<li><a href='#JSON_sample'><p>DiMA (v4.1.1) JSON Output File</p></a></li>
<li><a href='#json2csv'><p>JSON2CSV</p></a></li>
<li><a href='#metadata'><p>Metadata Input Sample</p></a></li>
<li><a href='#metadata_extraction'><p>Metadata Extraction from NCBI/GISAID EpiCoV FASTA file</p></a></li>
<li><a href='#plot_conservationLevel'><p>Conservation Levels Distribution Plot</p></a></li>
<li><a href='#plot_correlation'><p>Entropy and total variant incidence correlation plot</p></a></li>
<li><a href='#plot_dynamics_protein'><p>Dynamics of Diversity Motifs (Protein) Plot</p></a></li>
<li><a href='#plot_dynamics_proteome'><p>Dynamics of Diversity Motifs (Proteome) Plot</p></a></li>
<li><a href='#plot_entropy'><p>Entropy plot</p></a></li>
<li><a href='#plot_time'><p>Time Distribution of Sequences Plot</p></a></li>
<li><a href='#plot_worldmap'><p>Geographical Distribution of Sequences Plot</p></a></li>
<li><a href='#protein_2hosts'><p>DiMA (v4.1.1) JSON converted-CSV Output Sample 2</p></a></li>
<li><a href='#proteins_1host'><p>DiMA (v4.1.1) JSON converted-CSV Output Sample 1</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Visualization of Viral Protein Sequence Diversity Dynamics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>To ease the visualization of outputs from Diversity Motif Analyser ('DiMA';
    <a href="https://github.com/BVU-BILSAB/DiMA">https://github.com/BVU-BILSAB/DiMA</a>). 'vDiveR' allows visualization of the diversity 
    motifs (index and its variants – major, minor and unique) for elucidation of 
    the underlying inherent dynamics. Please refer <a href="https://vdiver-manual.readthedocs.io/en/latest/">https://vdiver-manual.readthedocs.io/en/latest/</a>
    for more information.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>DT, dplyr, gghalves, ggplot2, ggpubr, grid, gridExtra, ggtext,
magrittr, plyr, tidyr, stringr, rlang, rentrez, readr, scales,
maps</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-09 18:39:22 UTC; Kelvin</td>
</tr>
<tr>
<td>Author:</td>
<td>Pendy Tok [aut, cre],
  Li Chuin Chong [aut],
  Evgenia Chikina [aut],
  Yin Cheng Chen [aut],
  Mohammad Asif Khan [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pendy Tok &lt;pendytok0518@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-09 20:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='vDiveR-package'>vDiveR: Visualization of Viral Protein Sequence Diversity Dynamics</h2><span id='topic+vDiveR'></span><span id='topic+vDiveR-package'></span>

<h3>Description</h3>

<p>To ease the visualization of outputs from Diversity Motif Analyser ('DiMA'; <a href="https://github.com/BVU-BILSAB/DiMA">https://github.com/BVU-BILSAB/DiMA</a>). 'vDiveR' allows visualization of the diversity motifs (index and its variants – major, minor and unique) for elucidation of the underlying inherent dynamics. Please refer <a href="https://vdiver-manual.readthedocs.io/en/latest/">https://vdiver-manual.readthedocs.io/en/latest/</a> for more information.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Pendy Tok <a href="mailto:pendytok0518@gmail.com">pendytok0518@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Li Chuin Chong
</p>
</li>
<li><p> Evgenia Chikina
</p>
</li>
<li><p> Yin Cheng Chen
</p>
</li>
<li><p> Mohammad Asif Khan
</p>
</li></ul>


<hr>
<h2 id='concat_conserved_kmer'>k-mer sequences concatenation</h2><span id='topic+concat_conserved_kmer'></span>

<h3>Description</h3>

<p>This function concatenates completely (index incidence = 100
index incidence &lt; 100
k-mer position or are adjacent to each other and generate the CCS/HCS sequence
in either CSv or FASTA format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concat_conserved_kmer(
  data,
  conservation_level = "HCS",
  kmer = 9,
  threshold_pct = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concat_conserved_kmer_+3A_data">data</code></td>
<td>
<p>DiMA JSON converted csv file data</p>
</td></tr>
<tr><td><code id="concat_conserved_kmer_+3A_conservation_level">conservation_level</code></td>
<td>
<p>CCS (completely conserved) / HCS (highly conserved)</p>
</td></tr>
<tr><td><code id="concat_conserved_kmer_+3A_kmer">kmer</code></td>
<td>
<p>size of the k-mer window</p>
</td></tr>
<tr><td><code id="concat_conserved_kmer_+3A_threshold_pct">threshold_pct</code></td>
<td>
<p>manually set threshold of index.incidence for HCS</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list wit csv and fasta dataframes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>csv&lt;-concat_conserved_kmer(proteins_1host)$csv
csv_2hosts&lt;-concat_conserved_kmer(protein_2hosts, conservation_level = "CCS")$csv
fasta &lt;- concat_conserved_kmer(protein_2hosts, conservation_level = "HCS")$fasta
</code></pre>

<hr>
<h2 id='extract_from_GISAID'>Extract metadata via fasta file from GISAID</h2><span id='topic+extract_from_GISAID'></span>

<h3>Description</h3>

<p>This function get the metadata from each header of GISAID fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_from_GISAID(file_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_from_GISAID_+3A_file_path">file_path</code></td>
<td>
<p>path of fasta file</p>
</td></tr>
</table>

<hr>
<h2 id='extract_from_NCBI'>Extract metadata via fasta file from ncbi</h2><span id='topic+extract_from_NCBI'></span>

<h3>Description</h3>

<p>This function get the metadata from each head of fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_from_NCBI(file_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_from_NCBI_+3A_file_path">file_path</code></td>
<td>
<p>path of fasta file</p>
</td></tr>
</table>

<hr>
<h2 id='JSON_sample'>DiMA (v4.1.1) JSON Output File</h2><span id='topic+JSON_sample'></span>

<h3>Description</h3>

<p>A sample DiMA JSON Output File which acts as the input for JSON2CSV()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JSON_sample
</code></pre>


<h3>Format</h3>

<p>A Diversity Motif Analyzer (DiMA) tool JSON file
</p>

<hr>
<h2 id='json2csv'>JSON2CSV</h2><span id='topic+json2csv'></span>

<h3>Description</h3>

<p>This function converts DiMA (v4.1.1) JSON output file to a dataframe with 17
predefined columns which further acts as the input for other functions provided in this vDiveR package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>json2csv(
  json_data,
  host_name = "unknown host",
  protein_name = "unknown protein"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="json2csv_+3A_json_data">json_data</code></td>
<td>
<p>DiMA JSON output dataframe</p>
</td></tr>
<tr><td><code id="json2csv_+3A_host_name">host_name</code></td>
<td>
<p>name of the host species</p>
</td></tr>
<tr><td><code id="json2csv_+3A_protein_name">protein_name</code></td>
<td>
<p>name of the protein</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe which acts as input for the other functions in vDiveR package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>inputdf&lt;-json2csv(JSON_sample)
</code></pre>

<hr>
<h2 id='metadata'>Metadata Input Sample</h2><span id='topic+metadata'></span>

<h3>Description</h3>

<p>A dummy dataset that acts as an input for plot_worldmap() and plot_time()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows and 3 variables:
</p>

<dl>
<dt>ID</dt><dd><p>unique identifier of the sequence</p>
</dd>
<dt>country</dt><dd><p>country of the sequence collection</p>
</dd>
<dt>date</dt><dd><p>collection date of the sequence</p>
</dd>
</dl>


<hr>
<h2 id='metadata_extraction'>Metadata Extraction from NCBI/GISAID EpiCoV FASTA file</h2><span id='topic+metadata_extraction'></span>

<h3>Description</h3>

<p>This function retrieves metadata (ID, country, date) from the NCBI/GISAID EpiCoV FASTA file (default FASTA header expected).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata_extraction(file_path, source)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metadata_extraction_+3A_file_path">file_path</code></td>
<td>
<p>path of fasta file</p>
</td></tr>
<tr><td><code id="metadata_extraction_+3A_source">source</code></td>
<td>
<p>the source of fasta file, either &quot;ncbi&quot; or &quot;GISAID&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe that has three columns consisting ID, collected country and collected date
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filepath &lt;- system.file('extdata','GISAID_EpiCoV.faa', package = 'vDiveR')
meta_gisaid &lt;- metadata_extraction(filepath, 'GISAID')
</code></pre>

<hr>
<h2 id='plot_conservationLevel'>Conservation Levels Distribution Plot</h2><span id='topic+plot_conservationLevel'></span>

<h3>Description</h3>

<p>This function plots conservation levels distribution of k-mer positions, which consists of
completely conserved (black) (index incidence = 100%), highly conserved (blue)
(90% &lt;= index incidence &lt; 100%), mixed variable (green) (20% &lt; index incidence &lt;= 90%),
highly diverse (purple) (10% &lt; index incidence &lt;= 20%) and
extremely diverse (pink) (index incidence &lt;= 10%).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_conservationLevel(
  df,
  protein_order = "",
  conservation_label = 1,
  host = 1,
  base_size = 11,
  line_dot_size = 2,
  label_size = 2.6,
  alpha = 0.6
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_conservationLevel_+3A_df">df</code></td>
<td>
<p>DiMA JSON converted csv file data</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_protein_order">protein_order</code></td>
<td>
<p>order of proteins displayed in plot</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_conservation_label">conservation_label</code></td>
<td>
<p>0 (partial; show present conservation labels only) or 1 (full; show ALL conservation labels) in plot</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_host">host</code></td>
<td>
<p>number of host (1/2)</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_base_size">base_size</code></td>
<td>
<p>base font size in plot</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_line_dot_size">line_dot_size</code></td>
<td>
<p>lines and dots size</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_label_size">label_size</code></td>
<td>
<p>conservation labels font size</p>
</td></tr>
<tr><td><code id="plot_conservationLevel_+3A_alpha">alpha</code></td>
<td>
<p>any number from 0 (transparent) to 1 (opaque)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_conservationLevel(proteins_1host, conservation_label = 1,alpha=0.8, base_size = 15)
plot_conservationLevel(protein_2hosts, conservation_label = 0, host=2)
</code></pre>

<hr>
<h2 id='plot_correlation'>Entropy and total variant incidence correlation plot</h2><span id='topic+plot_correlation'></span>

<h3>Description</h3>

<p>This function plots the correlation between entropy and total variant incidence
of all the provided protein(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_correlation(
  df,
  host = 1,
  alpha = 1/3,
  line_dot_size = 3,
  base_size = 11,
  ylabel = "k-mer entropy (bits)\n",
  xlabel = "\nTotal variants (%)",
  ymax = ceiling(max(df$entropy)),
  ybreak = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_correlation_+3A_df">df</code></td>
<td>
<p>DiMA JSON converted csv file data</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_host">host</code></td>
<td>
<p>number of host (1/2)</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_alpha">alpha</code></td>
<td>
<p>any number from 0 (transparent) to 1 (opaque)</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_line_dot_size">line_dot_size</code></td>
<td>
<p>dot size in scatter plot</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_base_size">base_size</code></td>
<td>
<p>base font size in plot</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_ylabel">ylabel</code></td>
<td>
<p>y-axis label</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_xlabel">xlabel</code></td>
<td>
<p>x-axis label</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_ymax">ymax</code></td>
<td>
<p>maximum y-axis</p>
</td></tr>
<tr><td><code id="plot_correlation_+3A_ybreak">ybreak</code></td>
<td>
<p>y-axis breaks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scatter plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_correlation(proteins_1host)
plot_correlation(protein_2hosts, base_size = 2, ybreak=1, ymax=10, host = 2)
</code></pre>

<hr>
<h2 id='plot_dynamics_protein'>Dynamics of Diversity Motifs (Protein) Plot</h2><span id='topic+plot_dynamics_protein'></span>

<h3>Description</h3>

<p>This function compactly display the dynamics of diversity motifs (index and its variants: major, minor and unique)
in the form of dot plot(s) as well as violin plots for all the provided individual protein(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_dynamics_protein(
  df,
  host = 1,
  protein_order = "",
  base_size = 8,
  alpha = 1/3,
  line_dot_size = 3,
  bw = "nrd0",
  adjust = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_dynamics_protein_+3A_df">df</code></td>
<td>
<p>DiMA JSON converted csv file data</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_host">host</code></td>
<td>
<p>number of host (1/2)</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_protein_order">protein_order</code></td>
<td>
<p>order of proteins displayed in plot</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_base_size">base_size</code></td>
<td>
<p>base font size in plot</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_alpha">alpha</code></td>
<td>
<p>any number from 0 (transparent) to 1 (opaque)</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_line_dot_size">line_dot_size</code></td>
<td>
<p>dot size in scatter plot</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_bw">bw</code></td>
<td>
<p>smoothing bandwidth of violin plot (default: nrd0)</p>
</td></tr>
<tr><td><code id="plot_dynamics_protein_+3A_adjust">adjust</code></td>
<td>
<p>adjust the width of violin plot (default: 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_dynamics_protein(proteins_1host)
</code></pre>

<hr>
<h2 id='plot_dynamics_proteome'>Dynamics of Diversity Motifs (Proteome) Plot</h2><span id='topic+plot_dynamics_proteome'></span>

<h3>Description</h3>

<p>This function compactly display the dynamics of diversity motifs (index and its variants: major, minor and unique)
in the form of dot plot as well as violin plot for all the provided proteins at proteome level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_dynamics_proteome(
  df,
  host = 1,
  line_dot_size = 2,
  base_size = 15,
  alpha = 1/3,
  bw = "nrd0",
  adjust = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_dynamics_proteome_+3A_df">df</code></td>
<td>
<p>DiMA JSON converted csv file data</p>
</td></tr>
<tr><td><code id="plot_dynamics_proteome_+3A_host">host</code></td>
<td>
<p>number of host (1/2)</p>
</td></tr>
<tr><td><code id="plot_dynamics_proteome_+3A_line_dot_size">line_dot_size</code></td>
<td>
<p>size of dot in plot</p>
</td></tr>
<tr><td><code id="plot_dynamics_proteome_+3A_base_size">base_size</code></td>
<td>
<p>word size in plot</p>
</td></tr>
<tr><td><code id="plot_dynamics_proteome_+3A_alpha">alpha</code></td>
<td>
<p>any number from 0 (transparent) to 1 (opaque)</p>
</td></tr>
<tr><td><code id="plot_dynamics_proteome_+3A_bw">bw</code></td>
<td>
<p>smoothing bandwidth of violin plot (default: nrd0)</p>
</td></tr>
<tr><td><code id="plot_dynamics_proteome_+3A_adjust">adjust</code></td>
<td>
<p>adjust the width of violin plot (default: 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_dynamics_proteome(proteins_1host)
</code></pre>

<hr>
<h2 id='plot_entropy'>Entropy plot</h2><span id='topic+plot_entropy'></span>

<h3>Description</h3>

<p>This function plot entropy (black) and total variant (red) incidence of each
k-mer position across the studied proteins and highlight region(s) with zero entropy in yellow.
k-mer position with low support is marked with a red triangle underneath the x-axis line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_entropy(
  df,
  host = 1,
  protein_order = "",
  kmer_size = 9,
  ymax = 10,
  line_dot_size = 2,
  base_size = 8,
  all = TRUE,
  highlight_zero_entropy = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_entropy_+3A_df">df</code></td>
<td>
<p>DiMA JSON converted csv file data</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_host">host</code></td>
<td>
<p>number of host (1/2)</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_protein_order">protein_order</code></td>
<td>
<p>order of proteins displayed in plot</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_kmer_size">kmer_size</code></td>
<td>
<p>size of the k-mer window</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_ymax">ymax</code></td>
<td>
<p>maximum y-axis</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_line_dot_size">line_dot_size</code></td>
<td>
<p>size of the line and dot in plot</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_base_size">base_size</code></td>
<td>
<p>word size in plot</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_all">all</code></td>
<td>
<p>plot both the entropy and total variants (pass FALSE in to plot only the entropy)</p>
</td></tr>
<tr><td><code id="plot_entropy_+3A_highlight_zero_entropy">highlight_zero_entropy</code></td>
<td>
<p>highlight region with zero entropy (default: TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_entropy(proteins_1host)
plot_entropy(protein_2hosts, host = 2)
</code></pre>

<hr>
<h2 id='plot_time'>Time Distribution of Sequences Plot</h2><span id='topic+plot_time'></span>

<h3>Description</h3>

<p>This function plots the time distribution of provided sequences in the form of bar plot with 'Month' as x-axis and
'Number of Sequences' as y-axis. Aside from the plot, this function also returns a dataframe with 2 columns: 'Date' and 'Number of sequences'.
The input dataframe of this function is obtainable from metadata_extraction(), with NCBI Protein / GISAID EpiCoV FASTA file as input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_time(
  metadata,
  date_format = "%Y-%m-%d",
  base_size = 8,
  date_break = "2 month",
  scale = "count",
  only_plot = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_time_+3A_metadata">metadata</code></td>
<td>
<p>a dataframe with 3 columns, 'ID', 'country', and 'date'</p>
</td></tr>
<tr><td><code id="plot_time_+3A_date_format">date_format</code></td>
<td>
<p>date format of the input dataframe</p>
</td></tr>
<tr><td><code id="plot_time_+3A_base_size">base_size</code></td>
<td>
<p>word size in plot</p>
</td></tr>
<tr><td><code id="plot_time_+3A_date_break">date_break</code></td>
<td>
<p>date break for the scale_x_date</p>
</td></tr>
<tr><td><code id="plot_time_+3A_scale">scale</code></td>
<td>
<p>plot counts or log scale the data</p>
</td></tr>
<tr><td><code id="plot_time_+3A_only_plot">only_plot</code></td>
<td>
<p>logical, return only plot or dataframe info as well, default FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single plot or a list with 2 elements (a plot followed by a dataframe, default)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>time_plot &lt;- plot_time(metadata)$plot
time_df &lt;- plot_time(metadata)$df
</code></pre>

<hr>
<h2 id='plot_worldmap'>Geographical Distribution of Sequences Plot</h2><span id='topic+plot_worldmap'></span>

<h3>Description</h3>

<p>This function plots a worldmap and color the affected geographical region(s)
from light (lower) to dark (higher), depends on the cumulative number of sequences.
Aside from the plot, this function also returns a dataframe with 2 columns: 'Country' and 'Number of Sequences'.
The input dataframe of this function is obtainable from metadata_extraction(), with NCBI
Protein / GISAID EpiCoV FASTA file as input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_worldmap(meta, base_size = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_worldmap_+3A_meta">meta</code></td>
<td>
<p>a dataframe with 3 columns, 'ID', 'country', and 'date'</p>
</td></tr>
<tr><td><code id="plot_worldmap_+3A_base_size">base_size</code></td>
<td>
<p>word size in plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with 2 elements (a plot followed by a dataframe)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geographical_plot &lt;- plot_worldmap(metadata)$plot
geographical_df &lt;- plot_worldmap(metadata)$df
</code></pre>

<hr>
<h2 id='protein_2hosts'>DiMA (v4.1.1) JSON converted-CSV Output Sample 2</h2><span id='topic+protein_2hosts'></span>

<h3>Description</h3>

<p>A dummy dataset with 1 protein (Core) from two hosts, human and bat
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protein_2hosts
</code></pre>


<h3>Format</h3>

<p>A data frame with 200 rows and 17 variables:
</p>

<dl>
<dt>proteinName</dt><dd><p>name of the protein</p>
</dd>
<dt>position</dt><dd><p>starting position of the aligned, overlapping k-mer window</p>
</dd>
<dt>count</dt><dd><p>number of k-mer sequences at the given position</p>
</dd>
<dt>lowSupport</dt><dd><p>k-mer position with sequences lesser than the minimum support threshold (TRUE) are considered of low support, in terms of sample size</p>
</dd>
<dt>entropy</dt><dd><p>level of variability at the k-mer position, with zero representing completely conserved</p>
</dd>
<dt>indexSequence</dt><dd><p>the predominant sequence (index motif) at the given k-mer position</p>
</dd>
<dt>index.incidence</dt><dd><p>the fraction (in percentage) of the index sequences at the k-mer position</p>
</dd>
<dt>major.incidence</dt><dd><p>the fraction (in percentage) of the major sequence (the predominant variant to the index) at the k-mer position</p>
</dd>
<dt>minor.incidence</dt><dd><p>the fraction (in percentage) of minor sequences (of frequency lesser than the major variant, but not singletons) at the k-mer position</p>
</dd>
<dt>unique.incidence</dt><dd><p>the fraction (in percentage) of unique sequences (singletons, observed only once) at the k-mer position</p>
</dd>
<dt>totalVariants.incidence</dt><dd><p>the fraction (in percentage) of sequences at the k-mer position that are variants to the index (includes: major, minor and unique variants)</p>
</dd>
<dt>distinctVariant.incidence</dt><dd><p>incidence of the distinct k-mer peptides at the k-mer position</p>
</dd>
<dt>multiIndex</dt><dd><p>presence of more than one index sequence of equal incidence</p>
</dd>
<dt>host</dt><dd><p>species name of the organism host to the virus</p>
</dd>
<dt>highestEntropy.position</dt><dd><p>k-mer position that has the highest entropy value</p>
</dd>
<dt>highestEntropy</dt><dd><p>highest entropy values observed in the studied protein</p>
</dd>
<dt>averageEntropy</dt><dd><p>average entropy values across all the k-mer positions</p>
</dd>
</dl>


<hr>
<h2 id='proteins_1host'>DiMA (v4.1.1) JSON converted-CSV Output Sample 1</h2><span id='topic+proteins_1host'></span>

<h3>Description</h3>

<p>A dummy dataset with two proteins (A and B) from one host, human
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proteins_1host
</code></pre>


<h3>Format</h3>

<p>A data frame with 806 rows and 17 variables:
</p>

<dl>
<dt>proteinName</dt><dd><p>name of the protein</p>
</dd>
<dt>position</dt><dd><p>starting position of the aligned, overlapping k-mer window</p>
</dd>
<dt>count</dt><dd><p>number of k-mer sequences at the given position</p>
</dd>
<dt>lowSupport</dt><dd><p>k-mer position with sequences lesser than the minimum support threshold (TRUE) are considered of low support, in terms of sample size</p>
</dd>
<dt>entropy</dt><dd><p>level of variability at the k-mer position, with zero representing completely conserved</p>
</dd>
<dt>indexSequence</dt><dd><p>the predominant sequence (index motif) at the given k-mer position</p>
</dd>
<dt>index.incidence</dt><dd><p>the fraction (in percentage) of the index sequences at the k-mer position</p>
</dd>
<dt>major.incidence</dt><dd><p>the fraction (in percentage) of the major sequence (the predominant variant to the index) at the k-mer position</p>
</dd>
<dt>minor.incidence</dt><dd><p>the fraction (in percentage) of minor sequences (of frequency lesser than the major variant, but not singletons) at the k-mer position</p>
</dd>
<dt>unique.incidence</dt><dd><p>the fraction (in percentage) of unique sequences (singletons, observed only once) at the k-mer position</p>
</dd>
<dt>totalVariants.incidence</dt><dd><p>the fraction (in percentage) of sequences at the k-mer position that are variants to the index (includes: major, minor and unique variants)</p>
</dd>
<dt>distinctVariant.incidence</dt><dd><p>incidence of the distinct k-mer peptides at the k-mer position</p>
</dd>
<dt>multiIndex</dt><dd><p>presence of more than one index sequence of equal incidence</p>
</dd>
<dt>host</dt><dd><p>species name of the organism host to the virus</p>
</dd>
<dt>highestEntropy.position</dt><dd><p>k-mer position that has the highest entropy value</p>
</dd>
<dt>highestEntropy</dt><dd><p>highest entropy values observed in the studied protein</p>
</dd>
<dt>averageEntropy</dt><dd><p>average entropy values across all the k-mer positions</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
