<!DOCTYPE html><html><head><title>Help for package VDJgermlines</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {VDJgermlines}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#extractSequencesR'><p>Extracts the germline genes of interest and stores them as a dataframe</p></a></li>
<li><a href='#VDJgermlines'><p>VDJgermlines</p></a></li>
<li><a href='#VDJheatmap'><p>Calculates the distance between VDJ germlines of interest and produces a matrix that can be used for further analysis or viewed using Heatmap/pheatmap/ComplexHeatmap.</p></a></li>
<li><a href='#VDJphylo'><p>Infers basic phylogenetic trees of VDJ germlines of interest with basic distance methods.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Variable, Diversity and Joining Sequences from Various Species</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-12-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Alexander Yermanos</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Alexander Yermanos &lt;ayermanos@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.1.0), ape, stringdist</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains variable, diversity, and joining sequences and accompanying functions that enable both the extraction of and comparison between immune V-D-J genomic segments from a variety of species. Sources include IMGT from MP Lefranc (2009) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkn838">doi:10.1093/nar/gkn838</a>&gt; and Vgenerepertoire from publication DN Olivieri (2014) &lt;<a href="https://doi.org/10.1007%2Fs00251-014-0784-3">doi:10.1007/s00251-014-0784-3</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-12-10 13:36:52 UTC; ayermano</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-12-18 23:30:32 UTC</td>
</tr>
</table>
<hr>
<h2 id='extractSequencesR'>Extracts the germline genes of interest and stores them as a dataframe</h2><span id='topic+extractSequencesR'></span>

<h3>Description</h3>

<p>Extracts the germline genes of interest and stores them as a dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractSequencesR(species, chain, source, write.fasta, fasta.name,
  directory.string)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractSequencesR_+3A_species">species</code></td>
<td>
<p>Specifies desired species. Current species options can be found by running unique(VDJgermlines$species) after loading in data(VDJgermlines). &quot;all_species&quot; will select all available species. Species include (&quot;alpaca&quot;,&quot;bovine&quot;,&quot;crab_eating_macaque&quot;,&quot;dog&quot;,
&quot;human&quot;,&quot;mouse&quot;,&quot;pig&quot;,&quot;platypus&quot;,&quot;rabbit&quot;,raintrout&quot;,&quot;rat&quot;,&quot;rhesus monkey&quot;,&quot;salmon&quot;,
&quot;sheep&quot;,&quot;zebrafish&quot;,&quot;atlanticcod&quot;,&quot;catfish&quot;,&quot;teleostei&quot;,&quot;goat&quot;,&quot;dolphin&quot;,&quot;mamonkey&quot;,
&quot;camel&quot;,&quot;runny&quot;)</p>
</td></tr>
<tr><td><code id="extractSequencesR_+3A_chain">chain</code></td>
<td>
<p>Specifies the desired chain. &quot;all_chains&quot; will select all chains. The various chains can be listed by running unique(VDJgermlines$chain).Antibody loci start with IG, TCR starts with TR.NExt comes either heavy, light, alpha, beta etc. Examples include IGHD IGHJ IGHV IGIJ IGIV IGKJ IGKV IGLJ IGLV TRAJ TRAV TRBD TRBJ TRBV TRDD TRDJ TRDV TRGJ TRGV</p>
</td></tr>
<tr><td><code id="extractSequencesR_+3A_source">source</code></td>
<td>
<p>Specifies from which source the desired chains should be taken from.  Current sources can be found by running unique(VDJgermlines$source) after loading in data(VDJgermlines). IMGT and vgenerepertoire are currently present.</p>
</td></tr>
<tr><td><code id="extractSequencesR_+3A_write.fasta">write.fasta</code></td>
<td>
<p>A boolean (TRUE / FALSE) value that species if the output sequences should be also written as a fasta file.</p>
</td></tr>
<tr><td><code id="extractSequencesR_+3A_fasta.name">fasta.name</code></td>
<td>
<p>A string specifying the name of the fasta file. If this is left blank this will be defaulted to &quot;VDJgermlines.fasta&quot;. Manually including extension is not needed.</p>
</td></tr>
<tr><td><code id="extractSequencesR_+3A_directory.string">directory.string</code></td>
<td>
<p>A string stating the directory that the fasta file should be saved to. If left blank the
fasta file will be stored in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a dataframe containing the species and chains of interest from the desired source. The columns of the dataframe include sequence, names, species, chain, accession, official species name, source, and strain. Furthermore, if write.fasta is set to TRUE then the output will additionally include a fasta file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>extractSequencesR(species = "dog",chain = "IGHD",source = "IMGT",write.fasta = FALSE)
</code></pre>

<hr>
<h2 id='VDJgermlines'>VDJgermlines</h2><span id='topic+VDJgermlines'></span>

<h3>Description</h3>

<p>The germline information for a variety of species is contained in the VDJgermlines dataframe.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VDJgermlines
</code></pre>


<h3>Format</h3>

<p>A data frame with 7285 rows and 8 variables:
</p>

<dl>
<dt>sequence</dt><dd><p>The sequence directly from either IMGT or Vgenerepertoire</p>
</dd>
<dt>names</dt><dd><p>The corresponding name of the sequence including V gene number, species</p>
</dd>
<dt>species</dt><dd><p>The casual species name - e.g. dog, mouse, rat</p>
</dd>
<dt>chain</dt><dd><p>The corresponding chain names, including IGHV, IGHL, TRAV, etc</p>
</dd>
<dt>accession</dt><dd><p>The accession number of the given gene</p>
</dd>
<dt>species_name_official</dt><dd><p>The more scientific species name e.g. mus musculus</p>
</dd>
<dt>source</dt><dd><p>The database that annotated the sequence</p>
</dd>
<dt>strain</dt><dd><p>The accompanying strain if available (e.g. B6 vs BALB/c)</p>
</dd>
</dl>


<h3>Source</h3>

<p>IMGT and Vgenerepertoire IMGT (the international ImMunoGeneTics information system (founder and director: Marie-Paule Lefranc, Montpellier, France).&quot; : Lefranc, M.-P. et al., Nucleic Acids Research, 27, 209-212 (1999) Cover; Ruiz, M. et al.,  Nucleic Acids Research, 28, 219-221 (2000); Lefranc, M.-P., Nucleic Acids Research, 29, 207-209 (2001); Lefranc, M.-P., Nucleic Acids Res., 31, 307-310 (2003); Lefranc, M.-P. et al., In Silico Biol., 5, 0006 (2004) [Epub], 5:45-60 (2005); Lefranc, M.-P. et al., Nucleic Acids Res., 33, D593-D597 (2005) Full text; Lefranc, M.-P. et al., Nucleic Acids Research 2009 37(Database issue): D1006-D1012; doi:10.1093/nar/gkn838 Full text. Nucleic Acids Res. 2015 Jan 28;43(Database issue):D413-422. doi: 10.1093/nar/gku1056. Epub 2014 Nov 5. Full text) and Vgenerepertoire (Immunogenetics. 2014 Aug;66(7-8):479-92. doi: 10.1007/s00251-014-0784-3. Epub 2014 Jun 4. and Vgenerepertoire (Genomic V exons from whole genome shotgun data in reptiles. Olivieri DNx, von Haeften B, S&lt;U+00E1&gt;nchez-Espinel C, Faro J, Gamb&lt;U+00F3&gt;n-Deza F.).
</p>

<hr>
<h2 id='VDJheatmap'>Calculates the distance between VDJ germlines of interest and produces a matrix that can be used for further analysis or viewed using Heatmap/pheatmap/ComplexHeatmap.</h2><span id='topic+VDJheatmap'></span>

<h3>Description</h3>

<p>Calculates the distance between VDJ germlines of interest and produces a matrix that can be used for further analysis or viewed using Heatmap/pheatmap/ComplexHeatmap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VDJheatmap(species, chain, source, inference.method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VDJheatmap_+3A_species">species</code></td>
<td>
<p>Specifies desired species. Current species options can be found by running unique(VDJgermlines$species) after loading in data(VDJgermlines). &quot;all_species&quot; will select all available species. Species include (&quot;alpaca&quot;,&quot;bovine&quot;,&quot;crab_eating_macaque&quot;,&quot;dog&quot;,
&quot;human&quot;,&quot;mouse&quot;,&quot;pig&quot;,&quot;platypus&quot;,&quot;rabbit&quot;,raintrout&quot;,&quot;rat&quot;,&quot;rhesus monkey&quot;,&quot;salmon&quot;,
&quot;sheep&quot;,&quot;zebrafish&quot;,&quot;atlanticcod&quot;,&quot;catfish&quot;,&quot;teleostei&quot;,&quot;goat&quot;,&quot;dolphin&quot;,&quot;mamonkey&quot;,
&quot;camel&quot;,&quot;runny&quot;)</p>
</td></tr>
<tr><td><code id="VDJheatmap_+3A_chain">chain</code></td>
<td>
<p>Specifies the desired chain. &quot;all_chains&quot; will select all chains. The various chains can be listed by running unique(VDJgermlines$chain).Antibody loci start with IG, TCR starts with TR.NExt comes either heavy, light, alpha, beta etc. Examples include IGHD IGHJ IGHV IGIJ IGIV IGKJ IGKV IGLJ IGLV TRAJ TRAV TRBD TRBJ TRBV TRDD TRDJ TRDV TRGJ TRGV</p>
</td></tr>
<tr><td><code id="VDJheatmap_+3A_source">source</code></td>
<td>
<p>Specifies from which source the desired chains should be taken from.  Current sources can be found by running unique(VDJgermlines$source) after loading in data(VDJgermlines). IMGT and vgenerepertoire are currently present.</p>
</td></tr>
<tr><td><code id="VDJheatmap_+3A_inference.method">inference.method</code></td>
<td>
<p>Specifies the distance metric to be used to calculate distance. Currently includes parameters from stringdist::stringdistmatrix(), such as &quot;lv&quot;,&quot;jv&quot;,&quot;hamming&quot;,&quot;qgram&quot;,&quot;cosine&quot;,&quot;Jaccard&quot;,&quot;lcs&quot;, or &quot;soundex&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix where the entries correspond to the VDJ germlines used the neighbor joining algorithm. Can be viewed by heatmap() in base r, or pheatmap::pheatmap(VDJheatmap).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>VDJheatmap(species = "dog",chain = "IGHD",source = "IMGT")
</code></pre>

<hr>
<h2 id='VDJphylo'>Infers basic phylogenetic trees of VDJ germlines of interest with basic distance methods.</h2><span id='topic+VDJphylo'></span>

<h3>Description</h3>

<p>Infers basic phylogenetic trees of VDJ germlines of interest with basic distance methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VDJphylo(species, chain, source, write.fasta, fasta.name, directory.string,
  inference.method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VDJphylo_+3A_species">species</code></td>
<td>
<p>Specifies desired species. Current species options can be found by running unique(VDJgermlines$species) after loading in data(VDJgermlines). &quot;all_species&quot; will select all available species. Species include (&quot;alpaca&quot;,&quot;bovine&quot;,&quot;crab_eating_macaque&quot;,&quot;dog&quot;,
&quot;human&quot;,&quot;mouse&quot;,&quot;pig&quot;,&quot;platypus&quot;,&quot;rabbit&quot;,raintrout&quot;,&quot;rat&quot;,&quot;rhesus monkey&quot;,&quot;salmon&quot;,
&quot;sheep&quot;,&quot;zebrafish&quot;,&quot;atlanticcod&quot;,&quot;catfish&quot;,&quot;teleostei&quot;,&quot;goat&quot;,&quot;dolphin&quot;,&quot;mamonkey&quot;,
&quot;camel&quot;,&quot;runny&quot;)</p>
</td></tr>
<tr><td><code id="VDJphylo_+3A_chain">chain</code></td>
<td>
<p>Specifies the desired chain. &quot;all_chains&quot; will select all chains. The various chains can be listed by running unique(VDJgermlines$chain).Antibody loci start with IG, TCR starts with TR.NExt comes either heavy, light, alpha, beta etc. Examples include IGHD IGHJ IGHV IGIJ IGIV IGKJ IGKV IGLJ IGLV TRAJ TRAV TRBD TRBJ TRBV TRDD TRDJ TRDV TRGJ TRGV</p>
</td></tr>
<tr><td><code id="VDJphylo_+3A_source">source</code></td>
<td>
<p>Specifies from which source the desired chains should be taken from.  Current sources can be found by running unique(VDJgermlines$source) after loading in data(VDJgermlines). IMGT and vgenerepertoire are currently present.</p>
</td></tr>
<tr><td><code id="VDJphylo_+3A_write.fasta">write.fasta</code></td>
<td>
<p>A boolean (TRUE / FALSE) value that species if the output sequences should be also written as a fasta file.</p>
</td></tr>
<tr><td><code id="VDJphylo_+3A_fasta.name">fasta.name</code></td>
<td>
<p>A string specifying the name of the fasta file. If this is left blank this will be defaulted to &quot;VDJgermlines.fasta&quot;. Manually including extension is not needed.</p>
</td></tr>
<tr><td><code id="VDJphylo_+3A_directory.string">directory.string</code></td>
<td>
<p>A string stating the directory that the fasta file should be saved to. If left blank the
fasta file will be stored in the current working directory.</p>
</td></tr>
<tr><td><code id="VDJphylo_+3A_inference.method">inference.method</code></td>
<td>
<p>Specifies the method to be used to infer the phylogenetic tree. Currently includes
&quot;lv&quot; for Levenshtein distance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an unrooted tree inferred using the neighbor joining algorithm. Can be viewed by plot(output_tree) after loading ape package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>VDJphylo(species = "dog",chain = "IGHD",source = "IMGT",write.fasta = FALSE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
