<!DOCTYPE html><html><head><title>Help for package vermeulen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {vermeulen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#vermeulen-package'><p>vermeulen: Biomarker Data Set by Vermeulen et al. (2009)</p></a></li>
<li><a href='#get_biomarker_dataset'><p>Import the Biomarker data set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Biomarker Data Set by Vermeulen et al. (2009)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>The biomarker data set by Vermeulen et al. (2009)
    &lt;<a href="https://doi.org/10.1016%2FS1470-2045%2809%2970154-8">doi:10.1016/S1470-2045(09)70154-8</a>&gt; is provided. The data
    source, however, is by Ruijter et al. (2013)
    &lt;<a href="https://doi.org/10.1016%2Fj.ymeth.2012.08.011">doi:10.1016/j.ymeth.2012.08.011</a>&gt;. The original data set may
    be downloaded from
    <a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>.
    This data set is for a real-time quantitative polymerase chain reaction
    (PCR) experiment that comprises the raw fluorescence data of 24,576
    amplification curves. This data set comprises 59 genes of interest and 5
    reference genes. Each gene was assessed on 366 neuroblastoma complementary
    DNA (cDNA) samples and on 18 standard dilution series samples (10-fold
    5-point dilution series x 3 replicates + no template controls (NTC) x 3
    replicates).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by/4.0">CC BY 4.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>memoise</td>
</tr>
<tr>
<td>Suggests:</td>
<td>dplyr, ggplot2, rmarkdown, spelling, testthat (&ge; 3.0.0),
tibble</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ramiromagno/vermeulen">https://github.com/ramiromagno/vermeulen</a>,
<a href="https://rmagno.eu/vermeulen/">https://rmagno.eu/vermeulen/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ramiromagno/vermeulen/issues">https://github.com/ramiromagno/vermeulen/issues</a></td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-22 21:30:34 UTC; rmagno</td>
</tr>
<tr>
<td>Author:</td>
<td>Ramiro Magno <a href="https://orcid.org/0000-0001-5226-3441"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Pattern Institute [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ramiro Magno &lt;rmagno@pattern.institute&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-22 22:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='vermeulen-package'>vermeulen: Biomarker Data Set by Vermeulen et al. (2009)</h2><span id='topic+vermeulen'></span><span id='topic+vermeulen-package'></span>

<h3>Description</h3>

<p>The biomarker data set by Vermeulen et al. (2009) <a href="https://doi.org/10.1016/S1470-2045%2809%2970154-8">doi:10.1016/S1470-2045(09)70154-8</a> is provided. The data source, however, is by Ruijter et al. (2013) <a href="https://doi.org/10.1016/j.ymeth.2012.08.011">doi:10.1016/j.ymeth.2012.08.011</a>. The original data set may be downloaded from <a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>. This data set is for a real-time quantitative polymerase chain reaction (PCR) experiment that comprises the raw fluorescence data of 24,576 amplification curves. This data set comprises 59 genes of interest and 5 reference genes. Each gene was assessed on 366 neuroblastoma complementary DNA (cDNA) samples and on 18 standard dilution series samples (10-fold 5-point dilution series x 3 replicates + no template controls (NTC) x 3 replicates).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ramiro Magno <a href="mailto:rmagno@pattern.institute">rmagno@pattern.institute</a> (<a href="https://orcid.org/0000-0001-5226-3441">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Pattern Institute [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ramiromagno/vermeulen">https://github.com/ramiromagno/vermeulen</a>
</p>
</li>
<li> <p><a href="https://rmagno.eu/vermeulen/">https://rmagno.eu/vermeulen/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ramiromagno/vermeulen/issues">https://github.com/ramiromagno/vermeulen/issues</a>
</p>
</li></ul>


<hr>
<h2 id='get_biomarker_dataset'>Import the Biomarker data set</h2><span id='topic+get_biomarker_dataset'></span>

<h3>Description</h3>

<p>This function retrieves the Biomarker data set, a data set containing raw
fluorescence amplification data: 24,576 amplification curves, of 50 cycles
each.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biomarker_dataset()
</code></pre>


<h3>Details</h3>

<p>Data was gathered from Ruijter et al.
(2013), <a href="https://doi.org/10.1016/j.ymeth.2012.08.011">doi:10.1016/j.ymeth.2012.08.011</a> but original source is by Vermeulen
et al. (2009), <a href="https://doi.org/10.1016/S1470-2045%2809%2970154-8">doi:10.1016/S1470-2045(09)70154-8</a>.
The tidy version of the data is kept at the repository of the source of
<code>{vermeulen}</code> package. This function fetches such data and thus requires
internet connection. It takes a few seconds to run.
</p>
<p>The Biomarker data set comprises a set of 59 targets previously identified as
a 59-mRNA gene expression signature, that has been developed and validated for
improved outcome prediction of children with neuroblastoma. In short, 59
biomarkers and 5 reference genes were measured in 8 Âµl reactions in a
384-well plate using the LightCycler480 SYBR Green Master (Roche) in a sample
maximization experiment design. The 59 genes were carefully selected as being
previously reported as prognostic genes in neuroblastoma in at least two
independent studies. Each plate contained 366 cDNA samples (n = 1) from
primary tumor biopsies, a 5-point 10-fold serial dilution series based on an
external oligonucleotide standard (n = 3, from 150,000 to 15 copies), and a
no template control (NTC, n = 3). Raw (baseline uncorrected) fluorescent data
were exported from the LightCycler480 instrument software.
</p>


<h3>Value</h3>

<p>A data frame with 24,576 amplification curves, of 50 cycles each:
</p>

<dl>
<dt><code>plate</code></dt><dd><p>Plate identifier. Because one plate was used per gene, the
name of the plate is the same as the values in <code>target</code>.</p>
</dd>
<dt><code>well</code></dt><dd><p>Well identifier.</p>
</dd>
<dt><code>dye</code></dt><dd><p>In all reactions the SYBR Green I master mix (Roche) was used,
so the value is always <code>"SYBR"</code>.</p>
</dd>
<dt><code>target</code></dt><dd><p>Target identifier, in almost all cases the name of a gene.</p>
</dd>
<dt><code>target_type</code></dt><dd><p>Target type: either target of interest (<code>"toi"</code>) or
reference target (<code>"ref"</code>).</p>
</dd>
<dt><code>sample</code></dt><dd><p>Sample identifier.</p>
</dd>
<dt><code>sample_type</code></dt><dd><p>Sample type.</p>
</dd>
<dt><code>copies</code></dt><dd><p>Standard copy number.</p>
</dd>
<dt><code>dilution</code></dt><dd><p>Dilution factor. Higher number means greater dilution.</p>
</dd>
<dt><code>cycle</code></dt><dd><p>PCR cycle.</p>
</dd>
<dt><code>fluor</code></dt><dd><p>Raw fluorescence values.</p>
</dd>
</dl>



<h3>Source</h3>


<ul>
<li> <p><a href="https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip">https://medischebiologie.nl/wp-content/uploads/2019/02/qpcrdatamethods.zip</a>
</p>
</li>
<li> <p><a href="https://github.com/ramiromagno/vermeulen/tree/main/data-raw">https://github.com/ramiromagno/vermeulen/tree/main/data-raw</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
# Takes ~ 10-30 sec
head(get_biomarker_dataset())


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
