<!DOCTYPE html><html><head><title>Help for package vhcub</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {vhcub}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CAI.values'><p>Codon Adaptation  Index  (CAI)</p></a></li>
<li><a href='#dinuc.base'><p>Statistical dinucleotide over- and underrepresentation (base model).</p></a></li>
<li><a href='#dinuc.codon'><p>Statistical dinucleotide over- and underrepresentation (codon model).</p></a></li>
<li><a href='#dinuc.syncodon'><p>Statistical dinucleotide over- and underrepresentation (syncodon model).</p></a></li>
<li><a href='#ENc.GC3plot'><p>ENc-GC3 scatterplot.</p></a></li>
<li><a href='#ENc.values'><p>Effective Number of Codons (ENc).</p></a></li>
<li><a href='#fasta.read'><p>Read fasta formate and convert it to data frame</p></a></li>
<li><a href='#GC.content'><p>GC content</p></a></li>
<li><a href='#PR2.plot'><p>Parity rule 2 (PR2) plot</p></a></li>
<li><a href='#RCDI.values'><p>Relative Codon Deoptimization Index (RCDI)</p></a></li>
<li><a href='#RSCU.values'><p>Relative Synonymous Codon Usage (RSCU)</p></a></li>
<li><a href='#SCUO.values'><p>Synonymous codon usage eorderliness (SCUO)</p></a></li>
<li><a href='#SiD.value'><p>Similarity Index (SiD)</p></a></li>
<li><a href='#vhcub'><p>vhcub: A package to analysis the co-adaptation of codon usage between a virus and its host.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Virus-Host Codon Usage Co-Adaptation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Ali Mostafa Anwar [aut, cre],
  Mohamed Soudy [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ali Mostafa Anwar &lt;ali.mo.anwar@std.agr.cu.edu.eg&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analyze the co-adaptation of codon usage between a virus and its host, calculate various codon usage bias measurements as: effective number of codons (ENc) Novembre (2002) &lt;<a href="https://doi.org/10.1093%2Foxfordjournals.molbev.a004201">doi:10.1093/oxfordjournals.molbev.a004201</a>&gt;, codon adaptation index (CAI) Sharp  and  Li (1987) &lt;<a href="https://doi.org/10.1093%2Fnar%2F15.3.1281">doi:10.1093/nar/15.3.1281</a>&gt;, relative  codon deoptimization index (RCDI) Puigb√≤ et al (2010) &lt;<a href="https://doi.org/10.1186%2F1756-0500-3-87">doi:10.1186/1756-0500-3-87</a>&gt;, similarity index (SiD) Zhou et al (2013) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0077239">doi:10.1371/journal.pone.0077239</a>&gt;,  synonymous codon usage orderliness (SCUO) Wan et al (2004) &lt;<a href="https://doi.org/10.1186%2F1471-2148-4-19">doi:10.1186/1471-2148-4-19</a>&gt; and, relative synonymous    codon usage (RSCU) Sharp et al (1986) &lt;<a href="https://doi.org/10.1093%2Fnar%2F14.13.5125">doi:10.1093/nar/14.13.5125</a>&gt;. Also, it provides a statistical dinucleotide over- and underrepresentation with three different models. Implement several methods for visualization of codon usage as ENc.GC3plot() and PR2.plot().</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, coRdon , ggplot2, seqinr, stringr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-07 17:12:46 UTC; ali</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-15 12:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CAI.values'>Codon Adaptation  Index  (CAI)</h2><span id='topic+CAI.values'></span>

<h3>Description</h3>

<p>Measure the Codon Adaptation  Index  (CAI)  Sharp  and  Li  (1987), of DNA sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CAI.values(df.virus, ENc.set.host,
                       df.host,genetic.code = "1",set.len = 5, threshold = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CAI.values_+3A_df.virus">df.virus</code></td>
<td>
<p>a data frame with seq_name and its virus DNA sequence.</p>
</td></tr>
<tr><td><code id="CAI.values_+3A_enc.set.host">ENc.set.host</code></td>
<td>
<p>a data frame with ENc values of a host.</p>
</td></tr>
<tr><td><code id="CAI.values_+3A_df.host">df.host</code></td>
<td>
<p>a data frame with seq_name and its host DNA sequence.</p>
</td></tr>
<tr><td><code id="CAI.values_+3A_genetic.code">genetic.code</code></td>
<td>
<p>a single string that uniquely identifies a genetic code to use.</p>
</td></tr>
<tr><td><code id="CAI.values_+3A_set.len">set.len</code></td>
<td>
<p>a number represents a percent that will be used as reference genes from the total host genes.</p>
</td></tr>
<tr><td><code id="CAI.values_+3A_threshold">threshold</code></td>
<td>
<p>optional numeric, specifying sequence length, in codons, used for filtering.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about CAI <a href="https://academic.oup.com/nar/article-abstract/15/3/1281/1166844?redirectedFrom=fulltext">Sharp  and  Li, 1987</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed CAI values for each DNA sequences within df.fasta.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]
# Calculate CAI
enc.df.host &lt;- ENc.values(fasta.h) 


cai.df &lt;- CAI.values(fasta.v, enc.df.host, fasta.h)


</code></pre>

<hr>
<h2 id='dinuc.base'>Statistical dinucleotide over- and underrepresentation (base model).</h2><span id='topic+dinuc.base'></span>

<h3>Description</h3>

<p>A measure of statistical dinucleotide over- and underrepresentation; by allows for random sequence generation by shuffling (with/without replacement) of all bases in the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dinuc.base(df.virus,permutations=500,exact_numbers = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dinuc.base_+3A_df.virus">df.virus</code></td>
<td>
<p>data frame with seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="dinuc.base_+3A_permutations">permutations</code></td>
<td>
<p>the number of permutations for the z-score computation.</p>
</td></tr>
<tr><td><code id="dinuc.base_+3A_exact_numbers">exact_numbers</code></td>
<td>
<p>if TRUE exact analytical calculation will be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information <a href="https://www.rdocumentation.org/packages/seqinr/versions/3.6-1/topics/dinucleotides">seqinr</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed statistic for each dinucleotide in all DNA sequences within df.virus.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


# Calculate zscore using (base model)
base &lt;- dinuc.base(fasta.v, permutations = 10)

</code></pre>

<hr>
<h2 id='dinuc.codon'>Statistical dinucleotide over- and underrepresentation (codon model).</h2><span id='topic+dinuc.codon'></span>

<h3>Description</h3>

<p>A measure of statistical dinucleotide over- and underrepresentation; by allows for random sequence generation by shuffling (with/without replacement) of codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dinuc.codon(df.virus,permutations=500,exact_numbers = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dinuc.codon_+3A_df.virus">df.virus</code></td>
<td>
<p>data frame with seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="dinuc.codon_+3A_permutations">permutations</code></td>
<td>
<p>the number of permutations for the z-score computation.</p>
</td></tr>
<tr><td><code id="dinuc.codon_+3A_exact_numbers">exact_numbers</code></td>
<td>
<p>if TRUE exact analytical calculation will be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information <a href="https://www.rdocumentation.org/packages/seqinr/versions/3.6-1/topics/dinucleotides">seqinr</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed statistic for each dinucleotide in all DNA sequences within df.virus.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


# Calculate zscore using (codon model)
codon &lt;- dinuc.codon(fasta.v, permutations = 10)

</code></pre>

<hr>
<h2 id='dinuc.syncodon'>Statistical dinucleotide over- and underrepresentation (syncodon model).</h2><span id='topic+dinuc.syncodon'></span>

<h3>Description</h3>

<p>A measure of statistical dinucleotide over- and underrepresentation; by allows for random sequence generation by shuffling (with/without replacement) of synonymous codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dinuc.syncodon(df.virus,permutations=500,exact_numbers = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dinuc.syncodon_+3A_df.virus">df.virus</code></td>
<td>
<p>data frame with seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="dinuc.syncodon_+3A_permutations">permutations</code></td>
<td>
<p>the number of permutations for the z-score computation.</p>
</td></tr>
<tr><td><code id="dinuc.syncodon_+3A_exact_numbers">exact_numbers</code></td>
<td>
<p>if TRUE exact analytical calculation will be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information <a href="https://www.rdocumentation.org/packages/seqinr/versions/3.6-1/topics/dinucleotides">seqinr</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed statistic for each dinucleotide in all DNA sequences within df.virus.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


# Calculate zscore using (syncodon model)
syncodon &lt;- dinuc.syncodon(fasta.v, permutations = 10)

</code></pre>

<hr>
<h2 id='ENc.GC3plot'>ENc-GC3 scatterplot.</h2><span id='topic+ENc.GC3plot'></span>

<h3>Description</h3>

<p>Make an ENc-GC3 scatterplot. Where the y-axis represents the ENc values and the x-axis represents the GC3 content.
The red fitting line shows the expected ENc values when codon usage bias affected solely by GC3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ENc.GC3plot(enc.df, gc.df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ENc.GC3plot_+3A_enc.df">enc.df</code></td>
<td>
<p>a data frame with ENc values.</p>
</td></tr>
<tr><td><code id="ENc.GC3plot_+3A_gc.df">gc.df</code></td>
<td>
<p>a data frame with GC3 values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about ENc-GC3 plot <a href="https://www.tandfonline.com/doi/full/10.1038/emi.2016.106">Butt et al., 2016</a>.
</p>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
enc.df.virus &lt;- ENc.values(fasta.v)


gc.df &lt;- GC.content(fasta.v)

ENc.GC3plot(enc.df.virus, gc.df)

</code></pre>

<hr>
<h2 id='ENc.values'>Effective Number of Codons (ENc).</h2><span id='topic+ENc.values'></span>

<h3>Description</h3>

<p>Measure the Effective Number of Codons (ENc) of DNA sequence. Using its modified version (Novembre, 2002).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ENc.values(df.fasta,genetic.code = "1",threshold=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ENc.values_+3A_df.fasta">df.fasta</code></td>
<td>
<p>a data frame with seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="ENc.values_+3A_genetic.code">genetic.code</code></td>
<td>
<p>a single string that uniquely identifies a genetic code to use.</p>
</td></tr>
<tr><td><code id="ENc.values_+3A_threshold">threshold</code></td>
<td>
<p>optional numeric, specifying sequence length, in codons, used for filtering.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about ENc <a href="https://academic.oup.com/mbe/article/19/8/1390/997706">Novembre, 2002</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed ENc values for each DNA sequences within df.fasta.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


# Calculate ENc
enc.df.v &lt;- ENc.values(fasta.v)

enc.df.h &lt;- ENc.values(fasta.h)


</code></pre>

<hr>
<h2 id='fasta.read'>Read fasta formate and convert it to data frame</h2><span id='topic+fasta.read'></span>

<h3>Description</h3>

<p>Read fasta formate and convert it to data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasta.read(virus.fasta,host.fasta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fasta.read_+3A_virus.fasta">virus.fasta</code></td>
<td>
<p>directory path to the virus fasta file.</p>
</td></tr>
<tr><td><code id="fasta.read_+3A_host.fasta">host.fasta</code></td>
<td>
<p>directory path to the host fasta file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two data frames.
</p>


<h3>Note</h3>

<p>The list with two data.frames; the first one for virus DNA sequences and the second one for the host.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]

</code></pre>

<hr>
<h2 id='GC.content'>GC content</h2><span id='topic+GC.content'></span>

<h3>Description</h3>

<p>Calculates overall GC content as well as GC at first, second, and third codon positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GC.content(df.virus)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GC.content_+3A_df.virus">df.virus</code></td>
<td>
<p>data frame with seq_name and its DNA sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with overall GC content as well as GC at first, second, and third codon positions of all DNA sequence from df.virus.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


# Calculate GC content
gc.df &lt;- GC.content(fasta.v)

</code></pre>

<hr>
<h2 id='PR2.plot'>Parity rule 2 (PR2) plot</h2><span id='topic+PR2.plot'></span>

<h3>Description</h3>

<p>Make a Parity rule 2 (PR2) plot, where the AT-bias [A3/(A3 +T3)] at the third codon position of the four-codon amino acids of entire genes is the ordinate and the GC-bias [G3/(G3 +C3)] is the abscissa. The center of the plot, where both coordinates are 0.5, is where A = U and G = C (PR2), with no bias between the influence of the mutation and selection rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PR2.plot(fasta.df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PR2.plot_+3A_fasta.df">fasta.df</code></td>
<td>
<p>a data frame with seq_name and its DNA sequence.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about PR2 plot <a href="https://www.tandfonline.com/doi/full/10.1038/emi.2016.106">Butt et al., 2016</a>.
</p>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


PR2.plot(fasta.v)

</code></pre>

<hr>
<h2 id='RCDI.values'>Relative Codon Deoptimization Index (RCDI)</h2><span id='topic+RCDI.values'></span>

<h3>Description</h3>

<p>Measure the Relative Codon Deoptimization Index (RCDI) of DNA sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCDI.values(fasta.virus, fasta.host, enc.host, set.len= 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCDI.values_+3A_fasta.virus">fasta.virus</code></td>
<td>
<p>a data frame with virus seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="RCDI.values_+3A_fasta.host">fasta.host</code></td>
<td>
<p>a data frame with host seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="RCDI.values_+3A_enc.host">enc.host</code></td>
<td>
<p>a data frame of a hosts' ENc values.</p>
</td></tr>
<tr><td><code id="RCDI.values_+3A_set.len">set.len</code></td>
<td>
<p>a number represents a percent that will be used as reference genes from the total host genes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about RCDI <a href="https://bmcresnotes.biomedcentral.com/articles/10.1186/1756-0500-3-87">Puigb√≤ et al., 2010</a>
</p>


<h3>Value</h3>

<p>A data.frame containing the computed ENc values for each DNA sequences within df.fasta.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]

# Calculate RCDI

enc.df.host &lt;- ENc.values(fasta.h)


rcdi.df &lt;- RCDI.values(fasta.v, fasta.h, enc.df.host)

</code></pre>

<hr>
<h2 id='RSCU.values'>Relative Synonymous Codon Usage (RSCU)</h2><span id='topic+RSCU.values'></span>

<h3>Description</h3>

<p>Measure the Relative Synonymous Codon Usage (RSCU) of DNA sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RSCU.values(df.fasta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RSCU.values_+3A_df.fasta">df.fasta</code></td>
<td>
<p>a data frame with seq_name and its DNA sequence.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about ENc <a href="https://academic.oup.com/nar/article-abstract/14/13/5125/1143812?redirectedFrom=fulltext">Sharp et al., 1986</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed RSCU values for each codon for each DNA sequences within df.fasta.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]

# Calculate RSCU
RSCU.H &lt;- RSCU.values(fasta.h)
RSCU.V &lt;- RSCU.values(fasta.v)

</code></pre>

<hr>
<h2 id='SCUO.values'>Synonymous codon usage eorderliness (SCUO)</h2><span id='topic+SCUO.values'></span>

<h3>Description</h3>

<p>Measure the Synonymous Codon Usage Eorderliness (SCUO) of DNA sequence (Wan et al., 2004).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SCUO.values(df.fasta,genetic.code = "1",threshold=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SCUO.values_+3A_df.fasta">df.fasta</code></td>
<td>
<p>a data frame with seq_name and its DNA sequence.</p>
</td></tr>
<tr><td><code id="SCUO.values_+3A_genetic.code">genetic.code</code></td>
<td>
<p>a single string that uniquely identifies a genetic code to use.</p>
</td></tr>
<tr><td><code id="SCUO.values_+3A_threshold">threshold</code></td>
<td>
<p>optional numeric, specifying sequence length, in codons, used for filtering.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about ENc <a href="https://bmcevolbiol.biomedcentral.com/articles/10.1186/1471-2148-4-19">Wan et al., 2004</a>.
</p>


<h3>Value</h3>

<p>A data.frame containing the computed SCUO values for each DNA sequences within df.fasta.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]

# Calculate SCUO

SCUO.df &lt;- SCUO.values(fasta.v)

</code></pre>

<hr>
<h2 id='SiD.value'>Similarity Index (SiD)</h2><span id='topic+SiD.value'></span>

<h3>Description</h3>

<p>Measure the Similarity Index (SiD) between a virus and its host codon usage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SiD.value(rscu.host,rscu.virus)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SiD.value_+3A_rscu.host">rscu.host</code></td>
<td>
<p>a data frame with RSCU a host codon values.</p>
</td></tr>
<tr><td><code id="SiD.value_+3A_rscu.virus">rscu.virus</code></td>
<td>
<p>a data frame with RSCU a virus codon values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about SiD <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0077239">Zhou et al., 2013</a>.
</p>


<h3>Value</h3>

<p>A numeric represent a SiD value.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>


# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]
RSCU.H &lt;- RSCU.values(fasta.h)
RSCU.V &lt;- RSCU.values(fasta.v)

# Calculate SiD
SiD &lt;- SiD.value(RSCU.host, RSCU.virus)

</code></pre>

<hr>
<h2 id='vhcub'>vhcub: A package to analysis the co-adaptation of codon usage between a virus and its host.</h2><span id='topic+vhcub'></span><span id='topic+vhcub-package'></span>

<h3>Description</h3>

<p>vhcub can calculate various codon usage bias measurements as; effective number of codons (ENc),
codon adaptation index (CAI), relative  codon deoptimization index (RCDI), similarity index (SiD),
synonymous codon usage eorderliness (SCUO) and, relative synonymous codon usage (RSCU).
Also, it provides a statistical dinucleotide over- and underrepresentation with three different models.
Implement several methods for visualization of codon usage as ENc.GC3plot and PR2.plot.
</p>


<h3>vhcub functions</h3>

<p>fasta.read: read fasta format files and convert it to data.frame.
</p>
<p>GC.content: calculates overall GC content as well as GC at first, second, and third codon positions.
</p>
<p>RSCU.values: measure the Relative Synonymous Codon Usage (RSCU) of DNA sequence.
</p>
<p>SCUO.values: measure the Synonymous Codon Usage Eorderliness (SCUO) of DNA sequence.
</p>
<p>RCDI.values: measure the Relative Codon Deoptimization Index (RCDI) of DNA sequence.
</p>
<p>CAI.values: measure the Codon Adaptation  Index  (CAI)  Sharp  and  Li  (1987), of DNA sequence.
</p>
<p>ENc.values: measure the Effective Number of Codons (ENc) of DNA sequence. Using its modified version.
</p>
<p>dinuc.syncodon: measure of statistical dinucleotide over- and underrepresentation; by allows for random sequence generation by shuffling (with/without replacement) of synonymous codons.
</p>
<p>dinuc.codon: measure of statistical dinucleotide over- and underrepresentation; by allows for random sequence generation by shuffling (with/without replacement) of codons.
</p>
<p>dinuc.base: measure of statistical dinucleotide over- and underrepresentation; by allows for random sequence generation by shuffling (with/without replacement) of all bases in the sequence.
</p>
<p>ENc.GC3plot: make an ENc-GC3 scatterplot. Where the y-axis represents the ENc values and the x-axis represents the GC3 content. The red fitting line shows the expected ENc values when codon usage bias affected solely by GC3.
</p>
<p>PR2.plot: make a Parity rule 2 (PR2) plot, where the AT-bias [A3/(A3 +T3)] at the third codon position of the four-codon amino acids of entire genes is the ordinate and the GC-bias [G3/(G3 +C3)] is the abscissa. The center of the plot, where both coordinates are 0.5, is where A = U and G = C (PR2), with no bias between the influence of the mutation and selection rates.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read DNA from fasta files
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]
# calculate GC content
gc.df &lt;- GC.content(fasta.v)
# measure of statistical dinucleotide over- and underrepresentation
syncodon &lt;- dinuc.syncodon(fasta.v,permutations=10)
base &lt;- dinuc.base(fasta.v,permutations=10)
codon &lt;- dinuc.codon(fasta.v,permutations=10)
# calculate ENc
enc.df &lt;- ENc.values(fasta.v)
enc.df.h &lt;- ENc.values(fasta.h)
# calculate SCUO and CAI
SCUO.df &lt;- SCUO.values(fasta.v)
cai.df &lt;- CAI.values(fasta.v,enc.df.h, fasta.h)
# calculate RSCU
RSCU.H &lt;- RSCU.values(fasta.h)
RSCU.V &lt;- RSCU.values(fasta.v)
# calculate SiD
SiD &lt;- SiD.value(RSCU.H,RSCU.V)
# calculate RCDI
rcdi.df &lt;- RCDI.values(fasta.v,fasta.h, enc.df.h)
# plot ENc.GC3plot
ENc.GC3plot(enc.df,gc.df)
# plot PR2.plot
PR2.plot(fasta.v)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
