<!DOCTYPE html><html lang="en"><head><title>Help for package vhica</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {vhica}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#vhica-package'>
<p>Vertical and Horizontal Inheritance Consistence Analysis</p></a></li>
<li><a href='#CUB'>
<p>Computes the Codon Usage Bias of DNA sequences</p></a></li>
<li><a href='#div'>
<p>Computation of the synonymous divergence between sequences</p></a></li>
<li><a href='#image.vhica'>
<p>Consistency matrix for a transposable element in the VHICA analysis.</p></a></li>
<li><a href='#plot.vhica'>
<p>Plots a VHICA regression between two species.</p></a></li>
<li><a href='#read.vhica'>
<p>Reads divergence and codon usage data files for the VHICA method.</p></a></li>
<li><a href='#summary.vhicaimage'>
<p>Provides a data.frame that nicely displays the information returned by <code>image.vhica</code>.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Vertical and Horizontal Inheritance Consistence Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-03-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Arnaud Le Rouzic</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ape, plotrix, parallel, seqinr, gtools</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arnaud Le Rouzic &lt;arnaud.le-rouzic@universite-paris-saclay.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The "Vertical and Horizontal Inheritance Consistence Analysis" method is described in the following publication: "VHICA: a new method to discriminate between vertical and horizontal transposon transfer: application to the mariner family within Drosophila" by G. Wallau. et al. (2016) &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmsv341">doi:10.1093/molbev/msv341</a>&gt;. The purpose of the method is to detect horizontal transfers of transposable elements, by contrasting the divergence of transposable element sequences with that of regular genes. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-07 14:45:41 UTC; lerouzic</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-08 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='vhica-package'>
Vertical and Horizontal Inheritance Consistence Analysis
</h2><span id='topic+vhica-package'></span><span id='topic+vhica'></span>

<h3>Description</h3>

<p>The package implements the VHICA method described in Wallau et al. (in prep). The purpose of the method is to detect horizontal transfers of transposable elements, by contrasting the divergence of transposable element sequences with that of regular genes. Two files should be provided, for both a set of reference genes and transposable element sequences: (i) pairwise divergence across species (e.g., dS), (ii) codon usage bias for all genes and elements in all species.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> vhica</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-v2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package contains three main functions.
</p>

<ul>
<li> <p><code><a href="#topic+read.vhica">read.vhica</a></code>:  reads sequence files and generates an object of class <code>vhica</code> that will be used for further analysis.
</p>
</li>
<li> <p><code><a href="#topic+plot.vhica">plot.vhica</a></code>:  plots the VHICA regression between two species, and displays how far transposable elements (or any kind of other sequences) are from the reference genes.
</p>
</li>
<li> <p><code><a href="#topic+image.vhica">image.vhica</a></code>:  plots the consistency of a specific element across all species, which makes it possible to build evolutionary scenarios.
</p>
</li></ul>

<p>In addition, it provides tools to calculate divergence (<code><a href="#topic+div">div</a></code>) and codon usage bias (<code><a href="#topic+CUB">CUB</a></code>), which are necessary to apply the VHICA method.
</p>


<h3>Author(s)</h3>

<p>Implementation: Arnaud Le Rouzic &lt;arnaud.le-rouzic@universite-paris-saclay.fr&gt; <br />
Scientists who designed the method: Gabriel Wallau, Aurélie Hua-Van, Arnaud Le~Rouzic.
</p>
<p>Maintainer: Arnaud Le Rouzic &lt;arnaud.le-rouzic@universite-paris-saclay.fr&gt;
</p>
<p>Repository: https://github.com/lerouzic/vhica
</p>


<h3>References</h3>

<p>Gabriel Luz Wallau, Arnaud Le Rouzic, Pierre Capy, Elgion Loreto, Aurélie Hua-Van. VHICA: A new method to discriminate between vertical and horizontal transposon transfer: application to the mariner family within Drosophila. Molecular biology and evolution 33 (4), 1094-1109.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.cb &lt;- system.file("extdata", "mini-cbias.txt", package="vhica")
file.div &lt;- system.file("extdata", "mini-div.txt", package="vhica")
file.tree &lt;- if(require("ape")) system.file("extdata", "phylo.nwk", package="vhica") else NULL
vc &lt;- read.vhica(cb.filename=file.cb, div.filename=file.div)
plot(vc, "dere", "dana")
im &lt;- image(vc, "mellifera:6", treefile=file.tree, skip.void=TRUE)
summary(im)
</code></pre>

<hr>
<h2 id='CUB'>
Computes the Codon Usage Bias of DNA sequences
</h2><span id='topic+CUB'></span>

<h3>Description</h3>

<p>The function reads aligned sequences in a fasta file and estimates the codon usage bias for each sequence. Several methods exist to estimate CUB; so far, only the &quot;Effective Number of Codons&quot; (ENC) calculation is available. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CUB(file = NULL, sequence = NULL, method = "ENC")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CUB_+3A_file">file</code></td>
<td>
<p>FASTA file in which aligned sequences are stored.
</p>
</td></tr>
<tr><td><code id="CUB_+3A_sequence">sequence</code></td>
<td>
<p>Alternatively, the result of seqinr::read.fasta.
</p>
</td></tr>
<tr><td><code id="CUB_+3A_method">method</code></td>
<td>
<p>The method used to compute CUB. &quot;ENC&quot;: Effective Number of Codons, as described in Wright (1990). 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named vector of CUB scores. Names correspond to sequence names in the dataset. 
</p>


<h3>Author(s)</h3>

<p>Aurelie Hua-Van and Arnaud Le Rouzic.
</p>


<h3>References</h3>

<p>Wright, F. (1990). The 'effective number of codons' used in a gene. Gene, 87(1), 23-29.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+div">div</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	seq.file &lt;- system.file("extdata/Genes", "Amd.fas", package="vhica")
	CUB(seq.file)
</code></pre>

<hr>
<h2 id='div'>
Computation of the synonymous divergence between sequences
</h2><span id='topic+div'></span>

<h3>Description</h3>

<p>The divergence between DNA sequences can be synonymous (neutral) or non-synonymous. Synonymous differences are generally considered as a better proxy for evolutionary divergence, as it is not affected by selection. This function computes the synonymous divergence between sequences. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div(file = NULL, sequence = NULL, sqs = NULL, method = "LWL85", 
	pairwise = TRUE, max.lim = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="div_+3A_file">file</code></td>
<td>
<p>FASTA file in which aligned sequences are stored.
</p>
</td></tr>
<tr><td><code id="div_+3A_sequence">sequence</code></td>
<td>
<p>Alternatively, the result of seqinr::read.fasta.
</p>
</td></tr>
<tr><td><code id="div_+3A_sqs">sqs</code></td>
<td>
<p>Vector of sequence names to be compared. If not provided, all pairwise comparisons will be performed.
</p>
</td></tr>
<tr><td><code id="div_+3A_method">method</code></td>
<td>
<p>Method used to compute the divergence. So far, only the LWL85 method (from Li et al. 1985).
</p>
</td></tr>
<tr><td><code id="div_+3A_pairwise">pairwise</code></td>
<td>
<p>Boolean: should the divergence be calculated for each pair of sequences or on the whole dataset? This is of particular importance when indels (gaps) are present in sequences, as codons with gaps are generally discarded by most methods. Setting this option to <code>TRUE</code> is thus more likely to give accurate results with multiple-gap sequences, but the calculation will also be slower.
</p>
</td></tr>
<tr><td><code id="div_+3A_max.lim">max.lim</code></td>
<td>
<p>Maximum value for divergence. Depending on the algorithm, various corrections can bring the divergence value above 100%. Values larger than <code>max.lim</code> will be replaced by <code>NA</code>s, as they can be problematic for further statistical tests.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>LWL85</code> method is a wrapper around the <code>kaks</code> function from the <code>seqinr</code> package. 
</p>


<h3>Value</h3>

<p>A 3-column data frame with the following fields:
</p>

<ul>
<li><p>div: The divergence score	
</p>
</li>
<li><p>sq1: The first sequence in the comparison
</p>
</li>
<li><p>sq2: The second sequence in the comparison
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Aurelie Hua-Van and Arnaud Le Rouzic
</p>


<h3>References</h3>

<p>Li, W. H., Wu, C. I., &amp; Luo, C. C. (1985). A new method for estimating synonymous and nonsynonymous rates of nucleotide substitution considering the relative likelihood of nucleotide and codon changes. Molecular biology and evolution, 2(2), 150-174.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CUB">CUB</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	seq.file &lt;- system.file("extdata/Genes", "Amd.fas", package="vhica")
	div(seq.file)
</code></pre>

<hr>
<h2 id='image.vhica'>
Consistency matrix for a transposable element in the VHICA analysis.  
</h2><span id='topic+image.vhica'></span>

<h3>Description</h3>

<p>This function plots a composite figure summarizing the evolutionary properties of a transposable element in a group of related species. Discrepancies may indicate horizontal transfers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vhica'
image(x, element = "", H1.test = "bilat", treefile = NULL, 
skip.void = FALSE, species = NULL, p.threshold = 0.05, 
p.adjust.method = "bonferroni",  ncolors = 1024, 
main = element, threshcol=0.1, colsqueeze=1, 
species.font.family="mono", species.font.cex=1, 
    max.spname.length=10, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="image.vhica_+3A_x">x</code></td>
<td>

<p>An object of class <code>vhica</code>, created by the function <code><a href="#topic+read.vhica">read.vhica</a></code>.
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_element">element</code></td>
<td>

<p>The name of the transposable element, as specified in the data files. If the element is not present in the data, the program halts. 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_h1.test">H1.test</code></td>
<td>

<p>A value among <code>"bilat"</code>, <code>"lower"</code>, or <code>"greater"</code>. 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_treefile">treefile</code></td>
<td>

<p>A Newick file containing a phylogenetic tree. Species names in the tree need to match the data. If absent, the figure will not display the phylogenetic relationship (which makes the interpretation impossible). 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_skip.void">skip.void</code></td>
<td>

<p>Whether or not the figure should show species that do not contain the transposable element. 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_species">species</code></td>
<td>

<p>A named character vector to display pretty species names. The names of the vector are the real species names (as they will appear in the figure), the content of the vector are species codes as in the data files. 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_p.threshold">p.threshold</code></td>
<td>

<p>Threshold for the p-value (above which the color gradient increases). 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_p.adjust.method">p.adjust.method</code></td>
<td>

<p>As documented in <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>. 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_ncolors">ncolors</code></td>
<td>

<p>Number of colors in the gradient. 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_main">main</code></td>
<td>

<p>Main title of the figure (default: the name of the transposable element). 
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_threshcol">threshcol</code></td>
<td>

<p>Part of the color spectrum devoted to non-significant values.
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_colsqueeze">colsqueeze</code></td>
<td>

<p>Values larger than 1 shrink the color gradient around the threshold.
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_species.font.family">species.font.family</code></td>
<td>

<p>Font family for the species names.
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_species.font.cex">species.font.cex</code></td>
<td>

<p>Font size of the species names.
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_max.spname.length">max.spname.length</code></td>
<td>

<p>Maximum length of species names. Longer labels are truncated.
</p>
</td></tr>
<tr><td><code id="image.vhica_+3A_...">...</code></td>
<td>

<p>Further arguments to the generic function <code><a href="graphics.html#topic+image">image</a></code>. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The figure displays in blue TE copies that are more divergent than expected between species, and in red copies that are less divergent than expected. If several lineages of copies are present in a species, the table will be split to display both lineages. Keys for the interpretation of the pattern and the reconstruction of an evolutionary scenario are provided in the original publication. 
</p>


<h3>Value</h3>

<p>The function returns (invisibly) a list of elements (object of class <code>vhicaimage</code>) which can be used for further analysis: <code>tree</code> contains the phylogenetic tree (object of class <code>"phylo"</code>), <code>species</code> is the vector of species, <code>stats</code> is a matrix of log10(P-values) (positive elements are minus log10(P) corresponding to positive residuals), codedS is a matrix reminding the divergence rates from the data. Calling the method <code><a href="#topic+summary.vhicaimage">summary.vhicaimage</a></code> on this object returns a nicely formatted data frame. 
</p>


<h3>Author(s)</h3>

<p>Implementation: Arnaud Le Rouzic &lt;lerouzic@legs.cnrs-gif.fr&gt; <br />
Scientists who designed the method: Gabriel Wallau, Aurélie Hua-Van, Arnaud Le~Rouzic.
</p>


<h3>References</h3>

<p>Gabriel Luz Wallau, Arnaud Le Rouzic, Pierre Capy, Elgion Loreto, Aurélie Hua-Van. VHICA: A new method to discriminate between vertical and horizontal transposon transfer: application to the mariner family within Drosophila. Molecular biology and evolution 33 (4), 1094-1109.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.vhica">read.vhica</a></code>, <code><a href="#topic+plot.vhica">plot.vhica</a></code>, <code><a href="#topic+summary.vhicaimage">summary.vhicaimage</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.cb &lt;- system.file("extdata", "mini-cbias.txt", package="vhica")
file.div &lt;- system.file("extdata", "mini-div.txt", package="vhica")
file.tree &lt;- if(require("ape")) system.file("extdata", "phylo.nwk", package="vhica") else NULL
vc &lt;- read.vhica(cb.filename=file.cb, div.filename=file.div)
plot(vc, "dere", "dana")
im &lt;- image(vc, "mellifera:6", treefile=file.tree, skip.void=TRUE)
summary(im)
</code></pre>

<hr>
<h2 id='plot.vhica'>
Plots a VHICA regression between two species. 
</h2><span id='topic+plot.vhica'></span>

<h3>Description</h3>

<p>The VHICA method is based on a contrast between gene divergence and codon usage bias. A regression between divergence and codon usage provides a reference, and sequences of interest (typically, transposable elements) will be compared to the reference genes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vhica'
plot(x, sp1 = NULL, sp2 = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.vhica_+3A_x">x</code></td>
<td>

<p>An object of class <code>vhica</code>, created by <code><a href="#topic+read.vhica">read.vhica</a></code>. 
</p>
</td></tr>
<tr><td><code id="plot.vhica_+3A_sp1">sp1</code></td>
<td>

<p>Name of the first species, as in the data files. 
</p>
</td></tr>
<tr><td><code id="plot.vhica_+3A_sp2">sp2</code></td>
<td>

<p>Name of the second species, as in the data files. 
</p>
</td></tr>
<tr><td><code id="plot.vhica_+3A_...">...</code></td>
<td>

<p>Additional options for <code><a href="base.html#topic+plot">plot</a></code>. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The resulting figure displays genes as circles, and transposable elements as symbols. 
</p>


<h3>Author(s)</h3>

<p>Implementation: Arnaud Le Rouzic &lt;lerouzic@legs.cnrs-gif.fr&gt;<br />
Scientists who designed the method: Gabriel Wallau, Aurélie Hua-Van, Arnaud Le~Rouzic.
</p>


<h3>References</h3>

<p>Gabriel Luz Wallau, Arnaud Le Rouzic, Pierre Capy, Elgion Loreto, Aurélie Hua-Van. VHICA: A new method to discriminate between vertical and horizontal transposon transfer: application to the mariner family within Drosophila. Molecular biology and evolution 33 (4), 1094-1109. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.vhica">read.vhica</a></code>, <code><a href="#topic+image.vhica">image.vhica</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.cb &lt;- system.file("extdata", "mini-cbias.txt", package="vhica")
file.div &lt;- system.file("extdata", "mini-div.txt", package="vhica")
file.tree &lt;- if(require("ape")) system.file("extdata", "phylo.nwk", package="vhica") else NULL
vc &lt;- read.vhica(cb.filename=file.cb, div.filename=file.div)
plot(vc, "dere", "dana")
image(vc, "mellifera:6", treefile=file.tree, skip.void=TRUE)
</code></pre>

<hr>
<h2 id='read.vhica'>
Reads divergence and codon usage data files for the VHICA method. 
</h2><span id='topic+read.vhica'></span>

<h3>Description</h3>

<p>The VHICA method relies on two sources of information: (i) the divergence between sequences, and (ii) the codon usage bias. This function reads two data files and creates an object of class <code>vhica</code> that can be further explored by <code><a href="#topic+plot.vhica">plot.vhica</a></code> and <code><a href="#topic+image.vhica">image.vhica</a></code>. Input can be either (1) two vectors of fasta file names (one for the genes, one for the putatively transfered genes), or (2) already processed files containing codon usage bias and divergence data (see Details). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.vhica(gene.fasta=NULL, target.fasta=NULL, 
	cb.filename=NULL, div.filename=NULL, 
	reference = "Gene", divergence = "dS", 
	CUB.method="ENC", div.method="LWL85", div.pairwise=TRUE, 
	div.max.lim=3, species.sep="_", gene.sep=".", family.sep=".", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.vhica_+3A_gene.fasta">gene.fasta</code></td>
<td>

<p>Sequence files (FASTA format) containing the aligned sequences 
(respecting the translation phase) for all species of the reference genes.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_target.fasta">target.fasta</code></td>
<td>

<p>Sequence files (FASTA format) containing the aligned sequence 
of the putatively transfered genes.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_cb.filename">cb.filename</code></td>
<td>

<p>File name for the codon usage bias data. If FASTA files are provided, this file will be created.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_div.filename">div.filename</code></td>
<td>

<p>File name for the divergence data. If FASTA files are provided, this file will be created.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_reference">reference</code></td>
<td>

<p>Name of the reference type in the codon usage file. Default is &quot;Gene&quot;.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_divergence">divergence</code></td>
<td>

<p>Name of the divergence column in the divergence file. Default is &quot;dS&quot;.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_cub.method">CUB.method</code></td>
<td>

<p>Method to be used for Codon Usage Bias calculation (see <code><a href="#topic+CUB">CUB</a></code>).</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_div.method">div.method</code></td>
<td>

<p>Method to be used for divergence calculation (see <code><a href="#topic+div">div</a></code>).</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_div.pairwise">div.pairwise</code></td>
<td>

<p>Whether divergence should be calculated from the whole alignment of between pairs of sequences
(see <code><a href="#topic+div">div</a></code>).</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_div.max.lim">div.max.lim</code></td>
<td>
<p>Maximum divergence score. Estimated divergence much larger than 100% are likely 
to be problematic and should not be considered.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_species.sep">species.sep</code></td>
<td>

<p>Separator for species (or equivalent) labels in sequence names. Any character string following 
this separator will be disregarded &ndash; be careful about potential duplicates.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_gene.sep">gene.sep</code></td>
<td>

<p>Separator for gene names from gene sequence files.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_family.sep">family.sep</code></td>
<td>

<p>Separator for target sequence sub-families.</p>
</td></tr>
<tr><td><code id="read.vhica_+3A_...">...</code></td>
<td>

<p>Further parameters for the internal function <code>.reference.regression</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Details about CUB and divergence calculations can be found in <code><a href="#topic+CUB">CUB</a></code> and <code><a href="#topic+div">div</a></code>. If CUB and/or divergence need to be calculated by an external program, it is possible to provide them in the following format: 
</p>

<ul>
<li><p>Codon usage bias
Example of data file:
</p>
<pre>
        Type    sp1     sp2     sp3
CG4231  Gene    42.3    51.1    47.2
CG2214  Gene    47.2    44.9    53.2
Pelem1  TE      36.2    47.0    44.4
...</pre>

<ul>
<li> <p>Row names (or first column)sequence index
</p>
</li>
<li> <p>Type whether the sequence is a reference (default: Gene) or a focal sequence (transposable element, ...)
</p>
</li>
<li> <p>Following columns a measurement of codon bias (ENC, CBI...) for every species
</p>
</li></ul>


</li>
<li><p>Divergence
Example of data file:
</p>
<pre>
seq     dS      sp1     sp2
CG4231  0.84    Dmel    Dsim
CG4231  0.46    Dmel    Dana
CG4231  0.58    Dsim    Dana
CG2214  0.10    Dmel    Dsim
...</pre>

<ul>
<li> <p>First column (or row names): sequence index
</p>
</li>
<li> <p>Second column: divergence measurement
</p>
</li>
<li> <p>Columns 3 and 4: the pair of species on which the divergence is calculated
</p>
</li>
<li> <p>Row names and Col names are allowed but disregarded
</p>
</li></ul>


</li></ul>



<h3>Value</h3>

<p>The function returns an object of class <code>vhica</code>, a list containing:
</p>

<ul>
<li><p>cbias: A codon bias array
</p>
</li>
<li><p>div: The divergence matrix
</p>
</li>
<li><p>reg: The result of all pairwise regressions
</p>
</li>
<li><p>reference: The <code>reference</code> option
</p>
</li>
<li><p>target: The sequence type that is not the reference
</p>
</li>
<li><p>divergence: The <code>divergence</code> option
</p>
</li>
<li><p>family.sep: The character used to indicate TE sub-families
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Implementation: Arnaud Le Rouzic <br />
Scientists who designed the method: Gabriel Wallau, Aurelie Hua-Van, Arnaud Le Rouzic.
</p>


<h3>References</h3>

<p>Gabriel Luz Wallau, Arnaud Le Rouzic, Pierre Capy, Elgion Loreto, Aurelie Hua-Van. VHICA: A new method to discriminate between vertical and horizontal transposon transfer: application to the mariner family within Drosophila. Molecular biology and evolution 33 (4), 1094-1109. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.vhica">plot.vhica</a></code>, <code><a href="#topic+image.vhica">image.vhica</a></code>, <code><a href="#topic+CUB">CUB</a></code>, <code><a href="#topic+div">div</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.cb &lt;- system.file("extdata", "mini-cbias.txt", package="vhica")
file.div &lt;- system.file("extdata", "mini-div.txt", package="vhica")
file.tree &lt;- if(require("ape")) system.file("extdata", "phylo.nwk", package="vhica") else NULL
vc &lt;- read.vhica(cb.filename=file.cb, div.filename=file.div)
plot(vc, "dere", "dana")
image(vc, "mellifera:6", treefile=file.tree, skip.void=TRUE)
</code></pre>

<hr>
<h2 id='summary.vhicaimage'>
Provides a data.frame that nicely displays the information returned by <code><a href="#topic+image.vhica">image.vhica</a></code>. 
</h2><span id='topic+summary.vhicaimage'></span>

<h3>Description</h3>

<p>The <code><a href="#topic+image.vhica">image.vhica</a></code> routine displays visually the statistical support for horizontal transfers, and can return an object of class <code>vhicaimage</code>. The current <code>summary</code> method reorganizes this object into a dataframe that can be displayed or reused in further analysis. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vhicaimage'
summary(object, divrate=NA, p.thresh=1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.vhicaimage_+3A_object">object</code></td>
<td>

<p>An object of class <code>vhicaimage</code>, created by <code><a href="#topic+image.vhica">image.vhica</a></code>. 
</p>
</td></tr>
<tr><td><code id="summary.vhicaimage_+3A_divrate">divrate</code></td>
<td>

<p>Optional divergence rate (in neutral substitutions per Myr). 
</p>
</td></tr>
<tr><td><code id="summary.vhicaimage_+3A_p.thresh">p.thresh</code></td>
<td>

<p>Optional p-value threshold. By default, all data is returned. 
</p>
</td></tr>
<tr><td><code id="summary.vhicaimage_+3A_...">...</code></td>
<td>

<p>Additional options for <code><a href="base.html#topic+summary">summary</a></code> (unused). 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>	The resulting <code>data.frame</code> has 4 or 5 columns. The two first columns are <code>sp1</code> and <code>sp2</code>, the two species between which the horizontal transfer is tested, in an arbitrary order. The column <code>p.value</code> contains the p-value calculated as in <code><a href="#topic+image.vhica">image.vhica</a></code> (including the possible correction for multiple testing). The <code>dS</code> column is a copy of the corresponding divergence from the original data. The last, optional column <code> Time(Mya)</code> is a molecular clock estimate of the time of divergence between the two sequences, based on the divergence rate (when provided). </p>


<h3>Author(s)</h3>

<p>Implementation: Arnaud Le Rouzic &lt;lerouzic@legs.cnrs-gif.fr&gt; and Gabriel Wallau<br />
Scientists who designed the method: Gabriel Wallau, Aurélie Hua-Van, Arnaud Le~Rouzic.
</p>


<h3>References</h3>

<p>Gabriel Luz Wallau, Arnaud Le Rouzic, Pierre Capy, Elgion Loreto, Aurélie Hua-Van. VHICA: A new method to discriminate between vertical and horizontal transposon transfer: application to the mariner family within Drosophila. Molecular biology and evolution 33 (4), 1094-1109.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.vhica">read.vhica</a></code>, <code><a href="#topic+image.vhica">image.vhica</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file.cb &lt;- system.file("extdata", "mini-cbias.txt", package="vhica")
file.div &lt;- system.file("extdata", "mini-div.txt", package="vhica")
file.tree &lt;- if(require("ape")) system.file("extdata", "phylo.nwk", package="vhica") else NULL
vc &lt;- read.vhica(cb.filename=file.cb, div.filename=file.div)
plot(vc, "dere", "dana")
im &lt;- image(vc, "mellifera:6", treefile=file.tree, skip.void=TRUE)
summary(im)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
