<!DOCTYPE html><html><head><title>Help for package viewpoly</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {viewpoly}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#breeding_values'><p>Estimate breeding values - Adapted function from QTLpoly</p></a></li>
<li><a href='#calc_homologprob'><p>Calculates homologues probabilities - Adapted from MAPpoly</p></a></li>
<li><a href='#check_viewmap'><p>Viewmap object sanity check</p></a></li>
<li><a href='#check_viewpoly'><p>Viewpoly object sanity check</p></a></li>
<li><a href='#check_viewqtl'><p>viewqtl object sanity check</p></a></li>
<li><a href='#coord_radar'><p>Change ggplot coordinates to plot radar - From package see</p></a></li>
<li><a href='#data_effects'><p>Get effects information</p></a></li>
<li><a href='#dist_prob_to_class'><p>Returns the class with the highest probability in</p>
a genotype probability distribution. Function from MAPpoly.</a></li>
<li><a href='#draw_map_shiny'><p>Draws linkage map, parents haplotypes and marker doses</p>
Adapted from MAPpoly</a></li>
<li><a href='#filter_non_conforming_classes'><p>Filter non-conforming classes in F1, non double reduced population.</p>
Function from MAPpoly.</a></li>
<li><a href='#get_LOD'><p>Extract the LOD Scores in a <code>'mappoly.map'</code> object</p>
Function from MAPpoly.</a></li>
<li><a href='#gg_color_hue'><p>Color pallet ggplot-like - Adapted from MAPpoly</p></a></li>
<li><a href='#imf_h'><p>Map functions - from MAPpoly</p></a></li>
<li><a href='#import_data_from_polymapR'><p>Import data from polymapR</p></a></li>
<li><a href='#import_phased_maplist_from_polymapR'><p>Import phased map list from polymapR</p></a></li>
<li><a href='#is.prob.data'><p>Is it a probability dataset? Function from MAPpoly.</p></a></li>
<li><a href='#map_summary'><p>Gets summary information from map.</p>
Adapted from MAPpoly</a></li>
<li><a href='#mf_h'><p>Haldane map function. From MAPpoly</p></a></li>
<li><a href='#mrk_chisq_test'><p>Chi-square test. Function from MAPpoly.</p></a></li>
<li><a href='#only_plot_profile'><p>Only the plot part of plot_profile function</p></a></li>
<li><a href='#ph_list_to_matrix'><p>Linkage phase format conversion: list to matrix.</p>
Function from MAPpoly.</a></li>
<li><a href='#ph_matrix_to_list'><p>Linkage phase format conversion: matrix to list. Function from MAPpoly.</p></a></li>
<li><a href='#plot_cm_mb'><p>Scatter plot relating linkage map and genomic positions</p></a></li>
<li><a href='#plot_effects'><p>Plot effects data</p></a></li>
<li><a href='#plot_map_list'><p>Plot a genetic map - Adapted from MAPpoly</p></a></li>
<li><a href='#plot_one_map'><p>Plot a single linkage group with no phase - from MAPpoly</p></a></li>
<li><a href='#plot_profile'><p>Logarithm of <em>P</em>-value (LOP) profile plots. Modified version of QTLpoly function.</p></a></li>
<li><a href='#plot.mappoly.homoprob'><p>Plots mappoly.homoprob from MAPpoly</p></a></li>
<li><a href='#prepare_diaQTL'><p>Converts diaQTL output to viewqtl object</p></a></li>
<li><a href='#prepare_examples'><p>Upload example files</p></a></li>
<li><a href='#prepare_map'><p>prepare maps for plot - from MAPpoly</p></a></li>
<li><a href='#prepare_map_custom_files'><p>Converts map information in custom format files to viewmap object</p></a></li>
<li><a href='#prepare_MAPpoly'><p>Converts list of mappoly.map object into viewmap object</p></a></li>
<li><a href='#prepare_polymapR'><p>Converts polymapR ouputs to viewmap object</p></a></li>
<li><a href='#prepare_polyqtlR'><p>Converts polyqtlR outputs to viewqtl object</p></a></li>
<li><a href='#prepare_qtl_custom_files'><p>Converts QTL information in custom files to viewqtl object</p></a></li>
<li><a href='#prepare_QTLpoly'><p>Converts QTLpoly outputs to viewqtl object</p></a></li>
<li><a href='#run_app'><p>Run the Shiny Application</p></a></li>
<li><a href='#segreg_poly'><p>Polysomic segregation frequency - Function from MAPpoly</p></a></li>
<li><a href='#select_haplo'><p>Plot selected haplotypes</p></a></li>
<li><a href='#summary_maps'><p>Summary maps - adapted from MAPpoly</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Shiny App to Visualize Genetic Maps and QTL Analysis in
Polyploid Species</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cristiane Taniguti &lt;chtaniguti@tamu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a graphical user interface to integrate, visualize and explore results 
            from linkage and quantitative trait loci analysis, together with genomic information for autopolyploid 
            species. The app is meant for interactive use and allows users to optionally upload different sources 
            of information, including  gene annotation and alignment files, enabling the exploitation and search for 
            candidate genes in a genome browser. In its current version, 'VIEWpoly' supports inputs from 'MAPpoly', 
            'polymapR', 'diaQTL', 'QTLpoly' and 'polyqtlR' packages. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>shiny (&ge; 1.6.0), shinyjs, shinythemes, shinyWidgets,
shinydashboard, config (&ge; 0.3.1), golem (&ge; 0.3.1), JBrowseR,
dplyr, tidyr, DT, ggplot2, ggpubr, plotly, vroom, abind,
reshape2, markdown, stats, curl</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mmollina/viewpoly">https://github.com/mmollina/viewpoly</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mmollina/viewpoly/issues">https://github.com/mmollina/viewpoly/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), shinytest, rlang, pkgload, vdiffr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-01 17:10:05 UTC; chtan</td>
</tr>
<tr>
<td>Author:</td>
<td>Cristiane Taniguti [aut, cre],
  Gabriel de Siqueira Gesteira [aut],
  Jeekin Lau [ctb],
  Guilherme da Silva Pereira [ctb],
  David Byrne [ctb],
  Zhao-Bang Zeng [ctb],
  Oscar Riera-Lizarazu [ctb],
  Marcelo Mollinari [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-01 17:30:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='breeding_values'>Estimate breeding values - Adapted function from QTLpoly</h2><span id='topic+breeding_values'></span>

<h3>Description</h3>

<p>Estimate breeding values - Adapted function from QTLpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breeding_values(qtl_info, probs, selected_mks, blups, beta.hat, pos)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="breeding_values_+3A_qtl_info">qtl_info</code></td>
<td>
<p>data.frame with: LG - linkage group ID; Pos - position in linkage map (cM); 
Pheno - phenotype ID; Pos_lower - lower position of confidence interval; 
Pos_upper - upper position of the confidence interval; Pval - QTL p-value; h2 - herdability</p>
</td></tr>
<tr><td><code id="breeding_values_+3A_probs">probs</code></td>
<td>
<p>data.frame with first column (named 'ind') as individuals ID and next columns 
named with markers ID and containing the genotype probability at each marker</p>
</td></tr>
<tr><td><code id="breeding_values_+3A_selected_mks">selected_mks</code></td>
<td>
<p>data.frame with: LG - linkage group ID; mk - marker ID; pos - position in linkage map (cM)</p>
</td></tr>
<tr><td><code id="breeding_values_+3A_blups">blups</code></td>
<td>
<p>data.frame with: haplo - haplotype ID; pheno - phenotype ID; qtl - QTL ID; u.hat - QTL estimated BLUPs</p>
</td></tr>
<tr><td><code id="breeding_values_+3A_beta.hat">beta.hat</code></td>
<td>
<p>data.frame with: pheno - phenotype ID; beta.hat - estimated beta</p>
</td></tr>
<tr><td><code id="breeding_values_+3A_pos">pos</code></td>
<td>
<p>selected QTL position (cM)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame containing breeding values
</p>

<hr>
<h2 id='calc_homologprob'>Calculates homologues probabilities - Adapted from MAPpoly</h2><span id='topic+calc_homologprob'></span>

<h3>Description</h3>

<p>Calculates homologues probabilities - Adapted from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_homologprob(probs, selected_mks, selected_lgs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_homologprob_+3A_probs">probs</code></td>
<td>
<p>data.frame with first column (named 'ind') as individuals ID and next columns named with markers ID and containing the genotype probability at each marker</p>
</td></tr>
<tr><td><code id="calc_homologprob_+3A_selected_mks">selected_mks</code></td>
<td>
<p>data.frame with: LG - linkage group ID; mk - marker ID; pos - position in linkage map (cM)</p>
</td></tr>
<tr><td><code id="calc_homologprob_+3A_selected_lgs">selected_lgs</code></td>
<td>
<p>vector containing selected LGs ID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>mappoly.homoprob</code>
</p>

<hr>
<h2 id='check_viewmap'>Viewmap object sanity check</h2><span id='topic+check_viewmap'></span>

<h3>Description</h3>

<p>Viewmap object sanity check
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_viewmap(viewmap_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_viewmap_+3A_viewmap_obj">viewmap_obj</code></td>
<td>
<p>an object of class <code>viewmap</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>if consistent, returns 0. If not consistent, returns a 
vector with a number of tests, where <code>TRUE</code> indicates
a failed test.
</p>


<h3>Author(s)</h3>

<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>

<hr>
<h2 id='check_viewpoly'>Viewpoly object sanity check</h2><span id='topic+check_viewpoly'></span>

<h3>Description</h3>

<p>Viewpoly object sanity check
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_viewpoly(viewpoly_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_viewpoly_+3A_viewpoly_obj">viewpoly_obj</code></td>
<td>
<p>an object of class <code>viewpoly</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>if consistent, returns 0. If not consistent, returns a 
vector with a number of tests, where <code>TRUE</code> indicates
a failed test.
</p>


<h3>Author(s)</h3>

<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>

<hr>
<h2 id='check_viewqtl'>viewqtl object sanity check</h2><span id='topic+check_viewqtl'></span>

<h3>Description</h3>

<p>viewqtl object sanity check
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_viewqtl(viewqtl_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_viewqtl_+3A_viewqtl_obj">viewqtl_obj</code></td>
<td>
<p>an object of class <code>viewqtl</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>if consistent, returns 0. If not consistent, returns a 
vector with a number of tests, where <code>TRUE</code> indicates
a failed test.
</p>


<h3>Author(s)</h3>

<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>

<hr>
<h2 id='coord_radar'>Change ggplot coordinates to plot radar - From package see</h2><span id='topic+coord_radar'></span>

<h3>Description</h3>

<p>Change ggplot coordinates to plot radar - From package see
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_radar(theta = "x", start = 0, direction = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coord_radar_+3A_theta">theta</code></td>
<td>
<p>ariable to map angle to (x or y)</p>
</td></tr>
<tr><td><code id="coord_radar_+3A_start">start</code></td>
<td>
<p>offset of starting point from 12 o'clock in radians. Offset is applied clockwise or anticlockwise depending on value of direction.</p>
</td></tr>
<tr><td><code id="coord_radar_+3A_direction">direction</code></td>
<td>
<p>1, clockwise; -1, anticlockwise</p>
</td></tr>
</table>

<hr>
<h2 id='data_effects'>Get effects information</h2><span id='topic+data_effects'></span>

<h3>Description</h3>

<p>Get effects information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_effects(
  qtl_info,
  effects,
  pheno.col = NULL,
  parents = NULL,
  lgs = NULL,
  groups = NULL,
  position = NULL,
  software,
  design = c("bar", "circle", "digenic")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data_effects_+3A_qtl_info">qtl_info</code></td>
<td>
<p>data.frame with: LG - linkage group ID; Pos - position in linkage map (cM); 
Pheno - phenotype ID; Pos_lower - lower position of confidence interval; 
Pos_upper - upper position of the confidence interval; Pval - QTL p-value; h2 - herdability</p>
</td></tr>
<tr><td><code id="data_effects_+3A_effects">effects</code></td>
<td>
<p>data.frame with: pheno - phenotype ID; qtl.id - QTL ID; haplo - haplotype ID; effect - haplotype effect value</p>
</td></tr>
<tr><td><code id="data_effects_+3A_pheno.col">pheno.col</code></td>
<td>
<p>integer identifying phenotype</p>
</td></tr>
<tr><td><code id="data_effects_+3A_parents">parents</code></td>
<td>
<p>vector with parents ID</p>
</td></tr>
<tr><td><code id="data_effects_+3A_lgs">lgs</code></td>
<td>
<p>vector of integers with linkage group ID of selected QTL/s</p>
</td></tr>
<tr><td><code id="data_effects_+3A_groups">groups</code></td>
<td>
<p>vector of integers with selected linkage group ID</p>
</td></tr>
<tr><td><code id="data_effects_+3A_position">position</code></td>
<td>
<p>vector of centimorgan positions of selected QTL/s</p>
</td></tr>
<tr><td><code id="data_effects_+3A_software">software</code></td>
<td>
<p>character defining which software was used for QTL analysis. Currently support for: QTLpoly, diaQTL and polyqtlR.</p>
</td></tr>
<tr><td><code id="data_effects_+3A_design">design</code></td>
<td>
<p>character defining the graphic design. Options: 'bar' - barplot of the effects; 
'circle' - circular plot of the effects (useful to compare effects of different traits); 
'digenic' - heatmap plotting sum of additive effects (bottom diagonal) and digenic effects (top diagonal) when present</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot graphic
</p>

<hr>
<h2 id='dist_prob_to_class'>Returns the class with the highest probability in 
a genotype probability distribution. Function from MAPpoly.</h2><span id='topic+dist_prob_to_class'></span>

<h3>Description</h3>

<p>Returns the class with the highest probability in 
a genotype probability distribution. Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_prob_to_class(geno, prob.thres = 0.9)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_prob_to_class_+3A_geno">geno</code></td>
<td>
<p>the probabilistic genotypes contained in the object
<code>'mappoly.data'</code></p>
</td></tr>
<tr><td><code id="dist_prob_to_class_+3A_prob.thres">prob.thres</code></td>
<td>
<p>probability threshold to select the genotype. 
Values below this genotype are assumed as missing data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix containing the doses of each genotype and
marker. Markers are disposed in rows and individuals are 
disposed in columns. Missing data are represented by NAs
</p>

<hr>
<h2 id='draw_map_shiny'>Draws linkage map, parents haplotypes and marker doses
Adapted from MAPpoly</h2><span id='topic+draw_map_shiny'></span>

<h3>Description</h3>

<p>Draws linkage map, parents haplotypes and marker doses
Adapted from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_map_shiny(
  left.lim = 0,
  right.lim = 5,
  ch = 1,
  maps,
  ph.p1,
  ph.p2,
  d.p1,
  d.p2,
  snp.names = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_map_shiny_+3A_left.lim">left.lim</code></td>
<td>
<p>covered window in the linkage map start position</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_ch">ch</code></td>
<td>
<p>linkage group ID</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_maps">maps</code></td>
<td>
<p>list containing a vector for each linkage group markers with marker positions (named with marker names)</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_ph.p1">ph.p1</code></td>
<td>
<p>list containing a data.frame for each group with parent 1 estimated phases. The data.frame contain the columns:
1) Character vector with chromosome ID; 2) Character vector with marker ID;
3 to (ploidy number)*2 columns with each parents haplotypes</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_ph.p2">ph.p2</code></td>
<td>
<p>list containing a data.frame for each group with parent 2 estimated phases. See ph.p1 parameter description.</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_d.p1">d.p1</code></td>
<td>
<p>list containing a data.frame for each group with parent 1 dosages. The data.frame contain the columns: 
1) character vector with chromosomes ID; 
2) Character vector with markers ID; 3) Character vector with parent ID; 
4) numerical vector with dosage</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_d.p2">d.p2</code></td>
<td>
<p>list containing a data.frame for each group with parent 2 dosages. See d.p1 parameter description</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_snp.names">snp.names</code></td>
<td>
<p>logical TRUE/FALSE. If TRUE it includes the marker names in the plot</p>
</td></tr>
<tr><td><code id="draw_map_shiny_+3A_rigth.lim">rigth.lim</code></td>
<td>
<p>covered window in the linkage map end position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>graphic representing selected section of a linkage group
</p>

<hr>
<h2 id='filter_non_conforming_classes'>Filter non-conforming classes in F1, non double reduced population.
Function from MAPpoly.</h2><span id='topic+filter_non_conforming_classes'></span>

<h3>Description</h3>

<p>Filter non-conforming classes in F1, non double reduced population.
Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_non_conforming_classes(input.data, prob.thres = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_non_conforming_classes_+3A_input.data">input.data</code></td>
<td>
<p>object of class mappoly</p>
</td></tr>
<tr><td><code id="filter_non_conforming_classes_+3A_prob.thres">prob.thres</code></td>
<td>
<p>threshold for filtering genotypes by genotype probability values. If NULL, the filter is not applied.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>filtered <code>mappoly.data</code> object
</p>

<hr>
<h2 id='get_LOD'>Extract the LOD Scores in a <code>'mappoly.map'</code> object
Function from MAPpoly.</h2><span id='topic+get_LOD'></span>

<h3>Description</h3>

<p>Extract the LOD Scores in a <code>'mappoly.map'</code> object
Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_LOD(x, sorted = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_LOD_+3A_x">x</code></td>
<td>
<p>an object of class <code>mappoly.map</code></p>
</td></tr>
<tr><td><code id="get_LOD_+3A_sorted">sorted</code></td>
<td>
<p>logical. if <code>TRUE</code>, the LOD Scores are displayed
in a decreasing order</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector containing the LOD Scores
</p>

<hr>
<h2 id='gg_color_hue'>Color pallet ggplot-like - Adapted from MAPpoly</h2><span id='topic+gg_color_hue'></span>

<h3>Description</h3>

<p>Color pallet ggplot-like - Adapted from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_color_hue(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_color_hue_+3A_n">n</code></td>
<td>
<p>number of colors</p>
</td></tr>
</table>

<hr>
<h2 id='imf_h'>Map functions - from MAPpoly</h2><span id='topic+imf_h'></span>

<h3>Description</h3>

<p>Map functions - from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imf_h(r)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imf_h_+3A_r">r</code></td>
<td>
<p>vector with recombination fraction values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector with genetic distances
</p>

<hr>
<h2 id='import_data_from_polymapR'>Import data from polymapR</h2><span id='topic+import_data_from_polymapR'></span>

<h3>Description</h3>

<p>Function to import datasets from polymapR. Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_data_from_polymapR(
  input.data,
  ploidy,
  parent1 = "P1",
  parent2 = "P2",
  input.type = c("discrete", "probabilistic"),
  prob.thres = 0.95,
  pardose = NULL,
  offspring = NULL,
  filter.non.conforming = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_data_from_polymapR_+3A_input.data">input.data</code></td>
<td>
<p>a <code>polymapR</code> dataset</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_ploidy">ploidy</code></td>
<td>
<p>the ploidy level</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_parent1">parent1</code></td>
<td>
<p>a character string containing the name (or pattern of genotype IDs) of parent 1</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_parent2">parent2</code></td>
<td>
<p>a character string containing the name (or pattern of genotype IDs) of parent 2</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_input.type">input.type</code></td>
<td>
<p>Indicates whether the input is discrete (&quot;disc&quot;) or probabilistic (&quot;prob&quot;)</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_prob.thres">prob.thres</code></td>
<td>
<p>threshold probability to assign a dosage to offspring. If the probability 
is smaller than <code>thresh.parent.geno</code>, the data point is converted to 'NA'.</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_pardose">pardose</code></td>
<td>
<p>matrix of dimensions (n.mrk x 3) containing the name of the markers in the first column, and the 
dosage of parents 1 and 2 in columns 2 and 3. (see polymapR vignette)</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_offspring">offspring</code></td>
<td>
<p>a character string containing the name (or pattern of genotype IDs) of the offspring 
individuals. If <code>NULL</code> (default) it considers all individuals as offsprings, except 
<code>parent1</code> and <code>parent2</code>.</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_filter.non.conforming">filter.non.conforming</code></td>
<td>
<p>if <code>TRUE</code> exclude samples with non 
expected genotypes under no double reduction. Since markers were already filtered in polymapR, the default is 
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="import_data_from_polymapR_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code> (default), the current progress is shown; if
<code>FALSE</code>, no output is produced</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See examples at <a href="https://rpubs.com/mmollin/tetra_mappoly_vignette">https://rpubs.com/mmollin/tetra_mappoly_vignette</a>.
</p>


<h3>Value</h3>

<p>object of class <code>mappoly.data</code>
</p>


<h3>Author(s)</h3>

<p>Marcelo Mollinari <a href="mailto:mmollin@ncsu.edu">mmollin@ncsu.edu</a>
</p>


<h3>References</h3>

<p>Bourke PM et al: (2019) PolymapR — linkage analysis and genetic map 
construction from F1 populations of outcrossing polyploids. 
_Bioinformatics_ 34:3496–3502.
doi: <a href="https://doi.org/10.1093/bioinformatics/bty1002">10.1093/bioinformatics/bty1002</a>
</p>
<p>Mollinari, M., and Garcia, A.  A. F. (2019) Linkage
analysis and haplotype phasing in experimental autopolyploid
populations with high ploidy level using hidden Markov
models, _G3: Genes, Genomes, Genetics_. 
doi: <a href="https://doi.org/10.1534/g3.119.400378">10.1534/g3.119.400378</a>
</p>

<hr>
<h2 id='import_phased_maplist_from_polymapR'>Import phased map list from polymapR</h2><span id='topic+import_phased_maplist_from_polymapR'></span>

<h3>Description</h3>

<p>Function to import phased map lists from polymapR. Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_phased_maplist_from_polymapR(maplist, mappoly.data, ploidy = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_phased_maplist_from_polymapR_+3A_maplist">maplist</code></td>
<td>
<p>a list of phased maps obtained using function 
<code>create_phased_maplist</code> from package <code>polymapR</code></p>
</td></tr>
<tr><td><code id="import_phased_maplist_from_polymapR_+3A_mappoly.data">mappoly.data</code></td>
<td>
<p>a dataset used to obtain <code>maplist</code>, 
converted into class <code>mappoly.data</code></p>
</td></tr>
<tr><td><code id="import_phased_maplist_from_polymapR_+3A_ploidy">ploidy</code></td>
<td>
<p>the ploidy level</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See examples at <a href="https://rpubs.com/mmollin/tetra_mappoly_vignette">https://rpubs.com/mmollin/tetra_mappoly_vignette</a>.
</p>


<h3>Value</h3>

<p>object of class <code>mappoly.map</code>
</p>


<h3>Author(s)</h3>

<p>Marcelo Mollinari <a href="mailto:mmollin@ncsu.edu">mmollin@ncsu.edu</a>
</p>


<h3>References</h3>

<p>Bourke PM et al: (2019) PolymapR — linkage analysis and genetic map 
construction from F1 populations of outcrossing polyploids. 
_Bioinformatics_ 34:3496–3502.
doi: <a href="https://doi.org/10.1093/bioinformatics/bty1002">10.1093/bioinformatics/bty1002</a>
</p>
<p>Mollinari, M., and Garcia, A.  A. F. (2019) Linkage
analysis and haplotype phasing in experimental autopolyploid
populations with high ploidy level using hidden Markov
models, _G3: Genes, Genomes, Genetics_. 
doi: <a href="https://doi.org/10.1534/g3.119.400378">10.1534/g3.119.400378</a>
</p>

<hr>
<h2 id='is.prob.data'>Is it a probability dataset? Function from MAPpoly.</h2><span id='topic+is.prob.data'></span>

<h3>Description</h3>

<p>Is it a probability dataset? Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.prob.data(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.prob.data_+3A_x">x</code></td>
<td>
<p>object of class <code>mappoly.data</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE/FALSE indicating if genotype probability information is present
</p>

<hr>
<h2 id='map_summary'>Gets summary information from map.
Adapted from MAPpoly</h2><span id='topic+map_summary'></span>

<h3>Description</h3>

<p>Gets summary information from map.
Adapted from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_summary(left.lim = 0, right.lim = 5, ch = 1, maps, d.p1, d.p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map_summary_+3A_left.lim">left.lim</code></td>
<td>
<p>covered window in the linkage map start position</p>
</td></tr>
<tr><td><code id="map_summary_+3A_ch">ch</code></td>
<td>
<p>linkage group ID</p>
</td></tr>
<tr><td><code id="map_summary_+3A_maps">maps</code></td>
<td>
<p>list containing a vector for each linkage group markers with marker positions (named with marker names)</p>
</td></tr>
<tr><td><code id="map_summary_+3A_d.p1">d.p1</code></td>
<td>
<p>list containing a data.frame for each group with parent 1 dosages. The data.frame contain the columns: 
1) character vector with chromosomes ID; 
2) Character vector with markers ID; 3) Character vector with parent ID; 
4) numerical vector with dosage</p>
</td></tr>
<tr><td><code id="map_summary_+3A_d.p2">d.p2</code></td>
<td>
<p>list containing a data.frame for each group with parent 2 dosages. See d.p1 parameter description</p>
</td></tr>
<tr><td><code id="map_summary_+3A_rigth.lim">rigth.lim</code></td>
<td>
<p>covered window in the linkage map end position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with linkage map information: doses;  number snps by group; cM per snp; map size; number of linkage groups
</p>

<hr>
<h2 id='mf_h'>Haldane map function. From MAPpoly</h2><span id='topic+mf_h'></span>

<h3>Description</h3>

<p>Haldane map function. From MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mf_h(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mf_h_+3A_d">d</code></td>
<td>
<p>vector containing recombination fraction values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector with genetic distances estimated with Haldane function
</p>

<hr>
<h2 id='mrk_chisq_test'>Chi-square test. Function from MAPpoly.</h2><span id='topic+mrk_chisq_test'></span>

<h3>Description</h3>

<p>Chi-square test. Function from MAPpoly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrk_chisq_test(x, ploidy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrk_chisq_test_+3A_x">x</code></td>
<td>
<p>data.frame containing dosage information</p>
</td></tr>
<tr><td><code id="mrk_chisq_test_+3A_ploidy">ploidy</code></td>
<td>
<p>integer defining the specie ploidy</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector with p-values for each marker
</p>

<hr>
<h2 id='only_plot_profile'>Only the plot part of plot_profile function</h2><span id='topic+only_plot_profile'></span>

<h3>Description</h3>

<p>Only the plot part of plot_profile function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>only_plot_profile(pl.in)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="only_plot_profile_+3A_pl.in">pl.in</code></td>
<td>
<p>output object from <code>plot_profile</code> when plot == TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot graphic with QTL significance profile
</p>

<hr>
<h2 id='ph_list_to_matrix'>Linkage phase format conversion: list to matrix.
Function from MAPpoly.</h2><span id='topic+ph_list_to_matrix'></span>

<h3>Description</h3>

<p>This function converts linkage phase configurations from list
to matrix form
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ph_list_to_matrix(L, ploidy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ph_list_to_matrix_+3A_l">L</code></td>
<td>
<p>a list of configuration phases</p>
</td></tr>
<tr><td><code id="ph_list_to_matrix_+3A_ploidy">ploidy</code></td>
<td>
<p>ploidy level</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix whose columns represent homologous chromosomes and
the rows represent markers
</p>

<hr>
<h2 id='ph_matrix_to_list'>Linkage phase format conversion: matrix to list. Function from MAPpoly.</h2><span id='topic+ph_matrix_to_list'></span>

<h3>Description</h3>

<p>This function converts linkage phase configurations from matrix
form to list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ph_matrix_to_list(M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ph_matrix_to_list_+3A_m">M</code></td>
<td>
<p>matrix whose columns represent homologous chromosomes and
the rows represent markers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of linkage phase configurations
</p>

<hr>
<h2 id='plot_cm_mb'>Scatter plot relating linkage map and genomic positions</h2><span id='topic+plot_cm_mb'></span>

<h3>Description</h3>

<p>Scatter plot relating linkage map and genomic positions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cm_mb(viewmap, group, range.min, range.max)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_cm_mb_+3A_viewmap">viewmap</code></td>
<td>
<p>object of class <code>viewmap</code></p>
</td></tr>
<tr><td><code id="plot_cm_mb_+3A_group">group</code></td>
<td>
<p>selected group ID</p>
</td></tr>
<tr><td><code id="plot_cm_mb_+3A_range.min">range.min</code></td>
<td>
<p>minimum value of the selected position range</p>
</td></tr>
<tr><td><code id="plot_cm_mb_+3A_range.max">range.max</code></td>
<td>
<p>maximum value of the selected position range</p>
</td></tr>
</table>

<hr>
<h2 id='plot_effects'>Plot effects data</h2><span id='topic+plot_effects'></span>

<h3>Description</h3>

<p>Plot effects data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_effects(
  data_effects.obj,
  software,
  design = c("bar", "circle", "digenic")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_effects_+3A_data_effects.obj">data_effects.obj</code></td>
<td>
<p>output of function <code>data_effects</code></p>
</td></tr>
<tr><td><code id="plot_effects_+3A_software">software</code></td>
<td>
<p>character defining which software was used for QTL analysis. Currently support for: QTLpoly, diaQTL and polyqtlR.</p>
</td></tr>
<tr><td><code id="plot_effects_+3A_design">design</code></td>
<td>
<p>character defining the graphic design. Options: 'bar' - barplot of the effects; 
'circle' - circular plot of the effects (useful to compare effects of different traits); 
'digenic' - heatmap plotting sum of additive effects (bottom diagonal) and digenic effects (top diagonal) when present</p>
</td></tr>
</table>

<hr>
<h2 id='plot_map_list'>Plot a genetic map - Adapted from MAPpoly</h2><span id='topic+plot_map_list'></span>

<h3>Description</h3>

<p>This function plots a genetic linkage map(s)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_map_list(viewmap, horiz = TRUE, col = "ggstyle", title = "Linkage group")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_map_list_+3A_viewmap">viewmap</code></td>
<td>
<p>object of class <code>viewmap</code></p>
</td></tr>
<tr><td><code id="plot_map_list_+3A_horiz">horiz</code></td>
<td>
<p>logical. If FALSE, the maps are plotted vertically with the first map to the left. 
If TRUE  (default), the maps are plotted horizontally with the first at the bottom</p>
</td></tr>
<tr><td><code id="plot_map_list_+3A_col">col</code></td>
<td>
<p>a vector of colors for the bars or bar components (default = 'lightgrey')
<code>ggstyle</code> produces maps using the default <code>ggplot</code> color palette</p>
</td></tr>
<tr><td><code id="plot_map_list_+3A_title">title</code></td>
<td>
<p>a title (string) for the maps (default = 'Linkage group')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> object containing the name of the markers and their genetic position
</p>


<h3>Author(s)</h3>

<p>Marcelo Mollinari, <a href="mailto:mmollin@ncsu.edu">mmollin@ncsu.edu</a>
</p>
<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>


<h3>References</h3>

<p>Mollinari, M., and Garcia, A.  A. F. (2019) Linkage
analysis and haplotype phasing in experimental autopolyploid
populations with high ploidy level using hidden Markov
models, _G3: Genes, Genomes, Genetics_. 
doi: <a href="https://doi.org/10.1534/g3.119.400378">10.1534/g3.119.400378</a>
</p>

<hr>
<h2 id='plot_one_map'>Plot a single linkage group with no phase - from MAPpoly</h2><span id='topic+plot_one_map'></span>

<h3>Description</h3>

<p>Plot a single linkage group with no phase - from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_one_map(x, i = 0, horiz = FALSE, col = "lightgray")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_one_map_+3A_x">x</code></td>
<td>
<p>vector of genetic distances</p>
</td></tr>
<tr><td><code id="plot_one_map_+3A_i">i</code></td>
<td>
<p>margins size</p>
</td></tr>
<tr><td><code id="plot_one_map_+3A_horiz">horiz</code></td>
<td>
<p>logical TRUE/FALSE. If TRUE the map is plotted horizontally.</p>
</td></tr>
<tr><td><code id="plot_one_map_+3A_col">col</code></td>
<td>
<p>color pallete to be used</p>
</td></tr>
</table>

<hr>
<h2 id='plot_profile'>Logarithm of <em>P</em>-value (LOP) profile plots. Modified version of QTLpoly function.</h2><span id='topic+plot_profile'></span>

<h3>Description</h3>

<p>Plots profiled logarithm of score-based <em>P</em>-values (LOP) from individual or combined traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_profile(
  profile,
  qtl_info,
  selected_mks,
  pheno.col = NULL,
  lgs.id = NULL,
  by_range = TRUE,
  range.min = NULL,
  range.max = NULL,
  plot = TRUE,
  software = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_profile_+3A_profile">profile</code></td>
<td>
<p>data.frame with: pheno - phenotype ID; LOP - significance value for the QTL. 
It can be LOP, LOD or DIC depending of the software used</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_qtl_info">qtl_info</code></td>
<td>
<p>data.frame with: LG - linkage group ID; Pos - position in linkage map (cM); 
Pheno - phenotype ID; Pos_lower - lower position of confidence interval; 
Pos_upper - upper position of the confidence interval; Pval - QTL p-value; h2 - herdability</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_selected_mks">selected_mks</code></td>
<td>
<p>data.frame with: LG - linkage group ID; mk - marker ID; pos - position in linkage map (cM)</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_pheno.col">pheno.col</code></td>
<td>
<p>integer identifying phenotype</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_lgs.id">lgs.id</code></td>
<td>
<p>integer identifying linkage group</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_by_range">by_range</code></td>
<td>
<p>logical TRUE/FALSE. If TRUE range.min and range.max will set a colored window in the plot and the other positions will be gray.
If FALSE, range.min and range.max is ignored</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_range.min">range.min</code></td>
<td>
<p>position in centimorgan defining the start of the colored window</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_range.max">range.max</code></td>
<td>
<p>position in centimorgan defining the end of the colored window</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_plot">plot</code></td>
<td>
<p>logical TRUE/FALSE. If FALSE the function return a data.frame with information for <code>only_plot_profile</code> function. 
If TRUE, it returns a ggplot graphic.</p>
</td></tr>
<tr><td><code id="plot_profile_+3A_software">software</code></td>
<td>
<p>character defining which software was used for QTL analysis. Currently support for: QTLpoly, diaQTL and polyqtlR.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot graphic (if plot == TRUE) or data.frame (if plot == FALSE) with information 
from QTL significance profile
</p>

<hr>
<h2 id='plot.mappoly.homoprob'>Plots mappoly.homoprob from MAPpoly</h2><span id='topic+plot.mappoly.homoprob'></span>

<h3>Description</h3>

<p>Plots mappoly.homoprob from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mappoly.homoprob'
plot(x, stack = FALSE, lg = NULL, ind = NULL, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.mappoly.homoprob_+3A_x">x</code></td>
<td>
<p>an object of class <code>mappoly.homoprob</code></p>
</td></tr>
<tr><td><code id="plot.mappoly.homoprob_+3A_stack">stack</code></td>
<td>
<p>logical. If <code>TRUE</code>, probability profiles of all homologues
are stacked in the plot (default = FALSE)</p>
</td></tr>
<tr><td><code id="plot.mappoly.homoprob_+3A_lg">lg</code></td>
<td>
<p>indicates which linkage group should be plotted. If <code>NULL</code> 
(default), it plots the first linkage group. If 
<code>"all"</code>, it plots all linkage groups</p>
</td></tr>
<tr><td><code id="plot.mappoly.homoprob_+3A_ind">ind</code></td>
<td>
<p>indicates which individuals should be plotted. It can be the 
position of the individuals in the dataset or it's name. 
If <code>NULL</code> (default), the function plots the first 
individual</p>
</td></tr>
<tr><td><code id="plot.mappoly.homoprob_+3A_verbose">verbose</code></td>
<td>
<p>if <code>TRUE</code> (default), the current progress is shown; if
<code>FALSE</code>, no output is produced</p>
</td></tr>
<tr><td><code id="plot.mappoly.homoprob_+3A_...">...</code></td>
<td>
<p>unused arguments</p>
</td></tr>
</table>

<hr>
<h2 id='prepare_diaQTL'>Converts diaQTL output to viewqtl object</h2><span id='topic+prepare_diaQTL'></span>

<h3>Description</h3>

<p>Converts diaQTL output to viewqtl object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_diaQTL(scan1_list, scan1_summaries_list, fitQTL_list, BayesCI_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_diaQTL_+3A_scan1_list">scan1_list</code></td>
<td>
<p>list with results from diaQTL <code>scan1</code> function</p>
</td></tr>
<tr><td><code id="prepare_diaQTL_+3A_scan1_summaries_list">scan1_summaries_list</code></td>
<td>
<p>list with results from diaQTL <code>scan1_summaries</code> function</p>
</td></tr>
<tr><td><code id="prepare_diaQTL_+3A_fitqtl_list">fitQTL_list</code></td>
<td>
<p>list with results from diaQTL <code>fitQTL</code> function</p>
</td></tr>
<tr><td><code id="prepare_diaQTL_+3A_bayesci_list">BayesCI_list</code></td>
<td>
<p>list with results from diaQTL <code>BayesCI</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewqtl</code>
</p>

<hr>
<h2 id='prepare_examples'>Upload example files</h2><span id='topic+prepare_examples'></span>

<h3>Description</h3>

<p>Upload example files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_examples(example)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_examples_+3A_example">example</code></td>
<td>
<p>character indicating the example dataset selected</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewpoly</code>
</p>

<hr>
<h2 id='prepare_map'>prepare maps for plot - from MAPpoly</h2><span id='topic+prepare_map'></span>

<h3>Description</h3>

<p>prepare maps for plot - from MAPpoly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_map(input.map, config = "best")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_map_+3A_input.map">input.map</code></td>
<td>
<p>object of class <code>mappoly.map</code></p>
</td></tr>
<tr><td><code id="prepare_map_+3A_config">config</code></td>
<td>
<p>choose between 'best', 'all' or provide vector with defined configuration. 
'best' provide just the best estimated configuration. 'all' provides all possibles.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing phase and dosage information
</p>

<hr>
<h2 id='prepare_map_custom_files'>Converts map information in custom format files to viewmap object</h2><span id='topic+prepare_map_custom_files'></span>

<h3>Description</h3>

<p>Converts map information in custom format files to viewmap object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_map_custom_files(dosages, phases, genetic_map, mks_pos = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_map_custom_files_+3A_dosages">dosages</code></td>
<td>
<p>TSV or TSV.GZ file with both parents dosage information.
It should contain four columns: 1) character vector with chromosomes ID; 
2) Character vector with markers ID; 3) Character vector with parent ID; 
4) numerical vector with dosage.</p>
</td></tr>
<tr><td><code id="prepare_map_custom_files_+3A_phases">phases</code></td>
<td>
<p>TSV or TSV.GZ file with phases information. It should contain:
1) Character vector with chromosome ID; 2) Character vector with marker ID;
3 to (ploidy number)*2 columns with each parents haplotypes.</p>
</td></tr>
<tr><td><code id="prepare_map_custom_files_+3A_genetic_map">genetic_map</code></td>
<td>
<p>TSV or TSV.GZ file with the genetic map information</p>
</td></tr>
<tr><td><code id="prepare_map_custom_files_+3A_mks_pos">mks_pos</code></td>
<td>
<p>TSV or TSV.GZ file with table with three columns: 1) marker ID; 
2) genome position; 3) chromosome</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewmap</code>
</p>

<hr>
<h2 id='prepare_MAPpoly'>Converts list of mappoly.map object into viewmap object</h2><span id='topic+prepare_MAPpoly'></span>

<h3>Description</h3>

<p>Converts list of mappoly.map object into viewmap object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_MAPpoly(mappoly_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_MAPpoly_+3A_mappoly_list">mappoly_list</code></td>
<td>
<p>list with objects of class <code>mappoly.map</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewmap</code>
</p>

<hr>
<h2 id='prepare_polymapR'>Converts polymapR ouputs to viewmap object</h2><span id='topic+prepare_polymapR'></span>

<h3>Description</h3>

<p>Converts polymapR ouputs to viewmap object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_polymapR(polymapR.dataset, polymapR.map, input.type, ploidy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_polymapR_+3A_polymapr.dataset">polymapR.dataset</code></td>
<td>
<p>a <code>polymapR</code> dataset</p>
</td></tr>
<tr><td><code id="prepare_polymapR_+3A_polymapr.map">polymapR.map</code></td>
<td>
<p>output map sequence from polymapR</p>
</td></tr>
<tr><td><code id="prepare_polymapR_+3A_input.type">input.type</code></td>
<td>
<p>indicates whether the input is discrete (&quot;disc&quot;) or probabilistic (&quot;prob&quot;)</p>
</td></tr>
<tr><td><code id="prepare_polymapR_+3A_ploidy">ploidy</code></td>
<td>
<p>ploidy level</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewmap</code>
</p>

<hr>
<h2 id='prepare_polyqtlR'>Converts polyqtlR outputs to viewqtl object</h2><span id='topic+prepare_polyqtlR'></span>

<h3>Description</h3>

<p>Converts polyqtlR outputs to viewqtl object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_polyqtlR(polyqtlR_QTLscan_list, polyqtlR_qtl_info, polyqtlR_effects)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_polyqtlR_+3A_polyqtlr_qtlscan_list">polyqtlR_QTLscan_list</code></td>
<td>
<p>list containing results from polyqtlR <code>QTLscan_list</code> function</p>
</td></tr>
<tr><td><code id="prepare_polyqtlR_+3A_polyqtlr_qtl_info">polyqtlR_qtl_info</code></td>
<td>
<p>data.frame containing the QTL information:LG - group ID; Pos - QTL position (cM); 
pheno - phenotype ID; Pos_lower - lower position of confidence interval; Pos_upper - upper position of the confidence interval;
thresh - LOD threshold applied</p>
</td></tr>
<tr><td><code id="prepare_polyqtlR_+3A_polyqtlr_effects">polyqtlR_effects</code></td>
<td>
<p>data.frame with results from <code>visualiseQTLeffects</code> polyqtlR function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewqtl</code>
</p>

<hr>
<h2 id='prepare_qtl_custom_files'>Converts QTL information in custom files to viewqtl object</h2><span id='topic+prepare_qtl_custom_files'></span>

<h3>Description</h3>

<p>Converts QTL information in custom files to viewqtl object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_qtl_custom_files(
  selected_mks,
  qtl_info,
  blups,
  beta.hat,
  profile,
  effects,
  probs
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_qtl_custom_files_+3A_selected_mks">selected_mks</code></td>
<td>
<p>data.frame with: LG - linkage group ID; mk - marker ID; pos - position in linkage map (cM)</p>
</td></tr>
<tr><td><code id="prepare_qtl_custom_files_+3A_qtl_info">qtl_info</code></td>
<td>
<p>data.frame with: LG - linkage group ID; Pos - position in linkage map (cM); 
Pheno - phenotype ID; Pos_lower - lower position of confidence interval; 
Pos_upper - upper position of the confidence interval; Pval - QTL p-value; h2 - herdability</p>
</td></tr>
<tr><td><code id="prepare_qtl_custom_files_+3A_blups">blups</code></td>
<td>
<p>data.frame with: haplo - haplotype ID; pheno - phenotype ID; qtl - QTL ID; u.hat - QTL estimated BLUPs</p>
</td></tr>
<tr><td><code id="prepare_qtl_custom_files_+3A_beta.hat">beta.hat</code></td>
<td>
<p>data.frame with: pheno - phenotype ID; beta.hat - estimated beta</p>
</td></tr>
<tr><td><code id="prepare_qtl_custom_files_+3A_profile">profile</code></td>
<td>
<p>data.frame with: pheno - phenotype ID; LOP - significance value for the QTL, in this case LOP (can be LOD or DIC depending of the software used)</p>
</td></tr>
<tr><td><code id="prepare_qtl_custom_files_+3A_effects">effects</code></td>
<td>
<p>data.frame with: pheno - phenotype ID; qtl.id - QTL ID; haplo - haplotype ID; effect - haplotype effect value</p>
</td></tr>
<tr><td><code id="prepare_qtl_custom_files_+3A_probs">probs</code></td>
<td>
<p>data.frame with first column (named 'ind') as individuals ID and next columns named with markers ID and containing the genotype probability at each marker</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewqtl</code>
</p>

<hr>
<h2 id='prepare_QTLpoly'>Converts QTLpoly outputs to viewqtl object</h2><span id='topic+prepare_QTLpoly'></span>

<h3>Description</h3>

<p>Converts QTLpoly outputs to viewqtl object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_QTLpoly(data, remim.mod, est.effects, fitted.mod)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_QTLpoly_+3A_data">data</code></td>
<td>
<p>object of class &quot;qtlpoly.data&quot;</p>
</td></tr>
<tr><td><code id="prepare_QTLpoly_+3A_remim.mod">remim.mod</code></td>
<td>
<p>object of class &quot;qtlpoly.model&quot; &quot;qtlpoly.remim&quot;.</p>
</td></tr>
<tr><td><code id="prepare_QTLpoly_+3A_est.effects">est.effects</code></td>
<td>
<p>object of class &quot;qtlpoly.effects&quot;</p>
</td></tr>
<tr><td><code id="prepare_QTLpoly_+3A_fitted.mod">fitted.mod</code></td>
<td>
<p>object of class &quot;qtlpoly.fitted&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of class <code>viewqtl</code>
</p>


<h3>Author(s)</h3>

<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>

<hr>
<h2 id='run_app'>Run the Shiny Application</h2><span id='topic+run_app'></span>

<h3>Description</h3>

<p>Run the Shiny Application
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_app(
  onStart = NULL,
  options = list(),
  enableBookmarking = NULL,
  uiPattern = "/",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_app_+3A_onstart">onStart</code></td>
<td>
<p>A function that will be called before the app is actually run.
This is only needed for <code>shinyAppObj</code>, since in the <code>shinyAppDir</code>
case, a <code>global.R</code> file can be used for this purpose.</p>
</td></tr>
<tr><td><code id="run_app_+3A_options">options</code></td>
<td>
<p>Named options that should be passed to the <code>runApp</code> call
(these can be any of the following: &quot;port&quot;, &quot;launch.browser&quot;, &quot;host&quot;, &quot;quiet&quot;,
&quot;display.mode&quot; and &quot;test.mode&quot;). You can also specify <code>width</code> and
<code>height</code> parameters which provide a hint to the embedding environment
about the ideal height/width for the app.</p>
</td></tr>
<tr><td><code id="run_app_+3A_enablebookmarking">enableBookmarking</code></td>
<td>
<p>Can be one of <code>"url"</code>, <code>"server"</code>, or
<code>"disable"</code>. The default value, <code>NULL</code>, will respect the setting from
any previous calls to  <code><a href="shiny.html#topic+enableBookmarking">enableBookmarking()</a></code>. See <code><a href="shiny.html#topic+enableBookmarking">enableBookmarking()</a></code>
for more information on bookmarking your app.</p>
</td></tr>
<tr><td><code id="run_app_+3A_uipattern">uiPattern</code></td>
<td>
<p>A regular expression that will be applied to each <code>GET</code>
request to determine whether the <code>ui</code> should be used to handle the
request. Note that the entire request path must match the regular
expression in order for the match to be considered successful.</p>
</td></tr>
<tr><td><code id="run_app_+3A_...">...</code></td>
<td>
<p>arguments to pass to golem_opts. 
See '?golem::get_golem_options' for more details.</p>
</td></tr>
</table>

<hr>
<h2 id='segreg_poly'>Polysomic segregation frequency - Function from MAPpoly</h2><span id='topic+segreg_poly'></span>

<h3>Description</h3>

<p>Computes the polysomic segregation frequency given a ploidy level
and the dosage of the locus in both parents. It does not consider
double reduction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segreg_poly(ploidy, d.p1, d.p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="segreg_poly_+3A_ploidy">ploidy</code></td>
<td>
<p>the ploidy level</p>
</td></tr>
<tr><td><code id="segreg_poly_+3A_d.p1">d.p1</code></td>
<td>
<p>the dosage in parent P</p>
</td></tr>
<tr><td><code id="segreg_poly_+3A_d.p2">d.p2</code></td>
<td>
<p>the dosage in parent Q</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector containing the expected segregation frequency for
all possible genotypic classes.
</p>


<h3>Author(s)</h3>

<p>Marcelo Mollinari, <a href="mailto:mmollin@ncsu.edu">mmollin@ncsu.edu</a>
</p>


<h3>References</h3>

<p>Mollinari, M., and Garcia, A.  A. F. (2019) Linkage
analysis and haplotype phasing in experimental autopolyploid
populations with high ploidy level using hidden Markov
models, _G3: Genes, Genomes, Genetics_. 
doi: <a href="https://doi.org/10.1534/g3.119.400378">10.1534/g3.119.400378</a>
</p>
<p>Serang O, Mollinari M, Garcia AAF (2012) Efficient Exact 
Maximum a Posteriori Computation for Bayesian SNP 
Genotyping in Polyploids. _PLoS ONE_ 7(2): 
e30906.
</p>

<hr>
<h2 id='select_haplo'>Plot selected haplotypes</h2><span id='topic+select_haplo'></span>

<h3>Description</h3>

<p>Plot selected haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_haplo(
  input.haplo,
  probs,
  selected_mks,
  effects.data,
  exclude.haplo = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_haplo_+3A_input.haplo">input.haplo</code></td>
<td>
<p>character vector with selected haplotypes. It contains the information: &quot;Trait:&lt;trait ID&gt;_LG:&lt;linkage group ID_Pos:&lt;QTL position&gt;&quot;</p>
</td></tr>
<tr><td><code id="select_haplo_+3A_probs">probs</code></td>
<td>
<p>data.frame with first column (named 'ind') as individuals ID and next columns named with markers ID and containing the genotype probability at each marker</p>
</td></tr>
<tr><td><code id="select_haplo_+3A_selected_mks">selected_mks</code></td>
<td>
<p>data.frame with: LG - linkage group ID; mk - marker ID; pos - position in linkage map (cM)</p>
</td></tr>
<tr><td><code id="select_haplo_+3A_effects.data">effects.data</code></td>
<td>
<p>output object from <code>data_effects</code> function</p>
</td></tr>
<tr><td><code id="select_haplo_+3A_exclude.haplo">exclude.haplo</code></td>
<td>
<p>character vector with haplotypes to be excluded. It contains the information: &quot;Trait:&lt;trait ID&gt;_LG:&lt;linkage group ID_Pos:&lt;QTL position&gt;&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot graphic
</p>

<hr>
<h2 id='summary_maps'>Summary maps - adapted from MAPpoly</h2><span id='topic+summary_maps'></span>

<h3>Description</h3>

<p>This function generates a brief summary table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_maps(viewmap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_maps_+3A_viewmap">viewmap</code></td>
<td>
<p>a list of objects of class <code>viewmap</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing a brief summary of all maps
</p>


<h3>Author(s)</h3>

<p>Gabriel Gesteira, <a href="mailto:gabrielgesteira@usp.br">gabrielgesteira@usp.br</a>
</p>
<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
