<!DOCTYPE html><html lang="en"><head><title>Help for package Virusparies</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Virusparies}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Virusparies'><p>Virusparies Package</p></a></li>
<li><a href='#CombineHittables'><p>CombineHittables: Combine hittables.</p></a></li>
<li><a href='#ExportVirusDataFrame'><p>ExportVirusDataFrame: Export processed hittables and summary stats data frames</p></a></li>
<li><a href='#ExportVirusGt'><p>ExportVirusGt: (Experimental) Export Graphical Tables</p></a></li>
<li><a href='#ExportVirusPlot'><p>Export Virusparies Plots</p></a></li>
<li><a href='#get_plot_parameters'><p>Internal function to set default title and cutoff in boxplot function</p></a></li>
<li><a href='#ImportVirusTable'><p>ImportVirusTable: Import VirusHunterGatherer hittables into R</p></a></li>
<li><a href='#SummarizeViralStats'><p>SummarizeViralStats: Generate summary stats outside of plot functions</p></a></li>
<li><a href='#VgConLenViolin'><p>VgConLenViolin: Generate a Violinplot of contig length for each group (Gatherer only)</p></a></li>
<li><a href='#VhgAddPhylum'><p>VhgAddPhylum: extract Phylum information</p></a></li>
<li><a href='#VhgBoxplot'><p>VhgBoxplot: Generate box plots comparing E-values,identity or contig length (Gatherer only) for each virus group</p></a></li>
<li><a href='#VhgGetSubject'><p>VhgGetSubject: Process and Count Viral Subjects within Groups</p></a></li>
<li><a href='#VhgIdenFacetedScatterPlot'><p>VhgIdenFacetedScatterPlot: Create a scatter plot of Viral refseq identity vs. -log10 of viral refseq E-value.</p></a></li>
<li><a href='#VhgIdentityScatterPlot'><p>VhgIdentityScatterPlot: Scatter plot for refseq identity vs -log10 of refseq E-value</p></a></li>
<li><a href='#VhgPreprocessTaxa'><p>VhgPreprocessTaxa: preprocess ViralRefSeq_taxonomy elements</p></a></li>
<li><a href='#VhgRunsBarplot'><p>VhgRunsBarplot: Generate a bar plot showing the number of data sets with unique runs found for</p>
each Virus group.</a></li>
<li><a href='#VhgRunsTable'><p>VhgRunsTable: Generate a gt summary table of unique runs for each virus group</p></a></li>
<li><a href='#VhgSubsetHittable'><p>VhgSubsetHittable: Filter VirusHunter and VirusGatherer hittables</p></a></li>
<li><a href='#VhgSumHitsBarplot'><p>VhgSumHitsBarplot: Generate a bar plot showing the sum of reads/contigs for each virus family</p></a></li>
<li><a href='#VhgTabularRasa'><p>VhgTabularRasa: Generate Custom Formatted Graphical Tables</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Visualize and Process Output from 'VirusHunterGatherer'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ruff Sergej &lt;serijnh@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of tools for downstream analysis of
    'VirusHunterGatherer' output. Processing of hittables 
    and plotting of results, enabling better interpretation,
    is made easier with the provided functions. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SergejRuff/Virusparies">https://github.com/SergejRuff/Virusparies</a>,
<a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/SergejRuff/Virusparies/issues">https://github.com/SergejRuff/Virusparies/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>chromote, cowplot, dplyr, ggplot2, gt, parallel, readr, rlang,
scales, stats, stringr, tidyr, tools, utils</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-14 12:36:18 UTC; sergej</td>
</tr>
<tr>
<td>Author:</td>
<td>Ruff Sergej <a href="https://orcid.org/0009-0000-8264-6347"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Lauber Chris <a href="https://orcid.org/0000-0002-2265-2953"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Chong Li Chuin <a href="https://orcid.org/0000-0002-3574-1365"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-14 13:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='Virusparies'>Virusparies Package</h2><span id='topic+Virusparies'></span><span id='topic+Virusparies-package'></span>

<h3>Description</h3>

<p>Virusparies designed for visualizing output from VirusHunterGatherer.
</p>


<h3>Details</h3>

<p>VirusHunterGatherer is a pipeline designed for data-driven virus discovery (DDVD).
It involves two steps: (i) Virushunter conducts sensitive homology-based detection of viral sequence reads in unprocessed data,
identifying the most conserved regions of a virus, which serve as seeds for the
(ii) Virusgatherer step that assembles full-length viral genome sequences.
</p>
<p>The Virusparies package provides a set of plotting functions tailored for visualizing
VirushunterGatherer hittables. The name draws inspiration from the hunter-gatherer metaphor,
with &quot;paries&quot; derived from Latin meaning &quot;wall&quot;. It symbolizes the parietal art left by
ancient hunters and gatherers on walls, summarizing their stories and beliefs.
</p>


<h3>Functions</h3>

<p>This package includes the following functions:
</p>
<p><strong>Import</strong>:
</p>

<ul>
<li> <p><code><a href="#topic+ImportVirusTable">ImportVirusTable</a></code>: Import VirusHunterGatherer hittables into R.
</p>
</li></ul>

<p><strong>VirusHunterGatherer Plots</strong>:
</p>

<ul>
<li> <p><code><a href="#topic+VhgBoxplot">VhgBoxplot</a></code>: Box plot plotting refseq sequence identity, E-values or contig length for each group.
</p>
</li>
<li> <p><code><a href="#topic+VhgIdenFacetedScatterPlot">VhgIdenFacetedScatterPlot</a></code>: Faceted scatter plot for reference sequence identity vs -log10 of reference E-value.
</p>
</li>
<li> <p><code><a href="#topic+VhgIdentityScatterPlot">VhgIdentityScatterPlot</a></code>: Scatter plot for reference sequence identity vs -log10 of reference E-value.
</p>
</li>
<li> <p><code><a href="#topic+VhgRunsBarplot">VhgRunsBarplot</a></code>: Bar plot showing how many unique runs map against each virus.
</p>
</li>
<li> <p><code><a href="#topic+VhgSumHitsBarplot">VhgSumHitsBarplot</a></code>: Bar plot for the sum of hits for each virus found in group.
</p>
</li></ul>

<p><strong>VirusGatherer only plots</strong>:
</p>

<ul>
<li> <p><code><a href="#topic+VgConLenViolin">VgConLenViolin</a></code>: Violin plot to visualize the distribution of contig lengths.
</p>
</li></ul>

<p><strong>Graphical Tables(gt)</strong>:
</p>

<ul>
<li> <p><code><a href="#topic+VhgRunsTable">VhgRunsTable</a></code>: Generate a gt-table for VhgRunsBarplot.
</p>
</li>
<li> <p><code><a href="#topic+VhgTabularRasa">VhgTabularRasa</a></code>: Generate custome gt-tables.
</p>
</li></ul>

<p><strong>Export</strong>:
</p>

<ul>
<li> <p><code><a href="#topic+ExportVirusDataFrame">ExportVirusDataFrame</a></code>: Export data frames.
</p>
</li>
<li> <p><code><a href="#topic+ExportVirusGt">ExportVirusGt</a></code>: Export graphical tables.
</p>
</li>
<li> <p><code><a href="#topic+ExportVirusPlot">ExportVirusPlot</a></code>: Export plots.
</p>
</li></ul>

<p><strong>Utils</strong>:
</p>

<ul>
<li> <p><code><a href="#topic+CombineHittables">CombineHittables</a></code>: Combine hittables.
</p>
</li>
<li> <p><code><a href="#topic+SummarizeViralStats">SummarizeViralStats</a></code>: Generate summary stats outside of plot functions.
</p>
</li>
<li> <p><code><a href="#topic+VhgAddPhylum">VhgAddPhylum</a></code>: Extract phylum information.
</p>
</li>
<li> <p><code><a href="#topic+VhgGetSubject">VhgGetSubject</a></code>: Process and Count Viral Subjects within Groups.
</p>
</li>
<li> <p><code><a href="#topic+VhgPreprocessTaxa">VhgPreprocessTaxa</a></code>: Process ViralRefSeq_taxonomy column.
</p>
</li>
<li> <p><code><a href="#topic+VhgSubsetHittable">VhgSubsetHittable</a></code>: Filter VirusHunterGatherer data based on userâ€™s own criteria.
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Sergej Ruff (<a href="mailto:serijnh@gmail.com">serijnh@gmail.com</a>)
</p>
<p><strong>Other contributors</strong>:
</p>

<ul>
<li><p> Chris Lauber (<a href="mailto:chris.lauber@twincore.de">chris.lauber@twincore.de</a>)
</p>
</li>
<li><p> Li Chuin, Chong (<a href="mailto:lichuin.chong@twincore.de">lichuin.chong@twincore.de</a>)
</p>
</li></ul>

<p>If you have any questions, suggestions, or issues, please feel free to contact Sergej Ruff (<a href="mailto:serijnh@gmail.com">serijnh@gmail.com</a>).
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available on: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>
<p>If you encounter any errors or issues, please report them at: <a href="https://github.com/SergejRuff/Virusparies/issues">https://github.com/SergejRuff/Virusparies/issues</a>.
</p>

<hr>
<h2 id='CombineHittables'>CombineHittables: Combine hittables.</h2><span id='topic+CombineHittables'></span>

<h3>Description</h3>

<p>CombineHittables combines multiple hittables by row-binding them, provided they have the same column names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CombineHittables(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CombineHittables_+3A_...">...</code></td>
<td>
<p>Hittables to be combined.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single hittable resulting from the row-binding of all input Hittables.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)
file2 &lt;- ImportVirusTable(path)  # both files have 180 observations

combined_file &lt;- CombineHittables(file,file2)

print(nrow(combined_file))
</code></pre>

<hr>
<h2 id='ExportVirusDataFrame'>ExportVirusDataFrame: Export processed hittables and summary stats data frames</h2><span id='topic+ExportVirusDataFrame'></span>

<h3>Description</h3>

<p>Export data frames generated by Virusparies functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExportVirusDataFrame(
  df,
  file_name,
  dir_path = NULL,
  file_type = NULL,
  create_path = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExportVirusDataFrame_+3A_df">df</code></td>
<td>
<p>A summary statistics or VirusHunterGatherer hittable data frame.</p>
</td></tr>
<tr><td><code id="ExportVirusDataFrame_+3A_file_name">file_name</code></td>
<td>
<p>A character string naming the file, optionally including &quot;.tsv&quot; or &quot;.csv&quot; file extensions.
If suffix is not provided, file_type will be used to determine the file type.</p>
</td></tr>
<tr><td><code id="ExportVirusDataFrame_+3A_dir_path">dir_path</code></td>
<td>
<p>A character string indicating the directory path where the file will be saved (default: current corking directory).</p>
</td></tr>
<tr><td><code id="ExportVirusDataFrame_+3A_file_type">file_type</code></td>
<td>
<p>A character vector specifying the type of file to export. Can be &quot;csv&quot; or &quot;tsv&quot;.
If NULL and file_name does not specify a suffix, the function infers the file type based on the prefix in file_name.</p>
</td></tr>
<tr><td><code id="ExportVirusDataFrame_+3A_create_path">create_path</code></td>
<td>
<p>Logical indicating whether to create the directory path
specified in dir_path if it does not already exist (default: FALSE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Functions in the Virusparies package can generate both plots and new data frames.
Data frames contain either summary statistics for contig length, E-value or identity in percentage
or a processed hittable for example outlier or observations below threshold, when running
<code><a href="#topic+VhgBoxplot">VhgBoxplot</a></code>.
</p>
<p>Both types of data frames can be exported via <code>ExportVirusDataFrame</code>. Summary stats and hittables
can be exported as CSV files or in TSV format, if the user prefers the file type used in VirusHunterGatherer
hittables.
</p>


<h3>Value</h3>

<p>A message indicating that export was successful.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)


# generate a plot that returns both processed hittables (outlier) and summary stats
plot1 &lt;- VhgBoxplot(file, x_column = "best_query", y_column = "ViralRefSeq_E")




# export hittable as tsv (same format as input hittables)
ExportVirusDataFrame(df=plot1$outlier,file_name="outlier",file_type="tsv",
dir_path=tempdir())

# export summary stats as csv
ExportVirusDataFrame(df=plot1$summary_stats,file_name="summarystats",
file_type="csv",dir_path=tempdir())



</code></pre>

<hr>
<h2 id='ExportVirusGt'>ExportVirusGt: (Experimental) Export Graphical Tables</h2><span id='topic+ExportVirusGt'></span>

<h3>Description</h3>

<p>ExportVirusGt allows the user to export graphical tables in different formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExportVirusGt(
  gtable,
  filename = "table.docx",
  export_gt_obj = FALSE,
  path = NULL,
  create.dir = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExportVirusGt_+3A_gtable">gtable</code></td>
<td>
<p>A graphical table object.</p>
</td></tr>
<tr><td><code id="ExportVirusGt_+3A_filename">filename</code></td>
<td>
<p>Name of the output file (default: table.docx&quot;). Make sure to provide an extension
compatible with the output types: .html, .tex, .ltx, .rtf, .docx. If a custom save function is provided,
the file extension is ignored.</p>
</td></tr>
<tr><td><code id="ExportVirusGt_+3A_export_gt_obj">export_gt_obj</code></td>
<td>
<p>(optional): If TRUE, exports the input data frame in .rds format with the same name as specified in filename (default: FALSE).</p>
</td></tr>
<tr><td><code id="ExportVirusGt_+3A_path">path</code></td>
<td>
<p>Path of the directory to save plot to: path and filename are combined to create the
fully qualified file name (default: current corking directory).</p>
</td></tr>
<tr><td><code id="ExportVirusGt_+3A_create.dir">create.dir</code></td>
<td>
<p>Whether to create new directories if a non-existing directory is specified in
the filename or path (TRUE) or return an error (FALSE, default). If FALSE and run in an interactive session,
a prompt will appear asking to create a new directory when necessary.</p>
</td></tr>
<tr><td><code id="ExportVirusGt_+3A_...">...</code></td>
<td>
<p>Pass any other options to the corresponding internal saving function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Export graphical tables (gt) generated by functions within the Virusparies package.
</p>
<p>This feature is in an experimental phase and may not currently function as expected.
</p>
<p>The exportVirusGt function utilizes the gt package for table manipulation and formatting.
</p>
<p>For HTML output file names with .html or .htm extensions, an HTML document is generated using the gt package.
Pass TRUE or FALSE to inline_css to include or exclude CSS styles inline (default is FALSE).
Additional options can be passed through .... For RTF output file names with .rtf extension,
an RTF file is created. Use the page_numbering option to control page numbering (default is none).
</p>
<p>For image files, use .png for PNG and .pdf for PDF. The gt package relies on Google Chrome installation
for PNG and PDF images. Pass options to webshot2::webshot() through .... Useful PNG options include zoom
(default: 2) and expand (default: 5).
</p>
<p>For LaTeX output file names with .tex, .ltx, or .rnw extensions, and .rtf for RTF, the corresponding
documents are generated. No additional options available.
</p>
<p>For .docx output, requires rmarkdown package.
</p>
<p>When <code>create.dir</code> is set to TRUE, it generates a directory at the specified 'path' argument if the path doesn't already exist.
</p>
<p>The optional <code>export_gt_obj</code> argument enables the user to export the data frame as a .rds file alongside the graphical table.
</p>


<h3>Value</h3>

<p>A message indicating that export was successful.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
vh_file &lt;- ImportVirusTable(path)


# using first 10 rows of SRA_run,num_hits,bestquery,ViralRefSeq_E and Identity col.
vh_file_part &lt;- vh_file[c(1:10),c(1,7,9,10,11)]

table &lt;- VhgTabularRasa(vh_file_part,title = "first 10 rows of vh_file",subtitle =
"example for any table",names_ = c("Runs","Number of Contigs","Best Query Result",
"Reference E-Value","Reference Identity"))




ExportVirusGt(gtable=table,filename="vh_parttable.docx",path=tempdir())



</code></pre>

<hr>
<h2 id='ExportVirusPlot'>Export Virusparies Plots</h2><span id='topic+ExportVirusPlot'></span>

<h3>Description</h3>

<p>ExportVirusPlot allows the user to export plots in different formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExportVirusPlot(
  file_name,
  export_plotobj = FALSE,
  plot = NULL,
  device = NULL,
  path = NULL,
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  ...,
  align = "none",
  axis = "none",
  nrow = NULL,
  ncol = NULL,
  rel_widths = 1,
  rel_heights = 1,
  labels = NULL,
  label_size = 14,
  label_fontfamily = NULL,
  label_fontface = "bold",
  label_colour = NULL,
  label_x = 0,
  label_y = 1,
  hjust = -0.5,
  vjust = 1.5,
  scale_grid = 1,
  greedy = TRUE,
  byrow = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExportVirusPlot_+3A_file_name">file_name</code></td>
<td>
<p>Name of the output file.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_export_plotobj">export_plotobj</code></td>
<td>
<p>(optional): If TRUE, exports the plot object in rds format with the same name as specified in file_name (default: FALSE).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_plot">plot</code></td>
<td>
<p>The plot to be exported.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_device">device</code></td>
<td>
<p>The device used for output.
Can be one of &quot;eps&quot;, &quot;ps&quot;, &quot;tex&quot;, &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;png&quot;, &quot;bmp&quot;, &quot;svg&quot;, or &quot;wmf&quot; (windows only).
If NULL (default), the device is guessed based on the filename extension.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_path">path</code></td>
<td>
<p>The directory where the plot will be saved. Default is NULL (working directory).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_scale">scale</code></td>
<td>
<p>A scaling factor (default: 1).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_width">width</code></td>
<td>
<p>The width of the output file.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_height">height</code></td>
<td>
<p>The height of the output file.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_units">units</code></td>
<td>
<p>The units of the width and height parameters.
Can be one of &quot;in&quot;, &quot;cm&quot;, &quot;mm&quot;, or &quot;px&quot;.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_dpi">dpi</code></td>
<td>
<p>The resolution of the output device in dots per inch (default: 300).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_limitsize">limitsize</code></td>
<td>
<p>Whether to limit the size of the output file to the dimensions of the plot (default: TRUE).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_align">align</code></td>
<td>
<p>Specifies alignment of plots in the grid: &quot;none&quot; (default), &quot;hv&quot; (both directions), &quot;h&quot; (horizontally), or &quot;v&quot; (vertically).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_axis">axis</code></td>
<td>
<p>Specifies alignment of plots by margins: &quot;none&quot; (default), or any combo of left (&quot;l&quot;), right (&quot;r&quot;), top (&quot;t&quot;), or bottom (&quot;b&quot;)(e.g., &quot;tblr&quot; or &quot;rlbt&quot;).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_nrow">nrow</code></td>
<td>
<p>(optional): Number of rows in the plot grid (default: NULL).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_ncol">ncol</code></td>
<td>
<p>(optional): Number of columns in the plot grid (default: NULL).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_rel_widths">rel_widths</code></td>
<td>
<p>Numeric vector of relative column widths. Default is 1 (equal widths).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_rel_heights">rel_heights</code></td>
<td>
<p>Numeric vector of relative row heights. Default is 1 (equal heights).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_labels">labels</code></td>
<td>
<p>List of labels for plots. Default is NULL (no labels).labels=&quot;auto&quot; and labels=&quot;AUTO&quot; auto-generate lower and upper-case labels.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_label_size">label_size</code></td>
<td>
<p>Numeric label size (default: 14).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_label_fontfamily">label_fontfamily</code></td>
<td>
<p>Font family for labels. Default is NULL (theme default).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_label_fontface">label_fontface</code></td>
<td>
<p>Font face for labels (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_label_colour">label_colour</code></td>
<td>
<p>Color for labels. Default is NULL (theme default).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_label_x">label_x</code></td>
<td>
<p>Single value/vector of x positions for labels,relative to each subplot. Default is 0 (left).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_label_y">label_y</code></td>
<td>
<p>Single value/vector of y positions for labels, relative to each subplot. Default is 1 (top).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_hjust">hjust</code></td>
<td>
<p>Horizontal adjustment for labels (default: -0.5).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_vjust">vjust</code></td>
<td>
<p>Vertical adjustment for labels (default: 1.5).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_scale_grid">scale_grid</code></td>
<td>
<p>Single value/vector &gt; 0. Enables you to scale the size of all or select plots.</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_greedy">greedy</code></td>
<td>
<p>Margin adjustment during alignment (default: TRUE).</p>
</td></tr>
<tr><td><code id="ExportVirusPlot_+3A_byrow">byrow</code></td>
<td>
<p>Arrange plots by row (TRUE, default) or column (FALSE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Export plots generated by functions within the Virusparies package.
</p>
<p>ExportVirusPlot exports plots in various formats supported by Virusparies.
Supported devices include &quot;eps&quot;, &quot;ps&quot;, &quot;tex&quot;, &quot;pdf&quot;, &quot;jpeg&quot;, &quot;tiff&quot;, &quot;png&quot;, &quot;bmp&quot;, &quot;svg&quot;, or &quot;wmf&quot; (Windows only).
When 'device' is set to NULL, the file extension in <code>filename</code> is used to determine the device.
</p>
<p>Depending on the plot, the final image might be cropped or truncated.
We recommend experimenting with height, width, and resolution.
</p>
<p>In addition, users can generate a grid layout containing multiple plots when a list containing multiple plots is provided as input.
This will then be exported using the chosen device.
</p>
<p>The following arguments are only used for export with grid layout:
</p>

<ul>
<li> <p><code>align</code>: Specifies how plots are aligned within the grid.
</p>
</li>
<li> <p><code>axis</code>: Controls alignment of plots by margins.
</p>
</li>
<li> <p><code>nrow</code>, <code>ncol</code>: Define the structure of the plot grid.
</p>
</li>
<li> <p><code>rel_widths</code>, <code>rel_heights</code>: Adjust relative column and row sizes.
</p>
</li>
<li> <p><code>labels</code>, <code>label_size</code>, <code>label_fontfamily</code>, <code>label_fontface</code>, <code>label_colour</code>, <code>label_x</code>, <code>label_y</code>, <code>hjust</code>, <code>vjust</code>: Customize plot labels.
</p>
</li>
<li> <p><code>scale_grid</code>: Enables you to scale the size of all or select plots.
</p>
</li>
<li> <p><code>greedy</code>: Determines margin adjustments during alignment.
</p>
</li>
<li> <p><code>byrow</code>: Specifies the arrangement of plots in the grid.
</p>
</li></ul>

<p><code>export_plotobj</code> = TRUE exports the plot in the specified format (e.g., PNG, PDF, etc.)
and also saves the plot object in .rds format with the same file name. This allows the user to
import the plot object into R using the <code>readRDS</code> function and modify the plot further as needed.
</p>


<h3>Value</h3>

<p>a message indicating that export was successful.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
vh_file &lt;- ImportVirusTable(path)

# Basic plot
plot &lt;- VhgIdentityScatterPlot(vh_file,cutoff = 1e-5)



# first export
ExportVirusPlot(plot=plot$plot,file_name="testplot.png",width=8,height=6,
units="in",path=tempdir())

# second export with device argument
ExportVirusPlot(plot=plot$plot,file_name="testplot",width=8,height=6,
units="in",device = "png",path=tempdir())



## example 2 for multiple plots in 1 pdf file.

path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)

# Generate 3 plots
violinplot &lt;- VgConLenViolin(vg_file=vg_file,cut = 1e-5,log10_scale = TRUE,
legend_position = "none",title = "",xlabel = "",reorder_criteria = NULL,
                             theme_choice = "minimal")
srarun &lt;- VhgRunsBarplot(file = vg_file,groupby = "ViralRefSeq_taxonomy",
legend_position = "none",title = "",xlabel = "",reorder_criteria = NULL,
                         theme_choice = "minimal")
boxplot &lt;- VhgBoxplot(vg_file,x_column = "ViralRefSeq_taxonomy",y_column = "ViralRefSeq_ident",
legend_position = "bottom",title = "",xlabel = "",
                      reorder_criteria = NULL,theme_choice = "minimal")

# add plots to a list
plot_list &lt;- list(violinplot$plot,srarun$plot,boxplot$boxp)



ExportVirusPlot(plot=plot_list,file_name="grid_testplot.pdf",width=16,height=12,
units="in",nrow = 3,ncol = 1,path=tempdir())



</code></pre>

<hr>
<h2 id='get_plot_parameters'>Internal function to set default title and cutoff in boxplot function</h2><span id='topic+get_plot_parameters'></span>

<h3>Description</h3>

<p>Internal function to set default title and cutoff in boxplot function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_plot_parameters(y_column, cut)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_plot_parameters_+3A_y_column">y_column</code></td>
<td>
<p>y-column</p>
</td></tr>
<tr><td><code id="get_plot_parameters_+3A_cut">cut</code></td>
<td>
<p>cutoff value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing cutoff and title
</p>

<hr>
<h2 id='ImportVirusTable'>ImportVirusTable: Import VirusHunterGatherer hittables into R</h2><span id='topic+ImportVirusTable'></span>

<h3>Description</h3>

<p>ImportVirusTable imports VirusHunterGatherer hittables into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ImportVirusTable(path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ImportVirusTable_+3A_path">path</code></td>
<td>
<p>A character string specifying the path to the VirusHunter or VirusGatherer hittable file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ImportVirusTable reads the VirusHunter or VirusGatherer hittable file specified by 'path'
into a data frame in R. The file should be in tab-separated values (TSV) format.
The first row should contain column headers.
</p>


<h3>Value</h3>

<p>A data frame containing the data from the VirusHunter or VirusGatherer hittable file.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
vh_file &lt;- ImportVirusTable(path)

print(head(vh_file))

# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)

print(head(vg_file))

</code></pre>

<hr>
<h2 id='SummarizeViralStats'>SummarizeViralStats: Generate summary stats outside of plot functions</h2><span id='topic+SummarizeViralStats'></span>

<h3>Description</h3>

<p>Summarizes data by grouping it according to a specified metric (contig length, E-value or Identity).
SummarizeViralStats generates a summary table that includes counts of observations based on a specified metric cutoff.
It computes relevant summary statistics depending on the selected metric, with options to filter rows
based on a cutoff value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SummarizeViralStats(
  file,
  groupby = "best_query",
  metric,
  metric_cutoff,
  filter_cutoff = NULL,
  show_total = FALSE,
  extra_stats = NULL,
  sort_by = NULL,
  top_n = NULL,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SummarizeViralStats_+3A_file">file</code></td>
<td>
<p>VirusHunterGatherer hittable.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_metric">metric</code></td>
<td>
<p>A character string specifying the name of the metric column to be used for calculations.
This column must be present in <code>file</code>. Supported metric columns include:
</p>

<ul>
<li><p> &quot;contig_len&quot;
</p>
</li>
<li><p> &quot;ViralRefSeq_E&quot;
</p>
</li>
<li><p> &quot;ViralRefSeq_ident&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_metric_cutoff">metric_cutoff</code></td>
<td>
<p>A numeric value used to classify the metric into two categories: below cutoff and above or equal to cutoff.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_filter_cutoff">filter_cutoff</code></td>
<td>
<p>A numeric value for optional filtering of the data based on E-value. Rows where the specified filtering column has a value
less than this cutoff are retained. If <code>NULL</code>, no filtering is applied. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_show_total">show_total</code></td>
<td>
<p>A logical value indicating whether to include a row with the total sums for each column in the summary table.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_extra_stats">extra_stats</code></td>
<td>
<p>A character vector specifying additional summary statistics to include in the output. Options include:
</p>

<ul>
<li> <p><code>"mean"</code>
</p>
</li>
<li> <p><code>"median"</code>
</p>
</li>
<li> <p><code>"Q1"</code>
</p>
</li>
<li> <p><code>"Q3"</code>
</p>
</li>
<li> <p><code>"sd"</code>
</p>
</li>
<li> <p><code>"min"</code>
</p>
</li>
<li> <p><code>"max"</code>
</p>
</li></ul>

<p>If <code>NULL</code> (the default), only the basic counts are included.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_sort_by">sort_by</code></td>
<td>
<p>(optional): A character string specifying the column name by which to sort the results.
Supported values include:
</p>

<ul>
<li><p> &quot;less_than_X&quot;: The count of observations below the specified <code>metric_cutoff</code> (X is replaced by the cutoff value).
</p>
</li>
<li><p> &quot;equal_or_more_than_X&quot;: The count of observations greater than or equal to the specified <code>metric_cutoff</code> (X is replaced by the cutoff value).
</p>
</li>
<li><p> &quot;total&quot;: The total count of observations in each group.
</p>
</li>
<li><p> &quot;mean&quot;: The mean value of the specified metric in each group (if <code>"mean"</code> is included in <code>extra_stats</code>).
</p>
</li>
<li><p> &quot;median&quot;: The median value of the specified metric in each group (if <code>"median"</code> is included in <code>extra_stats</code>).
</p>
</li>
<li><p> &quot;Q1&quot;: The first quartile (25th percentile) of the specified metric in each group (if <code>"Q1"</code> is included in <code>extra_stats</code>).
</p>
</li>
<li><p> &quot;Q3&quot;: The third quartile (75th percentile) of the specified metric in each group (if <code>"Q3"</code> is included in <code>extra_stats</code>).
</p>
</li>
<li><p> &quot;sd&quot;: The standard deviation of the specified metric in each group (if <code>"sd"</code> is included in <code>extra_stats</code>).
</p>
</li>
<li><p> &quot;min&quot;: The minimum value of the specified metric in each group (if <code>"min"</code> is included in <code>extra_stats</code>).
</p>
</li>
<li><p> &quot;max&quot;: The maximum value of the specified metric in each group (if <code>"max"</code> is included in <code>extra_stats</code>).
</p>
</li></ul>

<p>If <code>NULL</code> (the default), no sorting is applied.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_top_n">top_n</code></td>
<td>
<p>(optional): A numeric value indicating the number of top rows to return based on the selected metric.
If <code>NULL</code> (the default), all rows are returned.</p>
</td></tr>
<tr><td><code id="SummarizeViralStats_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarizing the viral stats. The output includes:
</p>

<ul>
<li><p> The count of observations below and above or equal to the <code>metric_cutoff</code>.
</p>
</li>
<li><p> Optional additional summary statistics as specified by <code>extra_stats</code>.
</p>
</li>
<li><p> An optional total row if <code>show_total</code> is <code>TRUE</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

stats &lt;- SummarizeViralStats(file=file,
groupby = "best_query",
metric = "ViralRefSeq_ident",
metric_cutoff = 90,
show_total = TRUE,
filter_cutoff = 1e-5,
extra_stats = c("median","Q1","Q3"))

print(stats)



</code></pre>

<hr>
<h2 id='VgConLenViolin'>VgConLenViolin: Generate a Violinplot of contig length for each group (Gatherer only)</h2><span id='topic+VgConLenViolin'></span>

<h3>Description</h3>

<p>VgConLenViolin creates a violin plot to visualize the distribution of contig lengths
for each group in VirusGatherer hittables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VgConLenViolin(
  vg_file = vg_file,
  taxa_rank = "Family",
  cut = 1e-05,
  log10_scale = TRUE,
  reorder_criteria = "median",
  adjust_bw = 1,
  jitter_point = FALSE,
  theme_choice = "linedraw",
  flip_coords = TRUE,
  title = "Violinplot of contig length for each group",
  title_size = 16,
  title_face = "bold",
  title_colour = "#2a475e",
  subtitle = NULL,
  subtitle_size = 12,
  subtitle_face = "bold",
  subtitle_colour = "#1b2838",
  xlabel = "Viral group",
  ylabel = "Contig length (nt)",
  axis_title_size = 12,
  xtext_size = 10,
  x_angle = NULL,
  ytext_size = 10,
  y_angle = NULL,
  remove_group_labels = FALSE,
  legend_title = "Phylum",
  legend_position = "bottom",
  legend_title_size = 12,
  legend_title_face = "bold",
  legend_text_size = 10,
  min_observations = 1,
  facet_ncol = NULL,
  add_boxplot = FALSE,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VgConLenViolin_+3A_vg_file">vg_file</code></td>
<td>
<p>A data frame containing VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): Specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_cut">cut</code></td>
<td>
<p>(optional): A numeric value representing the cutoff for the refseq E-value (default: 1e-5).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_log10_scale">log10_scale</code></td>
<td>
<p>(optinal): transform y-axis to log10 scale (default: TRUE).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_reorder_criteria">reorder_criteria</code></td>
<td>
<p>Character string specifying the criteria for reordering the x-axis ('max', 'min', 'median'(Default),'mean','phylum').
NULL sorts alphabetically. You can also specify criteria with 'phylum_' prefix (e.g., 'phylum_median') to sort by phylum first and then by the specified statistic within each phylum.</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_adjust_bw">adjust_bw</code></td>
<td>
<p>(optional): control the bandwidth of the kernel density estimator used to create the violin plot.
A higher value results in a smoother plot by increasing the bandwidth, while a lower value can make the plot more detailed but potentially noisier (default: 1).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_jitter_point">jitter_point</code></td>
<td>
<p>(optional): logical: TRUE to show all observations, FALSE to show only groups with less than 2 observations (default: FALSE).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_theme_choice">theme_choice</code></td>
<td>
<p>(optional): A character indicating the ggplot2 theme to apply. Options include &quot;minimal&quot;,
&quot;classic&quot;, &quot;light&quot;, &quot;dark&quot;, &quot;void&quot;, &quot;grey&quot; (or &quot;gray&quot;), &quot;bw&quot;, &quot;linedraw&quot; (default), and &quot;test&quot;.
Append &quot;_dotted&quot; to any theme to add custom dotted grid lines (e.g., &quot;classic_dotted&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_flip_coords">flip_coords</code></td>
<td>
<p>(optional): Logical indicating whether to flip the coordinates of the plot (default: TRUE).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_title">title</code></td>
<td>
<p>(optional): The title of the plot (default: &quot;Violinplot of contig length for each group&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_title_size">title_size</code></td>
<td>
<p>(optional): The size of the title text (default: 16).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_title_face">title_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): The color of the title text (default: &quot;#2a475e&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): A character specifying the subtitle of the plot (default: NULL).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the subtitle text(default: 12).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_subtitle_face">subtitle_face</code></td>
<td>
<p>(optional): A character specifying the font face for the subtitle text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_subtitle_colour">subtitle_colour</code></td>
<td>
<p>(optional): A character specifying the color for the subtitle text (default: &quot;#1b2838&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_xlabel">xlabel</code></td>
<td>
<p>(optional): The label for the x-axis (default: &quot;Viral group&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_ylabel">ylabel</code></td>
<td>
<p>(optional): The label for the y-axis (default: &quot;Contig length (nt)&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>(optional): The size of the axis titles (default: 12).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_xtext_size">xtext_size</code></td>
<td>
<p>(optional): The size of the x-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_x_angle">x_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the x-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_ytext_size">ytext_size</code></td>
<td>
<p>(optional): The size of the y-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_y_angle">y_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the y-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_remove_group_labels">remove_group_labels</code></td>
<td>
<p>(optional): If <code>TRUE</code>, the group labels will be removed; if <code>FALSE</code> or omitted, the labels will be displayed.</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_legend_title">legend_title</code></td>
<td>
<p>(optional): A character specifying the title for the legend (default: &quot;Phylum&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_legend_position">legend_position</code></td>
<td>
<p>(optional): The position of the legend (default: &quot;bottom).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend title text (default: 12).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_legend_title_face">legend_title_face</code></td>
<td>
<p>(optional): A character specifying the font face for the legend title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend text (default: 10).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_min_observations">min_observations</code></td>
<td>
<p>(optional): Minimum number of observations required per group to be included in the plot (default: 1).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>(optional):  The number of columns for faceting (default: NULL).
It is recommended to specify this when the number of viral groups is high, to ensure they fit well in one plot.</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_add_boxplot">add_boxplot</code></td>
<td>
<p>(optional): Add a boxplot to the violin plot (default: FALSE).</p>
</td></tr>
<tr><td><code id="VgConLenViolin_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VgConLenViolin creates a violin plot to visualize the distribution of contig lengths
for each group in the &quot;ViralRefSeq_taxonomy&quot; column of the VirusGatherer hittable.
The x-axis represents the groups as defined by the &quot;ViralRefSeq_taxonomy&quot; column,
and the y-axis represents the contig lengths.
</p>
<p>By default, the y-axis is transformed to a log10 scale to better visualize differences
in contig lengths across groups. This transformation can be disabled by setting the
<code>log10_scale</code> argument to FALSE.
</p>
<p><code>min_observations</code> filters the data sets to include only groups with at least the specified number of observations
before plotting them. This feature allows users to exclude groups with insufficient data.
By default, every group is plotted, as the minimum requirement is set to at least one observation per group.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>

<ul>
<li><p> plot: A plot object representing the violin plot.
</p>
</li>
<li><p> contiglen_stats: A tibble data frame with summary statistics for &quot;contig_len&quot; values.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)

# create a violinplot.
violinplot &lt;- VgConLenViolin(vg_file=vg_file,cut = 1e-5,log10_scale = TRUE)

violinplot


</code></pre>

<hr>
<h2 id='VhgAddPhylum'>VhgAddPhylum: extract Phylum information</h2><span id='topic+VhgAddPhylum'></span>

<h3>Description</h3>

<p>VhgAddPhylum adds a <code>Phylum</code> column to the provided VirusHunter or VirusGatherer hittable,
with each entry in the column reflecting the phylum name derived from the <code>groupby</code> column for each observation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgAddPhylum(file, groupby = "best_query")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgAddPhylum_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunter or VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VhgAddPhylum_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Hittable with Phylum column
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
vh_file &lt;- ImportVirusTable(path)

vh_file_filtered &lt;- VhgPreprocessTaxa(vh_file,"Family")

processed_taxa &lt;- VhgAddPhylum(vh_file_filtered,"ViralRefSeq_taxonomy")

print(unique(processed_taxa$Phylum))

</code></pre>

<hr>
<h2 id='VhgBoxplot'>VhgBoxplot: Generate box plots comparing E-values,identity or contig length (Gatherer only) for each virus group</h2><span id='topic+VhgBoxplot'></span>

<h3>Description</h3>

<p>VhgBoxplot generates box plots comparing either E-values,identity or contig length (Gatherer only)
for each group from VirusHunter or VirusGatherer hittable results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgBoxplot(
  file,
  x_column = "best_query",
  taxa_rank = "Family",
  y_column = "ViralRefSeq_E",
  contiglen_log10_scale = FALSE,
  cut = 1e-05,
  add_cutoff_line = TRUE,
  cut_colour = "#990000",
  reorder_criteria = "median",
  theme_choice = "linedraw",
  flip_coords = TRUE,
  add_mean_point = FALSE,
  mean_color = "white",
  mean_point_size = 2,
  title = "default",
  title_size = 16,
  title_face = "bold",
  title_colour = "#2a475e",
  subtitle = "default",
  subtitle_size = 12,
  subtitle_face = "bold",
  subtitle_colour = "#1b2838",
  xlabel = NULL,
  ylabel = NULL,
  axis_title_size = 12,
  xtext_size = 10,
  x_angle = NULL,
  ytext_size = 10,
  y_angle = NULL,
  remove_group_labels = FALSE,
  legend_title = "Phylum",
  legend_position = "bottom",
  legend_title_size = 12,
  legend_title_face = "bold",
  legend_text_size = 10,
  facet_ncol = NULL,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgBoxplot_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunter or VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_x_column">x_column</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default:&quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): When <code>x_column</code> is set to &quot;ViralRefSeq_taxonomy&quot;, specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_y_column">y_column</code></td>
<td>
<p>A character specifying the column containing the values to be compared. Currently &quot;ViralRefSeq_ident&quot;,
&quot;contig_len&quot; (column in Gatherer hittable) and &quot;ViralRefSeq_E&quot; are supported columns (default:&quot;ViralRefSeq_E&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_contiglen_log10_scale">contiglen_log10_scale</code></td>
<td>
<p>(optional): When <code>y_column</code> is set to &quot;contig_len&quot;, this parameter enables logarithmic scaling (log10) of the y-axis (TRUE). By default, this feature is disabled (FALSE).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_cut">cut</code></td>
<td>
<p>(optional): The significance cutoff value for E-values (default: 1e-5).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_add_cutoff_line">add_cutoff_line</code></td>
<td>
<p>(optional): Whether to add a horizontal line based on <code>cut</code> for <code>"ViralRefSeq_E"</code> column (default: TRUE).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_cut_colour">cut_colour</code></td>
<td>
<p>(optional): The color for the significance cutoff line (default: &quot;#990000&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_reorder_criteria">reorder_criteria</code></td>
<td>
<p>Character string specifying the criteria for reordering the x-axis ('max', 'min', 'median'(Default),'mean','phylum').
NULL sorts alphabetically. You can also specify criteria with 'phylum_' prefix (e.g., 'phylum_median') to sort by phylum first and then by the specified statistic within each phylum.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_theme_choice">theme_choice</code></td>
<td>
<p>(optional): A character indicating the ggplot2 theme to apply. Options include &quot;minimal&quot;,
&quot;classic&quot;, &quot;light&quot;, &quot;dark&quot;, &quot;void&quot;, &quot;grey&quot; (or &quot;gray&quot;), &quot;bw&quot;, &quot;linedraw&quot; (default), and &quot;test&quot;.
Append &quot;_dotted&quot; to any theme to add custom dotted grid lines (e.g., &quot;classic_dotted&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_flip_coords">flip_coords</code></td>
<td>
<p>(optional): Logical indicating whether to flip the coordinates of the plot (default: TRUE).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_add_mean_point">add_mean_point</code></td>
<td>
<p>(optional): Logical indicating whether to add mean points to the box plot (default: FALSE).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_mean_color">mean_color</code></td>
<td>
<p>(optional): Change color of point indicating mean value in box plot (default: &quot;white&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_mean_point_size">mean_point_size</code></td>
<td>
<p>(optional): Change size of point indicating mean value in box plot (default: 2).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_title">title</code></td>
<td>
<p>(optional): A character specifying the title of the plot. Default title is set based on y_column.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_title_size">title_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the title text (default: 16).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_title_face">title_face</code></td>
<td>
<p>(optional): A character specifying the font face for the title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): A character specifying the color for the title text (default: &quot;#2a475e&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): A character specifying the subtitle of the plot. Default subtitle is set based on y_column.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the subtitle text(default: 12).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_subtitle_face">subtitle_face</code></td>
<td>
<p>(optional): A character specifying the font face for the subtitle text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_subtitle_colour">subtitle_colour</code></td>
<td>
<p>(optional): A character specifying the color for the subtitle text (default: &quot;#1b2838&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_xlabel">xlabel</code></td>
<td>
<p>(optional): A character specifying the label for the x-axis (default: &quot;Virus found in query&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_ylabel">ylabel</code></td>
<td>
<p>(optional): A character specifying the label for the y-axis. Default is set based on y_column.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the axis title text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_xtext_size">xtext_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the x-axis tick labels (default: 10).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_x_angle">x_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the x-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_ytext_size">ytext_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the y-axis tick labels (default: 10).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_y_angle">y_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the y-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_remove_group_labels">remove_group_labels</code></td>
<td>
<p>(optional): If <code>TRUE</code>, the group labels will be removed; if <code>FALSE</code> or omitted, the labels will be displayed.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_legend_title">legend_title</code></td>
<td>
<p>(optional): A character specifying the title for the legend (default: &quot;Phylum&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_legend_position">legend_position</code></td>
<td>
<p>(optional): A character specifying the position of the legend (default: &quot;bottom&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend title text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_legend_title_face">legend_title_face</code></td>
<td>
<p>(optional): A character specifying the font face for the legend title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>(optional):  The number of columns for faceting (default: NULL).
It is recommended to specify this when the number of viral groups is high, to ensure they fit well in one plot.</p>
</td></tr>
<tr><td><code id="VhgBoxplot_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VhgBoxplot generates box plots comparing either E-values, identity, or contig length (Gatherer only) for each virus group from the VirusHunter or Gatherer hittable.
</p>
<p>The user can specify whether to generate box plots for E-values, identity, or contig length (Gatherer only) by specifying the 'y_column'.
This means that 'VhgBoxplot' can generate three different types of box plots.
By default, 'y_column' is set to &quot;ViralRefSeq_E&quot; and will plot the reference E-Value on the y-axis.
Grouping on the x-axis is done by the 'x_column' argument. By default, the &quot;best_query&quot; will be used.
</p>
<p>Additionally, the function calculates summary statistics and identifies outliers for further analysis (&quot;ViralRefSeq_E&quot; and &quot;contig_len&quot; only).
When 'y_column' is set to &quot;ViralRefSeq_E&quot;, the output also includes 'rows_belowthres', which contains the hittable filtered for the rows below the threshold specified in the 'cut' argument.
</p>
<p>The 'cut' argument is used differently depending on the 'y_column' value:
</p>

<ul>
<li><p> For 'y_column' set to &quot;contig_len&quot; or &quot;ViralRefSeq_ident&quot;, the 'cut' argument filters the data to plot only the values with a &quot;ViralRefSeq_E&quot; below the specified threshold (default: 1e-5).
</p>
</li>
<li><p> For 'y_column' set to &quot;ViralRefSeq_E&quot;, the rows are not filtered. Instead, a horizontal line (h_line) is shown in the plot to indicate the cutoff value.
</p>
</li></ul>

<p>This allows the user to plot only the significant contig lengths and identities while also visualizing the number of non-significant and significant values for comparison.
</p>
<p>Warning: In some cases, E-values might be exactly 0. When these values are transformed using -log10, R
returns &quot;inf&quot; as the output. To avoid this issue, we replace all E-values that are 0 with the smallest e-value that is greater than 0.
If the smallest E-value is above the user-defined cutoff, we use a value of <code>cutoff * 10^-10</code> to replace the zeros.
</p>


<h3>Value</h3>

<p>A list containing:
</p>

<ul>
<li><p> The generated box plot.
</p>
</li>
<li><p> Summary statistics.
</p>
</li>
<li><p> Outliers (&quot;ViralRefSeq_E&quot; and &quot;contig_len&quot; only).
</p>
</li>
<li><p> rows_belowthres (&quot;ViralRefSeq_E&quot; only).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

# plot 1 for E-values
plot1 &lt;- VhgBoxplot(file, x_column = "best_query", y_column = "ViralRefSeq_E")
plot1

# plot 2 for identity
plot2 &lt;- VhgBoxplot(file, x_column = "best_query", y_column = "ViralRefSeq_ident")
plot2

# plot 3 custom arguments used
plot3 &lt;- VhgBoxplot(file,
                  x_column = "best_query",
                  y_column = "ViralRefSeq_E",
                  theme_choice = "grey",
                  subtitle = "Custom subtitle: Identity for custom query",
                  xlabel = "Custom x-axis label: Custom query",
                  ylabel = "Custom y-axis label: Viral Reference Evalue in -log10 scale",
                  legend_position = "right")
plot3

# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)


# plot 4: Virusgatherer plot for ViralRefSeq_taxonomy agains contig length
plot5 &lt;- VhgBoxplot(vg_file,x_column = "ViralRefSeq_taxonomy",y_column = "contig_len")
plot5



</code></pre>

<hr>
<h2 id='VhgGetSubject'>VhgGetSubject: Process and Count Viral Subjects within Groups</h2><span id='topic+VhgGetSubject'></span>

<h3>Description</h3>

<p>VhgGetSubject: Process and Count Viral Subjects within Groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgGetSubject(
  file,
  groupby = "best_query",
  remove_identifiers = TRUE,
  include_run_ids = FALSE,
  extract_brackets = FALSE,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgGetSubject_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunter or VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VhgGetSubject_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgGetSubject_+3A_remove_identifiers">remove_identifiers</code></td>
<td>
<p>(optional): if <code>TRUE</code> (default), removes the identifiers in the ViralRefSeq_subject cells.</p>
</td></tr>
<tr><td><code id="VhgGetSubject_+3A_include_run_ids">include_run_ids</code></td>
<td>
<p>(optional): If <code>TRUE</code> (default is <code>TRUE</code>), adds a fourth column named <code>run_ids</code> to the output.
This column contains a comma-separated list of unique identifiers from either the <code>SRA_run</code> or <code>run_id</code> column,
aggregated for each combination of group and subject.</p>
</td></tr>
<tr><td><code id="VhgGetSubject_+3A_extract_brackets">extract_brackets</code></td>
<td>
<p>(optional): extract content within square brackets [].</p>
</td></tr>
<tr><td><code id="VhgGetSubject_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function <code>VhgGetSubject</code> counts the number of viral subjects in the <code>ViralRefSeq_subject</code> column
for each group specified by the <code>groupby</code> argument.
It returns a tibble with three columns: the first column contains the viral group specified by the <code>groupby</code> argument,
the second column lists the viral subjects found in that group, and the third column shows how many times each viral subject appears in that group.
</p>


<h3>Value</h3>

<p>a processed tibble object.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# import data
path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

# process column and filter for significant groups
file &lt;- VhgPreprocessTaxa(file,taxa_rank = "Family")
file_filtered &lt;- VhgSubsetHittable(file,ViralRefSeq_E_criteria = 1e-5)

subject_df &lt;- VhgGetSubject(file_filtered,groupby = "ViralRefSeq_taxonomy",
remove_identifiers = TRUE)

print(subject_df)

</code></pre>

<hr>
<h2 id='VhgIdenFacetedScatterPlot'>VhgIdenFacetedScatterPlot: Create a scatter plot of Viral refseq identity vs. -log10 of viral refseq E-value.</h2><span id='topic+VhgIdenFacetedScatterPlot'></span>

<h3>Description</h3>

<p>VhgIdenFacetedScatterPlot generates a scatter plot of viral refseq identity versus -log10 of refseq E-value
for each virus group in the <code>best_query</code> or <code>ViralRefSeq_taxonomy</code> column . The points are colored based on whether the
E-value meets a specified cutoff and are faceted by the viral groups in the <code>best_query</code> or <code>ViralRefSeq_taxonomy</code> column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgIdenFacetedScatterPlot(
  file,
  groupby = "best_query",
  taxa_rank = "Family",
  cutoff = 1e-05,
  conlen_bubble_plot = FALSE,
  contiglen_breaks = 5,
  theme_choice = "linedraw",
  title = "Faceted scatterplot of viral reference E-values and sequence identity",
  title_size = 16,
  title_face = "bold",
  title_colour = "#2a475e",
  subtitle = NULL,
  subtitle_size = 12,
  subtitle_face = "bold",
  subtitle_colour = "#1b2838",
  xlabel = "Viral reference sequence identity (%)",
  ylabel = "-log10 of viral reference E-values",
  axis_title_size = 12,
  xtext_size = 10,
  x_angle = NULL,
  ytext_size = 10,
  y_angle = NULL,
  legend_position = "bottom",
  legend_title_size = 12,
  legend_title_face = "bold",
  legend_text_size = 10,
  true_colour = "blue",
  false_colour = "red",
  wrap_ncol = 2,
  filter_group_criteria = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_file">file</code></td>
<td>
<p>VirusHunterGatherer hittable.</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): When <code>groupby</code> is set to &quot;ViralRefSeq_taxonomy&quot;, specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_cutoff">cutoff</code></td>
<td>
<p>(optional):  A numeric value representing the cutoff for the refseq E-value. Points with <code>ViralRefSeq_E</code>
less than or equal to this value will be colored blue; otherwise, they will be colored red (default: 1e-5).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_conlen_bubble_plot">conlen_bubble_plot</code></td>
<td>
<p>(optional):  Logical value indicating whether the <code>contig_len</code> column
should be used to size the bubbles in the plot. Applicable only to VirusGatherer hittables input (default: FALSE).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_contiglen_breaks">contiglen_breaks</code></td>
<td>
<p>(optional): Number of breaks (default: 5) for the bubble plot (for <code>conlen_bubble_plot</code>=TRUE).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_theme_choice">theme_choice</code></td>
<td>
<p>(optional): A character indicating the ggplot2 theme to apply. Options include &quot;minimal&quot;,
&quot;classic&quot;, &quot;light&quot;, &quot;dark&quot;, &quot;void&quot;, &quot;grey&quot; (or &quot;gray&quot;), &quot;bw&quot;, &quot;linedraw&quot; (default), and &quot;test&quot;.
Append &quot;_dotted&quot; to any theme to add custom dotted grid lines (e.g., &quot;classic_dotted&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_title">title</code></td>
<td>
<p>(optional):  The title of the plot (default: &quot;Faceted scatter plot of viral reference E-values and sequence identity&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_title_size">title_size</code></td>
<td>
<p>(optional):  The size of the title text (default: 16).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_title_face">title_face</code></td>
<td>
<p>(optional):  The face (bold, italic, etc.) of the title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional):  The color of the title text (default: &quot;#2a475e&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional):  The subtitle of the plot (default: NULL).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional):  The size of the subtitle text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_subtitle_face">subtitle_face</code></td>
<td>
<p>(optional):  The face (bold, italic, etc.) of the subtitle text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_subtitle_colour">subtitle_colour</code></td>
<td>
<p>(optional):  The color of the subtitle text (default: &quot;#1b2838&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_xlabel">xlabel</code></td>
<td>
<p>(optional):  The label for the x-axis (default: &quot;Viral reference sequence identity (%)&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_ylabel">ylabel</code></td>
<td>
<p>(optional):  The label for the y-axis (default: &quot;-log10 of viral reference E-values&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>(optional):  The size of the axis titles (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_xtext_size">xtext_size</code></td>
<td>
<p>(optional):  The size of the x-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_x_angle">x_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the x-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_ytext_size">ytext_size</code></td>
<td>
<p>(optional):  The size of the y-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_y_angle">y_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the y-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_legend_position">legend_position</code></td>
<td>
<p>(optional):  The position of the legend (default: &quot;bottom).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>(optional):  The size of the legend title text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_legend_title_face">legend_title_face</code></td>
<td>
<p>(optional):  The face (bold, italic, etc.) of the legend title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>(optional):  The size of the legend text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_true_colour">true_colour</code></td>
<td>
<p>(optional):  The color for points that meet the cutoff condition (default: &quot;blue&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_false_colour">false_colour</code></td>
<td>
<p>(optional):  The color for points that do not meet the cutoff condition (default: &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_wrap_ncol">wrap_ncol</code></td>
<td>
<p>(optional):  The number of columns for faceting (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdenFacetedScatterPlot_+3A_filter_group_criteria">filter_group_criteria</code></td>
<td>
<p>(optional):  Character vector, numeric vector, or single character/numeric value.
</p>

<ul>
<li><p> Character vector: Names of viral groups to filter.
</p>
</li>
<li><p> Numeric vector: Indices of viral groups to filter.
</p>
</li>
<li><p> Single character or numeric value: Filter a single viral group.
</p>
</li>
<li><p> NULL: No filtering is performed (default).
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>'VhgIdenFacetedScatterPlot' takes a VirusHunter or VirusGatherer hittable and a cutoff value as inputs.
The plot includes:
</p>

<ul>
<li><p> Points colored based on whether they meet the cutoff condition.
</p>
</li>
<li><p> Faceting by the <code>best_query</code> column as the default column. The user can provide their own column
for grouping.
</p>
</li>
<li><p> The option <code>conlen_bubble_plot</code> = TRUE generates a bubble plot where the size of points corresponds to &quot;contig_len&quot; (exclusive to VirusGatherer).
</p>
</li></ul>

<p><code>filter_group_criteria</code>: Allows filtering of viral groups by specifying either a single character
string or a vector of character strings that match unique entries in <code>groupby</code>.
Alternatively, a single numeric value, a range, or a vector of numeric values can be used to filter groups.
</p>
<p>For example, if <code>groupby</code> is &quot;best_query&quot; with the following unique groups:
</p>

<ul>
<li><p> Anello_ORF1core
</p>
</li>
<li><p> Gemini_Rep
</p>
</li>
<li><p> Genomo_Rep
</p>
</li>
<li><p> Hepadna-Nackedna_TP
</p>
</li></ul>

<p>Setting <code>filter_group_criteria</code> to <code>c("Anello_ORF1core", "Genomo_Rep")</code> will filter the data to only
include observations where the &quot;best_query&quot; column has 'Anello_ORF1core' or 'Genomo_Rep'.
Alternatively, setting <code>filter_group_criteria</code> to <code>2:3</code> will return only the second and third
alphabetically ordered viral groups from &quot;best_query&quot;.
The order also matches the order of the viral groups in the faceted scatter plot.
</p>
<p>This is particularly useful when there are too many viral groups to be plotted in a single plot,
allowing for separation into different groups.
It also enables the user to focus on specific groups of interest for more detailed analysis.
</p>
<p>Tibble data frames containing summary statistics (median, Q1, Q3, mean, sd, min, and max) for 'ViralRefSeq_E'
and 'ViralRefSeq_ident' values are generated. Optionally, summary statistics for 'contig_len' values
are also included if applicable. These summary statistics, along with the plot object, are returned within a list object.
</p>
<p>Warning: In some cases, E-values might be exactly 0. When these values are transformed using -log10, R
returns &quot;inf&quot; as the output. To avoid this issue, we replace all E-values that are 0 with the smallest E-value that is greater than 0.
If the smallest E-value is above the user-defined cutoff, we use a value of <code>cutoff * 10^-10</code> to replace the zeros.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>

<ul>
<li><p> Plot: A plot object representing the faceted scatter plot.
</p>
</li>
<li><p> evalue_stats: A tibble data frame with summary statistics for &quot;ViralRefSeq_E&quot; values.
</p>
</li>
<li><p> identity_stats: A tibble data frame with summary statistics for &quot;ViralRefSeq_ident&quot; values.
</p>
</li>
<li><p> contig_stats (optional): A tibble data frame with summary statistics for &quot;contig_len&quot; values, included only if VirusGatherer is used with <code>conlen_bubble_plot=TRUE</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

# plot 1
plot &lt;- VhgIdenFacetedScatterPlot(file,cutoff = 1e-5)

plot

# plot 2 with custom data
custom_plot &lt;- VhgIdenFacetedScatterPlot(file,
                                         cutoff = 1e-4,
                                         theme_choice = "dark",
                                         title = "Custom Scatterplot",
                                         title_size = 18,
                                         title_face = "italic",
                                         title_colour = "orange",
                                         xlabel = "Custom X Label",
                                         ylabel = "Custom Y Label",
                                         axis_title_size = 14,
                                         legend_position = "right",
                                         true_colour = "green",
                                         false_colour = "purple")

custom_plot

# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)

# vgplot: virusgatherer plot with ViralRefSeq_taxonomy as custom grouping
vgplot &lt;- VhgIdenFacetedScatterPlot(vg_file,groupby = "ViralRefSeq_taxonomy")
vgplot


</code></pre>

<hr>
<h2 id='VhgIdentityScatterPlot'>VhgIdentityScatterPlot: Scatter plot for refseq identity vs -log10 of refseq E-value</h2><span id='topic+VhgIdentityScatterPlot'></span>

<h3>Description</h3>

<p>VhgIdentityScatterPlot generates a scatter plot of viral refSeq identity vs. -log10 of viral refseq E-value.
It colors the points based on  phylum  and adds a horizontal line representing the cutoff value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgIdentityScatterPlot(
  file,
  groupby = "best_query",
  taxa_rank = "Family",
  cutoff = 1e-05,
  conlen_bubble_plot = FALSE,
  contiglen_breaks = 5,
  theme_choice = "linedraw",
  cut_colour = "#990000",
  title = "Scatterplot of viral reference E-values and sequence identity",
  title_size = 16,
  title_face = "bold",
  title_colour = "#2a475e",
  subtitle = NULL,
  subtitle_size = 12,
  subtitle_face = "bold",
  subtitle_colour = "#1b2838",
  xlabel = "Viral reference sequence identity (%)",
  ylabel = "-log10 of viral reference E-values",
  axis_title_size = 12,
  xtext_size = 10,
  x_angle = NULL,
  ytext_size = 10,
  y_angle = NULL,
  legend_title = "Group",
  legend_position = "bottom",
  legend_title_size = 12,
  legend_title_face = "bold",
  legend_text_size = 10,
  highlight_groups = NULL,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgIdentityScatterPlot_+3A_file">file</code></td>
<td>
<p>VirusHunterGatherer hittable.</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): When <code>groupby</code> is set to &quot;ViralRefSeq_taxonomy&quot;, specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_cutoff">cutoff</code></td>
<td>
<p>(optional): A numeric value representing the cutoff for the refseq E-value (default: 1e-5).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_conlen_bubble_plot">conlen_bubble_plot</code></td>
<td>
<p>(optional):  Logical value indicating whether the <code>contig_len</code> column
should be used to size the bubbles in the plot. Applicable only to VirusGatherer hittables input (default: FALSE).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_contiglen_breaks">contiglen_breaks</code></td>
<td>
<p>(optional): Number of breaks (default: 5) for the bubble plot (for <code>conlen_bubble_plot</code>=TRUE).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_theme_choice">theme_choice</code></td>
<td>
<p>(optional): A character indicating the ggplot2 theme to apply. Options include &quot;minimal&quot;,
&quot;classic&quot;, &quot;light&quot;, &quot;dark&quot;, &quot;void&quot;, &quot;grey&quot; (or &quot;gray&quot;), &quot;bw&quot;, &quot;linedraw&quot; (default), and &quot;test&quot;.
Append &quot;_dotted&quot; to any theme to add custom dotted grid lines (e.g., &quot;classic_dotted&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_cut_colour">cut_colour</code></td>
<td>
<p>(optional): The color for the horizontal cutoff line (default: &quot;#990000&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_title">title</code></td>
<td>
<p>(optional): The title of the plot (default: &quot;Scatterplot of viral reference E-values and sequence identity&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_title_size">title_size</code></td>
<td>
<p>(optional): The size of the title text (default: 16).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_title_face">title_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): The color of the title text (default: &quot;#2a475e&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): The subtitle of the plot (default: NULL).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): The size of the subtitle text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_subtitle_face">subtitle_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the subtitle text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_subtitle_colour">subtitle_colour</code></td>
<td>
<p>(optional): The color of the subtitle text (default: &quot;#1b2838&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_xlabel">xlabel</code></td>
<td>
<p>(optional): The label for the x-axis (default: &quot;Viral reference sequence identity (%)&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_ylabel">ylabel</code></td>
<td>
<p>(optional): The label for the y-axis (default: &quot;-log10 of viral reference E-values&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>(optional): The size of the axis titles (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_xtext_size">xtext_size</code></td>
<td>
<p>(optional): The size of the x-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_x_angle">x_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the x-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_ytext_size">ytext_size</code></td>
<td>
<p>(optional): The size of the y-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_y_angle">y_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the y-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_legend_title">legend_title</code></td>
<td>
<p>(optional): The title of the legend (default: &quot;Group&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_legend_position">legend_position</code></td>
<td>
<p>(optional): The position of the legend (default: &quot;bottom).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>(optional):  The size of the legend title text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_legend_title_face">legend_title_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the legend title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>(optional): The size of the legend text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_highlight_groups">highlight_groups</code></td>
<td>
<p>(optional): A character vector specifying the names of viral groups to be highlighted in the plot (Default:NULL).</p>
</td></tr>
<tr><td><code id="VhgIdentityScatterPlot_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VhgIdentityScatterPlot generates a scatter plot for refseq sequence identity vs -log10 of refseq E-value.
It accepts both VirusHunter and VirusGatherer hittables as input.
The plot includes:
</p>

<ul>
<li><p> A line indicates whether the observed values are above or below the cutoff specified by the 'cutoff' argument (default: 1e-5).
</p>
</li>
<li><p> The option <code>conlen_bubble_plot</code> = TRUE generates a bubble plot where the size of points corresponds to &quot;contig_len&quot; (exclusive to VirusGatherer).
</p>
</li></ul>

<p>Tibble data frames containing summary statistics (median, Q1, Q3, mean, sd, min, and max) for 'ViralRefSeq_E'
and 'ViralRefSeq_ident' values are generated. Optionally, summary statistics for 'contig_len' values
are also included if applicable. These summary statistics, along with the plot object, are returned within a list object.
</p>
<p><code>highlight_groups</code> enables the user to specify one or more viral groups from the column indicated in the <code>groupby</code> argument. These groups will be highlighted in the plot.
</p>
<p>Warning: In some cases, E-values might be exactly 0. When these values are transformed using -log10, R
returns &quot;inf&quot; as the output. To avoid this issue, we replace all E-values that are 0 with the smallest e-value that is greater than 0.
If the smallest E-value is above the user-defined cutoff, we use a value of <code>cutoff * 10^-10</code> to replace the zeros.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>

<ul>
<li><p> Plot: A plot object representing the faceted scatterplot.
</p>
</li>
<li><p> evalue_stats: A tibble data frame with summary statistics for &quot;ViralRefSeq_E&quot; values.
</p>
</li>
<li><p> identity_stats: A tibble data frame with summary statistics for &quot;ViralRefSeq_ident&quot; values.
</p>
</li>
<li><p> contig_stats (optional): A tibble data frame with summary statistics for &quot;contig_len&quot; values, included only if VirusGatherer is used with <code>conlen_bubble_plot=TRUE</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)


# Basic plot
plot &lt;- VhgIdentityScatterPlot(file,cutoff = 1e-5)

plot(plot$plot)

# Custom plot with additional arguments
custom_plot &lt;- VhgIdentityScatterPlot(file,
                                     cutoff = 1e-5,
                                     theme_choice = "dark",
                                     cut_colour = "blue",
                                     title = "Custom Scatter Plot",
                                     title_size = 18,
                                     title_face = "italic",
                                     title_colour = "darkred",
                                     subtitle = "Custom Subtitle",
                                     subtitle_size = 14,
                                     subtitle_face = "italic",
                                     subtitle_colour = "purple",
                                     xlabel = "Custom X Label",
                                     ylabel = "Custom Y Label",
                                     axis_title_size = 14,
                                     xtext_size = 12,
                                     ytext_size = 12,
                                     legend_title = "Custom Legend",
                                     legend_position = "top",
                                     legend_title_size = 14,
                                     legend_title_face = "italic",
                                     legend_text_size = 12)

plot(custom_plot$plot)

# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)

# vgplot: virusgatherer plot with ViralRefSeq_taxonomy as custom grouping
vgplot &lt;- VhgIdentityScatterPlot(vg_file,groupby = "ViralRefSeq_taxonomy")
vgplot$plot

# plot as bubble plot with contig length as size
vgplot_con &lt;- VhgIdentityScatterPlot(vg_file,groupby = "ViralRefSeq_taxonomy",
conlen_bubble_plot = TRUE,contiglen_breaks = 4,legend_position = "right")

vgplot_con

</code></pre>

<hr>
<h2 id='VhgPreprocessTaxa'>VhgPreprocessTaxa: preprocess ViralRefSeq_taxonomy elements</h2><span id='topic+VhgPreprocessTaxa'></span>

<h3>Description</h3>

<p>VhgPreprocessTaxa: preprocess ViralRefSeq_taxonomy elements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgPreprocessTaxa(file, taxa_rank, num_cores = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgPreprocessTaxa_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunter or VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VhgPreprocessTaxa_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): Specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgPreprocessTaxa_+3A_num_cores">num_cores</code></td>
<td>
<p>(optional): Number of cores used (default: 1)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Process the <code>ViralRefSeq_taxonomy</code> column.
</p>
<p>Besides <code>best_query</code>, the user can utilize the <code>ViralRefSeq_taxonomy</code> column as <code>x_column</code> or <code>groupby</code> in plots.
This column needs preprocessing because it is too long and has too many unique elements for effective grouping.
The element containing the taxa suffix specified by the <code>taxa_rank</code> argument is used. NA values are replaced by &quot;unclassified&quot;.
</p>
<p>This function is used internally by every function that can use the <code>ViralRefSeq_taxonomy</code> column as input.
The user can also apply this function independently to process the taxonomy column and filter for the selected taxa rank in their data.
</p>
<p>For datasets with significantly more than 100,000 observations, it is recommended to use this function to ensure it is skipped in the plot functions.
</p>
<p>The <code>num_cores</code> parameter allows you to divide the dataset into multiple parts, corresponding to the number of cores available.
This enables parallel processing across multiple threads, thereby speeding up the overall processing time.
</p>


<h3>Value</h3>

<p>file with preprocessed ViralRefSeq_taxonomy elements
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

file_filtered &lt;- VhgPreprocessTaxa(file,"Family")

cat("ViralRefSeq_taxonomy before processing:\n")
print(head(file$ViralRefSeq_taxonomy,5))

cat("ViralRefSeq_taxonomy after processing:\n")
print(head(file_filtered$ViralRefSeq_taxonomy,5))



</code></pre>

<hr>
<h2 id='VhgRunsBarplot'>VhgRunsBarplot: Generate a bar plot showing the number of data sets with unique runs found for
each Virus group.</h2><span id='topic+VhgRunsBarplot'></span>

<h3>Description</h3>

<p>VhgRunsBarplot: Generate a bar plot showing the number of data sets with unique runs found for
each Virus group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgRunsBarplot(
  file,
  groupby = "best_query",
  taxa_rank = "Family",
  cut = 1e-05,
  reorder_criteria = "max",
  theme_choice = "linedraw",
  flip_coords = TRUE,
  title = "Distribution of viral groups detected across query sequences",
  title_size = 16,
  title_face = "bold",
  title_colour = "#2a475e",
  subtitle = "default",
  subtitle_size = 12,
  subtitle_face = "bold",
  subtitle_colour = "#1b2838",
  xlabel = "Viral group",
  ylabel = "Number of datasets with hits for group",
  axis_title_size = 12,
  xtext_size = 10,
  x_angle = NULL,
  ytext_size = 10,
  y_angle = NULL,
  remove_group_labels = FALSE,
  legend_title = "Phylum",
  legend_position = "bottom",
  legend_title_size = 12,
  legend_title_face = "bold",
  legend_text_size = 10,
  plot_text = 3,
  plot_text_size = 3.5,
  plot_text_position_dodge = 0.9,
  plot_text_hjust = -0.1,
  plot_text_vjust = 0.5,
  plot_text_colour = "black",
  facet_ncol = NULL,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgRunsBarplot_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunter or VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): When <code>groupby</code> is set to &quot;ViralRefSeq_taxonomy&quot;, specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_cut">cut</code></td>
<td>
<p>(optional): A numeric value representing the cutoff for the refseq E-value (default: 1e-5).
Removes rows in file with values larger than cutoff value in &quot;ViralRefSeq_E&quot; column.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_reorder_criteria">reorder_criteria</code></td>
<td>
<p>(optional): Character string specifying the criteria for reordering the x-axis ('max' (default), 'min','phylum',phylum_max,phylum_min).
NULL sorts alphabetically.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_theme_choice">theme_choice</code></td>
<td>
<p>(optional): A character indicating the ggplot2 theme to apply. Options include &quot;minimal&quot;,
&quot;classic&quot;, &quot;light&quot;, &quot;dark&quot;, &quot;void&quot;, &quot;grey&quot; (or &quot;gray&quot;), &quot;bw&quot;, &quot;linedraw&quot; (default), and &quot;test&quot;.
Append &quot;_dotted&quot; to any theme to add custom dotted grid lines (e.g., &quot;classic_dotted&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_flip_coords">flip_coords</code></td>
<td>
<p>(optional): Logical indicating whether to flip the coordinates of the plot (default: TRUE).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_title">title</code></td>
<td>
<p>(optional): The title of the plot (default: &quot;Distribution of viral groups detected across query sequences&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_title_size">title_size</code></td>
<td>
<p>(optional): The size of the title text (default: 16).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_title_face">title_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): The color of the title text (default: &quot;#2a475e&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): A character specifying the subtitle of the plot.
Default is &quot;default&quot;, which calculates the total number of data sets with hits and returns it as
&quot;total number of data sets with hits: &quot; followed by the calculated number. an empty string (&quot;&quot;) removes the subtitle.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): The size of the subtitle text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_subtitle_face">subtitle_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the subtitle text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_subtitle_colour">subtitle_colour</code></td>
<td>
<p>(optional): The color of the subtitle text (default: &quot;#1b2838&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_xlabel">xlabel</code></td>
<td>
<p>(optional): The label for the x-axis (default: &quot;Viral group&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_ylabel">ylabel</code></td>
<td>
<p>(optional): The label for the y-axis (default: &quot;Number of data sets with hits for group&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>(optional): The size of the axis titles (default: 12).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_xtext_size">xtext_size</code></td>
<td>
<p>(optional): The size of the x-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_x_angle">x_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the x-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_ytext_size">ytext_size</code></td>
<td>
<p>(optional): The size of the y-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_y_angle">y_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the y-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_remove_group_labels">remove_group_labels</code></td>
<td>
<p>(optional): If <code>TRUE</code>, the group labels will be removed; if <code>FALSE</code> or omitted, the labels will be displayed.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_legend_title">legend_title</code></td>
<td>
<p>(optional): A character specifying the title for the legend (default: &quot;Phylum&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_legend_position">legend_position</code></td>
<td>
<p>(optional): The position of the legend (default: &quot;bottom).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend title text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_legend_title_face">legend_title_face</code></td>
<td>
<p>(optional): A character specifying the font face for the legend title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_plot_text">plot_text</code></td>
<td>
<p>(optional): An index (0-3) to select the variable for text labels.
</p>

<ul>
<li><p> 0 = None.
</p>
</li>
<li><p> 1 = Number of viral groups detected across query sequences.
</p>
</li>
<li><p> 2 = Only the percentage.
</p>
</li>
<li><p> 3 = Both (Default).
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_plot_text_size">plot_text_size</code></td>
<td>
<p>(optional): The size of the text labels added to the plot (default: 3.5).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_plot_text_position_dodge">plot_text_position_dodge</code></td>
<td>
<p>(optional): The degree of dodging for positioning text labels (default: 0.9).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_plot_text_hjust">plot_text_hjust</code></td>
<td>
<p>(optional): The horizontal justification of text labels (default: -0.1).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_plot_text_vjust">plot_text_vjust</code></td>
<td>
<p>(optional): The vertical justification of text labels (default: 0.5).
It is recommended to change <code>vjust</code> when setting <code>flip_coords = FALSE</code>.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_plot_text_colour">plot_text_colour</code></td>
<td>
<p>(optional): The color of the text labels added to the plot (default: &quot;black&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>(optional):  The number of columns for faceting (default: NULL).
It is recommended to specify this when the number of viral groups is high, to ensure they fit well in one plot.</p>
</td></tr>
<tr><td><code id="VhgRunsBarplot_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VhgRunsBarplot generates a bar plot showing the number of data sets with unique runs found for
each Virus group. It takes VirusHunter and VirusGatherer hittables as Input.
</p>
<p>Only significant values below the threshold specified by the 'cut' argument (default: 1e-5) are included in the plot.
</p>


<h3>Value</h3>

<p>A list containing the bar plot and tabular data with information from the plot.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># import data
path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

# plot 1: plot boxplot for "identity"
plot &lt;- VhgRunsBarplot(file,cut = 1e-5)
plot


</code></pre>

<hr>
<h2 id='VhgRunsTable'>VhgRunsTable: Generate a gt summary table of unique runs for each virus group</h2><span id='topic+VhgRunsTable'></span>

<h3>Description</h3>

<p>VhgRunsTable generates a summary table of unique runs for each virus group
based on the input data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgRunsTable(
  vh_file,
  groupby = "best_query",
  taxa_rank = "Family",
  cut = 1e-05,
  title = "Summary of unique runs by virus group",
  title_align = "left",
  names_ = NULL,
  align = "left",
  subtitle = NULL,
  data_row.pad = 6,
  column_colour = "dodgerblue4",
  title_size = 26,
  subtitle_size = 14,
  title_weight = "bold",
  title_colour = "dodgerblue4",
  table_font_size = 14,
  cell_colour = "grey90",
  col_everyrow = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgRunsTable_+3A_vh_file">vh_file</code></td>
<td>
<p>A data frame containing the Virushunter hittables results.</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): When <code>groupby</code> is set to &quot;ViralRefSeq_taxonomy&quot;, specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_cut">cut</code></td>
<td>
<p>(optional): A numeric value representing the cutoff for the refseq E-value (default: 1e-5).
Removes rows in file with values larger than cutoff value in &quot;ViralRefSeq_E&quot; column.</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_title">title</code></td>
<td>
<p>(optional): The title of the plot (default: &quot;Summary of unique runs by virus group&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_title_align">title_align</code></td>
<td>
<p>(optional): A character vector specifying the alignment of title (and subtitle) text.
Possible values are &quot;left&quot; (default), &quot;center&quot;, or &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_names_">names_</code></td>
<td>
<p>(optional): A vector of length 3 containing column names (default: c(&quot;Virus Group&quot;,&quot;Number of Unique SRA Runs&quot;,&quot;SRAs Found&quot;)).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_align">align</code></td>
<td>
<p>(optional): A character vector specifying the alignment of text in the table columns.
Possible values are &quot;left&quot; (default), &quot;center&quot;, or &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): A character specifying the subtitle of the plot (default: NULL).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_data_row.pad">data_row.pad</code></td>
<td>
<p>(optional): Numeric value specifying the row padding (default: 6).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_column_colour">column_colour</code></td>
<td>
<p>(optional): character specifying the background colour for the column header (default: &quot;dodgerblue4&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_title_size">title_size</code></td>
<td>
<p>(optional): The size of the title text (default: 26).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the subtitle text (default: 14).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_title_weight">title_weight</code></td>
<td>
<p>(optional): Character or numeric value specifying title font weight.
The weight of the font can be modified thorough a text-based option such as &quot;normal&quot;, &quot;bold&quot; (default),
&quot;lighter&quot;, &quot;bolder&quot;, or, a numeric value between 1 and 1000, inclusive.</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): A character specifying the color for the title text (default: &quot;dodgerblue4&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_table_font_size">table_font_size</code></td>
<td>
<p>(optional): Numeric value specifying table font size. This will change font
size for the column header and for all values in each cell (default: 14).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_cell_colour">cell_colour</code></td>
<td>
<p>(optional): Character specifying cell colour (default: &quot;grey90&quot;).</p>
</td></tr>
<tr><td><code id="VhgRunsTable_+3A_col_everyrow">col_everyrow</code></td>
<td>
<p>(optional): Bool value specifying if every row or every second row of the table
should be filled with the colour from the cell_colour argument. col_everyrow = TRUE colors every
row and col_everyrow = FALSE (default) colors every second row.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VhgRunsTable calculates the number of unique runs for each virus group from the input data set.
It takes VirusHunter hittables as input and determines how many runs (SRA runs or local FASTQ files) found a specific virus family.
</p>
<p>A graphical table is returned with two columns. The first column contains the name of the virus group,
while the second column contains all IDs of SRA runs or local files that found that virus group.
</p>


<h3>Value</h3>

<p>A formatted gt table summarizing unique runs for each virus group
</p>


<h3>See Also</h3>

<p><code><a href="#topic+VhgRunsBarplot">VhgRunsBarplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
vh_file &lt;- ImportVirusTable(path)

# example 1: generate table with defaul arguments
table &lt;- VhgRunsTable(vh_file,cut = 1e-5)

table

# example 2: generate table with custom arguments

table_2 &lt;- VhgRunsTable(vh_file,title = "test",title_align="right",
names_ = c("column_1","column_2","column_3"),align = "right",subtitle="subtitlele")

table_2

# example 3: virusgatherer example
# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)


table_3 &lt;- VhgRunsTable(vg_file,groupby = "ViralRefSeq_taxonomy")

table_3

</code></pre>

<hr>
<h2 id='VhgSubsetHittable'>VhgSubsetHittable: Filter VirusHunter and VirusGatherer hittables</h2><span id='topic+VhgSubsetHittable'></span>

<h3>Description</h3>

<p>VhgSubsetHittable filters a VirusHunter or VirusGatherer hittable based on specified criteria,
including specific virus groups, minimum number of hits, and observations below certain
E-value or identity percentage criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgSubsetHittable(
  file,
  group_column = "best_query",
  virus_groups = NULL,
  num_hits_min = NULL,
  ViralRefSeq_E_criteria = NULL,
  ViralRefSeq_ident_criteria = NULL,
  contig_len_criteria = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgSubsetHittable_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunter or VirusGatherer hittable results.</p>
</td></tr>
<tr><td><code id="VhgSubsetHittable_+3A_group_column">group_column</code></td>
<td>
<p>A string indicating the column containing the virus groups specified in the virus_groups argument.
Note: Gatherer hittables do not have a &quot;best_query&quot; column. Please provide an appropriate column for grouping.</p>
</td></tr>
<tr><td><code id="VhgSubsetHittable_+3A_virus_groups">virus_groups</code></td>
<td>
<p>A character vector specifying virus groups to filter by.</p>
</td></tr>
<tr><td><code id="VhgSubsetHittable_+3A_num_hits_min">num_hits_min</code></td>
<td>
<p>Minimum number of hits required. Default is NULL, which means no filter based on num_hits.</p>
</td></tr>
<tr><td><code id="VhgSubsetHittable_+3A_viralrefseq_e_criteria">ViralRefSeq_E_criteria</code></td>
<td>
<p>Maximum E-value threshold for ViralRefSeq_E criteria. Default is NULL, which means no filter based on ViralRefSeq_E.</p>
</td></tr>
<tr><td><code id="VhgSubsetHittable_+3A_viralrefseq_ident_criteria">ViralRefSeq_ident_criteria</code></td>
<td>
<p>Maximum or minimum sequence identity percentage threshold for ViralRefSeq_ident criteria. Default is NULL, which means no filter based on ViralRefSeq_ident.
If positive, filters where ViralRefSeq_ident is above the threshold. If negative, filters where ViralRefSeq_ident is below the absolute value of the threshold.</p>
</td></tr>
<tr><td><code id="VhgSubsetHittable_+3A_contig_len_criteria">contig_len_criteria</code></td>
<td>
<p>(Gatherer only): Minimum contig length required.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function filters the input VirusHunter or VirusGatherer data (<code>file</code>) based on specified criteria:
</p>

<ul>
<li> <p><code>group_column</code>: Specifies the column to filter by, which must be either &quot;ViralRefSeq_taxonomy&quot; or &quot;best_query&quot;.
</p>
</li>
<li> <p><code>virus_groups</code>: Allows filtering by specific virus groups. If NULL, all virus groups are included.
</p>
</li>
<li> <p><code>num_hits_min</code>: Filters rows where the number of hits (&quot;num_hits&quot;) is greater than or equal to the specified minimum.
</p>
</li>
<li> <p><code>ViralRefSeq_E_criteria</code>: Filters rows where the E-value (&quot;ViralRefSeq_E&quot;) is below the specified maximum threshold.
</p>
</li>
<li> <p><code>ViralRefSeq_ident_criteria</code>: Filters rows where the sequence identity percentage (&quot;ViralRefSeq_ident&quot;) is above or below the specified threshold.
Use a positive value to filter where ViralRefSeq_ident is above the threshold, and a negative value to filter where ViralRefSeq_ident is below the absolute value of the threshold.
</p>
</li>
<li> <p><code>contig_len_criteria</code>: (Gatherer only) Filters rows where the contig length (&quot;contig_len&quot;) is greater than or equal to the specified threshold.
</p>
</li></ul>



<h3>Value</h3>

<p>A filtered dataframe based on the specified criteria.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

cat("The dimensions of the VirusHunter hittable before filtering are: \n");dim(file)

file_filtered &lt;- VhgSubsetHittable(file,group_column = "best_query",
virus_groups = "Anello_ORF1core",
num_hits_min = 4,ViralRefSeq_ident_criteria = -90,ViralRefSeq_E_criteria = 0.00001)

cat("The dimensions of the VirusHunter Hittable after filtering are: \n");dim(file_filtered)

# other examples for viral_group

# Include a single group:
result1 &lt;- VhgSubsetHittable(file, virus_groups = "Hepadna-Nackedna_TP")
# Include multiple groups:
result2 &lt;- VhgSubsetHittable(file, virus_groups = c("Hepadna-Nackedna_TP", "Gemini_Rep"))
# Exclude a single group:
result3 &lt;- VhgSubsetHittable(file, virus_groups = list(exclude = "Hepadna-Nackedna_TP"))
# Exclude multiple groups:
result4 &lt;- VhgSubsetHittable(file, virus_groups = list(exclude =
 c("Hepadna-Nackedna_TP", "Anello_ORF1core")))


</code></pre>

<hr>
<h2 id='VhgSumHitsBarplot'>VhgSumHitsBarplot: Generate a bar plot showing the sum of reads/contigs for each virus family</h2><span id='topic+VhgSumHitsBarplot'></span>

<h3>Description</h3>

<p>VhgSumHitsBarplot preprocesses virus data for plotting and generates a bar plot showing the sum of reads/contigs
for each virus family from the input data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgSumHitsBarplot(
  file,
  groupby = "best_query",
  taxa_rank = "Family",
  y_column = "num_hits",
  cut = 1e-05,
  reorder_criteria = "max",
  theme_choice = "linedraw",
  flip_coords = TRUE,
  title = "Distribution of hits for each virus group",
  title_size = 16,
  title_face = "bold",
  title_colour = "#2a475e",
  subtitle = "default",
  subtitle_size = 12,
  subtitle_face = "bold",
  subtitle_colour = "#1b2838",
  xlabel = "Viral group",
  ylabel = "Total number of hits",
  axis_title_size = 12,
  xtext_size = 10,
  x_angle = NULL,
  ytext_size = 10,
  y_angle = NULL,
  remove_group_labels = FALSE,
  legend_title = "Phylum",
  legend_position = "bottom",
  legend_title_size = 12,
  legend_title_face = "bold",
  legend_text_size = 10,
  plot_text = 3,
  plot_text_size = 3.5,
  plot_text_position_dodge = 0.9,
  plot_text_hjust = -0.1,
  plot_text_vjust = 0.5,
  plot_text_colour = "black",
  facet_ncol = NULL,
  group_unwanted_phyla = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgSumHitsBarplot_+3A_file">file</code></td>
<td>
<p>A data frame containing VirusHunters hittable results.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_groupby">groupby</code></td>
<td>
<p>(optional): A character specifying the column containing the groups (default: &quot;best_query&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_taxa_rank">taxa_rank</code></td>
<td>
<p>(optional): When <code>groupby</code> is set to &quot;ViralRefSeq_taxonomy&quot;, specify the taxonomic rank to group your data by.
Supported ranks are:
</p>

<ul>
<li><p> &quot;Subphylum&quot;
</p>
</li>
<li><p> &quot;Class&quot;
</p>
</li>
<li><p> &quot;Subclass&quot;
</p>
</li>
<li><p> &quot;Order&quot;
</p>
</li>
<li><p> &quot;Suborder&quot;
</p>
</li>
<li><p> &quot;Family&quot; (default)
</p>
</li>
<li><p> &quot;Subfamily&quot;
</p>
</li>
<li><p> &quot;Genus&quot; (including Subgenus)
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_y_column">y_column</code></td>
<td>
<p>A character specifying the column containing the values to be compared. Currently &quot;ViralRefSeq_contigs&quot; (micro-contigs),&quot;contigs&quot;,
and &quot;num_hits&quot; (reads) are supported columns (default:&quot;num_hits&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_cut">cut</code></td>
<td>
<p>(optional): A numeric value representing the cutoff for the refseq E-value (default: 1e-5). Removes rows in file
with values larger than cutoff value in &quot;ViralRefSeq_E&quot; column.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_reorder_criteria">reorder_criteria</code></td>
<td>
<p>(optional): Character string specifying the criteria for reordering the x-axis ('max' (default), 'min','phylum',phylum_max,phylum_min).
NULL sorts alphabetically.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_theme_choice">theme_choice</code></td>
<td>
<p>(optional): A character indicating the ggplot2 theme to apply. Options include &quot;minimal&quot;,
&quot;classic&quot;, &quot;light&quot;, &quot;dark&quot;, &quot;void&quot;, &quot;grey&quot; (or &quot;gray&quot;), &quot;bw&quot;, &quot;linedraw&quot; (default), and &quot;test&quot;.
Append &quot;_dotted&quot; to any theme to add custom dotted grid lines (e.g., &quot;classic_dotted&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_flip_coords">flip_coords</code></td>
<td>
<p>(optional): Logical indicating whether to flip the coordinates of the plot (default: TRUE).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_title">title</code></td>
<td>
<p>(optional): The title of the plot (default: &quot;Distribution of hits for each virus group&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_title_size">title_size</code></td>
<td>
<p>(optional): The size of the title text (default: 16).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_title_face">title_face</code></td>
<td>
<p>(optional): The face (bold, italic, etc.) of the title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): The color of the title text (default: &quot;#2a475e&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): A character specifying the subtitle of the plot.
Default is &quot;total number of hits/micro-contigs: &quot; followed by the calculated number.empty string (&quot;&quot;) removes subtitle.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the subtitle text(default: 12).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_subtitle_face">subtitle_face</code></td>
<td>
<p>(optional): A character specifying the font face for the subtitle text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_subtitle_colour">subtitle_colour</code></td>
<td>
<p>(optional): A character specifying the color for the subtitle text (default: &quot;#1b2838&quot;).
.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_xlabel">xlabel</code></td>
<td>
<p>(optional): The label for the x-axis (default: &quot;Viral group&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_ylabel">ylabel</code></td>
<td>
<p>(optional): The label for the y-axis (default: &quot;Total number of hits&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>(optional): The size of the axis titles (default: 12).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_xtext_size">xtext_size</code></td>
<td>
<p>(optional): The size of the x-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_x_angle">x_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the x-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_ytext_size">ytext_size</code></td>
<td>
<p>(optional): The size of the y-axis text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_y_angle">y_angle</code></td>
<td>
<p>(optional): An integer specifying the angle (in degrees) for the y-axis text labels. Default is NULL, meaning no change.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_remove_group_labels">remove_group_labels</code></td>
<td>
<p>(optional): If <code>TRUE</code>, the group labels will be removed; if <code>FALSE</code> or omitted, the labels will be displayed.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_legend_title">legend_title</code></td>
<td>
<p>(optional): A character specifying the title for the legend (default: &quot;Phylum&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_legend_position">legend_position</code></td>
<td>
<p>(optional): A character specifying the position of the legend (default: &quot;bottom&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_legend_title_size">legend_title_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend title text (default: 12).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_legend_title_face">legend_title_face</code></td>
<td>
<p>(optional): A character specifying the font face for the legend title text (default: &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_legend_text_size">legend_text_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the legend text (default: 10).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_plot_text">plot_text</code></td>
<td>
<p>(optional): An index (0-3) to select the variable for text labels.
</p>

<ul>
<li><p> 0 = None.
</p>
</li>
<li><p> 1 = Number of hits for each viral group.
</p>
</li>
<li><p> 2 = Only the percentage.
</p>
</li>
<li><p> 3 = Both (Default).
</p>
</li></ul>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_plot_text_size">plot_text_size</code></td>
<td>
<p>(optional): The size of the text labels added to the plot (default: 3.5).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_plot_text_position_dodge">plot_text_position_dodge</code></td>
<td>
<p>(optional): The degree of dodging for positioning text labels (default: 0.9).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_plot_text_hjust">plot_text_hjust</code></td>
<td>
<p>(optional): The horizontal justification of text labels (default: -0.1).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_plot_text_vjust">plot_text_vjust</code></td>
<td>
<p>(optional): The vertical justification of text labels (default: 0.5).
It is recommended to change <code>vjust</code> when setting <code>flip_coords = FALSE</code>.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_plot_text_colour">plot_text_colour</code></td>
<td>
<p>(optional): The color of the text labels added to the plot (default: &quot;black&quot;).</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_facet_ncol">facet_ncol</code></td>
<td>
<p>(optional):  The number of columns for faceting (default: NULL).
It is recommended to specify this when the number of viral groups is high, to ensure they fit well in one plot.</p>
</td></tr>
<tr><td><code id="VhgSumHitsBarplot_+3A_group_unwanted_phyla">group_unwanted_phyla</code></td>
<td>
<p>(optional): A character string specifying which group of viral phyla to retain in the analysis.
Valid values are:
</p>

<dl>
<dt>&quot;rna&quot;</dt><dd><p>Retain only the phyla specified for RNA viruses.</p>
</dd>
<dt>&quot;smalldna&quot;</dt><dd><p>Retain only the phyla specified for small DNA viruses.</p>
</dd>
<dt>&quot;largedna&quot;</dt><dd><p>Retain only the phyla specified for large DNA viruses.</p>
</dd>
<dt>&quot;others&quot;</dt><dd><p>Retain only the phyla that match small DNA, Large DNA and RNA viruses.</p>
</dd>
</dl>

<p>All other phyla not in the specified group will be grouped into a single category:
&quot;Non-RNA-virus&quot; for <code>"rna"</code>, &quot;Non-Small-DNA-Virus&quot; for <code>"smalldna"</code>,&quot;Non-Large-DNA-Virus&quot; for <code>"largedna"</code>,or &quot;Other Viruses&quot; for <code>"others"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VhgSumHitsBarplot preprocesses virus data for plotting by calculating the sum of
hits for each virus family from the input data set (accepts only VirusHunter hittables).
It then generates a bar plot showing the sum of hits for each virus family.
Additionally, it returns the processed data for further analysis.
</p>


<h3>Value</h3>

<p>A list containing the generated bar plot and processed data.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
file &lt;- ImportVirusTable(path)

# plot 1: plot boxplot for reads
plot &lt;- VhgSumHitsBarplot(file,cut = 1e-5)
plot

# plot 2: plot boxplot for micro_reads
plot_reads &lt;- VhgSumHitsBarplot(file,cut = 1e-5,
y_column = "ViralRefSeq_contigs")
plot_reads

# import gatherer files
path2 &lt;- system.file("extdata", "virusgatherer.tsv", package = "Virusparies")
vg_file &lt;- ImportVirusTable(path2)


# plot 3: contigs in Gatherer
contig_plot &lt;- VhgSumHitsBarplot(vg_file,groupby = "ViralRefSeq_taxonomy",
y_column = "contig")
contig_plot

</code></pre>

<hr>
<h2 id='VhgTabularRasa'>VhgTabularRasa: Generate Custom Formatted Graphical Tables</h2><span id='topic+VhgTabularRasa'></span>

<h3>Description</h3>

<p>VhgTabularRasa creates a formatted table using the <code>gt</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VhgTabularRasa(
  file,
  title = "Graphical Table",
  title_align = "left",
  names_ = NULL,
  align = "left",
  subtitle = NULL,
  data_row.pad = 6,
  column_colour = "dodgerblue4",
  title_size = 26,
  subtitle_size = 14,
  title_weight = "bold",
  title_colour = "dodgerblue4",
  table_font_size = 14,
  cell_colour = "grey90",
  col_everyrow = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="VhgTabularRasa_+3A_file">file</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_title">title</code></td>
<td>
<p>(optional): The title of the plot (default: &quot;Graphical Table&quot;).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_title_align">title_align</code></td>
<td>
<p>(optional): A character vector specifying the alignment of title (and subtitle) text.
Possible values are &quot;left&quot; (default), &quot;center&quot;, or &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_names_">names_</code></td>
<td>
<p>(optional): A vector containing column names, with a length matching the number
of columns in the file argument (default: names(file)).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_align">align</code></td>
<td>
<p>(optional): A character vector specifying the alignment of text in the table columns.
Possible values are &quot;left&quot; (default), &quot;center&quot;, or &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_subtitle">subtitle</code></td>
<td>
<p>(optional): A character specifying the subtitle of the plot (default: NULL).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_data_row.pad">data_row.pad</code></td>
<td>
<p>(optional): Numeric value specifying the row padding (default: 6).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_column_colour">column_colour</code></td>
<td>
<p>(optional): character specifying the background colour for the column header (default: &quot;dodgerblue4&quot;).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_title_size">title_size</code></td>
<td>
<p>(optional): The size of the title text (default: 26).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_subtitle_size">subtitle_size</code></td>
<td>
<p>(optional): Numeric specifying the size of the subtitle text (default: 14).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_title_weight">title_weight</code></td>
<td>
<p>(optional): Character or numeric value specifying title font weight.
The weight of the font can be modified thorough a text-based option such as &quot;normal&quot;, &quot;bold&quot; (default),
&quot;lighter&quot;, &quot;bolder&quot;, or, a numeric value between 1 and 1000, inclusive.</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_title_colour">title_colour</code></td>
<td>
<p>(optional): A character specifying the color for the title text (default: &quot;dodgerblue4&quot;).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_table_font_size">table_font_size</code></td>
<td>
<p>(optional): Numeric value specifying table font size. This will change font
size for the column header and for all values in each cell (default: 14).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_cell_colour">cell_colour</code></td>
<td>
<p>(optional): Character specifying cell colour (default: &quot;grey90&quot;).</p>
</td></tr>
<tr><td><code id="VhgTabularRasa_+3A_col_everyrow">col_everyrow</code></td>
<td>
<p>(optional): Bool value specifying if every row or every second row of the table
should be filled with the colour from the cell_colour argument. col_everyrow = TRUE colors every
row and col_everyrow = FALSE (default) colors every second row.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>VhgTabularRasa creates a formatted table using the <code>gt</code> package, based on input data with specified column names.
It is particularly useful for generating tables that cannot be produced with <code>vhRunsTable</code>,
when the input data does not originate from the <code>vhRunsBarplot</code> functions.
</p>
<p>The <code>VhgTabularRasa</code> function allows users to generate tables styled like Virusparies tables using their own input data.
Additionally, users can create custom tables by adjusting the parameters within this function.
</p>
<p><code>VhgTabularRasa</code> just like <code>vhRunsTable</code> empowers users to tailor their tables to suit their needs.
With its customizable features, users can effortlessly modify titles, subtitles, and column names.
By default, column names are derived from the data frame's structure using <code>names(file)</code>.
If the data frame lacks column names, an error message is triggered. However, users can
supply their own column names via the <code>names_</code> argument, requiring a vector of names matching the data frame's column count.
</p>
<p><code>VhgTabularRasa</code> takes data frames as Input. Passing any other object type results in an error message.
Users can fine-tune their tables with options to adjust text attributes, alignment, and size, as well as row padding
and color schemes for titles, subtitles, columns, and backgrounds.
If that is not enough, <code>VhgTabularRasa</code> returns an gt tables object, which
can be further manipulated with the <code>gt</code> package available on CRAN.
</p>


<h3>Value</h3>

<p>Returns a <code>gt</code> table object formatted according to the specified parameters.
</p>


<h3>Author(s)</h3>

<p>Sergej Ruff
</p>


<h3>References</h3>

<p>Iannone R, Cheng J, Schloerke B, Hughes E, Lauer A, Seo J(2024). <em>gt: Easily Create Presentation-Ready DisplayTables</em>.
R package version 0.10.1,<a href="https://CRAN.R-project.org/package=gt">https://CRAN.R-project.org/package=gt</a>.
</p>


<h3>See Also</h3>

<p>VirusHunterGatherer is available here: <a href="https://github.com/lauberlab/VirusHunterGatherer">https://github.com/lauberlab/VirusHunterGatherer</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path &lt;- system.file("extdata", "virushunter.tsv", package = "Virusparies")
vh_file &lt;- ImportVirusTable(path)

# plot 1: plot boxplot for "identity"
identity &lt;- VhgBoxplot(vh_file,y_column = "ViralRefSeq_ident")

# generate table
VhgTabularRasa(identity$summary_stats)


# example 2: plot part of Vh_file (could be any other table)
# using first 10 rows of SRA_run,num_hits,bestquery,ViralRefSeq_E and Identity col.
vh_file_part &lt;- vh_file[c(1:10),c(1,7,9,10,11)]

VhgTabularRasa(vh_file_part,title = "first 10 rows of vh_file",subtitle =
"example for any table",names_ = c("Runs","Number of Contigs","Best Query Result",
"Refrence E-Value","Refrence Identity"))



</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
