<!DOCTYPE html><html><head><title>Help for package vivaldi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {vivaldi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_metadata'><p>add_metadata</p></a></li>
<li><a href='#af_distribution'><p>af_distribution</p></a></li>
<li><a href='#arrange_data'><p>arrange_data</p></a></li>
<li><a href='#dNdS_segment'><p>dNdS_segment</p></a></li>
<li><a href='#example_filtered_SNV_df'><p>Example Dataframe</p>
The DF_filt_SNVs dataframe created in the vignette</a></li>
<li><a href='#filter_variants'><p>filter_variants</p></a></li>
<li><a href='#merge_replicates'><p>merge_replicates</p></a></li>
<li><a href='#plot_shannon'><p>plot_shannon</p></a></li>
<li><a href='#position_allele_freq'><p>position_allele_freq</p></a></li>
<li><a href='#prepare_annotations'><p>prepare_annotations</p></a></li>
<li><a href='#read_reference_fasta_dna'><p>read_reference_fasta_dna</p></a></li>
<li><a href='#shannon_entropy'><p>shannon_entropy</p></a></li>
<li><a href='#shared_snv_plot'><p>shared_snv_plot</p></a></li>
<li><a href='#shared_snv_table'><p>shared_snv_table</p></a></li>
<li><a href='#snpeff_info'><p>snpeff_info</p></a></li>
<li><a href='#snv_genome'><p>snv_genome</p></a></li>
<li><a href='#snv_location'><p>snv_location</p></a></li>
<li><a href='#snv_segment'><p>snv_segment</p></a></li>
<li><a href='#tally_it'><p>tally_it</p></a></li>
<li><a href='#tstv_plot'><p>tstv_plot</p></a></li>
<li><a href='#tstv_ratio'><p>tstv_ratio</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Viral Variant Location and Diversity</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of minor alleles in Illumina sequencing data of viral
    genomes. Functions in 'vivaldi' primarily operate on vcf files.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/GreshamLab/vivaldi">https://github.com/GreshamLab/vivaldi</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/GreshamLab/vivaldi/issues">https://github.com/GreshamLab/vivaldi/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr (&ge; 1.0.2), ggplot2 (&ge; 3.3.2), glue (&ge; 1.4.2),
magrittr (&ge; 2.0.1), plotly (&ge; 4.10.0), seqinr (&ge; 4.2-8),
tidyr (&ge; 1.1.2), tidyselect (&ge; 1.1.2), vcfR (&ge; 1.12.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>kableExtra, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-21 19:27:42 UTC; esbor</td>
</tr>
<tr>
<td>Author:</td>
<td>Marissa Knoll [aut],
  Katherine Johnson [aut],
  Megan Hockman [aut],
  Eric Borenstein [aut],
  Mohammed Khalfan [aut],
  Elodie Ghedin [aut],
  David Gresham [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Gresham &lt;dg107@nyu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-21 20:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_metadata'>add_metadata</h2><span id='topic+add_metadata'></span>

<h3>Description</h3>

<p>Adds metadata information to the vcf dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_metadata(df, metadf, by_vcf, by_meta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_metadata_+3A_df">df</code></td>
<td>
<p>A rearranged vcf dataframe (arrange_data)</p>
</td></tr>
<tr><td><code id="add_metadata_+3A_metadf">metadf</code></td>
<td>
<p>A metadata dataframe</p>
</td></tr>
<tr><td><code id="add_metadata_+3A_by_vcf">by_vcf</code></td>
<td>
<p>A vector of column names in the vcf dataframe that should be used to merge the vcf data with the metadata</p>
</td></tr>
<tr><td><code id="add_metadata_+3A_by_meta">by_meta</code></td>
<td>
<p>A vector of column names in the metadata dataframe that should be used to merge the metadata with the vcf data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vcf dataframe with metadata included
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(CHROM = c("A", "B"),
                 POS = c(234, 240),
                 REF = c("G", "A"),
                 ALT = c("A", "G")
)

sizes &lt;- data.frame(segment = c("A", "B"),
                    SegmentSize = c(2280, 2274)
)

df

sizes

# Add a new column of sizes of the segments which are necessary for
# downstream calculations such as transition:transversion (tstv) and dNdS.
add_metadata(df, sizes, c('CHROM'), c('segment'))

</code></pre>

<hr>
<h2 id='af_distribution'>af_distribution</h2><span id='topic+af_distribution'></span>

<h3>Description</h3>

<p>Plots distribution of all minor variants
</p>


<h3>Usage</h3>

<pre><code class='language-R'>af_distribution(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="af_distribution_+3A_df">df</code></td>
<td>
<p>A dataframe that has been arranged (arrange_data) and filtered (filter_variants)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots with the distribution of all minor variants
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1:
df &lt;- data.frame(sample = c("m1", "m2", "m1", "m2", "m1"),
                 CHROM = c("PB1", "PB1", "PB2", "PB2", "NP"),
                 POS = c(234, 234, 240, 240, 254),
                 REF = c("G", "G", "A", "A", "C"),
                 ALT = c("A", "A", "G", "G", "T"),
                 minorfreq = c(0.010, 0.022, 0.043, 0.055, 0.011),
                 majorfreq = c(0.990, 0.978, 0.957, 0.945, 0.989),
                 minorcount = c(7, 15, 26, 32, 7),
                 majorcount = c(709, 661, 574, 547, 610),
                 gt_DP = c(716, 676, 600, 579, 617)
)

af_distribution(df)

# Example 2:
af_distribution(example_filtered_SNV_df)

</code></pre>

<hr>
<h2 id='arrange_data'>arrange_data</h2><span id='topic+arrange_data'></span>

<h3>Description</h3>

<p>Reads in a directory of VCF files and converts them into a single dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arrange_data(
  vardir,
  reference_fasta,
  annotated = "yes",
  ntlist = c("A", "G", "T", "C", "-"),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="arrange_data_+3A_vardir">vardir</code></td>
<td>
<p>Directory path containing vcf files</p>
</td></tr>
<tr><td><code id="arrange_data_+3A_reference_fasta">reference_fasta</code></td>
<td>
<p>Reference fasta file used for alignment</p>
</td></tr>
<tr><td><code id="arrange_data_+3A_annotated">annotated</code></td>
<td>
<p>Whether the VCF files are annotated using snpeff &quot;yes&quot; or &quot;no&quot; (default &quot;yes&quot;)</p>
</td></tr>
<tr><td><code id="arrange_data_+3A_ntlist">ntlist</code></td>
<td>
<p>Nucleotides (default A, T, G, C) used for finding multiple alt alleles</p>
</td></tr>
<tr><td><code id="arrange_data_+3A_verbose">verbose</code></td>
<td>
<p>set verbosity of the vcfR commands</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A large dataframe containing information from all input VCF files
</p>

<hr>
<h2 id='dNdS_segment'>dNdS_segment</h2><span id='topic+dNdS_segment'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data), filtered (filter_variants), and annotated (prepare_annotations), calculates dNdS, and outputs plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dNdS_segment(annotation_df, SPLICEFORMS)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dNdS_segment_+3A_annotation_df">annotation_df</code></td>
<td>
<p>A rearranged, filtered, and annotated vcf dataframe - must be for amino-acid specific calculations, cannot be the same as the dataframe used for SNP calculations</p>
</td></tr>
<tr><td><code id="dNdS_segment_+3A_spliceforms">SPLICEFORMS</code></td>
<td>
<p>A character vector of isoform names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the dN/dS ratio for each splice form (rather than segment) for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample Data
head(example_filtered_SNV_df)
dim(example_filtered_SNV_df)

# Plot showing the dN/dS ratio for each splice form
SPLICEFORMS = c("H1N1_PB2.1", "H1N1_PB1.1","H1N1_NS.2")
dNdS_segment(example_filtered_SNV_df, SPLICEFORMS)
</code></pre>

<hr>
<h2 id='example_filtered_SNV_df'>Example Dataframe
The DF_filt_SNVs dataframe created in the vignette</h2><span id='topic+example_filtered_SNV_df'></span>

<h3>Description</h3>

<p>Example Dataframe
The DF_filt_SNVs dataframe created in the vignette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_filtered_SNV_df
</code></pre>


<h3>Format</h3>

<p>## 'example_filtered_SNV_df'
A data frame with 735 rows and 57 columns:
</p>

<hr>
<h2 id='filter_variants'>filter_variants</h2><span id='topic+filter_variants'></span>

<h3>Description</h3>

<p>Filters single-nucleotide variants using a coverage and frequency cutoff
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_variants(df, coverage_cutoff = 200, frequency_cutoff = 0.03)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_variants_+3A_df">df</code></td>
<td>
<p>A rearranged VCF dataframe (rearranged using the arrange_data function)</p>
</td></tr>
<tr><td><code id="filter_variants_+3A_coverage_cutoff">coverage_cutoff</code></td>
<td>
<p>The coverage cutoff for calling a SNV (default: 200x)</p>
</td></tr>
<tr><td><code id="filter_variants_+3A_frequency_cutoff">frequency_cutoff</code></td>
<td>
<p>Frequency cutoff for calling a SNV (default: 3%)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A filtered VCF dataframe
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(CHROM = c("A", "B", "C"),
                POS = c(234, 240, 255),
                ALT_FREQ = c(0.016, 0.049, 0.031),
                gt_DP = c(716, 600, 187)
)

df

# Default: filter by 3% frequency threshold and 200 coverage cutoff
filter_variants(df)

# Example 1: A 1% allele frequency threshold and 200 coverage cutoff
filter_variants(df, coverage_cutoff = 200, frequency_cutoff = 0.01)

# Example 2: A 2% allele frequency threshold and 100 coverage cutoff
filter_variants(df, coverage_cutoff = 100, frequency_cutoff = 0.02)


</code></pre>

<hr>
<h2 id='merge_replicates'>merge_replicates</h2><span id='topic+merge_replicates'></span>

<h3>Description</h3>

<p>Merges replicate VCF files into a single dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_replicates(vardf, repdata, nameofrep1, nameofrep2, commoncols)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge_replicates_+3A_vardf">vardf</code></td>
<td>
<p>Data frame of variants</p>
</td></tr>
<tr><td><code id="merge_replicates_+3A_repdata">repdata</code></td>
<td>
<p>Data frame of replicate information</p>
</td></tr>
<tr><td><code id="merge_replicates_+3A_nameofrep1">nameofrep1</code></td>
<td>
<p>Name of variable representing the first replicate, must be written with quotes</p>
</td></tr>
<tr><td><code id="merge_replicates_+3A_nameofrep2">nameofrep2</code></td>
<td>
<p>Name of variable representing the second replicate</p>
</td></tr>
<tr><td><code id="merge_replicates_+3A_commoncols">commoncols</code></td>
<td>
<p>List of columns to merge the replicates by</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing replicate information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(sample = c("m1", "m2", "m1", "m2", "m1"),
                 CHROM = c("PB1", "PB1", "PB2", "PB2", "NP"),
                 POS = c(234, 234, 240, 240, 254),
                 REF = c("G", "G", "A", "A", "C"),
                 ALT = c("A", "A", "G", "G", "T"),
                 minorfreq = c(0.010, 0.022, 0.043, 0.055, 0.011),
                 majorfreq = c(0.990, 0.978, 0.957, 0.945, 0.989),
                 minorcount = c(7, 15, 26, 32, 7),
                 majorcount = c(709, 661, 574, 547, 610),
                 gt_DP = c(716, 676, 600, 579, 617)
)

# Dataframe shows a pair of replicates and their variants at 3 positions.
df

replicates &lt;- data.frame(filename = c("m1","m2"),
                         replicate = c("rep1", "rep2"),
                         sample = c("a_2_iv", "a_2_iv")
)

# Dataframe showing relationship between filename, replicate, and sample name
replicates

# Merge by the following columns
cols = c("sample","CHROM","POS","REF","ALT")

merge_replicates(df, replicates, "rep1", "rep2", cols)
# The dataframe now contains the 2 variants at positions 234 &amp; 240 that were
# detected in both sequencing replicates whereas the variant at position 254
# was only in a single replicate so it was removed during the merge.

</code></pre>

<hr>
<h2 id='plot_shannon'>plot_shannon</h2><span id='topic+plot_shannon'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data), filtered (filter_variants), and piped through the Shannon calculations (shannon_entropy) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_shannon(shannon_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_shannon_+3A_shannon_df">shannon_df</code></td>
<td>
<p>A dataframe that has been arranged (arrange_data), filtered (filter_variants), and piped through the Shannon calculations (shannon_entropy)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'plot_shannon()' function takes the variant dataframe and generates three plots.
1. The Shannon entropy, or amount of diversity, at each position in the genome at which a variant was found.
2. The Shannon entropy summed over each segment
3. The Shannon entropy summed over each genome
A higher value indicates more diversity.
</p>


<h3>Value</h3>

<p>Three plots showing the nt Shannon, chrom Shannon, and full genome Shannon calculations
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample dataframe
df &lt;- data.frame(sample = c("m1", "m2", "m1", "m2", "m1"),
                 CHROM = c("PB1", "PB1", "PB2", "PB2", "NP"),
                 POS = c(234, 234, 240, 240, 254),
                 minorfreq = c(0.010, 0.022, 0.043, 0.055, 0.011),
                 majorfreq = c(0.990, 0.978, 0.957, 0.945, 0.989),
                 SegmentSize = c(2280, 2280, 2274, 2274, 1809)
)

df

genome_size = 13133

# Modify the dataframe to add 5 new columns of shannon entropy data:
# 1. shannon_ntpos
# 2. chrom_shannon
# 3. genome_shannon
# 4. shannon_chrom_perkb
# 5. genome_shannon_perkb
shannon_df = shannon_entropy(df, genome_size)

# Plot
plot_shannon(shannon_df)

</code></pre>

<hr>
<h2 id='position_allele_freq'>position_allele_freq</h2><span id='topic+position_allele_freq'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data) and filtered (filter_variants) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>position_allele_freq(vardf, segment, nt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="position_allele_freq_+3A_vardf">vardf</code></td>
<td>
<p>A rearranged (arrange_data) and filtered (filtered_variants) vcf dataframe</p>
</td></tr>
<tr><td><code id="position_allele_freq_+3A_segment">segment</code></td>
<td>
<p>Name of segment (must be in quotes)</p>
</td></tr>
<tr><td><code id="position_allele_freq_+3A_nt">nt</code></td>
<td>
<p>Position on segment (must be in quotes)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the the frequencies of the major and minor allele at the given position across all samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>position_allele_freq(example_filtered_SNV_df,"H1N1_NP", "1247")
</code></pre>

<hr>
<h2 id='prepare_annotations'>prepare_annotations</h2><span id='topic+prepare_annotations'></span>

<h3>Description</h3>

<p>Separates the SNPeff annotations found in an annotated and rearranged VCF dataframe (arranged using arrange_data)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_annotations(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_annotations_+3A_df">df</code></td>
<td>
<p>A rearranged and annotated VCF dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing each annotation on a separate column
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example: Shows the separation of the ANN column based on | delimiter.
test &lt;- data.frame( ANN = c("A|B|C|D|E|F|G|H|I|J|K|L|M|N|O|P"))

# The ANN column will be split based on the strings in `snpeff_info()` and
# an additional "error" column.
snpeff_info()

# Split the SNPeff annotations in "ANN" column and save to dataframe `df`
df &lt;- prepare_annotations(df)

# The one "ANN" column is split into 16 columns
dim(test)
dim(df)

</code></pre>

<hr>
<h2 id='read_reference_fasta_dna'>read_reference_fasta_dna</h2><span id='topic+read_reference_fasta_dna'></span>

<h3>Description</h3>

<p>Imports reference fasta, generates a dataframe with chroms and chrom lengths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_reference_fasta_dna(reference_fasta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_reference_fasta_dna_+3A_reference_fasta">reference_fasta</code></td>
<td>
<p>The name and location of the reference fasta file used for alignment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing the chroms and chrom lengths of a reference fasta
</p>

<hr>
<h2 id='shannon_entropy'>shannon_entropy</h2><span id='topic+shannon_entropy'></span>

<h3>Description</h3>

<p>Takes a rearranged vcf dataframe and calculates the Shannon entropy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shannon_entropy(df, genome_size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shannon_entropy_+3A_df">df</code></td>
<td>
<p>A rearranged vcf dataframe (arrange_data)</p>
</td></tr>
<tr><td><code id="shannon_entropy_+3A_genome_size">genome_size</code></td>
<td>
<p>Size of whole genome being used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Shannon entropy is a commonly used metric to describe the amount of genetic
diversity in sequencing data. It is calculated by considering the frequency
of the ALT and REF allele at every position and then summing those values
over 1) a segment or 2) the entire genome. These values can then be
normalized by sequence length (kb) in order to compare across different
segments or samples.
</p>


<h3>Value</h3>

<p>A dataframe with Shannon entropy/kb calculations for the chroms and entire genome
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample dataframe
df &lt;- data.frame(sample = c("m1", "m2", "m1", "m2", "m1"),
                 CHROM = c("PB1", "PB1", "PB2", "PB2", "NP"),
                 minorfreq = c(0.010, 0.022, 0.043, 0.055, 0.011),
                 majorfreq = c(0.990, 0.978, 0.957, 0.945, 0.989),
                 SegmentSize = c(2280, 2280, 2274, 2274, 1809)
)

df

genome_size = 13133

# MOdify the dataframe to add 5 new columns of shannon entropy data:
# 1. shannon_ntpos
# 2. chrom_shannon
# 3. genome_shannon
# 4. shannon_chrom_perkb
# 5. genome_shannon_perkb
shannon_entropy(df, genome_size)

</code></pre>

<hr>
<h2 id='shared_snv_plot'>shared_snv_plot</h2><span id='topic+shared_snv_plot'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data) and filtered (filter_variants) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shared_snv_plot(vardf, samples = unique(DF_filt$sample))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shared_snv_plot_+3A_vardf">vardf</code></td>
<td>
<p>A rearranged (arrange_data) and filtered (filtered_variants) vcf dataframe</p>
</td></tr>
<tr><td><code id="shared_snv_plot_+3A_samples">samples</code></td>
<td>
<p>A vector of samples to be compared (default:all samples in DF_filt)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the location of variants and the number of samples that contain each variant
</p>


<h3>Examples</h3>

<pre><code class='language-R'>samples = c("a_1_fb", "a_1_iv", "a_2_fb", "a_2_iv", "a_3_fb", "a_3_iv", "b_1_fb", "b_1_iv")
shared_snv_plot(example_filtered_SNV_df, samples)

</code></pre>

<hr>
<h2 id='shared_snv_table'>shared_snv_table</h2><span id='topic+shared_snv_table'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data) and filtered
(filter_variants) and outputs a table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shared_snv_table(vardf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shared_snv_table_+3A_vardf">vardf</code></td>
<td>
<p>A rearranged (arrange_data) and filtered (filtered_variants) vcf dataframe</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'shared_snv_table()' function takes the variant dataframe and creates a
new table, listing the variants in descending order of frequency how many
samples they are found in. This function is meant to simplify further
investigation of visual patterns in the previous plot.
</p>


<h3>Value</h3>

<p>A table listing variants in order by how many samples they are found in
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample dataframe has 57 columns
dim(example_filtered_SNV_df)

# Simplify sample dataframe
df &lt;- shared_snv_table(example_filtered_SNV_df)

# Dataframe created has 15 columns
df
dim(df)

</code></pre>

<hr>
<h2 id='snpeff_info'>snpeff_info</h2><span id='topic+snpeff_info'></span>

<h3>Description</h3>

<p>Returns vector containing information in snpeff annotations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snpeff_info()
</code></pre>


<h3>Value</h3>

<p>Returns vector containing information in snpeff annotations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>snpeff_info()
</code></pre>

<hr>
<h2 id='snv_genome'>snv_genome</h2><span id='topic+snv_genome'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data) and filtered (filter_variants) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snv_genome(vardf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snv_genome_+3A_vardf">vardf</code></td>
<td>
<p>A rearranged (arrange_data) and filtered (filtered_variants) vcf dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A bar plot showing the number of variants per sample colored by their SNPEff annotation
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Simple dataframe
df &lt;- data.frame(sample = c("m1", "m1", "m1", "m1", "m1",
                            "m2", "m2", "m2", "m2", "m2"),
			    annotation = c("downstrean_gene_variant", "synonymous_variant",
				                 "synonymous_variant", "stop_gained",
				                 "missense_variant", "downstrean_gene_variant",
				                 "downstrean_gene_variant", "synonymous_variant",
				                 "stop_gained", "missense_variant")
)

df

snv_genome(df)

# Example 2: Sample dataframe
snv_genome(example_filtered_SNV_df)

</code></pre>

<hr>
<h2 id='snv_location'>snv_location</h2><span id='topic+snv_location'></span>

<h3>Description</h3>

<p>Reads in the vcf dataframe and generates a plot showing the frequency and location of SNVs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snv_location(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snv_location_+3A_df">df</code></td>
<td>
<p>A rearranged dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the location and frequency of SNVs found across samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1:
df &lt;- data.frame(sample = c("m1", "m1", "m1", "m1", "m1",
                            "m2", "m2", "m2", "m2", "m2"),
                 CHROM = c("PB1", "PB1", "PB2", "PB2", "PB2",
                           "PB1", "PB1", "PB2", "PB2", "PB2"),
                 POS = c(234, 266, 117, 134, 180,
                         234, 266, 199, 88, 180),
                 major = c("G", "G", "A", "A", "C",
                           "G", "G", "A", "G", "C"),
                 minor = c("A", "A", "G", "G", "T",
                           "A", "A", "G", "A", "T"),
                 ALT_TYPE = c("minor", "minor", "minor", "minor", "minor",
                              "minor", "minor", "minor", "major", "minor"),
                 minorfreq = c(0.010, 0.022, 0.043, 0.055, 0.011,
                               0.010, 0.022, 0.043, 0.055, 0.011),
                 majorfreq = c(0.990, 0.978, 0.957, 0.945, 0.989,
                               0.990, 0.978, 0.957, 0.945, 0.989)
)

df

snv_location(df)

# Example 2:

snv_location(example_filtered_SNV_df)

</code></pre>

<hr>
<h2 id='snv_segment'>snv_segment</h2><span id='topic+snv_segment'></span>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data) and filtered (filter_variants) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snv_segment(vardf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snv_segment_+3A_vardf">vardf</code></td>
<td>
<p>A rearranged (arrange_data) and filtered (filtered_variants) vcf dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A bar plot showing the number of variants colored by their SNPEff annotation
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: Simple data
df &lt;- data.frame(sample = c("m1", "m1", "m1", "m1", "m1",
                            "m2", "m2", "m2", "m2", "m2"),
  CHROM = c("PB1", "PB1", "PB2", "PB2", "PB2",
	           "PB1", "PB1", "PB2", "PB2", "PB2"),
  annotation = c("downstrean_gene_variant", "synonymous_variant",
		              "synonymous_variant", "stop_gained", "missense_variant",
		          	  "downstrean_gene_variant", "downstrean_gene_variant",
				          "synonymous_variant", "stop_gained", "missense_variant")
)

df

snv_segment(df)

# Example 2: Sample data
snv_segment(example_filtered_SNV_df)

</code></pre>

<hr>
<h2 id='tally_it'>tally_it</h2><span id='topic+tally_it'></span>

<h3>Description</h3>

<p>Groups the input vcf data frame using a list of variables and tallies the number of occurrences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tally_it(df, groupit, new_colname)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tally_it_+3A_df">df</code></td>
<td>
<p>A rearranged vcf dataframe (arrange_data)</p>
</td></tr>
<tr><td><code id="tally_it_+3A_groupit">groupit</code></td>
<td>
<p>A vector containing column names that data should be grouped by</p>
</td></tr>
<tr><td><code id="tally_it_+3A_new_colname">new_colname</code></td>
<td>
<p>The name of the count column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with columns from the 'groupit' vector and the number of times each unique grouping occurs in the data
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Sample dataframe of 7 variants across 2 samples
df &lt;- data.frame(
  sample = c( "sample1", "sample1", "sample1", "sample2",
              "sample2", "sample2", "sample2"),
  CHROM = c("PB1", "PB2", "PB2", "LEO", "LEO", "LEO", "ALE"),
  SegmentSize = c(2280, 2274, 2274, 1701, 1701, 1701, 1888 ),
  minorfreq = c(0.04422785, 0.03738175, 0.01390202, 0.02927786,
                0.03071955, 0.02626025, 0.02875321)
)

# Example 1: to get the sum of variants on every segment:
groupit = c('sample','CHROM', "SegmentSize")
tally_it(df, groupit, "snv_count")

# Example 2: to get the count across genomes:
groupit = c('sample')
tally_it(df, groupit, "snv_count")

</code></pre>

<hr>
<h2 id='tstv_plot'>tstv_plot</h2><span id='topic+tstv_plot'></span>

<h3>Description</h3>

<p>Plots Ts/Tv ratios
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tstv_plot(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tstv_plot_+3A_df">df</code></td>
<td>
<p>TsTv dataframe generated using the tstv_ratio function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>two plots showing the K2P and simple Ts/Tv ratios
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- tstv_ratio(example_filtered_SNV_df,1300)
tstv_plot(df)
</code></pre>

<hr>
<h2 id='tstv_ratio'>tstv_ratio</h2><span id='topic+tstv_ratio'></span>

<h3>Description</h3>

<p>Inputs a filtered and rearranged vcf dataframe and calculates the transition/transversion ratio
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tstv_ratio(df, genome_size)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tstv_ratio_+3A_df">df</code></td>
<td>
<p>The filtered and rearranged variant dataframe</p>
</td></tr>
<tr><td><code id="tstv_ratio_+3A_genome_size">genome_size</code></td>
<td>
<p>Size of whole genome being used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing the calculated transition/transversion ratio (R or basic_tstv)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tstv_ratio(example_filtered_SNV_df, 13000)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
