<!DOCTYPE html><html><head><title>Help for package VLF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {VLF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aa.compare'>
<p>Amino Acid Comparison</p></a></li>
<li><a href='#aa.conservation_first'>
<p>First Modal Amino Acid Conservation</p></a></li>
<li><a href='#aa.conservation_two'>
<p>First and Second Modal Amino Acid Conservation</p></a></li>
<li><a href='#aa.count.function'>
<p>Amino Acid Count</p></a></li>
<li><a href='#aa.find.singles'>
<p>Find amino acid singles</p></a></li>
<li><a href='#aa.frequency.matrix.function'>
<p>Amino Acid Frequency Matrix</p></a></li>
<li><a href='#aa.MODE'>
<p>Amino Acid Modal Sequence</p></a></li>
<li><a href='#aa.MODE.freq'>
<p>Amino Acid Modal Frequencies</p></a></li>
<li><a href='#aa.MODE.second.freq'>
<p>Amino Acid Second Modal Frequency</p></a></li>
<li><a href='#aa.specimen.frequencies'>
<p>Specimen Nucleotide Frequencies</p></a></li>
<li><a href='#aa.VLF.convert.matrix'>
<p>Convert Amino Acid Matrix</p></a></li>
<li><a href='#aa.VLF.count.pos'>
<p>VLF position count</p></a></li>
<li><a href='#aa.VLF.count.spec'>
<p>VLF Specimen Count</p></a></li>
<li><a href='#aa.VLF.reduced'>
<p>Amino Acid Reduced</p></a></li>
<li><a href='#aaVLFs.to.modalchanges'>
<p>Amino Acid Changes</p></a></li>
<li><a href='#aminoAcid.matching.positions'>
<p>Matching Amino Acid Positions</p></a></li>
<li><a href='#aminoAcidFun'><p>Amino Acid VLF Analysis Function</p></a></li>
<li><a href='#birds'>
<p>Bird Nucleotide Sequences</p></a></li>
<li><a href='#birds_aminoAcids'>
<p>Bird Amino Acid Sequences</p></a></li>
<li><a href='#compare'>
<p>Compare VLFs within Species</p></a></li>
<li><a href='#concordanceFun'>
<p>VLF Concordance Check Function</p></a></li>
<li><a href='#concordant.to.modalchanges'>
<p>Concordant Residue Changes</p></a></li>
<li><a href='#conservation_first'>
<p>First Modal Conserved</p></a></li>
<li><a href='#conservation_two'>
<p>First and Second Modal Conserved</p></a></li>
<li><a href='#count.function'>
<p>Nucleotide Count</p></a></li>
<li><a href='#Decile.Plot'>
<p>VLF Decile Plot</p></a></li>
<li><a href='#Error.Rate'>
<p>Error Rate</p></a></li>
<li><a href='#fasta.read'>
<p>Read Fasta Files</p></a></li>
<li><a href='#ffrequency.matrix.function'><p>Frequency Matrix</p></a></li>
<li><a href='#find.matching'>
<p>Find Matching ntVLF and aaVLF Specimen.</p></a></li>
<li><a href='#find.singles'>
<p>Single and Shared VLF Find</p></a></li>
<li><a href='#matched.codon.position'>
<p>Codon Position of Matching aa and ntVLFs</p></a></li>
<li><a href='#MODE'>
<p>Modal Sequence</p></a></li>
<li><a href='#MODE.freq'>
<p>Modal Frequencies</p></a></li>
<li><a href='#MODE.second.freq'>
<p>Second Modal Frequency</p></a></li>
<li><a href='#nucleotide.matching.positions'>
<p>Matching Nucleotide Positions</p></a></li>
<li><a href='#overall.matched'>
<p>Final Matching</p></a></li>
<li><a href='#separate'>
<p>Separate Specimen by Species Names</p></a></li>
<li><a href='#Sliding.Window'>
<p>Sliding Window</p></a></li>
<li><a href='#specimen.frequencies'>
<p>Specimen Nucleotide Frequencies</p></a></li>
<li><a href='#VLF-package'>
<p>Frequency Matrix Approach for Assessing Very Low Frequency Variants in Sequence Records</p></a></li>
<li><a href='#VLF.aminoAcids'>
<p>Amino Acid VLFs</p></a></li>
<li><a href='#VLF.convert.matrix'>
<p>VLF Matrix Convert</p></a></li>
<li><a href='#VLF.count.pos'>
<p>VLF Count for Sequence Positions</p></a></li>
<li><a href='#VLF.count.spec'>
<p>VLF Count for Specimens</p></a></li>
<li><a href='#VLF.nucleotides'>
<p>Nucleotide VLF Convert</p></a></li>
<li><a href='#VLF.reduced'>
<p>Reduced VLF Matrix</p></a></li>
<li><a href='#vlfFun'>
<p>Nucleotide VLF Assessment Function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Frequency Matrix Approach for Assessing Very Low Frequency
Variants in Sequence Records</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-08-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Taryn B. T. Athey [aut, cre], Paul D. McNicholas [aut, cre], Jarrett D. Phillips [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Taryn B. T. Athey &lt;taryn.athey@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Using frequency matrices, very low frequency variants (VLFs) are assessed for amino acid and nucleotide sequences.  The VLFs are then compared to see if they occur in only one member of a species, singleton VLFs, or if they occur in multiple members of a species, shared VLFs.  The amino acid and nucleotide VLFs are then compared to see if they are concordant with one another.  Amino acid VLFs are also assessed to determine if they lead to a change in amino acid residue type, and potential changes to protein structures. Based on Stoeckle and Kerr (2012) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0043992">doi:10.1371/journal.pone.0043992</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-17 21:23:08 UTC; jarrettphillips</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-18 12:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='aa.compare'>
Amino Acid Comparison
</h2><span id='topic+aa.compare'></span>

<h3>Description</h3>

<p>Compares amino acid very low frequency variants between specimen of the same species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.compare(x, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.compare_+3A_x">x</code></td>
<td>

<p>A list of amino acid sequences separated by species name.
</p>
</td></tr>
<tr><td><code id="aa.compare_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequences within the list.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument x can be calculated using the separate function.
</p>


<h3>Value</h3>

<p>A matrix containing two vectors, one with singleton VLF counts for each position of the sequence, and one with shared VLF counts for each position of the sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B.T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.conservation_first'>
First Modal Amino Acid Conservation
</h2><span id='topic+aa.conservation_first'></span>

<h3>Description</h3>

<p>Calculates the conservation of the first amino acids that occur most often in a matrix of amino acid sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.conservation_first(modal, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.conservation_first_+3A_modal">modal</code></td>
<td>

<p>A vector of the frequencies for the amino acids in the first modal sequence.
</p>
</td></tr>
<tr><td><code id="aa.conservation_first_+3A_p">p</code></td>
<td>

<p>A conservation value for the amino acid frequencies to be compared to
</p>
</td></tr>
<tr><td><code id="aa.conservation_first_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the amino acid sequence
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The item modal can be calculated using the aa.MODE.freq function.
</p>


<h3>Value</h3>

<p>A vector that contains how many amino acids from the first modal sequence are conserved at the specified conservation level.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
aminoAcid_firstModalFreq &lt;- aa.MODE.freq(aminoAcid_frequency.Matrix, 216)
aminoAcid_firstConservation_100 &lt;- aa.conservation_first(aminoAcid_firstModalFreq, 1, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.conservation_two'>
First and Second Modal Amino Acid Conservation</h2><span id='topic+aa.conservation_two'></span>

<h3>Description</h3>

<p>Calculates the conservation of the amino acids that occur first and second most often in a matrix of sequences</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.conservation_two(modal1, modal2, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.conservation_two_+3A_modal1">modal1</code></td>
<td>

<p>A vector of the frequencies for the amino acids in the first modal sequence.
</p>
</td></tr>
<tr><td><code id="aa.conservation_two_+3A_modal2">modal2</code></td>
<td>

<p>A vector of the frequencies for the amino acids in the second modal sequence
</p>
</td></tr>
<tr><td><code id="aa.conservation_two_+3A_p">p</code></td>
<td>

<p>A conservation value for the amino acid frequencies to be compared to.
</p>
</td></tr>
<tr><td><code id="aa.conservation_two_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument modal1 can be calculated using the aa.MODE.freq function, and the argument modal2 can be calculated using he aa.MODE.second.freq function.</p>


<h3>Value</h3>

<p>A vector that contains how many amino acids from the first and second modal sequences are conserved at the specified conservation level.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
aminoAcid_firstModalFreq &lt;- aa.MODE.freq(aminoAcid_frequency.Matrix, 216)
aminoAcid_secondModalFreq &lt;- aa.MODE.second.freq(aminoAcid_frequency.Matrix, 216)
aminoAcid_secondConservation_99.9 &lt;- aa.conservation_two(aminoAcid_firstModalFreq, 
    aminoAcid_secondModalFreq, 0.999, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.count.function'>
Amino Acid Count
</h2><span id='topic+aa.count.function'></span>

<h3>Description</h3>

<p>Counts the number of each amino acid in each positino of the barcode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.count.function(aminoAcids, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.count.function_+3A_aminoacids">aminoAcids</code></td>
<td>

<p>A matrix of barcode amino acid sequences.
</p>
</td></tr>
<tr><td><code id="aa.count.function_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the amino acid sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first and second column of the aminoAcid argument must contain the unique specimen identifier and the species name, respectively, followed by the amino acid sequence.</p>


<h3>Value</h3>

<p>A matrix containing the number of each amino acid in each position of the sequence.  Each row is a different amino acid count, while the columns represent the sequence position.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.find.singles'>
Find amino acid singles</h2><span id='topic+aa.find.singles'></span>

<h3>Description</h3>

<p>Determines the number of shared and singleton amino acid VLFs.</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.find.singles(aaSpecies, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.find.singles_+3A_aaspecies">aaSpecies</code></td>
<td>

<p>List of amino acid sequences separated by species name.</p>
</td></tr>
<tr><td><code id="aa.find.singles_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of amino acid sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument aaSpecies contains only amino acid VLFs, and NAs in any other position in the sequence.  The list can be created using the separate function.</p>


<h3>Value</h3>

<p>A matrix containing the number of singleton and shared aaVLFs in each position of the barcode.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.frequency.matrix.function'>
Amino Acid Frequency Matrix</h2><span id='topic+aa.frequency.matrix.function'></span>

<h3>Description</h3>

<p>Calculates the frequency of each amino acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.frequency.matrix.function(aa.count, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.frequency.matrix.function_+3A_aa.count">aa.count</code></td>
<td>

<p>A matrix containing the number of each amino acid in each position.
</p>
</td></tr>
<tr><td><code id="aa.frequency.matrix.function_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequence
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The aa.count argument can be calculated using the function aa.count.function
</p>


<h3>Value</h3>

<p>A matrix of the frequencies for each amino acid in each position of the barcode sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.MODE'>
Amino Acid Modal Sequence
</h2><span id='topic+aa.MODE'></span>

<h3>Description</h3>

<p>Calculates the amino acid sequence that occurs most often in a matrix of amino acid sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.MODE(freq.matrix, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.MODE_+3A_freq.matrix">freq.matrix</code></td>
<td>

<p>Freuqncy matrix for amino acids.
</p>
</td></tr>
<tr><td><code id="aa.MODE_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of amino acid sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq.matrix can be calculated using the function aa.frequency.matrix.function</p>


<h3>Value</h3>

<p>A vector containing the amino acid sequence that occurs most often.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.MODE.freq'>
Amino Acid Modal Frequencies
</h2><span id='topic+aa.MODE.freq'></span>

<h3>Description</h3>

<p>Returns the frequencies of the amino acids that occur most often in each position of the sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.MODE.freq(freq.matrix, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.MODE.freq_+3A_freq.matrix">freq.matrix</code></td>
<td>

<p>Frequency matrix for amino acids.
</p>
</td></tr>
<tr><td><code id="aa.MODE.freq_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the amino acid sequences
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq.matrix can be calculated using the function aa.frequency.matrix.function
</p>


<h3>Value</h3>

<p>A vector of frequencies for the first modal sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
aminoAcid_firstModalFreq &lt;- aa.MODE.freq(aminoAcid_frequency.Matrix, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.MODE.second.freq'>
Amino Acid Second Modal Frequency
</h2><span id='topic+aa.MODE.second.freq'></span>

<h3>Description</h3>

<p>Returns the frequencies of the amino acids that occur second most often in each position of a matrix of amino acid sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.MODE.second.freq(freq.matrix, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.MODE.second.freq_+3A_freq.matrix">freq.matrix</code></td>
<td>

<p>Frequency matrix for amino acids.
</p>
</td></tr>
<tr><td><code id="aa.MODE.second.freq_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of amino acid sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq.matrix can be calculated using the function aa.frequency.matrix.function
</p>


<h3>Value</h3>

<p>A vector containing the frequencies of the second modal amino acid sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
aminoAcid_secondModalFreq &lt;- aa.MODE.second.freq(aminoAcid_frequency.Matrix, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.specimen.frequencies'>
Specimen Nucleotide Frequencies
</h2><span id='topic+aa.specimen.frequencies'></span>

<h3>Description</h3>

<p>Converts a matrix of amino acid sequences into a matrix of amino acid frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.specimen.frequencies(freq, seq.matrix, spec.names, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.specimen.frequencies_+3A_freq">freq</code></td>
<td>

<p>Frequency matrix for amino acids.
</p>
</td></tr>
<tr><td><code id="aa.specimen.frequencies_+3A_seq.matrix">seq.matrix</code></td>
<td>

<p>Matrix of specimen amino acid sequences.
</p>
</td></tr>
<tr><td><code id="aa.specimen.frequencies_+3A_spec.names">spec.names</code></td>
<td>

<p>A vector of the species names for each specimen in aminoAcids in the ordfer they appear in the matrix.
</p>
</td></tr>
<tr><td><code id="aa.specimen.frequencies_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of amino acid sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function aa.frequency.matrix.function.
</p>


<h3>Value</h3>

<p>A matrix containing the frequencies of each amino acid in the sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids,216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count,216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.VLF.convert.matrix'>
Convert Amino Acid Matrix
</h2><span id='topic+aa.VLF.convert.matrix'></span>

<h3>Description</h3>

<p>Converts a matrix of amino acid frequencies for each specimen into a matrix consisting of only VLF values and NAs in every non-VLF position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.VLF.convert.matrix(seq.matrix, freq, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.VLF.convert.matrix_+3A_seq.matrix">seq.matrix</code></td>
<td>

<p>A matrix of aligned DNA barcode amino acid sequences.
</p>
</td></tr>
<tr><td><code id="aa.VLF.convert.matrix_+3A_freq">freq</code></td>
<td>

<p>A matrix of amino acid frequencies for each specimen.
</p>
</td></tr>
<tr><td><code id="aa.VLF.convert.matrix_+3A_p">p</code></td>
<td>

<p>A very low frequency variant cut-off frequency.  Any frequency in the freq matrix below this value is considered to be a very low frequency variant.
</p>
</td></tr>
<tr><td><code id="aa.VLF.convert.matrix_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequences.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of VLF amino acid frequencies, containing only those nucleotide frequencies that occur less than the designation p value, and NAs in each other position of the matrix.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.VLF.count.pos'>
VLF position count
</h2><span id='topic+aa.VLF.count.pos'></span>

<h3>Description</h3>

<p>Calculates the number of very low frequency variants in each position in a matrix of sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.VLF.count.pos(freq, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.VLF.count.pos_+3A_freq">freq</code></td>
<td>

<p>A matrix of frequencies for each speicmen.
</p>
</td></tr>
<tr><td><code id="aa.VLF.count.pos_+3A_p">p</code></td>
<td>

<p>A very low frequency variant cut off frequency.  Any frequency in the freq matrix below this value is considered to be a very low frequency variant.
</p>
</td></tr>
<tr><td><code id="aa.VLF.count.pos_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequences.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the amino acid VLF count for each position of the sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
birds_aminoAcid_positionVLFcount &lt;- aa.VLF.count.pos(bird_aminoAcid_frequencies, 0.001, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.VLF.count.spec'>
VLF Specimen Count
</h2><span id='topic+aa.VLF.count.spec'></span>

<h3>Description</h3>

<p>Calculates the number of very low frequency variants for each specimen in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.VLF.count.spec(freq, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.VLF.count.spec_+3A_freq">freq</code></td>
<td>

<p>A matrix of amino acid frequencies for each specimen.
</p>
</td></tr>
<tr><td><code id="aa.VLF.count.spec_+3A_p">p</code></td>
<td>

<p>A very low frequency variant cut-off frequency.  Any frequency in the freq matrix below this value is considered to be a very low frequency variant.
</p>
</td></tr>
<tr><td><code id="aa.VLF.count.spec_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequences.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the aaVLF count for every specimen.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aa.VLF.reduced'>
Amino Acid Reduced
</h2><span id='topic+aa.VLF.reduced'></span>

<h3>Description</h3>

<p>Reduces a matrix of amino acid very low frequency variants (aaVLFs) so that only those specimen that contain aaVLFs remain
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aa.VLF.reduced(NA.matrix, sCount, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aa.VLF.reduced_+3A_na.matrix">NA.matrix</code></td>
<td>

<p>A matrix with values for amino acid very low frequency variants for each specimen and NAs in all other positions.
</p>
</td></tr>
<tr><td><code id="aa.VLF.reduced_+3A_scount">sCount</code></td>
<td>

<p>A vector for amino acid very low frequency variant (VLF) counts for each specimen in the NA.matrix.</p>
</td></tr>
<tr><td><code id="aa.VLF.reduced_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the amino acid sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument NA.matrix can be calculated using aa.VLF.convert.matrix and VLF.aminoAcids, the sCount argument can be calculaed using the aa.VLF.count.spec function.
</p>


<h3>Value</h3>

<p>A matrix containing only specimen with aaVLFs, and only the aaVLF values in the sequences.  All other positions of the sequence contain NAs.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aaVLFs.to.modalchanges'>
Amino Acid Changes
</h2><span id='topic+aaVLFs.to.modalchanges'></span>

<h3>Description</h3>

<p>Determines how many aaVLFs have changed &ldquo;type&rdquo; of amino acid from the modal amino acid sequence.  Amino acid types are polar charged, polar uncharged, non-polar, and those with a unique side group.</p>


<h3>Usage</h3>

<pre><code class='language-R'>aaVLFs.to.modalchanges(modal, AminoAcidList, aalength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aaVLFs.to.modalchanges_+3A_modal">modal</code></td>
<td>

<p>The modal amino acid sequence (i.e., the amino acid sequence that occurs most often based on the amino acid frequency matrix)</p>
</td></tr>
<tr><td><code id="aaVLFs.to.modalchanges_+3A_aminoacidlist">AminoAcidList</code></td>
<td>

<p>Matrix of VLF amino acid sequences containing only aaVLFs and NAs anywhere else</p>
</td></tr>
<tr><td><code id="aaVLFs.to.modalchanges_+3A_aalength">aalength</code></td>
<td>

<p>Amino Acid sequence length.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument modal can be created using the MODE function.  The argument AminoAcidList can be created using the aa.VLF.convert.matrix, VLF.aminoAcids, and aa.VLF.reduced functions.</p>


<h3>Value</h3>

<p>A sameAll value representative of the number of amino acids that were the same type as the modal, a changedAll value representative of the number of amino acids that changed amino acid type from the modal.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
All_aaType_change &lt;- aaVLFs.to.modalchanges(aminoAcid_Modal, birds_aaVLFreduced, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aminoAcid.matching.positions'>
Matching Amino Acid Positions
</h2><span id='topic+aminoAcid.matching.positions'></span>

<h3>Description</h3>

<p>Gives the position and residue of the amino acid VLFs in a matrix containing amino acid VLFs and NAs.</p>


<h3>Usage</h3>

<pre><code class='language-R'>aminoAcid.matching.positions(matchAA, aalength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aminoAcid.matching.positions_+3A_matchaa">matchAA</code></td>
<td>

<p>A matrix containing aaVLFs and NAs in all other positions of the sequences.</p>
</td></tr>
<tr><td><code id="aminoAcid.matching.positions_+3A_aalength">aalength</code></td>
<td>

<p>Amino acid sequence length.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument matchAA can be calculated usingthe find.matching function and taking the first argument from the returned value.</p>


<h3>Value</h3>

<p>A list for each aaVLF with a matching specimen identifier to a ntVLF.  The first position in each entry of the list contains the specimen identifier, the second position contains the species name, the third position contains the sequence position of the aaVLF, and the fourth position contains the aaVLF.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)

#Amino Acid VLF Analysis
data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)

#Concordance Analysis
VLF_match &lt;- find.matching(bird_VLFreduced, birds_aaVLFreduced, 648, 216)
position_matchingNuc &lt;- nucleotide.matching.positions(VLF_match[[2]], 648)
position_matchingAA &lt;- aminoAcid.matching.positions(VLF_match[[1]], 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='aminoAcidFun'>Amino Acid VLF Analysis Function
</h2><span id='topic+aminoAcidFun'></span>

<h3>Description</h3>

<p>Runs the full amino acid VLF analysis for the user and outputs results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aminoAcidFun(x, p = 0.001, seqlength = 216, own = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aminoAcidFun_+3A_x">x</code></td>
<td>

<p>A matrix of amino acid sequences with the first column containing the unique specimen identifier, the second column containing the species name and the remaining columns containing the amino acid sequence.
</p>
</td></tr>
<tr><td><code id="aminoAcidFun_+3A_p">p</code></td>
<td>

<p>A VLF designation frequency cut-off to be used within the analysis.  By default p = 0.001.
</p>
</td></tr>
<tr><td><code id="aminoAcidFun_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequence.  By default seqlength = 216.
</p>
</td></tr>
<tr><td><code id="aminoAcidFun_+3A_own">own</code></td>
<td>

<p>If the user wants to compare their own sequences separate from reference sequences, then argument can be used.  Similar to x, this argument is a matrix of amino acid sequences with the first column containing the unique specimen identifier, the second column contains the species name and the remaining columns containing the nucleotide sequence.  By default own = NULL.)
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>modal</code></td>
<td>
<p>A vector containing the amino acid sequence that occurs most often in the dataset.</p>
</td></tr>
<tr><td><code>con100</code></td>
<td>
<p>The number of amino acid positions that are 100% conserved in the sequence</p>
</td></tr>
<tr><td><code>conp</code></td>
<td>
<p>The number of amino acid positions that are (1-p)% conserved in the sequence</p>
</td></tr>
<tr><td><code>combine</code></td>
<td>
<p>The number of amino acid positions that are (1-p)% conserved when combining the first and second modal sequences.</p>
</td></tr>
<tr><td><code>specimen</code></td>
<td>
<p>A vector containing the number of VLFs for each specimen in the dataset.</p>
</td></tr>
<tr><td><code>position</code></td>
<td>
<p>A vector containing the number of VLFs for each position in the sequences.</p>
</td></tr>
<tr><td><code>sas</code></td>
<td>
<p>A matrix containing vectors of single and shared amino acid VLF counts for each position of the sequence.</p>
</td></tr>
<tr><td><code>VLFmatrix</code></td>
<td>
<p>A matrix containing only those specimen that have VLFs as well as the amino acid at the positions that contain VLFs and NAs in all other positions.</p>
</td></tr>
<tr><td><code>ownSpecCount</code></td>
<td>
<p>A vector containing the number of VLFs for each specimen in the users own specified dataset.  Only appears if own is not NULL.</p>
</td></tr>
<tr><td><code>ownPosCount</code></td>
<td>
<p>A vector containing the number of VLFs for each position in the sequences of the users own specified dataset.  Only appears if own is not NULL.</p>
</td></tr>
<tr><td><code>ownVLFMatrix</code></td>
<td>
<p>A matrix containing only those amino acids at the positions that contain VLFs and NAs in all other positions of the sequence.  Only appears if own is not NULL.</p>
</td></tr>
<tr><td><code>ownVLFreduced</code></td>
<td>
<p>ownVLFMatrix</p>
</td></tr></table>
<p>A matrix containing only those specimen that have VLFs as well as the amino acids at the positions that contain VLFs and NAs in all other positions of the sequence.  Only appears if own is not NULL.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
bird_aaAnalysis &lt;- aminoAcidFun(birds_aminoAcids)
## End(Not run)
</code></pre>

<hr>
<h2 id='birds'>
Bird Nucleotide Sequences</h2><span id='topic+birds'></span>

<h3>Description</h3>

<p>Data set containing nucleotide sequences for 11,333 bird barcodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(birds)</code></pre>


<h3>Format</h3>

<p>The format is:
chr [1:11333, 1:650] &quot;gi|359280039|gb|JQ173884.1|&quot; ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : NULL
..$ : chr [1:650] &quot;V1&quot; &quot;V2&quot; &quot;V3&quot; &quot;V4&quot; ...
</p>


<h3>Source</h3>

<p>Stoeckle, M. Y. and Kerr, K. C. R. (2012) Frequency Matrix Approach Demonstrates High Sequence Quality in Avian BARCODEs and Highlights Cryptic Pseudogenes. PLoS ONE. 7, e43992.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
</code></pre>

<hr>
<h2 id='birds_aminoAcids'>
Bird Amino Acid Sequences
</h2><span id='topic+birds_aminoAcids'></span>

<h3>Description</h3>

<p>Data set containing amino acid sequences for 11,333 bird barcodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(birds_aminoAcids)</code></pre>


<h3>Format</h3>

<p>The format is:
chr [1:11333, 1:218] &quot;gi|359280039|gb|JQ173884.1|&quot; ...
- attr(*, &quot;dimnames&quot;)=List of 2
..$ : NULL
..$ : chr [1:218] &quot;V1&quot; &quot;V2&quot; &quot;V3&quot; &quot;V4&quot; ...
</p>


<h3>Source</h3>

<p>Stoeckle, M. Y. and Kerr, K. C. R. (2012) Frequency Matrix Approach Demonstrates High Sequence Quality in Avian BARCODEs and Highlights Cryptic Pseudogenes. PLoS ONE. 7, e43992.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
</code></pre>

<hr>
<h2 id='compare'>
Compare VLFs within Species
</h2><span id='topic+compare'></span>

<h3>Description</h3>

<p>Compares VLFs between specimen of the same species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare(x, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_+3A_x">x</code></td>
<td>

<p>A list of sequences separated by species name.  Each entry in the list contains a matrix of sequences from the same species.
</p>
</td></tr>
<tr><td><code id="compare_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>List of sequences by species names, x, can be created using the separate function
</p>


<h3>Value</h3>

<p>A matrix containing two vectors, one with singleton VLF counts for each position of the sequence, and one with shared VLF counts for each position of the sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)

#The compare function is called on from within the find.singles function
birds_singleAndShared &lt;- find.singles(bird_species, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='concordanceFun'>
VLF Concordance Check Function
</h2><span id='topic+concordanceFun'></span>

<h3>Description</h3>

<p>Compares ntVLFs to aaVLFs to see if they are concordant (i.e., if the ntVLF causes the aaVLF).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concordanceFun(nuc, aa, nuclength = 648, aalength = 216, aminoAcid_Modal)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concordanceFun_+3A_nuc">nuc</code></td>
<td>

<p>A matrix of ntVLFs, that contains only those specimen with VLFs, and a sequence with only VLF nucleotides and NAs in all other positions of the nucleotide sequences.
</p>
</td></tr>
<tr><td><code id="concordanceFun_+3A_aa">aa</code></td>
<td>

<p>A matrix of aaVLFs, that contains only those specimen with VLFs, and a sequence with only VLF amino acids and NAs in all other positions of the amino acid sequence.
</p>
</td></tr>
<tr><td><code id="concordanceFun_+3A_nuclength">nuclength</code></td>
<td>

<p>The length of the nucleotide sequence.  By default is 648.
</p>
</td></tr>
<tr><td><code id="concordanceFun_+3A_aalength">aalength</code></td>
<td>

<p>The length of the amino acid sequence.  By default is 216.
</p>
</td></tr>
<tr><td><code id="concordanceFun_+3A_aminoacid_modal">aminoAcid_Modal</code></td>
<td>

<p>The modal amino acid sequence (i.e., the amino acid sequence that occurs most often in the given sequences)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument nuc can be taken from the VLFmatrix output from the vlfFun function.  The argument aa can be taken from the VLFmatrix output from the aminoAcidFun function.  The argument aminoAcid_Modal can be taken from the modal output from the aminoAcidFun function.</p>


<h3>Value</h3>

<table>
<tr><td><code>matched</code></td>
<td>
<p>A list of the concordant ntVLFs and aaVLFs.  Contains the specimen identifier, the species name, the concordant amino acid, the amino acid position, and the concordant amino acid position.  There may be mulitple entries for the same aaVLF if that VLF is concordant to more than one ntVLF.</p>
</td></tr>
<tr><td><code>codons</code></td>
<td>
<p>A vector containing calculations for how many of the concordant amino acids were caused by changes in each of the nucleotide codon positions.</p>
</td></tr>
<tr><td><code>concordantType</code></td>
<td>
<p>Contains information on how many of the concordant aaVLFs had a change in amino acid residue type and how many remained in the same amino acid residue category.</p>
</td></tr>
<tr><td><code>aminoAcidType</code></td>
<td>
<p>Contains information on how many of the aaVLFs had a change in amino acid residue type and how many remained in the same amino acid residue category.</p>
</td></tr>
<tr><td><code>concordNuc</code></td>
<td>
<p>Gives the number of ntVLFs that showed concordance to aaVLFs.</p>
</td></tr>
<tr><td><code>concordAA</code></td>
<td>
<p>Gives the number of aaVLFs that showed concordance to ntVLFs.</p>
</td></tr>
<tr><td><code>sequences</code></td>
<td>
<p>Gives the number of sequences that had both ntVLFs and aaVLFs.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #VLF analysis
data(birds)
bird_vlfAnalysis &lt;- vlfFun(birds)

#Amino Acid analysis
data(birds_aminoAcids)
bird_aaAnalysis &lt;- aminoAcidFun(birds_aminoAcids)

#Concordance analysis
bird_Concordance &lt;- concordanceFun(bird_vlfAnalysis$VLFmatrix, bird_aaAnalysis$VLFmatrix, 648, 216,
    bird_aaAnalysis$modal)
## End(Not run)
</code></pre>

<hr>
<h2 id='concordant.to.modalchanges'>
Concordant Residue Changes
</h2><span id='topic+concordant.to.modalchanges'></span>

<h3>Description</h3>

<p>Deternubes how many concordant aaVLFs have changed type of amino acid from the modal amino acid sequence.  Amino acid residue types are polar charged, polar uncharged, non-polar, and amino acids with a unique side group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concordant.to.modalchanges(matched, modal)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concordant.to.modalchanges_+3A_matched">matched</code></td>
<td>

<p>A list containing the concordant aaVLFs and their properties (e.g., sequence position).
</p>
</td></tr>
<tr><td><code id="concordant.to.modalchanges_+3A_modal">modal</code></td>
<td>

<p>A vector containing the modal amino acid sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The matched argument can be calculated using the overall.matched function.  The modal argument can be calculated using the aa.MODE function.</p>


<h3>Value</h3>

<p>A vector containing the number of concordant aaVLFs that changed amino acid residue type, and the number that contained the same residue type.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)

#Amino Acid VLF Analysis
data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids, 
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)

#Concordance Analysis
VLF_match &lt;- find.matching(bird_VLFreduced, birds_aaVLFreduced, 648, 216)
position_matchingNuc &lt;- nucleotide.matching.positions(VLF_match[[2]], 648)
position_matchingAA &lt;- aminoAcid.matching.positions(VLF_match[[1]], 216)
matching_comparison &lt;- overall.matched(position_matchingNuc, position_matchingAA, 648, 216)
concordant_aaType_change &lt;- concordant.to.modalchanges(matching_comparison, aminoAcid_Modal)
## End(Not run)
</code></pre>

<hr>
<h2 id='conservation_first'>
First Modal Conserved
</h2><span id='topic+conservation_first'></span>

<h3>Description</h3>

<p>Calculates the conservation of the nucleotides that occur most often in a matrix of sequences/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conservation_first(modal, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conservation_first_+3A_modal">modal</code></td>
<td>

<p>A vector of the frequencies of the nucleotides in the first modal sequences.
</p>
</td></tr>
<tr><td><code id="conservation_first_+3A_p">p</code></td>
<td>

<p>A conservation value for the nucleotide frequencies to be compared to.
</p>
</td></tr>
<tr><td><code id="conservation_first_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the nucleotide sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument modal can be calculated using the MODE.freq function.
</p>


<h3>Value</h3>

<p>A vector that contains how many nucleotides from the first modal sequence are conserved at the specified conservation level for each codon position.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
nucleotide.modalSequence &lt;- MODE(frequency.matrix, 648)
Bird_first.modal.frequencies &lt;- MODE.freq(frequency.matrix, 648)
First_conserved_100 &lt;- conservation_first(Bird_first.modal.frequencies, 1, 648)
First_conserved_99.9 &lt;- conservation_first(Bird_first.modal.frequencies, 0.999, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='conservation_two'>
First and Second Modal Conserved</h2><span id='topic+conservation_two'></span>

<h3>Description</h3>

<p>Calculates the conservation of the nucleotides that occur first and second most often in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>conservation_two(modal1, modal2, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="conservation_two_+3A_modal1">modal1</code></td>
<td>

<p>A vector of the frequencies for the nucleotides in the first modal sequence.
</p>
</td></tr>
<tr><td><code id="conservation_two_+3A_modal2">modal2</code></td>
<td>

<p>A vector of the frequencies for the nucleotides that occur second most often.
</p>
</td></tr>
<tr><td><code id="conservation_two_+3A_p">p</code></td>
<td>

<p>A conservation value for the nucleotide frequencies to be compared to.
</p>
</td></tr>
<tr><td><code id="conservation_two_+3A_seqlength">seqlength</code></td>
<td>

<p>The nucleotide sequence length.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument modal1 can be calculated using the function MODE.freq.  The argument modal2 can be calculated using the function MODE.second.freq.
</p>


<h3>Value</h3>

<p>A vector that contains how many nucleotides from the first and second modal sequences are conserved at the specified conservation level for each codon position.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
nucleotide.modalSequence &lt;- MODE(frequency.matrix, 648)
Bird_first.modal.frequencies &lt;- MODE.freq(frequency.matrix, 648)
Bird_second.modal.frequencies &lt;- MODE.second.freq(frequency.matrix, 648)
FirstAndSecond_conserved_99.9 &lt;- conservation_two(Bird_first.modal.frequencies, 
    Bird_second.modal.frequencies, 0.999, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='count.function'>
Nucleotide Count
</h2><span id='topic+count.function'></span>

<h3>Description</h3>

<p>Counts the number of each dNTP in each position of an aligned barcode matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count.function(nucleotides, spec.no, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count.function_+3A_nucleotides">nucleotides</code></td>
<td>

<p>A matrix of aligned DNA barcode sequences.  DNA sequences should start at the third column of the matrix, while the first column contains a unique specimen identifier and the second column contains the species name.
</p>
</td></tr>
<tr><td><code id="count.function_+3A_spec.no">spec.no</code></td>
<td>

<p>The number of specimen/sequences in the nucldeotide matrix.
</p>
</td></tr>
<tr><td><code id="count.function_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the nucleotide sequences.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing the number of each nucleotide in each position of the sequence.  Each row is a different dNTP count, while the columns represent the sequence position.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='Decile.Plot'>
VLF Decile Plot</h2><span id='topic+Decile.Plot'></span>

<h3>Description</h3>

<p>Creates a plot of VLF distributions summed for each decile segment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Decile.Plot(VLF, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Decile.Plot_+3A_vlf">VLF</code></td>
<td>

<p>A list of VLFs in each barcode position.  May be a matrix containing vectors of singleton and shared VLFs, or can be a single vector of total VLFs.
</p>
</td></tr>
<tr><td><code id="Decile.Plot_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the sequence.  Usually 648 for nucleotide sequences and 216 for amino acid sequences.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A barplot containing the sum of the VLFs for each decile barcode segment.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_position_VLFcount &lt;- VLF.count.pos(birdSpec.freq, 0.001, 648)
Decile.Plot(Bird_position_VLFcount, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='Error.Rate'>
Error Rate</h2><span id='topic+Error.Rate'></span>

<h3>Description</h3>

<p>Calculates shared, singleton, and total second codon position error rate in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Error.Rate(single, shared, spec, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Error.Rate_+3A_single">single</code></td>
<td>

<p>A vector of singleton very low frequency variant (VLF) counts for each position in the sequence.</p>
</td></tr>
<tr><td><code id="Error.Rate_+3A_shared">shared</code></td>
<td>

<p>A vector of shared very low frequenct variant (VLF) counts for each position in the sequence.</p>
</td></tr>
<tr><td><code id="Error.Rate_+3A_spec">spec</code></td>
<td>

<p>The number of specimen being considered in the dataset.</p>
</td></tr>
<tr><td><code id="Error.Rate_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the barcode sequence.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The arguments single and shared can be calculated simultaneously using the find.singles function.  The spec argument can be calculated by using the nrow() function on the sequence matrix.</p>


<h3>Value</h3>

<p>A vector containing the single, shared, and total error rate based on the number of second position VLFs.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
Bird_position_VLFcount &lt;- VLF.count.pos(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)
Bird_error &lt;- Error.Rate(birds_singleAndShared[1,], birds_singleAndShared[2,], specimen.Number, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='fasta.read'>
Read Fasta Files</h2><span id='topic+fasta.read'></span>

<h3>Description</h3>

<p>Reads in fasta files and converts into sequence matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasta.read(file, seqlength = 648, pos1 = 1, pos2 = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fasta.read_+3A_file">file</code></td>
<td>

<p>A fasta file to be read in.
</p>
</td></tr>
<tr><td><code id="fasta.read_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of sequence.
</p>
</td></tr>
<tr><td><code id="fasta.read_+3A_pos1">pos1</code></td>
<td>

<p>The position within the fasta title of the unique specimen identifier.  By default pos1 = 1.
</p>
</td></tr>
<tr><td><code id="fasta.read_+3A_pos2">pos2</code></td>
<td>

<p>The position within the fasta title of the species name.  By default pos2 = 3.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of sequences, with the unique specimen identifers in the first column, the species names in the second column, and the sequence starting in the third column.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>

<hr>
<h2 id='ffrequency.matrix.function'>Frequency Matrix</h2><span id='topic+ffrequency.matrix.function'></span>

<h3>Description</h3>

<p>Calculates the frequency of each dNTP in each position of a nucleotide count matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ffrequency.matrix.function(count.matrix, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ffrequency.matrix.function_+3A_count.matrix">count.matrix</code></td>
<td>

<p>A matrix of the counts for each dNTP from a matrix of aligned sequences.
</p>
</td></tr>
<tr><td><code id="ffrequency.matrix.function_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument count.matrix can be calculated using the function count.function.</p>


<h3>Value</h3>

<p>A matrix of the frequencies for each dNTP in each position of the barcode sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='find.matching'>
Find Matching ntVLF and aaVLF Specimen.
</h2><span id='topic+find.matching'></span>

<h3>Description</h3>

<p>Compares a list of aaVLF and ntVLF matrices for common specimen identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.matching(NucleotideList, AminoAcidList, nuclength, aalength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.matching_+3A_nucleotidelist">NucleotideList</code></td>
<td>

<p>Matrix of VLF nucleotide sequences containing the only the nucleotidies that are VLFs and NAs in the other positions of the sequences.
</p>
</td></tr>
<tr><td><code id="find.matching_+3A_aminoacidlist">AminoAcidList</code></td>
<td>

<p>Matrix of VLF amino acid sequences containing only the aaVLFs and NAs in the other positions of the sequences.
</p>
</td></tr>
<tr><td><code id="find.matching_+3A_nuclength">nuclength</code></td>
<td>

<p>Length of the nucleotide sequence (should be 3X the length of the amino acid sequence).
</p>
</td></tr>
<tr><td><code id="find.matching_+3A_aalength">aalength</code></td>
<td>

<p>Length of the amino acid sequence (should be 1/3 the length of the nucleotide sequence).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument NucleotideList can be calculated using the VLF.convert.matrix, VLF.nucleotides, and VLF.reduced functions.  The argument AminoAcidList can be calculated using the aa.VLF.convert.matrix, VLF.aminoAcids, and aa.VLF.reduced functions.</p>


<h3>Value</h3>

<p>A list containing matrices of aaVLFs in the first position and ntVLFs in the second position who have matching specimen identifiers.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)

#Amino Acid VLF Analysis
data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids, 
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)

#Concordance Analysis
VLF_match &lt;- find.matching(bird_VLFreduced, birds_aaVLFreduced, 648, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='find.singles'>
Single and Shared VLF Find</h2><span id='topic+find.singles'></span>

<h3>Description</h3>

<p>Calculates the number of singleton and shared VLFs for each position of the nucleotide, by first seeing if there is only one specimen for a species, and then calling on the compare() function to calculate the number of singleton and shared VLFs for those species with multiple specimen.</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.singles(species, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.singles_+3A_species">species</code></td>
<td>

<p>A list of sequences separated byh species name.  Each entry in the list contains a matrix of sequences from the same species.
</p>
</td></tr>
<tr><td><code id="find.singles_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the nucleotide sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument species can be calculated using the separate function.</p>


<h3>Value</h3>

<p>A matrix containing the number of singleton and shared ntVLFs in each position of the barcode.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='matched.codon.position'>
Codon Position of Matching aa and ntVLFs
</h2><span id='topic+matched.codon.position'></span>

<h3>Description</h3>

<p>Counts which codon positions of ntVLFs lead to the concordant aaVLF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matched.codon.position(matched)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matched.codon.position_+3A_matched">matched</code></td>
<td>

<p>A list of the nucleotide position of concordant ntVLFs and their associated aaVLFs.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument matched can be calculated using the function overall.matched.
</p>


<h3>Value</h3>

<p>A vector containing the number of concordant VLFs caused by ntVLFs in each codon position.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)

#Amino Acid VLF Analysis
data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)

#Concordance Analysis
VLF_match &lt;- find.matching(bird_VLFreduced, birds_aaVLFreduced, 648, 216)
position_matchingNuc &lt;- nucleotide.matching.positions(VLF_match[[2]], 648)
position_matchingAA &lt;- aminoAcid.matching.positions(VLF_match[[1]], 216)
matching_comparison &lt;- overall.matched(position_matchingNuc, position_matchingAA, 648, 216)
matching_codons &lt;- matched.codon.position(matching_comparison)
## End(Not run)
</code></pre>

<hr>
<h2 id='MODE'>
Modal Sequence
</h2><span id='topic+MODE'></span>

<h3>Description</h3>

<p>Calculates the nucleotide sequence that occurs most often in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MODE(freq, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MODE_+3A_freq">freq</code></td>
<td>

<p>Frequenct matrix for nucleotides.
</p>
</td></tr>
<tr><td><code id="MODE_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of nucleotide sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function ffrequency.matrix.function.
</p>


<h3>Value</h3>

<p>A vector containing the first modal sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
nucleotide.modalSequence &lt;- MODE(frequency.matrix, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='MODE.freq'>
Modal Frequencies</h2><span id='topic+MODE.freq'></span>

<h3>Description</h3>

<p>Returns the frequencies of the nucleotides in each position of nucleotide sequence that occurs most often.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MODE.freq(freq, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MODE.freq_+3A_freq">freq</code></td>
<td>

<p>Frequency matrix for nucleotides.
</p>
</td></tr>
<tr><td><code id="MODE.freq_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the nucleotide sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function ffrequency.matrix.function.
</p>


<h3>Value</h3>

<p>A vector of frequencies for the first modal sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
nucleotide.modalSequence &lt;- MODE(frequency.matrix, 648)
Bird_first.modal.frequencies &lt;- MODE.freq(frequency.matrix, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='MODE.second.freq'>
Second Modal Frequency</h2><span id='topic+MODE.second.freq'></span>

<h3>Description</h3>

<p>Calculates the frequencies of the nucleotides that occur second most often in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MODE.second.freq(freq, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MODE.second.freq_+3A_freq">freq</code></td>
<td>

<p>Frequency matrix for nucleotides.
</p>
</td></tr>
<tr><td><code id="MODE.second.freq_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of nucleotide sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function ffrequency.matrix.function.
</p>


<h3>Value</h3>

<p>A vector containing the frequencies of the nucleotide sequence that occurs second most often.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
nucleotide.modalSequence &lt;- MODE(frequency.matrix, 648)
Bird_first.modal.frequencies &lt;- MODE.freq(frequency.matrix, 648)
Bird_second.modal.frequencies &lt;- MODE.second.freq(frequency.matrix, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='nucleotide.matching.positions'>
Matching Nucleotide Positions
</h2><span id='topic+nucleotide.matching.positions'></span>

<h3>Description</h3>

<p>Calculates the position of the VLFs in a matrix contain ntVLFs whose specimen identifiers match identifiers of a matrix containing aaVLFs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nucleotide.matching.positions(matchNuc, nuclength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nucleotide.matching.positions_+3A_matchnuc">matchNuc</code></td>
<td>

<p>A matrix containing only of the nucleotides that are VLFs and NAs in all other positions of the sequences.
</p>
</td></tr>
<tr><td><code id="nucleotide.matching.positions_+3A_nuclength">nuclength</code></td>
<td>

<p>The length of the nucleotide sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument matchNuc can be calculated using the function find.matching.
</p>


<h3>Value</h3>

<p>A list for each ntVLF containing the specimen identifier in the first position of each list entry, the species name in the second position of each list entry, and the position of the ntVLF in the third position of each list entry.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)

#Amino Acid VLF Analysis
data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)

#Concordance Analysis
VLF_match &lt;- find.matching(bird_VLFreduced, birds_aaVLFreduced, 648, 216)
position_matchingNuc &lt;- nucleotide.matching.positions(VLF_match[[2]], 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='overall.matched'>
Final Matching
</h2><span id='topic+overall.matched'></span>

<h3>Description</h3>

<p>Compares the ntVLFs and aaVLFs with the same specimen identifier, and determines which ntVLFs are concordant with aaVLFs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overall.matched(positionNuc, positionAA, nuclength, aalength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overall.matched_+3A_positionnuc">positionNuc</code></td>
<td>

<p>A list containing the names of the specimen and the ntVLF positions for specimens that have both aaVLFs and ntVLFs.
</p>
</td></tr>
<tr><td><code id="overall.matched_+3A_positionaa">positionAA</code></td>
<td>

<p>A list containing the names of the specimen, the aaVLF, and the position of the aaVLF for specimens that have both aaVLFs and ntVLFs.
</p>
</td></tr>
<tr><td><code id="overall.matched_+3A_nuclength">nuclength</code></td>
<td>

<p>The length of the nucleotide sequence (should by 3X the length of the amino acid sequence)
</p>
</td></tr>
<tr><td><code id="overall.matched_+3A_aalength">aalength</code></td>
<td>

<p>The length of the amino acid sequence (should be 1/3 the length of the nucleotide sequence)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument positionNuc can be calculated using the function nucleotide.matching.positions.  The argument positionAA can be calculated using the function aminoAcid.matching.positions.
</p>


<h3>Value</h3>

<p>A list of each ntVLF containing the specimen identifier in the first position of each list entry, the species name in the second position of each list entry, the aaVLF in the third position of each entry, the amino acid position of the aaVLF in the fourth entry, and the codon position of the concordant ntVLF in each position of the entry.  If multiple ntVLFs have concordance with one aaVLF, then that aaVLF may contain multiple entries in the list, one for each ntVLF.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #Nucleotide VLF analysis
data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)

#Amino Acid VLF Analysis
data(birds_aminoAcids)
birds_aa_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aa_freq.Mat &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aa_freq &lt;- aa.specimen.frequencies(aa_freq.Mat, birds_aminoAcids, birds_aa_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aa_freq.Mat, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aa_freq, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aa_freq, 0.001, 216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
birds_aaVLFreduced &lt;- aa.VLF.reduced(birds_aminoAcidVLFs, birds_aminoAcid_specimenVLFcount, 216)
birds_aaSpecies &lt;- separate(birds_aaVLFreduced)
birds_aminoAcid_singleAndShared &lt;- aa.find.singles(birds_aaSpecies, 216)

#Concordance Analysis
VLF_match &lt;- find.matching(bird_VLFreduced, birds_aaVLFreduced, 648, 216)
position_matchingNuc &lt;- nucleotide.matching.positions(VLF_match[[2]], 648)
position_matchingAA &lt;- aminoAcid.matching.positions(VLF_match[[1]], 216)
matching_comparison &lt;- overall.matched(position_matchingNuc, position_matchingAA, 648, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='separate'>
Separate Specimen by Species Names</h2><span id='topic+separate'></span>

<h3>Description</h3>

<p>Separates specimen into lists by species names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>separate(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="separate_+3A_x">x</code></td>
<td>

<p>A matrix of sequences, usually reduced sequences containing only VLFs, where the second position of the matrix contains the species name for the specimen.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the argument x needs to be a reduced matrix, it can be calculated using the function VLF.reduced.
</p>


<h3>Value</h3>

<p>A list containing a matrix of sequences for each species.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
Bird_position_VLFcount &lt;- VLF.count.pos(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
## End(Not run)
</code></pre>

<hr>
<h2 id='Sliding.Window'>
Sliding Window
</h2><span id='topic+Sliding.Window'></span>

<h3>Description</h3>

<p>Creates a sliding window analysis plot for the VLFs in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sliding.Window(VLF, seqlength, n = 30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Sliding.Window_+3A_vlf">VLF</code></td>
<td>

<p>A vector of VLFs per position across the barcocde.  Can be a single vector of all VLFs per positions, or can be a matrix containing singleton and shared VLFs.
</p>
</td></tr>
<tr><td><code id="Sliding.Window_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of the barcode sequence.
</p>
</td></tr>
<tr><td><code id="Sliding.Window_+3A_n">n</code></td>
<td>

<p>The number of positions to average the window across (n = 30 by default).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument VLF can be calculated using the function VLF.count.pos for all VLFs, or find.singles for singleton and shared VLFs.
</p>


<h3>Value</h3>

<p>A sliding window plot for the VLFs in each position of the barcode averaged over n.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_position_VLFcount &lt;- VLF.count.pos(birdSpec.freq, 0.001, 648)
Sliding.Window(Bird_position_VLFcount, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='specimen.frequencies'>
Specimen Nucleotide Frequencies
</h2><span id='topic+specimen.frequencies'></span>

<h3>Description</h3>

<p>Converts a matrix of sequences into a matrix of nucleotide frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>specimen.frequencies(freq, seq.matrix, no.spec, spec.names, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="specimen.frequencies_+3A_freq">freq</code></td>
<td>

<p>Frequency matrix for nucleotides.
</p>
</td></tr>
<tr><td><code id="specimen.frequencies_+3A_seq.matrix">seq.matrix</code></td>
<td>

<p>Matrix of specimen sequences, where the sequence starts in the third position of the matrix and the first and second position contain the unique specimen identifier and the species name, respectively.
</p>
</td></tr>
<tr><td><code id="specimen.frequencies_+3A_no.spec">no.spec</code></td>
<td>

<p>The number of specimen in seq.matrix.
</p>
</td></tr>
<tr><td><code id="specimen.frequencies_+3A_spec.names">spec.names</code></td>
<td>

<p>A vector containing the names of the specimen in the seq.matrix, in the order they appear in the matrix.
</p>
</td></tr>
<tr><td><code id="specimen.frequencies_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the nucleotide sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function ffrequency.matrix.function.  The number of specimen can be calculated by using the nrow() function on seq.matrix.
</p>


<h3>Value</h3>

<p>A matrix containing the unique specimen identifer in the first position, the species name in the second position, and the frequencies for each nucleotide in the sequences starting at the third position.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='VLF-package'>
Frequency Matrix Approach for Assessing Very Low Frequency Variants in Sequence Records
</h2><span id='topic+VLF-package'></span><span id='topic+VLF'></span>

<h3>Description</h3>

<p>Using frequency matrices, very low frequency variants (VLFs) are assessed for amino acid and nucleotide sequences.  The VLFs are then compared to see if they occur in only one member of a species, singleton VLFs, or if they occur in multiple members of a species, shared VLFs.  The amino acid and nucleotide VLFs are then compared to see if they are concordant with one another.  Amino acid VLFs are also assessed to determine if they lead to a change in amino acid residue type, and potential changes to protein structures.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> VLF</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2013-10-25</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=3)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>vlfFun()
aminoAcidFun()
concordanceFun()
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>
<p>Maintainer: Taryn B. T. Athey &lt;tathey@uoguelph.ca&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: #VLF analysis
data(birds)
bird_vlfAnalysis &lt;- vlfFun(birds)

#Amino Acid analysis
data(birds_aminoAcids)
bird_aaAnalysis &lt;- aminoAcidFun(birds_aminoAcids)

#Concordance analysis
nuc_matrix &lt;- bird_vlfAnalysis$VLFmatrix
aa_matrix &lt;- bird_aaAnalysis$VLFmatrix
aa_modal &lt;- bird_aaAnalysis$modal
bird_Concordance &lt;- concordanceFun(nuc_matrix, aa_matrix, 648, 216, aa_modal)
## End(Not run)</code></pre>

<hr>
<h2 id='VLF.aminoAcids'>
Amino Acid VLFs</h2><span id='topic+VLF.aminoAcids'></span>

<h3>Description</h3>

<p>Converts a matrix of amino acid frequencies into a matrix of amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VLF.aminoAcids(convert.matrix, seq.matrix, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VLF.aminoAcids_+3A_convert.matrix">convert.matrix</code></td>
<td>

<p>A matrix consisting of only aaVLF frequencies for each specimen, and NAs in every other position of the sequence.
</p>
</td></tr>
<tr><td><code id="VLF.aminoAcids_+3A_seq.matrix">seq.matrix</code></td>
<td>

<p>A matrix of amino acid sequences.
</p>
</td></tr>
<tr><td><code id="VLF.aminoAcids_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the amino acid sequence.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument convert.matrix can be calculated using the function aa.VLF.convert.matrix</p>


<h3>Value</h3>

<p>A matrix containing only aaVLFs and NAs in every other position of the sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids, 
birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies, 0.001, 
    216)
birds_aminoAcidVLFs &lt;- VLF.aminoAcids(birds_aaVLFconvert, birds_aminoAcids, 216)
## End(Not run)
</code></pre>

<hr>
<h2 id='VLF.convert.matrix'>
VLF Matrix Convert
</h2><span id='topic+VLF.convert.matrix'></span>

<h3>Description</h3>

<p>Converts a matrix of nucleotide frequencies for each specimen into a matrix consisting entirely of very low frequency variant (VLF) frequencies and NAs in each other position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VLF.convert.matrix(seq.matrix, freq, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VLF.convert.matrix_+3A_seq.matrix">seq.matrix</code></td>
<td>

<p>A matrix of aligned DNA barcode sequences.
</p>
</td></tr>
<tr><td><code id="VLF.convert.matrix_+3A_freq">freq</code></td>
<td>

<p>A matrix of nucleotide frequencies for each specimen.
</p>
</td></tr>
<tr><td><code id="VLF.convert.matrix_+3A_p">p</code></td>
<td>

<p>A very low frequency variant designation cut off frequency.  Any frequency in the freq matrix below this value is considered to be a very low frequency variant.
</p>
</td></tr>
<tr><td><code id="VLF.convert.matrix_+3A_seqlength">seqlength</code></td>
<td>

<p>Length of nucleotide sequence.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function specimen.frequencies.
</p>


<h3>Value</h3>

<p>A matrix of VLF nucleotide frequencies, containing only those nucleotide frequencies that occur less than the designation p value, and NAs in each other position of the matrix.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds_aminoAcids)
birds_aminoAcid_speciesNames &lt;- birds_aminoAcids[,2]
aminoAcids_specimenNumber &lt;- nrow(birds_aminoAcids)
birds_aminoAcid_count &lt;- aa.count.function(birds_aminoAcids, 216)
aminoAcid_frequency.Matrix &lt;- aa.frequency.matrix.function(birds_aminoAcid_count, 216)
bird_aminoAcid_frequencies &lt;- aa.specimen.frequencies(aminoAcid_frequency.Matrix, birds_aminoAcids,
    birds_aminoAcid_speciesNames, 216)
aminoAcid_Modal &lt;- aa.MODE(aminoAcid_frequency.Matrix, 216)
birds_aminoAcid_specimenVLFcount &lt;- aa.VLF.count.spec(bird_aminoAcid_frequencies, 0.001, 216)
birds_aaVLFconvert &lt;- aa.VLF.convert.matrix(birds_aminoAcids, bird_aminoAcid_frequencies,0.001,216)
## End(Not run)
</code></pre>

<hr>
<h2 id='VLF.count.pos'>
VLF Count for Sequence Positions
</h2><span id='topic+VLF.count.pos'></span>

<h3>Description</h3>

<p>Calculates the number of very low frequency variants (VLFs) in each position in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VLF.count.pos(freq, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VLF.count.pos_+3A_freq">freq</code></td>
<td>

<p>A matrix of frequencies for each specimen.
</p>
</td></tr>
<tr><td><code id="VLF.count.pos_+3A_p">p</code></td>
<td>

<p>A very low frequency variant designation cut off frequency.  Any frequency in the freq matrix below this value is considered to be a very low frequency variant.
</p>
</td></tr>
<tr><td><code id="VLF.count.pos_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the specimen.frequencies function.
</p>


<h3>Value</h3>

<p>A vector containing the number of VLFs for each position in the sequence.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
Bird_position_VLFcount &lt;- VLF.count.pos(birdSpec.freq, 0.001, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='VLF.count.spec'>
VLF Count for Specimens</h2><span id='topic+VLF.count.spec'></span>

<h3>Description</h3>

<p>Calculates the number of very low frequency variants (VLFs) for each specimen in a matrix of sequence nucleotide frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VLF.count.spec(freq, p, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VLF.count.spec_+3A_freq">freq</code></td>
<td>

<p>A matrix of nucleotide frequencies for each specimen.
</p>
</td></tr>
<tr><td><code id="VLF.count.spec_+3A_p">p</code></td>
<td>

<p>A very low frequency variant designation vut off frequency.  Any frequency in the freq matrix below this value is considered to be a very low frequency variant.
</p>
</td></tr>
<tr><td><code id="VLF.count.spec_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument freq can be calculated using the function specimen.frequencies.
</p>


<h3>Value</h3>

<p>A vector containing the number of VLFs for each specimen in the matrix.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='VLF.nucleotides'>
Nucleotide VLF Convert</h2><span id='topic+VLF.nucleotides'></span>

<h3>Description</h3>

<p>Converts a matrix of nucleotide frequencies for each specimen into a matrix of nucleotides for each specimen.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VLF.nucleotides(convert.matrix, seq.matrix, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VLF.nucleotides_+3A_convert.matrix">convert.matrix</code></td>
<td>

<p>A matrix consisting of only very low frequency cariant frequencies for each specimen, and NAs in all other positions of the sequence.
</p>
</td></tr>
<tr><td><code id="VLF.nucleotides_+3A_seq.matrix">seq.matrix</code></td>
<td>

<p>A matrix of DNA sequences.
</p>
</td></tr>
<tr><td><code id="VLF.nucleotides_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument convert.matrix can be calculated using the function VLF.convert.matrix.
</p>


<h3>Value</h3>

<p>A matrix containing only ntVLFs in each position of the sequences, and NAs in all other positions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='VLF.reduced'>
Reduced VLF Matrix
</h2><span id='topic+VLF.reduced'></span>

<h3>Description</h3>

<p>Reduces a matrix of very low frequency variants (VLFs) so that only those specimen that contain VLFs remain in the matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VLF.reduced(NA.matrix, sCount, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VLF.reduced_+3A_na.matrix">NA.matrix</code></td>
<td>

<p>A matrix with values for very low frequency variants (VLFs) for each specimen and NAs in all other positions of the sequence.
</p>
</td></tr>
<tr><td><code id="VLF.reduced_+3A_scount">sCount</code></td>
<td>

<p>A vector of the very low frequency variant (VLF) counts for each specimen in the NA.matrix.
</p>
</td></tr>
<tr><td><code id="VLF.reduced_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the sequences.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument NA.matrix can be calculated using the function VLF.convert.matrix and VLF.nucleotides.  The argument sCount can be calculated using the function VLF.count.spec.
</p>


<h3>Value</h3>

<p>A matrix containing only those specimen that have VLFs, and only VLFs in their positions in the sequence, all other positions contain NAs.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
## End(Not run)
</code></pre>

<hr>
<h2 id='vlfFun'>
Nucleotide VLF Assessment Function</h2><span id='topic+vlfFun'></span>

<h3>Description</h3>

<p>Runs the full nucleotide VLF analysis for the user and outputs the results. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vlfFun(x, p = 0.001, seqlength = 648, own = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vlfFun_+3A_x">x</code></td>
<td>

<p>A matrix of nucleotide sequences with the first column containing the unique specimen identifier, the second column containing the species name and the remaining columns containing the nucleotide sequence.</p>
</td></tr>
<tr><td><code id="vlfFun_+3A_p">p</code></td>
<td>

<p>A VLF designation frequency cut-off to be used within the analysis.  By default p = 0.001.
</p>
</td></tr>
<tr><td><code id="vlfFun_+3A_seqlength">seqlength</code></td>
<td>

<p>The length of the nucleotide sequence.  By default seqlength = 648.
</p>
</td></tr>
<tr><td><code id="vlfFun_+3A_own">own</code></td>
<td>

<p>If the user wants to compare their own sequences separate from reference sequences, then argument can be used.  Similar to x, this argument is a matrix of nucleotide sequences with the first column containing the unique specimen identifier, the second column contains the species name and the remaining columns containing the nucleotide sequence.  By default own = NULL.)
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>modal</code></td>
<td>
<p>A vector containing the nucleotide sequence that occurs most often in the dataset.</p>
</td></tr>
<tr><td><code>con100</code></td>
<td>
<p>The number of nucleotide positions that are 100% conserved in the sequence, separated by codon position.</p>
</td></tr>
<tr><td><code>conp</code></td>
<td>
<p>The number of nucleotide positions that are (1-p)% conserved in the sequence, separated by codon position.</p>
</td></tr>
<tr><td><code>combine</code></td>
<td>
<p>The number of nucleotide positions that are (1-p)% conserved when combining the first and second modal sequences.</p>
</td></tr>
<tr><td><code>specimen</code></td>
<td>
<p>A vector containing the number of VLFs for each specimen in the dataset.</p>
</td></tr>
<tr><td><code>position</code></td>
<td>
<p>A vector containing the number of VLFs for each position in the sequences.</p>
</td></tr>
<tr><td><code>sas</code></td>
<td>
<p>A matrix containing vectors of single and shared ntVLF counts for each position in the sequences.</p>
</td></tr>
<tr><td><code>VLFmatrix</code></td>
<td>
<p>A matrix containing only those specimen that have VLFs as well as the nucleotides at the positions that contain VLFs and NAs in all other positions of the sequence.</p>
</td></tr>
<tr><td><code>ownSpecCount</code></td>
<td>
<p>A vector containing the number of VLFs for each specimen in the users own specified dataset.  Only appears if own is not NULL.</p>
</td></tr>
<tr><td><code>ownPosCount</code></td>
<td>
<p>A vector containing the number of VLFs for each position in the sequences of the users own specified dataset.  Only appears if own is not NULL.</p>
</td></tr>
<tr><td><code>ownVLFMatrix</code></td>
<td>
<p>A matrix containing only those nucleotides at the positions that contain VLFs and NAs in all other positions of the sequence.  Only appears if own is not NULL.</p>
</td></tr>
<tr><td><code>ownVLFreduced</code></td>
<td>
<p>ownVLFMatrix</p>
</td></tr></table>
<p>A matrix containing only those specimen that have VLFs as well as the nucleotides at the positions that contain VLFs and NAs in all other positions of the sequence.  Only appears if own is not NULL.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: data(birds)
bird_vlfAnalysis &lt;- vlfFun(birds)
## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
