<!DOCTYPE html><html><head><title>Help for package volcano3D</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {volcano3D}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_animation'><p>Add mode bar button to rotate the plot</p></a></li>
<li><a href='#boxplot_trio'><p>Boxplot to compare groups</p></a></li>
<li><a href='#calc_pvals'><p>Calculate one-way test and pairwise tests</p></a></li>
<li><a href='#calc_stats_2x3'><p>Calculate p-values for 2x3-way analysis</p></a></li>
<li><a href='#deseq_2x3'><p>2 x 3 factor DESeq2 analysis</p></a></li>
<li><a href='#deseq_2x3_polar'><p>Convert 2x3 design DESeq2 objects to a volcano3d object</p></a></li>
<li><a href='#deseq_polar'><p>Convert DESeq2 objects to a volcano3d object</p></a></li>
<li><a href='#forest_plot'><p>Forest plot individual gene from 2x3 factor analysis</p></a></li>
<li><a href='#grid-class'><p>An S4 class to define the polar grid coordinates system.</p></a></li>
<li><a href='#polar_coords'><p>Coordinates for Three Way Polar Plot</p></a></li>
<li><a href='#polar_coords_2x3'><p>Coordinates for three way polar plot from 2x3 factor analysis</p></a></li>
<li><a href='#polar_grid'><p>Grid required for 3D volcano plot and 2D radial plots</p></a></li>
<li><a href='#radial_ggplot'><p>'Ggplot' for Three Way Polar Plot</p></a></li>
<li><a href='#radial_plotly'><p>Three-way radial comparison Polar Plot (using plotly)</p></a></li>
<li><a href='#show_grid'><p>Plots grid objects for inspection using plotly</p></a></li>
<li><a href='#significance_subset'><p>Extract a subset population</p></a></li>
<li><a href='#syn_example_meta'><p>PEAC synovial sample data</p></a></li>
<li><a href='#syn_example_rld'><p>PEAC synovial gene expression data</p></a></li>
<li><a href='#volc3d-class'><p>An S4 class to define the polar coordinates.</p></a></li>
<li><a href='#volcano3D'><p>Three-Dimensional Volcano Plot</p></a></li>
<li><a href='#voom_polar'><p>Convert RNA-Seq count data to a volcano3d object using 'limma voom'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>3D Volcano Plots and Polar Plots for Three-Class Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.9</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Katriona Goldmann &lt;katrionagoldmann@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://katrionagoldmann.github.io/volcano3D/index.html">https://katrionagoldmann.github.io/volcano3D/index.html</a>,
<a href="https://github.com/KatrionaGoldmann/volcano3D">https://github.com/KatrionaGoldmann/volcano3D</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/KatrionaGoldmann/volcano3D/issues">https://github.com/KatrionaGoldmann/volcano3D/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Generates interactive plots for analysing and visualising 
    three-class high dimensional data. It is particularly suited to visualising 
    differences in continuous attributes such as gene/protein/biomarker 
    expression levels between three groups. Differential gene/biomarker 
    expression analysis between two classes is typically shown as a volcano 
    plot. However, with three groups this type of visualisation is particularly 
    difficult to interpret. This package generates 3D volcano plots and 3-way 
    polar plots for easier interpretation of three-class data.</td>
</tr>
<tr>
<td>Language:</td>
<td>en-gb</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-17 10:02:31 UTC; kgoldmann</td>
</tr>
<tr>
<td>Imports:</td>
<td>plotly, ggplot2, ggpubr, htmlwidgets, magrittr, methods,
Rfast, rlang, matrixTests</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DESeq2, edgeR, limma, qvalue, SummarizedExperiment, knitr,
rmarkdown, kableExtra, usethis, easylabel</td>
</tr>
<tr>
<td>Author:</td>
<td>Katriona Goldmann <a href="https://orcid.org/0000-0002-9073-6323"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Myles Lewis <a href="https://orcid.org/0000-0001-9365-5345"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-17 11:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_animation'>Add mode bar button to rotate the plot</h2><span id='topic+add_animation'></span>

<h3>Description</h3>

<p>Add mode bar button to rotate the plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_animation(
  p,
  rotate_icon_path = NULL,
  stop_icon_path = NULL,
  rotate_colour = "#c7c7c7",
  stop_colour = "#ff6347",
  scale = "scale(0.4) translate(-4, -4)",
  speed = 320,
  shiny_event_names = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_animation_+3A_p">p</code></td>
<td>
<p>The <code><a href="#topic+volcano3D">volcano3D</a></code> plot</p>
</td></tr>
<tr><td><code id="add_animation_+3A_rotate_icon_path">rotate_icon_path</code></td>
<td>
<p>The svg icon path for rotation. If <code>NULL</code> a play
button is used</p>
</td></tr>
<tr><td><code id="add_animation_+3A_stop_icon_path">stop_icon_path</code></td>
<td>
<p>The svg icon path for stop button. If <code>NULL</code> a pause
button is used</p>
</td></tr>
<tr><td><code id="add_animation_+3A_rotate_colour">rotate_colour</code></td>
<td>
<p>The colour for the rotate button (default=&quot;#c7c7c7&quot;)</p>
</td></tr>
<tr><td><code id="add_animation_+3A_stop_colour">stop_colour</code></td>
<td>
<p>The colour for the stop button (default='#ff6347',
a.k.a 'tomato')</p>
</td></tr>
<tr><td><code id="add_animation_+3A_scale">scale</code></td>
<td>
<p>Scaling for rotation button</p>
</td></tr>
<tr><td><code id="add_animation_+3A_speed">speed</code></td>
<td>
<p>The rotation speed</p>
</td></tr>
<tr><td><code id="add_animation_+3A_shiny_event_names">shiny_event_names</code></td>
<td>
<p>If using shiny, pass in any shiny event names which
should stop rotation when triggered (e.g. shiny_event_names = c('replot'))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a rotating cylindrical 3D plotly plot featuring variables on
a tri-axis radial graph with the -log10(multi-group test p-value) on the
z-axis
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>See Also</h3>

<p><code><a href="#topic+volcano3D">volcano3D</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

p &lt;- volcano3D(syn_polar, 
    label_rows = c("COBL", "TREX2"))
add_animation(p)

</code></pre>

<hr>
<h2 id='boxplot_trio'>Boxplot to compare groups</h2><span id='topic+boxplot_trio'></span>

<h3>Description</h3>

<p>Plots the expression of a specific row in expression to compare the three
groups in a boxplot using either ggplot or plotly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxplot_trio(
  polar,
  value,
  box_colours = c("green3", "blue", "red"),
  test = "polar_pvalue",
  levels_order = NULL,
  my_comparisons = NULL,
  text_size = 10,
  stat_colour = "black",
  stat_size = 3,
  step_increase = 0.05,
  plot_method = "ggplot",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boxplot_trio_+3A_polar">polar</code></td>
<td>
<p>A 'volc3d' object including expression data from groups of
interest. Created by <code><a href="#topic+polar_coords">polar_coords</a></code>.</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_value">value</code></td>
<td>
<p>The column name or number in <code>polar@data</code> to be
analysed</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_box_colours">box_colours</code></td>
<td>
<p>The fill colours for each box assigned in order of
levels_order. Default = c('green3', 'blue', 'red') ).</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_test">test</code></td>
<td>
<p>The statistical test used to compare expression.
Allowed values include: </p>

<ul>
<li> <p><code>polar_pvalue</code> (default) and 'polar_padj' for the pvalues
and adjusted pvalues in the polar object.
</p>
</li>
<li> <p><code>polar_multi_pvalue</code> and <code>polar_multi_padj</code> for the pvalues
and adjusted pvalues across all groups using the
<code>polar@multi_group_test </code> columns.
</p>
</li>
<li> <p><code><a href="stats.html#topic+t.test">t.test</a></code> (parametric) and
<code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code> (non-parametric). Perform comparison
between groups of samples.
</p>
</li>
<li> <p><code><a href="stats.html#topic+anova">anova</a></code> (parametric) and
<code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> (non-parametric). Perform one-way ANOVA
test comparing multiple groups. </p>
</li></ul>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_levels_order">levels_order</code></td>
<td>
<p>A character vector stating the contrast groups to be
plotted, in order. If <code>NULL</code> this defaults to the levels in
<code>polar@outcome</code>.</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_my_comparisons">my_comparisons</code></td>
<td>
<p>A list of contrasts to pass to
<code><a href="ggpubr.html#topic+stat_compare_means">stat_compare_means</a></code>. If <code>NULL</code> (default) all contrast
pvalues are calculated and plotted.</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_text_size">text_size</code></td>
<td>
<p>The font size of text (default = 10)</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_stat_colour">stat_colour</code></td>
<td>
<p>Colour to print statistics (default=&quot;black&quot;).</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_stat_size">stat_size</code></td>
<td>
<p>The font size of statistical parameter (default = 3).</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_step_increase">step_increase</code></td>
<td>
<p>The distance between statistics on the y-axis
(default = 0.05).</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_plot_method">plot_method</code></td>
<td>
<p>Whether to use 'plotly' or 'ggplot'. Default is 'ggplot'</p>
</td></tr>
<tr><td><code id="boxplot_trio_+3A_...">...</code></td>
<td>
<p>Other parameters for <code><a href="ggpubr.html#topic+stat_compare_means">stat_compare_means</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a boxplot featuring the differential expression
between groups in comparison with annotated pvalues.
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

boxplot_trio(syn_polar, value = "COBL", plot_method="plotly")
boxplot_trio(syn_polar, value = "COBL")
</code></pre>

<hr>
<h2 id='calc_pvals'>Calculate one-way test and pairwise tests</h2><span id='topic+calc_pvals'></span>

<h3>Description</h3>

<p>Internal function for calculating 3-class group test (either one-way ANOVA or
Kruskal-Wallis test) and pairwise tests (either t-test or Wilcoxon test) on
multi-column data against an outcome parameter with 3 levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_pvals(
  outcome,
  data,
  pcutoff = 0.05,
  padj.method = "BH",
  group_test = c("anova", "kruskal.test"),
  pairwise_test = c("t.test", "wilcoxon"),
  exact = FALSE,
  filter_pairwise = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_pvals_+3A_outcome">outcome</code></td>
<td>
<p>Outcome vector with 3 groups, ideally as a factor. If it is
not a factor, this will be coerced to a factor. This must have exactly 3
levels.</p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_data">data</code></td>
<td>
<p>Dataframe or matrix with variables in columns</p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_padj.method">padj.method</code></td>
<td>
<p>Can be any method available in <code>p.adjust</code> or <code>"qvalue"</code>.
The option &quot;none&quot; is a pass-through.</p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_group_test">group_test</code></td>
<td>
<p>Specifies statistical test for 3-class group comparison.
&quot;anova&quot; means one-way ANOVA, &quot;kruskal.test&quot; means Kruskal-Wallis test.</p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_pairwise_test">pairwise_test</code></td>
<td>
<p>Specifies statistical test for pairwise comparisons</p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_exact">exact</code></td>
<td>
<p>Logical which is only used with <code>pairwise_test = "wilcoxon"</code></p>
</td></tr>
<tr><td><code id="calc_pvals_+3A_filter_pairwise">filter_pairwise</code></td>
<td>
<p>Logical. If <code>TRUE</code> (the default) p-value adjustment on
pairwise statistical tests is only conducted on attributes which reached
the threshold for significance after p-value adjustment on the group
statistical test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with first element representing a data frame of
unadjusted p-values and the second element adjusted p-values. Each dataframe
contains 4 columns: the first column is the 3-way comparison (LRT or ANOVA).
Columns 2-4 are pairwise comparisons between groups A vs B, A vs C and
B vs C, where A, B, C are the 3 levels in the outcome factor.
</p>

<hr>
<h2 id='calc_stats_2x3'>Calculate p-values for 2x3-way analysis</h2><span id='topic+calc_stats_2x3'></span>

<h3>Description</h3>

<p>Calculate p-values for 2x3-way analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_stats_2x3(
  data,
  outcome,
  group,
  padj.method,
  test = c("t.test", "wilcoxon"),
  exact = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_stats_2x3_+3A_data">data</code></td>
<td>
<p>Dataframe or matrix with variables in columns and samples in rows</p>
</td></tr>
<tr><td><code id="calc_stats_2x3_+3A_outcome">outcome</code></td>
<td>
<p>Factor with 2 levels</p>
</td></tr>
<tr><td><code id="calc_stats_2x3_+3A_group">group</code></td>
<td>
<p>Factor with 3 levels</p>
</td></tr>
<tr><td><code id="calc_stats_2x3_+3A_padj.method">padj.method</code></td>
<td>
<p>Can be <code>"qvalue"</code> or any method available in <code>p.adjust</code>.
The option <code>"none"</code> is a pass-through.</p>
</td></tr>
<tr><td><code id="calc_stats_2x3_+3A_test">test</code></td>
<td>
<p>Character value specifying the statistical test between the 2
level response outcome. Current options are &quot;t.test&quot; or &quot;wilcoxon&quot;.</p>
</td></tr>
<tr><td><code id="calc_stats_2x3_+3A_exact">exact</code></td>
<td>
<p>Logical for whether to use an exact test (Wilcoxon test only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing a data frame with summary statistics for the
comparisons between the outcome, for each group level.
</p>

<hr>
<h2 id='deseq_2x3'>2 x 3 factor DESeq2 analysis</h2><span id='topic+deseq_2x3'></span>

<h3>Description</h3>

<p>Experimental function for performing 2x3 factor DESeq2 analyses. Output can
be passed to <code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar()</a></code> and subsequently plotted. Example usage
would include comparing gene expression against a binary outcome e.g.
response vs non-response, across 3 drugs: the design would be <code>~ response</code>
and <code>group</code> would refer to the medication column in the metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deseq_2x3(object, design, group, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deseq_2x3_+3A_object">object</code></td>
<td>
<p>An object of class 'DESeqDataSet' containing full dataset</p>
</td></tr>
<tr><td><code id="deseq_2x3_+3A_design">design</code></td>
<td>
<p>Design formula. The main contrast is taken from the last term
of the formula and must be a binary factor.</p>
</td></tr>
<tr><td><code id="deseq_2x3_+3A_group">group</code></td>
<td>
<p>Character value for the column with the 3-way grouping factor
within the sample information data <code>colData</code></p>
</td></tr>
<tr><td><code id="deseq_2x3_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code>DESeq()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of 3 DESeq2 results objects which can be passed onto
<code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Basic DESeq2 set up
library(DESeq2)

counts &lt;- matrix(rnbinom(n=3000, mu=100, size=1/0.5), ncol=30)
rownames(counts) &lt;- paste0("gene", 1:100)
cond &lt;- rep(factor(rep(1:3, each=5), labels = c('A', 'B', 'C')), 2)
resp &lt;- factor(rep(1:2, each=15), labels = c('non.responder', 'responder'))
metadata &lt;- data.frame(drug = cond, response = resp)

# Full dataset object construction
dds &lt;- DESeqDataSetFromMatrix(counts, metadata, ~response)

# Perform 3x DESeq2 analyses comparing binary response for each drug
res &lt;- deseq_2x3(dds, ~response, "drug")

# Generate polar object
obj &lt;- deseq_2x3_polar(res)

# 2d plot
radial_plotly(obj)

# 3d plot
volcano3D(obj)


</code></pre>

<hr>
<h2 id='deseq_2x3_polar'>Convert 2x3 design DESeq2 objects to a volcano3d object</h2><span id='topic+deseq_2x3_polar'></span>

<h3>Description</h3>

<p>This function is used instead of <code><a href="#topic+polar_coords">polar_coords</a></code> if you have raw
RNA-Seq count data. It takes results from a 2x3 design DESeq2 analysis using
<code><a href="#topic+deseq_2x3">deseq_2x3()</a></code>. Alternatively it will accept a list of 3 <code>DESeqDataSet</code> or
<code>DESeqResults</code> objects each with the same type of binary comparison across 3
groups. Statistical results are extracted and converted to a 'volc3d' object,
which can be directly plotted. Example usage would include comparing gene
expression against a binary outcome e.g. response vs non-response, across 3
drugs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deseq_2x3_polar(
  object,
  pcutoff = 0.05,
  fc_cutoff = NULL,
  padj.method = "BH",
  process = c("positive", "negative", "two.sided"),
  scheme = c("grey60", "red", "gold2", "green3", "cyan", "blue", "purple", "black"),
  labs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deseq_2x3_polar_+3A_object">object</code></td>
<td>
<p>A named list of 3 objects of class 'DESeqDataSet', or a list of
3 DESeq2 results dataframes generated by calling <code>DESeq2::results()</code>. Each
object should contain the binary outcome comparison for each of the 3
groups, e.g. group A response vs non-response, group B response vs
non-response, group C etc. The names of the groups are taken from the list
names.</p>
</td></tr>
<tr><td><code id="deseq_2x3_polar_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td></tr>
<tr><td><code id="deseq_2x3_polar_+3A_fc_cutoff">fc_cutoff</code></td>
<td>
<p>Cut-off for fold change on radial axis</p>
</td></tr>
<tr><td><code id="deseq_2x3_polar_+3A_padj.method">padj.method</code></td>
<td>
<p>Can be <code>"qvalue"</code> or any method available in <code>p.adjust</code>.
The option <code>"none"</code> is a pass-through.</p>
</td></tr>
<tr><td><code id="deseq_2x3_polar_+3A_process">process</code></td>
<td>
<p>Character value specifying colour process for statistical
significant genes: &quot;positive&quot; specifies genes are coloured if fold change
is &gt;0; &quot;negative&quot; for genes with fold change &lt;0 (note that for clarity the
polar position is altered so that genes along each axis have the most
strongly negative fold change values); or &quot;two.sided&quot; which is a compromise
in which positive genes are labelled as before but genes with negative fold
changes and significant p-values have an inverted colour scheme.</p>
</td></tr>
<tr><td><code id="deseq_2x3_polar_+3A_scheme">scheme</code></td>
<td>
<p>Vector of colours starting with non-significant genes/variables</p>
</td></tr>
<tr><td><code id="deseq_2x3_polar_+3A_labs">labs</code></td>
<td>
<p>Optional character vector for labelling groups. Default <code>NULL</code>
leads to abbreviated labels based on levels in <code>outcome</code> using
<code><a href="base.html#topic+abbreviate">abbreviate()</a></code>. A vector of length 3 with custom abbreviated names for the
outcome levels can be supplied. Otherwise a vector length 8 is expected, of
the form &quot;ns&quot;, &quot;A+&quot;, &quot;A+B+&quot;, &quot;B+&quot;, &quot;B+C+&quot;, &quot;C+&quot;, &quot;A+C+&quot;, &quot;A+B+C+&quot;, where
&quot;ns&quot; means non-significant and A, B, C refer to levels 1, 2, 3 in
<code>outcome</code>, and must be in the correct order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a 'volc3d' class object for visualising a 2x3 way
analysis for RNA-Seq data. For usual workflow it is typically preceded by a
call to <code><a href="#topic+deseq_2x3">deseq_2x3()</a></code> which runs the 3x DESeq2 analyses required.
</p>
<p>Scaled polar coordinates are based on the DESeq2 statistic for each group
comparison. Unscaled polar coordinates are generated using the log2 fold
change for each group comparison.
</p>
<p>The z axis for 3d volcano plots does not have as clear a corollary in 2x3
analysis as for the standard 3-way analysis (which uses the likelihood ratio
test for the 3 groups). For 2x3 polar analysis the smallest p-value from the
3 group pairwise comparisons for each gene is used to generate a z coordinate
as -log10(p-value).
</p>
<p>The colour scheme is not as straightforward as for the standard polar plot
and volcano3D plot since genes (or attributes) can be significantly up or
downregulated in the response comparison for each of the 3 groups.
<code>process = "positive"</code> means that genes are labelled with colours if a gene
is significantly upregulated in the response for that group. This uses the
primary colours (RGB) so that if a gene is upregulated in both red and blue
groups it becomes purple etc with secondary colours. If the gene is
upregulated in all 3 groups it is labelled black. Non-significant genes are
in grey.
</p>
<p>With <code>process = "negative"</code> genes are coloured when they are significantly
downregulated. With <code>process = "two.sided"</code> the colour scheme means that both
significantly up- and down-regulated genes are coloured with downregulated
genes labelled with inverted colours (i.e. cyan is the inverse of red etc).
However, significant upregulation in a group takes precedence.
</p>


<h3>Value</h3>

<p>Returns an S4 'volc3d' object containing:
</p>

<ul>
<li><p>'df' A list of 2 dataframes. Each dataframe contains both x,y,z
coordinates as well as polar coordinates r, angle. The first dataframe has
coordinates based on the DESeq2 statistic. The 2nd dataframe is unscaled
and represents log2 fold change for gene expression. The <code>type</code> argument in
<code><a href="#topic+volcano3D">volcano3D</a></code>, <code><a href="#topic+radial_plotly">radial_plotly</a></code> and
<code><a href="#topic+radial_ggplot">radial_ggplot</a></code> corresponds to these dataframes.
</p>
</li>
<li><p>'outcome' An empty factor whose levels are the three-group contrast
factor for comparisons
</p>
</li>
<li><p>'data' Empty dataframe for compatibility
</p>
</li>
<li><p>'pvals' A dataframe containing p-values. Columns 1-3 are pairwise
comparisons between the outcome factor for groups A, B, C respectively.
</p>
</li>
<li><p>'padj' A dataframe containing p-values adjusted for multiple testing
</p>
</li>
<li><p>'pcutoff Numeric value for cut-off for p-value significance
</p>
</li>
<li><p>'scheme' Character vector with colour scheme for plotting
</p>
</li>
<li><p>'labs' Character vector with labels for colour groups
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+deseq_2x3">deseq_2x3()</a></code>
</p>

<hr>
<h2 id='deseq_polar'>Convert DESeq2 objects to a volcano3d object</h2><span id='topic+deseq_polar'></span>

<h3>Description</h3>

<p>This function is used instead of <code><a href="#topic+polar_coords">polar_coords</a></code> if you have raw
RNA-Seq count data. It takes 2 <code>DESeqDataSet</code> objects, extracts statistical
results and converts the results to a 'volc3d' object, which can be directly
plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deseq_polar(
  object,
  objectLRT,
  contrast = NULL,
  data = NULL,
  pcutoff = 0.05,
  padj.method = "BH",
  filter_pairwise = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deseq_polar_+3A_object">object</code></td>
<td>
<p>An object of class 'DESeqDataSet' with the full design formula.
The function <code>DESeq</code> needs to have been run.</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_objectlrt">objectLRT</code></td>
<td>
<p>An object of class 'DESeqDataSet' with the reduced design
formula. The function <code>DESeq</code> needs to have been run on this object with
argument <code>test="LRT"</code>.</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_contrast">contrast</code></td>
<td>
<p>Character value specifying column within the metadata stored
in the DESeq2 dataset objects is the outcome variable. This column must
contain a factor with 3 levels. If not set, the function will select the
last term in the design formula of <code>object</code> as per DESeq2 convention.</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_data">data</code></td>
<td>
<p>Optional matrix containing gene expression data. If not supplied,
the function will pull the expression data from within the DESeq2 object
using the DESeq2 function <code>assay()</code>. NOTE: for consistency with gene
expression datasets, genes are in rows.</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_padj.method">padj.method</code></td>
<td>
<p>Can be any method available in <code>p.adjust</code> or <code>"qvalue"</code>.
The option <code>"none"</code> is a pass-through.</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_filter_pairwise">filter_pairwise</code></td>
<td>
<p>Logical whether adjusted p-value pairwise statistical
tests are only conducted on genes which reach significant adjusted p-value
cut-off on the group likelihood ratio test</p>
</td></tr>
<tr><td><code id="deseq_polar_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="#topic+polar_coords">polar_coords</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Calls <code><a href="#topic+polar_coords">polar_coords</a></code> to return an S4 'volc3d' object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+polar_coords">polar_coords</a></code>, <code><a href="#topic+voom_polar">voom_polar</a></code>,
<code><a href="DESeq2.html#topic+DESeq">DESeq</a></code> in the DESeq2 package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

  library(DESeq2)

  counts &lt;- matrix(rnbinom(n=1500, mu=100, size=1/0.5), ncol=15)
  cond &lt;- factor(rep(1:3, each=5), labels = c('A', 'B', 'C'))

  # object construction
  dds &lt;- DESeqDataSetFromMatrix(counts, DataFrame(cond), ~ cond)

  # standard analysis
  dds &lt;- DESeq(dds)

  # Likelihood ratio test
  ddsLRT &lt;- DESeq(dds, test="LRT", reduced= ~ 1)

  polar &lt;- deseq_polar(dds, ddsLRT, "cond")
  volcano3D(polar)
  radial_ggplot(polar)


</code></pre>

<hr>
<h2 id='forest_plot'>Forest plot individual gene from 2x3 factor analysis</h2><span id='topic+forest_plot'></span><span id='topic+forest_plotly'></span><span id='topic+forest_ggplot'></span>

<h3>Description</h3>

<p>Forest plot individual gene from 2x3 factor analysis using either base
graphics, plotly or ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forest_plot(
  object,
  genes,
  scheme = c("red", "green3", "blue"),
  labs = NULL,
  error_type = c("ci", "se"),
  error_width = 0.05,
  gap = 1,
  transpose = FALSE,
  mar = if (transpose) c(5, 7, 5, 4) else c(5, 5, 5, 3),
  ...
)

forest_plotly(
  object,
  genes,
  scheme = c("red", "green3", "blue"),
  labs = NULL,
  error_type = c("ci", "se"),
  error_width = 4,
  gap = 1,
  transpose = FALSE,
  ...
)

forest_ggplot(
  object,
  genes,
  scheme = c("red", "green3", "blue"),
  labs = NULL,
  error_type = c("ci", "se"),
  error_width = 0.3,
  facet = TRUE,
  gap = 1,
  transpose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forest_plot_+3A_object">object</code></td>
<td>
<p>A 'volc3d' class object from a 2x3 analysis generated by
<code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar()</a></code></p>
</td></tr>
<tr><td><code id="forest_plot_+3A_genes">genes</code></td>
<td>
<p>Vector of genes to plot</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_scheme">scheme</code></td>
<td>
<p>Vector of 3 colours for plotting</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_labs">labs</code></td>
<td>
<p>Optional character vector of labels for the groups</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_error_type">error_type</code></td>
<td>
<p>Either &quot;ci&quot; or &quot;se&quot; to specify whether error bars use 95%
confidence intervals or standard error</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_error_width">error_width</code></td>
<td>
<p>Width of error bars</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_gap">gap</code></td>
<td>
<p>Size of gap between groupings for each gene</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_transpose">transpose</code></td>
<td>
<p>Logical whether to transpose the plot</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_mar">mar</code></td>
<td>
<p>Vector of margins on four sides. See <code><a href="graphics.html#topic+par">par()</a></code></p>
</td></tr>
<tr><td><code id="forest_plot_+3A_...">...</code></td>
<td>
<p>Optional arguments</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_facet">facet</code></td>
<td>
<p>Logical whether to use facets for individual genes (ggplot2
only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot using either base graphics (<code>forest_plot</code>), plotly
(<code>forest_plotly</code>) or ggplot2 (<code>forest_ggplot</code>). <code>forest_plot</code> also
invisibly returns the dataframe used for plotting.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar()</a></code>
</p>

<hr>
<h2 id='grid-class'>An S4 class to define the polar grid coordinates system.</h2><span id='topic+grid-class'></span>

<h3>Description</h3>

<p>An S4 class to define the polar grid coordinates system.
</p>


<h3>Slots</h3>


<dl>
<dt><code>polar_grid</code></dt><dd><p>The coordinates for the cylindrical grid segments with
x,y,z coordinates</p>
</dd>
<dt><code>axes</code></dt><dd><p>The axes features for 'plotly'</p>
</dd>
<dt><code>axis_labs</code></dt><dd><p>The axis labels</p>
</dd>
<dt><code>r</code></dt><dd><p>The grid radius</p>
</dd>
<dt><code>z</code></dt><dd><p>The grid height</p>
</dd>
<dt><code>text_coords</code></dt><dd><p>data frame for axis label cartesian coordinates (x, y, z)</p>
</dd>
<dt><code>n_r_breaks</code></dt><dd><p>The number of ticks on the r axis</p>
</dd>
<dt><code>n_z_breaks</code></dt><dd><p>The number of ticks on the z axis</p>
</dd>
<dt><code>r_breaks</code></dt><dd><p>The r axis ticks as a numeric</p>
</dd>
<dt><code>z_breaks</code></dt><dd><p>The z axis ticks as a numeric</p>
</dd>
</dl>

<hr>
<h2 id='polar_coords'>Coordinates for Three Way Polar Plot</h2><span id='topic+polar_coords'></span>

<h3>Description</h3>

<p>This function creates a 'volc3d' object of S4 class for downstream plots
containing the p-values from a three-way group comparison, expression data
sample data and polar coordinates. For RNA-Seq count data, two functions
<code><a href="#topic+deseq_polar">deseq_polar</a></code> or <code><a href="#topic+voom_polar">voom_polar</a></code> can be used instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polar_coords(
  outcome,
  data,
  pvals = NULL,
  padj = pvals,
  pcutoff = 0.05,
  fc_cutoff = NULL,
  scheme = c("grey60", "red", "gold2", "green3", "cyan", "blue", "purple"),
  labs = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="polar_coords_+3A_outcome">outcome</code></td>
<td>
<p>Outcome vector with 3 groups, ideally as a factor. If it is
not a factor, this will be coerced to a factor. This must have exactly 3
levels. NOTE: if <code>pvals</code> is given, the order of the levels in <code>outcome</code>
must correspond to the order of columns in <code>pvals</code>.</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_data">data</code></td>
<td>
<p>Dataframe or matrix with variables in columns</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_pvals">pvals</code></td>
<td>
<p>Matrix or dataframe with p-values. The first column represents a
test across all 3 categories such as one-way ANOVA or likelihood ratio
test. Columns 2-4 represent pairwise tests comparing groups A vs B, A vs C
and B vs C, where A, B, C represent levels 1, 2, 3 in <code>outcome</code>. Columns
2-4 must be provided in the correct order. If <code>pvals</code> is not given, it is
calculated using the function <code><a href="#topic+calc_pvals">calc_pvals</a></code>.</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_padj">padj</code></td>
<td>
<p>Matrix or dataframe with adjusted p-values. If not supplied,
defaults to use nominal p-values from <code>pvals</code>.</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_fc_cutoff">fc_cutoff</code></td>
<td>
<p>Cut-off for fold change on radial axis</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_scheme">scheme</code></td>
<td>
<p>Vector of colours starting with non-significant variables</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_labs">labs</code></td>
<td>
<p>Optional character vector for labelling groups. Default <code>NULL</code>
leads to abbreviated labels based on levels in <code>outcome</code> using
<code><a href="base.html#topic+abbreviate">abbreviate()</a></code>. A vector of length 3 with custom abbreviated names for the
outcome levels can be supplied. Otherwise a vector length 7 is expected, of
the form &quot;ns&quot;, &quot;B+&quot;, &quot;B+C+&quot;, &quot;C+&quot;, &quot;A+C+&quot;, &quot;A+&quot;, &quot;A+B+&quot;, where &quot;ns&quot; means
non-significant and A, B, C refer to levels 1, 2, 3 in <code>outcome</code>, and must
be in the correct order.</p>
</td></tr>
<tr><td><code id="polar_coords_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="#topic+calc_pvals">calc_pvals</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S4 'volc3d' object containing:
</p>

<ul>
<li><p>'df' A list of 2 dataframes. Each dataframe contains both x,y,z
coordinates as well as polar coordinates r, angle. The first dataframe has
coordinates on scaled data. The 2nd dataframe has unscaled data (e.g. log2
fold change for gene expression). The <code>type</code> argument in
<code><a href="#topic+volcano3D">volcano3D</a></code>, <code><a href="#topic+radial_plotly">radial_plotly</a></code> and
<code><a href="#topic+radial_ggplot">radial_ggplot</a></code> corresponds to these dataframes.
</p>
</li>
<li><p>'outcome' The three-group contrast factor used for comparisons
</p>
</li>
<li><p>'data' Dataframe or matrix containing the expression data
</p>
</li>
<li><p>'pvals' A dataframe containing p-values. First column is the 3-way
comparison (LRT or ANOVA). Columns 2-4 are pairwise comparisons between
groups A vs B, A vs C and B vs C, where A, B, C are the 3 levels in the
outcome factor.
</p>
</li>
<li><p>'padj' A dataframe containing p-values adjusted for multiple testing
</p>
</li>
<li><p>'pcutoff Numeric value for cut-off for p-value significance
</p>
</li>
<li><p>'scheme' Character vector with colour scheme for plotting
</p>
</li>
<li><p>'labs' Character vector with labels for colour groups
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+deseq_polar">deseq_polar</a></code>, <code><a href="#topic+voom_polar">voom_polar</a></code>,
<code><a href="#topic+calc_pvals">calc_pvals</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

</code></pre>

<hr>
<h2 id='polar_coords_2x3'>Coordinates for three way polar plot from 2x3 factor analysis</h2><span id='topic+polar_coords_2x3'></span>

<h3>Description</h3>

<p>This function creates a 'volc3d' object of S4 class for downstream plots
containing the p-values from a 2x3 factor analysis, expression data
sample data and polar coordinates. For RNA-Seq count data, two functions
<code><a href="#topic+deseq_2x3">deseq_2x3</a></code> followed by <code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar()</a></code> can be used instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polar_coords_2x3(
  data,
  metadata = NULL,
  outcome,
  group,
  pvals = NULL,
  padj = pvals,
  pcutoff = 0.05,
  fc_cutoff = NULL,
  padj.method = "BH",
  process = c("positive", "negative", "two.sided"),
  scheme = c("grey60", "red", "gold2", "green3", "cyan", "blue", "purple", "black"),
  labs = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="polar_coords_2x3_+3A_data">data</code></td>
<td>
<p>Dataframe or matrix with variables in columns and samples in rows</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_metadata">metadata</code></td>
<td>
<p>Dataframe of sample information with samples in rows</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_outcome">outcome</code></td>
<td>
<p>Either the name of column in <code>metadata</code> containing the binary
outcome data. Or a vector with 2 groups, ideally a factor. If it is not a
factor, this will be coerced to a factor. This must have exactly 2 levels.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_group">group</code></td>
<td>
<p>Either the name of column in <code>metadata</code> containing the 3-way
grouping data. Or a vector with 3 groups, ideally a factor. If it is not a
factor, this will be coerced to a factor. This must have exactly 3 levels.
NOTE: if <code>pvals</code> is given, the order of the levels in <code>group</code> must
correspond to the order of columns in <code>pvals</code>.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_pvals">pvals</code></td>
<td>
<p>Optional matrix or dataframe with p-values in 3 columns. If
<code>pvals</code> is not given, it is calculated using the function
<code><a href="#topic+calc_stats_2x3">calc_stats_2x3</a></code>. The p-values in 3 columns represent the
comparison between the binary outcome with each column for the 3 groups as
specified in <code>group</code>.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_padj">padj</code></td>
<td>
<p>Matrix or dataframe with adjusted p-values. If not supplied,
defaults to use nominal p-values from <code>pvals</code>.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_fc_cutoff">fc_cutoff</code></td>
<td>
<p>Cut-off for fold change on radial axis</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_padj.method">padj.method</code></td>
<td>
<p>Can be <code>"qvalue"</code> or any method available in <code>p.adjust</code>.
The option <code>"none"</code> is a pass-through.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_process">process</code></td>
<td>
<p>Character value specifying colour process for statistical
significant genes: &quot;positive&quot; specifies genes are coloured if fold change
is &gt;0; &quot;negative&quot; for genes with fold change &lt;0 (note that for clarity the
polar position is altered so that genes along each axis have the most
strongly negative fold change values); or &quot;two.sided&quot; which is a compromise
in which positive genes are labelled as before but genes with negative fold
changes and significant p-values have an inverted colour scheme.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_scheme">scheme</code></td>
<td>
<p>Vector of colours starting with non-significant variables</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_labs">labs</code></td>
<td>
<p>Optional character vector for labelling groups. Default <code>NULL</code>
leads to abbreviated labels based on levels in <code>outcome</code> using
<code><a href="base.html#topic+abbreviate">abbreviate()</a></code>. A vector of length 3 with custom abbreviated names for the
outcome levels can be supplied. Otherwise a vector length 8 is expected, of
the form &quot;ns&quot;, &quot;A+&quot;, &quot;A+B+&quot;, &quot;B+&quot;, &quot;B+C+&quot;, &quot;C+&quot;, &quot;A+C+&quot;, &quot;A+B+C+&quot;, where
&quot;ns&quot; means non-significant and A, B, C refer to levels 1, 2, 3 in
<code>outcome</code>, and must be in the correct order.</p>
</td></tr>
<tr><td><code id="polar_coords_2x3_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="#topic+calc_stats_2x3">calc_stats_2x3</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is designed for manually generating a 'volc3d' class object for
visualising a 2x3 way analysis comparing a large number of attributes such as
genes. For RNA-Seq data we suggest using <code><a href="#topic+deseq_2x3">deseq_2x3()</a></code> and
<code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar()</a></code> functions in sequence instead.
</p>
<p>Scaled polar coordinates are generated using the t-score for each group
comparison. Unscaled polar coordinates are generated as difference between
means for each group comparison. If p-values are not supplied they are
calculated by <code><a href="#topic+calc_stats_2x3">calc_stats_2x3()</a></code> using either t-tests or wilcoxon tests.
</p>
<p>The z axis for 3d volcano plots does not have as clear a corollary in 2x3
analysis as for the standard 3-way analysis (which uses the likelihood ratio
test for the 3 groups). For 2x3 polar analysis the smallest p-value from the
3 group pairwise comparisons for each gene is used to generate a z coordinate
as -log10(p-value).
</p>
<p>The colour scheme is not as straightforward as for the standard polar plot
and volcano3D plot since genes (or attributes) can be significantly up or
downregulated in the response comparison for each of the 3 groups.
<code>process = "positive"</code> means that genes are labelled with colours if a gene
is significantly upregulated in the response for that group. This uses the
primary colours (RGB) so that if a gene is upregulated in both red and blue
group it becomes purple etc with secondary colours. If the gene is
upregulated in all 3 groups it is labelled black. Non-significant genes are
in grey.
</p>
<p>With <code>process = "negative"</code> genes are coloured when they are significantly
downregulated. With <code>process = "two.sided"</code> the colour scheme means that both
significantly up- and down-regulated genes are coloured with downregulated
genes labelled with inverted colours (i.e. cyan is the inverse of red etc).
However, significant upregulation in a group takes precedence.
</p>


<h3>Value</h3>

<p>Returns an S4 'volc3d' object containing:
</p>

<ul>
<li><p>'df' A list of 2 dataframes. Each dataframe contains both x,y,z
coordinates as well as polar coordinates r, angle. The first dataframe has
coordinates on scaled data. The 2nd dataframe has unscaled data (e.g. log2
fold change for gene expression). The <code>type</code> argument in
<code><a href="#topic+volcano3D">volcano3D</a></code>, <code><a href="#topic+radial_plotly">radial_plotly</a></code> and
<code><a href="#topic+radial_ggplot">radial_ggplot</a></code> corresponds to these dataframes.
</p>
</li>
<li><p>'outcome' The three-group contrast factor used for comparisons,
linked to the <code>group</code> column
</p>
</li>
<li><p>'data' Dataframe or matrix containing the expression data
</p>
</li>
<li><p>'pvals' A dataframe containing p-values in 3 columns representing
the binary comparison for the outcome for each of the 3 groups.
</p>
</li>
<li><p>'padj' A dataframe containing p-values adjusted for multiple testing
</p>
</li>
<li><p>'pcutoff Numeric value for cut-off for p-value significance
</p>
</li>
<li><p>'scheme' Character vector with colour scheme for plotting
</p>
</li>
<li><p>'labs' Character vector with labels for colour groups
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+deseq_2x3">deseq_2x3</a></code>, <code><a href="#topic+deseq_2x3_polar">deseq_2x3_polar</a></code>,
<code><a href="#topic+calc_stats_2x3">calc_stats_2x3</a></code>
</p>

<hr>
<h2 id='polar_grid'>Grid required for 3D volcano plot and 2D radial plots</h2><span id='topic+polar_grid'></span>

<h3>Description</h3>

<p>Generates a cylindrical grid of the appropriate dimensions for a 3D volcano
plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polar_grid(
  r_vector = NULL,
  z_vector = NULL,
  r_axis_ticks = NULL,
  z_axis_ticks = NULL,
  axis_angle = 5/6,
  n_spokes = 12,
  axes_from_origin = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="polar_grid_+3A_r_vector">r_vector</code></td>
<td>
<p>An optional numerical vector for the  radial coordinates.
This is used to calculate breaks on the r axis using
<code><a href="base.html#topic+pretty">pretty</a></code>. If this is NULL the r_axis_ticks are used as
breaks.</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_z_vector">z_vector</code></td>
<td>
<p>An optional numerical vector for the z coordinates.
This is used to calculate breaks on the z axis using <code>pretty</code>. If this
is NULL the z_axis_ticks are used as breaks.</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_r_axis_ticks">r_axis_ticks</code></td>
<td>
<p>A numerical vector of breaks for the radial axis (used
if r_vector is NULL).</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_z_axis_ticks">z_axis_ticks</code></td>
<td>
<p>A numerical vector of breaks for the z axis (used
if z_vector is NULL).</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_axis_angle">axis_angle</code></td>
<td>
<p>angle in radians to position the radial axis
(default = 5/6)</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_n_spokes">n_spokes</code></td>
<td>
<p>the number of outward spokes to be plotted (default = 12)</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_axes_from_origin">axes_from_origin</code></td>
<td>
<p>Whether the axis should start at 0 or the first
break (default = TRUE)</p>
</td></tr>
<tr><td><code id="polar_grid_+3A_...">...</code></td>
<td>
<p>optional parameters for <code><a href="base.html#topic+pretty">pretty</a></code> on the r axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S4 grid object containing:
</p>

<ul>
<li><p>'polar_grid' The coordinates for a radial grid
</p>
</li>
<li><p>'axes' The axes features for 'plotly'
</p>
</li>
<li><p>'axis_labels' The axis labels
</p>
</li>
<li><p>'r' The grid radius
</p>
</li>
<li><p>'z' The grid height
</p>
</li>
<li><p>'text_coords' The coordinates for text labels
</p>
</li>
<li><p>'n_r_breaks' The number of ticks on the r axis
</p>
</li>
<li><p>'n_r_breaks' The number of ticks on the z axis
</p>
</li></ul>



<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

grid &lt;- polar_grid(r_vector=syn_polar@df[[1]]$r,
           z_vector=NULL,
           r_axis_ticks = NULL,
           z_axis_ticks = c(0, 8, 16, 32),
           n_spokes = 4)
</code></pre>

<hr>
<h2 id='radial_ggplot'>'Ggplot' for Three Way Polar Plot</h2><span id='topic+radial_ggplot'></span>

<h3>Description</h3>

<p>This function creates a 3-way polar plot using 'ggplot' for a three-class
comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>radial_ggplot(
  polar,
  type = 1,
  colours = NULL,
  label_rows = NULL,
  arrow_length = 1,
  label_size = 5,
  colour_code_labels = FALSE,
  label_colour = "black",
  grid_colour = "grey80",
  grid_width = 0.7,
  axis_colour = "black",
  axis_width = 1,
  axis_title_size = 5,
  axis_label_size = 3,
  marker_alpha = 0.7,
  marker_size = 3,
  marker_outline_colour = "white",
  marker_outline_width = 0.5,
  legend_size = 20,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="radial_ggplot_+3A_polar">polar</code></td>
<td>
<p>A 'volc3d' object with the p-values between groups of interest
and polar coordinates created by <code><a href="#topic+polar_coords">polar_coords</a></code>,
<code><a href="#topic+deseq_polar">deseq_polar</a></code> or <code><a href="#topic+voom_polar">voom_polar</a></code>.</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_type">type</code></td>
<td>
<p>Numeric value whether to use scaled (z-score) or unscaled (fold
change) as magnitude. Options are 1 = z-score (default) or 2 =
unscaled/fold change.</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_colours">colours</code></td>
<td>
<p>A vector of colours for the non-significant points and each of
the six groups.</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_label_rows">label_rows</code></td>
<td>
<p>A vector of row names or indices to label</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_arrow_length">arrow_length</code></td>
<td>
<p>The length of label arrows</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_label_size">label_size</code></td>
<td>
<p>Font size of labels/annotations (default = 5).</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_colour_code_labels">colour_code_labels</code></td>
<td>
<p>Logical whether label annotations should be colour
coded. If FALSE <code>label_colour</code> is used.</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_label_colour">label_colour</code></td>
<td>
<p>Colour of annotation labels if not colour coded</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_grid_colour">grid_colour</code></td>
<td>
<p>The colour of the grid (default=&quot;grey80&quot;)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_grid_width">grid_width</code></td>
<td>
<p>The width of the axis lines (default=0.6)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_axis_colour">axis_colour</code></td>
<td>
<p>The colour of the grid axes and labels (default=&quot;black&quot;)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_axis_width">axis_width</code></td>
<td>
<p>The width of the axis lines (default=1)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>Font size for axis titles (default = 5)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_axis_label_size">axis_label_size</code></td>
<td>
<p>Font size for axis labels (default = 3)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_marker_alpha">marker_alpha</code></td>
<td>
<p>The alpha parameter for markers (default = 0.7)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_marker_size">marker_size</code></td>
<td>
<p>Size of the markers (default = 3)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_marker_outline_colour">marker_outline_colour</code></td>
<td>
<p>Colour for marker outline (default = white)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_marker_outline_width">marker_outline_width</code></td>
<td>
<p>Width for marker outline (default = 0.5)</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_legend_size">legend_size</code></td>
<td>
<p>Size for the legend text (default = 20).</p>
</td></tr>
<tr><td><code id="radial_ggplot_+3A_...">...</code></td>
<td>
<p>Optional parameters passed to <code><a href="#topic+polar_grid">polar_grid</a></code>
e.g. <code>r_axis_ticks</code> or <code>axis_angle</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a polar 'ggplot' object featuring variables on a tri-axis
radial graph
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>See Also</h3>

<p><code><a href="#topic+polar_coords">polar_coords</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

radial_ggplot(polar = syn_polar, label_rows = c("COBL"))
</code></pre>

<hr>
<h2 id='radial_plotly'>Three-way radial comparison Polar Plot (using plotly)</h2><span id='topic+radial_plotly'></span>

<h3>Description</h3>

<p>This function creates an interactive plotly object which maps differential
expression onto a polar coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>radial_plotly(
  polar,
  type = 1,
  colours = polar@scheme,
  label_rows = NULL,
  arrow_length = 80,
  label_size = 14,
  colour_code_labels = FALSE,
  label_colour = "black",
  grid_colour = "grey80",
  grid_width = 1,
  marker_size = 7,
  marker_alpha = 0.8,
  marker_outline_colour = "white",
  marker_outline_width = 0.5,
  axis_title_size = 16,
  axis_label_size = 10,
  axis_colour = "black",
  axis_width = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="radial_plotly_+3A_polar">polar</code></td>
<td>
<p>A 'volc3d' object with the p-values between groups of interest
and polar coordinates created by <code><a href="#topic+polar_coords">polar_coords</a></code>,
<code><a href="#topic+deseq_polar">deseq_polar</a></code> or <code><a href="#topic+voom_polar">voom_polar</a></code>.</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_type">type</code></td>
<td>
<p>Numeric value whether to use scaled (Z-score) or unscaled (fold
change) as magnitude. Options are 1 = Z-score (default) or 2 =
unscaled/fold change.</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_colours">colours</code></td>
<td>
<p>A vector of colour names or hex triplets for the
non-significant points and each of the six groups.</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_label_rows">label_rows</code></td>
<td>
<p>A vector of row names or numbers to label.</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_arrow_length">arrow_length</code></td>
<td>
<p>The length of label arrows (default = 80).</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_label_size">label_size</code></td>
<td>
<p>Font size of labels/annotations (default = 14)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_colour_code_labels">colour_code_labels</code></td>
<td>
<p>Logical whether label annotations should be colour
coded. If FALSE label_colour is used.</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_label_colour">label_colour</code></td>
<td>
<p>HTML colour of annotation labels if not colour coded.</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_grid_colour">grid_colour</code></td>
<td>
<p>The colour of the grid (default=&quot;grey80&quot;)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_grid_width">grid_width</code></td>
<td>
<p>The width of the grid lines (default=1)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_marker_size">marker_size</code></td>
<td>
<p>Size of the markers (default = 6)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_marker_alpha">marker_alpha</code></td>
<td>
<p>Opacity for the markers (default = 0.7)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_marker_outline_colour">marker_outline_colour</code></td>
<td>
<p>Colour for marker outline (default = white)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_marker_outline_width">marker_outline_width</code></td>
<td>
<p>Width for marker outline (default = 0.5)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_axis_title_size">axis_title_size</code></td>
<td>
<p>Font size for axis titles (default = 16)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_axis_label_size">axis_label_size</code></td>
<td>
<p>Font size for axis labels (default = 10)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_axis_colour">axis_colour</code></td>
<td>
<p>The colour of the grid axes and labels (default=&quot;black&quot;)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_axis_width">axis_width</code></td>
<td>
<p>The width of the axis lines (default=2)</p>
</td></tr>
<tr><td><code id="radial_plotly_+3A_...">...</code></td>
<td>
<p>Optional parameters passed to <code><a href="#topic+polar_grid">polar_grid</a></code>
e.g. <code>r_axis_ticks</code> or <code>axis_angle</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function builds a layered plotly object. By default this produces an SVG
output, but this can be slow with 1000s of points. For large number of points
we recommend switching to webGL by piping to <code>toWebGL()</code> as shown in the
examples.
</p>


<h3>Value</h3>

<p>Returns a plotly plot featuring variables on a tri-axis radial graph
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>See Also</h3>

<p><code><a href="#topic+polar_coords">polar_coords</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

radial_plotly(polar = syn_polar, label_rows = c("COBL"))

## Faster webGL version for large numbers of points
library(plotly)
radial_plotly(polar = syn_polar, label_rows = c("COBL")) %&gt;%
  toWebGL()

</code></pre>

<hr>
<h2 id='show_grid'>Plots grid objects for inspection using plotly</h2><span id='topic+show_grid'></span>

<h3>Description</h3>

<p>This function creates an interactive grids in polar and cylindrical
coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_grid(grid, plot_height = 700, axis_angle = 0, z_axis_title_offset = 1.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show_grid_+3A_grid">grid</code></td>
<td>
<p>A grid object produced by <code><a href="#topic+polar_grid">polar_grid</a></code>.</p>
</td></tr>
<tr><td><code id="show_grid_+3A_plot_height">plot_height</code></td>
<td>
<p>The plot height in px (default=700),</p>
</td></tr>
<tr><td><code id="show_grid_+3A_axis_angle">axis_angle</code></td>
<td>
<p>The angle in radians at which to add axis (default=0).</p>
</td></tr>
<tr><td><code id="show_grid_+3A_z_axis_title_offset">z_axis_title_offset</code></td>
<td>
<p>Offset for z axis title (default=1.2).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing a polar and cylindrical coordinate system.
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

grid &lt;- polar_grid(r_vector=syn_polar@df[[1]]$r,
        z_vector=syn_polar@df[[1]]$z,
        r_axis_ticks = NULL,
        z_axis_ticks = NULL)
p &lt;- show_grid(grid)
p$polar
p$cyl
</code></pre>

<hr>
<h2 id='significance_subset'>Extract a subset population</h2><span id='topic+significance_subset'></span>

<h3>Description</h3>

<p>Subsets data according to the significance groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>significance_subset(polar, significance = NULL, output = "pvals")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="significance_subset_+3A_polar">polar</code></td>
<td>
<p>A polar object including expression data from groups of
interest. Created by <code><a href="#topic+polar_coords">polar_coords</a></code>.</p>
</td></tr>
<tr><td><code id="significance_subset_+3A_significance">significance</code></td>
<td>
<p>Which significance factors to subset to. If <code>NULL</code>
all levels except 'ns' (non-significant) are selected.</p>
</td></tr>
<tr><td><code id="significance_subset_+3A_output">output</code></td>
<td>
<p>What object to return. Options are &quot;pvals&quot;, &quot;padj&quot;, &quot;data&quot;,
&quot;df&quot; for subset dataframes, or &quot;polar&quot; to subset the entire 'volc3d' class
object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object (type defined by <code>output</code>) with rows susbet to
those which satisfy the significance condition.
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))

subset &lt;- significance_subset(syn_polar, "L+", "df")
</code></pre>

<hr>
<h2 id='syn_example_meta'>PEAC synovial sample data</h2><span id='topic+syn_example_meta'></span>

<h3>Description</h3>

<p>A dataset containing sample data for 81 synovial biopsies from the PEAC
cohort
</p>


<h3>Usage</h3>

<pre><code class='language-R'>syn_example_meta
</code></pre>


<h3>Format</h3>

<p>A data frame with 81 rows and 1 variables:
</p>

<dl>
<dt>Pathotype</dt><dd><p>The synovial biopsy histological pathotype</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">https://pubmed.ncbi.nlm.nih.gov/31461658/</a>
</p>

<hr>
<h2 id='syn_example_rld'>PEAC synovial gene expression data</h2><span id='topic+syn_example_rld'></span>

<h3>Description</h3>

<p>A dataset containing the gene expression data for 81 synovial biopsies from
the PEAC cohort
</p>


<h3>Usage</h3>

<pre><code class='language-R'>syn_example_rld
</code></pre>


<h3>Format</h3>

<p>A data frame with 500 rows representing the most significant
genes/probes and 81 columns representing samples.
</p>


<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">https://pubmed.ncbi.nlm.nih.gov/31461658/</a>
</p>

<hr>
<h2 id='volc3d-class'>An S4 class to define the polar coordinates.</h2><span id='topic+volc3d-class'></span>

<h3>Description</h3>

<p>An S4 class to define the polar coordinates.
</p>


<h3>Slots</h3>


<dl>
<dt><code>df</code></dt><dd><p>List of coordinate data frames for scaled and unscaled expression</p>
</dd>
<dt><code>outcome</code></dt><dd><p>Outcome vector</p>
</dd>
<dt><code>data</code></dt><dd><p>Expression data</p>
</dd>
<dt><code>pvals</code></dt><dd><p>Matrix or dataframe with p-values</p>
</dd>
<dt><code>padj</code></dt><dd><p>Matrix adjusted p-values</p>
</dd>
<dt><code>pcutoff</code></dt><dd><p>Cut-off for p-value significance</p>
</dd>
<dt><code>scheme</code></dt><dd><p>Vector for colour scheme</p>
</dd>
<dt><code>labs</code></dt><dd><p>Character vector for labelling groups</p>
</dd>
</dl>

<hr>
<h2 id='volcano3D'>Three-Dimensional Volcano Plot</h2><span id='topic+volcano3D'></span>

<h3>Description</h3>

<p>Plots the three-way comparisons of variables such as gene expression data in
3D space using plotly. x, y position represents polar position on 3 axes
representing the amount each variable or gene tends to each of the 3
categories. The z axis represents -log10 P value for the one-way test
comparing each variable across the 3 groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcano3D(
  polar,
  type = 1,
  label_rows = c(),
  label_size = 14,
  arrow_length = 100,
  colour_code_labels = FALSE,
  label_colour = "black",
  grid_colour = "grey80",
  grid_width = 2,
  grid_options = NULL,
  axis_colour = "black",
  axis_width = 2,
  marker_size = 3,
  marker_outline_width = 0,
  marker_outline_colour = "white",
  z_axis_title_offset = 1.2,
  z_axis_title_size = 12,
  z_axis_angle = 0.5,
  radial_axis_title_size = 14,
  radial_axis_title_offset = 1.2,
  xy_aspectratio = 1,
  z_aspectratio = 0.8,
  camera_eye = list(x = 0.9, y = 0.9, z = 0.9),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcano3D_+3A_polar">polar</code></td>
<td>
<p>Object of S4 class 'volc3d' following call to either
<code><a href="#topic+polar_coords">polar_coords</a></code>, <code><a href="#topic+deseq_polar">deseq_polar</a></code> or
<code><a href="#topic+voom_polar">voom_polar</a></code></p>
</td></tr>
<tr><td><code id="volcano3D_+3A_type">type</code></td>
<td>
<p>Either <code>1</code> or <code>2</code> specifying type of polar coordinates: <code>1</code> =
Z-scaled, <code>2</code> = unscaled (equivalent to log2 fold change for gene
expression).</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_label_rows">label_rows</code></td>
<td>
<p>A vector of row names or numbers to label</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_label_size">label_size</code></td>
<td>
<p>font size for labels (default 14).</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_arrow_length">arrow_length</code></td>
<td>
<p>The length of label arrows (default 100)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_colour_code_labels">colour_code_labels</code></td>
<td>
<p>Logical whether label annotations should be colour
coded. If <code>FALSE</code> <code>label_colour</code> is used.</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_label_colour">label_colour</code></td>
<td>
<p>HTML colour of annotation labels if not colour coded.</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_grid_colour">grid_colour</code></td>
<td>
<p>The colour of the cylindrical grid (default &quot;grey80&quot;)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_grid_width">grid_width</code></td>
<td>
<p>The width of the grid lines (default 2)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_grid_options">grid_options</code></td>
<td>
<p>Optional list of additional arguments to pass to
<code><a href="#topic+polar_grid">polar_grid</a></code>, eg. <code>z_axis_ticks</code> and <code>r_axis_ticks</code></p>
</td></tr>
<tr><td><code id="volcano3D_+3A_axis_colour">axis_colour</code></td>
<td>
<p>The colour of the grid axes and labels (default &quot;black&quot;)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_axis_width">axis_width</code></td>
<td>
<p>The width of axis lines (default 2)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_marker_size">marker_size</code></td>
<td>
<p>Size of the markers (default 3)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_marker_outline_width">marker_outline_width</code></td>
<td>
<p>Width for marker outline (default 0 means no
outline)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_marker_outline_colour">marker_outline_colour</code></td>
<td>
<p>Colour for marker outline (default white)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_z_axis_title_offset">z_axis_title_offset</code></td>
<td>
<p>The position scaling between grid and z axis title
(default=1.2)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_z_axis_title_size">z_axis_title_size</code></td>
<td>
<p>The font size for the z axis title (default=12)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_z_axis_angle">z_axis_angle</code></td>
<td>
<p>Angle in radians for the position of z axis (default
0.5)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_radial_axis_title_size">radial_axis_title_size</code></td>
<td>
<p>The font size for the radial (default=15)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_radial_axis_title_offset">radial_axis_title_offset</code></td>
<td>
<p>The position scaling between grid and radial
axis title (default=1.2)</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_xy_aspectratio">xy_aspectratio</code></td>
<td>
<p>The aspect ratio for the xy axis compared to z (default
1). Increasing this makes the grid wider in the plot window.</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_z_aspectratio">z_aspectratio</code></td>
<td>
<p>The aspect ratio for the z axis compared to x and y
(default 0.8). Decreasing this makes the plot appear more squat.</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_camera_eye">camera_eye</code></td>
<td>
<p>The (x,y,z) components of the start 'eye' camera vector.
This vector determines the view point about the origin of this scene.</p>
</td></tr>
<tr><td><code id="volcano3D_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="plotly.html#topic+plot_ly">plot_ly</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a cylindrical 3D plotly plot featuring variables on a
tri-axis radial graph with the -log10(multi-group test p-value) on the
z-axis
</p>


<h3>References</h3>

<p>Lewis, Myles J., et al. (2019).
<a href="https://pubmed.ncbi.nlm.nih.gov/31461658/">
Molecular portraits of early rheumatoid arthritis identify clinical and
treatment response phenotypes.</a>
<em>Cell reports</em>, <strong>28</strong>:9
</p>


<h3>See Also</h3>

<p><code><a href="#topic+polar_coords">polar_coords</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_data)
syn_polar &lt;- polar_coords(outcome = syn_example_meta$Pathotype,
                          data = t(syn_example_rld))
volcano3D(syn_polar)

</code></pre>

<hr>
<h2 id='voom_polar'>Convert RNA-Seq count data to a volcano3d object using 'limma voom'</h2><span id='topic+voom_polar'></span>

<h3>Description</h3>

<p>This function is used instead of <code><a href="#topic+polar_coords">polar_coords</a></code> if you have raw
RNA-Seq count data. The function takes a design formula, metadata and raw
RNA-Seq count data and uses 'limma voom' to analyse the data. The results are
converted to a 'volc3d' object ready for plotting a 3d volcano plot or polar
plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>voom_polar(
  formula,
  metadata,
  counts,
  pcutoff = 0.05,
  padj.method = "BH",
  filter_pairwise = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="voom_polar_+3A_formula">formula</code></td>
<td>
<p>Design formula which must be of the form <code>~ 0 + outcome + ...</code>. The 3-way outcome variable must be the first variable after the '0',
and this variable must be a factor with exactly 3 levels.</p>
</td></tr>
<tr><td><code id="voom_polar_+3A_metadata">metadata</code></td>
<td>
<p>Matrix or dataframe containing metadata as referenced by
<code>formula</code></p>
</td></tr>
<tr><td><code id="voom_polar_+3A_counts">counts</code></td>
<td>
<p>Matrix containing raw gene expression count data</p>
</td></tr>
<tr><td><code id="voom_polar_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td></tr>
<tr><td><code id="voom_polar_+3A_padj.method">padj.method</code></td>
<td>
<p>Can be any method available in <code>p.adjust</code> or <code>"qvalue"</code>.
The option &quot;none&quot; is a pass-through.</p>
</td></tr>
<tr><td><code id="voom_polar_+3A_filter_pairwise">filter_pairwise</code></td>
<td>
<p>Logical whether adjusted p-value pairwise statistical
tests are only conducted on genes which reach significant adjusted p-value
cut-off on the group likelihood ratio test</p>
</td></tr>
<tr><td><code id="voom_polar_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="#topic+polar_coords">polar_coords</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Statistical results for the group and pairwise comparisons are calculated
using the 'limma voom' pipeline and the results passed to
<code><a href="#topic+polar_coords">polar_coords</a></code> to generate a 'volc3d' object ready for plotting a
3d volcano plot or polar plot.
</p>


<h3>Value</h3>

<p>Calls <code><a href="#topic+polar_coords">polar_coords</a></code> to return an S4 'volc3d' object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+polar_coords">polar_coords</a></code>, <code><a href="#topic+deseq_polar">deseq_polar</a></code>,
<code><a href="limma.html#topic+voom">voom</a></code> in the limma package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("limma", quietly = TRUE) &amp; 
    requireNamespace("edgeR", quietly = TRUE)) {
  library(limma)
  library(edgeR)
  
  counts &lt;- matrix(rnbinom(n=1500, mu=100, size=1/0.5), ncol=15)
  cond &lt;- factor(rep(1:3, each=5), labels = c('A', 'B', 'C'))
  cond &lt;- data.frame(cond)
  
  polar &lt;- voom_polar(~0 + cond, cond, counts)
  
  volcano3D(polar)
  radial_ggplot(polar)
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
