<!DOCTYPE html><html lang="en"><head><title>Help for package WallomicsData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {WallomicsData}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#WallomicsData-package'><p>WallomicsData: Datasets for Multi-Omics Integration in a Plant Abiotic Stress Context</p></a></li>
<li><a href='#Altitude_Cluster'><p>Altitude Cluster</p></a></li>
<li><a href='#Ecotype'><p>Ecotype</p></a></li>
<li><a href='#Genetic_Cluster'><p>Genetic Cluster</p></a></li>
<li><a href='#Metabolomics_Rosettes'><p>Metabolomics Rosettes</p></a></li>
<li><a href='#Metabolomics_Stems'><p>Metabolomics Stems</p></a></li>
<li><a href='#Metadata'><p>Metadata</p></a></li>
<li><a href='#Phenomics_Rosettes'><p>Phenomics Rosettes</p></a></li>
<li><a href='#Phenomics_Stems'><p>Phenomics Stems</p></a></li>
<li><a href='#Proteomics_Rosettes_CW'><p>Proteomics Rosettes Cell Wall</p></a></li>
<li><a href='#Proteomics_Stems_CW'><p>Proteomics Stems Cell Wall</p></a></li>
<li><a href='#Temperature'><p>Temperature</p></a></li>
<li><a href='#Transcriptomics_Rosettes'><p>Transcriptomics Rosettes</p></a></li>
<li><a href='#Transcriptomics_Rosettes_CW'><p>Transcriptomics Rosettes Cell Wall</p></a></li>
<li><a href='#Transcriptomics_Stems'><p>Transcriptomics Stems</p></a></li>
<li><a href='#Transcriptomics_Stems_CW'><p>Transcriptomics Stems Cell Wall</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Datasets for Multi-Omics Integration in a Plant Abiotic Stress
Context</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Datasets from the WallOmics project. Contains phenomics, metabolomics, proteomics and transcriptomics data collected from two organs of five ecotypes of the model plant Arabidopsis thaliana exposed to two temperature growth conditions. Exploratory and integrative analyses of these data are presented in Durufle et al (2020) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbaa166">doi:10.1093/bib/bbaa166</a>&gt; and Durufle et al (2020) &lt;<a href="https://doi.org/10.3390%2Fcells9102249">doi:10.3390/cells9102249</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-14 12:41:58 UTC; sdejean</td>
</tr>
<tr>
<td>Author:</td>
<td>Sébastien Déjean [aut, cre],
  Harold Duruflé [aut],
  Aurélie Mercadié [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sébastien Déjean &lt;sebastien.dejean@math.univ-toulouse.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-15 10:32:29 UTC</td>
</tr>
</table>
<hr>
<h2 id='WallomicsData-package'>WallomicsData: Datasets for Multi-Omics Integration in a Plant Abiotic Stress Context</h2><span id='topic+WallomicsData'></span><span id='topic+WallomicsData-package'></span>

<h3>Description</h3>

<p>Datasets from the WallOmics project. Contains phenomics, metabolomics, proteomics and transcriptomics data collected from two organs of five ecotypes of the model plant Arabidopsis thaliana exposed to two temperature growth conditions. Exploratory and integrative analyses of these data are presented in Durufle et al (2020) &lt;doi:10.1093/bib/bbaa166&gt; and Durufle et al (2020) &lt;doi:10.3390/cells9102249&gt;.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Sébastien Déjean <a href="mailto:sebastien.dejean@math.univ-toulouse.fr">sebastien.dejean@math.univ-toulouse.fr</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Harold Duruflé <a href="mailto:harold.durufle@inrae.fr">harold.durufle@inrae.fr</a>
</p>
</li>
<li><p> Aurélie Mercadié <a href="mailto:aurelie.mercadie@inrae.fr">aurelie.mercadie@inrae.fr</a>
</p>
</li></ul>


<hr>
<h2 id='Altitude_Cluster'>Altitude Cluster</h2><span id='topic+Altitude_Cluster'></span>

<h3>Description</h3>

<p>The <strong>Altitude Cluster factor</strong> identifies the environment height
from which is originated a given plant from the sample under study, either high altitude (denoted <em>High</em>),
moderate altitude (<em>Low</em>) or the reference group's environment height (<em>Col</em>, the lowest of all).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Altitude_Cluster")
</code></pre>


<h3>Format</h3>

<p>A factor with 3 levels.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("Altitude_Cluster")

# Count how many samples are in each group
table(Altitude_Cluster)

</code></pre>

<hr>
<h2 id='Ecotype'>Ecotype</h2><span id='topic+Ecotype'></span>

<h3>Description</h3>

<p>The <strong>Ecotype factor</strong> identifies the genotype specifically designed for a
given ecosystem of the <em>A. thaliana</em> from which the studied sample comes from. We have a population of reference as well as
4 newly-described Pyrenean populations, namely:
</p>

<ul>
<li><p> Columbia, denoted <em>Col</em> (originating from Poland, acts as the reference ecotype)
</p>
</li>
<li><p> Grip, denoted <em>Grip</em>
</p>
</li>
<li><p> Herran, denoted <em>Hern</em>
</p>
</li>
<li><p> L’Hospitalet-près-l’Andorre, denoted <em>Hosp</em>
</p>
</li>
<li><p> Chapelle Saint Roch, denoted <em>Roch</em>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data("Ecotype")
</code></pre>


<h3>Format</h3>

<p>A factor with 5 levels of <em>A. thaliana</em> genotypes.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("Ecotype")

# Count how many samples are in each group
table(Ecotype)

</code></pre>

<hr>
<h2 id='Genetic_Cluster'>Genetic Cluster</h2><span id='topic+Genetic_Cluster'></span>

<h3>Description</h3>

<p>The <strong>Genetic Cluster factor</strong> identifies the genetic group
from which comes from the studied sample, either <strong>Genetics Cluster 1</strong> (constitued of <em>Grip</em> and <em>Roch</em> genotypes),
<strong>Genetics Cluster 2</strong> (<em>Hern</em> and <em>Hosp</em> genotypes) or <strong>Genetics Cluster 3</strong> (<em>Col</em> genotype).
See <a href="#topic+Ecotype">Ecotype</a> for more information on genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Genetic_Cluster")
</code></pre>


<h3>Format</h3>

<p>A factor with 3 levels.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("Genetic_Cluster")

# Count how many samples are in each group
table(Genetic_Cluster)

</code></pre>

<hr>
<h2 id='Metabolomics_Rosettes'>Metabolomics Rosettes</h2><span id='topic+Metabolomics_Rosettes'></span>

<h3>Description</h3>

<p>A dataset containing metabolomics variables measured on rosettes of
five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Metabolomics_Rosettes")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 6 variables:
</p>

<ul>
<li> <p><strong>Pectin_RGI</strong>: Rhamnogalacturonan I (µg/100mg)
</p>
</li>
<li> <p><strong>Pectin_HG</strong>: Homogalacturonan (µg/100mg)
</p>
</li>
<li> <p><strong>XG</strong>: Xyloglucan (µg/100mg)
</p>
</li>
<li> <p><strong>Pectin_linearity</strong>: Linearity of pectin (Ratio)
</p>
</li>
<li> <p><strong>Contribution_RG</strong>: Contribution of rhamnogalacturonan to pectin population (Ratio)
</p>
</li>
<li> <p><strong>RGI_branching</strong>: Branching of Rhamnogalacturonan I (Ratio)
</p>
</li></ul>



<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Metabolomics_Rosettes")

# Look at simple statistics
summary(Metabolomics_Rosettes)

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# A graphical representation
plot(x = as.factor(substr(row.names(Metabolomics_Rosettes), 1, 7)),
     y = Metabolomics_Rosettes$Pectin_linearity, col = "white", lty = 0,
     xlab = "Genotype x Temperature groups",
     ylab = "Pectin linearity (Ratio)",
     main = "Pectin linearity distribution by genotype and growth temperature")
grid()
abline(h = 1, lty = 2)
points(x = as.factor(substr(row.names(Metabolomics_Rosettes), 1, 7)),
       y = Metabolomics_Rosettes$Pectin_linearity, type = "p", pch = 19, lwd = 5,
       col = colors)

</code></pre>

<hr>
<h2 id='Metabolomics_Stems'>Metabolomics Stems</h2><span id='topic+Metabolomics_Stems'></span>

<h3>Description</h3>

<p>A dataset containing metabolomics variables measured on floral stems of
five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Metabolomics_Stems")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 6 variables:
</p>

<ul>
<li> <p><strong>Pectin_RGI</strong>: Rhamnogalacturonan I (µg/100mg)
</p>
</li>
<li> <p><strong>Pectin_HG</strong>: Homogalacturonan (µg/100mg)
</p>
</li>
<li> <p><strong>XG</strong>: Xyloglucan (µg/100mg)
</p>
</li>
<li> <p><strong>Pectin_linearity</strong>: Linearity of pectin (Ratio)
</p>
</li>
<li> <p><strong>Contribution_RG</strong>: Contribution of rhamnogalacturonan to pectin population (Ratio)
</p>
</li>
<li> <p><strong>RGI_branching</strong>: Branching of Rhamnogalacturonan I (Ratio)
</p>
</li></ul>



<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Metabolomics_Stems")

# Look at simple statistics
summary(Metabolomics_Stems)

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# A graphical representation
plot(x = as.factor(substr(row.names(Metabolomics_Stems), 1, 7)),
     y = Metabolomics_Stems$Pectin_linearity, col = "white", lty = 0,
     xlab = "Genotype x Temperature groups",
     ylab = "Pectin linearity (Ratio)",
     main = "Pectin linearity distribution by genotype and growth temperature")
grid()
abline(h = 1, lty = 2)
points(x = as.factor(substr(row.names(Metabolomics_Stems), 1, 7)),
       y = Metabolomics_Stems$Pectin_linearity, type = "p", pch = 19, lwd = 5,
       col = colors)

</code></pre>

<hr>
<h2 id='Metadata'>Metadata</h2><span id='topic+Metadata'></span>

<h3>Description</h3>

<p>Bioinformatics Annotation and description, using the WallProtDB database, of all the Cell Wall Proteins (CWPs)
identified on rosettes and floral stems of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for additionnal information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Metadata")
</code></pre>


<h3>Format</h3>

<p>A data frame with 474 rows and 4 variables:
</p>

<ul>
<li> <p><strong>Acc_number</strong>: GenBank accession number (gene name)
</p>
</li>
<li> <p><strong>Functional_classes</strong>: Functional classes of the CWPs
</p>
</li>
<li> <p><strong>Protein_families</strong>: Protein families of the CWPs
</p>
</li>
<li> <p><strong>Putative_functions</strong>: Putative functions of the CWPs
</p>
</li></ul>



<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Metadata")

# Look at the dataset's dimensions
dim(Metadata)
head(Metadata)

# How many functional classes ?
table(Metadata$Functional_classes)

# How many protein families ?
table(Metadata$Protein_families)

</code></pre>

<hr>
<h2 id='Phenomics_Rosettes'>Phenomics Rosettes</h2><span id='topic+Phenomics_Rosettes'></span>

<h3>Description</h3>

<p>A dataset containing phenotypic variables measured on rosettes of
five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Phenomics_Rosettes")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 5 variables:
</p>

<ul>
<li> <p><strong>Mass</strong>: rosette mass (g)
</p>
</li>
<li> <p><strong>Diameter</strong>: rosette diameter (cm)
</p>
</li>
<li> <p><strong>Leaves_number</strong>: total number of leaves
</p>
</li>
<li> <p><strong>Density</strong>: rosette density (g/cm²)
</p>
</li>
<li> <p><strong>Area</strong>: projected rosette area (cm²)
</p>
</li></ul>



<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("Phenomics_Rosettes")

# Look at simple statistics
dim(Phenomics_Rosettes)
summary(Phenomics_Rosettes)

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# A graphical representation: Leaves number distribution
plot(x = as.factor(substr(row.names(Phenomics_Rosettes), 1, 7)),
     y = Phenomics_Rosettes$Leaves_number, col = "white", lty = 0,
     xlab = "Genotype x Temperature groups",
     ylab = "Number of rosette leaves",
     main = "Rosette leaves' distribution by genotype and growth temperature"
     )
grid()
points(x = as.factor(substr(row.names(Phenomics_Rosettes), 1, 7)),
       y = Phenomics_Rosettes$Leaves_number, type = "p", pch = 19, lwd = 5,
       col = colors)

</code></pre>

<hr>
<h2 id='Phenomics_Stems'>Phenomics Stems</h2><span id='topic+Phenomics_Stems'></span>

<h3>Description</h3>

<p>A dataset containing phenotypic variables measured on floral stems of
five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Phenomics_Stems")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 4 variables:
</p>

<ul>
<li> <p><strong>Mass</strong>: floral stems mass (g)
</p>
</li>
<li> <p><strong>Diameter</strong>: floral stems diameter (mm)
</p>
</li>
<li> <p><strong>Length</strong>: length of the floral stems (cm)
</p>
</li>
<li> <p><strong>Number_lateral_stems</strong>: number of lateral stems)
</p>
</li></ul>



<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("Phenomics_Stems")

# Look at simple statistics
dim(Phenomics_Stems)
summary(Phenomics_Stems)

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# A graphical representation: Lateral stems distribution
plot(x = as.factor(substr(row.names(Phenomics_Stems), 1, 7)),
     y = Phenomics_Stems$Number_lateral_stems, col = "white", lty = 0,
     xlab = "Genotype x Temperature groups",
     ylab = "Number of lateral stems",
     main = "Lateral stems' distribution by genotype and growth temperature"
     )
grid()
points(x = as.factor(substr(row.names(Phenomics_Stems), 1, 7)),
       y = Phenomics_Stems$Number_lateral_stems, type = "p", pch = 19, lwd = 5,
       col = colors)

</code></pre>

<hr>
<h2 id='Proteomics_Rosettes_CW'>Proteomics Rosettes Cell Wall</h2><span id='topic+Proteomics_Rosettes_CW'></span>

<h3>Description</h3>

<p>A dataset containing the identification and quantification of Cell Wall Proteins (CWPs) performed using LC-MS/MS
analysis on rosettes of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for additional information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Proteomics_Rosettes_CW")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 364 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Proteomics_Rosettes_CW")

# Look at data frame dimensions
dim(Proteomics_Rosettes_CW)

# Look at the first rows and columns
head(Proteomics_Rosettes_CW[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on proteomics
res.pca &lt;- prcomp(Proteomics_Rosettes_CW, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Rosettes Cell Wall Proteomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>

<hr>
<h2 id='Proteomics_Stems_CW'>Proteomics Stems Cell Wall</h2><span id='topic+Proteomics_Stems_CW'></span>

<h3>Description</h3>

<p>A dataset containing the identification and quantification of Cell Wall Proteins (CWPs) performed using LC-MS/MS
analysis on floral stems of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for additionnal information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Proteomics_Stems_CW")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 414 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Proteomics_Stems_CW")

# Look at data frame dimensions
dim(Proteomics_Stems_CW)

# Look at the first rows and columns
head(Proteomics_Stems_CW[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on proteomics
res.pca &lt;- prcomp(Proteomics_Stems_CW, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Stems Cell Wall Proteomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>

<hr>
<h2 id='Temperature'>Temperature</h2><span id='topic+Temperature'></span>

<h3>Description</h3>

<p>The <strong>Temperature factor</strong> identifies the temperature at which the studied sample was exposed
all along its growth, either 22°C (optimal condition) or 15°C (high altitude condition).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Temperature")
</code></pre>


<h3>Format</h3>

<p>A factor with 2 levels.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the data
data("Temperature")

# Count how many samples are in each group
table(Temperature)

</code></pre>

<hr>
<h2 id='Transcriptomics_Rosettes'>Transcriptomics Rosettes</h2><span id='topic+Transcriptomics_Rosettes'></span>

<h3>Description</h3>

<p>A dataset containing all the transcripts obtained by RNA-seq performed, according to the standard Illumina protocols,
on rosettes of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Transcriptomics_Rosettes")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 19763 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Transcriptomics_Rosettes")

# Look at data frame dimensions
dim(Transcriptomics_Rosettes)

# Look at the first rows and columns
head(Transcriptomics_Rosettes[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on transcriptomics
res.pca &lt;- prcomp(Transcriptomics_Rosettes, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Rosettes' Transcriptomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>

<hr>
<h2 id='Transcriptomics_Rosettes_CW'>Transcriptomics Rosettes Cell Wall</h2><span id='topic+Transcriptomics_Rosettes_CW'></span>

<h3>Description</h3>

<p>A dataset containing the transcripts encoding Cell Wall Proteins (CWPs)
sorted from the 19 763 transcripts (see <a href="#topic+Transcriptomics_Rosettes">Transcriptomics_Rosettes</a>) obtained by RNA-seq performed,
according to the standard Illumina protocols, on rosettes of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Transcriptomics_Rosettes_CW")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 364 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Transcriptomics_Rosettes_CW")

# Look at data frame dimensions
dim(Transcriptomics_Rosettes_CW)

# Look at the first rows and columns
head(Transcriptomics_Rosettes_CW[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on transcriptomics
res.pca &lt;- prcomp(Transcriptomics_Rosettes_CW, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Rosettes Cell Wall Transcriptomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>

<hr>
<h2 id='Transcriptomics_Stems'>Transcriptomics Stems</h2><span id='topic+Transcriptomics_Stems'></span>

<h3>Description</h3>

<p>A dataset containing all the transcripts obtained by RNA-seq performed, according to the standard Illumina protocols,
on floral stems of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Transcriptomics_Stems")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 22570 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Transcriptomics_Stems")

# Look at data frame dimensions
dim(Transcriptomics_Stems)

# Look at the first rows and columns
head(Transcriptomics_Stems[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on transcriptomics
res.pca &lt;- prcomp(Transcriptomics_Stems, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Stems' Transcriptomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>

<hr>
<h2 id='Transcriptomics_Stems_CW'>Transcriptomics Stems Cell Wall</h2><span id='topic+Transcriptomics_Stems_CW'></span>

<h3>Description</h3>

<p>A dataset containing the transcripts encoding Cell Wall Proteins (CWPs) sorted
from the 22 570 transcripts (see <a href="#topic+Transcriptomics_Stems">Transcriptomics_Stems</a>) obtained by RNA-seq performed,
according to the standard Illumina protocols,
on floral stems of five <em>A. thaliana</em> genotypes at two growth temperatures.
See <a href="#topic+Ecotype">Ecotype</a> and <a href="#topic+Temperature">Temperature</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Transcriptomics_Stems_CW")
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 414 variables.
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.3390/cells9102249">10.3390/cells9102249</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the dataset
data("Transcriptomics_Stems_CW")

# Look at data frame dimensions
dim(Transcriptomics_Stems_CW)

# Look at the first rows and columns
head(Transcriptomics_Stems_CW[,c(1:10)])

# Create a colors' vector
colors &lt;- c(rep("#A6CEE3",3), rep("#1F78B4",3), rep("#B2DF8A",3), rep("#33A02C",3),
            rep("#FB9A99",3), rep("#E31A1C",3), rep("#FDBF6F",3), rep("#FF7F00",3),
            rep("#CAB2D6",3), rep("#6A3D9A",3))

# PCA on transcriptomics
res.pca &lt;- prcomp(Transcriptomics_Stems_CW, center = TRUE, scale. = TRUE)
plot(res.pca$x[,"PC1"], res.pca$x[,"PC2"], pch = 19, xlab = "PC1", ylab = "PC2", lwd = 5,
     main = "Individuals' plot (1 x 2) - PCA on Stems Cell Wall Transcriptomics",
     col = colors)
text(res.pca$x[,"PC1"], res.pca$x[,"PC2"], labels = row.names(res.pca$x), cex = 0.8, pos = 3)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
