<!DOCTYPE html><html><head><title>Help for package warbleR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {warbleR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[.extended_selection_table'><p>extract method for class <code>extended_selection_table</code></p></a></li>
<li><a href='#[.selection_table'><p>extract method for class <code>selection_table</code></p></a></li>
<li><a href='#auto_detec'><p><code>auto_detec</code> automatically detects the start and end of vocalizations in sound files based</p>
on amplitude, duration, and frequency range attributes.</a></li>
<li><a href='#autodetec'><p>alternative name for <code>auto_detec</code></p></a></li>
<li><a href='#by_element_est'><p>Convert a by-song extended selection table to by-element</p></a></li>
<li><a href='#catalog'><p>Create catalogs of vocal signals</p></a></li>
<li><a href='#catalog2pdf'><p>Combine <code>catalog</code> images into pdfs</p></a></li>
<li><a href='#check_sels'><p>Check selection data frames</p></a></li>
<li><a href='#check_sound_files'><p>Check sound files</p></a></li>
<li><a href='#check_wavs'><p>alternative name for <code>check_sound_files</code></p></a></li>
<li><a href='#checksels'><p>alternative name for <code>check_sels</code></p></a></li>
<li><a href='#color_spectro'><p>Highlight spectrogram regions</p></a></li>
<li><a href='#color.spectro'><p>alternative name for <code>color_spectro</code></p></a></li>
<li><a href='#comp_matrix'><p>Example matrix listing selections to be compared by <code>cross_correlation</code></p></a></li>
<li><a href='#compare_methods'><p>Assessing the performance of acoustic distance measurements</p></a></li>
<li><a href='#consolidate'><p>Consolidate (sound) files into a single directory</p></a></li>
<li><a href='#coor.graph'><p>alternative name for <code>plot_coordination</code></p></a></li>
<li><a href='#coor.test'><p>alternative name for <code>test_coordination</code></p></a></li>
<li><a href='#cross_correlation'><p>Time-frequency cross-correlation</p></a></li>
<li><a href='#cut_sels'><p>Cut selections into individual sound files</p></a></li>
<li><a href='#df_DTW'><p>alternative name for <code>freq_DTW</code></p></a></li>
<li><a href='#dfDTW'><p>alternative name for <code>freq_DTW</code></p></a></li>
<li><a href='#duration_sound_files'><p>Measure the duration of sound files</p></a></li>
<li><a href='#duration_wavs'><p>alternative name for <code>duration_sound_files</code></p></a></li>
<li><a href='#envelope'><p>Calculates the absolute amplitude envelope</p></a></li>
<li><a href='#ffDTW'><p>deprecated functions and function names</p></a></li>
<li><a href='#filter_sels'><p>Subset selection data frames based on manually filtered image files</p></a></li>
<li><a href='#filtersels'><p>alternative name for <code>filter_sels</code></p></a></li>
<li><a href='#find_clipping'><p>Find clipped selections</p></a></li>
<li><a href='#find_peaks'><p>Find cross-correlation peaks</p></a></li>
<li><a href='#fix_extended_selection_table'><p>Fix extended selection tables</p></a></li>
<li><a href='#fix_wavs'><p>Fix .wav files to allow importing them into R</p></a></li>
<li><a href='#fixwavs'><p>alternative name for <code>fix_wavs</code></p></a></li>
<li><a href='#frange'><p>alternative name for <code>freq_range</code></p></a></li>
<li><a href='#frange.detec'><p>alternative name for <code>freq_range_detec</code></p></a></li>
<li><a href='#freq_DTW'><p>Acoustic dissimilarity using dynamic time warping on dominant frequency contours</p></a></li>
<li><a href='#freq_range'><p>Detect frequency range iteratively</p></a></li>
<li><a href='#freq_range_detec'><p>Detect frequency range on wave objects</p></a></li>
<li><a href='#freq_ts'><p>Extract frequency contours as time series</p></a></li>
<li><a href='#full_spectrogram2pdf'><p><code>full_spectrogram2pdf</code> combines <code>full_spectrograms</code> images in .jpeg format to a single pdf file.</p></a></li>
<li><a href='#full_spectrograms'><p>Create long spectrograms of entire sound files</p></a></li>
<li><a href='#gaps'><p>Gap duration</p></a></li>
<li><a href='#image_to_wave'><p>Convert images into wave objects</p></a></li>
<li><a href='#inflections'><p>Count number of inflections in a frequency contour</p></a></li>
<li><a href='#info_sound_files'><p>Get sound file parameter information</p></a></li>
<li><a href='#info_wavs'><p>alternative name for <code>info_sound_files</code></p></a></li>
<li><a href='#is_extended_selection_table'><p>Class 'extended_selection_table': selection table containing wave objects</p></a></li>
<li><a href='#is_selection_table'><p>Class 'selection_table': double-checked frequency/time coordinates of selections</p></a></li>
<li><a href='#lbh_selec_table'><p>Example data frame of selections (i.e. selection table).</p></a></li>
<li><a href='#lbh_selec_table2'><p>Example data frame of selections (i.e. selection table).</p></a></li>
<li><a href='#lspec'><p>alternative name for <code>full_spectrograms</code></p></a></li>
<li><a href='#lspec2pdf'><p>alternative name for <code>full_spectrogram2pdf</code></p></a></li>
<li><a href='#make.selection.table'><p>Old name for <code>selection_table</code></p></a></li>
<li><a href='#map_xc'><p>Maps of 'Xeno-Canto' recordings by species</p></a></li>
<li><a href='#mfcc_stats'><p>Calculate descriptive statistics on Mel-frequency cepstral coefficients</p></a></li>
<li><a href='#move_images'><p>Move/copy image files between directories</p></a></li>
<li><a href='#move_imgs'><p>alternative name for <code>move_images</code></p></a></li>
<li><a href='#mp3_2_wav'><p>alternative name for <code>mp32wav</code></p></a></li>
<li><a href='#mp32wav'><p>Convert .mp3 files to .wav</p></a></li>
<li><a href='#multi_DTW'><p>A wrapper on <code>dtwDist</code> for comparing multivariate contours</p></a></li>
<li><a href='#new_function_names'><p>Data frame detailing function name changes</p></a></li>
<li><a href='#open_wd'><p>Open working directory</p></a></li>
<li><a href='#overlapping_sels'><p>Find overlapping selections</p></a></li>
<li><a href='#ovlp_sels'><p>alternative name for <code>overlapping_sels</code></p></a></li>
<li><a href='#phylo_spectro'><p>Add spectrograms onto phylogenetic trees</p></a></li>
<li><a href='#plot_coordination'><p>Coordinated singing graphs</p></a></li>
<li><a href='#print.autodetec.output'><p>print method for class <code>autodetec.output</code></p></a></li>
<li><a href='#print.extended_selection_table'><p>print method for class <code>extended_selection_table</code></p></a></li>
<li><a href='#print.find_peaks.output'><p>print method for class <code>xcorr.output</code></p></a></li>
<li><a href='#print.selection_table'><p>print method for class <code>selection_table</code></p></a></li>
<li><a href='#print.xcorr.output'><p>print method for class <code>xcorr.output</code></p></a></li>
<li><a href='#querxc'><p>alternative name for <code>query_xc</code></p></a></li>
<li><a href='#query_xc'><p>Access 'Xeno-Canto' recordings and metadata</p></a></li>
<li><a href='#rbind.extended_selection_table'><p>rbind method for class <code>extended_selection_table</code></p></a></li>
<li><a href='#rbind.selection_table'><p>rbind method for class <code>selection_table</code></p></a></li>
<li><a href='#read_sound_file'><p>An extended version of read_wave that reads several sound file formats and files from selection tables</p></a></li>
<li><a href='#read_wave'><p>A wrapper for tuneR's readWave that read sound files listed within selection tables</p></a></li>
<li><a href='#remove_channels'><p>Remove channels in wave files</p></a></li>
<li><a href='#remove_silence'><p>Remove silence in wave files</p></a></li>
<li><a href='#rename_est_waves'><p>Rename wave objects and associated metadata in extended selection tables</p></a></li>
<li><a href='#rename_waves_est'><p>alternative name for <code>rename_est_waves</code></p></a></li>
<li><a href='#resample_est'><p>Resample wave objects in a extended selection table</p></a></li>
<li><a href='#resample_est_waves'><p>alternative name for <code>resample_est</code></p></a></li>
<li><a href='#rm_channels'><p>alternative name for <code>remove_channels</code></p></a></li>
<li><a href='#rm_sil'><p>alternative name for <code>remove_silence</code></p></a></li>
<li><a href='#selec.table'><p>Alternative name for <code>lbh_selec_table</code></p></a></li>
<li><a href='#selection_table'><p>Create 'selection_table' and 'extended_selection_table' objects</p></a></li>
<li><a href='#seltailor'><p>alternative name for <code>tailor_sels</code></p></a></li>
<li><a href='#sig2noise'><p>Measure signal-to-noise ratio</p></a></li>
<li><a href='#signal_2_noise'><p>alternative name for <code>sig2noise</code></p></a></li>
<li><a href='#sim_coor_sing'><p>Simulated coordinated singing events.</p></a></li>
<li><a href='#sim_songs'><p>alternative name for <code>simulate_songs</code></p></a></li>
<li><a href='#simulate_songs'><p>Simulate animal vocalizations</p></a></li>
<li><a href='#snr_specs'><p>alternative name for <code>snr_spectrograms</code></p></a></li>
<li><a href='#snr_spectrograms'><p>Spectrograms with background noise margins</p></a></li>
<li><a href='#snrspecs'><p>alternative name for <code>snr_spectrograms</code></p></a></li>
<li><a href='#song_analysis'><p>Calculates acoustic parameters at the song level</p></a></li>
<li><a href='#song_param'><p>alternative name for <code>song_analysis</code></p></a></li>
<li><a href='#sort_colms'><p>Sort columns in a more intuitive order</p></a></li>
<li><a href='#sound_pressure_level'><p>Measure relative sound pressure level</p></a></li>
<li><a href='#spec_param'><p>alternative name for <code>tweak_spectro</code></p></a></li>
<li><a href='#specan'><p>alternative name for <code>spectro_analysis</code></p></a></li>
<li><a href='#specreator'><p>alternative name for <code>spectrograms</code></p></a></li>
<li><a href='#spectro_analysis'><p>Measure acoustic parameters in batches of sound files</p></a></li>
<li><a href='#spectrograms'><p>Spectrograms of selected signals</p></a></li>
<li><a href='#split_sound_files'><p>Splits sound files</p></a></li>
<li><a href='#split_wavs'><p>alternative name for <code>split_sound_files</code></p></a></li>
<li><a href='#tailor_sels'><p>Interactive view of spectrograms to tailor selections</p></a></li>
<li><a href='#test_coordination'><p>Randomization test for singing coordination</p></a></li>
<li><a href='#track_freq_contour'><p>Spectrograms with frequency measurements</p></a></li>
<li><a href='#track_harmonic'><p>Track harmonic frequency contour</p></a></li>
<li><a href='#trackfreqs'><p>alternative name for <code>track_freq_contour</code></p></a></li>
<li><a href='#try_na'><p>Wrapper for &quot;try&quot; function</p></a></li>
<li><a href='#tweak_spectro'><p>Plot a mosaic of spectrograms with varying display parameters</p></a></li>
<li><a href='#warbleR'><p>warbleR: A package to streamline bioacoustic analysis</p></a></li>
<li><a href='#warbleR_options'><p>Setting warbleR options</p></a></li>
<li><a href='#warbleR-internals'>
<p><code>warbleR</code> Internal Functions</p></a></li>
<li><a href='#wav_2_flac'><p>Convert .wav files to .flac</p></a></li>
<li><a href='#wav_dur'><p>alternative name for <code>duration_sound_files</code></p></a></li>
<li><a href='#wav_info'><p>alternative name for <code>info_sound_files</code></p></a></li>
<li><a href='#wavdur'><p>alternative name for <code>duration_sound_files</code></p></a></li>
<li><a href='#wpd_features'><p>Measure wavelet packet decomposition features (EXPERIMENTAL)</p></a></li>
<li><a href='#xcmaps'><p>alternative name for <code>map_xc</code></p></a></li>
<li><a href='#xcorr'><p>alternative name for <code>cross_correlation</code></p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Streamline Bioacoustic Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.30</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions aiming to facilitate the analysis of the structure of animal acoustic signals in 'R'. 'warbleR' makes use of the basic sound analysis tools from the package 'seewave', and offers new tools for acoustic structure analysis. The main features of the package are the use of loops to apply tasks through acoustic signals referenced in a selection (annotation) table and the production of spectrograms in image files that allow to organize data and verify acoustic analyzes. The package offers functions to explore, organize and manipulate multiple sound files, explore and download 'Xeno-Canto' recordings, detect signals automatically, create spectrograms of complete recordings or individual signals, run different measures of acoustic signal structure, evaluate the performance of measurement methods, catalog signals, characterize different structural levels in acoustic signals, run statistical analysis of duet coordination and consolidate databases and annotation tables, among others.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>dtw, fftw, graphics, grDevices, monitoR, parallel, pbapply,
RCurl, rjson, stats, utils, methods, Rcpp, knitr, bioacoustics,
cli, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.1), tuneR, seewave (&ge; 2.0.1), NatureSounds</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>fftw3, GNU make, sox, Ghostscript, libsndfile, GDAL</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://marce10.github.io/warbleR/">https://marce10.github.io/warbleR/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/maRce10/warbleR/issues/">https://github.com/maRce10/warbleR/issues/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2, rmarkdown, jpeg, ape, soundgen, wavethresh, png,
pracma, Sim.DiffProc, maps, leaflet, kableExtra, covr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 15:22:32 UTC; marce</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-23 18:40:02 UTC</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Author:</td>
<td>Marcelo Araya-Salas
    <a href="https://orcid.org/0000-0003-3594-619X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Grace Smith-Vidaurre [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marcelo Araya-Salas &lt;marcelo.araya@ucr.ac.cr&gt;</td>
</tr>
</table>
<hr>
<h2 id='+5B.extended_selection_table'>extract method for class <code>extended_selection_table</code></h2><span id='topic++5B.extended_selection_table'></span>

<h3>Description</h3>

<p>extract method for class <code>extended_selection_table</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'extended_selection_table'
X[i = NULL, j = NULL, drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.extended_selection_table_+3A_x">X</code></td>
<td>
<p>Object of class <code>extended_selection_table</code>, generated by <code><a href="#topic+selection_table">selection_table</a></code>.</p>
</td></tr>
<tr><td><code id="+2B5B.extended_selection_table_+3A_i">i</code></td>
<td>
<p>j,    indices specifying elements to extract or replace. Indices are numeric or character vectors or empty
(missing) or NULL.</p>
</td></tr>
</table>

<hr>
<h2 id='+5B.selection_table'>extract method for class <code>selection_table</code></h2><span id='topic++5B.selection_table'></span>

<h3>Description</h3>

<p>extract method for class <code>selection_table</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selection_table'
X[i = NULL, j = NULL, drop = FALSE]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B.selection_table_+3A_x">X</code></td>
<td>
<p>Object of class <code>selection_table</code>, generated by <code><a href="#topic+selection_table">selection_table</a></code>.</p>
</td></tr>
<tr><td><code id="+2B5B.selection_table_+3A_i">i</code></td>
<td>
<p>j,    indices specifying elements to extract or replace. Indices are numeric or character vectors or empty
(missing) or NULL.</p>
</td></tr>
</table>

<hr>
<h2 id='auto_detec'><code>auto_detec</code> automatically detects the start and end of vocalizations in sound files based
on amplitude, duration, and frequency range attributes.</h2><span id='topic+auto_detec'></span>

<h3>Description</h3>

<p><code>auto_detec</code> automatically detects the start and end of vocalizations in sound files based
on amplitude, duration, and frequency range attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auto_detec(X = NULL, wl = 512, threshold = 15, parallel = 1, power = 1,
   output = 'data.frame', thinning = 1, path = NULL, pb = TRUE, ssmooth = 0,
   bp = NULL, flist = NULL, hold.time = 0, mindur = NULL, maxdur = NULL, envt = NULL,
   msmooth = NULL, osci = NULL, xl = NULL, picsize = NULL, res = NULL, flim = NULL,
   ls = NULL, sxrow = NULL, rows = NULL,  redo = NULL, img = NULL, it = NULL,
   set = NULL, smadj = NULL, pal = NULL, fast.spec = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="auto_detec_+3A_x">X</code></td>
<td>
<p>'selection_table' object or a data frame with columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end). If provided the detection will be conducted only within
the selections in 'X'. Alternatively, an 'autodetec.output' object can be input. These objects are also generated by this function when <code>output = "list"</code>. If so the detection runs much faster as envelopes have been already calculated.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window used internally by
<code><a href="seewave.html#topic+ffilter">ffilter</a></code> for bandpass filtering (so only applied when 'bp' is supplied). Default is 512.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_threshold">threshold</code></td>
<td>
<p>A numeric vector of length 1 specifying the amplitude threshold for detecting
signals (in %).</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_power">power</code></td>
<td>
<p>A numeric vector of length 1 indicating a power factor applied to the amplitude envelope. Increasing power will reduce low amplitude modulations and increase high amplitude modulations, in order to reduce background noise. Default is 1 (no change).</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_output">output</code></td>
<td>
<p>Character string indicating if the output should be a 'data.frame' with the detections (default) or a list (of class 'autodetec.output') containing both 1) the detections and 2) the amplitude envelopes (time vs amplitude) for each sound file. The list can be input into <code><a href="#topic+full_spectrograms">full_spectrograms</a></code> to explore detections and associated amplitude envelopes.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_thinning">thinning</code></td>
<td>
<p>Numeric vector of length 1 in the range 0~1 indicating the proportional reduction of the number of
samples used to represent amplitude envelopes (i.e. the thinning of the envelopes). Usually amplitude envelopes have many more samples
than those needed to accurately represent amplitude variation in time, which affects the size of the
output (usually very large R objects / files). Default is  <code>1</code> (no thinning). Higher sampling rates can afford higher size reduction (e.g. lower thinning values). Reduction is conducted by interpolation using <code><a href="stats.html#topic+approx">approx</a></code>. Note that thinning may decrease time precision, and the higher the thinning the less precise the time detection.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_ssmooth">ssmooth</code></td>
<td>
<p>A numeric vector of length 1 to smooth the amplitude envelope
with a sum smooth function. Default is 0. Note that smoothing is applied before thinning (see 'thinning' argument).</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_bp">bp</code></td>
<td>
<p>Numeric vector of length 2 giving the lower and upper limits of a
frequency bandpass filter (in kHz). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_flist">flist</code></td>
<td>
<p>character vector or factor indicating the subset of files that will be analyzed. Ignored
if X is provided.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_hold.time">hold.time</code></td>
<td>
<p>Numeric vector of length 1. Specifies the time range at which selections will be merged (i.e. if 2 selections are separated by less than the specified hold.time they will be merged in to a single selection). Default is  <code>0</code>.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_mindur">mindur</code></td>
<td>
<p>Numeric vector of length 1 giving the shortest duration (in
seconds) of the signals to be detected. It removes signals below that
threshold.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_maxdur">maxdur</code></td>
<td>
<p>Numeric vector of length 1 giving the longest duration (in
seconds) of the signals to be detected. It removes signals above that
threshold.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_envt">envt</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_msmooth">msmooth</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_osci">osci</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_xl">xl</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_picsize">picsize</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_res">res</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_flim">flim</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_ls">ls</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_sxrow">sxrow</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_rows">rows</code></td>
<td>
<p>DEPRECATED</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_redo">redo</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_img">img</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_it">it</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_set">set</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_smadj">smadj</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_pal">pal</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="auto_detec_+3A_fast.spec">fast.spec</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function determines the start and end of signals in the sound file selections listed
in the input data frame ('X'). Alternatively, if no data frame is provided, the function detects signals across
each entire sound file. It can also create long spectrograms highlighting the start and of the detected
signals for all sound files in the working directory (if <code>img = TRUE</code>). Sound files should be located in the
working directory or the path to the sound files should be provided using the 'path' argument. The input
data frame should have the following columns: c(&quot;sound.files&quot;,&quot;selec&quot;,&quot;start&quot;,&quot;end&quot;). This function uses a modified version of the <code><a href="seewave.html#topic+timer">timer</a></code> function from seewave package to detect signals. Note that warbleR function for signal detection will be deprecated in future warbleR versions. Look at the ohun package for automatic signal detection functions.
</p>


<h3>Value</h3>

<p>A data frame containing the start and end of each signal by
sound file and selection number. If 'output = &quot;list&quot;' then a list including 1) a detection data frame, 2) amplitude envelopes and 3) parameters will be return. An additional column 'org.selec' is added when 'X' is provided (so detection can be traced back to the selections in 'X').
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>). Implements a
modified version of the timer function from seewave.
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+cross_correlation">cross_correlation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # Save to temporary working directory
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  ad &lt;- auto_detec(
    threshold = 5, ssmooth = 300,
    bp = c(2, 9), wl = 300, path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='autodetec'>alternative name for <code><a href="#topic+auto_detec">auto_detec</a></code></h2><span id='topic+autodetec'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+auto_detec">auto_detec</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autodetec(
  X = NULL,
  wl = 512,
  threshold = 15,
  parallel = 1,
  power = 1,
  output = "data.frame",
  thinning = 1,
  path = NULL,
  pb = TRUE,
  ssmooth = 0,
  bp = NULL,
  flist = NULL,
  hold.time = 0,
  mindur = NULL,
  maxdur = NULL,
  envt = NULL,
  msmooth = NULL,
  osci = NULL,
  xl = NULL,
  picsize = NULL,
  res = NULL,
  flim = NULL,
  ls = NULL,
  sxrow = NULL,
  rows = NULL,
  redo = NULL,
  img = NULL,
  it = NULL,
  set = NULL,
  smadj = NULL,
  pal = NULL,
  fast.spec = NULL
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+auto_detec">auto_detec</a></code> for documentation. <code><a href="#topic+autodetec">autodetec</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='by_element_est'>Convert a by-song extended selection table to by-element</h2><span id='topic+by_element_est'></span>

<h3>Description</h3>

<p><code>by_element_est</code> converts a by-song extended selection table to by-element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>by_element_est(X, mar = 0.1, pb = FALSE, parallel = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="by_element_est_+3A_x">X</code></td>
<td>
<p>object of class 'extended_selection_table' (see <code><a href="#topic+selection_table">selection_table</a></code>).</p>
</td></tr>
<tr><td><code id="by_element_est_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1 specifying the margins (in seconds)
adjacent to the start and end points of the selections when creating the &rdquo;by element' extended
selection table. Default is 0.1.</p>
</td></tr>
<tr><td><code id="by_element_est_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="by_element_est_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function converts extended selection tables in 'by song' format (several selection per wave object) to a 'by element' format (one wave object per selection).
</p>


<h3>Value</h3>

<p>A 'by element' extended selection table (see <code><a href="#topic+selection_table">selection_table</a></code>).
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
#last modification on nov-9-2022 (MAS)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+mp32wav">mp32wav</a></code>, <code><a href="#topic+fix_wavs">fix_wavs</a></code>
</p>
<p>Other extended selection table manipulation: 
<code><a href="#topic+rename_est_waves">rename_est_waves</a>()</code>,
<code><a href="#topic+resample_est">resample_est</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# create extended selection table
by_song_est &lt;- selection_table(lbh_selec_table,
  path = tempdir(),
  extended = TRUE, by.song = "song", confirm.extended = FALSE
)

# conver o by element
by_element_est &lt;- by_element_est(by_song_est, mar = 0.05)

## End(Not run)
</code></pre>

<hr>
<h2 id='catalog'>Create catalogs of vocal signals</h2><span id='topic+catalog'></span>

<h3>Description</h3>

<p><code>catalog</code> produces spectrograms of selections (signals) split into multiple rows and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>catalog(X, flim = NULL, nrow = 4, ncol = 3, same.time.scale = TRUE, 
collevels = seq(-40, 0, 1), ovlp = 50, parallel = 1, mar = 0.05, prop.mar = NULL, 
lab.mar = 1, wl = 512, wn = "hanning", gr = FALSE, pal = reverse.gray.colors.2, 
it = "jpeg", path = NULL, pb = TRUE, fast.spec = FALSE, res = 100, 
orientation = "v", labels = c("sound.files", "selec"), height = NULL, 
width = NULL, tags = NULL, tag.pal = list(temp.colors, heat.colors, topo.colors),
legend = 3, cex = 1, leg.wd = 1, img.suffix = NULL, img.prefix = NULL, 
tag.widths = c(1, 1), hatching = 0, breaks = c(5, 5), group.tag = NULL, 
spec.mar = 0, spec.bg = "white", max.group.cols = NULL, sub.legend = FALSE, 
rm.axes = FALSE, title = NULL, by.row = TRUE, box = TRUE, highlight = FALSE, alpha = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="catalog_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame with columns for sound file name (sound.files), selection number (selec), 
and start and end time of signal (start and end). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 indicating the highest and lowest 
frequency limits (kHz) of the spectrogram, as in 
<code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_nrow">nrow</code></td>
<td>
<p>A numeric vector of length 1. Specifies number of rows. Default is 4.</p>
</td></tr>
<tr><td><code id="catalog_+3A_ncol">ncol</code></td>
<td>
<p>A numeric vector of length 1.  Specifies number of columns. Default is 3.</p>
</td></tr>
<tr><td><code id="catalog_+3A_same.time.scale">same.time.scale</code></td>
<td>
<p>Logical. Controls if all spectrograms are in the same time scale 
(i.e. have the same duration).</p>
</td></tr>
<tr><td><code id="catalog_+3A_collevels">collevels</code></td>
<td>
<p>A numeric vector of length 3. Specifies levels to partition the 
amplitude range of the spectrogram (in dB). The more levels the higher the
resolution of the spectrogram. Default is seq(-40, 0, 1). seq(-115, 0, 1) will produces spectrograms
similar to other acoustic analysis software packages.</p>
</td></tr>
<tr><td><code id="catalog_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two 
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 50. High values of ovlp 
slow down the function but produce more accurate selection limits (when X is provided).</p>
</td></tr>
<tr><td><code id="catalog_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="catalog_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins (in seconds) adjacent to the start and end points of selections,
delineating spectrogram limits. Default is 0.05.</p>
</td></tr>
<tr><td><code id="catalog_+3A_prop.mar">prop.mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the 
start and end points of selections as a proportion of the duration of the signal. If
provided 'mar' argument is ignored. Default is <code>NULL</code>. Useful when having high 
variation in signal duration. Ignored if <code>same.time.scale = FALSE</code>. Must be &gt; 0 and &lt;= 1.</p>
</td></tr>
<tr><td><code id="catalog_+3A_lab.mar">lab.mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the space allocated to labels and tags (the upper margin). Default is 1.</p>
</td></tr>
<tr><td><code id="catalog_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default 
is 512.</p>
</td></tr>
<tr><td><code id="catalog_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying the window function name. See <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code>
for name options. Default is &quot;hanning&quot;.</p>
</td></tr>
<tr><td><code id="catalog_+3A_gr">gr</code></td>
<td>
<p>Logical argument to add grid to spectrogram. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_pal">pal</code></td>
<td>
<p>Color palette function for spectrogram. Default is reverse.gray.colors.2. See 
<code><a href="seewave.html#topic+spectro">spectro</a></code> for more palettes. Palettes as <code><a href="monitoR.html#topic+specCols">gray.2</a></code> may work better when <code>fast.spec = TRUE</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="catalog_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located. 
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="catalog_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially 
increases performance (much faster), although some options become unavailable, as collevels, and sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>, 
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast.spec' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>, 
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer 
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="catalog_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution. Default is 100 (faster)
although 300 is recommended for publication/presentation quality. Note that high resolution
produce significantly bigger image files. This could be problematic when creating pdf files
using <code><a href="#topic+catalog">catalog</a></code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_orientation">orientation</code></td>
<td>
<p>String. Indicates whether a letter page size image is produced in vertical ('v' option) or
horizontal orientation ('h' option). Note that width and height can also be specified.</p>
</td></tr>
<tr><td><code id="catalog_+3A_labels">labels</code></td>
<td>
<p>String vector. Provides the column names that will be used as labels above the corresponding spectrograms.</p>
</td></tr>
<tr><td><code id="catalog_+3A_height">height</code></td>
<td>
<p>Numeric. Single value (in inches) indicating the height of the output image files. Default is 11 
for vertical orientation.</p>
</td></tr>
<tr><td><code id="catalog_+3A_width">width</code></td>
<td>
<p>Numeric. Single value (in inches) indicating the width of the output image files.  Default is 8.5
for vertical orientation.</p>
</td></tr>
<tr><td><code id="catalog_+3A_tags">tags</code></td>
<td>
<p>String vector. Provides the column names that will be used for the color tagging legend above. Tags can also be numeric. Continuous variables would be break down in 10 color classes.</p>
</td></tr>
<tr><td><code id="catalog_+3A_tag.pal">tag.pal</code></td>
<td>
<p>List of color palette function for tags. Should be of length 1, 2 or 3.  Default is <code>list(temp.colors, heat.colors, topo.colors)</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_legend">legend</code></td>
<td>
<p>A numeric vector of length 1 controlling a legend for color tags is added.
Ignored if no tags are provided. Four values are allowed: 
</p>

<ul>
<li> <p><code>0</code>: No label
</p>
</li>
<li> <p><code>1</code>: Label for the first color tag
</p>
</li>
<li> <p><code>2</code>: Label for the second color tag
</p>
</li>
<li> <p><code>3</code>: Labels both color tags
</p>
</li></ul>

<p>Default is 3. Currently no legend can be set for group tags. Use labels instead.</p>
</td></tr>
<tr><td><code id="catalog_+3A_cex">cex</code></td>
<td>
<p>A numeric vector of length 1 giving the amount by which text 
(including labels and axis) should be magnified. Default is 1.</p>
</td></tr>
<tr><td><code id="catalog_+3A_leg.wd">leg.wd</code></td>
<td>
<p>Numeric. Controls the width of the legend column. Default is 1.</p>
</td></tr>
<tr><td><code id="catalog_+3A_img.suffix">img.suffix</code></td>
<td>
<p>A character vector of length 1 with a suffix (label) to add at the end of the names of 
image files. Default is <code>NULL</code> (no suffix). Useful to label catalogs from different individuals, 
species or sites.</p>
</td></tr>
<tr><td><code id="catalog_+3A_img.prefix">img.prefix</code></td>
<td>
<p>A character vector of length 1 with a prefix (label) to add at the beginning of the names of 
image files. Default is <code>NULL</code> (no prefix). Useful to label catalogs from different individuals, 
species or sites and ensure they will be grouped together when sorted by file name.</p>
</td></tr>
<tr><td><code id="catalog_+3A_tag.widths">tag.widths</code></td>
<td>
<p>A numeric vector of length 2 to control the relative width of the color tags (when 2 tags are provided).</p>
</td></tr>
<tr><td><code id="catalog_+3A_hatching">hatching</code></td>
<td>
<p>A numeric vector of length 1 controlling cross-hatching is used for color tags. Several cross-hatching 
patterns are used to make tags with similar colors more distinguishable. Four values are allowed: 
</p>

<ul>
<li> <p><code>0</code>: No cross-hatching
</p>
</li>
<li> <p><code>1</code>: Cross-hatching the first color tag
</p>
</li>
<li> <p><code>2</code>: Cross-hatching the second color tag
</p>
</li>
<li> <p><code>3</code>: Cross-hatching both color tags
</p>
</li></ul>
</td></tr>
<tr><td><code id="catalog_+3A_breaks">breaks</code></td>
<td>
<p>Numeric vector of length 1 or 2 controlling the number of intervals in which a 
numeric tag will be divided. The numbers control the first and second tags respectively. 
Ignored if tags are not numeric. Default is <code>c(5, 5)</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_group.tag">group.tag</code></td>
<td>
<p>Character vector of length 1 indicating the column name to be used to color
the empty plot areas around the spectrograms. If provided selections that belong to the same
tag level are clumped together in the catalog (the 'X' data frame is sorted by that column).
This tags cannot be included in the legend so it would be better to use the label field to identify the different levels.</p>
</td></tr>
<tr><td><code id="catalog_+3A_spec.mar">spec.mar</code></td>
<td>
<p>Numeric vector of length 1 to add space at the top, left and right sides of
the spectrogram. Useful to better display the grouping of selections when 'group.tag' is 
provided. Internally applied for setting 'mar' using <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_spec.bg">spec.bg</code></td>
<td>
<p>Character vector of length 1 to control the background color of the spectrogram. Default is 'white'. Ignored if <code>group.tag = NULL</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_max.group.cols">max.group.cols</code></td>
<td>
<p>Numeric vector of length 1 indicating the number of different colors 
that will be used for group tags (see 'group.tag' argument). If provided (and the number is 
smaller than the number of levels in the 'group.tag' column) the colors will be recycled, 
although ensuring that adjacent groups do not share the same color. Useful when the 
'group.tag' has many levels and the colors assigned become very similar. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_sub.legend">sub.legend</code></td>
<td>
<p>Logical. If <code>TRUE</code> then only the levels present on each
page are shown in the legend. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_rm.axes">rm.axes</code></td>
<td>
<p>Logical. If <code>TRUE</code> frequency and time axes are excluded. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_title">title</code></td>
<td>
<p>Character vector of length 1 to set the title of catalogs.</p>
</td></tr>
<tr><td><code id="catalog_+3A_by.row">by.row</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default) catalogs are filled by rows.</p>
</td></tr>
<tr><td><code id="catalog_+3A_box">box</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default) a box is drawn around spectrograms and 
corresponding labels and tags.</p>
</td></tr>
<tr><td><code id="catalog_+3A_highlight">highlight</code></td>
<td>
<p>Logical. If <code>TRUE</code> a transparent white layer is plotted on the spectrogram areas outside the selection. The level of transparency is controlled with the argument 'alpha'. Default is <code>FAlSE</code>.</p>
</td></tr>
<tr><td><code id="catalog_+3A_alpha">alpha</code></td>
<td>
<p>Numeric vector of length 1 controlling the level of transparency when highlighting selections (i.e. when <code>highlight = TRUE</code>, see highlight argument. Default is 0.5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions aims to simplify the visual exploration of multiple vocalizations. The function plots a
matrix of spectrograms from a selection table. Spectrograms can be labeled or color tagged to facilitate
exploring variation related to a parameter of interest (e.g. location, song type). A legend will be added to 
help match colors with tag levels (if legend is &gt; 0). Different color palettes can
be used for each tag. Numeric tags are split in intervals (the number of intervals can be
controlled with break argument). The width and height can also be adjusted to fit more column and/or rows.
This files can be put together in a single pdf file with <code><a href="#topic+catalog2pdf">catalog2pdf</a></code>.
We recommend using low resolution (~60-100) and smaller dimensions (width &amp; height &lt; 10) if
aiming to generate pdfs (otherwise pdfs could be pretty big).
</p>


<h3>Value</h3>

<p>Image files with spectrogram catalogs in the working directory. Multiple pages
can be returned, depending on the length of each sound file.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+catalog2pdf">catalog2pdf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# save sound file examples
data(list = c("Phae.long1", "Phae.long2","lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) 
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
 writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
 writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))


catalog(X = lbh_selec_table, flim = c(1, 10), nrow = 4, ncol = 2, same.time.scale = T,
 ovlp = 90, parallel = 1, mar = 0.01, wl = 200, gr = FALSE,
 orientation = "v", labels = c("sound.files", "selec"), legend = 0, 
 path = tempdir())
 
 #different time scales and tag palette
catalog(X = lbh_selec_table, flim = c(1, 10), nrow = 4, ncol = 2, same.time.scale = F,
 ovlp = 90, parallel = 1, mar = 0.01, wl = 200, 
 orientation = "v",  labels = c("sound.files", "selec"), legend = 0, 
 tag.pal = list(terrain.colors), 
 path = tempdir())
 
 #adding tags and changing spectro palette
catalog(X = lbh_selec_table, flim = c(1, 10), nrow = 4, ncol = 2, same.time.scale = F,
 ovlp = 90, parallel = 1, mar = 0.01, wl = 200, pal = reverse.heat.colors,
 orientation = "v",  labels = c("sound.files", "selec"), legend = 1, 
 tag.pal = list(terrain.colors), tags = "sound.files", 
 path = tempdir())

 #create a bigger selection table
 X &lt;- rbind(lbh_selec_table, lbh_selec_table, lbh_selec_table, lbh_selec_table)
 X &lt;- rbind(X, X)
 
 #create some simulated labels
 X$songtype &lt;- sample(letters[13:15], nrow(X), replace = T)
 X$indiv &lt;- sample(letters[1:12], nrow(X), replace = T)

# 12 columns in 5 rows, 2 tags
catalog(X = X, flim = c(1, 10), nrow = 5, ncol = 12, same.time.scale = F,
 ovlp = 90, parallel = 1, mar = 0.01, wl = 200, 
 orientation = "v",  labels = c("sound.files", "selec"), legend = 3, 
 collevels = seq(-65, 0, 5), tag.pal = list(terrain.colors), tags = c("songtype", "indiv"), 
 path = tempdir())

# with legend
catalog(X = X, flim = c(1, 10), nrow = 5, ncol = 12, same.time.scale = F,
 ovlp = 90, parallel = 1, mar = 0.01, wl = 200, gr = FALSE,
 orientation = "v",  labels = c("sound.files", "selec"), legend = 3, 
 width = 20, collevels = seq(-65, 0, 5), tag.pal = list(terrain.colors),
  tags = c("songtype", "indiv"), 
  path = tempdir())
  
  # horizontal orientation
catalog(X = X, flim = c(1, 10), nrow = 5, ncol = 12, same.time.scale = F,
 ovlp = 90, parallel = 1, mar = 0.01, wl = 200, gr = FALSE,
 orientation = "h",  labels = c("sound.files", "selec"), legend = 3, 
 width = 20, collevels = seq(-65, 0, 5), tag.pal = list(terrain.colors),
  tags = c("songtype", "indiv"), 
  path = tempdir())

check this floder
tempdir()

## End(Not run)
</code></pre>

<hr>
<h2 id='catalog2pdf'>Combine <code><a href="#topic+catalog">catalog</a></code> images into pdfs</h2><span id='topic+catalog2pdf'></span>

<h3>Description</h3>

<p><code>catalog2pdf</code> combines <code><a href="#topic+catalog">catalog</a></code> jpeg images into pdfs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>catalog2pdf(keep.img = TRUE, overwrite = FALSE, parallel = 1, path = NULL,
pb = TRUE, by.img.suffix = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="catalog2pdf_+3A_keep.img">keep.img</code></td>
<td>
<p>Logical argument. Indicates whether jpeg files should be kept (default) or remove.</p>
</td></tr>
<tr><td><code id="catalog2pdf_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical argument. If <code>TRUE</code> all jpeg pdf will be produced again
when code is rerun. If <code>FALSE</code> only the ones missing will be produced. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="catalog2pdf_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="catalog2pdf_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the catalog image files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="catalog2pdf_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="catalog2pdf_+3A_by.img.suffix">by.img.suffix</code></td>
<td>
<p>Logical. If <code>TRUE</code> catalogs with the same image suffix will be
put together in a single pdf (so one pdf per image suffix in the catalog
images). Default is <code>FALSE</code> (i.e. no suffix).</p>
</td></tr>
<tr><td><code id="catalog2pdf_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal pdf creating function <code><a href="grDevices.html#topic+pdf">pdf</a></code> for customizing output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function combines catalog images in .jpeg format from the <code><a href="#topic+catalog">catalog</a></code> function into pdfs. Images must be saved in .jpeg format. Note that using lower resolution and smaller dimension (width and height) when creating catalogs will substantially decrease the size of pdf files (which could be pretty big).
</p>


<h3>Value</h3>

<p>Image files in pdf format with spectrogram catalogs in the working directory.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+full_spectrogram2pdf">full_spectrogram2pdf</a></code>, <code><a href="#topic+catalog">catalog</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# save sound file examples
data(list = c("Phae.long1", "Phae.long2"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

catalog(X = lbh_selec_table, nrow = 2, ncol = 4)

# now create single pdf removing jpeg
catalog2pdf(keep.img = FALSE, path = tempdir())

# check this floder
tempdir()

## End(Not run)
</code></pre>

<hr>
<h2 id='check_sels'>Check selection data frames</h2><span id='topic+check_sels'></span>

<h3>Description</h3>

<p><code>check_sels</code> checks whether selections can be read by subsequent functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_sels(X, parallel = 1, path = NULL, check.header = FALSE, pb = TRUE,
wav.size = FALSE, verbose = TRUE, fix.selec = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_sels_+3A_x">X</code></td>
<td>
<p>'selection_table' object or data frame with the following columns: 1) &quot;sound.files&quot;: name of the .wav
files, 2) &quot;sel&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections. Alternatively, a 'selection_table' class object can be input to double check selections.</p>
</td></tr>
<tr><td><code id="check_sels_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="check_sels_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="check_sels_+3A_check.header">check.header</code></td>
<td>
<p>Logical. Controls whether sound file headers correspond to the actual file properties
(i.e. if is corrupted). This could significantly affect the performance of the function (much slower) particularly
with long sound files.</p>
</td></tr>
<tr><td><code id="check_sels_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="check_sels_+3A_wav.size">wav.size</code></td>
<td>
<p>Logical argument to control if the size of the wave object
when the selection is imported into R (as when using <code><a href="tuneR.html#topic+readWave">readWave</a></code>
is calculated and added as a column. Size is return in MB. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="check_sels_+3A_verbose">verbose</code></td>
<td>
<p>Logical to control whether the summary messages are printed to the console. Defaut is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="check_sels_+3A_fix.selec">fix.selec</code></td>
<td>
<p>Logical to control if labels in 'selec' column should be fixed. This column should not be duplicated within a sound file. If that happens and <code>fix.selec = TRUE</code> duplicated labels will be changed. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function checks the information in a selection data frame or selection table (i.e. data frame with annotations on sound files)
to avoid problems in any warbleR analysis downstream. It specifically checks if:
</p>

<ul>
<li><p> 'X' is an object of class 'data.frame' or 'selection_table' (see <code><a href="#topic+selection_table">selection_table</a></code>) and contains
the required columns to be used on any warbleR function ('sound.files', 'selec', 'start', 'end', if not returns an error)
</p>
</li>
<li><p>  'sound.files' in 'X' correspond to sound files in the working directory or in the provided 'path'
(if no file is found returns an error, if some files are not found returns error info in the ouput data frame)
</p>
</li>
<li><p> time ('start', 'end') and frequency ('bottom.freq', 'top.freq', if provided) limit parameters are numeric and
don't contain NAs (if not returns an error)
</p>
</li>
<li><p> there are no duplicated selection labels ('selec') within a sound file (if not returns an error)
</p>
</li>
<li><p> sound files can be read (error info in the ouput data frame)
</p>
</li>
<li><p> the start and end time of the selections are found within the duration of the sound files (error info in the ouput data frame)
</p>
</li>
<li><p> sound files can be read (error info in the ouput data frame)
</p>
</li>
<li><p> sound files header is not corrupted (only if <code>header = TRUE</code>, error info in the ouput data frame)
</p>
</li>
<li><p> selection time position (start and end) doesn't exceeds sound file length (error info in the ouput data frame)
</p>
</li>
<li><p> 'top.freq' is lower than half the sample rate (nyquist frequency, error info in the ouput data frame)
</p>
</li>
<li><p> negative values aren't found in time or frequency limit parameters (error info in the ouput data frame)
</p>
</li>
<li><p> 'start' higher than 'end' or 'bottom.freq' higher than 'top.freq' (error info in the ouput data frame)
</p>
</li>
<li><p> 'channel' value is not higher than number of channels in sound files (error info in the ouput data frame)
</p>
</li></ul>

<p>The function returns a data frame that includes the information in 'X' plus additional columns about the format of sound
files (see 'Value') as well as the result of the checks ('check.res' column, value is 'OK' if everything is fine).
Sound files should be in the working directory (or the directory provided in 'path'). Corrupt files can be fixed using
<code><a href="#topic+fix_wavs">fix_wavs</a></code>.
</p>


<h3>Value</h3>

<p>A data frame including the columns in the input data frame (X) and the following additional columns:
</p>

<ul>
<li> <p><code>check.res</code>: diagnose for each selection
</p>
</li>
<li> <p><code>duration</code>: duration of selection in seconds
</p>
</li>
<li> <p><code>min.n.samples</code> number of samples in a selection. Note the number of samples available
in a selection limits the minimum window length (wl argument in other functions) that can be used in batch analyses.
</p>
</li>
<li> <p><code>sample.rate</code>: sampling rate in kHz
</p>
</li>
<li> <p><code>channels</code>: number of channels
</p>
</li>
<li> <p><code>bits</code>: bit depth
</p>
</li>
<li> <p><code>sound.file.samples</code>: number of samples in the sound file
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+check_wavs">check_wavs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# save wav file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

check_sels(X = lbh_selec_table, path = tempdir())
}
</code></pre>

<hr>
<h2 id='check_sound_files'>Check sound files</h2><span id='topic+check_sound_files'></span>

<h3>Description</h3>

<p><code>check_sound_files</code> checks whether sound files can be read by subsequent functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_sound_files(X = NULL, path = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_sound_files_+3A_x">X</code></td>
<td>
<p>Optional. 'selection_table' object or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;sel&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections. If provided the function also returns the
smallest number of samples from the listed selections, which limits the minimum window
length (wl argument in other functions) that can be used in batch analyses.
This could be useful for avoiding errors in downstream functions (e.g. <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>).</p>
</td></tr>
<tr><td><code id="check_sound_files_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function checks if sound files in the working directory can be read.
Users must set the working directory where they wish to check sound files beforehand.
If X is provided it also returns the smallest number of samples from
the selections listed in X (if all files can be read). Note that corrupt files can be
fixed using <code><a href="#topic+fix_wavs">fix_wavs</a></code>) ('sox' must be installed to be able to run this function).
The function is intended for a &quot;quick and dirty&quot; check of the sound files in a selections data
frame. For a more thorough analysis see <code><a href="#topic+check_sels">check_sels</a></code>.
</p>


<h3>Value</h3>

<p>If all sound files are ok, returns message &quot;All files can be read&quot;.
Otherwise returns the names of the corrupted sound files.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+check_sels">check_sels</a></code> <code><a href="#topic+tailor_sels">tailor_sels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# save wav file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# without selection data frame
check_sound_files(path = tempdir())

# with selection data frame
check_sound_files(X = lbh_selec_table, path = tempdir())
}
</code></pre>

<hr>
<h2 id='check_wavs'>alternative name for <code><a href="#topic+check_sound_files">check_sound_files</a></code></h2><span id='topic+check_wavs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+check_sound_files">check_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_wavs(X = NULL, path = NULL)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+check_sound_files">check_sound_files</a></code> for documentation. <code><a href="#topic+check_wavs">check_wavs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='checksels'>alternative name for <code><a href="#topic+check_sels">check_sels</a></code></h2><span id='topic+checksels'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+check_sels">check_sels</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checksels(
  X = NULL,
  parallel = 1,
  path = NULL,
  check.header = FALSE,
  pb = TRUE,
  wav.size = FALSE,
  verbose = TRUE,
  fix.selec = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+check_sels">check_sels</a></code> for documentation. <code><a href="#topic+checksels">checksels</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='color_spectro'>Highlight spectrogram regions</h2><span id='topic+color_spectro'></span>

<h3>Description</h3>

<p><code>color_spectro</code> highlights spectrogram regions specified by users
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_spectro(wave, wl = 512, wn = "hanning", ovlp = 70,
dB = "max0", collevels = NULL, selec.col = "red2", col.clm = NULL,
base.col = "black", bg.col = "white", strength = 1,
cexlab = 1, cexaxis = 1, tlab = "Time (s)", flab = "Frequency (kHz)",
title = NULL, axisX = TRUE, axisY = TRUE, flim = NULL,
rm.zero = FALSE, X = NULL, fast.spec = FALSE, t.mar = NULL,
f.mar = NULL, interactive = NULL, add = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="color_spectro_+3A_wave">wave</code></td>
<td>
<p>A 'wave' object produced by  <code><a href="tuneR.html#topic+readWave">readWave</a></code> or similar functions.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram. Default
is 512.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying the percent overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_db">dB</code></td>
<td>
<p>Character vector of length 1 controlling the amplitude weights as in
<code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 'max0'.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_collevels">collevels</code></td>
<td>
<p>Numeric. Levels used to partition amplitude range as in <code><a href="seewave.html#topic+spectro">spectro</a></code>.
Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_selec.col">selec.col</code></td>
<td>
<p>Character vector of length 1 specifying the color to be used to highlight selection.
See 'col.clm' for specifying unique colors for each selection. Default is 'red2'. Ignored if 'col.cm'
and 'X' are provided.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_col.clm">col.clm</code></td>
<td>
<p>Character vector of length 1 indicating the name of the column in 'X' that contains the
color names for each selection. Ignored if <code>X == NULL</code> or <code>interactive != NULL</code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_base.col">base.col</code></td>
<td>
<p>Character vector of length 1 specifying the color of the background spectrogram.
Default is 'black'.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_bg.col">bg.col</code></td>
<td>
<p>Character vector of length 1 specifying the background color for both base
and highlighted spectrograms. Default is 'white'.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_strength">strength</code></td>
<td>
<p>Numeric vector of length 1 controlling the strength of the highlighting color (actually how many times it is repeated in the internal color palette). Must be a positive integer. Default is 1.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_cexlab">cexlab</code></td>
<td>
<p>Numeric vector of length 1 specifying the relative size of axis
labels. See <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 1.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_cexaxis">cexaxis</code></td>
<td>
<p>Numeric vector of length 1 specifying the relative size of axis. See
<code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 1.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_tlab">tlab</code></td>
<td>
<p>Character vector of length 1 specifying the label of the time axis.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_flab">flab</code></td>
<td>
<p>Character vector of length 1 specifying the label of the frequency axis.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_title">title</code></td>
<td>
<p>Logical argument to add a title to individual spectrograms.
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_axisx">axisX</code></td>
<td>
<p>Logical to control whether time axis is plotted. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_axisy">axisY</code></td>
<td>
<p>Logical to control whether frequency axis is plotted. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit (in kHz) of
the spectrogram, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_rm.zero">rm.zero</code></td>
<td>
<p>Logical indicated if the 0 at the start of the time axis should be removed. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_x">X</code></td>
<td>
<p>Optional. Data frame containing columns for start and end time of signals ('start' and 'end') and low and high frequency ('bottom.freq' and 'top.freq').</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially
increases performance (much faster), although some options become unavailable, as collevels, and sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_t.mar">t.mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the start and end points to be added when highlighting selection. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_f.mar">f.mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the low and high frequencies to be added when highlighting selection. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_interactive">interactive</code></td>
<td>
<p>Numeric. Allow user to interactively select the signals to be highlighted by clicking
on the graphic device. Users must select the opposite corners of a square delimiting the spectrogram region
to be highlighted. Controls the number of signals that users would be able to select (2 clicks per signal).</p>
</td></tr>
<tr><td><code id="color_spectro_+3A_add">add</code></td>
<td>
<p>Logical. If <code>TRUE</code> new highlighting can be applied to the current plot (which means
that the function with <code>add = FALSE</code> should be run first). Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function highlights regions of the spectrogram with different colors. The regions to be
highlighted can be provided in a selection table (as the example data 'lbh_selec_table') or interactively ('interactive' argument).
</p>


<h3>Value</h3>

<p>A plot is produced in the graphic device.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for creating spectrograms to visualize
frequency measurements by <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>, <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> for
creating spectrograms to optimize noise margins used in <code><a href="#topic+sig2noise">sig2noise</a></code>
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+freq_DTW">freq_DTW</a>()</code>,
<code><a href="#topic+multi_DTW">multi_DTW</a>()</code>,
<code><a href="#topic+phylo_spectro">phylo_spectro</a>()</code>,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a>()</code>,
<code><a href="#topic+spectrograms">spectrograms</a>()</code>,
<code><a href="#topic+track_freq_contour">track_freq_contour</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(list = c("Phae.long1", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound files

# subset selection table
st &lt;- lbh_selec_table[lbh_selec_table$sound.files == "Phae.long1.wav", ]

# read wave file as an R object
sgnl &lt;- tuneR::readWave(file.path(tempdir(), st$sound.files[1]))

# create color column
st$colors &lt;- c("red2", "blue", "green")

# highlight selections
color_spectro(
  wave = sgnl, wl = 300, ovlp = 90, flim = c(1, 8.6), collevels = seq(-40, 0, 5),
  dB = "B", X = st, col.clm = "colors", base.col = "skyblue", t.mar = 0.07, f.mar = 0.1,
  interactive = NULL
)

# interactive (selected manually: you have to select them by clicking on the spectrogram)
color_spectro(
  wave = sgnl, wl = 300, ovlp = 90, flim = c(1, 8.6), collevels = seq(-40, 0, 5),
  dB = "B", col.clm = "colors", t.mar = 0.07, f.mar = 1, interactive = 2
)

## End(Not run)

</code></pre>

<hr>
<h2 id='color.spectro'>alternative name for <code><a href="#topic+color_spectro">color_spectro</a></code></h2><span id='topic+color.spectro'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+color_spectro">color_spectro</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color.spectro(
  wave,
  wl = 512,
  wn = "hanning",
  ovlp = 70,
  dB = "max0",
  collevels = NULL,
  selec.col = "red2",
  col.clm = NULL,
  base.col = "black",
  bg.col = "white",
  strength = 1,
  cexlab = 1,
  cexaxis = 1,
  tlab = "Time (s)",
  flab = "Frequency (kHz)",
  title = NULL,
  axisX = TRUE,
  axisY = TRUE,
  flim = NULL,
  rm.zero = FALSE,
  X = NULL,
  fast.spec = FALSE,
  t.mar = NULL,
  f.mar = NULL,
  interactive = NULL,
  add = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+color_spectro">color_spectro</a></code> for documentation. <code><a href="#topic+color.spectro">color.spectro</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='comp_matrix'>Example matrix listing selections to be compared by <code><a href="#topic+cross_correlation">cross_correlation</a></code></h2><span id='topic+comp_matrix'></span>

<h3>Description</h3>

<p><code>comp_matrix</code> is a character matrix with 2 columns indicating the selections to be compared (column 1 vs column 2) by <code><a href="#topic+cross_correlation">cross_correlation</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(comp_matrix)
</code></pre>


<h3>Format</h3>

<p>A data frame with 11 rows and 6 variables: </p>

<dl>
<dt>sound.files</dt><dd><p>recording names</p>
</dd>
<dt>channel</dt><dd><p>channel in which signal is found</p>
</dd>
<dt>selec</dt><dd><p>selection numbers within recording</p>
</dd>
<dt>start</dt><dd><p>start times of selected signal</p>
</dd>
<dt>end</dt><dd><p>end times of selected signal</p>
</dd>
<dt>bottom.freq</dt><dd><p>lower limit of frequency range</p>
</dd>
<dt>top.freq</dt><dd><p>upper limit of frequency range</p>
</dd>
</dl>



<h3>Details</h3>

<p>A character matrix with 2 columns indicating the selections to be compared (column 1 vs column 2) by <code><a href="#topic+cross_correlation">cross_correlation</a></code>. The first column contain the ID of the selection, which is given by combining the 'sound.files' and 'selec' columns of 'X', separated by '-' (i.e. <code>paste(X$sound.files, X$selec, sep = "-")</code>). The selection id's refer to those on the example data &quot;lbh_selec_table&quot;. The second column refers to the sound files in which to search for the templates.
</p>


<h3>Source</h3>

<p>Marcelo Araya Salas, warbleR
</p>

<hr>
<h2 id='compare_methods'>Assessing the performance of acoustic distance measurements</h2><span id='topic+compare_methods'></span>

<h3>Description</h3>

<p><code>compare_methods</code> creates graphs to visually assess performance of acoustic distance measurements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_methods(X = NULL, flim = NULL, bp = NULL, mar = 0.1, wl = 512, ovlp = 90,
res = 150, n = 10, length.out = 30,
methods = NULL,
it = "jpeg", parallel = 1, path = NULL, sp = NULL, custom1 = NULL,
custom2 = NULL, pb = TRUE, grid = TRUE,  clip.edges = TRUE,
threshold = 15, na.rm = FALSE, scale = FALSE, pal = reverse.gray.colors.2,
img = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_methods_+3A_x">X</code></td>
<td>
<p>'selection_table' object or data frame with results from <code><a href="#topic+auto_detec">auto_detec</a></code>
function, or any data frame with columns for sound file name (sound.files),
selection number (selec), and start and end time of signal (start and end).
Default <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit in kHz of
the spectrogram, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_bp">bp</code></td>
<td>
<p>numeric vector of length 2 giving the lower and upper limits of the
frequency bandpass filter (in kHz) used in the acoustic distance methods. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies plot margins around selection in seconds. Default is 0.1.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram and cross-correlation, default
is 512.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying the percent overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 90.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution.
Default is 150.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_n">n</code></td>
<td>
<p>Numeric argument of length 1. Defines the number of plots to be produce.
Default is 10.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_length.out">length.out</code></td>
<td>
<p>A character vector of length 1 giving the number of measurements of fundamental or dominant
frequency desired (the length of the time series). Default is 30.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_methods">methods</code></td>
<td>
<p>A character vector of length 2 giving the names of the acoustic distance
methods that would be compared. The methods available are:
</p>

<ul>
<li> <p><code>XCORR</code>: cross-correlation (<code><a href="#topic+cross_correlation">cross_correlation</a></code> function)
</p>
</li>
<li> <p><code>dfDTW</code>: dynamic time warping on dominant frequency contours (<code><a href="#topic+freq_DTW">freq_DTW</a></code> function)
</p>
</li>
<li> <p><code>ffDTW</code>: dynamic time warping on fundamental frequency contours (<code><a href="#topic+freq_DTW">freq_DTW</a></code> function)
</p>
</li>
<li> <p><code>SP</code>: spectral parameters (<code><a href="#topic+spectro_analysis">spectro_analysis</a></code> function)
</p>
</li>
<li> <p><code>SPharm</code>: spectral parameters (<code><a href="#topic+spectro_analysis">spectro_analysis</a></code> function with argument <code>harmonicity  = TRUE</code>)
</p>
</li>
<li> <p><code>MFCC</code>: statistical descriptors of Mel frequency cepstral coefficients (<code><a href="#topic+mfcc_stats">mfcc_stats</a></code> function)
</p>
</li></ul>

<p>Default <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_sp">sp</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_custom1">custom1</code></td>
<td>
<p>Data frame containing user parameters. The data frame must have 4 columns: the first 2 columns are 'sound.files'
and &quot;selec' columns as in 'X', the other 2 (columns 3 and 4) are
2 numeric columns to be used as the 2 parameters representing custom measurements. If the data has more than 2 parameters try using PCA (i.e. <code><a href="stats.html#topic+prcomp">prcomp</a></code> function)to summarize it in 2 dimensions before using it as an input. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_custom2">custom2</code></td>
<td>
<p>Data frame containing user parameters with the same format as 'custom1'. 'custom1' must be provided first. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_grid">grid</code></td>
<td>
<p>Logical argument to control the presence of a grid on the spectrograms (default is <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_clip.edges">clip.edges</code></td>
<td>
<p>Logical argument to control whether edges (start or end of signal) in
which amplitude values above the threshold were not detected will be removed when using dfDTW and
ffDTW methods. If <code>TRUE</code> this edges will be excluded and signal contour will be calculated on the
remaining values. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_threshold">threshold</code></td>
<td>
<p>amplitude threshold (%) for dominant and/or fundamental frequency detection when using dfDTW, ffDTW
and SP methods. Default is 15.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_na.rm">na.rm</code></td>
<td>
<p>Logical. If <code>TRUE</code> all NAs produced when pairwise cross-correlations failed are removed from the
results. This means that all selections with at least 1 cross-correlation that failed are excluded in both methods under
comparison. Only apply if XCORR is one of the methods being compared.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_scale">scale</code></td>
<td>
<p>Logical. If <code>TRUE</code> dominant and/or fundamental frequency values are z-transformed using the <code><a href="base.html#topic+scale">scale</a></code> function, which &quot;ignores&quot; differences in absolute frequencies between the signals in order to focus the
comparison in the frequency contour, regardless of the pitch of signals. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_pal">pal</code></td>
<td>
<p>A color palette function to be used to assign colors in the
spectrograms, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is reverse.gray.colors.2.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_img">img</code></td>
<td>
<p>A logical argument specifying whether an image files would be produced. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="compare_methods_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to a modified version of <code><a href="seewave.html#topic+spectro">spectro</a></code> for customizing
graphical output. This includes fast.spec, an argument that speeds up the plotting of spectrograms (see description in
<code><a href="#topic+spectrograms">spectrograms</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces graphs with spectrograms from 4 signals in the
provided data frame that allow visual inspection of the performance of acoustic
distance methods at comparing those signals. The signals are randomly picked up
from the provided data frame (X argument).The spectrograms are all plotted with
the same frequency and time scales. The function compares 2 methods at a time. The
methods available are: cross-correlation
(XCORR, from <code><a href="#topic+cross_correlation">cross_correlation</a></code>), dynamic time warping on dominant frequency time
series (dfDTW, from <code><a href="dtw.html#topic+dtw">dtw</a></code> applied on <code><a href="#topic+freq_ts">freq_ts</a></code> output), dynamic time
warping on dominant frequency time series (ffDTW, from <code><a href="dtw.html#topic+dtw">dtw</a></code> applied on
<code><a href="#topic+freq_ts">freq_ts</a></code> output), spectral parameters (SP, from <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>). The graph also
contains 2 scatterplots (1 for each method) of the acoustic space of all signals in the
input data frame 'X', including the centroid as black dot. The compared selections are randomly picked up from the pool of
selections in the input data frame. The argument 'n' defines the number of comparisons (i.e. graphs)
to be produced. The acoustic pairwise distance between signals is shown next
to the arrows linking them. The font color of a distance value correspond to the font
color of the method that generated it, as shown in the scatterplots. Distances are
standardized, being 0 the distance of a signal to itself and 1 the farthest pairwise
distance in the pool of signals. Principal Component Analysis (<code><a href="stats.html#topic+prcomp">prcomp</a></code>)
is applied to calculate distances when using spectral parameters (SP) and descriptors of cepstral coefficients (MFCC). In those cases the first 2 PC's are used. Classical
Multidimensional Scalling (also known as Principal Coordinates Analysis,
(<code><a href="stats.html#topic+cmdscale">cmdscale</a></code>)) is used for cross-correlation (XCORR) and any dynamic time warping method. The graphs are return as image files in the
working directory. The file name contains the methods being compared and the
row number of the selections. This function uses internally a modified version
of the <code><a href="seewave.html#topic+spectro">spectro</a></code> function from seewave package to create spectrograms. Custom data can also be compared against the available methods (or against each other) using the arguments 'custom1' and 'custom2'.
</p>


<h3>Value</h3>

<p>Image files with 4 spectrograms of the selection being compared and scatterplots
of the acoustic space of all signals in the input data frame 'X'.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>). It uses
internally a modified version of the <code><a href="seewave.html#topic+spectro">spectro</a></code> function from
seewave package to create spectrograms.
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+catalog">catalog</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Save to temporary working directory
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

compare_methods(
  X = lbh_selec_table, flim = c(0, 10), bp = c(0, 10), mar = 0.1, wl = 300,
  ovlp = 90, res = 200, n = 10, length.out = 30,
  methods = c("XCORR", "dfDTW"), parallel = 1, it = "jpeg", path = tempdir()
)

# remove progress bar
compare_methods(
  X = lbh_selec_table, flim = c(0, 10), bp = c(0, 10), mar = 0.1, wl = 300,
  ovlp = 90, res = 200, n = 10, length.out = 30,
  methods = c("XCORR", "dfDTW"), parallel = 1, it = "jpeg", pb = FALSE, path = tempdir()
)

# check this folder!
getwd()


# compare SP and XCORR
compare_methods(
  X = lbh_selec_table, flim = c(0, 10), bp = c(0, 10), mar = 0.1, wl = 300,
  ovlp = 90, res = 200, n = 10, length.out = 30,
  methods = c("XCORR", "SP"), parallel = 1, it = "jpeg", path = tempdir()
)

# compare SP method against dfDTW
compare_methods(
  X = lbh_selec_table, flim = c(0, 10), bp = c(0, 10), mar = 0.1, wl = 300,
  ovlp = 90, res = 200, n = 10, length.out = 30,
  methods = c("dfDTW", "SP"), parallel = 1, it = "jpeg",
  path = tempdir()
)

# alternatively we can provide our own SP matrix
Y &lt;- spectro_analysis(lbh_selec_table, path = tempdir())

# selec a subset of variables
Y &lt;- Y[, 1:7]

# PCA
Y &lt;- prcomp(Y[, 3:ncol(Y)])$x

# add sound files and selec columns
Y &lt;- data.frame(lbh_selec_table[, c(1, 3)], Y[, 1:2])

compare_methods(
  X = lbh_selec_table, methods = c("dfDTW"), custom1 = Y,
  path = tempdir()
)

## End(Not run)

</code></pre>

<hr>
<h2 id='consolidate'>Consolidate (sound) files into a single directory</h2><span id='topic+consolidate'></span>

<h3>Description</h3>

<p><code>consolidate</code> copies (sound) files scattered in several directories into a single one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consolidate(files = NULL, path = NULL, dest.path = NULL, pb = TRUE, file.ext = ".wav$",
parallel = 1, save.csv = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="consolidate_+3A_files">files</code></td>
<td>
<p>character vector indicating the subset of files that will be consolidated. File names should include the full file path. Optional.</p>
</td></tr>
<tr><td><code id="consolidate_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
'wav.path' set by <code><a href="#topic+warbleR_options">warbleR_options</a></code> is ignored.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="consolidate_+3A_dest.path">dest.path</code></td>
<td>
<p>Character string containing the directory path where the sound files will be saved.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="consolidate_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="consolidate_+3A_file.ext">file.ext</code></td>
<td>
<p>Character string defining the file extension (i.e. format) for the files to be consolidated. Default is <code>'.wav$'</code> ignoring case. Several formats can be used: <code>"\.wav$|\.wac$|\.mp3$|\.flac$"</code>.</p>
</td></tr>
<tr><td><code id="consolidate_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="consolidate_+3A_save.csv">save.csv</code></td>
<td>
<p>Logical. Controls whether a data frame containing sound file information is saved in the new directory. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="consolidate_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal <code><a href="base.html#topic+files">file.copy</a></code> function for customizing file copying.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allows users to put files scattered in several directories into a
single one. By default it works on sound files in '.wav' format but can work with
other type of files (for instance '.txt' selection files).
</p>


<h3>Value</h3>

<p>All (sound) files are consolidated (copied) to a single directory (&quot;consolidated_files&quot;). The function returns a data frame with each of the files that were copied in a row and the following information:
</p>

<ul>
<li> <p><code>original_dir</code> the path to the original file
</p>
</li>
<li> <p><code>old_name</code> the name of the original file
</p>
</li>
<li> <p><code>new_name</code> the name of the new file. This will be the same as 'old_name' if the name was not duplicated (i.e. no files in other directories with the same name).
</p>
</li>
<li> <p><code>file_size_bytes</code> size of the file in bytes.
</p>
</li>
<li> <p><code>duplicate</code> indicates whether a file is likely to be duplicated (i.e. if files with the same name were found in other directories). If so it will be labeled as 'possible.dupl', otherwise it will contain NAs.
</p>
</li></ul>

<p>If <code>csv = TRUE</code> (default)
a 'file_names_info.csv' file with the same information as the output data frame is also saved in the consolidated directory.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fix_wavs">fix_wavs</a></code> for making sound files readable in R
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# save wav file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))

# create first folder with 2 sound files
dir.create(file.path(tempdir(), "folder1"))
writeWave(Phae.long1, file.path(tempdir(), "folder1", "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "folder1", "Phae.long2.wav"))

# create second folder with 2 sound files
dir.create(file.path(tempdir(), "folder2"))
writeWave(Phae.long3, file.path(tempdir(), "folder2", "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "folder2", "Phae.long4.wav"))

# consolidate in a single folder
# consolidate(path = tempdir(), dest.path = tempdir())

# check this folder
tempdir()
}

</code></pre>

<hr>
<h2 id='coor.graph'>alternative name for <code><a href="#topic+plot_coordination">plot_coordination</a></code></h2><span id='topic+coor.graph'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+plot_coordination">plot_coordination</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coor.graph(
  X = NULL,
  only.coor = FALSE,
  ovlp = TRUE,
  xl = 1,
  res = 80,
  it = "jpeg",
  img = TRUE,
  tlim = NULL,
  pb = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+plot_coordination">plot_coordination</a></code> for documentation. <code><a href="#topic+coor.graph">coor.graph</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='coor.test'>alternative name for <code><a href="#topic+test_coordination">test_coordination</a></code></h2><span id='topic+coor.test'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+test_coordination">test_coordination</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coor.test(
  X = NULL,
  iterations = 1000,
  ovlp.method = "count",
  randomization = "keep.gaps",
  less.than.chance = TRUE,
  parallel = 1,
  pb = TRUE,
  rm.incomp = FALSE,
  cutoff = 2,
  rm.solo = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+test_coordination">test_coordination</a></code> for documentation. <code><a href="#topic+coor.test">coor.test</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='cross_correlation'>Time-frequency cross-correlation</h2><span id='topic+cross_correlation'></span>

<h3>Description</h3>

<p><code>cross_correlation</code> estimates the similarity of two sound waves by means of time-frequency cross-correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cross_correlation(X, wl = 512, bp = "pairwise.freq.range", ovlp = 70,
dens = NULL, wn = 'hanning', cor.method = "pearson", parallel = 1,
path = NULL, pb = TRUE, na.rm = FALSE, cor.mat = NULL, output = "cor.mat",
templates = NULL, surveys = NULL, compare.matrix = NULL, type = "fourier",
 nbands = 40, method = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cross_correlation_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame containing columns for sound files (sound.files),
selection number (selec), and start and end time of signal (start and end).
All selections must have the same sampling rate.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz). If columns for bottom and top frequency ('bottom.freq' and 'top.freq') are supplied &quot;pairwise.freq.range&quot; ca be used (default). If so, the lowest values in 'bottom.freq'
and the highest values in 'top.freq' for the selections involved in a pairwise comparison will be used as bandpass limits.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70. High values of ovlp
slow down the function but produce more accurate results.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_dens">dens</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_wn">wn</code></td>
<td>
<p>A character vector of length 1 specifying the window name as in <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code>.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_cor.method">cor.method</code></td>
<td>
<p>A character vector of length 1 specifying the correlation method as in <code><a href="stats.html#topic+cor">cor</a></code>.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_na.rm">na.rm</code></td>
<td>
<p>Logical. If <code>TRUE</code> all NAs produced when pairwise cross-correlations failed are removed from the
results. This means that all selections with at least 1 cross-correlation that failed are excluded.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_cor.mat">cor.mat</code></td>
<td>
<p>DEPRECATED. Use 'compare.matrix' instead.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_output">output</code></td>
<td>
<p>Character vector of length 1 to determine if only the correlation matrix is returned ('cor.mat', default) or a list ('list') containing 1) the correlation matrix and 2) a data frame with correlation values at each sliding step for each comparison. The list, which is also of class 'xcorr.output', can be used to find detection peaks with <code><a href="#topic+find_peaks">find_peaks</a></code> or to graphically explore detections using <code><a href="#topic+lspec">lspec</a></code>.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_templates">templates</code></td>
<td>
<p>Character vector with the selections in 'X' to be used as templates for cross-correlation detection. To refer to specific selections in 'X' the user must use the format &quot;sound.file-selec&quot; (e.g. &quot;file1.wav-1&quot;). If only the sound file name is included then the entire sound file is used as template.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_surveys">surveys</code></td>
<td>
<p>Character vector with the selections in 'X' to be used as surveys for cross-correlation detection. To refer to specific selections in 'X' the user must use the format &quot;sound.file-selec&quot; (e.g. &quot;file1.wav-1&quot;). If only the sound file name is included then the entire sound file is used as survey.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_compare.matrix">compare.matrix</code></td>
<td>
<p>A character matrix with 2 columns indicating the selections to be compared (column 1 vs column 2). The columns must contained the ID of the selection, which is given by combining the 'sound.files' and 'selec' columns of 'X',  separated by '-' (i.e. <code>paste(X$sound.files, X$selec, sep = "-")</code>). Default is <code>NULL</code>. If the 'selec' ID is not supplied then the entire sound file is used instead. If 'compare.matrix' is supplied only those comparisons will be calculated (as opposed to all pairwise comparisons as the default behavior) and the output will be a data frame composed of the supplied matrix and the correspondent cross-correlation values. Note that 'method' is automatically set to 2 (create spectrograms on the fly) when 'compare.matrix' is supplied but can be set back to 1. When supplied 'output' is forced to be 'list'. Note that the use of this function for signal detection (with 'compare.matrix') will be deprecated in future warbleR versions. Look at the ohun package for automatic signal detection functions.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_type">type</code></td>
<td>
<p>A character vector of length 1 specifying the type of cross-correlation; either &quot;fourier&quot; (i.e. spectrographic cross-correlation using Fourier transform; internally using <code><a href="seewave.html#topic+spectro">spectro</a></code>; default) or &quot;mfcc&quot; (Mel cepstral coefficient spectrogram cross-correlation; internally using <code><a href="tuneR.html#topic+melfcc">melfcc</a></code>).</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_nbands">nbands</code></td>
<td>
<p>Numeric vector of length 1 controlling the number of warped spectral bands to calculate when using <code>type = "mfcc"</code> (see <code><a href="tuneR.html#topic+melfcc">melfcc</a></code>). Default is 40.</p>
</td></tr>
<tr><td><code id="cross_correlation_+3A_method">method</code></td>
<td>
<p>Numeric vector of length 1 to control the method used to create spectrogram matrices. Two option are available:
</p>

<ul>
<li> <p><code>1</code>:  matrices are created first (keeping them internally as a list) and cross-correlation is calculated on a second step. Note that this method may require lots of memory if selection and or sound files are large.
</p>
</li>
<li> <p><code>2</code>: matrices are created &quot;on the fly&quot; (i.e. at the same time that cross-correlation is calculated). More memory efficient but may require extracting the same matrix several times, which will affect performance. Note that when using this method the function does not check if sound files have the same sampling rate which if not, may produce an error.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the pairwise similarity of multiple signals by means of time-frequency cross-correlation. Fourier or Mel frequency ceptral coefficients spectrograms can be used as time-frequency representations of sound.
This method &quot;slides&quot; the spectrogram of the shortest selection over the longest one calculating a correlation of the amplitude values at each step.
The function runs pairwise cross-correlations on several signals and returns a list including the correlation statistic
for each &quot;sliding&quot; step as well as the maximum (peak) correlation for each pairwise comparison.
The correlation matrix could have NA's if some of the pairwise correlation did not work (common when sound files have been modified by band-pass filters). Make sure all sound files have the same sampling rate (can be checked with <code><a href="#topic+check_sels">check_sels</a></code> or <code><a href="#topic+check_sound_files">check_sound_files</a></code>).
</p>


<h3>Value</h3>

<p>If <code>output = "cor.mat"</code> the function returns a matrix with
the maximum (peak) correlation for each pairwise comparison (if 'compare.matrix' is not supplied) or the peak correlation for each comparison in the supplied 'compare.matrix'. Otherwise it will return a list that includes 1) a matrix with
the maximum correlation for each pairwise comparison ('max.xcorr.matrix') and 2) a data frame with the correlation statistic for each &quot;sliding&quot; step ('scores').
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas <a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>
<p>H. Khanna, S.L.L. Gaunt &amp; D.A. McCallum (1997). Digital spectrographic cross-correlation: tests of sensitivity. Bioacoustics 7(3): 209-234
</p>
<p>Lyon, R. H., &amp; Ordubadi, A. (1982). Use of cepstra in acoustical signal analysis. Journal of Mechanical Design, 104(2), 303-306.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+mfcc_stats">mfcc_stats</a></code>, <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>, <code><a href="#topic+freq_DTW">freq_DTW</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))

  # save sound files
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  # run cross correlation on spectrograms (SPCC)
  xcor &lt;- cross_correlation(X = lbh_selec_table, wl = 300, ovlp = 90, path = tempdir())

  # run cross correlation on Mel cepstral coefficients (mfccs)
  xcor &lt;- cross_correlation(
    X = lbh_selec_table, wl = 300, ovlp = 90, path = tempdir(),
    type = "mfcc"
  )

  # using the 'compare.matrix' argument to specify pairwise comparisons
  # create matrix with ID of signals to compare
  cmp.mt &lt;- cbind(
    paste(lbh_selec_table$sound.files[1:10], lbh_selec_table$selec[1:10], sep = "-"),
    paste(lbh_selec_table$sound.files[2:11], lbh_selec_table$selec[2:11], sep = "-")
  )

  # run cross-correlation on the selected pairwise comparisongs
  xcor &lt;- cross_correlation(
    X = lbh_selec_table, compare.matrix = cmp.mt,
    wl = 300, ovlp = 90, path = tempdir()
  )
}
</code></pre>

<hr>
<h2 id='cut_sels'>Cut selections into individual sound files</h2><span id='topic+cut_sels'></span>

<h3>Description</h3>

<p><code>cut_sels</code> cuts selections from a selection table into individual sound files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cut_sels(X, mar = 0.05, parallel = 1, path = NULL, dest.path = NULL, pb = TRUE,
labels = c("sound.files", "selec"), overwrite = FALSE, norm = FALSE,
keep.stereo = FALSE, ...)

cut_sels(
  X,
  mar = 0.05,
  parallel = 1,
  path = NULL,
  dest.path = NULL,
  pb = TRUE,
  labels = c("sound.files", "selec"),
  overwrite = FALSE,
  norm = FALSE,
  keep.stereo = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cut_sels_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data frame containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signals (start and end).</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the start and end points of selections,
delineating spectrogram limits. Default is 0.05.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_dest.path">dest.path</code></td>
<td>
<p>Character string containing the directory path where the cut sound files will be saved.
If <code>NULL</code> (default) then the directory containing the sound files will be used instead.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_labels">labels</code></td>
<td>
<p>String vector. Provides the column names that will be used as labels to
create sound file names. Note that they should provide unique names (otherwise
sound files will be overwritten). Default is <code>c("sound.files", "selec")</code>.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical. If <code>TRUE</code> sound files with the same name will be
overwritten. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_norm">norm</code></td>
<td>
<p>Logical indicating whether wave objects must be normalized first using the function <code><a href="tuneR.html#topic+normalize">normalize</a></code>. Additional arguments can be passed to <code><a href="tuneR.html#topic+normalize">normalize</a></code> using '...'.' Default is <code>FALSE</code>. See <code><a href="tuneR.html#topic+normalize">normalize</a></code> for available options.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_keep.stereo">keep.stereo</code></td>
<td>
<p>Logical. If <code>TRUE</code> both channels are kept in the clips, oterwise it will keep the channel referenced in the channel column (if supplied) or the first channel if a 'channel' column is not found in 'X'. Only applies to stereo (2-channel) files.</p>
</td></tr>
<tr><td><code id="cut_sels_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal <code><a href="tuneR.html#topic+normalize">normalize</a></code> function for customizing sound file output. Ignored if  <code>norm = FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allow users to produce individual sound files from the selections
listed in a selection table as in <code><a href="#topic+lbh_selec_table">lbh_selec_table</a></code>. Note that wave objects with a bit depth of 32 might not be readable by some programs after exporting. In this case they should be &quot;normalized&quot; (argument 'norm&quot;) with a lower bit depth. The function keeps the original number of channels in the output clips only for 1- and 2-channel files.
</p>
<p>see <code><a href="#topic+cut_sels">cut_sels</a></code> for documentation. <code><a href="#topic+cut_sels">cut_sels</a></code> will be deprecated in future versions.
</p>


<h3>Value</h3>

<p>Sound files of the signals listed in the input data frame.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+seltailor">seltailor</a></code> for tailoring selections
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# save wav file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# cut selections
cut_sels(lbh_selec_table, path = tempdir())

#check this folder!!
tempdir()
}

</code></pre>

<hr>
<h2 id='df_DTW'>alternative name for <code><a href="#topic+freq_DTW">freq_DTW</a></code></h2><span id='topic+df_DTW'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+freq_DTW">freq_DTW</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df_DTW(
  X = NULL,
  type = "dominant",
  wl = 512,
  wl.freq = 512,
  length.out = 20,
  wn = "hanning",
  ovlp = 70,
  bp = NULL,
  threshold = 15,
  threshold.time = NULL,
  threshold.freq = NULL,
  img = TRUE,
  parallel = 1,
  path = NULL,
  ts.df = NULL,
  img.suffix = "dfDTW",
  pb = TRUE,
  clip.edges = TRUE,
  window.type = "none",
  open.end = FALSE,
  scale = FALSE,
  frange.detec = FALSE,
  fsmooth = 0.1,
  adjust.wl = TRUE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+freq_DTW">freq_DTW</a></code> for documentation. <code><a href="#topic+freq_DTW">freq_DTW</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='dfDTW'>alternative name for <code><a href="#topic+freq_DTW">freq_DTW</a></code></h2><span id='topic+dfDTW'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+freq_DTW">freq_DTW</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfDTW(
  X = NULL,
  type = "dominant",
  wl = 512,
  wl.freq = 512,
  length.out = 20,
  wn = "hanning",
  ovlp = 70,
  bp = NULL,
  threshold = 15,
  threshold.time = NULL,
  threshold.freq = NULL,
  img = TRUE,
  parallel = 1,
  path = NULL,
  ts.df = NULL,
  img.suffix = "dfDTW",
  pb = TRUE,
  clip.edges = TRUE,
  window.type = "none",
  open.end = FALSE,
  scale = FALSE,
  frange.detec = FALSE,
  fsmooth = 0.1,
  adjust.wl = TRUE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+freq_DTW">freq_DTW</a></code> for documentation. <code><a href="#topic+dfDTW">dfDTW</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='duration_sound_files'>Measure the duration of sound files</h2><span id='topic+duration_sound_files'></span>

<h3>Description</h3>

<p><code>duration_sound_files</code> measures the duration of sound files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>duration_sound_files(files = NULL, path = NULL, skip.error = FALSE,
file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="duration_sound_files_+3A_files">files</code></td>
<td>
<p>Character vector with the names of the sound files to be measured. The sound files should be in the working directory or in the directory provided in 'path'.</p>
</td></tr>
<tr><td><code id="duration_sound_files_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.</p>
</td></tr>
<tr><td><code id="duration_sound_files_+3A_skip.error">skip.error</code></td>
<td>
<p>Logical to control if errors are omitted. If so, files that could not be read will return <code>NA</code> in the 'duration' column. Default is <code>FALSE</code>, which will return an error if some files are problematic.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="duration_sound_files_+3A_file.format">file.format</code></td>
<td>
<p>Character string with the format of sound files. By default all sound file formats supported by warbleR are included (&quot;\.wav$|\.wac$|\.mp3$|\.flac$&quot;). Note that several formats can be included using regular expression syntax as in <code><a href="base.html#topic+grep">grep</a></code>. For instance <code>"\.wav$|\.mp3$"</code> will only include .wav and .mp3 files.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns the duration (in seconds) of sound files.
</p>


<h3>Value</h3>

<p>A data frame with the duration (in seconds) of the sound files.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

  duration_sound_files(path = tempdir())
}

</code></pre>

<hr>
<h2 id='duration_wavs'>alternative name for <code><a href="#topic+duration_sound_files">duration_sound_files</a></code></h2><span id='topic+duration_wavs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+duration_sound_files">duration_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>duration_wavs(
  files = NULL,
  path = NULL,
  skip.error = FALSE,
  file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$"
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+duration_sound_files">duration_sound_files</a></code> for documentation. <code><a href="#topic+duration_wavs">duration_wavs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='envelope'>Calculates the absolute amplitude envelope</h2><span id='topic+envelope'></span>

<h3>Description</h3>

<p>Calculates the absolute amplitude envelope
</p>


<h3>Usage</h3>

<pre><code class='language-R'>envelope(x, ssmooth = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="envelope_+3A_x">x</code></td>
<td>
<p>Numeric vector with amplitude values. Required.</p>
</td></tr>
<tr><td><code id="envelope_+3A_ssmooth">ssmooth</code></td>
<td>
<p>Numeric vector of length 1 indicating the size of the sliding window use to smooth envelopes. Default is 0 (no smoothing).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates the absolute amplitude envelope of an amplitude vector using compiled C code which is usually several times faster.
</p>


<h3>Value</h3>

<p>An amplitude envelope.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) &amp; Paula Monge
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="seewave.html#topic+env">env</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data(tico)

amp_env &lt;- envelope(tico@left, ssmooth = 100)
}
</code></pre>

<hr>
<h2 id='ffDTW'>deprecated functions and function names</h2><span id='topic+ffDTW'></span>

<h3>Description</h3>

<p>deprecated functions and function names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ffDTW(...)
</code></pre>


<h3>Details</h3>

<p>functions that have been deprecated or moved to other packages.
</p>

<hr>
<h2 id='filter_sels'>Subset selection data frames based on manually filtered image files</h2><span id='topic+filter_sels'></span>

<h3>Description</h3>

<p><code>filter_sels</code> subsets selection data frames based on image files that have been manually filtered.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_sels(X, path = NULL, lspec = FALSE, img.suffix = NULL, it = "jpeg",
 incl.wav = TRUE, missing = FALSE, index = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_sels_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data frame with the following columns: 1) &quot;sound.files&quot;: name of the .wav
files, 2) &quot;sel&quot;: number of the selections.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the image files are located.
If <code>NULL</code> (default) then the current working directory is used.
<code><a href="#topic+warbleR_options">warbleR_options</a></code> 'wav.path' argument does not apply.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_lspec">lspec</code></td>
<td>
<p>A logical argument indicating if the image files to be use for filtering were produced by the function <code><a href="#topic+full_spectrograms">full_spectrograms</a></code>.
All the image files that correspond to a sound file must be deleted in order to be
filtered out.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_img.suffix">img.suffix</code></td>
<td>
<p>A character vector of length 1 with the suffix (label) at the end
of the names of the image files. Default is <code>NULL</code> (i.e. no suffix as in the images
produced by <code><a href="#topic+spectrograms">spectrograms</a></code>). Ignored if <code>lspec = TRUE</code>.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type (&quot;tiff&quot;, &quot;jpeg&quot; or &quot;pdf&quot;) Default is &quot;jpeg&quot;. Note that pdf files can only be generated by <code><a href="#topic+lspec2pdf">lspec2pdf</a></code>.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_incl.wav">incl.wav</code></td>
<td>
<p>Logical. To indicate if sound files extensions are included ( <code>TRUE</code>, default) or not in the image
file names.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_missing">missing</code></td>
<td>
<p>Logical. Controls whether the output data frame (or row index if is <code>index = TRUE</code>)
contains the selections with images in the working directory (Default, <code>missing = FALSE</code>)
or the ones with no image.</p>
</td></tr>
<tr><td><code id="filter_sels_+3A_index">index</code></td>
<td>
<p>Logical. If <code>TRUE</code> and <code>missing = FALSE</code> the row index for the selections with images in the working directory is returned. If <code>missing = TRUE</code>) then the row index of the ones with no image is returned instead. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function subsets selections (or sound files if <code>lspec</code> is <code>TRUE</code>) listed in a data frame
based on the image files from spectrogram-creating functions (e.g. <code><a href="#topic+spectrograms">spectrograms</a></code>) in the
working directory. Only the selections/sound files with and image in the working directory will remain.
This is useful for excluding selections from undesired signals. Note that the
image files should be in the working directory (or the directory provided in 'path').
</p>


<h3>Value</h3>

<p>If all sound files are ok, returns message &quot;All files are ok!&quot;.
Otherwise returns &quot;These file(s) cannot be read&quot; message with names of the
corrupted sound files.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# save wav file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

spectrograms(lbh_selec_table,
  flim = c(0, 11), inner.mar = c(4, 4.5, 2, 1), outer.mar = c(4, 2, 2, 1),
  picsize = 2, res = 300, cexlab = 2, mar = 0.05, wl = 300, path = tempdir()
)

# go to the working directory (tempdir()) and delete some images

# filter selection data frame
fmloc &lt;- filter_sels(X = lbh_selec_table, path = tempdir())

# this data frame does not have the selections corresponding to the images that were deleted
fmloc

# now using lspec images
full_spectrograms(
  sxrow = 2, rows = 8, pal = reverse.heat.colors, wl = 300, ovlp = 10,
  path = tempdir()
)

# go to the working directory (tempdir()) and delete lspec
# images (the ones with several rows of spectrograms)

# filter selection data frame
fmloc2 &lt;- filter_sels(
  X = lbh_selec_table, lspec = TRUE,
  path = tempdir()
)

## End(Not run)

</code></pre>

<hr>
<h2 id='filtersels'>alternative name for <code><a href="#topic+filter_sels">filter_sels</a></code></h2><span id='topic+filtersels'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+filter_sels">filter_sels</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filtersels(
  X,
  path = NULL,
  lspec = FALSE,
  img.suffix = NULL,
  it = "jpeg",
  incl.wav = TRUE,
  missing = FALSE,
  index = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+filter_sels">filter_sels</a></code> for documentation. <code><a href="#topic+filtersels">filtersels</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='find_clipping'>Find clipped selections</h2><span id='topic+find_clipping'></span>

<h3>Description</h3>

<p><code>find_clipping</code> gets the proportion of samples that are clipped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_clipping(X, path = NULL, parallel = 1, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_clipping_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;selec&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections.</p>
</td></tr>
<tr><td><code id="find_clipping_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="find_clipping_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="find_clipping_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Clipping (i.e. saturation) occurs when an audio signal is amplified above the maximum limit of the recorder. This leads to distortion and a lowering of audio quality. If stereo the mean proportion of both channels is returned. The function assumes specific range values for different bit depths as detailed in <code><a href="tuneR.html#topic+normalize">normalize</a></code>.
</p>


<h3>Value</h3>

<p>A data frame with the 'sound.files' and 'selec' columns in X plus an additional column ('prop.clipped') indicating
the proportion of clipped samples for each row. If sound files are stereo the average proportion of the two channels is returned.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+sig2noise">sig2noise</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data(list = c("Phae.long1", "Phae.long2", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound files
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

  find_clipping(X = lbh_selec_table[1:5, ], path = tempdir())
}
</code></pre>

<hr>
<h2 id='find_peaks'>Find cross-correlation peaks</h2><span id='topic+find_peaks'></span>

<h3>Description</h3>

<p><code>find_peaks</code> find peaks in cross-correlation scores from <code><a href="#topic+cross_correlation">cross_correlation</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_peaks(xc.output, parallel = 1, cutoff = 0.4, path = NULL, pb = TRUE,
max.peak = FALSE, output = "data.frame")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_peaks_+3A_xc.output">xc.output</code></td>
<td>
<p>output of <code><a href="#topic+cross_correlation">cross_correlation</a></code> after setting <code>output = "list"</code>.</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_cutoff">cutoff</code></td>
<td>
<p>Numeric vector of length 1 with a value between 0 and 1 specifying the correlation cutoff for detecting peaks. Default is 0.4.</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_max.peak">max.peak</code></td>
<td>
<p>Logical argument to control whether only the peak with the highest correlation value is returned (if TRUE; cutoff will be ignored). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="find_peaks_+3A_output">output</code></td>
<td>
<p>Character vector of length 1 to determine if only the detected peaks are returned ('cormat') or a list ('list') containing 1) the peaks  and 2) a data frame with correlation values at each sliding step for each comparison. The list, which is also of class 'peaks.output', can be used to graphically explore detections using <code><a href="#topic+full_spectrograms">full_spectrograms</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function finds cross-correlation peaks along signals (analogous to <code><a href="monitoR.html#topic+findPeaks">findPeaks</a></code>).
</p>


<h3>Value</h3>

<p>The function returns a data frame with time and correlation score for the
detected peaks.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas <a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>
<p>H. Khanna, S.L.L. Gaunt &amp; D.A. McCallum (1997). Digital spectrographic cross-correlation: tests of sensitivity. Bioacoustics 7(3): 209-234

</p>


<h3>See Also</h3>

<p><code><a href="#topic+auto_detec">auto_detec</a></code>, <code><a href="monitoR.html#topic+findPeaks">findPeaks</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data(list = c("Phae.long4", "Phae.long2", "lbh_selec_table2", "comp_matrix"))

  # save sound files
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

  # run cross-correlation
  xc.output &lt;- cross_correlation(
    X = lbh_selec_table2, output = "list",
    compare.matrix = comp_matrix, path = tempdir()
  )

  # find peaks
  pks &lt;- find_peaks(xc.output = xc.output, path = tempdir())
}
</code></pre>

<hr>
<h2 id='fix_extended_selection_table'>Fix extended selection tables</h2><span id='topic+fix_extended_selection_table'></span>

<h3>Description</h3>

<p><code>fix_extended_selection_table</code> fixes extended selection tables that have lost their attributes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fix_extended_selection_table(X, Y, to.by.song = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fix_extended_selection_table_+3A_x">X</code></td>
<td>
<p>an object of class 'selection_table' or data frame that contains columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end).</p>
</td></tr>
<tr><td><code id="fix_extended_selection_table_+3A_y">Y</code></td>
<td>
<p>an object of class 'extended_selection_table'</p>
</td></tr>
<tr><td><code id="fix_extended_selection_table_+3A_to.by.song">to.by.song</code></td>
<td>
<p>Logical argument to control if the attributes are formatted to a match a 'by.song' extended selection table. This is required when 'X' is created by collapsing an Y by song (see 'by.song' argument in <code><a href="#topic+selection_table">selection_table</a></code>). Mostly needed internally by some warbleR functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An extended selection table.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))

writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# create extended selection table
ext_st &lt;- selection_table(lbh_selec_table, extended = TRUE,
path = tempdir())

# remove attributes
st &lt;- as.data.frame(ext_st)

# check class
class(st)

# fix selection table
st &lt;- fix_extended_selection_table(X = st, Y = ext_st)

# check class
class(st)
}
</code></pre>

<hr>
<h2 id='fix_wavs'>Fix .wav files to allow importing them into R</h2><span id='topic+fix_wavs'></span>

<h3>Description</h3>

<p><code>fix_wavs</code> fixes sound files in .wav format so they can be imported into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fix_wavs(checksels = NULL, files = NULL, samp.rate = NULL, bit.depth = NULL,
 path = NULL, mono = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fix_wavs_+3A_checksels">checksels</code></td>
<td>
<p>Data frame with results from <code><a href="#topic+check_sels">check_sels</a></code>. Default is <code>NULL</code>. If  both 'checksels' and 'files' are <code>NULL</code>
then all files in 'path' are converted. Note that it only fixes/convert sound files in .wav format.</p>
</td></tr>
<tr><td><code id="fix_wavs_+3A_files">files</code></td>
<td>
<p>Character vector with the names of the .wav files to fix. Default is <code>NULL</code>. If  both 'checksels' and 'files'  are <code>NULL</code>
then all files in 'path' are converted.</p>
</td></tr>
<tr><td><code id="fix_wavs_+3A_samp.rate">samp.rate</code></td>
<td>
<p>Numeric vector of length 1 with the sampling rate (in kHz) for output files. Default is <code>NULL</code>.
(remain unchanged).</p>
</td></tr>
<tr><td><code id="fix_wavs_+3A_bit.depth">bit.depth</code></td>
<td>
<p>Numeric vector of length 1 with the dynamic interval (i.e. bit depth) for output files.
Default is <code>NULL</code> (remain unchanged).</p>
</td></tr>
<tr><td><code id="fix_wavs_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="fix_wavs_+3A_mono">mono</code></td>
<td>
<p>Logical indicating if stereo (2 channel) files should be converted to mono (1 channel). Default is <code>FALSE</code> (remain unchanged).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aims to simplify the process of converting sound files that cannot be imported into R and/or homogenizing sound files. Problematic files can be determined using <code><a href="#topic+check_wavs">check_wavs</a></code> or <code><a href="#topic+check_sels">check_sels</a></code>. The
<code><a href="#topic+check_sels">check_sels</a></code> output can be directly input using the argument 'checksels'. Alternatively a vector of file
names to be &quot;fixed&quot; can be provided (argument 'files'). If neither of those 2 are provided the function will convert
all .wav sound files in the working directory to the specified sample rate/bit depth. Files are saved in a new directory
('converted_sound_files'). Internally the function calls <a href="https://sourceforge.net/projects/sox/">SOX</a> (<a href="https://sourceforge.net/projects/sox/">SOX</a> must be installed). If  both 'checksels' and 'files' are <code>NULL</code>
then all files in 'path' are converted. Note that it only fixes/convert sound files in .wav format.
</p>


<h3>Value</h3>

<p>A folder inside the working directory (or path provided) all 'converted_sound_files', containing
sound files in a format that can be imported in R.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Load example files and save to temporary working directory
#

# check this folder
tempdir()

## End(Not run)

</code></pre>

<hr>
<h2 id='fixwavs'>alternative name for <code><a href="#topic+fix_wavs">fix_wavs</a></code></h2><span id='topic+fixwavs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+fix_wavs">fix_wavs</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fixwavs(
  checksels = NULL,
  files = NULL,
  samp.rate = NULL,
  bit.depth = NULL,
  path = NULL,
  mono = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+fix_wavs">fix_wavs</a></code> for documentation. <code><a href="#topic+fixwavs">fixwavs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='frange'>alternative name for <code><a href="#topic+freq_range">freq_range</a></code></h2><span id='topic+frange'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+freq_range">freq_range</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frange(
  X,
  wl = 512,
  it = "jpeg",
  line = TRUE,
  fsmooth = 0.1,
  threshold = 10,
  dB.threshold = NULL,
  wn = "hanning",
  flim = NULL,
  bp = NULL,
  propwidth = FALSE,
  xl = 1,
  picsize = 1,
  res = 100,
  fast.spec = FALSE,
  ovlp = 50,
  pal = reverse.gray.colors.2,
  parallel = 1,
  widths = c(2, 1),
  main = NULL,
  img = TRUE,
  mar = 0.05,
  path = NULL,
  pb = TRUE,
  impute = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+freq_range">freq_range</a></code> for documentation. <code><a href="#topic+frange">frange</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='frange.detec'>alternative name for <code><a href="#topic+freq_range_detec">freq_range_detec</a></code></h2><span id='topic+frange.detec'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+freq_range_detec">freq_range_detec</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frange.detec(
  wave,
  wl = 512,
  fsmooth = 0.1,
  threshold = 10,
  dB.threshold = NULL,
  wn = "hanning",
  flim = NULL,
  bp = NULL,
  fast.spec = FALSE,
  ovlp = 50,
  pal = reverse.gray.colors.2,
  widths = c(2, 1),
  main = NULL,
  plot = TRUE,
  all.detec = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+freq_range_detec">freq_range_detec</a></code> for documentation. <code><a href="#topic+frange.detec">frange.detec</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='freq_DTW'>Acoustic dissimilarity using dynamic time warping on dominant frequency contours</h2><span id='topic+freq_DTW'></span>

<h3>Description</h3>

<p><code>freq_DTW</code> calculates acoustic dissimilarity of frequency contours using dynamic
time warping. Internally it applies the <code><a href="dtw.html#topic+dtwDist">dtwDist</a></code> function from the <code>dtw</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freq_DTW(X = NULL, type = "dominant", wl = 512, wl.freq = 512, length.out = 20,
wn = "hanning", ovlp = 70, bp = NULL, threshold = 15, threshold.time = NULL,
threshold.freq = NULL, img = TRUE, parallel = 1, path = NULL, ts.df = NULL,
img.suffix = "dfDTW", pb = TRUE, clip.edges = TRUE, window.type = "none",
open.end = FALSE, scale = FALSE, frange.detec = FALSE,  fsmooth = 0.1,
adjust.wl = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freq_DTW_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data
frame containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signal (start and end).
The output of <code><a href="#topic+auto_detec">auto_detec</a></code> can be used as the input data frame.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_type">type</code></td>
<td>
<p>Character string to determine the type of contour to be detected. Three options are available, &quot;dominant&quot; (default), &quot;fundamental&quot; and &quot;entropy&quot;.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_wl.freq">wl.freq</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram
for measurements on the frequency spectrum. Default is 512. Higher values would provide
more accurate measurements.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_length.out">length.out</code></td>
<td>
<p>A numeric vector of length 1 giving the number of measurements of frequency desired (the length of the time series).</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_threshold">threshold</code></td>
<td>
<p>amplitude threshold (%) for frequency detection. Default is 15.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_threshold.time">threshold.time</code></td>
<td>
<p>amplitude threshold (%) for the time domain. Use for frequency detection. If <code>NULL</code> (default) then the 'threshold' value is used.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_threshold.freq">threshold.freq</code></td>
<td>
<p>amplitude threshold (%) for the frequency domain. Use for frequency range detection from the spectrum (see 'frange.detec'). If <code>NULL</code> (default) then the
'threshold' value is used.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_img">img</code></td>
<td>
<p>Logical argument. If <code>FALSE</code>, image files are not produced. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_ts.df">ts.df</code></td>
<td>
<p>Optional. Data frame with frequency contour time series of signals to be compared. If provided &quot;X&quot; is ignored.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_img.suffix">img.suffix</code></td>
<td>
<p>A character vector of length 1 with a suffix (label) to add at the end of the names of
image files. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_clip.edges">clip.edges</code></td>
<td>
<p>Logical argument to control whether edges (start or end of signal) in
which amplitude values above the threshold were not detected will be removed. If
<code>TRUE</code> (default) this edges will be excluded and contours will be calculated on the
remaining values. Note that DTW cannot be applied if missing values (e.i. when amplitude is not detected).</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_window.type">window.type</code></td>
<td>
<p><code><a href="dtw.html#topic+dtw">dtw</a></code> windowing control parameter. Character: &quot;none&quot;, &quot;itakura&quot;, or a function (see <code><a href="dtw.html#topic+dtw">dtw</a></code>).</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_open.end">open.end</code></td>
<td>
<p><code><a href="dtw.html#topic+dtw">dtw</a></code> control parameter. Performs
open-ended alignments (see <code><a href="dtw.html#topic+dtw">dtw</a></code>).</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_scale">scale</code></td>
<td>
<p>Logical. If <code>TRUE</code> frequency values are z-transformed using the <code><a href="base.html#topic+scale">scale</a></code> function, which &quot;ignores&quot; differences in absolute frequencies between the signals in order to focus the
comparison in the frequency contour, regardless of the pitch of signals. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_frange.detec">frange.detec</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_fsmooth">fsmooth</code></td>
<td>
<p>A numeric vector of length 1 to smooth the frequency spectrum with a mean
sliding window (in kHz) used for frequency range detection (when <code>frange.detec = TRUE</code>). This help to average amplitude &quot;hills&quot; to minimize the effect of
amplitude modulation. Default is 0.1.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_adjust.wl">adjust.wl</code></td>
<td>
<p>Logical. If <code>TRUE</code> 'wl' (window length) is reset to be lower than the
number of samples in a selection if the number of samples is less than 'wl'. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_DTW_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for customizing
graphical output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extracts the dominant frequency values as a time series and
then calculates the pairwise acoustic dissimilarity using dynamic time warping.
The function uses the <code><a href="stats.html#topic+approxfun">approx</a></code> function to interpolate values between dominant
frequency  measures. If 'img' is  <code>TRUE</code> the function also produces image files
with the spectrograms of the signals listed in the input data frame showing the
location of the dominant frequencies.
</p>


<h3>Value</h3>

<p>A matrix with the pairwise dissimilarity values. If img is
<code>FALSE</code> it also produces image files with the spectrograms of the signals listed in the
input data frame showing the location of the dominant frequencies.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+spectrograms">spectrograms</a></code> for creating spectrograms from selections,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> for creating spectrograms to
optimize noise margins used in <code><a href="#topic+sig2noise">sig2noise</a></code> and <code><a href="#topic+freq_ts">freq_ts</a></code>, <code><a href="#topic+freq_ts">freq_ts</a></code>, for frequency contour overlaid spectrograms.
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+color_spectro">color_spectro</a>()</code>,
<code><a href="#topic+multi_DTW">multi_DTW</a>()</code>,
<code><a href="#topic+phylo_spectro">phylo_spectro</a>()</code>,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a>()</code>,
<code><a href="#topic+spectrograms">spectrograms</a>()</code>,
<code><a href="#topic+track_freq_contour">track_freq_contour</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data(list = c("Phae.long1", "Phae.long2", "lbh_selec_table"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav")) # save sound files
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))

  # dominant frequency
  freq_DTW(lbh_selec_table,
    length.out = 30, flim = c(1, 12), bp = c(2, 9),
    wl = 300, path = tempdir()
  )

  # fundamental frequency
  freq_DTW(lbh_selec_table,
    type = "fundamental", length.out = 30, flim = c(1, 12),
    bp = c(2, 9), wl = 300, path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='freq_range'>Detect frequency range iteratively</h2><span id='topic+freq_range'></span>

<h3>Description</h3>

<p><code>freq_range</code> detect frequency range iteratively from signals in a selection table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freq_range(X, wl = 512, it = "jpeg", line = TRUE, fsmooth = 0.1, threshold = 10,
dB.threshold = NULL, wn = "hanning", flim = NULL, bp = NULL,
propwidth = FALSE, xl = 1, picsize = 1, res = 100, fast.spec = FALSE, ovlp = 50,
pal = reverse.gray.colors.2, parallel = 1, widths = c(2, 1), main = NULL,
img = TRUE, mar = 0.05, path = NULL, pb = TRUE, impute = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freq_range_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;sel&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections. The output of <code><a href="#topic+auto_detec">auto_detec</a></code> can
also be used as the input data frame.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>)
and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>img = TRUE</code>).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_line">line</code></td>
<td>
<p>Logical argument to add red lines (or box if bottom.freq and top.freq columns are provided) at start and end times of selection. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_fsmooth">fsmooth</code></td>
<td>
<p>A numeric vector of length 1 to smooth the frequency spectrum with a mean
sliding window in kHz. This help to average amplitude &quot;hills&quot; to minimize the effect of
amplitude modulation. Default is 0.1.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_threshold">threshold</code></td>
<td>
<p>Amplitude threshold (%) for frequency range detection. The frequency range (not the cumulative amplitude) is represented as percentage (100% = highest amplitude). Default is 10. Ignored if 'dB.threshold' is supplied.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_db.threshold">dB.threshold</code></td>
<td>
<p>Amplitude threshold for frequency range detection (in dB). The value indicates the decrease in dB in relation to the highest amplitude (e.g. the peak frequency) in which range will be detected. For instance a dB.threshold = 20 means that the amplitude threshold would be 20 dB below the highest amplitude. If provided 'threshold' is ignored. Default is <code>NULL</code>.
Note that the power spectrum is normalized when using a dB scale, so it looks different than the one produced when no dB scale is used (e.g. when using 'threshold' argument).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>) and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>img = TRUE</code>).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit of
the spectrogram (in kHz), as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz) or &quot;frange&quot; to indicate that values in 'bottom.freq'
and 'top.freq' columns will be used as bandpass limits. Default is c(0, 22).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_propwidth">propwidth</code></td>
<td>
<p>Logical argument to scale the width of spectrogram
proportionally to duration of the selected call. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_xl">xl</code></td>
<td>
<p>Numeric vector of length 1. A constant by which to scale
spectrogram width. Default is 1.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_picsize">picsize</code></td>
<td>
<p>Numeric argument of length 1. Controls relative size of
spectrogram. Default is 1.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution.
Default is 100 (faster) although 300 - 400 is recommended for publication/
presentation quality.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially
increases performance (much faster), although some options become unavailable, as collevels, and sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast.spec' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 50. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>) and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>img = TRUE</code>).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_pal">pal</code></td>
<td>
<p>Color palette function for spectrogram. Default is reverse.gray.colors.2. See
<code><a href="seewave.html#topic+spectro">spectro</a></code> for more palettes. Palettes as <code><a href="monitoR.html#topic+specCols">gray.2</a></code> may work better when <code>fast.spec = TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_widths">widths</code></td>
<td>
<p>Numeric vector of length 2 to control the relative widths of the spectro (first element) and spectrum (second element).</p>
</td></tr>
<tr><td><code id="freq_range_+3A_main">main</code></td>
<td>
<p>Character vector of length 1 specifying the img title. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_img">img</code></td>
<td>
<p>Logical. Controls whether a plot is produced. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the selections
to set spectrogram limits. Default is 0.05.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_+3A_impute">impute</code></td>
<td>
<p>Logical. If <code>TRUE</code> then missing range values are imputed
with the corresponding bandpass value (hence ignored when <code>bp = NULL</code>). Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions aims to automatize the detection of frequency ranges. The frequency range is calculated as follows:
</p>

<ul>
<li><p> bottom.freq = the start frequency of the amplitude 'hill' containing the highest amplitude at the given threshold.
</p>
</li>
<li><p> top.freq = the end frequency of the amplitude 'hill' containing the highest amplitude at the given threshold.
</p>
</li></ul>

<p>If <code>img = TRUE</code> a graph including a spectrogram and a frequency spectrum is
generated for each selection (saved as an image file in the working directory). The graph would include gray areas in the frequency ranges excluded by the bandpass ('bp' argument), dotted lines highlighting the detected range. The function <code><a href="#topic+freq_range_detec">freq_range_detec</a></code> is used internally.
</p>


<h3>Value</h3>

<p>The original data frame with an additional 2 columns for low and high frequency values. A plot is produced in the working directory if <code>img = TRUE</code> (see details).
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+freq_range_detec">freq_range_detec</a></code>, <code><a href="#topic+auto_detec">auto_detec</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  freq_range(
    X = lbh_selec_table, wl = 112, fsmooth = 1, threshold = 13, widths = c(4, 1),
    img = TRUE, pb = TRUE, it = "tiff", line = TRUE, mar = 0.1, bp = c(1, 10.5),
    flim = c(0, 11), path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='freq_range_detec'>Detect frequency range on wave objects</h2><span id='topic+freq_range_detec'></span>

<h3>Description</h3>

<p><code>freq_range_detec</code> detects the frequency range of acoustic signals on wave objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freq_range_detec(wave, wl = 512, fsmooth = 0.1, threshold = 10,
dB.threshold = NULL, wn = "hanning", flim = NULL, bp = NULL,
fast.spec = FALSE, ovlp = 50, pal = reverse.gray.colors.2,
 widths = c(2, 1), main = NULL, plot = TRUE, all.detec = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freq_range_detec_+3A_wave">wave</code></td>
<td>
<p>A 'wave' object produced by  <code><a href="tuneR.html#topic+readWave">readWave</a></code> or similar functions.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>)
and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>plot = TRUE</code>).</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_fsmooth">fsmooth</code></td>
<td>
<p>A numeric vector of length 1 to smooth the frequency spectrum with a mean
sliding window in kHz. This help to average amplitude &quot;hills&quot; to minimize the effect of
amplitude modulation. Default is 0.1.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_threshold">threshold</code></td>
<td>
<p>Amplitude threshold (%) for frequency range detection. The frequency range (not the cumulative amplitude) is represented as percentage (100% = highest amplitude). Default is 10. Ignored if 'dB.threshold' is supplied.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_db.threshold">dB.threshold</code></td>
<td>
<p>Amplitude threshold for frequency range detection (in dB). The
value indicates the decrease in dB in relation to the highest amplitude (e.g.
the peak frequency) in which range will be detected. For instance a
<code>dB.threshold = 20</code> means that the amplitude threshold would be 20 dB below
the highest amplitude. If provided 'threshold' is ignored. Default is <code>NULL</code>.
Note that the power spectrum is normalized when using a dB scale, so it looks different than the one produced when no dB scale is used (e.g. when using 'threshold' argument).</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>) and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>plot = TRUE</code>).</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit of
the spectrogram (in kHz), as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz) or &quot;frange&quot; to indicate that values in 'bottom.freq'
and 'top.freq' columns will be used as bandpass limits. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially
increases performance (much faster), although some options become unavailable, as collevels, and sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast.spec' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 50. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>) and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>plot = TRUE</code>).</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_pal">pal</code></td>
<td>
<p>Color palette function for spectrogram. Default is reverse.gray.colors.2. See
<code><a href="seewave.html#topic+spectro">spectro</a></code> for more palettes. Palettes as <code><a href="monitoR.html#topic+specCols">gray.2</a></code> may work better when <code>fast.spec = TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_widths">widths</code></td>
<td>
<p>Numeric vector of length 2 to control the relative widths of the spectro (first element) and spectrum (second element).</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_main">main</code></td>
<td>
<p>Character vector of length 1 specifying the plot title. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_plot">plot</code></td>
<td>
<p>Logical. Controls whether an image file is produced for each selection (in the
working directory). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_range_detec_+3A_all.detec">all.detec</code></td>
<td>
<p>Logical. If <code>TRUE</code> returns the start and end of all detected amplitude
&quot;hills&quot;. Otherwise only the range is returned. Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions aims to automatize the detection of frequency ranges. The frequency range is calculated as follows:
</p>

<ul>
<li><p> bottom.freq = the start frequency of the amplitude 'hill' containing the highest amplitude at the given threshold.
</p>
</li>
<li><p> top.freq = the end frequency of the amplitude 'hill' containing the highest amplitude at the given threshold.
</p>
</li></ul>

<p>If <code>plot = TRUE</code> a graph including a spectrogram and a frequency spectrum is
produced in the graphic device. The graph would include gray areas in the frequency ranges excluded by the bandpass ('bp' argument), dotted lines highlighting the detected range.
</p>


<h3>Value</h3>

<p>A data frame with 2 columns for low and high frequency values. A plot is produced (in the graphic device) if <code>plot = TRUE</code> (see details).
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+frange">frange</a></code>, <code><a href="#topic+autodetec">autodetec</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(tico)
  freq_range_detec(
    wave = tico, wl = 512, fsmooth = 0.01, threshold = 1, bp = c(2, 8),
    widths = c(4, 2)
  )

  data(sheep)
  freq_range_detec(
    wave = sheep, wl = 512, fsmooth = 0.2, threshold = 50, bp = c(0.3, 1),
    flim = c(0, 1.5), pal = reverse.heat.colors, main = "sheep"
  )
}

</code></pre>

<hr>
<h2 id='freq_ts'>Extract frequency contours as time series</h2><span id='topic+freq_ts'></span>

<h3>Description</h3>

<p><code>freq_ts</code> extracts the fundamental frequency values as a time series.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>freq_ts(X, type = "dominant", wl = 512, length.out = 20, wn = "hanning",
ovlp = 70, bp = NULL, threshold = 15, img = TRUE, parallel = 1, path = NULL,
img.suffix = "frequency.ts", pb = TRUE, clip.edges = FALSE, leglab = "frequency.ts",
track.harm = FALSE, raw.contour = FALSE, adjust.wl = TRUE,
ff.method = "seewave", entropy.range = c(2, 10), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freq_ts_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data
frame containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signal (start and end).</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_type">type</code></td>
<td>
<p>Character string to determine the type of contour to be detected. Three options are available, &quot;dominant&quot; (default), &quot;fundamental&quot; and &quot;entropy&quot;.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_length.out">length.out</code></td>
<td>
<p>A numeric vector of length 1 giving the number of measurements of fundamental
frequency desired (the length of the time series).</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_threshold">threshold</code></td>
<td>
<p>amplitude threshold (%) for fundamental frequency detection. Default is 15.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_img">img</code></td>
<td>
<p>Logical argument. If <code>FALSE</code>, image files are not produced. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_img.suffix">img.suffix</code></td>
<td>
<p>A character vector of length 1 with a suffix (label) to add at the end of the names of
image files.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_clip.edges">clip.edges</code></td>
<td>
<p>Logical argument to control whether edges (start or end of signal) in
which amplitude values above the threshold were not detected will be removed. If
<code>TRUE</code> this edges will be excluded and signal contour will be calculated on the
remaining values. Default is <code>FALSE</code>.
#' @param leglab A character vector of length 1 or 2 containing the label(s) of the frequency contour legend
in the output image.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_leglab">leglab</code></td>
<td>
<p>A character vector of length 1 or 2 containing the label(s) of the frequency contour legend
in the output image.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_track.harm">track.harm</code></td>
<td>
<p>Logical. If <code>TRUE</code> warbleR's <code><a href="#topic+track_harmonic">track_harmonic</a></code> function is
used to track dominant frequency contours. Otherwise seewave's <code><a href="seewave.html#topic+dfreq">dfreq</a></code> is used by default. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_raw.contour">raw.contour</code></td>
<td>
<p>Logical. If <code>TRUE</code> then a list with the original contours
(i.e. without interpolating values to make all contours of equal length) is returned (and no images are produced).</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_adjust.wl">adjust.wl</code></td>
<td>
<p>Logical. If <code>TRUE</code> 'wl' (window length) is reset to be lower than the
number of samples in a selection if the number of samples is less than 'wl'. Default is <code>TRUE</code>. Used only for dominant frequency detection.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_ff.method">ff.method</code></td>
<td>
<p>Character. Selects the method used to detect fundamental
frequency contours. Either 'tuneR' (using <code><a href="tuneR.html#topic+FF">FF</a></code>) or 'seewave' (using
<code><a href="seewave.html#topic+fund">fund</a></code>). Default is 'seewave'. 'tuneR' performs
faster (and seems to be more accurate) than 'seewave'.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_entropy.range">entropy.range</code></td>
<td>
<p>Numeric vector of length 2. Range of frequency in which to display the entropy values
on the spectrogram (when img = TRUE). Default is c(2, 10). Negative values can be used in order to stretch more
the range.</p>
</td></tr>
<tr><td><code id="freq_ts_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code><a href="#topic+track_freq_contour">track_freq_contour</a></code>. for customizing
graphical output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extracts the dominant frequency, fundamental frequency or spectral entropy contours as time series.
The function uses the <code><a href="stats.html#topic+approxfun">approx</a></code> function to interpolate values between frequency measures. If there are no frequencies above the amplitude threshold (for dominant and fundamental) at the beginning or end
of the signals then NAs will be generated. On the other hand, if there are no frequencies
above the amplitude theshold in between signal segments in which amplitude was
detected then the values of this adjacent segments will be interpolated
to fill out the missing values (e.g. no NAs in between detected amplitude segments).
</p>


<h3>Value</h3>

<p>A data frame with the fundamental frequency values measured across the signals. If img is
<code>TRUE</code> it also produces image files with the spectrograms of the signals listed in the
input data frame showing the location of the fundamental frequencies
(see <code><a href="#topic+track_freq_contour">track_freq_contour</a></code> description for more details).
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sig2noise">sig2noise</a></code>, <code><a href="#topic+track_freq_contour">track_freq_contour</a></code>, <code><a href="#topic+freq_ts">freq_ts</a></code>, <code><a href="#topic+freq_DTW">freq_DTW</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
#load data
data(list = c("Phae.long1", "Phae.long2","lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) #save sound files
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav")) #save sound files

# run function with dominant frequency
freq_ts(X = lbh_selec_table, length.out = 30, flim = c(1, 12), bp = c(2, 9),
wl = 300, pb = FALSE, path = tempdir())

# note a NA in the row 4 column 3 (dfreq-1)
# this can be removed by clipping edges (removing NAs at the start and/or end
# when no freq was detected)

freq_ts(X = lbh_selec_table, length.out = 30, flim = c(1, 12), bp = c(2, 9),
wl = 300, pb = FALSE, clip.edges = TRUE, path = tempdir())

# run function with fundamental frequency
freq_ts(lbh_selec_table, type = "fundamental", length.out = 50,
flim = c(1, 12),  bp = c(2, 9), wl = 300, path = tempdir())

# run function with spectral entropy
# without clip edges
freq_ts(X = lbh_selec_table, type = "entropy", threshold = 10,
clip.edges = FALSE, length.out = 10, sp.en.range = c(-25, 10), path = tempdir(),
 img = FALSE)

# with clip edges and length.out 10
freq_ts(X = lbh_selec_table, type = "entropy", threshold = 10, bp = c(2, 12),
clip.edges = TRUE, length.out = 10, path = tempdir(), img = FALSE)
}
</code></pre>

<hr>
<h2 id='full_spectrogram2pdf'><code>full_spectrogram2pdf</code> combines <code><a href="#topic+full_spectrograms">full_spectrograms</a></code> images in .jpeg format to a single pdf file.</h2><span id='topic+full_spectrogram2pdf'></span>

<h3>Description</h3>

<p><code>full_spectrogram2pdf</code> combines <code><a href="#topic+full_spectrograms">full_spectrograms</a></code> images in .jpeg format to a single pdf file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>full_spectrogram2pdf(keep.img = TRUE, overwrite = FALSE,
parallel = 1, path = NULL, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="full_spectrogram2pdf_+3A_keep.img">keep.img</code></td>
<td>
<p>Logical argument. Indicates whether jpeg files should be kept (default) or remove.
(including sound file and page number) should be magnified. Default is 1.</p>
</td></tr>
<tr><td><code id="full_spectrogram2pdf_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical argument. If <code>TRUE</code> all jpeg pdf will be produced again
when code is rerun. If <code>FALSE</code> only the ones missing will be produced. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="full_spectrogram2pdf_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="full_spectrogram2pdf_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="full_spectrogram2pdf_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function combines  spectrograms for complete sound files from the <code><a href="#topic+full_spectrograms">full_spectrograms</a></code> function into
a single pdf (for each sound file).
</p>


<h3>Value</h3>

<p>Image files in pdf format with spectrograms of entire sound files in the working directory.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+full_spectrograms">full_spectrograms</a></code>, <code><a href="#topic+catalog2pdf">catalog2pdf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# save sound file examples
data(list = c("Phae.long1", "Phae.long2"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

full_spectrograms(
  sxrow = 2, rows = 8, pal = reverse.heat.colors, wl = 300,
  it = "jpeg", path = tempdir()
)

# now create single pdf removing jpeg
full_spectrogram2pdf(keep.img = FALSE, path = tempdir())

# check this floder
tempdir()

## End(Not run)

</code></pre>

<hr>
<h2 id='full_spectrograms'>Create long spectrograms of entire sound files</h2><span id='topic+full_spectrograms'></span>

<h3>Description</h3>

<p><code>full_spectrograms</code> produces image files with spectrograms of entire sound files split into multiple
rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>full_spectrograms(X = NULL, flim = NULL, sxrow = 5, rows = 10,
collevels = seq(-40, 0, 1), ovlp = 50, parallel = 1, wl = 512, gr = FALSE,
pal = reverse.gray.colors.2, cex = 1, it = "jpeg", flist = NULL,
overwrite = TRUE, path = NULL, pb = TRUE, fast.spec = FALSE, labels = "selec",
 horizontal = FALSE, song = NULL, suffix = NULL, dest.path = NULL,
 only.annotated = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="full_spectrograms_+3A_x">X</code></td>
<td>
<p>'selection_table' object or any data frame with columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end). If given, a transparent box is  plotted around each selection and the selections are labeled with the selection number
(and selection comment, if available). Default is <code>NULL</code>. Alternatively, it can also take the output of <code><a href="#topic+cross_correlation">cross_correlation</a></code> or <code><a href="#topic+auto_detec">auto_detec</a></code> (when 'output' is a 'list', see <code><a href="#topic+cross_correlation">cross_correlation</a></code> or <code><a href="#topic+auto_detec">auto_detec</a></code>). If supplied a secondary row is displayed under each spectrogram showing the detection (either cross-correlation scores or wave envelopes) values across time.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 indicating the highest and lowest
frequency limits (kHz) of the spectrogram, as in
<code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>. Alternatively, a character vector similar to <code>c("-1", "1")</code> in which the first number is the value to be added to the minimum bottom frequency in 'X' and the second the value to be added to the maximum top frequency in 'X'. This is computed independently for each sound file so the frequency limit better fits the frequency range of the annotated signals. This is useful when plotting annotated spectrograms with marked differences in the frequency range of annotations among sond files. Note that top frequency adjustment is ignored if 'song' labels are included (argument 'song').</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_sxrow">sxrow</code></td>
<td>
<p>A numeric vector of length 1. Specifies seconds of spectrogram
per row. Default is 5.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_rows">rows</code></td>
<td>
<p>A numeric vector of length 1. Specifies number of rows per
image file. Default is 10.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_collevels">collevels</code></td>
<td>
<p>A numeric vector of length 3. Specifies levels to partition the
amplitude range of the spectrogram (in dB). The more levels the higher the
resolution of the spectrogram. Default is seq(-40, 0, 1).</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 50. High values of ovlp
slow down the function but produce more accurate selection limits (when X is provided).</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_gr">gr</code></td>
<td>
<p>Logical argument to add grid to spectrogram. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_pal">pal</code></td>
<td>
<p>Color palette function for spectrogram. Default is reverse.gray.colors.2. See
<code><a href="seewave.html#topic+spectro">spectro</a></code> for more palettes.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_cex">cex</code></td>
<td>
<p>A numeric vector of length 1 giving the amount by which text
(including sound file and page number) should be magnified. Default is 1.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_flist">flist</code></td>
<td>
<p>character vector or factor indicating the subset of files that will be analyzed. Ignored
if X is provided.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical argument. If <code>TRUE</code> all selections will be analyzed again
when code is rerun. If <code>FALSE</code> only the selections that do not have a image
file in the working directory will be analyzed. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially
increases performance (much faster), although some options become unavailable, as collevels, and sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_labels">labels</code></td>
<td>
<p>Character string with the name of the column(s) for selection
labeling. Default is 'selec'. Set to <code>NULL</code> to remove labels.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_horizontal">horizontal</code></td>
<td>
<p>Logical. Controls if the images are produced as horizontal or vertical pages. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_song">song</code></td>
<td>
<p>Character string with the name of the column to used as a label a for higher
organization level in the song (similar to 'song_colm' in <code><a href="#topic+song_analysis">song_analysis</a></code>). If supplied then lines above the selections belonging to the same
'song' are plotted. Ignored if 'X' is not provided.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_suffix">suffix</code></td>
<td>
<p>Character vector of length 1. Suffix for the output image file (to be added at the end of the default file name). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_dest.path">dest.path</code></td>
<td>
<p>Character string containing the directory path where the image files will be saved.
If <code>NULL</code> (default) then the folder containing the sound files will be used instead.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_only.annotated">only.annotated</code></td>
<td>
<p>Logical argument to control if only the pages that contained annotated sounds (from 'X') are printed. Only used if 'X' is supplied.</p>
</td></tr>
<tr><td><code id="full_spectrograms_+3A_...">...</code></td>
<td>
<p>Additional arguments for image formatting. It accepts 'width', 'height' (which will overwrite 'horizontal') and 'res' as in <code><a href="grDevices.html#topic+png">png</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function creates spectrograms for complete sound files, printing
the name of the sound files and the &quot;page&quot; number (p1-p2...) at the upper
right corner of the image files. If 'X' is
supplied, the function delimits and labels the selections.
This function aims to facilitate visual inspection of multiple files as well as visual classification
of vocalization units and the analysis of animal vocal sequences.
</p>


<h3>Value</h3>

<p>image files with spectrograms of entire sound files in the working directory. Multiple pages
can be returned, depending on the length of each sound file.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+full_spectrogram2pdf">full_spectrogram2pdf</a></code>, <code><a href="#topic+catalog2pdf">catalog2pdf</a></code>, <code><a href="#topic+cross_correlation">cross_correlation</a></code>, <code><a href="#topic+auto_detec">auto_detec</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# save sound file examples to temporary working directory
data(list = c("Phae.long1", "Phae.long2", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

full_spectrograms(
  sxrow = 2, rows = 8, pal = reverse.heat.colors, wl = 300,
  path = tempdir()
)

# including selections
full_spectrograms(
  sxrow = 2, rows = 8, X = lbh_selec_table,
  pal = reverse.heat.colors, overwrite = TRUE, wl = 300, path = tempdir()
)

# check this floder
# tempdir()

## End(Not run)

</code></pre>

<hr>
<h2 id='gaps'>Gap duration</h2><span id='topic+gaps'></span>

<h3>Description</h3>

<p><code>gaps</code> measures gap duration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaps(X = NULL, by = "sound.files", parallel = 1, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gaps_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' (created 'by.song') or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;selec&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections.</p>
</td></tr>
<tr><td><code id="gaps_+3A_by">by</code></td>
<td>
<p>Character vector with column names. Controls the levels at which gaps will be measured. &quot;sound.files&quot; must always be included.</p>
</td></tr>
<tr><td><code id="gaps_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="gaps_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function measures the time intervals (i.e. gaps) between selections.  The gap for a given selection is calculated as the time interval to the selection immediately after. Hence, there is no gap for the last selection in a sound file (or level determined by the 'by' argument). Gap is set to 0  when selections overlap in time. Note that the sound files are not required.
</p>


<h3>Value</h3>

<p>A data frame identical to that supplied in 'X', with and additional column ('gaps') with the duration of the time interval between selections.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+inflections">inflections</a></code>, <code><a href="#topic+song_analysis">song_analysis</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# get warbleR sound file examples
data(list = "lbh_selec_table")

# get gaps
gaps(X = lbh_selec_table)
}

</code></pre>

<hr>
<h2 id='image_to_wave'>Convert images into wave objects</h2><span id='topic+image_to_wave'></span>

<h3>Description</h3>

<p><code>image_to_wave</code> converts images in 'png' format into wave objects using the inverse Fourier transformation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>image_to_wave(file, duration = 1, samp.rate = 44.1,
bit.depth = 16, flim = c(0, samp.rate / 2), plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="image_to_wave_+3A_file">file</code></td>
<td>
<p>Character with the name of image file to be converted. File must be in 'png' format.</p>
</td></tr>
<tr><td><code id="image_to_wave_+3A_duration">duration</code></td>
<td>
<p>duration of the output wave object (in s).</p>
</td></tr>
<tr><td><code id="image_to_wave_+3A_samp.rate">samp.rate</code></td>
<td>
<p>Numeric vector of length 1 indicating the sampling rate of the output wave object (in kHz). Default is 44.1.</p>
</td></tr>
<tr><td><code id="image_to_wave_+3A_bit.depth">bit.depth</code></td>
<td>
<p>Numeric vector of length 1 with the dynamic interval (i.e. bit depth) for output files. Default is 16.</p>
</td></tr>
<tr><td><code id="image_to_wave_+3A_flim">flim</code></td>
<td>
<p>Numeric vector of length 2 indicating the highest and lowest
frequency limits (kHz) in which the image would be located. Default is <code>c(0, samp.rate / 2)</code>.</p>
</td></tr>
<tr><td><code id="image_to_wave_+3A_plot">plot</code></td>
<td>
<p>Logical argument to control if image is plotted after being imported into R.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function converts images in 'png' format into wave objects using the inverse Fourier transformation.
</p>


<h3>Value</h3>

<p>A single wave object.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### create image with text to use in the spectrogram
# remove margins of plot
par(mar = rep(0, 4))

# empty plot
plot(0, type = "n", axes = FALSE, ann = FALSE, xlim = c(0, 1), ylim = c(0, 1))

# text to include
text &lt;- " warbleR "

# add text
text(x = 0.5, y = 0.5, labels = text, cex = 11, font = 1)

# save image in temporary directory
dev2bitmap(file.path(tempdir(), "temp-img.png"), type = "pngmono", res = 30)

# read it
wv &lt;- image_to_wave(file = file.path(tempdir(), "temp-img.png"), plot = TRUE, flim = c(1, 12))

# output wave object
# wv

## plot it
# reset margins
par(mar = c(5, 4, 4, 2) + 0.1)

# plot spectrogram
# spectro(wave = wv, scale = FALSE, collevels = seq(-30, 0, 5),
# palette = reverse.terrain.colors, ovlp = 90, grid = FALSE, flim = c(2, 11))


</code></pre>

<hr>
<h2 id='inflections'>Count number of inflections in a frequency contour</h2><span id='topic+inflections'></span>

<h3>Description</h3>

<p><code>inflections</code> counts the number of inflections in a frequency contour (or any time series)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inflections(X = NULL, parallel = 1, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inflections_+3A_x">X</code></td>
<td>
<p>data frame with the columns for &quot;sound.files&quot; (sound file name), &quot;selec&quot; (unique identifier for each selection) and columns for each of the frequency values of the contours. No other columns should be included.</p>
</td></tr>
<tr><td><code id="inflections_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="inflections_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function counts the number of inflections in a frequency contour.
</p>


<h3>Value</h3>

<p>A data frame with 3 columns: &quot;sound.files&quot;, &quot;selec&quot; and &quot;infls&quot; (number of inflections).
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+freq_ts">freq_ts</a></code>, <code><a href="#topic+track_freq_contour">track_freq_contour</a></code>,  <code><a href="#topic+gaps">gaps</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# get warbleR sound file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# measure frequency contours
dom.freq.ts &lt;- freq_ts(X = lbh_selec_table, path = tempdir())

# get number of inflections
inflections(X = dom.freq.ts)
}

</code></pre>

<hr>
<h2 id='info_sound_files'>Get sound file parameter information</h2><span id='topic+info_sound_files'></span>

<h3>Description</h3>

<p><code>info_sound_files</code> summariz sound file information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>info_sound_files(path = NULL, files = NULL, parallel = 1, pb = TRUE, skip.error = FALSE,
file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="info_sound_files_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="info_sound_files_+3A_files">files</code></td>
<td>
<p>character vector indicating the set of files that will be consolidated. File names should not include the full file path. Optional.</p>
</td></tr>
<tr><td><code id="info_sound_files_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="info_sound_files_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="info_sound_files_+3A_skip.error">skip.error</code></td>
<td>
<p>Logical to control if errors are omitted. If so, files that could not be read will be excluded and their name printed in the console. Default is <code>FALSE</code>, which will return an error if some files are problematic.</p>
</td></tr>
<tr><td><code id="info_sound_files_+3A_file.format">file.format</code></td>
<td>
<p>Character string with the format of sound files. By default all sound file formats supported by warbleR are included (&quot;\\.wav$|\\.wac$|\\.mp3$|\\.flac$&quot;). Note that several formats can be included using regular expression syntax as in <code><a href="base.html#topic+grep">grep</a></code>. For instance <code>"\\.wav$|\\.mp3$"</code> will only include .wav and .mp3 files.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper for <code><a href="#topic+selection_table">selection_table</a></code> that returns a data frame with the following descriptive parameters for each sound file in the working directory (or 'path'):
</p>

<ul>
<li> <p><code>duration</code>: duration of selection in seconds
</p>
</li>
<li> <p><code>sample.rate</code>: sampling rate in kHz
</p>
</li>
<li> <p><code>channels</code>: number of channels
</p>
</li>
<li> <p><code>bits</code>: bit depth
</p>
</li>
<li> <p><code>wav.size</code>: sound file size in MB
</p>
</li>
<li> <p><code>samples</code>: number of samples in the sound file
</p>
</li></ul>



<h3>Value</h3>

<p>A data frame with descriptive information about the sound files in the working directory (or 'path'). See &quot;details&quot;.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+fix_wavs">fix_wavs</a></code>, <code><a href="#topic+selection_table">selection_table</a></code> &amp; <code><a href="#topic+check_sels">check_sels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

#get info
info_sound_files(path = tempdir())
}

</code></pre>

<hr>
<h2 id='info_wavs'>alternative name for <code><a href="#topic+info_sound_files">info_sound_files</a></code></h2><span id='topic+info_wavs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+info_sound_files">info_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>info_wavs(
  path = NULL,
  files = NULL,
  parallel = 1,
  pb = TRUE,
  skip.error = FALSE,
  file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$"
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+info_sound_files">info_sound_files</a></code> for documentation. <code><a href="#topic+info_wavs">info_wavs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='is_extended_selection_table'>Class 'extended_selection_table': selection table containing wave objects</h2><span id='topic+is_extended_selection_table'></span>

<h3>Description</h3>

<p>Class for selections of signals in sound files and corresponding wave objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_extended_selection_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_extended_selection_table_+3A_x">x</code></td>
<td>
<p>R object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An object of class <code>extended_selection_table</code> created by <code><a href="#topic+selection_table">selection_table</a></code> is a list with the following elements:
</p>

<ul>
<li> <p><code>selections</code>: data frame containing the frequency/time coordinates of the selections, sound file names, and any  additional information
</p>
</li>
<li> <p><code>check.resutls</code>: results of the checks on data consistency using <a href="#topic+check_sels">check_sels</a>
</p>
</li>
<li> <p><code>wave.objects</code>: list of wave objects corresponding to each selection
</p>
</li>
<li> <p><code>by.song</code>: a list with 1) a logical argument defining if the 'extended_selection_table' was created 'by song'
and 2) the name of the song column (see <code><a href="#topic+selection_table">selection_table</a></code>)
</p>
</li></ul>



<h3>Value</h3>

<p>A logical argument indicating whether the object class is 'extended_selection_table'
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+selection_table">selection_table</a></code>, <code><a href="#topic+selection_table">selection_table</a></code>
Check if object is of class &quot;extended_selection_table&quot;
</p>
<p><code>is_extended_selection_table</code> Check if the object belongs to the class &quot;extended_selection_table&quot;
</p>
<p><code><a href="#topic+selection_table">selection_table</a></code>; <code><a href="#topic+is_selection_table">is_selection_table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))

  is_extended_selection_table(lbh_selec_table)

  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  st &lt;- selection_table(lbh_selec_table,
    extended = TRUE,
    path = tempdir()
  )

  is_extended_selection_table(st)

  class(st)
}
</code></pre>

<hr>
<h2 id='is_selection_table'>Class 'selection_table': double-checked frequency/time coordinates of selections</h2><span id='topic+is_selection_table'></span>

<h3>Description</h3>

<p>Class for selections of signals in sound files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_selection_table(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_selection_table_+3A_x">x</code></td>
<td>
<p>R object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An object of class <code>selection_table</code> created by <code><a href="#topic+selection_table">selection_table</a></code> is a list with the following elements:
</p>

<ul>
<li><p><code>selections</code>: data frame containing the frequency/time coordinates of the selections, sound file names, and any  additional information
</p>
</li>
<li> <p><code>check.resutls</code>: results of the checks on data consistency using <a href="#topic+check_sels">check_sels</a>
</p>
</li></ul>



<h3>Value</h3>

<p>A logical argument indicating whether the object class is 'selection_table'
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+selection_table">selection_table</a></code>
Check if object is of class &quot;selection_table&quot;
</p>
<p><code>is_selection_table</code> Check if the object belongs to the class &quot;selection_table&quot;
</p>
<p><code><a href="#topic+selection_table">selection_table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))

  is_selection_table(lbh_selec_table)

  # save wave files in temporary directory
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  st &lt;- selection_table(lbh_selec_table, path = tempdir())

  is_selection_table(st)

  class(st)
}
</code></pre>

<hr>
<h2 id='lbh_selec_table'>Example data frame of selections (i.e. selection table).</h2><span id='topic+lbh_selec_table'></span>

<h3>Description</h3>

<p><code>lbh_selec_table</code> alternative name for <code>selec.table</code>. <code>selec.table</code> will be deprecated in
future versions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lbh_selec_table)
</code></pre>


<h3>Format</h3>

<p>A data frame with 11 rows and 7 variables: </p>

<dl>
<dt>sound.files</dt><dd><p>recording names</p>
</dd>
<dt>channel</dt><dd><p>channel in which signal is found</p>
</dd>
<dt>selec</dt><dd><p>selection numbers within recording</p>
</dd>
<dt>start</dt><dd><p>start times of selected signal</p>
</dd>
<dt>end</dt><dd><p>end times of selected signal</p>
</dd>
<dt>bottom.freq</dt><dd><p>lower limit of frequency range</p>
</dd>
<dt>top.freq</dt><dd><p>upper limit of frequency range</p>
</dd>
</dl>



<h3>Details</h3>

<p>A data frame containing the start, end, low and high frequency of
<em>Phaethornis longirostris</em> (Long-billed Hermit) songs from the
example sound files included in this package. Same data than 'selec_table'.
'selec_table' will be removed in future package version.
</p>


<h3>Source</h3>

<p>Marcelo Araya-Salas, warbleR
</p>

<hr>
<h2 id='lbh_selec_table2'>Example data frame of selections (i.e. selection table).</h2><span id='topic+lbh_selec_table2'></span>

<h3>Description</h3>

<p><code>lbh_selec_table2</code> is a data frame containing the start, end, low and high frequency of 2 selections. Mostly to be used as an example in  <code><a href="#topic+find_peaks">find_peaks</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lbh_selec_table2)
</code></pre>


<h3>Format</h3>

<p>A data frame with 2 rows and 7 variables: </p>

<dl>
<dt>sound.files</dt><dd><p>recording names</p>
</dd>
<dt>channel</dt><dd><p>channel in which signal is found</p>
</dd>
<dt>selec</dt><dd><p>selection numbers within recording</p>
</dd>
<dt>start</dt><dd><p>start times of selected signal</p>
</dd>
<dt>end</dt><dd><p>end times of selected signal</p>
</dd>
<dt>bottom.freq</dt><dd><p>lower limit of frequency range</p>
</dd>
<dt>top.freq</dt><dd><p>upper limit of frequency range</p>
</dd>
</dl>



<h3>Details</h3>

<p>A data frame containing the start, end, low and high frequency of
<em>Phaethornis longirostris</em> (Long-billed Hermit) songs from the
example sound files included in this package. Similar than 'lbh_selec_table'.
but it contains only 2 selections.
</p>


<h3>Source</h3>

<p>Marcelo Araya-Salas, warbleR
</p>

<hr>
<h2 id='lspec'>alternative name for <code><a href="#topic+full_spectrograms">full_spectrograms</a></code></h2><span id='topic+lspec'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+full_spectrograms">full_spectrograms</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lspec(
  X = NULL,
  flim = NULL,
  sxrow = 5,
  rows = 10,
  collevels = seq(-40, 0, 1),
  ovlp = 50,
  parallel = 1,
  wl = 512,
  gr = FALSE,
  pal = reverse.gray.colors.2,
  cex = 1,
  it = "jpeg",
  flist = NULL,
  overwrite = TRUE,
  path = NULL,
  pb = TRUE,
  fast.spec = FALSE,
  labels = "selec",
  horizontal = FALSE,
  song = NULL,
  suffix = NULL,
  dest.path = NULL,
  only.annotated = FALSE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+full_spectrograms">full_spectrograms</a></code> for documentation. <code><a href="#topic+lspec">lspec</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='lspec2pdf'>alternative name for <code><a href="#topic+full_spectrogram2pdf">full_spectrogram2pdf</a></code></h2><span id='topic+lspec2pdf'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+full_spectrogram2pdf">full_spectrogram2pdf</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lspec2pdf(
  keep.img = TRUE,
  overwrite = FALSE,
  parallel = 1,
  path = NULL,
  pb = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+full_spectrogram2pdf">full_spectrogram2pdf</a></code> for documentation
</p>

<hr>
<h2 id='make.selection.table'>Old name for <code><a href="#topic+selection_table">selection_table</a></code></h2><span id='topic+make.selection.table'></span>

<h3>Description</h3>

<p>Old name for <code><a href="#topic+selection_table">selection_table</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.selection.table(
  X,
  max.dur = 10,
  path = NULL,
  whole.recs = FALSE,
  extended = FALSE,
  confirm.extended = FALSE,
  mar = 0.1,
  by.song = NULL,
  pb = TRUE,
  parallel = 1,
  verbose = TRUE,
  skip.error = FALSE,
  file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$",
  files = NULL,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+selection_table">selection_table</a></code> for documentation
</p>

<hr>
<h2 id='map_xc'>Maps of 'Xeno-Canto' recordings by species</h2><span id='topic+map_xc'></span>

<h3>Description</h3>

<p><code>map_xc</code> creates maps to visualize the geographic spread of 'Xeno-Canto'
recordings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_xc(X, img = TRUE, it = "jpeg", res = 100, labels = FALSE,
 path = NULL, leaflet.map = FALSE,
 leaflet.cluster = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map_xc_+3A_x">X</code></td>
<td>
<p>Data frame output from <code><a href="#topic+query_xc">query_xc</a></code>.</p>
</td></tr>
<tr><td><code id="map_xc_+3A_img">img</code></td>
<td>
<p>A logical argument specifying whether an image file of each species
map should be returned, default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="map_xc_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="map_xc_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution.
Default is 100 (faster) although 300 - 400 is recommended for publication/
presentation quality.</p>
</td></tr>
<tr><td><code id="map_xc_+3A_labels">labels</code></td>
<td>
<p>A logical argument defining whether dots depicting recording locations are labeled.
If <code>TRUE</code> then the Recording_ID is used as label.</p>
</td></tr>
<tr><td><code id="map_xc_+3A_path">path</code></td>
<td>
<p>Character string with the directory path where the image files will be saved.
If <code>NULL</code> (default) then the current working directory is used.
Ignored if <code>img = FALSE</code>.</p>
</td></tr>
<tr><td><code id="map_xc_+3A_leaflet.map">leaflet.map</code></td>
<td>
<p>Logical to control whether the package 'leaflet' is used for displaying the maps. 'leaflet' maps are interactive and display information about recordings and links to the Xeno-Canto website. If <code>TRUE</code> a single map is displayed regardless of the number of species and all other image related arguments are ignored. Default is <code>FALSE</code>. The hovering label shows the species scientific name (or the subspecies if only 1 species is present in 'X'). Note that colors will be recycled if more after 18 species (or subspecies).</p>
</td></tr>
<tr><td><code id="map_xc_+3A_leaflet.cluster">leaflet.cluster</code></td>
<td>
<p>Logical to control if icons are clustered by locality (as in Xeno-Canto maps). Default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates maps for visualizing the geographic spread of recordings from the open-access
online repository <a href="https://www.xeno-canto.org/">Xeno-Canto</a>. The function takes the output of
<code><a href="#topic+query_xc">query_xc</a></code> as input. Maps can be displayed in the graphic device (or Viewer if 'leaflet.map = TRUE') or saved as images in the
working directory. Note that only recordings with geographic coordinates are displayed.
</p>


<h3>Value</h3>

<p>A map of 'Xeno-Canto' recordings per species (image file), or a faceted
plot of species map(s) in the active graphic device.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search in xeno-canto
X &lt;- query_xc("Phaethornis anthophilus", download = FALSE)

# create image in R graphic device
map_xc(X, img = FALSE)

# create leaflet map
map_xc(X, leaflet.map = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='mfcc_stats'>Calculate descriptive statistics on Mel-frequency cepstral coefficients</h2><span id='topic+mfcc_stats'></span>

<h3>Description</h3>

<p><code>mfcc_stats</code> calculates descriptive statistics on Mel-frequency cepstral coefficients and its derivatives.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mfcc_stats(X, ovlp = 50, wl = 512, bp = 'frange', path = NULL, numcep = 25,
nbands = 40, parallel = 1, pb = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mfcc_stats_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;sel&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections. The output of <code><a href="#topic+auto_detec">auto_detec</a></code> can
be used as the input data frame.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows. Internally this is used to set the 'hoptime' argument in <code><a href="tuneR.html#topic+melfcc">melfcc</a></code>. Default is 50.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the spectrogram window length. Default is 512. See 'wl.freq' for setting windows length independently in the frequency domain.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz) or &quot;frange&quot; (default) to indicate that values in minimum of 'bottom.freq'
and maximum of 'top.freq' columns will be used as bandpass limits.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_numcep">numcep</code></td>
<td>
<p>Numeric vector of length 1 controlling the number of cepstra to
return (see <code><a href="tuneR.html#topic+melfcc">melfcc</a></code>).</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_nbands">nbands</code></td>
<td>
<p>Numeric vector of length 1 controlling the number of warped spectral bands to use (see <code><a href="tuneR.html#topic+melfcc">melfcc</a></code>). Default is 40.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="mfcc_stats_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <code><a href="tuneR.html#topic+melfcc">melfcc</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates descriptive statistics on Mel-frequency cepstral coefficients (MFCCs) for each of the signals (rows) in a selection
data frame. The descriptive statistics are: minimum, maximum, mean, median, skewness, kurtosis and
variance.
It also returns the mean and variance for the first and second derivatives of the coefficients. These parameters are commonly used in acoustic signal processing and detection (e.g. Salamon et al 2014).
</p>


<h3>Value</h3>

<p>A data frame containing the descriptive statistics for each of the Mel-frequency
cepstral coefficients (set by 'numcep' argument). See details.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>
<p>Lyon, R. H., &amp; Ordubadi, A. (1982). Use of cepstra in acoustical signal analysis. Journal of Mechanical Design, 104(2), 303-306.
</p>
<p>Salamon, J., Jacoby, C., &amp; Bello, J. P. (2014). A dataset and taxonomy for urban sound research. In Proceedings of the 22nd ACM international conference on Multimedia. 1041-1044.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+fix_wavs">fix_wavs</a></code>, <code><a href="#topic+remove_silence">remove_silence</a></code>,  <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# run function
mel_st &lt;- mfcc_stats(X = lbh_selec_table, pb = FALSE, path = tempdir())

head(mel_st)

# measure 12 coefficients
mel_st12 &lt;- mfcc_stats(X = lbh_selec_table, numcep = 12, pb = FALSE, path = tempdir())

 head(mel_st)
}
</code></pre>

<hr>
<h2 id='move_images'>Move/copy image files between directories</h2><span id='topic+move_images'></span>

<h3>Description</h3>

<p><code>move_images</code> moves/copies image files created by <code><a href="#topic+warbleR">warbleR</a></code> between
directories (folders).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>move_images(from = NULL, to = NULL, it = "all", cut = TRUE,
overwrite = FALSE, create.folder = TRUE, folder.name = "image_files",
parallel = 1, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="move_images_+3A_from">from</code></td>
<td>
<p>Directory path where image files to be copied are found.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="move_images_+3A_to">to</code></td>
<td>
<p>Directory path where image files will be copied to.</p>
</td></tr>
<tr><td><code id="move_images_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. &quot;all&quot;,
&quot;tiff&quot;, &quot;jpeg&quot; and &quot;pdf&quot; are admitted (&quot;all&quot; includes all the rest). Default is &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="move_images_+3A_cut">cut</code></td>
<td>
<p>Logical. Determines if files are removed from the original location after
being copied (cut) or not (just copied). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="move_images_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical. Determines if files that already exist in the destination directory
should be overwritten. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="move_images_+3A_create.folder">create.folder</code></td>
<td>
<p>Logical. Determines if files are moved to a new folder (which is named with the
&quot;folder.name&quot; argument). Ignored if 'to' is provided. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="move_images_+3A_folder.name">folder.name</code></td>
<td>
<p>Character string with the name of the new folder where the files will be
copied to. Ignored if 'to' is provided. Default is <code>"image_files"</code>.</p>
</td></tr>
<tr><td><code id="move_images_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="move_images_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aims to simplify the manipulation of the image files generated by many
of the <code><a href="#topic+warbleR">warbleR</a></code> function. It copies/cuts files between directories.
</p>


<h3>Value</h3>

<p>Image files moved into user-defined folders.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+filtersels">filtersels</a></code>
</p>
<p>Other data manipulation: 
<code><a href="#topic+open_wd">open_wd</a>()</code>,
<code><a href="#topic+split_sound_files">split_sound_files</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

  # create spectrograms
  spectrograms(lbh_selec_table[1:5, ], path = tempdir(), pb = FALSE)

  # create folder to move image files
  dir.create(file.path(tempdir(), "imgs"))

  # copy files
  move_images(cut = FALSE, from = tempdir(), to = file.path(tempdir(), "imgs"))

  # cut files
  move_images(
    cut = TRUE, from = tempdir(),
    to = file.path(tempdir(), "imgs"), overwrite = TRUE
  )

  # Check this folder
  file.path(tempdir(), "imgs")
}

</code></pre>

<hr>
<h2 id='move_imgs'>alternative name for <code><a href="#topic+move_images">move_images</a></code></h2><span id='topic+move_imgs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+move_images">move_images</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>move_imgs(
  from = NULL,
  to = NULL,
  it = "all",
  cut = TRUE,
  overwrite = FALSE,
  create.folder = TRUE,
  folder.name = "image_files",
  parallel = 1,
  pb = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+move_images">move_images</a></code> for documentation. <code><a href="#topic+move_imgs">move_imgs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='mp3_2_wav'>alternative name for <code><a href="#topic+mp32wav">mp32wav</a></code></h2><span id='topic+mp3_2_wav'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+mp32wav">mp32wav</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mp3_2_wav(
  samp.rate = NULL,
  parallel = 1,
  path = NULL,
  dest.path = NULL,
  bit.depth = 16,
  pb = TRUE,
  overwrite = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+mp32wav">mp32wav</a></code> for documentation. <code><a href="#topic+mp32wav">mp32wav</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='mp32wav'>Convert .mp3 files to .wav</h2><span id='topic+mp32wav'></span>

<h3>Description</h3>

<p><code>mp32wav</code> converts several .mp3 files in working directory to .wav format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mp32wav(samp.rate = NULL, parallel = 1, path = NULL,
 dest.path = NULL, bit.depth = 16, pb = TRUE, overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mp32wav_+3A_samp.rate">samp.rate</code></td>
<td>
<p>Sampling rate in kHz at which the .wav files should be written. If not provided the sample rate of the original .mp3 file is used. THIS FEATURE IS CURRENTLY NOT AVAILABLE. However, downsampling can be done after .mp3's have been converted using the <code><a href="#topic+fix_wavs">fix_wavs</a></code> function (which uses <a href="https://sourceforge.net/projects/sox/">SOX</a> instead). Default is <code>NULL</code> (e.g. keep original sampling rate).</p>
</td></tr>
<tr><td><code id="mp32wav_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="mp32wav_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the .mp3 files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="mp32wav_+3A_dest.path">dest.path</code></td>
<td>
<p>Character string containing the directory path where the .wav files will be saved.
If <code>NULL</code> (default) then the folder containing the sound files will be used.</p>
</td></tr>
<tr><td><code id="mp32wav_+3A_bit.depth">bit.depth</code></td>
<td>
<p>Character string containing the units to be used for amplitude normalization. Check
<code><a href="tuneR.html#topic+normalize">normalize</a></code> for details. Default is 16.</p>
</td></tr>
<tr><td><code id="mp32wav_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="mp32wav_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical. Control whether a .wav sound file that is already in the working directory should be
overwritten.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will convert all mp3 files in  working directory or 'path' supplied to wav format. <a href="https://cran.r-project.org/package=bioacoustics">bioacoustics package</a> must be installed when changing sampling rates (i.e. if 'samp.rate' is supplied). Note that sound files are normalized using <code><a href="tuneR.html#topic+normalize">normalize</a></code> so they can be written by <code><a href="tuneR.html#topic+writeWave">writeWave</a></code>.
</p>
<p>convert all .mp3 files in working directory to .wav format. Function used internally to read .mp3 files (<code><a href="tuneR.html#topic+readMP3">readMP3</a></code>) sometimes crashes.
</p>


<h3>Value</h3>

<p>.wav files saved in the working directory with same name as original mp3 files.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download mp3 files from xeno-canto
query_xc(qword = "Phaethornis aethopygus", download = TRUE, path = tempdir())

# Convert all files to .wav format
mp32wav(path = tempdir(), dest.path = tempdir())

# check this folder!!
tempdir()

## End(Not run)

</code></pre>

<hr>
<h2 id='multi_DTW'>A wrapper on <code><a href="dtw.html#topic+dtwDist">dtwDist</a></code> for comparing multivariate contours</h2><span id='topic+multi_DTW'></span>

<h3>Description</h3>

<p><code>multi_DTW</code> is a wrapper on <code><a href="dtw.html#topic+dtwDist">dtwDist</a></code> that simplify applying dynamic time warping on multivariate contours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_DTW(ts.df1 = NULL, ts.df2 = NULL, pb = TRUE,  parallel = 1,
window.type = "none", open.end = FALSE, scale = FALSE, dist.mat = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_DTW_+3A_ts.df1">ts.df1</code></td>
<td>
<p>Optional. Data frame with frequency contour time series of signals to be compared.</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_ts.df2">ts.df2</code></td>
<td>
<p>Optional. Data frame with frequency contour time series of signals to be compared.</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>. Note that progress bar is only used
when parallel = 1.</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing). Not available in Windows OS.</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_window.type">window.type</code></td>
<td>
<p><code><a href="dtw.html#topic+dtw">dtw</a></code> windowing control parameter. Character: &quot;none&quot;, &quot;itakura&quot;, or a function (see <code><a href="dtw.html#topic+dtw">dtw</a></code>).</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_open.end">open.end</code></td>
<td>
<p><code><a href="dtw.html#topic+dtw">dtw</a></code> control parameter. Performs
open-ended alignments (see <code><a href="dtw.html#topic+dtw">dtw</a></code>).</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_scale">scale</code></td>
<td>
<p>Logical. If <code>TRUE</code> dominant frequency values are z-transformed using the <code><a href="base.html#topic+scale">scale</a></code> function, which &quot;ignores&quot; differences in absolute frequencies between the signals in order to focus the
comparison in the frequency contour, regardless of the pitch of signals. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_dist.mat">dist.mat</code></td>
<td>
<p>Logical controlling whether a distance matrix (<code>TRUE</code>,
default) or a tabular data frame (<code>FALSE</code>) is returned.</p>
</td></tr>
<tr><td><code id="multi_DTW_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for customizing
graphical output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extracts the dominant frequency values as a time series and
then calculates the pairwise acoustic dissimilarity using dynamic time warping.
The function uses the <code><a href="stats.html#topic+approxfun">approx</a></code> function to interpolate values between dominant
frequency  measures. If 'img' is  <code>TRUE</code> the function also produces image files
with the spectrograms of the signals listed in the input data frame showing the
location of the dominant frequencies.
</p>


<h3>Value</h3>

<p>A matrix with the pairwise dissimilarity values. If img is
<code>FALSE</code> it also produces image files with the spectrograms of the signals listed in the
input data frame showing the location of the dominant frequencies.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+freq_ts">freq_ts</a></code>
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+color_spectro">color_spectro</a>()</code>,
<code><a href="#topic+freq_DTW">freq_DTW</a>()</code>,
<code><a href="#topic+phylo_spectro">phylo_spectro</a>()</code>,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a>()</code>,
<code><a href="#topic+spectrograms">spectrograms</a>()</code>,
<code><a href="#topic+track_freq_contour">track_freq_contour</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# load data
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))

writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound files
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# measure
df &lt;- freq_ts(X = lbh_selec_table, threshold = 10, img = FALSE, path = tempdir())
se &lt;- freq_ts(X = lbh_selec_table, threshold = 10, img = FALSE, path = tempdir(), type = "entropy")

# run function
multi_DTW(df, se)

## End(Not run)

</code></pre>

<hr>
<h2 id='new_function_names'>Data frame detailing function name changes</h2><span id='topic+new_function_names'></span>

<h3>Description</h3>

<p>A data frame containing the old and new names for <code><a href="#topic+warbleR">warbleR</a></code> functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(new_function_names)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 34 rows and 2 columns.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>

<hr>
<h2 id='open_wd'>Open working directory</h2><span id='topic+open_wd'></span>

<h3>Description</h3>

<p><code>open_wd</code> opens the working directory in the default file browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_wd(path = getwd(), verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_wd_+3A_path">path</code></td>
<td>
<p>Directory path to be opened. By default it's the working directory. 
'wav.path' set by <code><a href="#topic+warbleR_options">warbleR_options</a></code> is ignored in this case.</p>
</td></tr>
<tr><td><code id="open_wd_+3A_verbose">verbose</code></td>
<td>
<p>Logical to control whether the 'path' is printed in the console. Defaut is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function opens the working directory using the default file browser 
and prints the working directory in the R console. This function aims to simplify
the manipulation of sound files and other files produced by many of the <code><a href="#topic+warbleR">warbleR</a></code> function.
</p>


<h3>Value</h3>

<p>Opens the working directory using the default file browser.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+move_imgs">move_imgs</a></code>
</p>
<p>Other data manipulation: 
<code><a href="#topic+move_images">move_images</a>()</code>,
<code><a href="#topic+split_sound_files">split_sound_files</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
open_wd()
}

</code></pre>

<hr>
<h2 id='overlapping_sels'>Find overlapping selections</h2><span id='topic+overlapping_sels'></span>

<h3>Description</h3>

<p><code>overlapping_sels</code> finds which selections overlap in time within a given sound file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlapping_sels(X, index = FALSE, pb = TRUE, max.ovlp = 0, relabel = FALSE,
drop = FALSE, priority = NULL, priority.col = NULL, unique.labs = NULL,
indx.row = FALSE, parallel = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlapping_sels_+3A_x">X</code></td>
<td>
<p>'selection_table' object or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;selec&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_index">index</code></td>
<td>
<p>Logical. Indicates if only the index of the overlapping selections would be returned.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_max.ovlp">max.ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying the maximum overlap allowed (in seconds)
. Default is 0.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_relabel">relabel</code></td>
<td>
<p>Logical. If <code>TRUE</code> then selection names ('selec' column) are reset within each sound files.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_drop">drop</code></td>
<td>
<p>Logical. If <code>TRUE</code>, when 2 or more selections overlap the function will remove
all but one of the overlapping selection. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_priority">priority</code></td>
<td>
<p>Character vector. Controls the priority criteria used for removing overlapped selections. It
must list the levels of the column used to determine priority (argument priority.col) in the desired
priority order. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_priority.col">priority.col</code></td>
<td>
<p>Character vector of length 1 with the name of the column use to determine the priority of
overlapped selections. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_unique.labs">unique.labs</code></td>
<td>
<p>DEPRECATED.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_indx.row">indx.row</code></td>
<td>
<p>Logical. If <code>TRUE</code> then a character column with the indices of all selections that overlapped with
each selection is added to the ouput data frame (if <code>index = TRUE</code>). For instance, if the selections in rows 1,2
and 3 all overlapped with each other, the 'indx.row' value would be &quot;1/2/3&quot; for all. However, if selection 3 only overlaps
with 2 but not with 1, then it returns, &quot;1/2&quot; for row 1, &quot;1/2/3&quot; for row 2, and &quot;2/3&quot; for row 3. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="overlapping_sels_+3A_verbose">verbose</code></td>
<td>
<p>Logical to control if messages are printed to the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function detects selections within a selection table that overlap in time. Selections must be
listed in a data frame similar to <code><a href="#topic+lbh_selec_table">lbh_selec_table</a></code>. Note that row names are set to <code>1:nrow(X)</code>.
</p>


<h3>Value</h3>

<p>A data frame with the columns in X plus an additional column ('ovlp.sels') indicating
which selections overlap. For instance, if the selections in rows 1,2
overlap and 2 and 3 also overlap, the 'ovlp.sels' label would be the same for all 3 selections. If
<code>drop = TRUE</code> only the non-overlapping selections are returned.and if 2 or more selections
overlap only the first one is kept. The arguments 'priority' and 'priority.col' can be used to modified the criterium for droping overlapping selections.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+filtersels">filtersels</a></code>, <code><a href="#topic+lbh_selec_table">lbh_selec_table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # no overlap
  overlapping_sels(X = lbh_selec_table)

  # modified lbh_selec_table to make the first and second selection overlap
  Y &lt;- lbh_selec_table
  Y$end[4] &lt;- 1.5

  overlapping_sels(X = Y)

  # drop overlapping
  overlapping_sels(X = Y, drop = TRUE)

  # get index instead
  overlapping_sels(X = Y, index = TRUE)
}
</code></pre>

<hr>
<h2 id='ovlp_sels'>alternative name for <code><a href="#topic+overlapping_sels">overlapping_sels</a></code></h2><span id='topic+ovlp_sels'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+overlapping_sels">overlapping_sels</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ovlp_sels(
  X,
  index = FALSE,
  pb = TRUE,
  max.ovlp = 0,
  relabel = FALSE,
  drop = FALSE,
  priority = NULL,
  priority.col = NULL,
  unique.labs = NULL,
  indx.row = FALSE,
  parallel = 1,
  verbose = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+overlapping_sels">overlapping_sels</a></code> for documentation. <code><a href="#topic+ovlp_sels">ovlp_sels</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='phylo_spectro'>Add spectrograms onto phylogenetic trees</h2><span id='topic+phylo_spectro'></span>

<h3>Description</h3>

<p><code>phylo_spectro</code> Add spectrograms to the tips of an objects of class phylo.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo_spectro(X, tree, type = "phylogram", par.mar = rep(1, 4), 
size = 1, offset = 0, path = NULL, ladder = NULL, horizontal = TRUE, axis = TRUE, 
box = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo_spectro_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame containing columns for sound file name 
(sound.files), selection number (selec), and start and end time of signals (start and end). 
'top.freq' and 'bottom.freq' columns are optional. In addition, the data frame must include the column 'tip.label' that contains the names of the tip labels found in the tree (e.g. '<code>tree$tip.label</code>). This column is used to match rows and tip labels. If using an 
'extended_selection_table' the sound files are not required (see <code><a href="#topic+selection_table">selection_table</a></code>).</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_tree">tree</code></td>
<td>
<p>Object of class 'phylo' (i.e. a phylogenetic tree). Ultrametric trees may produce better results.
If <code>NULL</code> (default) then the current working directory is used. Tip labels must match the names provided in the 'tip.label' column in 'X' (see 'X' argument).</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_type">type</code></td>
<td>
<p>Character string of length 1 specifying the type of phylogeny to be drawn 
(as in <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>). Only 'phylogram' (default) and 'fan' are allowed.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_par.mar">par.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is <code>rep(1, 4)</code>. Specifies the number of lines 
in inner plot margins where axis labels fall, with form c(bottom, left, top, right). 
See <code><a href="graphics.html#topic+par">par</a></code>. See 'inner.par' argument for controlling spectrogram margins.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_size">size</code></td>
<td>
<p>Numeric vector of length 1 controlling the relative size of spectrograms. Higher numbers increase the height of spectrograms. Default is 1. 
Numbers between range <code>c(&gt;0, Inf)</code> are allowed.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_offset">offset</code></td>
<td>
<p>Numeric vector of length 1 controlling the space between tips and spectrograms. Default is 0.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located. 
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_ladder">ladder</code></td>
<td>
<p>Character string controlling whether the phylogeny is ladderized (i.e. the internal structure of the 
tree is reorganized to get the ladderized effect when plotted). Only 'left' of 'right' values are accepted. Default is 
<code>NULL</code> (no ladderization). See <code><a href="ape.html#topic+ladderize">ladderize</a></code> for more details.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_horizontal">horizontal</code></td>
<td>
<p>Logical. Controls whether spectrograms in a fan phylogeny are place in a horizontal position 
<code>FALSE</code> or in the same angle as the tree tips. Currently only horizontal spectrograms are available.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_axis">axis</code></td>
<td>
<p>Logical to control if the Y and X axis of spectrograms are plotted (see <code><a href="graphics.html#topic+box">box</a></code>). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_box">box</code></td>
<td>
<p>Logical to control if the box around spectrograms is plotted (see <code><a href="graphics.html#topic+box">box</a></code>). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="phylo_spectro_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal spectrogram 
creating function (<code><a href="#topic+spectrograms">spectrograms</a></code>) or phylogeny plotting function (<code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>) for 
customizing graphical output. Only rightwards phylogenies can be plotted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function add the spectrograms of sounds annotated in a selection table ('X' argument) onto the tips of a phylogenetic tree. 
The 'tip.label' column in 'X' is used to match spectrograms and tree tips. The function uses internally the <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code> function to plot the tree 
and the <code><a href="#topic+spectrograms">spectrograms</a></code> function to create the spectrograms. Arguments for both of these functions
can be provided for further customization.
</p>


<h3>Value</h3>

<p>A phylogenetic tree with spectrograms on tree tips is plotted in the current graphical device.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+spectrograms">spectrograms</a></code>, <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+color_spectro">color_spectro</a>()</code>,
<code><a href="#topic+freq_DTW">freq_DTW</a>()</code>,
<code><a href="#topic+multi_DTW">multi_DTW</a>()</code>,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a>()</code>,
<code><a href="#topic+spectrograms">spectrograms</a>()</code>,
<code><a href="#topic+track_freq_contour">track_freq_contour</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{ 

# First set empty folder


# save example sound files
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

# set spectrogram options (can be done at the phylo_spectro() function too)
warbleR_options(wl = 200, ovlp = 90, flim = "frange", wav.path = tempdir())

# subset example selection table
X &lt;- lbh_selec_table[1:8, ]

# create random tree (need ape to be installed)
set.seed(1)
tree &lt;- ape::rtree(nrow(X))

# Force tree to be ultrametric
tree &lt;- ape::chronoMPL(tree)

# add tip label column to example selection table (just for the sake of the example)
X$tip.label &lt;- tree$tip.label

# print phylogram with spectros
phylo_spectro(X = X, tree = tree, par.mar = c(0, 0, 0, 8), size = 2)

# no margin in spectrograms and showing tip labels (higher offset)
phylo_spectro(X = X, tree = tree, offset = 0.1, par.mar = c(0, 0, 0, 6), 
inner.mar = rep(0, 4), size = 2, box = FALSE, axis = FALSE)

# print fan tree and no margin in spectrograms
phylo_spectro(X = X, tree = tree, offset = 0.6, par.mar = rep(3, 4),
inner.mar = rep(0, 4), size = 2, type = "fan", show.tip.label = FALSE, box = FALSE, axis = FALSE)

# changing edge color and witdh
phylo_spectro(X = X, tree = tree, offset = 0.2, par.mar = rep(3, 4), inner.mar = rep(0, 4), 
size = 2, type = "fan", show.tip.label = FALSE, edge.color = "red", edge.width = 2, 
box = FALSE, axis = FALSE)

# plotting a tree representing cross-correlation distances 
xcorr_mat &lt;- cross_correlation(X, bp = c(1, 10))

xc.tree &lt;- ape::chronoMPL(ape::as.phylo(hclust(as.dist(1 - xcorr_mat))))

X$tip.label &lt;- xc.tree$tip.label

phylo_spectro(X = X, tree = xc.tree, offset = 0.03, par.mar = rep(3, 4), 
inner.mar = rep(0, 4), size = 0.3, type = "fan", show.tip.label = FALSE, 
edge.color = "red", edge.width = 2, box = FALSE, axis = FALSE)
  
}
</code></pre>

<hr>
<h2 id='plot_coordination'>Coordinated singing graphs</h2><span id='topic+plot_coordination'></span>

<h3>Description</h3>

<p><code>plot_coordination</code> creates graphs of coordinated singing and highlights the signals that overlap
in time. The signals are represented by polygons of different colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_coordination(X, only.coor = FALSE, ovlp = TRUE, xl = 1, res= 80, it = "jpeg",
img = TRUE, tlim = NULL, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_coordination_+3A_x">X</code></td>
<td>
<p>Data frame containing columns for singing event (sing.event),
individual (indiv), and start and end time of signal (start and end).</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_only.coor">only.coor</code></td>
<td>
<p>Logical. If <code>TRUE</code> only the segment in which both individuals are singing is
included (solo singing is removed). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_ovlp">ovlp</code></td>
<td>
<p>Logical. If <code>TRUE</code> the vocalizations that overlap in time are highlighted.
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_xl">xl</code></td>
<td>
<p>Numeric vector of length 1, a constant by which to scale
spectrogram width. Default is 1.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution. Default is 80.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_img">img</code></td>
<td>
<p>Logical argument. If <code>FALSE</code>, image files are not produced and
the graphs are shown in the current graphic device. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_tlim">tlim</code></td>
<td>
<p>Numeric vector of length 2 indicating the start and end time of the coordinated singing events
to be displayed in the graphs.</p>
</td></tr>
<tr><td><code id="plot_coordination_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is
<code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides visualization for coordination of acoustic signals. Signals are shown as
polygon across a time axis. It also shows which signals overlap, the amount of overlap, and
highlights the individual responsible for the overlap using a color code. The width of the polygons
depicting the time of overlap.
</p>


<h3>Value</h3>

<p>The function returns a list of graphs, one for each singing event in the input data frame. The graphs can be plotted by simply calling the list. If 'img' is <code>TRUE</code> then the graphs are also saved in the working
directory as files.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# load simulate singing events (see data documentation)
data(sim_coor_sing)

#' # make plot_coordination in graphic device format
cgs &lt;- plot_coordination(X = sim_coor_sing, ovlp = TRUE, only.coor = FALSE, img = FALSE)

cgs

## End(Not run)

</code></pre>

<hr>
<h2 id='print.autodetec.output'>print method for class <code>autodetec.output</code></h2><span id='topic+print.autodetec.output'></span>

<h3>Description</h3>

<p>print method for class <code>autodetec.output</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'autodetec.output'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.autodetec.output_+3A_x">x</code></td>
<td>
<p>Object of class <code>autodetec.output</code>, generated by <code><a href="#topic+auto_detec">auto_detec</a></code>.</p>
</td></tr>
<tr><td><code id="print.autodetec.output_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods. Ignored when printing selection tables.</p>
</td></tr>
</table>

<hr>
<h2 id='print.extended_selection_table'>print method for class <code>extended_selection_table</code></h2><span id='topic+print.extended_selection_table'></span>

<h3>Description</h3>

<p>print method for class <code>extended_selection_table</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'extended_selection_table'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.extended_selection_table_+3A_x">x</code></td>
<td>
<p>Object of class <code>extended_selection_table</code>, generated by <code><a href="#topic+selection_table">selection_table</a></code>.</p>
</td></tr>
<tr><td><code id="print.extended_selection_table_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods. Ignored when printing extended selection tables.</p>
</td></tr>
</table>

<hr>
<h2 id='print.find_peaks.output'>print method for class <code>xcorr.output</code></h2><span id='topic+print.find_peaks.output'></span>

<h3>Description</h3>

<p>print method for class <code>xcorr.output</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'find_peaks.output'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.find_peaks.output_+3A_x">x</code></td>
<td>
<p>Object of class <code>find_peaks.output</code>, generated by <code><a href="#topic+find_peaks">find_peaks</a></code>.</p>
</td></tr>
<tr><td><code id="print.find_peaks.output_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods. Ignored when printing selection tables.</p>
</td></tr>
</table>

<hr>
<h2 id='print.selection_table'>print method for class <code>selection_table</code></h2><span id='topic+print.selection_table'></span>

<h3>Description</h3>

<p>print method for class <code>selection_table</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selection_table'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.selection_table_+3A_x">x</code></td>
<td>
<p>Object of class <code>selection_table</code>, generated by <code><a href="#topic+selection_table">selection_table</a></code>.</p>
</td></tr>
<tr><td><code id="print.selection_table_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods. Ignored when printing selection tables.</p>
</td></tr>
</table>

<hr>
<h2 id='print.xcorr.output'>print method for class <code>xcorr.output</code></h2><span id='topic+print.xcorr.output'></span>

<h3>Description</h3>

<p>print method for class <code>xcorr.output</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'xcorr.output'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.xcorr.output_+3A_x">x</code></td>
<td>
<p>Object of class <code>xcorr.output</code>, generated by <code><a href="#topic+cross_correlation">cross_correlation</a></code>.</p>
</td></tr>
<tr><td><code id="print.xcorr.output_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods. Ignored when printing selection tables.</p>
</td></tr>
</table>

<hr>
<h2 id='querxc'>alternative name for <code><a href="#topic+query_xc">query_xc</a></code></h2><span id='topic+querxc'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+query_xc">query_xc</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>querxc(
  qword,
  download = FALSE,
  X = NULL,
  file.name = c("Genus", "Specific_epithet"),
  parallel = 1,
  path = NULL,
  pb = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+query_xc">query_xc</a></code> for documentation. <code><a href="#topic+querxc">querxc</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='query_xc'>Access 'Xeno-Canto' recordings and metadata</h2><span id='topic+query_xc'></span>

<h3>Description</h3>

<p><code>query_xc</code> downloads recordings and metadata from <a href="https://www.xeno-canto.org/">Xeno-Canto</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>query_xc(qword, download = FALSE, X = NULL, file.name = c("Genus", "Specific_epithet"),
parallel = 1, path = NULL, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="query_xc_+3A_qword">qword</code></td>
<td>
<p>Character vector of length one indicating the genus, or genus and
species, to query 'Xeno-Canto' database. For example, <em>Phaethornis</em> or <em>Phaethornis longirostris</em>.
More complex queries can be done by using search terms that follow the
xeno-canto advance query syntax. This syntax uses tags to search within a particular aspect of the recordings
(e.g. country, location, sound type). Tags are of the form tag:searchterm'. For instance, 'type:song'
will search for all recordings in which the sound type description contains the word 'song'.
Several tags can be included in the same query. The query &quot;phaethornis cnt:belize' will only return
results for birds in the genus <em>Phaethornis</em> that were recorded in Belize. Queries are case insensitive. Make sure taxonomy related tags (Genus or scientific name) are found first in multi-tag queries. See <a href="https://www.xeno-canto.org/help/search">Xeno-Canto's search help</a> for a full description and see examples below
for queries using terms with more than one word.</p>
</td></tr>
<tr><td><code id="query_xc_+3A_download">download</code></td>
<td>
<p>Logical argument. If <code>FALSE</code> only the recording file names and
associated metadata are downloaded. If <code>TRUE</code>, recordings are also downloaded to the working
directory as .mp3 files. Default is <code>FALSE</code>. Note that if the recording is already in the
working directory (as when the downloading process has been interrupted) it will be skipped.
Hence, resuming downloading processes will not start from scratch.</p>
</td></tr>
<tr><td><code id="query_xc_+3A_x">X</code></td>
<td>
<p>Data frame with a 'Recording_ID' column and any other column listed in the file.name argument. Only the recordings listed in the data frame
will be download (<code>download</code> argument is automatically set to <code>TRUE</code>). This can be used to select
the recordings to be downloaded based on their attributes.</p>
</td></tr>
<tr><td><code id="query_xc_+3A_file.name">file.name</code></td>
<td>
<p>Character vector indicating the tags (or column names) to be included in the sound file names (if download = <code>TRUE</code>). Several tags can be included. If <code>NULL</code> only the 'Xeno-Canto' recording identification number (&quot;Recording_ID&quot;) is used. Default is c(&quot;Genus&quot;, &quot;Specific_epithet&quot;).
Note that recording id is always used (whether or not is listed by users) to avoid duplicated names.</p>
</td></tr>
<tr><td><code id="query_xc_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied when downloading mp3 files.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing). Applied both when getting metadata and downloading files.</p>
</td></tr>
<tr><td><code id="query_xc_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files will be saved.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="query_xc_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is slated for deprecation in future versions. We recommend exploring the advanced functionalities offered by the <a href="https://github.com/maRce10/suwo">suwo package</a> as an alternative solution. This function queries for avian vocalization recordings in the open-access
online repository <a href="https://www.xeno-canto.org/">Xeno-Canto</a>. It can return recordings metadata
or download the associated sound files. Complex queries can be done by using search terms that follow the
xeno-canto advance query syntax (check &quot;qword&quot; argument description).
Files are double-checked after downloading and &quot;empty&quot; files are re-downloaded.
File downloading process can be interrupted and resume later as long as the working directory is the same.
Maps of recording coordinates can be produced using
<code><a href="#topic+map_xc">map_xc</a></code>.
</p>


<h3>Value</h3>

<p>If X is not provided the function returns a data frame with the following recording information: recording ID, Genus, Specific epithet, Subspecies, English name, Recordist, Country, Locality, Latitude, Longitude, Vocalization type, Audio file, License, URL, Quality, Time, Date. Sound files in .mp3 format are downloaded into the working directory if download = <code>TRUE</code> or if X is provided; a column indicating the  names of the downloaded files is included in the output data frame.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+map_xc">map_xc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# search without downloading
df1 &lt;- query_xc(qword = "Phaethornis anthophilus", download = FALSE)
View(df1)

# downloading files
query_xc(qword = "Phaethornis anthophilus", download = TRUE, path = tempdir())

# check this folder
tempdir()

## search using xeno-canto advance query ###
orth.pap &lt;- query_xc(qword = "gen:orthonyx cnt:papua loc:tari", download = FALSE)

# download file using the output data frame as input
query_xc(X = orth.pap, path = tempdir())

# use quotes for queries with more than 1 word (e.g. Costa Rica),note that the
# single quotes are used for the whole 'qword' and double quotes for the 2-word term inside
# Phaeochroa genus in Costa Rica
phae.cr &lt;- query_xc(qword = 'gen:phaeochroa cnt:"costa rica"', download = FALSE)

# several terms can be searched for in the same field
# search for all female songs in sound type
femsong &lt;- query_xc(qword = "type:song type:female", download = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='rbind.extended_selection_table'>rbind method for class <code>extended_selection_table</code></h2><span id='topic+rbind.extended_selection_table'></span>

<h3>Description</h3>

<p>rbind method for class <code>extended_selection_table</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'extended_selection_table'
rbind(..., deparse.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind.extended_selection_table_+3A_...">...</code></td>
<td>
<p>Objects of class <code>extended_selection_table</code>, generated by <code><a href="#topic+selection_table">selection_table</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='rbind.selection_table'>rbind method for class <code>selection_table</code></h2><span id='topic+rbind.selection_table'></span>

<h3>Description</h3>

<p>rbind method for class <code>selection_table</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'selection_table'
rbind(..., deparse.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbind.selection_table_+3A_...">...</code></td>
<td>
<p>Objects of class <code>selection_table</code>, generated by <code><a href="#topic+selection_table">selection_table</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='read_sound_file'>An extended version of read_wave that reads several sound file formats and files from selection tables</h2><span id='topic+read_sound_file'></span>

<h3>Description</h3>

<p><code>read_sound_file</code> reads several sound file formats as well as files referenced in selection tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_sound_file(X, index, from = X$start[index], to = X$end[index],
channel = X$channel[index], header = FALSE, path = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_sound_file_+3A_x">X</code></td>
<td>
<p>'data.frame', 'selection_table' or 'extended_selection_table' containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signals (start and end). Alternatively, the name of a sound file or URL address to sound file can be provided. The function can read sound files in 'wav', 'mp3', 'flac' and 'wac' format. The file name can contain the directory path.
'top.freq' and 'bottom.freq' columns are optional. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="read_sound_file_+3A_index">index</code></td>
<td>
<p>Index of the selection in 'X' that will be read. Ignored if 'X' is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="read_sound_file_+3A_from">from</code></td>
<td>
<p>Where to start reading, in seconds. Default is <code>X$start[index]</code>.</p>
</td></tr>
<tr><td><code id="read_sound_file_+3A_to">to</code></td>
<td>
<p>Where to stop reading, in seconds. Default is <code>X$end[index]</code>.</p>
</td></tr>
<tr><td><code id="read_sound_file_+3A_channel">channel</code></td>
<td>
<p>Channel to be read from sound file (1 = left, 2 = right, or higher number for multichannel waves). Default is <code>X$channel[index]</code>. If a 'channel' column does not exist it will read the first channel.</p>
</td></tr>
<tr><td><code id="read_sound_file_+3A_header">header</code></td>
<td>
<p>If <code>TRUE</code>, only the header information of the Wave object is returned, otherwise (the default) the whole Wave object.</p>
</td></tr>
<tr><td><code id="read_sound_file_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used. If 'X' refers to a sound file including its directory 'path' is ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a wrapper for <code><a href="tuneR.html#topic+readWave">readWave</a></code> that read sound files, including those referenced in selection tables. It
is also used internally by warbleR functions to read wave objects from extended selection tables (see <code><a href="#topic+selection_table">selection_table</a></code> for details). For reading 'flac' files on windows the path to the .exe is required. This can be set globally using the 'flac.path' argument in <code><a href="#topic+warbleR_options">warbleR_options</a></code>. Note that reading 'flac' files requires creating a temporary copy in 'wav' format, which can be particularly slow for long files.
</p>


<h3>Value</h3>

<p>An object of class &quot;Wave&quot;.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# write wave files with lower case file extension
data(list = c("Phae.long1"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))

# read from selection table
read_sound_file(X = lbh_selec_table, index = 1, path = tempdir())

# from extended selection table
library(NatureSounds)
read_sound_file(X = lbh.est, index = 1)

# read from selection table
read_sound_file(X = lbh_selec_table, index = 1, path = tempdir())

# read WAV
filepath &lt;- system.file("extdata", "recording.wav", package = "bioacoustics")
read_sound_file(filepath)

# read MP3
filepath &lt;- system.file("extdata", "recording.mp3", package = "bioacoustics")
read_sound_file(filepath)

# read WAC
filepath &lt;- system.file("extdata", "recording_20170716_230503.wac", package = "bioacoustics")
read_sound_file(filepath, from = 0, to = 0.2)

# URL file
read_sound_file(X = "https://www.xeno-canto.org/513948/download")

## End(Not run)

</code></pre>

<hr>
<h2 id='read_wave'>A wrapper for tuneR's readWave that read sound files listed within selection tables</h2><span id='topic+read_wave'></span>

<h3>Description</h3>

<p><code>read_wave</code> is a wrapper for tuneR's <code><a href="tuneR.html#topic+readWave">readWave</a></code> function that read sound files listed in data frames and selection tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_wave(X, index, from = X$start[index], to = X$end[index], channel = NULL,
header = FALSE, path = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_wave_+3A_x">X</code></td>
<td>
<p>'data.frame', 'selection_table' or 'extended_selection_table' containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signals (start and end). Alternatively, the name of a '.wav' file or URL address to a '.wav' or '.mp3' file can be provided. The file name can contain the directory path.
'top.freq' and 'bottom.freq' columns are optional. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="read_wave_+3A_index">index</code></td>
<td>
<p>Index of the selection in 'X' that will be read. Ignored if 'X' is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="read_wave_+3A_from">from</code></td>
<td>
<p>Where to start reading, in seconds. Default is <code>X$start[index]</code>.</p>
</td></tr>
<tr><td><code id="read_wave_+3A_to">to</code></td>
<td>
<p>Where to stop reading, in seconds. Default is <code>X$end[index]</code>.  <code>Inf</code> can be used for reading the entire sound file (when 'X' is a sound file name),</p>
</td></tr>
<tr><td><code id="read_wave_+3A_channel">channel</code></td>
<td>
<p>Channel to be read from sound file (1 = left, 2 = right, or higher number for multichannel waves). If
<code>NULL</code> (default) or higher than the number of channels in a wave then the first channel is used. Only applies to '.wav' files in local directories.</p>
</td></tr>
<tr><td><code id="read_wave_+3A_header">header</code></td>
<td>
<p>If <code>TRUE</code>, only the header information of the Wave object is returned, otherwise (the default) the whole Wave object.</p>
</td></tr>
<tr><td><code id="read_wave_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used. If 'X' refers to a sound file including its directory 'path' is ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is a wrapper for <code><a href="tuneR.html#topic+readWave">readWave</a></code> that read sound files listed within selection tables. It
is also used internally by warbleR functions to read wave objects from extended selection tables (see <code><a href="#topic+selection_table">selection_table</a></code> for details).
</p>


<h3>Value</h3>

<p>An object of class &quot;Wave&quot;.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # write wave files with lower case file extension
  data(list = c("Phae.long1"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))

  # read from selection table
  read_wave(X = lbh_selec_table, index = 1, path = tempdir())

  # from extended selection table
  library(NatureSounds)
  read_wave(X = lbh.est, index = 1)

  # read WAV
  filepath &lt;- system.file("extdata", "recording.wav", package = "bioacoustics")
  read_wave(filepath)

  # read MP3
  filepath &lt;- system.file("extdata", "recording.mp3", package = "bioacoustics")
  read_wave(filepath)

  # URL file
  read_wave(X = "https://www.xeno-canto.org/513948/download")
}

</code></pre>

<hr>
<h2 id='remove_channels'>Remove channels in wave files</h2><span id='topic+remove_channels'></span>

<h3>Description</h3>

<p><code>remove_channels</code> remove channels in wave files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_channels(files = NULL, channels, path = NULL, parallel = 1, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_channels_+3A_files">files</code></td>
<td>
<p>Character vector indicating the files that will be analyzed. If not provided. Optional.
then all wave files in the working directory (or path) will be processed.</p>
</td></tr>
<tr><td><code id="remove_channels_+3A_channels">channels</code></td>
<td>
<p>Numeric vector indicating the index (or channel number) for the channels that will be kept (left = 1, right = 2; 3 to inf for multichannel sound files).</p>
</td></tr>
<tr><td><code id="remove_channels_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="remove_channels_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="remove_channels_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function removes channels from wave files. It works on regular and
multichannel wave files. Converted files are saved in a new directory (&quot;converted_sound_files&quot;)
and original files are not modified.
</p>


<h3>Value</h3>

<p>Sound files that have been converted are saved in the new
folder &quot;converted_sound_files&quot;. If 'img = TRUE' then spectrogram images highlighting the silence segments
that were removed are also saved.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+fix_wavs">fix_wavs</a></code>, <code><a href="#topic+info_wavs">info_wavs</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# save sound file examples
data("Phae.long1")
Phae.long1.2 &lt;- stereo(Phae.long1, Phae.long1)

writeWave(Phae.long1.2, file.path(tempdir(), "Phae.long1.2.wav"))

remove_channels(channels = 1, path = tempdir())

#check this floder
tempdir()
}

</code></pre>

<hr>
<h2 id='remove_silence'>Remove silence in wave files</h2><span id='topic+remove_silence'></span>

<h3>Description</h3>

<p><code>remove_silence</code> Removes silences in wave files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_silence(path = NULL, min.sil.dur = 2, img = TRUE, it = "jpeg", flim = NULL,
files = NULL, flist = NULL, parallel = 1, pb = TRUE, downsample = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_silence_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_min.sil.dur">min.sil.dur</code></td>
<td>
<p>Numeric. Controls the minimum duration of silence segments that would be removed.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_img">img</code></td>
<td>
<p>Logical argument. If <code>FALSE</code>, image files are not produced. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_it">it</code></td>
<td>
<p>A character vector of length 1  giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 indicating the highest and lowest
frequency limits (kHz) of the spectrogram as in
<code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>. Ignored if 'img = FALSE'.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_files">files</code></td>
<td>
<p>character vector or factor indicating the subset of files that will be analyzed. If not provided
then all wave files in the working directory (or path) will be processed.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_flist">flist</code></td>
<td>
<p>DEPRECATED. Please use 'files' instead.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="remove_silence_+3A_downsample">downsample</code></td>
<td>
<p>Logical indicating whether files are downsampled to a 5000 kHz sampling rate. This can be used to speed up the process, but will make the function ignore sound/noise above 2500 kHz. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function removes silence segments (i.e. segments with very low amplitude values) from wave files.
</p>


<h3>Value</h3>

<p>Sound files for which silence segments have been removed are saved in the new
folder &quot;silence-removed_files&quot; in .wav format. If 'img = TRUE' then spectrogram images highlighting the silence segments
that were removed are also saved.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+fix_wavs">fix_wavs</a></code>, <code><a href="#topic+auto_detec">auto_detec</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# save sound file examples
data(list = c("Phae.long1", "Phae.long2","lbh_selec_table"))
sil &lt;- silence(samp.rate = 22500, duration = 3, xunit = "time")


wv1 &lt;- pastew(pastew(Phae.long1, sil, f = 22500, output = "Wave"),
Phae.long2, f = 22500, output = "Wave")

#check silence in between amplitude peaks
env(wv1)

 #save wave file
 writeWave(object = wv1, filename = file.path(tempdir(), "wv1.wav"),
  extensible = FALSE)

#remove silence
# remove_silence(files = "wv1.wav", pb = FALSE, path = tempdir())

#check this floder
tempdir()
}

</code></pre>

<hr>
<h2 id='rename_est_waves'>Rename wave objects and associated metadata in extended selection tables</h2><span id='topic+rename_est_waves'></span>

<h3>Description</h3>

<p><code>rename_est_waves</code> rename wave objects and associated metadata in extended selection tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_est_waves(X, new.sound.files, new.selec = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_est_waves_+3A_x">X</code></td>
<td>
<p>object of class 'extended_selection_table'.</p>
</td></tr>
<tr><td><code id="rename_est_waves_+3A_new.sound.files">new.sound.files</code></td>
<td>
<p>Character vector of length equals to the number of wave objects in the extended selection table (<code>length(attr(X, "wave.objects"))</code>).Specifies the new names to be used for wave objects and sound file column. Note that this will rename wave objects and associated attributes and data in 'X'.</p>
</td></tr>
<tr><td><code id="rename_est_waves_+3A_new.selec">new.selec</code></td>
<td>
<p>Numeric or character vector of length equals to the number of rows in 'X' to specify the 'selec' column labels. Default is <code>NULL</code>. If not provided the 'selec' column is kept unchanged. Note that the combination of 'sound.files' and 'selec' columns must produce unique IDs for each selection (row).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allow users to change the names of 'sound.files' and 'selec' columns in extended selection tables. These names can become very long after manipulations used to produce extended tables.
</p>


<h3>Value</h3>

<p>An extended selection table with rename sound files names in data frame and attributes. The function adds columns with the previous sound file names (and 'selec' if provided).
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p>Other extended selection table manipulation: 
<code><a href="#topic+by_element_est">by_element_est</a>()</code>,
<code><a href="#topic+resample_est">resample_est</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("lbh.est")

# order by sound file name
lbh.est &lt;- lbh.est[order(lbh.est$sound.files),]

# create new sound file name
nsf &lt;- sapply(strsplit(lbh.est$sound.files, ".wav",fixed = TRUE), "[",1)

slc &lt;- vector(length = nrow(lbh.est))
slc[1] &lt;- 1

for(i in 2:length(slc))
if (nsf[i - 1] == nsf[i]) slc[i] &lt;- slc[i - 1] + 1 else
slc[i] &lt;- 1

nsf &lt;- paste(nsf, slc, sep = "_")

# rename sound files
Y &lt;- rename_est_waves(X = lbh.est, new.sound.files = nsf)
}

</code></pre>

<hr>
<h2 id='rename_waves_est'>alternative name for <code><a href="#topic+rename_est_waves">rename_est_waves</a></code></h2><span id='topic+rename_waves_est'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+rename_est_waves">rename_est_waves</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_waves_est(X, new.sound.files, new.selec = NULL)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+rename_est_waves">rename_est_waves</a></code> for documentation. <code><a href="#topic+rename_waves_est">rename_waves_est</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='resample_est'>Resample wave objects in a extended selection table</h2><span id='topic+resample_est'></span>

<h3>Description</h3>

<p><code>resample_est</code> changes sampling rate and bit depth of wave objects in a extended selection table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resample_est(X, samp.rate = 44.1, bit.depth = 16,
 avoid.clip = TRUE, pb = FALSE, parallel = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resample_est_+3A_x">X</code></td>
<td>
<p>object of class 'extended_selection_table' (see <code><a href="#topic+selection_table">selection_table</a></code>).</p>
</td></tr>
<tr><td><code id="resample_est_+3A_samp.rate">samp.rate</code></td>
<td>
<p>Numeric vector of length 1 with the sampling rate (in kHz) for output files. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="resample_est_+3A_bit.depth">bit.depth</code></td>
<td>
<p>Numeric vector of length 1 with the dynamic interval (i.e. bit depth) for output files.</p>
</td></tr>
<tr><td><code id="resample_est_+3A_avoid.clip">avoid.clip</code></td>
<td>
<p>Logical to control whether the volume is automatically
adjusted to avoid clipping high amplitude samples when resampling. Ignored if
'<code>sox = FALSE</code>. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="resample_est_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="resample_est_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aims to simplify the process of homogenizing sound
files (sampling rate and bit depth). This is a necessary step before running
any further (bio)acoustic analysis. <a href="https://sourceforge.net/projects/sox/">SOX</a> must be installed.
</p>


<h3>Value</h3>

<p>An extended selection table with the modified wave objects.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
#last modification on oct-15-2018 (MAS)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+mp32wav">mp32wav</a></code>, <code><a href="#topic+fix_wavs">fix_wavs</a></code>
</p>
<p>Other extended selection table manipulation: 
<code><a href="#topic+by_element_est">by_element_est</a>()</code>,
<code><a href="#topic+rename_est_waves">rename_est_waves</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# create extended selection table
X &lt;- selection_table(
  X = lbh_selec_table, extended = TRUE, confirm.extended = FALSE, pb = FALSE,
  path = tempdir()
)

# resample
Y &lt;- resample_est(X)

## End(Not run)
</code></pre>

<hr>
<h2 id='resample_est_waves'>alternative name for <code><a href="#topic+resample_est">resample_est</a></code></h2><span id='topic+resample_est_waves'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+resample_est">resample_est</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resample_est_waves(
  X,
  samp.rate = 44.1,
  bit.depth = 16,
  avoid.clip = TRUE,
  pb = FALSE,
  parallel = 1
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+resample_est">resample_est</a></code> for documentation. <code><a href="#topic+resample_est_waves">resample_est_waves</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='rm_channels'>alternative name for <code><a href="#topic+remove_channels">remove_channels</a></code></h2><span id='topic+rm_channels'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+remove_channels">remove_channels</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm_channels(files = NULL, channels, path = NULL, parallel = 1, pb = TRUE)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+remove_channels">remove_channels</a></code> for documentation. <code><a href="#topic+rm_channels">rm_channels</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='rm_sil'>alternative name for <code><a href="#topic+remove_silence">remove_silence</a></code></h2><span id='topic+rm_sil'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+remove_silence">remove_silence</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm_sil(
  path = NULL,
  min.sil.dur = 2,
  img = TRUE,
  it = "jpeg",
  flim = NULL,
  files = NULL,
  flist = NULL,
  parallel = 1,
  pb = TRUE,
  downsample = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+remove_silence">remove_silence</a></code> for documentation. <code><a href="#topic+rm_sil">rm_sil</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='selec.table'>Alternative name for <code>lbh_selec_table</code></h2><span id='topic+selec.table'></span>

<h3>Description</h3>

<p><code>selec.table</code> alternative name for <code>lbh_selec_table</code>. <code>selec.table</code> will be deprecated in
future versions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selec.table
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 11 rows and 9 columns.
</p>


<h3>Details</h3>

<p>Selection table of long-billed hermit songs
</p>


<h3>Source</h3>

<p>Marcelo Araya Salas, warbleR
</p>

<hr>
<h2 id='selection_table'>Create 'selection_table' and 'extended_selection_table' objects</h2><span id='topic+selection_table'></span>

<h3>Description</h3>

<p><code>selection_table</code> converts data frames into an object of classes 'selection_table' or 'extended_selection_table'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selection_table(X, max.dur = 10, path = NULL, whole.recs = FALSE,
extended = FALSE, confirm.extended = FALSE, mar = 0.1, by.song = NULL,
pb = TRUE, parallel = 1, verbose = TRUE, skip.error = FALSE,
file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$", files = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selection_table_+3A_x">X</code></td>
<td>
<p>data frame with the following columns: 1) &quot;sound.files&quot;: name of the .wav
files, 2) &quot;selec&quot;: unique selection identifier (within a sound file), 3) &quot;start&quot;: start time and 4) &quot;end&quot;:
end time of selections. Columns for 'top.freq', 'bottom.freq' and 'channel' are optional. Note that, when 'channel' is
not provided the first channel (i.e. left channel) would be used by default.
Frequency parameters (including top and bottom frequency) should be provided in kHz. Alternatively, a 'selection_table' class object can be input.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_max.dur">max.dur</code></td>
<td>
<p>the maximum duration of expected for a selection  (ie. end - start). If surpassed then an error message
will be generated. Useful for detecting errors in selection tables.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_whole.recs">whole.recs</code></td>
<td>
<p>Logical. If <code>TRUE</code> the function will create a selection
table for all sound files in the working directory (or &quot;path&quot;) with 'start = 0'
and 'end = duration_wavs()'. Default is if <code>FALSE</code>. Note that this will not create
a extended selection table. If provided 'X' is ignored.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_extended">extended</code></td>
<td>
<p>Logical. If <code>TRUE</code>, the function will create an object of class 'extended_selection_table'
which included the wave objects of the selections as an additional attribute ('wave.objects') to the data set. This is
and self-contained format that does not require the original sound files for running most acoustic analysis in
<code><a href="#topic+warbleR">warbleR</a></code>. This can largely facilitate the storing and sharing of (bio)acoustic data. Default
is if <code>FALSE</code>. An extended selection table won't be created if there is any issue with the selection. See
'details'.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_confirm.extended">confirm.extended</code></td>
<td>
<p>Logical. If <code>TRUE</code> then the size of the 'extended_selection_table'
will be estimated and the user will be asked for confirmation (in the console)
before proceeding. Ignored if 'extended' is <code>FALSE</code>. This is used to prevent
generating objects too big to be dealt with by R. See 'details' for more information about extended selection table size. THIS ARGUMENT WILL BE DEPRECATED IN FUTURE VERSIONS.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1 specifying the margins (in seconds)
adjacent to the start and end points of the selections when creating extended
selection tables. Default is 0.1. Ignored if 'extended' is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_by.song">by.song</code></td>
<td>
<p>Character string with the column name containing song labels. If provided a wave object containing for
all selection belonging to a single song would be saved in the extended selection table (hence only applicable for
extended selection tables). Note that the function assumes that song labels are not repeated within a sound file.
If <code>NULL</code> (default), wave objects are created for each selection (e.g. by selection).
Ignored if <code>extended = FALSE</code>.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="selection_table_+3A_verbose">verbose</code></td>
<td>
<p>Logical argument to control if summary messages are printed to the console. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_skip.error">skip.error</code></td>
<td>
<p>Logical to control if errors are omitted. If so, files that could not be read will be excluded and their name printed in the console. Default is <code>FALSE</code>, which will return an error if some files are problematic.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_file.format">file.format</code></td>
<td>
<p>Character string with the format of sound files. By default all sound file formats supported by warbleR are included (&quot;\.wav$|\.wac$|\.mp3$|\.flac$&quot;). Note that several formats can be included using regular expression syntax as in <code><a href="base.html#topic+grep">grep</a></code>. For instance <code>"\.wav$|\.mp3$"</code> will only include .wav and .mp3 files. Ignored if <code>whole.recs = FALSE</code>.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_files">files</code></td>
<td>
<p>character vector indicating the set of files that will be consolidated. File names should not include the full file path. Optional.</p>
</td></tr>
<tr><td><code id="selection_table_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code><a href="#topic+check_sels">check_sels</a></code> for customizing
checking routine.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates and object of class 'selection_table' or 'extended_selection_table' (if <code>extended = TRUE</code>, see below). First, the function checks:
</p>

<ul>
<li><p> 1) if the selections listed in the data frame correspond to .wav files
in the working directory
</p>
</li>
<li><p> 2) if the sound files can be read and if so,
</p>
</li>
<li><p> 3) if the start and end time of the selections are found within the duration of the sound files
</p>
</li></ul>

<p>If no errors are found the a selection table or extended selection table will be generated.
Note that the sound files should be in the working directory (or the directory provided in 'path').
This is useful for avoiding errors in downstream functions (e.g. <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>, <code><a href="#topic+cross_correlation">cross_correlation</a></code>,  <code><a href="#topic+catalog">catalog</a></code>, <code><a href="#topic+freq_DTW">freq_DTW</a></code>). Note also that corrupt files can be
fixed using <code><a href="#topic+fix_wavs">fix_wavs</a></code> ('sox' must be installed to be able to run this function).
The 'selection_table' class can be input in subsequent functions.
</p>
<p>When <code>extended = TRUE</code> the function will generate an object of class 'extended_selection_table' which
will also contain the wave objects for each of the selections in the data frame.
This transforms selection tables into self-contained objects as they no longer need the original
sound files to run acoustic analysis. This can largely facilitate the storing and sharing of (bio)acoustic data.
Extended selection table size will be a function of the number of selections <code>nrow(X)</code>, sampling rate, selection
duration and margin duration. As a guide, a selection table
with 1000 selections similar to the ones in 'lbh_selec_table' (mean duration ~0.15
seconds) at 22.5 kHz sampling rate and the default margin (<code>mar = 0.1</code>)
will generate a extended selection table of ~31 MB (~310 MB for a 10000 rows
selection table). You can check the size of the output extended selection table
with the <code><a href="utils.html#topic+object.size">object.size</a></code> function. Note that extended selection table created 'by.song' could be
considerable larger.
</p>


<h3>Value</h3>

<p>An object of class selection_table which includes the original data frame plus the following additional attributes:
</p>

<ul>
<li><p> 1) A data frame with the output of <code><a href="#topic+check_sels">check_sels</a></code> run on the input data frame. If a extended selection table is created it will also include the original values in the input data frame for each selections. This are used by downstream warbleR functions to improve efficiency and avoid
errors due to missing or mislabeled data, or selection out of the ranges of the original sound files.
</p>
</li>
<li><p> 2) A list indicating if the selection table has been created by song (see 'by.song argument).
</p>
</li>
<li><p> 3) If a extended selection table is created a list containing the wave objects for each selection (or song if 'by.song').
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+check_wavs">check_wavs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c(
    "Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4",
    "lbh_selec_table"
  ))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  # make selection table
  st &lt;- selection_table(X = lbh_selec_table, path = tempdir())

  is_selection_table(st)

  #' # make extended selection table
  st &lt;- selection_table(
    X = lbh_selec_table, extended = TRUE,
    path = tempdir()
  )

  is_extended_selection_table(st)

  ### make extended selection by song
  # create a song variable
  lbh_selec_table$song &lt;- as.numeric(as.factor(lbh_selec_table$sound.files))

  st &lt;- selection_table(
    X = lbh_selec_table, extended = TRUE,
    by.song = "song", path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='seltailor'>alternative name for <code><a href="#topic+tailor_sels">tailor_sels</a></code></h2><span id='topic+seltailor'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+tailor_sels">tailor_sels</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seltailor(
  X = NULL,
  wl = 512,
  flim = c(0, 22),
  wn = "hanning",
  mar = 0.5,
  osci = TRUE,
  pal = reverse.gray.colors.2,
  ovlp = 70,
  auto.next = FALSE,
  pause = 1,
  comments = TRUE,
  path = NULL,
  frange = TRUE,
  fast.spec = FALSE,
  ext.window = TRUE,
  width = 15,
  height = 5,
  index = NULL,
  collevels = NULL,
  title = c("sound.files", "selec"),
  ts.df = NULL,
  col = "#E37222",
  alpha = 0.7,
  auto.contour = FALSE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+tailor_sels">tailor_sels</a></code> for documentation. <code><a href="#topic+seltailor">seltailor</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='sig2noise'>Measure signal-to-noise ratio</h2><span id='topic+sig2noise'></span>

<h3>Description</h3>

<p><code>sig2noise</code> measures signal-to-noise ratio across multiple files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sig2noise(X, mar, parallel = 1, path = NULL, pb = TRUE, type = 1, eq.dur = FALSE,
in.dB = TRUE, before = FALSE, lim.dB = TRUE, bp = NULL, wl = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sig2noise_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or any data frame with columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end).</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_mar">mar</code></td>
<td>
<p>numeric vector of length 1. Specifies the margins adjacent to
the start and end points of selection over which to measure noise.</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing). It can also be
set globally using the 'parallel' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used. It can also be
set globally using the 'wav.path' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control if progress bar is shown. Default is <code>TRUE</code>. It can also be
set globally using the 'pb' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_type">type</code></td>
<td>
<p>Numeric. Determine the formula to be used to calculate the signal-to-noise ratio (S = signal
, N = background noise):
</p>

<ul>
<li> <p><code>1</code>: ratio of S mean amplitude envelope to
N mean amplitude envelope (<code>mean(env(S))/mean(env(N))</code>)
</p>
</li>
<li> <p><code>2</code>: ratio of S amplitude envelope RMS (root mean square) to N amplitude envelope RMS
(<code>rms(env(S))/rms(env(N))</code>)
</p>
</li>
<li> <p><code>3</code>: ratio of the difference between S amplitude envelope RMS and N amplitude envelope RMS to N amplitude envelope RMS (<code>(rms(env(S)) - rms(env(N)))/rms(env(N))</code>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="sig2noise_+3A_eq.dur">eq.dur</code></td>
<td>
<p>Logical. Controls whether the noise segment that is measured has the same duration
than the signal (if <code>TRUE</code>, default <code>FALSE</code>). If <code>TRUE</code> then 'mar' argument is ignored.</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_in.db">in.dB</code></td>
<td>
<p>Logical. Controls whether the signal-to-noise ratio is returned in decibels (20*log10(SNR)).
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_before">before</code></td>
<td>
<p>Logical. If <code>TRUE</code> noise is only measured right before the signal (instead of before and after). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_lim.db">lim.dB</code></td>
<td>
<p>Logical. If <code>TRUE</code> the lowest signal-to-noise would be limited to -40 dB (if <code>in.dB = TRUE</code>). This would remove NA's that can be produced when noise segments have a higher amplitude than the signal
itself. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_bp">bp</code></td>
<td>
<p>Numeric vector of length 2 giving the lower and upper limits of a frequency bandpass filter (in kHz). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="sig2noise_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram for applying bandpass. Default
is 10. Ignored if <code>bp = NULL</code>. It can also be
set globally using the 'wl' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).
Note that lower values will increase time resolution, which is more important for signal-to-noise ratio calculations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Signal-to-noise ratio (SNR) is a measure of the level of a desired signal compared to
background noise. The function divides the mean amplitude of the signal by
the mean amplitude of the background noise adjacent to the signal.
A general margin to apply before and after the acoustic signal must
be specified. Setting margins for individual signals that have been
previously clipped from larger files may take some optimization, as
for calls within a larger file that are irregularly separated. When
margins overlap with another acoustic signal nearby, the signal-to-noise
ratio (SNR) will be inaccurate. Any SNR less than or equal to one suggests
background noise is equal to or overpowering the acoustic signal.
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> can be used to troubleshoot different noise margins.
</p>


<h3>Value</h3>

<p>Data frame similar to <code><a href="#topic+auto_detec">auto_detec</a></code> output, but also includes a new variable
with the signal-to-noise values.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
<a href="https://en.wikipedia.org/wiki/Signal-to-noise_ratio">Wikipedia: Signal-to-noise ratio</a>

</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c("Phae.long1", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound files

  # specifying the correct margin is important
  # use snr_spectrograms to troubleshoot margins for sound files
  sig2noise(lbh_selec_table[grep("Phae.long1", lbh_selec_table$sound.files), ],
    mar = 0.2,
    path = tempdir()
  )

  # this smaller margin doesn't overlap neighboring signals
  sig2noise(lbh_selec_table[grep("Phae.long1", lbh_selec_table$sound.files), ],
    mar = 0.1,
    path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='signal_2_noise'>alternative name for <code><a href="#topic+sig2noise">sig2noise</a></code></h2><span id='topic+signal_2_noise'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+sig2noise">sig2noise</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>signal_2_noise(
  X,
  mar,
  parallel = 1,
  path = NULL,
  pb = TRUE,
  type = 1,
  eq.dur = FALSE,
  in.dB = TRUE,
  before = FALSE,
  lim.dB = TRUE,
  bp = NULL,
  wl = 10
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+sig2noise">sig2noise</a></code> for documentation. <code><a href="#topic+sig2noise">sig2noise</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='sim_coor_sing'>Simulated coordinated singing events.</h2><span id='topic+sim_coor_sing'></span>

<h3>Description</h3>

<p><code>sim_coor_sing</code> contains selections of simulated interactive singing events.The simulated events use the mean
and standard deviation of real lekking <em>Phaethornis longirostris</em> (Long-billed
Hermit hummingbird) songs and intervals between songs (e.i gaps). Three events are simulated: overlapping signals (ovlp),
alternating signals (altern) and non-synchronized signals (uncoor).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sim_coor_sing)
</code></pre>


<h3>Format</h3>


<dl>
<dt>sim_coor_sing</dt><dd><p>Simulated coordinated singing events that overlap and do not overlap most of the time,
for use with <code>test_coordination</code></p>
</dd>
</dl>


<hr>
<h2 id='sim_songs'>alternative name for <code><a href="#topic+simulate_songs">simulate_songs</a></code></h2><span id='topic+sim_songs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+simulate_songs">simulate_songs</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_songs(
  n = 1,
  durs = 0.2,
  harms = 3,
  harm.amps = c(1, 0.5, 0.2),
  am.amps = 1,
  gaps = 0.1,
  freqs = 5,
  samp.rate = 44.1,
  sig2 = 0.5,
  steps = 10,
  bgn = 0.5,
  seed = NULL,
  diff.fun = "GBM",
  fin = 0.1,
  fout = 0.2,
  shape = "linear",
  selec.table = FALSE,
  file.name = NULL,
  path = NULL,
  hrm.freqs = c(1/2, 1/3, 2/3, 1/4, 3/4, 1/5, 1/6, 1/7, 1/8, 1/9, 1/10),
  freq.range = 4
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+simulate_songs">simulate_songs</a></code> for documentation. <code><a href="#topic+sim_songs">sim_songs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='simulate_songs'>Simulate animal vocalizations</h2><span id='topic+simulate_songs'></span>

<h3>Description</h3>

<p><code>simulate_songs</code> simulate animal vocalizations in a wave object under brownian motion frequency drift.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_songs(n = 1, durs = 0.2, harms = 3, harm.amps = c(1, 0.5, 0.2), am.amps = 1,
gaps = 0.1, freqs = 5, samp.rate = 44.1, sig2 = 0.5,
steps = 10, bgn = 0.5, seed = NULL, diff.fun = "GBM",
fin = 0.1, fout = 0.2, shape = "linear", selec.table = FALSE,
file.name = NULL, path = NULL,
hrm.freqs = c(1/2, 1/3, 2/3, 1/4, 3/4, 1/5, 1/6, 1/7, 1/8, 1/9, 1/10),
freq.range = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_songs_+3A_n">n</code></td>
<td>
<p>Number of song subunits (e.g. elements). Default is 1.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_durs">durs</code></td>
<td>
<p>Numeric vector with the duration of subunits in seconds. It should either be a single value (which would
be used for all subunits) or a vector of length <code>n</code>.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_harms">harms</code></td>
<td>
<p>NUmeric vector of length 1 specifying the number of harmonics to simulate. 1 indicates that only the fundamental
frequency harmonic will be simulated.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_harm.amps">harm.amps</code></td>
<td>
<p>Numeric vector with the relative amplitude of each of the harmonics (including the fundamental frequency).</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_am.amps">am.amps</code></td>
<td>
<p>Numeric vector with the relative amplitude for each step (see 'step' argument) to simulate amplitude modulation (only applied to the fundamental frequency). Should have the same length as the number of steps. Default is 1 (no amplitude modulation).</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_gaps">gaps</code></td>
<td>
<p>Numeric vector with the duration of gaps (silence between subunits) in seconds. It should either be a single value
(which would be used for all subunits) or a vector  of length <code>n + 1</code>.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_freqs">freqs</code></td>
<td>
<p>Numeric vector with the initial frequency of the subunits (and ending frequency if <code>diff.fun == "BB"</code>) in kHz.
It should either be a single value (which would  be used for all subunits) or a vector of length <code>n</code>.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_samp.rate">samp.rate</code></td>
<td>
<p>Numeric vector of length 1. Sets the sampling frequency of the wave object (in kHz). Default is 44.1.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_sig2">sig2</code></td>
<td>
<p>Numeric vector defining the sigma value of the brownian motion model. It should either be a single value
(which would be used for all subunits) or a vector of length <code>n + 1</code>.  Higher values will produce faster
frequency modulations. Only applied if <code>diff.fun == "GBM"</code>. Default is 0.1. Check the <code><a href="Sim.DiffProc.html#topic+GBM">GBM</a></code> function from the Sim.DiffProc package
for more details.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_steps">steps</code></td>
<td>
<p>Numeric vector of length 1. Controls the mean number of segments in which each song subunit is split during
the brownian motion process. If not all subunits have the same duration, longer units will be split in more steps (although
the average duration subunit will have the predefined number of steps). Default is 10.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_bgn">bgn</code></td>
<td>
<p>Numeric vector of length 1 indicating the background noise level. 0 means no additional noise will 1 means
noise at the same amplitude than the song subunits. Default is 0.5.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_seed">seed</code></td>
<td>
<p>Numeric vector of length 1. This allows users to get the same results in different runs (using <code><a href="base.html#topic+Random">set.seed</a></code> internally). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_diff.fun">diff.fun</code></td>
<td>
<p>Character vector of length 1 controlling the function used to simulate the brownian motion process of
frequency drift across time. Only &quot;BB&quot;, &quot;GBM&quot; and &quot;pure.tone&quot; are accepted at this time. Check the <code><a href="Sim.DiffProc.html#topic+GBM">GBM</a></code> function from the Sim.DiffProc package
for more details.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_fin">fin</code></td>
<td>
<p>Numeric vector of length 1 setting the proportion of the sub-unit to fade-in amplitude (value between 0 and 1).
Default is 0.1. Note that 'fin' + 'fout' cannot be higher than 1.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_fout">fout</code></td>
<td>
<p>Numeric vector of length 1 setting the proportion of the sub-unit to fade-out amplitude (value between 0 and 1).
Default is 0.2. Note that 'fin' + 'fout' cannot be higher than 1.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_shape">shape</code></td>
<td>
<p>Character string of length 1 controlling the shape of in and out amplitude fading of the song sub-units
('fin' and 'fout'). &quot;linear&quot; (default), &quot;exp&quot; (exponential), and &quot;cos&quot; (cosine) are currently allowed.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_selec.table">selec.table</code></td>
<td>
<p>Logical. If <code>TRUE</code> the function returns a list with two elements: 1) a data frame containing the start/end time, and bottom/top frequency of the sub-units and  2) the wave object containing the simulated songs. If <code>FALSE</code> (default) no objects are returned. Regardless of the value of this argument a .wav file is always saved in the working directory.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_file.name">file.name</code></td>
<td>
<p>Character string for naming the &quot;.wav&quot; file. Ignored if
'selec.table' is <code>FALSE</code>. If not provided the date-time stamp will be used.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_path">path</code></td>
<td>
<p>Character string with the directory path where the sound file should be saved. Ignored if 'selec.table' is <code>FALSE</code>.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_hrm.freqs">hrm.freqs</code></td>
<td>
<p>Numeric vector with the frequencies of the harmonics relative to the fundamental frequency. The default values are c(1/2, 1/3, 2/3, 1/4, 3/4, 1/5, 1/6, 1/7, 1/8, 1/9, 1/10).</p>
</td></tr>
<tr><td><code id="simulate_songs_+3A_freq.range">freq.range</code></td>
<td>
<p>Numeric vector of length 1 with the frequency range around the simulated frequency in which signals will modulate. Default is 4 which means that sounds will range +/- 2 kHz around the target frequency. If <code>NULL</code> the frequency range is not constrained.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions uses a geometric (<code>diff.fun == "GBM"</code>) or Brownian bridge (<code>diff.fun == "BB"</code>) motion stochastic process to simulate modulation in animal vocalizations (i.e. frequency traces across time).
The function can also simulate pure tones (<code>diff.fun == "pure.tone"</code>, 'sig2' is ignored).
Several song subunits (e.g. elements) can be simulated as well as the corresponding harmonics. Modulated sounds are adjusted so the mean frequency of the frequency contour is equal to the target frequency supplied by the user.
</p>


<h3>Value</h3>

<p>A wave object containing the simulated songs. If 'selec.table' is <code>TRUE</code> the function saves the wave object as a '.wav' sound file in the working directory (or 'path') and returns a list including 1) a selection table with the start/end time, and bottom/top frequency of the sub-units and 2) the wave object.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+query_xc">query_xc</a></code> for for downloading bird vocalizations from an online repository.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# simulate a song with 3 elements and no harmonics
sm_sng &lt;- simulate_songs(n = 3, harms = 1)

# plot spectro
seewave::spectro(sm_sng)

# simulate a song with 5 elements and 2 extra harmonics
sm_sng2 &lt;- simulate_songs(n = 5, harms = 3)

# plot spectrogram
seewave::spectro(sm_sng2)

# six pure tones with frequency ranging form 4 to 6 and returning selection table
sm_sng &lt;- simulate_songs(
  n = 6, harms = 1, seed = 1, diff.fun = "pure.tone",
  freqs = seq(4, 6, length.out = 6), selec.table = TRUE,
  path = tempdir()
)

# plot spectro
seewave::spectro(sm_sng$wave, flim = c(2, 8))

# selection table
sm_sng$selec.table

## End(Not run)

</code></pre>

<hr>
<h2 id='snr_specs'>alternative name for <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code></h2><span id='topic+snr_specs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snr_specs(
  X,
  wl = 512,
  flim = NULL,
  wn = "hanning",
  ovlp = 70,
  inner.mar = c(5, 4, 4, 2),
  outer.mar = c(0, 0, 0, 0),
  picsize = 1,
  res = 100,
  cexlab = 1,
  title = TRUE,
  before = FALSE,
  eq.dur = FALSE,
  propwidth = FALSE,
  xl = 1,
  osci = FALSE,
  gr = FALSE,
  sc = FALSE,
  mar = 0.2,
  snrmar = 0.1,
  it = "jpeg",
  parallel = 1,
  path = NULL,
  pb = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> for documentation. <code><a href="#topic+snr_specs">snr_specs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='snr_spectrograms'>Spectrograms with background noise margins</h2><span id='topic+snr_spectrograms'></span>

<h3>Description</h3>

<p><code>snr_spectrograms</code> creates spectrograms to visualize margins over which background noise
will be measured by <code><a href="#topic+sig2noise">sig2noise</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snr_spectrograms(X, wl = 512, flim = NULL, wn = "hanning", ovlp = 70,
inner.mar = c(5, 4, 4, 2), outer.mar = c(0, 0, 0, 0), picsize = 1,
res = 100, cexlab = 1, title = TRUE, before = FALSE, eq.dur = FALSE,
  propwidth= FALSE, xl = 1, osci = FALSE, gr = FALSE, sc = FALSE, mar = 0.2,
   snrmar = 0.1, it = "jpeg", parallel = 1, path = NULL, pb = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snr_spectrograms_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or any data frame with columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end).</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit in kHz of
the spectrogram, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_inner.mar">inner.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is c(5,4,4,2).
Specifies number of lines in inner plot margins where axis labels fall,
with form c(bottom, left, top, right). See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_outer.mar">outer.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is c(0,0,0,0).
Specifies number of lines in outer plot margins beyond axis labels, with
form c(bottom, left, top, right). See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_picsize">picsize</code></td>
<td>
<p>Numeric argument of length 1, controls relative size of
spectrogram. Default is 1.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1 that controls image resolution.
Default is 100 (faster) although 300 - 400 is recommended for publication/
presentation quality.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_cexlab">cexlab</code></td>
<td>
<p>Numeric vector of length 1 specifying relative size of axis
labels. See <code><a href="seewave.html#topic+spectro">spectro</a></code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_title">title</code></td>
<td>
<p>Logical argument to add a title to individual spectrograms.
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_before">before</code></td>
<td>
<p>Logical. If <code>TRUE</code> noise is only measured right before the signal (instead of before and after). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_eq.dur">eq.dur</code></td>
<td>
<p>Logical. Controls whether the noise segment that is measured has the same duration
than the signal (if <code>TRUE</code>, default <code>FALSE</code>). If <code>TRUE</code> then 'snrmar' argument is ignored.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_propwidth">propwidth</code></td>
<td>
<p>Logical argument to scale the width of spectrogram
proportionally to duration of the selected call. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_xl">xl</code></td>
<td>
<p>Numeric vector of length 1, a constant by which to scale
spectrogram width if propwidth = <code>TRUE</code>. Default is 1.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_osci">osci</code></td>
<td>
<p>Logical argument to add an oscillogram underneath spectrogram, as
in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_gr">gr</code></td>
<td>
<p>Logical argument to add grid to spectrogram. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_sc">sc</code></td>
<td>
<p>Logical argument to add amplitude scale to spectrogram, default is
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the
start and end points of the selections to define spectrogram limits. Default is 0.2. If snrmar
is larger than mar, then mar is set to be equal to snrmar.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_snrmar">snrmar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the start and end
points of the selections where noise will be measured. Default is 0.1.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="snr_spectrograms_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to test different margins to facilitate
accurate SNR measurements when using <code><a href="#topic+sig2noise">sig2noise</a></code> down the line.
Setting margins for individual calls that have been previously clipped from
larger files may take some optimization, as for calls within a
larger file that are irregularly separated. Setting inner.mar to
c(4,4.5,2,1) and outer.mar to c(4,2,2,1) works well when picsize = 2 or 3.
Title font size, inner.mar and outer.mar (from <code>mar</code> and <code>oma</code> in <code>par</code>) don't work well
when osci or sc = <code>TRUE</code>, this may take some optimization by the user.
</p>


<h3>Value</h3>

<p>Spectrograms per selection marked with margins where background noise will be measured.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
<a href="https://en.wikipedia.org/wiki/Signal-to-noise_ratio">Wikipedia: Signal-to-noise ratio</a>

</p>


<h3>See Also</h3>

<p><code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for creating spectrograms to visualize
frequency measurements by <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>, <code><a href="#topic+spectrograms">spectrograms</a></code> for
creating spectrograms
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+color_spectro">color_spectro</a>()</code>,
<code><a href="#topic+freq_DTW">freq_DTW</a>()</code>,
<code><a href="#topic+multi_DTW">multi_DTW</a>()</code>,
<code><a href="#topic+phylo_spectro">phylo_spectro</a>()</code>,
<code><a href="#topic+spectrograms">spectrograms</a>()</code>,
<code><a href="#topic+track_freq_contour">track_freq_contour</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(list = c("Phae.long1", "Phae.long2", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound.files
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

# make Phae.long1 and Phae.long2 spectrograms
# snrmar needs to be smaller before moving on to sig2noise()

snr_spectrograms(lbh_selec_table,
  flim = c(0, 14), inner.mar = c(4, 4.5, 2, 1),
  outer.mar = c(4, 2, 2, 1), picsize = 2, res = 300, cexlab = 2, mar = 0.2,
  snrmar = 0.1, it = "jpeg", wl = 300, path = tempdir()
)

# make only Phae.long1 spectrograms
# snrmar now doesn't overlap neighboring signals

snr_spectrograms(lbh_selec_table[grepl(c("Phae.long1"), lbh_selec_table$sound.files), ],
  flim = c(3, 14), inner.mar = c(4, 4.5, 2, 1), outer.mar = c(4, 2, 2, 1),
  picsize = 2, res = 300, cexlab = 2, mar = 0.2, snrmar = 0.01, wl = 300,
  path = tempdir()
)

# check this folder!
tempdir()

## End(Not run)
</code></pre>

<hr>
<h2 id='snrspecs'>alternative name for <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code></h2><span id='topic+snrspecs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snrspecs(
  X,
  wl = 512,
  flim = NULL,
  wn = "hanning",
  ovlp = 70,
  inner.mar = c(5, 4, 4, 2),
  outer.mar = c(0, 0, 0, 0),
  picsize = 1,
  res = 100,
  cexlab = 1,
  title = TRUE,
  before = FALSE,
  eq.dur = FALSE,
  propwidth = FALSE,
  xl = 1,
  osci = FALSE,
  gr = FALSE,
  sc = FALSE,
  mar = 0.2,
  snrmar = 0.1,
  it = "jpeg",
  parallel = 1,
  path = NULL,
  pb = TRUE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> for documentation. <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='song_analysis'>Calculates acoustic parameters at the song level</h2><span id='topic+song_analysis'></span>

<h3>Description</h3>

<p><code>song_analysis</code> calculates descriptive statistics of songs or other higher levels of organization in the signals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>song_analysis(X = NULL, song_colm = "song",mean_colm = NULL, min_colm = NULL,
max_colm = NULL, elm_colm = NULL, elm_fun = NULL, sd = FALSE, parallel = 1, pb = TRUE,
na.rm = FALSE, weight = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="song_analysis_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' (created 'by.song') or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;selec&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_song_colm">song_colm</code></td>
<td>
<p>Character string with the column name containing song labels.
It can be used to label any hierarchical level at which parameters need to be calculated (e.g. syllables, phrases).
Note that
the function assumes that song labels are not repeated within a sound file.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_mean_colm">mean_colm</code></td>
<td>
<p>Numeric vector with the index of the columns that will be averaged. If <code>NULL</code> the mean of all numeric columns in 'X' is returned.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_min_colm">min_colm</code></td>
<td>
<p>Character vector with the name(s) of the columns for which the minimum
value is needed. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_max_colm">max_colm</code></td>
<td>
<p>Character vector with the name(s) of the columns for which the maximum
value is needed. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_elm_colm">elm_colm</code></td>
<td>
<p>Character vector with the name(s) of the columns identifying the element labels (i.e. element types). If supplied 'unq.elms' and 'mean.elm.count' are returned. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_elm_fun">elm_fun</code></td>
<td>
<p>Function to be applied to the sequence of elements composing a song. Default is <code>NULL</code>. Ignored if 'elm_colm' is not supplied. The name of the column containing the function's output is &quot;elm_fun'.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_sd">sd</code></td>
<td>
<p>Logical value indicating whether standard deviation is also returned for
variables in which averages are reported. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_na.rm">na.rm</code></td>
<td>
<p>Logical value indicating whether 'NA' values should be ignored for calculations.</p>
</td></tr>
<tr><td><code id="song_analysis_+3A_weight">weight</code></td>
<td>
<p>Character vector defining 1 or more numeric vectors to weight average
measurements (i.e. song parameters). Names of numeric columns in 'X' can also be used. See <code><a href="stats.html#topic+weighted.mean">weighted.mean</a></code>.
for more details.  Default is <code>NULL</code> (unweighted average).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates average or extreme values of acoustic parameters of
elements in a song or other level of organization in the signals.
</p>


<h3>Value</h3>

<p>A data frame similar to the input 'X' data frame, but in this case each row corresponds to a single song. The data frame contains the mean or extreme
values for numeric columns for each song. Columns that will be averaged can be defined with
'mean_colm' (otherwise all numeric columns are used). Columns can be
weighted by other columns in the data set (e.g. duration, frequency range). In addition, the function returns the following song level parameters:
</p>

<ul>
<li> <p><code>elm.duration</code>: mean length of elements (in s)
</p>
</li>
<li> <p><code>song.duration</code>: length of song (in s)
</p>
</li>
<li> <p><code>num.elms</code>: number of elements (or song units)
</p>
</li>
<li> <p><code>start</code>: start time of song (in s)
</p>
</li>
<li> <p><code>end</code>: end time of song (in s)
</p>
</li>
<li> <p><code>bottom.freq</code>: lowest 'bottom.freq' from all song elements (in kHz)
</p>
</li>
<li> <p><code>top.freq</code>: highest 'top.freq' from all song elements (in kHz)
</p>
</li>
<li> <p><code>freq.range</code>: difference between song's 'top.freq' and 'bottom.freq' (in kHz)
</p>
</li>
<li> <p><code>song.rate</code>: number of elements per second (NA if only 1 element). Calculated as the number of elements in the 'song' divided by the duration of the song. In this case song duration is calculated as the time between the start of the first element and the start of the last element, which provides a rate that is less affected by the duration of individual elements. Note that this calculation is different than that from 'song.duration' above.
</p>
</li>
<li> <p><code>gap.duration</code>: average length of gaps (i.e. silences) in between elements
(in s, NA if only 1 element)
</p>
</li>
<li> <p><code>elm.types</code>: number of element types (i.e. number of unique types, only if 'elm_colm' is supplied)
</p>
</li>
<li> <p><code>mean.elm.count</code>: mean number of times element types are found (only if 'elm_colm' is supplied)
</p>
</li></ul>

<p>This function assumes that song labels are not repeated within a sound file.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+spectro_analysis">spectro_analysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# get warbleR sound file examples
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

# add a 'song' column
lbh_selec_table$song &lt;- c("song1", "song1", "song1", "song2",
  "song2", "song3", "song3", "song3", "song4", "song4", "song4")

# measure acoustic parameters
sp &lt;- spectro_analysis(lbh_selec_table[1:8, ], bp = c(1, 11), 300, fast = TRUE, path = tempdir())

# add song data
sp &lt;- merge(sp, lbh_selec_table[1:8, ], by = c("sound.files", "selec"))

# caculate song-level parameters for all numeric parameters
song_analysis(X = sp, song_colm = "song", parallel = 1, pb = TRUE)

# caculate song-level parameters selecting parameters with mean_colm
song_analysis(X = sp, song_colm = "song",mean_colm = c("dfrange", "duration"),
 parallel = 1, pb = TRUE)

# caculate song-level parameters for selecting parameters with mean_colm, max_colm
# and min_colm and weighted by duration
song_analysis(X = sp, weight = "duration", song_colm = "song",
mean_colm =  c("dfrange", "duration"), min_colm =  "mindom", max_colm = "maxdom",
  parallel = 1, pb = TRUE)

# with two weights
song_analysis(X = sp, weight = c("duration", "dfrange"), song_colm = "song",
mean_colm = c("kurt", "sp.ent"), parallel = 1, pb = TRUE)

# with two weights no progress bar
song_analysis(X = sp, weight = c("duration", "dfrange"), song_colm = "song",
mean_colm = c("kurt", "sp.ent"), parallel = 1, pb = FALSE)
}

</code></pre>

<hr>
<h2 id='song_param'>alternative name for <code><a href="#topic+song_analysis">song_analysis</a></code></h2><span id='topic+song_param'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+song_analysis">song_analysis</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>song_param(
  X = NULL,
  song_colm = "song",
  mean_colm = NULL,
  min_colm = NULL,
  max_colm = NULL,
  elm_colm = NULL,
  elm_fun = NULL,
  sd = FALSE,
  parallel = 1,
  pb = TRUE,
  na.rm = FALSE,
  weight = NULL
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+song_analysis">song_analysis</a></code> for documentation. <code><a href="#topic+song_param">song_param</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='sort_colms'>Sort columns in a more intuitive order</h2><span id='topic+sort_colms'></span>

<h3>Description</h3>

<p><code>sort_colms</code> sorts selection table columns in a more intuitive order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_colms(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sort_colms_+3A_x">X</code></td>
<td>
<p>Data frame containing columns for sound file (sound.files), selection
(selec), start and end time of signals ('start' and 'end') and low and high
frequency ('bottom.freq' and 'top.freq', optional). See the example data 'lbh_selec_table'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns the data from the input data frame with the most relevant information
for acoustic analysis located in the first columns. The priority order for column names is: &quot;sound.files&quot;, &quot;channel&quot;, &quot;selec&quot;, &quot;start&quot;, &quot;end&quot;, &quot;top.freq&quot;, and &quot;bottom.freq&quot;.
</p>


<h3>Value</h3>

<p>The same data as in the input data frame but with the most relevant information
for acoustic analysis located in the first columns.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(warbleR)
data("selec.table")

# mess column order
selec.table &lt;- selec.table[, sample(seq_len(ncol(selec.table)))]

# check names
names(selec.table)

selec.table &lt;- sort_colms(X = selec.table)

# check names again
names(selec.table)

</code></pre>

<hr>
<h2 id='sound_pressure_level'>Measure relative sound pressure level</h2><span id='topic+sound_pressure_level'></span>

<h3>Description</h3>

<p><code>sound_pressure_level</code> measures relative (uncalibrated) sound pressure level in signals referenced in a selection table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sound_pressure_level(X, reference = 20, parallel = 1, path = NULL, pb = TRUE,
type = "single", wl = 100, bp = NULL, remove.bgn = FALSE, mar = NULL, envelope = "abs")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sound_pressure_level_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or any data frame with columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end).</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_reference">reference</code></td>
<td>
<p>Numeric vector of length 1 indicating the pressure (in Pa) to be used as reference. Alternatively, a character vector with the name of a numeric column containing reference values for each row can be supplied. Default is 20 (Pa). NOT YET IMPLEMENTED.</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing). It can also be
set globally using the 'parallel' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used. It can also be
set globally using the 'wav.path' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control if progress bar is shown. Default is <code>TRUE</code>. It can also be
set globally using the 'pb' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_type">type</code></td>
<td>
<p>Character string controlling how SPL is measured: #' </p>

<ul>
<li> <p><code>single</code>: single SPL value obtained on the entire signal. Default.
</p>
</li>
<li> <p><code>mean</code>: average of SPL values measured across the signal.
</p>
</li>
<li> <p><code>peak</code>: maximum of several SPL values measured across the signal.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the spectrogram window length. Default is 512.</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_bp">bp</code></td>
<td>
<p>Numeric vector of length 2 giving the lower and upper limits of a frequency bandpass filter (in kHz). Alternatively, when set to 'freq.range', the function will use the 'bottom.freq' and 'top.freq' for each signal as the bandpass range. Default is <code>NULL</code> (no bandpass filter).</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_remove.bgn">remove.bgn</code></td>
<td>
<p>Logical argument to control if SPL from background noise is excluded from the measured signal SPL. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_mar">mar</code></td>
<td>
<p>numeric vector of length 1. Specifies the margins adjacent to
the start point of selection over which to measure background noise.</p>
</td></tr>
<tr><td><code id="sound_pressure_level_+3A_envelope">envelope</code></td>
<td>
<p>Character string vector with the method to calculate amplitude envelopes (in which SPL is measured), as in <code><a href="seewave.html#topic+env">env</a></code>. Must be either 'abs' (absolute envelope, default) or 'hil' (Hilbert transformation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sound pressure level (SPL) is a logarithmic measure of the effective pressure of a sound relative to a reference, so it's a measure of sound intensity. SPL is measured as the root mean square of the amplitude vector, and as such is only a useful metric of the variation in loudness for signals within the same recording (or recorded with the same equipment and gain).
</p>


<h3>Value</h3>

<p>The object supplied in 'X' with a new variable
with the sound pressure level values ('SPL' or 'peak.amplitude' column, see argument 'peak.amplitude') in decibels.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>

<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
<a href="https://en.wikipedia.org/wiki/Sound_pressure">Wikipedia: Sound pressure level</a>

</p>


<h3>See Also</h3>

<p><code><a href="#topic+sig2noise">sig2noise</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c("Phae.long1", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound files

  spl &lt;- sound_pressure_level(
    X = lbh_selec_table[grep("Phae.long1", lbh_selec_table$sound.files), ],
    parallel = 1, pb = TRUE, path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='spec_param'>alternative name for <code><a href="#topic+tweak_spectro">tweak_spectro</a></code></h2><span id='topic+spec_param'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+tweak_spectro">tweak_spectro</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spec_param(
  X,
  length.out = 5,
  ovlp = 90,
  wl = c(100, 1000),
  wn = "hanning",
  collev.min = -40,
  pal = "reverse.gray.colors.2",
  path = NULL,
  rm.axes = TRUE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+tweak_spectro">tweak_spectro</a></code> for documentation. <code><a href="#topic+spec_param">spec_param</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='specan'>alternative name for <code><a href="#topic+spectro_analysis">spectro_analysis</a></code></h2><span id='topic+specan'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>specan(
  X,
  bp = "frange",
  wl = 512,
  wl.freq = NULL,
  threshold = 15,
  parallel = 1,
  fast = TRUE,
  path = NULL,
  pb = TRUE,
  ovlp = 50,
  wn = "hanning",
  fsmooth = 0.1,
  harmonicity = FALSE,
  nharmonics = 3,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+spectro_analysis">spectro_analysis</a></code> for documentation. <code><a href="#topic+specan">specan</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='specreator'>alternative name for <code><a href="#topic+spectrograms">spectrograms</a></code></h2><span id='topic+specreator'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+spectrograms">spectrograms</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>specreator(
  X,
  wl = 512,
  flim = "frange",
  wn = "hanning",
  pal = reverse.gray.colors.2,
  ovlp = 70,
  inner.mar = c(5, 4, 4, 2),
  outer.mar = c(0, 0, 0, 0),
  picsize = 1,
  res = 100,
  cexlab = 1,
  propwidth = FALSE,
  xl = 1,
  osci = FALSE,
  gr = FALSE,
  sc = FALSE,
  line = TRUE,
  col = "#07889B",
  fill = adjustcolor("#07889B", alpha.f = 0.15),
  lty = 3,
  mar = 0.05,
  it = "jpeg",
  parallel = 1,
  path = NULL,
  pb = TRUE,
  fast.spec = FALSE,
  by.song = NULL,
  sel.labels = "selec",
  title.labels = NULL,
  dest.path = NULL,
  box = TRUE,
  axis = TRUE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+spectrograms">spectrograms</a></code> for documentation. <code><a href="#topic+specreator">specreator</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='spectro_analysis'>Measure acoustic parameters in batches of sound files</h2><span id='topic+spectro_analysis'></span>

<h3>Description</h3>

<p><code>spectro_analysis</code> measures acoustic parameters on acoustic signals for which the start and end times 
are provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spectro_analysis(X, bp = "frange", wl = 512, wl.freq = NULL, threshold = 15,
 parallel = 1, fast = TRUE, path = NULL, pb = TRUE, ovlp = 50,
wn = "hanning", fsmooth = 0.1, harmonicity = FALSE, nharmonics = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spectro_analysis_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound 
files, 2) &quot;sel&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;: 
end time of selections. The output <code><a href="#topic+auto_detec">auto_detec</a></code> can
be used as the input data frame.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a 
frequency bandpass filter (in kHz) or &quot;frange&quot; (default) to indicate that values in bottom.freq
and top.freq columns will be used as bandpass limits.  Lower limit of
bandpass filter is not applied to fundamental frequencies.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the spectrogram window length. Default is 512. See 'wl.freq' for setting windows length independently in the frequency domain.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_wl.freq">wl.freq</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram
for measurements on the frequency spectrum. Default is 512. Higher values would provide 
more accurate measurements. Note that this allows to increase measurement precision independently in the time and frequency domain. If <code>NULL</code> (default) then the 'wl' value is used.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_threshold">threshold</code></td>
<td>
<p>amplitude threshold (%) for fundamental frequency and 
dominant frequency detection. Default is 15.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_fast">fast</code></td>
<td>
<p>Logical. If <code>TRUE</code> (default) then the peakf acoustic parameter (see below) is not computed, which 
substantially increases performance (~9 times faster). This argument will be removed in future version.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located. 
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two 
consecutive windows, used for fundamental frequency (using <code><a href="seewave.html#topic+fund">fund</a></code> or <code><a href="tuneR.html#topic+FF">FF</a></code>) and dominant frequency (using <code><a href="seewave.html#topic+dfreq">dfreq</a></code>). 
Default is 50.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is hanning'. 
See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_fsmooth">fsmooth</code></td>
<td>
<p>A numeric vector of length 1 to smooth the frequency spectrum with a mean
sliding window (in kHz) used for mean peak frequency detection. This help to average 
amplitude &quot;hills&quot; to minimize the effect of amplitude modulation. Default is 0.1.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_harmonicity">harmonicity</code></td>
<td>
<p>Logical. If <code>TRUE</code> harmonicity related parameters (fundamental frequency parameters [meanfun, minfun, maxfun], hn_freq, 
hn_width, harmonics and HNR) are measured. Note that measuring these parameters 
considerably increases computing time.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_nharmonics">nharmonics</code></td>
<td>
<p>Numeric vector of length 1 setting the number of harmonics to analyze.</p>
</td></tr>
<tr><td><code id="spectro_analysis_+3A_...">...</code></td>
<td>
<p>Additional parameters to be passed to <code><a href="soundgen.html#topic+analyze">analyze</a></code>, which measures parameters related to harmonicity.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function measures 29 acoustic parameters (if <code>fast = TRUE</code>) on 
each selection in the data frame. Most parameters are produced internally by 
<code><a href="seewave.html#topic+specprop">specprop</a></code>, <code><a href="seewave.html#topic+fpeaks">fpeaks</a></code>, <code><a href="seewave.html#topic+fund">fund</a></code>,
and <code><a href="seewave.html#topic+dfreq">dfreq</a></code> from the package seewave and <code><a href="soundgen.html#topic+analyze">analyze</a></code> 
from the package soundgen. NAs are produced for fundamental and dominant 
frequency measures when there are no amplitude values above the threshold. 
Additional parameters can be provided to the internal function <code><a href="soundgen.html#topic+analyze">analyze</a></code>, which measures parameters related to harmonicity.
</p>


<h3>Value</h3>

<p>Data frame with 'sound.files' and 'selec' as in the input data frame, plus the following acoustic parameters: 
</p>

<ul>
<li> <p><code>duration</code>: length of signal (in s)
</p>
</li>
<li> <p><code>meanfreq</code>: mean frequency (in kHz). Calculated as the weighted average of the frequency spectrum (i.e. weighted by the amplitude within the supplied band pass).  
</p>
</li>
<li> <p><code>sd</code>: standard deviation of frequency (in kHz). Calculated as the weighted standard deviation of the frequency spectrum (i.e. weighted by the amplitude within the supplied band pass).  
</p>
</li>
<li> <p><code>freq.median</code>: median frequency. The frequency at which the frequency spectrum is divided in two frequency
intervals of equal energy (in kHz) 
</p>
</li>
<li> <p><code>freq.Q25</code>: first quartile frequency. The frequency at which the frequency spectrum is divided in two 
frequency intervals of 25% and 75% energy respectively (in kHz) 
</p>
</li>
<li> <p><code>freq.Q75</code>: third quartile frequency. The frequency at which the frequency spectrum is divided in two
frequency intervals of 75% and 25% energy respectively (in kHz) 
</p>
</li>
<li> <p><code>freq.IQR</code>: interquartile frequency range. Frequency range between 'freq.Q25' and 'freq.Q75' 
(in kHz) 
</p>
</li>
<li> <p><code>time.median</code>: median time. The time at which the time envelope is divided in two time
intervals of equal energy (in s) 
</p>
</li>
<li> <p><code>time.Q25</code>: first quartile time. The time at which the time envelope is divided in two
time intervals of 25% and 75% energy respectively (in s). See <code><a href="seewave.html#topic+acoustat">acoustat</a></code>
</p>
</li>
<li> <p><code>time.Q75</code>: third quartile time. The time at which the time envelope is divided in two
time intervals of 75% and 25% energy respectively (in s). See <code><a href="seewave.html#topic+acoustat">acoustat</a></code>
</p>
</li>
<li> <p><code>time.IQR</code>: interquartile time range. Time range between 'time.Q25' and 'time.Q75' 
(in s). See <code><a href="seewave.html#topic+acoustat">acoustat</a></code>
</p>
</li>
<li> <p><code>skew</code>: skewness. Asymmetry of the frequency spectrum (see note in <code><a href="seewave.html#topic+specprop">specprop</a></code> description) 
</p>
</li>
<li> <p><code>kurt</code>: kurtosis. Peakedness of the frequency spectrum (see note in <code><a href="seewave.html#topic+specprop">specprop</a></code> description)
</p>
</li>
<li> <p><code>sp.ent</code>: spectral entropy. Energy distribution of the frequency spectrum. Pure tone ~ 0; 
noisy ~ 1. See <code><a href="seewave.html#topic+sh">sh</a></code>
</p>
</li>
<li> <p><code>time.ent</code>: time entropy. Energy distribution on the time envelope. ~0 means amplitude concentrated in a specific time point, 1 means amplitude equally distributed across time. See <code><a href="seewave.html#topic+th">th</a></code>
</p>
</li>
<li> <p><code>entropy</code>: spectrographic entropy. Product of time and spectral entropy <code>sp.ent * time.ent</code>. 
See <code><a href="seewave.html#topic+H">H</a></code>
</p>
</li>
<li> <p><code>sfm</code>: spectral flatness. Similar to sp.ent (Pure tone ~ 0; 
noisy ~ 1). See <code><a href="seewave.html#topic+sfm">sfm</a></code>
</p>
</li>
<li> <p><code>meandom</code>: average of dominant frequency measured across the spectrogram 
</p>
</li>
<li> <p><code>mindom</code>: minimum of dominant frequency measured across the spectrogram
</p>
</li>
<li> <p><code>maxdom</code>: maximum of dominant frequency measured across the spectrogram 
</p>
</li>
<li> <p><code>dfrange</code>: range of dominant frequency measured across the spectrogram 
</p>
</li>
<li> <p><code>modindx</code>: modulation index. Calculated as the cumulative absolute
difference between adjacent measurements of dominant frequencies divided
by the dominant frequency range (measured on the spectrogram). 1 means the signal is not modulated. 
</p>
</li>
<li> <p><code>startdom</code>:  dominant frequency measurement at the start of the signal (measured on the spectrogram).
</p>
</li>
<li> <p><code>enddom</code>: dominant frequency measurement at the end of the signal(measured on the spectrogram).
</p>
</li>
<li> <p><code>dfslope</code>: slope of the change in dominant frequency (measured on the spectrogram) through time ((enddom-startdom)/duration). Units are kHz/s.  
</p>
</li>
<li> <p><code>peakf</code>: peak frequency. Frequency with the highest energy. This 
parameter can take a considerable amount of time to measure. It's only 
generated if <code>fast = FALSE</code>. It provides a more accurate measure of peak
frequency than 'meanpeakf' but can be more easily affected by background noise. Measured on the frequency spectrum.
</p>
</li>
<li> <p><code>meanpeakf</code>: mean peak frequency. Frequency with highest energy from the 
mean frequency spectrum (see <code><a href="seewave.html#topic+meanspec">meanspec</a></code>). Typically more consistent than peakf in the presence of noise.
</p>
</li>
<li> <p><code>meanfun</code>: average of fundamental frequency measured across the acoustic signal. Only measured if <code>harmonicity = TRUE</code>.
</p>
</li>
<li> <p><code>minfun</code>: minimum fundamental  frequency measured across the acoustic signal. Only measured if <code>harmonicity = TRUE</code>.
</p>
</li>
<li> <p><code>maxfun</code>: maximum fundamental frequency measured across the acoustic signal. Only measured if <code>harmonicity = TRUE</code>.
</p>
</li>
<li> <p><code>hn_freq</code>: mean frequency of the 'n' upper harmonics (kHz) (see <code><a href="soundgen.html#topic+analyze">analyze</a></code>). 
Number of harmonics is defined with the argument 'nharmonics'. Only measured if <code>harmonicity = TRUE</code>.
</p>
</li>
<li> <p><code>hn_width</code>: mean bandwidth of the 'n' upper harmonics (kHz) (see <code><a href="soundgen.html#topic+analyze">analyze</a></code>). Number of harmonics is defined with the argument 'nharmonics'. Only measured if <code>harmonicity = TRUE</code>.  
</p>
</li>
<li> <p><code>harmonics</code>: the amount of energy in upper harmonics, namely the 
ratio of total spectral power above 1.25 x F0 to the total spectral power 
below 1.25 x F0 (dB) (see <code><a href="soundgen.html#topic+analyze">analyze</a></code>). Number of 
harmonics is defined with the argument 'nharmonics'. Only measured if <code>harmonicity = TRUE</code>.
</p>
</li>
<li> <p><code>HNR</code>: harmonics-to-noise ratio (dB). A measure of the harmonic content generated by <code><a href="soundgen.html#topic+getPitchAutocor">getPitchAutocor</a></code>. Only measured if <code>harmonicity = TRUE</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

# measure acoustic parameters
sp_param &lt;- spectro_analysis(X = lbh_selec_table[1:8,], pb = FALSE, path = tempdir())

# measuring peakf
sp_param &lt;- spectro_analysis(X = lbh_selec_table[1:8,], pb = FALSE, fast = FALSE, path = tempdir())


# measuring harmonic-related parameters using progress bar
sp_param &lt;- spectro_analysis(X = lbh_selec_table[1:8,], harmonicity = TRUE, 
path = tempdir(), ovlp = 70)

}
</code></pre>

<hr>
<h2 id='spectrograms'>Spectrograms of selected signals</h2><span id='topic+spectrograms'></span>

<h3>Description</h3>

<p><code>spectrograms</code> creates spectrograms of signals from selection tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spectrograms(X, wl = 512, flim = "frange", wn = "hanning", pal  = reverse.gray.colors.2,
ovlp = 70, inner.mar = c(5, 4, 4, 2), outer.mar = c(0, 0, 0, 0), picsize = 1, res = 100,
cexlab = 1, propwidth = FALSE, xl = 1, osci = FALSE, gr = FALSE,  sc = FALSE, line = TRUE,
col = "#07889B", fill = adjustcolor("#07889B", alpha.f = 0.15), lty = 3,
mar = 0.05, it = "jpeg", parallel = 1, path = NULL, pb = TRUE, fast.spec = FALSE,
by.song = NULL, sel.labels = "selec", title.labels = NULL, dest.path = NULL,
box = TRUE, axis = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spectrograms_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' or data frame containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signals (start and end).
'top.freq' and 'bottom.freq' columns are optional. If using an
'extended_selection_table' the sound files are not required (see <code><a href="#topic+selection_table">selection_table</a></code>).</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit (in kHz) of
the spectrogram, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. The function also
accepts 'frange' (default) which produces spectrograms with a frequency
limit around the range of each signal (adding a 1 kHz margin).</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_pal">pal</code></td>
<td>
<p>A color palette function to be used to assign colors in the
plot, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is reverse.gray.colors.2.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying the percent overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_inner.mar">inner.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is c(5,4,4,2).
Specifies number of lines in inner plot margins where axis labels fall,
with form c(bottom, left, top, right). See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_outer.mar">outer.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is c(0,0,0,0).
Specifies number of lines in outer plot margins beyond axis labels, with
form c(bottom, left, top, right). See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_picsize">picsize</code></td>
<td>
<p>Numeric argument of length 1. Controls relative size of
spectrogram. Default is 1. Ignored when propwidth is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution.
Default is 100 (faster) although 300 - 400 is recommended for publication/
presentation quality.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_cexlab">cexlab</code></td>
<td>
<p>Numeric vector of length 1 specifying the relative size of axis
labels. See <code><a href="seewave.html#topic+spectro">spectro</a></code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_propwidth">propwidth</code></td>
<td>
<p>Logical argument to scale the width of spectrogram
proportionally to duration of the selection. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_xl">xl</code></td>
<td>
<p>Numeric vector of length 1. A constant by which to scale
spectrogram width if propwidth = <code>TRUE</code>. Default is 1.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_osci">osci</code></td>
<td>
<p>Logical argument to add an oscillogram underneath spectrogram, as
in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_gr">gr</code></td>
<td>
<p>Logical argument to add grid to spectrogram. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_sc">sc</code></td>
<td>
<p>Logical argument to add amplitude scale to spectrogram, default is
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_line">line</code></td>
<td>
<p>Logical argument to add lines at start and end times of selection
(or box if bottom.freq and top.freq columns are provided). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_col">col</code></td>
<td>
<p>Color of 'line'. Default is &quot;#07889B&quot;.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_fill">fill</code></td>
<td>
<p>Fill color of box around selections. Default is  <code>adjustcolor("#07889B", alpha.f = 0.15)</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_lty">lty</code></td>
<td>
<p>Type of 'line' as in <code><a href="graphics.html#topic+par">par</a></code>. Default is 1.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the start and end points of selections,
dealineating spectrogram limits. Default is 0.05.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms,
which substantially increases performance (much faster), although some options become unavailable,
as collevels, and sc (amplitude scale). This option is indicated for signals with high background noise
levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code>
(which should be imported from the package monitoR) seem to work better with 'fast' spectrograms.
Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>, <code><a href="monitoR.html#topic+specCols">gray.3</a></code>
offer decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_by.song">by.song</code></td>
<td>
<p>Character string with the column name containing song labels. If
provide a single spectrogram containing all elements for each song will be produce. Note that
the function assumes that each song has a unique label within a sound file. If <code>NULL</code> (default), spectrograms are produced for single selections.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_sel.labels">sel.labels</code></td>
<td>
<p>Character string with the name of the column(s) for selection
labeling. Default is 'selec'. Set to <code>NULL</code> to remove labels.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_title.labels">title.labels</code></td>
<td>
<p>Character string with the name(s) of the column(s) to use as title. Default is <code>NULL</code> (no title). Only sound file and song included if 'by.song' is provided.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_dest.path">dest.path</code></td>
<td>
<p>Character string containing the directory path where the image files will be saved.
If <code>NULL</code> (default) then the folder containing the sound files will be used instead.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_box">box</code></td>
<td>
<p>Logical to control if the box around the spectrogram is plotted (see <code><a href="graphics.html#topic+box">box</a></code>). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_axis">axis</code></td>
<td>
<p>Logical to control if the Y and X axis are of the spectrogram are plotted (see <code><a href="graphics.html#topic+box">box</a></code>). Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="spectrograms_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal spectrogram
creating function for customizing graphical output. The function is a modified
version of <code><a href="seewave.html#topic+spectro">spectro</a></code>, so it takes the same arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides access to batch process of (a modified version of) the <code><a href="seewave.html#topic+spectro">spectro</a></code> function from the 'seewave' package. The function creates spectrograms for visualization of vocalizations.
Setting inner.mar to c(4,4.5,2,1) and outer.mar to c(4,2,2,1) works well when picsize = 2 or 3.
Title font size, inner.mar and outer.mar (from mar and oma) don't work well when osci or sc = TRUE,
this may take some optimization by the user. Setting 'fast' argument to TRUE significantly increases speed, although
some options become unavailable, as collevels, and sc (amplitude scale). This option is indicated for signals with
high background noise levels.
</p>


<h3>Value</h3>

<p>Image files containing spectrograms of the signals listed in the input data frame.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>) and Grace Smith Vidaurre
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for creating spectrograms to visualize
frequency measurements by <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>, <code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> for
creating spectrograms to optimize noise margins used in <code><a href="#topic+sig2noise">sig2noise</a></code>
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+color_spectro">color_spectro</a>()</code>,
<code><a href="#topic+freq_DTW">freq_DTW</a>()</code>,
<code><a href="#topic+multi_DTW">multi_DTW</a>()</code>,
<code><a href="#topic+phylo_spectro">phylo_spectro</a>()</code>,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a>()</code>,
<code><a href="#topic+track_freq_contour">track_freq_contour</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load and save data
  data(list = c("Phae.long1", "Phae.long2", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) # save sound files
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

  # make spectrograms
  spectrograms(
    X = lbh_selec_table, flim = c(0, 11), res = 300, mar = 0.05,
    wl = 300, path = tempdir()
  )

  # check this folder
  tempdir()
}

</code></pre>

<hr>
<h2 id='split_sound_files'>Splits sound files</h2><span id='topic+split_sound_files'></span>

<h3>Description</h3>

<p><code>split_sound_files</code> splits sound files in shorter segments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_sound_files(path = NULL, sgmt.dur = 10, sgmts = NULL, files = NULL,
 parallel = 1, pb = TRUE, only.sels = FALSE, X = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split_sound_files_+3A_path">path</code></td>
<td>
<p>Directory path where sound files are found.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_sgmt.dur">sgmt.dur</code></td>
<td>
<p>Numeric. Duration (in s) of segments in which sound files would be split. Sound files shorter than 'sgmt.dur' won't be split. Ignored if 'sgmts' is supplied.</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_sgmts">sgmts</code></td>
<td>
<p>Numeric. Number of segments in which to split each sound file. If supplied 'sgmt.dur' is ignored.</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_files">files</code></td>
<td>
<p>Character vector indicating the subset of files that will be split.</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>. Only used when</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_only.sels">only.sels</code></td>
<td>
<p>Logical argument to control if only the data frame is returned (no wave files are saved). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="split_sound_files_+3A_x">X</code></td>
<td>
<p>'selection_table' object or a data frame with columns
for sound file name (sound.files), selection number (selec), and start and end time of signal
(start and end). If supplied the data frame/selection table is modified to reflect the position of the selections in the new sound files. Note that some selections could split between 2 segments. To deal with this, a 'split.sels' column is added to the data frame in which those selection are labeled as 'split'. Default is <code>NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function aims to reduce the size of sound files in order to simplify some processes that are limited by sound file size (big files can be manipulated, e.g. <code><a href="#topic+auto_detec">auto_detec</a></code>). The function keeps the original number of channels in the output clips only for 1- and 2-channel files.
</p>


<h3>Value</h3>

<p>Wave files for each segment in the working directory (if <code>only.sels = FALSE</code>, named as 'sound.file.name-#.wav') and a data frame in the R environment containing the name of the original sound files (org.sound.files), the name of the clips (sound.files) and the start and end of clips in the original files. Clips are saved in .wav format. If 'X' is supplied then a data frame with the position of the selections in the newly created clips is returned instead.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+cut_sels">cut_sels</a></code>
</p>
<p>Other data manipulation: 
<code><a href="#topic+move_images">move_images</a>()</code>,
<code><a href="#topic+open_wd">open_wd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data and save to temporary working directory
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))

  # split files in 1 s files
  split_sound_files(sgmt.dur = 1, path = tempdir())

  # Check this folder
  tempdir()
}

</code></pre>

<hr>
<h2 id='split_wavs'>alternative name for <code><a href="#topic+split_sound_files">split_sound_files</a></code></h2><span id='topic+split_wavs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+split_sound_files">split_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_wavs(
  path = NULL,
  sgmt.dur = 10,
  sgmts = NULL,
  files = NULL,
  parallel = 1,
  pb = TRUE,
  only.sels = FALSE,
  X = NULL
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+split_sound_files">split_sound_files</a></code> for documentation. <code><a href="#topic+split_wavs">split_wavs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='tailor_sels'>Interactive view of spectrograms to tailor selections</h2><span id='topic+tailor_sels'></span>

<h3>Description</h3>

<p><code>tailor_sels</code> produces an interactive spectrographic view in
which the start/end times and frequency range of acoustic signals listed in a data frame can be adjusted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tailor_sels(X = NULL, wl = 512, flim = c(0,22), wn = "hanning", mar = 0.5,
 osci = TRUE, pal = reverse.gray.colors.2, ovlp = 70, auto.next = FALSE, pause = 1,
  comments = TRUE, path = NULL, frange = TRUE, fast.spec = FALSE, ext.window = TRUE,
  width = 15, height = 5, index = NULL, collevels = NULL,
  title = c("sound.files", "selec"), ts.df = NULL, col = "#E37222",
  alpha = 0.7, auto.contour = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tailor_sels_+3A_x">X</code></td>
<td>
<p>'selection_table', 'extended_selection_table' object or data frame with the following columns: 1) &quot;sound.files&quot;: name of the .wav
files, 2) &quot;selec&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections. Notice that, if an output file (&quot;seltailor_output.csv&quot;) is found in the working directory it will be given priority over an input data frame.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the spectrogram window length. Default is 512.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 specifying the frequency limit (in kHz) of
the spectrogram, as in the function <code><a href="seewave.html#topic+spectro">spectro</a></code>.
Default is c(0,22).</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_wn">wn</code></td>
<td>
<p>A character vector of length 1 specifying the window function (by default &quot;hanning&quot;).
See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the
start and end points of the selections to define spectrogram limits. Default is 0.5.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_osci">osci</code></td>
<td>
<p>Logical argument. If <code>TRUE</code> adds a oscillogram whenever the spectrograms are produced
with higher resolution (see seltime). Default is <code>TRUE</code>.
The external program must be closed before resuming analysis. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_pal">pal</code></td>
<td>
<p>A color palette function to be used to assign colors in the
plot, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is reverse.gray.colors.2. See Details.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying the percent overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_auto.next">auto.next</code></td>
<td>
<p>Logical argument to control whether the functions moves automatically to the
next selection. The time interval before moving to the next selection is controlled by the 'pause' argument. Ignored if <code>ts.df = TRUE</code>.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_pause">pause</code></td>
<td>
<p>Numeric vector of length 1. Controls the duration of the waiting period before
moving to the next selection (in seconds). Default is 1.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_comments">comments</code></td>
<td>
<p>Logical argument specifying if 'sel.comment' (when in data frame) should be included
in the title of the spectrograms. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_frange">frange</code></td>
<td>
<p>Logical argument specifying whether limits on frequency range should be
recorded.
If <code>TRUE</code> (default) time and frequency limits are recorded.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially
increases performance (much faster), although some options become unavailable, as sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_ext.window">ext.window</code></td>
<td>
<p>Logical. If <code>TRUE</code> then and external graphic window is used. Default
dimensions can be set using the 'width' and 'height' arguments. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_width">width</code></td>
<td>
<p>Numeric of length 1 controlling the width of the external graphic window. Ignored
if <code>ext.window = FALSE</code>. Default is 15.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_height">height</code></td>
<td>
<p>Numeric of length 1 controlling the height of the external graphic window.
Ignored if <code>ext.window = FALSE</code>. Default is 5.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_index">index</code></td>
<td>
<p>Numeric vector indicating which selections (rows) of 'X' should be tailored.
Default is <code>NULL</code>. Ignored when the process is resumed. This can be useful when combined
with <code><a href="#topic+filtersels">filtersels</a></code>) output (see 'index' argument in <code><a href="#topic+filtersels">filtersels</a></code>).</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_collevels">collevels</code></td>
<td>
<p>Numeric. Set of levels used to partition the amplitude range (see
<code><a href="seewave.html#topic+spectro">spectro</a></code>).</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_title">title</code></td>
<td>
<p>Character vector with the names of the columns to be included in the title for each
selection.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_ts.df">ts.df</code></td>
<td>
<p>Optional. Data frame with frequency contour time series of signals to be tailored. If provided then
'autonext' is set to <code>FALSE</code>. Default is <code>NULL</code>. The data frame must include the 'sound.files' and 'selec'
columns for the same selections included in 'X'.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_col">col</code></td>
<td>
<p>Character vector defining the color of the points when 'ts.df' is provided. Default is &quot;#E37222&quot; (orange).</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_alpha">alpha</code></td>
<td>
<p>Numeric of length one to adjust transparency of points when adjusting frequency contours.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_auto.contour">auto.contour</code></td>
<td>
<p>Logical. If <code>TRUE</code> contours are displayed automatically
(without having to click on 'contour'). Note that adjusting the selection box
(frequency/time limits) won't be available. Default is <code>FALSE</code>. Ignored if
'ts.df' is not provided.</p>
</td></tr>
<tr><td><code id="tailor_sels_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal spectrogram creating function for customizing graphical output. The function is a modified version of <code><a href="seewave.html#topic+spectro">spectro</a></code>, so it takes the same arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces an interactive spectrographic
view in which users can select new time/frequency
coordinates the selections. 4 &quot;buttons&quot; are provided at the upper right side of the spectrogram that
allow to stop the analysis (stop symbol, a solid rectangle), go to the next sound file (&quot;&gt;&gt;&quot;), return to the
previous selection (&quot;&lt;&lt;&quot;) or delete
the current selection (&quot;X&quot;). An additional &quot;button&quot; to tailored frequency contour is shown
when 'ts.df' is provided. The button contains a symbol with a 4 point contour. When a unit has been selected, the function plots
dotted lines in the start and end of the selection in the spectrogram (or a box if
<code>frange = TRUE</code>). Only the last selection is kept for each
selection that is adjusted. The function produces a .csv file (seltailor_output.csv)
with the same information than the input data frame, except for the new time
coordinates, plus a new column (X$tailored) indicating if the selection
has been tailored. The file is saved in the working directory  and is updated every time the user
moves into the next sound file (&quot;&gt;&gt;&quot;) or stop the process
(stop &quot;button&quot;). It also return the same data frame as and object in the R environment.
If no selection is made (by clicking on &quot;&gt;&gt;&quot;) the
original time/frequency coordinates are kept. When resuming the process (after &quot;stop&quot; and re-running
the function in the same working directory), the function will continue working on the
selections that have not been analyzed. When deleting a file (X button) an orange &quot;X&quot; when returning to that selection. If X is used again the selection is recovered.
The function also displays a progress bar right on
top of the spectrogram. The zoom can be adjusted by setting the <code>mar</code> argument.
To fix contours a data.frame containing the 'sound.files' and 'selec' columns as in 'X' as well
as the frequency values at each contour step must be provided. The function plots points corresponding to the
time/frequency coordinates of each element of the contour. Clicking on the spectrogram will substitute the
frequency value of the points. The contour point closest in time to the &quot;click&quot; will be replaced by the
frequency value of the &quot;click&quot;.
</p>


<h3>Value</h3>

<p>data frame similar to X with the and a .csv file saved in the working directory with start and end time of
selections.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

tailor_sels(X = lbh_selec_table, flim = c(1, 12), wl = 300, auto.next = TRUE, path = tempdir())

# Read output .csv file
seltailor.df &lt;- read.csv(file.path(tempdir(), "seltailor_output.csv"))
seltailor.df

# check this directory for .csv file after stopping function
tempdir()

## End(Not run)
</code></pre>

<hr>
<h2 id='test_coordination'>Randomization test for singing coordination</h2><span id='topic+test_coordination'></span>

<h3>Description</h3>

<p>Monte Carlo randomization test to assess the statistical significance of overlapping or alternating singing (or any other simultaneously occurring behavior).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_coordination(X, iterations = 1000, ovlp.method = "count",
randomization = "keep.gaps", less.than.chance = TRUE, parallel = 1, pb = TRUE,
rm.incomp = FALSE, cutoff = 2, rm.solo = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test_coordination_+3A_x">X</code></td>
<td>
<p>Data frame containing columns for singing event (sing.event),
individual (indiv), and start and end time of signal (start and end).</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_iterations">iterations</code></td>
<td>
<p>number of iterations for shuffling and calculation of the expected number of overlaps. Default is 1000.</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_ovlp.method">ovlp.method</code></td>
<td>
<p>Character string defining the method to measure the amount of overlap. Three methods are available:
</p>

<ul>
<li> <p><code>count</code>: count the number of overlapping signals (default)
</p>
</li>
<li> <p><code>time.overlap</code>: measure the total duration (in s) in which signals overlap
</p>
</li>
<li> <p><code>time.distance</code>: measure the time (in s) to the other individual's closest signal. This is the only method that can take more than 2 individuals.
</p>
</li></ul>
</td></tr>
<tr><td><code id="test_coordination_+3A_randomization">randomization</code></td>
<td>
<p>Character string defining the procedure for signal randomization. Three methods are available:
</p>

<ul>
<li> <p><code>keep.gaps</code> the position of both signals and gaps (i.e. intervals between signals) are randomized. Default.
</p>
</li>
<li> <p><code>sample.gaps</code> gaps are simulated using a lognormal distribution with
mean and standard deviation derived from the observed gaps. Signal position is randomized.
</p>
</li>
<li> <p><code>keep.song.order</code> only the position of gaps is randomized.
</p>
</li></ul>

<p>More details in Masco et al. (2015).</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_less.than.chance">less.than.chance</code></td>
<td>
<p>Logical. If <code>TRUE</code> the test evaluates whether overlaps occur less often than expected by chance.
If <code>FALSE</code> the opposite pattern is evaluated (whether overlaps occur more often than expected by chance).
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_rm.incomp">rm.incomp</code></td>
<td>
<p>Logical. If <code>TRUE</code> removes the events that don't have 2 interacting individuals. Default is
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_cutoff">cutoff</code></td>
<td>
<p>Numeric. Determines the minimum number of signals per individual in a singing event. Events not meeting
this criterium are removed. Default is 2.
Note that randomization tests are not reliable with very small sample sizes. Ideally 10 or more signals per individual
should be available in each singing event.</p>
</td></tr>
<tr><td><code id="test_coordination_+3A_rm.solo">rm.solo</code></td>
<td>
<p>Logical. Controls if signals that are not alternated at the start or end of the
sequence are removed (if <code>TRUE</code>). For instance, the sequence of signals A-A-A-B-A-B-A-B-B-B (in which A and B represent different individuals, as in the 'indiv' column) would be subset to
A-B-A-B-A-B. Default is  <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the probability of finding an equal or more extreme amount of song overlap (higher or lower) in a coordinated singing event (or any pair-coordinated behavior).
The function shuffles the sequences of signals and silence-between-signals for both individuals to produce
a null distribution of overlaps expected by chance. The observed overlaps is compared to this
expected values. The p-values are calculated as the proportion of random expected values that were lower (or higher)
than the observed value. All procedures described in Masco et al. (2015) are implemented. In addition, either the number (<code>ovlp.method = "count"</code>) or the total duration (<code>ovlp.method = "time.overlap"</code>) in which signals overlap can be used for estimating the overall degree of overlap. The function runs one test for each singing event in the input data frame. This function assumes that there are no overlaps between signals belonging to the same individual. See Masco et al. (2015) for recommendations on randomization procedures for specific signal structures.
</p>


<h3>Value</h3>

<p>A data frame with the following columns:
</p>

<ul>
<li> <p><code>sing.event</code>: singing event ID
</p>
</li>
<li> <p><code>obs.overlap</code>: observed amount of overlap (counts or total duration, depending on overlap method, see 'ovlp.method' argument)
</p>
</li>
<li> <p><code>mean.random.ovlp</code>: mean amount of overlap expected by chance
</p>
</li>
<li> <p><code>p.value</code>: p value
</p>
</li>
<li> <p><code>coor.score</code>: coordination score (<em>sensu</em> Araya-Salas et al. 2017),
calculated as:
</p>
<p style="text-align: center;"><code class="reqn">(obs.overlap - mean.random.ovlp) / mean.random.ovlp</code>
</p>

<p>Positive values indicate a tendency to overlap while negative values indicate a tendency to alternate. NA values will be returned when events cannot be randomized (e.g. too few signals).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals.
Methods in Ecology and Evolution, 8(2), 184-191.
</p>
<p>Araya-Salas M., Wojczulanis-Jakubas K., Phillips E.M., Mennill D.J., Wright T.F.
(2017) To overlap or not to overlap: context-dependent coordinated singing in
lekking long-billed hermits. Animal Behavior  124, 57-65.
</p>
<p>Keenan EL, Odom KJ, M Araya-Salas, KG Horton, M Strimas-Mackey,MA Meatte, NI Mann,PJ Slater,JJ Price, and CN Templeton . 2020. Breeding season length predicts duet coordination and consistency in Neotropical wrens (Troglodytidae). Proceeding of the Royal Society B. 20202482.
</p>
<p>Masco, C., Allesina, S., Mennill, D. J., and Pruett-Jones, S. (2015). The Song
Overlap Null model Generator (SONG): a new tool for distinguishing between random
and non-random song overlap. Bioacoustics.
</p>
<p>Rivera-Caceres K, E Quiros-Guerrero E, M Araya-Salas, C Templeton &amp; W Searcy. (2018). Early development of vocal interaction rules in a duetting songbird. Royal Society Open Science. 5, 171791.
</p>
<p>Rivera-Caceres K, E Quiros-Guerrero, M Araya-Salas &amp; W Searcy. (2016). Neotropical wrens learn new duet as adults. Proceedings of the Royal Society B. 285, 20161819

</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
#load  simulated singing data (see data documentation)
data(sim_coor_sing)

# set global options (this can also be set within the function call)
warbleR_options(iterations = 100, pb = FALSE)

# testing if coordination happens less than expected by chance
test_coordination(sim_coor_sing)

# testing if coordination happens more than expected by chance
test_coordination(sim_coor_sing, less.than.chance = FALSE)

# using "duration" method and "keep.song.order" as randomization procedure
test_coordination(sim_coor_sing, ovlp.method =  "time.overlap",
randomization = "keep.song.order")
}

</code></pre>

<hr>
<h2 id='track_freq_contour'>Spectrograms with frequency measurements</h2><span id='topic+track_freq_contour'></span>

<h3>Description</h3>

<p><code>track_freq_contour</code> creates spectrograms to visualize dominant and fundamental frequency measurements (contours)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_freq_contour(X, wl = 512, wl.freq = 512, flim = NULL, wn = "hanning", pal =
  reverse.gray.colors.2, ovlp = 70, inner.mar = c(5, 4, 4, 2), outer.mar =
  c(0, 0, 0, 0), picsize = 1, res = 100, cexlab = 1, title = TRUE, propwidth = FALSE,
  xl = 1, osci = FALSE, gr = FALSE, sc = FALSE, bp = NULL, cex = c(0.6, 1),
  threshold = 15, threshold.time = NULL, threshold.freq = NULL, contour = "both",
  col = c("#E37222B3", "#07889BB3"), pch = c(21, 24),  mar = 0.05, lpos = "topright",
  it = "jpeg", parallel = 1, path = NULL, img.suffix = NULL, custom.contour = NULL,
  pb = TRUE, type = "p", leglab = c("Ffreq", "Dfreq"), col.alpha = 0.6, line = TRUE,
   fast.spec = FALSE, ff.method = "seewave", frange.detec = FALSE, fsmooth = 0.1,
   widths = c(2, 1), freq.continuity = NULL, clip.edges = 2, track.harm = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_freq_contour_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data frame containing columns for sound file name (sound.files),
selection number (selec), and start and end time of signal (start and end).
The output <code><a href="#topic+auto_detec">auto_detec</a></code> can also be used as the input data frame.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram, default
is 512.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_wl.freq">wl.freq</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length of the spectrogram
for measurements on the frequency spectrum. Default is 512. Higher values would provide
more accurate measurements.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_flim">flim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit of
the spectrogram (in kHz), as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_pal">pal</code></td>
<td>
<p>A color palette function to be used to assign colors in the
plot, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is reverse.gray.colors.2.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 70.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_inner.mar">inner.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is c(5,4,4,2).
Specifies number of lines in inner plot margins where axis labels fall,
with form c(bottom, left, top, right). See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_outer.mar">outer.mar</code></td>
<td>
<p>Numeric vector with 4 elements, default is c(0,0,0,0).
Specifies number of lines in outer plot margins beyond axis labels, with
form c(bottom, left, top, right). See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_picsize">picsize</code></td>
<td>
<p>Numeric argument of length 1. Controls relative size of
spectrogram. Default is 1.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_res">res</code></td>
<td>
<p>Numeric argument of length 1. Controls image resolution.
Default is 100 (faster) although 300 - 400 is recommended for publication/
presentation quality.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_cexlab">cexlab</code></td>
<td>
<p>Numeric vector of length 1 specifying the relative size of axis
labels. See <code><a href="seewave.html#topic+spectro">spectro</a></code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_title">title</code></td>
<td>
<p>Logical argument to add a title to individual spectrograms.
Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_propwidth">propwidth</code></td>
<td>
<p>Logical argument to scale the width of spectrogram
proportionally to duration of the selected call. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_xl">xl</code></td>
<td>
<p>Numeric vector of length 1. A constant by which to scale
spectrogram width. Default is 1.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_osci">osci</code></td>
<td>
<p>Logical argument to add an oscillogram underneath spectrogram, as
in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_gr">gr</code></td>
<td>
<p>Logical argument to add grid to spectrogram. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_sc">sc</code></td>
<td>
<p>Logical argument to add amplitude scale to spectrogram, default is
<code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_bp">bp</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a
frequency bandpass filter (in kHz) or &quot;frange&quot; to indicate that values in bottom.freq
and top.freq columns will be used as bandpass limits. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_cex">cex</code></td>
<td>
<p>Numeric vector of length 2, specifies relative size of points
plotted for frequency measurements and legend font/points, respectively.
See <code><a href="seewave.html#topic+spectro">spectro</a></code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_threshold">threshold</code></td>
<td>
<p>amplitude threshold (%) for fundamental and dominant frequency detection as well as frequency range from the spectrum (see 'frange.detec'). Default is 15. WILL BE DEPRECATED. Use 'threshold.time' and 'threshold.time' instead.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_threshold.time">threshold.time</code></td>
<td>
<p>amplitude threshold (%) for the time domain. Use for fundamental and dominant frequency detection. If <code>NULL</code> (default) then the 'threshold' value is used.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_threshold.freq">threshold.freq</code></td>
<td>
<p>amplitude threshold (%) for the frequency domain. Use for frequency range detection from the spectrum (see 'frange.detec'). If <code>NULL</code> (default) then the
'threshold' value is used.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_contour">contour</code></td>
<td>
<p>Character vector, one of &quot;df&quot;, &quot;ff&quot; or &quot;both&quot;, specifying whether the
dominant or fundamental frequencies or both should be plotted. Default is &quot;both&quot;.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_col">col</code></td>
<td>
<p>Vector of length 1 or 2 specifying colors of points plotted to mark
fundamental and dominant frequency measurements respectively (if both are plotted). Default is <code>c("#E37222B3",
"#07889BB3")</code>. Extreme values (lowest and highest) are highlighted in yellow.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_pch">pch</code></td>
<td>
<p>Numeric vector of length 1 or 2 specifying plotting characters for
the frequency measurements. Default is c(21, 24).</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_mar">mar</code></td>
<td>
<p>Numeric vector of length 1. Specifies the margins adjacent to the selections
to set spectrogram limits. Default is 0.05.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_lpos">lpos</code></td>
<td>
<p>Character vector of length 1 or numeric vector of length 2,
specifying position of legend. If the former, any keyword accepted by
xy.coords can be used (see below). If the latter, the first value will be the x
coordinate and the second value the y coordinate for the legend's position.
Default is &quot;topright&quot;.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_it">it</code></td>
<td>
<p>A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted. Default is &quot;jpeg&quot;.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_img.suffix">img.suffix</code></td>
<td>
<p>A character vector of length 1 with a suffix (label) to add at the end of the names of
image files. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_custom.contour">custom.contour</code></td>
<td>
<p>A data frame with frequency contours for exactly the same sound files and selection as in X.
The frequency values are assumed to be equally spaced in between the start and end of the signal. The
first 2 columns of the data frame should contain the 'sound.files' and 'selec' columns and should be
identical to the corresponding columns in X (same order).</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_type">type</code></td>
<td>
<p>A character vector of length 1 indicating the type of frequency contour plot to be drawn.
Possible types are &quot;p&quot; for points, &quot;l&quot; for lines and &quot;b&quot; for both.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_leglab">leglab</code></td>
<td>
<p>A character vector of length 1 or 2 containing the label(s) of the frequency contour legend
in the output image.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_col.alpha">col.alpha</code></td>
<td>
<p>A numeric vector of length 1  within [0,1] indicating how transparent the lines/points should be.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_line">line</code></td>
<td>
<p>Logical argument to add red lines (or box if bottom.freq and top.freq columns are provided) at start and end times of selection. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_fast.spec">fast.spec</code></td>
<td>
<p>Logical. If <code>TRUE</code> then image function is used internally to create spectrograms, which substantially
increases performance (much faster), although some options become unavailable, as collevels, and sc (amplitude scale).
This option is indicated for signals with high background noise levels. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code>, <code><a href="monitoR.html#topic+specCols">topo.1</a></code> and <code><a href="monitoR.html#topic+specCols">rainbow.1</a></code> (which should be imported from the package monitoR) seem
to work better with 'fast' spectrograms. Palette colors <code><a href="monitoR.html#topic+specCols">gray.1</a></code>, <code><a href="monitoR.html#topic+specCols">gray.2</a></code>,
<code><a href="monitoR.html#topic+specCols">gray.3</a></code> offer
decreasing darkness levels.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_ff.method">ff.method</code></td>
<td>
<p>Character. Selects the method used to calculate the fundamental
frequency. Either 'tuneR' (using <code><a href="tuneR.html#topic+FF">FF</a></code>) or 'seewave' (using
<code><a href="seewave.html#topic+fund">fund</a></code>). Default is 'seewave'. 'tuneR' performs faster (and seems to be more accurate) than 'seewave'.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_frange.detec">frange.detec</code></td>
<td>
<p>Logical. Controls whether frequency range of signal is automatically
detected  using the <code><a href="#topic+freq_range_detec">freq_range_detec</a></code> function. If so, the range is used as the
bandpass filter (overwriting 'bp' argument). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_fsmooth">fsmooth</code></td>
<td>
<p>A numeric vector of length 1 to smooth the frequency spectrum with a mean
sliding window (in kHz) used for frequency range detection (when <code>frange.detec = TRUE</code>). This help to average amplitude &quot;hills&quot; to minimize the effect of
amplitude modulation. Default is 0.1.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_widths">widths</code></td>
<td>
<p>Numeric vector of length 2 to control the relative widths of the spectro (first element) and spectrum (second element,  (when <code>frange.detec = TRUE</code>)).</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_freq.continuity">freq.continuity</code></td>
<td>
<p>Numeric vector of length 1 to control whether dominant frequency detections
outliers(i.e that differ from the frequency of the detections right before and after) would be removed. Should be given in kHz. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_clip.edges">clip.edges</code></td>
<td>
<p>Integer vector of length 1 to control if how many 'frequency-wise discontinuous' detection would be remove at the start and end of signals (see
'freq.continuity' argument). Default is 2. Ignored if <code>freq.continuity = NULL</code>.</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_track.harm">track.harm</code></td>
<td>
<p>Logical to control if <code><a href="#topic+track_harmonic">track_harmonic</a></code> or a modified version of <code><a href="seewave.html#topic+dfreq">dfreq</a></code> is used for dominant frequency detection. Default is <code>FALSE</code> (use <code><a href="seewave.html#topic+dfreq">dfreq</a></code>).</p>
</td></tr>
<tr><td><code id="track_freq_contour_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the internal spectrogram creating function for customizing graphical output. The function is a modified version of <code><a href="seewave.html#topic+spectro">spectro</a></code>, so it takes the same arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides visualization of frequency measurements as the ones
made by <code><a href="#topic+spectro_analysis">spectro_analysis</a></code>, <code><a href="#topic+freq_ts">freq_ts</a></code> and <code><a href="#topic+freq_DTW">freq_DTW</a></code>. Frequency measures can be made by the function or input by the
user (see 'custom.contour' argument). If <code>frange = TRUE</code> the function uses <code><a href="#topic+freq_range_detec">freq_range_detec</a></code> to detect the frequency range. In this case the graphical output includes a
frequency spectrum showing the detection threshold. Extreme values (lowest and highest) are highlighted in yellow.
Note that, unlike other warbleR functions that measure frequency contours, track_freq_contour do not interpolate frequency values.
</p>


<h3>Value</h3>

<p>Spectrograms of the signals listed in the input data frame showing the location of
the dominant and fundamental frequencies.
</p>


<h3>Author(s)</h3>

<p>Grace Smith Vidaurre and Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+spectrograms">spectrograms</a></code> for creating spectrograms from selections,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code> for creating spectrograms to
optimize noise margins used in <code><a href="#topic+sig2noise">sig2noise</a></code>
</p>
<p>Other spectrogram creators: 
<code><a href="#topic+color_spectro">color_spectro</a>()</code>,
<code><a href="#topic+freq_DTW">freq_DTW</a>()</code>,
<code><a href="#topic+multi_DTW">multi_DTW</a>()</code>,
<code><a href="#topic+phylo_spectro">phylo_spectro</a>()</code>,
<code><a href="#topic+snr_spectrograms">snr_spectrograms</a>()</code>,
<code><a href="#topic+spectrograms">spectrograms</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data
  data("Cryp.soui")
  writeWave(Cryp.soui, file.path(tempdir(), "Cryp.soui.wav")) # save sound files

  # autodetec location of signals
  ad &lt;- auto_detec(
    threshold = 6, bp = c(1, 3), mindur = 1.2, flim = c(0, 5),
    maxdur = 3, img = FALSE, ssmooth = 600, wl = 300, flist = "Cryp.soui.wav",
    path = tempdir()
  )

  # track dominant frequency graphs with freq range detection
  track_freq_contour(
    X = ad[!is.na(ad$start), ], flim = c(0, 5), ovlp = 90,
    it = "tiff", bp = c(1, 3), contour = "df", wl = 300, frange = TRUE,
    path = tempdir()
  )

  # using users frequency data (custom.contour argument)
  # first get contours using freq_ts
  df &lt;- freq_ts(
    X = ad[!is.na(ad$start), ], flim = c(0, 5), ovlp = 90, img = FALSE,
    bp = c(1, 3), wl = 300, path = tempdir()
  )

  # now input the freq_ts output into track_freq_contour
  track_freq_contour(
    X = ad[!is.na(ad$start), ], custom.contour = df, flim = c(0, 5), ovlp = 90,
    it = "tiff", path = tempdir()
  )

  # Check this folder
  tempdir()

  # track both frequencies
  track_freq_contour(
    X = ad[!is.na(ad$start), ], flim = c(0, 5), ovlp = 90,
    it = "tiff", bp = c(1, 3), contour = "both", wl = 300, path = tempdir()
  )
}

</code></pre>

<hr>
<h2 id='track_harmonic'>Track harmonic frequency contour</h2><span id='topic+track_harmonic'></span>

<h3>Description</h3>

<p><code>track_harmonic</code> tracks the frequency contour of the dominant harmonic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_harmonic(wave, f, wl = 512, wn = "hanning", ovlp = 0, fftw = FALSE, at = NULL,
tlim = NULL, threshold = 10, bandpass = NULL, clip = NULL, plot = TRUE,
xlab = "Times (s)", ylab = "Frequency (kHz)",  ylim = c(0, f/2000),
adjust.wl = FALSE, dfrq = FALSE, ...)

track_harmonic(
  wave,
  f,
  wl = 512,
  wn = "hanning",
  ovlp = 0,
  fftw = FALSE,
  at = NULL,
  tlim = NULL,
  threshold = 10,
  bandpass = NULL,
  clip = NULL,
  plot = TRUE,
  xlab = "Times (s)",
  ylab = "Frequency (kHz)",
  ylim = c(0, f/2000),
  adjust.wl = FALSE,
  dfrq = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_harmonic_+3A_wave">wave</code></td>
<td>
<p>A 'wave' object produced by <code><a href="tuneR.html#topic+readWave">readWave</a></code> or similar functions.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_f">f</code></td>
<td>
<p>Sampling frequency of the wave object (in Hz). Does not need to be specified if embedded in wave.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 specifying the window length for the FFT, default
is 512.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_wn">wn</code></td>
<td>
<p>Character vector of length 1 specifying window name. Default is
&quot;hanning&quot;. See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for more options. This is used for calculating the frequency spectrum (using <code><a href="seewave.html#topic+meanspec">meanspec</a></code>) and producing the spectrogram (using <code><a href="seewave.html#topic+spectro">spectro</a></code>, if <code>plot = TRUE</code>).</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 specifying % of overlap between two
consecutive time windows, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 0.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_fftw">fftw</code></td>
<td>
<p>if TRUE calls the function FFT of the library fftw. See Notes of the <code><a href="seewave.html#topic+spectro">spectro</a></code> function.
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_at">at</code></td>
<td>
<p>Time position where the harmonic frequency contour has to be computed (in seconds). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_tlim">tlim</code></td>
<td>
<p>time range in which to measure frequency contours. Default is <code>NULL</code> (which means it will measure
across the entire wave object).</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_threshold">threshold</code></td>
<td>
<p>Amplitude threshold (%) for dominant frequency and detection. Default is 10.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_bandpass">bandpass</code></td>
<td>
<p>A numeric vector of length 2 for the lower and upper limits of a frequency bandpass filter (in kHz).</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_clip">clip</code></td>
<td>
<p>A numeric value to select dominant frequency values according to their amplitude in reference to a maximal value of 1 for the whole signal (has to be &gt;0 &amp; &lt; 1).</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_plot">plot</code></td>
<td>
<p>Logical, if TRUE plots the dominant frequency against time. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_xlab">xlab</code></td>
<td>
<p>Label of the time axis.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_ylab">ylab</code></td>
<td>
<p>Label of the frequency axis.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_ylim">ylim</code></td>
<td>
<p>A numeric vector of length 2 for the frequency limit of
the spectrogram (in kHz), as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is c(0, f/2000).</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_adjust.wl">adjust.wl</code></td>
<td>
<p>Logical. If <code>TRUE</code> 'wl' (window length) is reset to be lower than the
number of samples in a selection if the number of samples is less than 'wl'. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_dfrq">dfrq</code></td>
<td>
<p>Logical. If <code>TRUE</code> seewave's <code><a href="seewave.html#topic+dfreq">dfreq</a></code> is used instead. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="track_harmonic_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the plotting function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a modified version of seewave's <code><a href="seewave.html#topic+dfreq">dfreq</a></code> function that allows to track the frequency
contour of a dominant harmonic even when the highest amplitude jumps between harmonics. The arguments and default values of the
original <code><a href="seewave.html#topic+dfreq">dfreq</a></code> function have been kept unchanged to facilitate switching between the 2 functions.
</p>
<p>see <code><a href="#topic+track_harmonic">track_harmonic</a></code> for documentation. <code><a href="#topic+track_harmonic">track_harmonic</a></code> will be deprecated in future versions.
</p>


<h3>Author(s)</h3>

<p>Jerome Sueur, modified by Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for tracking frequencies iteratively on selections tables.
</p>

<hr>
<h2 id='trackfreqs'>alternative name for <code><a href="#topic+track_freq_contour">track_freq_contour</a></code></h2><span id='topic+trackfreqs'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+track_freq_contour">track_freq_contour</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trackfreqs(
  X,
  wl = 512,
  wl.freq = 512,
  flim = NULL,
  wn = "hanning",
  pal = reverse.gray.colors.2,
  ovlp = 70,
  inner.mar = c(5, 4, 4, 2),
  outer.mar = c(0, 0, 0, 0),
  picsize = 1,
  res = 100,
  cexlab = 1,
  title = TRUE,
  propwidth = FALSE,
  xl = 1,
  osci = FALSE,
  gr = FALSE,
  sc = FALSE,
  bp = NULL,
  cex = c(0.6, 1),
  threshold = 15,
  threshold.time = NULL,
  threshold.freq = NULL,
  contour = "both",
  col = c("#E37222B3", "#07889BB3"),
  pch = c(21, 24),
  mar = 0.05,
  lpos = "topright",
  it = "jpeg",
  parallel = 1,
  path = NULL,
  img.suffix = NULL,
  custom.contour = NULL,
  pb = TRUE,
  type = "p",
  leglab = c("Ffreq", "Dfreq"),
  col.alpha = 0.6,
  line = TRUE,
  fast.spec = FALSE,
  ff.method = "seewave",
  frange.detec = FALSE,
  fsmooth = 0.1,
  widths = c(2, 1),
  freq.continuity = NULL,
  clip.edges = 2,
  track.harm = FALSE,
  ...
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+track_freq_contour">track_freq_contour</a></code> for documentation. <code><a href="#topic+trackfreqs">trackfreqs</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='try_na'>Wrapper for &quot;try&quot; function</h2><span id='topic+try_na'></span>

<h3>Description</h3>

<p><code>try_na</code> silly wrapper for <code><a href="base.html#topic+try">try</a></code> function that returns an NA if an error is found. TO BE DEPRECATED IN FUTURE VERSIONS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>try_na(expr, silent = TRUE, outFile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="try_na_+3A_expr">expr</code></td>
<td>
<p>An R expression to try.</p>
</td></tr>
<tr><td><code id="try_na_+3A_silent">silent</code></td>
<td>
<p>Logical to control whether the report of error messages is suppressed. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="try_na_+3A_outfile">outFile</code></td>
<td>
<p>A connection, or a character string naming the file to print to
(via message2(*, file = outFile)); used only if silent is false, as by default.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a silly wrapper on <code><a href="base.html#topic+try">try</a></code> that returns an 'NA' if any error occurs during the evaluation of a expression. See <code><a href="base.html#topic+try">try</a></code> for details.
</p>


<h3>Value</h3>

<p>Returns an 'NA' if any error occurs during the evaluation of a expression.
If not, it will return the result of the evaluation.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.

</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
# try a function that does not exists to produce an error
try_na(crazy78(12))

# try a real function (no error)
try_na(mean(1:5))
}

</code></pre>

<hr>
<h2 id='tweak_spectro'>Plot a mosaic of spectrograms with varying display parameters</h2><span id='topic+tweak_spectro'></span>

<h3>Description</h3>

<p><code>tweak_spectro</code> plots a mosaic of spectrograms with varying display parameters to
facilitate selection of display parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tweak_spectro(X, length.out = 5, ovlp = 90, wl = c(100, 1000), wn = "hanning",
 collev.min = -40, pal = "reverse.gray.colors.2", path = NULL, rm.axes = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tweak_spectro_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data frame with a single row and columns for sound file name (sound.files), selection number (selec), 
and start and end time of signal (start and end). Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_length.out">length.out</code></td>
<td>
<p>Numeric vector of length 1 controlling the number of sublevels of
the numeric arguments for which a range has been provided. Ranges are allowed for
'ovlp', 'wl', and 'collev.min' arguments.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_ovlp">ovlp</code></td>
<td>
<p>Numeric vector of length 1 or 2 specifying % of overlap (or 
lower/upper values the desired range) between two consecutive windows, as in 
<code><a href="seewave.html#topic+spectro">spectro</a></code>. Default is 90.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_wl">wl</code></td>
<td>
<p>A numeric vector of length 1 or 2 specifying the window length (length 1)
or the lower and upper range limits of the desired window length range (length 2) for creating spectrograms.
Default is c(100, 1000).</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_wn">wn</code></td>
<td>
<p>Character vector specifying the window function names to be used. Several
names can be provided. See <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code>
for name options. Default is &quot;hanning&quot;. If &quot;all&quot;, then all window functions available are used.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_collev.min">collev.min</code></td>
<td>
<p>A (negative) numeric vector of length 1 or 2. Determines the first argument
to use in 'collevels' for the internal spectrogram creating function. This replaces the 
first element in the 'collevels' as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. Note that
'collevels' is not available in this function <code><a href="#topic+tweak_spectro">tweak_spectro</a></code>.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_pal">pal</code></td>
<td>
<p>Color palette function for spectrogram. Default is &quot;reverse.gray.colors.2&quot;. 
Several palettes can be provided in a character vector. Note that, contrary to
other <code>warbleR</code> and <code>seewave</code> functions, the 
palette most be provided as character string rather than as a function. See 
<code><a href="seewave.html#topic+spectro">spectro</a></code> for more palettes.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound file are located.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_rm.axes">rm.axes</code></td>
<td>
<p>Logical. If <code>TRUE</code> frequency and time axes are excluded. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="tweak_spectro_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to <code><a href="#topic+catalog">catalog</a></code> function for customizing
graphical output. Check out <code><a href="#topic+catalog">catalog</a></code> for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions aims to simplify the selection of spectrogram parameters. 
The function plots, for a single selection, a mosaic of spectrograms with varying 
display parameters. For numeric arguments the upper and lower limits of a range can
be provided. The following arguments accept can have varying values:
</p>

<ul>
<li> <p><code>wl</code>: Windows length (numeric range)
</p>
</li>
<li> <p><code>ovlp</code>: Overlap (numeric range)
</p>
</li>
<li> <p><code>collev.min</code>: Minimum value of the color levels (numeric range) 
</p>
</li>
<li> <p><code>wn</code>: window function names (character)
</p>
</li>
<li> <p><code>pal</code>: palette (character)
</p>
</li></ul>

<p>Outputs are similar to those of <code><a href="#topic+catalog">catalog</a></code>. The output image files can be put together in a single pdf file with <code><a href="#topic+catalog2pdf">catalog2pdf</a></code>.
We recommend using low resolution (~60-100) and smaller dimensions (width &amp; height &lt; 10) if
aiming to generate pdfs (otherwise pdfs could be pretty big).
</p>


<h3>Value</h3>

<p>Image files with spectrograms of entire sound files in the working directory. Multiple pages
can be returned, depending on the length of each sound file.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+catalog2pdf">catalog2pdf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Save to temporary working directory

# save sound file examples
data(list = c("Phae.long1", "lbh_selec_table"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav")) 

# variable collevels
tweak_spectro(X = lbh_selec_table, wl = 164, ovlp = c(90), wn = c("flattop"), 
length.out = 16, nrow = 4, ncol = 4, width = 20, height = 11.3, rm.axes = TRUE, 
cex = 1, box = F, collev.min = c(-20, -150), path = tempdir(), flim = c(0, 10))

# variable overlap and wn
tweak_spectro(X = lbh_selec_table, wl = 164, ovlp = c(50, 90), 
wn = c("hanning", "hamming", "rectangle", "bartlett", "blackman", "flattop"),
length.out = 7, nrow = 6, ncol = 7, width = 20, height = 11.3, rm.axes = TRUE, 
cex = 1, box = F, path = tempdir(), flim = c(0, 10))

# variable wl and wn
tweak_spectro(X = lbh_selec_table, wl = c(100, 1000), ovlp = c(50, 90), wn = "all", 
length.out = 5, nrow = 10, ncol = 14, width = 20, height = 11.3, rm.axes = TRUE, 
cex = 0.7, path = tempdir(), flim = c(0, 10))

# variable wl, collev.min and wn 
tweak_spectro(X = lbh_selec_table, wl = c(100, 1000), ovlp = 90, 
wn = c("hanning", "hamming", "rectangle"), collev.min = c(-110, -25), 
length.out = 3, nrow = 10, ncol = 14, width = 20, height = 11.3, rm.axes = TRUE,
 cex = 0.7, path = tempdir(), flim = c(0, 10))
 
 # variable wl, wn and pal
 tweak_spectro(X = lbh_selec_table, wl = c(100, 1000), ovlp = 90, 
 wn = c("hanning", "hamming", "rectangle"), 
 pal = c("reverse.gray.colors.2", "reverse.topo.colors", 
 "reverse.terrain.colors", "reverse.cm.colors"), 
 length.out = 4, nrow = 5, ncol = 10, width = 20, height = 11.3,
  rm.axes = TRUE, cex = 0.7, lab.mar = 2, path = tempdir(), flim = c(0, 10))
  
  # wl, wn and pal
  tweak_spectro(X = lbh_selec_table, wl = c(100, 1000), ovlp = 90,
   wn = c("hanning", "hamming", "rectangle"), 
  pal = c("reverse.gray.colors.2", "reverse.topo.colors", 
  "reverse.terrain.colors", "reverse.cm.colors"), 
  length.out = 4, nrow = 5, ncol = 10, width = 20, height = 11.3, rm.axes = TRUE,
   cex = 0.7, group.tag = "wn",  spec.mar = 0.4, lab.mar = 0.8, box = FALSE, 
   tag.pal = list(reverse.cm.colors), path = tempdir(), flim = c(0, 10))

check this floder
tempdir()

## End(Not run)
</code></pre>

<hr>
<h2 id='warbleR'>warbleR: A package to streamline bioacoustic analysis</h2><span id='topic+warbleR-package'></span><span id='topic+warbleR'></span>

<h3>Description</h3>

<p>warbleR is intended to facilitate the analysis of the structure of animal acoustic signals in R. Users can collect open-access avian recordings or enter their own data into a workflow that facilitates spectrographic visualization and measurement of acoustic parameters. warbleR makes use of the fundamental sound analysis tools of the seewave package, and offers new tools for acoustic structure analysis. These tools are available for batch analysis of acoustic signals.
</p>


<h3>Details</h3>

<p>The main features of the package are:
</p>

<ul>
<li><p> The use of loops to apply tasks through acoustic signals referenced in a selection table
</p>
</li>
<li><p> The production of images in the working folder with spectrograms that allow to organize data and verify acoustic analyzes
</p>
</li></ul>

<p>The package offers functions to:
</p>

<ul>
<li><p> Explore and download Xeno Canto recordings
</p>
</li>
<li><p> Explore, organize and manipulate multiple sound files
</p>
</li>
<li><p> Detect signals automatically (in frequency and time)
</p>
</li>
<li><p> Create spectrograms of complete recordings or individual signals
</p>
</li>
<li><p> Run different measures of acoustic signal structure
</p>
</li>
<li><p> Evaluate the performance of measurement methods
</p>
</li>
<li><p> Catalog signals
</p>
</li>
<li><p> Characterize different structural levels in acoustic signals
</p>
</li>
<li><p> Statistical analysis of duet coordination
</p>
</li>
<li><p> Consolidate databases and annotation tables
</p>
</li></ul>

<p>Most of the functions allow the parallelization of tasks, which distributes the tasks among several processors to improve computational efficiency. Tools to evaluate the performance of the analysis at each step are also available. In addition, warbleR satisfies the need for rigorous open source bioacoustic analysis, which facilitates opportunities for use in research and innovation of additional custom analyzes.
</p>
<p>The warbleR package offers three overarching categories of
functions:
</p>
<p>License: GPL (&gt;= 2)
</p>


<h3>Obtaining animal vocalization data</h3>

<p><code><a href="#topic+query_xc">query_xc</a></code>: Download recordings and/or metadata from 'Xeno-Canto'
</p>
<p><code><a href="#topic+sim_songs">sim_songs</a></code>: Simulate animal vocalizations
</p>


<h3>Managing sound files</h3>

<p><code><a href="#topic+read_wave">read_wave</a></code>: Read wave files into 'wave' objects
</p>
<p><code><a href="#topic+read_sound_file">read_sound_file</a></code>: Read sound files into 'wave' objects
</p>
<p><code><a href="#topic+selection_table">selection_table</a></code>: Create 'selection_table' class objects
</p>
<p><code><a href="#topic+mp32wav">mp32wav</a></code>: Convert several .mp3 files in working directory to .wav
format
</p>
<p><code><a href="#topic+check_sels">check_sels</a></code>: Check whether selections can be read by subsequent functions
</p>
<p><code><a href="#topic+check_wavs">check_wavs</a></code>: Check whether .wav files can be read by subsequent
functions and the minimum windows length (&quot;wl&quot; argument) that can be used
</p>
<p><code><a href="#topic+fix_wavs">fix_wavs</a></code>: Fix .wav files so they can be read by other functions
</p>
<p><code><a href="#topic+split_wavs">split_wavs</a></code>: Split .wav fies in several sound files
</p>
<p><code><a href="#topic+resample_est_waves">resample_est_waves</a></code>: Resample wave objects in extended selection tables
</p>
<p><code><a href="#topic+duration_wavs">duration_wavs</a></code>: Determine the duration of sound files
</p>
<p><code><a href="#topic+cut_sels">cut_sels</a></code>: Cut selections from a selection table into individual sound files
</p>
<p><code><a href="#topic+remove_silence">remove_silence</a></code>: Remove silence segments from wave files
</p>
<p><code><a href="#topic+remove_channels">remove_channels</a></code>: Remove channels in wave files
</p>
<p><code><a href="#topic+consolidate">consolidate</a></code>: Consolidate sound files into a single folder
</p>
<p><code><a href="#topic+selection_table">selection_table</a></code>: Create double-checked and self-contained selection tables
</p>
<p><code><a href="#topic+fix_extended_selection_table">fix_extended_selection_table</a></code>: Fix attributes of extended selection tables
</p>


<h3>Exploring/analyzing signal structure</h3>

<p><code><a href="#topic+auto_detec">auto_detec</a></code>: Automatically detect start and
end of acoustic signals
</p>
<p><code><a href="#topic+tailor_sels">tailor_sels</a></code>: Interactive view of spectrograms to tailor start and end of selections
</p>
<p><code><a href="#topic+sig2noise">sig2noise</a></code>: Measure signal-to-noise ratio across multiple files
</p>
<p><code><a href="#topic+track_freq_contour">track_freq_contour</a></code>: Create spectrograms to visualize frequency
measurements
</p>
<p><code><a href="#topic+filter_sels">filter_sels</a></code>: Filter selection data frames based on filtered image files
</p>
<p><code><a href="#topic+freq_range">freq_range</a></code>: Detect frequency range iteratively from signals in a selection table
</p>
<p><code><a href="#topic+freq_range_detec">freq_range_detec</a></code>: Detect frequency range in a Wave object
</p>
<p><code><a href="#topic+spectro_analysis">spectro_analysis</a></code>: Measure acoustic parameters on selected acoustic
signals
</p>
<p><code><a href="#topic+mfcc_stats">mfcc_stats</a></code>: Calculate descriptive statistics on Mel-frequency cepstral coefficients
</p>
<p><code><a href="#topic+song_analysis">song_analysis</a></code>: Measure acoustic parameters at higher levels of organization
</p>
<p><code><a href="#topic+cross_correlation">cross_correlation</a></code>: Pairwise cross-correlation of multiple signals
</p>
<p><code><a href="#topic+gaps">gaps</a></code>: Measures gap duration
</p>
<p><code><a href="#topic+freq_ts">freq_ts</a></code>: Extract frequency contours the signal as a time series
</p>
<p><code><a href="#topic+find_peaks">find_peaks</a></code>: Find peaks in cross-correlation scores from <code><a href="#topic+cross_correlation">cross_correlation</a></code>
</p>
<p><code><a href="#topic+freq_DTW">freq_DTW</a></code>: Calculate acoustic dissimilarity using dynamic time warping
on frequency contours
</p>
<p><code><a href="#topic+wpd_features">wpd_features</a></code>: Measure wavelet packet decomposition features
</p>
<p><code><a href="#topic+compare_methods">compare_methods</a></code>: Produce graphs to visually assess performance of acoustic
distance measurements
</p>
<p><code><a href="#topic+test_coordination">test_coordination</a></code>: Assess statistical significance of singing coordination
</p>
<p><code><a href="#topic+overlapping_sels">overlapping_sels</a></code>: Find selections that overlap in time within a given sound file
</p>
<p><code><a href="#topic+track_harmonic">track_harmonic</a></code>:  Track harmonic frequency contour
</p>


<h3>Graphical outputs</h3>

<p><code><a href="#topic+map_xc">map_xc</a></code>: Create maps to visualize the geographic spread of 'Xeno-Canto' recordings
</p>
<p><code><a href="#topic+catalog">catalog</a></code>: Produce a vocalization catalog with spectrograms in and array with
several rows and columns
</p>
<p><code><a href="#topic+catalog2pdf">catalog2pdf</a></code>: Combine catalog images to single pdf files
</p>
<p><code><a href="#topic+plot_coordination">plot_coordination</a></code>: Create graphs of coordinated singing
</p>
<p><code><a href="#topic+color_spectro">color_spectro</a></code>: Highlight spectrogram regions
</p>
<p><code><a href="#topic+full_spectrograms">full_spectrograms</a></code>: Produce spectrograms of whole recordings split into
multiple rows
</p>
<p><code><a href="#topic+full_spectrogram2pdf">full_spectrogram2pdf</a></code>: Combine lspec images to single pdf files
</p>
<p><code><a href="#topic+spectrograms">spectrograms</a></code>: Create spectrograms of selections
</p>
<p><code><a href="#topic+snr_spectrograms">snr_spectrograms</a></code>: Create spectrograms to visualize margins over which
noise will be measured by sig2noise
</p>
<p><code><a href="#topic+phylo_spectro">phylo_spectro</a></code>: Add spectrograms onto phylogenetic trees
</p>
<p><code><a href="#topic+tweak_spectro">tweak_spectro</a></code>: Visually inspect effect of different settings for creating (and improving) spectrograms
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas &amp; Grace Smith Vidaurre
</p>
<p>Maintainer: Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://marce10.github.io/warbleR/">https://marce10.github.io/warbleR/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/maRce10/warbleR/issues/">https://github.com/maRce10/warbleR/issues/</a>
</p>
</li></ul>


<hr>
<h2 id='warbleR_options'>Setting warbleR options</h2><span id='topic+warbleR_options'></span>

<h3>Description</h3>

<p><code>warbleR_options</code> sets global parameters for warbleR functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>warbleR_options(reset = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="warbleR_options_+3A_reset">reset</code></td>
<td>
<p>Logical. If <code>TRUE</code> then all global parameters are removed. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="warbleR_options_+3A_...">...</code></td>
<td>
<p>Arguments in 'parameter = value' form, or a list of tagged values. The tags (i.e. parameters)
must come from the list of parameters described below.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function aims to simplify the use of parameters that apply to many warbleR functions (i.e. global parameters)
by setting a default value that will be used to any function in downstream
analyses. Tags that are set with warbleR_options will be used iby the functions
that share those arguments. However, if an argument is set within a function call
it will overwrite the values set by warbleR_options. Hence, the functions remain
'flexible' as their parameters can also be modified 'on the fly'. The following tags are available:
</p>

<ul>
<li> <p><code>bp</code>: Numeric vector of length 2 giving the lower and upper limits of a
frequency bandpass filter (in kHz).
</p>
</li>
<li> <p><code>collevels</code>:  A numeric vector of length 3. Specifies levels to partition the
amplitude range of the spectrogram (in dB) as in <code><a href="seewave.html#topic+spectro">spectro</a></code>. The more levels the higher the
resolution of the spectrogram. The lower the first value the darker the spectrograms.
</p>
</li>
<li> <p><code>flim</code>: A numeric vector of length 2 for the frequency limit in kHz of
the spectrogram, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>.
</p>
</li>
<li> <p><code>it</code>: A character vector of length 1 giving the image type to be used. Currently only
&quot;tiff&quot; and &quot;jpeg&quot; are admitted.
</p>
</li>
<li> <p><code>osci</code>: Logical argument to add an oscillogram underneath spectrogram, as
in <code><a href="seewave.html#topic+spectro">spectro</a></code>.
</p>
</li>
<li> <p><code>pal</code>: A color palette function to be used to assign colors in the
plot, as in <code><a href="seewave.html#topic+spectro">spectro</a></code>.
</p>
</li>
<li> <p><code>parallel</code>: Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used in iterative functions.
</p>
</li>
<li> <p><code>pb</code>: Logical argument to control whether progress bar is used.
</p>
</li>
<li> <p><code>res</code>: Numeric argument of length 1. Controls image resolution in all image creating functions.
</p>
</li>
<li> <p><code>wav.path</code>: Character string containing the directory path where the
sound files are located. Used as 'path' in all functions in which sound files are read.
</p>
</li>
<li> <p><code>wl</code>: A numeric vector of length 1 specifying the window length for creating spectrogram (either for plotting or for measuring spectrogram parameters).
</p>
</li>
<li> <p><code>wn</code>: Character vector of length 1 specifying the window name for creating spectrogram (either for plotting or for measuring spectrogram parameters). See function <code><a href="seewave.html#topic+ftwindow">ftwindow</a></code> for options.
</p>
</li></ul>



<h3>Value</h3>

<p>When parameters are set by warbleR_options, their former values are
returned in an invisible named list. Such a list can be passed as an argument to
pboptions to restore the parameter values. If the function is called with no arguments the current option values are printed.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # load data and save in temporary working directory
  data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
  writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
  writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

  # sig2noise with progress bar (by default is TRUE)
  a &lt;- sig2noise(X = lbh_selec_table, mar = 0.1, path = tempdir())

  # set progress bar to FALSE with warbleR_options
  warbleR_options(pb = FALSE, path = tempdir())

  # sig2noise without progress bar
  a &lt;- sig2noise(X = lbh_selec_table, mar = 0.1)

  # sig2noise with progress bar by setting it within the function call (overwritting options)
  a &lt;- sig2noise(X = lbh_selec_table, pb = TRUE, mar = 0.1)

  # sig2noise without progress bar using warbleR_options setting again
  a &lt;- sig2noise(X = lbh_selec_table, mar = 0.1)
}
</code></pre>

<hr>
<h2 id='warbleR-internals'>
<code>warbleR</code> Internal Functions
</h2><span id='topic+filled.contour.color.INTFUN'></span><span id='topic+filled.contour.INTFUN'></span><span id='topic+frd.INTFUN'></span><span id='topic+frd.plot.INTFUN'></span><span id='topic+spectro.rectw.INTFUN.2'></span><span id='topic+spectro.rectw.INTFUN'></span><span id='topic+boxw.INTFUN'></span>

<h3>Description</h3>

<p>These functions are used by other functions in the warbleR package, but are not intended to be called by users. 
</p>
<p>These 2 functions are modified versions of <code><a href="seewave.html#topic+spectro">spectro</a></code>.
The functions are:
<code>spectro.rectw.INTFUN.2</code>
<code>spectro.rectw.INTFUN</code>
</p>
<p>These 2 functions are modified versions of seewave's <code>filled.contour.modif2()</code>.
<code><a href="seewave.html#topic+spectro">spectro</a></code>.
The functions are:
<code>filled.contour.color.INTFUN</code>
<code>filled.contour.INTFUN</code>
</p>
<p><code>boxw.INTFUN</code> to create boxes around spectrograms in <code><a href="#topic+catalog">catalog</a></code>
</p>
<p><code>rectw</code> is a modified version of <code><a href="graphics.html#topic+rect">rect</a></code> to add density (cross-hatching lines)
</p>
<p>These 2 functions are used for frequency range detection:
<code>frd.INTFUN</code>
<code>frd.plot.INTFUN</code>
</p>


<h3>Note</h3>

<p>These functions are not to be called by the user.
</p>

<hr>
<h2 id='wav_2_flac'>Convert .wav files to .flac</h2><span id='topic+wav_2_flac'></span>

<h3>Description</h3>

<p><code>wav_2_flac</code> converts several .wav files to .flac compressed lossless format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wav_2_flac(files = NULL, path = NULL, overwrite = FALSE,
pb = TRUE, parallel = 1, reverse = FALSE, compression = 5, flac.path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wav_2_flac_+3A_files">files</code></td>
<td>
<p>character vector with the names of files to be converted. If <code>NULL</code> all files in the working directory (or 'path' if supplied) are converted.</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the .wav files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical. Control whether a .flac sound file that is already in the working directory should be
overwritten.</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control if progress bar is shown. Default is <code>TRUE</code>. It can also be
set globally using the 'pb' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing). It can also be
set globally using the 'parallel' option (see <code><a href="#topic+warbleR_options">warbleR_options</a></code>).</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_reverse">reverse</code></td>
<td>
<p>Logical argument to control if .wav files are converted into .flac files (default, <code>reverse = FALSE</code>) or .flac files are converted into .wav files <code>reverse = TRUE</code>.</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_compression">compression</code></td>
<td>
<p>Numeric string on length 1 indicating the level of compression for .flac files. Must a number between 0 (lowest) to 8 (highest compression). Default is 5.</p>
</td></tr>
<tr><td><code id="wav_2_flac_+3A_flac.path">flac.path</code></td>
<td>
<p>Path to the flac program, mostly needed for windows OS.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will convert all .wav files in working directory or 'path' supplied to .flac format (or the opposite if <code>reverse = TRUE</code>). For reading 'flac' files on windows the path to the .exe is required. This can be set using the 'flac.path' argument (or globally using the same argument in <code><a href="#topic+warbleR_options">warbleR_options</a></code>). Note that reading 'flac' files requires creating a temporary copy in 'wav' format, which can be particularly slow for long files.
</p>
<p>convert all .wav files in working directory to .flac compressed lossless format. It's just a silly wrapper over (<code><a href="seewave.html#topic+wav2flac">wav2flac</a></code>) to simplify converting several files at once. The function works recursively, converting files within all subfolders.
</p>


<h3>Value</h3>

<p>.flac files saved in the working directory with same name as original wav files.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# create some .wav files
data(list = c("Phae.long1", "Phae.long2", "Phae.long3", "Phae.long4"))
writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))
writeWave(Phae.long3, file.path(tempdir(), "Phae.long3.wav"))
writeWave(Phae.long4, file.path(tempdir(), "Phae.long4.wav"))

# Convert all files to .flac format
wav_2_flac(path = tempdir())

# check this folder!!
open_wd(tempdir())

## End(Not run)

</code></pre>

<hr>
<h2 id='wav_dur'>alternative name for <code><a href="#topic+duration_sound_files">duration_sound_files</a></code></h2><span id='topic+wav_dur'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+duration_sound_files">duration_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wav_dur(
  files = NULL,
  path = NULL,
  skip.error = FALSE,
  file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$"
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+duration_sound_files">duration_sound_files</a></code> for documentation. <code><a href="#topic+wavdur">wavdur</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='wav_info'>alternative name for <code><a href="#topic+info_sound_files">info_sound_files</a></code></h2><span id='topic+wav_info'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+info_sound_files">info_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wav_info(
  path = NULL,
  files = NULL,
  parallel = 1,
  pb = TRUE,
  skip.error = FALSE,
  file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$"
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+info_sound_files">info_sound_files</a></code> for documentation. <code><a href="#topic+wav_info">wav_info</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='wavdur'>alternative name for <code><a href="#topic+duration_sound_files">duration_sound_files</a></code></h2><span id='topic+wavdur'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+duration_sound_files">duration_sound_files</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wavdur(
  files = NULL,
  path = NULL,
  skip.error = FALSE,
  file.format = "\\.wav$|\\.wac$|\\.mp3$|\\.flac$"
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+duration_sound_files">duration_sound_files</a></code> for documentation. <code><a href="#topic+wavdur">wavdur</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='wpd_features'>Measure wavelet packet decomposition features (EXPERIMENTAL)</h2><span id='topic+wpd_features'></span>

<h3>Description</h3>

<p><code>wpd_features</code> Measure wavelet packet decomposition features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wpd_features(X, normalize = TRUE, threshold1 = 6,
threshold2 = 0.5, path = NULL, pb = TRUE, parallel = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wpd_features_+3A_x">X</code></td>
<td>
<p>object of class 'selection_table', 'extended_selection_table' or data frame with the following columns: 1) &quot;sound.files&quot;: name of the sound
files, 2) &quot;sel&quot;: number of the selections, 3) &quot;start&quot;: start time of selections, 4) &quot;end&quot;:
end time of selections. The output of <code><a href="#topic+auto_detec">auto_detec</a></code> can
also be used as the input data frame.</p>
</td></tr>
<tr><td><code id="wpd_features_+3A_normalize">normalize</code></td>
<td>
<p>Logical to determine if features are normalized by signal duration.</p>
</td></tr>
<tr><td><code id="wpd_features_+3A_threshold1">threshold1</code></td>
<td>
<p>Threshold (%) for wavelet coefficient detection. Equivalent to denominator of equation 6 in Selin et al (2007). Must be a value between 0 and 1.</p>
</td></tr>
<tr><td><code id="wpd_features_+3A_threshold2">threshold2</code></td>
<td>
<p>Threshold for width detection. Equivalent to threshold 2 (th2) in equation 7 in Selin et al (2007).</p>
</td></tr>
<tr><td><code id="wpd_features_+3A_path">path</code></td>
<td>
<p>Character string containing the directory path where the sound files are located.
If <code>NULL</code> (default) then the current working directory is used.</p>
</td></tr>
<tr><td><code id="wpd_features_+3A_pb">pb</code></td>
<td>
<p>Logical argument to control progress bar and messages. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="wpd_features_+3A_parallel">parallel</code></td>
<td>
<p>Numeric. Controls whether parallel computing is applied.
It specifies the number of cores to be used. Default is 1 (i.e. no parallel computing).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Measures wavelet packet decomposition features. STILL UNDER DEVELOPMENT. USE IT UNDER YOUR OWN RISK.
</p>


<h3>Value</h3>

<p>A data frame with rows for each of the selections in 'X' in addition to four wavelet packet decomposition features: max.energy, position, spread and width.
</p>


<h3>Author(s)</h3>

<p>Marcelo Araya-Salas (<a href="mailto:marcelo.araya@ucr.ac.cr">marcelo.araya@ucr.ac.cr</a>)
</p>


<h3>References</h3>


<p>Araya-Salas, M., &amp; Smith-Vidaurre, G. (2017). warbleR: An R package to streamline analysis of animal acoustic signals. Methods in Ecology and Evolution, 8(2), 184-191.
</p>
<p>Selin A., J. Turunen, and J. T. Tanttu, 2007. Wavelets in recognition of bird sounds. EURASIP Journal on Advances in Signal Processing.

</p>


<h3>See Also</h3>

<p><code><a href="#topic+mfcc_stats">mfcc_stats</a></code>, <code><a href="#topic+mfcc_stats">mfcc_stats</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  data(list = c("Phae.long1", "Phae.long2", "lbh_selec_table"))
  writeWave(Phae.long1, file.path(tempdir(), "Phae.long1.wav"))
  writeWave(Phae.long2, file.path(tempdir(), "Phae.long2.wav"))

  # not normalize
  wpd_features(lbh_selec_table[1:5, ], threshold2 = 0.3, nor = FALSE, path = tempdir())
}

</code></pre>

<hr>
<h2 id='xcmaps'>alternative name for <code><a href="#topic+map_xc">map_xc</a></code></h2><span id='topic+xcmaps'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+map_xc">map_xc</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xcmaps(
  X,
  img = TRUE,
  it = "jpeg",
  res = 100,
  labels = FALSE,
  path = NULL,
  leaflet.map = FALSE,
  leaflet.cluster = FALSE
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+map_xc">map_xc</a></code> for documentation. <code><a href="#topic+xcmaps">xcmaps</a></code> will be deprecated in future versions.
</p>

<hr>
<h2 id='xcorr'>alternative name for <code><a href="#topic+cross_correlation">cross_correlation</a></code></h2><span id='topic+xcorr'></span>

<h3>Description</h3>

<p>alternative name for <code><a href="#topic+cross_correlation">cross_correlation</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xcorr(
  X = NULL,
  wl = 512,
  bp = "pairwise.freq.range",
  ovlp = 70,
  dens = NULL,
  wn = "hanning",
  cor.method = "pearson",
  parallel = 1,
  path = NULL,
  pb = TRUE,
  na.rm = FALSE,
  cor.mat = NULL,
  output = "cor.mat",
  templates = NULL,
  surveys = NULL,
  compare.matrix = NULL,
  type = "fourier",
  nbands = 40,
  method = 1
)
</code></pre>


<h3>Details</h3>

<p>see <code><a href="#topic+cross_correlation">cross_correlation</a></code> for documentation. <code><a href="#topic+xcorr">xcorr</a></code> will be deprecated in future versions.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
