<!DOCTYPE html><html lang="en"><head><title>Help for package webchem</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {webchem}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#as.cas'><p>Format numbers as CAS numbers</p></a></li>
<li><a href='#bcpc_query'><p>Query https://pesticidecompendium.bcpc.org</p></a></li>
<li><a href='#chebi_comp_entity'><p>Retrieve Complete Entity from ChEBI</p></a></li>
<li><a href='#chembl_atc_classes'><p>Retrieve all ATC classes</p></a></li>
<li><a href='#chembl_query'><p>Query ChEMBL using ChEMBL IDs</p></a></li>
<li><a href='#chembl_resources'><p>List ChEMBL Resources</p></a></li>
<li><a href='#cir_img'><p>Query Chemical Identifier Resolver Images</p></a></li>
<li><a href='#cir_query'><p>Query Chemical Identifier Resolver</p></a></li>
<li><a href='#cs_check_key'><p>Retrieve ChemSpider API key</p></a></li>
<li><a href='#cs_compinfo'><p>Retrieve record details by ChemSpider ID</p></a></li>
<li><a href='#cs_control'><p>Control ChemSpider API requests</p></a></li>
<li><a href='#cs_convert'><p>Convert identifiers using ChemSpider</p></a></li>
<li><a href='#cs_datasources'><p>Retrieve ChemSpider data sources</p></a></li>
<li><a href='#cs_extcompinfo'><p>Get extended record details by ChemSpider ID</p></a></li>
<li><a href='#cs_img'><p>Download images from ChemSpider</p></a></li>
<li><a href='#cts_compinfo'><p>Get record details from Chemical Translation Service (CTS)</p></a></li>
<li><a href='#cts_convert'><p>Convert Ids using Chemical Translation Service (CTS)</p></a></li>
<li><a href='#cts_from'><p>Return a list of all possible ids</p></a></li>
<li><a href='#cts_to'><p>Return a list of all possible ids</p></a></li>
<li><a href='#etox_basic'><p>Get basic information from a ETOX ID</p></a></li>
<li><a href='#etox_targets'><p>Get Quality Targets from a ETOX ID</p></a></li>
<li><a href='#etox_tests'><p>Get Tests from a ETOX ID</p></a></li>
<li><a href='#extractors'><p>Extract parts from webchem objects</p></a></li>
<li><a href='#find_db'><p>Check data source coverage of compounds</p></a></li>
<li><a href='#fn_percept'><p>Retrieve flavor percepts from www.flavornet.org</p></a></li>
<li><a href='#get_chebiid'><p>Retrieve Lite Entity (identifiers) from ChEBI</p></a></li>
<li><a href='#get_cid'><p>Retrieve Pubchem Compound ID (CID)</p></a></li>
<li><a href='#get_csid'><p>ChemSpider ID from compound name, formula, SMILES, InChI or InChIKey</p></a></li>
<li><a href='#get_etoxid'><p>Get ETOX ID</p></a></li>
<li><a href='#get_wdid'><p>Get Wikidata Item ID</p></a></li>
<li><a href='#is.cas'><p>Check if input is a valid CAS</p></a></li>
<li><a href='#is.inchikey'><p>Check if input is a valid inchikey</p></a></li>
<li><a href='#is.inchikey_cs'><p>Check if input is a valid inchikey using ChemSpider API</p></a></li>
<li><a href='#is.inchikey_format'><p>Check if input is a valid inchikey using format</p></a></li>
<li><a href='#is.smiles'><p>Check if input is a SMILES string</p></a></li>
<li><a href='#jagst'><p>Organic plant protection products in the river Jagst / Germany in 2013</p></a></li>
<li><a href='#lc50'><p>Acute toxicity data from U.S. EPA ECOTOX</p></a></li>
<li><a href='#nist_ri'><p>Retrieve retention indices from NIST</p></a></li>
<li><a href='#opsin_query'><p>OPSIN web interface</p></a></li>
<li><a href='#parse_mol'><p>Parse Molfile (as returned by ChemSpider) into a R-object.</p></a></li>
<li><a href='#pc_prop'><p>Retrieve compound properties from a pubchem CID</p></a></li>
<li><a href='#pc_sect'><p>Retrieve data from PubChem content pages</p></a></li>
<li><a href='#pc_synonyms'><p>Search synonyms in pubchem</p></a></li>
<li><a href='#ping_service'><p>Ping an API used in webchem to see if it's working.</p></a></li>
<li><a href='#srs_query'><p>Get record details from U.S. EPA Substance Registry Servives (SRS)</p></a></li>
<li><a href='#wd_ident'><p>Retrieve identifiers from Wikidata</p></a></li>
<li><a href='#webchem'><p>webchem: An R package to retrieve chemical information from the web.</p></a></li>
<li><a href='#webchem-defunct'><p>Defunct function(s) in the webchem package</p></a></li>
<li><a href='#webchem-deprecated'><p>Deprecated function(s) in the webchem package</p></a></li>
<li><a href='#with_cts'><p>Auto-translate identifiers and search databases</p></a></li>
<li><a href='#write_mol'><p>Export a Chemical Structure in .mol Format.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Chemical Information from the Web</td>
</tr>
<tr>
<td>Description:</td>
<td>Chemical information from around the web. This package interacts 
    with a suite of web services for chemical information. Sources include: Alan
    Wood's Compendium of Pesticide Common Names, Chemical Identifier Resolver,
    ChEBI, Chemical Translation Service, ChemSpider, ETOX, Flavornet,
    NIST Chemistry WebBook, OPSIN, PubChem, SRS, Wikidata.</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://docs.ropensci.org/webchem/">https://docs.ropensci.org/webchem/</a>,
<a href="https://github.com/ropensci/webchem">https://github.com/ropensci/webchem</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/webchem/issues">https://github.com/ropensci/webchem/issues</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tamás Stirling &lt;stirling.tamas@gmail.com&gt;</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>xml2, httr, rvest, jsonlite, stringr, methods, dplyr, purrr,
data.tree, tibble, base64enc, rlang, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, rcdk, covr, robotstxt, knitr, rmarkdown,
plot.matrix, usethis, vcr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Config/testthat/parallel:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-09 13:40:10 UTC; tamas</td>
</tr>
<tr>
<td>Author:</td>
<td>Eduard Szöcs [aut],
  Robert Allaway [ctb],
  Daniel Muench [ctb],
  Johannes Ranke [ctb],
  Andreas Scharmüller [ctb],
  Eric R Scott [ctb],
  Jan Stanstrup [ctb],
  João Vitor F Cavalcante [ctb],
  Gordon Getzinger [ctb],
  Ethan Bass [ctb],
  Tamás Stirling [ctb, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-06-09 14:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='as.cas'>Format numbers as CAS numbers</h2><span id='topic+as.cas'></span>

<h3>Description</h3>

<p>This function attempts to format numeric (or character) vectors
as character vectors of CAS numbers.  If they cannot be converted to CAS
format or don't pass <code><a href="#topic+is.cas">is.cas</a></code>, <code>NA</code> is returned
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.cas(x, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.cas_+3A_x">x</code></td>
<td>
<p>numeric vector, or character vector of CAS numbers missing the
hyphens</p>
</td></tr>
<tr><td><code id="as.cas_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector of valid CAS numbers
</p>


<h3>See Also</h3>

<p><code><a href="#topic+is.cas">is.cas</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x = c(58082, 123456, "hexenol")
as.cas(x)

</code></pre>

<hr>
<h2 id='bcpc_query'>Query https://pesticidecompendium.bcpc.org</h2><span id='topic+bcpc_query'></span>

<h3>Description</h3>

<p>Query the BCPC Compendium of Pesticide Common Names
<a href="https://pesticidecompendium.bcpc.org">https://pesticidecompendium.bcpc.org</a>
formerly known as Alan Woods Compendium of Pesticide Common Names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcpc_query(
  query,
  from = c("name", "cas"),
  verbose = getOption("verbose"),
  type,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bcpc_query_+3A_query">query</code></td>
<td>
<p>character; search string</p>
</td></tr>
<tr><td><code id="bcpc_query_+3A_from">from</code></td>
<td>
<p>character; type of input ('cas' or 'name')</p>
</td></tr>
<tr><td><code id="bcpc_query_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
<tr><td><code id="bcpc_query_+3A_type">type</code></td>
<td>
<p>deprecated</p>
</td></tr>
<tr><td><code id="bcpc_query_+3A_...">...</code></td>
<td>
<p>additional arguments to internal utility functions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of eight entries: common-name, status, preferred IUPAC Name,
IUPAC Name, cas, formula, activity, subactivity, inchikey, inchi and source
url.
</p>


<h3>Note</h3>

<p>for from = 'cas' only the first matched link is returned.
Please respect Copyright, Terms and Conditions
<a href="https://pesticidecompendium.bcpc.org/legal.html">https://pesticidecompendium.bcpc.org/legal.html</a>!
</p>


<h3>References</h3>

<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bcpc_query('Fluazinam', from = 'name')
out &lt;- bcpc_query(c('Fluazinam', 'Diclofop'), from = 'name')
out
# extract subactivity from object
sapply(out, function(y) y$subactivity[1])

# use CAS-numbers
bcpc_query("79622-59-6", from = 'cas')

## End(Not run)
</code></pre>

<hr>
<h2 id='chebi_comp_entity'>Retrieve Complete Entity from ChEBI</h2><span id='topic+chebi_comp_entity'></span>

<h3>Description</h3>

<p>Returns a list of Complete ChEBI entities.
ChEBI data are parsed as data.frames (&quot;properties&quot;, &quot;chebiid_snd&quot;,
&quot;synonyms&quot;, &quot;iupacnames&quot;, &quot;formulae&quot;, &quot;regnumbers&quot;, &quot;citations&quot;, &quot;dblinks&quot;,
&quot;parents&quot;, &quot;children&quot;, &quot;comments&quot;, &quot;origins&quot;) or
as a list (&quot;chem_structure&quot;) in the list.
The SOAP protocol is used <a href="https://www.ebi.ac.uk/chebi/webServices.do">https://www.ebi.ac.uk/chebi/webServices.do</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chebi_comp_entity(chebiid, verbose = getOption("verbose"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chebi_comp_entity_+3A_chebiid">chebiid</code></td>
<td>
<p>character; search term (i.e. chebiid).</p>
</td></tr>
<tr><td><code id="chebi_comp_entity_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="chebi_comp_entity_+3A_...">...</code></td>
<td>
<p>optional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of data.frames or lists containing a complete ChEBI
entity
</p>


<h3>References</h3>

<p>Hastings J, Owen G, Dekker A, Ennis M, Kale N, Muthukrishnan V,
Turner S, Swainston N, Mendes P, Steinbeck C. (2016). ChEBI in 2016:
Improved services and an expanding collection of metabolites. Nucleic Acids
Res.
</p>
<p>Hastings, J., de Matos, P., Dekker, A., Ennis, M., Harsha, B., Kale, N.,
Muthukrishnan, V., Owen, G., Turner, S., Williams, M., and Steinbeck, C.
(2013) The ChEBI reference database and ontology for biologically relevant
chemistry: enhancements for 2013. Nucleic Acids Res.
</p>
<p>de Matos, P., Alcantara, R., Dekker, A., Ennis, M., Hastings, J., Haug, K.,
Spiteri, I., Turner, S., and Steinbeck, C. (2010) Chemical entities of
biological interest: an update. Nucleic Acids Res. Degtyarenko, K.,
Hastings, J., de Matos, P., and Ennis, M. (2009). ChEBI: an open
bioinformatics and cheminformatics resource. Current protocols in
bioinformatics / editoral board, Andreas D. Baxevanis et al., Chapter 14.
</p>
<p>Degtyarenko, K., de Matos, P., Ennis, M., Hastings, J., Zbinden, M.,
McNaught, A., Alcántara, R., Darsow, M., Guedj, M. and Ashburner, M. (2008)
ChEBI: a database and ontology for chemical entities of biological
interest. Nucleic Acids Res. 36, D344–D350.
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
chebi_comp_entity('CHEBI:27744')

# multiple inputs
comp &lt;- c('CHEBI:27744', 'CHEBI:27744')
chebi_comp_entity(comp)


## End(Not run)
</code></pre>

<hr>
<h2 id='chembl_atc_classes'>Retrieve all ATC classes</h2><span id='topic+chembl_atc_classes'></span>

<h3>Description</h3>

<p>Retrieve all available classes within the Anatomical Therapeutic Chemical
(ATC) classification system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chembl_atc_classes(verbose = getOption("verbose"), test_service_down = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chembl_atc_classes_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="chembl_atc_classes_+3A_test_service_down">test_service_down</code></td>
<td>
<p>logical; this argument is only used for testing.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Gaulton, A., Bellis, L. J., Bento, A. P., Chambers, J.,
Davies, M., Hersey, A., ... &amp; Overington, J. P. (2012). ChEMBL: a large-scale
bioactivity database for drug discovery. Nucleic acids research, 40(D1),
D1100-D1107.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Might fail if API is not available

atc &lt;- atc_classes()

## End(Not run)
</code></pre>

<hr>
<h2 id='chembl_query'>Query ChEMBL using ChEMBL IDs</h2><span id='topic+chembl_query'></span>

<h3>Description</h3>

<p>Retrieve ChEMBL data using a vector of ChEMBL IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chembl_query(
  query,
  resource = "molecule",
  cache_file = NULL,
  verbose = getOption("verbose"),
  test_service_down = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chembl_query_+3A_query">query</code></td>
<td>
<p>character; a vector of ChEMBL IDs.</p>
</td></tr>
<tr><td><code id="chembl_query_+3A_resource">resource</code></td>
<td>
<p>character; the ChEMBL resource to query. Use
[chembl_resources()] to see all available resources.</p>
</td></tr>
<tr><td><code id="chembl_query_+3A_cache_file">cache_file</code></td>
<td>
<p>character; the name of the cache file without the file
extension. If <code>NULL</code>, results are not cached.</p>
</td></tr>
<tr><td><code id="chembl_query_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="chembl_query_+3A_test_service_down">test_service_down</code></td>
<td>
<p>logical; this argument is only used for testing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each entry in ChEMBL has a unique ID. Data in ChEMBL is organized in
databases called resources. An entry may or may not have a record in a
particular resource. An entry may have a record in more than one resource,
e.g. a compound may be present in both the &quot;molecule&quot; and the &quot;drug&quot;
resource. This function queries a vector of ChEMBL IDs from a specific ChEMBL
resource.
</p>
<p>If you are unsure which ChEMBL resource contains your ChEMBL ID,
use this function with the <code>"chembl_id_lookup"</code> resource to find the
appropriate resource for a ChEMBL ID. Note that <code>"chembl_id_lookup"</code> is
not a separate function but a resource used by <code>chembl_query</code>.
</p>
<p>If <code>cache_file</code> is not <code>NULL</code> the function creates a
cache directory in the working directory and a cache file in the cache
directory. This file is used in subsequent calls of the function. The
function first tries to retrieve query results from the cache file and only
accesses the webservice if the ChEMBL ID cannot be found in the cache file.
The cache file is extended as new ChEMBL ID-s are queried during the session.
</p>


<h3>Value</h3>

<p>The function returns a list of lists, where each element of the list
contains a list of respective query results. Results are simplified, if
possible.
</p>


<h3>Note</h3>

<p>Links to the webservice documentation:
</p>

<ul>
<li> <p><a href="https://chembl.gitbook.io/chembl-interface-documentation">https://chembl.gitbook.io/chembl-interface-documentation</a>,
</p>
</li>
<li> <p><a href="https://www.ebi.ac.uk/chembl/api/data/docs">https://www.ebi.ac.uk/chembl/api/data/docs</a>
</p>
</li></ul>



<h3>References</h3>

<p>Gaulton, A., Bellis, L. J., Bento, A. P., Chambers, J.,
Davies, M., Hersey, A., ... &amp; Overington, J. P. (2012). ChEMBL: a large-scale
bioactivity database for drug discovery. Nucleic acids research, 40(D1),
D1100-D1107.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Might fail if API is not available

# Search molecules
chembl_query("CHEMBL1082", resource = "molecule")
chembl_query(c("CHEMBL25", "CHEMBL1082"), resource = "molecule")

# Look up ChEMBL IDs in ChEMBL "resources", returns one resource per query.
chembl_query("CHEMBL771355", "chembl_id_lookup")

# Search assays
chembl_query("CHEMBL771355", resource = "assay")

## End(Not run)
</code></pre>

<hr>
<h2 id='chembl_resources'>List ChEMBL Resources</h2><span id='topic+chembl_resources'></span>

<h3>Description</h3>

<p>Data in ChEMBL is organized in databases called resources. This function
lists available ChEMBL resources.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chembl_resources()
</code></pre>


<h3>Note</h3>

<p>The list was compiled manually using the following url:
<a href="https://chembl.gitbook.io/chembl-interface-documentation/web-services/chembl-data-web-services">https://chembl.gitbook.io/chembl-interface-documentation/web-services/chembl-data-web-services</a>
</p>


<h3>References</h3>

<p>Gaulton, A., Bellis, L. J., Bento, A. P., Chambers, J.,
Davies, M., Hersey, A., ... &amp; Overington, J. P. (2012). ChEMBL: a large-scale
bioactivity database for drug discovery. Nucleic acids research, 40(D1),
D1100-D1107.
</p>

<hr>
<h2 id='cir_img'>Query Chemical Identifier Resolver Images</h2><span id='topic+cir_img'></span>

<h3>Description</h3>

<p>A interface to the Chemical Identifier Resolver (CIR).
(<a href="https://cactus.nci.nih.gov/chemical/structure_documentation">https://cactus.nci.nih.gov/chemical/structure_documentation</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cir_img(
  query,
  dir,
  format = c("png", "gif"),
  width = 500,
  height = 500,
  linewidth = 2,
  symbolfontsize = 16,
  bgcolor = NULL,
  antialiasing = TRUE,
  atomcolor = NULL,
  bondcolor = NULL,
  csymbol = c("special", "all"),
  hsymbol = c("special", "all"),
  hcolor = NULL,
  header = NULL,
  footer = NULL,
  frame = NULL,
  verbose = getOption("verbose"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cir_img_+3A_query">query</code></td>
<td>
<p>character; Search term. Can be any common chemical identifier
(e.g. CAS, INCHI(KEY), SMILES etc.)</p>
</td></tr>
<tr><td><code id="cir_img_+3A_dir">dir</code></td>
<td>
<p>character; Directory to save the image.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_format">format</code></td>
<td>
<p>character; Output format of the image. Can be one of &quot;png&quot;,
&quot;gif&quot;.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_width">width</code></td>
<td>
<p>integer; Width of the image.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_height">height</code></td>
<td>
<p>integer; Height of the image.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_linewidth">linewidth</code></td>
<td>
<p>integer; Width of lines.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_symbolfontsize">symbolfontsize</code></td>
<td>
<p>integer; Fontsize of atoms in the image.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_bgcolor">bgcolor</code></td>
<td>
<p>character; E.g. transparent, white, %23AADDEE</p>
</td></tr>
<tr><td><code id="cir_img_+3A_antialiasing">antialiasing</code></td>
<td>
<p>logical; Should antialiasing be used?</p>
</td></tr>
<tr><td><code id="cir_img_+3A_atomcolor">atomcolor</code></td>
<td>
<p>character; Color of the atoms in the image.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_bondcolor">bondcolor</code></td>
<td>
<p>character; Color of the atom bond lines.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_csymbol">csymbol</code></td>
<td>
<p>character; Can be one of &quot;special&quot; (default - i.e. only
hydrogen atoms in functional groups or defining stereochemistry) or &quot;all&quot;.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_hsymbol">hsymbol</code></td>
<td>
<p>character; Can be one of &quot;special&quot; (default - i.e. none are
shown) or &quot;all&quot; (all are printed).</p>
</td></tr>
<tr><td><code id="cir_img_+3A_hcolor">hcolor</code></td>
<td>
<p>character; Color of the hydrogen atoms.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_header">header</code></td>
<td>
<p>character; Should a header text be added to the image? Can be
any string.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_footer">footer</code></td>
<td>
<p>character; Should a footer text be added to the image? Can be
any string.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_frame">frame</code></td>
<td>
<p>integer; Should a frame be plotted? Can be on of NULL (default)
or 1.</p>
</td></tr>
<tr><td><code id="cir_img_+3A_verbose">verbose</code></td>
<td>
<p>logical; Should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cir_img_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>CIR can resolve can be of the following <code>identifier</code>: Chemical Names,
IUPAC names,
CAS Numbers, SMILES strings, IUPAC InChI/InChIKeys, NCI/CADD Identifiers,
CACTVS HASHISY, NSC number, PubChem SID, ZINC Code, ChemSpider ID,
ChemNavigator SID, eMolecule VID.
</p>
<p>For an image with transparent background use ‘transparent’ as color name and
switch off antialiasing (i.e. antialiasing = 0).
</p>


<h3>Value</h3>

<p>image written to disk
</p>


<h3>Note</h3>

<p>You can only make 1 request per second (this is a hard-coded feature).
</p>


<h3>References</h3>

<p><code>cir</code> relies on the great CIR web service created by the CADD
Group at NCI/NIH! <br />
<a href="https://cactus.nci.nih.gov/chemical/structure_documentation">https://cactus.nci.nih.gov/chemical/structure_documentation</a>, <br />
<a href="https://cactus.nci.nih.gov/blog/?cat=10">https://cactus.nci.nih.gov/blog/?cat=10</a>, <br />
<a href="https://cactus.nci.nih.gov/blog/?p=1386">https://cactus.nci.nih.gov/blog/?p=1386</a>, <br />
<a href="https://cactus.nci.nih.gov/blog/?p=1456">https://cactus.nci.nih.gov/blog/?p=1456</a>, <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
cir_img("CCO", dir = tempdir()) # SMILES

# multiple query strings and different formats
query = c("Glyphosate", "Isoproturon", "BSYNRYMUTXBXSQ-UHFFFAOYSA-N")
cir_img(query, dir = tempdir(), bgcolor = "transparent", antialising = 0)

# all parameters
query  = "Triclosan"
cir_img(query,
        dir = tempdir(),
        format = "png",
        width = 600,
        height = 600,
        linewidth = 5,
        symbolfontsize = 30,
        bgcolor = "red",
        antialiasing = FALSE,
        atomcolor = "green",
        bondcolor = "yellow",
        csymbol = "all",
        hsymbol = "all",
        hcolor = "purple",
        header = "My funky chemical structure..",
        footer = "..is just so awesome!",
        frame = 1,
        verbose = getOption("verbose"))

## End(Not run)
</code></pre>

<hr>
<h2 id='cir_query'>Query Chemical Identifier Resolver</h2><span id='topic+cir_query'></span>

<h3>Description</h3>

<p>A interface to the Chemical Identifier Resolver (CIR).
(<a href="https://cactus.nci.nih.gov/chemical/structure_documentation">https://cactus.nci.nih.gov/chemical/structure_documentation</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cir_query(
  identifier,
  representation = "smiles",
  resolver = NULL,
  match = c("all", "first", "ask", "na"),
  verbose = getOption("verbose"),
  choices = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cir_query_+3A_identifier">identifier</code></td>
<td>
<p>character; chemical identifier.</p>
</td></tr>
<tr><td><code id="cir_query_+3A_representation">representation</code></td>
<td>
<p>character; what representation of the identifier should
be returned. See details for possible representations.</p>
</td></tr>
<tr><td><code id="cir_query_+3A_resolver">resolver</code></td>
<td>
<p>character; what resolver should be used? If NULL (default)
the identifier type is detected and the different resolvers are used in turn.
See details for possible resolvers.</p>
</td></tr>
<tr><td><code id="cir_query_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handled? <code>"all"</code>
returns all matches, <code>"first"</code> returns only the first result,
<code>"ask"</code> enters an interactive mode and the user is asked for input,
<code>"na"</code> returns <code>NA</code> if multiple hits are found.</p>
</td></tr>
<tr><td><code id="cir_query_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cir_query_+3A_choices">choices</code></td>
<td>
<p>deprecated.  Use the <code>match</code> argument instead.</p>
</td></tr>
<tr><td><code id="cir_query_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>CIR can resolve can be of the following <code>identifier</code>: Chemical Names,
IUPAC names,
CAS Numbers, SMILES strings, IUPAC InChI/InChIKeys, NCI/CADD Identifiers,
CACTVS HASHISY, NSC number, PubChem SID, ZINC Code, ChemSpider ID,
ChemNavigator SID, eMolecule VID.
</p>
<p><code>cir_query()</code> can handle only a part of all possible conversions of CIR.
Possible <code>representations</code> are:
</p>

<ul>
<li> <p><code>'smiles'</code>(SMILES strings),
</p>
</li>
<li> <p><code>'names'</code> (Names),
</p>
</li>
<li> <p><code>'cas'</code> (CAS numbers),
</p>
</li>
<li> <p><code>'stdinchikey'</code> (Standard InChIKey),
</p>
</li>
<li> <p><code>'stdinchi'</code> (Standard InChI),
</p>
</li>
<li> <p><code>'ficts'</code> (FICTS Identifier),
</p>
</li>
<li> <p><code>'ficus'</code> (FICuS Indetifier),
</p>
</li>
<li> <p><code>'uuuuu'</code> (uuuuu Identifier),
</p>
</li>
<li> <p><code>'mw'</code> (Molecular weight),
</p>
</li>
<li> <p><code>'monoisotopic_mass'</code> (Monoisotopic Mass),
</p>
</li>
<li> <p><code>'formula'</code> (Chemical Formula),
</p>
</li>
<li> <p><code>'chemspider_id'</code> (ChemSpider ID),
</p>
</li>
<li> <p><code>'pubchem_sid'</code> (PubChem SID),
</p>
</li>
<li> <p><code>'chemnavigator_sid'</code> (ChemNavigator SID),
</p>
</li>
<li> <p><code>'h_bond_donor_count'</code> (Number of Hydrogen Bond Donors),
</p>
</li>
<li> <p><code>'h_bond_acceptor_count'</code> (Number of Hydrogen Bond Acceptors),
</p>
</li>
<li> <p><code>'h_bond_center_count'</code> (Number of Hydrogen Bond Centers),
</p>
</li>
<li> <p><code>'rule_of_5_violation_count'</code> (Number of Rule of 5 Violations),
</p>
</li>
<li> <p><code>'rotor_count'</code> (Number of Freely Rotatable Bonds),
</p>
</li>
<li> <p><code>'effective_rotor_count'</code> (Number of Effectively Rotatable Bonds),
</p>
</li>
<li> <p><code>'ring_count'</code> (Number of Rings),
</p>
</li>
<li> <p><code>'ringsys_count'</code> (Number of Ring Systems),
</p>
</li>
<li> <p><code>'xlogp2'</code> (octanol-water partition coefficient),
</p>
</li>
<li> <p><code>'aromatic'</code> (is the compound aromatic),
</p>
</li>
<li> <p><code>'macrocyclic'</code> (is the compound macrocyclic),
</p>
</li>
<li> <p><code>'heteroatom_count'</code> (heteroatom count),
</p>
</li>
<li> <p><code>'hydrogen_atom_count'</code> (H atom count),
</p>
</li>
<li> <p><code>'heavy_atom_count'</code> ( Heavy atom count),
</p>
</li>
<li> <p><code>'deprotonable_group_count'</code> (Number of deprotonable groups),
</p>
</li>
<li> <p><code>'protonable_group_count'</code> (Number of protonable groups).
</p>
</li></ul>

<p>CIR first tries to determine the identifier type submitted and then
uses 'resolvers' to look up the data.
If no <code>resolver</code> is supplied, CIR tries different resolvers in
turn till a hit is found.
E.g. for names CIR tries first to look up in OPSIN and if this fails
the local name index of CIR.
However, it can be also specified which resolvers to use
(if you know e.g. know your identifier type)
Possible <code>resolvers</code> are:
</p>

<ul>
<li> <p><code>'name_by_cir'</code> (Lookup in name index of CIR),
</p>
</li>
<li> <p><code>'name_by_opsin'</code> (Lookup in OPSIN),
</p>
</li>
<li> <p><code>'name_by_chemspider'</code> (Lookup in ChemSpider,
<a href="https://cactus.nci.nih.gov/blog/?p=1386">https://cactus.nci.nih.gov/blog/?p=1386</a>),
</p>
</li>
<li> <p><code>'smiles'</code> (Lookup SMILES),
</p>
</li>
<li> <p><code>'stdinchikey'</code>, <code>'stdinchi'</code> (InChI),
</p>
</li>
<li> <p><code>'cas_number'</code> (CAS Number),
</p>
</li>
<li> <p><code>'name_pattern'</code> (Google-like pattern search
(<a href="https://cactus.nci.nih.gov/blog/?p=1456">https://cactus.nci.nih.gov/blog/?p=1456</a>)
Note, that the pattern search can be combined with other resolvers,
e.g. <code>resolver = 'name_by_chemspider,name_pattern'</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble with a 'query' column and a column for the requested representation.
</p>


<h3>Note</h3>

<p>You can only make 1 request per second (this is a hard-coded feature).
</p>


<h3>References</h3>

<p><code>cir</code> relies on the great CIR web service created by the CADD
Group at NCI/NIH! <br />
<a href="https://cactus.nci.nih.gov/chemical/structure_documentation">https://cactus.nci.nih.gov/chemical/structure_documentation</a>, <br />
<a href="https://cactus.nci.nih.gov/blog/?cat=10">https://cactus.nci.nih.gov/blog/?cat=10</a>, <br />
<a href="https://cactus.nci.nih.gov/blog/?p=1386">https://cactus.nci.nih.gov/blog/?p=1386</a>, <br />
<a href="https://cactus.nci.nih.gov/blog/?p=1456">https://cactus.nci.nih.gov/blog/?p=1456</a>, <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
cir_query("Triclosan", "cas")
cir_query("3380-34-5", "cas", match = "first")
cir_query("3380-34-5", "cas", resolver = "cas_number")
cir_query("3380-34-5", "smiles")
cir_query("Triclosan", "mw")

# multiple inputs
comp &lt;- c("Triclosan", "Aspirin")
cir_query(comp, "cas", match = "first")


## End(Not run)
</code></pre>

<hr>
<h2 id='cs_check_key'>Retrieve ChemSpider API key</h2><span id='topic+cs_check_key'></span>

<h3>Description</h3>

<p>Look for and retrieve ChemSpider API key stored in .Renviron or .Rprofile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_check_key()
</code></pre>


<h3>Details</h3>

<p>To use the any of the functions in <code>webchem</code> that access the
ChemSpider database, you'll need to obtain an API key. Register at
<a href="https://developer.rsc.org/">https://developer.rsc.org/</a> for an API key. Please respect the Terms &amp;
Conditions <a href="https://developer.rsc.org/terms">https://developer.rsc.org/terms</a>.
</p>
<p>You can store your API key as <code>CHEMSPIDER_KEY = &lt;your key&gt;</code> in
.Renviron or as <code>options(chemspider_key = &lt;your key&gt;)</code> in .Rprofile.
This will allow you to use ChemSpider without adding your API key in the
beginning of each session, and will also allow you to share your analysis
without sharing your API key. Keeping your API key hidden is good practice.
</p>


<h3>Value</h3>

<p>an API key
</p>


<h3>See Also</h3>

<p><code><a href="usethis.html#topic+edit_r_environ">edit_r_environ</a></code>
<code><a href="usethis.html#topic+edit_r_profile">edit_r_profile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cs_check_key()

## End(Not run)
</code></pre>

<hr>
<h2 id='cs_compinfo'>Retrieve record details by ChemSpider ID</h2><span id='topic+cs_compinfo'></span>

<h3>Description</h3>

<p>Submit a ChemSpider ID (CSID) and the fields you are interested in, and
retrieve the record details for your query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_compinfo(csid, fields, verbose = getOption("verbose"), apikey = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_compinfo_+3A_csid">csid</code></td>
<td>
<p>numeric; can be obtained using <code><a href="#topic+get_csid">get_csid</a></code></p>
</td></tr>
<tr><td><code id="cs_compinfo_+3A_fields">fields</code></td>
<td>
<p>character; see details.</p>
</td></tr>
<tr><td><code id="cs_compinfo_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cs_compinfo_+3A_apikey">apikey</code></td>
<td>
<p>character; your API key. If NULL (default),
<code>cs_check_key()</code> will look for it in .Renviron or .Rprofile.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Valid values for <code>fields</code> are <code>"SMILES"</code>,
<code>"Formula"</code>, <code>"InChI"</code>, <code>"InChIKey"</code>, <code>"StdInChI"</code>,
<code>"StdInChIKey"</code>, <code>"AverageMass"</code>, <code>"MolecularWeight"</code>,
<code>"MonoisotopicMass"</code>, <code>"NominalMass"</code>, <code>"CommonName"</code>,
<code>"ReferenceCount"</code>, <code>"DataSourceCount"</code>, <code>"PubMedCount"</code>,
<code>"RSCCount"</code>, <code>"Mol2D"</code>, <code>"Mol3D"</code>. You can specify any
number of fields.
</p>


<h3>Value</h3>

<p>Returns a data frame.
</p>


<h3>Note</h3>

<p>An API key is needed. Register at <a href="https://developer.rsc.org/">https://developer.rsc.org/</a>
for an API key. Please respect the Terms &amp; Conditions. The Terms &amp; Conditions
can be found at <a href="https://developer.rsc.org/terms">https://developer.rsc.org/terms</a>.
</p>


<h3>References</h3>

<p><a href="https://developer.rsc.org/docs/compounds-v1-trial/1/overview">https://developer.rsc.org/docs/compounds-v1-trial/1/overview</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cs_compinfo(171, c("SMILES", "CommonName"))
cs_compinfo(171:182, "SMILES")

## End(Not run)
</code></pre>

<hr>
<h2 id='cs_control'>Control ChemSpider API requests</h2><span id='topic+cs_control'></span>

<h3>Description</h3>

<p>For some ChemSpider API requests, you can also specify various control
options. This function is used to set these control options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_control(
  datasources = vector(),
  order_by = "default",
  order_direction = "default",
  include_all = FALSE,
  complexity = "any",
  isotopic = "any"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_control_+3A_datasources">datasources</code></td>
<td>
<p>character; specifies the databases to query. Use
<code>cs_datasources()</code> to retrieve available ChemSpider data sources.</p>
</td></tr>
<tr><td><code id="cs_control_+3A_order_by">order_by</code></td>
<td>
<p>character; specifies the sort order for the results.
Valid values are <code>"default"</code>, <code>"recordId"</code>, <code>"massDefect"</code>,
<code>"molecularWeight"</code>, <code>"referenceCount"</code>, <code>"dataSourceCount"</code>,
<code>"pubMedCount"</code>, <code>"rscCount"</code>.</p>
</td></tr>
<tr><td><code id="cs_control_+3A_order_direction">order_direction</code></td>
<td>
<p>character; specifies the sort order for the results.
Valid values are <code>"default"</code>, <code>"ascending"</code>, <code>"descending"</code>.</p>
</td></tr>
<tr><td><code id="cs_control_+3A_include_all">include_all</code></td>
<td>
<p>logical; see details.</p>
</td></tr>
<tr><td><code id="cs_control_+3A_complexity">complexity</code></td>
<td>
<p>character; see details.
Valid values are <code>"any"</code> <code>"single"</code>, <code>"multiple"</code>.</p>
</td></tr>
<tr><td><code id="cs_control_+3A_isotopic">isotopic</code></td>
<td>
<p>character; see details.
Valid values are <code>"any"</code>, <code>"labeled"</code>, <code>"unlabeled"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The only function that currently uses <code>databases</code> is
<code>get_csid()</code> and only when you query a CSID from a formula. This
parameter is disregarded in all other queries.
</p>
<p>Setting <code>include_all</code> to <code>TRUE</code> will consider records
which contain all of the filter criteria specified in the request. Setting
it to <code>FALSE</code> will consider records which contain any of the filter
criteria.
</p>
<p>A compound with a  <code>complexity</code> of <code>"multiple"</code> has more
than one disconnected system in it or a metal atom or ion.
</p>


<h3>Value</h3>

<p>Returns a list of specified control options.
</p>


<h3>Note</h3>

<p>This is a full list of all API control options.
However, not all of these options are used in all functions.
Each API uses a subset of these controls.
The controls that are available for a given function are indicated within the
documentation of the function.
</p>


<h3>References</h3>

<p><a href="https://developer.rsc.org/docs/compounds-v1-trial/1/overview">https://developer.rsc.org/docs/compounds-v1-trial/1/overview</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_csid">get_csid</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cs_control()
cs_control(order_direction = "descending")
</code></pre>

<hr>
<h2 id='cs_convert'>Convert identifiers using ChemSpider</h2><span id='topic+cs_convert'></span>

<h3>Description</h3>

<p>Submit one or more identifiers (CSID, SMILES, InChI, InChIKey or Mol) and
return one or more identifiers in another format (CSID, SMILES, InChI,
InChIKey or Mol).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_convert(query, from, to, verbose = getOption("verbose"), apikey = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_convert_+3A_query">query</code></td>
<td>
<p>character; query ID.</p>
</td></tr>
<tr><td><code id="cs_convert_+3A_from">from</code></td>
<td>
<p>character; type of query ID.</p>
</td></tr>
<tr><td><code id="cs_convert_+3A_to">to</code></td>
<td>
<p>character; type to convert to.</p>
</td></tr>
<tr><td><code id="cs_convert_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cs_convert_+3A_apikey">apikey</code></td>
<td>
<p>character; your API key. If NULL (default),
<code>cs_check_key()</code> will look for it in .Renviron or .Rprofile.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Not all conversions are supported. Allowed conversions:
</p>

<ul>
<li><p> CSID &lt;-&gt; InChI
</p>
</li>
<li><p> CSID &lt;-&gt; InChIKey
</p>
</li>
<li><p> CSID &lt;-&gt; SMILES
</p>
</li>
<li><p> CSID -&gt; Mol file
</p>
</li>
<li><p> InChI &lt;-&gt; InChIKey
</p>
</li>
<li><p> InChI &lt;-&gt; SMILES
</p>
</li>
<li><p> InChI -&gt; Mol file
</p>
</li>
<li><p> InChIKey &lt;-&gt; Mol file
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a vector containing the converted identifier(s).
</p>


<h3>Note</h3>

<p>An API key is needed. Register at <a href="https://developer.rsc.org/">https://developer.rsc.org/</a>
for an API key. Please respect the Terms &amp; Conditions. The Terms &amp; Conditions
can be found at <a href="https://developer.rsc.org/terms">https://developer.rsc.org/terms</a>.
</p>


<h3>References</h3>

<p><a href="https://developer.rsc.org/docs/compounds-v1-trial/1/overview">https://developer.rsc.org/docs/compounds-v1-trial/1/overview</a>
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cs_convert("BQJCRHHNABKAKU-KBQPJGBKSA-N",
  from = "inchikey", to = "csid"
)
cs_convert("BQJCRHHNABKAKU-KBQPJGBKSA-N",
  from = "inchikey", to = "inchi"
)
cs_convert("BQJCRHHNABKAKU-KBQPJGBKSA-N",
  from = "inchikey", to = "mol"
)
cs_convert(160, from = "csid", to = "smiles")

## End(Not run)
</code></pre>

<hr>
<h2 id='cs_datasources'>Retrieve ChemSpider data sources</h2><span id='topic+cs_datasources'></span>

<h3>Description</h3>

<p>The function returns a vector of available data sources used by ChemSpider.
Some ChemSpider functions allow you to restrict which sources are used to
lookup the requested query. Restricting the sources makes these queries
faster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_datasources(apikey = NULL, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_datasources_+3A_apikey">apikey</code></td>
<td>
<p>character; your API key. If NULL (default),
<code>cs_check_key()</code> will look for it in .Renviron or .Rprofile.</p>
</td></tr>
<tr><td><code id="cs_datasources_+3A_verbose">verbose</code></td>
<td>
<p>should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector.
</p>


<h3>Note</h3>

<p>An API key is needed. Register at <a href="https://developer.rsc.org/">https://developer.rsc.org/</a>
for an API key. Please respect the Terms &amp; Conditions. The Terms &amp; Conditions
can be found at <a href="https://developer.rsc.org/terms">https://developer.rsc.org/terms</a>.
</p>


<h3>References</h3>

<p><a href="https://developer.rsc.org/docs/compounds-v1-trial/1/overview">https://developer.rsc.org/docs/compounds-v1-trial/1/overview</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cs_datasources()

## End(Not run)
</code></pre>

<hr>
<h2 id='cs_extcompinfo'>Get extended record details by ChemSpider ID</h2><span id='topic+cs_extcompinfo'></span>

<h3>Description</h3>

<p>Get extended info from ChemSpider, see <a href="https://www.chemspider.com/">https://www.chemspider.com/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_extcompinfo(csid, token, verbose = getOption("verbose"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_extcompinfo_+3A_csid">csid</code></td>
<td>
<p>character,  ChemSpider ID.</p>
</td></tr>
<tr><td><code id="cs_extcompinfo_+3A_token">token</code></td>
<td>
<p>character; security token.</p>
</td></tr>
<tr><td><code id="cs_extcompinfo_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cs_extcompinfo_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with entries: 'csid', 'mf' (molecular formula),
'smiles', 'inchi' (non-standard), 'inchikey' (non-standard), 'average_mass',
'mw' (Molecular weight), 'monoiso_mass' (MonoisotopicMass), nominal_mass',
'alogp', 'xlogp', 'common_name' and 'source_url'
</p>


<h3>Note</h3>

<p>A security token is needed. Please register at RSC
<a href="https://www.rsc.org/rsc-id/register">https://www.rsc.org/rsc-id/register</a>
for a security token.
Please respect the Terms &amp; conditions
<a href="https://www.rsc.org/help-legal/legal/terms-conditions/">https://www.rsc.org/help-legal/legal/terms-conditions/</a>.
</p>
<p>use <code><a href="#topic+cs_compinfo">cs_compinfo</a></code> to retrieve standard inchikey.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_csid">get_csid</a></code> to retrieve ChemSpider IDs,
<code><a href="#topic+cs_compinfo">cs_compinfo</a></code> for extended compound information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
token &lt;- "&lt;redacted&gt;"
csid &lt;- get_csid("Triclosan")
cs_extcompinfo(csid, token)

csids &lt;- get_csid(c('Aspirin', 'Triclosan'))
cs_compinfo(csids)

## End(Not run)
</code></pre>

<hr>
<h2 id='cs_img'>Download images from ChemSpider</h2><span id='topic+cs_img'></span>

<h3>Description</h3>

<p>Retrieve images of substances from ChemSpider and export them
in PNG format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cs_img(
  csid,
  dir,
  overwrite = TRUE,
  apikey = NULL,
  verbose = getOption("verbose")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cs_img_+3A_csid">csid</code></td>
<td>
<p>numeric; the ChemSpider ID (CSID) of the substance. This will
also be the name of the image file.</p>
</td></tr>
<tr><td><code id="cs_img_+3A_dir">dir</code></td>
<td>
<p>character; the download directory. <code>dir</code> accepts both
absolute and relative paths.</p>
</td></tr>
<tr><td><code id="cs_img_+3A_overwrite">overwrite</code></td>
<td>
<p>logical; should existing files in the directory with the
same name be overwritten?</p>
</td></tr>
<tr><td><code id="cs_img_+3A_apikey">apikey</code></td>
<td>
<p>character; your API key. If NULL (default),
<code>cs_check_key()</code> will look for it in .Renviron or .Rprofile.</p>
</td></tr>
<tr><td><code id="cs_img_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Note</h3>

<p>An API key is needed. Register at <a href="https://developer.rsc.org/">https://developer.rsc.org/</a>
for an API key. Please respect the Terms &amp; Conditions. The Terms &amp; Conditions
can be found at <a href="https://developer.rsc.org/terms">https://developer.rsc.org/terms</a>.
</p>


<h3>References</h3>

<p><a href="https://developer.rsc.org/docs/compounds-v1-trial/1/overview">https://developer.rsc.org/docs/compounds-v1-trial/1/overview</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_csid">get_csid</a></code>, <code><a href="#topic+cs_check_key">cs_check_key</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cs_img(c(582, 682), dir = tempdir())

## End(Not run)
</code></pre>

<hr>
<h2 id='cts_compinfo'>Get record details from Chemical Translation Service (CTS)</h2><span id='topic+cts_compinfo'></span>

<h3>Description</h3>

<p>Get record details from CTS, see <a href="http://cts.fiehnlab.ucdavis.edu/">http://cts.fiehnlab.ucdavis.edu/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cts_compinfo(
  query,
  from = "inchikey",
  verbose = getOption("verbose"),
  inchikey
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cts_compinfo_+3A_query">query</code></td>
<td>
<p>character; InChIkey.</p>
</td></tr>
<tr><td><code id="cts_compinfo_+3A_from">from</code></td>
<td>
<p>character; currently only accepts &quot;inchikey&quot;.</p>
</td></tr>
<tr><td><code id="cts_compinfo_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cts_compinfo_+3A_inchikey">inchikey</code></td>
<td>
<p>deprecated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of lists (for each supplied inchikey):
a list of 7. inchikey, inchicode, molweight, exactmass, formula, synonyms and
externalIds
</p>


<h3>References</h3>

<p>Wohlgemuth, G., P. K. Haldiya, E. Willighagen, T. Kind, and O.
Fiehn 2010The Chemical Translation Service &ndash; a Web-Based Tool to Improve
Standardization of Metabolomic Reports. Bioinformatics 26(20): 2647–2648.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
out &lt;- cts_compinfo("XEFQLINVKFYRCS-UHFFFAOYSA-N")
# = Triclosan
str(out)
out[[1]][1:5]

### multiple inputs
inchikeys &lt;- c("XEFQLINVKFYRCS-UHFFFAOYSA-N","BSYNRYMUTXBXSQ-UHFFFAOYSA-N" )
out2 &lt;- cts_compinfo(inchikeys)
str(out2)
# a list of two
# extract molecular weight
sapply(out2, function(y) y$molweight)

## End(Not run)
</code></pre>

<hr>
<h2 id='cts_convert'>Convert Ids using Chemical Translation Service (CTS)</h2><span id='topic+cts_convert'></span>

<h3>Description</h3>

<p>Convert Ids using Chemical Translation Service (CTS), see
<a href="http://cts.fiehnlab.ucdavis.edu/">http://cts.fiehnlab.ucdavis.edu/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cts_convert(
  query,
  from,
  to,
  match = c("all", "first", "ask", "na"),
  verbose = getOption("verbose"),
  choices = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cts_convert_+3A_query">query</code></td>
<td>
<p>character; query ID.</p>
</td></tr>
<tr><td><code id="cts_convert_+3A_from">from</code></td>
<td>
<p>character; type of query ID, e.g. <code>'Chemical Name'</code> ,
<code>'InChIKey'</code>, <code>'PubChem CID'</code>, <code>'ChemSpider'</code>, <code>'CAS'</code>.</p>
</td></tr>
<tr><td><code id="cts_convert_+3A_to">to</code></td>
<td>
<p>character; type to convert to.</p>
</td></tr>
<tr><td><code id="cts_convert_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handled? <code>"all"</code>
returns all matches, <code>"first"</code> returns only the first result,
<code>"ask"</code> enters an interactive mode and the user is asked for input,
<code>"na"</code> returns <code>NA</code> if multiple hits are found.</p>
</td></tr>
<tr><td><code id="cts_convert_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="cts_convert_+3A_choices">choices</code></td>
<td>
<p>deprecated.  Use the <code>match</code> argument instead.</p>
</td></tr>
<tr><td><code id="cts_convert_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See also <a href="http://cts.fiehnlab.ucdavis.edu/">http://cts.fiehnlab.ucdavis.edu/</a>
for possible values of from and to.
</p>


<h3>Value</h3>

<p>a list of character vectors or if <code>choices</code> is used, then a
single named vector.
</p>


<h3>References</h3>

<p>Wohlgemuth, G., P. K. Haldiya, E. Willighagen, T. Kind, and O.
Fiehn 2010The Chemical Translation Service &ndash; a Web-Based Tool to Improve
Standardization of Metabolomic Reports. Bioinformatics 26(20): 2647–2648.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cts_from">cts_from</a></code> for possible values in the 'from' argument
and <code><a href="#topic+cts_to">cts_to</a></code> for possible values in the 'to' argument.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
cts_convert("XEFQLINVKFYRCS-UHFFFAOYSA-N", "inchikey", "Chemical Name")

### multiple inputs
keys &lt;- c("XEFQLINVKFYRCS-UHFFFAOYSA-N", "VLKZOEOYAKHREP-UHFFFAOYSA-N")
cts_convert(keys, "inchikey", "cas")

## End(Not run)
</code></pre>

<hr>
<h2 id='cts_from'>Return a list of all possible ids</h2><span id='topic+cts_from'></span>

<h3>Description</h3>

<p>Return a list of all possible ids that can be used in the 'from' argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cts_from(verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cts_from_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See also <a href="http://cts.fiehnlab.ucdavis.edu/services">http://cts.fiehnlab.ucdavis.edu/services</a>
</p>


<h3>Value</h3>

<p>a character vector.
</p>


<h3>References</h3>

<p>Wohlgemuth, G., P. K. Haldiya, E. Willighagen, T. Kind, and O.
Fiehn 2010The Chemical Translation Service &ndash; a Web-Based Tool to Improve
Standardization of Metabolomic Reports. Bioinformatics 26(20): 2647–2648.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cts_convert">cts_convert</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cts_from()

## End(Not run)
</code></pre>

<hr>
<h2 id='cts_to'>Return a list of all possible ids</h2><span id='topic+cts_to'></span>

<h3>Description</h3>

<p>Return a list of all possible ids that can be used in the 'to' argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cts_to(verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cts_to_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See also <a href="http://cts.fiehnlab.ucdavis.edu/services">http://cts.fiehnlab.ucdavis.edu/services</a>
</p>


<h3>Value</h3>

<p>a character vector.
</p>


<h3>References</h3>

<p>Wohlgemuth, G., P. K. Haldiya, E. Willighagen, T. Kind, and O.
Fiehn 2010The Chemical Translation Service &ndash; a Web-Based Tool to Improve
Standardization of Metabolomic Reports. Bioinformatics 26(20): 2647–2648.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+cts_convert">cts_convert</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cts_from()

## End(Not run)
</code></pre>

<hr>
<h2 id='etox_basic'>Get basic information from a ETOX ID</h2><span id='topic+etox_basic'></span>

<h3>Description</h3>

<p>Query ETOX: Information System Ecotoxicology and Environmental Quality
Targets <a href="https://webetox.uba.de/webETOX/index.do">https://webetox.uba.de/webETOX/index.do</a> for basic information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>etox_basic(id, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="etox_basic_+3A_id">id</code></td>
<td>
<p>character; ETOX ID</p>
</td></tr>
<tr><td><code id="etox_basic_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with lists of four entries: cas (the CAS numbers), ec (the EC
number), gsbl (the gsbl number), a data.frame synonys with synonyms and the
source url.
</p>


<h3>Note</h3>

<p>Before using this function, please read the disclaimer
<a href="https://webetox.uba.de/webETOX/disclaimer.do">https://webetox.uba.de/webETOX/disclaimer.do</a>.
</p>


<h3>References</h3>

<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_etoxid">get_etoxid</a></code> to retrieve ETOX IDs,
<code><a href="#topic+etox_basic">etox_basic</a></code> for basic information, <code><a href="#topic+etox_targets">etox_targets</a></code>
for quality targets and <code><a href="#topic+etox_tests">etox_tests</a></code> for test results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
id &lt;- get_etoxid('Triclosan', match = 'best')
etox_basic(id$etoxid)

# Retrieve data for multiple inputs
ids &lt;- c("20179", "9051")
out &lt;- etox_basic(ids)
out

# extract cas numbers
sapply(out, function(y) y$cas)

## End(Not run)
</code></pre>

<hr>
<h2 id='etox_targets'>Get Quality Targets from a ETOX ID</h2><span id='topic+etox_targets'></span>

<h3>Description</h3>

<p>Query ETOX: Information System Ecotoxicology and Environmental Quality
Targets <a href="https://webetox.uba.de/webETOX/index.do">https://webetox.uba.de/webETOX/index.do</a> for quality targets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>etox_targets(id, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="etox_targets_+3A_id">id</code></td>
<td>
<p>character; ETOX ID</p>
</td></tr>
<tr><td><code id="etox_targets_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of lists of two: <code>res</code> a data.frame with quality targets
from the ETOX database, and source_url.
</p>


<h3>Note</h3>

<p>Before using this function, please read the disclaimer
<a href="https://webetox.uba.de/webETOX/disclaimer.do">https://webetox.uba.de/webETOX/disclaimer.do</a>.
</p>


<h3>References</h3>

<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_etoxid">get_etoxid</a></code> to retrieve ETOX IDs,
<code><a href="#topic+etox_basic">etox_basic</a></code> for basic information, <code><a href="#topic+etox_targets">etox_targets</a></code>
for quality targets and <code><a href="#topic+etox_tests">etox_tests</a></code> for test results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
id &lt;- get_etoxid('Triclosan', match = 'best')
out &lt;- etox_targets(id$etoxid)
out[ , c('Substance', 'CAS_NO', 'Country_or_Region', 'Designation',
'Value_Target_LR', 'Unit')]
etox_targets( c("20179", "9051"))


## End(Not run)
</code></pre>

<hr>
<h2 id='etox_tests'>Get Tests from a ETOX ID</h2><span id='topic+etox_tests'></span>

<h3>Description</h3>

<p>Query ETOX: Information System Ecotoxicology and Environmental Quality Targets
<a href="https://webetox.uba.de/webETOX/index.do">https://webetox.uba.de/webETOX/index.do</a> for tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>etox_tests(id, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="etox_tests_+3A_id">id</code></td>
<td>
<p>character; ETOX ID</p>
</td></tr>
<tr><td><code id="etox_tests_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of lists of two: A data.frame with test results from the ETOX database and the source_url.
</p>


<h3>Note</h3>

<p>Before using this function, please read the disclaimer
<a href="https://webetox.uba.de/webETOX/disclaimer.do">https://webetox.uba.de/webETOX/disclaimer.do</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_etoxid">get_etoxid</a></code> to retrieve ETOX IDs, <code><a href="#topic+etox_basic">etox_basic</a></code> for basic information,
<code><a href="#topic+etox_targets">etox_targets</a></code> for quality targets and <code><a href="#topic+etox_tests">etox_tests</a></code> for test results
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
id &lt;- get_etoxid('Triclosan', match = 'best')
out &lt;- etox_tests(id$etoxid)
out[ , c('Organism', 'Effect', 'Duration', 'Time_Unit',
'Endpoint', 'Value', 'Unit')]
etox_tests( c("20179", "9051"))

## End(Not run)
</code></pre>

<hr>
<h2 id='extractors'>Extract parts from webchem objects</h2><span id='topic+extractors'></span><span id='topic+cas'></span><span id='topic+inchikey'></span><span id='topic+smiles'></span>

<h3>Description</h3>

<p>Extract parts from webchem objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cas(x, ...)

inchikey(x, ...)

smiles(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractors_+3A_x">x</code></td>
<td>
<p>object</p>
</td></tr>
<tr><td><code id="extractors_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector.
</p>


<h3>References</h3>

<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>

<hr>
<h2 id='find_db'>Check data source coverage of compounds</h2><span id='topic+find_db'></span>

<h3>Description</h3>

<p>Checks if entries are found in (most) data sources included in webchem
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_db(
  query,
  from,
  sources = c("etox", "pc", "chebi", "cs", "bcpc", "fn", "srs"),
  plot = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find_db_+3A_query">query</code></td>
<td>
<p>character; the search term</p>
</td></tr>
<tr><td><code id="find_db_+3A_from">from</code></td>
<td>
<p>character; the format or type of query.  Commonly accepted values
are &quot;name&quot;, &quot;cas&quot;, &quot;inchi&quot;, and &quot;inchikey&quot;</p>
</td></tr>
<tr><td><code id="find_db_+3A_sources">sources</code></td>
<td>
<p>character; which data sources to check.  Data sources are
identified by the prefix associated with webchem functions that query those
databases.  If not specified, all data sources listed will be checked.</p>
</td></tr>
<tr><td><code id="find_db_+3A_plot">plot</code></td>
<td>
<p>logical; plot a graphical representation of results.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble of logical values where <code>TRUE</code> indicates that a data
source contains a record for the query
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
find_db("hexane", from = "name")

## End(Not run)
</code></pre>

<hr>
<h2 id='fn_percept'>Retrieve flavor percepts from www.flavornet.org</h2><span id='topic+fn_percept'></span>

<h3>Description</h3>

<p>Retrieve flavor percepts from <a href="http://www.flavornet.org">http://www.flavornet.org</a>.  Flavornet is a database of 738 compounds with odors
perceptible to humans detected using gas chromatography olfactometry (GCO).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fn_percept(query, from = "cas", verbose = getOption("verbose"), CAS, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fn_percept_+3A_query">query</code></td>
<td>
<p>character; CAS number to search by. See <code><a href="#topic+is.cas">is.cas</a></code> for correct formatting</p>
</td></tr>
<tr><td><code id="fn_percept_+3A_from">from</code></td>
<td>
<p>character; currently only CAS numbers are accepted.</p>
</td></tr>
<tr><td><code id="fn_percept_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="fn_percept_+3A_cas">CAS</code></td>
<td>
<p>deprecated</p>
</td></tr>
<tr><td><code id="fn_percept_+3A_...">...</code></td>
<td>
<p>currently unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named character vector containing flavor percepts or NA's in the case of CAS numbers that are not found
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if website is not available
fn_percept("123-32-0")

CASs &lt;- c("75-07-0",  "64-17-5",  "109-66-0", "78-94-4",  "78-93-3")
fn_percept(CASs)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_chebiid'>Retrieve Lite Entity (identifiers) from ChEBI</h2><span id='topic+get_chebiid'></span>

<h3>Description</h3>

<p>Returns a data.frame with a ChEBI entity ID (chebiid),
a ChEBI entity name (chebiasciiname), a search score (searchscore) and
stars (stars) using the SOAP protocol:
<a href="https://www.ebi.ac.uk/chebi/webServices.do">https://www.ebi.ac.uk/chebi/webServices.do</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_chebiid(
  query,
  from = c("all", "chebi id", "chebi name", "definition", "name", "iupac name",
    "citations", "registry numbers", "manual xrefs", "automatic xrefs", "formula",
    "mass", "monoisotopic mass", "charge", "inchi", "inchikey", "smiles", "species"),
  match = c("all", "best", "first", "ask", "na"),
  max_res = 200,
  stars = c("all", "two only", "three only"),
  verbose = getOption("verbose"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_chebiid_+3A_query">query</code></td>
<td>
<p>character; search term.</p>
</td></tr>
<tr><td><code id="get_chebiid_+3A_from">from</code></td>
<td>
<p>character; type of input.  <code>"all"</code> searches all types and
<code>"name"</code> searches all names. Other options include <code>'chebi id'</code>,
<code>'chebi name'</code>, <code>'definition'</code>, <code>'iupac name'</code>,
<code>'citations'</code>, <code>'registry numbers'</code>, <code>'manual xrefs'</code>,
<code>'automatic xrefs'</code>, <code>'formula'</code>, <code>'mass'</code>,
<code>'monoisotopic mass'</code>,<code>'charge'</code>, <code>'inchi'</code>,
<code>'inchikey'</code>, <code>'smiles'</code>, and <code>'species'</code></p>
</td></tr>
<tr><td><code id="get_chebiid_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handled?, <code>"all"</code>
all matches are returned, <code>"best"</code> the best matching (by the ChEBI
searchscore) is returned, <code>"ask"</code> enters an interactive mode and the
user is asked for input, <code>"na"</code> returns NA if multiple hits are found.</p>
</td></tr>
<tr><td><code id="get_chebiid_+3A_max_res">max_res</code></td>
<td>
<p>integer; maximum number of results to be retrieved from the
web service</p>
</td></tr>
<tr><td><code id="get_chebiid_+3A_stars">stars</code></td>
<td>
<p>character; &quot;three only&quot; restricts results to those manualy
annotated by the ChEBI team.</p>
</td></tr>
<tr><td><code id="get_chebiid_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="get_chebiid_+3A_...">...</code></td>
<td>
<p>currently unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of data.frames containing a chebiid, a chebiasciiname,
a searchscore and stars if matches were found. If not, data.frame(NA) is
returned
</p>


<h3>References</h3>

<p>Hastings J, Owen G, Dekker A, Ennis M, Kale N, Muthukrishnan V,
Turner S, Swainston N, Mendes P, Steinbeck C. (2016). ChEBI in 2016:
Improved services and an expanding collection of metabfolites. Nucleic
Acids Res.
</p>
<p>Hastings, J., de Matos, P., Dekker, A., Ennis, M., Harsha, B., Kale, N.,
Muthukrishnan, V., Owen, G., Turner, S., Williams, M., and Steinbeck, C.
(2013) The ChEBI reference database and ontology for biologically relevant
chemistry: enhancements for 2013. Nucleic Acids Res.
</p>
<p>de Matos, P., Alcantara, R., Dekker, A., Ennis, M., Hastings, J., Haug, K.,
Spiteri, I., Turner, S., and Steinbeck, C. (2010) Chemical entities of
biological interest: an update. Nucleic Acids Res. Degtyarenko, K.,
Hastings, J., de Matos, P., and Ennis, M. (2009). ChEBI: an open
bioinformatics and cheminformatics resource. Current protocols in
bioinformatics / editoral board, Andreas D. Baxevanis et al., Chapter 14.
</p>
<p>Degtyarenko, K., de Matos, P., Ennis, M., Hastings, J., Zbinden, M.,
McNaught, A., Alcántara, R., Darsow, M., Guedj, M. and Ashburner, M. (2008)
ChEBI: a database and ontology for chemical entities of biological
interest. Nucleic Acids Res. 36, D344–D350.
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
get_chebiid('Glyphosate')
get_chebiid('BPGDAMSIGCZZLK-UHFFFAOYSA-N')

# multiple inputs
comp &lt;- c('Iron', 'Aspirin', 'BPGDAMSIGCZZLK-UHFFFAOYSA-N')
get_chebiid(comp)


## End(Not run)
</code></pre>

<hr>
<h2 id='get_cid'>Retrieve Pubchem Compound ID (CID)</h2><span id='topic+get_cid'></span>

<h3>Description</h3>

<p>Retrieve compound IDs (CIDs) from PubChem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cid(
  query,
  from = "name",
  domain = c("compound", "substance", "assay"),
  match = c("all", "first", "ask", "na"),
  verbose = getOption("verbose"),
  arg = NULL,
  first = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_cid_+3A_query">query</code></td>
<td>
<p>character; search term, one or more compounds.</p>
</td></tr>
<tr><td><code id="get_cid_+3A_from">from</code></td>
<td>
<p>character; type of input. See details for more information.</p>
</td></tr>
<tr><td><code id="get_cid_+3A_domain">domain</code></td>
<td>
<p>character; query domain, can be one of <code>"compound"</code>,
<code>"substance"</code>, <code>"assay"</code>.</p>
</td></tr>
<tr><td><code id="get_cid_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handled?, <code>"all"</code>
all matches are returned, <code>"first"</code> the first matching is returned,
<code>"ask"</code> enters an interactive mode and the user is asked for input,
<code>"na"</code> returns NA if multiple hits are found.</p>
</td></tr>
<tr><td><code id="get_cid_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="get_cid_+3A_arg">arg</code></td>
<td>
<p>character; optinal arguments like &quot;name_type=word&quot; to match
individual words.</p>
</td></tr>
<tr><td><code id="get_cid_+3A_first">first</code></td>
<td>
<p>deprecated. Use 'match' instead.</p>
</td></tr>
<tr><td><code id="get_cid_+3A_...">...</code></td>
<td>
<p>currently unused.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Valid values for the <code>from</code> argument depend on the
<code>domain</code>:
</p>

<ul>
<li><p><code>compound</code>: <code>"name"</code>, <code>"smiles"</code>, <code>"inchi"</code>,
<code>"inchikey"</code>, <code>"formula"</code>, <code>"sdf"</code>, <code>"cas"</code> (an alias for
<code>"xref/RN"</code>), &lt;xref&gt;, &lt;structure search&gt;, &lt;fast search&gt;.
</p>
</li>
<li><p><code>substance</code>: <code>"name"</code>, <code>"sid"</code>,
<code>&lt;xref&gt;</code>, <code>"sourceid/&lt;source id&gt;"</code> or <code>"sourceall"</code>.
</p>
</li>
<li><p><code>assay</code>: <code>"aid"</code>, <code>&lt;assay target&gt;</code>.
</p>
</li></ul>

<p>&lt;structure search&gt; is assembled as &quot;<code>substructure</code> |
<code>superstructure</code> | <code>similarity</code> | <code>identity</code> / <code>smiles</code>
| <code>inchi</code> | <code>sdf</code> | <code>cid</code>&quot;, e.g.
<code>from = "substructure/smiles"</code>.
</p>
<p><code>&lt;xref&gt;</code> is assembled as &quot;<code>xref</code>/{<code>RegistryID</code> |
<code>RN</code> | <code>PubMedID</code> | <code>MMDBID</code> | <code>ProteinGI</code>,
<code>NucleotideGI</code> | <code>TaxonomyID</code> | <code>MIMID</code> | <code>GeneID</code> |
<code>ProbeID</code> | <code>PatentID</code>}&quot;, e.g. <code>from = "xref/RN"</code> will query
by CAS RN.
</p>
<p>&lt;fast search&gt; is either <code>fastformula</code> or it is assembled as
&quot;<code>fastidentity</code> | <code>fastsimilarity_2d</code> | <code>fastsimilarity_3d</code> |
<code>fastsubstructure</code> | <code>fastsuperstructure</code>/<code>smiles</code> |
<code>smarts</code> | <code>inchi</code> | <code>sdf</code> | <code>cid</code>&quot;, e.g.
<code>from = "fastidentity/smiles"</code>.
</p>
<p><code>&lt;source id&gt;</code> is any valid PubChem Data Source ID. When
<code>from = "sourceid/&lt;source id&gt;"</code>, the query is the ID of the substance in
the depositor's database.
</p>
<p>If <code>from = "sourceall"</code> the query is one or more valid Pubchem
depositor names. Depositor names are not case sensitive.
</p>
<p>Depositor names and Data Source IDs can be found at
<a href="https://pubchem.ncbi.nlm.nih.gov/sources/">https://pubchem.ncbi.nlm.nih.gov/sources/</a>.
</p>
<p><code>&lt;assay target&gt;</code> is assembled as &quot;<code>target</code>/{<code>gi</code> |
<code>proteinname</code> | <code>geneid</code> | <code>genesymbol</code> | <code>accession</code>}&quot;,
e.g. <code>from = "target/geneid"</code> will query by GeneID.
</p>


<h3>Value</h3>

<p>a tibble.
</p>


<h3>Note</h3>

<p>Please respect the Terms and Conditions of the National Library of
Medicine, <a href="https://www.nlm.nih.gov/databases/download.html">https://www.nlm.nih.gov/databases/download.html</a> the data
usage policies of National Center for Biotechnology Information,
<a href="https://www.ncbi.nlm.nih.gov/home/about/policies/">https://www.ncbi.nlm.nih.gov/home/about/policies/</a>,
<a href="https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access">https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access</a>, and the data
usage policies of the indicidual data sources
<a href="https://pubchem.ncbi.nlm.nih.gov/sources/">https://pubchem.ncbi.nlm.nih.gov/sources/</a>.
</p>


<h3>References</h3>

<p>Wang, Y., J. Xiao, T. O. Suzek, et al. 2009 PubChem: A Public
Information System for
Analyzing Bioactivities of Small Molecules. Nucleic Acids Research 37:
623–633.
</p>
<p>Kim, Sunghwan, Paul A. Thiessen, Evan E. Bolton, et al. 2016
PubChem Substance and Compound Databases. Nucleic Acids Research 44(D1):
D1202–D1213.
</p>
<p>Kim, S., Thiessen, P. A., Bolton, E. E., &amp; Bryant, S. H. (2015).
PUG-SOAP and PUG-REST: web services for programmatic access to chemical
information in PubChem. Nucleic acids research, gkv396.
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
get_cid("Triclosan")
get_cid("Triclosan", arg = "name_type=word")
# from SMILES
get_cid("CCCC", from = "smiles")
# from InChI
get_cid("InChI=1S/CH5N/c1-2/h2H2,1H3", from = "inchi")
# from InChIKey
get_cid("BPGDAMSIGCZZLK-UHFFFAOYSA-N", from = "inchikey")
# from formula
get_cid("C26H52NO6P", from = "formula")
# from CAS RN
get_cid("56-40-6", from = "xref/rn")
# similarity
get_cid(5564, from = "similarity/cid")
get_cid("CCO", from = "similarity/smiles")
# from SID
get_cid("126534046", from = "sid", domain = "substance")
# sourceid
get_cid("VCC957895", from = "sourceid/23706", domain = "substance")
# sourceall
get_cid("Optopharma Ltd", from = "sourceall", domain = "substance")
# from AID (CIDs of substances tested in the assay)
get_cid(170004, from = "aid", domain = "assay")
# from GeneID (CIDs of substances tested on the gene)
get_cid(25086, from = "target/geneid", domain = "assay")

# multiple inputs
get_cid(c("Triclosan", "Aspirin"))


## End(Not run)
</code></pre>

<hr>
<h2 id='get_csid'>ChemSpider ID from compound name, formula, SMILES, InChI or InChIKey</h2><span id='topic+get_csid'></span>

<h3>Description</h3>

<p>Query one or more compunds by name, formula, SMILES, InChI or InChIKey and
return a vector of ChemSpider IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_csid(
  query,
  from = c("name", "formula", "inchi", "inchikey", "smiles"),
  match = c("all", "first", "ask", "na"),
  verbose = getOption("verbose"),
  apikey = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_csid_+3A_query">query</code></td>
<td>
<p>character; search term.</p>
</td></tr>
<tr><td><code id="get_csid_+3A_from">from</code></td>
<td>
<p>character; the type of the identifier to convert from. Valid
values are <code>"name"</code>, <code>"formula"</code>, <code>"smiles"</code>,
<code>"inchi"</code>, <code>"inchikey"</code>. The default value is <code>"name"</code>.</p>
</td></tr>
<tr><td><code id="get_csid_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handled?, &quot;all&quot; all
matches are returned, &quot;best&quot; the best matching is returned, &quot;ask&quot; enters an
interactive mode and the user is asked for input, &quot;na&quot; returns NA if
multiple hits are found.</p>
</td></tr>
<tr><td><code id="get_csid_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="get_csid_+3A_apikey">apikey</code></td>
<td>
<p>character; your API key. If NULL (default),
<code>cs_check_key()</code> will look for it in .Renviron or .Rprofile.</p>
</td></tr>
<tr><td><code id="get_csid_+3A_...">...</code></td>
<td>
<p>furthrer arguments passed to <code><a href="#topic+cs_control">cs_control</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Queries by SMILES, InChI or InChiKey do not use <code>cs_control</code>
options. Queries by name use <code>order_by</code> and <code>order_direction</code>.
Queries by formula also use <code>datasources</code>. See <code>cs_control()</code> for
a full list of valid values for these control options.
</p>
<p><code>formula</code> can be expressed with and without LaTeX syntax.
</p>


<h3>Value</h3>

<p>Returns a tibble.
</p>


<h3>Note</h3>

<p>An API key is needed. Register at <a href="https://developer.rsc.org/">https://developer.rsc.org/</a> for
an API key. Please respect the Terms &amp; conditions:
<a href="https://developer.rsc.org/terms">https://developer.rsc.org/terms</a>.
</p>


<h3>References</h3>

<p><a href="https://developer.rsc.org/docs/compounds-v1-trial/1/overview">https://developer.rsc.org/docs/compounds-v1-trial/1/overview</a>
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_csid("triclosan")
get_csid(c("carbamazepine", "naproxene","oxygen"))
get_csid("C2H6O", from = "formula")
get_csid("C_{2}H_{6}O", from = "formula")
get_csid("CC(O)=O", from = "smiles")
get_csid("InChI=1S/C2H4O2/c1-2(3)4/h1H3,(H,3,4)", from = "inchi")
get_csid("QTBSBXVTEAMEQO-UHFFFAOYAR", from = "inchikey")

## End(Not run)
</code></pre>

<hr>
<h2 id='get_etoxid'>Get ETOX ID</h2><span id='topic+get_etoxid'></span>

<h3>Description</h3>

<p>Query ETOX: Information System Ecotoxicology and Environmental Quality
Targets <a href="https://webetox.uba.de/webETOX/index.do">https://webetox.uba.de/webETOX/index.do</a> for their substance ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_etoxid(
  query,
  from = c("name", "cas", "ec", "gsbl", "rtecs"),
  match = c("all", "best", "first", "ask", "na"),
  verbose = getOption("verbose")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_etoxid_+3A_query">query</code></td>
<td>
<p>character; The searchterm</p>
</td></tr>
<tr><td><code id="get_etoxid_+3A_from">from</code></td>
<td>
<p>character; Type of input, can be one of &quot;name&quot; (chemical name),
&quot;cas&quot; (CAS Number), &quot;ec&quot; (European Community number for regulatory purposes),
&quot;gsbl&quot; (Identifier used by <a href="https://www.chemikalieninfo.de/">https://www.chemikalieninfo.de/</a>) and &quot;rtecs&quot;
(Identifier used by the Registry of Toxic Effects of Chemical Substances
database).</p>
</td></tr>
<tr><td><code id="get_etoxid_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handeled? &quot;all&quot; returns
all matched IDs, &quot;first&quot; only the first match, &quot;best&quot; the best matching (by
name) ID, &quot;ask&quot; is a interactive mode and the user is asked for input, &quot;na&quot;
returns <code>NA</code> if multiple hits are found.</p>
</td></tr>
<tr><td><code id="get_etoxid_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble with 3 columns: the query, the match, and the etoxID
</p>


<h3>Note</h3>

<p>Before using this function, please read the disclaimer
<a href="https://webetox.uba.de/webETOX/disclaimer.do">https://webetox.uba.de/webETOX/disclaimer.do</a>.
</p>


<h3>References</h3>

<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+etox_basic">etox_basic</a></code> for basic information,
<code><a href="#topic+etox_targets">etox_targets</a></code> for quality targets and
<code><a href="#topic+etox_tests">etox_tests</a></code> for test results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
get_etoxid("Triclosan")
# multiple inputs
comps &lt;- c("Triclosan", "Glyphosate")
get_etoxid(comps)
get_etoxid(comps, match = "all")
get_etoxid("34123-59-6", from = "cas") # Isoproturon
get_etoxid("133483", from = "gsbl") # 3-Butin-1-ol
get_etoxid("203-157-5", from = "ec") # Paracetamol

## End(Not run)
</code></pre>

<hr>
<h2 id='get_wdid'>Get Wikidata Item ID</h2><span id='topic+get_wdid'></span>

<h3>Description</h3>

<p>Search www.wikidata.org for wikidata item identifiers.  Note that this search
is currently not limited to chemical substances, so be sure to check your
results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wdid(
  query,
  match = c("best", "first", "all", "ask", "na"),
  verbose = getOption("verbose"),
  language = "en"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_wdid_+3A_query">query</code></td>
<td>
<p>character; The searchterm</p>
</td></tr>
<tr><td><code id="get_wdid_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handeled? 'all' returns
all matched IDs, 'first' only the first match, 'best' the best matching (by
name) ID, 'ask' is a interactive mode and the user is asked for input, na'
returns NA if multiple hits are found.</p>
</td></tr>
<tr><td><code id="get_wdid_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
<tr><td><code id="get_wdid_+3A_language">language</code></td>
<td>
<p>character; the language to search in</p>
</td></tr>
</table>


<h3>Value</h3>

<p>if match = 'all' a list with ids, otherwise a dataframe with 4 columns:
id, matched text, string distance to match and the queried string
</p>


<h3>Note</h3>

<p>Only matches in labels are returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_wdid('Triclosan', language = 'de')
get_wdid('DDT')
get_wdid('DDT', match = 'all')

# multiple inputs
comps &lt;- c('Triclosan', 'Glyphosate')
get_wdid(comps)

## End(Not run)
</code></pre>

<hr>
<h2 id='is.cas'>Check if input is a valid CAS</h2><span id='topic+is.cas'></span>

<h3>Description</h3>

<p>This function checks if a string is a valid CAS registry number.
A valid CAS is 1) separated by two hyphes into three parts; 2) the first part
consists from two up to seven digits; 3) the second of two digits; 4) the
third of one digit (check digit); 5) the check digits corresponds the
checksum. The checksum is found by taking the last digit (excluding the check
digit) multiplyingit with 1, the second last multiplied with 2, the
third-last multiplied with 3 etc. The modulo 10 of the sum of these is the
checksum.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.cas(x, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.cas_+3A_x">x</code></td>
<td>
<p>character; input CAS</p>
</td></tr>
<tr><td><code id="is.cas_+3A_verbose">verbose</code></td>
<td>
<p>logical; print messages during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical
</p>


<h3>Note</h3>

<p>This function can only handle one CAS string
</p>


<h3>References</h3>

<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is.cas('64-17-5')
is.cas('64175')
is.cas('4-17-5')
is.cas('64-177-6')
is.cas('64-17-55')
is.cas('64-17-6')
</code></pre>

<hr>
<h2 id='is.inchikey'>Check if input is a valid inchikey</h2><span id='topic+is.inchikey'></span>

<h3>Description</h3>

<p>This function checks if a string is a valid inchikey.
Inchikey must fulfill the following criteria:
1) consist of 27 characters;
2) be all uppercase, all letters (no numbers);
3) contain two hyphens at positions 15 and 26;
4) 24th character (flag character) be 'S' (Standard InChI) or 'N' (non-standard)
5) 25th character (version character) must be 'A' (currently).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.inchikey(
  x,
  type = c("format", "chemspider"),
  verbose = getOption("verbose")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.inchikey_+3A_x">x</code></td>
<td>
<p>character; input InChIKey</p>
</td></tr>
<tr><td><code id="is.inchikey_+3A_type">type</code></td>
<td>
<p>character; How should be checked? Either, by format (see above)
('format') or by ChemSpider ('chemspider').</p>
</td></tr>
<tr><td><code id="is.inchikey_+3A_verbose">verbose</code></td>
<td>
<p>logical; print messages during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical
</p>


<h3>Note</h3>

<p>This function can handle only one inchikey string.
</p>


<h3>References</h3>

<p>Heller, Stephen R., et al. &quot;InChI, the IUPAC International
Chemical Identifier.&quot; Journal of Cheminformatics 7.1 (2015): 23.
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is.inchikey('BQJCRHHNABKAKU-KBQPJGBKSA-N')
is.inchikey('BQJCRHHNABKAKU-KBQPJGBKSA')
is.inchikey('BQJCRHHNABKAKU-KBQPJGBKSA-5')
is.inchikey('BQJCRHHNABKAKU-KBQPJGBKSA-n')
is.inchikey('BQJCRHHNABKAKU/KBQPJGBKSA/N')
is.inchikey('BQJCRHHNABKAKU-KBQPJGBKXA-N')
is.inchikey('BQJCRHHNABKAKU-KBQPJGBKSB-N')
</code></pre>

<hr>
<h2 id='is.inchikey_cs'>Check if input is a valid inchikey using ChemSpider API</h2><span id='topic+is.inchikey_cs'></span>

<h3>Description</h3>

<p>Check if input is a valid inchikey using ChemSpider API
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.inchikey_cs(x, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.inchikey_cs_+3A_x">x</code></td>
<td>
<p>character; input string</p>
</td></tr>
<tr><td><code id="is.inchikey_cs_+3A_verbose">verbose</code></td>
<td>
<p>logical; print messages during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical
</p>


<h3>See Also</h3>

<p><code><a href="#topic+is.inchikey">is.inchikey</a></code> for a pure-R implementation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
is.inchikey_cs('BQJCRHHNABKAKU-KBQPJGBKSA-N')
is.inchikey_cs('BQJCRHHNABKAKU-KBQPJGBKSA')
is.inchikey_cs('BQJCRHHNABKAKU-KBQPJGBKSA-5')
is.inchikey_cs('BQJCRHHNABKAKU-KBQPJGBKSA-n')
is.inchikey_cs('BQJCRHHNABKAKU/KBQPJGBKSA/N')
is.inchikey_cs('BQJCRHHNABKAKU-KBQPJGBKXA-N')
is.inchikey_cs('BQJCRHHNABKAKU-KBQPJGBKSB-N')

## End(Not run)
</code></pre>

<hr>
<h2 id='is.inchikey_format'>Check if input is a valid inchikey using format</h2><span id='topic+is.inchikey_format'></span>

<h3>Description</h3>

<p>Inchikey must fulfill the following criteria:
1) consist of 27 characters;
2) be all uppercase, all letters (no numbers);
3) contain two hyphens at positions 15 and 26;
4) 24th character (flag character) be 'S' (Standard InChI) or 'N'
(non-standard)
5) 25th character (version character) must be 'A' (currently).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.inchikey_format(x, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.inchikey_format_+3A_x">x</code></td>
<td>
<p>character; input string</p>
</td></tr>
<tr><td><code id="is.inchikey_format_+3A_verbose">verbose</code></td>
<td>
<p>logical; print messages during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical
</p>


<h3>See Also</h3>

<p><code><a href="#topic+is.inchikey">is.inchikey</a></code> for a pure-R implementation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
is.inchikey_format('BQJCRHHNABKAKU-KBQPJGBKSA-N')
is.inchikey_format('BQJCRHHNABKAKU-KBQPJGBKSA')
is.inchikey_format('BQJCRHHNABKAKU-KBQPJGBKSA-5')
is.inchikey_format('BQJCRHHNABKAKU-KBQPJGBKSA-n')
is.inchikey_format('BQJCRHHNABKAKU/KBQPJGBKSA/N')
is.inchikey_format('BQJCRHHNABKAKU-KBQPJGBKXA-N')
is.inchikey_format('BQJCRHHNABKAKU-KBQPJGBKSB-N')

## End(Not run)
</code></pre>

<hr>
<h2 id='is.smiles'>Check if input is a SMILES string</h2><span id='topic+is.smiles'></span>

<h3>Description</h3>

<p>This function checks if a string is a valid SMILES by checking
if (R)CDK can parse it. If it cannot be parsed by rcdk FALSE is returned,
else TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.smiles(x, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is.smiles_+3A_x">x</code></td>
<td>
<p>character; input SMILES.</p>
</td></tr>
<tr><td><code id="is.smiles_+3A_verbose">verbose</code></td>
<td>
<p>logical; print messages during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical
</p>


<h3>Note</h3>

<p>This function can handle only one SMILES string.
</p>


<h3>References</h3>

<p>Egon Willighagen (2015). How to test SMILES strings in
Supplementary Information.
<a href="https://chem-bla-ics.blogspot.nl/2015/10/how-to-test-smiles-strings-in.html">https://chem-bla-ics.blogspot.nl/2015/10/how-to-test-smiles-strings-in.html</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if rcdk is not working properly
is.smiles('Clc(c(Cl)c(Cl)c1C(=O)O)c(Cl)c1Cl')
is.smiles('Clc(c(Cl)c(Cl)c1C(=O)O)c(Cl)c1ClJ')

## End(Not run)
</code></pre>

<hr>
<h2 id='jagst'>Organic plant protection products in the river Jagst / Germany in 2013</h2><span id='topic+jagst'></span>

<h3>Description</h3>

<p>This dataset comprises environmental monitoring data of organic plant protection products
in the year 2013 in the river Jagst, Germany.
The data is publicly available and can be retrieved from the
LUBW Landesanstalt für Umwelt, Messungen und Naturschutz Baden-Württemberg.
It has been preprocessed and comprises measurements of 34 substances.
Substances without detects have been removed.
on 13 sampling occasions.
Values are given in ug/L.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jagst
</code></pre>


<h3>Format</h3>

<p>A data frame with 442 rows and 4 variables:
</p>

<dl>
<dt>date</dt><dd><p>sampling data</p>
</dd>
<dt>substance</dt><dd><p>substance names</p>
</dd>
<dt>value</dt><dd><p>concentration in ug/L</p>
</dd>
<dt>qual</dt><dd><p>qualifier, indicating values &lt; LOQ</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://udo.lubw.baden-wuerttemberg.de/public/pages/home/index.xhtml">https://udo.lubw.baden-wuerttemberg.de/public/pages/home/index.xhtml</a>
</p>

<hr>
<h2 id='lc50'>Acute toxicity data from U.S. EPA ECOTOX</h2><span id='topic+lc50'></span>

<h3>Description</h3>

<p>This dataset comprises acute ecotoxicity data of 124 insecticides.
The data is publicly available and can be retrieved from the EPA ECOTOX database
(<a href="https://cfpub.epa.gov/ecotox/">https://cfpub.epa.gov/ecotox/</a>)
It comprises acute toxicity data (D. magna, 48h, Laboratory, 48h) and has been
preprocessed (remove non-insecticides, aggregate multiple value, keep only numeric data etc).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lc50
</code></pre>


<h3>Format</h3>

<p>A data frame with 124 rows and 2 variables:
</p>

<dl>
<dt>cas</dt><dd><p>CAS registry number</p>
</dd>
<dt>value</dt><dd><p>LC50value</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://cfpub.epa.gov/ecotox/">https://cfpub.epa.gov/ecotox/</a>
</p>

<hr>
<h2 id='nist_ri'>Retrieve retention indices from NIST</h2><span id='topic+nist_ri'></span>

<h3>Description</h3>

<p>This function scrapes NIST for literature retention indices
given a query or vector of queries as input. The query can be a cas
number, IUPAC name, or International Chemical Identifier (<code>inchikey</code>),
according to the value of the <code>from</code> argument. Retention indices are
stored in tables by <code>type</code>, <code>polarity</code> and temperature program
(<code>temp_prog</code>). The function can take multiple arguments for these
parameters and will return any retention times matching the specified
criteria in a single table.
</p>
<p>If a non-cas query is provided, the function will try to resolve the query
by searching the NIST WebBook for a corresponding CAS number. If
<code>from == "name"</code>, phonetic spellings of Greek stereo-descriptors
(e.g. &quot;alpha&quot;, &quot;beta&quot;, &quot;gamma&quot;) will be automatically
converted to the corresponding letters to match the form used by NIST. If
a CAS number is found, it will be returned in a <code>tibble</code> with the
corresponding information from the NIST retention index database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nist_ri(
  query,
  from = c("cas", "inchi", "inchikey", "name"),
  type = c("kovats", "linear", "alkane", "lee"),
  polarity = c("polar", "non-polar"),
  temp_prog = c("isothermal", "ramp", "custom"),
  cas = NULL,
  verbose = getOption("verbose")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nist_ri_+3A_query">query</code></td>
<td>
<p>character; the search term</p>
</td></tr>
<tr><td><code id="nist_ri_+3A_from">from</code></td>
<td>
<p>character; type of search term. can be one of <code>"name"</code>,
<code>"inchi"</code>, <code>"inchikey"</code>, or <code>"cas"</code>. Using an identifier is
preferred to <code>"name"</code> since <code>NA</code> is returned in the event of
multiple matches to a query. Using an identifier other than a CAS number
will cause this function to run slower as CAS numbers are used as internal
identifiers by NIST.</p>
</td></tr>
<tr><td><code id="nist_ri_+3A_type">type</code></td>
<td>
<p>Retention index type: <code>"kovats"</code>, <code>"linear"</code>,
<code>"alkane"</code>, and/or <code>"lee"</code>. See details for more.</p>
</td></tr>
<tr><td><code id="nist_ri_+3A_polarity">polarity</code></td>
<td>
<p>Column polarity: <code>"polar"</code> and/or <code>"non-polar"</code>
to get RIs calculated for polar or non-polar columns.</p>
</td></tr>
<tr><td><code id="nist_ri_+3A_temp_prog">temp_prog</code></td>
<td>
<p>Temperature program: <code>"isothermal"</code>,
<code>"ramp"</code>, and/or <code>"custom"</code>.</p>
</td></tr>
<tr><td><code id="nist_ri_+3A_cas">cas</code></td>
<td>
<p>deprecated.  Use <code>query</code> instead.</p>
</td></tr>
<tr><td><code id="nist_ri_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The types of retention indices included in NIST include Kovats
(<code>"kovats"</code>), Van den Dool and Kratz (<code>"linear"</code>), normal alkane
(<code>"alkane"</code>), and Lee (<code>"lee"</code>). Details about how these are
calculated are available on the NIST website:
<a href="https://webbook.nist.gov/chemistry/gc-ri/">https://webbook.nist.gov/chemistry/gc-ri/</a>
</p>


<h3>Value</h3>

<p>returns a tibble of literature RIs with the following columns:
</p>

<ul>
<li><p><code>query</code> is the query provided to the NIST server
</p>
</li>
<li><p><code>cas</code> is the CAS number or unique record identified used by NIST
</p>
</li>
<li><p><code>RI</code> is retention index
</p>
</li>
<li><p><code>type</code> is the type of RI (e.g. &quot;kovats&quot;, &quot;linear&quot;, &quot;alkane&quot;, or &quot;lee&quot;)
</p>
</li>
<li><p><code>polarity</code> is the polarity of the column (either &quot;polar&quot; or &quot;non-polar&quot;)
</p>
</li>
<li><p><code>temp_prog</code> is the type of temperature program (e.g. &quot;isothermal&quot;, &quot;ramp&quot;, or &quot;custom&quot;)
</p>
</li>
<li><p><code>column</code> is the column type, e.g. &quot;capillary&quot;
</p>
</li>
<li><p><code>phase</code> is the stationary phase (column phase)
</p>
</li>
<li><p><code>length</code> is column length in meters
</p>
</li>
<li><p><code>gas</code> is the carrier gas used
</p>
</li>
<li><p><code>substrate</code>
</p>
</li>
<li><p><code>diameter</code> is the column diameter in mm
</p>
</li>
<li><p><code>thickness</code> is the phase thickness in µm
</p>
</li>
<li><p><code>program</code>. various columns depending on the value of
<code>temp_prog</code>
</p>
</li>
<li><p><code>reference</code> is where this retention index was published
</p>
</li>
<li><p><code>comment</code>. I believe this denotes the database these data
were aggregated from
</p>
</li></ul>



<h3>Note</h3>

<p>Copyright for NIST Standard Reference Data is governed by the Standard
Reference Data Act, <a href="https://www.nist.gov/srd/public-law">https://www.nist.gov/srd/public-law</a>.
</p>


<h3>References</h3>

<p>NIST Mass Spectrometry Data Center, William E. Wallace, director,
&quot;Retention Indices&quot; in NIST Chemistry WebBook, NIST Standard Reference
Database Number 69, Eds. P.J. Linstrom and W.G. Mallard,
National Institute of Standards and Technology, Gaithersburg MD, 20899,
<a href="https://doi.org/10.18434/T4D303">doi:10.18434/T4D303</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+is.cas">is.cas</a></code> <code><a href="#topic+as.cas">as.cas</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
myRIs &lt;-
  nist_ri(
    c("78-70-6", "13474-59-4"),
    from = "cas",
    type = c("linear", "kovats"),
    polarity = "non-polar",
    temp_prog = "ramp"
  )
myRIs
## End(Not run)
</code></pre>

<hr>
<h2 id='opsin_query'>OPSIN web interface</h2><span id='topic+opsin_query'></span>

<h3>Description</h3>

<p>Query the OPSIN  (Open Parser for Systematic IUPAC nomenclature) web service
<a href="https://opsin.ch.cam.ac.uk/instructions.html">https://opsin.ch.cam.ac.uk/instructions.html</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opsin_query(query, verbose = getOption("verbose"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="opsin_query_+3A_query">query</code></td>
<td>
<p>character;  chemical name that should be queryed.</p>
</td></tr>
<tr><td><code id="opsin_query_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="opsin_query_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble with six columnns: &quot;query&quot;, inchi&quot;, &quot;stdinchi&quot;, &quot;stdinchikey&quot;, &quot;smiles&quot;, &quot;message&quot;, and &quot;status&quot;
</p>


<h3>References</h3>

<p>Lowe, D. M., Corbett, P. T., Murray-Rust, P., &amp; Glen, R. C. (2011).
Chemical Name to Structure: OPSIN, an Open Source Solution. Journal of Chemical Information and Modeling,
51(3), 739–753. <a href="https://doi.org/10.1021/ci100384d">doi:10.1021/ci100384d</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
opsin_query('Cyclopropane')
opsin_query(c('Cyclopropane', 'Octane'))
opsin_query(c('Cyclopropane', 'Octane', 'xxxxx'))

## End(Not run)
</code></pre>

<hr>
<h2 id='parse_mol'>Parse Molfile (as returned by ChemSpider) into a R-object.</h2><span id='topic+parse_mol'></span>

<h3>Description</h3>

<p>Parse Molfile (as returned by ChemSpider) into a R-object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parse_mol(string)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parse_mol_+3A_string">string</code></td>
<td>
<p>molfile as one string</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with of four entries: header (eh), counts line (cl), atom
block (ab) and bond block (bb).
</p>
<p>header: a = number of atoms, b = number of bonds, l = number of atom lists,
f = obsolete, c = chiral flag (0=not chiral, 1 = chiral), s = number of stext
entries, x, r, p, i = obsolete, m = 999, v0 version
</p>
<p>atom block: x, y, z = atom coordinates, a = mass difference, c= charge,
s= stereo parity, h = hydrogen count 1, b = stereo care box, v = valence,
h = h0 designator, r, i = not used, m = atom-atom mapping number,
n = inversion/retention flag, e = exact change flag
</p>
<p>bond block:
1 = first atom, 2 = second atom, t = bond type, s = stereo type, x = not
used, r = bond typology, c = reacting center status.
</p>


<h3>References</h3>

<p>Grabner, M., Varmuza, K., &amp; Dehmer, M. (2012). RMol:
a toolset for transforming SD/Molfile structure information into R objects.
Source Code for Biology and Medicine, 7, 12.
<a href="https://doi.org/10.1186/1751-0473-7-12">doi:10.1186/1751-0473-7-12</a>
</p>

<hr>
<h2 id='pc_prop'>Retrieve compound properties from a pubchem CID</h2><span id='topic+pc_prop'></span>

<h3>Description</h3>

<p>Retrieve compound information from pubchem CID, see
<a href="https://pubchem.ncbi.nlm.nih.gov/">https://pubchem.ncbi.nlm.nih.gov/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pc_prop(cid, properties = NULL, verbose = getOption("verbose"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pc_prop_+3A_cid">cid</code></td>
<td>
<p>character; Pubchem ID (CID).</p>
</td></tr>
<tr><td><code id="pc_prop_+3A_properties">properties</code></td>
<td>
<p>character vector; properties to retrieve, e.g.
c(&quot;MolecularFormula&quot;, &quot;MolecularWeight&quot;). If NULL (default) all available
properties are retrieved. See
<a href="https://pubchem.ncbi.nlm.nih.gov/docs/pug-rest">https://pubchem.ncbi.nlm.nih.gov/docs/pug-rest</a>
for a list of all available properties.</p>
</td></tr>
<tr><td><code id="pc_prop_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed to the console?</p>
</td></tr>
<tr><td><code id="pc_prop_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>


<h3>Note</h3>

<p>Please respect the Terms and Conditions of the National Library of
Medicine, <a href="https://www.nlm.nih.gov/databases/download.html">https://www.nlm.nih.gov/databases/download.html</a> the data
usage policies of National Center for Biotechnology Information,
<a href="https://www.ncbi.nlm.nih.gov/home/about/policies/">https://www.ncbi.nlm.nih.gov/home/about/policies/</a>,
<a href="https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access">https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access</a>, and the data
usage policies of the indicidual data sources
<a href="https://pubchem.ncbi.nlm.nih.gov/sources/">https://pubchem.ncbi.nlm.nih.gov/sources/</a>.
</p>


<h3>References</h3>

<p>Wang, Y., J. Xiao, T. O. Suzek, et al. 2009 PubChem: A Public
Information System for
Analyzing Bioactivities of Small Molecules. Nucleic Acids Research 37:
623–633.
</p>
<p>Kim, Sunghwan, Paul A. Thiessen, Evan E. Bolton, et al. 2016
PubChem Substance and Compound Databases. Nucleic Acids Research 44(D1):
D1202–D1213.
</p>
<p>Kim, S., Thiessen, P. A., Bolton, E. E., &amp; Bryant, S. H. (2015).
PUG-SOAP and PUG-REST: web services for programmatic access to chemical
information in PubChem. Nucleic acids research, gkv396.
</p>
<p>Eduard Szöcs, Tamás Stirling, Eric R. Scott, Andreas Scharmüller,
Ralf B. Schäfer (2020). webchem: An R Package to Retrieve Chemical
Information from the Web. Journal of Statistical Software, 93(13).
<a href="https://doi.org/10.18637/jss.v093.i13">doi:10.18637/jss.v093.i13</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_cid">get_cid</a></code>, <code><a href="#topic+pc_sect">pc_sect</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
pc_prop(5564)

###
# multiple CIDS
comp &lt;- c("Triclosan", "Aspirin")
cids &lt;- get_cid(comp)
pc_prop(cids$cid, properties = c("MolecularFormula", "MolecularWeight",
"CanonicalSMILES"))

## End(Not run)
</code></pre>

<hr>
<h2 id='pc_sect'>Retrieve data from PubChem content pages</h2><span id='topic+pc_sect'></span>

<h3>Description</h3>

<p>When you search for an entity at <a href="https://pubchem.ncbi.nlm.nih.gov/">https://pubchem.ncbi.nlm.nih.gov/</a>,
e.g. a compound or a substance, and select the record you are interested in,
you will be forwarded to a PubChem content page. When you look at a PubChem
content page, you can see that chemical information is organised into
sections, subsections, etc. The chemical data live at the lowest levels of
these sections. Use this function to retrieve the lowest level information
from PubChem content pages.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pc_sect(
  id,
  section,
  domain = c("compound", "substance", "assay", "gene", "protein", "patent"),
  verbose = getOption("verbose")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pc_sect_+3A_id">id</code></td>
<td>
<p>numeric or character; a vector of PubChem identifiers to search
for.</p>
</td></tr>
<tr><td><code id="pc_sect_+3A_section">section</code></td>
<td>
<p>character; the section of the content page to be imported.</p>
</td></tr>
<tr><td><code id="pc_sect_+3A_domain">domain</code></td>
<td>
<p>character; the query domain. Can be one of <code>"compound"</code>,
<code>"substance"</code>, <code>"assay"</code>, <code>"gene"</code>, <code>"protein"</code> or
<code>"patent"</code>.</p>
</td></tr>
<tr><td><code id="pc_sect_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>section</code> is not case sensitive but it is sensitive to typing
errors and it requires the full name of the section as it is printed on the
content page. The PubChem Table of Contents Tree can also be found at
<a href="https://pubchem.ncbi.nlm.nih.gov/classification/#hid=72">https://pubchem.ncbi.nlm.nih.gov/classification/#hid=72</a>.
</p>


<h3>Value</h3>

<p>Returns a tibble of query results. In the returned tibble,
<code>SourceName</code> is the name of the depositor, and <code>SourceID</code> is the
ID of the search term within the depositor's database. You can browse
<a href="https://pubchem.ncbi.nlm.nih.gov/sources/">https://pubchem.ncbi.nlm.nih.gov/sources/</a> for more information about
the depositors.
</p>


<h3>Note</h3>

<p>Please respect the Terms and Conditions of the National Library of
Medicine, <a href="https://www.nlm.nih.gov/databases/download.html">https://www.nlm.nih.gov/databases/download.html</a> the data
usage policies of National Center for Biotechnology Information,
<a href="https://www.ncbi.nlm.nih.gov/home/about/policies/">https://www.ncbi.nlm.nih.gov/home/about/policies/</a>,
<a href="https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access">https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access</a>, and the data
usage policies of the individual data sources
<a href="https://pubchem.ncbi.nlm.nih.gov/sources/">https://pubchem.ncbi.nlm.nih.gov/sources/</a>.
</p>


<h3>References</h3>

<p>Kim, S., Thiessen, P.A., Cheng, T. et al. PUG-View: programmatic
access to chemical annotations integrated in PubChem. J Cheminform 11, 56
(2019). <a href="https://doi.org/10.1186/s13321-019-0375-2">doi:10.1186/s13321-019-0375-2</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_cid">get_cid</a></code>, <code><a href="#topic+pc_prop">pc_prop</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># might fail if API is not available
## Not run: 
pc_sect(176, "Dissociation Constants")
pc_sect(c(176, 311), "density")
pc_sect(2231, "depositor-supplied synonyms", "substance")
pc_sect(780286, "modify date", "assay")
pc_sect(9023, "Ensembl ID", "gene")
pc_sect("1ZHY_A", "Sequence", "protein")

## End(Not run)
</code></pre>

<hr>
<h2 id='pc_synonyms'>Search synonyms in pubchem</h2><span id='topic+pc_synonyms'></span>

<h3>Description</h3>

<p>Search synonyms using PUG-REST,
see <a href="https://pubchem.ncbi.nlm.nih.gov/">https://pubchem.ncbi.nlm.nih.gov/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pc_synonyms(
  query,
  from = c("name", "cid", "sid", "aid", "smiles", "inchi", "inchikey"),
  match = c("all", "first", "ask", "na"),
  verbose = getOption("verbose"),
  arg = NULL,
  choices = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pc_synonyms_+3A_query">query</code></td>
<td>
<p>character; search term.</p>
</td></tr>
<tr><td><code id="pc_synonyms_+3A_from">from</code></td>
<td>
<p>character; type of input, can be one of &quot;name&quot; (default), &quot;cid&quot;,
&quot;sid&quot;, &quot;aid&quot;, &quot;smiles&quot;, &quot;inchi&quot;, &quot;inchikey&quot;</p>
</td></tr>
<tr><td><code id="pc_synonyms_+3A_match">match</code></td>
<td>
<p>character; How should multiple hits be handled? <code>"all"</code>
returns all matches, <code>"first"</code> returns only the first result,
<code>"ask"</code> enters an interactive mode and the user is asked for input,
<code>"na"</code> returns <code>NA</code> if multiple hits are found.</p>
</td></tr>
<tr><td><code id="pc_synonyms_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="pc_synonyms_+3A_arg">arg</code></td>
<td>
<p>character; optional arguments like &quot;name_type=word&quot; to match
individual words.</p>
</td></tr>
<tr><td><code id="pc_synonyms_+3A_choices">choices</code></td>
<td>
<p>deprecated.  Use the <code>match</code> argument instead.</p>
</td></tr>
<tr><td><code id="pc_synonyms_+3A_...">...</code></td>
<td>
<p>currently unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list.
</p>


<h3>Note</h3>

<p>Please respect the Terms and Conditions of the National Library of
Medicine, <a href="https://www.nlm.nih.gov/databases/download.html">https://www.nlm.nih.gov/databases/download.html</a> the data
usage policies of National Center for Biotechnology Information,
<a href="https://www.ncbi.nlm.nih.gov/home/about/policies/">https://www.ncbi.nlm.nih.gov/home/about/policies/</a>,
<a href="https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access">https://pubchem.ncbi.nlm.nih.gov/docs/programmatic-access</a>, and the data
usage policies of the indicidual data sources
<a href="https://pubchem.ncbi.nlm.nih.gov/sources/">https://pubchem.ncbi.nlm.nih.gov/sources/</a>.
</p>


<h3>References</h3>

<p>Wang, Y., J. Xiao, T. O. Suzek, et al. 2009 PubChem: A Public
Information System for
Analyzing Bioactivities of Small Molecules. Nucleic Acids Research 37:
623–633.
</p>
<p>Kim, Sunghwan, Paul A. Thiessen, Evan E. Bolton, et al. 2016
PubChem Substance and Compound Databases. Nucleic Acids Research 44(D1):
D1202–D1213.
</p>
<p>Kim, S., Thiessen, P. A., Bolton, E. E., &amp; Bryant, S. H. (2015).
PUG-SOAP and PUG-REST: web services for programmatic access to chemical
information in PubChem. Nucleic acids research, gkv396.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pc_synonyms("Aspirin")
pc_synonyms(c("Aspirin", "Triclosan"))
pc_synonyms(5564, from = "cid")
pc_synonyms(c("Aspirin", "Triclosan"), match = "ask")

## End(Not run)
</code></pre>

<hr>
<h2 id='ping_service'>Ping an API used in webchem to see if it's working.</h2><span id='topic+ping_service'></span>

<h3>Description</h3>

<p>Ping an API used in webchem to see if it's working.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ping_service(
  service = c("bcpc", "chebi", "chembl", "cs", "cs_web", "cir", "cts", "etox", "fn",
    "nist", "opsin", "pc", "srs", "wd"),
  apikey = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ping_service_+3A_service">service</code></td>
<td>
<p>character; the same abbreviations used as prefixes in
<code>webchem</code> functions, with the exception of <code>"cs_web"</code>, which only
checks if the ChemSpider website is up, and thus doesn't require an API key.</p>
</td></tr>
<tr><td><code id="ping_service_+3A_apikey">apikey</code></td>
<td>
<p>character; API key for services that require API keys</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical, TRUE if the service is available or FALSE if it isn't
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ping_service("chembl")

## End(Not run)
</code></pre>

<hr>
<h2 id='srs_query'>Get record details from U.S. EPA Substance Registry Servives (SRS)</h2><span id='topic+srs_query'></span>

<h3>Description</h3>

<p>Get record details from SRS, see <a href="https://cdxnodengn.epa.gov/cdx-srs-rest/">https://cdxnodengn.epa.gov/cdx-srs-rest/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>srs_query(
  query,
  from = c("itn", "cas", "epaid", "tsn", "name"),
  verbose = getOption("verbose"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="srs_query_+3A_query">query</code></td>
<td>
<p>character; query ID.</p>
</td></tr>
<tr><td><code id="srs_query_+3A_from">from</code></td>
<td>
<p>character; type of query ID, e.g. <code>'itn'</code> , <code>'cas'</code>,
<code>'epaid'</code>, <code>'tsn'</code>, <code>'name'</code>.</p>
</td></tr>
<tr><td><code id="srs_query_+3A_verbose">verbose</code></td>
<td>
<p>logical; should a verbose output be printed on the console?</p>
</td></tr>
<tr><td><code id="srs_query_+3A_...">...</code></td>
<td>
<p>not currently used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of lists (for each supplied query): a list of 22. subsKey,
internalTrackingNumber, systematicName, epaIdentificationNumber,
currentCasNumber, currentTaxonomicSerialNumber, epaName, substanceType,
categoryClass, kingdomCode, iupacName, pubChemId, molecularWeight,
molecularFormula, inchiNotation, smilesNotation, classifications,
characteristics, synonyms, casNumbers, taxonomicSerialNumbers, relationships
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# might fail if API is not available
srs_query(query = '50-00-0', from = 'cas')

### multiple inputs
casrn &lt;- c('50-00-0', '67-64-1')
srs_query(query = casrn, from = 'cas')

## End(Not run)
</code></pre>

<hr>
<h2 id='wd_ident'>Retrieve identifiers from Wikidata</h2><span id='topic+wd_ident'></span>

<h3>Description</h3>

<p>Retrieve identifiers from Wikidata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wd_ident(id, verbose = getOption("verbose"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wd_ident_+3A_id">id</code></td>
<td>
<p>character; identifier, as returned by <code><a href="#topic+get_wdid">get_wdid</a></code></p>
</td></tr>
<tr><td><code id="wd_ident_+3A_verbose">verbose</code></td>
<td>
<p>logical; print message during processing to console?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame of identifiers. Currently these are 'smiles', 'cas', 'cid', 'einecs', 'csid', 'inchi', 'inchikey',
'drugbank', 'zvg', 'chebi', 'chembl', 'unii', 'lipidmaps', 'swisslipids' and source_url.
</p>


<h3>Note</h3>

<p>Only matches in labels are returned. If more than one unique hit is found,
only the first is returned.
</p>


<h3>References</h3>

<p>Willighagen, E., 2015. Getting CAS registry numbers out of WikiData. The Winnower.
<a href="https://doi.org/10.15200/winn.142867.72538">doi:10.15200/winn.142867.72538</a>
</p>
<p>Mitraka, Elvira, Andra Waagmeester, Sebastian Burgstaller-Muehlbacher, et al. 2015
Wikidata: A Platform for Data Integration and Dissemination for the Life Sciences and beyond. bioRxiv: 031971.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_wdid">get_wdid</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 id &lt;- c("Q408646", "Q18216")
 wd_ident(id)

## End(Not run)
</code></pre>

<hr>
<h2 id='webchem'>webchem: An R package to retrieve chemical information from the web.</h2><span id='topic+webchem'></span>

<h3>Description</h3>

<p>Chemical information from around the web. This package interacts with a suite
of web APIs for chemical information.
</p>

<hr>
<h2 id='webchem-defunct'>Defunct function(s) in the webchem package</h2><span id='topic+webchem-defunct'></span><span id='topic+ppdb_query'></span><span id='topic+ppdb_parse'></span><span id='topic+ppdb'></span><span id='topic+cir'></span><span id='topic+pp_query'></span><span id='topic+cs_prop'></span><span id='topic+ci_query'></span><span id='topic+pan_query'></span>

<h3>Description</h3>

<p>These functions are defunct and no longer available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ppdb_query()

ppdb_parse()

ppdb()

cir()

pp_query()

cs_prop()

ci_query()

pan_query()
</code></pre>

<hr>
<h2 id='webchem-deprecated'>Deprecated function(s) in the webchem package</h2><span id='topic+webchem-deprecated'></span><span id='topic+cid_compinfo'></span><span id='topic+aw_query'></span>

<h3>Description</h3>

<p>These functions are provided for compatibility with older version of
the webchem package.  They may eventually be completely
removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cid_compinfo(...)

aw_query(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="webchem-deprecated_+3A_...">...</code></td>
<td>
<p>Parameters to be passed to the modern version of the function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Deprecated functions are:
</p>

<table>
<tr>
 <td style="text-align: right;">
  <code>pc_prop</code> </td><td style="text-align: left;"> was formerly <code><a href="#topic+cid_compinfo">cid_compinfo</a></code></td>
</tr>
<tr>
 <td style="text-align: right;">
  <code>bcpc_query</code> </td><td style="text-align: left;"> was formerly <code><a href="#topic+aw_query">aw_query</a></code></td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>


<hr>
<h2 id='with_cts'>Auto-translate identifiers and search databases</h2><span id='topic+with_cts'></span>

<h3>Description</h3>

<p>Supply a query of any type (e.g. SMILES, CAS, name, InChI, etc.) along with
any webchem function that has <code>query</code> and <code>from</code> arguments.  If the
function doesn't accept the type of query you've supplied, this will try to
automatically translate it using CTS and run the query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>with_cts(query, from, .f, .verbose = getOption("verbose"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="with_cts_+3A_query">query</code></td>
<td>
<p>character; the search term</p>
</td></tr>
<tr><td><code id="with_cts_+3A_from">from</code></td>
<td>
<p>character; the format or type of query.  Commonly accepted values
are &quot;name&quot;, &quot;cas&quot;, &quot;inchi&quot;, and &quot;inchikey&quot;</p>
</td></tr>
<tr><td><code id="with_cts_+3A_.f">.f</code></td>
<td>
<p>character; the (quoted) name of a webchem function</p>
</td></tr>
<tr><td><code id="with_cts_+3A_.verbose">.verbose</code></td>
<td>
<p>logical; print a message when translating query?</p>
</td></tr>
<tr><td><code id="with_cts_+3A_...">...</code></td>
<td>
<p>other arguments passed to the function specified with <code>.f</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns results from <code>.f</code>
</p>


<h3>Note</h3>

<p>During the translation step, only the first hit from CTS is used.
Therefore, using this function to translate on the fly is not foolproof and
care should be taken to verify the results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
with_cts("XDDAORKBJWWYJS-UHFFFAOYSA-N", from = "inchikey", .f = "get_etoxid")

## End(Not run)
</code></pre>

<hr>
<h2 id='write_mol'>Export a Chemical Structure in .mol Format.</h2><span id='topic+write_mol'></span>

<h3>Description</h3>

<p>Some webchem functions return character strings that contain a chemical
structure in Mol format. This function exports a character string as a .mol
file so it can be imported with other chemistry software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_mol(x, file = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_mol_+3A_x">x</code></td>
<td>
<p>a character string of a chemical structure in mol format.</p>
</td></tr>
<tr><td><code id="write_mol_+3A_file">file</code></td>
<td>
<p>a character vector of file names</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# export Mol file
csid &lt;- get_csid("bergapten")
mol3d &lt;- cs_compinfo(csid$csid, field = "Mol3D")
write_mol(mol3d$mol3D, file = mol3d$id)

# export multiple Mol files
csids &lt;- get_csid(c("bergapten", "xanthotoxin"))
mol3ds &lt;- cs_compinfo(csids$csid, field = "Mol3D")
mapply(function(x, y) write_mol(x, y), x = mol3ds$mol3D, y = mol3ds$id)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
