<!DOCTYPE html><html lang="en"><head><title>Help for package WeightedTreemaps</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {WeightedTreemaps}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#as.data.frame.sunburstResult'><p>as.data.frame.sunburstResult</p></a></li>
<li><a href='#as.data.frame.voronoiResult'><p>as.data.frame.voronoiResult</p></a></li>
<li><a href='#cropped_voronoi'><p>cropped_voronoi</p></a></li>
<li><a href='#drawTreemap'><p>drawTreemap</p></a></li>
<li><a href='#get_polygons'><p>get_polygons</p></a></li>
<li><a href='#Jahn_CellReports_2018'><p>Data from the publication of Jahn et al., CellReports, 2018</p></a></li>
<li><a href='#poly_area'><p>poly_area</p></a></li>
<li><a href='#poly_centroid'><p>poly_centroid</p></a></li>
<li><a href='#poly_transform_shape'><p>poly_transform_shape</p></a></li>
<li><a href='#print.sunburstResult'><p>print.sunburstResult</p></a></li>
<li><a href='#print.voronoiResult'><p>print.voronoiResult</p></a></li>
<li><a href='#rounded_rect'><p>Coordinates to draw a rounded rectangle as parent cell for treemaps</p></a></li>
<li><a href='#summary.sunburstResult'><p>summary.sunburstResult</p></a></li>
<li><a href='#summary.voronoiResult'><p>summary.voronoiResult</p></a></li>
<li><a href='#sunburstTreemap'><p>sunburstTreemap</p></a></li>
<li><a href='#voronoiTreemap'><p>voronoiTreemap</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Generate and Plot Voronoi or Sunburst Treemaps from Hierarchical
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Treemaps are a visually appealing graphical representation of
    numerical data using a space-filling approach. A plane or 'map' is
    subdivided into smaller areas called cells.  The cells in the map are
    scaled according to an underlying metric which allows to grasp the
    hierarchical organization and relative importance of many objects at
    once. This package contains two different implementations of treemaps,
    Voronoi treemaps and Sunburst treemaps.  The Voronoi treemap function
    subdivides the plot area in polygonal cells according to the highest
    hierarchical level, then continues to subdivide those parental cells
    on the next lower hierarchical level, and so on. The Sunburst treemap
    is a computationally less demanding treemap that does not require
    iterative refinement, but simply generates circle sectors that are
    sized according to predefined weights.  The Voronoi tesselation is
    based on functions from Paul Murrell (2012)
    <a href="https://www.stat.auckland.ac.nz/~paul/Reports/VoronoiTreemap/voronoiTreeMap.html">https://www.stat.auckland.ac.nz/~paul/Reports/VoronoiTreemap/voronoiTreeMap.html</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/m-jahn/WeightedTreemaps">https://github.com/m-jahn/WeightedTreemaps</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/m-jahn/WeightedTreemaps/issues">https://github.com/m-jahn/WeightedTreemaps/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>colorspace, dplyr, grid, lattice, methods, Rcpp, scales, sf,
sp, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, parallel, rmarkdown</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>BH, Rcpp, RcppCGAL</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++17</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-12 08:31:44 UTC; michael</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Jahn <a href="https://orcid.org/0000-0002-3913-153X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  David Leslie [aut],
  Ahmadou Dicko <a href="https://orcid.org/0000-0002-9654-7582"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Dunipace Eric <a href="https://orcid.org/0000-0001-8909-213X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Paul Murrell <a href="https://orcid.org/0000-0002-3224-8858"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Jahn &lt;jahn@mpusp.mpg.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-12 09:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='as.data.frame.sunburstResult'>as.data.frame.sunburstResult</h2><span id='topic+as.data.frame.sunburstResult'></span>

<h3>Description</h3>

<p>Coerces a sunburstResult object to data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sunburstResult'
as.data.frame(x, ..., stringsAsFactors = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.data.frame.sunburstResult_+3A_x">x</code></td>
<td>
<p>(sunburstResult) A sunburst treemap results object</p>
</td></tr>
<tr><td><code id="as.data.frame.sunburstResult_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
<tr><td><code id="as.data.frame.sunburstResult_+3A_stringsasfactors">stringsAsFactors</code></td>
<td>
<p>(logical) Transform strings to factors, default FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sunburstTreemap">sunburstTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='as.data.frame.voronoiResult'>as.data.frame.voronoiResult</h2><span id='topic+as.data.frame.voronoiResult'></span>

<h3>Description</h3>

<p>Coerces a voronoiResult object to data frame, omitting polygon data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'voronoiResult'
as.data.frame(x, ..., stringsAsFactors = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.data.frame.voronoiResult_+3A_x">x</code></td>
<td>
<p>(voronoiResult) A voronoi treemap results object</p>
</td></tr>
<tr><td><code id="as.data.frame.voronoiResult_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
<tr><td><code id="as.data.frame.voronoiResult_+3A_stringsasfactors">stringsAsFactors</code></td>
<td>
<p>(logical) Transform strings to factors, default FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='cropped_voronoi'>cropped_voronoi</h2><span id='topic+cropped_voronoi'></span>

<h3>Description</h3>

<p>Tesselates a plane using a set of XY coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cropped_voronoi(sites)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cropped_voronoi_+3A_sites">sites</code></td>
<td>
<p>(numeric matrix) The only input parameter for the function.
A matrix with 3 columns: X and Y coordinates, as well as weights
that are used for tesselation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is only intended for
internal use. However, one can also use it directly for
test purposes.
</p>


<h3>Value</h3>

<p>A list of cell coordinates; one cell for each
set of input coordinates.
</p>

<hr>
<h2 id='drawTreemap'>drawTreemap</h2><span id='topic+drawTreemap'></span>

<h3>Description</h3>

<p>Draws the treemap object that was obtained by running <code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> or
<code><a href="#topic+sunburstTreemap">sunburstTreemap</a></code>. Many graphical parameters can be customized but some
settings that determine the appearance of treemaps are already made
during treemap generation. Such parameters are primarily cell size and
initial shape of the treemap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawTreemap(
  treemap,
  levels = 1:length(treemap@call$levels),
  color_type = "categorical",
  color_level = NULL,
  color_palette = NULL,
  color_steps = 10,
  border_level = levels,
  border_size = 6,
  border_color = grey(0.9),
  label_level = max(levels),
  label_size = 1,
  label_color = grey(0.9),
  label_autoscale = TRUE,
  title = NULL,
  title_size = 1,
  title_color = grey(0.5),
  legend = FALSE,
  legend_position = "left",
  legend_size = 0.1,
  custom_range = NULL,
  width = 0.9,
  height = 0.9,
  layout = c(1, 1),
  position = c(1, 1),
  add = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="drawTreemap_+3A_treemap">treemap</code></td>
<td>
<p>(treemapResult) Either a <code>voronoiResult</code> or <code>sunburstResult</code>
object that contains polygons and metadata as output from running
<code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> or <code><a href="#topic+sunburstTreemap">sunburstTreemap</a></code>.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_levels">levels</code></td>
<td>
<p>(numeric) A numeric vector representing the hierarchical levels
that are drawn. The default is to draw all levels.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_color_type">color_type</code></td>
<td>
<p>(character) One of &quot;categorical&quot;, &quot;cell_size&quot;, &quot;both&quot;, or &quot;custom_color&quot;.
For &quot;categorical&quot;, each cell is colored based on the (parent) category it belongs.
Colors may repeat if there are many cells. For &quot;cell_size&quot;, cells are colored
according to their relative area. For &quot;both&quot;, cells are be colored the same
way as for &quot;categorical&quot;, but lightness is adjusted according to cell area.
For &quot;custom_color&quot;, a color index is used that was specified by
<code>custom_color</code> during treemap generation. Use <code>NULL</code> to omit drawing colors.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_color_level">color_level</code></td>
<td>
<p>(numeric) A numeric vector representing the hierarchical level
that should be used for cell coloring. Must be one of <code>levels</code>.
Default is to use the lowest level cells for Voronoi treemaps and all levels
for sunburst treemaps.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_color_palette">color_palette</code></td>
<td>
<p>(character) A character vector of colors used to fill cells.
The default is to use <code><a href="colorspace.html#topic+rainbow_hcl">rainbow_hcl</a></code></p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_color_steps">color_steps</code></td>
<td>
<p>(numeric) Approximate number of steps for the color gradient
to be used when drawing cells with <code>color_type = "cell_size"</code>.
Default step number is 10, and final step number can a vary a bit because
<code>pretty()</code> is used to calculate a decent color range.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_border_level">border_level</code></td>
<td>
<p>(numeric) A numeric vector representing the hierarchical level that should be
used for drawing cell borders, or NULL to omit drawing borders, The default is
that all borders are drawn.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_border_size">border_size</code></td>
<td>
<p>(numeric) A single number indicating initial line width of the highest level
cells. Is reduced each level, default is 6 pts. Alternatively a vector of
<code>length(border_level)</code>, then each border is drawn with the specified width.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_border_color">border_color</code></td>
<td>
<p>(character) A single character indicating color for cell borders,
default is a light grey. Alternatively a vector of <code>length(border_level)</code>,
then each border is drawn with the specified color.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_label_level">label_level</code></td>
<td>
<p>(numeric) A numeric vector representing the hierarchical level that should be
used for drawing cell labels, or NULL to omit drawing labels. The default is the
deepest level (every cell has a label).</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_label_size">label_size</code></td>
<td>
<p>(numeric) A single number indicating relative size of each label
in relation to its parent cell. Alternatively a numeric vector of
<code>length(label_level)</code> that specifies relative size of labels for each level
individually.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_label_color">label_color</code></td>
<td>
<p>(character) A single character indicating color for cell labels.
Alternatively a vector of <code>length(label_level)</code>, then each label
is drawn with the specified color.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_label_autoscale">label_autoscale</code></td>
<td>
<p>(logical) Whether to automatically scale labels based on
their estimated width. Default is TRUE.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_title">title</code></td>
<td>
<p>(character) An optional title, default to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_title_size">title_size</code></td>
<td>
<p>(numeric) The size (or 'character expansion') of the title.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_title_color">title_color</code></td>
<td>
<p>(character) Color for title.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_legend">legend</code></td>
<td>
<p>(logical) Set to TRUE if a color key should be drawn. Default is FALSE.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_legend_position">legend_position</code></td>
<td>
<p>(character) The position of the legend, one of &quot;left&quot; (default),
&quot;right&quot;, &quot;top&quot;, or &quot;bottom&quot;.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_legend_size">legend_size</code></td>
<td>
<p>(numeric) The relative size of the legend (0 to 1), default is 0.1.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_custom_range">custom_range</code></td>
<td>
<p>(numeric) A numeric vector of length 2 that can be used
to rescale the values in <code>custom_color</code> to the range of choice.
The default is <code>NULL</code> and it only has an effect if <code>custom_color</code>
was specified when generating the treemap.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_width">width</code></td>
<td>
<p>(numeric) The width (0 to 0.9) of the viewport that the treemap will occupy.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_height">height</code></td>
<td>
<p>(numeric) The height (0 to 0.9) of the viewport that the treemap will occupy.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_layout">layout</code></td>
<td>
<p>(numeric) Vector of length 2 indicating the number of rows and columns
that the plotting area is supposed to be subdivided in. Useful only together with
<code>position</code>, which indicates the position of the specific treemap. Use <code>add = TRUE</code>
to omit starting a new page every time you call <code>drawTreemap()</code>.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_position">position</code></td>
<td>
<p>(numeric) Vector of length 2 indicating the position where the current
treemap should be drawn. Useful only together with <code>layout</code>, which indicates
the number of rows and columns the plotting area is subdivided into. Use <code>add = TRUE</code>
to omit starting a new page every time you call <code>drawTreemap()</code>.</p>
</td></tr>
<tr><td><code id="drawTreemap_+3A_add">add</code></td>
<td>
<p>(logical) Defaults to <code>FALSE</code>, creating a new page when drawing
a treemap. When multiple treemaps should be plotted on the same page, this should be
set to TRUE, and position of treemaps specified by <code>layout</code> and <code>position</code> arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function does not return a value (except NULL). It creates a grid viewport and
draws the treemap.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> for generating the treemap that is
the input for the drawing function
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load package
library(WeightedTreemaps)

# generate dummy data
df &lt;- data.frame(
  A = rep(c("abcd", "efgh"), each = 4),
  B = letters[1:8],
  size = c(37, 52, 58, 27, 49, 44, 34, 45)
)

# compute treemap
tm &lt;- voronoiTreemap(
  data = df,
  levels = c("B"),
  cell_size = "size",
  shape = "circle",
  positioning = "regular",
  seed = 123
)

# plot treemap with each cell colored by name (default)
drawTreemap(tm, label_size = 1, color_type = "categorical")

# plot treemap with each cell colored by name, but larger cells
# lighter and smaller cells darker
drawTreemap(tm, label_size = 1, color_type = "both")

# plot treemap with different color palette and style
drawTreemap(tm, label_size = 1, label_color = grey(0.3),
            border_color = grey(0.3), color_palette = heat.colors(6)
)

# ---------------------------------------------

# load example data
data(mtcars)
mtcars$car_name = gsub(" ", "\n", row.names(mtcars))

# generate sunburst treemap
tm &lt;- sunburstTreemap(
  data = mtcars,
  levels = c("gear", "cyl"),
  cell_size = "hp"
)

# draw treemap
drawTreemap(tm,
  title = "A sunburst treemap",
  legend = TRUE,
  border_size = 2,
  label_color = grey(0.6)
)

</code></pre>

<hr>
<h2 id='get_polygons'>get_polygons</h2><span id='topic+get_polygons'></span>

<h3>Description</h3>

<p>Extract the list of polygons from a voronoiResult object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_polygons(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_polygons_+3A_x">x</code></td>
<td>
<p>(voronoiResult) A voronoi treemap results object</p>
</td></tr>
<tr><td><code id="get_polygons_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of polygons, as class POLYGON from sf
</p>


<h3>See Also</h3>

<p><code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='Jahn_CellReports_2018'>Data from the publication of Jahn et al., CellReports, 2018</h2><span id='topic+Jahn_CellReports_2018'></span>

<h3>Description</h3>

<p>The dataset contains protein abundances of the Synechocystis sp.
PCC6803 proteome. Protein abundance was determined using shotgun
mass spectrometry. The data set also contains pathway information according to
the cyanobase hierarchical annotation:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Jahn_CellReports_2018)
</code></pre>


<h3>Format</h3>

<p>A data frame with 19790 rows and 12 variables
</p>


<h3>Details</h3>


<ul>
<li><p> protein - protein ID
</p>
</li>
<li><p> condition - combination from light and CO2
</p>
</li>
<li><p> light - light intensity in umol photons / m2 * s
</p>
</li>
<li><p> co2_concentration - CO2 concentration in 
</p>
</li>
<li><p> mean_intensity - mean MS1 ion intensity
</p>
</li>
<li><p> mean_mass_fraction_norm - normalized mean mass fraction of protein
</p>
</li>
<li><p> sd_intensity - standard deviation from mean
</p>
</li>
<li><p> Process - functional annotation 1st level
</p>
</li>
<li><p> Pathway - functional annotation 2nd level
</p>
</li>
<li><p> Protein - functional annotation 3rd level
</p>
</li>
<li><p> Process.abbr - abbreviated Process
</p>
</li>
<li><p> Pathway.abbr - abbreviated Pathway
</p>
</li>
<li><p> Gene.names - trivial names of genes, if available
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://pubmed.ncbi.nlm.nih.gov/30304686/">https://pubmed.ncbi.nlm.nih.gov/30304686/</a>
</p>

<hr>
<h2 id='poly_area'>poly_area</h2><span id='topic+poly_area'></span>

<h3>Description</h3>

<p>Mainly for internal use. Determines the area of a polygon based on
its x and y coordinates.
This function is a reimplementation of 'soiltexture::TT.polygon.area()'
and only re-implemented to avoid extra dependencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poly_area(poly_x, poly_y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="poly_area_+3A_poly_x">poly_x</code></td>
<td>
<p>(numeric)  X coordinates of each vertices of the polygon</p>
</td></tr>
<tr><td><code id="poly_area_+3A_poly_y">poly_y</code></td>
<td>
<p>(numeric)  Y coordinates of each vertices of the polygon</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric, the area of the polygon.
</p>


<h3>See Also</h3>

<p>soiltexture::TT.polygon.area()
</p>

<hr>
<h2 id='poly_centroid'>poly_centroid</h2><span id='topic+poly_centroid'></span>

<h3>Description</h3>

<p>Mainly for internal use. Determines the centroids of a polygon based on
its x and y coordinates.
This function is a reimplementation of 'soiltexture::TT.polygon.centroids()'
and only re-implemented to avoid extra dependencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poly_centroid(poly_x, poly_y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="poly_centroid_+3A_poly_x">poly_x</code></td>
<td>
<p>(numeric)  X coordinates of each vertices of the polygon</p>
</td></tr>
<tr><td><code id="poly_centroid_+3A_poly_y">poly_y</code></td>
<td>
<p>(numeric)  Y coordinates of each vertices of the polygon</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric, the area of the polygon.
</p>


<h3>See Also</h3>

<p>soiltexture::TT.polygon.centroids()
</p>

<hr>
<h2 id='poly_transform_shape'>poly_transform_shape</h2><span id='topic+poly_transform_shape'></span>

<h3>Description</h3>

<p>Mainly for internal use. The function transforms ('projects') arbitrary input
coordinates that are used as parent polygon or shape of the treemap.
By default it scales and centers polygon coordinates on a square
of 0 to 2000 units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>poly_transform_shape(poly, xy_max = 2000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="poly_transform_shape_+3A_poly">poly</code></td>
<td>
<p>(list)  named list of 'x' and 'y' coordinates of the polygon</p>
</td></tr>
<tr><td><code id="poly_transform_shape_+3A_xy_max">xy_max</code></td>
<td>
<p>(numeric)  maximum desired expansion of the polygon (default: 2000)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with slots 'x' and 'y' containing numeric coordinates
</p>

<hr>
<h2 id='print.sunburstResult'>print.sunburstResult</h2><span id='topic+print.sunburstResult'></span>

<h3>Description</h3>

<p>Print method for sunburstResult
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sunburstResult'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.sunburstResult_+3A_x">x</code></td>
<td>
<p>(sunburstResult) A sunburst treemap results object</p>
</td></tr>
<tr><td><code id="print.sunburstResult_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sunburstTreemap">sunburstTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='print.voronoiResult'>print.voronoiResult</h2><span id='topic+print.voronoiResult'></span>

<h3>Description</h3>

<p>Print method for voronoiResult
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'voronoiResult'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.voronoiResult_+3A_x">x</code></td>
<td>
<p>(voronoiResult) A voronoi treemap results object</p>
</td></tr>
<tr><td><code id="print.voronoiResult_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='rounded_rect'>Coordinates to draw a rounded rectangle as parent cell for treemaps</h2><span id='topic+rounded_rect'></span>

<h3>Description</h3>

<p>Set of coordinates for a rounded rectangle as parent cell for treemaps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rounded_rect)
</code></pre>


<h3>Format</h3>

<p>A data frame with 50 rows and 2 variables
</p>


<h3>Details</h3>


<ul>
<li><p> X - coordinate
</p>
</li>
<li><p> Y - coordinate
</p>
</li></ul>


<hr>
<h2 id='summary.sunburstResult'>summary.sunburstResult</h2><span id='topic+summary.sunburstResult'></span>

<h3>Description</h3>

<p>Summary method for sunburstResult
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sunburstResult'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.sunburstResult_+3A_object">object</code></td>
<td>
<p>(sunburstResult) A sunburst treemap results object</p>
</td></tr>
<tr><td><code id="summary.sunburstResult_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sunburstTreemap">sunburstTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='summary.voronoiResult'>summary.voronoiResult</h2><span id='topic+summary.voronoiResult'></span>

<h3>Description</h3>

<p>Summary method for voronoiResult.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'voronoiResult'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.voronoiResult_+3A_object">object</code></td>
<td>
<p>(voronoiResult) A voronoi treemap results object</p>
</td></tr>
<tr><td><code id="summary.voronoiResult_+3A_...">...</code></td>
<td>
<p>(none) Not used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data.frame
</p>


<h3>See Also</h3>

<p><code><a href="#topic+voronoiTreemap">voronoiTreemap</a></code> for generating the treemap that is
the input for this function
</p>

<hr>
<h2 id='sunburstTreemap'>sunburstTreemap</h2><span id='topic+sunburstTreemap'></span>

<h3>Description</h3>

<p>Create sunburst treemaps where variables are encoded by size of circular sectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sunburstTreemap(
  data,
  levels,
  fun = sum,
  sort = TRUE,
  filter = 0,
  cell_size = NULL,
  custom_color = NULL,
  diameter_inner = 0.3,
  diameter_outer = 0.8,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sunburstTreemap_+3A_data">data</code></td>
<td>
<p>(data.frame) A data.frame with one column for each hierarchical level</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_levels">levels</code></td>
<td>
<p>(character) Character vector indicating the column names to 
be used. The order of names must correspond to the hierarchical levels, 
going from broad to fine</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_fun">fun</code></td>
<td>
<p>(function) Function to be used to aggregate cell sizes of parental cells</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_sort">sort</code></td>
<td>
<p>(logical) Should the columns of the data.frame be sorted before?</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_filter">filter</code></td>
<td>
<p>(logical) Filter the supplied data frame to remove very small
sectors that may not be visible. The default is to not remove any sectors.</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_cell_size">cell_size</code></td>
<td>
<p>(character) The name of the column used to control sector size. 
Can be one of <code>levels</code> or any other column with numerical data. NA or
values equal or less than zero are not allowed.
The values in this column are aggregated by the function specified by <code>fun</code>.
If <code>sector_size = NULL</code>, sector size is simply computed by the number of members
for the respective sector (corresponding to rows in the data.frame).</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_custom_color">custom_color</code></td>
<td>
<p>(character) An optional column that can be specified to
control cell color. Cell colors are determined when drawing the treemap
using <code><a href="#topic+drawTreemap">drawTreemap</a></code>, but the default is to use one of 
<code>levels</code> or <code>cell size</code>. Any other data source that shall be used
instead has to be included in the treemap generation and explicitly 
specified here. The default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_diameter_inner">diameter_inner</code></td>
<td>
<p>(numeric) The minimum inner diameter of the drawn map. 
Defaults to 0.3,</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_diameter_outer">diameter_outer</code></td>
<td>
<p>(numeric) The maximum outer diameter of the drawn map. 
Defaults to 0.8</p>
</td></tr>
<tr><td><code id="sunburstTreemap_+3A_verbose">verbose</code></td>
<td>
<p>(logical) If verbose is TRUE (default is FALSE), basic information
such as a success message is printed to the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns a treemap object instead of a plot. In order 
to actually draw the treemap, use <code><a href="#topic+drawTreemap">drawTreemap</a></code>.
</p>


<h3>Value</h3>

<p>'sunburstTreemap' returns an object of the formal class 'sunburstResult'.
It is essentially a list of objects related to the graphical
representation of the treemap (polygons, labels, cell data) as well as data from the call
of the function. It contains the following slots:
</p>
<table role = "presentation">
<tr><td><code>cells</code></td>
<td>
<p>'list' of polygons for drawing a treemap</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>'data.frame', the original data that was supplied to calling 'voronoiTreemap'</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>'list' of arguments used to call 'voronoiTreemap'</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+drawTreemap">drawTreemap</a></code> for drawing the treemap.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example data
data(mtcars)
mtcars$car_name = gsub(" ", "\n", row.names(mtcars))

# generate treemap;
# by default cell (sector) size is encoded by number of members per group
tm &lt;- sunburstTreemap(
  data = mtcars,
  levels = c("gear", "cyl"),
  cell_size = "hp"
)

# draw treemap with default options
drawTreemap(tm,
  title = "A sunburst treemap",
  legend = TRUE,
  border_size = 2,
  layout = c(1, 3),
  position = c(1, 1)
)

# use custom color palette
drawTreemap(tm,
  title = "Use custom palette",
  legend = TRUE,
  color_palette = rep(c("#81E06E", "#E68CFF", "#76BBF7"), c(3, 4, 5)),
  border_size = 2,
  label_level = 2,
  label_size = 0.7,
  label_color = grey(0.5),
  layout = c(1, 3),
  position = c(1, 2),
  add = TRUE
)

# color cells (sectors) based on cell size
drawTreemap(tm,
  title = "Coloring encoded by cell size",
  color_type = "cell_size",
  legend = TRUE,
  color_palette = rev(heat.colors(10)),
  border_size = 3,
  border_color = grey(0.3),
  label_level = 1,
  label_size = 2,
  label_color = grey(0.5),
  layout = c(1, 3),
  position = c(1, 3),
  add = TRUE
)

</code></pre>

<hr>
<h2 id='voronoiTreemap'>voronoiTreemap</h2><span id='topic+voronoiTreemap'></span>

<h3>Description</h3>

<p>Create nested additively weighted Voronoi treemaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>voronoiTreemap(
  data,
  levels,
  fun = sum,
  sort = TRUE,
  filter = 0,
  cell_size = NULL,
  custom_color = NULL,
  shape = "rectangle",
  maxIteration = 100,
  error_tol = 0.01,
  convergence = "intermediate",
  seed = NULL,
  positioning = "regular",
  verbose = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="voronoiTreemap_+3A_data">data</code></td>
<td>
<p>(data.frame) A data.frame with one column for each hierarchical level</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_levels">levels</code></td>
<td>
<p>(character) Character vector indicating the column names to
be used. The order of names must correspond to the hierarchical levels,
going from broad to fine</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_fun">fun</code></td>
<td>
<p>(function) Function to be used to aggregate cell sizes of parental cells</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_sort">sort</code></td>
<td>
<p>(logical) Should the columns of the data.frame be sorted before treemap generation?</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_filter">filter</code></td>
<td>
<p>(numeric) Filter the supplied data frame to remove very small
cells that may not be visible. The default is to remove cells with a
relative target area below a threshold of zero (no negative values allowed).
Computation time can increase when many small cells are present. For example,
a threshold of 0.01 filters out all observations/cells below 1 % of the total area.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_cell_size">cell_size</code></td>
<td>
<p>(character) The name of the column used to control cell size.
Can be one of <code>levels</code> or any other column with numerical data. NA or
values equal or less than zero are not allowed as the cell area needs to be positive.
The values in this column are aggregated by the function specified by <code>fun</code>.
If <code>cell_size = NULL</code>, cell area is simply computed by the number of members
for the respective cell (corresponding to rows in the data.frame).</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_custom_color">custom_color</code></td>
<td>
<p>(character) An optional column that can be specified to
control cell color. Cell colors are determined when drawing the treemap
using <code><a href="#topic+drawTreemap">drawTreemap</a></code>, but the default is to use one of
<code>levels</code> or <code>cell size</code>. Any other data source that shall be used
instead has to be included in the treemap generation and explicitly
specified here. The default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_shape">shape</code></td>
<td>
<p>(list or character) Set the initial shape of the treemap. Currently
supported are the keywords &quot;rectangle&quot;, &quot;rounded_rect&quot;, &quot;circle&quot; or &quot;hexagon&quot;.
Alternatively the user can supply a named list with coordinates for a custom polygon.
The slots of the list must be labeled 'x' and 'y'. The coordinates are not tested
for validity, use on your own risk.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_maxiteration">maxIteration</code></td>
<td>
<p>(numeric) Force algorithm to stop at this number of iterations
for each parent cell. The algorithm usually converges to an acceptable
solution fairly quickly, so it seems reasonable to restrict this number
in order to save computation time. However, more iterations give higher
accuracy.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_error_tol">error_tol</code></td>
<td>
<p>(numeric) The allowed maximum error tolerance of a cell.
The algorithm will stop when all cells have lower error than this value.
It is calculated as the absolute difference of a cell's area to its target
area. The default is 0.01 (or 1 %) of the total parental area. Note: this
is is different from a relative per-cell error, where 1 % would be more
strict.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_convergence">convergence</code></td>
<td>
<p>(character) One of &quot;slow&quot;, &quot;intermediate&quot;, or &quot;fast&quot;.
Intermediate (default) and fast try to adjust cell weights stronger such
that the algorithm converges faster towards the final size of the cell.
However this comes at the price of stability, with a larger number of
polygons possibly being misformed, e.g. by having self-intersections.
Set convergence to &quot;slow&quot; if you experience problems to calculate treemaps
with very unequal cell sizes or very large treemaps.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_seed">seed</code></td>
<td>
<p>(integer) The default seed is NULL, which will lead to a new
random sampling of cell coordinates for each tesselation. If you want
a reproducible arrangement of cells, set seed to an arbitrary number.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_positioning">positioning</code></td>
<td>
<p>(character) Algorithm for positioning of starting
coordinates of child cells in the parental cell using <code>spsample()</code>;
&quot;random&quot; for completely random positions, &quot;regular&quot; for cells aligned
to a grid sorted from bottom to top by name, &quot;clustered&quot; with regular
positions of cells but sorted by name from inside out. Two variants
&quot;regular_by_area&quot; and &quot;clustered_by_area&quot; will work as their counterparts
but will sort by cell target area instead of cell name. <code>positioning</code>
can be a single character or a vector of <code>length(levels)</code> to allow
different positioning algorithms for each level.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_verbose">verbose</code></td>
<td>
<p>(logical) If verbose is TRUE (default is FALSE), messages
with statistics for each iteration of a treemap as well as a success message
are printed to the console.</p>
</td></tr>
<tr><td><code id="voronoiTreemap_+3A_debug">debug</code></td>
<td>
<p>(logical) If debug is TRUE (default is FALSE), the solution
for each iteration is drawn to the viewport to allow some visual
inspection. The weights, target area, and difference are printed to the
console. It is not recommended to set this option to TRUE unless you know
what you are doing, as it makes treemap generation much slower.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a recursive wrapper function, making use of the original implementation
of the voronoi tesselation from Paul Murrell, University of Auckland.
The original functions were obtained and slightly modified from
<a href="https://www.stat.auckland.ac.nz/~paul/Reports/VoronoiTreemap/voronoiTreeMap.html">https://www.stat.auckland.ac.nz/~paul/Reports/VoronoiTreemap/voronoiTreeMap.html</a>
This function returns a treemap object instead of a plot. In order
to actually draw the treemap, use <code><a href="#topic+drawTreemap">drawTreemap</a></code>.
</p>


<h3>Value</h3>

<p>'voronoiTreemap' returns an object of the formal class 'voronoiResult'.
It is essentially a list of objects related to the graphical
representation of the treemap (polygons, labels, cell data) as well as data from the call
of the function. It contains the following slots:
</p>
<table role = "presentation">
<tr><td><code>cells</code></td>
<td>
<p>'list' of polygons for drawing a treemap</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>'data.frame', the original data that was supplied to calling 'voronoiTreemap'</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>'list' of arguments used to call 'voronoiTreemap'</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+drawTreemap">drawTreemap</a></code> for drawing the treemap.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load package
library(WeightedTreemaps)

# generate dummy data
df &lt;- data.frame(
  A = rep(c("abcd", "efgh"), each = 4),
  B = letters[1:8],
  size = c(37, 52, 58, 27, 49, 44, 34, 45)
)

# compute treemap
tm &lt;- voronoiTreemap(
  data = df,
  levels = c("B"),
  cell_size = "size",
  shape = "circle",
  positioning = "regular",
  seed = 123
)

# plot treemap with each cell colored by name (default)
drawTreemap(tm, label_size = 1, color_type = "categorical")

# plot treemap with each cell colored by name, but larger cells
# lighter and smaller cells darker
drawTreemap(tm, label_size = 1, color_type = "both")

# plot treemap with different color palette and style
drawTreemap(tm, label_size = 1, label_color = grey(0.3),
            border_color = grey(0.3), color_palette = heat.colors(6)
)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
