<!DOCTYPE html><html><head><title>Help for package WES</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {WES}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#apply_amplification_efficiency'><p>Apply PCR amplification efficiency estimation to a data.frame</p></a></li>
<li><a href='#apply_delta_delta_ct'><p>Apply the delta delta Ct calculation to a data.frame</p></a></li>
<li><a href='#calc_delta_delta_ct'><p>Calculate delta delta Ct</p></a></li>
<li><a href='#calc_n_copies'><p>Calculate number of target copies</p></a></li>
<li><a href='#calc_sample_sizes'><p>Calculate sample sizes</p></a></li>
<li><a href='#coords_to_iso3'><p>Convert coordinates to ISO country code</p></a></li>
<li><a href='#download_admin_data'><p>Download administrative boundaries from geoBoundaries API</p></a></li>
<li><a href='#download_elevation_data'><p>Download DEM from AWS Terrain Tiles</p></a></li>
<li><a href='#download_worldpop_data'><p>Download WorldPop population raster data</p></a></li>
<li><a href='#est_amplification_efficiency'><p>Estimate PCR amplification efficiency</p></a></li>
<li><a href='#get_admin_data'><p>Get administrative data for a set of points</p></a></li>
<li><a href='#get_elevation_data'><p>Get elevation data</p></a></li>
<li><a href='#get_esi_data'><p>Get Evaporative Stress Index (ESI) data</p></a></li>
<li><a href='#get_geoboundaries_api_data'><p>Request metadata from GeoBoundaries API</p></a></li>
<li><a href='#get_hydro_data'><p>Get hydrological data</p></a></li>
<li><a href='#get_population_catchment'><p>Get population counts within catchments of sampling sites</p></a></li>
<li><a href='#get_population_radius'><p>Get population counts within a radius of sampling sites</p></a></li>
<li><a href='#get_precip_data'><p>Get precipitation data</p></a></li>
<li><a href='#get_river_discharge_data'><p>Get river discharge data</p></a></li>
<li><a href='#get_temp_data'><p>Get temperature data</p></a></li>
<li><a href='#template_WES_data'><p>Template environmental sampling data</p></a></li>
<li><a href='#template_WES_standard_curve'><p>Template standard curve data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for Analyzing Wastewater and Environmental Sampling Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-07-03</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides reproducible functions for collating and analyzing 
     data from environmental sampling studies. Environmental Sampling (ES) 
     of infectious diseases involves collecting samples from various sources 
     (such as sewage, water, air, soil, or surfaces) to monitor the presence 
     of pathogens in the environment. Analysis of ES data often requires the 
     calculation of Real-Time Quantitative PCR (qPCR) variables, normalizing 
     ES observations, and analyzing sampling site characteristics. To help 
     reduce the complexity of these analyses we have implemented tools that 
     assist with establishing standardized ES data formats, absolute and 
     relative quantification of qPCR data, estimation of qPCR amplification 
     efficiency, and collating open-source spatial data for sampling sites.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-wes.com/">https://www.r-wes.com/</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by/4.0">CC BY 4.0</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>glue, foreach, data.table, dplyr, HDInterval, zoo, openmeteo,
chirps, whitebox, exactextractr, elevatr, raster, sp, sf,
rworldmap, stars, RCurl, XML, httr, jsonlite</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.1), stats, graphics, grDevices, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, rworldxtra, progress</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-11 02:31:38 UTC; johngiles</td>
</tr>
<tr>
<td>Author:</td>
<td>John R Giles <a href="https://orcid.org/0000-0002-0954-4093"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John R Giles &lt;john.giles@gatesfoundation.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-12 14:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='apply_amplification_efficiency'>Apply PCR amplification efficiency estimation to a data.frame</h2><span id='topic+apply_amplification_efficiency'></span>

<h3>Description</h3>

<p>This function applies the <code>est_amplification_efficiency()</code> function to a data.frame object which follows the
standardized format shown in the <code>template_WES_standard_curve</code> data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_amplification_efficiency(standard_curves)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apply_amplification_efficiency_+3A_standard_curves">standard_curves</code></td>
<td>
<p>A data.frame giving the target name, serial diluted concentration of target nucleic acid,
and Ct value from a standard curve assay. Must follow the <code>template_WES_standard_curve</code> standardized format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing the mean, and low and high of the 95% confidence interval of the percentile
amplification efficiency for each target name.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>apply_amplification_efficiency(template_WES_standard_curve)

</code></pre>

<hr>
<h2 id='apply_delta_delta_ct'>Apply the delta delta Ct calculation to a data.frame</h2><span id='topic+apply_delta_delta_ct'></span>

<h3>Description</h3>

<p>This function will calculate the delta delta Ct metric for all applicable observations in a data.frame
by applying the <code>calc_delta_delta_ct</code> function. The data.frame must have the following columns:
'location_id', 'sample_date', 'target_name', and 'ct_value'. The relevant target_names and associated reference_names
must be provided. The result is a data.frame containing a 'delta_delta_ct' column which can be merge into the source data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_delta_delta_ct(
  df,
  target_names,
  reference_names,
  pae_names = NULL,
  pae_values = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="apply_delta_delta_ct_+3A_df">df</code></td>
<td>
<p>A data.frame containing the following columns: 'location_id', 'sample_date', 'target_name', and 'ct_value'.</p>
</td></tr>
<tr><td><code id="apply_delta_delta_ct_+3A_target_names">target_names</code></td>
<td>
<p>Character vector giving the names of the target genes.</p>
</td></tr>
<tr><td><code id="apply_delta_delta_ct_+3A_reference_names">reference_names</code></td>
<td>
<p>Character vector giving the names of the reference genes associated with each target gene.</p>
</td></tr>
<tr><td><code id="apply_delta_delta_ct_+3A_pae_names">pae_names</code></td>
<td>
<p>Character vector giving the names of the target genes and reference genes for which the percentile amplification efficiency has been estimated. Default is NULL.</p>
</td></tr>
<tr><td><code id="apply_delta_delta_ct_+3A_pae_values">pae_values</code></td>
<td>
<p>A numeric scalar giving the estimated PCR amplification efficiency for each of the names in <code>pae_names</code>. Defaults is NULL, which assumes 100% efficiency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
pae &lt;- apply_amplification_efficiency(template_WES_standard_curve)

ddct_standard &lt;- apply_delta_delta_ct(df = template_WES_data,
                                      target_names = c('target_1', 'target_2', 'target_3'),
                                      reference_names = rep('target_0', 3))

ddct_adjusted &lt;- apply_delta_delta_ct(df = template_WES_data,
                                      target_names = c('target_1', 'target_2', 'target_3'),
                                      reference_names = rep('target_0', 3),
                                      pae_names = pae$target_name,
                                      pae_values = pae$mean)

head(ddct_adjusted)

</code></pre>

<hr>
<h2 id='calc_delta_delta_ct'>Calculate delta delta Ct</h2><span id='topic+calc_delta_delta_ct'></span>

<h3>Description</h3>

<p>This function calculates relative gene expression using the delta delta Ct method described in
<a href="https://www.sciencedirect.com/science/article/abs/pii/S1046202301912629?via%3Dihub">Livak and Schmittgen (2001)</a>.
Adjusted delta delta Ct values following <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-85">Yuan et al. (2008)</a>
can be calculated by providing estimated values for the percentile amplification efficiency in <code style="white-space: pre;">&#8288;pae_*&#8288;</code> arguments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_delta_delta_ct(
  ct_target_treatment,
  ct_target_control,
  ct_reference_treatment,
  ct_reference_control,
  pae_target_treatment = 1,
  pae_target_control = 1,
  pae_reference_treatment = 1,
  pae_reference_control = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_delta_delta_ct_+3A_ct_target_treatment">ct_target_treatment</code></td>
<td>
<p>A numeric scalar providing the Ct value of the target gene for an observation in the treatment group</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_ct_target_control">ct_target_control</code></td>
<td>
<p>A numeric scalar providing the Ct value of the target gene for the reference observation in the control group</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_ct_reference_treatment">ct_reference_treatment</code></td>
<td>
<p>A numeric scalar providing the Ct value of the reference gene for an observation in the treatment group</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_ct_reference_control">ct_reference_control</code></td>
<td>
<p>A numeric scalar providing the Ct value of the reference gene for the reference observation in the control group</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_pae_target_treatment">pae_target_treatment</code></td>
<td>
<p>A numeric scalar providing the percentile amplification efficiency for the target gene and the treatment group. Defaults to 1.</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_pae_target_control">pae_target_control</code></td>
<td>
<p>A numeric scalar providing the percentile amplification efficiency for the target gene and the control group. Defaults to 1.</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_pae_reference_treatment">pae_reference_treatment</code></td>
<td>
<p>A numeric scalar providing the percentile amplification efficiency for the reference gene and the treatment group. Defaults to 1.</p>
</td></tr>
<tr><td><code id="calc_delta_delta_ct_+3A_pae_reference_control">pae_reference_control</code></td>
<td>
<p>A numeric scalar providing the percentile amplification efficiency for the reference gene and the control group. Defaults to 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scalar
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Traditional method
calc_delta_delta_ct(ct_target_treatment = 32.5,
                    ct_reference_treatment = 25,
                    ct_target_control = 34,
                    ct_reference_control = 30)

# Adjusted calculation incorporating amplification efficiency
calc_delta_delta_ct(ct_target_treatment = 32.5,
                    ct_reference_treatment = 25,
                    ct_target_control = 34,
                    ct_reference_control = 30,
                    pae_target_treatment=0.97,
                    pae_target_control=0.98,
                    pae_reference_treatment=0.98,
                    pae_reference_control=0.99)

</code></pre>

<hr>
<h2 id='calc_n_copies'>Calculate number of target copies</h2><span id='topic+calc_n_copies'></span>

<h3>Description</h3>

<p>This function calculates the quantitative value of the qPCR Ct value. Cycle threshold here is converted into
the estimated number of gene target copies (e.g. viral load for a viral pathogen) by fitting a log linear model
to the standard curve data and then using that model to find a point estimate for the provided Ct values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_n_copies(ct_values, target_names, standard_curves)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_n_copies_+3A_ct_values">ct_values</code></td>
<td>
<p>A numeric vector giving the Ct value for each observation.</p>
</td></tr>
<tr><td><code id="calc_n_copies_+3A_target_names">target_names</code></td>
<td>
<p>A character vector giving the target names for each element in 'ct_values'.</p>
</td></tr>
<tr><td><code id="calc_n_copies_+3A_standard_curves">standard_curves</code></td>
<td>
<p>A data.frame containing results from standard curve dilution experiment.
Elements in 'target_names' must map to either 'target_name_unique' or 'target_name_concise'. See package
data object <code>template_WES_standard_curves</code> for template.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
df &lt;- template_WES_data[template_WES_data$target_name == 'target_1',]
sc &lt;- template_WES_standard_curve[template_WES_standard_curve$target_name == 'target_1',]

tmp &lt;- calc_n_copies(ct_values = df$ct_value,
                     target_names = df$target_name,
                     standard_curves = sc)

df$n_copies &lt;- tmp
head(df)

</code></pre>

<hr>
<h2 id='calc_sample_sizes'>Calculate sample sizes</h2><span id='topic+calc_sample_sizes'></span>

<h3>Description</h3>

<p>This function takes a compiled data.frame following the format shown in the <code>template_WES_data</code> object
and calculates basic sample sizes and detection rates for all gene targets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_sample_sizes(df, cutoff = 40)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_sample_sizes_+3A_df">df</code></td>
<td>
<p>A data.frame produced by the <code>compile_tac_data</code> function containing 'target_name' and 'ct_value' columns.</p>
</td></tr>
<tr><td><code id="calc_sample_sizes_+3A_cutoff">cutoff</code></td>
<td>
<p>Numeric scalar giving the cutoff Ct value over which a gene target is deemed absent from a sample. Default is 40.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calc_sample_sizes(template_WES_data)
</code></pre>

<hr>
<h2 id='coords_to_iso3'>Convert coordinates to ISO country code</h2><span id='topic+coords_to_iso3'></span>

<h3>Description</h3>

<p>This function takes a set of longitude and latitude coordinates and retrieves the administrative
units that each point lies within. The administrative units are given in the ISO-3166 Alpha-3 country code
standard (<a href="https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3">https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coords_to_iso3(lon, lat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coords_to_iso3_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="coords_to_iso3_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
coords_to_iso3(lon = c(90.37, 90.38, 90.37),
               lat = c(23.80, 23.80, 23.81))

</code></pre>

<hr>
<h2 id='download_admin_data'>Download administrative boundaries from geoBoundaries API</h2><span id='topic+download_admin_data'></span>

<h3>Description</h3>

<p>This function takes a single ISO country code and downloads the corresponding
high resolution administrative boundary <a href="https://en.wikipedia.org/wiki/GeoJSON">GeoJSON</a> files from the <a href="https://www.geoboundaries.org/">www.geoBoundaries.org</a> API
hosted at GitHub <a href="https://github.com/wmgeolab/geoBoundaries/tree/9469f09592ced973a3448cf66b6100b741b64c0d">HERE</a>. If the
desired administrative level is not available the next most detailed administrative level is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_admin_data(
  iso3,
  release,
  path_output,
  simplified = FALSE,
  keep_geojson = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_admin_data_+3A_iso3">iso3</code></td>
<td>
<p>A three-letter capitalized character string. Must follow the ISO-3166 Alpha-3 country code</p>
</td></tr>
<tr><td><code id="download_admin_data_+3A_release">release</code></td>
<td>
<p>A character string specifying the release type on the geoBoundaries API. It should be one of 'gbOpen',
'gbHumanitarian', or 'gbAuthoritative'.  Release types are described at <a href="https://www.geoboundaries.org/api.html#api">https://www.geoboundaries.org/api.html#api</a>.</p>
</td></tr>
<tr><td><code id="download_admin_data_+3A_path_output">path_output</code></td>
<td>
<p>A character string giving the file path of an output directory to save downloaded data.</p>
</td></tr>
<tr><td><code id="download_admin_data_+3A_simplified">simplified</code></td>
<td>
<p>Logical indicating whether to download simplified administrative boundaries instead of high resolution. Default is FALSE.</p>
</td></tr>
<tr><td><code id="download_admin_data_+3A_keep_geojson">keep_geojson</code></td>
<td>
<p>Logical indicating whether to keep the raw geojson files downloaded from geoBoundaries API. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character string giving path to downloaded data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

download_admin_data(iso3 = 'MCO',
                    release = 'gbOpen',
                    path_output = tempdir())


</code></pre>

<hr>
<h2 id='download_elevation_data'>Download DEM from AWS Terrain Tiles</h2><span id='topic+download_elevation_data'></span>

<h3>Description</h3>

<p>This function takes the coordinates of sampling sites (longitude and latitude) and downloads a Digital Elevation Model (DEM)
for the surrounding area. The DEM has an approximate spatial resolution of 100 meters. These data are derived from the
Shuttle Radar Topography Mission (SRTM) DEM, which is accessible through the Amazon Web Services (AWS) API and the
<a href="https://CRAN.R-project.org/package=elevatr"><code>elevatr</code></a> R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_elevation_data(lon, lat, path_output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_elevation_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.Can accept a vector of multiple ISO codes.</p>
</td></tr>
<tr><td><code id="download_elevation_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="download_elevation_data_+3A_path_output">path_output</code></td>
<td>
<p>A character string giving the file path of an output directory to save downloaded data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character string giving path to downloaded data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

download_elevation_data(lon = template_WES_data$lon,
                        lat = template_WES_data$lat,
                        path_output = tempdir())


</code></pre>

<hr>
<h2 id='download_worldpop_data'>Download WorldPop population raster data</h2><span id='topic+download_worldpop_data'></span>

<h3>Description</h3>

<p>This function takes a single ISO country code and downloads the appropriate population count raster data (100m grid cell resolution)
from the WorldPop FTP data server. Note that these data are spatial disaggregations of census data using random forest models described in
<a href="https://www.tandfonline.com/doi/full/10.1080/20964471.2019.1625151">Lloyd et al. 2019</a> and available for manual download at
<a href="https://hub.worldpop.org/geodata/listing?id=29">https://hub.worldpop.org/geodata/listing?id=29</a>. Downloaded data sets are
saved to the <code>path_output</code> directory in .tif format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_worldpop_data(
  iso3,
  year,
  constrained = FALSE,
  UN_adjusted = FALSE,
  path_output
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_worldpop_data_+3A_iso3">iso3</code></td>
<td>
<p>A three-letter capitalized character string. Must follow the ISO-3166 Alpha-3 country code
standard (<a href="https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3">https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3</a>). Can accept a vector of multiple ISO codes.</p>
</td></tr>
<tr><td><code id="download_worldpop_data_+3A_year">year</code></td>
<td>
<p>A numeric or integer scalar giving the year of WorldPop data to download (as of 2024-05-15, years 2000-2020 are available)</p>
</td></tr>
<tr><td><code id="download_worldpop_data_+3A_constrained">constrained</code></td>
<td>
<p>Logical indicating whether to get population counts estimated using constrained models (details <a href="https://www.worldpop.org/methods/top_down_constrained_vs_unconstrained/">HERE</a>).
Default is FALSE.</p>
</td></tr>
<tr><td><code id="download_worldpop_data_+3A_un_adjusted">UN_adjusted</code></td>
<td>
<p>Logical indicating whether to get population counts that are adjusted to match United Nations national population estimates
(details <a href="https://hub.worldpop.org/project/categories?id=3">HERE</a>). Default is FALSE.</p>
</td></tr>
<tr><td><code id="download_worldpop_data_+3A_path_output">path_output</code></td>
<td>
<p>A character string giving the file path of an output directory to save downloaded data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character string giving path to downloaded data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

download_worldpop_data(iso3 = 'MCO',
                       year = 2020,
                       constrained = TRUE,
                       UN_adjusted = FALSE,
                       path_output = tempdir())


</code></pre>

<hr>
<h2 id='est_amplification_efficiency'>Estimate PCR amplification efficiency</h2><span id='topic+est_amplification_efficiency'></span>

<h3>Description</h3>

<p>This function takes a set of serial diluted concentrations of target nucleic acid from a standard curve assay and their associated Ct values and
estimates the percentile amplification efficiency using a linear model as described in <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-85">Yuan et al. (2008)</a>.
Note that the model uses a log base 2 transform which assumes that serial dilutions double with each increase in concentration. The function also
requires a minimum of 5 observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_amplification_efficiency(n_copies, ct_value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="est_amplification_efficiency_+3A_n_copies">n_copies</code></td>
<td>
<p>A numeric vector giving the serial diluted concentration of target nucleic acid</p>
</td></tr>
<tr><td><code id="est_amplification_efficiency_+3A_ct_value">ct_value</code></td>
<td>
<p>A numeric vector giving the measured Ct value for each serial dilution in the standard curve design</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List containing the mean, and low and high of the 95% confidence interval for the percentile amplification efficiency.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
sel &lt;- template_WES_standard_curve$target_name == 'target_1'
tmp_n_copies &lt;- template_WES_standard_curve$n_copies[sel]
tmp_ct_value &lt;- template_WES_standard_curve$ct_value[sel]

est_amplification_efficiency(n_copies = tmp_n_copies,
                             ct_value = tmp_ct_value)

</code></pre>

<hr>
<h2 id='get_admin_data'>Get administrative data for a set of points</h2><span id='topic+get_admin_data'></span>

<h3>Description</h3>

<p>This function takes a set of longitude and latitude coordinates and retrieves the administrative
units that each point lies within.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_admin_data(lon, lat, path_admin_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_admin_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_admin_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_admin_data_+3A_path_admin_data">path_admin_data</code></td>
<td>
<p>The file path to the admin data. Note that the function expects .shp
format output from the <code>download_admin_data</code> function or from another user supplied source.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

download_admin_data(iso3 = "MCO",
                    release = 'gbOpen',
                    path_output = tempdir())

get_admin_data(lon = c(7.416, 7.434),
               lat = c(43.734, 43.747),
               path_admin_data = file.path(tempdir(), 'MCO_admin_levels.shp'))


</code></pre>

<hr>
<h2 id='get_elevation_data'>Get elevation data</h2><span id='topic+get_elevation_data'></span>

<h3>Description</h3>

<p>This function takes information of where and when a set of environmental samples were
collected and retrieves the elevation (in meters) for those locations at an approximate 100m spatial resolution.
Data come from the <a href="https://www.usgs.gov/centers/eros/science/usgs-eros-archive-digital-elevation-shuttle-radar-topography-mission-srtm-1">SRTM</a>
DEM which are accessed through the Amazon Web Services (AWS) API and the <a href="https://CRAN.R-project.org/package=elevatr"><code>elevatr</code></a>
R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_elevation_data(lon, lat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_elevation_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_elevation_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

get_elevation_data(lon = template_WES_data$lon,
                   lat = template_WES_data$lat)


</code></pre>

<hr>
<h2 id='get_esi_data'>Get Evaporative Stress Index (ESI) data</h2><span id='topic+get_esi_data'></span>

<h3>Description</h3>

<p>This function takes information of where and when a set of environmental samples were
collected and retrieves the Evaporative Stress Index (ESI) for those locations and times.
For more information about ESI, see description <a href="https://climateserv.readthedocs.io/en/latest/user/datasets.html#evaporative-stress-index-esi">HERE</a>.
Data come from the Climate Hazards Center InfraRed Precipitation with Station data (<a href="https://www.chc.ucsb.edu/data">CHIRPS</a>) via
the <a href="https://docs.ropensci.org/chirps/"><code>chirps</code></a> R package. Additionally, the optional <code>intervals</code> argument
specifies a set of intervals over which the function will calculate the average ESI for the previous X number
of days for each location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_esi_data(lon, lat, dates, intervals = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_esi_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_esi_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_esi_data_+3A_dates">dates</code></td>
<td>
<p>A character or date vector of dates giving the date when each sample was
collected (format is YYYY-MM-DD)</p>
</td></tr>
<tr><td><code id="get_esi_data_+3A_intervals">intervals</code></td>
<td>
<p>An integer vector giving a set of time intervals over which to calculate
the average ESI. Default is NULL where the interval is 0 (returns the ESI value at time t). If <code>intervals</code>=3
then the average ESI over the preceding 3 days is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

tmp &lt;- get_esi_data(lon = c(-54.9857, -52.9857),
                    lat = c(-5.9094, -25.8756),
                    dates = c("2020-06-01", "2020-10-31"),
                    intervals = c(5,10,20))

head(tmp)


</code></pre>

<hr>
<h2 id='get_geoboundaries_api_data'>Request metadata from GeoBoundaries API</h2><span id='topic+get_geoboundaries_api_data'></span>

<h3>Description</h3>

<p>This function retrieves GeoBoundaries data from the API based on the specified release, ISO3 country code, and administrative level.
If data is not found at the specified administrative level, it attempts to retrieve data from a lower administrative level until data
is found or the lowest level is reached.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_geoboundaries_api_data(iso3, admin_level, release = "gbOpen")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_geoboundaries_api_data_+3A_iso3">iso3</code></td>
<td>
<p>A three-letter capitalized character string. Must follow the ISO-3166 Alpha-3 country code
standard (<a href="https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3">https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3</a>).</p>
</td></tr>
<tr><td><code id="get_geoboundaries_api_data_+3A_admin_level">admin_level</code></td>
<td>
<p>An integer specifying the administrative level. It should be between 0 and 5.</p>
</td></tr>
<tr><td><code id="get_geoboundaries_api_data_+3A_release">release</code></td>
<td>
<p>A character string specifying the release type. It should be one of 'gbOpen', 'gbHumanitarian', or 'gbAuthoritative'. Default is 'gbOpen'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the GeoBoundaries API data and file paths to admin boundaries in .geojson format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

tmp &lt;- get_geoboundaries_api_data(iso3 = 'MCO', admin_level = 2, release = 'gbOpen')
head(tmp)


</code></pre>

<hr>
<h2 id='get_hydro_data'>Get hydrological data</h2><span id='topic+get_hydro_data'></span>

<h3>Description</h3>

<p>This function takes information of where and when a set of environmental samples were
collected and retrieves a suite of topographical and hydrological variables for each unique
location. The variables include: elevation, slope, aspect, Topographical Wetness Index (TWI),
flow accumulation, total flow accumulation within 500m, and distance to the nearest stream.
If a DEM is not provided, then a DEM is acquired via <a href="https://www.rdocumentation.org/packages/elevatr/versions/0.99.0/topics/get_elev_raster"><code>elevatr::get_elev_raster</code></a>
and the suite of variables are calculated using functions from the <a href="https://CRAN.R-project.org/package=whitebox">'WhiteboxTools'</a>
R frontend.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hydro_data(lon, lat, path_dem_raster = NULL, path_output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_hydro_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_hydro_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_hydro_data_+3A_path_dem_raster">path_dem_raster</code></td>
<td>
<p>The file path to a Digital Elevation Model (DEM) raster. See <code>download_elevation_data</code>
for methods to download DEM raster data. If NULL, a DEM is downloaded automatically using this function.</p>
</td></tr>
<tr><td><code id="get_hydro_data_+3A_path_output">path_output</code></td>
<td>
<p>The file path of an output directory where spatial data will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

MCO_lon &lt;- c(7.416, 7.434)
MCO_lat &lt;- c(43.734, 43.747)

download_elevation_data(lon = MCO_lon,
                        lat = MCO_lat,
                        path_output = tempdir())

get_hydro_data(lon = MCO_lon,
               lat = MCO_lat,
               path_dem_raster = file.path(tempdir(), 'dem.tif'),
               path_output = tempdir())


</code></pre>

<hr>
<h2 id='get_population_catchment'>Get population counts within catchments of sampling sites</h2><span id='topic+get_population_catchment'></span>

<h3>Description</h3>

<p>This function takes vectors of sampling site longitude and latitude and calculates the total population
residing within the drainage catchment of each coordinate pair. Raster data giving population counts per grid cell
and a Digital Elevation Model (DEM) are required. By default, the function delineates streams based on the
provided DEM. However, an optional shapefile (such as an urban sewer network) can be specified using the
<code>path_stream_shp</code> argument and will be used instead of the natural stream network calculated from the DEM.
Note that the delineation of catchments along streams (or sewer networks) still depends on the directional flow
from the provided DEM. Intermediate spatial variables are written to the directory specified in <code>path_output</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_population_catchment(
  lon,
  lat,
  path_pop_raster,
  path_dem_raster,
  path_stream_shp = NULL,
  path_output
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_population_catchment_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitudes of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_population_catchment_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitudes of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_population_catchment_+3A_path_pop_raster">path_pop_raster</code></td>
<td>
<p>The file path to a raster object providing population counts in each grid cell.
See <code>download_worldpop_data</code> for methods to download population raster data.</p>
</td></tr>
<tr><td><code id="get_population_catchment_+3A_path_dem_raster">path_dem_raster</code></td>
<td>
<p>The file path to a Digital Elevation Model (DEM) raster. See <code>download_elevation_data</code>
for methods to download DEM raster data.</p>
</td></tr>
<tr><td><code id="get_population_catchment_+3A_path_stream_shp">path_stream_shp</code></td>
<td>
<p>An optional file path to a stream or sewer network shapefile. If NULL (the default), streams are delineated
based on flow accumulation in the provided DEM.</p>
</td></tr>
<tr><td><code id="get_population_catchment_+3A_path_output">path_output</code></td>
<td>
<p>The file path of an output directory where spatial data will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> containing the catchment area and population counts for each sampling site.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

MCO_lon &lt;- c(7.416, 7.434)
MCO_lat &lt;- c(43.734, 43.747)

download_worldpop_data(iso3 = 'MCO',
                       year = 2020,
                       constrained = TRUE,
                       UN_adjusted = FALSE,
                       path_output = tempdir())

download_elevation_data(lon = MCO_lon,
                        lat = MCO_lat,
                        path_output = tempdir())

get_population_catchment(lon = MCO_lon,
                         lat = MCO_lat,
                         path_pop_raster = file.path(tempdir(), 'mco_ppp_2020_constrained.tif'),
                         path_dem_raster = file.path(tempdir(), 'dem.tif'),
                         path_output = tempdir())


</code></pre>

<hr>
<h2 id='get_population_radius'>Get population counts within a radius of sampling sites</h2><span id='topic+get_population_radius'></span>

<h3>Description</h3>

<p>This function takes vectors of sampling site longitude and latitude and calculates the total population
residing within a given radius around each sampling site. Intermediate spatial variables are written to
the directory specified in <code>path_output</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_population_radius(lon, lat, radius, path_pop_raster, path_output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_population_radius_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitudes of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_population_radius_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitudes of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_population_radius_+3A_radius">radius</code></td>
<td>
<p>Numeric giving the radius (in meters) around each point to calculate total population</p>
</td></tr>
<tr><td><code id="get_population_radius_+3A_path_pop_raster">path_pop_raster</code></td>
<td>
<p>The file path to a raster object providing population counts in each grid cell.
See <code>download_worldpop_data</code> for methods to download population raster data.</p>
</td></tr>
<tr><td><code id="get_population_radius_+3A_path_output">path_output</code></td>
<td>
<p>The file path of an output directory where spatial data will be saved.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> containing the total population counts for the given radius around each sampling site.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

download_worldpop_data(iso3 = 'MCO',
                       year = 2020,
                       constrained = TRUE,
                       UN_adjusted = FALSE,
                       path_output = tempdir())

get_population_radius(lon = c(7.416, 7.434),
                      lat = c(43.734, 43.747),
                      radius = 100,
                      path_pop_raster = file.path(tempdir(), 'mco_ppp_2020_constrained.tif'),
                      path_output = tempdir())


</code></pre>

<hr>
<h2 id='get_precip_data'>Get precipitation data</h2><span id='topic+get_precip_data'></span>

<h3>Description</h3>

<p>This function takes information of where and when a set of environmental samples were
collected and retrieves precipitation data (in millimeters) for those locations and times. Data come from
the Open-Meteo Historical Weather API (<a href="https://open-meteo.com/en/docs/historical-weather-api">https://open-meteo.com/en/docs/historical-weather-api</a>)
via the <a href="https://CRAN.R-project.org/package=openmeteo"><code>openmeteo</code></a> R package.
Additionally, the optional <code>intervals</code> argument specifies a set of intervals over which the function
will calculate the cumulative sum of precipitation in millimeters (mm) for the previous X number of
days for each location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_precip_data(lon, lat, dates, intervals = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_precip_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_precip_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_precip_data_+3A_dates">dates</code></td>
<td>
<p>A character or date vector of dates giving the date when each sample was
collected (format is YYYY-MM-DD)</p>
</td></tr>
<tr><td><code id="get_precip_data_+3A_intervals">intervals</code></td>
<td>
<p>An integer vector giving a set of time intervals over which to sum the
precipitation data. Default is NULL where the interval is 0 (returns the precipitation value at time t). If <code>intervals</code>=3
then the cumulative precipitation over the preceding 3 days is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

tmp &lt;- get_precip_data(lon = c(-56.0281, -54.9857),
                       lat = c(-2.9094, -2.8756),
                       dates = c("2017-12-01", "2017-12-31"),
                       intervals = c(1,3,7))

head(tmp)


</code></pre>

<hr>
<h2 id='get_river_discharge_data'>Get river discharge data</h2><span id='topic+get_river_discharge_data'></span>

<h3>Description</h3>

<p>This function takes information of where and when a set of environmental samples were
collected and retrieves daily river discharge data from the nearest river (<code class="reqn">m^3/s</code>) for those locations and times. Data come from
the Open-Meteo Global Flood API (<a href="https://open-meteo.com/en/docs/flood-api">https://open-meteo.com/en/docs/flood-api</a>)
via the <a href="https://CRAN.R-project.org/package=openmeteo"><code>openmeteo</code></a> R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_river_discharge_data(lon, lat, dates)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_river_discharge_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_river_discharge_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_river_discharge_data_+3A_dates">dates</code></td>
<td>
<p>A character or date vector of dates giving the date when each sample was
collected (format is YYYY-MM-DD)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

tmp &lt;- get_river_discharge_data(lon = c(-54.9857, -52.9857),
                                lat = c(-10.9094, -25.8756),
                                dates = c("2020-06-01", "2020-10-31"))

head(tmp)


</code></pre>

<hr>
<h2 id='get_temp_data'>Get temperature data</h2><span id='topic+get_temp_data'></span>

<h3>Description</h3>

<p>This function takes information of where and when a set of environmental samples were
collected and retrieves temperature data (measured in accumulated degree-days) for those locations and times. Data come from
the Open-Meteo Historical Weather API (<a href="https://open-meteo.com/en/docs/historical-weather-api">https://open-meteo.com/en/docs/historical-weather-api</a>)
via the <a href="https://CRAN.R-project.org/package=openmeteo"><code>openmeteo</code></a> R package. The optional <code>intervals</code> argument
specifies a set of intervals over which the function will calculate the accumulated temperature in the form of Accumulated Thermal Units (ATUs) for each interval.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_temp_data(lon, lat, dates, intervals = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_temp_data_+3A_lon">lon</code></td>
<td>
<p>A numeric vector giving the longitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_temp_data_+3A_lat">lat</code></td>
<td>
<p>A numeric vector giving the latitude of the sampling sites in Decimal Degrees.</p>
</td></tr>
<tr><td><code id="get_temp_data_+3A_dates">dates</code></td>
<td>
<p>A character or date vector of dates giving the date when each sample was
collected (format is YYYY-MM-DD)</p>
</td></tr>
<tr><td><code id="get_temp_data_+3A_intervals">intervals</code></td>
<td>
<p>An integer vector giving a set of time intervals over to calculate accumulated degree-days. Default
is NULL where the interval is 0 (returns the daily temperature in degrees Celsius at time t). If <code>intervals</code>=3 then the accumulated
degree-days for the preceding 3 days is returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

tmp &lt;- get_temp_data(lon = c(30.0281, -52.9857),
                     lat = c(15.9094, -25.8756),
                     dates = c("2020-08-01", "2020-12-31"),
                     intervals = c(1,5,10))

head(tmp)


</code></pre>

<hr>
<h2 id='template_WES_data'>Template environmental sampling data</h2><span id='topic+template_WES_data'></span>

<h3>Description</h3>

<p>The <code>template_WES_data</code> object provides a template of the data format required by the 'WES' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>template_WES_data
</code></pre>


<h3>Format</h3>



<h4><code>template_WES_data</code></h4>

<p>A data frame with 6 columns:
</p>

<dl>
<dt>date</dt><dd><p>The date each sample was collected. Formate is &quot;YYYY-MM-DD&quot;.</p>
</dd>
<dt>location_id</dt><dd><p>A unique identifier for each of the sampling locations.</p>
</dd>
<dt>lat</dt><dd><p>The lattitude of the sampling location in decimal degrees.</p>
</dd>
<dt>lon</dt><dd><p>The longitude of the sampling location in decimal degrees.</p>
</dd>
<dt>target_name</dt><dd><p>The unique name of the gene target for which the Ct values correspond.</p>
</dd>
<dt>ct_value</dt><dd><p>The Cycle Threshold (Ct) of the qPCR assay.</p>
</dd>
</dl>



<hr>
<h2 id='template_WES_standard_curve'>Template standard curve data</h2><span id='topic+template_WES_standard_curve'></span>

<h3>Description</h3>

<p>The <code>template_WES_standard_curve</code> object provides a template of the data format required
by the 'WES' package for standard curve values. These data are only required when calculating the
number of gene copies using the <code>calc_n_copies</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>template_WES_standard_curve
</code></pre>


<h3>Format</h3>



<h4><code>template_WES_standard_curve</code></h4>

<p>A data frame with 3 columns:
</p>

<dl>
<dt>target_name</dt><dd><p>The unique name of the gene target for which the Ct values correspond.</p>
</dd>
<dt>n_copies</dt><dd><p>The number of gene copies represented in the particular dilution.</p>
</dd>
<dt>ct_value</dt><dd><p>The Cycle Threshold (Ct) of the qPCR assay.</p>
</dd>
</dl>



</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
