<!DOCTYPE html><html lang="en"><head><title>Help for package wgaim</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wgaim}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#wgaim-package'><p>Whole Genome Average Interval Mapping (wgaim) for QTL detection</p>
and estimation</a></li>
<li><a href='#cross2int'><p>Convert a cross genetic object to an interval object</p></a></li>
<li><a href='#genoCxR'><p>Genotypic marker data for Cascades x RAC875-2 doubled haploid</p>
population in R/qtl format</a></li>
<li><a href='#genoRxK'><p>Genotypic marker data for RAC875 x Kukri doubled haploid</p>
population in R/qtl format</a></li>
<li><a href='#genoSxT'><p>Genotypic marker data for Sunco x Tasman doubled haploid</p>
population in R/qtl format</a></li>
<li><a href='#linkMap.cross'><p>Plot a genetic linkage map</p></a></li>
<li><a href='#linkMap.default'><p>Plot a genetic linkage map with QTL for multiple traits</p></a></li>
<li><a href='#linkMap.wgaim'><p>Plot a genetic linkage map with QTL</p></a></li>
<li><a href='#outStat'><p>A faceted ggplot of the chromosome outlier statistics or the</p>
interval blups/outlier statistics obtained from specified iteratons of wgaim.</a></li>
<li><a href='#phenoCxR'><p>Phenotypic Cascades x RAC875-2 zinc experiment data frame</p></a></li>
<li><a href='#phenoRxK'><p>Phenotypic RAC875 x Kukri trial data frame</p></a></li>
<li><a href='#phenoSxT'><p>Phenotypic Sunco x Tasman trial data frame</p></a></li>
<li><a href='#qtlTable'><p>Stack QTL summary information into a super table</p></a></li>
<li><a href='#summary.wgaim'><p>Summary and print methods for the class <code>"wgaim"</code></p></a></li>
<li><a href='#tr.wgaim'><p>Display diagnostic information about the QTL detected.</p></a></li>
<li><a href='#wgaim-internal'><p>Internal wgaim functions</p></a></li>
<li><a href='#wgaim.asreml'><p>wgaim method for class &quot;asreml&quot;</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Whole Genome Average Interval Mapping for QTL Detection and
Estimation using ASReml-R</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0-6</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), qtl</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, prettydoc</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>asreml-R 4.x</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/DrJ001/wgaim">https://github.com/DrJ001/wgaim</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/DrJ001/wgaim/issues">https://github.com/DrJ001/wgaim/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>A computationally efficient whole genome approach to detecting and estimating significant QTL in linkage maps using the flexible linear mixed modelling functionality of ASReml-R.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-27 00:34:25 UTC; a1193984</td>
</tr>
<tr>
<td>Author:</td>
<td>Julian Taylor [aut, cre],
  Ari Verbyla [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Julian Taylor &lt;julian.taylor@adelaide.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-27 04:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='wgaim-package'>Whole Genome Average Interval Mapping (wgaim) for QTL detection
and estimation
</h2><span id='topic+wgaim-package'></span>

<h3>Description</h3>

<p>This package provides an efficient computational implementation of the QTL
whole genome QTL analysis algorithm (wgaim) discussed in Verbyla et. al
(2007, 2012) using extensions of the functionality provided in the
linear mixed modelling R package <span class="pkg">ASReml-R</span> V4.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> wgaim</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0-0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-08-12</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL 2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Welcome to version 2.0 of <span class="pkg">wgaim</span>! The documentation given in this help file is
only brief and users should consult the vignette available with the
package by typing <code>vignette("wgaim")</code> at the prompt. Alternatively,
users can also consult the individual help files for the main functions of the
package.
</p>
<p>The package provides a user friendly function <code><a href="#topic+cross2int">cross2int</a></code>
for the conversion of <code>"cross"</code> objects created using
<code>read.cross</code> in Bromans <span class="pkg">qtl</span> package into an <code>"interval"</code>
object ready for use in <span class="pkg">wgaim</span>. Specifically, <code>cross2int()</code> performs additional
calculations to impute missing marker values on each of the
chromosomes across the full linkage map and also provides users with
genetic distances and recombination fractions for the intervals. The
returned object retains the class structure
of an object created with <code>read.cross</code> and therefore allows further
use with the <span class="pkg">qtl</span> package if desired.
</p>
<p>The package also provides a function for the graphical display of the
chromosomes of a <code>"cross"</code> object. The method function
<code><a href="#topic+linkMap">linkMap</a></code> displays the full or subsetted linkage map
according to chromosome or distance as well as displays non-overlapping
marker names on the right hand side.
</p>
<p>QTL analysis is conducted using the function <code><a href="#topic+wgaim">wgaim</a></code> which,
as its first argument, requires an <code>asreml</code> base model. High
dimensional genetic components are allowed (See <code>wgaim.asreml</code> for
more details). For convenience the default tracing of results from the
asreml models can be outputted to a file for further inspection. Outlier
statistics and marker effects from each iteration can be viewed using
<code><a href="#topic+outStat">outStat</a></code>. Diagnostics of the likelihood ratio test
performed for each forward step can be displayed using
<code><a href="#topic+tr.wgaim">tr.wgaim</a></code>. The function also displays an incremental
probability value matrix of the QTL ascertained at each forward step of the algorithm.
</p>
<p>Summary and print methods are available for the returned <code>"wgaim"</code>
object and provide users with a detailed report on the QTL, their
size, their flanking markers, significance (including LOD score if
desired) and approximate contribution to the genetic variance. The
returned <code>"wgaim"</code> object may also be plotted using the method
function <code><a href="#topic+linkMap">linkMap</a></code>. This function plots the full linkage map
subsetted for chromosome and distance as well as provides shaded
QTL regions and highlighted flanking markers. Plotting of QTL for
multiple traits is also possible (see <code><a href="#topic+linkMap.default">linkMap.default</a></code>).
</p>


<h3>Author(s)</h3>

<p>Julian Taylor and Ari Verbyla
Maintainer: Julian Taylor &lt;julian.taylor@adelaide.edu.au&gt;
</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical And
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="qtl.html#topic+qtl-package">qtl-package</a></code>
</p>

<hr>
<h2 id='cross2int'>Convert a cross genetic object to an interval object</h2><span id='topic+cross2int'></span>

<h3>Description</h3>

<p>Converts an object of class &quot;<code>cross</code>&quot; to an object with class
&quot;<code>interval</code>&quot;. The function also imputes missing markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cross2int(object, impute = "MartinezCurnow", consensus.mark = TRUE,
     id = "id", subset = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cross2int_+3A_object">object</code></td>
<td>
<p>an object of class <code>"cross"</code> that inherits one of the class
structures <code>"bc"</code>, <code>"dh"</code>, <code>"f2"</code>, <code>"riself"</code>.</p>
</td></tr>
<tr><td><code id="cross2int_+3A_impute">impute</code></td>
<td>
<p>a character string determining how missing values in
the linkage map should be imputed. If <code>"Broman"</code>, then missing values are
imputed according to Bromans rules. If <code>"MartinezCurnow"</code> then
missing values are imputed according to the rules of Martinez &amp;
Curnow (1994) (see reference list). The default is <code>"MartinezCurnow"</code> (see Details).</p>
</td></tr>
<tr><td><code id="cross2int_+3A_consensus.mark">consensus.mark</code></td>
<td>
<p>logical value. If <code>TRUE</code> co-locating marker sets
are condensed to form consensus markers (see Details). Defaults to
<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="cross2int_+3A_id">id</code></td>
<td>
<p>a character string or name of the unique identifier for each row of genotype
data (see Details). Defaults to <code>"id"</code></p>
</td></tr>
<tr><td><code id="cross2int_+3A_subset">subset</code></td>
<td>
<p>a possible character vector naming the subset of
chromosomes to be returned. Defaults to <code>NULL</code> implying return
all chromosomes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides the conversion of genetic data objects that have
already been generated using <code>read.cross()</code> from Bromans <span class="pkg">qtl</span>
package, to <code>"interval"</code> objects ready for use with
<code>wgaim</code>. Users should be aware that this function is restricted to
certain populations.  <code>object</code> must inherit one of the class
structures <code>"bc"</code>, <code>"dh"</code>, <code>"f2"</code>, <code>"riself"</code>.
</p>
<p>During the conversion process three important linkage map attributes are
assessed.
</p>

<ol>
<li><p> The map may be subsetted using the <code>subset</code> argument
</p>
</li>
<li><p> If <code>consensus.mark = TRUE</code> then co-located marker sets are reduced
to form single consensus markers before missing values are
imputed. The marker similarity is determined by
the genetic distances that are given in the map component for each linkage
group. If a set of markers co-locate the name of the first marker is
chosen and a single consensus marker is determined by coalescing the
genetic information from all markers in the set. A &quot;(C)&quot; is placed
after the marker name for easy identification. The markers removed
from each set are returned with the object and placed under
<code>"colocated.markers"</code> for inspection if required.
</p>
</li>
<li><p> Missing values are imputed according to
the argument given by <code>impute</code>. This imputation results in a
complete version of the marker data for each chromosome which is then
used to create the interval data component &quot;<code>interval.data</code>&quot;. The complete
marker data for each chromosome can be obtained from the &quot;<code>imputed.data</code>&quot; element of the
returned list. It is therefore also possible to perform whole genome <em>marker</em>
analysis using <code>wgaim</code>. See <code>wgaim.asreml</code> for more details.
</p>
</li></ol>



<h3>Value</h3>

<p>a list of class <code>"cross"</code> that also inherits the class
<code>"interval"</code>. The list contains the following components
</p>
<table role = "presentation">
<tr><td><code>geno</code></td>
<td>

<p>A list with elements named by the corresponding names of the
chromosomes. Each chromosome is itself a list with six
elements: <code>"data"</code> is the actual estimated map matrix with rows
as individuals named by <code>"id"</code> and markers as columns;
<code>"map"</code> is a vector of marker positions on the
corresponding chromosome; <code>"imputed.data"</code> is identical to <code>"data"</code>
matrix but with all NAs replaced by imputed values according to the
rules of <code>"impute"</code>; <code>"dist"</code> contains the genetic
distance between adjacent markers or the genetic distances of the
intervals; <code>"theta"</code> contains the recombination fractions for
each interval; <code>"interval.data"</code> contains the recalculated intervals
based on the recombination fractions and the missing marker information.</p>
</td></tr>
<tr><td><code>colocated.markers</code></td>
<td>

<p>If <code>consensus.mark = TRUE</code>, a four column data frame containing
stacked binned sets of co-located markers. In each binned set the first
marker is the unique consensus marker name used in the linkage map and the
others are the co-located marker names that were omitted. Additionally
for each binned set, the data frame also contains linkage group
identification and marker position information.</p>
</td></tr>
<tr><td><code>pheno</code></td>
<td>

<p>A data.frame of phenotypic information with rows as individuals read
in from <code>read.cross</code>. A copy of the column named by the <code>"id"</code> argument
can be found here (see the help for the <code>read.cross() function</code>).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Julian Taylor and Ari Verblya</p>


<h3>References</h3>

<p>Martinez, O., Curnow. R. N. (1994) Missing markers when estimating
quantitative trait loci using regression mapping. <em>Heredity</em>,
<b>73</b>, 198-206.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical And
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="qtl.html#topic+read.cross">read.cross</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# read in linkage map from a rotated .CSV file with "id" as the
# identifier for each unique row

wgpath &lt;- system.file("extdata", package = "wgaim")
genoSxT &lt;- read.cross("csvr", file="genoSxT.csv", genotypes=c("AA","BB"),
         na.strings = c("-", "NA"), dir = wgpath)
genoSxT &lt;- cross2int(genoSxT, impute="MartinezCurnow", id = "id")

# plot linkage map

linkMap(genoSxT, cex = 0.5)


## End(Not run)
</code></pre>

<hr>
<h2 id='genoCxR'>Genotypic marker data for Cascades x RAC875-2 doubled haploid
population in R/qtl format</h2><span id='topic+genoCxR'></span>

<h3>Description</h3>

<p>Linkage map marker data for the Cascades x RAC875-2 doubled
haploid population in the form of an R/qtl cross object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genoCxR)</code></pre>


<h3>Format</h3>

<p>This data relates to a linkage map of 663 markers genotyped on 93
individuals. The linkage map consists of 42 linkage groups spanning
the whole genome. Coincident markers have been removed reducing the
linkage map to 458 markers. Map distances have been
estimated using <code>read.cross()</code> with the haldane
mapping function. The data object is therefore an R/qtl
cross object. See <code>read.cross()</code> documentation for more details on the
format of this object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(genoCxR, package = "wgaim")
linkMap(genoCxR, cex = 0.5)

</code></pre>

<hr>
<h2 id='genoRxK'>Genotypic marker data for RAC875 x Kukri doubled haploid
population in R/qtl format</h2><span id='topic+genoRxK'></span>

<h3>Description</h3>

<p>Linkage map marker data for the RAC875 x Kukri doubled haploid
population in the form of an R/qtl cross object.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenoSxT)</code></pre>


<h3>Format</h3>

<p>This data relates to a linkage map of 500 genetic markers genotyped on 368
individuals from the RAC875 x Kukri population. The linkage map
consists of 21 linkage groups with varying numbers of markers. Map
distances have been estimated using <code>read.cross()</code> with the kosambi
mapping function. The data is therefore an R/qtl
cross object. See <code>read.cross()</code> documentation for more details on the
format of this object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(genoRxK, package = "wgaim")
linkMap(genoRxK, cex = 0.5)

</code></pre>

<hr>
<h2 id='genoSxT'>Genotypic marker data for Sunco x Tasman doubled haploid
population in R/qtl format</h2><span id='topic+genoSxT'></span>

<h3>Description</h3>

<p>Linkage map marker data for the Sunco x Tasman doubled haploid
population in the form of an R/qtl cross object.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenoSxT)</code></pre>


<h3>Format</h3>

<p>This data relates to a linkage map of 287 genetic markers genotyped on 190
individuals from the Sunco x Tasman population. This set is reduced from
the original 345 markers (a mixture of AFLP, RFLP and microsatellite
markers and protein analysis). The reduction was created by discarding
58 markers which were co-located with one or more other markers. The
linkage map consists of 21 linkage groups with varying numbers of
markers. Map distances have been estimated using <code>read.cross()</code>
with the kosambi mapping function. The data is therefore an R/qtl
cross object. See <code>read.cross()</code> documentation for more details on the
format of this object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(genoSxT, package = "wgaim")
linkMap(genoSxT, cex = 0.5)

</code></pre>

<hr>
<h2 id='linkMap.cross'>Plot a genetic linkage map</h2><span id='topic+linkMap.cross'></span><span id='topic+linkMap'></span>

<h3>Description</h3>

<p>Neatly plots the genetic linkage map with marker locations and marker names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cross'
linkMap(object, chr, chr.dist, marker.names = "markers",
     tick = FALSE, squash = TRUE, m.cex = 0.6, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="linkMap.cross_+3A_object">object</code></td>
<td>
<p>object of class <code>"cross"</code></p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_chr">chr</code></td>
<td>
<p>character string naming the subset of chromosomes to plot</p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_chr.dist">chr.dist</code></td>
<td>
<p>a list containing named elements <code>"start"</code> and
<code>"end"</code> containing the start and end distances in cM the
genetic map should be subsetted by. Each of these may also be
a vector of distances equal to the length of the number of linkage
groups to be plotted.</p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_marker.names">marker.names</code></td>
<td>
<p>a character string naming the type of marker
information to plot. If <code>"dist"</code> then distances are
plotted alongside each chromosome on the left. If <code>"markers"</code>
then marker names are plotted instead. Defaults to <code>"markers"</code></p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_tick">tick</code></td>
<td>
<p>logical value. If <code>TRUE</code> then an axis with tick marks are
generated for the chromosome names. Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_squash">squash</code></td>
<td>
<p>logical value. if <code>TRUE</code> then creates extra room on
the left side of the chromosomes. This is useful for plotting trait
names for QTL using <code>linkMap.wgaim()</code> and
<code>linkMap.default()</code></p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_m.cex">m.cex</code></td>
<td>
<p>the expansion factor to use for the marker names</p>
</td></tr>
<tr><td><code id="linkMap.cross_+3A_...">...</code></td>
<td>
<p>arguments passed to <code>plot()</code> function to set up the plot
region. Arguments may also be passed to <code>text()</code> function for the
manipulation of the marker names</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plotting procedure provides a visual display of the
chromosomes without marker names overlapping vertically. The plotting
region will adjust itself to ensure that all marker names are in the region. For
this reason the value for <code>"m.cex"</code> is passed to the <code>text()</code>
function and should be manipulated until an aesthetic genetic map is reached.
</p>
<p>For large maps with many chromosomes, marker names and adjacent
chromosomes will overlap horizontally. For the interest of readability
this has not been corrected. For this particular situation it is
suggested that the user horizontally maximise the plotting window
until no overlapping occurs or subset the genetic map to achieve the
desired result.
</p>


<h3>Value</h3>

<p>This invisibly returns the following list for manipulation with
<code>linkMap.wgaim()</code>
</p>
<table role = "presentation">
<tr><td><code>mt</code></td>
<td>
<p>A list named by the chromosomes with each element containing the locations of
the marker names after correcting for overlapping</p>
</td></tr>
<tr><td><code>map</code></td>
<td>
<p>A list named by the chromosomes with each element
containing the locations of markers on the chromosomes</p>
</td></tr>
<tr><td><code>chrpos</code></td>
<td>
<p>The numerical position of the chromosomes on the
plotting region</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+linkMap.wgaim">linkMap.wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(genoSxT, package = "wgaim")

## plot linkage map with marker names

linkMap(genoSxT, cex = 0.5)

## plot linkage map with distances

linkMap(genoSxT, cex = 0.5, marker.names = "dist")

</code></pre>

<hr>
<h2 id='linkMap.default'>Plot a genetic linkage map with QTL for multiple traits</h2><span id='topic+linkMap.default'></span>

<h3>Description</h3>

<p>Neatly plots the genetic linkage map with marker locations,
marker names and highlights QTL with their associated flanking markers
for multiple traits obtained from a list of <code>wgaim</code> models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## Default S3 method:
linkMap(object, intervalObj, chr, chr.dist, marker.names
    = "markers", flanking = TRUE, list.col = list(q.col = rainbow(length(object)),
    m.col = "red", t.col = rainbow(length(object))), list.cex =
    list(m.cex = 0.6, t.cex = 0.6), trait.labels = NULL, tick = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="linkMap.default_+3A_object">object</code></td>
<td>
<p>a list object with elements inheriting the class <code>"wgaim"</code></p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_intervalobj">intervalObj</code></td>
<td>
<p>object of class <code>"cross"</code> or <code>"interval"</code></p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_chr">chr</code></td>
<td>
<p>character string naming the subset of chromosomes to plot</p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_chr.dist">chr.dist</code></td>
<td>
<p>a list containing named elements <code>"start"</code> and
<code>"end"</code> containing the start and end distances in cM the
genetic map should be subsetted by. Each of these may also be
a vector of distances equal to the length of the number of linkage
groups to be plotted.</p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_marker.names">marker.names</code></td>
<td>
<p>a character string naming the type of marker
information to plot. If <code>"dist"</code> then distances are
plotted alongside each chromosome on the left. If <code>"markers"</code>
then marker names are plotted instead. Defaults to
<code>"markers"</code>.</p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_flanking">flanking</code></td>
<td>
<p>logical value. If <code>TRUE</code> then only plot marker
names or distances for flanking markers of the QTL. Defaults to
<code>TRUE</code></p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_list.col">list.col</code></td>
<td>
<p>named list of colors used to highlight the QTL regions and
their flanking markers. <code>q.col</code> is the colors of the QTL
regions (defaults to <code>rainbow(n)</code> where <code>n</code> is the length
of <code>object</code>). <code>m.col</code> is the color the flanking markers. <code>t.col</code>
is the color of the trait names used in each model (defaults to the
same color as the QTL regions). See <code>par</code> for color options</p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_list.cex">list.cex</code></td>
<td>
<p>a named list object containing the character expansion
factors for the marker names <code>m.cex</code> and the trait labels <code>t.cex</code></p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_trait.labels">trait.labels</code></td>
<td>
<p>character string naming the trait used in the
model object, defaults to the response names extracted from the
fixed compoenent of the base model.</p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_tick">tick</code></td>
<td>
<p>logical value. If <code>TRUE</code> then an axis with tick marks are
generated for the chromosome names</p>
</td></tr>
<tr><td><code id="linkMap.default_+3A_...">...</code></td>
<td>
<p>arguments passed to the <code>plot()</code> function to set up the plot
region. Arguments may also be passed to the <code>text()</code> function for the
manipulation of the marker names</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plotting procedure is a wrapper for <code>linkMap.wgaim</code> and displays
QTL for multiple traits obtained from a list of models given by <code>object</code>.
Alternative labels for the traits can be given, in model order, using
<code>trait.labels</code>.
</p>
<p>Color specific highligting of the QTL is also available using
<code>clist</code>. This differs slightly from <code>linkMap.wgaim</code>. Here
the <code>q.col</code> and <code>t.col</code> should be given a set of colors equal to
the length of <code>object</code>. Let <code>n</code> be the length of <code>object</code>.
Then if <code>q.col</code> is <code>NULL</code> or length of <code>q.col</code> is not equal
to <code>n</code> then it defaults to <code>rainbow(n)</code>. If <cite>t.col</cite> is
<code>NULL</code> or length of <code>t.col</code> is not equal to <code>n</code> or 1 then
it defaults to the colors of <code>q.col</code>. Examples of different color
combinations are given below in the examples.
</p>
<p>The <code>list.cex</code> argument can be used to manipulate the character expansion of
the marker names using <code>m.cex</code> or the character expansion of the
<code>trait.labels</code> using <code>t.cex</code>. If a set of <code>"marker"</code> analyses has been
performed then <code>pch</code> is used to plot a symbol at the
location of the QTL. This character can be changed using the usual
arguments such as <code>pch</code> or <code>cex</code> that are passed through the
... argument.
</p>


<h3>Value</h3>

<p>For a set of <code>"interval"</code> analyses, the genetic linkage map is
plotted with shaded QTL regions and highlighted flanking markers. For
a set of <code>"marker"</code> analyses, symbols are
placed at the QTL locations and the markers are highlighted.
</p>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+linkMap.cross">linkMap.cross</a></code>, <code><a href="#topic+linkMap.wgaim">linkMap.wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
## fit wgaim models

rktgw.qtl &lt;- wgaim(rktgw.asf, intervalObj = genoRxK, merge.by = "Genotype",
                 trace = "trace.txt", na.action = na.method(x = "include"))

rkyld.qtl &lt;- wgaim(rkyld.asf, intervalObj = genoRxK, merge.by = "Genotype",
                 trace = "trace.txt", na.action = na.method(x = "include"))

## plot QTL intervals

# matching rainbow QTL color and trait names, red flanking markers
# (default) and gray background markers.

linkMap(list(rktgw.qtl,rkyld.qtl), genoRxK, col = "gray")

# rainbow QTL color and black trait names, red flanking markers
# (default) and gray background markers.

linkMap(list(rktgw.qtl,rkyld.qtl), genoRxK, list.col = list(t.col =
"black", m.col = "red"), col = "gray")

# monochromatic plot: gray QTLs, black trait names, black flanking
# markers and gray background markers

linkMap(list(rktgw.qtl,rkyld.qtl), genoRxK, list.col = list(q.col =
rep(gray(0.8), 2), t.col = "black", mcol = "black"), col = "gray")


## End(Not run)
</code></pre>

<hr>
<h2 id='linkMap.wgaim'>Plot a genetic linkage map with QTL</h2><span id='topic+linkMap.wgaim'></span>

<h3>Description</h3>

<p>Neatly plots the genetic linkage map with marker locations,
marker names and highlights QTL with their associated flanking markers
obtained from a <code>wgaim</code> model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'wgaim'
linkMap(object, intervalObj, chr, chr.dist,
    marker.names = "markers", flanking = TRUE, list.col = list(q.col = "light blue",
    m.col = "red", t.col = "light blue"), list.cex = list(t.cex = 0.6,
    m.cex = 0.6), trait.labels = NULL, tick = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="linkMap.wgaim_+3A_object">object</code></td>
<td>
<p>object of class <code>"wgaim"</code></p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_intervalobj">intervalObj</code></td>
<td>
<p>object of class <code>"cross"</code> or <code>"interval"</code></p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_chr">chr</code></td>
<td>
<p>character string naming the subset of chromosomes to plot</p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_chr.dist">chr.dist</code></td>
<td>
<p>a list containing named elements <code>"start"</code> and
<code>"end"</code> containing the start and end distances in cM the
genetic map should be subsetted by. Each of these may also be
a vector of distances equal to the length of the number of linkage
groups to be plotted.</p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_marker.names">marker.names</code></td>
<td>
<p>a character string naming the type of marker
information to plot. If <code>"dist"</code> then distances names
plotted alongside each chromosome on the left. If <code>"markers"</code>
then marker names are plotted instead. Defaults to
<code>"markers"</code>.</p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_flanking">flanking</code></td>
<td>
<p>logical value. If <code>TRUE</code> then only plot marker
names or distances for flanking markers of the QTL. Defualts to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_list.col">list.col</code></td>
<td>
<p>named list of colours used to highlight the QTL regions and
their flanking markers. <code>q.col</code> is the color of the QTL
regions. <code>m.col</code> is the color the flanking markers. <code>t.col</code>
is the color of the trait name used in the model object (see
<code>par</code> for colour options)</p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_list.cex">list.cex</code></td>
<td>
<p>a named list object containing the character expansion
factors for the marker names <code>m.cex</code> and the trait labels <code>t.cex</code></p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_trait.labels">trait.labels</code></td>
<td>
<p>character string naming the trait used in the
model object</p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_tick">tick</code></td>
<td>
<p>logical value. If <code>TRUE</code> then an axis with tick marks are
generated for the chromosome names</p>
</td></tr>
<tr><td><code id="linkMap.wgaim_+3A_...">...</code></td>
<td>
<p>arguments passed to the <code>plot()</code> function to set up the plot
region and plot any symbols if required</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plotting procedure builds on <code>linkMap.cross()</code> by adding the
QTL regions to the map and highlighting the appropriate markers obtained
from a fit to <code>wgaim</code>. If the linkage map is subsetted and QTL
regions fall outside the remaining map a warning will be given that
the QTL have been omitted from the display.
</p>
<p>The <code>list.col</code> arguments <code>q.col</code>, <code>m.col</code> and
<code>t.col</code> have been added for personal colour highlighting of the QTL
regions, flanking markers and trait names. For greater flexibility the
procedure may also be given the usual <code>col</code> argument that will be
passed to the other markers.
</p>
<p>The <code>list.cex</code> argument can be used to manipulate the character expansion of
the marker names using <code>m.cex</code> or the character expansion of the
<code>trait.labels</code> using <code>t.cex</code>. If a <code>"marker"</code> analysis has been
performed then <code>pch</code> is used to plot a symbol at the
location of the QTL. This character can be changed using the usual
arguments such as <code>pch</code> or <code>cex</code> that are passed through the
... argument.
</p>


<h3>Value</h3>

<p>For an <code>"interval"</code> analysis, the genetic linkage map is
plotted with shaded QTL regions and highlighted flanking markers. For
a <code>"marker"</code> analysis, a symbol is placed at the QTL locations and
the markers are highlighted.
</p>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+linkMap.cross">linkMap.cross</a></code>, <code><a href="#topic+wgaim">wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# fit wgaim model

yield.qtl &lt;- wgaim(yield.fm, intervalObj = genoRxK, merge.by = "Genotype",
                  trace = "trace.txt", na.action = na.method(x = "include"))

# plot QTL

linkMap(yield.qtl, genoRxK, list.col = list(m.col = "red"), col = "gray")


## End(Not run)
</code></pre>

<hr>
<h2 id='outStat'>A faceted ggplot of the chromosome outlier statistics or the
interval blups/outlier statistics obtained from specified iteratons of wgaim.</h2><span id='topic+outStat'></span>

<h3>Description</h3>

<p>A faceted <code>ggplot()</code> of the chromosome outlier statistics or the
interval blups/outlier statistics from specified iteratons of
<code>wgaim</code>. The interval blups/outlier statistics appear as
a trace across the genome separated by chromosomes and appropriately
spaced by their cM distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outStat(object, intervalObj, iter = NULL, chr = NULL, statistic =
        "outlier", plot.chr = FALSE, chr.lines = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outStat_+3A_object">object</code></td>
<td>
<p>object of class <code>"wgaim"</code>.
</p>
</td></tr>
<tr><td><code id="outStat_+3A_intervalobj">intervalObj</code></td>
<td>
<p>object of class <code>"interval"</code>.
</p>
</td></tr>
<tr><td><code id="outStat_+3A_iter">iter</code></td>
<td>
<p>range of integers determining which iterations will be
plotted.
</p>
</td></tr>
<tr><td><code id="outStat_+3A_chr">chr</code></td>
<td>
<p>character vector naming the subset of chromosomes to
plot.
</p>
</td></tr>
<tr><td><code id="outStat_+3A_statistic">statistic</code></td>
<td>
<p>character string naming the type of diagnostic statistic to be
plotted. Default is <code>"outlier"</code> (outlier statistics). Other option is
<code>"blups"</code> for the scaled empirical blups calculated during each
iteration.
</p>
</td></tr>
<tr><td><code id="outStat_+3A_plot.chr">plot.chr</code></td>
<td>
<p>logical value, if <code>TRUE</code> then plot chromosome outlier
statistics. If <code>FALSE</code> then plot interval outlier
statistics (see Details). Defaults to <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="outStat_+3A_chr.lines">chr.lines</code></td>
<td>
<p>logical value, if <code>TRUE</code> then plot vertical
lines to show separation of linkage groups. This is only useful if
<code>plot.chr = FALSE</code>. Defaults to <code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>plot.chr = TRUE</code> then outlier statistics for each chromosome
are plotted in separate faceted panels for specified values of <code>chr</code> and
<code>iter</code>. This option requies <code>selection="chromosome"</code> to be set
in the <code>wgaim.asreml()</code> call. If <code>plot.chr = FALSE</code> then interval blups or
outlier statistics are plotted in separate faceted panels for specified
values of <code>chr</code> and <code>iter</code>.
</p>
<p>Additionally, the set of significant QTL (chromosome and interval position) are
extracted from the model <code>object</code> and annotated on the plot in
their appropriate positions in each facet panel. Graphical aesthetics,
such as themes, text, font etc. can be further manipulated through
the inclusion of additional overlays to the returned <code>ggplot()</code> object.
</p>


<h3>Value</h3>

<p>The blups or outlier statistics are plotted in a faceted <code>ggplot()</code>
with information of significant QTL overlayed.
</p>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tr.wgaim">tr.wgaim</a></code>, <code><a href="#topic+wgaim">wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# fit wgaim model

rkyld.qtl &lt;- wgaim(rkyld.asf, intervalObj = genoRxK, merge.by = "Genotype",
                  trace = "trace.txt", na.action = na.method(x = "include"))

# plot QTL interval outlier statistics

outStat(rkyld.qtl, genoRxK, iter = 1:5)


## End(Not run)
</code></pre>

<hr>
<h2 id='phenoCxR'>Phenotypic Cascades x RAC875-2 zinc experiment data frame</h2><span id='topic+phenoCxR'></span>

<h3>Description</h3>

<p>Zinc concentration data of a Doubled Haploid wheat population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenoCxR)</code></pre>


<h3>Format</h3>

<p>This data relates to a glasshouse experiment involving a set of 90
Doubled Haploid (DH) lines from a crossing of Cascades x Rac875-2. The
DH lines were allocated randomly to pots in the glasshouse using a
randomised complete block design. There were also additional pots
that contained 5 of each of the parents (Cascades and Rac-875-2). Two
measurments were made, namely zinc concentration and shoot length.
The data frame consists of 200 rows and 5 columns decribed below
</p>

<dl>
<dt>id:</dt><dd><p>A factor of 92 levels conataining the unique
identification of the DH lines and parents.</p>
</dd>
<dt>Block:</dt><dd><p>A factor of two levels indexing the blocks in the experiment</p>
</dd>
<dt>Type:</dt><dd><p>A factor of 3 levels indexing the wheat variety (Doubled
Haploid, Cascades, Rac875-2)</p>
</dd>
<dt>shoot:</dt><dd><p>A numerical variable of shoot lengths for each plant</p>
</dd>
<dt>znconc:</dt><dd><p>A numerical variable of zinc concentration levels for each plant</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(phenoCxR, package = "wgaim")
plot(phenoCxR$shoot, phenoCxR$znconc)

</code></pre>

<hr>
<h2 id='phenoRxK'>Phenotypic RAC875 x Kukri trial data frame</h2><span id='topic+phenoRxK'></span>

<h3>Description</h3>

<p>Phenotype data arising from a field trial of a
Doubled Haploid population involving a crossing of the wheat varities
RAC875 and Kukri</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenoRxK)</code></pre>


<h3>Format</h3>

<p>This data relates to a field trial conducted in 2007 at the
Roseworthy Cmapus of the University of Adelaide. The trial consisted
of 2 replicates of 254 Doubled Haploid lines from a cross between wheat
varieties RAC875 and Kukri. The DH lines, the parents
(RAC875, Kukri) and control varieties (ATIL, SOKOLL, WEEBIL) were
randomly allocated to 520 plots using a randomized complete bloack
design. The trial was laid out in a 20 by 26 rectangular array. The
data frame consists of 520 rows with 9 columns described by:
</p>

<dl>
<dt>Genotype:</dt><dd><p>A 254 level factor containing a unique
identification for the wheat varieties involved in the experiment.</p>
</dd>
<dt>Type:</dt><dd><p>A factor of four levels indexing the wheat varieties
(Doubled Haploid, RAC875, Kukri, ATIL, SOKOLL, WEEBIL).</p>
</dd>
<dt>Range:</dt><dd><p>A factor of 20 numeric levels indexing the field Range.</p>
</dd>
<dt>Row:</dt><dd><p>A factor of 26 numeric levels indexing the field Rows.</p>
</dd>
<dt>Block:</dt><dd><p>A factor of 2 levels indexing the Blocks of the
experiment.</p>
</dd>
<dt>yield:</dt><dd><p>A numeric vector of yield observations in kg/ha.</p>
</dd>
<dt>tgw:</dt><dd><p>A numeric vector of thousand grain weight observations.</p>
</dd>
<dt>lrange:</dt><dd><p>A centred numerical vector of the field Ranges.</p>
</dd>
<dt>lrow:</dt><dd><p>A centred numerical vector of the field Rows.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(phenoRxK, package = "wgaim")

</code></pre>

<hr>
<h2 id='phenoSxT'>Phenotypic Sunco x Tasman trial data frame</h2><span id='topic+phenoSxT'></span>

<h3>Description</h3>

<p>Phenotype data arising from a two-phase experiment
involving a Doubled Haploid population from a crossing of the wheat varities
Sunco and Tasman</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenoSxT)</code></pre>


<h3>Format</h3>

<p>This data relates to a two-phase epxeriment involving a set of 175
Doubled Haploid lines. In the first phase DH lines were randomly
allocated to plots using a complete block design with additional plots
containing the parents (Sunco, Tasman) as well as commercial lines
(Frame, Janz, Krichauff, Machete, RAC820, Trident). The trial was laid
out in a rectangular array of 31 rows and 12 columns. In the second
phase 23% of the field samples were replicated in the milling
process producing a total of 456 milling samples. These partially
replicated field samples were then randomly allocated to 38
mill days with 12 samples per mill day. The data frame
consists of 456 rows with 11 columns. These columns are
</p>

<dl>
<dt>Expt:</dt><dd><p>A one level of factor containing a unique
identification for the experiment.</p>
</dd>
<dt>Type:</dt><dd><p>A factor of nine levels indexing the wheat variety
(Doubled Haploid, Sunco, Tasman, (Frame, Janz, Krichauff, Machete, RAC820, Trident))</p>
</dd>
<dt>id:</dt><dd><p>A factor of 183 levels uniquely identifying the wheat
varieties involved in the experiment.</p>
</dd>
<dt>Range:</dt><dd><p>A factor of 12 numeric levels indexing the field Range.</p>
</dd>
<dt>Row:</dt><dd><p>A factor of 31 numeric levels indexing the field Rows.</p>
</dd>
<dt>Rep:</dt><dd><p>A factor of 2 levels indexing the Block of the
experiment</p>
</dd>
<dt>Millday:</dt><dd><p>A factor of 38 numeric levels indexing the milling
day</p>
</dd>
<dt>Millord:</dt><dd><p>A factor of 12 levels indexing the milling order</p>
</dd>
<dt>myield:</dt><dd><p>A numeric vector of milling yield observations from
the second phase of the experiment.</p>
</dd>
<dt>lord:</dt><dd><p>A centered numerical vector of milling orders, Millord</p>
</dd>
<dt>lrow:</dt><dd><p>A centered numerical vector of Rows</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(phenoSxT, package = "wgaim")

</code></pre>

<hr>
<h2 id='qtlTable'>Stack QTL summary information into a super table</h2><span id='topic+qtlTable'></span>

<h3>Description</h3>

<p>Stack QTL summary information into a super table ready for
simple exporting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtlTable(..., intervalObj = NULL, labels = NULL, columns = "all")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qtlTable_+3A_...">...</code></td>
<td>
<p>list of objects of class <code>"wgaim"</code>. All
models must have been analysed with the same <code>gen.type</code>
(see help for <code>wgaim.asreml()</code>).</p>
</td></tr>
<tr><td><code id="qtlTable_+3A_intervalobj">intervalObj</code></td>
<td>
<p>a genetic object of class <code>"interval"</code>
reequire in a <code>wgaim</code> analysis  (see help for
<code>wgaim.asreml()</code>). This is required to be non NULL</p>
</td></tr>
<tr><td><code id="qtlTable_+3A_labels">labels</code></td>
<td>
<p>a vector of character strings describing the trait
names of each model QTL table.</p>
</td></tr>
<tr><td><code id="qtlTable_+3A_columns">columns</code></td>
<td>
<p>this can be either a numeric vector determining which columns of the QTL
summaries should be outputted or <code>"all"</code> for all
columns. The default is <code>"all"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The super table is created by obtaining the QTL summaries for each model
in <code>...</code> using <code>summary.wgaim()</code> and then row binding them
together. An extra column is created on the left hand side of the
super table for the trait names given in the <code>labels</code>
argument. If <code>labels = NULL</code> then trait names are extracted from
the left hand-side of the fixed component of the associated <code>wgaim</code>
model. The returned super table allows simple exporting to spreadsheet software packages
or with the R/LaTeX package <span class="pkg">xtable</span>.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> object with stacked QTL summaries
</p>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wgaim">wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 

## fit wgaim models

rktgw.qtl &lt;- wgaim(rktgw.asf, intervalObj = genoRxK, merge.by = "Genotype",
                   trace = "trace.txt", na.action = na.method(x = "include"))

rkyld.qtl &lt;- wgaim(rkyld.asf, intervalObj = genoRxK, merge.by = "Genotype",
                   trace = "trace.txt", na.action = na.method(x = "include"))

## create super table and export

qtlt &lt;- qtlTable(rktgw.qtl, rkyld.qtl, labels = c("Conc.", "Shoot"))
print(xtable(qtlt), file = "superQTL.tex", include.rownames = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='summary.wgaim'>Summary and print methods for the class <code>"wgaim"</code></h2><span id='topic+summary.wgaim'></span><span id='topic+print.wgaim'></span><span id='topic+print.summary.wgaim'></span>

<h3>Description</h3>

<p>Prints a QTL summary form the <code>"wgaim"</code> object in a presentable format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'wgaim'
summary(object, intervalObj, LOD = TRUE, ...)
## S3 method for class 'wgaim'
print(x, intervalObj, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.wgaim_+3A_object">object</code></td>
<td>
<p>an object of class <code>"wgaim"</code> (see Details)</p>
</td></tr>
<tr><td><code id="summary.wgaim_+3A_x">x</code></td>
<td>
<p>an object of class <code>"wgaim"</code></p>
</td></tr>
<tr><td><code id="summary.wgaim_+3A_intervalobj">intervalObj</code></td>
<td>
<p>a data structure of class <code>"cross"</code> or
<code>"interval"</code> containing the genotypic data</p>
</td></tr>
<tr><td><code id="summary.wgaim_+3A_lod">LOD</code></td>
<td>
<p>logical value. If TRUE LOD scores for QTL are calculated, defaults to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="summary.wgaim_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is important that the <code>intervalObj</code> is not missing in
<code>summary.wgaim()</code> or <code>print.wgaim()</code> as it
contains vital summary information about each of the QTL
detected.
</p>
<p>The summary of the QTL differs depending on the method chosen
in the <code>wgaim.asreml</code> call. If <code>method = "random"</code>
then the significance of the QTL are summarized using a probablistic
argument based on the conditional distribution of the QTL sizes given
the data (see Verbyla et. al, 2012 in References) Thus, for each
QTL, a value is calculated that represents the probability that the
QTL size is greater than zero (or less than zero if the effect is
negative). If <code>method = "fixed"</code> then the significance of the QTL is
summarized using a one degree of freedom Wald statistic.
</p>


<h3>Value</h3>

<p>A summary of the QTL component of the <code>"wgaim"</code> object is
printed to the screen. For each QTL detected, if an <code>"interval"</code>
analysis was performed then <code>summary.wgaim()</code> prints which
chromosome, name and distance of each flanking marker, size,
probability/p-value, contribution of genetic variance and LOD
score if desired. If a <code>"marker"</code> analysis was performed then the
chromosome, name and distance of the associated marker, size,
probability/p-value, contribution of genetic variance
and LOD score are printed. <code>print.wgaim()</code> provides a narrative
brief of the QTL detected.
</p>


<h3>Author(s)</h3>

<p>Julian Taylor and Ari Verbyla</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 195-211.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wgaim.asreml">wgaim.asreml</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# read in data

data(phenoRxK, package = "wgaim")
data(genoRxK, package = "wgaim")

# subset linkage map and convert to "interval" object

genoRxK &lt;- subset(genoRxK, chr = c("1A", "2D1", "2D2", "3B"))
genoRxK &lt;- cross2int(genoRxK, impute = "Martinez", id = "Genotype")

# base model

rkyld.asf &lt;- asreml(yld ~ lrow, random = ~ Genotype + Range,
                   residual = ~ ar1(Range):ar1(Row), data = phenoRxK)

# find QTL

rkyld.qtl &lt;- wgaim(rkyld.asf, intervalObj = genoRxK, merge.by = "Genotype",
                   trace = "trace.txt", na.action = na.method(x = "include"))

# summarise

print(rkyld.qtl, genoRxK)
summary(rkyld.qtl, genoRxK)


## End(Not run)
</code></pre>

<hr>
<h2 id='tr.wgaim'>Display diagnostic information about the QTL detected.</h2><span id='topic+tr'></span><span id='topic+tr.wgaim'></span>

<h3>Description</h3>

<p>Displays diagnostic infomation about QTL detection and
significance for the sequence of models generated in a <code>wgaim</code>
analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'wgaim'
tr(object, iter = 1:length(object$QTL$effects),
      lik.out = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tr.wgaim_+3A_object">object</code></td>
<td>
<p>an object of class <code>"wgaim"</code></p>
</td></tr>
<tr><td><code id="tr.wgaim_+3A_iter">iter</code></td>
<td>
<p>a vector of integers specifying what rows of the p-value matrix to display</p>
</td></tr>
<tr><td><code id="tr.wgaim_+3A_lik.out">lik.out</code></td>
<td>
<p>logical value. If <code>TRUE</code> then diagnostic information
about the testing of the genetic variance is given for all iterations.</p>
</td></tr>
<tr><td><code id="tr.wgaim_+3A_...">...</code></td>
<td>
<p>arguments passed to <code>print.default</code> for displaying
of information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the printing of the objects occur with arguments <code>quote = FALSE</code>
and <code>right = TRUE</code>. Users should avoid using these arguments.
</p>


<h3>Value</h3>

<p>For the selected QTL, a probability value matrix is displayed
with rows specified by <code>iter</code>. If <code>lik.out =
TRUE</code> then a matrix with rows consisting of the likelihood with
additive genetic variance, the likelihood without additive genetic
variance (NULL model), the test statistic and the p-value for the statistic.
</p>


<h3>Author(s)</h3>

<p>Julian Taylor</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wgaim">wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# read in data

data(phenoRxK, package = "wgaim")
data(genoRxK, package = "wgaim")

# subset linkage map and convert to "interval" object

genoRxK &lt;- subset(genoRxK, chr = c("1A", "2D1", "2D2", "3B"))
genoRxK &lt;- cross2int(genoRxK, impute = "Martinez", id = "Genotype")

# base model

rkyld.asf &lt;- asreml(yld ~ lrow, random = ~ Genotype + Range,
                   residual = ~ ar1(Range):ar1(Row), data = phenoRxK)

# find QTL

rkyld.qtl &lt;- wgaim(rkyld.asf, intervalObj = genoRxK, merge.by = "Genotype",
                   trace = "trace.txt", na.action = na.method(x = "include"))

# diagnostic check

tr(rkyld.qtl, digits = 4)

## End(Not run)
</code></pre>

<hr>
<h2 id='wgaim-internal'>Internal wgaim functions</h2><span id='topic+constructCM'></span><span id='topic+fixMap'></span><span id='topic+envFix'></span><span id='topic+getQTL'></span><span id='topic+addiag'></span><span id='topic+imputeGen'></span><span id='topic+wgaim.default'></span><span id='topic+qtlSelect'></span><span id='topic+vModify'></span><span id='topic+theme_scatter'></span>

<h3>Description</h3>

<p>These are internal <code>wgaim</code> functions that are not intended to be
directly called by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>constructCM(genoData, scale.method = "diag")
fixMap(full.data, rd = 3)
envFix(model, asremlEnv)
getQTL(object, intervalObj)
addiag(x = 1, di = 0, nrow.arg, ncol.arg = n)
imputeGen(theta, chr, dom = TRUE)
## Default S3 method:
wgaim(baseModel, ...)
qtlSelect(asm, phenoData, intervalObj, gen.type, selection,
          exclusion.window, state, verboseLev)
vModify(model, merge.by)
theme_scatter(base_size = 11, base_family = "")
</code></pre>


<h3>Author(s)</h3>

<p>Julian Taylor and Ari Verbyla</p>

<hr>
<h2 id='wgaim.asreml'>wgaim method for class &quot;asreml&quot;</h2><span id='topic+wgaim.asreml'></span><span id='topic+wgaim'></span>

<h3>Description</h3>

<p>Implements the iterative Whole Genome Average Interval Mapping
(wgaim) algorithm using the functionality of the flexible linear mixed
modelling R package <span class="pkg">ASReml-R</span> V4.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'asreml'
wgaim(baseModel, intervalObj, merge.by = NULL,
         fix.lines = TRUE, gen.type = "interval", method = "fixed",
         selection = "interval", force = FALSE, exclusion.window = 20,
         breakout = -1, TypeI = 0.05, trace = TRUE, verboseLev = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wgaim.asreml_+3A_basemodel">baseModel</code></td>
<td>
<p>a linear mixed model object of class <code>"asreml"</code> usually representing
a base <code>asreml()</code> model to be extended.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_intervalobj">intervalObj</code></td>
<td>
<p>a list object containing the genotypic data, usually an <code>"interval"</code>
object obtained from using <code>cross2int</code> (see Details).
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_merge.by">merge.by</code></td>
<td>
<p>a character string or name of the column(s) in <code>phenoData</code> and
<code>intervalObj</code> to merge the phenotypic and genotypic data sets.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_fix.lines">fix.lines</code></td>
<td>
<p>a logical value. If <code>TRUE</code> then lines existing in
the phenotype data that do not exist in <code>intervalObj</code> are fixed and placed in
the <code>fixed</code> component of the <code>asreml()</code> models (see
Details). It is recommended to set this to TRUE. Defaults to <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_gen.type">gen.type</code></td>
<td>
<p>a character string determining the type of genetic data to
be used in the analysis. Possibilities are <code>"interval"</code> and
<code>"markers"</code>. The default is <code>"interval"</code>. (see Details).
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_method">method</code></td>
<td>
<p>a character string determining the type of algorithm to
be used in the analysis. Possibilities are <code>"random"</code> and
<code>"fixed"</code>. The default is <code>"random"</code>. (see Details).
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_selection">selection</code></td>
<td>
<p>a character string determining the type of selection
method that is used to select QTL in the analysis. Possibilities are
<code>"interval"</code> and <code>"chromosome"</code>. The default is
<code>"interval"</code>. (see Details).
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_force">force</code></td>
<td>
<p>a logical value. If <code>TRUE</code> then force the algorithm to
to envoke the low-dimensional model that is normally used when the
number of markers is less than the number of lines. Defaults to <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_exclusion.window">exclusion.window</code></td>
<td>
<p>For each QTL, the distance in centimorgans on
the left and right side of each QTL that is excluded from further analysis.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_breakout">breakout</code></td>
<td>
<p>a numerical integer equivalent to the iteration
where the algorithm breaks out. The default is
-1 which ensures the algorithm finds all QTL before halting. (see
Details)
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_typei">TypeI</code></td>
<td>
<p>a numerical value determining the familywise alpha level of significance
for detecting a QTL. The default is 0.05.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_trace">trace</code></td>
<td>
<p>a automatic tracing facility. If <code>trace = TRUE</code> then
all <code>asreml</code> output is piped to the screen during the analysis.
If <code>trace = "file.txt"</code>, then output from all asreml models is
piped to &quot;<code>file.txt</code>&quot;. Both trace machanisms will display a
message if a QTL is detected.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_verboselev">verboseLev</code></td>
<td>
<p>Numerical value, either 0 or 1, determining the level of tracing
outputted during execution of the algorithm A 0 value will produce
the standard model fitting output from the fitted ASReml models
involved in the forward selection. A value of 1 will add a table of
chromosome and interval outlier statistics for each iteration.
</p>
</td></tr>
<tr><td><code id="wgaim.asreml_+3A_...">...</code></td>
<td>
<p>Any extra arguments to be passed to the <code>asreml</code>
calls. (see <code>?asreml</code> and <code>?asreml.options</code> for more
information).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the initial call to <code>wgaim.asreml()</code>, the marker or interval
information is collated from <code>intervalObj</code>. If <code>gen.type =
  "interval"</code> then midpoints of intervals are collated from the
<code>"interval.data"</code> components of the chromosomes in
<code>intervalObj</code>. If <code>gen.type = "markers"</code> then markers are
collated from the <code>"imputed.data"</code> components of the chromosomes in
<code>intervalObj</code>.
</p>
<p>It is recommended to set <code>fix.lines = TRUE</code> to ensure additive and
non-additive genetic variances are estimated from lines in the
<code>merge.by</code> component of the phenotypic data that have genetic
marker data in <code>intervalObj</code>. Lines in the phenotype
<code>merge.by</code> factor not existing in <code>intervalObj</code> will be placed
as a fixed factor (called <code>Gomit</code>) in the <code>asreml</code> model.
Note, if there are others factors in the model that have some
potential confounding with <code>Gomit</code> then <code>asreml</code> will
indicate this with a simple message <code>'Terms with zero df listed
in attribute 'zerodf' of the wald table'</code> at the end of its iterative
maximisation. This confounding will have no effect on the outcome and
can be safely ignored. If <code>fix.lines = FALSE</code> is set then all
available lines in the <code>merge.by</code> component of the phenotypic data
will be used to estimate the non-additive genetic variance.
In this instance, users also need to be aware that <code>asreml</code> will
output a large number of warnings due to an inherent mismatch in the levels
of the lines contained in the phenotype data compared to the lines in
<code>intervalObj</code>.
</p>
<p>The <code>method</code> argument in <code>wgaim.asreml()</code> allows the user access
to two algorithms. If <code>method = "fixed"</code> the algorithm
places selected QTL as an additive set of fixed
effects in the model as the forward selection algorithm proceeds. If <code>method = "random"</code>
places selected QTL in the random part of the model as
an additive set of random effects. This new formulation is outlined in
Verbyla et. al (2012).
</p>
<p>The <code>selection</code> argument determines the type of selection algorithm
for the analysis. If <code>selection = "chromosome"</code> then outlier
statistics for each chromosome are calculated and the largest
chromosome or linkage group is chosen. The largest marker/interval outlier
statistic in this linkage group is then selected as the putative QTL. If
<code>selection = "interval"</code>, only marker/interval statistics are calculated
and the largest marker/interval is chosen as the putative QTL.
</p>
<p>Note: If a genetic map has a small number of markers on a linkage group
then using <code>selection = "chromosome"</code> as the selection algorithm
is known to be flawed (see Verbyla et. al, 2012). For this reason it is suggested
that this option only be used when there are a moderate number of
markers on each linkage group.
</p>
<p>Users can break out of the algorithm using the <code>breakout</code>
argument. If a numerical value greater than zero is given, then the forward
selection algorithm breaks at the iteration equal to that value and
returns the collected information to this point. This includes
fixed/random QTL effects, diaganostic components such as interval/marker
BLUPs and outlier statistics as well as the trace components of the
algorithm. It should be noted that the algorithm breaks out before a QTL
has been moved to the fixed/random effects and estimated. Therefore a
positive integer, say <code>n</code> will not return an estimate of the nth
QTL but it will return the outlier statistics or BLUPs for the nth iteration.
</p>
<p>It is recommended that <code>trace = "file.txt"</code> be used to pipe the
sometimes invasive tracing of <code>asreml</code> licensing and fitting
numerics for each model to a file. Errors, warnings and messages will
still appear on screen during this process. Note some warnings that
appear may be passed through from an asreml call and are outputted upon
exit. These may be ignored as they are handled during the execution of
the function.
</p>


<h3>Value</h3>

<p>An object of class <code>"wgaim"</code> which also inherits the class
<code>"asreml"</code> by default. The object returned is actually an <code>asreml</code>
object (see <code>asreml.object</code>) with the addition of components from
the QTL detection listed below.
</p>
<table role = "presentation">
<tr><td><code>QTL</code></td>
<td>
<p>A list of components from the significant QTL detected
including a character vector of the significant QTL along with a
vector of the QTL effect sizes. There are also a number of diagnostic meausres that
can be found in <code>diag</code> that are used in conjunction with
<code>tr.wgaim</code> and <code>outStat</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Julian Taylor and Ari Verbyla</p>


<h3>References</h3>

<p>Verbyla, A. P &amp; Taylor, J. D, Verbyla, K. L (2012). RWGAIM: An efficient high
dimensional random whole genome average (QTL) interval mapping approach.
<em>Genetics Research</em>. <b>94</b>, 291-306.
</p>
<p>Julian Taylor, Arunas Vebyla (2011). R Package wgaim: QTL Analysis in
Bi-Parental Populations Using Linear Mixed Models. <em>Journal of
Statistical Software</em>, <b>40</b>(7), 1-18. URL <a href="https://www.jstatsoft.org/v40/i07/">https://www.jstatsoft.org/v40/i07/</a>.
</p>
<p>Verbyla, A. P., Cullis, B. R., Thompson, R (2007) The analysis of QTL
by simultaneous use of the full linkage map. <em>Theoretical and
Applied Genetics</em>, <b>116</b>, 95-111.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.wgaim">print.wgaim</a></code>, <code><a href="#topic+summary.wgaim">summary.wgaim</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
# read in data

data(phenoRxK, package = "wgaim")
data(genoRxK, package = "wgaim")

# subset linkage map and convert to "interval" object

genoRxK &lt;- subset(genoRxK, chr = c("1A", "2D1", "2D2", "3B"))
genoRxK &lt;- cross2int(genoRxK, impute = "Martinez", id = "Genotype")

# base model

rkyld.asf &lt;- asreml(yld ~ Type + lrow, random = ~ Genotype + Range,
                   residual = ~ ar1(Range):ar1(Row), data = phenoRxK)

# detect and estimate QTL

rkyld.qtl &lt;- wgaim(rkyld.asf, intervalObj = genoRxK, merge.by = "Genotype",
                   trace = "trace.txt", na.action = na.method(x = "include"))


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
