<!DOCTYPE html><html><head><title>Help for package wingen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wingen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#wingen-package'><p>wingen: Continuous Mapping of Genetic Diversity</p></a></li>
<li><a href='#circle_gd'><p>Create a moving window map of genetic diversity using a circle window</p></a></li>
<li><a href='#circle_general'><p>General function for making circular moving window maps</p></a></li>
<li><a href='#coords_to_raster'><p>Create a raster from coordinates</p></a></li>
<li><a href='#get_geodist'><p>Get a matrix of geographic distances for circle_gd</p></a></li>
<li><a href='#get_resdist'><p>Get a matrix of resistance distances for resist_gd</p></a></li>
<li><a href='#ggplot_count'><p>Plot moving window map of sample counts</p></a></li>
<li><a href='#ggplot_gd'><p>Plot moving window map of genetic diversity</p></a></li>
<li><a href='#krig_gd'><p>Krige moving window maps</p></a></li>
<li><a href='#load_middle_earth_ex'><p>Middle earth example</p></a></li>
<li><a href='#load_mini_ex'><p>Mini middle earth example</p></a></li>
<li><a href='#lotr_coords'><p>Middle earth example coordinates</p></a></li>
<li><a href='#lotr_lyr'><p>Middle earth example raster</p></a></li>
<li><a href='#lotr_range'><p>Middle earth example range polygon</p></a></li>
<li><a href='#lotr_vcf'><p>Middle earth example vcf</p></a></li>
<li><a href='#mask_gd'><p>Mask moving window maps</p></a></li>
<li><a href='#mini_coords'><p>Mini middle earth example coordinates</p></a></li>
<li><a href='#mini_lyr'><p>Mini middle earth example raster</p></a></li>
<li><a href='#mini_vcf'><p>Mini middle earth example vcf</p></a></li>
<li><a href='#mini_vcf_NA'><p>Mini middle earth example vcf with NA values</p></a></li>
<li><a href='#plot_count'><p>Plot moving window map of sample counts</p></a></li>
<li><a href='#plot_gd'><p>Plot moving window map of genetic diversity</p></a></li>
<li><a href='#preview_gd'><p>Preview moving window and sample counts</p></a></li>
<li><a href='#resist_gd'><p>Create a moving window map of genetic diversity based on resistance</p></a></li>
<li><a href='#resist_general'><p>General function for making resistance-based maps</p></a></li>
<li><a href='#vcf_to_dosage'><p>Convert a vcf to a dosage matrix</p></a></li>
<li><a href='#window_gd'><p>Create a moving window map of genetic diversity</p></a></li>
<li><a href='#window_general'><p>General function for making moving window maps</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Continuous Mapping of Genetic Diversity</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Generate continuous maps of genetic diversity using moving windows with options for rarefaction, interpolation, and masking as described in Bishop et al. (2023) &lt;<a href="https://doi.org/10.1111%2F2041-210X.14090">doi:10.1111/2041-210X.14090</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>automap, crayon, dplyr, furrr, gdistance, graphics, grDevices,
ggplot2, hierfstat, magrittr, pegas, purrr, raster, rlang, sf,
terra, tidyr, tidyselect, utils, vcfR, viridis</td>
</tr>
<tr>
<td>Suggests:</td>
<td>adegenet, covr, devtools, future, knitr, MASS, rmarkdown,
stringr, SpatialKDE, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-25 18:16:05 UTC; wanglab</td>
</tr>
<tr>
<td>Author:</td>
<td>Anusha Bishop <a href="https://orcid.org/0000-0003-1731-8683"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Anne Chambers <a href="https://orcid.org/0000-0002-7369-0108"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ian Wang <a href="https://orcid.org/0000-0003-2554-9414"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anusha Bishop &lt;anusha.bishop@berkeley.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-27 16:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='wingen-package'>wingen: Continuous Mapping of Genetic Diversity</h2><span id='topic+wingen'></span><span id='topic+wingen-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Generate continuous maps of genetic diversity using moving windows with options for rarefaction, interpolation, and masking as described in Bishop et al. (2023) <a href="https://doi.org/10.1111/2041-210X.14090">doi:10.1111/2041-210X.14090</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Anusha Bishop <a href="mailto:anusha.bishop@berkeley.edu">anusha.bishop@berkeley.edu</a> (<a href="https://orcid.org/0000-0003-1731-8683">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Anne Chambers <a href="mailto:eachambers@berkeley.edu">eachambers@berkeley.edu</a> (<a href="https://orcid.org/0000-0002-7369-0108">ORCID</a>)
</p>
</li>
<li><p> Ian Wang <a href="mailto:ianwang@berkeley.edu">ianwang@berkeley.edu</a> (<a href="https://orcid.org/0000-0003-2554-9414">ORCID</a>)
</p>
</li></ul>


<hr>
<h2 id='circle_gd'>Create a moving window map of genetic diversity using a circle window</h2><span id='topic+circle_gd'></span>

<h3>Description</h3>

<p>Generate a continuous raster map of genetic diversity using circle moving windows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circle_gd(
  gen,
  coords,
  lyr,
  maxdist,
  distmat = NULL,
  stat = "pi",
  fact = 0,
  rarify = FALSE,
  rarify_n = 2,
  rarify_nit = 5,
  min_n = 2,
  fun = mean,
  L = "nvariants",
  rarify_alleles = TRUE,
  sig = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circle_gd_+3A_gen">gen</code></td>
<td>
<p>genetic data either as an object of type vcf or a path to a vcf file (<em>note:</em> order matters! The coordinate and genetic data should be in the same order; there are currently no checks for this)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer to slide the window across (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_maxdist">maxdist</code></td>
<td>
<p>maximum geographic distance used to define neighborhood; any samples further than this distance will not be included (this can be thought of as the neighborhood radius)
Can either be (1) a single numeric value or (2) a SpatRaster where each pixel is the maximum distance to be used for that cell on the landscape (must be the same spatial scale as <code>lyr</code>).</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_distmat">distmat</code></td>
<td>
<p>distance matrix output from <a href="#topic+get_geodist">get_geodist</a> (optional; can be used to save time on distance calculations)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_stat">stat</code></td>
<td>
<p>genetic diversity statistic(s) to calculate (see Details, defaults to <code>"pi"</code>). Can be a single statistic or a vector of statistics</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_rarify">rarify</code></td>
<td>
<p>if rarify = TRUE, rarefaction is performed (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_rarify_n">rarify_n</code></td>
<td>
<p>if rarify = TRUE, number of points to use for rarefaction (defaults to min_n)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_rarify_nit">rarify_nit</code></td>
<td>
<p>if rarify = TRUE, number of iterations to use for rarefaction (defaults to 5). Can also be set to <code>"all"</code> to use all possible combinations of samples of size <code>rarify_n</code> within the window.</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA; defaults to 2)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_fun">fun</code></td>
<td>
<p>function to use to summarize rarefaction results (defaults to mean, must take <code>na.rm = TRUE</code> as an argument)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_l">L</code></td>
<td>
<p>for calculating <code>"pi"</code>, L argument in <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> function. Return the average nucleotide diversity per nucleotide given the length L of the sequence. The wingen default is L = &quot;nvariants&quot; which sets L to the number of variants in the VCF. If L = NULL, returns the sum over SNPs of nucleotide diversity (<em>note:</em> L = NULL is the <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> default which wingen does not use)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_rarify_alleles">rarify_alleles</code></td>
<td>
<p>for calculating <code>"biallelic_richness"</code>, whether to perform rarefaction of allele counts as in <a href="hierfstat.html#topic+allelic.richness">allelic.richness</a> (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="circle_gd_+3A_sig">sig</code></td>
<td>
<p>for calculating <code>"hwe"</code>, significance threshold (i.e., alpha level) to use for hardy-weinberg equilibrium tests (defaults to 0.05)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Coordinates and rasters should be in a Euclidean coordinate system (i.e., UTM coordinates) such that raster cell width and height are equal distances.
As such, longitude-latitude systems should be transformed before using dist_gd. Transformation can be performed using <a href="sf.html#topic+st_set_crs">st_set_crs</a> for coordinates or <a href="terra.html#topic+project">project</a> for rasters (see vignette for more details).
</p>


<h3>Value</h3>

<p>SpatRaster that includes a raster layer of genetic diversity and a raster layer of the number of samples within the window for each cell
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
load_mini_ex()
cpi &lt;- circle_gd(mini_vcf, mini_coords, mini_lyr, fact = 2, maxdist = 5)

</code></pre>

<hr>
<h2 id='circle_general'>General function for making circular moving window maps</h2><span id='topic+circle_general'></span>

<h3>Description</h3>

<p>Generate a continuous raster map using circular moving windows.
While <a href="#topic+resist_gd">resist_gd</a> is built specifically for making maps
of genetic diversity from vcfs,<code>circle_general</code> can be used to make maps
from different data inputs. Unlike <code>resist_gd</code>, <code>resist_general</code>
will not convert your data into the correct format for calculations of different
diversity metrics. See details for how to format data inputs for different statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circle_general(
  x,
  coords,
  lyr,
  maxdist,
  distmat = NULL,
  stat,
  fact = 0,
  rarify = FALSE,
  rarify_n = 2,
  rarify_nit = 5,
  min_n = 2,
  fun = mean,
  L = NULL,
  rarify_alleles = TRUE,
  sig = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circle_general_+3A_x">x</code></td>
<td>
<p>data to be summarized by the moving window (<em>note:</em> order matters! <code>coords</code> should be in the same order, there are currently no checks for this). The class of <code>x</code> required depends on the statistic being calculated (see the <code>stat</code> argument and the function description for more details)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer to slide the window across (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_maxdist">maxdist</code></td>
<td>
<p>maximum geographic distance used to define neighborhood; any samples further than this distance will not be included (this can be thought of as the neighborhood radius)
Can either be (1) a single numeric value or (2) a SpatRaster where each pixel is the maximum distance to be used for that cell on the landscape (must be the same spatial scale as <code>lyr</code>).</p>
</td></tr>
<tr><td><code id="circle_general_+3A_distmat">distmat</code></td>
<td>
<p>distance matrix output from <a href="#topic+get_geodist">get_geodist</a> (optional; can be used to save time on distance calculations)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_stat">stat</code></td>
<td>
<p>moving window statistic to calculate (see details). <code>stat</code> can generally be set to any function that will take <code>x</code>as input and return a single numeric value (for example, <code>x</code> can be a vector and <code>stat</code> can be set equal to a summary statistic like <code>mean</code>, <code>sum</code>, or <code>sd</code>)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_rarify">rarify</code></td>
<td>
<p>if rarify = TRUE, rarefaction is performed (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_rarify_n">rarify_n</code></td>
<td>
<p>if rarify = TRUE, number of points to use for rarefaction (defaults to min_n)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_rarify_nit">rarify_nit</code></td>
<td>
<p>if rarify = TRUE, number of iterations to use for rarefaction (defaults to 5). Can also be set to <code>"all"</code> to use all possible combinations of samples of size <code>rarify_n</code> within the window.</p>
</td></tr>
<tr><td><code id="circle_general_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA; defaults to 2)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_fun">fun</code></td>
<td>
<p>function to use to summarize rarefaction results (defaults to mean, must take <code>na.rm = TRUE</code> as an argument)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_l">L</code></td>
<td>
<p>for calculating <code>"pi"</code>, L argument in <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> function. Return the average nucleotide diversity per nucleotide given the length L of the sequence. The wingen default is L = &quot;nvariants&quot; which sets L to the number of variants in the VCF. If L = NULL, returns the sum over SNPs of nucleotide diversity (<em>note:</em> L = NULL is the <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> default which wingen does not use)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_rarify_alleles">rarify_alleles</code></td>
<td>
<p>for calculating <code>"biallelic_richness"</code>, whether to perform rarefaction of allele counts as in <a href="hierfstat.html#topic+allelic.richness">allelic.richness</a> (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_sig">sig</code></td>
<td>
<p>for calculating <code>"hwe"</code>, significance threshold (i.e., alpha level) to use for hardy-weinberg equilibrium tests (defaults to 0.05)</p>
</td></tr>
<tr><td><code id="circle_general_+3A_...">...</code></td>
<td>
<p>if a function is provided for <code>stat</code>, additional arguments to pass to the <code>stat</code> function (e.g. if <code>stat = mean</code>, users may want to set <code>na.rm = TRUE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To calculate genetic diversity statistics with the built in wingen functions, data must be formatted as such:
</p>

<ul>
<li><p> for <code>"pi"</code> or  <code>"biallelic_richness"</code>, <code>x</code> must be a dosage matrix with values of 0, 1, or 2
</p>
</li>
<li><p> for <code>"Ho"</code>, <code>x</code> must be a heterozygosity matrix where values of 0 = homozygosity and values of 1 = heterozygosity
</p>
</li>
<li><p> for <code>"allelic_richness"</code> or <code style="white-space: pre;">&#8288;"hwe&#8288;</code>, <code>x</code> must be a <code>genind</code> type object
</p>
</li>
<li><p> for <code>"basic_stats"</code>, <code>x</code> must be a <code>hierfstat</code> type object
</p>
</li></ul>

<p>Otherwise, <code>stat</code> can be any function that takes a matrix or data frame and outputs a
single numeric value (e.g., a function that produces a custom diversity index);
however, this should be attempted with caution since this functionality has
not have been tested extensively and may produce errors.
</p>


<h3>Value</h3>

<p>SpatRaster that includes a raster layer of genetic diversity and a raster layer of the number of samples within the window for each cell
</p>

<hr>
<h2 id='coords_to_raster'>Create a raster from coordinates</h2><span id='topic+coords_to_raster'></span>

<h3>Description</h3>

<p>Generate a raster layer from coordinates which can be used in <a href="#topic+window_gd">window_gd</a> as the RasterLayer to move the window across
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coords_to_raster(
  coords,
  buffer = 0,
  res = 1,
  agg = NULL,
  disagg = NULL,
  plot = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coords_to_raster_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a SpatVector, a two-column matrix, or a data.frame with x and y coordinates</p>
</td></tr>
<tr><td><code id="coords_to_raster_+3A_buffer">buffer</code></td>
<td>
<p>size of buffer to add to edge of raster (defaults to 0)</p>
</td></tr>
<tr><td><code id="coords_to_raster_+3A_res">res</code></td>
<td>
<p>desired resolution of raster (defaults to 1). Can be a single value for square cells or a vector with two values representing x and y resolutions</p>
</td></tr>
<tr><td><code id="coords_to_raster_+3A_agg">agg</code></td>
<td>
<p>aggregation factor to apply to raster (defaults to NULL)</p>
</td></tr>
<tr><td><code id="coords_to_raster_+3A_disagg">disagg</code></td>
<td>
<p>disaggregation factor to apply to raster (defaults to NULL)</p>
</td></tr>
<tr><td><code id="coords_to_raster_+3A_plot">plot</code></td>
<td>
<p>whether to plot resulting raster with coords (defaults to FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>RasterLayer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
coords_to_raster(mini_coords, buffer = 1, plot = TRUE)
</code></pre>

<hr>
<h2 id='get_geodist'>Get a matrix of geographic distances for <a href="#topic+circle_gd">circle_gd</a></h2><span id='topic+get_geodist'></span>

<h3>Description</h3>

<p>Create a distance matrix based on coordinates and a raster layer.
The output is a distance matrix where rows represent cells on the landscape
and columns represent individual locations on the landscape. Each value is
the geographic distance between each individual and each cell calculated
using <a href="sf.html#topic+st_distance">st_distance</a>. This matrix is used by <a href="#topic+circle_gd">circle_gd</a>.
If coords_only = TRUE, the result is a distance matrix for the sample coordinates
only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_geodist(coords, lyr = NULL, fact = 0, coords_only = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_geodist_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="get_geodist_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer for generating distances (not required if coords_only = TRUE)</p>
</td></tr>
<tr><td><code id="get_geodist_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="get_geodist_+3A_coords_only">coords_only</code></td>
<td>
<p>whether to return distances only for sample coordinates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a distance matrix used by <a href="#topic+circle_gd">circle_gd</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
distmat &lt;- get_geodist(mini_coords, mini_lyr)

</code></pre>

<hr>
<h2 id='get_resdist'>Get a matrix of resistance distances for <a href="#topic+resist_gd">resist_gd</a></h2><span id='topic+get_resdist'></span>

<h3>Description</h3>

<p>Create a distance matrix based on coordinates and a connectivity layer.
The output is a distance matrix where rows represent cells on the landscape
and columns represent individual locations on the landscape. Each value is
the resistance distance between each sample and each cell calculated
using the gdistance package. This matrix is used by <a href="#topic+resist_gd">resist_gd</a>.
If coords_only = TRUE, the result is a distance matrix for the sample coordinates
only.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_resdist(
  coords,
  lyr,
  fact = 0,
  transitionFunction = mean,
  directions = 8,
  geoCorrection = TRUE,
  coords_only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_resdist_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="get_resdist_+3A_lyr">lyr</code></td>
<td>
<p>conductivity layer (higher values should mean greater conductivity) for generating distances. Can be either a SpatRaster or RasterLayer.</p>
</td></tr>
<tr><td><code id="get_resdist_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="get_resdist_+3A_transitionfunction">transitionFunction</code></td>
<td>
<p>function to calculate transition values from grid values (defaults to mean)</p>
</td></tr>
<tr><td><code id="get_resdist_+3A_directions">directions</code></td>
<td>
<p>directions in which cells are connected (4, 8, 16, or other), see <a href="raster.html#topic+adjacent">adjacent</a> (defaults to 8)</p>
</td></tr>
<tr><td><code id="get_resdist_+3A_geocorrection">geoCorrection</code></td>
<td>
<p>whether to apply correction to account for local distances (defaults to TRUE). Geographic correction is necessary for all objects of the class Transition that are either: (1) based on a grid in a geographic (lonlat) projection and covering a large area; (2) made with directions &gt; 4 (see <a href="gdistance.html#topic+geoCorrection">geoCorrection</a> for more details).</p>
</td></tr>
<tr><td><code id="get_resdist_+3A_coords_only">coords_only</code></td>
<td>
<p>whether to return distances only for sample coordinates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a distance matrix used by <a href="#topic+resist_gd">resist_gd</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
distmat &lt;- get_resdist(mini_coords, mini_lyr)
</code></pre>

<hr>
<h2 id='ggplot_count'>Plot moving window map of sample counts</h2><span id='topic+ggplot_count'></span>

<h3>Description</h3>

<p>Plot sample counts layer produced by <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_count(x, index = NULL, col = viridis::mako(100))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_count_+3A_x">x</code></td>
<td>
<p>single SpatRaster of counts or SpatRaster where indexed layer is sample counts</p>
</td></tr>
<tr><td><code id="ggplot_count_+3A_index">index</code></td>
<td>
<p>index of raster layers to plot (defaults to plotting the one called &quot;sample_count&quot;, if more than one layer is provided)</p>
</td></tr>
<tr><td><code id="ggplot_count_+3A_col">col</code></td>
<td>
<p>color palette to use for plotting (defaults to viridis::mako palette)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of ggplots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("mini_lyr")
ggplot_count(mini_lyr)
</code></pre>

<hr>
<h2 id='ggplot_gd'>Plot moving window map of genetic diversity</h2><span id='topic+ggplot_gd'></span>

<h3>Description</h3>

<p>Plot genetic diversity layer produced by <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_gd(x, bkg = NULL, index = NULL, col = viridis::magma(100))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_gd_+3A_x">x</code></td>
<td>
<p>output from <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a> (RasterStack where first layer is genetic diversity)</p>
</td></tr>
<tr><td><code id="ggplot_gd_+3A_bkg">bkg</code></td>
<td>
<p>optional raster or sf polygon</p>
</td></tr>
<tr><td><code id="ggplot_gd_+3A_index">index</code></td>
<td>
<p>index of raster layers to plot (defaults to plotting all of the layers except the one called &quot;sample_count&quot;, if more than one layer is provided)</p>
</td></tr>
<tr><td><code id="ggplot_gd_+3A_col">col</code></td>
<td>
<p>color palette to use for plotting (defaults to <a href="viridis.html#topic+magma">magma</a> palette)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of ggplots
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("mini_lyr")
ggplot_gd(mini_lyr)

</code></pre>

<hr>
<h2 id='krig_gd'>Krige moving window maps</h2><span id='topic+krig_gd'></span>

<h3>Description</h3>

<p>Perform interpolation of the raster(s) produced by <a href="#topic+window_gd">window_gd</a> using <a href="automap.html#topic+autoKrige">autoKrige</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>krig_gd(
  r,
  grd = NULL,
  index = 1,
  coords = NULL,
  agg_grd = NULL,
  disagg_grd = NULL,
  agg_r = NULL,
  disagg_r = NULL,
  autoKrige_output = FALSE,
  lower_bound = TRUE,
  upper_bound = TRUE,
  krig_method = "ordinary",
  resample = FALSE,
  resample_first = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="krig_gd_+3A_r">r</code></td>
<td>
<p>SpatRaster produced by <a href="#topic+window_gd">window_gd</a></p>
</td></tr>
<tr><td><code id="krig_gd_+3A_grd">grd</code></td>
<td>
<p>object to create grid for kriging; can be a SpatRaster or RasterLayer. If undefined, will use <code>r</code> to create a grid.</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_index">index</code></td>
<td>
<p>integer indices of layers in raster stack to krige (defaults to 1; i.e., the first layer)</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_coords">coords</code></td>
<td>
<p>if provided, kriging will occur based only on values at these coordinates. Can be provided as an sf points, a two-column matrix, or a data.frame representing x and y coordinates</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_agg_grd">agg_grd</code></td>
<td>
<p>factor to use for aggregation of <code>grd</code>, if provided (this will decrease the resolution of the final kriged raster; defaults to NULL)</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_disagg_grd">disagg_grd</code></td>
<td>
<p>factor to use for disaggregation of <code>grd</code>, if provided (this will increase the resolution of the final kriged raster; defaults to NULL)</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_agg_r">agg_r</code></td>
<td>
<p>factor to use for aggregation of <code>r</code>, if provided (this will decrease the number of points used in the kriging model; defaults to NULL)</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_disagg_r">disagg_r</code></td>
<td>
<p>factor to use for disaggregation, of <code>r</code> if provided (this will increase the number of points used in the kriging model; defaults to NULL)</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_autokrige_output">autoKrige_output</code></td>
<td>
<p>whether to return full output from <a href="automap.html#topic+autoKrige">autoKrige</a> including uncertainty rasters (defaults to FALSE). If TRUE, returns a list with the kriged input raster layer (&quot;raster&quot;), kriged variance (&quot;var&quot;), kriged standard deviation (&quot;stdev&quot;), and full autoKrige output (&quot;autoKrige_output&quot;).</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_lower_bound">lower_bound</code></td>
<td>
<p>if TRUE (default), converts all values in the kriged raster less than the minimum value of the input raster, to that minimum</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_upper_bound">upper_bound</code></td>
<td>
<p>if TRUE (default), converts all values in the kriged raster greater than the maximum value of the input raster, to that maximum</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_krig_method">krig_method</code></td>
<td>
<p>method to use for kriging. If <code>ordinary</code>, ordinary/simple kriging is performed (formula: ~ 1; default). If <code>universal</code>,  universal kriging is performed (formula = ~ x + y).</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_resample">resample</code></td>
<td>
<p>whether to resample <code>grd</code> or <code>r</code>. Set to <code>"r"</code> to resample <code>r</code> to <code>grd</code>. Set to <code>"grd"</code> to resample <code>grd</code> to <code>r</code> (defaults to FALSE for no resampling)</p>
</td></tr>
<tr><td><code id="krig_gd_+3A_resample_first">resample_first</code></td>
<td>
<p>if aggregation or disaggregation is used in addition to resampling, specifies whether to resample before (resample_first = TRUE) or after (resample_first = FALSE) aggregation/disaggregation (defaults to TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a SpatRaster object or a list of <a href="automap.html#topic+autoKrige">autoKrige</a> outputs (if autoKrige_output = TRUE)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
wpi &lt;- window_gd(mini_vcf, mini_coords, mini_lyr, L = 10, rarify = TRUE)
kpi &lt;- krig_gd(wpi, mini_lyr)
plot_gd(kpi, main = "Kriged Pi")

</code></pre>

<hr>
<h2 id='load_middle_earth_ex'>Middle earth example</h2><span id='topic+load_middle_earth_ex'></span>

<h3>Description</h3>

<p>Loads middle earth example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_middle_earth_ex(quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_middle_earth_ex_+3A_quiet">quiet</code></td>
<td>
<p>whether to hide message (defaults to FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>three objects are loaded (lotr_vcf, lotr_coords, and lotr_lyrs)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_middle_earth_ex()
</code></pre>

<hr>
<h2 id='load_mini_ex'>Mini middle earth example</h2><span id='topic+load_mini_ex'></span>

<h3>Description</h3>

<p>Loads mini middle earth example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_mini_ex(quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_mini_ex_+3A_quiet">quiet</code></td>
<td>
<p>whether to hide message (defaults to FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>three objects are assigned in the GlobalEnv (vcf, coords, and lyr)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
</code></pre>

<hr>
<h2 id='lotr_coords'>Middle earth example coordinates</h2><span id='topic+lotr_coords'></span>

<h3>Description</h3>

<p>Middle earth example coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lotr_coords
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 rows and 2 columns
</p>

<dl>
<dt>x</dt><dd><p>x coordinate</p>
</dd>
<dt>y</dt><dd><p>y coordinate</p>
</dd>
</dl>



<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='lotr_lyr'>Middle earth example raster</h2><span id='topic+lotr_lyr'></span>

<h3>Description</h3>

<p>RasterLayer of middle earth based on an example digital elevation model of Tolkien’s
Middle Earth produced by the Center for Geospatial Analysis at William &amp; Mary
(<a href="https://scholarworks.wm.edu/asoer/3/">Robert, 2020</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lotr_lyr
</code></pre>


<h3>Format</h3>

<p>RasterLayer
</p>


<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023) based on Rose, Robert A. (2020) GIS &amp; Middle Earth Presentation &amp; Data Set. William &amp; Mary. https://doi.org/10.21220/RKEZ-X707
</p>

<hr>
<h2 id='lotr_range'>Middle earth example range polygon</h2><span id='topic+lotr_range'></span>

<h3>Description</h3>

<p>sf polygon of range map
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lotr_range
</code></pre>


<h3>Format</h3>

<p>sf
</p>


<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='lotr_vcf'>Middle earth example vcf</h2><span id='topic+lotr_vcf'></span>

<h3>Description</h3>

<p>A Variant Call Format data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lotr_vcf
</code></pre>


<h3>Format</h3>

<p>Object of class vcfR with 100 individuals and 1000 loci
</p>


<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='mask_gd'>Mask moving window maps</h2><span id='topic+mask_gd'></span>

<h3>Description</h3>

<p>Mask genetic diversity layer produced by <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mask_gd(x, y, minval = NULL, maxval = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mask_gd_+3A_x">x</code></td>
<td>
<p>Raster object to mask</p>
</td></tr>
<tr><td><code id="mask_gd_+3A_y">y</code></td>
<td>
<p>Raster object or Spatial object to use as mask</p>
</td></tr>
<tr><td><code id="mask_gd_+3A_minval">minval</code></td>
<td>
<p>if y is a Raster object, value of y below which to mask</p>
</td></tr>
<tr><td><code id="mask_gd_+3A_maxval">maxval</code></td>
<td>
<p>if y is a Raster object, value of y above which to mask</p>
</td></tr>
</table>


<h3>Value</h3>

<p>RasterLayer
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("mini_lyr")
mpi &lt;- mask_gd(mini_lyr, mini_lyr, minval = 0.01)

</code></pre>

<hr>
<h2 id='mini_coords'>Mini middle earth example coordinates</h2><span id='topic+mini_coords'></span>

<h3>Description</h3>

<p>Mini middle earth example coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mini_coords
</code></pre>


<h3>Format</h3>

<p>A data frame with 10 rows and 2 columns
</p>

<dl>
<dt>x</dt><dd><p>x coordinate</p>
</dd>
<dt>y</dt><dd><p>y coordinate</p>
</dd>
</dl>



<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='mini_lyr'>Mini middle earth example raster</h2><span id='topic+mini_lyr'></span>

<h3>Description</h3>

<p>Small RasterLayer of middle earth based on an example digital elevation model of Tolkien's
Middle Earth produced by the Center for Geospatial Analysis at William &amp; Mary
(<a href="https://scholarworks.wm.edu/asoer/3/">Robert, 2020</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mini_lyr
</code></pre>


<h3>Format</h3>

<p>A RasterLayer of middle earth
</p>


<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='mini_vcf'>Mini middle earth example vcf</h2><span id='topic+mini_vcf'></span>

<h3>Description</h3>

<p>A Variant Call Format data set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mini_vcf
</code></pre>


<h3>Format</h3>

<p>Object of class vcfR with 10 individuals and 10 loci
</p>


<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='mini_vcf_NA'>Mini middle earth example vcf with NA values</h2><span id='topic+mini_vcf_NA'></span>

<h3>Description</h3>

<p>A Variant Call Format data set with NA values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mini_vcf_NA
</code></pre>


<h3>Format</h3>

<p>Object of class vcfR with 10 individuals and 10 loci
</p>


<h3>Source</h3>

<p>created from simulations in Bishop et al. (2023)
</p>

<hr>
<h2 id='plot_count'>Plot moving window map of sample counts</h2><span id='topic+plot_count'></span>

<h3>Description</h3>

<p>Plot sample counts layer produced by <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_count(
  x,
  index = NULL,
  breaks = 100,
  col = viridis::mako(breaks),
  main = NULL,
  box = FALSE,
  range = NULL,
  legend = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_count_+3A_x">x</code></td>
<td>
<p>single SpatRaster of counts or SpatRaster where indexed layer is sample counts</p>
</td></tr>
<tr><td><code id="plot_count_+3A_index">index</code></td>
<td>
<p>if a raster stack is provided, index of the sample count layer to plot
(defaults to plotting the layer named &quot;sample_count&quot; or the last layer of the stack)</p>
</td></tr>
<tr><td><code id="plot_count_+3A_breaks">breaks</code></td>
<td>
<p>number of breaks to use in color scale (defaults to 10)</p>
</td></tr>
<tr><td><code id="plot_count_+3A_col">col</code></td>
<td>
<p>color palette to use for plotting (defaults to viridis::magma palette)</p>
</td></tr>
<tr><td><code id="plot_count_+3A_main">main</code></td>
<td>
<p>character. Main plot titles (one for each layer to be plotted). You can use arguments <code>cex.main</code>, <code>font.main</code>, <code>col.main</code> to change the appearance; and <code>loc.main</code> to change the location of the main title (either two coordinates, or a character value such as &quot;topleft&quot;)</p>
</td></tr>
<tr><td><code id="plot_count_+3A_box">box</code></td>
<td>
<p>whether to include a box around the raster plot (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="plot_count_+3A_range">range</code></td>
<td>
<p>numeric. minimum and maximum values to be used for the continuous legend</p>
</td></tr>
<tr><td><code id="plot_count_+3A_legend">legend</code></td>
<td>
<p>whether to include legend</p>
</td></tr>
<tr><td><code id="plot_count_+3A_...">...</code></td>
<td>
<p>arguments passed to <code>plot("SpatRaster", "numeric")</code> and additional graphical arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot of sample counts
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("mini_lyr")
plot_count(mini_lyr)
</code></pre>

<hr>
<h2 id='plot_gd'>Plot moving window map of genetic diversity</h2><span id='topic+plot_gd'></span>

<h3>Description</h3>

<p>Plot genetic diversity layer produced by <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gd(
  x,
  bkg = NULL,
  index = NULL,
  col = viridis::magma(breaks),
  breaks = 100,
  main = NULL,
  box = FALSE,
  range = NULL,
  legend = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gd_+3A_x">x</code></td>
<td>
<p>output from <a href="#topic+window_gd">window_gd</a> or <a href="#topic+krig_gd">krig_gd</a> (SpatRaster where first layer is genetic diversity)</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_bkg">bkg</code></td>
<td>
<p>optional SpatRaster or other spatial object that will be plotted as the &quot;background&quot; in gray</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_index">index</code></td>
<td>
<p>if a raster stack is provided, index of the layer to plot (defaults to plotting all layers except layers named &quot;sample_count&quot;)</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_col">col</code></td>
<td>
<p>color palette to use for plotting (defaults to <a href="viridis.html#topic+magma">magma</a> palette)</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_breaks">breaks</code></td>
<td>
<p>number of breaks to use in color scale (defaults to 100)</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_main">main</code></td>
<td>
<p>character. Main plot titles (one for each layer to be plotted). You can use arguments <code>cex.main</code>, <code>font.main</code>, <code>col.main</code> to change the appearance; and <code>loc.main</code> to change the location of the main title (either two coordinates, or a character value such as &quot;topleft&quot;)</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_box">box</code></td>
<td>
<p>whether to include a box around the Raster plot (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_range">range</code></td>
<td>
<p>numeric. minimum and maximum values to be used for the continuous legend</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_legend">legend</code></td>
<td>
<p>whether to include legend</p>
</td></tr>
<tr><td><code id="plot_gd_+3A_...">...</code></td>
<td>
<p>arguments passed to <code>plot("SpatRaster", "numeric")</code> and additional graphical arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot of genetic diversity
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("mini_lyr")
plot_gd(mini_lyr)

</code></pre>

<hr>
<h2 id='preview_gd'>Preview moving window and sample counts</h2><span id='topic+preview_gd'></span>

<h3>Description</h3>

<p>Generate a preview of moving window size and sample counts based on the coordinates and
parameters to be supplied to <a href="#topic+window_gd">window_gd</a>, <a href="#topic+circle_gd">circle_gd</a>, or  <a href="#topic+resist_gd">resist_gd</a>.
The method to be used should be specified with <code>method = "window"</code>, <code>"circle"</code>, or <code>"resist"</code>. For <code>method = "window"</code>,
<code>wdim</code> must be specified. For <code>method = "circle"</code> or <code>"resist"</code>, <code>maxdist</code> must be specified and
<code>distmat</code> can also optionally be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preview_gd(
  lyr,
  coords,
  method = "window",
  wdim = 3,
  maxdist = NULL,
  distmat = NULL,
  fact = 0,
  sample_count = TRUE,
  min_n = 0,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preview_gd_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer to slide the window across (see Details for important information about projections). For <code>method = "resist"</code> this should also be the conductivity layer (see <a href="#topic+resist_gd">resist_gd</a>)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_method">method</code></td>
<td>
<p>which method to use to create preview (<code>"window"</code> for <a href="#topic+window_gd">window_gd</a>, <code>"circle"</code> for <a href="#topic+circle_gd">circle_gd</a>, or <code>"resist"</code> for <a href="#topic+resist_gd">resist_gd</a>; defaults to <code>"window"</code>)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_wdim">wdim</code></td>
<td>
<p>if <code>method = "window"</code>, dimensions (height x width) of window; if only one value is provided, a square window is created (defaults to 3 x 3 window)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_maxdist">maxdist</code></td>
<td>
<p>if <code>method = "circle"</code> or <code style="white-space: pre;">&#8288;method = "resist&#8288;</code>, the maximum geographic distance used to define the neighborhood; any samples further than this distance will not be included (see <a href="#topic+get_geodist">get_geodist</a> or <a href="#topic+get_resdist">get_resdist</a>)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_distmat">distmat</code></td>
<td>
<p>if <code>method = "circle"</code> or <code>method = "resist"</code>, an optional distance matrix to be used output from either <a href="#topic+get_geodist">get_geodist</a> or <a href="#topic+get_resdist">get_resdist</a>, respectively. If not provided, one will be automatically calculated.</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_sample_count">sample_count</code></td>
<td>
<p>whether to create plot of sample counts for each cell (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA)</p>
</td></tr>
<tr><td><code id="preview_gd_+3A_plot">plot</code></td>
<td>
<p>whether to plot results (default = TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Coordinates and rasters should be in a projected (planar) coordinate system such that raster cells are of equal sizes.
Therefore, spherical systems (including latitute-longitude coordinate systems) should be projected prior to use.
Transformation can be performed using <a href="sf.html#topic+st_set_crs">st_set_crs</a> for coordinates or <a href="terra.html#topic+project">project</a> for rasters (see vignette for more details).
</p>


<h3>Value</h3>

<p>Plots preview of window and returns SpatRaster with sample counts layer (if sample_count = TRUE)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
preview_gd(mini_lyr, mini_coords, wdim = 3, fact = 3, sample_count = TRUE, min_n = 2)
</code></pre>

<hr>
<h2 id='resist_gd'>Create a moving window map of genetic diversity based on resistance</h2><span id='topic+resist_gd'></span>

<h3>Description</h3>

<p>Generate a continuous raster map of genetic diversity using resistance distances calculated with a conductivity surface
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resist_gd(
  gen,
  coords,
  lyr,
  maxdist,
  distmat = NULL,
  stat = "pi",
  fact = 0,
  rarify = FALSE,
  rarify_n = 2,
  rarify_nit = 5,
  min_n = 2,
  fun = mean,
  L = "nvariants",
  rarify_alleles = TRUE,
  sig = 0.05,
  transitionFunction = mean,
  directions = 8,
  geoCorrection = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resist_gd_+3A_gen">gen</code></td>
<td>
<p>genetic data either as an object of type vcf or a path to a vcf file (<em>note:</em> order matters! The coordinate and genetic data should be in the same order; there are currently no checks for this)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_lyr">lyr</code></td>
<td>
<p>conductivity layer (higher values should mean greater conductivity) to move window across. Can be either a SpatRaster or RasterLayer.</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_maxdist">maxdist</code></td>
<td>
<p>maximum cost distance used to define neighborhood; any samples further than this cost distance will not be included (this can be thought of as the neighborhood radius, but in terms of cost distance).
Can either be (1) a single numeric value or (2) a SpatRaster where each pixel is the maximum distance to be used for that cell on the landscape (must be the same spatial scale as <code>lyr</code>).</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_distmat">distmat</code></td>
<td>
<p>distance matrix output from <a href="#topic+get_resdist">get_resdist</a> (optional; can be used to save time on distance calculations)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_stat">stat</code></td>
<td>
<p>genetic diversity statistic(s) to calculate (see Details, defaults to <code>"pi"</code>). Can be a single statistic or a vector of statistics</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_rarify">rarify</code></td>
<td>
<p>if rarify = TRUE, rarefaction is performed (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_rarify_n">rarify_n</code></td>
<td>
<p>if rarify = TRUE, number of points to use for rarefaction (defaults to min_n)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_rarify_nit">rarify_nit</code></td>
<td>
<p>if rarify = TRUE, number of iterations to use for rarefaction (defaults to 5). Can also be set to <code>"all"</code> to use all possible combinations of samples of size <code>rarify_n</code> within the window.</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA; defaults to 2)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_fun">fun</code></td>
<td>
<p>function to use to summarize rarefaction results (defaults to mean, must take <code>na.rm = TRUE</code> as an argument)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_l">L</code></td>
<td>
<p>for calculating <code>"pi"</code>, L argument in <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> function. Return the average nucleotide diversity per nucleotide given the length L of the sequence. The wingen default is L = &quot;nvariants&quot; which sets L to the number of variants in the VCF. If L = NULL, returns the sum over SNPs of nucleotide diversity (<em>note:</em> L = NULL is the <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> default which wingen does not use)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_rarify_alleles">rarify_alleles</code></td>
<td>
<p>for calculating <code>"biallelic_richness"</code>, whether to perform rarefaction of allele counts as in <a href="hierfstat.html#topic+allelic.richness">allelic.richness</a> (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_sig">sig</code></td>
<td>
<p>for calculating <code>"hwe"</code>, significance threshold (i.e., alpha level) to use for hardy-weinberg equilibrium tests (defaults to 0.05)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_transitionfunction">transitionFunction</code></td>
<td>
<p>function to calculate transition values from grid values (defaults to mean)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_directions">directions</code></td>
<td>
<p>directions in which cells are connected (4, 8, 16, or other), see <a href="raster.html#topic+adjacent">adjacent</a> (defaults to 8)</p>
</td></tr>
<tr><td><code id="resist_gd_+3A_geocorrection">geoCorrection</code></td>
<td>
<p>whether to apply correction to account for local distances (defaults to TRUE). Geographic correction is necessary for all objects of the class Transition that are either: (1) based on a grid in a geographic (lonlat) projection and covering a large area; (2) made with directions &gt; 4 (see <a href="gdistance.html#topic+geoCorrection">geoCorrection</a> for more details).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Coordinates and rasters should be in a Euclidean coordinate system (i.e., UTM coordinates) such that raster cell width and height are equal distances.
As such, longitude-latitude systems should be transformed before using dist_gd. Transformation can be performed using <a href="sf.html#topic+st_set_crs">st_set_crs</a> for coordinates or <a href="terra.html#topic+project">project</a> for rasters (see vignette for more details).
</p>


<h3>Value</h3>

<p>SpatRaster that includes a raster layer of genetic diversity and a raster layer of the number of samples within the window for each cell
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load_mini_ex()
rpi &lt;- resist_gd(mini_vcf, mini_coords, mini_lyr, maxdist = 50)
</code></pre>

<hr>
<h2 id='resist_general'>General function for making resistance-based maps</h2><span id='topic+resist_general'></span>

<h3>Description</h3>

<p>Generate a continuous raster map using resistance distances.
While <a href="#topic+resist_gd">resist_gd</a> is built specifically for making maps
of genetic diversity from vcfs,<code>resist_general</code> can be used to make maps
from different data inputs. Unlike <code>resist_gd</code>, <code>resist_general</code>
will not convert your data into the correct format for calculations of different
diversity metrics. See details for how to format data inputs for different statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resist_general(
  x,
  coords,
  lyr,
  maxdist,
  distmat = NULL,
  stat,
  fact = 0,
  rarify = FALSE,
  rarify_n = 2,
  rarify_nit = 5,
  min_n = 2,
  fun = mean,
  L = NULL,
  rarify_alleles = TRUE,
  sig = 0.05,
  transitionFunction = mean,
  directions = 8,
  geoCorrection = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resist_general_+3A_x">x</code></td>
<td>
<p>data to be summarized by the moving window (<em>note:</em> order matters! <code>coords</code> should be in the same order, there are currently no checks for this). The class of <code>x</code> required depends on the statistic being calculated (see the <code>stat</code> argument and the function description for more details)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer to slide the window across (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_maxdist">maxdist</code></td>
<td>
<p>maximum cost distance used to define neighborhood; any samples further than this cost distance will not be included (this can be thought of as the neighborhood radius, but in terms of cost distance).
Can either be (1) a single numeric value or (2) a SpatRaster where each pixel is the maximum distance to be used for that cell on the landscape (must be the same spatial scale as <code>lyr</code>).</p>
</td></tr>
<tr><td><code id="resist_general_+3A_distmat">distmat</code></td>
<td>
<p>distance matrix output from <a href="#topic+get_resdist">get_resdist</a> (optional; can be used to save time on distance calculations)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_stat">stat</code></td>
<td>
<p>moving window statistic to calculate (see details). <code>stat</code> can generally be set to any function that will take <code>x</code>as input and return a single numeric value (for example, <code>x</code> can be a vector and <code>stat</code> can be set equal to a summary statistic like <code>mean</code>, <code>sum</code>, or <code>sd</code>)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_rarify">rarify</code></td>
<td>
<p>if rarify = TRUE, rarefaction is performed (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_rarify_n">rarify_n</code></td>
<td>
<p>if rarify = TRUE, number of points to use for rarefaction (defaults to min_n)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_rarify_nit">rarify_nit</code></td>
<td>
<p>if rarify = TRUE, number of iterations to use for rarefaction (defaults to 5). Can also be set to <code>"all"</code> to use all possible combinations of samples of size <code>rarify_n</code> within the window.</p>
</td></tr>
<tr><td><code id="resist_general_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA; defaults to 2)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_fun">fun</code></td>
<td>
<p>function to use to summarize rarefaction results (defaults to mean, must take <code>na.rm = TRUE</code> as an argument)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_l">L</code></td>
<td>
<p>for calculating <code>"pi"</code>, L argument in <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> function. Return the average nucleotide diversity per nucleotide given the length L of the sequence. The wingen default is L = &quot;nvariants&quot; which sets L to the number of variants in the VCF. If L = NULL, returns the sum over SNPs of nucleotide diversity (<em>note:</em> L = NULL is the <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> default which wingen does not use)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_rarify_alleles">rarify_alleles</code></td>
<td>
<p>for calculating <code>"biallelic_richness"</code>, whether to perform rarefaction of allele counts as in <a href="hierfstat.html#topic+allelic.richness">allelic.richness</a> (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_sig">sig</code></td>
<td>
<p>for calculating <code>"hwe"</code>, significance threshold (i.e., alpha level) to use for hardy-weinberg equilibrium tests (defaults to 0.05)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_transitionfunction">transitionFunction</code></td>
<td>
<p>function to calculate transition values from grid values (defaults to mean)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_directions">directions</code></td>
<td>
<p>directions in which cells are connected (4, 8, 16, or other), see <a href="raster.html#topic+adjacent">adjacent</a> (defaults to 8)</p>
</td></tr>
<tr><td><code id="resist_general_+3A_geocorrection">geoCorrection</code></td>
<td>
<p>whether to apply correction to account for local distances (defaults to TRUE). Geographic correction is necessary for all objects of the class Transition that are either: (1) based on a grid in a geographic (lonlat) projection and covering a large area; (2) made with directions &gt; 4 (see <a href="gdistance.html#topic+geoCorrection">geoCorrection</a> for more details).</p>
</td></tr>
<tr><td><code id="resist_general_+3A_...">...</code></td>
<td>
<p>if a function is provided for <code>stat</code>, additional arguments to pass to the <code>stat</code> function (e.g. if <code>stat = mean</code>, users may want to set <code>na.rm = TRUE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To calculate genetic diversity statistics with the built in wingen functions, data must be formatted as such:
</p>

<ul>
<li><p> for <code>"pi"</code> or <code>"biallelic_richness"</code>, <code>x</code> must be a dosage matrix with values of 0, 1, or 2
</p>
</li>
<li><p> for <code>"Ho"</code>, <code>x</code> must be a heterozygosity matrix where values of 0 = homozygosity and values of 1 = heterozygosity
</p>
</li>
<li><p> for <code>"allelic_richness"</code> or <code style="white-space: pre;">&#8288;"hwe&#8288;</code>, <code>x</code> must be a <code>genind</code> type object
</p>
</li>
<li><p> for <code>"basic_stats"</code>, <code>x</code> must be a <code>hierfstat</code> type object
</p>
</li></ul>

<p>Otherwise, <code>stat</code> can be any function that takes a matrix or data frame and outputs a
single numeric value (e.g., a function that produces a custom diversity index);
however, this should be attempted with caution since this functionality has
not have been tested extensively and may produce errors.
</p>


<h3>Value</h3>

<p>SpatRaster that includes a raster layer of genetic diversity and a raster layer of the number of samples within the window for each cell
</p>

<hr>
<h2 id='vcf_to_dosage'>Convert a vcf to a dosage matrix</h2><span id='topic+vcf_to_dosage'></span>

<h3>Description</h3>

<p>Convert a vcf to a dosage matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_to_dosage(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcf_to_dosage_+3A_x">x</code></td>
<td>
<p>can either be an object of class 'vcfR' or a path to a .vcf file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dosage matrix
</p>

<hr>
<h2 id='window_gd'>Create a moving window map of genetic diversity</h2><span id='topic+window_gd'></span>

<h3>Description</h3>

<p>Generate a continuous raster map of genetic diversity using moving windows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>window_gd(
  gen,
  coords,
  lyr,
  stat = "pi",
  wdim = 3,
  fact = 0,
  rarify = FALSE,
  rarify_n = NULL,
  rarify_nit = 5,
  min_n = 2,
  fun = mean,
  L = "nvariants",
  rarify_alleles = TRUE,
  sig = 0.05,
  crop_edges = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="window_gd_+3A_gen">gen</code></td>
<td>
<p>genetic data either as an object of type vcf or a path to a vcf file (<em>note:</em> order matters! The coordinate and genetic data should be in the same order; there are currently no checks for this)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer to slide the window across (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_stat">stat</code></td>
<td>
<p>genetic diversity statistic(s) to calculate (see Details, defaults to <code>"pi"</code>). Can be a single statistic or a vector of statistics</p>
</td></tr>
<tr><td><code id="window_gd_+3A_wdim">wdim</code></td>
<td>
<p>dimensions (height x width) of window; if only one value is provided, a square window is created (defaults to 3 x 3 window)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_rarify">rarify</code></td>
<td>
<p>if rarify = TRUE, rarefaction is performed (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_rarify_n">rarify_n</code></td>
<td>
<p>if rarify = TRUE, number of points to use for rarefaction (defaults to min_n)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_rarify_nit">rarify_nit</code></td>
<td>
<p>if rarify = TRUE, number of iterations to use for rarefaction (defaults to 5). Can also be set to <code>"all"</code> to use all possible combinations of samples of size <code>rarify_n</code> within the window.</p>
</td></tr>
<tr><td><code id="window_gd_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA; defaults to 2)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_fun">fun</code></td>
<td>
<p>function to use to summarize rarefaction results (defaults to mean, must take <code>na.rm = TRUE</code> as an argument)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_l">L</code></td>
<td>
<p>for calculating <code>"pi"</code>, L argument in <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> function. Return the average nucleotide diversity per nucleotide given the length L of the sequence. The wingen default is L = &quot;nvariants&quot; which sets L to the number of variants in the VCF. If L = NULL, returns the sum over SNPs of nucleotide diversity (<em>note:</em> L = NULL is the <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> default which wingen does not use)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_rarify_alleles">rarify_alleles</code></td>
<td>
<p>for calculating <code>"biallelic_richness"</code>, whether to perform rarefaction of allele counts as in <a href="hierfstat.html#topic+allelic.richness">allelic.richness</a> (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_sig">sig</code></td>
<td>
<p>for calculating <code>"hwe"</code>, significance threshold (i.e., alpha level) to use for hardy-weinberg equilibrium tests (defaults to 0.05)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_crop_edges">crop_edges</code></td>
<td>
<p>whether to remove cells on the edge of the raster where the window is incomplete (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="window_gd_+3A_...">...</code></td>
<td>
<p><a href="base.html#topic+deprecated">deprecated</a> this was intended to be used to pass additional arguments to the <code>stat</code> function, however now formal arguments are used instead (see <code>L</code>, <code>rarify_alleles</code>, and <code>sig</code>). Passing additional arguments using <code>...</code> is still possible with the <code style="white-space: pre;">&#8288;*_general()&#8288;</code> functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Coordinates and rasters should be in a projected (planar) coordinate system such that raster cells are of equal sizes.
Therefore, spherical systems (including latitute-longitude coordinate systems) should be projected prior to use.
Transformation can be performed using <a href="sf.html#topic+st_set_crs">st_set_crs</a> for coordinates or <a href="terra.html#topic+project">project</a> for rasters (see vignette for more details).
</p>
<p>Current genetic diversity metrics that can be specified with <code>stat</code> include:
</p>

<ul>
<li> <p><code>"pi"</code> for nucleotide diversity (default) calculated using <code>hierfstat</code> <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a>
</p>
</li>
<li> <p><code>"Ho"</code> for average observed heterozygosity across all sites
</p>
</li>
<li> <p><code>"allelic_richness"</code> for average number of alleles across all sites
</p>
</li>
<li> <p><code>"biallelic_richness"</code> for average allelic richness across all sites for a biallelic dataset (this option is faster than <code>"allelic_richness"</code>)
</p>
</li>
<li> <p><code>"hwe"</code> for the proportion of sites that are not in Hardy–Weinberg equilibrium, calculated using <code>pegas</code> <a href="pegas.html#topic+hw.test">hw.test</a> at the 0.05 level (other alpha levels  can be specified by adding the sig argument; e.g., <code>sig = 0.10</code>).
</p>
</li>
<li> <p><code>"basic_stats"</code> for a series of statistics produced by <code>hierfstat</code> <a href="hierfstat.html#topic+basic.stats">basic.stats</a> including
mean observed heterozygosity (same as Ho), mean gene diversities within population (Hs),
Gene diversities overall (Ht), and Fis following Nei (1987). Population-based statistics (e.g., FST) normally reported by <a href="hierfstat.html#topic+basic.stats">basic.stats</a>
are not included as they are not meaningful within the individual-based moving windows.
</p>
</li></ul>



<h3>Value</h3>

<p>SpatRaster that includes raster layers of genetic diversity and a raster layer of the number of samples within the window for each cell
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
load_mini_ex()
wpi &lt;- window_gd(mini_vcf, mini_coords, mini_lyr, rarify = TRUE)

</code></pre>

<hr>
<h2 id='window_general'>General function for making moving window maps</h2><span id='topic+window_general'></span>

<h3>Description</h3>

<p>Generate a continuous raster map using moving windows. While <a href="#topic+window_gd">window_gd</a> is
built specifically for making moving window maps of genetic diversity from vcfs,
<code>window_general</code> can be used to make moving window maps from different data inputs.
See details for how to format data inputs for different statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>window_general(
  x,
  coords,
  lyr,
  stat,
  wdim = 3,
  fact = 0,
  rarify = FALSE,
  rarify_n = NULL,
  rarify_nit = 5,
  min_n = 2,
  fun = mean,
  L = "nvariants",
  rarify_alleles = TRUE,
  sig = 0.05,
  crop_edges = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="window_general_+3A_x">x</code></td>
<td>
<p>data to be summarized by the moving window (<em>note:</em> order matters! <code>coords</code> should be in the same order, there are currently no checks for this). The class of <code>x</code> required depends on the statistic being calculated (see the <code>stat</code> argument and the function description for more details)</p>
</td></tr>
<tr><td><code id="window_general_+3A_coords">coords</code></td>
<td>
<p>coordinates of samples as sf points, a two-column matrix, or a data.frame representing x and y coordinates (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="window_general_+3A_lyr">lyr</code></td>
<td>
<p>SpatRaster or RasterLayer to slide the window across (see Details for important information about projections)</p>
</td></tr>
<tr><td><code id="window_general_+3A_stat">stat</code></td>
<td>
<p>moving window statistic to calculate (can either be <code>"pi"</code> for nucleotide diversity (<code>x</code> must be a dosage matrix), <code>"Ho"</code> for average observed heterozygosity across all loci (<code>x</code> must be a heterozygosity matrix) , <code>"allelic_richness"</code> for average allelic richness across all loci (<code>x</code> must be a <code>genind</code> type object), <code>"biallelic_richness"</code> to get average allelic richness across all loci for a biallelic dataset (<code>x</code> must be a dosage matrix). <code>stat</code> can also be set to any function that will take <code>x</code>as input and return a single numeric value (for example, <code>x</code> can be a vector and <code>stat</code> can be set equal to a summary statistic like <code>mean</code>, <code>sum</code>, or <code>sd</code>)</p>
</td></tr>
<tr><td><code id="window_general_+3A_wdim">wdim</code></td>
<td>
<p>dimensions (height x width) of window; if only one value is provided, a square window is created (defaults to 3 x 3 window)</p>
</td></tr>
<tr><td><code id="window_general_+3A_fact">fact</code></td>
<td>
<p>aggregation factor to apply to <code>lyr</code> (defaults to 0; <em>note:</em> increasing this value reduces computational time)</p>
</td></tr>
<tr><td><code id="window_general_+3A_rarify">rarify</code></td>
<td>
<p>if rarify = TRUE, rarefaction is performed (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="window_general_+3A_rarify_n">rarify_n</code></td>
<td>
<p>if rarify = TRUE, number of points to use for rarefaction (defaults to min_n)</p>
</td></tr>
<tr><td><code id="window_general_+3A_rarify_nit">rarify_nit</code></td>
<td>
<p>if rarify = TRUE, number of iterations to use for rarefaction (defaults to 5). Can also be set to <code>"all"</code> to use all possible combinations of samples of size <code>rarify_n</code> within the window.</p>
</td></tr>
<tr><td><code id="window_general_+3A_min_n">min_n</code></td>
<td>
<p>minimum number of samples to use in calculations (any focal cell with a window containing less than this number of samples will be assigned a value of NA; defaults to 2)</p>
</td></tr>
<tr><td><code id="window_general_+3A_fun">fun</code></td>
<td>
<p>function to use to summarize rarefaction results (defaults to mean, must take <code>na.rm = TRUE</code> as an argument)</p>
</td></tr>
<tr><td><code id="window_general_+3A_l">L</code></td>
<td>
<p>for calculating <code>"pi"</code>, L argument in <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> function. Return the average nucleotide diversity per nucleotide given the length L of the sequence. The wingen default is L = &quot;nvariants&quot; which sets L to the number of variants in the VCF. If L = NULL, returns the sum over SNPs of nucleotide diversity (<em>note:</em> L = NULL is the <a href="hierfstat.html#topic+pi.dosage">pi.dosage</a> default which wingen does not use)</p>
</td></tr>
<tr><td><code id="window_general_+3A_rarify_alleles">rarify_alleles</code></td>
<td>
<p>for calculating <code>"biallelic_richness"</code>, whether to perform rarefaction of allele counts as in <a href="hierfstat.html#topic+allelic.richness">allelic.richness</a> (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="window_general_+3A_sig">sig</code></td>
<td>
<p>for calculating <code>"hwe"</code>, significance threshold (i.e., alpha level) to use for hardy-weinberg equilibrium tests (defaults to 0.05)</p>
</td></tr>
<tr><td><code id="window_general_+3A_crop_edges">crop_edges</code></td>
<td>
<p>whether to remove cells on the edge of the raster where the window is incomplete (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="window_general_+3A_...">...</code></td>
<td>
<p>if a function is provided for <code>stat</code>, additional arguments to pass to the <code>stat</code> function (e.g. if <code>stat = mean</code>, users may want to set <code>na.rm = TRUE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To calculate genetic diversity statistics with the built in wingen functions, data must be formatted as such:
</p>

<ul>
<li><p> for <code>"pi"</code> or  <code>"biallelic_richness"</code>, <code>x</code> must be a dosage matrix with values of 0, 1, or 2
</p>
</li>
<li><p> for <code>"Ho"</code>, <code>x</code> must be a heterozygosity matrix where values of 0 = homozygosity and values of 1 = heterozygosity
</p>
</li>
<li><p> for <code>"allelic_richness"</code> or <code style="white-space: pre;">&#8288;"hwe&#8288;</code>, <code>x</code> must be a <code>genind</code> type object
</p>
</li>
<li><p> for <code>"basic_stats"</code>, <code>x</code> must be a <code>hierfstat</code> type object
</p>
</li></ul>

<p>Otherwise, <code>stat</code> can be any function that takes a matrix or data frame and outputs a
single numeric value (e.g., a function that produces a custom diversity index);
however, this should be attempted with caution since this functionality has
not have been tested extensively and may produce errors.
</p>


<h3>Value</h3>

<p>SpatRaster that includes a raster layer of genetic diversity and a raster layer of the number of samples within the window for each cell
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
