<!DOCTYPE html><html><head><title>Help for package wISAM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wISAM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#check_eigen_decomposition'><p>check_eigen_decomposition</p></a></li>
<li><a href='#covariate_mat'><p>Example covariate matrix</p></a></li>
<li><a href='#GenomeScan_calc_multiplier_eigen'><p>calc_multiplier_eigen</p></a></li>
<li><a href='#GenomeScan_conduct_scan'><p>conduct_scan</p></a></li>
<li><a href='#GenomeScan_fit_locus'><p>fit_locus</p></a></li>
<li><a href='#GenomeScan_fit_locus_given_h2'><p>fit_locus_given_h2</p></a></li>
<li><a href='#GenomeScan_initialize'><p>initialize</p></a></li>
<li><a href='#GenomeScan_prep_scan'><p>prep_scan</p></a></li>
<li><a href='#GenomeScan-class'><p>GenomeScan</p></a></li>
<li><a href='#kinship_mat'><p>Example kinship matrix</p></a></li>
<li><a href='#locus_list'><p>Example locus list</p></a></li>
<li><a href='#phenotype'><p>Example phenotype</p></a></li>
<li><a href='#se_mean_per_strain'><p>Example standard error of the mean per strain</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Weighted Inbred Strain Association Mapping</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.8</td>
</tr>
<tr>
<td>Author:</td>
<td>Robert W. Cort &lt;rcorty@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Robert W. Cort &lt;rcorty@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>In the course of a genome-wide association study, the situation often arises that some phenotypes are known with greater precision than others.
    It could be that some individuals are known to harbor more micro-environmental variance than others.
    In the case of inbred strains of model organisms, it could be the case that more organisms were observed from some strains than others, so the strains with more organisms have better-estimated means.
    Package 'wISAM' handles this situation by allowing for weighting of each observation according to residual variance.
    Specifically, the 'weight' parameter to the function conduct_scan() takes the precision of each observation (one over the variance). </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, MASS</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-31 17:18:15 UTC; rcorty</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-04 08:12:54 UTC</td>
</tr>
</table>
<hr>
<h2 id='check_eigen_decomposition'>check_eigen_decomposition</h2><span id='topic+check_eigen_decomposition'></span>

<h3>Description</h3>

<p>Grabbed from MASS.  Useful to sparsify matrices when some eigenvalues are essentially zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_eigen_decomposition(e, tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_eigen_decomposition_+3A_e">e</code></td>
<td>
<p>the eigen decomposition to check</p>
</td></tr>
<tr><td><code id="check_eigen_decomposition_+3A_tol">tol</code></td>
<td>
<p>the threshold below which a number is said to be effectively zero, defaults to 1e-6</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The eigen values with any values with absolute value less than tol zeroed.
</p>

<hr>
<h2 id='covariate_mat'>Example covariate matrix</h2><span id='topic+covariate_mat'></span>

<h3>Description</h3>

<p>Example covariate matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covariate_mat
</code></pre>


<h3>Format</h3>

<p>A matrix with 200 rows and 4 variables</p>

<hr>
<h2 id='GenomeScan_calc_multiplier_eigen'>calc_multiplier_eigen</h2><span id='topic+GenomeScan_calc_multiplier_eigen'></span>

<h3>Description</h3>

<p>Compute a multiplier (aka rotation) matrix.  Details in in h2lmm_math_RWC.Rmd.
</p>


<h3>Value</h3>

<p>an object of class GenomeScan
</p>

<hr>
<h2 id='GenomeScan_conduct_scan'>conduct_scan</h2><span id='topic+GenomeScan_conduct_scan'></span>

<h3>Description</h3>

<p>Conducts the GenomeScan.
</p>


<h3>Value</h3>

<p>an object of class GenomeScan
</p>


<h3>Note</h3>

<p>TODO: allow user to specify subset of chromosomes or loci
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(wISAM)

wgs &lt;- GenomeScan$new(y = phenotype,
                      X = covariate_mat,
                      G = locus_list,
                      K = kinship_mat,
                      w = 1/se_mean_per_strain)

result &lt;- wgs$prep_scan()$conduct_scan()

</code></pre>

<hr>
<h2 id='GenomeScan_fit_locus'>fit_locus</h2><span id='topic+GenomeScan_fit_locus'></span>

<h3>Description</h3>

<p>Fit one locus of a GenomeScan.  Should not typically be called by a user.
</p>


<h3>Value</h3>

<p>an object of class GenomeScan
</p>

<hr>
<h2 id='GenomeScan_fit_locus_given_h2'>fit_locus_given_h2</h2><span id='topic+GenomeScan_fit_locus_given_h2'></span>

<h3>Description</h3>

<p>Fit one locus at a specified value of h2.  Should not typically be called by a user.
</p>


<h3>Value</h3>

<p>an object of class GenomeScan
</p>

<hr>
<h2 id='GenomeScan_initialize'>initialize</h2><span id='topic+GenomeScan_initialize'></span>

<h3>Description</h3>

<p>Initialize a GenomeScan
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="GenomeScan_initialize_+3A_y">y</code></td>
<td>
<p>vector of length n - the phenotype of each of n genomes (individuals or strains)</p>
</td></tr>
<tr><td><code id="GenomeScan_initialize_+3A_x">X</code></td>
<td>
<p>matrix of dimension n-by-c - the covariate value of each individual for c covariates</p>
</td></tr>
<tr><td><code id="GenomeScan_initialize_+3A_g">G</code></td>
<td>
<p>a list where each element is of length n - the genotype of each individual at p loci</p>
</td></tr>
<tr><td><code id="GenomeScan_initialize_+3A_k">K</code></td>
<td>
<p>matrix of dimension n-by-n - the covariance of the phenotype</p>
</td></tr>
<tr><td><code id="GenomeScan_initialize_+3A_w">w</code></td>
<td>
<p>matrix of dimension n-by-n - the inverse variance of the phenotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class GenomeScan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(wISAM)

wgs &lt;- GenomeScan$new(y = phenotype,
                      X = covariate_mat,
                      G = locus_list,
                      K = kinship_mat,
                      w = 1/se_mean_per_strain)

</code></pre>

<hr>
<h2 id='GenomeScan_prep_scan'>prep_scan</h2><span id='topic+GenomeScan_prep_scan'></span>

<h3>Description</h3>

<p>Prepare a GenomeScan for running.  Does all the computations that need to be done exactly once per genome scan.
</p>


<h3>Value</h3>

<p>an object of class GenomeScan
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(wISAM)

wgs &lt;- GenomeScan$new(y = phenotype,
                      X = covariate_mat,
                      G = locus_list,
                      K = kinship_mat,
                      w = 1/se_mean_per_strain)

result &lt;- wgs$prep_scan()

</code></pre>

<hr>
<h2 id='GenomeScan-class'>GenomeScan</h2><span id='topic+GenomeScan-class'></span><span id='topic+GenomeScan'></span>

<h3>Description</h3>

<p>A Reference Class implementing a Genome Scan
</p>


<h3>Fields</h3>


<dl>
<dt><code>Options</code></dt><dd><p>these are options</p>
</dd>
<dt><code>Data</code></dt><dd><p>Things the user inputs.
They have interpretable meaning and define the GenomeScan.
Currently: y, X, G, K, weights (inverse variances), and variances.</p>
</dd>
<dt><code>Intermediates_per_scan</code></dt><dd><p>Things the GenomeScan will compute once per scan.
They are mathetmatical tools that can't really be interpreted.
Currently: L, eigen_L, and LL_null.</p>
</dd>
<dt><code>Intermediates_per_locus</code></dt><dd><p>Things the GenomeScan will compute once per locus.
They are mathematical tools that can't really be interpreted.
Currently: XG</p>
</dd>
<dt><code>Intermediates_per_fit</code></dt><dd><p>Things the GenomeScan will compute many times per locus (once per trial fit on that locus).
These are interpretable but rapidly changing and not guaranteed to be finalized or optimal.
Currently: M, LDV, and h2</p>
</dd>
<dt><code>Results</code></dt><dd><p>The results of the GenomeScan.
Currently: The h2 that maximizes the LL at each locus and the LR as compared with the no-locus (null) model.</p>
</dd>
</dl>

<hr>
<h2 id='kinship_mat'>Example kinship matrix</h2><span id='topic+kinship_mat'></span>

<h3>Description</h3>

<p>Example kinship matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinship_mat
</code></pre>


<h3>Format</h3>

<p>A matrix with 200 rows and 200 columns</p>

<hr>
<h2 id='locus_list'>Example locus list</h2><span id='topic+locus_list'></span>

<h3>Description</h3>

<p>Example locus list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_list
</code></pre>


<h3>Format</h3>

<p>A list of matrices, where each matrix has 200 rows and 1 column</p>

<hr>
<h2 id='phenotype'>Example phenotype</h2><span id='topic+phenotype'></span>

<h3>Description</h3>

<p>Example phenotype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenotype
</code></pre>


<h3>Format</h3>

<p>A vector of length 200</p>

<hr>
<h2 id='se_mean_per_strain'>Example standard error of the mean per strain</h2><span id='topic+se_mean_per_strain'></span>

<h3>Description</h3>

<p>Example standard error of the mean per strain
</p>


<h3>Usage</h3>

<pre><code class='language-R'>se_mean_per_strain
</code></pre>


<h3>Format</h3>

<p>A vector of length 200</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
