<!DOCTYPE html><html><head><title>Help for package wrGraph</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wrGraph}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.addTextToPoints'><p>Add text to points on plot</p></a></li>
<li><a href='#.bestLegendLoc'><p>Search best corner of plot for placing for legend</p>
</p>
<p>This function aims to find best corner for plotting a legend.</p></a></li>
<li><a href='#.colorByPvalue'><p>Colors based on p-Values</p></a></li>
<li><a href='#.determFigMargPix'><p>Estimate size/distance of margin to edge of image (png) in pixels</p></a></li>
<li><a href='#.levIndex'><p>Transform levels into index</p></a></li>
<li><a href='#.predPointsPix'><p>Predict and return pixel location of points of current plot</p></a></li>
<li><a href='#.sampNoDeMArrayLM'><p>Locate sample index from index or name of pair-wise comparisons</p></a></li>
<li><a href='#.serachColName'><p>Search Column Name</p>
</p>
<p>This function provides help when seraching column names</p></a></li>
<li><a href='#addBagPlot'><p>Add bagplot to existing plot</p></a></li>
<li><a href='#checkForLegLoc'><p>Find best place on plot for placing legend</p></a></li>
<li><a href='#convertPlotCoordPix'><p>Convert points of plot to coordinates in pixels</p></a></li>
<li><a href='#cumFrqPlot'><p>Cumulative (or sorted) frequency plot (takes columns of 'dat' as separate series)</p></a></li>
<li><a href='#foldChangeArrow'><p>Add arrow for expected Fold-Change to VolcanoPlot or MA-plot</p></a></li>
<li><a href='#histW'><p>Histogram (version by WR)</p></a></li>
<li><a href='#imageW'><p>Display numeric content of matrix as image</p></a></li>
<li><a href='#legendHist'><p>Add histogram to existing plot</p></a></li>
<li><a href='#MAplotW'><p>MA-plot (differential intensity versus average intensity)</p></a></li>
<li><a href='#mouseOverHtmlFile'><p>Create mouse-over interactive html-pages (with links)</p></a></li>
<li><a href='#partitionPlot'><p>Make matrix for layout to partition plotting area</p></a></li>
<li><a href='#plotBy2Groups'><p>Separate and plot data by 2 groups</p></a></li>
<li><a href='#plotLinReg'><p>Plot linear regression and confidence interval of regression</p></a></li>
<li><a href='#plotPCAw'><p>PCA plot with bag-plot to highlight groups</p></a></li>
<li><a href='#plotW2Leg'><p>x-y plot with 2 legends</p></a></li>
<li><a href='#profileAsClu'><p>Plot profiles according to CLustering</p></a></li>
<li><a href='#staggerdCountsPlot'><p>Staggered Chart for Ploting Counts to Multiple Leveles of the Threshold used</p></a></li>
<li><a href='#vioplotW'><p>Violin-plots version W</p></a></li>
<li><a href='#VolcanoPlotW'><p>Volcano-Plot (Statistical Test Outcome versus Relative Change)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.3.7</td>
</tr>
<tr>
<td>Title:</td>
<td>Graphics in the Context of Analyzing High-Throughput Data</td>
</tr>
<tr>
<td>Author:</td>
<td>Wolfgang Raffelsberger [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wolfgang Raffelsberger &lt;w.raffelsberger@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Additional options for making graphics in the context of analyzing high-throughput data are available here.
  This includes automatic segmenting of the current device (eg window) to accommodate multiple new plots, 
  automatic checking for optimal location of legends in plots, small histograms to insert as legends, 
  histograms re-transforming axis labels to linear when plotting log2-transformed data,
  a violin-plot &lt;<a href="https://doi.org/10.1080%2F00031305.1998.10480559">doi:10.1080/00031305.1998.10480559</a>&gt; function for a wide variety of input-formats, 
  principal components analysis (PCA) &lt;<a href="https://doi.org/10.1080%2F14786440109462720">doi:10.1080/14786440109462720</a>&gt; with bag-plots &lt;<a href="https://doi.org/10.1080%2F00031305.1999.10474494">doi:10.1080/00031305.1999.10474494</a>&gt; to highlight and compare the center areas for groups of samples, 
  generic MA-plots (differential- versus average-value plots) &lt;<a href="https://doi.org/10.1093%2Fnar%2F30.4.e15">doi:10.1093/nar/30.4.e15</a>&gt;, 
  staggered count plots and generation of mouse-over interactive html pages.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, grDevices, grid, lattice, RColorBrewer, stats,
wrMisc</td>
</tr>
<tr>
<td>Suggests:</td>
<td>dplyr, factoextra, FactoMineR, knitr, limma, rmarkdown, sm</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-22 15:15:39 UTC; wraff</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-22 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.addTextToPoints'>Add text to points on plot</h2><span id='topic+.addTextToPoints'></span>

<h3>Description</h3>

<p>This function allows to add custom text as lables to points on standard xy-plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.addTextToPoints(
  x,
  paramLst = NULL,
  labels = NULL,
  cex = NULL,
  col = NULL,
  adj = "auto",
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".addTextToPoints_+3A_x">x</code></td>
<td>
<p>(matrix, data.frame) coordinated of ponts on plot</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_paramlst">paramLst</code></td>
<td>
<p>(list) additional parameters for plotting (priority over separate cex or col), otherwise names displayed will be taken from 'labels' or rownames of coordinates 'x'</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_labels">labels</code></td>
<td>
<p>(character) the text to be displayed (has not priority over 'paramLst'), if nothing valid found numbers will be displayed</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_cex">cex</code></td>
<td>
<p>(character) (numeric) size of text, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_col">col</code></td>
<td>
<p>(character or integer) use custom colors</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_adj">adj</code></td>
<td>
<p>(character) values other than 0,0.5 or 1 will lead to 'auto' where text is displayed only to left of sufficient space available</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".addTextToPoints_+3A_debug">debug</code></td>
<td>
<p>(logical) display additional messages for debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function make a plot and may retiurn an optional matrix of outlyer-data (depending on argument <code>getOutL</code>)
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+prcomp">prcomp</a></code> (used here for the PCA underneith) , <code><a href="stats.html#topic+princomp">princomp</a></code>, see the package <a href="https://CRAN.R-project.org/package=FactoMineR">FactoMineR</a> for multiple plotting options or ways of combining categorical and numeric data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2019); dat1 &lt;- matrix(round(runif(30),2), ncol=2)
plot(dat1)        # traditional plot
.addTextToPoints(dat1, labels=letters[1:nrow(dat1)])
</code></pre>

<hr>
<h2 id='.bestLegendLoc'>Search best corner of plot for placing for legend 
This function aims to find best corner for plotting a legend.</h2><span id='topic+.bestLegendLoc'></span>

<h3>Description</h3>

<p>Search best corner of plot for placing for legend 
</p>
<p>This function aims to find best corner for plotting a legend.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.bestLegendLoc(
  dat,
  txtLen,
  txtHi,
  displayPlotSearch = FALSE,
  silent = TRUE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".bestLegendLoc_+3A_dat">dat</code></td>
<td>
<p>(matrix, list or data.frame) main data of plot</p>
</td></tr>
<tr><td><code id=".bestLegendLoc_+3A_txtlen">txtLen</code></td>
<td>
<p>(numeric, length=1)</p>
</td></tr>
<tr><td><code id=".bestLegendLoc_+3A_txthi">txtHi</code></td>
<td>
<p>(numeric, length=1) text height from graphics::strheight() (including inter-line)</p>
</td></tr>
<tr><td><code id=".bestLegendLoc_+3A_displayplotsearch">displayPlotSearch</code></td>
<td>
<p>(logical) decide if lines to mark area where data is searched for legend should be drawn</p>
</td></tr>
<tr><td><code id=".bestLegendLoc_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".bestLegendLoc_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id=".bestLegendLoc_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector with counts of umber of points expected to enter legend-location for each corner (ie legend-localization)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+checkForLegLoc">checkForLegLoc</a></code>,  <code><a href="graphics.html#topic+legend">legend</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat1 &lt;- matrix(c(1:5,1,1:5,5), ncol=2)
(legLoc &lt;- .bestLegendLoc(dat1, txtLen=0.4, txtHi=28))
</code></pre>

<hr>
<h2 id='.colorByPvalue'>Colors based on p-Values</h2><span id='topic+.colorByPvalue'></span>

<h3>Description</h3>

<p>This function helps defining color-gradient based on p-Values.
This fuction requires package RColorBrewer being installed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.colorByPvalue(
  x,
  br = NULL,
  col = NULL,
  asIndex = FALSE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".colorByPvalue_+3A_x">x</code></td>
<td>
<p>(numeric) p-values (main input)</p>
</td></tr>
<tr><td><code id=".colorByPvalue_+3A_br">br</code></td>
<td>
<p>(numeric) custom breaks (used with cut)</p>
</td></tr>
<tr><td><code id=".colorByPvalue_+3A_col">col</code></td>
<td>
<p>custom colors (must be of length(br) -1)</p>
</td></tr>
<tr><td><code id=".colorByPvalue_+3A_asindex">asIndex</code></td>
<td>
<p>(logical) custom breaks (used with cut)</p>
</td></tr>
<tr><td><code id=".colorByPvalue_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".colorByPvalue_+3A_debug">debug</code></td>
<td>
<p>(logical) supplemental messages for debugging</p>
</td></tr>
<tr><td><code id=".colorByPvalue_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function retruns a color-gradient based on p-Values
</p>


<h3>See Also</h3>

<p>(for PCA) <code><a href="#topic+plotPCAw">plotPCAw</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.colorByPvalue((1:10)/10)
</code></pre>

<hr>
<h2 id='.determFigMargPix'>Estimate size/distance of margin to edge of image (png) in pixels</h2><span id='topic+.determFigMargPix'></span>

<h3>Description</h3>

<p>This function allows estimating size/distance of margin to edge of image (png) in pixel and return numeric vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.determFigMargPix(marg, res, callFrom = NULL, silent = FALSE, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".determFigMargPix_+3A_marg">marg</code></td>
<td>
<p>(numeric) distance of margin in inch (as given in <code>par(mar=c(  ))</code>)</p>
</td></tr>
<tr><td><code id=".determFigMargPix_+3A_res">res</code></td>
<td>
<p>(numeric, length=1) resolution of image (png)</p>
</td></tr>
<tr><td><code id=".determFigMargPix_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
<tr><td><code id=".determFigMargPix_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".determFigMargPix_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a numeric vetctor matrix with (estimated) distance to figure margins in pixels
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convertPlotCoordPix">convertPlotCoordPix</a></code>, <code><a href="#topic+mouseOverHtmlFile">mouseOverHtmlFile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.determFigMargPix(c(5,4,4,2),100)
</code></pre>

<hr>
<h2 id='.levIndex'>Transform levels into index</h2><span id='topic+.levIndex'></span>

<h3>Description</h3>

<p>This function transforms levels into index. This function has been depreciated, please use wrMisc::levIndex() insted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.levIndex(dat, asSortedLevNa = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".levIndex_+3A_dat">dat</code></td>
<td>
<p>(numeric) initial levels (main input)</p>
</td></tr>
<tr><td><code id=".levIndex_+3A_assortedlevna">asSortedLevNa</code></td>
<td>
<p>(logical)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function retruns a color-gradient based on p-Values
</p>


<h3>See Also</h3>

<p>(for PCA) <code><a href="#topic+plotPCAw">plotPCAw</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(wrMisc)
</code></pre>

<hr>
<h2 id='.predPointsPix'>Predict and return pixel location of points of current plot</h2><span id='topic+.predPointsPix'></span>

<h3>Description</h3>

<p>This function allows predicting the pixel location of points of current plot.
Note: may be imprecise in case of x or y with all same values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.predPointsPix(
  x,
  y,
  dimPng,
  res,
  marg,
  fromTop = TRUE,
  scExt = 0.04,
  displ = FALSE,
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".predPointsPix_+3A_x">x</code></td>
<td>
<p>(numeric) initial coordinates for plot</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_y">y</code></td>
<td>
<p>(numeric) initial coordinates for plot</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_dimpng">dimPng</code></td>
<td>
<p>(numeric, length=2) width and hight of png</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_res">res</code></td>
<td>
<p>(numeric, length=1) resultion of png</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_marg">marg</code></td>
<td>
<p>(numeric, length=4) margins in inches (as given by <code>par(mar=c(...))</code>)</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_fromtop">fromTop</code></td>
<td>
<p>(logical) default counting in html is from top</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_scext">scExt</code></td>
<td>
<p>(numeric, length=1) extending scale (default at 0.04 ie 4 %)</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_displ">displ</code></td>
<td>
<p>(logical) optional plot</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".predPointsPix_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a numeric matrix with 2 columns 'xPix' and 'yPix' (with length(x) rows); and optionally a plot (if argument <code>displ=TRUE</code>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convertPlotCoordPix">convertPlotCoordPix</a></code>, <code><a href="#topic+mouseOverHtmlFile">mouseOverHtmlFile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>.predPointsPix(x=c(1,100), y=c(1,100), dimPng=c(700,600), res=200, marg=c(5,4,4,2))
</code></pre>

<hr>
<h2 id='.sampNoDeMArrayLM'>Locate sample index from index or name of pair-wise comparisons</h2><span id='topic+.sampNoDeMArrayLM'></span>

<h3>Description</h3>

<p>This function helps locating sample index from index or name of pair-wise comparisons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.sampNoDeMArrayLM(
  MArrayObj,
  useComp,
  groupSep = "-",
  lstMeans = "means",
  lstP = "BH",
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".sampNoDeMArrayLM_+3A_marrayobj">MArrayObj</code></td>
<td>
<p>(MArray type objct) main input</p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_usecomp">useComp</code></td>
<td>
<p>(matrix) types of pair-wise comparisons to be performed</p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_groupsep">groupSep</code></td>
<td>
<p>(character) separator used with pair-wise grouping</p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_lstmeans">lstMeans</code></td>
<td>
<p>(character) type of summarization, default is 'means'</p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_lstp">lstP</code></td>
<td>
<p>(character) type of multiple testing correction data to choose from <code>MArrayObj</code></p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_debug">debug</code></td>
<td>
<p>(logical) supplemental messages for debugging</p>
</td></tr>
<tr><td><code id=".sampNoDeMArrayLM_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a integer vector of indexes
</p>


<h3>See Also</h3>

<p>(for PCA) <code><a href="#topic+plotPCAw">plotPCAw</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aa &lt;- 1:5
</code></pre>

<hr>
<h2 id='.serachColName'>Search Column Name	 
This function provides help when seraching column names</h2><span id='topic+.serachColName'></span>

<h3>Description</h3>

<p>Search Column Name	 
</p>
<p>This function provides help when seraching column names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.serachColName(
  x,
  searchColNa,
  plusLowerCaps = TRUE,
  returnList = TRUE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".serachColName_+3A_x">x</code></td>
<td>
<p>(matrix or data.frame) main input</p>
</td></tr>
<tr><td><code id=".serachColName_+3A_searchcolna">searchColNa</code></td>
<td>
<p>(character)</p>
</td></tr>
<tr><td><code id=".serachColName_+3A_pluslowercaps">plusLowerCaps</code></td>
<td>
<p>(logical) add lower caps to search</p>
</td></tr>
<tr><td><code id=".serachColName_+3A_returnlist">returnList</code></td>
<td>
<p>(logical)</p>
</td></tr>
<tr><td><code id=".serachColName_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id=".serachColName_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
<tr><td><code id=".serachColName_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>integer vector with index of colnames found or list with $foundNa and $remainNa
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convertPlotCoordPix">convertPlotCoordPix</a></code>; use <code><a href="wrMisc.html#topic+htmlSpecCharConv">htmlSpecCharConv</a></code> to convert special characters for proper html display
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mat1 &lt;- matrix(1:6, ncol=3, dimnames=list(NULL, LETTERS[1:3]))
.serachColName(mat1, c("C","F","A"))
</code></pre>

<hr>
<h2 id='addBagPlot'>Add bagplot to existing plot</h2><span id='topic+addBagPlot'></span>

<h3>Description</h3>

<p>This function adds a bagplot on an existing (scatter-)plot allowing to highlight the central area of the data.
Briefly, a bagplot is a bivariate boxplot, see <a href="https://en.wikipedia.org/wiki/Bagplot">Bagplot</a>, following the basic idea of a boxplot in two dimensions.
Of course, multimodal distributions - if not separated first - may likely lead to mis-interpretation, similarly as it is known for interpreting boxplots.
If a group of data consists only of 2 data-points, they will be conected using a straight line.
It is recommended using transparent colors to highlight the core part of a group (if only 2 points are available, they will be conected using a straight line),
in addition, one could use the option to re-plot all (non-outlyer) points (arguments <code>reCol</code>, <code>rePch</code> and <code>reCex</code> must be used).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addBagPlot(
  x,
  lev1 = 0.5,
  outCoef = 2,
  bagCol = NULL,
  bagCont = bagCol,
  bagLwd = 1.5,
  nCore = 4,
  outlCol = 2,
  outlPch = NULL,
  outlCex = 0.6,
  reCol = NULL,
  rePch = NULL,
  reCex = NULL,
  ctrPch = NULL,
  ctrCol = NULL,
  ctrCex = NULL,
  addSubTi = TRUE,
  returnOutL = FALSE,
  silent = TRUE,
  callFrom = NULL,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addBagPlot_+3A_x">x</code></td>
<td>
<p>(matrix, list or data.frame) main numeric input of data/points to plot</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_lev1">lev1</code></td>
<td>
<p>(numeric) min content of data for central area (default 0.5 for 50 percent)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_outcoef">outCoef</code></td>
<td>
<p>(numeric) parameter for defining outliers (equivalent to <code>range</code> in <code><a href="graphics.html#topic+boxplot">boxplot</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_bagcol">bagCol</code></td>
<td>
<p>(character or integer) color for filling center part of bagplot (default light transparent grey); Note: It is highly suggested to use transparency, otherwise points underneith will be covered</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_bagcont">bagCont</code></td>
<td>
<p>(character) color for inner and outer contours of bagplot</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_baglwd">bagLwd</code></td>
<td>
<p>(numeric) line width for outer contour, set to <code>NULL</code> for not diaplaying outer contour (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_ncore">nCore</code></td>
<td>
<p>(integer) decide when center should be determined by median or mean: if number of points reach <code>nCore</code> the median will be used</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_outlcol">outlCol</code></td>
<td>
<p>(character or integer) color for highlighting outlyers (for text and replottig outlyers points), set to <code>NULL</code> for not highlighting outlyers at all</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_outlpch">outlPch</code></td>
<td>
<p>(integer) symbol replotting highlighted outlyers (for text and replottig outlyers points), set to <code>NULL</code> for not replotting outlyer-points (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_outlcex">outlCex</code></td>
<td>
<p>(numeric) cex type expansion factor for labels of highlighted outlyers, set to <code>NULL</code> for not printing (row)names of outlyers (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_recol">reCol</code></td>
<td>
<p>(character or integer) color for replotting (non-outlyer) points, default set to <code>NULL</code> for not replotting</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_repch">rePch</code></td>
<td>
<p>(integer) symbol for replotting (non-outlyer) points, default set to <code>NULL</code> for not re-plotting (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_recex">reCex</code></td>
<td>
<p>(numeric) cex type expansion factor for lfor replotting (non-outlyer) points, default set to <code>NULL</code> for not replotting</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_ctrpch">ctrPch</code></td>
<td>
<p>(integer) symbol for showing group center (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_ctrcol">ctrCol</code></td>
<td>
<p>(character or integer) color for group center symbol</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_ctrcex">ctrCex</code></td>
<td>
<p>(numeric) cex type expansion factor for size of group center (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_addsubti">addSubTi</code></td>
<td>
<p>(logical) decide if subtitle (stating that potential outlyers were displayed separatetly) should be added in plot</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_returnoutl">returnOutL</code></td>
<td>
<p>(logical) decide if rownames of (potential) outlyer values should be returned when running the function</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
<tr><td><code id="addBagPlot_+3A_debug">debug</code></td>
<td>
<p>(logical) display additional messages for debugging</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The outlyer detection works similar to the one used in <code>boxplot</code>: The distance of a given point is compared to the median distance of all points to their respective group-center plus
the 25 - 75 quantile-distance (of all points) times the multiplicative factor of argument <code>outCoef</code>.
</p>


<h3>Value</h3>

<p>This function returns primarily a plot, optionally it may return of matrix with outlyers (if argument <code>returnOutL=TRUE</code>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotPCAw">plotPCAw</a></code>, <code><a href="stats.html#topic+princomp">princomp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2020); dat1 &lt;- matrix(round(rnorm(2000),3),ncol=2); rownames(dat1) &lt;- 1:nrow(dat1)
dat1 &lt;- dat1 + 5*matrix(rep(c(0,1,1,0,0,0,1,1),nrow(dat1)/4), byrow=TRUE, ncol=2)
col1 &lt;- rgb(red=c(153,90,203,255), green=c(143,195,211,125), blue=c(204,186,78,115),
  alpha=90, maxColorValue=255)
## suppose we know the repartition into 4 subgroups which we would like to highlight them
grp1 &lt;- rep(1:4, nrow(dat1)/4)
plot(dat1, col=grey(0.8), xlab="x", ylab="y", las=1, pch=grp1)
for(i in 1:4) addBagPlot(dat1[which(grp1==i),], bagCol=col1[i])
## slightly improved
library(wrMisc)
col2 &lt;- convColorToTransp(col1, 255)
plot(dat1, col=grey(0.8), xlab="x", ylab="y", las=1, pch=grp1)
for(i in 1:4) addBagPlot(dat1[which(grp1==i),], bagCol=col1[i], outlPch=i,
  outlCol=col2[i], bagLwd=3)
</code></pre>

<hr>
<h2 id='checkForLegLoc'>Find best place on plot for placing legend</h2><span id='topic+checkForLegLoc'></span>

<h3>Description</h3>

<p>This function tries to find the best location for placing a legend of a bivariate plot, ie scatter-plot.
All 4 corners of the data to plot are inspected for the least occupation by data plotted while displaying the content of <code>sampleGrp</code>.
Alternatively, by setting the argument <code>showLegend</code> the user-defined legend will be returned
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkForLegLoc(
  matr,
  sampleGrp = NULL,
  showLegend = TRUE,
  suplSpace = 4,
  testCorner = 1:4,
  silent = TRUE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkForLegLoc_+3A_matr">matr</code></td>
<td>
<p>(matrix, list or data.frame) main data of plot</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_samplegrp">sampleGrp</code></td>
<td>
<p>(character or factor) with this option the text to be displayed in the legend may be taken into consideration for its length</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_showlegend">showLegend</code></td>
<td>
<p>(logical or character) decide if <code>matr</code> should be checked for best location; if <code>showLegend</code> contains any of the standard legend-location designations (eg 'topleft') it will be used in the output</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_suplspace">suplSpace</code></td>
<td>
<p>(numeric) allows to consider extra room taken in legend by symbol and surrounding space, interpreted as n additional characters</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_testcorner">testCorner</code></td>
<td>
<p>(integer) which corners should be considered (1=left-top, 2=right-top, right-bottom, left-bottom)</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="checkForLegLoc_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with $showL indicating if legend is desired and $loc for the proposition of the best location, $nConflicts gives the counts of conflicts
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+legend">legend</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat1 &lt;- matrix(c(1:5,1,1:5,5), ncol=2)
grp &lt;- c("abc","efghijk")
(legLoc &lt;- checkForLegLoc(dat1, grp)) 
plot(dat1, cex=3)
legend(legLoc$loc, legend=grp, text.col=2:3, pch=1, cex=0.8)
</code></pre>

<hr>
<h2 id='convertPlotCoordPix'>Convert points of plot to coordinates in pixels</h2><span id='topic+convertPlotCoordPix'></span>

<h3>Description</h3>

<p>This function allows conversion the plotting positions ('x' and 'y' coordinates) of points in a given plot into coordinates in pixels (of the entire plotting region).
It was designed to be used as coordinates in an html file for mouse-over interactivity (display of names of points and links).
Of course, the size of the plotting region is crucial and may not be changed afterwards (if the plot is not written to file using <code>png</code> etc).
In turn the function <code><a href="#topic+mouseOverHtmlFile">mouseOverHtmlFile</a></code> will use the pixel-coordinates, allowing to annotate given points of a plot for mouse-over interactive html.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertPlotCoordPix(
  x,
  y,
  useMar = c(6.2, 4, 4, 2),
  plotDim = c(1400, 800),
  plotRes = 100,
  fromTop = TRUE,
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convertPlotCoordPix_+3A_x">x</code></td>
<td>
<p>(numeric) initial plotting coordinates on x-axis, names of vector - if available- will be used as IDs</p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_y">y</code></td>
<td>
<p>(numeric) initial plotting coordinates on y-axis</p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_usemar">useMar</code></td>
<td>
<p>(numeric,length=4) margins defined with plot, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_plotdim">plotDim</code></td>
<td>
<p>(integer, length=2) dimension of the plotting device in pixels, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_plotres">plotRes</code></td>
<td>
<p>(integer) resoltion of plotting device, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_fromtop">fromTop</code></td>
<td>
<p>(logical) toggle if poordinates should start from top</p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="convertPlotCoordPix_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with x- and y-coordinates in pixels
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mouseOverHtmlFile">mouseOverHtmlFile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df1 &lt;- data.frame(id=letters[1:10], x=1:10, y=rep(5,10),mou=paste("point",letters[1:10]),
  link=file.path(tempdir(),paste0(LETTERS[1:10],".html")), stringsAsFactors=FALSE)
## Typically one wants to get pixel-coordinates for plots written to file.
## Here we'll use R's tempdir, later you may want to choose other locations
pngFile &lt;- file.path(tempdir(),"test01.png")
png(pngFile, width=800, height=600, res=72)
## here we'll just plot a set of horiontal points at default parameters ...
plot(df1[,2:3], las=1, main="test01")
dev.off()
## Note: Special characters should be converted for proper display in html during mouse-over
library(wrMisc)
df1$mou &lt;- htmlSpecCharConv(df1$mou)
## Let's add the x- and y-coordiates of the points in pixels to the data.frame
df1 &lt;- cbind(df1,convertPlotCoordPix(x=df1[,2], y=df1[,3], plotD=c(800,600),plotRes=72))
head(df1)
## using mouseOverHtmlFile() one could now make an html document with interactive
## display of names and clockable links to the coordinates we determined here ...
</code></pre>

<hr>
<h2 id='cumFrqPlot'>Cumulative (or sorted) frequency plot (takes columns of 'dat' as separate series)</h2><span id='topic+cumFrqPlot'></span>

<h3>Description</h3>

<p>Display data as sorted or cumulative frequency plot. This type of plot represents an alternative to plotting data as histograms.
Histograms are very universal and which are very intuitive. However,fine-tuning the bandwith (ie width of the bars) may be very delicate, 
fine resultion details may often remain hidden. 
One of the advanges of directly displaying all data-points is that subtile differences may be revealed easier, compared to calssical histograms. 
Furthermore, the plot prensented her offeres more options to display multiple series of data simultaneaously. 
Thus, this type of plot may be useful to compare eg results of data normalization.
Of course, with very large data-sets (eg &gt; 3000 values) this gain of 'details' will be less important (compared to histograms) and will penalize speed.
In such cases the argument <code>thisResol</code> will get useful as it allows to reduce the resultion and introduce binning.
Alternatively for very large data-sets one may looking into density-plots or vioplots (eg <code><a href="#topic+vioplotW">vioplotW</a></code>).
The argument <code>CVlimit</code> allows optionally excluding extreme values. 
If numeric (&amp; &gt; 2 columns), its value will be used <code><a href="wrMisc.html#topic+exclExtrValues">exclExtrValues</a></code> to identify series with column-median &gt; 'CVlimit'.
Of course, exclusion of extreme values should be done with great care, important features of the data may get lost.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cumFrqPlot(
  dat,
  cumSum = FALSE,
  exclCol = NULL,
  colNames = NULL,
  displColNa = TRUE,
  tit = NULL,
  xLim = NULL,
  yLim = NULL,
  xLab = NULL,
  yLab = NULL,
  col = NULL,
  CVlimit = NULL,
  thisResol = NULL,
  supTxtAdj = 0,
  supTxtYOffs = 0,
  useLog = "",
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cumFrqPlot_+3A_dat">dat</code></td>
<td>
<p>(matrix or data.frame) data to plot/inspect</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_cumsum">cumSum</code></td>
<td>
<p>(logical) for either plotting cumulates Sums (then <code>thisResol</code> for number of breaks) or (if <code>=FALSE</code>) simply sorted values -&gt; max resolution</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_exclcol">exclCol</code></td>
<td>
<p>(integer) columlns to exclude</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_colnames">colNames</code></td>
<td>
<p>(character) for alternative column/series names in display, as long as <code>displColNa=TRUE</code></p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_displcolna">displColNa</code></td>
<td>
<p>(logical) display column-names</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_tit">tit</code></td>
<td>
<p>(character) custom title</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_xlim">xLim</code></td>
<td>
<p>(numeric) custom limit for x-axis (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_ylim">yLim</code></td>
<td>
<p>(numeric) custom limit for y-axis (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_xlab">xLab</code></td>
<td>
<p>(character) custom x-axis label</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_ylab">yLab</code></td>
<td>
<p>(character) custom y-axis label</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_col">col</code></td>
<td>
<p>(integer or character) custom colors</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_cvlimit">CVlimit</code></td>
<td>
<p>(numeric) for the tag 'outlier column' (uses <code><a href="wrMisc.html#topic+exclExtrValues">exclExtrValues</a></code>)  identify &amp; mark column with median row-CV &gt; <code>CVlimit</code></p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_thisresol">thisResol</code></td>
<td>
<p>(integer) resolution <code>res</code> for binning large data-sets</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_suptxtadj">supTxtAdj</code></td>
<td>
<p>(numeric) parameter <code>adj</code> for supplemetal text</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_suptxtyoffs">supTxtYOffs</code></td>
<td>
<p>(numeric) supplemental offset for text on y axis</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_uselog">useLog</code></td>
<td>
<p>(character) default=&quot;&quot;, otherwise for setting axis in log-scale &quot;x&quot;, &quot;y&quot; or &quot;xy&quot;</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="cumFrqPlot_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function plots to the current garphical device
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+layout">layout</a></code>, <code><a href="wrMisc.html#topic+exclExtrValues">exclExtrValues</a></code> for decision of potential outliers; <code><a href="graphics.html#topic+hist">hist</a></code>, <code><a href="#topic+vioplotW">vioplotW</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2017); dat0 &lt;- matrix(rnorm(500), ncol=5, dimnames=list(NULL,1:5))
cumFrqPlot(dat0, tit="Sorted values")
cumFrqPlot(dat0, cumSum=TRUE, tit="Sum of sorted values")
</code></pre>

<hr>
<h2 id='foldChangeArrow'>Add arrow for expected Fold-Change to VolcanoPlot or MA-plot</h2><span id='topic+foldChangeArrow'></span>

<h3>Description</h3>

<p>This function allows adding an arrow indicating a fold-change to MA- or Volcano-plots. 
When comparing mutiple concentratios of standards in benchmark-tests it may be useful to indicate the expected ratio in a pair-wise comparison.
In case of main input as list or MArrayLM-object (as generated from limma), the colum-names of multiple pairwise comparisons can be used 
for extracting a numeric content (supposed as concentrations in sample-names) which will be used to determine the expected ratio used for plotting. 
Optionally the ratio used for plotting can be returned as numeric value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>foldChangeArrow(
  FC,
  useComp = 1,
  isLin = TRUE,
  asX = TRUE,
  col = 2,
  arr = c(0.005, 0.15),
  lwd = NULL,
  addText = c(line = -0.9, cex = 0.7, txt = "expected", loc = "toright"),
  returnRatio = FALSE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="foldChangeArrow_+3A_fc">FC</code></td>
<td>
<p>(numeric, list or MArrayLM-object) main information for drawing arrow : either numeric value for fold-change/log2-ratio of object to search for colnames of statistical testing for extracting numeric part</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_usecomp">useComp</code></td>
<td>
<p>(integer) only used in case FC is list or MArrayLM-object an has multiple pairwise-comparisons</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_islin">isLin</code></td>
<td>
<p>(logical) inidicate if <code>FC</code> is log2 or not</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_asx">asX</code></td>
<td>
<p>(logical) indicate if arrow should be on x-axis</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_col">col</code></td>
<td>
<p>(integer or character) custom color</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_arr">arr</code></td>
<td>
<p>(numeric, length=2) start- and end-points of arrow (as relative to entire plot)</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_lwd">lwd</code></td>
<td>
<p>(numeric) line-width of arrow</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_addtext">addText</code></td>
<td>
<p>(logical or named vector) indicate if text explaining arrow should be displayed, use <code>TRUE</code> for default (on top right of plot), 
or any combination of 'loc','line','cex','side','adj','col','text' (or 'txt') for customizing specific elements</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_returnratio">returnRatio</code></td>
<td>
<p>(logical) return ratio</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="foldChangeArrow_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Details</h3>

<p>#' @details The argument <code>addText</code> also allows specifying a fixed position when using <code>addText=c(loc="bottomleft")</code>, also bottomright, topleft, topright, toleft and toright may be used.
In this case the elemts <code>side</code> and <code>adjust</code> will be redefined to accomodate the text in the corner specified.
</p>


<h3>Value</h3>

<p>This function plots only an arrow onto current plotting device (and some explicative text), if <code>returnRatio=TRUE</code> also returns numeric value for extracted ratio
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MAplotW">MAplotW</a></code>, <code><a href="#topic+VolcanoPlotW">VolcanoPlotW</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(rnorm(20,1.5,0.1), 1:20)
foldChangeArrow(FC=1.5) 

</code></pre>

<hr>
<h2 id='histW'>Histogram (version by WR)</h2><span id='topic+histW'></span>

<h3>Description</h3>

<p>This function proposes a few special tweaks to the general <code><a href="graphics.html#topic+hist">hist</a></code> function :
In a number of settings data are treated and plotted as log-data. This function allows feeding directly log2-data and displaying the x-axis   
(re-translated) in linear scale (see argument <code>isLog</code>). 
The default settings allow making (very) small histograms ('low resolution'), which may be used as a rough overview of bandwidth and distribution of values in <code>dat</code>. 
Similar to <code><a href="graphics.html#topic+hist">hist</a></code>, by changing the parameters <code>nBars</code> and/or <code>breaks</code> very 'high resolution' histograms can be produced.
By default it displays n per set of data (on the top of the figure). 
Note that the argument for (costom) title <code>main</code> is now called <code>tit</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>histW(
  dat,
  fileName = "histW",
  output = "screen",
  nBars = 8,
  breaks = NULL,
  tit = NULL,
  subTi = NULL,
  xLab = NULL,
  yLab = NULL,
  las = NULL,
  xcex = 0.7,
  imgxSize = 900,
  useCol = NULL,
  useBord = NULL,
  isLog = TRUE,
  cexSubTi = NULL,
  cropHist = TRUE,
  parDefault = TRUE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="histW_+3A_dat">dat</code></td>
<td>
<p>(matrix, list or data.frame) data to plot</p>
</td></tr>
<tr><td><code id="histW_+3A_filename">fileName</code></td>
<td>
<p>(character) name of file for saving graphics</p>
</td></tr>
<tr><td><code id="histW_+3A_output">output</code></td>
<td>
<p>(character, length=1) options for output on 'screen' or saving image in various formats (set to 'jpg','png' or 'tif')</p>
</td></tr>
<tr><td><code id="histW_+3A_nbars">nBars</code></td>
<td>
<p>(integer) number of bars in histogram (default for 'low resolution' plot to give rough overview)</p>
</td></tr>
<tr><td><code id="histW_+3A_breaks">breaks</code></td>
<td>
<p>(integer) for (partial) compatibility with hist() : use only for number of breaks (or 'FD'), gets priority over 'nBars'</p>
</td></tr>
<tr><td><code id="histW_+3A_tit">tit</code></td>
<td>
<p>(character) custom title</p>
</td></tr>
<tr><td><code id="histW_+3A_subti">subTi</code></td>
<td>
<p>(character) may be <code>FALSE</code> for NOT displaying, or any text, otherwise range</p>
</td></tr>
<tr><td><code id="histW_+3A_xlab">xLab</code></td>
<td>
<p>(character) custom x-axes label</p>
</td></tr>
<tr><td><code id="histW_+3A_ylab">yLab</code></td>
<td>
<p>(character) custom y-axes label</p>
</td></tr>
<tr><td><code id="histW_+3A_las">las</code></td>
<td>
<p>(integer) optional fixed text orientation of x-axis numbers : use 1 for horizontal and 2 for perpendicular, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_xcex">xcex</code></td>
<td>
<p>(numeric) cex-type expansion factor for x-axis numbers, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_imgxsize">imgxSize</code></td>
<td>
<p>(integer) width of image when saving to file, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_usecol">useCol</code></td>
<td>
<p>(character or integer) custom colors, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_usebord">useBord</code></td>
<td>
<p>(character) custom histogram elements border color, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_islog">isLog</code></td>
<td>
<p>(logical) for lin scale signal intensity values where repesentation needs log, assume log2 if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="histW_+3A_cexsubti">cexSubTi</code></td>
<td>
<p>(numerical) subtitle size (expansion factor cex), see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_crophist">cropHist</code></td>
<td>
<p>(logical) -not implemented yet- designed for cutting off bars with very low ('insignificant') values</p>
</td></tr>
<tr><td><code id="histW_+3A_pardefault">parDefault</code></td>
<td>
<p>(logical) to automatic adjusting par(marg=,cex.axis=0.8), see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="histW_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="histW_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="histW_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function produces a histogram type graphic (to the ccurrent graphical device)
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+hist">hist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2016); dat1 &lt;- round(c(rnorm(200,6,0.5),rlnorm(300,2,0.5),rnorm(100,17)),2)
dat1 &lt;- dat1[which(dat1 &lt;50 &amp; dat1 &gt; 0.2)]
histW(dat1, br="FD", isLog=FALSE)
histW(log2(dat1), br="FD", isLog=TRUE)

## quick overview of distributions  
layout(partitionPlot(4))
for(i in 1:4) histW(iris[,i], isLog=FALSE, tit=colnames(iris)[i])
</code></pre>

<hr>
<h2 id='imageW'>Display numeric content of matrix as image</h2><span id='topic+imageW'></span>

<h3>Description</h3>

<p>To get a quick overview of the spatial distribution of smaller data-sets it may be useful to display numeric values as colored boxes.
Such an output may also be referred to as heatmap (note that the term 'heatmap' is frequently associated with graphical display of hierarchcal clustering results).
The function <code><a href="graphics.html#topic+image">image</a></code> provides the basic support to do so (ie heatmap without rearranging rows and columns by clustering).  
To do this more conveniently, the function <code>imageW</code> offers additional options for displaying row- and column-names or displaying NA-values as custom-color.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imageW(
  data,
  latticeVersion = FALSE,
  transp = TRUE,
  NAcol = "grey95",
  tit = NULL,
  rowNa = NULL,
  colNa = NULL,
  xLab = NULL,
  yLab = NULL,
  las = 2,
  col = NULL,
  nColor = 9,
  balanceCol = TRUE,
  gridCol = "grey75",
  gridLty = 1,
  centColShift = 0,
  cexDispl = NULL,
  panel.background.col = "white",
  supLat = list(),
  rotXlab = 0,
  rotYlab = 0,
  cexXlab = 0.7,
  cexAxs = NULL,
  cexYlab = 0.9,
  Xtck = 0,
  Ytck = 0,
  cexTit = 1.6,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imageW_+3A_data">data</code></td>
<td>
<p>(matrix or data.frame) main input</p>
</td></tr>
<tr><td><code id="imageW_+3A_latticeversion">latticeVersion</code></td>
<td>
<p>(logical) use lattice for plotting (this will include a color-legend)</p>
</td></tr>
<tr><td><code id="imageW_+3A_transp">transp</code></td>
<td>
<p>(logical) decide if data should get transposed (if <code>TRUE</code> the data will be displayed exacetly same order as when printing the values as table); 
set to <code>FALSE</code> to get behaviour prior to version 1.3.0.</p>
</td></tr>
<tr><td><code id="imageW_+3A_nacol">NAcol</code></td>
<td>
<p>(character or integer) custom color for NA-values, default is light grey</p>
</td></tr>
<tr><td><code id="imageW_+3A_tit">tit</code></td>
<td>
<p>(character) custom figure title</p>
</td></tr>
<tr><td><code id="imageW_+3A_rowna">rowNa</code></td>
<td>
<p>(character) optional custom rownames</p>
</td></tr>
<tr><td><code id="imageW_+3A_colna">colNa</code></td>
<td>
<p>(character) optional custom colnames</p>
</td></tr>
<tr><td><code id="imageW_+3A_xlab">xLab</code></td>
<td>
<p>(character) optional custom text for x-axis 'values'</p>
</td></tr>
<tr><td><code id="imageW_+3A_ylab">yLab</code></td>
<td>
<p>(character) optional custom text for y-axis 'values'</p>
</td></tr>
<tr><td><code id="imageW_+3A_las">las</code></td>
<td>
<p>(numeric) style of axis labels (see also <code><a href="graphics.html#topic+par">par</a></code>); in case of <code>latticeVersion=TRUE</code> this argument will override default <code>rotXlab=0</code> and/or  <code>rotYlab=0</code></p>
</td></tr>
<tr><td><code id="imageW_+3A_col">col</code></td>
<td>
<p>(character or integer) colors; in lattice version 2 or 3 color-names to define central- and end-points of gradient (starting with color for lowest values, optional central color and color for highest values), default is 60 shades 'RdYlBu' RColorBrewer, if 'heat.colors' use  heat.colors in min 15 shades</p>
</td></tr>
<tr><td><code id="imageW_+3A_ncolor">nColor</code></td>
<td>
<p>(integer, only used in lattice version) number of color-blocks in color gradient (made based on central- and end-points from <code>col</code></p>
</td></tr>
<tr><td><code id="imageW_+3A_balancecol">balanceCol</code></td>
<td>
<p>(logical, only used in lattice version) if <code>TRUE</code> the color-radient aims to color the value closest to 0 with the center color (from <code>col</code> (default gray)</p>
</td></tr>
<tr><td><code id="imageW_+3A_gridcol">gridCol</code></td>
<td>
<p>(character, only used in lattice version) define color of grid</p>
</td></tr>
<tr><td><code id="imageW_+3A_gridlty">gridLty</code></td>
<td>
<p>(integer, only used in lattice version) define line-type of grid (see also lty <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="imageW_+3A_centcolshift">centColShift</code></td>
<td>
<p>(integer, only used in lattice version) shift central (default grey) color element for negative scale up or down (ie increase or reduce number of color-blocks for negatve values), 
used for correcting automatic scaling rounding issues to ensure the central elements captures 0</p>
</td></tr>
<tr><td><code id="imageW_+3A_cexdispl">cexDispl</code></td>
<td>
<p>(numeric, length=1, only used in lattice version) define cex size for displaying (rounded) values in plot, set to <code>NULL</code> for omitting</p>
</td></tr>
<tr><td><code id="imageW_+3A_panel.background.col">panel.background.col</code></td>
<td>
<p>(character, only used in lattice version)</p>
</td></tr>
<tr><td><code id="imageW_+3A_suplat">supLat</code></td>
<td>
<p>(list, only used in lattice version) additional arguments/parameters passed to <code>levelplot</code></p>
</td></tr>
<tr><td><code id="imageW_+3A_rotxlab">rotXlab</code></td>
<td>
<p>(numeric, 0 - 360, lattice version only) control rotation of x-axis labels</p>
</td></tr>
<tr><td><code id="imageW_+3A_rotylab">rotYlab</code></td>
<td>
<p>(numeric, 0 - 360, lattice version only) control rotation of y-axis labels</p>
</td></tr>
<tr><td><code id="imageW_+3A_cexxlab">cexXlab</code></td>
<td>
<p>(numeric) cex-like expansion factor for x-axis labels  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="imageW_+3A_cexaxs">cexAxs</code></td>
<td>
<p>(numeric) cex-like expansion factor for x- and y-axis text/labels (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="imageW_+3A_cexylab">cexYlab</code></td>
<td>
<p>(numeric) cex-like expansion factor for y-axis labels  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="imageW_+3A_xtck">Xtck</code></td>
<td>
<p>(numeric or logical) expansion factor for length of tick-marks on x-axis (default=0 for no tick-marks)</p>
</td></tr>
<tr><td><code id="imageW_+3A_ytck">Ytck</code></td>
<td>
<p>(numeric or logical) expansion factor for length of tick-marks on y-axis</p>
</td></tr>
<tr><td><code id="imageW_+3A_cextit">cexTit</code></td>
<td>
<p>(numeric) cex-like expansion factor for title  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="imageW_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="imageW_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
<tr><td><code id="imageW_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allows two modes of operation : 1) plotting using standard R -graphics or 2) using the framework of grid- and lattice-graphics (since version 1.2.6). 
The latter version allows integrating a legend for the color-scale and adding grid-lines, rotation of axis-labels or removing tick-marks.
Please note that sometimes the center-color segment may not end up directly with the center-color, in this case you may adjust using the argument <code>centColShift=-1</code>
</p>


<h3>Value</h3>

<p>This function plots in image (to the current graphical device) as <code>image</code> does
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+image">image</a></code>, for the lattice version <code><a href="lattice.html#topic+levelplot">levelplot</a></code>, heatmaps including hierarchical clustering <code><a href="stats.html#topic+heatmap">heatmap</a></code> or <code>heatmap.2</code> from package <a href="https://CRAN.R-project.org/package=gplots">gplots</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>imageW(as.matrix(iris[1:40,1:4]), transp=FALSE, tit="Iris (head)")
imageW(as.matrix(iris[1:20,1:4]), latticeVersion=TRUE, col=c("blue","red"), 
  rotXlab=45, yLab="Observation no", tit="Iris (head)")
</code></pre>

<hr>
<h2 id='legendHist'>Add histogram to existing plot</h2><span id='topic+legendHist'></span>

<h3>Description</h3>

<p>Add histogram at pleace of legend using colors from 'colorRamp'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legendHist(
  x,
  colRamp = NULL,
  location = "bottomright",
  legTit = NULL,
  cex = 0.7,
  srt = 67,
  offS = NULL,
  border = TRUE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="legendHist_+3A_x">x</code></td>
<td>
<p>(numeric) main input/component of plot</p>
</td></tr>
<tr><td><code id="legendHist_+3A_colramp">colRamp</code></td>
<td>
<p>(character or integer) set of colors, default is rainbow-like</p>
</td></tr>
<tr><td><code id="legendHist_+3A_location">location</code></td>
<td>
<p>(character) for location of histogram inside existing plot (may be 'br','bl','tl','tr','bottomright', 'bottomleft','topleft','topright')</p>
</td></tr>
<tr><td><code id="legendHist_+3A_legtit">legTit</code></td>
<td>
<p>(character, length=1) optional title for histogram-insert</p>
</td></tr>
<tr><td><code id="legendHist_+3A_cex">cex</code></td>
<td>
<p>(numeric) expansion factor (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="legendHist_+3A_srt">srt</code></td>
<td>
<p>(numeric) angle for histogram text labels (90 will give vertical label) (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="legendHist_+3A_offs">offS</code></td>
<td>
<p>(<code>NULL</code> or numeric, length=5) fine-tuning of where histogram-insert will be placed and how elements therein are ditributed
(default c(xOff=0.2,yOff=0.25,leftOffS=0.05, upperBarEnd=1.05,txtOff=0.02),
1st and 2nd determine proportio of insert relative to entire plotting region, 3rd defines space left on bottom for text,
4th if bars hit ceiling of insert or proportion to leave, 5th for shifting text towards top when turned other than 90 degrees )</p>
</td></tr>
<tr><td><code id="legendHist_+3A_border">border</code></td>
<td>
<p>(logical) decide of draw gray rectangle or not around legend</p>
</td></tr>
<tr><td><code id="legendHist_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="legendHist_+3A_debug">debug</code></td>
<td>
<p>(logical) display additional messages for debugging</p>
</td></tr>
<tr><td><code id="legendHist_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function produces a histogram on the current plottig device
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- rnorm(90); plot(dat)
legendHist(dat, col=1:5)
</code></pre>

<hr>
<h2 id='MAplotW'>MA-plot (differential intensity versus average intensity)</h2><span id='topic+MAplotW'></span>

<h3>Description</h3>

<p>This type of plot for display of relative changes versus (mean) absolute abundance is very common in high-throughput biology, see <a href="https://en.wikipedia.org/wiki/MA_plot">MA-plot</a>.
Basically one compares two independent series of measures (ie gene transcript or protein abundance values) of 2 samples/data-sets or the means of 2 groups of replicates.
And the log-fold-change ('Minus'=M) is plotted againts the absolute mean value ('Average'=A).
Furthermore, output from statistical testing by <code><a href="wrMisc.html#topic+moderTest2grp">moderTest2grp</a></code> or <code><a href="wrMisc.html#topic+moderTestXgrp">moderTestXgrp</a></code> can be directly read to produce MA plots for diagnostic purpose.
Please note, that plotting a very high number of points in transparency (eg &gt;10000) may take several seconds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAplotW(
  Mvalue,
  Avalue = NULL,
  useComp = 1,
  filtFin = NULL,
  ProjNa = NULL,
  FCthrs = NULL,
  subTxt = NULL,
  grayIncrem = TRUE,
  col = NULL,
  pch = 16,
  compNa = NULL,
  batchFig = FALSE,
  cexMa = 1.8,
  cexLa = 1.1,
  limM = NULL,
  limA = NULL,
  annotColumn = c("SpecType", "GeneName", "EntryName", "Accession", "Species", "Contam"),
  annColor = NULL,
  cexPt = NULL,
  cexSub = NULL,
  cexTxLab = 0.7,
  namesNBest = NULL,
  NbestCol = 1,
  NaSpecTypeAsContam = TRUE,
  useMar = c(6.2, 4, 4, 2),
  returnData = FALSE,
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAplotW_+3A_mvalue">Mvalue</code></td>
<td>
<p>(numeric, list or MArrayLM-object) main data to plot; if numeric, the content will be used as M-values (and A-values must be provided separateley);
if list or MArrayLM-object, it must conatin list-elements named <code>Mvalue</code> and <code>means</code> to extract all information needed for plotting</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_avalue">Avalue</code></td>
<td>
<p>(numeric, list or data.frame) if <code>NULL</code> it is assumed that M-values can be extracted form argument <code>Avalue</code></p>
</td></tr>
<tr><td><code id="MAplotW_+3A_usecomp">useComp</code></td>
<td>
<p>(integer) choice of one of multiple comparisons present in <code>Mvalue</code> (if generated using <code>moderTestXgrp()</code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_filtfin">filtFin</code></td>
<td>
<p>(matrix or logical) The data may get filtered before plotting: If <code>FALSE</code> no filtering will get applied; if matrix of <code>TRUE</code>/<code>FALSE</code> it will be used as optional custom filter, otherwise (if <code>Mvalue</code> if an <code>MArrayLM</code>-object eg from limma) a default filtering based on the <code>filtFin</code> element will be applied</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_projna">ProjNa</code></td>
<td>
<p>(character) custom title</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_fcthrs">FCthrs</code></td>
<td>
<p>(numeric) Fold-Change threshold (display as line) give as Fold-change and NOT log2(FC)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_subtxt">subTxt</code></td>
<td>
<p>(character) custom sub-title</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_grayincrem">grayIncrem</code></td>
<td>
<p>(logical) if <code>TRUE</code>, display overlay of points (not exceeding threshold) as increased shades of gray</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_col">col</code></td>
<td>
<p>(character) custom color(s) for points of plot (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_pch">pch</code></td>
<td>
<p>(integer) type of symbol(s) to plot (default=16) (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_compna">compNa</code></td>
<td>
<p>depreciated, please use <code>useComp</code> instead</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_batchfig">batchFig</code></td>
<td>
<p>(logical) if <code>TRUE</code> figure title and axes legends will be kept shorter for display on fewer splace</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_cexma">cexMa</code></td>
<td>
<p>(numeric) font-size of title, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_cexla">cexLa</code></td>
<td>
<p>(numeric) size of axis-labels, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_limm">limM</code></td>
<td>
<p>(numeric, length=2) range of axis M-values</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_lima">limA</code></td>
<td>
<p>(numeric, length=2) range of axis A-values</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_annotcolumn">annotColumn</code></td>
<td>
<p>(character) column names of annotation to be extracted (only if <code>Mvalue</code> is <code>MArrayLM</code>-object containing matrix $annot).
The first entry (typically 'SpecType') is used for different symbols in figure, the second (typically 'GeneName') is used as prefered text for annotating the best points (if <code>namesNBest</code> allows to do so.)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_anncolor">annColor</code></td>
<td>
<p>(character or integer) colors for specific groups of annotation (only if <code>Mvalue</code> is <code>MArrayLM</code>-object containing matrix $annot)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_cexpt">cexPt</code></td>
<td>
<p>(numeric) size of points, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_cexsub">cexSub</code></td>
<td>
<p>(numeric) size of subtitle, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_cextxlab">cexTxLab</code></td>
<td>
<p>(numeric) size of text-labels for points, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_namesnbest">namesNBest</code></td>
<td>
<p>(integer or character, length=1) number of best points to add names in figure; if 'passThr' all points passing FC-filter will be selected; 
if the initial object <code>Mvalue</code> contains a list-element called 'annot' the second of the column specified in argument <code>annotColumn</code> will be used as text</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_nbestcol">NbestCol</code></td>
<td>
<p>(character or integer) colors for text-labels of best points</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_naspectypeascontam">NaSpecTypeAsContam</code></td>
<td>
<p>(logical) consider lines/proteins with <code>NA</code> in Mvalue$annot[,&quot;SpecType&quot;] as contaminants (if a 'SpecType' for contaminants already exits)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_usemar">useMar</code></td>
<td>
<p>(numeric,length=4) custom margings (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_returndata">returnData</code></td>
<td>
<p>(logical) optional returning data.frame with (ID, Mvalue, Avalue, FDRvalue, passFilt)</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="MAplotW_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function plots an MA-plot (to the current graphical device); if <code>returnData=TRUE</code>, a data.frame with ($ID, $Mvalue, $Avalue, $FDRvalue, $passFilt) gets returned
</p>


<h3>See Also</h3>

<p>(for PCA) <code><a href="#topic+plotPCAw">plotPCAw</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(wrMisc)
set.seed(2005); mat &lt;- matrix(round(runif(600),2), ncol=6)
rownames(mat) &lt;- c(rep(letters[1:25],each=3), letters[2:26])
MAplotW(mat[,2] -mat[,1], A=rowMeans(mat))
## assume 2 groups with 3 samples each
matMeans &lt;- rowGrpMeans(mat, gr=gl(2,3,labels=LETTERS[3:4]))
MAplotW(M=matMeans[,2] -matMeans[,1], A=matMeans) 
## assume 2 groups with 3 samples each and run moderated t-test (from package 'limma')
tRes &lt;- moderTest2grp(mat, gl(2,3))
MAplotW(tRes$Mval, tRes$Amean)                          
MAplotW(M=tRes$Mval, A=tRes$means, FCth=1.3) 
MAplotW(tRes)
MAplotW(tRes, limM=c(-2,2), FCth=1.3) 

</code></pre>

<hr>
<h2 id='mouseOverHtmlFile'>Create mouse-over interactive html-pages (with links)</h2><span id='topic+mouseOverHtmlFile'></span>

<h3>Description</h3>

<p>This function allows generating html pages with interactive mouse-over to display information for the points of the plot and www-links when clicking based on embedded png file.  
Basically, an html page will be generated which contains a call to display to an image file specified in <code>pngFileNa</code> and in the body below pixel-coordinated will be 
given for disply of information at mouse-over and embedded links.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mouseOverHtmlFile(
  myCoor,
  pngFileNa,
  HtmFileNa = NULL,
  mouseOverTxt = NULL,
  displSi = c(800, 600),
  colNa = NULL,
  tit = "",
  myHtmTit = "",
  myComment = NULL,
  textAtStart = NULL,
  textAtEnd = NULL,
  pxDiam = 5,
  addLinks = NULL,
  linkExt = NULL,
  htmlExt = "htm",
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mouseOverHtmlFile_+3A_mycoor">myCoor</code></td>
<td>
<p>(matrix or data.frame) with initial x&amp;y coordinates of points for plot; with IDs (1st column !!) &amp; coordinates (2nd &amp; 3rd col), data for mouse-over &amp; link (4th &amp; 5th); 
NOTE : if 'colNa' NOT given, colnames of 'myCoor' will be inspected &amp; filtered (columns of non-conform names may get lost) !!!  
Associated with (already existing) figure file 'pngFileNa' and make html page where points may be indicated by mouse-over</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_pngfilena">pngFileNa</code></td>
<td>
<p>(character, length=1) filename for complementary png figure (must already exist)</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_htmfilena">HtmFileNa</code></td>
<td>
<p>(character, length=1) filename for html file produced</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_mouseovertxt">mouseOverTxt</code></td>
<td>
<p>(character, length=1) text for interactive mouse-over in html, if <code>NULL</code>, will use col specified by 1st 'colNa' or (if NULL) rownames of 'myCoor'</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_displsi">displSi</code></td>
<td>
<p>(integer, length=2)  size of image ('pngFileNa') at display in html (width,height), see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_colna">colNa</code></td>
<td>
<p>(character) if not <code>NULL</code> min length of 3 to custom specify the column-names to be used : 1st for mouse-over and 2nd+3rd for coordinates associated (and optional 4th for links)</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_tit">tit</code></td>
<td>
<p>(character) title to be displayed on top of figure</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_myhtmtit">myHtmTit</code></td>
<td>
<p>(character) title of Html page; 'htmlExt' .. checking and correcting filename-extension (only main Html page)</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_mycomment">myComment</code></td>
<td>
<p>(character) modify comment embedded in html-document</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_textatstart">textAtStart</code></td>
<td>
<p>(character) text in html before figure</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_textatend">textAtEnd</code></td>
<td>
<p>(character) text in html after figure</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_pxdiam">pxDiam</code></td>
<td>
<p>(integer, length=1) diameter for mouse-over tip to appear (single val or vector), simpler version/solution than with 'Tooltip' package</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_addlinks">addLinks</code></td>
<td>
<p>(character) for clickable links, either 1) vector of links or 2) single character-chain to be used for pasting to rownames (eg https://www.uniprot.org/uniprot/)
or 3) <code>TRUE</code> to check presence of 4th name specified in 'colNa' to be useed as columname from 'myCoor'   dominates over eventual presence of 4th name in 'colNa'</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_linkext">linkExt</code></td>
<td>
<p>(character) if specified : links will get specified ending, define as <code>NULL</code> or &quot;&quot; for taking 'addLinks' asIs</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_htmlext">htmlExt</code></td>
<td>
<p>(character, length=1) extension used when making html files</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="mouseOverHtmlFile_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Basically theer are two options for defining the path to the image embedded : 
1) Absolute path : I turn you can moove the html to different locations, as long as it still can see the png-file the image can be displayed. However, this may 
not be any more the case when the html file is sent to another person. If the png-file is accessible as url, it should be easily visible. 
2) Relative path : The simplest case would be to give only the file-name with no path at all, thus the png-file is supposed to be in the same directory as the html-file. 
This option is very 'transportable'. 
Basically the same applies to the clickable links which may be provided. In high-throughput biology one typically points here to data-bases accessible
over the internet where urls to specific pages. With UniProt such links can easily be constructed when using protein identifiers as rownames.
</p>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convertPlotCoordPix">convertPlotCoordPix</a></code>; use <code><a href="wrMisc.html#topic+htmlSpecCharConv">htmlSpecCharConv</a></code> to convert special characters for proper html display
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Note, this example writes files to R's tempdir,
## Otherwise, if you simply work in the current directory without spcifying paths you'll 
##  get an html with relatove paths, which simply needs the png file in the same path 
df1 &lt;- data.frame(id=letters[1:10], x=1:10, y=rep(5,10), mou=paste("point",letters[1:10]),
  link=file.path(tempdir(),paste0(LETTERS[1:10],".html")), stringsAsFactors=FALSE)  
## here we'll use R's tempdir, later you may want to choose other locations
pngFile &lt;- file.path(tempdir(),"test01.png")
png(pngFile,width=800, height=600,res=72)
## here we'll just plot a set of horiontal points ...
plot(df1[,2:3],las=1,main="test01")
dev.off()
## Note : Special characters should be converted for display in html pages during mouse-over
library(wrMisc)
df1$mou &lt;- htmlSpecCharConv(df1$mou)
## Let's add the x- and y-coordiates of the points in pixels to the data.frame
df1 &lt;- cbind(df1,convertPlotCoordPix(x=df1[,2],y=df1[,3],plotD=c(800,600),plotRes=72))
head(df1)
## Now make the html-page allowing to display mouse-over to the png made before
htmFile &lt;- file.path(tempdir(),"test01.html")
mouseOverHtmlFile(df1,pngFile,HtmFileNa=htmFile,pxDiam=15,
  textAtStart="Points in the figure are interactive to mouse-over ...",
  textAtEnd="and/or may contain links")
## We still need to make some toy links
for(i in 1:nrow(df1)) cat(paste0("point no ",i," : ",df1[i,1]," x=",df1[i,2]," y=",
  df1[i,3]), file=df1$link[i]) 
## Now we are ready to open the html file using any browser
## Not run:  
browseURL(htmFile)

## End(Not run)
</code></pre>

<hr>
<h2 id='partitionPlot'>Make matrix for layout to partition plotting area</h2><span id='topic+partitionPlot'></span>

<h3>Description</h3>

<p>This function proposes a matrix for use with <code><a href="graphics.html#topic+layout">layout</a></code> to arrange given number of plots to be placed on a page/plotting area.
In certain instances the proposed layout may accomodate slightly more plots, eg <code>nFig=5</code> can not be arranged in 2 or 3 columns without an empty last spot.
Portrait (vertival) or lanscape (horizontal) layout proportions can be chosen.  The user can also impose a given number of columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partitionPlot(
  nFig,
  returnMatr = TRUE,
  horiz = TRUE,
  figNcol = NULL,
  byrow = TRUE,
  silent = TRUE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="partitionPlot_+3A_nfig">nFig</code></td>
<td>
<p>(integer) number of figures to be arrages on single plotting surface (ie window or plotting device)</p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_returnmatr">returnMatr</code></td>
<td>
<p>(logical) will return matrix ready for use by  <code><a href="graphics.html#topic+layout">layout</a></code>; returns vector with nRow and nCol if <code>=FALSE</code></p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_horiz">horiz</code></td>
<td>
<p>(logical) will priviledge horizontal layout if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_figncol">figNcol</code></td>
<td>
<p>(integer) optional number of columns</p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_byrow">byrow</code></td>
<td>
<p>(logical) toggle if output is in order of rows or columns (equivament to <code><a href="base.html#topic+matrix">matrix</a></code></p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="partitionPlot_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix for use with <code>layout</code> or (if <code>returnMatr=FALSE</code> numeric vector with number of segements in x- an y-axis)
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+layout">layout</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>partitionPlot(5); partitionPlot(14,horiz=TRUE)
</code></pre>

<hr>
<h2 id='plotBy2Groups'>Separate and plot data by 2 groups</h2><span id='topic+plotBy2Groups'></span>

<h3>Description</h3>

<p>Plot series of data as membership of 2 different grouping vectors (eg by grp=patient and grp2=age-group).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBy2Groups(
  dat,
  grp,
  grp2 = NULL,
  col = NULL,
  pch = NULL,
  tit = NULL,
  cex = 2,
  lwd = 0.5,
  lty = 2,
  yLab = NULL,
  cexLab = NULL,
  sepLines = FALSE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotBy2Groups_+3A_dat">dat</code></td>
<td>
<p>(numeric) main data (may contain <code>NA</code>)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_grp">grp</code></td>
<td>
<p>(character or factor) grouping of columns of 'dat', eg replicate association</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_grp2">grp2</code></td>
<td>
<p>(character or factor) aadditional/secondary grouping of columns of 'dat'</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_col">col</code></td>
<td>
<p>(character or integer) use custom colors, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_pch">pch</code></td>
<td>
<p>(integer) symbol to mark group-center  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_tit">tit</code></td>
<td>
<p>(character) custom title</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_cex">cex</code></td>
<td>
<p>(numeric) expansion factor for text (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_lwd">lwd</code></td>
<td>
<p>(integer) line-width  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_lty">lty</code></td>
<td>
<p>(integer) line-type  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_ylab">yLab</code></td>
<td>
<p>(character) custom y-axis label</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_cexlab">cexLab</code></td>
<td>
<p>(numeric) expansion factor for labels: 1st value for main groups (<code>grp</code>, eg genotypes), 2nd for detailed text (<code>grp2</code>, eg animal IDs) (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_seplines">sepLines</code></td>
<td>
<p>(logical) optional drawing of horizontal lines aiming to separate groups (in analogy to support vectors)</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
<tr><td><code id="plotBy2Groups_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with <code>$annot</code>, <code>$abund</code> for initial/raw abundance values and <code>$quant</code> with final normalized quantitations, or returns data.frame with annot and quant if <code>separateAnnot=FALSE</code>
</p>


<h3>See Also</h3>

<p><code><a href="utils.html#topic+read.table">read.table</a></code>, <code><a href="wrMisc.html#topic+normalizeThis">normalizeThis</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2020); rand1 &lt;- round(runif(12),2) +rep(1:3,each=4)
plotBy2Groups(rand1, gl(2,6,labels=LETTERS[5:6]), gl(4,3,labels=letters[1:4]))
 
</code></pre>

<hr>
<h2 id='plotLinReg'>Plot linear regression and confidence interval of regression</h2><span id='topic+plotLinReg'></span>

<h3>Description</h3>

<p>This function provides help to display a series of bivariate points given in 'dat' (multiple data formats possible), to model a linear regression and plot the results.
Furthermore, a confidence interval to the regression may be added to the plot, regression parameters get be displayed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotLinReg(
  dat,
  indepVarLst = NULL,
  dependVar = NULL,
  cusTxt = NULL,
  regrLty = 1,
  regrLwd = 1,
  regrCol = 1,
  confInt = 0.95,
  confCol = NULL,
  xLab = NULL,
  yLab = NULL,
  xLim = NULL,
  yLim = NULL,
  tit = NULL,
  nSignif = 3,
  col = 1,
  pch = 1,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotLinReg_+3A_dat">dat</code></td>
<td>
<p>(numeric, data.frame or list) main data to plot/inspect. If numeric 'dat' will be used as dependent variable (y-data)
together with numeric 'indepVarLst' (independent variable); if list, then list-elments <code>indepVarLst</code> and <code>dependVar</code> will be used; if matrix, the the 1st and 2nd colum will be used</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_indepvarlst">indepVarLst</code></td>
<td>
<p>(character) if 'dat' is list, this designes the list element with the explanatory or independent variable (ie the variable used for explaining, typically x-data)</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_dependvar">dependVar</code></td>
<td>
<p>(character) if 'dat' is list, this designes the list element with dependent variable (ie the variable to be explained, typically y-data) to test</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_custxt">cusTxt</code></td>
<td>
<p>(character) optional custom text to display in subtitle (instead of p-value to H0: slope.regression=0)</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_regrlty">regrLty</code></td>
<td>
<p>(integer) line type for regression</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_regrlwd">regrLwd</code></td>
<td>
<p>(integer) line width for regression</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_regrcol">regrCol</code></td>
<td>
<p>(integer) color of regression-line</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_confint">confInt</code></td>
<td>
<p>(numeric, between 0 and 1) the probabiity alpha for the regression interval, if <code>NULL</code> no confidence intervall will be plotted/calculated</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_confcol">confCol</code></td>
<td>
<p>(character) (background) color for confidence-interval</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_xlab">xLab</code></td>
<td>
<p>(character) optional custom x-label</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_ylab">yLab</code></td>
<td>
<p>(character) optional custom y-label</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_xlim">xLim</code></td>
<td>
<p>(numeric) custom limit for x-axis (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_ylim">yLim</code></td>
<td>
<p>(numeric) custom limit for y-axis (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_tit">tit</code></td>
<td>
<p>(character) optional title</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_nsignif">nSignif</code></td>
<td>
<p>(integer) number of significant digits for regression parameters in subtitle of plot</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_col">col</code></td>
<td>
<p>(integer or character) custom color for points (choose <code>NULL</code> for not plotting the actual data)</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_pch">pch</code></td>
<td>
<p>(integer or character) type of symbol for points (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
<tr><td><code id="plotLinReg_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This functions simply plots (to the current graphical devce); an invisible list containing $data, $linRegr, $confInterval (if calculated) may be returned, too
</p>


<h3>See Also</h3>

<p><code><a href="wrMisc.html#topic+exclExtrValues">exclExtrValues</a></code> for decision of potential outliers; <code><a href="graphics.html#topic+hist">hist</a></code>, <code><a href="#topic+vioplotW">vioplotW</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2020); dat1 &lt;- rep(1:6,each=2) +runif(12,0,1)
plotLinReg(dat1, gl(6,2))
## extract elements out of list :
li2 &lt;- list(aa=gl(5,2), bb=dat1[1:10])
plotLinReg(li2, indepVarLst="aa", dependVar="bb")
</code></pre>

<hr>
<h2 id='plotPCAw'>PCA plot with bag-plot to highlight groups</h2><span id='topic+plotPCAw'></span>

<h3>Description</h3>

<p>This function allows to plot <a href="https://en.wikipedia.org/wiki/Principal_component_analysis">principal components analysis (PCA)</a>,
with options to show center and potential outliers for each of the groups (columns of data).
The main points of this implementation consist in offering bagplots to highlight groups of columns/samples and support to (object-oriented)
output from <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> and <a href="https://CRAN.R-project.org/package=wrProteo">wrProteo</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPCAw(
  dat,
  sampleGrp,
  tit = NULL,
  useSymb = c(21:25, 9:12, 3:4),
  center = TRUE,
  scale. = TRUE,
  colBase = NULL,
  useSymb2 = NULL,
  cexTxt = 1,
  cexSub = 0.6,
  displBagPl = TRUE,
  outCoef = 2,
  getOutL = FALSE,
  showLegend = TRUE,
  nGrpForMedian = 6,
  pointLabelPar = NULL,
  rowTyName = "genes",
  rotatePC = NULL,
  suplFig = TRUE,
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPCAw_+3A_dat">dat</code></td>
<td>
<p>(matrix, data.frame, MArrayLM-object or list) data to plot. Note: <code>NA</code>-values cannot be processed - all lines with non-finite data (eg <code>NA</code>) will be omitted !
In case of  MArrayLM-object or list <code>dat</code> must conatain list-element named 'datImp','dat' or 'data'.</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_samplegrp">sampleGrp</code></td>
<td>
<p>(character or factor) should be factor describing groups of replicates, NAs are not supported</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_tit">tit</code></td>
<td>
<p>(character) custom title</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_usesymb">useSymb</code></td>
<td>
<p>(integer) symbols to use (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_center">center</code></td>
<td>
<p>(logical or numeric) decide if variables should be shifted to be zero centered, argument passed to <code><a href="stats.html#topic+prcomp">prcomp</a></code></p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_scale.">scale.</code></td>
<td>
<p>(logical or numeric) decide if scaling to obtain unit variance, argument passed to <code><a href="stats.html#topic+prcomp">prcomp</a></code>
Alternatively, a vector of length equal the number of columns of x can be supplied. The value is passed to scale.</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_colbase">colBase</code></td>
<td>
<p>(character or integer) use custom colors</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_usesymb2">useSymb2</code></td>
<td>
<p>(integer) symbol to mark group-center (no mark of group-center if default NULL) (equivalent to <code>pch</code>, see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_cextxt">cexTxt</code></td>
<td>
<p>(integer) expansion factor for text (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_cexsub">cexSub</code></td>
<td>
<p>(integer) expansion factor for subtitle line text (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_displbagpl">displBagPl</code></td>
<td>
<p>(logical) if <code>TRUE</code>, show bagPlot (group-center) if &gt;3 points per group otherwise the average-confidence-interval</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_outcoef">outCoef</code></td>
<td>
<p>(numeric) parameter for defining outliers, see  <code><a href="#topic+addBagPlot">addBagPlot</a></code> (equivalent to <code>range</code> in <code><a href="graphics.html#topic+boxplot">boxplot</a></code>)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_getoutl">getOutL</code></td>
<td>
<p>(logical) return outlyer samples/values</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_showlegend">showLegend</code></td>
<td>
<p>(logical or character) toggle to display legend, if character it designes the location within the plot to display the legend ('bottomleft','topright', etc..)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_ngrpformedian">nGrpForMedian</code></td>
<td>
<p>(integer) decide if group center should be displayed via its average or median value: If group has less than 'nGrpForMedian' values, the average will be used, otherwise the median; if <code>NULL</code> no group centers will be displayed</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_pointlabelpar">pointLabelPar</code></td>
<td>
<p>(character) define formatting for optional labels next to points in main figure (ie PC1 vs PC2); may be <code>TRUE</code> or list containing elments 'textLabel', 'textCol', 'textCex',
'textOffSet', 'textAdj' for fine-tuning</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_rowtyname">rowTyName</code></td>
<td>
<p>(character) for subtitle : specify nature of rows (genes, proteins, probesets,...)</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_rotatepc">rotatePC</code></td>
<td>
<p>(integer) optional rotation (by -1) for fig&amp;ure of the principal components specified by index</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_suplfig">suplFig</code></td>
<td>
<p>(logical) to include plots vs 3rd principal component (PC) and Screeplot</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="plotPCAw_+3A_debug">debug</code></td>
<td>
<p>(logical) display additional messages for debugging</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One motivation for this implementation of plotting PCA was to provide a convenient way for doing so with of MArrayLM-objects or lists
as created by <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> and <a href="https://CRAN.R-project.org/package=wrProteo">wrProteo</a>.
</p>
<p>Another motivation for this implementation come from integrating the idea of bag-plots to better visualize different groups of points
(if they can be organized so beforehand as distinct groups) :
The main body of data is shown as 'bag-plots' (a bivariate boxplot, see <a href="https://en.wikipedia.org/wiki/Bagplot">Bagplot</a>)
with different transparent colors to highlight the core part of different groups (if they contain more than 2 values per group).
Furthermore, group centers are shown as average or median (see 'nGrpForMedian') with stars &amp; index-number (if &lt;25 groups).
</p>
<p>Layout is automatically set to 2 or 4 subplots (if plotting more than 2 principal components makes sense).
</p>
<p>Note : This function uses <code><a href="stats.html#topic+prcomp">prcomp</a></code> for calculating Eigenvectors and principal components, with default <code>center=TRUE</code> and <code>scale.=FALSE</code> (different to <code>princomp()</code>. which standardizes by default).
This way the user has to option to intervene on arguments <code>center</code> and <code>scale.</code>. However, this should be done with care.
</p>
<p>Note: <code>NA</code>-values cannot (by definition) be processed by (any) PCA - all lines with any non-finite values/content (eg <code>NA</code>) will be omitted !
</p>
<p>Note : Package RColorBrewer may be used if available.
</p>
<p>For more options with PCA (and related methods) you may also see also the package  <a href="https://CRAN.R-project.org/package=FactoMineR">FactoMineR</a>
which provides a very wide spectrum of possibiities, in particular for combined numeric and categorical data.
</p>


<h3>Value</h3>

<p>This function make a plot and may retiurn an optional matrix of outlyer-data (depending on argument <code>getOutL</code>)
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+prcomp">prcomp</a></code> (used here for the PCA underneith) , <code><a href="stats.html#topic+princomp">princomp</a></code>, see the package <a href="https://CRAN.R-project.org/package=FactoMineR">FactoMineR</a> for multiple plotting options or ways of combining categorical and numeric data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2019); dat1 &lt;- matrix(round(c(rnorm(1000), runif(1000,-0.9,0.9)),2),
  ncol=20, byrow=TRUE) + matrix(rep(rep(1:5,6:2), each=100), ncol=20)
biplot(prcomp(dat1))        # traditional plot
(grp = factor(rep(LETTERS[5:1],6:2)))
plotPCAw(dat1, grp)
</code></pre>

<hr>
<h2 id='plotW2Leg'>x-y plot with 2 legends</h2><span id='topic+plotW2Leg'></span>

<h3>Description</h3>

<p>This is a modified version of <code>plot</code> for 2-dimensional data,
allowing to choose symbols and colors of points according to two additional columns of <code>dat</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotW2Leg(
  dat,
  useCol = c("logp", "slope", "medAbund", "startFr"),
  tit = NULL,
  subTi = NULL,
  subCex = 0.9,
  pch = 21:25,
  xlim = NULL,
  ylim = NULL,
  xlab = NULL,
  ylab = NULL,
  ablines = NULL,
  legendloc = "topright",
  txtLegend = NULL,
  histLoc = "bottomleft",
  legHiTi = NULL,
  silent = TRUE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotW2Leg_+3A_dat">dat</code></td>
<td>
<p>(matrix or data.frame) main input</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_usecol">useCol</code></td>
<td>
<p>(character or integer) columns form <code>dat</code>: The 1st and 2nd column are used as x- and y-axis</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_tit">tit</code></td>
<td>
<p>(character) optional custom title</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_subti">subTi</code></td>
<td>
<p>(character) optional custom subtitle</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_subcex">subCex</code></td>
<td>
<p>(numeric) cex-like expansion factor for subtitle (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_pch">pch</code></td>
<td>
<p>(integer) symbols to use for plotting (see also <code><a href="graphics.html#topic+par">par</a></code>), will be associated to 4th column of <code>useCol</code></p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_xlim">xlim</code></td>
<td>
<p>(numeric, length=2) x- axis limits (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_ylim">ylim</code></td>
<td>
<p>(numeric, length=2) y- axis limits (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_xlab">xlab</code></td>
<td>
<p>(character) custom x-axis label</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_ylab">ylab</code></td>
<td>
<p>(character) custom x-axis label</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_ablines">ablines</code></td>
<td>
<p>(list) optional horzontal and/or vertical gray dashed guide-lines</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_legendloc">legendloc</code></td>
<td>
<p>(character) location of legend (of symbols)</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_txtlegend">txtLegend</code></td>
<td>
<p>(character) optional label for legend (of symbols)</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_histloc">histLoc</code></td>
<td>
<p>(character) location of histomgram-legend (of 3rd column of <code>useCol</code>)</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_leghiti">legHiTi</code></td>
<td>
<p>(character) optional title for histomgram-legend</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="plotW2Leg_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>graphical output only
</p>


<h3>See Also</h3>

<p>(standard plots) <code>plot</code> from the package <code>base</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x1 &lt;- cbind(x=c(2,1:7), y=8:1 +runif(8), grade=rep(1:4,2))
plotW2Leg(x1,useCol=c("x","y","y","grade"))
</code></pre>

<hr>
<h2 id='profileAsClu'>Plot profiles according to CLustering</h2><span id='topic+profileAsClu'></span>

<h3>Description</h3>

<p>This function was made for visualuzing the result of clustering of a numeric vector or clustering along multiple columns of a matrix.
The data will be plotted like a reglar scatter-plot, but some extra space is added to separate clusters and dashed lines highlight cluster-borders.
If no mean/representative value is spacified, a geometric mean will be calculated along all columns of  <code>dat</code>.
In case <code>dat</code> has multiple columns, a legend and a representative (default geometric mean) dashed grey line will be displayed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>profileAsClu(
  dat,
  clu,
  meanD = NULL,
  tit = NULL,
  col = NULL,
  pch = NULL,
  xlab = NULL,
  ylab = NULL,
  meCol = "grey",
  meLty = 1,
  meLwd = 1,
  cex = NULL,
  cexTit = NULL,
  legLoc = "bottomleft",
  silent = TRUE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="profileAsClu_+3A_dat">dat</code></td>
<td>
<p>(matrix or data.frame) main input with data to plot as points</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_clu">clu</code></td>
<td>
<p>(numeric or character) clustering results; if length=1 and character this term will be understood as colum-name with cluster-numbers from <code>dat</code></p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_meand">meanD</code></td>
<td>
<p>(numeric) mean/representative of multiple series for display as lines; if length=1 and character this term will be understood as columname with cluster-numbers from <code>dat</code></p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_tit">tit</code></td>
<td>
<p>(character) optional custom title</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_col">col</code></td>
<td>
<p>(character) custom colors</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_pch">pch</code></td>
<td>
<p>(integer) custom plotting symbols (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_xlab">xlab</code></td>
<td>
<p>(character) custom x-axis label</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_ylab">ylab</code></td>
<td>
<p>(character) custom y-axis label</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_mecol">meCol</code></td>
<td>
<p>(character) color for (dashed) line of mean/representative values</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_melty">meLty</code></td>
<td>
<p>(integer) line-type line of mean/representative values (see also <code>lty</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_melwd">meLwd</code></td>
<td>
<p>(numeric) line-width line of mean/representative values (see also <code>lwd</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_cex">cex</code></td>
<td>
<p>(numeric) cex-like expansion factor  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_cextit">cexTit</code></td>
<td>
<p>(numeric) cex-like expansion factor for title  (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_legloc">legLoc</code></td>
<td>
<p>(character) legend location</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="profileAsClu_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This functin returns a plot only
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2020); dat1 &lt;- runif(12)/2 + rep(6:8, each=4)
dat1Cl &lt;- stats::kmeans(dat1, 3)$cluster
dat1Cl &lt;- 5- dat1Cl              # bring cluster-numbers in ascending form
dat1Cl[which(dat1Cl &gt;3)] &lt;- 1    # bring cluster-numbers in ascending form
profileAsClu(dat1, clu=dat1Cl)
</code></pre>

<hr>
<h2 id='staggerdCountsPlot'>Staggered Chart for Ploting Counts to Multiple Leveles of the Threshold used</h2><span id='topic+staggerdCountsPlot'></span>

<h3>Description</h3>

<p>The basic idea of this plot is to show how counts data change while shifting a threshold-criterium.
At each given threshold the counts are plotted like a staggered bar-chart (or staggered histogram) but without vertical lines to illustrated the almost continuous change
from preceedig or following threshold-value.
Initially this plot was designed for showing the absolute count-data used when constructing roc-curves (eg using 
the function <code>summarizeForROC</code> of package <a href="https://CRAN.R-project.org/package=wrProteo">wrProteo</a> ).
The main input should furnish the panel of threshold as one column and the coresponding counts data as min 2 columns. 
The threshold coumns gets specified using the argument <code>threColumn</code>, the counts-data may either be specified using argument <code>countsCol</code>
or be searched using <code><a href="base.html#topic+grep">grep</a></code> using column-names containing the text given in argument <code>varCountNa</code> with may be combined with 
a fixed preceeding part given as argument <code>fixedCountPat</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>staggerdCountsPlot(
  roc,
  threColumn = 1,
  countsCol = NULL,
  fixedCountPat = "n.pos.",
  varCountNa = NULL,
  sortAscending = TRUE,
  vertLine = NULL,
  col = NULL,
  tit = NULL,
  logScale = FALSE,
  las.alph = 2,
  displMaxSpec = TRUE,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="staggerdCountsPlot_+3A_roc">roc</code></td>
<td>
<p>(numeric matrix or data.frame) main input: one column with thresholds and multiple columns of assoicated count data</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_threcolumn">threColumn</code></td>
<td>
<p>(integer or character) to specify the column with threshold-data, in typica proteomics benchmark studies this would be 'alph' (for the statistical test threshold)</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_countscol">countsCol</code></td>
<td>
<p>(character of integer, min length=2) choice of column(s) with count-data in 'roc' to be used for display, if not <code>NULL</code> will override alternative search of columns using 'varCountNa' and 'fixedCountPat'</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_fixedcountpat">fixedCountPat</code></td>
<td>
<p>(character) optional pattern to help identifying counts-data: if not <code>NULL</code> it will be used as fixed part in column names to get pasted to <code>varCountNa</code>.
In proteomics benchmark studies this would typically be 'n.pos.'</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_varcountna">varCountNa</code></td>
<td>
<p>(character) alternative way to select the columns from 'roc': searched using <code><a href="base.html#topic+grep">grep</a></code> using column-names containing the text given in argument <code>varCountNa</code> with may be combined with a fixed preceeding part given as argument <code>fixedCountPat</code>
In proteomics benchmark studies this would typically be the species-abbreciations (eg 'H','S','E')</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_sortascending">sortAscending</code></td>
<td>
<p>(logical) decide if data should be sorted ascending or descending</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_vertline">vertLine</code></td>
<td>
<p>(numeric) for optional vertical line, typically used to highlight alpha 0.05</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_col">col</code></td>
<td>
<p>(character) custom colors, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_tit">tit</code></td>
<td>
<p>(character) cutom title</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_logscale">logScale</code></td>
<td>
<p>(logical) display threshld values (x-axis) on log-scale</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_las.alph">las.alph</code></td>
<td>
<p>(numeric) orientation of label of alpha-cutoff, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_displmaxspec">displMaxSpec</code></td>
<td>
<p>(logical) display on right side of figure max count value of contributing group species</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_debug">debug</code></td>
<td>
<p>(logical) additonal messages for debugging</p>
</td></tr>
<tr><td><code id="staggerdCountsPlot_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allows easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Investigate count data prepared for plotting ROC curves : cumulative counts plot by species (along different statistical test thresholds).
Note : Package <a href="https://CRAN.R-project.org/package=wrProteo">wrProteo</a> may be used to prepare input (matrix of ROC data).
</p>


<h3>Value</h3>

<p>plot only
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+ecdf">ecdf</a></code>, for preparing input to ROC: function <code>summarizeForROC</code> in package <a href="https://CRAN.R-project.org/package=wrProteo">wrProteo</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2019); test1 &lt;- cbind(a=sample.int(n=7,size=50,repl=TRUE),
  b=sample.int(n=11,size=50,repl=TRUE),c=sample.int(n=18,size=50,repl=TRUE))
test1 &lt;- cbind(alph=seq(0,1,length.out=50),a=cumsum(test1[,1]),b=cumsum(test1[,2]),
  c=cumsum(test1[,3]))
staggerdCountsPlot(test1,countsCol=c("a","b","c"))
## example below requires the package wrProteo
</code></pre>

<hr>
<h2 id='vioplotW'>Violin-plots version W</h2><span id='topic+vioplotW'></span>

<h3>Description</h3>

<p>This function allows generating <a href="https://en.wikipedia.org/wiki/Violin_plot">Violin plots</a>) using a variety of input formats and offers additional options for colors.
Main input may be multiple vectors, a matrix or list of multiple data-elements (entries may be of variable length),
individual colors for different sets of data or color-gradients can be specified, and the display of n per set of data was integtated
(based on an inspiration from the discussion 'Removing-NAs-from-dataframe-for-use-in-Vioplot' on the forum Nabble).
It is also possible to plot pairwise half-violins for easier pairwise-comparisons (using <code>halfViolin="pairwise"</code>).
Many arguments are kept similar to <a href="https://CRAN.R-project.org/package=vioplot">vioplot</a> (here, the package <code>vioplot</code> is not required/used).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vioplotW(
  x,
  ...,
  finiteOnly = TRUE,
  removeEmpty = FALSE,
  halfViolin = FALSE,
  boxCol = "def",
  hh = NULL,
  xlim = NULL,
  ylim = NULL,
  nameSer = NULL,
  cexNameSer = NULL,
  horizontal = FALSE,
  col = "rainbow",
  border = "black",
  xlab = NULL,
  ylab = NULL,
  cexLab = NULL,
  cexAxis = NULL,
  lty = 1,
  pointCol = NULL,
  cexPt = NULL,
  tit = NULL,
  las = 1,
  lwd = 1,
  rectCol = "black",
  at = 0,
  add = FALSE,
  wex = NULL,
  silent = FALSE,
  debug = FALSE,
  callFrom = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vioplotW_+3A_x">x</code></td>
<td>
<p>(matrix, list or data.frame) data to plot, or first series of data</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_...">...</code></td>
<td>
<p>(numeric) additional sets of data to plot</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_finiteonly">finiteOnly</code></td>
<td>
<p>(logical) eliminate non-finite elements to avoid potential errors (eg when encountering <code>NA</code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_removeempty">removeEmpty</code></td>
<td>
<p>(logical) omit empty series (or less than 4 finite numeric entries) of data from plot</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_halfviolin">halfViolin</code></td>
<td>
<p>(logical or character) decide with <code>TRUE</code> or <code>FALSE</code> if full or only half of violins should be plotted, if &quot;pairwise&quot; always 2 data-sets will be plotted back-to-back</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_boxcol">boxCol</code></td>
<td>
<p>(character) decide if boxplot should be adde inside the violin, use &quot;def&quot; for default transparent grey</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_hh">hh</code></td>
<td>
<p>(numeric, length &lt;4) smoothing parameter (standard deviation to kernel function, if omited anormal optimal smoothing parameter is used); equivalent to argument <code>h</code> in package  <a href="https://CRAN.R-project.org/package=vioplot">vioplot</a> ; see also <code><a href="sm.html#topic+sm.density">sm.density</a></code></p>
</td></tr>
<tr><td><code id="vioplotW_+3A_xlim">xlim</code></td>
<td>
<p>(<code>NULL</code> or numeric, length=2) custom limit on x-axis, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="vioplotW_+3A_ylim">ylim</code></td>
<td>
<p>(<code>NULL</code> or numeric, length=2) custom limit on y-axis, see also <code><a href="graphics.html#topic+par">par</a></code></p>
</td></tr>
<tr><td><code id="vioplotW_+3A_nameser">nameSer</code></td>
<td>
<p>(character) custom label for data-sets or columns (length must match number of data-sets)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_cexnameser">cexNameSer</code></td>
<td>
<p>(numeric) size of individual data-series labels as cex-expansion factor (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_horizontal">horizontal</code></td>
<td>
<p>(logical) orientation of plot</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_col">col</code></td>
<td>
<p>(character or integer) custom colors or gradients like 'rainbow', 'grayscale', 'heat.colors', 'topo.colors', 'Spectral' or 'Paired',  or you may use colors made by the package <a href="https://CRAN.R-project.org/package=colorRamps">colorRamps</a></p>
</td></tr>
<tr><td><code id="vioplotW_+3A_border">border</code></td>
<td>
<p>(character) custom color for figure border</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_xlab">xlab</code></td>
<td>
<p>(character) custom x-axis label</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_ylab">ylab</code></td>
<td>
<p>(character) custom y-axis label</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_cexlab">cexLab</code></td>
<td>
<p>(numeric) size of axis labels as cex-expansion factor (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_cexaxis">cexAxis</code></td>
<td>
<p>(numeric) size of numeric y-axis labels as cex-expansion factor (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_lty">lty</code></td>
<td>
<p>(integer) line-type for linear regression line (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_pointcol">pointCol</code></td>
<td>
<p>(character or numeric) display of median: color (defauly white)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_cexpt">cexPt</code></td>
<td>
<p>(numeric) display of median : size of point as cex-expansion factor (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_tit">tit</code></td>
<td>
<p>(character) custom title to figure</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_las">las</code></td>
<td>
<p>(integer) orientation of axis labels (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_lwd">lwd</code></td>
<td>
<p>(integer) width of line(s) (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_rectcol">rectCol</code></td>
<td>
<p>(character) color of rectangle</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_at">at</code></td>
<td>
<p>(numeric) custom locoation of data-series names, ie the points at which tick-marks are to be drawn, will be passed to <code><a href="graphics.html#topic+axis">axis</a></code>, it's length ust match the number of data-sets</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_add">add</code></td>
<td>
<p>(logical) add to existing plot if <code>TRUE</code></p>
</td></tr>
<tr><td><code id="vioplotW_+3A_wex">wex</code></td>
<td>
<p>(integer) relative expansion factor of the violin</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
<tr><td><code id="vioplotW_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The (relative) width of the density-profiles ('Violins') may be manually adjusted using the parameter <code>wex</code> which applieds to all profiles drawn.
Please note that different n (eg for different columns) will not be shown, so far.
</p>
<p>Note : Arguments have to be given with full names, lazy evaluation of arguments will not work properly with this function (since '...' is used to capture additional data-sets).
Note : <a href="https://CRAN.R-project.org/package=vioplot">vioplot</a> offers better options for plotting formulas
</p>


<h3>Value</h3>

<p>This function plots a figure (to the current graphical device)
</p>


<h3>See Also</h3>

<p>the package <a href="https://CRAN.R-project.org/package=vioplot">vioplot</a>, <code><a href="sm.html#topic+sm">sm</a></code> is used for the density estimation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2013)
dat6 &lt;- matrix(round(rnorm(300) +3, 1), ncol=6,
	 dimnames=list(paste0("li",1:50), letters[19:24]))
vioplotW(dat6)
## variable number of elements (each n is displayed)
dat6b &lt;- apply(dat6, 2, function(x) x[which(x &lt; 5)])
dat6b[[4]] &lt;- dat6b[[4]][dat6b[[4]] &lt; 4]
vioplotW(dat6b, col="Spectral")
vioplotW(dat6b, col="Spectral" ,halfViolin="pairwise", horizontal=TRUE)
vioplotW(dat6b, col="Spectral", halfViolin="pairwise", horizontal=FALSE)
</code></pre>

<hr>
<h2 id='VolcanoPlotW'>Volcano-Plot (Statistical Test Outcome versus Relative Change)</h2><span id='topic+VolcanoPlotW'></span>

<h3>Description</h3>

<p>This type of plot is very common in high-throughput biology, see <a href="https://en.wikipedia.org/wiki/Volcano_plot_(statistics)">Volcano-plot</a>.
Basically, this plot allows comparing the outcome of a statistical test to the relative change (ie log fold-change, M-value).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VolcanoPlotW(
  Mvalue,
  pValue = NULL,
  useComp = 1,
  filtFin = NULL,
  ProjNa = NULL,
  FCthrs = NULL,
  FdrList = NULL,
  FdrThrs = NULL,
  FdrType = NULL,
  subTxt = NULL,
  grayIncrem = TRUE,
  col = NULL,
  pch = 16,
  compNa = NULL,
  batchFig = FALSE,
  cexMa = 1.8,
  cexLa = 1.1,
  limM = NULL,
  limp = NULL,
  annotColumn = c("SpecType", "GeneName", "EntryName", "Accession", "Species", "Contam"),
  annColor = NULL,
  expFCarrow = FALSE,
  cexPt = NULL,
  cexSub = NULL,
  cexTxLab = 0.7,
  namesNBest = NULL,
  NbestCol = 1,
  sortLeg = "descend",
  NaSpecTypeAsContam = TRUE,
  useMar = c(6.2, 4, 4, 2),
  returnData = FALSE,
  callFrom = NULL,
  silent = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VolcanoPlotW_+3A_mvalue">Mvalue</code></td>
<td>
<p>(MArrayLM-object, numeric or matrix) data to plot; M-values are typically calculated as difference of log2-abundance values and 'pValue' the mean of log2-abundance values;
M-values and p-values may be given as 2 columsn of a matrix, in this case the argument <code>pValue</code> should remain NULL.
One may also furnish MArrayLM-objects created bpackage wrProteo or limma.</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_pvalue">pValue</code></td>
<td>
<p>(numeric, list or data.frame) if <code>NULL</code> it is assumed that 2nd column of 'Mvalue' contains the p-values to be used</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_usecomp">useComp</code></td>
<td>
<p>(integer, length=1) choice of which of multiple comparisons to present in <code>Mvalue</code> (if generated using <code>moderTestXgrp()</code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_filtfin">filtFin</code></td>
<td>
<p>(matrix or logical) The data may get filtered before plotting: If <code>FALSE</code> no filtering will get applied; if matrix of <code>TRUE</code>/<code>FALSE</code> it will be used as optional custom filter, otherwise (if <code>Mvalue</code> if an <code>MArrayLM</code>-object eg from limma) a default filtering based on the <code>filtFin</code> element will be applied</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_projna">ProjNa</code></td>
<td>
<p>(character) custom title</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_fcthrs">FCthrs</code></td>
<td>
<p>(numeric) Fold-Change threshold (display as line) give as Fold-change and NOT log2(FC), default at 1.5, set to <code>NA</code> for omitting</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_fdrlist">FdrList</code></td>
<td>
<p>(numeric) FDR data or name of list-element</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_fdrthrs">FdrThrs</code></td>
<td>
<p>(numeric) FDR threshold (display as line), default at 0.05, set to <code>NA</code> for omitting</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_fdrtype">FdrType</code></td>
<td>
<p>(character) FDR-type to extract if <code>Mvalue</code> is 'MArrayLM'-object (eg produced by from <code>moderTest2grp</code> etc);
if <code>NULL</code> it will search for suitable fields/values in this order : 'FDR','BH','lfdr' and 'BY'</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_subtxt">subTxt</code></td>
<td>
<p>(character) custom sub-title</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_grayincrem">grayIncrem</code></td>
<td>
<p>(logical) if <code>TRUE</code>, display overlay of points  (not exceeding thresholds) as increased shades of gray</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_col">col</code></td>
<td>
<p>(character) custom color(s) for points of plot (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_pch">pch</code></td>
<td>
<p>(integer) type of symbol(s) to plot (default=16) (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_compna">compNa</code></td>
<td>
<p>(character) names of groups compared</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_batchfig">batchFig</code></td>
<td>
<p>(logical) if <code>TRUE</code> figure title and axes legends will be kept shorter for display on fewer splace</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_cexma">cexMa</code></td>
<td>
<p>(numeric) font-size of title, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_cexla">cexLa</code></td>
<td>
<p>(numeric) size of axis-labels, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_limm">limM</code></td>
<td>
<p>(numeric, length=2) range of axis M-values</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_limp">limp</code></td>
<td>
<p>(numeric, length=2) range of axis FDR / p-values</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_annotcolumn">annotColumn</code></td>
<td>
<p>(character) column names of annotation to be extracted (only if <code>Mvalue</code> is <code>MArrayLM</code>-object containing matrix $annot).
The first entry (typically 'SpecType') is used for different symbols in figure, the second (typically 'GeneName') is used as prefered text for annotating the best points (if <code>namesNBest</code> allows to do so.)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_anncolor">annColor</code></td>
<td>
<p>(character or integer) colors for specific groups of annotation (only if <code>Mvalue</code> is <code>MArrayLM</code>-object containing matrix $annot)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_expfcarrow">expFCarrow</code></td>
<td>
<p>(logical, character or numeric) optional adding arrow for expected fold-change; if <code>TRUE</code> the expected ratio will be extracted from numeric concentration-indications from sample-names
if <code>numeric</code> an arrow will be drawn (M-value as 1st position, color of 2nd position of vector).</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_cexpt">cexPt</code></td>
<td>
<p>(numeric) size of points, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_cexsub">cexSub</code></td>
<td>
<p>(numeric) size of subtitle, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_cextxlab">cexTxLab</code></td>
<td>
<p>(numeric) size of text-labels for points, as expansion factor (see also <code>cex</code> in <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_namesnbest">namesNBest</code></td>
<td>
<p>(integer or character) for display of labels to points in figure: if 'pass','passThr' or 'signif' all points passing thresholds; if numeric (length=1) this number of best points will get labels
if the initial object <code>Mvalue</code> contains a list-element called 'annot' the second of the column specified in argument <code>annotColumn</code> will be used as text</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_nbestcol">NbestCol</code></td>
<td>
<p>(character or integer) colors for text-labels of best points, also used for arrow</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_sortleg">sortLeg</code></td>
<td>
<p>(character) sorting of 'SpecType' annotation either ascending ('ascend') or descending ('descend'), no sorting if <code>NULL</code></p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_naspectypeascontam">NaSpecTypeAsContam</code></td>
<td>
<p>(logical) consider lines/proteins with <code>NA</code> in Mvalue$annot[,&quot;SpecType&quot;] as contaminants (if a 'SpecType' for contaminants already exits)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_usemar">useMar</code></td>
<td>
<p>(numeric,length=4) custom margings (see also <code><a href="graphics.html#topic+par">par</a></code>)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_returndata">returnData</code></td>
<td>
<p>(logical) optional returning data.frame with (ID, Mvalue, pValue, FDRvalue, passFilt)</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_callfrom">callFrom</code></td>
<td>
<p>(character) allow easier tracking of messages produced</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_silent">silent</code></td>
<td>
<p>(logical) suppress messages</p>
</td></tr>
<tr><td><code id="VolcanoPlotW_+3A_debug">debug</code></td>
<td>
<p>(logical) additional messages for debugging</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In high-throughput biology data are typically already transformed to log2 and thus, the 'M'-values (obtained by subtrating two group means) represent a relative change.
Output from statistical testing by <code><a href="wrMisc.html#topic+moderTest2grp">moderTest2grp</a></code> or <code><a href="wrMisc.html#topic+moderTestXgrp">moderTestXgrp</a></code> can be directly read to produce Volcano plots for diagnostic reasons.
Please note, that plotting a very high number of points (eg &gt;10000) in transparency may take several seconds.
</p>


<h3>Value</h3>

<p>This function simply plots an MA-plot (to the current graphical device), if <code>returnData=TRUE</code> an optional data.frame with (ID, Mvalue, pValue, FDRvalue, passFilt) can be returned
</p>


<h3>See Also</h3>

<p>(for PCA) <code><a href="#topic+plotPCAw">plotPCAw</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(wrMisc)
set.seed(2005); mat &lt;- matrix(round(runif(900),2), ncol=9)
rownames(mat) &lt;- paste0(rep(letters[1:25], each=4), rep(letters[2:26],4))
mat[1:50,4:6] &lt;- mat[1:50,4:6] + rep(c(-1,1)*0.1,25)
mat[3:7,4:9] &lt;- mat[3:7,4:9] + 0.7
mat[11:15,1:6] &lt;- mat[11:15,1:6] - 0.7
## assume 2 groups with 3 samples each
gr3 &lt;- gl(3, 3, labels=c("C","A","B"))
tRes2 &lt;- moderTest2grp(mat[,1:6], gl(2,3), addResults = c("FDR","means"))
# Note: due to the small number of lines only FDR chosen to calculate
VolcanoPlotW(tRes2)
## Add names of points passing custom filters
VolcanoPlotW(tRes2, FCth=1.3, FdrThrs=0.2, namesNBest="passThr")

## assume 3 groups with 3 samples each
tRes &lt;- moderTestXgrp(mat, gr3, addResults = c("FDR","means"))
# Note: due to the small number of lines only FDR chosen to calculate
VolcanoPlotW(tRes)
VolcanoPlotW(tRes, FCth=1.3, FdrThrs=0.2)
VolcanoPlotW(tRes, FCth=1.3, FdrThrs=0.2, useComp=2)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
