<!DOCTYPE html><html><head><title>Help for package wsprv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wsprv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#wsprv'><p>A weighted selection probability is developed to locate individual rare variants associated with multiple phenotypes.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Weighted Selection Probability for Rare Variant Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Xianglong Liang [aut, cre],
  Hokeun Sun [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xianglong Liang &lt;xianglongliang@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A weighted selection probability to locate rare variants associated with multiple phenotypes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, mnormt, stats</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-12-17 05:14:18 UTC; xiang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-17 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='wsprv'>A weighted selection probability is developed to locate individual rare variants associated with multiple phenotypes.</h2><span id='topic+wsprv'></span><span id='topic+weight_sp'></span>

<h3>Description</h3>

<p>Recently, rare variant association studies with multiple phenotypes have drawn a lot of attentions because association signals can be boosted when rare variants are related with more than one phenotype. Most of existing statistical methods to identify rare variants associated with multiple phenotypes are based on a group test, where a gene or a genetic region is tested one at a time. However, these methods are not designed to locate individual rare variants within a gene or a genetic region. We propose a weighted selection probability to locate individual rare variants within a group after a multiple-phenotype based group test finds significance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weight_sp(
  x,
  y,
  alpha = 1,
  penalty.factor = NULL,
  standardize = TRUE,
  type.multinomial = c("grouped", "ungrouped"),
  rep = 100,
  rate = 0.05,
  gamma = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wsprv_+3A_x">x</code></td>
<td>
<p>A <code class="reqn">n \times (m+p)</code> matrix with <code class="reqn">n</code> samples, <code class="reqn">m</code> covariates and <code class="reqn">p</code> rare variants where <code class="reqn">m</code> can be zero, i.e., there does not exist covariates.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_y">y</code></td>
<td>
<p>A <code class="reqn">n \times Q</code> phenotype matrix with <code class="reqn">n</code> samples and <code class="reqn">Q</code> phenotypes where <code class="reqn">Q&gt;1</code>.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_alpha">alpha</code></td>
<td>
<p>The mixing parameter of elastic-net, <code>alpha=1</code> is the lasso, and <code>alpha=0</code> is the ridge. Default value is 1.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_penalty.factor">penalty.factor</code></td>
<td>
<p>Separate penalty factors factors can be applied to each coefficient. Can be <code>0</code> for some variables, which implies no shrinkage, and that variable is always included in the model.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_standardize">standardize</code></td>
<td>
<p>Genotype standardization. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_type.multinomial">type.multinomial</code></td>
<td>
<p>A group lasso penalty is used on the multinomial coefficients for a variable when 'grouped'. It ensures the multinomial coefficents are all in or out. Default is 'grouped'.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_rep">rep</code></td>
<td>
<p>The number of bootstrap replications. We recommend to use 100 or more to compute weighted selection probability. Default value is 100.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_rate">rate</code></td>
<td>
<p>A tuning parameter represents rate of degree of freedom to the number of rare variants. Default value is 0.05.</p>
</td></tr>
<tr><td><code id="wsprv_+3A_gamma">gamma</code></td>
<td>
<p>The upper <code>gamma</code> quantile of selection frequencies of individual variants each phenotype to compute the threshold. Default value is 0.01.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The penalty function of <code>elastic-net</code> is defined as </p>
<p style="text-align: center;"><code class="reqn">\lambda(\alpha||\beta||_1+\frac{(1-\alpha)}{2}||\beta||_2^2),</code>
</p>
<p> where <code class="reqn">\alpha</code> is a mixing proportion of ridge and the lasso, and <code class="reqn">\beta</code> is regression coefficients. This penalty is equivalent to the Lasso penalty if <code>alpha=1</code>. <br /> <br /> Let <code class="reqn">\eta</code> be the degree of freedom and it depends on the tuning parameter <code class="reqn">\lambda</code>, and <code>rate</code> is computed as </p>
<p style="text-align: center;"><code class="reqn">rate=\frac{\eta}{p},</code>
</p>
<p> Note that <code class="reqn">\eta \leq n</code> is set up in <code>weight_sp</code> function. <br /> <br /> Let <code class="reqn">\delta_{\gamma}</code> be a threshold of <code class="reqn">SF</code> and it depends on the upper <code class="reqn">\gamma^{th}</code> qunatile value of <code class="reqn">SF</code>. Where <code class="reqn">SF=\left\{SF_{11}(\eta),SF_{21}(\eta),\cdots,SF_{pQ}(\eta) \right\}</code> is a set that contains selection frequencies of individual rare variants each phenotype.
</p>


<h3>Value</h3>

<table>
<tr><td><code>res</code></td>
<td>
<p>A matrix contains the order of weighted selection probabilities from the largest to the smallest and the corresponding weighted selection probabilities.</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>eta used.</p>
</td></tr>
<tr><td><code>bootstrap.rep</code></td>
<td>
<p>The number of bootstrap replications used.</p>
</td></tr>
<tr><td><code>rate</code></td>
<td>
<p>The tuning parameter <code>rate</code> used.</p>
</td></tr>
<tr><td><code>gamma</code></td>
<td>
<p>The upper <code>gamma</code> quantile of selection frequencies of individual rare variants each phenotype used.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate simulation data
 n &lt;- 400
 p &lt;- 100
 q &lt;- 5
 MAF &lt;- 0.01
 geno.prob &lt;- rbind((1-MAF)^2,2*(1-MAF)*MAF,MAF^2)
 x &lt;- matrix(NA,n,p)
 set.seed(1)
 for(i in 1:p) x[,i] &lt;- sample(0:2,n,prob=geno.prob,replace=TRUE)
 beta &lt;- c(rep(3.0,10),rep(0,(p-10)))
 cova &lt;- matrix(0.75,q,q)
 diag(cova) &lt;- 1
 require(mnormt)
 err.mat &lt;- rmnorm(n,rep(0,q),cova)

 y1 &lt;- x %*% beta+err.mat[,1]
 y2 &lt;- x %*% beta+err.mat[,2]
 y &lt;- cbind(y1,y2,err.mat[,3:5])
 # Weighted selection probabilities for individual rare variants without covariates.
 #If rep=100, time consuming.
 wsp.rv1 &lt;- weight_sp(x,y,rep=5) # continuous phenotypes

 # Weighted selection probabilities for individual rare variants with covariates.
 #If rep=100, time consuming.
 cx &lt;- cbind(rnorm(n),sample(0:1,n,replace=TRUE))
 x &lt;- cbind(cx,x)
 penalty.factor &lt;- c(rep(0,2),rep(1,p))
 colnames(x) &lt;- c('Age','Gender',paste0('V',3:102))
 wsp.rv2 &lt;- weight_sp(x,y,penalty.factor=penalty.factor,rep=5) # continuous phenotypes


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
