<!DOCTYPE html><html lang="en"><head><title>Help for package wtest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wtest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CpG.pos'>
<p>Position of CpG Sites for the methylation Data</p></a></li>
<li><a href='#diabetes.geno'>
<p>Genotype Data of Candidate Diabetes Genes</p></a></li>
<li><a href='#genotype'>
<p>Example Genotype Data</p></a></li>
<li><a href='#hf'><p>Patameter Estimation for W-test Probability Distribution</p></a></li>
<li><a href='#hf.snps.meth'><p>Parameter Estimation for W-test Probability Distribution in Gene-methylation Data</p></a></li>
<li><a href='#maf'><p>Minor Allele Frequency</p></a></li>
<li><a href='#methylation'>
<p>Example Methylation Data</p></a></li>
<li><a href='#methylation.recode'><p>Recode Methylation Data</p></a></li>
<li><a href='#odds.ratio'><p>Odds Ratio</p></a></li>
<li><a href='#phenotype1'>
<p>Phenotype of the diabetes.geno Data</p></a></li>
<li><a href='#phenotype2'>
<p>Simulated Phenotype of the genotype-methylation Data</p></a></li>
<li><a href='#SNP.pos'>
<p>Pseudo Position of SNPs for the genotype Data</p></a></li>
<li><a href='#w.diagnosis'><p>W-test Probability Distribution Diagnostic Plot</p></a></li>
<li><a href='#w.qqplot'><p>W P-values Diagnosis by Q-Q Plot</p></a></li>
<li><a href='#wtest'><p>W-test</p></a></li>
<li><a href='#wtest.high'><p>W-test for High Order Interaction Analysis</p></a></li>
<li><a href='#wtest.snps.meth'><p>W-test for Gene-methylation Interaction Analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The W-Test for Genetic Interactions Testing</td>
</tr>
<tr>
<td>Version:</td>
<td>3.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Rui Sun, Maggie Haitian Wang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rui Sun &lt;rsunzju@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform the calculation of W-test, diagnostic checking, calculate
    minor allele frequency (MAF) and odds ratio.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-09-03 06:49:04 UTC; rui</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-09-03 07:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CpG.pos'>
Position of CpG Sites for the methylation Data
</h2><span id='topic+CpG.pos'></span>

<h3>Description</h3>

<p>This dataset contains pseudo positions of 200 CpG sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CpG.pos)</code></pre>


<h3>Format</h3>

<p>The format is:
CpG sites by rows; column 1: names of CpG sites; column 2: positions of CpG sites.
</p>

<hr>
<h2 id='diabetes.geno'>
Genotype Data of Candidate Diabetes Genes
</h2><span id='topic+diabetes.geno'></span>

<h3>Description</h3>

<p>A data frame contains 23 SNPs for 115 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(diabetes.geno)</code></pre>


<h3>Format</h3>

<p>The format is:
subjects by rows and genotypes by columns.
</p>


<h3>References</h3>

<p>Wang, M. H., Li, J., Yeung, V. S. Y., Zee, B. C. Y., Yu, R. H. Y., Ho, S., &amp; Waye, M. M. Y. (2014). Four pairs of gene-gene interactions associated with increased risk for type 2 diabetes (CDKN2BAS-KCNJ11), obesity (SLC2A9-IGF2BP2, FTO-APOA5), and hypertension (MC4R-IGF2BP2) in Chinese women. Meta Gene, 2, 384-391. http://doi.org/10.1016/j.mgene.2014.04.010</p>

<hr>
<h2 id='genotype'>
Example Genotype Data
</h2><span id='topic+genotype'></span>

<h3>Description</h3>

<p>This simulated data frame contains 300 individuals and 200 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genotype)</code></pre>


<h3>Format</h3>

<p>The format is:
subjects by rows, and genotype by columns.
</p>

<hr>
<h2 id='hf'>Patameter Estimation for W-test Probability Distribution</h2><span id='topic+hf'></span>

<h3>Description</h3>

<p>Estimate parameters (<em>h</em> and <em>f</em>) for <code>W-test</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hf(data, w.order, B = 400, n.sample = nrow(data),
  n.marker = "default.nmarker")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hf_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns and subjects in the rows. Genotypes should be coded as (0, 1, 2) or (0, 1).</p>
</td></tr>
<tr><td><code id="hf_+3A_w.order">w.order</code></td>
<td>
<p>a numeric number. <code>w.order</code> = 1 gives main effect calculation. <code>w.order</code> = 2 gives pairwise interaction calculation. <code>w.order</code> &gt; 2 gives high order interaction calculation.</p>
</td></tr>
<tr><td><code id="hf_+3A_b">B</code></td>
<td>
<p>a numeric number specifying the number of replicates. Default is 400.</p>
</td></tr>
<tr><td><code id="hf_+3A_n.sample">n.sample</code></td>
<td>
<p>a numeric number specifying the number of samples to be used for estimating parameters. Default is the total number of samples in the data.</p>
</td></tr>
<tr><td><code id="hf_+3A_n.marker">n.marker</code></td>
<td>
<p>a numeric value, the number of biomarkers to include in bootstrapping. For <code>order</code> = 1, the default = min(P, 1000), and for order = 2, default = min(P, 50). P is the total number of markers in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a set of <em>h</em> and <em>f</em> values indexed by <em>k</em>, estimated automatically. For main effect, <em>k</em> is the number of levels of a predictor variable. For interactions, <em>k</em> is the number of categorical combinations of a variable pair.
</p>


<h3>Author(s)</h3>

<p>Rui Sun, Maggie Haitian Wang
</p>


<h3>References</h3>

<p>Maggie Haitian Wang, Rui Sun, Junfeng Guo, Haoyi Weng, Jack Lee, Inchi Hu, Pak Sham and Benny C.Y. Zee (2016). A fast and powerful W-test for pairwise epistasis testing. Nucleic Acids Research. doi:10.1093/nar/gkw347.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wtest">wtest</a></code>, <code><a href="#topic+w.diagnosis">w.diagnosis</a></code>, <code><a href="#topic+w.qqplot">w.qqplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)

# Please note that parameter B is recommended to be greater than 400.
# For high order interaction analysis (w.order &gt; 2), it is recommended to use default n.sample.
hf1 &lt;- hf(data = diabetes.geno, w.order = 1, B = 100)
hf2 &lt;- hf(data = diabetes.geno, w.order = 2, B = 80)
</code></pre>

<hr>
<h2 id='hf.snps.meth'>Parameter Estimation for W-test Probability Distribution in Gene-methylation Data</h2><span id='topic+hf.snps.meth'></span>

<h3>Description</h3>

<p>Estimate parameters (<em>h</em> and <em>f</em>) for <code>W-test</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hf.snps.meth(B = 400, geno, meth, y, geno.pos, meth.pos, window.size,
  n.sample = nrow(geno), n.pair = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hf.snps.meth_+3A_b">B</code></td>
<td>
<p>a numeric number specifying the number of bootstrapping times. Default is 400.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_geno">geno</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1). SNP names should be stored as column names.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_meth">meth</code></td>
<td>
<p>a data frame or matrix containing methylation data in the columns. Methylation data should be recoded as (0, 1, 2) or (0, 1). Names of CpG sites should be stored as column names.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_y">y</code></td>
<td>
<p>a numeric vector of 0 or 1.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_geno.pos">geno.pos</code></td>
<td>
<p>a data frame containing SNP names and positions in two columns.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_meth.pos">meth.pos</code></td>
<td>
<p>a data frame containing CpG names and positions in two columns.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_window.size">window.size</code></td>
<td>
<p>a numeric number specifying the size of genome distance. Interaction of the SNPs and CpG sites located within the size of genome distance will be evaluated exhaustively.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_n.sample">n.sample</code></td>
<td>
<p>a numeric number specifying the number of samples to be included for estimating parameters. Default is the total number of samples.</p>
</td></tr>
<tr><td><code id="hf.snps.meth_+3A_n.pair">n.pair</code></td>
<td>
<p>a numeric value, the number of SNP-CpG pairs to use in bootstrapping. Default = min(P, 1000). P is the total number of pairs within the <code>window.size</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a set of <em>h</em> and <em>f</em> values indexed by <em>k</em>, estimated automatically. Variable <em>k</em> is the number of categorical combinations of a variable pair.
</p>


<h3>Author(s)</h3>

<p>Rui Sun, Maggie Haitian Wang
</p>


<h3>References</h3>

<p>Maggie Haitian Wang, Rui Sun, Junfeng Guo, Haoyi Weng, Jack Lee, Inchi Hu, Pak Sham and Benny C.Y. Zee (2016). A fast and powerful W-test for pairwise epistasis testing. Nucleic Acids Research. doi:10.1093/nar/gkw347.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNP.pos)
data(CpG.pos)
data(genotype)
data(methylation)
data(phenotype2)

# Please note that parameter B is recommended to be greater than 400.
hf.pair &lt;- hf.snps.meth(B = 80, geno = genotype, meth = methylation, y = phenotype2,
                        geno.pos = SNP.pos, meth.pos = CpG.pos, window.size = 1000)

</code></pre>

<hr>
<h2 id='maf'>Minor Allele Frequency</h2><span id='topic+maf'></span>

<h3>Description</h3>

<p>Calculate minor allele frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maf(data, which.snp = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maf_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1).</p>
</td></tr>
<tr><td><code id="maf_+3A_which.snp">which.snp</code></td>
<td>
<p>a numeric value, indicating which SNP to calculate. When which.snp = NULL, MAF of all the markers is calculated. Default is NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MAF of one marker.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)
result &lt;- maf(diabetes.geno, which.snp=10)
</code></pre>

<hr>
<h2 id='methylation'>
Example Methylation Data
</h2><span id='topic+methylation'></span>

<h3>Description</h3>

<p>This data frame contains 300 samples and 200 CpG sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(methylation)</code></pre>


<h3>Format</h3>

<p>The format is:
subjects by rows and methylation by columns.
</p>

<hr>
<h2 id='methylation.recode'>Recode Methylation Data</h2><span id='topic+methylation.recode'></span>

<h3>Description</h3>

<p>Code a CpG variable into two levels (high and low) by the two-mean clustering method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>methylation.recode(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="methylation.recode_+3A_data">data</code></td>
<td>
<p>a data frame or matrix contains methylation data in the columns.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(methylation)
data.recoded &lt;- methylation.recode(methylation)
</code></pre>

<hr>
<h2 id='odds.ratio'>Odds Ratio</h2><span id='topic+odds.ratio'></span>

<h3>Description</h3>

<p>Calculate odds ratio for a single SNP or a pair of SNPs. Single marker odds ratio is computed by contigency table as the odds of disease at minor allele vs the odds of diseases at major allele. Odds ratio of a pair of SNPs is calculated by the Logistic Regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>odds.ratio(data, y, w.order, which.marker)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="odds.ratio_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1), according to minor allele count.</p>
</td></tr>
<tr><td><code id="odds.ratio_+3A_y">y</code></td>
<td>
<p>binary values.</p>
</td></tr>
<tr><td><code id="odds.ratio_+3A_w.order">w.order</code></td>
<td>
<p>a numeric number taking values 1 or 2. If w.order = 1, odds ratio of main effect is calculated. If w.order = 2, odds ratio of pairwise interaction is calculated.</p>
</td></tr>
<tr><td><code id="odds.ratio_+3A_which.marker">which.marker</code></td>
<td>
<p>a numeric vector, when w.order = 1, a single value indicating the column index of the variable to calculate; when w.order = 2, a vector indicating the column index of a SNP-pair to calculate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The odds ratio of a SNP or a SNP-pair.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)
data(phenotype1)
y &lt;- as.numeric(phenotype1)
OR.snp4.snp8 &lt;- odds.ratio(diabetes.geno, y, w.order=2, which.marker = c(4,8))
OR.snp4 &lt;- odds.ratio(diabetes.geno, y, w.order = 1, which.marker = 4)
</code></pre>

<hr>
<h2 id='phenotype1'>
Phenotype of the diabetes.geno Data
</h2><span id='topic+phenotype1'></span>

<h3>Description</h3>

<p>A binary variable indicate the status of 115 individuals (1 = affacted, 0 = unaffacted).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenotype1)</code></pre>


<h3>Format</h3>

<p>The format is:
115 rows and 1 column.
</p>


<h3>References</h3>

<p>Wang, M. H., Li, J., Yeung, V. S. Y., Zee, B. C. Y., Yu, R. H. Y., Ho, S., &amp; Waye, M. M. Y. (2014). Four pairs of gene-gene interactions associated with increased risk for type 2 diabetes (CDKN2BAS-KCNJ11), obesity (SLC2A9-IGF2BP2, FTO-APOA5), and hypertension (MC4R-IGF2BP2) in Chinese women. Meta Gene, 2, 384-391. http://doi.org/10.1016/j.mgene.2014.04.010</p>

<hr>
<h2 id='phenotype2'>
Simulated Phenotype of the genotype-methylation Data
</h2><span id='topic+phenotype2'></span>

<h3>Description</h3>

<p>The phenotype of the 300 individuals (1 = affacted, 0 = unaffacted).</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenotype2)</code></pre>


<h3>Format</h3>

<p>The format is:
300 rows and 1 column.
</p>

<hr>
<h2 id='SNP.pos'>
Pseudo Position of SNPs for the genotype Data
</h2><span id='topic+SNP.pos'></span>

<h3>Description</h3>

<p>Positions of 200 SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNP.pos)</code></pre>


<h3>Format</h3>

<p>The format is:
SNPs by rows; column 1: names of SNPs, column 2: positions of SNPs.
</p>

<hr>
<h2 id='w.diagnosis'>W-test Probability Distribution Diagnostic Plot</h2><span id='topic+w.diagnosis'></span>

<h3>Description</h3>

<p>Diagnostic checking of W-test probability distribution estimation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>w.diagnosis(data, w.order = c(1, 2), n.rep = 10,
  n.sample = nrow(data), n.marker = ncol(data), hf1 = "default.hf1",
  hf2 = "default.hf2", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="w.diagnosis_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1).</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_w.order">w.order</code></td>
<td>
<p>an integer value of 0 or 1. <code>w.order</code> = 1 gives main effect calculation; <code>w.order</code> = 2 gives pairwise calculation.</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_n.rep">n.rep</code></td>
<td>
<p>a numeric value, the number of bootstrapping times.</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_n.sample">n.sample</code></td>
<td>
<p>a numeric value, the number of samples to use in bootstrapping. Default is the total number of samples in the data.</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_n.marker">n.marker</code></td>
<td>
<p>a numeric value, the number of markers to use in bootstrapping. Default is the total number of markers.</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_hf1">hf1</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate main effect, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 3. Needed when <code>w.order</code> = 1.</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_hf2">hf2</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate interaction associations, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 9. Needed when <code>w.order</code> = 2.</p>
</td></tr>
<tr><td><code id="w.diagnosis_+3A_...">...</code></td>
<td>
<p>graphical parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function evaluates the input W values of main or interaction effects using a set of null Y by the <code>W-test</code>, and the evaluation is performed in several bootstrap samples to achieve fast and stable output. The W histogram and its theoretical Chi-squared distribution density with <em>f</em> degrees of freedom are plotted indexed by <em>k</em>. Close overlaying of the histogram and the probability density curve indicates that the estimated <em>h</em> and <em>f</em> give a good test statistic probability distribution.
</p>


<h3>Author(s)</h3>

<p>Rui Sun, Maggie Haitian Wang
</p>


<h3>References</h3>

<p>Maggie Haitian Wang, Rui Sun, Junfeng Guo, Haoyi Weng, Jack Lee, Inchi Hu, Pak Sham and Benny C.Y. Zee (2016). A fast and powerful W-test for pairwise epistasis testing. Nucleic Acids Research. doi:10.1093/nar/gkw347.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wtest">wtest</a></code>, <code><a href="#topic+hf">hf</a></code>, <code><a href="#topic+w.qqplot">w.qqplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)
# Please note that parameter B is recommended to be greater than 400.
hf1 &lt;- hf(data = diabetes.geno, w.order = 1, B = 100)
hf2 &lt;- hf(data = diabetes.geno, w.order = 2, B = 50)
w.diagnosis(diabetes.geno, w.order = 1, n.rep = 100, hf1 = hf1, main=NULL, xlab=NULL, ylab=NULL)
w.diagnosis(diabetes.geno, w.order = 2, n.rep = 100, hf2 = hf2, main=NULL, xlab=NULL, ylab=NULL)
</code></pre>

<hr>
<h2 id='w.qqplot'>W P-values Diagnosis by Q-Q Plot</h2><span id='topic+w.qqplot'></span>

<h3>Description</h3>

<p>Draw a Q-Q plot for W-test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>w.qqplot(data, y, w.order = c(1, 2), input.poolsize = 200,
  hf1 = "default.hf1", hf2 = "default.hf2", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="w.qqplot_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1).</p>
</td></tr>
<tr><td><code id="w.qqplot_+3A_y">y</code></td>
<td>
<p>a numeric vector of 0 or 1.</p>
</td></tr>
<tr><td><code id="w.qqplot_+3A_w.order">w.order</code></td>
<td>
<p>a numeric number taking values 1 or 2. <code>w.order</code> = 1 gives main effect Q-Q plot. <code>w.order</code> = 2 gives interaction Q-Q plot.</p>
</td></tr>
<tr><td><code id="w.qqplot_+3A_input.poolsize">input.poolsize</code></td>
<td>
<p>a numeric number; The maximum number of SNPs to calculate the Q-Q plot. Default is 200. The <code>input.poolsize</code> is suggested to set as 1000 for <code>w.order</code> = 1, and 200 for <code>w.order</code> = 2.</p>
</td></tr>
<tr><td><code id="w.qqplot_+3A_hf1">hf1</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate main effect, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 3. Needed when <code>w.order</code> = 1.</p>
</td></tr>
<tr><td><code id="w.qqplot_+3A_hf2">hf2</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate interaction associations, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 9. Needed when <code>w.order</code> = 2.</p>
</td></tr>
<tr><td><code id="w.qqplot_+3A_...">...</code></td>
<td>
<p>graphical parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With a given data and y, the p-value of W-test is calculated at given <em>h</em> and <em>f</em> values, which are plotted against the theoretical distribution.
</p>


<h3>Value</h3>

<p>Q-Q plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)
data(phenotype1)
## Step 1. HF Calculation
# Please note that parameter B is recommended to be greater than 400.
hf1&lt;-hf(data = diabetes.geno, w.order = 1, B = 200)

## Step 2. Q-Q Plot
w.qqplot(data = diabetes.geno, y = phenotype1, w.order = 1, hf1 = hf1, cex =.5)
abline(0,1)
</code></pre>

<hr>
<h2 id='wtest'>W-test</h2><span id='topic+wtest'></span>

<h3>Description</h3>

<p>This function performs the <code>W-test</code> to calculate main effect or pairwise interactions in case-control studies
for categorical data sets. The test measures target variables' distributional difference between cases and controls via a combined
log of odds ratio. It follows a Chi-squared probability distribution with data-adaptive degrees of freedom. For pairwise interaction
calculation, the user has 3 options: (1) calculate a single pair's W-value, (2) calculate pairwise interaction for a list of variables,
which p-values are smaller than a threshold (<code>input.pval</code>); (3) calculate the pairwise interaction exhaustively for all variables.
For both main and interaction calculation, the output can be filtered by p-values, such that only sets with smaller p-value
than a threshold (<code>output.pval</code>) will be returned. An extension of the W-test for rare variant analysis is available in <code>zfa</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wtest(data, y, w.order = c(1, 2), hf1 = "default.hf1",
  hf2 = "default.hf2", which.marker = NULL, output.pval = NULL,
  sort = TRUE, input.pval = 0.1, input.poolsize = 150)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wtest_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1).</p>
</td></tr>
<tr><td><code id="wtest_+3A_y">y</code></td>
<td>
<p>a numeric vector of 0 or 1.</p>
</td></tr>
<tr><td><code id="wtest_+3A_w.order">w.order</code></td>
<td>
<p>an integer value of 0 or 1. <code>w.order</code> = 1 for main effect calculation; <code>w.order</code> = 2 for pairwise calculation.</p>
</td></tr>
<tr><td><code id="wtest_+3A_hf1">hf1</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate main effect, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 3. Needed when <code>w.order</code> = 1.</p>
</td></tr>
<tr><td><code id="wtest_+3A_hf2">hf2</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate interaction associations, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 9. Needed when <code>w.order</code> = 2.</p>
</td></tr>
<tr><td><code id="wtest_+3A_which.marker">which.marker</code></td>
<td>
<p>a numeric vector, when <code>w.order</code> = 1, a single value indicating the column index of a SNP to calculate, when <code>w.order</code> = 2, a vector indicating the column index of a SNP-pair to calculate. Default <code>which.marker</code> = NULL means main or interaction effect will be calculated exhaustively.</p>
</td></tr>
<tr><td><code id="wtest_+3A_output.pval">output.pval</code></td>
<td>
<p>a p-value threshold for filtering the output. If NULL, all the results will be listed; otherwise, the function will only output the results with p-values smaller than the <code>output.pval</code>.</p>
</td></tr>
<tr><td><code id="wtest_+3A_sort">sort</code></td>
<td>
<p>a logical value indicating whether or not to sort the output by p-values in ascending order. Default = TRUE.</p>
</td></tr>
<tr><td><code id="wtest_+3A_input.pval">input.pval</code></td>
<td>
<p>a p-value threshold to select markers for pairwise calculation, used only when <code>w.order</code> = 2. When specified, only markers with main effect p-value smaller than <code>input.pval</code> will be passed to interaction effect calculation. Default = 0.10. Set <code>input.pval</code> = NULL or 1 for exhaustive pairwise calculation.</p>
</td></tr>
<tr><td><code id="wtest_+3A_input.poolsize">input.poolsize</code></td>
<td>
<p>an integer, with value less than the number of input variables. It is an optional filter to control the maximum number of variables to include in pairwise calculation, used only when <code>w.order</code> = 2. When specified, the function selects top <code>input.poolsize</code> number of variables to calculate pairwise interactions. It can be used separately or jointly with <code>input.pval</code>, whichever gives smaller input variable pool size. Default = 50. Set <code>input.poolsize</code> = NULL for exhaustive pairwise calculation. It can be useful for data exploration, when there are a large number of variables with extremely small main effect p-values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>W-test is a model-free statistical test to measure main effect or pairwise interactions in case-control studies with categorical variables.
Theoretically, the test statistic follows a Chi-squared distribution with <em>f</em> degrees of freedom. The data-adaptive degree of freedom <em>f</em>,
and a scalar <em>h</em> in the test statistics allow the W-test to correct for distributional bias due to sparse data and small sample size.
Let <em>k</em> be the number of columns of the 2 by <em>k</em> contingency table formed by a single variable or a variable pair.
When the sample size is large and there is no population stratification, the <em>h</em> and <em>f</em> will approximate well to the theoretical
value <em>h</em> = (<em>k</em>-1)/<em>k</em>, and <em>f</em> = <em>k</em>-1. When sample size is small and there is population stratification, the <em>h</em> and
<em>f</em> will vary to correct for distributional bias caused by the data structure.
</p>
<p>When <code>w.order</code> =2, the <code>wtest()</code> will automatically calculate the main effect first and then do a pre-filter before calculating interactions.
This filtering is to avoid overloading the memory before having a better understanding of the data. User can specify a smaller input.pval such as 0.05 or 0.001
for less output, or <code>input.pval</code>=1 or NULL for exhaustive pairwise calculation. Another optional filter is <code>input.poolsize</code>. It will take the top <code>input.poolsize</code>
number of variables to calculated pairwise effect exhaustively, selected by smallest p-value; when used together with <code>input.pval</code>, the smaller set will be passed to pairwise calculation.
</p>


<h3>Value</h3>

<p>An object <code>"wtest"</code> containing:
</p>
<table role = "presentation">
<tr><td><code>order</code></td>
<td>
<p>the &quot;w.order&quot; specified.</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>When <code>w.order</code> = 1, the test results include: the ID of SNP, the W value, <em>k</em>, and p-value. When <code>w.order</code> = 2 and <code>which.marker</code> = NULL, the test results include: (information of the pair, column 1-5) [SNP1 name, SNP2, name, W-value, k, p-value]; (Information of the first variable in the pair, column 6-8) [W-value, k, p-value]; (Information of the second variable in the pair, column 9-11) [W-value, k, p-value].</p>
</td></tr>
<tr><td><code>hf1</code></td>
<td>
<p>The <em>h</em> and <em>f</em> values used in main effect calculation.</p>
</td></tr>
<tr><td><code>hf2</code></td>
<td>
<p>The <em>h</em> and <em>f</em> values used in pairwise interaction calculation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rui Sun, Maggie Haitian Wang
</p>


<h3>References</h3>

<p>Maggie Haitian Wang, Rui Sun, Junfeng Guo, Haoyi Weng, Jack Lee, Inchi Hu, Pak Sham and Benny C.Y. Zee (2016). A fast and powerful W-test for pairwise epistasis testing. Nucleic Acids Research. doi:10.1093/nar/gkw347.
</p>
<p>Maggie Haitian Wang, Haoyi Weng, Rui Sun, Jack Lee, William K.K. Wu, Ka Chun Chong, Benny C.Y. Zee. (2017). A Zoom-Focus algorithm (ZFA) to locate the optimal testing region for rare variant association tests. Bioinformatics, 33(15), 2330-2336.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hf">hf</a></code>, <code><a href="#topic+w.diagnosis">w.diagnosis</a></code>, <code><a href="#topic+w.qqplot">w.qqplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)
data(phenotype1)

## Step 1. HF Calculation
# Please note that parameter B is recommended to be greater than 400.
hf1 &lt;- hf(data = diabetes.geno, w.order = 1, B = 100)
hf2 &lt;- hf(data = diabetes.geno, w.order = 2, B = 50)

## Step 2. W-test Calculation
w1 &lt;- wtest(diabetes.geno, phenotype1, w.order = 1, hf1 = hf1)
w2 &lt;- wtest(diabetes.geno, phenotype1, w.order = 2, input.pval = 0.3,
            input.poolsize = 50, output.pval = 0.01, hf1 = hf1, hf2 = hf2)
w.pair &lt;- wtest(diabetes.geno, phenotype1, w.order = 2, which.marker = c(10,13), hf2 = hf2)
</code></pre>

<hr>
<h2 id='wtest.high'>W-test for High Order Interaction Analysis</h2><span id='topic+wtest.high'></span>

<h3>Description</h3>

<p>This function performs the <code>W-test</code> to calculate high-order interactions in case-control studies
for categorical data sets. The test measures target variables' distributional difference between cases and controls via a combined
log of odds ratio. It follows a Chi-squared probability distribution with data-adaptive degrees of freedom. For high-order interaction
calculation, the user has 3 options: (1) calculate W-test of a set of SNPs, (2) calculate high-order interaction for a list of variables,
which p-values are smaller than a threshold (<code>input.pval</code>); (3) calculate high-order interaction exhaustively for all variables.
Output can be filtered by p-values, such that only sets with smaller p-value than a threshold (<code>output.pval</code>) will be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wtest.high(data, y, w.order = 3, hf1 = "default.hf1",
  hf.high.order = "default.high", which.marker = NULL,
  output.pval = NULL, sort = TRUE, input.pval = 0.1,
  input.poolsize = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wtest.high_+3A_data">data</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns. Genotypes should be coded as (0, 1, 2) or (0, 1).</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_y">y</code></td>
<td>
<p>a numeric vector of 0 or 1.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_w.order">w.order</code></td>
<td>
<p>an integer value, indicating the order of high-way interactions. For example, <code>w.order</code> = 3 for three-way interaction analysis.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_hf1">hf1</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate main effect, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 3.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_hf.high.order">hf.high.order</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate high-order interactions, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), where <em>k</em> is the number of genotype combinations of a set of SNPs.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_which.marker">which.marker</code></td>
<td>
<p>a numeric vector indicating the column index of a set of SNPs to calculate. Default <code>which.marker</code> = NULL gives an exhaustively high-order interaction calculation.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_output.pval">output.pval</code></td>
<td>
<p>a p-value threshold for filtering the output. If NULL, all the results will be listed; otherwise, the function will only output the results with p-values smaller than the <code>output.pval</code>.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_sort">sort</code></td>
<td>
<p>a logical value indicating whether or not to sort the output by p-values in ascending order. Default = TRUE.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_input.pval">input.pval</code></td>
<td>
<p>a p-value threshold to select markers for high-order interaction calculation, used only when <code>w.order</code> &gt; 2. When specified, only markers with main effect p-value smaller than <code>input.pval</code> will be passed to interaction effect calculation. Default = 0.10. Set <code>input.pval</code> = NULL or 1 for exhaustive calculation.</p>
</td></tr>
<tr><td><code id="wtest.high_+3A_input.poolsize">input.poolsize</code></td>
<td>
<p>an integer, with value less than the number of input variables. It is an optional filter to control the maximum number of variables to include in high-order interaction calculation, used only when <code>w.order</code> &gt; 2. When specified, the function selects top <code>input.poolsize</code> number of variables to calculate interactions. It can be used separately or jointly with <code>input.pval</code>, whichever gives smaller input pool size. Default = 10. Set <code>input.poolsize</code> = NULL for exhaustive calculation. It can be useful for data exploration, when there are a large number of variables with extremely small main effect p-values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>W-test is a model-free statistical test orginally proposed to measure main effect or pairwise interactions in case-control studies with categorical variables.
It can be extended to high-order interaction detection by the <em>wtest.high()</em> function. Theoretically, the test statistic follows a Chi-squared distribution with <em>f</em> degrees of freedom. The data-adaptive degree of freedom <em>f</em>,
and a scalar <em>h</em> in the test statistics allow the W-test to correct for distributional bias due to sparse data and small sample size.
Let <em>k</em> be the number of columns of the 2 by <em>k</em> contingency table formed by a single variable or a variable pair.
When the sample size is large and there is no population stratification, the <em>h</em> and <em>f</em> will approximate well to the theoretical
value <em>h</em> = (<em>k</em>-1)/<em>k</em>, and <em>f</em> = <em>k</em>-1. When sample size is small and there is population stratification, the <em>h</em> and
<em>f</em> will vary to correct for distributional bias caused by the data structure.
</p>
<p>When <code>w.order</code> &gt; 2, the <code>wtest()</code> will automatically calculate the main effect first and then do a pre-filter before calculating interactions.
This filtering is to avoid overloading the memory before having a better understanding of the data. User can specify a smaller input.pval such as 0.05 or 0.001
for less output, or <code>input.pval</code>=1 or NULL for exhaustive high-order interaction calculation. Another optional filter is <code>input.poolsize</code>. It will select the top <code>input.poolsize</code>
number of variables, ranked by p-values, to calculate high-order interactions. When used together with <code>input.pval</code>, the algorithm selects the smaller set in the high-order calculation.
</p>


<h3>Value</h3>

<p>An object <code>"wtest"</code> containing:
</p>
<table role = "presentation">
<tr><td><code>order</code></td>
<td>
<p>the &quot;w.order&quot; specified.</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>When order &gt; 2 and which.marker = NULL, the test results include: (information of a set) [SNPs name, W-value, k, p-value]; (Information of the first variable in the set) [W-value, k, p-value]; (Information of the second variable in the set) [W-value, k, p-value] ...</p>
</td></tr>
<tr><td><code>hf1</code></td>
<td>
<p>The <em>h</em> and <em>f</em> values used in main effect calculation.</p>
</td></tr>
<tr><td><code>hf2</code></td>
<td>
<p>The <em>h</em> and <em>f</em> values used in high-order interaction calculation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rui Sun, Maggie Haitian Wang
</p>


<h3>References</h3>

<p>Maggie Haitian Wang, Rui Sun, Junfeng Guo, Haoyi Weng, Jack Lee, Inchi Hu, Pak Sham and Benny C.Y. Zee (2016). A fast and powerful W-test for pairwise epistasis testing. Nucleic Acids Research. doi:10.1093/nar/gkw347.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+hf">hf</a></code>, <code><a href="#topic+w.diagnosis">w.diagnosis</a></code>, <code><a href="#topic+w.qqplot">w.qqplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(diabetes.geno)
data(phenotype1)

## Step 1. HF Calculation
# Please note that parameter B is recommended to be greater than 400 for w.order = 1 or 2.
# For high order interaction analysis (w.order &gt; 2), it is recommended to use default n.sample.
hf1 &lt;- hf(data = diabetes.geno, w.order = 1, B = 100)
hf.high &lt;- hf(data = diabetes.geno, w.order = 3, B = 30, n.marker = 10)

## Step 2. W-test Calculation
w1 &lt;- wtest.high(diabetes.geno, phenotype1, w.order = 1, hf1 = hf1)
w3 &lt;- wtest.high(diabetes.geno, phenotype1, w.order = 3, input.pval = 0.3,
            input.poolsize = 50, output.pval = 0.5, hf1 = hf1, hf.high.order = hf.high)
w.set &lt;- wtest.high(diabetes.geno, phenotype1, w.order = 3, which.marker = c(10,13,20),
            hf.high.order = hf.high)
</code></pre>

<hr>
<h2 id='wtest.snps.meth'>W-test for Gene-methylation Interaction Analysis</h2><span id='topic+wtest.snps.meth'></span>

<h3>Description</h3>

<p>Calculate cis-gene-methylation interaction of a (SNP, CpG) pair in user-defined window, and can run in a genome-wide manner. The output can be filtered by p-values, such that only sets with smaller
p-value than the threshold (<code>output.pval</code>) will be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wtest.snps.meth(geno, meth, y, geno.pos, meth.pos, window.size = 10000,
  hf = "default.hf", output.pval = NULL, sort = TRUE,
  which.marker = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wtest.snps.meth_+3A_geno">geno</code></td>
<td>
<p>a data frame or matrix containing genotypes in the columns and subjects in the rows. Genotypes should be coded as (0, 1, 2) or (0, 1). SNP names should be stored as column names of the data.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_meth">meth</code></td>
<td>
<p>a data frame or matrix containing methylation data in the columns. Methylation data should be recoded as (0, 1, 2) or (0, 1). Names of CpG sites should be stored as column names of the data.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_y">y</code></td>
<td>
<p>a numeric vector of 0 or 1.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_geno.pos">geno.pos</code></td>
<td>
<p>a data frame containing SNP names and positions in two columns.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_meth.pos">meth.pos</code></td>
<td>
<p>a data frame containing CpG names and positions in two columns.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_window.size">window.size</code></td>
<td>
<p>a numeric number specifying the size of genome distance. Interaction effects of the SNPs and CpG sites located within the size of genome distance will be evaluated exhaustively.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_hf">hf</code></td>
<td>
<p><em>h</em> and <em>f</em> values to calculate gene-methylation interaction associations, organized as a matrix, with columns (<em>k</em>, <em>h</em>, <em>f</em>), <em>k</em> = 2 to 6.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_output.pval">output.pval</code></td>
<td>
<p>a p-value threshold for filtering the output. If NULL, all results will be listed; otherwise, the function will only output the results with p-values smaller than <code>output.pval</code>.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_sort">sort</code></td>
<td>
<p>a logical value indicating whether or not to sort the output by p-values in ascending order. Default = TRUE.</p>
</td></tr>
<tr><td><code id="wtest.snps.meth_+3A_which.marker">which.marker</code></td>
<td>
<p>a vector indicating the column index of a SNP-CpG pair to calculate. Default <code>which.marker</code> = NULL means interaction pairs located within <code>window.size</code> will be calculated exhaustively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculate cis-gene-methylation interaction of a (SNP, CpG) pair in user-defined window, and can run in a genome-wide manner. The output can be filtered by p-values, such that only sets with smaller
p-value than the threshold (<code>output.pval</code>) will be returned.
</p>


<h3>Value</h3>

<p>An object <code>"wtest.snps.meth"</code> containing:
</p>
<table role = "presentation">
<tr><td><code>results</code></td>
<td>
<p>The test results include: SNP name, CpG name, SNP position, CpG position, W value, <em>k</em>, and p-value.</p>
</td></tr>
<tr><td><code>hf</code></td>
<td>
<p>The <em>h</em> and <em>f</em> values used for each <em>k</em> in pairwise calculation, where <em>k</em> = 2 to 6.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rui Sun, Maggie Haitian Wang
</p>


<h3>References</h3>

<p>Maggie Haitian Wang, Rui Sun, Junfeng Guo, Haoyi Weng, Jack Lee, Inchi Hu, Pak Sham and Benny C.Y. Zee (2016). A fast and powerful W-test for pairwise epistasis testing. Nucleic Acids Research. doi:10.1093/nar/gkw347.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+wtest">wtest</a></code>, <code><a href="#topic+hf.snps.meth">hf.snps.meth</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNP.pos)
data(CpG.pos)
data(genotype)
data(methylation)
data(phenotype2)

w &lt;- 13000

# Recode methylation data
methylation &lt;- methylation.recode(methylation)

## Step 1. HF Calculation.
# Please note that parameter B is recommended to be greater than 400.
hf.pair &lt;- hf.snps.meth(B = 80, geno = genotype, meth = methylation, y = phenotype2,
                        geno.pos = SNP.pos, meth.pos = CpG.pos, window.size = w)

## Step 2. Application
result &lt;- wtest.snps.meth(geno = genotype, meth = methylation, y = phenotype2, geno.pos = SNP.pos,
                          meth.pos = CpG.pos, window.size = w, hf = hf.pair, output.pval = 0.1)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
