<!DOCTYPE html><html lang="en"><head><title>Help for package wTO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {wTO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CorrelationOverlap'><p>CorrelationOverlap</p></a></li>
<li><a href='#Cut.off'><p>Cut.off</p></a></li>
<li><a href='#ExampleGRF'><p>ExampleGRF</p></a></li>
<li><a href='#metagenomics_abundance'><p>metagenomics_abundance</p></a></li>
<li><a href='#Microarray_Expression1'><p>Microarray_Expression1</p></a></li>
<li><a href='#Microarray_Expression2'><p>Microarray_Expression2</p></a></li>
<li><a href='#NetVis'><p>NetVis</p></a></li>
<li><a href='#rmcor'><p>wTO.aux.each</p></a></li>
<li><a href='#sample_ind'><p>sample_ind</p></a></li>
<li><a href='#wTO'><p>wTO</p></a></li>
<li><a href='#wTO.aux.each'><p>wTO.aux.each</p></a></li>
<li><a href='#wTO.Complete'><p>wTO.Complete</p></a></li>
<li><a href='#wTO.Consensus'><p>wTO.Consensus</p></a></li>
<li><a href='#wTO.export'><p>wTO.export</p></a></li>
<li><a href='#wTO.fast'><p>wTO.fast</p></a></li>
<li><a href='#wTO.in.line'><p>wTO.in.line</p></a></li>
<li><a href='#wTO.rep_measure'><p>wTO.rep_measure</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Computing Weighted Topological Overlaps (wTO) &amp; Consensus wTO
Network</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Deisy Morselli Gysi, Andre Voigt, Tiago Miranda Fragoso, Eivind Almaas and Katja Nowick.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Deisy Morselli Gysi &lt;deisy.ccnr@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Computes the Weighted Topological Overlap with positive and negative signs (wTO) networks given a data frame containing the mRNA count/ expression/ abundance per sample, and a vector containing the interested nodes of interaction (a subset of the elements of the full data frame). It also computes the cut-off threshold or p-value based on the individuals bootstrap or the values reshuffle per individual. It also allows the construction of a consensus network, based on multiple wTO networks. The package includes a visualization tool for the networks.  More about the methodology can be found at &lt;<a href="https://doi.org/10.1186%2Fs12859-018-2351-7">doi:10.1186/s12859-018-2351-7</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, igraph, magrittr, plyr, parallel, som, visNetwork,
reshape2, Rfast, HiClimR, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-04 19:54:14 UTC; deisygysi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-04 22:40:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='CorrelationOverlap'>CorrelationOverlap</h2><span id='topic+CorrelationOverlap'></span>

<h3>Description</h3>

<p>This function computes the correlation between Nodes and the Overlapping Nodes of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CorrelationOverlap(Data, Overlap, method)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CorrelationOverlap_+3A_data">Data</code></td>
<td>
<p>data.frame containing the expression data. Nodes on the Rows, Individuals on the Columns. Don't forget to give the names to the Nodes and to the Individuals. Nodes must have the row.names() with the Node Name.</p>
</td></tr>
<tr><td><code id="CorrelationOverlap_+3A_overlap">Overlap</code></td>
<td>
<p>A vector containg the names of the Nodes of interest.</p>
</td></tr>
<tr><td><code id="CorrelationOverlap_+3A_method">method</code></td>
<td>
<p>Spearman (&quot;s&quot;, &quot;spearman&quot;) or Pearson (&quot;p&quot;, &quot;pearson&quot;) correlation</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>

<hr>
<h2 id='Cut.off'>Cut.off</h2><span id='topic+Cut.off'></span>

<h3>Description</h3>

<p>Computes the cutoff threshold using the descriptive values generated by the bootstrap, also plots the empirical distribution and the sample wTO.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cut.off(wTO_value, type, plot)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Cut.off_+3A_wto_value">wTO_value</code></td>
<td>
<p>is the table returned by the wTO with all the simulated values.</p>
</td></tr>
<tr><td><code id="Cut.off_+3A_type">type</code></td>
<td>
<p>type of plot</p>
</td></tr>
<tr><td><code id="Cut.off_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether to plot the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plots in a new device the cutoff value, and the amount of links in each one of the bands.
</p>

<hr>
<h2 id='ExampleGRF'>ExampleGRF</h2><span id='topic+ExampleGRF'></span>

<h3>Description</h3>

<p>ExampleGRF data.frame containing data.frame containing names of GRFs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ExampleGRF)
</code></pre>


<h3>Format</h3>

<p>data.frame 184 lines, 1 column.
</p>

<hr>
<h2 id='metagenomics_abundance'>metagenomics_abundance</h2><span id='topic+metagenomics_abundance'></span>

<h3>Description</h3>

<p>metagenomics_abundance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data('metagenomics_abundance')
</code></pre>


<h3>Format</h3>

<p>data.frame from The USC Microbial Observatory. The data is public available at &lt;https://www.ebi.ac.uk/metagenomics/projects/ERP013549&gt;
</p>

<hr>
<h2 id='Microarray_Expression1'>Microarray_Expression1</h2><span id='topic+Microarray_Expression1'></span>

<h3>Description</h3>

<p>Microarray_Expression1 data.frame containing expression data for 1000 genes and 18 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Microarray_Expression1
</code></pre>


<h3>Format</h3>

<p>data.frame 1000 lines, 18 columns.
</p>

<hr>
<h2 id='Microarray_Expression2'>Microarray_Expression2</h2><span id='topic+Microarray_Expression2'></span>

<h3>Description</h3>

<p>Microarray_Expression2 data.frame containing expression data for 1000 genes and 18 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Microarray_Expression2
</code></pre>


<h3>Format</h3>

<p>data.frame 1000 lines, 18 columns.
</p>

<hr>
<h2 id='NetVis'>NetVis</h2><span id='topic+NetVis'></span>

<h3>Description</h3>

<p>Given a set of Nodes and the weight of the edges, a cutoff for the edges, it draws the networks. Returns a list with the nodes and edges  attributes. And plots the network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NetVis(
  Node.1,
  Node.2,
  wTO,
  pval = NULL,
  MakeGroups = FALSE,
  padj = NULL,
  cutoff = list(kind = "Threshold", value = 0.5),
  layout = NULL,
  smooth.edges = T,
  path = NULL,
  Cluster = F,
  legend = T,
  shape = list(shape = "triangle", names = NULL),
  manipulation = F
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NetVis_+3A_node.1">Node.1</code></td>
<td>
<p>Names of the Nodes.1 that are connected to the Nodes.2. It's the output from wTO.Complete or Consensus.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_node.2">Node.2</code></td>
<td>
<p>Names of the Nodes.2 that are connected to the Nodes.1. It's the output from wTO.Complete or Consensus.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_wto">wTO</code></td>
<td>
<p>weight of the links, the wTO output from wTO.Complete or wTO.Consensus.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_pval">pval</code></td>
<td>
<p>p-values for the wTO value. By default it is NULL.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_makegroups">MakeGroups</code></td>
<td>
<p>algorithm to find clusters. One of the followings: walktrap, optimal, spinglass, edge.betweenness, fast_greedy, infomap, louvain, label_prop, leading_eigen. Default to FALSE.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_padj">padj</code></td>
<td>
<p>Adjusted p-values for the wTO value. By default it is NULL.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_cutoff">cutoff</code></td>
<td>
<p>It's a list containing the kind of cutoff to be used (pval, Threshold or pval.adj)and it's value. Example: cutoff= list(kind = &quot;Threshold&quot;, value = 0.5)</p>
</td></tr>
<tr><td><code id="NetVis_+3A_layout">layout</code></td>
<td>
<p>a layout from the igraph package.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_smooth.edges">smooth.edges</code></td>
<td>
<p>If the edges should be smoothed or not.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_path">path</code></td>
<td>
<p>If the graph should be saved specify the name of the file.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_cluster">Cluster</code></td>
<td>
<p>TRUE or FALSE if the nodes should be clustered (double click to uncluster).</p>
</td></tr>
<tr><td><code id="NetVis_+3A_legend">legend</code></td>
<td>
<p>TRUE or FALSE if the legend should appear.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_shape">shape</code></td>
<td>
<p>a list shape=list(shape = &quot;triangle&quot;, names = NULL), with the shape and the IDs that should have a different shape, shape can be: diamond, star, triangle, triangleDown or square.</p>
</td></tr>
<tr><td><code id="NetVis_+3A_manipulation">manipulation</code></td>
<td>
<p>TRUE or FALSE if the graph should be editable.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
 X =  wTO.Complete( k =1, n = 5, Data = Microarray_Expression1,
 Overlap = ExampleGRF$x[1:10], method = "p",  plot = FALSE)
# Plot with the default aguments.
 NetVis(Node.1 = X$wTO$Node.1, Node.2 = X$wTO$Node.2,
 wTO = X$wTO$wTO_sign, cutoff = list(kind =
"Threshold", value = 0.50))

# Plotting just the edges with p-value &lt; 0.05, with straight edges, nodes clustered,
# no legend and mapipulation of the graph enabled.
  NetVis(Node.1 = X$wTO$Node.1, Node.2 = X$wTO$Node.2,
 wTO = X$wTO$wTO_sign, pval = X$wTO$pval_sign,
 padj = X$wTO$pval_sign,
  cutoff= list(kind = "pval", value = 0.05),
  smooth.edges = FALSE,
 Cluster = TRUE, legend = FALSE, manipulation = TRUE)
# Plotting just the edges with wTO &gt; 0.50, no legend and the nodes:
# "ZNF738", "ZNF677" with triagle shape,
# no legend and mapipulation of the graph enabled.
 NetVis(Node.1 = X$wTO$Node.1, Node.2 = X$wTO$Node.2,
 wTO = X$wTO$wTO_sign, pval = X$wTO$pval_sign,
 padj = X$wTO$pval_sign, cutoff= list(kind = "Threshold", value = 0.5),legend = FALSE,
 shape = list(shape = "triangle", names = c("ZNF738", "ZNF677")))

 
## End(Not run)
</code></pre>

<hr>
<h2 id='rmcor'>wTO.aux.each</h2><span id='topic+rmcor'></span>

<h3>Description</h3>

<p>Computes the repeated measures correlation. based on Blendman Altman (1995). Implemented by &lt;https://doi.org/10.3389/fpsyg.2017.00456&gt;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmcor(ID, Measure1, Measure2)
</code></pre>

<hr>
<h2 id='sample_ind'>sample_ind</h2><span id='topic+sample_ind'></span>

<h3>Description</h3>

<p>Computes the resufling of the expression values for the IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_ind(x, dfExpression)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_ind_+3A_x">x</code></td>
<td>
<p>Column to be resampled</p>
</td></tr>
<tr><td><code id="sample_ind_+3A_dfexpression">dfExpression</code></td>
<td>
<p>data.frame object containing the genes expression on the rows and the individuals (Individuals in the Columns)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>

<hr>
<h2 id='wTO'>wTO</h2><span id='topic+wTO'></span>

<h3>Description</h3>

<p>Calculates the weighted topologycal overlap (wTO)
between a set of Nodes and the Overlapping nodes. This function implements the method from Nowick (2009).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO(A_TF, sign = c("abs", "sign"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO_+3A_a_tf">A_TF</code></td>
<td>
<p>Is the weighted adjency matrix (correlation matrix).</p>
</td></tr>
<tr><td><code id="wTO_+3A_sign">sign</code></td>
<td>
<p>(&quot;abs&quot;, &quot;sign&quot;) if the user wants to use the absolute correlation or the signed correlation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing the wTO values.
</p>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>


<h3>References</h3>

<p>Katja Nowick, Tim Gernat, Eivind Almaas and Lisa Stubbs (2009) &lt;doi:10.1073/pnas.0911376106&gt;
</p>

<hr>
<h2 id='wTO.aux.each'>wTO.aux.each</h2><span id='topic+wTO.aux.each'></span>

<h3>Description</h3>

<p>wTO.aux.each calculate the wTO for each one of the resamplings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.aux.each(n, Data, Overlap, method, method_resampling, lag, ID)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.aux.each_+3A_n">n</code></td>
<td>
<p>Number of bootstraps / reshuffles to be run for the estimates of the &quot;Threshold&quot; or &quot;pval&quot;.</p>
</td></tr>
<tr><td><code id="wTO.aux.each_+3A_data">Data</code></td>
<td>
<p>data.frame containing the count / expression data for the correlation.</p>
</td></tr>
<tr><td><code id="wTO.aux.each_+3A_overlap">Overlap</code></td>
<td>
<p>Nodes of interested, where the Overlapping weights will be computed.</p>
</td></tr>
<tr><td><code id="wTO.aux.each_+3A_method">method</code></td>
<td>
<p>Type of the correlation that should be used. &quot;s&quot; / &quot;spearman&quot; will compute the rank spearman correlation, &quot;p&quot; / &quot;pearson&quot; will compute the linear correlation. If no value is given, the default is to use &quot;s&quot;.</p>
</td></tr>
<tr><td><code id="wTO.aux.each_+3A_method_resampling">method_resampling</code></td>
<td>
<p>method of the resampling. Bootstrap or Reshuffle. Bootstrap null hypothesis is that the wTO is random, and Reshuffle tests if the wTO is equal to zero.</p>
</td></tr>
</table>

<hr>
<h2 id='wTO.Complete'>wTO.Complete</h2><span id='topic+wTO.Complete'></span>

<h3>Description</h3>

<p>Compute the wTO and also the bootstraps. Proposed at: 	arXiv:1711.04702
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.Complete(
  k = 1,
  n = 100,
  Data,
  Overlap = row.names(Data),
  method = "p",
  method_resampling = "Bootstrap",
  pvalmethod = "BH",
  savecor = F,
  expected.diff = 0.2,
  lag = NULL,
  ID = NULL,
  normalize = F,
  plot = T
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.Complete_+3A_k">k</code></td>
<td>
<p>Number of threads to be used for computing the weight Topological Overlap. Default is set to 1.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_n">n</code></td>
<td>
<p>Number of resamplings, used to compute the empirical distribuitions of the links. Default is set to 100.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_data">Data</code></td>
<td>
<p>data.frame containing the count / expression data for the correlation.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_overlap">Overlap</code></td>
<td>
<p>Set of nodes of interest, where the Overlapping weights will be computed.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_method">method</code></td>
<td>
<p>Type of the correlation that should be used. &quot;s&quot; / &quot;spearman&quot; will compute the rank spearman correlation, &quot;p&quot; / &quot;pearson&quot; will compute the linear correlation. If no value is given, the default is to use &quot;p&quot;.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_method_resampling">method_resampling</code></td>
<td>
<p>method of the resampling. Bootstrap, BlockBootstrap or Reshuffle. Bootstrap null hypothesis is that the wTO is random, and Reshuffle tests if the wTO is equal to zero.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_pvalmethod">pvalmethod</code></td>
<td>
<p>method to compute the multiple test correction for the pvalue. for more information check the function <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_savecor">savecor</code></td>
<td>
<p>T/F if need to save the correlation.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_expected.diff">expected.diff</code></td>
<td>
<p>Difference expected between the real wTO and resampled wTO By default, it is set to 0.2.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_lag">lag</code></td>
<td>
<p>time dependency, lag, if you are using the BlockedBootstrap.</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_id">ID</code></td>
<td>
<p>ID of the samples for the blocked bootstrap (for repeated measures).</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_normalize">normalize</code></td>
<td>
<p>T/F Should the data be normalized?</p>
</td></tr>
<tr><td><code id="wTO.Complete_+3A_plot">plot</code></td>
<td>
<p>T/F Should the diagnosis plot be plotted?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with results.
</p>

<ul>
<li><p> wTO is a data.frame containig the Nodes, the wTO computed using the signed correlations, the pvalue and the adj.pvalue.
</p>
</li>
<li><p> abs.wTO is a data.frame containig the Nodes, the wTO computed using the absolute correlations, the pvalue and the adj.pvalue.
</p>
</li>
<li><p> Correlation is a data.frame containing the correlation between all the nodes.
</p>
</li>
<li><p> Empirical.Quantile quantile values for the empirical distribution.
</p>
</li>
<li><p> Quantile quantile values for the sample distribution.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Using spearman rank correlation and bonferroni correction for the pvalues.
wTO.Complete( k =8, n = 1000, Data = Microarray_Expression1,
 Overlap = ExampleGRF$x, method = "s", pvalmethod = "bonferroni")
 # Changing the resampling method to Reshuffle.
wTO.Complete( k =1, n = 1000, Data = Microarray_Expression1,
Overlap = ExampleGRF$x, method_resampling = "Reshuffle")
 # Changing the resampling method to BlockBootstrap, with a lag of 2.
 row.names(metagenomics_abundance) = metagenomics_abundance$OTU
 metagenomics_abundance = metagenomics_abundance[,-1]
wTO.Complete( k =1, n = 1000, Data = metagenomics_abundance, method = "s",
Overlap = row.names(metagenomics_abundance), method_resampling = "BlockBootstrap", lag = 2)
wTO.Complete( k =2, n = 1000, Data = Microarray_Expression1, method = "s",
Overlap = ExampleGRF$x, method_resampling = "BlockBootstrap", ID = rep(1:9,each = 2))
X = wTO.Complete( k =1, n = 1000, Data = Microarray_Expression1,
Overlap = ExampleGRF$x, method = "p", plot = FALSE)
 
## End(Not run)
</code></pre>

<hr>
<h2 id='wTO.Consensus'>wTO.Consensus</h2><span id='topic+wTO.Consensus'></span>

<h3>Description</h3>

<p>Consensus requires a list of data.frame containing the pair of nodes, and the wTO values for all networks that need to be joined. Reference: 	arXiv:1711.04702
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.Consensus(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.Consensus_+3A_data">data</code></td>
<td>
<p>list of data.frame containing the &quot;Node.1&quot;, &quot;Node.2&quot; and &quot;wTO&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
EXAMPLE =  wTO.Complete( k =1, n = 200, Data = Microarray_Expression1,
                                      Overlap = ExampleGRF$x, method = "p")

# Constructing the consensus network
data = list(data.frame(Node.1 = EXAMPLE$wTO$Node.1,
Node.2 = EXAMPLE$wTO$Node.2,
wto_sig = EXAMPLE$wTO$wTO_sign,
pvalsig = EXAMPLE$wTO$pval_sig),
data.frame(Node.1 = EXAMPLE$wTO$Node.1,
Node.2 = EXAMPLE$wTO$Node.2,
wtoabs = EXAMPLE$wTO$wTO_abs,
pvalabs = EXAMPLE$wTO$pval_abs) )
CONS = wTO.Consensus(data)


## End(Not run)
</code></pre>

<hr>
<h2 id='wTO.export'>wTO.export</h2><span id='topic+wTO.export'></span>

<h3>Description</h3>

<p>Exports the significative interactions, the wTO weight and pvalues into a .txt file, tab separeted. This file can be imported in other visualization tools (Cytoscape for example).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.export(DATA, path, sign = TRUE, pvalue = 0.05, padj = 0.05, prop.NA = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.export_+3A_data">DATA</code></td>
<td>
<p>Output from the function wTO.Complete or wTO.Consensus.</p>
</td></tr>
<tr><td><code id="wTO.export_+3A_path">path</code></td>
<td>
<p>Path and file name where the .txt file should be saved.</p>
</td></tr>
<tr><td><code id="wTO.export_+3A_sign">sign</code></td>
<td>
<p>Should the network contain the results for the signed network or unsigned? Only for data coming from wTO.Complete.</p>
</td></tr>
<tr><td><code id="wTO.export_+3A_pvalue">pvalue</code></td>
<td>
<p>cutoff p-value for the network. Only for data coming from wTO.Complete.</p>
</td></tr>
<tr><td><code id="wTO.export_+3A_padj">padj</code></td>
<td>
<p>cutoff adjusted p-value for the network. Only for data coming from wTO.Complete.</p>
</td></tr>
<tr><td><code id="wTO.export_+3A_prop.na">prop.NA</code></td>
<td>
<p>cutoff proportion of NAs for the network. Only for data coming from wTO.Consensus.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
EXAMPLE =  wTO.Complete( k =1, n = 200, Data = Microarray_Expression2,
                                      Overlap = ExampleGRF$x, method = "p")
wTO.export(EXAMPLE , './EXAMPLE.txt')

#Selection of only the significative ones for the Consensus
Ex_k1_cor_p_boot_p005_sig = subset(EXAMPLE$wTO,
EXAMPLE$wTO$pval_sig &lt; 0.05,
select = c("Node.1", "Node.2", "wTO_sign"))
Ex_k1_cor_p_boot_p005_abs = subset(EXAMPLE$wTO,
EXAMPLE$wTO$pval_abs &lt; 0.05,
select = c("Node.1", "Node.2", "wTO_abs"))
# Constructing the consensus network
CN = wTO.Consensus(data = list(Ex_k1_cor_p_boot_p005_sig,
Ex_k1_cor_p_boot_p005_abs))
wTO.export(CN, './CN.txt')
### You can store the result on the workspace.
y = wTO.export(CN, './CN.txt')
head(y)

## End(Not run)



</code></pre>

<hr>
<h2 id='wTO.fast'>wTO.fast</h2><span id='topic+wTO.fast'></span>

<h3>Description</h3>

<p>Compute the wTO and also the bootstraps. Proposed at arXiv:1711.04702. This is a quicker version of the wTO.Complete. It doesn't contain diagnose plots nor a parallel version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.fast(
  Data,
  Overlap = row.names(Data),
  method = "p",
  sign = "sign",
  delta = 0.2,
  n = 10,
  method_resampling = "Bootstrap",
  lag = NULL,
  ID = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.fast_+3A_data">Data</code></td>
<td>
<p>data.frame containing the count / expression data for the correlation.</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_overlap">Overlap</code></td>
<td>
<p>Set of nodes of interest, where the Overlapping weights will be computed.</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_method">method</code></td>
<td>
<p>Type of the correlation that should be used. &quot;s&quot; / &quot;spearman&quot; will compute the rank spearman correlation, &quot;p&quot; / &quot;pearson&quot; will compute the linear correlation. If no value is given, the default is to use &quot;p&quot;.</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_sign">sign</code></td>
<td>
<p>Should the wTO be signed?</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_delta">delta</code></td>
<td>
<p>expected difference between the real wTO and the bootstraped.</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_n">n</code></td>
<td>
<p>Number of resamplings, used to compute the empirical distribuitions of the links. Default is set to 100.</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_method_resampling">method_resampling</code></td>
<td>
<p>method of the resampling. Bootstrap or BlockBootstrap.If the second is used, please give the lag (time dependency among the data).</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_lag">lag</code></td>
<td>
<p>Time dependency for the blocked bootstrap (for time series).</p>
</td></tr>
<tr><td><code id="wTO.fast_+3A_id">ID</code></td>
<td>
<p>ID of the samples for the blocked bootstrap (for repeated measures).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'> wTO.fast(Data = Microarray_Expression1,
 Overlap = ExampleGRF$x, 
 method = "p")

#  For a time series with lag = 4
# wTO.fast(Data = Microarray_Expression1,
# Overlap = ExampleGRF$x, 
# method = "p", 
# method_resampling = 'BlockBootstrap', 
# lag = 4)

# For a study where the individuals were measured multiple times.
# wTO.fast(Data = Microarray_Expression1,
# Overlap = ExampleGRF$x, 
# method = "p", 
# method_resampling = 'BlockBootstrap', 
# ID = rep(1:9, each= 2))
</code></pre>

<hr>
<h2 id='wTO.in.line'>wTO.in.line</h2><span id='topic+wTO.in.line'></span>

<h3>Description</h3>

<p>Transforms a correlation matrix into the line format.
</p>
<p>Transforms a correlation matrix into the line format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.in.line(d)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.in.line_+3A_d">d</code></td>
<td>
<p>correlation matrix to be converted into the line format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the wTO matrix into a data.frame: Node1, Node2 and wTO.
</p>
<p>the wTO matrix into a data.frame: Node1, Node2 and wTO.
</p>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>

<hr>
<h2 id='wTO.rep_measure'>wTO.rep_measure</h2><span id='topic+wTO.rep_measure'></span>

<h3>Description</h3>

<p>Compute the wTO for a repeated measures expermiment and also the bootstraps. Proposed at arXiv:1711.04702. This is a quicker version of the wTO.Complete. It doesn'T contain diagnose plots nor a parallel version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wTO.rep_measure(
  Data,
  Overlap = row.names(Data),
  ID,
  sign = "sign",
  delta = 0.2,
  n = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wTO.rep_measure_+3A_data">Data</code></td>
<td>
<p>data.frame containing the count / expression data for the correlation.</p>
</td></tr>
<tr><td><code id="wTO.rep_measure_+3A_overlap">Overlap</code></td>
<td>
<p>Set of nodes of interest, where the Overlapping weights will be computed.</p>
</td></tr>
<tr><td><code id="wTO.rep_measure_+3A_id">ID</code></td>
<td>
<p>a vector with the individuals identification</p>
</td></tr>
<tr><td><code id="wTO.rep_measure_+3A_sign">sign</code></td>
<td>
<p>Should the wTO be signed?</p>
</td></tr>
<tr><td><code id="wTO.rep_measure_+3A_delta">delta</code></td>
<td>
<p>expected difference between the real wTO and the bootstraped.</p>
</td></tr>
<tr><td><code id="wTO.rep_measure_+3A_n">n</code></td>
<td>
<p>Number of resamplings, used to compute the empirical distribuitions of the links. Default is set to 100.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Deisy Morselli Gysi &lt;deisy at bioinf.uni-leipzig.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#wTO.rep_measure(Data = Microarray_Expression1, ID = rep(c(1:9),2), 
#Overlap = ExampleGRF$x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
