<!DOCTYPE html><html><head><title>Help for package xegaPermGene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {xegaPermGene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Decay'><p>Exponential decay.</p></a></li>
<li><a href='#lFxegaPermGene'><p>Generate local functions and objects.</p></a></li>
<li><a href='#without'><p>Returns elements of</p>
vector <code>x</code> without elements in <code>y</code>.</a></li>
<li><a href='#xegaPermCross2Gene'><p>Position based crossover of 2 genes.</p></a></li>
<li><a href='#xegaPermCrossGene'><p>Position based crossover of 2 genes.</p></a></li>
<li><a href='#xegaPermCrossoverFactory'><p>Configure the crossover function of a genetic algorithm.</p></a></li>
<li><a href='#xegaPermDecodeGene'><p>Decode a permutation.</p></a></li>
<li><a href='#xegaPermGene'><p>Package xegaPermGene.</p></a></li>
<li><a href='#xegaPermInitGene'><p>Initialize a gene with a permutation of integers</p></a></li>
<li><a href='#xegaPermMutateGene2Opt'><p>Mutate a gene (by a random 2-Opt move).</p></a></li>
<li><a href='#xegaPermMutateGeneBestGreedy'><p>Mutate a gene (by inserting a greedy path at start of random length k).</p></a></li>
<li><a href='#xegaPermMutateGeneGreedy'><p>Mutate a gene (by inserting a greedy path at start of random length k).</p></a></li>
<li><a href='#xegaPermMutateGenekInversion'><p>Mutate a gene (k random inversions).</p></a></li>
<li><a href='#xegaPermMutateGenekOptLK'><p>Mutate a gene (by a random Lin-Kernighan k-OPT move).</p></a></li>
<li><a href='#xegaPermMutateGeneOrderBased'><p>Mutate a gene (generalized order based mutation).</p></a></li>
<li><a href='#xegaPermMutateMix'><p>Mutation by a random mutation function.</p></a></li>
<li><a href='#xegaPermMutationFactory'><p>Configure the mutation function of a genetic algorithm.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Operations on Permutation Genes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Andreas Geyer-Schulz &lt;Andreas.Geyer-Schulz@kit.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of 
        representation-dependent gene level operations for
        genetic algorithms with genes which represent permutations: 
        initialization of genes, mutation and crossover.
        The crossover operation provided is position-based crossover 
        (Syswerda, G., Chap. 21 in Davis, L. (1991, ISBN:0-442-00173-8). 
        For mutation, several variants are included: Order-based mutation
        (Syswerda, G., Chap. 21 in Davis, L. (1991, ISBN:0-442-00173-8), 
        randomized Lin-Kernighan heuristics
        (Croes, G. A. (1958) &lt;<a href="https://doi.org/10.1287%2Fopre.6.6.791">doi:10.1287/opre.6.6.791</a>&gt; and
         Lin, S. and Kernighan. B. W. (1973) 
        &lt;<a href="https://doi.org/10.1287%2Fopre.21.2.498">doi:10.1287/opre.21.2.498</a>&gt;), 
        and randomized greedy operators.
        A random mix operator for mutation selects a mutation variant 
        randomly.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/ageyerschulz/xegaPermGene&amp;gt;">https://github.com/ageyerschulz/xegaPermGene&gt;</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>xegaSelectGene</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-04 21:21:12 UTC; dj2333</td>
</tr>
<tr>
<td>Author:</td>
<td>Andreas Geyer-Schulz
    <a href="https://orcid.org/0009-0000-5237-3579"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-05 21:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Decay'>Exponential decay.</h2><span id='topic+Decay'></span>

<h3>Description</h3>

<p>Exponential decay.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Decay(t, lambda = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Decay_+3A_t">t</code></td>
<td>
<p>Number of objects.</p>
</td></tr>
<tr><td><code id="Decay_+3A_lambda">lambda</code></td>
<td>
<p>Exponential decay constant.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector with t elements with values of exponential decay.
</p>


<h3>See Also</h3>

<p>Other Utility: 
<code><a href="#topic+without">without</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Decay(5, 0.4)
Decay(10, 0.4)

</code></pre>

<hr>
<h2 id='lFxegaPermGene'>Generate local functions and objects.</h2><span id='topic+lFxegaPermGene'></span>

<h3>Description</h3>

<p><code>lFxegaPermGene</code> is a list of functions 
which contains a definition of all local objects 
required for the use of genetic operators with the 
We refer to this object as local configuration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lFxegaPermGene
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 21.
</p>


<h3>Details</h3>

<p>We use the local function list (the local configuration) for 
</p>

<ol>
<li>
<p>replacing all constants by constant functions.
</p>
<p>Rationale: We need one formal argument (the local function list lF)
and we can dispatch multiple functions. E.g.  <code>lF$verbose()</code>
</p>
</li>
<li>    
<p>for dynamically binding a local function with a definition from a
proper function factory. E.g. the selection methods 
<code>lf$SelectGene</code> and <code>SelectMate</code>.
</p>
</li>
<li><p> for gene representation specific special functions:
<code>lf$InitGene</code>, <code>lF$DecodeGene</code>, <code>lf$EvalGene</code>
<code>lf$ReplicateGene</code>, ...
</p>
</li></ol>



<h3>See Also</h3>

<p>Other Configuration: 
<code><a href="#topic+xegaPermCrossoverFactory">xegaPermCrossoverFactory</a>()</code>,
<code><a href="#topic+xegaPermMutationFactory">xegaPermMutationFactory</a>()</code>
</p>

<hr>
<h2 id='without'>Returns elements of 
vector <code>x</code> without elements in <code>y</code>.</h2><span id='topic+without'></span>

<h3>Description</h3>

<p>Returns elements of 
vector <code>x</code> without elements in <code>y</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>without(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="without_+3A_x">x</code></td>
<td>
<p>Vector.</p>
</td></tr>
<tr><td><code id="without_+3A_y">y</code></td>
<td>
<p>Vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector.
</p>


<h3>See Also</h3>

<p>Other Utility: 
<code><a href="#topic+Decay">Decay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a&lt;-sample(1:15,15, replace=FALSE)
b&lt;-c(1, 3, 5)
without(a, b)
</code></pre>

<hr>
<h2 id='xegaPermCross2Gene'>Position based crossover of 2 genes.</h2><span id='topic+xegaPermCross2Gene'></span>

<h3>Description</h3>

<p><code>xegaPermCross2Gene</code> determines 
a random subschedule of random length.
</p>
<p>It copies the random subschedule into a new gene.
The rest of the positions of the new scheme is filled 
with the elements of the other gene to complete the
permutation. This is done for each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermCross2Gene(gg1, gg2, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermCross2Gene_+3A_gg1">gg1</code></td>
<td>
<p>Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermCross2Gene_+3A_gg2">gg2</code></td>
<td>
<p>Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermCross2Gene_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of 2 permutations.
</p>


<h3>References</h3>

<p>Syswerda, G. (1991): 
Schedule Optimization Using Genetic Algorithms.
In: Davis, L. (Ed.): 
Handbook of Genetic Algorithms, Chapter 21, p. 343.
Van Nostrand Reinhold, New York.
(ISBN:0-442-00173-8)
</p>


<h3>See Also</h3>

<p>Other Crossover: 
<code><a href="#topic+xegaPermCrossGene">xegaPermCrossGene</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
gene2&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene2, lFxegaPermGene)
newgenes&lt;-xegaPermCross2Gene(gene1, gene2)
xegaPermDecodeGene(newgenes[[1]], lFxegaPermGene)
xegaPermDecodeGene(newgenes[[2]], lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermCrossGene'>Position based crossover of 2 genes.</h2><span id='topic+xegaPermCrossGene'></span>

<h3>Description</h3>

<p><code>xegaPermCrossGene</code> determines 
a random subschedule of random length.
</p>
<p>It copies the random subschedule into a new gene.
The rest of the positions of the new scheme is filled 
with the elements of the other gene to complete the
permutation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermCrossGene(gg1, gg2, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermCrossGene_+3A_gg1">gg1</code></td>
<td>
<p>Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermCrossGene_+3A_gg2">gg2</code></td>
<td>
<p>Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermCrossGene_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 2 permutations.
</p>


<h3>References</h3>

<p>Syswerda, G. (1991): 
Schedule Optimization Using Genetic Algorithms.
In: Davis, L. (Ed.): 
Handbook of Genetic Algorithms, Chapter 21, p. 343.
Van Nostrand Reinhold, New York.
(ISBN:0-442-00173-8)
</p>


<h3>See Also</h3>

<p>Other Crossover: 
<code><a href="#topic+xegaPermCross2Gene">xegaPermCross2Gene</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
gene2&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene2, lFxegaPermGene)
newgenes&lt;-xegaPermCrossGene(gene1, gene2)
xegaPermDecodeGene(newgenes[[1]], lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermCrossoverFactory'>Configure the crossover function of a genetic algorithm.</h2><span id='topic+xegaPermCrossoverFactory'></span>

<h3>Description</h3>

<p><code>xegaPermCrossoverFactory</code> implements the selection
of one of the crossover functions in this
package by specifying a text string.
The selection fails ungracefully (produces
a runtime error), if the label does not match.
The functions are specified locally.
</p>
<p>Current support:
</p>

<ol>
<li><p> Crossover functions with two kids:
</p>

<ol>
<li><p> &quot;Cross2Gene&quot; returns <code>xegaPermCross2Gene</code>.
</p>
</li></ol>

</li>
<li><p> Crossover functions with one kid:
</p>

<ol>
<li><p> &quot;CrossGene&quot; returns <code>xegaPermCrossGene</code>.
</p>
</li></ol>

</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>xegaPermCrossoverFactory(method = "Cross2Gene")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermCrossoverFactory_+3A_method">method</code></td>
<td>
<p>A string specifying the crossover function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A crossover function for genes.
</p>


<h3>See Also</h3>

<p>Other Configuration: 
<code><a href="#topic+lFxegaPermGene">lFxegaPermGene</a></code>,
<code><a href="#topic+xegaPermMutationFactory">xegaPermMutationFactory</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>XGene&lt;-xegaPermCrossoverFactory("Cross2Gene")
gene1&lt;-xegaPermInitGene(lFxegaPermGene)
gene2&lt;-xegaPermInitGene(lFxegaPermGene)
XGene(gene1, gene2, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermDecodeGene'>Decode a permutation.</h2><span id='topic+xegaPermDecodeGene'></span>

<h3>Description</h3>

<p><code>xegaPermDecodeGene</code> decodes a permutation gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermDecodeGene(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermDecodeGene_+3A_gene">gene</code></td>
<td>
<p>Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermDecodeGene_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>xegaPermDecodeGene</code> is the identy function.
</p>


<h3>Value</h3>

<p>A permutation gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(g)
</code></pre>

<hr>
<h2 id='xegaPermGene'>Package xegaPermGene.</h2><span id='topic+xegaPermGene'></span>

<h3>Description</h3>

<p>Genetic operations for permutation genes.
</p>


<h3>Details</h3>

<p>Permutation genes are  a representation of a tour of a 
Traveling Salesman Problem (TSP).
</p>
<p>For permutation genes, the <code>xegaPermGene</code> package provides
</p>

<ul>
<li><p> Gene initiatilization.
</p>
</li>
<li><p> Decoding of parameters.
</p>
</li>
<li><p> Mutation functions as well as a function factory for configuration.
</p>
</li>
<li><p> Crossover functions as well as a function factory for configuration.
</p>
</li></ul>



<h3>Permutation Gene Representation</h3>

<p>A permutation gene is a named list with at least the following elements:
</p>

<ul>
<li> <p><code>$gene1</code>:      The gene must be a permutation vector.
</p>
</li>
<li> <p><code>$fit</code>:        The fitness value of the gene
(for EvalGeneDet and EvalGeneU) or
the mean fitness (for stochastic functions
evaluated with EvalGeneStoch).
</p>
</li>
<li> <p><code>$evaluated</code>:  Boolean. Has the gene been evaluated?
</p>
</li>
<li> <p><code>$evalFail</code>:   Boolean. Has the evaluation of the gene failed?
</p>
</li></ul>



<h3>Abstract Interface of a Problem Environment for the TSP</h3>

<p>A problem environment <code>penv</code> for the TSP must provide:
</p>

<ul>
<li> <p><code>$name()</code>: Returns the name of the problem environment.
</p>
</li>
<li> <p><code>$genelength()</code>: The number of bits of the binary coded
real parameter vector. Used in <code>InitGene</code>.
</p>
</li>
<li> <p><code>$dist()</code>: The distance matrix of the TSP.
</p>
</li>
<li> <p><code>$cities()</code>: A list of city names or <code>1:numberOfCities</code>.
</p>
</li>
<li> <p><code>$f(permutation, gene, lF)</code>: 
Returns the fitness of the permutation (the length of a tour).
</p>
</li>
<li> <p><code>$solution()</code>: The minimal tour length (if known).
</p>
</li>
<li> <p><code>$path()</code>: An optimal TSP tour.
</p>
</li>
<li> <p><code>$show(permutation)</code>: Prints the tour with the distances
and the cumulative distances between the cities.  
</p>
</li>
<li><p> TSP Heuristics:
</p>

<ul>
<li> <p><code>$greedy(startposition, k)</code>: 
Computes a greedy tour of length k.
</p>
</li>
<li> <p><code>$kBestgreedy(k)</code>: 
Computes the best greedy tour of length k.
</p>
</li>
<li> <p><code>$rnd2Opt(permutation, maxTries)</code>: 
Generate a new permutation by a random 2-change.
<code>maxTries</code> is the maximal number of trials 
to find a better permutation. 
<code>$rnd2Opt</code> either returns
a better permutation or, if no better permutation can be found 
in <code>maxTries</code> attempts, the original permutation.   
</p>
</li>
<li> <p><code>$LinKernighan(permutation, maxTries)</code>: 
Returns a permutation generated by a random sequence of 2-changes 
with improving performance. The optimality criterion of the 
k Lin-Kernighan heuristics is replaced by the necessity of 
finding a sequence of random 2-changes with strictly 
increasing performance. 
</p>
</li></ul>
    
</li></ul>



<h3>Abstract Interface of Mutation Functions</h3>

<p>Each mutation function has the following function signature:
</p>
<p><code>newGene&lt;-Mutate(gene, lF)</code>
</p>
<p>All local parameters of the mutation function configured are 
expected in the local configuration <code>lF</code>.
</p>


<h3>Local Constants of Mutation Functions</h3>

<p>The local constants of a mutation function determine the 
the behavior of the function. 
</p>

<table>
<tr>
 <td style="text-align: right;"> 
<strong>Constant</strong> </td><td style="text-align: center;"> <strong>Default</strong> </td><td style="text-align: left;"> <strong>Used in</strong> </td>
</tr>
<tr>
 <td style="text-align: right;"> 
<code>lF$BitMutationRate1()</code> </td><td style="text-align: center;"> 0.005 </td><td style="text-align: left;"> xegaPermMutateGeneOrderBased </td>
</tr>
<tr>
 <td style="text-align: right;"> 
<code>lF$Lambda()</code>  </td><td style="text-align: center;"> 0.05         </td><td style="text-align: left;"> xegaPermMutateGenekInversion </td>
</tr>
<tr>
 <td style="text-align: right;"> 
                    </td><td style="text-align: center;">              </td><td style="text-align: left;"> xegaPermMutateGenekGreedy </td>
</tr>
<tr>
 <td style="text-align: right;"> 
                    </td><td style="text-align: center;">              </td><td style="text-align: left;"> xegaPermMutateGeneBestGreedy </td>
</tr>
<tr>
 <td style="text-align: right;"> 
<code>lF$max2Opt()</code> </td><td style="text-align: center;"> 100          </td><td style="text-align: left;"> xegaPermMutateGene2Opt </td>
</tr>
<tr>
 <td style="text-align: right;">
                    </td><td style="text-align: center;">              </td><td style="text-align: left;"> xegaPermMutateGenekOptLK </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Abstract Interface of Crossover Functions</h3>

<p>The signatures of the abstract interface to the 2 families 
of crossover functions are:
</p>
<p><code>ListOfTwoGenes&lt;-Crossover2(gene1, gene2, lF)</code> 
</p>
<p><code>newGene&lt;-Crossover(gene1, gene2, lF)</code>
</p>


<h3>The Architecture of the xegaX-Packages</h3>

<p>The xegaX-packages are a family of R-packages which implement 
eXtended Evolutionary and Genetic Algorithms (xega).  
The architecture has 3 layers, 
namely the user interface layer,
the population layer, and the gene layer: 
</p>

<ul>
<li>
<p>The user interface layer (package <code>xega</code>) 
provides a function call interface and configuration support
for several algorithms: genetic algorithms (sga), 
permutation-based genetic algorithms (sgPerm), 
derivation free algorithms as e.g. differential evolution (sgde), 
grammar-based genetic programming (sgp) and grammatical evolution
(sge). 
</p>
</li>
<li>
<p>The population layer (package <code>xegaPopulation</code>) contains
population related functionality as well as support for 
population statistics dependent adaptive mechanisms and parallelization.
</p>
</li>
<li> 
<p>The gene layer is split in a representation independent and 
a representation dependent part:
</p>

<ol>
<li> 
<p>The representation indendent part (package <code>xegaSelectGene</code>)
is responsible for variants of selection operators, evaluation 
strategies for genes, as well as profiling and timing capabilities.        
</p>
</li>
<li> 
<p>The representation dependent part consists of the following packages: 
</p>

<ul>
<li> <p><code>xegaGaGene</code> for binary coded genetic algorithms.
</p>
</li>
<li> <p><code>xegaPermGene</code> for permutation-based genetic algorithms.
</p>
</li>
<li> <p><code>xegaDfGene</code> for derivation free algorithms as e.g. 
differential evolution.
</p>
</li>
<li> <p><code>xegaGpGene</code> for grammar-based genetic algorithms.
</p>
</li>
<li> <p><code>xegaGeGene</code> for grammatical evolution algorithms.
</p>
</li></ul>

<p>The packages <code>xegaDerivationTrees</code> and <code>xegaBNF</code> support
the last two packages:
<code>xegaBNF</code> essentially provides a grammar compiler and 
<code>xegaDerivationTrees</code> an abstract data type for derivation trees.
</p>
</li></ol>
</li></ul>



<h3>Copyright</h3>

<p>(c) 2023 Andreas Geyer-Schulz
</p>


<h3>License</h3>

<p>MIT
</p>


<h3>URL</h3>

<p>&lt;https://github.com/ageyerschulz/xegaPermGene&gt;
</p>


<h3>Installation</h3>

<p>From CRAN by <code>install.packages('xegaPermGene')</code>
</p>


<h3>Author(s)</h3>

<p>Andreas Geyer-Schulz
</p>

<hr>
<h2 id='xegaPermInitGene'>Initialize a gene with a permutation of integers</h2><span id='topic+xegaPermInitGene'></span>

<h3>Description</h3>

<p><code>xegaPermInitGene</code> generates a random permutation 
with a given length n.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermInitGene(lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermInitGene_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the permutation representation of 
package <code>xegaPerm</code>, <em>gene</em> is a list with 
</p>

<ol>
<li> <p><code>$evaluated</code>: Boolean: TRUE if the fitness is known.
</p>
</li>
<li> <p><code>$fit</code>:       The fitness of the genotype of 
<code>$gene1</code>.         
</p>
</li>
<li> <p><code>$gene1</code>:     The permutation (the genetopye).
</p>
</li></ol>

<p>This representation makes several code optimizations
and generalizations easier.
</p>


<h3>Value</h3>

<p>A permutation gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xegaPermInitGene(lFxegaPermGene)

</code></pre>

<hr>
<h2 id='xegaPermMutateGene2Opt'>Mutate a gene (by a random 2-Opt move).</h2><span id='topic+xegaPermMutateGene2Opt'></span>

<h3>Description</h3>

<p><code>xegaPermMutateGene2Opt</code> mutates a permutation.
The per position mutation rate is given by MutationRate().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateGene2Opt(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateGene2Opt_+3A_gene">gene</code></td>
<td>
<p>A Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateGene2Opt_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This operator is an implementation of the 2-Opt move
due to Croes (1958).
</p>
<p>Two edges are exchanged, if the exchange improves the result.
</p>


<h3>Value</h3>

<p>A Permutation.
</p>


<h3>References</h3>

<p>Croes, G. A. (1958): 
A Method for Solving Traveling-Salesman Problems.
Operations Research, 6(6), pp. 791-812.
&lt;doi:10.1287/opre.6.6.791&gt;
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGeneBestGreedy">xegaPermMutateGeneBestGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneGreedy">xegaPermMutateGeneGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneOrderBased">xegaPermMutateGeneOrderBased</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekInversion">xegaPermMutateGenekInversion</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekOptLK">xegaPermMutateGenekOptLK</a>()</code>,
<code><a href="#topic+xegaPermMutateMix">xegaPermMutateMix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateGene2Opt(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermMutateGeneBestGreedy'>Mutate a gene (by inserting a greedy path at start of random length k).</h2><span id='topic+xegaPermMutateGeneBestGreedy'></span>

<h3>Description</h3>

<p><code>xegaPermMutateGeneGreedy</code> mutates a permutation 
by inserting a greedy path of length <code>k</code> 
at a random position <code>start</code>.
The mutation rate for a gene is given by <code>MutationRate()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateGeneBestGreedy(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateGeneBestGreedy_+3A_gene">gene</code></td>
<td>
<p>A Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateGeneBestGreedy_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The path length <code>k</code> is expontially decaying
with exponential decay constant <code>lF$lambda()</code>.
</p>


<h3>Value</h3>

<p>A Permutation
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGene2Opt">xegaPermMutateGene2Opt</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneGreedy">xegaPermMutateGeneGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneOrderBased">xegaPermMutateGeneOrderBased</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekInversion">xegaPermMutateGenekInversion</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekOptLK">xegaPermMutateGenekOptLK</a>()</code>,
<code><a href="#topic+xegaPermMutateMix">xegaPermMutateMix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateGeneGreedy(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermMutateGeneGreedy'>Mutate a gene (by inserting a greedy path at start of random length k).</h2><span id='topic+xegaPermMutateGeneGreedy'></span>

<h3>Description</h3>

<p><code>xegaPermMutateGeneGreedy</code> mutates a permutation 
by inserting a greedy path of length <code>k</code> 
at a random position <code>start</code>.
The mutation rate for a gene is given by MutationRate().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateGeneGreedy(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateGeneGreedy_+3A_gene">gene</code></td>
<td>
<p>A Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateGeneGreedy_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The path length <code>k</code> is expontially decaying
with exponential decay constant <code>lambda</code>.
</p>


<h3>Value</h3>

<p>A Permutation.
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGene2Opt">xegaPermMutateGene2Opt</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneBestGreedy">xegaPermMutateGeneBestGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneOrderBased">xegaPermMutateGeneOrderBased</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekInversion">xegaPermMutateGenekInversion</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekOptLK">xegaPermMutateGenekOptLK</a>()</code>,
<code><a href="#topic+xegaPermMutateMix">xegaPermMutateMix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateGeneGreedy(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermMutateGenekInversion'>Mutate a gene (k random inversions).</h2><span id='topic+xegaPermMutateGenekInversion'></span>

<h3>Description</h3>

<p><code>xegaPermMutateGenekInversion</code> performs k random inversions.
The number of inversions is expontially decaying
with exponential decay constant <code>lambda</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateGenekInversion(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateGenekInversion_+3A_gene">gene</code></td>
<td>
<p>A Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateGenekInversion_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The only difference to the order based mutation 
operator (Syswerda, 1991) is the exponential decay 
in the number of inversions. 
</p>

<ol>
<li><p> The indices of a random subschedule are extracted.
</p>
</li>
<li><p> The subschedule is extracted, permuted, and reinserted.
</p>
</li></ol>



<h3>Value</h3>

<p>A Permutation.
</p>


<h3>References</h3>

<p>Syswerda, G. (1991): 
Schedule Optimization Using Genetic Algorithms.
In: Davis, L. (Ed.): 
Handbook of Genetic Algorithms, Chapter 21, pp. 332-349.
Van Nostrand Reinhold, New York.
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGene2Opt">xegaPermMutateGene2Opt</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneBestGreedy">xegaPermMutateGeneBestGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneGreedy">xegaPermMutateGeneGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneOrderBased">xegaPermMutateGeneOrderBased</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekOptLK">xegaPermMutateGenekOptLK</a>()</code>,
<code><a href="#topic+xegaPermMutateMix">xegaPermMutateMix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateGenekInversion(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)

</code></pre>

<hr>
<h2 id='xegaPermMutateGenekOptLK'>Mutate a gene (by a random Lin-Kernighan k-OPT move).</h2><span id='topic+xegaPermMutateGenekOptLK'></span>

<h3>Description</h3>

<p><code>xegaPermMutateGenekOptLK</code> mutates a permutation.
The mutation rate of a gene is given by MutationRate().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateGenekOptLK(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateGenekOptLK_+3A_gene">gene</code></td>
<td>
<p>A Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateGenekOptLK_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This operator is an implementation of the random k-Opt move
version of the Lin-Kernighan heuristic.
</p>
<p>A sequence of random 2-Opt moves, all of which improve 
the result is executed.
</p>


<h3>Value</h3>

<p>A Permutation.
</p>


<h3>References</h3>

<p>Lin, S. and Kernighan. B. W. (1973): 
An Effective Heuristic Algorithm 
for the Traveling-Salesman Problem.
Operations Research, 21(2), pp. 791-812.
&lt;doi:10.1287/opre.21.2.498&gt;
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGene2Opt">xegaPermMutateGene2Opt</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneBestGreedy">xegaPermMutateGeneBestGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneGreedy">xegaPermMutateGeneGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneOrderBased">xegaPermMutateGeneOrderBased</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekInversion">xegaPermMutateGenekInversion</a>()</code>,
<code><a href="#topic+xegaPermMutateMix">xegaPermMutateMix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateGenekOptLK(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermMutateGeneOrderBased'>Mutate a gene (generalized order based mutation).</h2><span id='topic+xegaPermMutateGeneOrderBased'></span>

<h3>Description</h3>

<p><code>xegaPermMutateGene</code> mutates a permutation.
The per position mutation rate is given by 
<code>lF$BitMutationRate1()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateGeneOrderBased(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateGeneOrderBased_+3A_gene">gene</code></td>
<td>
<p>A Permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateGeneOrderBased_+3A_lf">lF</code></td>
<td>
<p>Local configuration of the genetic algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This operator is an implementation of a generalized 
order based mutation operator (Syswerda, 1991).
</p>

<ol>
<li><p> The indices of a random subschedule are extracted.
</p>
</li>
<li><p> The subschedule is extracted, permuted, and reinserted.
</p>
</li></ol>



<h3>Value</h3>

<p>A Permutation.
</p>


<h3>References</h3>

<p>Syswerda, G. (1991): 
Schedule Optimization Using Genetic Algorithms.
In: Davis, L. (Ed.): 
Handbook of Genetic Algorithms, Chapter 21, pp. 332-349.
Van Nostrand Reinhold, New York.
(ISBN:0-442-00173-8)
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGene2Opt">xegaPermMutateGene2Opt</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneBestGreedy">xegaPermMutateGeneBestGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneGreedy">xegaPermMutateGeneGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekInversion">xegaPermMutateGenekInversion</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekOptLK">xegaPermMutateGenekOptLK</a>()</code>,
<code><a href="#topic+xegaPermMutateMix">xegaPermMutateMix</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateGeneOrderBased(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermMutateMix'>Mutation by a random mutation function.</h2><span id='topic+xegaPermMutateMix'></span>

<h3>Description</h3>

<p>A mutation function is randomly selected from the following list:
xegaPermMutateGeneOrderBased, xegaPermMutateGenekInversion,
xegaPermMutateGene2Opt, xegaPermMutateGenekOptLK, xegaPermMutateGeneGreedy,
xegaPermMutateGeneBestGreedy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutateMix(gene, lF)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutateMix_+3A_gene">gene</code></td>
<td>
<p>A permutation.</p>
</td></tr>
<tr><td><code id="xegaPermMutateMix_+3A_lf">lF</code></td>
<td>
<p>Local configuration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A permutation.
</p>


<h3>See Also</h3>

<p>Other Mutation: 
<code><a href="#topic+xegaPermMutateGene2Opt">xegaPermMutateGene2Opt</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneBestGreedy">xegaPermMutateGeneBestGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneGreedy">xegaPermMutateGeneGreedy</a>()</code>,
<code><a href="#topic+xegaPermMutateGeneOrderBased">xegaPermMutateGeneOrderBased</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekInversion">xegaPermMutateGenekInversion</a>()</code>,
<code><a href="#topic+xegaPermMutateGenekOptLK">xegaPermMutateGenekOptLK</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene1&lt;-xegaPermInitGene(lFxegaPermGene)
xegaPermDecodeGene(gene1, lFxegaPermGene)
gene&lt;-xegaPermMutateMix(gene1, lFxegaPermGene)
xegaPermDecodeGene(gene, lFxegaPermGene)
</code></pre>

<hr>
<h2 id='xegaPermMutationFactory'>Configure the mutation function of a genetic algorithm.</h2><span id='topic+xegaPermMutationFactory'></span>

<h3>Description</h3>

<p><code>xegaPermMutationFactory</code> implements the selection
of one of the gene mutation functions in this 
package by specifying a text string.
The selection fails ungracefully (produces
a runtime error), if the label does not match.
The functions are specified locally.             
</p>
<p>Current Support:
</p>

<ol>
<li><p> &quot;MutateGene&quot; returns <code>xegaPermMutateGeneOrderBased</code>.
</p>
</li>
<li><p> &quot;MutateGeneOrderBased&quot; returns <code>xegaPermMutateGeneOrderBased</code>.
</p>
</li>
<li><p> &quot;MutateGenekInversion&quot; returns <code>xegaPermMutateGenekInversion</code>.
</p>
</li>
<li><p> &quot;MutateGene2Opt&quot; returns <code>xegaPermMutateGene2Opt</code>.
</p>
</li>
<li><p> &quot;MutateGenekOptLK&quot; returns <code>xegaPermMutateGenekOptLK</code>.
</p>
</li>
<li><p> &quot;MutateGeneGreedy&quot; returns <code>xegaPermMutateGeneGreedy</code>.
</p>
</li>
<li><p> &quot;MutateGeneBestGreedy&quot; returns <code>xegaPermMutateGeneBestGreedy</code>.
</p>
</li>
<li><p> &quot;MutateGeneMix&quot; returns <code>xegaPermMutateMix</code>.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>xegaPermMutationFactory(method = "MutateGene")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xegaPermMutationFactory_+3A_method">method</code></td>
<td>
<p>The name of the mutation method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A permutation based mutation function.
</p>


<h3>See Also</h3>

<p>Other Configuration: 
<code><a href="#topic+lFxegaPermGene">lFxegaPermGene</a></code>,
<code><a href="#topic+xegaPermCrossoverFactory">xegaPermCrossoverFactory</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xegaPermMutationFactory(method="MutateGene")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
