<!DOCTYPE html><html><head><title>Help for package xlink</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {xlink}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#xlink'><p>xlink: A package for genetic association models for X-chromosome SNPS on continuous, binary and survival outcomes.</p></a></li>
<li><a href='#fit_all_models'><p>Model fitting results for each SNP considering as XCI, XCI-E and XCI-S type</p></a></li>
<li><a href='#fit_XCI_E_model'><p>Model fitting results for each SNP considering as XCI-E type</p></a></li>
<li><a href='#fit_XCI_model'><p>Model fitting results for each SNP considering as XCI type</p></a></li>
<li><a href='#infor_table'><p>Information table.</p></a></li>
<li><a href='#MAF'><p>Select SNP by MAF.</p></a></li>
<li><a href='#Rdata'><p>Simulation data for Genetic association models for X-chromosome SNPS</p></a></li>
<li><a href='#select_output'><p>Selected results table by P value</p></a></li>
<li><a href='#xlink_fit'><p>Genetic association models for X-chromosome SNPs on continuous, binary and survival outcomes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Genetic Association Models for X-Chromosome SNPS on Continuous,
Binary and Survival Outcomes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yi Zhu &lt;yizhu87@gmail.com&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/qiuanzhu/xlink">https://github.com/qiuanzhu/xlink</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/qiuanzhu/xlink/issues">https://github.com/qiuanzhu/xlink/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>The expression of X-chromosome undergoes three possible biological processes: X-chromosome
             inactivation (XCI), escape of the X-chromosome inactivation (XCI-E),and skewed X-chromosome                    
             inactivation (XCI-S). To analyze the X-linked genetic association for phenotype such as                         
             continuous, binary, and time-to-event outcomes with the actual process unknown, we propose a unified            
             approach of maximizing the likelihood or partial likelihood over all of the potential biological             
             processes. The methods are described in Wei Xu, Meiling Hao (2017) &lt;<a href="https://doi.org/10.1002%2Fgepi.22097">doi:10.1002/gepi.22097</a>&gt;. And 
             also see Dongxiao Han, Meiling Hao, Lianqiang Qu, Wei Xu (2019) &lt;<a href="https://doi.org/10.1177%2F0962280219859037">doi:10.1177/0962280219859037</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival (&ge; 2.41.3)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-08-20 14:44:08 UTC; Yi</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Xu [aut],
  Meiling Hao [aut],
  Yi Zhu [cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-08-20 21:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='xlink'>xlink: A package for genetic association models for X-chromosome SNPS on continuous, binary and survival outcomes.</h2><span id='topic+xlink'></span><span id='topic+xlink-package'></span>

<h3>Description</h3>

<p>The expression of X-chromosome undergoes three possible biological processes: X-chromosome inactivation (XCI),
escape of the X-chromosome inactivation (XCI-E),and skewed X-chromosome inactivation (XCI-S).
To analyze the X-linked genetic association for phenotype  such as continuous, binary, and time-to-event outcomes with
the actual process unknown, we propose a unified approach of maximizing the likelihood or partial likelihood over
all of the potential biological processes.
</p>


<h3>xlink functions</h3>

<p>xlink_fit, select_output.
</p>


<h3>References</h3>

<p>Xu, Wei, and Meiling Hao. 'A unified partial likelihood approach for X-chromosome association on time-to-event outcomes.' Genetic epidemiology 42.1 (2018): 80-94.
</p>

<hr>
<h2 id='fit_all_models'>Model fitting results for each SNP considering as XCI, XCI-E and XCI-S type</h2><span id='topic+fit_all_models'></span>

<h3>Description</h3>

<p><code>fit_all_models</code> returns model fitting results for each SNP understanding as XCI, XCI-E and XCI-S type respectively. Model comparison results is provided by using AIC as a criterion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_all_models(resp, os, ostime, snp, gender, male, female, covars, model,
  data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_all_models_+3A_resp">resp</code></td>
<td>
<p>Response variable for continuous or binary model fitting.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_os">os</code></td>
<td>
<p>Survival indicator, 1 for death, 0 for censoring.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_ostime">ostime</code></td>
<td>
<p>Duration time of survival.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_snp">snp</code></td>
<td>
<p>Single SNP name.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_gender">gender</code></td>
<td>
<p>Gender variable.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_male">male</code></td>
<td>
<p>Male indicator in gender variable.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_female">female</code></td>
<td>
<p>Female indicator in gender variable.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_covars">covars</code></td>
<td>
<p>Covariates list.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_model">model</code></td>
<td>
<p>Fitting model type. For 'linear', fitting linear model. For 'binary', fitting logistic regression model. For 'survival', fitting survival model.</p>
</td></tr>
<tr><td><code id="fit_all_models_+3A_data">data</code></td>
<td>
<p>Data set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns estimated parameters, confidence interval and P value for each variable. Baseline model and full model maximum likelihood estimation are provided.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a>{stats}</code> for linear model, <code><a href="stats.html#topic+glm">glm</a>{stats}</code> for logistic regression model, and <code><a href="survival.html#topic+coxph">coxph</a>{survival}</code> for survival model.
</p>

<hr>
<h2 id='fit_XCI_E_model'>Model fitting results for each SNP considering as XCI-E type</h2><span id='topic+fit_XCI_E_model'></span>

<h3>Description</h3>

<p><code>fit_XCI_E_model</code> returns model fitting results for each SNP understanding as XCI-E type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_XCI_E_model(resp, os, ostime, snp, gender, male, female, covars, model,
  data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_XCI_E_model_+3A_resp">resp</code></td>
<td>
<p>Response variable for continuous or binary model fitting.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_os">os</code></td>
<td>
<p>Survival indicator, 1 for death, 0 for censoring.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_ostime">ostime</code></td>
<td>
<p>Duration time of survival.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_snp">snp</code></td>
<td>
<p>Single SNP name.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_gender">gender</code></td>
<td>
<p>Gender variable.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_male">male</code></td>
<td>
<p>Male indicator in gender variable.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_female">female</code></td>
<td>
<p>Female indicator in gender variable.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_covars">covars</code></td>
<td>
<p>Covariates list.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_model">model</code></td>
<td>
<p>Fitting model type. For 'linear', fitting linear model. For 'binary', fitting logistic regression model. For 'survival', fitting survival model.</p>
</td></tr>
<tr><td><code id="fit_XCI_E_model_+3A_data">data</code></td>
<td>
<p>Data set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns estimated parameters, confidence interval and P value for each variable. Baseline model and full model maximum likelihood estimation are provided.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a>{stats}</code> for linear model, <code><a href="stats.html#topic+glm">glm</a>{stats}</code> for logistic regression model, and <code><a href="survival.html#topic+coxph">coxph</a>{survival}</code> for survival model.
</p>

<hr>
<h2 id='fit_XCI_model'>Model fitting results for each SNP considering as XCI type</h2><span id='topic+fit_XCI_model'></span>

<h3>Description</h3>

<p><code>fit_XCI_model</code> returns model fitting results for each SNP understanding as XCI type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_XCI_model(resp, os, ostime, snp, gender, male, female, covars, model,
  data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_XCI_model_+3A_resp">resp</code></td>
<td>
<p>Response variable for continuous or binary model fitting.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_os">os</code></td>
<td>
<p>Survival indicator, 1 for death, 0 for censoring.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_ostime">ostime</code></td>
<td>
<p>Duration time of survival.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_snp">snp</code></td>
<td>
<p>Single SNP name.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_gender">gender</code></td>
<td>
<p>Gender variable.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_male">male</code></td>
<td>
<p>Male indicator in gender variable.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_female">female</code></td>
<td>
<p>Female indicator in gender variable.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_covars">covars</code></td>
<td>
<p>Covariates list.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_model">model</code></td>
<td>
<p>Fitting model type. For 'linear', fitting linear model. For 'binary', fitting logistic regression model. For 'survival', fitting survival model.</p>
</td></tr>
<tr><td><code id="fit_XCI_model_+3A_data">data</code></td>
<td>
<p>Data set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns estimated parameters, confidence interval and P value for each variable. Baseline model and full model maximum likelihood estimation are provided.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a>{stats}</code> for linear model, <code><a href="stats.html#topic+glm">glm</a>{stats}</code> for logistic regression model, and <code><a href="survival.html#topic+coxph">coxph</a>{survival}</code> for survival model.
</p>

<hr>
<h2 id='infor_table'>Information table.</h2><span id='topic+infor_table'></span>

<h3>Description</h3>

<p><code>infor_table</code> returns information table of estimated coefficients/hazard ratio, confidence interval and P value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infor_table(x, snp, covar_n, MAF_value, model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infor_table_+3A_x">x</code></td>
<td>
<p>An output from continuous/binary/survival model.</p>
</td></tr>
<tr><td><code id="infor_table_+3A_snp">snp</code></td>
<td>
<p>Single SNP name.</p>
</td></tr>
<tr><td><code id="infor_table_+3A_covar_n">covar_n</code></td>
<td>
<p>Covariate names.</p>
</td></tr>
<tr><td><code id="infor_table_+3A_maf_value">MAF_value</code></td>
<td>
<p>A minimum value of minor allele frequency.</p>
</td></tr>
<tr><td><code id="infor_table_+3A_model">model</code></td>
<td>
<p>Model type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Information table. If linear or binary model is chosen, it returns estimated coefficients, 
confidence interval and P value. If survival model is chosen, it returns hazard ratio, confidence interval and P value.
</p>

<hr>
<h2 id='MAF'>Select SNP by MAF.</h2><span id='topic+MAF'></span>

<h3>Description</h3>

<p><code>MAF</code> returns SNPs with higher MAF than default value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAF(snp, gender, male, MAF_v, data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAF_+3A_snp">snp</code></td>
<td>
<p>SNP name.</p>
</td></tr>
<tr><td><code id="MAF_+3A_gender">gender</code></td>
<td>
<p>gender variable.</p>
</td></tr>
<tr><td><code id="MAF_+3A_male">male</code></td>
<td>
<p>male information.</p>
</td></tr>
<tr><td><code id="MAF_+3A_maf_v">MAF_v</code></td>
<td>
<p>minimum MAF value.</p>
</td></tr>
<tr><td><code id="MAF_+3A_data">data</code></td>
<td>
<p>a dataset.</p>
</td></tr>
</table>

<hr>
<h2 id='Rdata'>Simulation data for Genetic association models for X-chromosome SNPS</h2><span id='topic+Rdata'></span>

<h3>Description</h3>

<p>A simulated dataset containing 400 observations. The variables list as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Rdata)
</code></pre>


<h3>Format</h3>

<p>A data frame with 400 rows and 17 variables.</p>


<h3>Details</h3>


<ul>
<li><p> ID. Identification number.
</p>
</li>
<li><p> OS. Survival indicator, 1 for death, 0 for censoring.
</p>
</li>
<li><p> OS_time. Duration time of survival.
</p>
</li>
<li><p> gender. Binary value 0,1 with P(x=1)=0.5 and hazard ratio is 1.
</p>
</li>
<li><p> Age. Uniform distribution in [20,80] and hazard ratio is 1.02.
</p>
</li>
<li><p> Smoking. Binary value 0,1 with P(x=1)=0.3 and hazard ratio is 1.2.
</p>
</li>
<li><p> Treatment. Binary value 0,1 with P(X=1)=0.3 and hazard ratio is 1.2.
</p>
</li>
<li><p> snp_1. True type in coxph model is 'XCI', minor allele frequency is 0.2, hazard ratio is 1.5.
</p>
</li>
<li><p> snp_2. True type in coxph model is 'XCI-E', minor allele frequency is 0.3, hazard ratio is 1.5.
</p>
</li>
<li><p> snp_3. True type in coxph model is 'XCI-S', minor allele frequency is 0.4, hazard ratio is 1.5.
</p>
</li>
<li><p> snp_4. True type in coxph model is 'XCI', minor allele frequency is 0.3, hazard ratio is 1.
</p>
</li>
<li><p> snp_5. True type in coxph model is 'XCI-E', minor allele frequency is 0.1, hazard ratio is 1.
</p>
</li>
<li><p> snp_6. True type in coxph model is 'XCI', minor allele frequency is 0.2, hazard ratio is 1.
</p>
</li>
<li><p> snp_7. True type in coxph model is 'XCI', minor allele frequency is 0.15, hazard ratio is 1.
</p>
</li>
<li><p> snp_8. True type in coxph model is 'XCI-E', minor allele frequency is 0.1, hazard ratio is 1.
</p>
</li>
<li><p> snp_9. True type in coxph model is 'XCI', minor allele frequency is 0.2, hazard ratio is 1.
</p>
</li>
<li><p> snp_10. True type in coxph model is 'XCI', minor allele frequency is 0.15, hazard ratio is 1.
</p>
</li></ul>


<hr>
<h2 id='select_output'>Selected results table by P value</h2><span id='topic+select_output'></span>

<h3>Description</h3>

<p><code>select_output</code> returns selected SNP information by P value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_output(input, pv_thold = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_output_+3A_input">input</code></td>
<td>
<p>Input results from xlink fit.</p>
</td></tr>
<tr><td><code id="select_output_+3A_pv_thold">pv_thold</code></td>
<td>
<p>P value threshold for output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns estimated parameters, confidence interval, P value, MAF and Best model information.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+xlink_fit">xlink_fit</a>{xlink}</code> for input results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Covars&lt;-c("Age","Smoking","Treatment")
SNPs&lt;-c("snp_1","snp_2","snp_3")
result&lt;-xlink_fit(os="OS",ostime ="OS_time",snps=SNPs,gender ="gender",covars=Covars,
                  option =list(type="all",MAF_v=0.05), model="survival", data = Rdata)
select_output(input=result,pv_thold=10^-5)
</code></pre>

<hr>
<h2 id='xlink_fit'>Genetic association models for X-chromosome SNPs on continuous, binary and survival outcomes</h2><span id='topic+xlink_fit'></span>

<h3>Description</h3>

<p><code>xlink_fit</code> returns model fitting results for each SNP with the covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xlink_fit(resp = c(), os = c(), ostime = c(), snps = c(),
  gender = c(), covars = c(), option = c(type = c(), male = c(),
  female = c(), MAF_v = 0), model = c(), data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xlink_fit_+3A_resp">resp</code></td>
<td>
<p>Response variable for continuous or binary model fitting.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_os">os</code></td>
<td>
<p>Survival indicator, 1 for death, 0 for censoring.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_ostime">ostime</code></td>
<td>
<p>Duration time of survival.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_snps">snps</code></td>
<td>
<p>SNP name list for model fitting.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_gender">gender</code></td>
<td>
<p>Gender information must be included in the data. Default setting is male=1 and female=0. If not as default setting, please provide male and female information in the option.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_covars">covars</code></td>
<td>
<p>Covariates list if needed.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_option">option</code></td>
<td>
<p>There are three options. First, type has default 'all', which provides model fitting results for each SNP understanding as 'XCI', 'XCI-E' and 'XCI-S' type respectively. If type is chosen as 'XCI' or 'XCI-E', all the SNPS consider as 'XCI' or 'XCI-E' type in corresponding model. Secondly, if gender is not as default gender setting (male=1,female=0), male and female information should be provided here.The third one, MAF_v is the low bound of the minimum allele frequency, the SNP MAF below this value will not be used in xlink_fit.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_model">model</code></td>
<td>
<p>Fitting model. For 'linear', fitting linear model. For 'binary', fitting logistic regression model. For 'survival', fitting survival model.</p>
</td></tr>
<tr><td><code id="xlink_fit_+3A_data">data</code></td>
<td>
<p>Data set.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns estimated parameters, confidence interval and P value for each variable in the chosen model. The baseline and full model maximum likelihood estimation are provided. If type is 'all', best model choice is provided by using AIC as an benchmark.
</p>


<h3>References</h3>

<p>Xu, Wei, and Meiling Hao. 'A unified partial likelihood approach for X-chromosome association on time-to-event outcomes.' Genetic epidemiology 42.1 (2018): 80-94.
</p>
<p>Han, D., Hao, M., Qu, L., &amp; Xu, W. (2019). A novel model for the X-chromosome inactivation association on survival data. Statistical Methods in Medical Research.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a>{stats}</code> for linear model, <code><a href="stats.html#topic+glm">glm</a>{stats}</code> for logistic regression model, and <code><a href="survival.html#topic+coxph">coxph</a>{survival}</code> for survival model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Covars&lt;-c("Age","Smoking","Treatment")
SNPs&lt;-c("snp_1","snp_2","snp_3")
xlink_fit(os="OS",ostime="OS_time",snps=SNPs,gender="gender",covars=Covars,
          option =list(MAF_v=0.05),model="survival",data = Rdata)
xlink_fit(resp="OS_time",snps=SNPs,gender="gender",option =list(type="XCI",MAF_v=0.05),
          model="linear",data = Rdata)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
