<!DOCTYPE html><html lang="en"><head><title>Help for package xoi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {xoi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bssbsb'><p>BSS/BSB backcross data</p></a></li>
<li><a href='#chiasma'><p>Estimate chiasma distribution from crossover counts</p></a></li>
<li><a href='#coincidence'><p>Estimate coincidence function</p></a></li>
<li><a href='#convertxoloc'><p>Convert format of crossover locations data</p></a></li>
<li><a href='#countxo'><p>Estimate number of crossovers</p></a></li>
<li><a href='#distance.given.two'><p>Distance between crossovers given there are two</p></a></li>
<li><a href='#est.coi'><p>Estimate the coincidence function</p></a></li>
<li><a href='#est.coi.um'><p>Estimate the coincidence as a function of micron distance</p></a></li>
<li><a href='#est.recrate'><p>Estimate recombination rate</p></a></li>
<li><a href='#find.breaks'><p>Estimate crossover locations</p></a></li>
<li><a href='#first.given.two'><p>Location of first crossover given there are two</p></a></li>
<li><a href='#firstden'><p>Distance to a crossover</p></a></li>
<li><a href='#fitGamma'><p>Fit Gamma model</p></a></li>
<li><a href='#fitStahl'><p>Fit Stahl model</p></a></li>
<li><a href='#gammacoi'><p>Coincidence function for the gamma model</p></a></li>
<li><a href='#intensity'><p>Estimate intensity function</p></a></li>
<li><a href='#ioden'><p>Distance between crossovers</p></a></li>
<li><a href='#joint.given.two'><p>Crossover locations given there are two</p></a></li>
<li><a href='#kfunc'><p>estimate Ripley's K function</p></a></li>
<li><a href='#location.given.one'><p>Location of crossover given there is one</p></a></li>
<li><a href='#recrate2scanone'><p>Convert recrate to scanone format</p></a></li>
<li><a href='#simStahl'><p>Simulate crossover locations under the Stahl model</p></a></li>
<li><a href='#stahlcoi'><p>Coincidence function for the Stahl model</p></a></li>
<li><a href='#stahlLoglik'><p>Calculate log likelihood for Stahl model</p></a></li>
<li><a href='#xoiversion'><p>Installed version of R/xoi</p></a></li>
<li><a href='#xoprob'><p>Distribution of number of crossovers</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.72</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-03-21</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10.1), qtl</td>
</tr>
<tr>
<td>Title:</td>
<td>Tools for Analyzing Crossover Interference</td>
</tr>
<tr>
<td>Author:</td>
<td>Karl W Broman <a href="https://orcid.org/0000-0002-4914-6671"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Il-Youp Kwak <a href="https://orcid.org/0000-0002-7117-7669"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karl W Broman &lt;broman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of crossover interference in experimental crosses,
    particularly regarding the gamma model. See, for example,
    Broman and Weber (2000) &lt;<a href="https://doi.org/10.1086%2F302923">doi:10.1086/302923</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>devtools, roxygen2, testthat</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kbroman/xoi">https://github.com/kbroman/xoi</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kbroman/xoi/issues">https://github.com/kbroman/xoi/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-22 12:21:39 UTC; kbroman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-22 12:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bssbsb'>BSS/BSB backcross data</h2><span id='topic+bssbsb'></span>

<h3>Description</h3>

<p>Data from two densely genotyped backcrosses.
</p>


<h3>Format</h3>

<p>An object of class <code>cross</code>.  See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code>
for details.
</p>


<h3>Details</h3>

<p>There are 94 individuals from each of two interspecific backcross: (C57BL/6J
<code class="reqn">\times</code> <em>M. spretus</em>) <code class="reqn">\times</code> C57BL/6J and (C57BL/6J
<code class="reqn">\times</code> SPRET/Ei) <code class="reqn">\times</code> SPRET/Ei.  They were typed on 1372
and 4913 genetic markers, respectively, with 904 markers in common.
</p>
<p>These data are from September, 2000.  Updated data are available.
</p>


<h3>Source</h3>

<p>Lucy Rowe, Jackson Laboratory
</p>


<h3>References</h3>

<p>Rowe, L. B., Nadeau, J. H., Turner, R., Frankel, W. N., Letts,
V. A., Eppig, J. T., Ko, M. S., Thurston, S. J. and Birkenmeier, E. H.
(1994) Maps from two interspecific backcross DNA panels available as a
community genetic mapping resource.  <em>Mamm. Genome</em> <strong>5</strong>, 253&ndash;274.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)
summary(bssbsb)
## Not run: plot(bssbsb)

</code></pre>

<hr>
<h2 id='chiasma'>Estimate chiasma distribution from crossover counts</h2><span id='topic+chiasma'></span>

<h3>Description</h3>

<p>Fit several models, with an assumption of no chromatid interference, to
crossover count data to obtain fitted distributions of the number of
chiasmata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chiasma(
  xo,
  max.chiasma = max(xo) * 2 + 5,
  n.iter = 10000,
  tol = 0.000001,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chiasma_+3A_xo">xo</code></td>
<td>
<p>Vector of non-negative integers; the number of crossovers in a set
of meiotic products.</p>
</td></tr>
<tr><td><code id="chiasma_+3A_max.chiasma">max.chiasma</code></td>
<td>
<p>Maximum number of chiasmata to allow.</p>
</td></tr>
<tr><td><code id="chiasma_+3A_n.iter">n.iter</code></td>
<td>
<p>Maximum number of iterations in the EM algorithm.</p>
</td></tr>
<tr><td><code id="chiasma_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence of the EM algorithm.</p>
</td></tr>
<tr><td><code id="chiasma_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, print number of interations for each of the 4 models
at the end.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Broman and Weber (2000) for details of the method.
</p>
<p>We use R's <code><a href="stats.html#topic+integrate">stats::integrate()</a></code> function for numerical integrals,
<code><a href="stats.html#topic+optimize">stats::optimize()</a></code> for optimizing the likelihood, and
<code><a href="stats.html#topic+uniroot">stats::uniroot()</a></code> for identifying the endpoints of the likelihood
support interval.
</p>


<h3>Value</h3>

<p>A list with three components.
</p>
<p>First, a matrix containing the observed distribution of the numbers of
crossovers, followed by the fitted distributions under the Poisson model,
the truncated Poisson model (assuming an obligate chiasma), the obligate
chiasma model, and the freely varying model.  In all cases we assume no
chromatid interference.
</p>
<p>Second, a matrix containing the estimated distributions of the number of
chiasmata on the four-strand bundle for the above four models.
</p>
<p>Third, the estimated average number of crossovers under the Poisson and
truncated Poisson models.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>Yu, K. and Feinbold, E. (2001) Estimating the frequency distribution of
crossovers during meiosis from recombination data.  <em>Biometrics</em>
<strong>57</strong>, 427&ndash;434.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitGamma">fitGamma()</a></code>, <code><a href="qtl.html#topic+fitstahl">qtl::fitstahl()</a></code>,
<code><a href="#topic+countxo">countxo()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)

# estimated number of crossovers on chr 1
nxo &lt;- countxo(bssbsb, chr=1)

# estimate chiasma distribution
## Not run: chiasma(nxo)


</code></pre>

<hr>
<h2 id='coincidence'>Estimate coincidence function</h2><span id='topic+coincidence'></span>

<h3>Description</h3>

<p>Estimate coincidence function for a chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coincidence(cross, chr = NULL, window = 5, ncalc = 500)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coincidence_+3A_cross">cross</code></td>
<td>
<p>Cross object; must be a backcross.  See
<code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code> for format details.</p>
</td></tr>
<tr><td><code id="coincidence_+3A_chr">chr</code></td>
<td>
<p>Chromosome to consider (only one is allowed).  If NULL, the
first chromosome is considered.</p>
</td></tr>
<tr><td><code id="coincidence_+3A_window">window</code></td>
<td>
<p>Window size</p>
</td></tr>
<tr><td><code id="coincidence_+3A_ncalc">ncalc</code></td>
<td>
<p>Total number of points for calculations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with columns <code>distance</code> and <code>coincidence</code>.  The
input argument <code>window</code> is kept as an attribute.
</p>


<h3>Author(s)</h3>

<p>Il youp Kwak
</p>


<h3>See Also</h3>

<p><code><a href="#topic+intensity">intensity()</a></code>, <code><a href="#topic+est.coi">est.coi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
map1 &lt;- sim.map(103, n.mar=104, anchor=TRUE, include.x=FALSE, eq=TRUE)
x &lt;- sim.cross(map1, n.ind=2000, m=6, type="bc")

out &lt;- coincidence(x, ncalc=101)
plot(out, type="l", lwd=2, ylim=c(0, max(out[,2])))

</code></pre>

<hr>
<h2 id='convertxoloc'>Convert format of crossover locations data</h2><span id='topic+convertxoloc'></span>

<h3>Description</h3>

<p>Convert the format of data on crossover locations to that needed for the
function <code><a href="#topic+fitGamma">fitGamma</a>.</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convertxoloc(breaks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convertxoloc_+3A_breaks">breaks</code></td>
<td>
<p>A list of crossover locations, as output by
<code><a href="#topic+find.breaks">find.breaks()</a></code> or <code><a href="#topic+simStahl">simStahl()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with two columns: the inter-crossover and crossover-to
chromosome end differences (<code>"distance"</code>) and indicators of censoring
type (<code>"censor"</code>), with 0 = distance between crossovers, 1=start of
chromosome to first crossover, 2 = crossover to end of chromosome, and 3 =
whole chromosome.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+find.breaks">find.breaks()</a></code>, <code><a href="#topic+fitGamma">fitGamma()</a></code>,
<code><a href="#topic+simStahl">simStahl()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)

# crossover locations on chromosome 1
xoloc1 &lt;- convertxoloc(find.breaks(bssbsb, chr=1))

# crossover locations on all chromosomes
xoloc &lt;- convertxoloc(find.breaks(bssbsb))

</code></pre>

<hr>
<h2 id='countxo'>Estimate number of crossovers</h2><span id='topic+countxo'></span>

<h3>Description</h3>

<p>Estimate the number of crossovers in each meiosis in a backcross.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countxo(cross, chr = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="countxo_+3A_cross">cross</code></td>
<td>
<p>An object of class <code>cross</code>. (This must be a backcross.)
See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code> for details.</p>
</td></tr>
<tr><td><code id="countxo_+3A_chr">chr</code></td>
<td>
<p>Optional set of chromosomes across which to count crossovers.  If
NULL, the total number of crossovers, genome-wide, is counted.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This works only a backcross.  We use the internal function (within R/qtl)
<code>locate.xo</code>.
</p>


<h3>Value</h3>

<p>A vector with the estimated number of crossovers for each
individual.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+find.breaks">find.breaks()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)

# estimated number of crossovers on chr 1
nxo &lt;- countxo(bssbsb, chr=1)

# estimated number of crossovers genome-wide
nxo &lt;- countxo(bssbsb)

</code></pre>

<hr>
<h2 id='distance.given.two'>Distance between crossovers given there are two</h2><span id='topic+distance.given.two'></span>

<h3>Description</h3>

<p>Calculates the density of the distance between the crossovers on a meiotic
product, given that there are precisely two crossovers, for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distance.given.two(
  nu,
  L = 103,
  x = NULL,
  n = 400,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="distance.given.two_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_l">L</code></td>
<td>
<p>The length of the chromsome in cM.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="distance.given.two_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>
<p>The distribution of the distance from the start of the chromosome to the
first crossover is <code class="reqn">g^*(x;\nu) = 1 - F^*(x;\nu)</code> where <code class="reqn">F^*</code> is the cdf of <code class="reqn">f^*</code>.
</p>
<p>We calculate the distribution of the distance between crossovers on a
product with two crossovers by first calculating the joint distribution of
the location of the two crossovers, given that they both occur before L and
the third occurs after L, and then integrating out the location of the first
crossover.
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the distance (between 0
and <code>L</code>, in cM) at which the density was calculated and <code>f</code> is the
density.
</p>


<h3>Warning</h3>

<p><strong>We sometimes have difficulty with the numerical
integrals.  You may need to use large <code>min.subd</code> (e.g. 25) to get
accurate results.</strong>
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+location.given.one">location.given.one()</a></code>,
<code><a href="#topic+first.given.two">first.given.two()</a></code>,<code><a href="#topic+joint.given.two">joint.given.two()</a></code>,
<code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>, <code><a href="#topic+xoprob">xoprob()</a></code>,
<code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- distance.given.two(1, L=200, n=101)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,0.0122), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- distance.given.two(2.6, L=200, n=101)
lines(f2, col="blue", lwd=2)

## Not run: 
f3 &lt;- distance.given.two(4.3, L=200, n=101)
lines(f3, col="red", lwd=2)

f4 &lt;- distance.given.two(7.6, L=200, n=101)
lines(f4, col="green", lwd=2)

## End(Not run)

</code></pre>

<hr>
<h2 id='est.coi'>Estimate the coincidence function</h2><span id='topic+est.coi'></span>

<h3>Description</h3>

<p>Estimate the coincidence function from backcross data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est.coi(
  cross,
  chr = NULL,
  pos = NULL,
  window = 0,
  fill.method = c("imp", "argmax"),
  error.prob = 0.0000000001,
  map.function = c("haldane", "kosambi", "c-f", "morgan")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est.coi_+3A_cross">cross</code></td>
<td>
<p>Cross object; must be a backcross.  See
<code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code> for format details.</p>
</td></tr>
<tr><td><code id="est.coi_+3A_chr">chr</code></td>
<td>
<p>Chromosome to consider (only one is allowed).  If NULL, the
first chromosome is considered.</p>
</td></tr>
<tr><td><code id="est.coi_+3A_pos">pos</code></td>
<td>
<p>If provided, these are used as the marker positions.  (This could
be useful if you want to do things with respect to physical distance.)</p>
</td></tr>
<tr><td><code id="est.coi_+3A_window">window</code></td>
<td>
<p>Window size used to smooth the estimates.</p>
</td></tr>
<tr><td><code id="est.coi_+3A_fill.method">fill.method</code></td>
<td>
<p>Method used to impute missing data.</p>
</td></tr>
<tr><td><code id="est.coi_+3A_error.prob">error.prob</code></td>
<td>
<p>Genotyping error probability used in imputation of missing
data.</p>
</td></tr>
<tr><td><code id="est.coi_+3A_map.function">map.function</code></td>
<td>
<p>Map function used in imputation of missing data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The coincidence function is the probability of a recombination event in both
of two intervals, divided by the product of the two recombination fractions.
We estimate this as a function of the distance between the two intervals.
</p>
<p>Note that we first call <code><a href="qtl.html#topic+fill.geno">qtl::fill.geno()</a></code> to impute any missing
genotype data.
</p>


<h3>Value</h3>

<p>A data.frame containing the distance between intervals and the
corresponding estimate of the coincidence.  There are actually two columns
of estimates of the coincidence.  In the first estimate, we take a running
mean of each of the numerator and denominator and then divide.  In the
second estimate, we first take a ratio and then take a running mean.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in
genetic recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gammacoi">gammacoi()</a></code>, <code><a href="#topic+stahlcoi">stahlcoi()</a></code>, <code><a href="#topic+kfunc">kfunc()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
map1 &lt;- sim.map(103, n.mar=104, anchor=TRUE, include.x=FALSE, eq=TRUE)
x &lt;- sim.cross(map1, n.ind=2000, m=6, type="bc")

out &lt;- est.coi(x, window=5)
plot(coi1 ~ d, data=out, type="l", lwd=2, col="blue")
lines(coi2 ~ d, data=out, lwd=2, col="green")
lines(gammacoi(7), lwd=2, col="red", lty=2)

</code></pre>

<hr>
<h2 id='est.coi.um'>Estimate the coincidence as a function of micron distance</h2><span id='topic+est.coi.um'></span>

<h3>Description</h3>

<p>Estimate the coincidence as a function of micron distance, with
data on XO locations in microns plus SC length in microns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est.coi.um(
  xoloc,
  sclength,
  centromeres = NULL,
  group = NULL,
  intwindow = 0.05,
  coiwindow = NULL,
  intloc = NULL,
  coiloc = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est.coi.um_+3A_xoloc">xoloc</code></td>
<td>
<p>list of crossover locations (in microns) for each of several oocytes or spermatocytes.</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_sclength">sclength</code></td>
<td>
<p>vector of SC lengths (in microns).</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_centromeres">centromeres</code></td>
<td>
<p>vector of centromere locations (in microns). If NULL, taken to be <code>sclength/2</code>.</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_group">group</code></td>
<td>
<p>nominal vector of groups; the intensity function of
the crossover process will be estimated separately for each group,
but a joint coincidence function will be estimated.</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_intwindow">intwindow</code></td>
<td>
<p>Window size used to smooth the estimated intensity
function.</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_coiwindow">coiwindow</code></td>
<td>
<p>Window size used to smooth the estimated
coincidence function.</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_intloc">intloc</code></td>
<td>
<p>Locations at which to estimate the intensity
function, in the interval [0,1]</p>
</td></tr>
<tr><td><code id="est.coi.um_+3A_coiloc">coiloc</code></td>
<td>
<p>Values at which the coincidence function is to be
estimated, in microns, less than <code>max(sclength)</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The coincidence function is the probability of a recombination
event in both of two intervals, divided by the product of the two
intensity function for the two intervals.
</p>
<p>We estimate this as a function of the distance between the two
intervals in microns, taking account of varying SC lengths,.
</p>


<h3>Value</h3>

<p>A list containing the estimated coincidence (as a matrix
with two columns, micron distance and corresponding estimated
coincidence) and the estimated intensity functions (as a matrix
with <code>length(group)+1</code> columns (the locations at which the
intensity functions were estimated followed by the group-specific estimates).
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gammacoi">gammacoi()</a></code>, <code><a href="#topic+stahlcoi">stahlcoi()</a></code>,
<code><a href="#topic+kfunc">kfunc()</a></code>, <code><a href="#topic+est.coi">est.coi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simple example using data simulated with no crossover interference
ncells &lt;- 1000
L &lt;- 2                      # chr lengths in Morgans (constant here)
nchi &lt;- rpois(ncells, 2*L)  # number of chiasmata
xoloc &lt;- lapply(nchi, function(a) runif(a, 0, L)) # chi locations
coi &lt;- est.coi.um(xoloc, rep(L, ncells))

# plot estimated coincidence and intensity
#    (intensity is after scaling chromosome to length 1)
par(mfrow=c(2,1), las=1)
plot(coi$coincidence, type="l", lwd=2, ylim=c(0, max(coi$coincidence[,2])))
plot(coi$intensity, type="l", lwd=2, ylim=c(0, max(coi$intensity[,2])))

</code></pre>

<hr>
<h2 id='est.recrate'>Estimate recombination rate</h2><span id='topic+est.recrate'></span>

<h3>Description</h3>

<p>Obtain a smoothed estimate of the recombination rate along a chromosome,
using the cM and Mbp position of markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est.recrate(genmap, phymap, pos = NULL, window = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est.recrate_+3A_genmap">genmap</code></td>
<td>
<p>Vector of cM positions of markers, or a list of such vectors.</p>
</td></tr>
<tr><td><code id="est.recrate_+3A_phymap">phymap</code></td>
<td>
<p>Vector of Mbp positions of markers, or a list of such vectors;
same length as <code>genmap</code>.</p>
</td></tr>
<tr><td><code id="est.recrate_+3A_pos">pos</code></td>
<td>
<p>Vector of positions at which the recombination rate should be
estimated, or a list of such vectors.  If NULL, we use the physical
marker positions plus a grid with 4 positions per Mbp.</p>
</td></tr>
<tr><td><code id="est.recrate_+3A_window">window</code></td>
<td>
<p>Length of sliding window (in Mbp).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We assume constant recombination rate within each marker interval.
</p>


<h3>Value</h3>

<p>A data.frame containing the positions and estimate recombination
rates.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+est.coi">est.coi()</a></code>, <code><a href="#topic+intensity">intensity()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create equally-spaced map
pmap &lt;- sim.map(100, n.mar=51, anchor=TRUE, include.x=FALSE, eq.spacing=TRUE)

# simulate cross
x &lt;- sim.cross(pmap, type="bc", n.ind=501)

# estimate map for that cross
emap &lt;- est.map(x)

# empirical estimate of recombination rate
rr &lt;- est.recrate(emap[[1]], pmap[[1]], window=5)
plot(rr, type="l", lwd=2)

</code></pre>

<hr>
<h2 id='find.breaks'>Estimate crossover locations</h2><span id='topic+find.breaks'></span>

<h3>Description</h3>

<p>Estimate the locations of crossovers in a backcross.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.breaks(cross, chr = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="find.breaks_+3A_cross">cross</code></td>
<td>
<p>An object of class <code>cross</code>. (This must be a backcross,
RIL, or intercross.) See <code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code> for details.</p>
</td></tr>
<tr><td><code id="find.breaks_+3A_chr">chr</code></td>
<td>
<p>Optional set of chromosomes on which to look for crossovers.  If
NULL, all chromosomes are considered.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This works only a backcross, RIL, or intercross.  We use the function
<code><a href="qtl.html#topic+locateXO">qtl::locateXO()</a></code> in R/qtl.  Crossovers are estimated to be at the
midpoint of the interval between the nearest flanking typed markers.
</p>


<h3>Value</h3>

<p>If only one chromosome is considered, this is a list with one
component for each individual.  If multiple chromosomes were considered,
this is a list with one element for each chromosome, each of which is a list
with one element for each individual, as above.
</p>
<p>For backcrosses and RIL, the componenets for the individuals are
<code>numeric(0)</code> if there were no crossovers or a vector giving the
crossover locations.  The length of the chromosome (in cM) is saved as an
attribute.  (Note that the format is the same as the output of
<code><a href="#topic+simStahl">simStahl()</a></code>.)
</p>
<p>For an intercross, the components for the individuals are themselves lists
with all possible allocations of the crossovers to the two meiotic products;
each component of this list is itself a list with two components,
corresponding to the two meiotic products.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+convertxoloc">convertxoloc()</a></code>, <code><a href="#topic+fitGamma">fitGamma()</a></code>,
<code><a href="#topic+simStahl">simStahl()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)

# crossover locations on chromosome 1
xoloc1 &lt;- find.breaks(bssbsb, chr=1)

# crossover locations on all chromosomes
xoloc &lt;- find.breaks(bssbsb)

</code></pre>

<hr>
<h2 id='first.given.two'>Location of first crossover given there are two</h2><span id='topic+first.given.two'></span>

<h3>Description</h3>

<p>Calculates the density of the location of the first crossover on a random
meiotic product, given that there are precisely two crossovers, for the
gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>first.given.two(
  nu,
  L = 103,
  x = NULL,
  n = 400,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="first.given.two_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_l">L</code></td>
<td>
<p>The length of the chromsome in cM.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="first.given.two_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>
<p>The distribution of the distance from the start of the chromosome to the
first crossover is <code class="reqn">g^*(x;\nu) = 1 - F^*(x;\nu)</code> where <code class="reqn">F^*</code> is the cdf of <code class="reqn">f^*</code>.
</p>
<p>We calculate the distribution of the location of the first crossover in a
product with two crossovers by calculating the joint distribution of the
location of the two crossovers, given that they both occur before L and the
third occurs after L, and then integrating out the location of the second
crossover.
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the location (between 0
and <code>L</code>, in cM) at which the density was calculated and <code>f</code> is the
density.
</p>


<h3>Warning</h3>

<p><strong>We sometimes have difficulty with the numerical
integrals.  You may need to use large <code>min.subd</code> (e.g. 25) to get
accurate results.</strong>
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+location.given.one">location.given.one()</a></code>, <code><a href="#topic+distance.given.two">distance.given.two()</a></code>,
<code><a href="#topic+joint.given.two">joint.given.two()</a></code>, <code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>,
<code><a href="#topic+xoprob">xoprob()</a></code>, <code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- first.given.two(1, L=200, n=101)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,0.011), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- first.given.two(2.6, L=200, n=101)
lines(f2, col="blue", lwd=2)

## Not run: 
f3 &lt;- first.given.two(4.3, L=200, n=101)
lines(f3, col="red", lwd=2)

f4 &lt;- first.given.two(7.6, L=200, n=101)
lines(f4, col="green", lwd=2)

## End(Not run)

</code></pre>

<hr>
<h2 id='firstden'>Distance to a crossover</h2><span id='topic+firstden'></span>

<h3>Description</h3>

<p>Calculates the density of the distance from an arbitrary point to the next
crossover, for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>firstden(nu, L = 103, x = NULL, n = 400, max.conv = 25)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="firstden_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="firstden_+3A_l">L</code></td>
<td>
<p>Maximal distance (in cM) at which to calculate the density. Ignored
if <code>x</code> is specified.</p>
</td></tr>
<tr><td><code id="firstden_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="firstden_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="firstden_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>
<p>The distribution of the distance from an arbitrary point to the first
crossover is <code class="reqn">g^*(x;\nu) = 1 - F^*(x;\nu)</code>
where <code class="reqn">F^*</code> is the cdf of <code class="reqn">f^*</code>.
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the distance (between 0
and <code>L</code>, in cM) at which the density was calculated and <code>f</code> is the
density.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+location.given.one">location.given.one()</a></code>, <code><a href="#topic+first.given.two">first.given.two()</a></code>,
<code><a href="#topic+distance.given.two">distance.given.two()</a></code>, <code><a href="#topic+joint.given.two">joint.given.two()</a></code>,
<code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+xoprob">xoprob()</a></code>, <code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- firstden(1, L=200, n=201)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,0.012), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- firstden(2.6, L=200, n=201)
lines(f2, col="blue", lwd=2)

f3 &lt;- firstden(4.3, L=200, n=201)
lines(f3, col="red", lwd=2)

f4 &lt;- firstden(7.6, L=200, n=201)
lines(f4, col="green", lwd=2)

</code></pre>

<hr>
<h2 id='fitGamma'>Fit Gamma model</h2><span id='topic+fitGamma'></span>

<h3>Description</h3>

<p>Fit the gamma model for crossover interference to data on crossover
locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitGamma(
  d,
  censor = NULL,
  nu = NULL,
  lo = NULL,
  hi = NULL,
  se = FALSE,
  supint = FALSE,
  rescale = FALSE,
  drop = 1.5,
  tol = 0.00001,
  maxit = 1000,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10,
  h = 0.1,
  hstep = 1.5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitGamma_+3A_d">d</code></td>
<td>
<p>A vector of inter-crossover distances in cM.  This should include
distances from start of chromosome to first crossover, last crossover to end
of chromosome, and chromosome length, if there are no crossovers.
</p>
<p>Alternatively, this may be a matrix with the first column being the
distances and second column being the censoring types (<code>censor</code>).</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_censor">censor</code></td>
<td>
<p>A vector of the same length as <code>d</code>, indicating the
censoring type for each distance.  <code>0</code> = uncensored, <code>1</code> =
right-censored, <code>2</code> = initial crossover on chromosome, <code>3</code> = whole
chromosome.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_nu">nu</code></td>
<td>
<p>A vector of interference parameters (<code class="reqn">\nu</code>) at which to
calculate the log likelihood.  If NULL, <code>lo</code> and <code>hi</code> must be
specified.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_lo">lo</code></td>
<td>
<p>If <code>nu</code> is unspecified, <code>lo</code> indicates the lower value
of the interval in which to search for the MLE.  If <code>supint=TRUE</code>, this
should be below the lower limit of the support interval.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_hi">hi</code></td>
<td>
<p>If <code>nu</code> is unspecified, <code>hi</code> indicates the upper value
of the interval in which to search for the MLE.  If <code>supint=TRUE</code>, this
should be above the upper limit of the support interval.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_se">se</code></td>
<td>
<p>If TRUE and <code>nu</code> was not specified, an estimated SE (based on
the second derivative of the log likelihood) is estimated.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_supint">supint</code></td>
<td>
<p>If TRUE and <code>nu</code> was not specified, a likelihood support
interval is calculated, with <code>drop</code> being the amount to drop in log
(base 10).</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_rescale">rescale</code></td>
<td>
<p>If TRUE and <code>nu</code> was specified, re-scale the log
likelihoods so that the maximum is at 0.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_drop">drop</code></td>
<td>
<p>If <code>supint</code> was specified, this indicates the amount to
drop in log (base 10) for the likelihood support interval.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_tol">tol</code></td>
<td>
<p>Tolerance for converence to calculate the likelihood, SE, and
likelihood support interval.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations in estimating the SE and
likelihood support interval.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_h">h</code></td>
<td>
<p>Step used in estimating the second derivative of the log
likelihood.</p>
</td></tr>
<tr><td><code id="fitGamma_+3A_hstep">hstep</code></td>
<td>
<p>factor by which <code>h</code> is decreased in each iteration of the
estimation of the second derivative of the log likelihood.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Broman and Weber (2000) for details of the method.
</p>
<p>We use R's <code><a href="stats.html#topic+integrate">stats::integrate()</a></code> function for numerical integrals,
<code><a href="stats.html#topic+optimize">stats::optimize()</a></code> for optimizing the likelihood, and
<code><a href="stats.html#topic+uniroot">stats::uniroot()</a></code> for identifying the endpoints of the likelihood
support interval.
</p>


<h3>Value</h3>

<p>If <code>nu</code> is specified, we return a data frame with two columns:
<code>nu</code> and the corresponding log (base e) likelihood.  If
<code>rescale=TRUE</code>, the maximum log likelihood is subtracted off, so that
its maximum is at 0.
</p>
<p>If <code>lo</code> and <code>hi</code> is specified, the output contains a single row
with the MLE of <code class="reqn">\nu</code> and the corresponding log likelihood.  If
<code>se=TRUE</code>, we also include the estimated SE.  If <code>supint=TRUE</code>, we
include two additional rows with the lower and upper limits of the
likelihood support interval.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="qtl.html#topic+fitstahl">qtl::fitstahl()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)


xodist &lt;- convertxoloc(find.breaks(bssbsb, chr=1))

# plot a rough log likelihood curve
## Not run: out &lt;- fitGamma(xodist, nu=seq(1, 19, by=2))

plot(out, type="l", lwd=2)

# get MLE
## Not run: mle &lt;- fitGamma(xodist, lo=8, hi=12)

mle

abline(v=mle[1], h=mle[2], col="blue", lty=2)

# get MLE and SE
## Not run: mle &lt;- fitGamma(xodist, lo=9.5, hi=10.5, se=TRUE)

mle

# get MLE and 10^1.5 support interval
## Not run: int &lt;- fitGamma(xodist, lo=1, hi=20, supint=TRUE)

int
abline(v=mle[2:3,1], h=mle[2:3,2], col="red", lty=2)

</code></pre>

<hr>
<h2 id='fitStahl'>Fit Stahl model</h2><span id='topic+fitStahl'></span>

<h3>Description</h3>

<p>Fit the Stahl model for crossover interference to data on crossover
locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitStahl(
  xoloc,
  chrlen = NULL,
  nu = c(1, 20),
  p = 0.02,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fitStahl_+3A_xoloc">xoloc</code></td>
<td>
<p>A list of crossover locations (in cM), each component being a
vector of locations for a different meiotic product.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_chrlen">chrlen</code></td>
<td>
<p>Chromosome length (in cM), either of length 1 or the same
length as <code>xoloc</code>.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_nu">nu</code></td>
<td>
<p>Interference parameter (<code class="reqn">\nu</code>).  This should be a pair of
values to be used as endpoints to first do a 1-dimensional optimization with
<code class="reqn">p=0</code>.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_p">p</code></td>
<td>
<p>Starting value for the proportion of crossovers from the no
interference pathway, for the 2-dimensional optimization.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, print tracing information.  If &quot;...&quot; includes
<code>control</code>, this is ignored.</p>
</td></tr>
<tr><td><code id="fitStahl_+3A_...">...</code></td>
<td>
<p>Further arguments sent to <code><a href="stats.html#topic+optim">stats::optim()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Housworth and Stahl (2003) and Broman and Weber (2000) for details of
the method.
</p>
<p>We first use <code><a href="stats.html#topic+optimize">stats::optimize()</a></code> to find the MLE with the
contraint <code>p=0</code>, followed by use of <code><a href="stats.html#topic+optim">stats::optim()</a></code> to do a
2-dimensional optimization for the MLEs of the pair.
</p>


<h3>Value</h3>

<p>A vector with the estimates of <code class="reqn">\nu</code> (interference
parameter) and <code class="reqn">p</code> (proportion of crossovers coming from the no
interference pathway), the maximized log likelihood, the estimate of nu with
p constrained to be 0, the maximized log likelihood in this case, and the
log likelihood ratio for comparing the model with p allowed to vary freely
versus contrained to be 0.  (Note that it's the natural log of the
likelihood ratio, and not twice that.)
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Housworth, E. A. and Stahl, F. W. (2003) Crossover interference
in humans. <em>Am. J. Hum. Genet.</em> <strong>73</strong>, 188&ndash;197.
</p>
<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human crossover
interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitGamma">fitGamma()</a></code>, <code><a href="#topic+stahlLoglik">stahlLoglik()</a></code>,
<code><a href="#topic+simStahl">simStahl()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)


xoloc &lt;- find.breaks(bssbsb, chr=1)
L &lt;- attr(xoloc, "L")

# get MLE (limiting maximum iterations to 10, just for speed in this example)
## Not run: mle &lt;- fitStahl(xoloc, L, nu=c(9, 12), control=list(maxit=10))


</code></pre>

<hr>
<h2 id='gammacoi'>Coincidence function for the gamma model</h2><span id='topic+gammacoi'></span>

<h3>Description</h3>

<p>Calculates the coincidence function for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gammacoi(nu, L = 103, x = NULL, n = 400, max.conv = 25)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gammacoi_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="gammacoi_+3A_l">L</code></td>
<td>
<p>Maximal distance (in cM) at which to calculate the density. Ignored
if <code>x</code> is specified.</p>
</td></tr>
<tr><td><code id="gammacoi_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="gammacoi_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="gammacoi_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolution.  This should
be greater than the maximum number of chiasmata on the 4-strand bundle.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The coincidence function for the gamma model is <code class="reqn">C(x;\nu) =
\sum_{k=1}^{\infty} f_k(x;\nu)/2</code>.
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the distance (between 0
and <code>L</code>, in cM) at which the coicidence was calculated and
<code>coincidence</code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+stahlcoi">stahlcoi()</a></code>, <code><a href="#topic+location.given.one">location.given.one()</a></code>,
<code><a href="#topic+first.given.two">first.given.two()</a></code>, <code><a href="#topic+distance.given.two">distance.given.two()</a></code>,
<code><a href="#topic+joint.given.two">joint.given.two()</a></code>, <code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>,
<code><a href="#topic+xoprob">xoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- gammacoi(1, L=200)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,1.25), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- gammacoi(2.6, L=200)
lines(f2, col="blue", lwd=2)

f3 &lt;- gammacoi(4.3, L=200)
lines(f3, col="red", lwd=2)

f4 &lt;- gammacoi(7.6, L=200)
lines(f4, col="green", lwd=2)

</code></pre>

<hr>
<h2 id='intensity'>Estimate intensity function</h2><span id='topic+intensity'></span>

<h3>Description</h3>

<p>Estimate intensity function for a chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intensity(cross, chr = NULL, window = 2.5, ncalc = 500)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intensity_+3A_cross">cross</code></td>
<td>
<p>Cross object; must be a backcross.  See
<code><a href="qtl.html#topic+read.cross">qtl::read.cross()</a></code> for format details.</p>
</td></tr>
<tr><td><code id="intensity_+3A_chr">chr</code></td>
<td>
<p>Chromosome to consider (only one is allowed).  If NULL, the
first chromosome is considered.</p>
</td></tr>
<tr><td><code id="intensity_+3A_window">window</code></td>
<td>
<p>Window size</p>
</td></tr>
<tr><td><code id="intensity_+3A_ncalc">ncalc</code></td>
<td>
<p>Total number of points for calculations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with columns <code>position</code> and <code>intensity</code>.  The
input argument <code>window</code> is kept as an attribute.
</p>


<h3>Author(s)</h3>

<p>Il youp Kwak
</p>


<h3>See Also</h3>

<p><code><a href="#topic+coincidence">coincidence()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
map1 &lt;- sim.map(103, n.mar=104, anchor=TRUE, include.x=FALSE, eq=TRUE)
x &lt;- sim.cross(map1, n.ind=2000, m=6, type="bc")

out &lt;- intensity(x)
plot(out, type="l", lwd=2, ylim=c(0, max(out[,2])))

</code></pre>

<hr>
<h2 id='ioden'>Distance between crossovers</h2><span id='topic+ioden'></span>

<h3>Description</h3>

<p>Calculates the density of the distance from a given crossover to the next
crossover, for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ioden(nu, L = 103, x = NULL, n = 400, max.conv = 25)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ioden_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="ioden_+3A_l">L</code></td>
<td>
<p>Maximal distance (in cM) at which to calculate the density. Ignored
if <code>x</code> is specified.</p>
</td></tr>
<tr><td><code id="ioden_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="ioden_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="ioden_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the distance (between 0
and <code>L</code>, in cM) at which the density was calculated and <code>f</code> is the
density.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+location.given.one">location.given.one()</a></code>, <code><a href="#topic+first.given.two">first.given.two()</a></code>,
<code><a href="#topic+distance.given.two">distance.given.two()</a></code>, <code><a href="#topic+joint.given.two">joint.given.two()</a></code>,
<code><a href="#topic+firstden">firstden()</a></code>, <code><a href="#topic+xoprob">xoprob()</a></code>, <code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- ioden(1, L=200, n=201)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,0.014), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- ioden(2.6, L=200, n=201)
lines(f2, col="blue", lwd=2)

f3 &lt;- ioden(4.3, L=200, n=201)
lines(f3, col="red", lwd=2)

f4 &lt;- ioden(7.6, L=200, n=201)
lines(f4, col="green", lwd=2)

</code></pre>

<hr>
<h2 id='joint.given.two'>Crossover locations given there are two</h2><span id='topic+joint.given.two'></span>

<h3>Description</h3>

<p>Calculates the joint density of the crossover locations on a random meiotic
product, given that there are precisely two crossovers, for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>joint.given.two(
  nu,
  L = 103,
  x = NULL,
  y = NULL,
  n = 20,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="joint.given.two_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_l">L</code></td>
<td>
<p>The length of the chromsome in cM.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_x">x</code></td>
<td>
<p>If specified, locations of the first crossover.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_y">y</code></td>
<td>
<p>If specified, locations of the second crossover.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> and
<code>y</code> are specified.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="joint.given.two_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>
<p>The distribution of the distance from the start of the chromosome to the
first crossover is <code class="reqn">g^*(x;\nu) = 1 - F^*(x;\nu)</code> where <code class="reqn">F^*</code> is the cdf of <code class="reqn">f^*</code>.
</p>


<h3>Value</h3>

<p>A data frame with three columns: <code>x</code> and <code>y</code> are the
locations (between 0 and <code>L</code>, in cM) at which the density was
calculated and <code>f</code> is the density.
</p>


<h3>Warning</h3>

<p><strong>We sometimes have difficulty with the numerical
integrals.  You may need to use large <code>min.subd</code> (e.g. 25) to get
accurate results.</strong>
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+location.given.one">location.given.one()</a></code>, <code><a href="#topic+distance.given.two">distance.given.two()</a></code>,
<code><a href="#topic+first.given.two">first.given.two()</a></code>, <code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>,
<code><a href="#topic+xoprob">xoprob()</a></code>, <code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Calculate the distribution of the average of the crossover locations,
# given that there are two and that they are separated by 20 cM
# (for a chromosome of length 200 cM)
L &lt;- 200
d &lt;- 20
x &lt;- seq(0, L-d, by=0.5)
y &lt;- x+d

f &lt;- joint.given.two(4.3, L=L, x, y)
f$f &lt;- f$f / distance.given.two(4.3, L, d)$f
plot((f$x+f$y)/2, f$f, type="l", xlim=c(0, L), ylim=c(0,max(f$f)),
     lwd=2, xlab="Average location", ylab="Density")
abline(v=c(d/2,L-d/2), h=1/(L-d), lty=2, lwd=2)

</code></pre>

<hr>
<h2 id='kfunc'>estimate Ripley's K function</h2><span id='topic+kfunc'></span>

<h3>Description</h3>

<p>estimate the 1-d version of Ripley's K function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kfunc(
  x,
  d = seq(0, 100, by = 0.1),
  lengths = NULL,
  exclude = 0,
  tol = 0.000001
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kfunc_+3A_x">x</code></td>
<td>
<p>list with sorted locations of the data</p>
</td></tr>
<tr><td><code id="kfunc_+3A_d">d</code></td>
<td>
<p>values at which to calculate the function</p>
</td></tr>
<tr><td><code id="kfunc_+3A_lengths">lengths</code></td>
<td>
<p>lengths of segments studied</p>
</td></tr>
<tr><td><code id="kfunc_+3A_exclude">exclude</code></td>
<td>
<p>distance to exclude</p>
</td></tr>
<tr><td><code id="kfunc_+3A_tol">tol</code></td>
<td>
<p>tolerance value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with <code>d</code>, <code>k</code>, and <code>se</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gammacoi">gammacoi()</a></code>, <code><a href="#topic+stahlcoi">stahlcoi()</a></code>, <code><a href="#topic+coincidence">coincidence()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>L &lt;- 103
n &lt;- 2000
map1 &lt;- sim.map(L, n.mar=104, anchor=TRUE, include.x=FALSE, eq=TRUE)
x &lt;- sim.cross(map1, n.ind=n, m=6, type="bc")

xoloc &lt;- find.breaks(x)

d &lt;- seq(0, 100, by=0.1)[-1]
kf &lt;- kfunc(xoloc, d=d, lengths=rep(L, n))

plot(k ~ d, data=kf, type="n", yaxs="i", xaxs="i", las=1,
     ylim=c(0, max(kf$k + kf$se)))
polygon(c(kf$d, rev(kf$d)), c(kf$k + kf$se, rev(kf$k-kf$se)),
        border=NA, col="#add8e650")
lines(k ~ d, data=kf)

</code></pre>

<hr>
<h2 id='location.given.one'>Location of crossover given there is one</h2><span id='topic+location.given.one'></span>

<h3>Description</h3>

<p>Calculates the density of the location of the crossover on a random meiotic
product, given that there is precisely one crossover, for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>location.given.one(
  nu,
  L = 103,
  x = NULL,
  n = 400,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="location.given.one_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_l">L</code></td>
<td>
<p>The length of the chromsome in cM.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="location.given.one_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>
<p>The distribution of the distance from the start of the chromosome to the
first crossover is <code class="reqn">g^*(x;\nu) = 1 - F^*(x;\nu)</code> where <code class="reqn">F^*</code> is the cdf of <code class="reqn">f^*</code>.
</p>
<p>We calculate the distribution of the location of the crossover on a product
with a single crossover as the convolution of <code class="reqn">g^*</code> with itself, and
then rescaled to integrate to 1 on the interval (0,L).
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the location (between 0
and <code>L</code>, in cM) at which the density was calculated and <code>f</code> is the
density.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+first.given.two">first.given.two()</a></code>, <code><a href="#topic+distance.given.two">distance.given.two()</a></code>,
<code><a href="#topic+joint.given.two">joint.given.two()</a></code>, <code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>,
<code><a href="#topic+xoprob">xoprob()</a></code>, <code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- location.given.one(1, L=200, n=201)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,0.006), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- location.given.one(2.6, L=200, n=201)
lines(f2, col="blue", lwd=2)

f3 &lt;- location.given.one(4.3, L=200, n=201)
lines(f3, col="red", lwd=2)

f4 &lt;- location.given.one(7.6, L=200, n=201)
lines(f4, col="green", lwd=2)

</code></pre>

<hr>
<h2 id='recrate2scanone'>Convert recrate to scanone format</h2><span id='topic+recrate2scanone'></span>

<h3>Description</h3>

<p>Convert the result of <code><a href="#topic+est.recrate">est.recrate()</a></code> to the format
output by R/qtl's <code><a href="qtl.html#topic+scanone">qtl::scanone()</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recrate2scanone(recrate, phymap = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="recrate2scanone_+3A_recrate">recrate</code></td>
<td>
<p>A list of results from <code><a href="#topic+est.recrate">est.recrate()</a></code></p>
</td></tr>
<tr><td><code id="recrate2scanone_+3A_phymap">phymap</code></td>
<td>
<p>A list of vectors of Mbp positions of markers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with class <code>"scanone"</code>, in the format output by <code><a href="qtl.html#topic+scanone">qtl::scanone()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+est.recrate">est.recrate()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pmap &lt;- sim.map(100, n.mar=51, anchor=TRUE, include.x=FALSE, eq.spacing=TRUE)

# simulate cross
x &lt;- sim.cross(pmap, type="bc", n.ind=501)

# estimate map for that cross
emap &lt;- est.map(x)

# empirical estimate of recombination rate
rr &lt;- est.recrate(emap[[1]], pmap[[1]], window=5)

# make it a list (one component per chromosome, but here just the one chromosome)
rr &lt;- list("1"=rr)

# convert to scanone output and plot
rr_scanone &lt;- recrate2scanone(rr)
plot(rr_scanone)

</code></pre>

<hr>
<h2 id='simStahl'>Simulate crossover locations under the Stahl model</h2><span id='topic+simStahl'></span>

<h3>Description</h3>

<p>Simulate crossover locations under the Stahl model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simStahl(
  n.sim,
  nu = 1,
  p = 0,
  L = 100,
  obligate_chiasma = FALSE,
  n.bins4start = 10000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simStahl_+3A_n.sim">n.sim</code></td>
<td>
<p>Number of meiotic products to simulate.</p>
</td></tr>
<tr><td><code id="simStahl_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="simStahl_+3A_p">p</code></td>
<td>
<p>The proportion of chiasmata coming from the no-interference
mechanism.</p>
</td></tr>
<tr><td><code id="simStahl_+3A_l">L</code></td>
<td>
<p>Chromosome length (in cM).</p>
</td></tr>
<tr><td><code id="simStahl_+3A_obligate_chiasma">obligate_chiasma</code></td>
<td>
<p>Require an obligate chiasma (requires <code>nu</code> to
be an integer; if nu is not an integer, it is rounded.</p>
</td></tr>
<tr><td><code id="simStahl_+3A_n.bins4start">n.bins4start</code></td>
<td>
<p>We approximate the distribution of the location of the
first crossover from the mechanism exhibiting interference using a even grid
with this many bins. (Only if <code>nu</code> is not an integer.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Stahl model is an extension to the gamma model, in which chiasmata occur
according to two independent mechanisms.  A proportion <code class="reqn">p</code> come from a
mechanism exhibiting no interference, and a proportion 1-<code class="reqn">p</code> come from a
mechanism in which chiasma locations follow a gamma model with interference
parameter <code class="reqn">\nu</code>.
</p>


<h3>Value</h3>

<p>A vector of length <code>n.sim</code>, each element being empty (for
products with no crossovers) or a vector of crossover locations, in cM.  An
attribute, <code>L</code>, contains the chromosome length in cM.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Copenhaver, G. P., Housworth, E. A. and Stahl, F. W. (2002)
Crossover interference in Arabidopsis. <em>Genetics</em> <strong>160</strong>,
1631&ndash;1639.
</p>
<p>Housworth, E. A. and Stahl, F. W. (2003) Crossover interference in humans.
<em>Am J Hum Genet</em> <strong>73</strong>, 188&ndash;197.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fitGamma">fitGamma()</a></code>, <code><a href="qtl.html#topic+sim.cross">qtl::sim.cross()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# simulations with no interference, chromosome of length 80 cM
xoNI &lt;- simStahl(100, nu=1, p=0, L=80)

# simulations under gamma model with nu=7.6
xogamma &lt;- simStahl(100, nu=7.6, p=0, L=80)

# simulations under Stahl model with nu=7.6, p=0.1
xostahl &lt;- simStahl(100, nu=7.6, p=0.1, L=80)

# simulations under chi-square model with nu=11 (m=10) and obligate chiasma
xo_oblchi &lt;- simStahl(100, nu=11, p=0, L=80, obligate_chiasma=TRUE)

# simulations under Stahl model with nu=11, p=0.1, and obligate chiasma
xo_oblchi_stahl &lt;- simStahl(100, nu=11, p=0.1, L=80, obligate_chiasma=TRUE)
</code></pre>

<hr>
<h2 id='stahlcoi'>Coincidence function for the Stahl model</h2><span id='topic+stahlcoi'></span>

<h3>Description</h3>

<p>Calculates the coincidence function for the Stahl model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stahlcoi(nu, p = 0, L = 103, x = NULL, n = 400, max.conv = 25)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stahlcoi_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="stahlcoi_+3A_p">p</code></td>
<td>
<p>The proportion of chiasmata coming from the no-interference
mechanism.</p>
</td></tr>
<tr><td><code id="stahlcoi_+3A_l">L</code></td>
<td>
<p>Maximal distance (in cM) at which to calculate the density. Ignored
if <code>x</code> is specified.</p>
</td></tr>
<tr><td><code id="stahlcoi_+3A_x">x</code></td>
<td>
<p>If specified, points at which to calculate the density.</p>
</td></tr>
<tr><td><code id="stahlcoi_+3A_n">n</code></td>
<td>
<p>Number of points at which to calculate the density.  The points
will be evenly distributed between 0 and <code>L</code>. Ignored if <code>x</code> is
specified.</p>
</td></tr>
<tr><td><code id="stahlcoi_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolution.  This should
be greater than the maximum number of chiasmata on the 4-strand bundle.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Stahl model is an extension to the gamma model, in which chiasmata occur
according to two independent mechanisms.  A proportion <code class="reqn">p</code> come from a
mechanism exhibiting no interference, and a proportion 1-<code class="reqn">p</code> come from a
mechanism in which chiasma locations follow a gamma model with interference
parameter <code class="reqn">\nu</code>.
</p>
<p>Let <code class="reqn">f(x;\nu,\lambda)</code> denote the density of a gamma
random variable with parameters shape=<code class="reqn">\nu</code> and
rate=<code class="reqn">\lambda</code>.
</p>
<p>The coincidence function for the Stahl model is <code class="reqn">C(x;\nu,p) = [p +
\sum_{k=1}^{\infty} f(x;k\nu, </code><code class="reqn"> 2(1-p)\nu)]/2</code>.
</p>


<h3>Value</h3>

<p>A data frame with two columns: <code>x</code> is the distance (between 0
and <code>L</code>, in cM) at which the coicidence was calculated and
<code>coincidence</code>.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Copenhaver, G. P., Housworth, E. A. and Stahl, F. W. (2002)
Crossover interference in Arabidopsis. <em>Genetics</em> <strong>160</strong>,
1631&ndash;1639.
</p>
<p>Housworth, E. A. and Stahl, F. W. (2003) Crossover interference in humans.
<em>Am J Hum Genet</em> <strong>73</strong>, 188&ndash;197.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gammacoi">gammacoi()</a></code>, <code><a href="#topic+location.given.one">location.given.one()</a></code>,
<code><a href="#topic+first.given.two">first.given.two()</a></code>, <code><a href="#topic+distance.given.two">distance.given.two()</a></code>,
<code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>, <code><a href="#topic+xoprob">xoprob()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
f1 &lt;- stahlcoi(1, p=0, L=200)
plot(f1, type="l", lwd=2, las=1,
     ylim=c(0,1.25), yaxs="i", xaxs="i", xlim=c(0,200))

f2 &lt;- stahlcoi(2.6, p=0, L=200)
lines(f2, col="blue", lwd=2)

f2s &lt;- stahlcoi(2.6, p=0.1, L=200)
lines(f2s, col="blue", lwd=2, lty=2)

f3 &lt;- stahlcoi(4.3, p=0, L=200)
lines(f3, col="red", lwd=2)

f3s &lt;- stahlcoi(4.3, p=0.1, L=200)
lines(f3s, col="red", lwd=2, lty=2)

f4 &lt;- stahlcoi(7.6, p=0, L=200)
lines(f4, col="green", lwd=2)

f4s &lt;- stahlcoi(7.6, p=0.1, L=200)
lines(f4s, col="green", lwd=2, lty=2)

</code></pre>

<hr>
<h2 id='stahlLoglik'>Calculate log likelihood for Stahl model</h2><span id='topic+stahlLoglik'></span>

<h3>Description</h3>

<p>Calculate the log likelihood for the Stahl model for varying parameters,
with data on crossover locations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stahlLoglik(
  xoloc,
  chrlen = NULL,
  nu,
  p,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stahlLoglik_+3A_xoloc">xoloc</code></td>
<td>
<p>A list of crossover locations (in cM), each component being a
vector of locations for a different meiotic product.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_chrlen">chrlen</code></td>
<td>
<p>Chromosome length (in cM), either of length 1 or the same
length as <code>xoloc</code>.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_nu">nu</code></td>
<td>
<p>A vector of interference parameters (<code class="reqn">\nu</code>) at which to
calculate the log likelihood.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_p">p</code></td>
<td>
<p>A vector of parameter values for the proportion of crossovers from
the no interference pathway.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="stahlLoglik_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Housworth and Stahl (2003) and Broman and Weber (2000) for details of
the method.
</p>
<p>If neither <code>nu</code> nor <code>p</code> has length 1, they both must have the same
length.  If one has length 1 and the other does not, the one with length 1
is repeated so that they both have the same length.
</p>


<h3>Value</h3>

<p>A vector of log likelihoods.
</p>
<p>The corresponding values of nu and p are saved as attributes.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Housworth, E. A. and Stahl, F. W. (2003) Crossover interference
in humans. <em>Am. J. Hum. Genet.</em> <strong>73</strong>, 188&ndash;197.
</p>
<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human crossover
interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>


<h3>See Also</h3>

<p><code><a href="qtl.html#topic+fitstahl">qtl::fitstahl()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bssbsb)
xoloc &lt;- find.breaks(bssbsb, chr=1)

loglik &lt;- stahlLoglik(xoloc, nu=4, p=c(0.05, 0.1, 0.15))

</code></pre>

<hr>
<h2 id='xoiversion'>Installed version of R/xoi</h2><span id='topic+xoiversion'></span>

<h3>Description</h3>

<p>Print the version number of the currently installed version of R/xoi.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xoiversion()
</code></pre>


<h3>Value</h3>

<p>A character string with the version number of the currently
installed version of R/xoi.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xoiversion()

</code></pre>

<hr>
<h2 id='xoprob'>Distribution of number of crossovers</h2><span id='topic+xoprob'></span>

<h3>Description</h3>

<p>Calculates the probability of 0, 1, 2, or &gt;2 crossovers for a chromosome of
a given length, for the gamma model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xoprob(
  nu,
  L = 103,
  max.conv = 25,
  integr.tol = 0.00000001,
  max.subd = 1000,
  min.subd = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="xoprob_+3A_nu">nu</code></td>
<td>
<p>The interference parameter in the gamma model.</p>
</td></tr>
<tr><td><code id="xoprob_+3A_l">L</code></td>
<td>
<p>Length of the chromosome (in cM).</p>
</td></tr>
<tr><td><code id="xoprob_+3A_max.conv">max.conv</code></td>
<td>
<p>Maximum limit for summation in the convolutions to get
inter-crossover distance distribution from the inter-chiasma distance
distributions.  This should be greater than the maximum number of chiasmata
on the 4-strand bundle.</p>
</td></tr>
<tr><td><code id="xoprob_+3A_integr.tol">integr.tol</code></td>
<td>
<p>Tolerance for convergence of numerical integration.</p>
</td></tr>
<tr><td><code id="xoprob_+3A_max.subd">max.subd</code></td>
<td>
<p>Maximum number of subdivisions in numerical integration.</p>
</td></tr>
<tr><td><code id="xoprob_+3A_min.subd">min.subd</code></td>
<td>
<p>Minimum number of subdivisions in numerical integration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">f(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">\nu</code> and rate=<code class="reqn">2\nu</code>, and let
<code class="reqn">f_k(x;\nu)</code> denote the density of a gamma random variable
with parameters shape=<code class="reqn">k \nu</code> and rate=<code class="reqn">2\nu</code>.
</p>
<p>The distribution of the distance from one crossover to the next is
<code class="reqn">f^*(x;\nu) = \sum_{k=1}^{\infty} f_k(x;\nu)/2^k</code>.
</p>
<p>The distribution of the distance from the start of the chromosome to the
first crossover is <code class="reqn">g^*(x;\nu) = 1 - F^*(x;\nu)</code> where <code class="reqn">F^*</code> is the cdf of <code class="reqn">f^*</code>.
</p>
<p>We calculate the desired probabilities by numerical integration.
</p>


<h3>Value</h3>

<p>A vector of length 4, giving the probabilities of 0, 1, 2, or &gt;2
crossovers, respectively, on a chromosome of length <code>L</code> cM.
</p>


<h3>Author(s)</h3>

<p>Karl W Broman, <a href="mailto:broman@wisc.edu">broman@wisc.edu</a>
</p>


<h3>References</h3>

<p>Broman, K. W. and Weber, J. L. (2000) Characterization of human
crossover interference. <em>Am. J. Hum. Genet.</em> <strong>66</strong>, 1911&ndash;1926.
</p>
<p>Broman, K. W., Rowe, L. B., Churchill, G. A. and Paigen, K. (2002) Crossover
interference in the mouse. <em>Genetics</em> <strong>160</strong>, 1123&ndash;1131.
</p>
<p>McPeek, M. S. and Speed, T. P. (1995) Modeling interference in genetic
recombination.  <em>Genetics</em> <strong>139</strong>, 1031&ndash;1044.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+location.given.one">location.given.one()</a></code>, <code><a href="#topic+first.given.two">first.given.two()</a></code>,
<code><a href="#topic+distance.given.two">distance.given.two()</a></code>, <code><a href="#topic+joint.given.two">joint.given.two()</a></code>,
<code><a href="#topic+ioden">ioden()</a></code>, <code><a href="#topic+firstden">firstden()</a></code>, <code><a href="#topic+gammacoi">gammacoi()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
xoprob(1, L=103)
xoprob(4.3, L=103)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
