<!DOCTYPE html><html><head><title>Help for package yaConsensus</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {yaConsensus}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#consensus.diss'><p>Computes the consensus dissimilarity matrix.</p>
</a></li>
<li><a href='#plot.yaConsensus'><p>Plot the consensus dissimilarity matrix and provide the consensus clustering.</p>
</a></li>
<li><a href='#yaConsensus'><p>yaConsensus computes a number of hierarchical clusterings by sampling either samples or features.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Consensus Clustering of Omic Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-06-30</td>
</tr>
<tr>
<td>Description:</td>
<td>Procedures to perform consensus clustering starting from a dissimilarity matrix or a data matrix. It's allowed to select if the subsampling has to be by samples or features. In case of computational heavy load, the procedures can run in parallel.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), foreach (&ge; 1.5.1), pheatmap (&ge; 1.0.12),
doParallel (&ge; 1.0.16)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/stefanoMP/yaConsensus">https://github.com/stefanoMP/yaConsensus</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-30 13:19:29 UTC; stefano</td>
</tr>
<tr>
<td>Author:</td>
<td>Stefano Maria Pagnotta
    <a href="https://orcid.org/0000-0002-8298-9777"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre,
    cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stefano Maria Pagnotta &lt;pagnotta@unisannio.it&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-01 07:40:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='consensus.diss'>Computes the consensus dissimilarity matrix.

</h2><span id='topic+consensus.diss'></span>

<h3>Description</h3>

<p>Computes the consensus dissimilarity according to the algorithm of Monti et al. (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>consensus.diss(cclusters, similarity = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="consensus.diss_+3A_cclusters">cclusters</code></td>
<td>
<p>a matrix of integers where the column are the samples, and the rows are different clusterings of the samples.

</p>
</td></tr>
<tr><td><code id="consensus.diss_+3A_similarity">similarity</code></td>
<td>
<p>a logical value signaling if the similarity matrix is required. 

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In any row of the ccluster matrix, the value 0 means that the corresponding sample is not assigned to any cluster. In this case, the dissimilarity is computed consistently.

</p>


<h3>Value</h3>






<p>An object of the 'dist' class.
</p>


<h3>Author(s)</h3>

<p>Stefano M. Pagnotta
</p>


<h3>References</h3>

<p>Monti et al. (2003) - Consensus Clustering: A Resampling-Based Method for Class Discovery and Visualization of Gene Expression Microarray Data - Machine Learning 52(1-2):91-118 &lt;DOI: 10.1023/A:1023949509487&gt;
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+dist">dist</a></code>

</p>


<h3>Examples</h3>

<pre><code class='language-R'>clusters &lt;- rep(1:3, c(3, 9, 18))
clusterings &lt;- matrix(NA, ncol = 30, nrow = 50)
for(k in 1:50) clusterings[k,] &lt;- sample(clusters)
ddist &lt;- consensus.diss(clusterings)
class(ddist)
attr(ddist, "method")
</code></pre>

<hr>
<h2 id='plot.yaConsensus'>Plot the consensus dissimilarity matrix and provide the consensus clustering.

</h2><span id='topic+plot.yaConsensus'></span>

<h3>Description</h3>

<p>This function processes the ouput of yaConsensus and acts as a wrapper to the pheatmap function.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'yaConsensus'
plot(x, G = 2, annotation = NULL, annotation.colorCode = NULL, 
     matching_clustering = NULL, consensus_colors = NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.yaConsensus_+3A_x">x</code></td>
<td>
<p>an object coming from yaConsensus().

</p>
</td></tr>
<tr><td><code id="plot.yaConsensus_+3A_g">G</code></td>
<td>
<p>an integer value indicating the number of clusters required for the consensus clustering. Default is 2.

</p>
</td></tr>
<tr><td><code id="plot.yaConsensus_+3A_annotation">annotation</code></td>
<td>
<p>a data frame where the variables are annotations (as labels) of samples. The row-names have to match the names of the samples.

</p>
</td></tr>
<tr><td><code id="plot.yaConsensus_+3A_annotation.colorcode">annotation.colorCode</code></td>
<td>
<p>a string named list of color names. The names have to be values stored in the annotation data-frame.

</p>
</td></tr>
<tr><td><code id="plot.yaConsensus_+3A_matching_clustering">matching_clustering</code></td>
<td>
<p>a string value matching one of the annotation valiables in the annotation data-frame. The function tries to match at best the color coding of the selected variable in the data-frame.

</p>
</td></tr>
<tr><td><code id="plot.yaConsensus_+3A_consensus_colors">consensus_colors</code></td>
<td>
<p>a list of color provided to annotate the consensus clustering. If provided, the matching_clustering is overrided.

</p>
</td></tr>
<tr><td><code id="plot.yaConsensus_+3A_...">...</code></td>
<td>
<p>parameters compatible with pheatmap function. 

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the slot 'statistics', the function returns the same statistics of yaConsensus().

</p>


<h3>Value</h3>



<p>A named list with the following slots: 
</p>
<table>
<tr><td><code>annotation</code></td>
<td>
<p>a data frame. It is the same given in input, with 'consensus' and 'consensus.col' more variables.</p>
</td></tr>
<tr><td><code>ann_colors</code></td>
<td>
<p>a named list of colors associated with each variable in the annotation data-frame.</p>
</td></tr>
<tr><td><code>hclust</code></td>
<td>
<p>an object of hclust clust. It's the result of the hclust() applied to the consensus dissimilarity with the complete linkage.</p>
</td></tr>
<tr><td><code>statistics</code></td>
<td>
<p>see Note</p>
</td></tr>
</table>


<h3>Note</h3>

<p>In case an 'annotation' is provided, with summary(), additional statistics are provided. 
</p>
<p>The variable specified as given in summary(..., given = &quot;some clustering&quot;) is assumed as a theoretical clustering, while the consensus clustering is the empirical one. The entropy accuracy, precision, and average (Risso and Pagnotta, 2021) are computed with the former assumptions. summary() itself returns all the statistics. &quot;some clustering&quot; has to be one of the column of the annotation data.frame. 

</p>


<h3>Author(s)</h3>

<p>Stefano M. Pagnotta
</p>


<h3>References</h3>

<p>Risso and Pagnotta (2021) - Per-sample standardization and asymmetric winsorization lead to accurate clustering of RNA-seq expression profiles - Bioinformatics, btab091, &lt;DOI: 10.1093/bioinformatics/btab091&gt;

</p>


<h3>See Also</h3>

<p><code>pheatmap</code>, <code><a href="#topic+yaConsensus">yaConsensus</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># see the examples in yaConsensus help.
</code></pre>

<hr>
<h2 id='yaConsensus'>yaConsensus computes a number of hierarchical clusterings by sampling either samples or features.
</h2><span id='topic+yaConsensus'></span>

<h3>Description</h3>

<p>This function mainly generates a list of &quot;hclust&quot; objects for downstream analysis.

</p>


<h3>Usage</h3>

<pre><code class='language-R'>yaConsensus(ddata, runs = 1000, epsilon = 0.65, is_by_sample = TRUE, 
            distMethod = "euclidean", hcMethod = "ward.D2", prefix = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yaConsensus_+3A_ddata">ddata</code></td>
<td>
<p>either a data matrix (samples in rows, and features in columns), or a &quot;dist&quot; object.

</p>
</td></tr>
<tr><td><code id="yaConsensus_+3A_runs">runs</code></td>
<td>
<p>an integer value for the number of samplings.

</p>
</td></tr>
<tr><td><code id="yaConsensus_+3A_epsilon">epsilon</code></td>
<td>
<p>a real value indicating the sampling rate.

</p>
</td></tr>
<tr><td><code id="yaConsensus_+3A_is_by_sample">is_by_sample</code></td>
<td>
<p>a logical value indicating if the sampling is by samples (TRUE) or features (FALSE).

</p>
</td></tr>
<tr><td><code id="yaConsensus_+3A_distmethod">distMethod</code></td>
<td>
<p>a character indicating the kind of distance for the inner clustering. It can be any of the methods from the <code><a href="stats.html#topic+dist">dist</a></code> function.  

</p>
</td></tr>
<tr><td><code id="yaConsensus_+3A_hcmethod">hcMethod</code></td>
<td>
<p>a character indicating the linkage mathod of the inner clustering. It can be any of the methods from the <code><a href="stats.html#topic+hclust">hclust</a></code> function.

</p>
</td></tr>
<tr><td><code id="yaConsensus_+3A_prefix">prefix</code></td>
<td>
<p>string specifying a prefix to store the results in a .RData file.  

</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can run sequentially or in parallel. In this case, it is necessary to register a cluster of CPUs according to doParallel protocol.  
</p>
<p>To get the consensus clustering, the output of the function has to be processed with the plot() function. The consensus dissimilarity follows from the algorithm of Monti et al. (2003). The consensus clustering is from a hierarchical procedure (hclust) with &quot;complete&quot; linkage (outer hc method). 
</p>



<h3>Value</h3>



<p>A named list with the following slots: 
</p>
<table>
<tr><td><code>distMethod</code></td>
<td>
<p>matches the input</p>
</td></tr>
<tr><td><code>hcMethod</code></td>
<td>
<p>matches the input</p>
</td></tr>
<tr><td><code>lables</code></td>
<td>
<p>a string list with the names of the samples</p>
</td></tr>
<tr><td><code>bySample</code></td>
<td>
<p>matches 'is_by_sample' input parameter</p>
</td></tr>
<tr><td><code>epsilon</code></td>
<td>
<p>matches the input</p>
</td></tr>
<tr><td><code>subsetDimension</code></td>
<td>
<p>actual dimension of the subsets</p>
</td></tr>
<tr><td><code>runs</code></td>
<td>
<p>matches the input</p>
</td></tr>
<tr><td><code>hclust</code></td>
<td>
<p>a list of 'hclust' objects</p>
</td></tr>
<tr><td><code>elapsed_time</code></td>
<td>
<p>time (in seconds) required</p>
</td></tr>
<tr><td><code>ncores</code></td>
<td>
<p>the number of cores used</p>
</td></tr>

</table>


<h3>Note</h3>


<p>The plot function in the example provides an invisible result with detail ans statistics of the experiment.  
</p>


<h3>Author(s)</h3>

<p>Stefano M. Pagnotta
</p>


<h3>References</h3>

<p>Monti et al. (2003) - Consensus Clustering: A Resampling-Based Method for Class Discovery and Visualization of Gene Expression Microarray Data - Machine Learning 52(1-2):91-118 &lt;DOI: 10.1023/A:1023949509487&gt;
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="stats.html#topic+hclust">hclust</a></code>, <code>doParallel</code>

</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Generate data and annotation
n &lt;- 50; m &lt;- 3000
ddata &lt;- matrix(rnorm(n * m), ncol = m)  
ddata[1:20, ] &lt;- ddata[1:20, ] + 0.2
row.names(ddata) &lt;- c(paste0("A", 1:20), paste0("B", 1:30))
ddist &lt;- dist(ddata)

annotation &lt;- data.frame(row.names = rownames(ddata), clust = substr(rownames(ddata), 1, 1))
annotation.colorCode &lt;- c("red", "blue")
names(annotation.colorCode) &lt;- c("A", "B")

####### run in sequential mode
####### sampling the samples ....
(aConsensus &lt;- yaConsensus(ddist))
plot(aConsensus, G = 2)

ans &lt;- plot(aConsensus, G = 2, 
            annotation = annotation, 
            annotation.colorCode = annotation.colorCode)
summary(ans)
summary(ans, given = "clust")

####### sampling the features ....
(aConsensus &lt;- yaConsensus(ddata, runs= 20, epsilon = 0.2, is_by_sample = FALSE))
ans &lt;- plot(aConsensus, G = 2, 
            annotation = annotation, 
            annotation.colorCode = annotation.colorCode,
            matching_clustering = "clust")

summary(ans, given = "clust")


####### run in parallel mode
## uncomment to run

# require(doParallel)
# cpu_cluster &lt;- makeCluster(3)
# registerDoParallel(cpu_cluster)

# (aConsensus &lt;- yaConsensus(ddist))
# plot(aConsensus, G = 2)

#stopCluster(cpu_cluster)





</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
