<!DOCTYPE html><html><head><title>Help for package zalpha</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {zalpha}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#create_LDprofile'><p>Creates an LD profile</p></a></li>
<li><a href='#L_plus_R'><p>Runs the L_plus_R function</p></a></li>
<li><a href='#LDprofile'><p>Dataset containing an example LD profile</p></a></li>
<li><a href='#LR'><p>Runs the LR function</p></a></li>
<li><a href='#snps'><p>Dataset containing details on simulated SNPs</p></a></li>
<li><a href='#Zalpha'><p>Runs the Zalpha function</p></a></li>
<li><a href='#Zalpha_all'><p>Runs all the statistics in the zalpha package</p></a></li>
<li><a href='#Zalpha_BetaCDF'><p>Runs the Zalpha function using a cumulative beta distribution function on the r-squared values for the region</p></a></li>
<li><a href='#Zalpha_expected'><p>Runs the Zalpha function on the expected r-squared values for the region</p></a></li>
<li><a href='#Zalpha_log_rsq_over_expected'><p>Runs the Zalpha function on the log of the r-squared values over the expected r-squared values for the region</p></a></li>
<li><a href='#Zalpha_rsq_over_expected'><p>Runs the Zalpha function on the r-squared values over the expected r-squared values for the region</p></a></li>
<li><a href='#Zalpha_Zscore'><p>Runs the Zalpha function using the Z score of the r-squared values for the region</p></a></li>
<li><a href='#Zbeta'><p>Runs the Zbeta function</p></a></li>
<li><a href='#Zbeta_BetaCDF'><p>Runs the Zbeta function using a cumulative beta distribution function on the r-squared values for the region</p></a></li>
<li><a href='#Zbeta_expected'><p>Runs the Zbeta function on the expected r-squared values for the region</p></a></li>
<li><a href='#Zbeta_log_rsq_over_expected'><p>Runs the Zbeta function on the log of the r-squared values over the expected r-squared values for the region</p></a></li>
<li><a href='#Zbeta_rsq_over_expected'><p>Runs the Zbeta function on the r-squared values over the expected r-squared values for the region</p></a></li>
<li><a href='#Zbeta_Zscore'><p>Runs the Zbeta function using the Z score of the r-squared values for the region</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Run a Suite of Selection Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Clare Horscroft</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Clare Horscroft &lt;chorscroft@aol.co.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A suite of statistics for identifying areas of the genome under selective pressure. See Jacobs, Sluckin and Kivisild (2016) &lt;<a href="https://doi.org/10.1534%2Fgenetics.115.185900">doi:10.1534/genetics.115.185900</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0), knitr, rmarkdown, fitdistrplus</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-27 06:14:24 UTC; ch19g17</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-27 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='create_LDprofile'>Creates an LD profile</h2><span id='topic+create_LDprofile'></span>

<h3>Description</h3>

<p>An LD (linkage disequilibrium) profile is a look-up table containing the expected correlation between SNPs given the genetic distance between them. The use of an LD profile can increase the accuracy of results by taking into account the expected correlation between SNPs. This function aids the user in creating their own LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_LDprofile(dist, x, bin_size, max_dist = NULL, beta_params = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_LDprofile_+3A_dist">dist</code></td>
<td>
<p>A numeric vector, or a list of numeric vectors, containing the genetic distance for each SNP.</p>
</td></tr>
<tr><td><code id="create_LDprofile_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values, or a list of matrices. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>dist</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="create_LDprofile_+3A_bin_size">bin_size</code></td>
<td>
<p>The size of each bin, in the same units as <code>dist</code>.</p>
</td></tr>
<tr><td><code id="create_LDprofile_+3A_max_dist">max_dist</code></td>
<td>
<p>Optional. The maximum genetic distance to be considered. If this is not supplied, it will default to the maximum distance in the <code>dist</code> vector.</p>
</td></tr>
<tr><td><code id="create_LDprofile_+3A_beta_params">beta_params</code></td>
<td>
<p>Optional. Beta parameters are calculated if this is set to TRUE. Default is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The input for <code>dist</code> and <code>x</code> can be lists. This allows multiple datasets to be used in the creation of the LD profile. For example, using all 22 autosomes from the human genome would involve 22 different distance vectors and SNP matrices.
Both lists should be the same length and should correspond exactly to each other (i.e. the distances in each element of <code>dist</code> should go with the SNPs in the same element of x)
</p>
<p>In the output, bins represent lower bounds. The first bin contains pairs where the genetic distance is greater than or equal to 0 and less than <code>bin_size</code>. The final bin contains pairs where the genetic distance is greater than or equal to <code>max_dist</code>-<code>bin_size</code> and less than <code>max_dist</code>.
If the <code>max_dist</code> is not an increment of <code>bin_size</code>, it will be adjusted to the next highest increment. The final bin will be the bin that <code>max_dist</code> falls into. For example, if the <code>max_dist</code> is given as 4.5 and the <code>bin_size</code> is 1, the final bin will be 4.
<code>max_dist</code> should be big enough to cover the genetic distances between pairs of SNPs within the window size given when the <code class="reqn">Z_{\alpha}</code> statistics are run. Any pairs with genetic distances bigger than <code>max_dist</code> will be assigned the values in the maximum bin of the LD profile.<br />
</p>
<p>By default, Beta parameters are not calculated. To fit a Beta distribution to the expected correlations, needed for the <code><a href="#topic+Zalpha_BetaCDF">Zalpha_BetaCDF</a></code> and <code><a href="#topic+Zbeta_BetaCDF">Zbeta_BetaCDF</a></code> statistics, <code>beta_params</code> should be set to TRUE and the package 'fitdistrplus' must be installed.
</p>
<p>Ideally, an LD profile would be generated using data from a null population with no selection, For example by using a simulation if the other population parameters are known. However, often these are unknown or complex, so generating an LD profile using the same data as is being analysed is acceptable, as long as the bins are large enough.
</p>


<h3>Value</h3>

<p>A data frame containing an LD profile that can be used by other statistics in this package.
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Zalpha_expected">Zalpha_expected</a></code>, <code><a href="#topic+Zalpha_rsq_over_expected">Zalpha_rsq_over_expected</a></code>, <code><a href="#topic+Zalpha_log_rsq_over_expected">Zalpha_log_rsq_over_expected</a></code>, <code><a href="#topic+Zalpha_Zscore">Zalpha_Zscore</a></code>, <code><a href="#topic+Zalpha_BetaCDF">Zalpha_BetaCDF</a></code>, <code><a href="#topic+Zbeta_expected">Zbeta_expected</a></code>, <code><a href="#topic+Zbeta_rsq_over_expected">Zbeta_rsq_over_expected</a></code>, <code><a href="#topic+Zbeta_log_rsq_over_expected">Zbeta_log_rsq_over_expected</a></code>, <code><a href="#topic+Zbeta_Zscore">Zbeta_Zscore</a></code>, <code><a href="#topic+Zbeta_BetaCDF">Zbeta_BetaCDF</a></code>, <code><a href="#topic+Zalpha_all">Zalpha_all</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps example dataset
data(snps)
## Create an LD profile using this data
create_LDprofile(snps$cM_distances,as.matrix(snps[,3:12]),0.001)
## To get the Beta distribution parameter estimates, the fitdistrplus package is required
if (requireNamespace("fitdistrplus", quietly = TRUE)==TRUE) {
  create_LDprofile(snps$cM_distances,as.matrix(snps[,3:12]),0.001,beta_params=TRUE)
}


</code></pre>

<hr>
<h2 id='L_plus_R'>Runs the L_plus_R function</h2><span id='topic+L_plus_R'></span>

<h3>Description</h3>

<p>Returns the <code class="reqn">{|L| \choose 2} + {|R| \choose 2}</code> value for each SNP location supplied to the function. <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>.
For more information about the <code>L_plus_R</code> diversity statistic, please see Jacobs (2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L_plus_R(pos, ws, X = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L_plus_R_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="L_plus_R_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code>L_plus_R</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="L_plus_R_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code>L_plus_R</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate L_plus_R for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code>L_plus_R </code>values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps example dataset
data(snps)
## run L_plus_R over all the SNPs with a window size of 3000 bp
L_plus_R(snps$bp_positions,3000)
## only return results for SNPs between locations 600 and 1500 bp
L_plus_R(snps$bp_positions,3000,X=c(600,1500))

</code></pre>

<hr>
<h2 id='LDprofile'>Dataset containing an example LD profile</h2><span id='topic+LDprofile'></span>

<h3>Description</h3>

<p>A simulated LD profile, containing example LD statistics for
genetic distances of 0 to 0.0049, in bins of size 0.0001.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LDprofile)
</code></pre>


<h3>Format</h3>

<p>A data frame with 50 rows and 5 variables:
</p>

<dl>
<dt>bin</dt><dd><p>the lower bound of each bin</p>
</dd>
<dt>rsq</dt><dd><p>the expected <code class="reqn">r^2</code> value for a pair of SNPs, where the genetic distance between them falls in the given bin</p>
</dd>
<dt>sd</dt><dd><p>the standard deviation of the expected <code class="reqn">r^2</code> value</p>
</dd>
<dt>Beta_a</dt><dd><p>the first shape parameter for the Beta distribution fitted for this bin</p>
</dd>
<dt>Beta_b</dt><dd><p>the second shape parameter for the Beta distribution fitted for this bin</p>
</dd>
</dl>


<hr>
<h2 id='LR'>Runs the LR function</h2><span id='topic+LR'></span>

<h3>Description</h3>

<p>Returns the <code>|L||R|</code> value for each SNP location supplied to the function, where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>.
For more information about the <code>|L||R|</code> diversity statistic, please see Jacobs (2016).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LR(pos, ws, X = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LR_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="LR_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code>LR</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="LR_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate LR for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate LR for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code>LR</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps example dataset
data(snps)
## run LR over all the SNPs with a window size of 3000 bp
LR(snps$bp_positions,3000)
## only return results for SNPs between locations 600 and 1500 bp
LR(snps$bp_positions,3000,X=c(600,1500))

</code></pre>

<hr>
<h2 id='snps'>Dataset containing details on simulated SNPs</h2><span id='topic+snps'></span>

<h3>Description</h3>

<p>A dataset containing the positions, genetic distances and
alleles for 20 SNPs, across 10 simulated chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snps
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows and 12 variables:
</p>

<dl>
<dt>bp_positions</dt><dd><p>location of the SNP on the chromosome e.g. in base pairs</p>
</dd>
<dt>cM_distances</dt><dd><p>genetic distance of the SNP from the start of the chromosome e.g. in centimorgans</p>
</dd>
<dt>chrom_1</dt><dd><p>allele of the SNP on the first example chromosome</p>
</dd>
<dt>chrom_2</dt><dd><p>allele of the SNP on the second example chromosome</p>
</dd>
<dt>chrom_3</dt><dd><p>allele of the SNP on the third example chromosome</p>
</dd>
<dt>chrom_4</dt><dd><p>allele of the SNP on the fourth example chromosome</p>
</dd>
<dt>chrom_5</dt><dd><p>allele of the SNP on the fifth example chromosome</p>
</dd>
<dt>chrom_6</dt><dd><p>allele of the SNP on the sixth example chromosome</p>
</dd>
<dt>chrom_7</dt><dd><p>allele of the SNP on the seventh example chromosome</p>
</dd>
<dt>chrom_8</dt><dd><p>allele of the SNP on the eighth example chromosome</p>
</dd>
<dt>chrom_9</dt><dd><p>allele of the SNP on the ninth example chromosome</p>
</dd>
<dt>chrom_10</dt><dd><p>allele of the SNP on the tenth example chromosome</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>snps
</code></pre>

<hr>
<h2 id='Zalpha'>Runs the Zalpha function</h2><span id='topic+Zalpha'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\alpha}</code> value for each SNP location supplied to the function.
For more information about the <code class="reqn">Z_{\alpha}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\alpha}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\alpha}=\frac{{|L| \choose 2}^{-1}\sum_{i,j \in L}r^2_{i,j} + {|R| \choose 2}^{-1}\sum_{i,j \in L}r^2_{i,j}}{2}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, and <code class="reqn">r^2</code> is equal to the squared correlation between a pair of SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha(pos, ws, x, minRandL = 4, minRL = 25, X = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\alpha}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zalpha_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\alpha}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\alpha}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\alpha}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps example dataset
data(snps)
## run Zalpha over all the SNPs with a window size of 3000 bp
Zalpha(snps$bp_positions,3000,as.matrix(snps[,3:12]))
## only return results for SNPs between locations 600 and 1500 bp
Zalpha(snps$bp_positions,3000,as.matrix(snps[,3:12]),X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zalpha_all'>Runs all the statistics in the zalpha package</h2><span id='topic+Zalpha_all'></span>

<h3>Description</h3>

<p>Returns every statistic for each SNP location, given the appropriate parameters. See Details for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha_all(
  pos,
  ws,
  x = NULL,
  dist = NULL,
  LDprofile_bins = NULL,
  LDprofile_rsq = NULL,
  LDprofile_sd = NULL,
  LDprofile_Beta_a = NULL,
  LDprofile_Beta_b = NULL,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_all_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_ws">ws</code></td>
<td>
<p>The window size which the statistics will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_x">x</code></td>
<td>
<p>Optional. A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_dist">dist</code></td>
<td>
<p>Optional. A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>Optional. A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>Optional. A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_ldprofile_sd">LDprofile_sd</code></td>
<td>
<p>Optional. A numeric vector containing the standard deviation of the <code class="reqn">r^2</code> values for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_ldprofile_beta_a">LDprofile_Beta_a</code></td>
<td>
<p>Optional. A numeric vector containing the first estimated Beta parameter for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_ldprofile_beta_b">LDprofile_Beta_b</code></td>
<td>
<p>Optional. A numeric vector containing the second estimated Beta parameter for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistics to be calculated. L is the set of SNPs to the left of the target SNP and R to the right, within the given window size <code>ws</code>. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_all_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate the statistics for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate the statistics for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Not all statistics will be returned, depending on the parameters supplied to the function.<br />
If <code>x</code> is not supplied, only <code><a href="#topic+Zalpha_expected">Zalpha_expected</a></code>, <code><a href="#topic+Zbeta_expected">Zbeta_expected</a></code>, <code><a href="#topic+LR">LR</a></code> and <code><a href="#topic+L_plus_R">L_plus_R</a></code> will be calculated.<br />
For any of the statistics which use an expected <code class="reqn">r^2</code> value, the parameters <code>dist</code>, <code>LDprofile_bins</code> and <code>LDprofile_rsq</code> must be supplied.
This includes the statistics: <code><a href="#topic+Zalpha_expected">Zalpha_expected</a></code>, <code><a href="#topic+Zalpha_rsq_over_expected">Zalpha_rsq_over_expected</a></code>, <code><a href="#topic+Zalpha_log_rsq_over_expected">Zalpha_log_rsq_over_expected</a></code>, <code><a href="#topic+Zalpha_Zscore">Zalpha_Zscore</a></code>, <code><a href="#topic+Zalpha_BetaCDF">Zalpha_BetaCDF</a></code>, <code><a href="#topic+Zbeta_expected">Zbeta_expected</a></code>, <code><a href="#topic+Zbeta_rsq_over_expected">Zbeta_rsq_over_expected</a></code>, <code><a href="#topic+Zbeta_log_rsq_over_expected">Zbeta_log_rsq_over_expected</a></code>, <code><a href="#topic+Zbeta_Zscore">Zbeta_Zscore</a></code> and <code><a href="#topic+Zbeta_BetaCDF">Zbeta_BetaCDF</a></code>.
</p>

<ul>
<li><p> For <code><a href="#topic+Zalpha_Zscore">Zalpha_Zscore</a></code> and <code><a href="#topic+Zbeta_Zscore">Zbeta_Zscore</a></code> to be calculated, the parameter <code>LDprofile_sd</code> must also be supplied.
</p>
</li>
<li><p> For <code><a href="#topic+Zalpha_BetaCDF">Zalpha_BetaCDF</a></code> and <code><a href="#topic+Zbeta_BetaCDF">Zbeta_BetaCDF</a></code> to be calculated, the parameters <code>LDprofile_Beta_a</code> and <code>LDprofile_Beta_b</code> must also be supplied.
</p>
</li></ul>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
For more information about the statistics, please see Jacobs (2016).
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the statistics for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Zalpha">Zalpha</a></code>, <code><a href="#topic+Zalpha_expected">Zalpha_expected</a></code>, <code><a href="#topic+Zalpha_rsq_over_expected">Zalpha_rsq_over_expected</a></code>, <code><a href="#topic+Zalpha_log_rsq_over_expected">Zalpha_log_rsq_over_expected</a></code>, <code><a href="#topic+Zalpha_Zscore">Zalpha_Zscore</a></code>, <code><a href="#topic+Zalpha_BetaCDF">Zalpha_BetaCDF</a></code>, <code><a href="#topic+Zbeta">Zbeta</a></code>, <code><a href="#topic+Zbeta_expected">Zbeta_expected</a></code>, <code><a href="#topic+Zbeta_rsq_over_expected">Zbeta_rsq_over_expected</a></code>, <code><a href="#topic+Zbeta_log_rsq_over_expected">Zbeta_log_rsq_over_expected</a></code>, <code><a href="#topic+Zbeta_Zscore">Zbeta_Zscore</a></code>, <code><a href="#topic+Zbeta_BetaCDF">Zbeta_BetaCDF</a></code>, <code><a href="#topic+LR">LR</a></code>, <code><a href="#topic+L_plus_R">L_plus_R</a></code>, <code><a href="#topic+create_LDprofile">create_LDprofile</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zalpha_all over all the SNPs with a window size of 3000 bp
## will return all 15 statistics
Zalpha_all(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,LDprofile$sd,LDprofile$Beta_a,LDprofile$Beta_b)
## only return results for SNPs between locations 600 and 1500 bp
Zalpha_all(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,LDprofile$sd,LDprofile$Beta_a,LDprofile$Beta_b,X=c(600,1500))
## will only return statistics not requiring an LD profile
Zalpha_all(snps$bp_positions,3000,as.matrix(snps[,3:12]))

</code></pre>

<hr>
<h2 id='Zalpha_BetaCDF'>Runs the Zalpha function using a cumulative beta distribution function on the r-squared values for the region</h2><span id='topic+Zalpha_BetaCDF'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\alpha}^{BetaCDF}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\alpha}^{BetaCDF}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\alpha}^{BetaCDF}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">{Z_{\alpha}^{BetaCDF}}=\frac{{|L| \choose 2}^{-1}\sum_{i,j \in L}\frac{B(r^2_{i,j};a,b)}{B(a,b)} + {|R| \choose 2}^{-1}\sum_{i,j \in R}\frac{B(r^2_{i,j};a,b)}{B(a,b)}}{2}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, and <code class="reqn">\frac{B(r^2_{i,j};a,b)}{B(a,b)}</code> is the cumulative distribution function for the Beta distribution given
the estimated a and b parameters from the LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha_BetaCDF(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_Beta_a,
  LDprofile_Beta_b,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_BetaCDF_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\alpha}^{BetaCDF}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_ldprofile_beta_a">LDprofile_Beta_a</code></td>
<td>
<p>A numeric vector containing the first estimated Beta parameter for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_ldprofile_beta_b">LDprofile_Beta_b</code></td>
<td>
<p>A numeric vector containing the second estimated Beta parameter for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_BetaCDF_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\alpha}^{BetaCDF}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\alpha}^{BetaCDF}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\alpha}^{BetaCDF}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zalpha_BetaCDF over all the SNPs with a window size of 3000 bp
Zalpha_BetaCDF(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$Beta_a,LDprofile$Beta_b)
## only return results for SNPs between locations 600 and 1500 bp
Zalpha_BetaCDF(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$Beta_a,LDprofile$Beta_b,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zalpha_expected'>Runs the Zalpha function on the expected r-squared values for the region</h2><span id='topic+Zalpha_expected'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\alpha}^{E[r^2]}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\alpha}^{E[r^2]}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\alpha}^{E[r^2]}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">{Z_{\alpha}^{E[r^2]}}=\frac{{|L| \choose 2}^{-1}\sum_{i,j \in L}E[r^2_{i,j}] + {|R| \choose 2}^{-1}\sum_{i,j \in R}E[r^2_{i,j}]}{2}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>,
and <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha_expected(
  pos,
  ws,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_expected_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\alpha}^{E[r^2]}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_expected_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\alpha}^{E[r^2]}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\alpha}^{E[r^2]}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\alpha}^{E[r^2]}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zalpha_expected over all the SNPs with a window size of 3000 bp
Zalpha_expected(snps$bp_positions,3000,snps$cM_distances,LDprofile$bin,LDprofile$rsq)
## only return results for SNPs between locations 600 and 1500 bp
Zalpha_expected(snps$bp_positions,3000,snps$cM_distances,LDprofile$bin,LDprofile$rsq,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zalpha_log_rsq_over_expected'>Runs the Zalpha function on the log of the r-squared values over the expected r-squared values for the region</h2><span id='topic+Zalpha_log_rsq_over_expected'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">{Z_{\alpha}^{log_{10}(r^2/E[r^2])}}=\frac{{|L| \choose 2}^{-1}\sum_{i,j \in L}log_{10}(r^2_{i,j}/E[r^2_{i,j}]) + {|R| \choose 2}^{-1}\sum_{i,j \in R}log_{10}(r^2_{i,j}/E[r^2_{i,j}])}{2}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, and <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha_log_rsq_over_expected(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_log_rsq_over_expected_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\alpha}^{log_{10}(r^2/E[r^2])}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zalpha_log_rsq_over_expected over all the SNPs with a window size of 3000 bp
Zalpha_log_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq)
## only return results for SNPs between locations 600 and 1500 bp
Zalpha_log_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zalpha_rsq_over_expected'>Runs the Zalpha function on the r-squared values over the expected r-squared values for the region</h2><span id='topic+Zalpha_rsq_over_expected'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">{Z_{\alpha}^{r^2/E[r^2]}}=\frac{{|L| \choose 2}^{-1}\sum_{i,j \in L}r^2_{i,j}/E[r^2_{i,j}] + {|R| \choose 2}^{-1}\sum_{i,j \in R}r^2_{i,j}/E[r^2_{i,j}]}{2}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, and <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha_rsq_over_expected(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_rsq_over_expected_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\alpha}^{r^2/E[r^2]}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zalpha_rsq_over_expected over all the SNPs with a window size of 3000 bp
Zalpha_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq)
## only return results for SNPs between locations 600 and 1500 bp
Zalpha_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zalpha_Zscore'>Runs the Zalpha function using the Z score of the r-squared values for the region</h2><span id='topic+Zalpha_Zscore'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\alpha}^{Zscore}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\alpha}^{Zscore}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\alpha}^{Zscore}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">{Z_{\alpha}^{Zscore}}=\frac{{|L| \choose 2}^{-1}\sum_{i,j \in L}\frac{r^2_{i,j}-E[r^2_{i,j}]}{\sigma[r^2_{i,j}]} + {|R| \choose 2}^{-1}\sum_{i,j \in R}\frac{r^2_{i,j}-E[r^2_{i,j}]}{\sigma[r^2_{i,j}]}}{2}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile, and <code class="reqn">\sigma[r^2]</code> is the standard deviation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zalpha_Zscore(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  LDprofile_sd,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zalpha_Zscore_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\alpha}^{Zscore}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_ldprofile_sd">LDprofile_sd</code></td>
<td>
<p>A numeric vector containing the standard deviation of the <code class="reqn">r^2</code> values for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zalpha_Zscore_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\alpha}^{Zscore}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\alpha}^{Zscore}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\alpha}^{Zscore}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zalpha_Zscore over all the SNPs with a window size of 3000 bp
Zalpha_Zscore(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,LDprofile$sd)
## only return results for SNPs between locations 600 and 1500 bp
Zalpha_Zscore(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,LDprofile$sd,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zbeta'>Runs the Zbeta function</h2><span id='topic+Zbeta'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\beta}</code> value for each SNP location supplied to the function.
For more information about the <code class="reqn">Z_{\beta}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\beta}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\beta}=\frac{\sum_{i \in L,j \in R}r^2_{i,j}}{|L||R|}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, and <code class="reqn">r^2</code> is equal to the squared correlation between a pair of SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zbeta(pos, ws, x, minRandL = 4, minRL = 25, X = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zbeta_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zbeta_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\beta}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zbeta_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zbeta_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zbeta_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zbeta_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\beta}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\beta}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\beta}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps example dataset
data(snps)
## run Zbeta over all the SNPs with a window size of 3000 bp
Zbeta(snps$bp_positions,3000,as.matrix(snps[,3:12]))
## only return results for SNPs between locations 600 and 1500 bp
Zbeta(snps$bp_positions,3000,as.matrix(snps[,3:12]),X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zbeta_BetaCDF'>Runs the Zbeta function using a cumulative beta distribution function on the r-squared values for the region</h2><span id='topic+Zbeta_BetaCDF'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\beta}^{BetaCDF}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\beta}^{BetaCDF}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\beta}^{BetaCDF}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\beta}^{BetaCDF}=\frac{\sum_{i \in L,j \in R}\frac{B(r^2_{i,j};a,b)}{B(a,b)}}{|L||R|}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, and <code class="reqn">\frac{B(r^2_{i,j};a,b)}{B(a,b)}</code> is the cumulative distribution function for the Beta distribution given
the estimated a and b parameters from the LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zbeta_BetaCDF(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_Beta_a,
  LDprofile_Beta_b,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zbeta_BetaCDF_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\beta}^{BetaCDF}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_ldprofile_beta_a">LDprofile_Beta_a</code></td>
<td>
<p>A numeric vector containing the first estimated Beta parameter for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_ldprofile_beta_b">LDprofile_Beta_b</code></td>
<td>
<p>A numeric vector containing the second estimated Beta parameter for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zbeta_BetaCDF_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\beta}^{BetaCDF}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\beta}^{BetaCDF}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\beta}^{BetaCDF}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zbeta_BetaCDF over all the SNPs with a window size of 3000 bp
Zbeta_BetaCDF(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$Beta_a,LDprofile$Beta_b)
## only return results for SNPs between locations 600 and 1500 bp
Zbeta_BetaCDF(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$Beta_a,LDprofile$Beta_b,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zbeta_expected'>Runs the Zbeta function on the expected r-squared values for the region</h2><span id='topic+Zbeta_expected'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\beta}^{E[r^2]}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\beta}^{E[r^2]}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\beta}^{E[r^2]}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\beta}^{E[r^2]}=\frac{\sum_{i \in L,j \in R}E[r^2_{i,j}]}{|L||R|}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>,
and <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zbeta_expected(
  pos,
  ws,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zbeta_expected_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\beta}^{E[r^2]}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zbeta_expected_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\beta}^{E[r^2]}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\beta}^{E[r^2]}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\beta}^{E[r^2]}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zbeta_expected over all the SNPs with a window size of 3000 bp
Zbeta_expected(snps$bp_positions,3000,snps$cM_distances,LDprofile$bin,LDprofile$rsq)
## only return results for SNPs between locations 600 and 1500 bp
Zbeta_expected(snps$bp_positions,3000,snps$cM_distances,LDprofile$bin,LDprofile$rsq,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zbeta_log_rsq_over_expected'>Runs the Zbeta function on the log of the r-squared values over the expected r-squared values for the region</h2><span id='topic+Zbeta_log_rsq_over_expected'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}=\frac{\sum_{i \in L,j \in R}log_{10}(r^2_{i,j}/E[r^2_{i,j}])}{|L||R|}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, and <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zbeta_log_rsq_over_expected(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zbeta_log_rsq_over_expected_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\beta}^{log_{10}(r^2/E[r^2])}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zbeta_log_rsq_over_expected over all the SNPs with a window size of 3000 bp
Zbeta_log_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq)
## only return results for SNPs between locations 600 and 1500 bp
Zbeta_log_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zbeta_rsq_over_expected'>Runs the Zbeta function on the r-squared values over the expected r-squared values for the region</h2><span id='topic+Zbeta_rsq_over_expected'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\beta}^{r^2/E[r^2]}=\frac{\sum_{i \in L,j \in R}r^2_{i,j}/E[r^2_{i,j}]}{|L||R|}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, and <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zbeta_rsq_over_expected(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zbeta_rsq_over_expected_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\beta}^{r^2/E[r^2]}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zbeta_rsq_over_expected over all the SNPs with a window size of 3000 bp
Zbeta_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq)
## only return results for SNPs between locations 600 and 1500 bp
Zbeta_rsq_over_expected(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,X=c(600,1500))

</code></pre>

<hr>
<h2 id='Zbeta_Zscore'>Runs the Zbeta function using the Z score of the r-squared values for the region</h2><span id='topic+Zbeta_Zscore'></span>

<h3>Description</h3>

<p>Returns a <code class="reqn">Z_{\beta}^{Zscore}</code> value for each SNP location supplied to the function, based on
the expected <code class="reqn">r^2</code> values given an LD profile and genetic distances.
For more information about the <code class="reqn">Z_{\beta}^{Zscore}</code> statistic, please see Jacobs (2016).
The <code class="reqn">Z_{\beta}^{Zscore}</code> statistic is defined as:
</p>
<p style="text-align: center;"><code class="reqn">Z_{\beta}^{Zscore}=\frac{\sum_{i \in L,j \in R}\frac{r^2_{i,j}-E[r^2_{i,j}]}{\sigma[r^2_{i,j}]}}{|L||R|}</code>
</p>

<p>where <code>|L|</code> and <code>|R|</code> are the number of SNPs to the left and right of the current locus within the given window <code>ws</code>, <code class="reqn">r^2</code> is equal to
the squared correlation between a pair of SNPs, <code class="reqn">E[r^2]</code> is equal to the expected squared correlation between a pair of SNPs, given an LD profile, and <code class="reqn">\sigma[r^2]</code> is the standard deviation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zbeta_Zscore(
  pos,
  ws,
  x,
  dist,
  LDprofile_bins,
  LDprofile_rsq,
  LDprofile_sd,
  minRandL = 4,
  minRL = 25,
  X = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zbeta_Zscore_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of SNP locations</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_ws">ws</code></td>
<td>
<p>The window size which the <code class="reqn">Z_{\beta}^{Zscore}</code> statistic will be calculated over. This should be on the same scale as the <code>pos</code> vector.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_x">x</code></td>
<td>
<p>A matrix of SNP values. Columns represent chromosomes; rows are SNP locations. Hence, the number of rows should equal the length of the <code>pos</code> vector. SNPs should all be biallelic.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_dist">dist</code></td>
<td>
<p>A numeric vector of genetic distances (e.g. cM, LDU). This should be the same length as <code>pos</code>.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_ldprofile_bins">LDprofile_bins</code></td>
<td>
<p>A numeric vector containing the lower bound of the bins used in the LD profile. These should be of equal size.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_ldprofile_rsq">LDprofile_rsq</code></td>
<td>
<p>A numeric vector containing the expected <code class="reqn">r^2</code> values for the corresponding bin in the LD profile. Must be between 0 and 1.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_ldprofile_sd">LDprofile_sd</code></td>
<td>
<p>A numeric vector containing the standard deviation of the <code class="reqn">r^2</code> values for the corresponding bin in the LD profile.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_minrandl">minRandL</code></td>
<td>
<p>Minimum number of SNPs in each set R and L for the statistic to be calculated. Default is 4.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_minrl">minRL</code></td>
<td>
<p>Minimum value for the product of the set sizes for R and L. Default is 25.</p>
</td></tr>
<tr><td><code id="Zbeta_Zscore_+3A_x">X</code></td>
<td>
<p>Optional. Specify a region of the chromosome to calculate <code class="reqn">Z_{\beta}^{Zscore}</code> for in the format <code>c(startposition, endposition)</code>. The start position and the end position should be within the extremes of the positions given in the <code>pos</code> vector. If not supplied, the function will calculate <code class="reqn">Z_{\beta}^{Zscore}</code> for every SNP in the <code>pos</code> vector.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LD profile describes the expected correlation between SNPs at a given genetic distance, generated using simulations or
real data. Care should be taken to utilise an LD profile that is representative of the population in question. The LD
profile should consist of evenly sized bins of distances (for example 0.0001 cM per bin), where the value given is the (inclusive) lower
bound of the bin. Ideally, an LD profile would be generated using data from a null population with no selection, however one can be generated
using this data. See the <code><a href="#topic+create_LDprofile">create_LDprofile</a></code> function for more information on how to create an LD profile.
</p>


<h3>Value</h3>

<p>A list containing the SNP positions and the <code class="reqn">Z_{\beta}^{Zscore}</code> values for those SNPs
</p>


<h3>References</h3>

<p>Jacobs, G.S., T.J. Sluckin, and T. Kivisild, <em>Refining the Use of Linkage Disequilibrium as a Robust Signature of Selective Sweeps.</em> Genetics, 2016. <strong>203</strong>(4): p. 1807
</p>


<h3>See Also</h3>

<p><code><a href="#topic+create_LDprofile">create_LDprofile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load the snps and LDprofile example datasets
data(snps)
data(LDprofile)
## run Zbeta_Zscore over all the SNPs with a window size of 3000 bp
Zbeta_Zscore(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,LDprofile$sd)
## only return results for SNPs between locations 600 and 1500 bp
Zbeta_Zscore(snps$bp_positions,3000,as.matrix(snps[,3:12]),snps$cM_distances,
 LDprofile$bin,LDprofile$rsq,LDprofile$sd,X=c(600,1500))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
