<!DOCTYPE html><html><head><title>Help for package ZeBook</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ZeBook}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ZeBook-package'><p>Working with Dynamic Models for Agriculture and Environment</p></a></li>
<li><a href='#AICf'><p>Calculate AIC, Akaike's Information Criterion</p></a></li>
<li><a href='#Bean'><p>Bean gene-based models dataset</p></a></li>
<li><a href='#carbonsoil.model'><p>The CarbonSoil model - calculate daily values over designated time period</p></a></li>
<li><a href='#carbonsoil.update'><p>The CarbonSoil model - calculate change in soil carbon for one year</p></a></li>
<li><a href='#carcass_data'><p>Data of growth of beef cattle for Carcass model</p></a></li>
<li><a href='#carcass.define.param'><p>Define values of the parameters for the Carcass model</p></a></li>
<li><a href='#carcass.EMI.model'><p>The Carcass (growth of beef cattle) model with energy as input</p></a></li>
<li><a href='#carcass.EMI.model2'><p>The Carcass model function for use with carcass.EMI.simule</p></a></li>
<li><a href='#carcass.EMI.multi'><p>Wrapper function to run Carcass model on several animals with different conditions</p></a></li>
<li><a href='#carcass.EMI.simule'><p>Wrapper function to the Carcass model for multiple sets of parameter values</p></a></li>
<li><a href='#carcass.model'><p>The Carcass (growth of beef cattle) model</p></a></li>
<li><a href='#carrot.weevil.model'><p>Carrot weevil development model</p></a></li>
<li><a href='#chicks_data'><p>Data of growth of chicks</p></a></li>
<li><a href='#cotton.model'><p>The Cotton model (dynamic for numbers of Cotton fruiting points).</p></a></li>
<li><a href='#epirice.define.param'><p>Define values of the parameters for the Epirice model</p></a></li>
<li><a href='#epirice.model'><p>The Epirice model (Disease model for rice)</p></a></li>
<li><a href='#epirice.multi.simule'><p>Wrapper function to run Epirice multiple times (for multiple sets of inputs)</p></a></li>
<li><a href='#epirice.weather'><p>Read weather data for Epirice (southern Asia weather)</p></a></li>
<li><a href='#evaluation.criteria'><p>Calcule multiple goodness-of-fit criteria</p></a></li>
<li><a href='#exponential.model'><p>The Exponential growth model of dynamic of population</p></a></li>
<li><a href='#exponential.model.bis'><p>The Exponential growth model of dynamic of population - another form</p></a></li>
<li><a href='#exponential.model.ie'><p>The Exponential growth model of dynamic of population - with improved Euler integration</p></a></li>
<li><a href='#goodness.of.fit'><p>Calcule multiple goodness-of-fit criteria</p></a></li>
<li><a href='#graph_epid'><p>Plot output of a Classical SEIR model for plant disease</p></a></li>
<li><a href='#graph_epid_s'><p>Plot output of a Classical SEIR model for plant disease</p></a></li>
<li><a href='#lactation.calf.model'><p>The Lactation model</p></a></li>
<li><a href='#lactation.calf.model2'><p>The Lactation model for use with lactation.calf.simule</p></a></li>
<li><a href='#lactation.calf.simule'><p>Wrapper function to run the Lactation model for multiple sets of parameter values</p></a></li>
<li><a href='#lactation.define.param'><p>Define values of the parameters for the Lactation model</p></a></li>
<li><a href='#lactation.machine.model'><p>The Lactation model with milking machine</p></a></li>
<li><a href='#lactation.machine.model2'><p>The Lactation model for use with lactation.machine.simule</p></a></li>
<li><a href='#magarey.define.param'><p>Define values of the parameters for the Magarey model</p></a></li>
<li><a href='#magarey.model'><p>The Magarey model</p></a></li>
<li><a href='#magarey.model2'><p>The Magarey model, taking a vector of parameters as argument</p></a></li>
<li><a href='#magarey.simule'><p>Wrapper function to run the Magarey model multiple times (for multiple sets of inputs)</p></a></li>
<li><a href='#maize_cir_rue_ear.model'><p>The Maize model with temperature dependent RUE, CumInt and ear growth</p></a></li>
<li><a href='#maize_cir_rue.model'><p>The Maize model with temperature dependent RUE and CumInt</p></a></li>
<li><a href='#maize_cir.model'><p>The Maize model with additional state variable CumInt</p></a></li>
<li><a href='#maize.data_EuropeEU'><p>maize biomass and leaf area data</p></a></li>
<li><a href='#maize.data_MetaModelling'><p>dataset of simulation for maize final biomass</p></a></li>
<li><a href='#maize.define.param'><p>Define values of the parameters for the Maize model</p></a></li>
<li><a href='#maize.model'><p>The basic Maize model.</p></a></li>
<li><a href='#maize.model2'><p>The basic Maize model for use with maize.simule</p></a></li>
<li><a href='#maize.muchow.graph'><p>Plot dynamic output of Muchow Maize model.</p></a></li>
<li><a href='#maize.muchow.model'><p>Maize model, with harvest index and yield from Muchow et al. (1990)</p></a></li>
<li><a href='#maize.multisy'><p>Wrapping function to run maize model on several site-years</p></a></li>
<li><a href='#maize.multisy240'><p>Wrapper function to run Maize model for multiple sets of input variables (site-year) and give Biomass at day240.</p></a></li>
<li><a href='#maize.RUEtemp'><p>Calculate effect of temperature on RUE for Maize</p></a></li>
<li><a href='#maize.simule'><p>Wrapper function to run Maize model for multiple sets of parameter values</p></a></li>
<li><a href='#maize.simule_multisy240'><p>Wrapper function to run Maize model for multiple sets of parameter values (virtual design) and multiple sets of input variables (site-year) and give Biomass at day240</p></a></li>
<li><a href='#maize.simule240'><p>Wrapper function to run Maize model multiple times for multiple sets of parameter values and give Biomass at day240</p></a></li>
<li><a href='#maize.weather'><p>Read weather data for the Maize model</p></a></li>
<li><a href='#mm.A.fct'><p>Expanded leaf area function for Muchow et al. (1990) Maize model</p></a></li>
<li><a href='#mm.FAS.fct'><p>Senescence function for Muchow et al. (1990) Maize model</p></a></li>
<li><a href='#mm.HI.fct'><p>Harvest index function for Muchow et al. (1990) Maize model</p></a></li>
<li><a href='#mm.LN.fct'><p>Leaf number function for Muchow et al. (1990) Maize model</p></a></li>
<li><a href='#param.rtriangle'><p>Generate a random plan as a data frame. Columns are parameters. Values have triangle distribution</p></a></li>
<li><a href='#param.runif'><p>Generate a random plan as a data frame. Columns are parameters. Values have uniform distribution</p></a></li>
<li><a href='#population.age.matrix.model'><p>The PopulationAge model (Population Dynamics with Age Classes) - matrix form</p></a></li>
<li><a href='#population.age.model'><p>The PopulationAge model (Population Dynamics with Age Classes)</p></a></li>
<li><a href='#population.age.model.ode'><p>The PopulationAge model (Population Dynamics with Age Classes) - ode form</p></a></li>
<li><a href='#predator.prey.model'><p>The PredatorPrey model (Predator-Prey Lotka-Volterra with logistic equation for prey)</p></a></li>
<li><a href='#q.arg.fast.runif'><p>Build the q.arg argument for the  FAST function (sensitivity analysis)</p></a></li>
<li><a href='#seedweight.data'><p>Wheat grain weight measurements after anthesis</p></a></li>
<li><a href='#seedweight.model'><p>The SeedWeight model</p></a></li>
<li><a href='#Sunflower_Phomopsis'><p>Phomopsis stem canker observations for Sunflower</p></a></li>
<li><a href='#threshold.measures'><p>Computation of threshold.measures</p></a></li>
<li><a href='#verhulst.model'><p>The Verhulst (logistic) model - calculate daily values over designated time period</p></a></li>
<li><a href='#verhulst.update'><p>The Verhulst (logistic) model - calculate change for one day</p></a></li>
<li><a href='#watbal.define.param'><p>Define values of the parameters for the WaterBalance model</p></a></li>
<li><a href='#watbal.model'><p>WaterBalance model - calculate soil water over designated time period</p></a></li>
<li><a href='#watbal.model.arid'><p>WaterBalance model - Variant with another order of calculation and ARID index</p></a></li>
<li><a href='#watbal.simobsdata'><p>Soil water content measurements and associated simulations with WaterBalance model</p></a></li>
<li><a href='#watbal.update'><p>WaterBalance model - calculate change in soil water for one day</p></a></li>
<li><a href='#watbal.weather'><p>Read weather data for the WaterBalance model (West of France Weather)</p></a></li>
<li><a href='#weather_EuropeEU'><p>Weather serie for Europe EU from NASA POWER agroclimatology</p></a></li>
<li><a href='#weather_FranceWest'><p>Weather series for western France from NASA POWER agroclimatology</p></a></li>
<li><a href='#weather_GNS'><p>Weather series for Gainesville (FL, USA) years 1982 and 1983</p></a></li>
<li><a href='#weather_SouthAsia'><p>Weather series for southern Asia from NASA POWER agroclimatology</p></a></li>
<li><a href='#weed.define.param'><p>Define parameter values of the Weed model</p></a></li>
<li><a href='#weed.model'><p>The Weed model - calculate daily values over designated time period</p></a></li>
<li><a href='#weed.simule'><p>Wrapper function to run the Weed model multiple times (for multiple sets of inputs)</p></a></li>
<li><a href='#weed.update'><p>The Weed model - calculate change for one year</p></a></li>
<li><a href='#Wheat_GPC'><p>Grain Protein Contents in Wheat Grains</p></a></li>
<li><a href='#WheatYieldGreece'><p>National Wheat Yield evolution for Greece from FAO</p></a></li>
<li><a href='#zakoks.original.model'><p>Classical SEIR model for plant diseases from Zadoks (1971)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Working with Dynamic Models for Agriculture and Environment</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-11-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Francois Brun (ACTA), David Makowski (INRA), Daniel Wallach (INRA), James W. Jones (U.of Florida).</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francois Brun &lt;francois.brun@acta.asso.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>R package accompanying the book Working with dynamic models for agriculture and environment, by Daniel Wallach (INRA), David Makowski (INRA), James W. Jones (U.of Florida), Francois Brun (ACTA). 3rd edition 2018-09-27.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>triangle, deSolve, stats, graphics</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-11-09 17:27:17 UTC; fbrun</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-11-09 17:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ZeBook-package'>Working with Dynamic Models for Agriculture and Environment</h2><span id='topic+ZeBook-package'></span><span id='topic+ZeBook'></span>

<h3>Description</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ZeBook</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-11-08</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> LGPL-3 </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyData: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R(&gt;= 2.10.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> triangle, deSolve, stats, graphics</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>ZeBook</strong>
Working with dynamic models for agriculture and environment
(Working with Dynamic Crop Models)
</p>
<p>Linked to book <strong>Working with Dynamic Crop Models</strong> (Elsevier), Third edition, 27 septembre 2018 by Wallach, Makowski, Jones and Brun. <a href="http://www.modelia.org/moodle/course/view.php?id=61">http://www.modelia.org/moodle/course/view.php?id=61</a>
</p>
<p>A full description of the models is in the book in appendix of the book.
</p>
<p>Chapter numbers have changed between Second edition and Third Edition. Here the chapter numbers in the demo were changed to fit to Third edition. But all materials available in Second edition are still available in this version. 
</p>
<p><strong>ACKNOWLEDGMENTS</strong>
The project &quot;Associate a level of error in predictions of models for agronomy&quot; (CASDAR 2010-2013) and the French network &quot;RMT modeling and agriculture&quot;, <a href="http://www.modelia.org">http://www.modelia.org</a>) have contributed to the development of this R package. This project and network are lead by ACTA (French Technical Institute for Agriculture) and was funded by a grant from the Ministry of Agriculture and Fishing of France.
</p>
<p><strong>Other contributions</strong>
Juliette Adrian, Master2 internship (ACTA, <a href="http://www.modelia.org/moodle/mod/resource/view.php?id=1027">http://www.modelia.org/moodle/mod/resource/view.php?id=1027</a>), january-jully 2013.
</p>
<p>Sylvain Toulet, Master2 internship (INRA, <a href="http://www.modelia.org/moodle/mod/resource/view.php?id=965">http://www.modelia.org/moodle/mod/resource/view.php?id=965</a>), january-jully 2012.
</p>


<h3>Author(s)</h3>

<p>Francois Brun (ACTA)  <a href="mailto:francois.brun@acta.asso.fr">francois.brun@acta.asso.fr</a>, David Makowski (INRA), Daniel Wallach (INRA), James W. Jones (U.of Florida),
</p>


<h3>References</h3>

<p>Working with Dynamic Crop Models (Elsevier), Third edition
<a href="http://www.modelia.org">http://www.modelia.org</a>
</p>

<hr>
<h2 id='AICf'>Calculate AIC, Akaike's Information Criterion</h2><span id='topic+AICf'></span>

<h3>Description</h3>

<p>This function calculate AIC criterion given a vector of observation, a vector of prediction and number of parameter.
Note that number of parameters should include variance.
AICcomplete is the same calculation of the AIC function of R (AICcomplete = n*log(RSS/n)+n+n*log(2*pi)+2*p, with p including variance).
AICshort is the calculation described in chapter 6 Working with crop model (AICshort =n*log(RSS/n)+2*p, with p including variance).
difference between AICcomplete and AICshort is AICcomplete-AICshort=n+n*log(2*pi)
As you use AIC to compare models (with different number of parameters) on a same data (with same n, number of observation), you can use AICshort or AICcomplete.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AICf(Yobs, Ypred, npar)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AICf_+3A_yobs">Yobs</code></td>
<td>
<p>: observed values</p>
</td></tr>
<tr><td><code id="AICf_+3A_ypred">Ypred</code></td>
<td>
<p>: prediction values from the model</p>
</td></tr>
<tr><td><code id="AICf_+3A_npar">npar</code></td>
<td>
<p>: number of parameters (should include variance that count for one supplementary parameter)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with AICcomplete and AICshort
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x=c(1,2,3,4,5)
y=c(1.2,1.8,3.5,4.3,5.5)
fit = lm(y~x)
AIC(fit)
AICf(y,predict(fit),3) # 3 parameters : intercept, slope and variance
</code></pre>

<hr>
<h2 id='Bean'>Bean gene-based models dataset</h2><span id='topic+Bean'></span>

<h3>Description</h3>

<p>Genetic data for the common bean (Phaseolus vulgaris L) that was based on a population
created by C. E. Vallejos (personal communication; also see Bhakta et al., 2015, 2017)
by crossing two widely-differing cultivars of bean (Calima, an Andean type, with Jamapa,
a Mesoamerican type). 
Bean$marker : Bean Marker Data
Bean$MET : Weather data on 5 Locations
Bean$QTL : QTL Data
Bean$modelpar : Dynamic Model parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bean
</code></pre>


<h3>Format</h3>

<p>a <code>List</code> including 4 data.frame  Bean$marker, Bean$MET, Bean$QTL, Bean$modelpar.</p>


<h3>Source</h3>

<p>C. E. Vallejos (personal communication) and Bhakta et al. (2015, 2017).
Bhakta, M. S., V. A. Jones, C. E. Vallejos. 2015. Punctuated distribution of recombination hotspots and demarcation of pericentromeric regions in Phaseolus vulgaris L.  PLoS ONE 10(1): https://doi.org/10.1371/journal.pone.0116822 
Bhakta, M. S., S. A. Gezan, J. A. Clavijo Michelangeli, M. Carvalho, L. Zhang, J. W. Jones, K. J. Boote, M. J. Correll, J. Beaver, J. M. Osorno, R. Colbert, I. Rao, S. Beebe, A. Gonzalez, J. Ricaurte, and C. E.do Vallejos, 2017. A predictive model for time-to-flower in the common bean based on  QTL and environmental variables. G3: Genes, Genomes, Genetics 7(12) 3901-3912. https://doi.org/10.1534/g3.117.300229.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># show the maker of JC1 to JC9 values for both parents (JAM and CAL)
# and 5 cRILS (RIJC001 to RIJC005)
Bean$marker[2:8,1:10]
# show the first value of weather data
head(Bean$MET)
# show the value of QTL
Bean$QTL[4:10,1:10]
# show the value of
Bean$modelpar
</code></pre>

<hr>
<h2 id='carbonsoil.model'>The CarbonSoil model - calculate daily values over designated time period</h2><span id='topic+carbonsoil.model'></span>

<h3>Description</h3>

<p>Simple dynamic model of soil carbon content, with a time step of one year. The equations that describe the dynamics of this system are adapted from the Henin-Dupuis model described in Jones et al. (2004). The soil carbon content is represented by a single state variable: the mass of carbon per unit land area in the top 20 cm of soil in a given year (Z, kg.ha-1). It is assumed that soil C is known in some year, taken as the initial year. 
The yearly change in soil C is the difference between input from crop biomass and loss.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carbonsoil.model(R, b, U, Z1, duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carbonsoil.model_+3A_r">R</code></td>
<td>
<p>: the fraction of soil carbon content lost per year</p>
</td></tr>
<tr><td><code id="carbonsoil.model_+3A_b">b</code></td>
<td>
<p>: the fraction of yearly crop biomass production left in the soil</p>
</td></tr>
<tr><td><code id="carbonsoil.model_+3A_u">U</code></td>
<td>
<p>: the amount of C in crop biomass production (constant or time series)</p>
</td></tr>
<tr><td><code id="carbonsoil.model_+3A_z1">Z1</code></td>
<td>
<p>: initial soil carbon content</p>
</td></tr>
<tr><td><code id="carbonsoil.model_+3A_duration">duration</code></td>
<td>
<p>:  duration of simulation (year))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Soil carbon years.
</p>

<hr>
<h2 id='carbonsoil.update'>The CarbonSoil model - calculate change in soil carbon for one year</h2><span id='topic+carbonsoil.update'></span>

<h3>Description</h3>

<p>Simple dynamic model of soil carbon content, with a time step of one year. The equations that describe the dynamics of this system are adapted from the Henin-Dupuis model described in Jones et al. (2004). The soil carbon content is represented by a single state variable: the mass of carbon per unit land area in the top 20 cm of soil in a given year (Z, kg.ha-1). It is assumed that soil C is known in some year, taken as the initial year. 
The yearly change in soil C is the difference between input from crop biomass and loss.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carbonsoil.update(Zy, R, b, Uy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carbonsoil.update_+3A_zy">Zy</code></td>
<td>
<p>: Soil carbon content for year</p>
</td></tr>
<tr><td><code id="carbonsoil.update_+3A_r">R</code></td>
<td>
<p>: the fraction of soil carbon content lost per year</p>
</td></tr>
<tr><td><code id="carbonsoil.update_+3A_b">b</code></td>
<td>
<p>: the fraction of yearly crop biomass production left in the soil</p>
</td></tr>
<tr><td><code id="carbonsoil.update_+3A_uy">Uy</code></td>
<td>
<p>: the amount of C in crop biomass production in the given year</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Soil carbon content for year+1.
</p>

<hr>
<h2 id='carcass_data'>Data of growth of beef cattle for Carcass model</h2><span id='topic+carcass_data'></span>

<h3>Description</h3>

<p>This dataset is a list of 5 dataframe.
list_individuals : identifiant for each indivual
energy : Ration (type ration), Individu, time (week), energie (?)
init_condition : Individu, Pvi (initial liveweight)
observation_dynamic :Ration (3 levels &quot;C&quot;,&quot;EM&quot;,&quot;F&quot;), individu, time, PVobs
observation_slaughter : Ration, individu, time, PVVobs, PVobs, ProtCobs, ProtNCobs, LipCobs, LipNCobs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass_data
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per plot.</p>


<h3>Source</h3>

<p>Agabriel, J. (com.pers.)
</p>

<hr>
<h2 id='carcass.define.param'>Define values of the parameters for the Carcass model</h2><span id='topic+carcass.define.param'></span>

<h3>Description</h3>

<p>Define parameters values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass.define.param(full = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carcass.define.param_+3A_full">full</code></td>
<td>
<p>: if TRUE, return the full description of distribution(default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup). A data.frame if full=TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>carcass.define.param(full=TRUE)
</code></pre>

<hr>
<h2 id='carcass.EMI.model'>The Carcass (growth of beef cattle) model with energy as input</h2><span id='topic+carcass.EMI.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong>
This model is proposed by Hoch et. al (2004) to represent the growth of cattle and the relative body composition of diferent types of animals depending on nuritionnal conditions.
It simulates the dynamics of changes in the composition of the body fat and proteins according to nutrient intake. The system is represented by four state variables: the protein and fat in the carcass (resp. ProtC and LIPC) and other tissues (resp. ProtNC and LipNC) grouped under the name of the fifth district (again, gastrointestinal tract, skin .. .). These variables depend on time, the time step used is dt = 1 day.
The model is defined by 20 equations, with a total of 18 parameters for the described process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass.EMI.model(protcmax, protncmax, alphac, alphanc, gammac, gammanc,
  lip0, lipc1, lipnc1, beta, delta, amW, b0c, b1c, b0nc, b1nc, c0, c1,
  energie, PVi, duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carcass.EMI.model_+3A_protcmax">protcmax</code></td>
<td>
<p>: amounts of protein in the carcass of the adult animal (kg)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_protncmax">protncmax</code></td>
<td>
<p>: amounts of protein in the 5th district of the adult animal (kg)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_alphac">alphac</code></td>
<td>
<p>: maximum protein synthesis rate in the frame (excluding basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_alphanc">alphanc</code></td>
<td>
<p>: maximum rate of protein synthesis in the 5th district (except basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_gammac">gammac</code></td>
<td>
<p>: Maximum rate of protein degradation in the frame (excluding basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_gammanc">gammanc</code></td>
<td>
<p>: maximum rate of protein degradation in the 5th district (except basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_lip0">lip0</code></td>
<td>
<p>: maximum lipid concentration to the theoretical physiological age (percent)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_lipc1">lipc1</code></td>
<td>
<p>: increase coefficient of the maximum lipid concentration with the physiological age of the carcase (percent)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_lipnc1">lipnc1</code></td>
<td>
<p>: increase coefficient of the highest lipid concentration with physiological age area in the 5th (percent)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_beta">beta</code></td>
<td>
<p>: lipid synthesis rate (j-1)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_delta">delta</code></td>
<td>
<p>: lipid degradation rate (d-1)</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_amw">amW</code></td>
<td>
<p>:</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_b0c">b0c</code></td>
<td>
<p>: coefficient of the allometric equation linking mass and lipid-protein carcass</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_b1c">b1c</code></td>
<td>
<p>:  exponent allometric equation linking mass and defatted protein carcass</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_b0nc">b0nc</code></td>
<td>
<p>: coefficient of the allometric equation linking mass and lipid-protein 5th district</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_b1nc">b1nc</code></td>
<td>
<p>: exponent allometric equation linking mass and lipid-protein 5th district</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_c0">c0</code></td>
<td>
<p>: coefficient of the allometric equation between live weight and live weight empty</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_c1">c1</code></td>
<td>
<p>: exponent allometric equation linking body weight and live weight empty</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_energie">energie</code></td>
<td>
<p>: Metabolizable energy available</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_pvi">PVi</code></td>
<td>
<p>: initial liveweight</p>
</td></tr>
<tr><td><code id="carcass.EMI.model_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with ProtC,LipC,ProtNC,LipNC,PV
</p>

<hr>
<h2 id='carcass.EMI.model2'>The Carcass model function for use with carcass.EMI.simule</h2><span id='topic+carcass.EMI.model2'></span>

<h3>Description</h3>

<p>see carcass.EMI.model for model description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass.EMI.model2(param, energie, PVi, duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carcass.EMI.model2_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="carcass.EMI.model2_+3A_energie">energie</code></td>
<td>
<p>: Metabolizable energy available</p>
</td></tr>
<tr><td><code id="carcass.EMI.model2_+3A_pvi">PVi</code></td>
<td>
<p>: initial liveweight</p>
</td></tr>
<tr><td><code id="carcass.EMI.model2_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with PV,ProtC,ProtNC,LipC,LipNC
</p>

<hr>
<h2 id='carcass.EMI.multi'>Wrapper function to run Carcass model on several animals with different conditions</h2><span id='topic+carcass.EMI.multi'></span>

<h3>Description</h3>

<p>wrapper function for multisimulation with carcass.EMI.model2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass.EMI.multi(param, list_individuals, energy, init_condition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carcass.EMI.multi_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="carcass.EMI.multi_+3A_list_individuals">list_individuals</code></td>
<td>
<p>: list of individuals</p>
</td></tr>
<tr><td><code id="carcass.EMI.multi_+3A_energy">energy</code></td>
<td>
<p>: Metabolizable energy available for all individuals</p>
</td></tr>
<tr><td><code id="carcass.EMI.multi_+3A_init_condition">init_condition</code></td>
<td>
<p>: initial condition for all individuals</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with id, ration ,duration, day, PV,ProtC,ProtNC,LipC,LipNC
</p>

<hr>
<h2 id='carcass.EMI.simule'>Wrapper function to the Carcass model for multiple sets of parameter values</h2><span id='topic+carcass.EMI.simule'></span>

<h3>Description</h3>

<p>Wrapper function to the Carcass model for multiple sets of parameter values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass.EMI.simule(X, energy, PVi, duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carcass.EMI.simule_+3A_x">X</code></td>
<td>
<p>: parameter matrix</p>
</td></tr>
<tr><td><code id="carcass.EMI.simule_+3A_energy">energy</code></td>
<td>
<p>: Metabolizable energy available</p>
</td></tr>
<tr><td><code id="carcass.EMI.simule_+3A_pvi">PVi</code></td>
<td>
<p>: initial liveweight</p>
</td></tr>
<tr><td><code id="carcass.EMI.simule_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with PV,ProtC,ProtNC,LipC,LipNC
</p>

<hr>
<h2 id='carcass.model'>The Carcass (growth of beef cattle) model</h2><span id='topic+carcass.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong>
</p>
<p>The model is defined by 20 equations, with a total of 19 parameters for the described process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass.model(protcmax, protncmax, alphac, alphanc, gammac, gammanc, lip0,
  lipc1, lipnc1, beta, delta, k, b0c, b1c, b0nc, b1nc, c0, c1, cem,
  duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carcass.model_+3A_protcmax">protcmax</code></td>
<td>
<p>: amounts of protein in the carcass of the adult animal (kg)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_protncmax">protncmax</code></td>
<td>
<p>: amounts of protein in the 5th district of the adult animal (kg)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_alphac">alphac</code></td>
<td>
<p>: maximum protein synthesis rate in the frame (excluding basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_alphanc">alphanc</code></td>
<td>
<p>: maximum rate of protein synthesis in the 5th district (except basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_gammac">gammac</code></td>
<td>
<p>: Maximum rate of protein degradation in the frame (excluding basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_gammanc">gammanc</code></td>
<td>
<p>: maximum rate of protein degradation in the 5th district (except basal metabolism) (j-1)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_lip0">lip0</code></td>
<td>
<p>: maximum lipid concentration to the theoretical physiological age (percent)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_lipc1">lipc1</code></td>
<td>
<p>: increase coefficient of the maximum lipid concentration with the physiological age of the carcase (percent)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_lipnc1">lipnc1</code></td>
<td>
<p>: increase coefficient of the highest lipid concentration with physiological age area in the 5th (percent)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_beta">beta</code></td>
<td>
<p>: lipid synthesis rate (j-1)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_delta">delta</code></td>
<td>
<p>: lipid degradation rate (d-1)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_k">k</code></td>
<td>
<p>: Parameter coefficient between the half-saturation of the Michaelis-Menten equation of the metabolic weight (MJ.kg^0.75)</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_b0c">b0c</code></td>
<td>
<p>: coefficient of the allometric equation linking mass and lipid-protein carcass</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_b1c">b1c</code></td>
<td>
<p>:  exponent allometric equation linking mass and defatted protein carcass</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_b0nc">b0nc</code></td>
<td>
<p>: coefficient of the allometric equation linking mass and lipid-protein 5th district</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_b1nc">b1nc</code></td>
<td>
<p>: exponent allometric equation linking mass and lipid-protein 5th district</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_c0">c0</code></td>
<td>
<p>: coefficient of the allometric equation between live weight and live weight empty</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_c1">c1</code></td>
<td>
<p>: exponent allometric equation linking body weight and live weight empty</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_cem">cem</code></td>
<td>
<p>:</p>
</td></tr>
<tr><td><code id="carcass.model_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with ProtC,LipC,ProtNC,LipNC,PV
</p>

<hr>
<h2 id='carrot.weevil.model'>Carrot weevil development model</h2><span id='topic+carrot.weevil.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong> Simple model of developpement of carrot weevil.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carrot.weevil.model(tbase = 7, tteggs = 130, ttlarvae = 256,
  ttprepupae = 114, ttpupae = 130, ttadultpreovi = 91, weather,
  sdate = 1, ldate = 360)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="carrot.weevil.model_+3A_tbase">tbase</code></td>
<td>
<p>: base temperature</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_tteggs">tteggs</code></td>
<td>
<p>: duration of eggs stage in degre.day</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_ttlarvae">ttlarvae</code></td>
<td>
<p>: duration of larvae stage in degre.day</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_ttprepupae">ttprepupae</code></td>
<td>
<p>: duration of prepupae stage in degre.day</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_ttpupae">ttpupae</code></td>
<td>
<p>: duration of pupae stage in degre.day</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_ttadultpreovi">ttadultpreovi</code></td>
<td>
<p>: duration of adult stage until egg laying in degre.day</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_sdate">sdate</code></td>
<td>
<p>: date to begin simulation (day of year) (default 1)</p>
</td></tr>
<tr><td><code id="carrot.weevil.model_+3A_ldate">ldate</code></td>
<td>
<p>: date to end simulation (day of year) (default 360)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily state variable
</p>

<hr>
<h2 id='chicks_data'>Data of growth of chicks</h2><span id='topic+chicks_data'></span>

<h3>Description</h3>

<p>This dataset content dynamic measurements of growth of chicks for different individuals and different strains.
The data comes from a selection experiment on chicken initiated by F. Ricard Research Station on Poultry of INRA Nouzilly.
The selection focuses on weight at 8 and 36 weeks and allowed to differentiate
the following five strains:
</p>
<p>strain 1 : X-+ (low at 8, but high at 36 weeks)
</p>
<p>strain 2 : X+- (high at 8, but low at 36 weeks)
</p>
<p>strain 3 : X++ (high weight at both ages)
</p>
<p>strain 4 : X- (low weight for both ages)
</p>
<p>strain 5 : X0 (control).
</p>
<p>This is a sub-sample of 50 females in the last generation of selection with weight data (in g) at 12 different ages   to complete measurement (0, 4, 6, 8, 12, 16, 20, 24; 28, 32, 36 and 40 weeks).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>carcass_data
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row : strain ; id_animal ; time (day) ; liveweight (g).</p>


<h3>Source</h3>

<p>Duval M., Robert-Granie C., Foulley J.-L. (2009) Estimation of heterogeneous variances in non linear mixed models via the SAEM algorithm with applications to growth curves in poultry. Journal de la Societe Francaise de Statistique, 150,65-83
</p>
<p>Donnet S., Foulley J.-L., Samson A. (2010) Bayesian analysis of growth curves using mixed models defined by stochastic differential equations. Biometrics 66, 733-741
</p>
<p>Jaffrezic F., Meza C., Foulley .J.-L., Lavielle M. (2006) The SAEM algorithm for the analysis of non linear traits in genetic studies. Genetics, Selection, Evolution,38, 583-
</p>
<p>This dataset was used in a training session Biobayes (France, 2011) training session.
</p>
<p>Albert I., Ancelet S., David O., Denis J.B., Makowski D., Parent E., Soubeyrand S. (2012) Methodes statistiques bayesiennes. Bases theoriques et applications en alimentation, environnement et genetique. FormaScience. Ecole-chercheurs INRA.
</p>

<hr>
<h2 id='cotton.model'>The Cotton model (dynamic for numbers of Cotton fruiting points).</h2><span id='topic+cotton.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong> TO COMPLETE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cotton.model(TESQ, PMAX, AFL, AL, AOP, P1, P2, P3, P4, P5, PF, PSF, TSQ, P,
  PR, PT, tend)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cotton.model_+3A_tesq">TESQ</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_pmax">PMAX</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_afl">AFL</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_al">AL</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_aop">AOP</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_p1">P1</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_p2">P2</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_p3">P3</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_p4">P4</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_p5">P5</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_pf">PF</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_psf">PSF</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_tsq">TSQ</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_p">P</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_pr">PR</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_pt">PT</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="cotton.model_+3A_tend">tend</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily state variable
</p>

<hr>
<h2 id='epirice.define.param'>Define values of the parameters for the Epirice model</h2><span id='topic+epirice.define.param'></span>

<h3>Description</h3>

<p>Define parameters values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epirice.define.param()
</code></pre>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup)
</p>

<hr>
<h2 id='epirice.model'>The Epirice model (Disease model for rice)</h2><span id='topic+epirice.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong> Adapted from Savary et al.(2012)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epirice.model(param, weather, sdate = 1, ldate = 120, H0 = 600)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epirice.model_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="epirice.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="epirice.model_+3A_sdate">sdate</code></td>
<td>
<p>: date to begin simulation (day of year) (default 1)</p>
</td></tr>
<tr><td><code id="epirice.model_+3A_ldate">ldate</code></td>
<td>
<p>: date to end simulation (day of year) (default 120)</p>
</td></tr>
<tr><td><code id="epirice.model_+3A_h0">H0</code></td>
<td>
<p>: initial number of plant's healthy sites(default 600)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily state variable
</p>


<h3>See Also</h3>

<p><code><a href="#topic+epirice.multi.simule">epirice.multi.simule</a></code>
</p>

<hr>
<h2 id='epirice.multi.simule'>Wrapper function to run Epirice multiple times (for multiple sets of inputs)</h2><span id='topic+epirice.multi.simule'></span>

<h3>Description</h3>

<p>Wrapper function for epirice.model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epirice.multi.simule(param, multi.simule, all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epirice.multi.simule_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="epirice.multi.simule_+3A_multi.simule">multi.simule</code></td>
<td>
<p>: matrix of n row definition of input variable : site, year and date of transplantation.</p>
</td></tr>
<tr><td><code id="epirice.multi.simule_+3A_all">all</code></td>
<td>
<p>: if you want a matrix combining multi.simule and output (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with AUDPC for each input vector
</p>


<h3>See Also</h3>

<p><code><a href="#topic+epirice.model">epirice.model</a></code>
</p>

<hr>
<h2 id='epirice.weather'>Read weather data for Epirice (southern Asia weather)</h2><span id='topic+epirice.weather'></span>

<h3>Description</h3>

<p>Read weather data and format them for epirice.model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>epirice.weather(working.year = NA, working.site = NA,
  weather = weather_SouthAsia)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="epirice.weather_+3A_working.year">working.year</code></td>
<td>
<p>: year for the subset of weather data (default=NA : all the year)</p>
</td></tr>
<tr><td><code id="epirice.weather_+3A_working.site">working.site</code></td>
<td>
<p>: site for the subset of weather data (default=NA : all the site)</p>
</td></tr>
<tr><td><code id="epirice.weather_+3A_weather">weather</code></td>
<td>
<p>: weather table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily weather data for one or several site(s) and for one or several year(s)
</p>

<hr>
<h2 id='evaluation.criteria'>Calcule multiple goodness-of-fit criteria</h2><span id='topic+evaluation.criteria'></span>

<h3>Description</h3>

<p>This function is depreciated and will be remove from the package in future versions. Please use goodness.of.fit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluation.criteria(Ypred, Yobs, draw.plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluation.criteria_+3A_ypred">Ypred</code></td>
<td>
<p>: prediction values from the model</p>
</td></tr>
<tr><td><code id="evaluation.criteria_+3A_yobs">Yobs</code></td>
<td>
<p>: observed values</p>
</td></tr>
<tr><td><code id="evaluation.criteria_+3A_draw.plot">draw.plot</code></td>
<td>
<p>: draw evaluation plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with the different evaluation criteria
</p>


<h3>Examples</h3>

<pre><code class='language-R'># observed and simulated values
obs&lt;-c(78,110,92,75,110,108,113,155,150)
sim&lt;-c(126,126,126,105,105,105,147,147,147)
evaluation.criteria(sim,obs,draw.plot=TRUE)
</code></pre>

<hr>
<h2 id='exponential.model'>The Exponential growth model of dynamic of population</h2><span id='topic+exponential.model'></span>

<h3>Description</h3>

<p>Exponential growth model of dynamic of population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exponential.model(a, Y0, duration = 40, dt = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exponential.model_+3A_a">a</code></td>
<td>
<p>: growth rate</p>
</td></tr>
<tr><td><code id="exponential.model_+3A_y0">Y0</code></td>
<td>
<p>: initial condition</p>
</td></tr>
<tr><td><code id="exponential.model_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="exponential.model_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with Y for each time step
</p>


<h3>See Also</h3>

<p><code><a href="#topic+verhulst.update">verhulst.update</a></code> for the update function of the Verhulst model.
</p>

<hr>
<h2 id='exponential.model.bis'>The Exponential growth model of dynamic of population - another form</h2><span id='topic+exponential.model.bis'></span>

<h3>Description</h3>

<p>Exponential growth model of dynamic of population - another form
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exponential.model.bis(a, Y0, duration = 40, dt = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exponential.model.bis_+3A_a">a</code></td>
<td>
<p>: growth rate</p>
</td></tr>
<tr><td><code id="exponential.model.bis_+3A_y0">Y0</code></td>
<td>
<p>: initial condition</p>
</td></tr>
<tr><td><code id="exponential.model.bis_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="exponential.model.bis_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with Y for each time step
</p>


<h3>See Also</h3>

<p><code><a href="#topic+verhulst.update">verhulst.update</a></code> for the update function of the Verhulst model.
</p>

<hr>
<h2 id='exponential.model.ie'>The Exponential growth model of dynamic of population - with improved Euler integration</h2><span id='topic+exponential.model.ie'></span>

<h3>Description</h3>

<p>Exponential growth model of dynamic of population - with improved Euler integration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exponential.model.ie(a, Y0, duration = 40, dt = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exponential.model.ie_+3A_a">a</code></td>
<td>
<p>: growth rate</p>
</td></tr>
<tr><td><code id="exponential.model.ie_+3A_y0">Y0</code></td>
<td>
<p>: initial condition</p>
</td></tr>
<tr><td><code id="exponential.model.ie_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="exponential.model.ie_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with Y for each time step
</p>


<h3>See Also</h3>

<p><code><a href="#topic+verhulst.update">verhulst.update</a></code> for the update function of the Verhulst model.
</p>

<hr>
<h2 id='goodness.of.fit'>Calcule multiple goodness-of-fit criteria</h2><span id='topic+goodness.of.fit'></span>

<h3>Description</h3>

<p>Calcule multiple goodness-of-fit criteria
</p>


<h3>Usage</h3>

<pre><code class='language-R'>goodness.of.fit(Yobs, Ypred, draw.plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="goodness.of.fit_+3A_yobs">Yobs</code></td>
<td>
<p>: observed values</p>
</td></tr>
<tr><td><code id="goodness.of.fit_+3A_ypred">Ypred</code></td>
<td>
<p>: prediction values from the model</p>
</td></tr>
<tr><td><code id="goodness.of.fit_+3A_draw.plot">draw.plot</code></td>
<td>
<p>: draw evaluation plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with the different evaluation criteria
</p>


<h3>Examples</h3>

<pre><code class='language-R'># observed and simulated values
obs&lt;-c(78,110,92,75,110,108,113,155,150)
sim&lt;-c(126,126,126,105,105,105,147,147,147)
goodness.of.fit(obs,sim,draw.plot=TRUE)
</code></pre>

<hr>
<h2 id='graph_epid'>Plot output of a Classical SEIR model for plant disease</h2><span id='topic+graph_epid'></span>

<h3>Description</h3>

<p>Plot the output of the Zadoks classical SEIR model for plant disease.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph_epid(out, typel = "s", all = TRUE, param = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graph_epid_+3A_out">out</code></td>
<td>
<p>: output of the zadoks.original.model</p>
</td></tr>
<tr><td><code id="graph_epid_+3A_typel">typel</code></td>
<td>
<p>: type of plot (default : s)</p>
</td></tr>
<tr><td><code id="graph_epid_+3A_all">all</code></td>
<td>
<p>: if all=true (default), plot all the state variable</p>
</td></tr>
<tr><td><code id="graph_epid_+3A_param">param</code></td>
<td>
<p>: if param (default), add the values of param on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+zakoks.original.model">zakoks.original.model</a></code>
</p>

<hr>
<h2 id='graph_epid_s'>Plot output of a Classical SEIR model for plant disease</h2><span id='topic+graph_epid_s'></span>

<h3>Description</h3>

<p>Plot the output of the Zadoks classical SEIR model for plant disease.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph_epid_s(out, typel = "s", all = TRUE, param = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graph_epid_s_+3A_out">out</code></td>
<td>
<p>: output of the zadoks.original.model</p>
</td></tr>
<tr><td><code id="graph_epid_s_+3A_typel">typel</code></td>
<td>
<p>: type of plot (default : s)</p>
</td></tr>
<tr><td><code id="graph_epid_s_+3A_all">all</code></td>
<td>
<p>: if all=true (default), plot all the state variable</p>
</td></tr>
<tr><td><code id="graph_epid_s_+3A_param">param</code></td>
<td>
<p>: if param (default), add the values of param on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+zakoks.original.model">zakoks.original.model</a></code>
</p>

<hr>
<h2 id='lactation.calf.model'>The Lactation model</h2><span id='topic+lactation.calf.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong>
This model is a model of lactating mammary glands of cattle described by Heather et al. (1983). This model was then inspired more complex models based on these principles.
This model simulates the dynamics of the production of cow's milk.
the system is represented by 6 state variables: change in hormone levels (H), the production and loss of milk secreting cells (CS), and removing the secretion of milk (M), the average quantity of milk contained in the animal (Mmean), the amount of milk removed (RM) and yield (Y).
The model has a time step dt = 0.1 for regular consumption of milk by a calf.
The model is defined by a few equations, with a total of fourteen parameters for the described process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lactation.calf.model(cu, kdiv, kdl, kdh, km, ksl, kr, ks, ksm, mh, mm, p,
  mum, rc, duration, dt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lactation.calf.model_+3A_cu">cu</code></td>
<td>
<p>: number of undifferentiated cells</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_kdiv">kdiv</code></td>
<td>
<p>: cell division rate, Michaelis-Menten constant</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_kdl">kdl</code></td>
<td>
<p>: constant degradation of milk</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_kdh">kdh</code></td>
<td>
<p>: rate of decomposition of the hormone</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_km">km</code></td>
<td>
<p>:  constant secretion of milk</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_ksl">ksl</code></td>
<td>
<p>: milk secretion rate, Michaelis-Menten constant</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_kr">kr</code></td>
<td>
<p>: average milk constant</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_ks">ks</code></td>
<td>
<p>: rate of degradation of the basal cells</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_ksm">ksm</code></td>
<td>
<p>: constant rate of degradation of milk secreting cells</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_mh">mh</code></td>
<td>
<p>: parameter</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_mm">mm</code></td>
<td>
<p>: storage Capacity milk the animal</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_p">p</code></td>
<td>
<p>: parameter</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_mum">mum</code></td>
<td>
<p>: setting the maximum rate of cell division</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_rc">rc</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="lactation.calf.model_+3A_dt">dt</code></td>
<td>
<p>: time step</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with CS, M, Mmoy, RM, day, week
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lactation.calf.model2(lactation.define.param()["nominal",],300,0.1)
</code></pre>

<hr>
<h2 id='lactation.calf.model2'>The Lactation model for use with lactation.calf.simule</h2><span id='topic+lactation.calf.model2'></span>

<h3>Description</h3>

<p>see lactation.calf.model for model description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lactation.calf.model2(param, duration, dt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lactation.calf.model2_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="lactation.calf.model2_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="lactation.calf.model2_+3A_dt">dt</code></td>
<td>
<p>: time step</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with CS, M, Mmoy, RM, day, week
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim=lactation.calf.model2(lactation.define.param()["nominal",],6+2*7, 0.1)
</code></pre>

<hr>
<h2 id='lactation.calf.simule'>Wrapper function to run the Lactation model for multiple sets of parameter values</h2><span id='topic+lactation.calf.simule'></span>

<h3>Description</h3>

<p>Wrapper function to run the Lactation model for multiple sets of parameter values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lactation.calf.simule(X, duration, dt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lactation.calf.simule_+3A_x">X</code></td>
<td>
<p>: parameter matrix</p>
</td></tr>
<tr><td><code id="lactation.calf.simule_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="lactation.calf.simule_+3A_dt">dt</code></td>
<td>
<p>: time step</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with : number of paramter vector (line number from X), week, CS, M, Mmoy, RM, day, week
</p>

<hr>
<h2 id='lactation.define.param'>Define values of the parameters for the Lactation model</h2><span id='topic+lactation.define.param'></span>

<h3>Description</h3>

<p>values from Heather et al. (1983) for different scenarios
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lactation.define.param(type = "calf")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lactation.define.param_+3A_type">type</code></td>
<td>
<p>: for which model version ? &quot;calf&quot; or &quot;machine&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lactation.define.param()
</code></pre>

<hr>
<h2 id='lactation.machine.model'>The Lactation model with milking machine</h2><span id='topic+lactation.machine.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong>
This model is a model of lactating mammary glands of cattle described by Heather et al. (1983). This model was then inspired more complex models based on these principles.
This model simulates the dynamics of the production of cow's milk.
the system is represented by 6 state variables: change in hormone levels (H), the production and loss of milk secreting cells (CS), and removing the secretion of milk (M), the average quantity of milk contained in the animal (Mmean), the amount of milk removed (RM) and yield (Y).
The model has a time step dt = 0.001 for milking machines.
The model is defined by a few equations, with a total of twenty parameters for the described process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lactation.machine.model(cu, kdiv, kdl, kdh, km, ksl, kr, ks, ksm, mh, mm,
  p, mum, rma, t1, t2, t3, t4, t5, t6, duration, dt, CSi, Mi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lactation.machine.model_+3A_cu">cu</code></td>
<td>
<p>: number of undifferentiated cells</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_kdiv">kdiv</code></td>
<td>
<p>: cell division rate, Michaelis-Menten constant</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_kdl">kdl</code></td>
<td>
<p>: constant degradation of milk</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_kdh">kdh</code></td>
<td>
<p>: rate of decomposition of the hormone</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_km">km</code></td>
<td>
<p>:  constant secretion of milk</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_ksl">ksl</code></td>
<td>
<p>: milk secretion rate, Michaelis-Menten constant</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_kr">kr</code></td>
<td>
<p>: average milk constant</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_ks">ks</code></td>
<td>
<p>: rate of degradation of the basal cells</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_ksm">ksm</code></td>
<td>
<p>: constant rate of degradation of milk secreting cells</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_mh">mh</code></td>
<td>
<p>: parameter</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_mm">mm</code></td>
<td>
<p>: storage Capacity milk the animal</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_p">p</code></td>
<td>
<p>: parameter</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_mum">mum</code></td>
<td>
<p>: setting the maximum rate of cell division</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_rma">rma</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_t1">t1</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_t2">t2</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_t3">t3</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_t4">t4</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_t5">t5</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_t6">t6</code></td>
<td>
<p>: parameter of milk m (t) function</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_dt">dt</code></td>
<td>
<p>: time step</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_csi">CSi</code></td>
<td>
<p>: initial Number of secretory cells</p>
</td></tr>
<tr><td><code id="lactation.machine.model_+3A_mi">Mi</code></td>
<td>
<p>: initial Quantity of milk in animal (kg)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with CS,M,Mmoy,RM
</p>

<hr>
<h2 id='lactation.machine.model2'>The Lactation model for use with lactation.machine.simule</h2><span id='topic+lactation.machine.model2'></span>

<h3>Description</h3>

<p>see lactation.calf.model for model description.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lactation.machine.model2(param, duration, dt, CSi, Mi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lactation.machine.model2_+3A_param">param</code></td>
<td>
<p>: a vector of parameters containning (cu,kdiv,kdl,kdh,km,ksl,kr,ks,ksm,mh,mm,p,mum,rma,t1,t2,t3,t4,t5,t6)(see lactation.model.machine)</p>
</td></tr>
<tr><td><code id="lactation.machine.model2_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
<tr><td><code id="lactation.machine.model2_+3A_dt">dt</code></td>
<td>
<p>: time step</p>
</td></tr>
<tr><td><code id="lactation.machine.model2_+3A_csi">CSi</code></td>
<td>
<p>: initial Number of secretory cells</p>
</td></tr>
<tr><td><code id="lactation.machine.model2_+3A_mi">Mi</code></td>
<td>
<p>: initial Quantity of milk in animal (kg)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with CS, M, Mmoy, RM, day, week
</p>

<hr>
<h2 id='magarey.define.param'>Define values of the parameters for the Magarey model</h2><span id='topic+magarey.define.param'></span>

<h3>Description</h3>

<p>Define values of the parameters for the Magarey model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>magarey.define.param(species = "unkown")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="magarey.define.param_+3A_species">species</code></td>
<td>
<p>: name of a species. By default, value for an &quot;unkown&quot; species are given. Other possibility are &quot;G.citricarpa&quot; or &quot;Kaki.fungus&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>magarey.define.param(species="G.citricarpa")
magarey.define.param(species="Kaki.fungus")
</code></pre>

<hr>
<h2 id='magarey.model'>The Magarey model</h2><span id='topic+magarey.model'></span>

<h3>Description</h3>

<p>Generic model of infection for foliar diseases caused by fungi (from Magarey et al.,2005).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>magarey.model(T, Tmin, Topt, Tmax, Wmin, Wmax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="magarey.model_+3A_t">T</code></td>
<td>
<p>: input variable. Either a scalar or a vector (for a weather series).</p>
</td></tr>
<tr><td><code id="magarey.model_+3A_tmin">Tmin</code></td>
<td>
<p>: parameter of minimal temperature for infection (degC)</p>
</td></tr>
<tr><td><code id="magarey.model_+3A_topt">Topt</code></td>
<td>
<p>: parameter of optimal temperature for infection (degC)</p>
</td></tr>
<tr><td><code id="magarey.model_+3A_tmax">Tmax</code></td>
<td>
<p>: parameter of maximal temperature for infection (degC)</p>
</td></tr>
<tr><td><code id="magarey.model_+3A_wmin">Wmin</code></td>
<td>
<p>: parameter of minimal wetness duration for infection (hour)</p>
</td></tr>
<tr><td><code id="magarey.model_+3A_wmax">Wmax</code></td>
<td>
<p>: parameter of maximal wetness duration for infection (hour)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Wetness duration (W, hour). Either a scalar or a vector depending on T.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(1:35, magarey.model (1:35,7, 18, 30, 10, 42), type="l", xlab="T", ylab="W")
</code></pre>

<hr>
<h2 id='magarey.model2'>The Magarey model, taking a vector of parameters as argument</h2><span id='topic+magarey.model2'></span>

<h3>Description</h3>

<p>Generic model of infection for foliar diseases caused by fungi (from Magarey et al.,2005).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>magarey.model2(T, param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="magarey.model2_+3A_t">T</code></td>
<td>
<p>: input variable. Either a scalar or a vector (for a weather series).</p>
</td></tr>
<tr><td><code id="magarey.model2_+3A_param">param</code></td>
<td>
<p>: parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>W : Wetness duration (hour). Either a scalar or a vector depending on T.
</p>

<hr>
<h2 id='magarey.simule'>Wrapper function to run the Magarey model multiple times (for multiple sets of inputs)</h2><span id='topic+magarey.simule'></span>

<h3>Description</h3>

<p>Wrapper function to run the Magarey model multiple times (for multiple sets of inputs)
</p>
<p>Example magarey.simule(magarey.define.param(),15)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>magarey.simule(X, T, all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="magarey.simule_+3A_x">X</code></td>
<td>
<p>: parameter matrix</p>
</td></tr>
<tr><td><code id="magarey.simule_+3A_t">T</code></td>
<td>
<p>: input variable, temperature</p>
</td></tr>
<tr><td><code id="magarey.simule_+3A_all">all</code></td>
<td>
<p>: if you want a matrix combining X and output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table with wetness duration (W) for each parameter vector
</p>

<hr>
<h2 id='maize_cir_rue_ear.model'>The Maize model with temperature dependent RUE, CumInt and ear growth</h2><span id='topic+maize_cir_rue_ear.model'></span>

<h3>Description</h3>

<p>Variant of the maize.model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize_cir_rue_ear.model(Tbase, RUE_max, K, alpha, LAImax, TTM, TTL,
  weather, sdate, ldate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize_cir_rue_ear.model_+3A_tbase">Tbase</code></td>
<td>
<p>:    parameter the baseline temperature for growth (degreeCelsius)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_rue_max">RUE_max</code></td>
<td>
<p>: parameter maximum radiation use efficiency (?)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_k">K</code></td>
<td>
<p>: parameter extinction coefficient (relation between leaf area index and intercepted radiation) (-)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_alpha">alpha</code></td>
<td>
<p>: parameter the relative rate of leaf area index increase for small values of leaf area index (?)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_laimax">LAImax</code></td>
<td>
<p>: parameter maximum leaf area index (-)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_ttm">TTM</code></td>
<td>
<p>:    parameter temperature sum for crop maturity (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_ttl">TTL</code></td>
<td>
<p>:    parameter temperature sum at the end of leaf area increase (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize_cir_rue_ear.model_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily TT, LAI,B
</p>

<hr>
<h2 id='maize_cir_rue.model'>The Maize model with temperature dependent RUE and CumInt</h2><span id='topic+maize_cir_rue.model'></span>

<h3>Description</h3>

<p>Variant of the maize.model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize_cir_rue.model(Tbase, RUE_max, K, alpha, LAImax, TTM, TTL, weather,
  sdate, ldate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize_cir_rue.model_+3A_tbase">Tbase</code></td>
<td>
<p>:    parameter the baseline temperature for growth (degreeCelsius)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_rue_max">RUE_max</code></td>
<td>
<p>: parameter maximum radiation use efficiency (?)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_k">K</code></td>
<td>
<p>: parameter extinction coefficient (relation between leaf area index and intercepted radiation) (-)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_alpha">alpha</code></td>
<td>
<p>: parameter the relative rate of leaf area index increase for small values of leaf area index (?)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_laimax">LAImax</code></td>
<td>
<p>: parameter maximum leaf area index (-)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_ttm">TTM</code></td>
<td>
<p>:    parameter temperature sum for crop maturity (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_ttl">TTL</code></td>
<td>
<p>:    parameter temperature sum at the end of leaf area increase (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize_cir_rue.model_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily TT, LAI,B
</p>

<hr>
<h2 id='maize_cir.model'>The Maize model with additional state variable CumInt</h2><span id='topic+maize_cir.model'></span>

<h3>Description</h3>

<p>Variant of the maize model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize_cir.model(Tbase, RUE, K, alpha, LAImax, TTM, TTL, weather, sdate,
  ldate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize_cir.model_+3A_tbase">Tbase</code></td>
<td>
<p>:    parameter the baseline temperature for growth (degreeCelsius)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_rue">RUE</code></td>
<td>
<p>: parameter radiation use efficiency (?)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_k">K</code></td>
<td>
<p>: parameter extinction coefficient (relation between leaf area index and intercepted radiation) (-)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_alpha">alpha</code></td>
<td>
<p>: parameter the relative rate of leaf area index increase for small values of leaf area index (?)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_laimax">LAImax</code></td>
<td>
<p>: parameter maximum leaf area index (-)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_ttm">TTM</code></td>
<td>
<p>:    parameter temperature sum for crop maturity (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_ttl">TTL</code></td>
<td>
<p>:    parameter temperature sum at the end of leaf area increase (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize_cir.model_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily TT, LAI,B
</p>

<hr>
<h2 id='maize.data_EuropeEU'>maize biomass and leaf area data</h2><span id='topic+maize.data_EuropeEU'></span>

<h3>Description</h3>

<p>&quot;observation data&quot; for several site and year (site-year) in Europe.
This data are fake observed date, derived from simulations with an error model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.data_EuropeEU
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per measurement.</p>


<h3>Source</h3>

<p>NA
</p>

<hr>
<h2 id='maize.data_MetaModelling'>dataset of simulation for maize final biomass</h2><span id='topic+maize.data_MetaModelling'></span>

<h3>Description</h3>

<p>Simulation data for several site and year (site-year) in Europe.
This data are from run of the original maize crop model with the R function maize.model for the
30 French sites and 17 years included in the dataset weather_FranceWest of the package ZeBook.
Our training dataset includes 510 (30 sites * 17 years) simulated biomass values and the 510 corresponding series of input values.
The input values are the three average temperatures T1, T2, T3 and
the three average radiations RAD1, RAD2, RAD3 computed on 3 periods of the growing season.
Period 1 : from day 1 to dat 50 (day of the year), period 2: from day 51 to day 100, period 3 : from day 101 to day 150.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.data_EuropeEU
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per simulation. Site, Year, T1, T2, T3, RAD1, RAD2, RAD3, B</p>


<h3>Source</h3>

<p>NA
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maize.model">maize.model</a></code>, <code><a href="#topic+weather_FranceWest">weather_FranceWest</a></code>
</p>

<hr>
<h2 id='maize.define.param'>Define values of the parameters for the Maize model</h2><span id='topic+maize.define.param'></span>

<h3>Description</h3>

<p>Define parameters values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.define.param()
</code></pre>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup)
</p>

<hr>
<h2 id='maize.model'>The basic Maize model.</h2><span id='topic+maize.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong>
This model is a dynamic model of crop growth for Maize cultivated in potential conditions.
The crop growth is represented by three state variables, leaf area per unit ground area (leaf area index, LAI), total biomass (B) and cumulative thermal time since plant emergence (TT). It is based on key concepts included in most crop models, at least for the &quot;potential production&quot; part. In fact, this model does not take into account any effects of soil water, nutrients, pests, or diseases,...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.model(Tbase, RUE, K, alpha, LAImax, TTM, TTL, weather, sdate, ldate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.model_+3A_tbase">Tbase</code></td>
<td>
<p>:    parameter the baseline temperature for growth (degreeCelsius)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_rue">RUE</code></td>
<td>
<p>: parameter radiation use efficiency (?)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_k">K</code></td>
<td>
<p>: parameter extinction coefficient (relation between leaf area index and intercepted radiation) (-)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_alpha">alpha</code></td>
<td>
<p>: parameter the relative rate of leaf area index increase for small values of leaf area index (?)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_laimax">LAImax</code></td>
<td>
<p>: parameter maximum leaf area index (-)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_ttm">TTM</code></td>
<td>
<p>:    parameter temperature sum for crop maturity (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_ttl">TTL</code></td>
<td>
<p>:    parameter temperature sum at the end of leaf area increase (degreeC.day)</p>
</td></tr>
<tr><td><code id="maize.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize.model_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.model_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The tree state variables are dynamic variables depending on days after emergence: TT(day), B(day), and LAI(day). The model has a time step dt of one day.<br />
The model is defined by a few equations, with a total of seven parameters for the described process.
<br /> (1) <code class="reqn">TT(day+1) = TT(day)+dTT(day)</code>
<br /> (2) <code class="reqn">B(day+1) = B(day)+dB(day)</code>
<br /> (3) <code class="reqn">LAI(day+1) = LAI(day)+dLAI(day)</code>
<br /> (4) <code class="reqn">dTT(day) = \max(\frac{TMIN(day)+TMAX(day)}{2}-Tbase;0)</code>
<br /> (5) <code class="reqn">dB(day) = RUE*(1-e^{-K*LAI(day)*I(day)}),\ if\ TT(day)\le TTM</code> 
<br /> <code class="reqn">dB(day) = 0,\ if\  TT(day)&gt;TTM</code>
<br /> (6) <code class="reqn">dLAI(day) = alpha*dTT(day)*LAI(day)*\max(LAImax-LAI(day);0),\ if \ TT(day)\le TTL </code>
<br /> <code class="reqn">dLAI(day) = 0,\ if\  TT(day)&gt;TTL </code>
</p>


<h3>Value</h3>

<p>data.frame with daily TT, LAI,B
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maize.model2">maize.model2</a></code>, <code><a href="#topic+maize.define.param">maize.define.param</a></code>, <code><a href="#topic+maize.simule">maize.simule</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+maize.simule240">maize.simule240</a></code>,<code><a href="#topic+maize.simule_multisy240">maize.simule_multisy240</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>weather = maize.weather(working.year=2010, working.site=30,weather_all=weather_EuropeEU)
maize.model(Tbase=7, RUE=1.85, K=0.7, alpha=0.00243, LAImax=7, TTM=1200, TTL=700,
  weather, sdate=100, ldate=250)
</code></pre>

<hr>
<h2 id='maize.model2'>The basic Maize model for use with maize.simule</h2><span id='topic+maize.model2'></span>

<h3>Description</h3>

<p>Wrapper pour maize.model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.model2(param, weather, sdate, ldate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.model2_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="maize.model2_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize.model2_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.model2_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily TT, LAI,B
</p>


<h3>Examples</h3>

<pre><code class='language-R'>weather = maize.weather(working.year=2010, working.site=30,weather_all=weather_EuropeEU)
maize.model2(maize.define.param()["nominal",], weather, sdate=100, ldate=250)
</code></pre>

<hr>
<h2 id='maize.muchow.graph'>Plot dynamic output of Muchow Maize model.</h2><span id='topic+maize.muchow.graph'></span>

<h3>Description</h3>

<p>Plot 6 graphs of main output variables of the Muchow Maize model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.muchow.graph(res)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.muchow.graph_+3A_res">res</code></td>
<td>
<p>: list of result from maize.muchow.model</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+mm.A.fct">mm.A.fct</a></code>, <code><a href="#topic+mm.LN.fct">mm.LN.fct</a></code>, <code><a href="#topic+mm.FAS.fct">mm.FAS.fct</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+mm.HI.fct">mm.HI.fct</a></code>,<code><a href="#topic+maize.muchow.model">maize.muchow.model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not run in package test
# res = maize.muchow.model(weather=maize.weather(working.year=2010, working.site=1))
# maize.muchow.graph(res)
</code></pre>

<hr>
<h2 id='maize.muchow.model'>Maize model, with harvest index and yield from Muchow et al. (1990)</h2><span id='topic+maize.muchow.model'></span>

<h3>Description</h3>

<p>Maize model, with harvest index and yield.
from Muchow RC, Sinclair TR, and Bennett JM (1990). Temperature and Solar Radiation
Effects on Potential Maize Yield across Locations AGRONOMY JOURNAL, VOL. 82, MARCH-APRIL 1990
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.muchow.model(Tbase1 = 8, TTE = 87, TTS = 67, Tbase2 = 0,
  TTRUE = 500, TTM = 1150, TLN = 20, AM = 596, RUE1 = 1.6,
  RUE2 = 1.2, K = 0.4, HImax = 0.5, Population = 7, sdate = 100,
  ldate = 365, weather)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.muchow.model_+3A_tbase1">Tbase1</code></td>
<td>
<p>: base temperature before silking(degC)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_tte">TTE</code></td>
<td>
<p>: Thermal units from sowing to emergence/leaf growth (degC.day)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_tts">TTS</code></td>
<td>
<p>: Thermal units from end of leaf growth to silking (degC.day)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_tbase2">Tbase2</code></td>
<td>
<p>: base temperature after silking (degC)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_ttrue">TTRUE</code></td>
<td>
<p>: Thermal units from silking for RUE change (degC.day)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_ttm">TTM</code></td>
<td>
<p>: Thermal units from silking to physiological maturity (degC.day)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_tln">TLN</code></td>
<td>
<p>: total number of leaves initiated (-)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_am">AM</code></td>
<td>
<p>: area of the largest leaf (cm2)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_rue1">RUE1</code></td>
<td>
<p>: radiation use efficiency (g.MJ-1) from crop emergence until 500 thermal units (base 0 &quot;C) after silking</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_rue2">RUE2</code></td>
<td>
<p>: radiation use efficiency (g.MJ-1) from 500 thermal units (base 0 &quot;C) after silking</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_k">K</code></td>
<td>
<p>: radiation extinction coefficient (-)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_himax">HImax</code></td>
<td>
<p>: maximum harvest index - genetic potential (-)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_population">Population</code></td>
<td>
<p>: number of plant per square meter (-)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date (day)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_ldate">ldate</code></td>
<td>
<p>: end of simulation (day)</p>
</td></tr>
<tr><td><code id="maize.muchow.model_+3A_weather">weather</code></td>
<td>
<p>: daily weather dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with TT1, TT2, STADE, LN, LAI, B, HI, YIELD
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mm.A.fct">mm.A.fct</a></code>, <code><a href="#topic+mm.LN.fct">mm.LN.fct</a></code>, <code><a href="#topic+mm.FAS.fct">mm.FAS.fct</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+mm.HI.fct">mm.HI.fct</a></code>,<code><a href="#topic+maize.muchow.graph">maize.muchow.graph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not run in package test
# res = maize.muchow.model(weather=maize.weather(working.year=2010, working.site=1))
#res$FinalYield
</code></pre>

<hr>
<h2 id='maize.multisy'>Wrapping function to run maize model on several site-years</h2><span id='topic+maize.multisy'></span>

<h3>Description</h3>

<p>Wrapping function to run maize model on several site-years
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.multisy(param, list_site_year, sdate, ldate,
  weather_all = weather_EuropeEU)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.multisy_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="maize.multisy_+3A_list_site_year">list_site_year</code></td>
<td>
<p>: vector of site-year</p>
</td></tr>
<tr><td><code id="maize.multisy_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.multisy_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
<tr><td><code id="maize.multisy_+3A_weather_all">weather_all</code></td>
<td>
<p>: weather data.frame for corresponding site-years</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with simulation for all site-years, with the first column sy indicating the site-years
</p>

<hr>
<h2 id='maize.multisy240'>Wrapper function to run Maize model for multiple sets of input variables (site-year) and give Biomass at day240.</h2><span id='topic+maize.multisy240'></span>

<h3>Description</h3>

<p>Wrapper function to run Maize model for multiple sets of input variables (site-year) and give Biomass at day240.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.multisy240(param, liste_sy, sdate, ldate,
  weather_all = weather_EuropeEU)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.multisy240_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="maize.multisy240_+3A_liste_sy">liste_sy</code></td>
<td>
<p>: vector of site-year</p>
</td></tr>
<tr><td><code id="maize.multisy240_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.multisy240_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
<tr><td><code id="maize.multisy240_+3A_weather_all">weather_all</code></td>
<td>
<p>: weather data table used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>mean biomass at day=240
</p>


<h3>Examples</h3>

<pre><code class='language-R'>maize.multisy240(maize.define.param()["nominal",],c("18-2006","64-2004") , sdate=100, ldate=250)
</code></pre>

<hr>
<h2 id='maize.RUEtemp'>Calculate effect of temperature on RUE for Maize</h2><span id='topic+maize.RUEtemp'></span>

<h3>Description</h3>

<p>Function to compute effect of temperature on RUE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.RUEtemp(T, RUE_max, T0, T1, T2, T3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.RUEtemp_+3A_t">T</code></td>
<td>
<p>: temperature</p>
</td></tr>
<tr><td><code id="maize.RUEtemp_+3A_rue_max">RUE_max</code></td>
<td>
<p>: maximum value for RUE</p>
</td></tr>
<tr><td><code id="maize.RUEtemp_+3A_t0">T0</code></td>
<td>
<p>: temperature parameter</p>
</td></tr>
<tr><td><code id="maize.RUEtemp_+3A_t1">T1</code></td>
<td>
<p>: temperature parameter</p>
</td></tr>
<tr><td><code id="maize.RUEtemp_+3A_t2">T2</code></td>
<td>
<p>: temperature parameter</p>
</td></tr>
<tr><td><code id="maize.RUEtemp_+3A_t3">T3</code></td>
<td>
<p>: temperature parameter</p>
</td></tr>
</table>


<h3>Value</h3>

<p>RUE value
</p>

<hr>
<h2 id='maize.simule'>Wrapper function to run Maize model for multiple sets of parameter values</h2><span id='topic+maize.simule'></span>

<h3>Description</h3>

<p>wrapper for maize.model2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.simule(X, weather, sdate, ldate, all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.simule_+3A_x">X</code></td>
<td>
<p>: matrix of n row vectors of 7 parameters</p>
</td></tr>
<tr><td><code id="maize.simule_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize.simule_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.simule_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
<tr><td><code id="maize.simule_+3A_all">all</code></td>
<td>
<p>: if you want a matrix combining X and output (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with maximum biomass for each parameter vector
</p>

<hr>
<h2 id='maize.simule_multisy240'>Wrapper function to run Maize model for multiple sets of parameter values (virtual design) and multiple sets of input variables (site-year) and give Biomass at day240</h2><span id='topic+maize.simule_multisy240'></span>

<h3>Description</h3>

<p>Wrapper function to run Maize model for multiple sets of input variables (site-year) and give Biomass at day240.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.simule_multisy240(X, liste_sy, sdate, ldate, all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.simule_multisy240_+3A_x">X</code></td>
<td>
<p>: matrix of n row vectors of 7 parameters</p>
</td></tr>
<tr><td><code id="maize.simule_multisy240_+3A_liste_sy">liste_sy</code></td>
<td>
<p>: vector of site-year</p>
</td></tr>
<tr><td><code id="maize.simule_multisy240_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.simule_multisy240_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
<tr><td><code id="maize.simule_multisy240_+3A_all">all</code></td>
<td>
<p>: if you want a matrix combining X and output (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of mean biomass at day=240 for all combinations of parameters of X
</p>


<h3>Examples</h3>

<pre><code class='language-R'>maize.simule_multisy240(maize.define.param(),c("18-2006","64-2004"),
  sdate=100, ldate=250, all=FALSE)
</code></pre>

<hr>
<h2 id='maize.simule240'>Wrapper function to run Maize model multiple times for multiple sets of parameter values and give Biomass at day240</h2><span id='topic+maize.simule240'></span>

<h3>Description</h3>

<p>Wrapper function for multiple simulation with Maize model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.simule240(X, weather, sdate, ldate, all = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.simule240_+3A_x">X</code></td>
<td>
<p>: matrix of n row vectors of 7 parameters</p>
</td></tr>
<tr><td><code id="maize.simule240_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="maize.simule240_+3A_sdate">sdate</code></td>
<td>
<p>: sowing date</p>
</td></tr>
<tr><td><code id="maize.simule240_+3A_ldate">ldate</code></td>
<td>
<p>: last date</p>
</td></tr>
<tr><td><code id="maize.simule240_+3A_all">all</code></td>
<td>
<p>: if you want a matrix combining X and output (default = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of biomass at day=240 for all combinations of parameters of X
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sy="18-2006"
weather = maize.weather(working.year=strsplit(sy,"-")[[1]][2],
  working.site=strsplit(sy,"-")[[1]][1],weather_all=weather_EuropeEU)
maize.simule240(maize.define.param(),weather, sdate=100, ldate=250, all=FALSE)
</code></pre>

<hr>
<h2 id='maize.weather'>Read weather data for the Maize model</h2><span id='topic+maize.weather'></span>

<h3>Description</h3>

<p>Function to read weather data and format them for maize.model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maize.weather(working.year = NA, working.site = NA,
  weather_all = weather_FranceWest)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="maize.weather_+3A_working.year">working.year</code></td>
<td>
<p>: year for the subset of weather data (default=NA : all the year)</p>
</td></tr>
<tr><td><code id="maize.weather_+3A_working.site">working.site</code></td>
<td>
<p>: site for the subset of weather data (default=NA : all the site)</p>
</td></tr>
<tr><td><code id="maize.weather_+3A_weather_all">weather_all</code></td>
<td>
<p>: weather data base (default=weather_FranceWest)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily weather data for one or several site(s) and for one or several year(s)
</p>

<hr>
<h2 id='mm.A.fct'>Expanded leaf area function for Muchow et al. (1990) Maize model</h2><span id='topic+mm.A.fct'></span>

<h3>Description</h3>

<p>Compute fully expanded area by leaf number (A, cm2)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm.A.fct(LN, AM, LNM, a1 = -0.0344, a2 = 0.000731)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mm.A.fct_+3A_ln">LN</code></td>
<td>
<p>: Leaf number</p>
</td></tr>
<tr><td><code id="mm.A.fct_+3A_am">AM</code></td>
<td>
<p>: area of the largest leaf (cm2)</p>
</td></tr>
<tr><td><code id="mm.A.fct_+3A_lnm">LNM</code></td>
<td>
<p>: leaf number having the largest area (-)</p>
</td></tr>
<tr><td><code id="mm.A.fct_+3A_a1">a1</code></td>
<td>
<p>: coefficient of the statistical relation (default : -0.0344)</p>
</td></tr>
<tr><td><code id="mm.A.fct_+3A_a2">a2</code></td>
<td>
<p>: coefficient of the statistical relation (default : 0.000731)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of Expanded leaf area
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maize.muchow.model">maize.muchow.model</a></code>, <code><a href="#topic+mm.LN.fct">mm.LN.fct</a></code>, <code><a href="#topic+mm.FAS.fct">mm.FAS.fct</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+mm.HI.fct">mm.HI.fct</a></code>,<code><a href="#topic+maize.muchow.graph">maize.muchow.graph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>barplot(mm.A.fct(LN=1:20, AM=750, LNM=12),names.arg=1:20,
horiz=TRUE,xlab="leaf area (cm2)",ylab="leaf number")
</code></pre>

<hr>
<h2 id='mm.FAS.fct'>Senescence function for Muchow et al. (1990) Maize model</h2><span id='topic+mm.FAS.fct'></span>

<h3>Description</h3>

<p>Senesced fraction of total leaf area (FAS) increase with thermal units (TU) from emergence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm.FAS.fct(TT, TTE, c1 = 0.00161, c2 = 0.00328)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mm.FAS.fct_+3A_tt">TT</code></td>
<td>
<p>: Thermal time (degC.day)</p>
</td></tr>
<tr><td><code id="mm.FAS.fct_+3A_tte">TTE</code></td>
<td>
<p>: Thermal units from sowing to emergence/leaf growth (degC.day)</p>
</td></tr>
<tr><td><code id="mm.FAS.fct_+3A_c1">c1</code></td>
<td>
<p>: coefficient of the statistical relation (default : 0.00161)</p>
</td></tr>
<tr><td><code id="mm.FAS.fct_+3A_c2">c2</code></td>
<td>
<p>: coefficient of the statistical relation (default : 0.00328)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Senesced fraction of total leaf area
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maize.muchow.model">maize.muchow.model</a></code>, <code><a href="#topic+mm.A.fct">mm.A.fct</a></code>, <code><a href="#topic+mm.LN.fct">mm.LN.fct</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+mm.HI.fct">mm.HI.fct</a></code>,<code><a href="#topic+maize.muchow.graph">maize.muchow.graph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(1:2500,mm.FAS.fct(1:2500,TTE=87))
</code></pre>

<hr>
<h2 id='mm.HI.fct'>Harvest index function for Muchow et al. (1990) Maize model</h2><span id='topic+mm.HI.fct'></span>

<h3>Description</h3>

<p>Compute the harvest index.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm.HI.fct(day, daysilking, HImax, d1 = 0.015, d2 = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mm.HI.fct_+3A_day">day</code></td>
<td>
<p>: day of the year</p>
</td></tr>
<tr><td><code id="mm.HI.fct_+3A_daysilking">daysilking</code></td>
<td>
<p>: day of the year for silking (day)</p>
</td></tr>
<tr><td><code id="mm.HI.fct_+3A_himax">HImax</code></td>
<td>
<p>: maximum harvest index - genetic potential (-)</p>
</td></tr>
<tr><td><code id="mm.HI.fct_+3A_d1">d1</code></td>
<td>
<p>: coefficient of the statistical relation (day-1, default : 0.015)</p>
</td></tr>
<tr><td><code id="mm.HI.fct_+3A_d2">d2</code></td>
<td>
<p>: coefficient of the statistical relation (day, default : 3)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Harvest index
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maize.muchow.model">maize.muchow.model</a></code>, <code><a href="#topic+mm.A.fct">mm.A.fct</a></code>, <code><a href="#topic+mm.LN.fct">mm.LN.fct</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+mm.FAS.fct">mm.FAS.fct</a></code>,<code><a href="#topic+maize.muchow.graph">maize.muchow.graph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(1:350, mm.HI.fct(1:350, 200, 0.75), type="l")
</code></pre>

<hr>
<h2 id='mm.LN.fct'>Leaf number function for Muchow et al. (1990) Maize model</h2><span id='topic+mm.LN.fct'></span>

<h3>Description</h3>

<p>Leaf number as a function of thermal time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mm.LN.fct(TT1, TTE, b1 = 2.5, b2 = 0.00225, TLN = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mm.LN.fct_+3A_tt1">TT1</code></td>
<td>
<p>: Thermal time from sowing (degC.day)</p>
</td></tr>
<tr><td><code id="mm.LN.fct_+3A_tte">TTE</code></td>
<td>
<p>: Thermal units from sowing to emergence/leaf growth (degC.day)</p>
</td></tr>
<tr><td><code id="mm.LN.fct_+3A_b1">b1</code></td>
<td>
<p>: coefficient of the statistical relation (default : 2.5)</p>
</td></tr>
<tr><td><code id="mm.LN.fct_+3A_b2">b2</code></td>
<td>
<p>: coefficient of the statistical relation (default : 0.00225)</p>
</td></tr>
<tr><td><code id="mm.LN.fct_+3A_tln">TLN</code></td>
<td>
<p>: total number of leaves initiated (-)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Leaf number
</p>


<h3>See Also</h3>

<p><code><a href="#topic+maize.muchow.model">maize.muchow.model</a></code>, <code><a href="#topic+mm.A.fct">mm.A.fct</a></code>, <code><a href="#topic+mm.FAS.fct">mm.FAS.fct</a></code>, <code><a href="#topic+maize.multisy">maize.multisy</a></code>,
<code><a href="#topic+mm.HI.fct">mm.HI.fct</a></code>,<code><a href="#topic+maize.muchow.graph">maize.muchow.graph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(1:1000,mm.LN.fct(1:1000,TTE=87))
</code></pre>

<hr>
<h2 id='param.rtriangle'>Generate a random plan as a data frame. Columns are parameters. Values have triangle distribution</h2><span id='topic+param.rtriangle'></span>

<h3>Description</h3>

<p>according to nominal, minimal and maximal values defined in a model.factors matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.rtriangle(model.factors, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.rtriangle_+3A_model.factors">model.factors</code></td>
<td>
<p>: matrix defining nominal, minimal (binf), maximal values (bsup) for a set of p parameters</p>
</td></tr>
<tr><td><code id="param.rtriangle_+3A_n">N</code></td>
<td>
<p>: size of sample</p>
</td></tr>
</table>


<h3>Value</h3>

<p>parameter matrix of dim = (N, p)
</p>

<hr>
<h2 id='param.runif'>Generate a random plan as a data frame. Columns are parameters. Values have uniform distribution</h2><span id='topic+param.runif'></span>

<h3>Description</h3>

<p>according to minimal and maximal values defined in a model.factors matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>param.runif(model.factors, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="param.runif_+3A_model.factors">model.factors</code></td>
<td>
<p>: matrix defining minimal (binf) and maximal values (bsup) for a set of p parameters</p>
</td></tr>
<tr><td><code id="param.runif_+3A_n">N</code></td>
<td>
<p>: size of sample</p>
</td></tr>
</table>


<h3>Value</h3>

<p>parameter matrix of dim = (N, p)
</p>

<hr>
<h2 id='population.age.matrix.model'>The PopulationAge model (Population Dynamics with Age Classes) - matrix form</h2><span id='topic+population.age.matrix.model'></span>

<h3>Description</h3>

<p>Population Dynamics Model with Age Classes for an insect
Exactly the same model as population.age.model, but written as a matrix computation. It's possible for this model. It's really more efficient and reduce computer time by 6!
7 states variables
E : egg stage. homogenous population (density) (number per ha)
L1 : larvae1 stage. homogenous population (density) (number per ha)
L2 : larvae2 stage. homogenous population (density) (number per ha)
L3 : larvae3 stage. homogenous population (density) (number per ha)
L4 : larvae4 stage. homogenous population (density) (number per ha)
P : pupae stage. homogenous population (density) (number per ha)
A : adult stage. homogenous population (density) (number per ha)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>population.age.matrix.model(rb = 3.5, mE = 0.017, rE = 0.172,
  m1 = 0.06, r12 = 0.217, m2 = 0.032, r23 = 0.313, m3 = 0.022,
  r34 = 0.222, m4 = 0.02, r4P = 0.135, mP = 0.02, rPA = 0.099,
  mA = 0.027, iA = 0, duration = 100, dt = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="population.age.matrix.model_+3A_rb">rb</code></td>
<td>
<p>: eggs laid per adult per unit area (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_me">mE</code></td>
<td>
<p>: relative mortality rate of egg (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_re">rE</code></td>
<td>
<p>: eggs hatch (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_m1">m1</code></td>
<td>
<p>: relative mortality rate of larvae L1 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_r12">r12</code></td>
<td>
<p>: relative rate L1-&gt;L2 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_m2">m2</code></td>
<td>
<p>: relative mortality rate of larvae L2 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_r23">r23</code></td>
<td>
<p>: relative rate L2-&gt;L3 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_m3">m3</code></td>
<td>
<p>: relative mortality rate of larvae L3 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_r34">r34</code></td>
<td>
<p>: relative rate L3-&gt;L4 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_m4">m4</code></td>
<td>
<p>: relative mortality rate of larvae L4 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_r4p">r4P</code></td>
<td>
<p>: relative rate L4-&gt;P (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_mp">mP</code></td>
<td>
<p>: relative mortality rate of purpae (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_rpa">rPA</code></td>
<td>
<p>: relative rate P-&gt;A (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_ma">mA</code></td>
<td>
<p>: relative mortality rate of adult L1 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_ia">iA</code></td>
<td>
<p>: input rate of adult (unit.day-1)</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_duration">duration</code></td>
<td>
<p>: simulation duration</p>
</td></tr>
<tr><td><code id="population.age.matrix.model_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with values for state variables for each time step.
</p>

<hr>
<h2 id='population.age.model'>The PopulationAge model (Population Dynamics with Age Classes)</h2><span id='topic+population.age.model'></span>

<h3>Description</h3>

<p>Population Dynamics Model with Age Classes for an insect
</p>


<h3>Usage</h3>

<pre><code class='language-R'>population.age.model(rb = 3.5, mE = 0.017, rE = 0.172, m1 = 0.06,
  r12 = 0.217, m2 = 0.032, r23 = 0.313, m3 = 0.022, r34 = 0.222,
  m4 = 0.02, r4P = 0.135, mP = 0.02, rPA = 0.099, mA = 0.027,
  iA = 0, duration = 100, dt = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="population.age.model_+3A_rb">rb</code></td>
<td>
<p>: eggs laid per adult per unit area (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_me">mE</code></td>
<td>
<p>: relative mortality rate of egg (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_re">rE</code></td>
<td>
<p>: eggs hatch (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_m1">m1</code></td>
<td>
<p>: relative mortality rate of larvae L1 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_r12">r12</code></td>
<td>
<p>: relative rate L1-&gt;L2 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_m2">m2</code></td>
<td>
<p>: relative mortality rate of larvae L2 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_r23">r23</code></td>
<td>
<p>: relative rate L2-&gt;L3 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_m3">m3</code></td>
<td>
<p>: relative mortality rate of larvae L3 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_r34">r34</code></td>
<td>
<p>: relative rate L3-&gt;L4 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_m4">m4</code></td>
<td>
<p>: relative mortality rate of larvae L4 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_r4p">r4P</code></td>
<td>
<p>: relative rate L4-&gt;P (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_mp">mP</code></td>
<td>
<p>: relative mortality rate of purpae (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_rpa">rPA</code></td>
<td>
<p>: relative rate P-&gt;A (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_ma">mA</code></td>
<td>
<p>: relative mortality rate of adult L1 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_ia">iA</code></td>
<td>
<p>: input rate of adult (unit.day-1)</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_duration">duration</code></td>
<td>
<p>: simulation duration</p>
</td></tr>
<tr><td><code id="population.age.model_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with values for state variables for each time step.
</p>

<hr>
<h2 id='population.age.model.ode'>The PopulationAge model (Population Dynamics with Age Classes) - ode form</h2><span id='topic+population.age.model.ode'></span>

<h3>Description</h3>

<p>Population Dynamics Model with Age Classes for an insect
Exactly the same model as population.age.model, but written as an ordinary differential equation system (ode) with deSolve package.
7 states variables
E : egg stage. homogenous population (density) (number per ha)
L1 : larvae1 stage. homogenous population (density) (number per ha)
L2 : larvae2 stage. homogenous population (density) (number per ha)
L3 : larvae3 stage. homogenous population (density) (number per ha)
L4 : larvae4 stage. homogenous population (density) (number per ha)
P : pupae stage. homogenous population (density) (number per ha)
A : adult stage. homogenous population (density) (number per ha)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>population.age.model.ode(rb = 3.5, mE = 0.017, rE = 0.172,
  m1 = 0.06, r12 = 0.217, m2 = 0.032, r23 = 0.313, m3 = 0.022,
  r34 = 0.222, m4 = 0.02, r4P = 0.135, mP = 0.02, rPA = 0.099,
  mA = 0.027, iA = 0, duration = 100, dt = 1, method = "euler")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="population.age.model.ode_+3A_rb">rb</code></td>
<td>
<p>: eggs laid per adult per unit area (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_me">mE</code></td>
<td>
<p>: relative mortality rate of egg (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_re">rE</code></td>
<td>
<p>: eggs hatch (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_m1">m1</code></td>
<td>
<p>: relative mortality rate of larvae L1 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_r12">r12</code></td>
<td>
<p>: relative rate L1-&gt;L2 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_m2">m2</code></td>
<td>
<p>: relative mortality rate of larvae L2 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_r23">r23</code></td>
<td>
<p>: relative rate L2-&gt;L3 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_m3">m3</code></td>
<td>
<p>: relative mortality rate of larvae L3 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_r34">r34</code></td>
<td>
<p>: relative rate L3-&gt;L4 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_m4">m4</code></td>
<td>
<p>: relative mortality rate of larvae L4 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_r4p">r4P</code></td>
<td>
<p>: relative rate L4-&gt;P (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_mp">mP</code></td>
<td>
<p>: relative mortality rate of purpae (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_rpa">rPA</code></td>
<td>
<p>: relative rate P-&gt;A (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_ma">mA</code></td>
<td>
<p>: relative mortality rate of adult L1 (day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_ia">iA</code></td>
<td>
<p>: input rate of adult (unit.day-1)</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_duration">duration</code></td>
<td>
<p>: simulation duration</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
<tr><td><code id="population.age.model.ode_+3A_method">method</code></td>
<td>
<p>: integration method (euler, rk4,...)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with values for state variables for each time step.
</p>

<hr>
<h2 id='predator.prey.model'>The PredatorPrey model (Predator-Prey Lotka-Volterra with logistic equation for prey)</h2><span id='topic+predator.prey.model'></span>

<h3>Description</h3>

<p>Predator-Prey Lotka-Volterra model (with logistic prey)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predator.prey.model(grH = 1, kH = 10, mrH = 0.2, eff = 0.5,
  mrA = 0.2, H0 = 1, A0 = 2, duration = 200, dt = 1,
  method = "euler")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predator.prey.model_+3A_grh">grH</code></td>
<td>
<p>: relative rate of prey population growth</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_kh">kH</code></td>
<td>
<p>: environment carrying capacity for prey (number per ha)</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_mrh">mrH</code></td>
<td>
<p>: maximum predation rate (number per predator and per prey per day)</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_eff">eff</code></td>
<td>
<p>: efficiency, growth of predator population depending on predation (-)</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_mra">mrA</code></td>
<td>
<p>: mortality of predator (-)</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_h0">H0</code></td>
<td>
<p>: size of population of prey, at time 0</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_a0">A0</code></td>
<td>
<p>: size of population of predator, at time 0</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_duration">duration</code></td>
<td>
<p>: simulation duration</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_dt">dt</code></td>
<td>
<p>: time step for integration</p>
</td></tr>
<tr><td><code id="predator.prey.model_+3A_method">method</code></td>
<td>
<p>: integration method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily H and A
</p>

<hr>
<h2 id='q.arg.fast.runif'>Build the q.arg argument for the  FAST function (sensitivity analysis)</h2><span id='topic+q.arg.fast.runif'></span>

<h3>Description</h3>

<p>according to minimal and maximal values defined in a model.factors matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>q.arg.fast.runif(model.factors)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="q.arg.fast.runif_+3A_model.factors">model.factors</code></td>
<td>
<p>: matrix defining minimal (binf) and maximal values (bsup) for a set of p parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of list
</p>

<hr>
<h2 id='seedweight.data'>Wheat grain weight measurements after anthesis</h2><span id='topic+seedweight.data'></span>

<h3>Description</h3>

<p>Darroch and Baker (1990) studied grain filling in three spring wheat genotypes.
The data are seed weights of the spring wheat cultivar Neepawa in three different years 1986, 1987, 1988.
These data were numerised from figure of the article, so they present slight difference with original data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seedweight.data
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per measurement. DD = Degree Days after anthesis; seedweight = Wheat grain weight (mg)</p>


<h3>Source</h3>

<p>Darroch, B A, and R J Baker. 1990. &quot;Grain filling in three spring wheat genotypes: statistical analysis.&quot; Crop Science 30(3):525-529.
</p>

<hr>
<h2 id='seedweight.model'>The SeedWeight model</h2><span id='topic+seedweight.model'></span>

<h3>Description</h3>

<p>The SeedWeight model is a logistic model of grain weight over time in wheat. The model was proposed by Darroch &amp; Baker (1990) in a study of grain filling in three spring wheat genotypes.
This model has a single input variable, degree days after anthesis noted DD, and three parameters, noted W, B and C.
Parameters are estimated from observations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>seedweight.model(DD, W, B, C)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="seedweight.model_+3A_dd">DD</code></td>
<td>
<p>: degree days after anthesis</p>
</td></tr>
<tr><td><code id="seedweight.model_+3A_w">W</code></td>
<td>
<p>: parameter of the model</p>
</td></tr>
<tr><td><code id="seedweight.model_+3A_b">B</code></td>
<td>
<p>: parameter of the model</p>
</td></tr>
<tr><td><code id="seedweight.model_+3A_c">C</code></td>
<td>
<p>: parameter of the model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Seed Weight for each TT
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(1:500,seedweight.model(1:500, W=30,B=4,C=0.020),type="l",
   xlab="degree days after anthesis", ylab="grain weight")
</code></pre>

<hr>
<h2 id='Sunflower_Phomopsis'>Phomopsis stem canker observations for Sunflower</h2><span id='topic+Sunflower_Phomopsis'></span>

<h3>Description</h3>

<p>This dataset contains fraction intercepted photosynthetically active radiation (IPAR) and corresponding percent of girdling lesions at harvest (pclesions) for 43 fields.
Phomopsis stem canker is a worldwide fungal disease of sunflower, which causes stem girdling lesions and a consequent reduction in yield.  One wants to decide if the number of girdling lesions at harvest in the absence of early treatment will exceed 15
This data frame consist of a sample of fields with values for IPAR  (ipar) and for the percent of girdling lesions at harvest (pclesions).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sunflower_Phomopsis
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per observation.</p>


<h3>Source</h3>

<p>Debaeke, P., &amp; Estragnat, A. (2009). Crop canopy indicators for the early prediction of Phomopsis stem canker (Diaporthe helianthi) in sunflower. Crop Protection, 28(9), 792-801. doi:10.1016/j.cropro.2009.04.011
</p>

<hr>
<h2 id='threshold.measures'>Computation of threshold.measures</h2><span id='topic+threshold.measures'></span>

<h3>Description</h3>

<p>Computation of threshold.measures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshold.measures(Yobs, Ypred, p, d, units = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threshold.measures_+3A_yobs">Yobs</code></td>
<td>
<p>: observed values</p>
</td></tr>
<tr><td><code id="threshold.measures_+3A_ypred">Ypred</code></td>
<td>
<p>: prediction values from the model</p>
</td></tr>
<tr><td><code id="threshold.measures_+3A_p">p</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="threshold.measures_+3A_d">d</code></td>
<td>
<p>: TO COMPLETE</p>
</td></tr>
<tr><td><code id="threshold.measures_+3A_units">units</code></td>
<td>
<p>: units</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with the different evaluation criteria
</p>


<h3>Examples</h3>

<pre><code class='language-R'># observed and simulated values
obs&lt;-c(78,110,92,75,110,108,113,155,150)
sim&lt;-c(126,126,126,105,105,105,147,147,147)
threshold.measures(obs,sim,80,1.0)
</code></pre>

<hr>
<h2 id='verhulst.model'>The Verhulst (logistic) model - calculate daily values over designated time period</h2><span id='topic+verhulst.model'></span>

<h3>Description</h3>

<p>The Verhulst (logistic) model - calculate daily values over designated time period
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verhulst.model(a, k, Y0, duration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verhulst.model_+3A_a">a</code></td>
<td>
<p>: growth rate</p>
</td></tr>
<tr><td><code id="verhulst.model_+3A_k">k</code></td>
<td>
<p>: capacity</p>
</td></tr>
<tr><td><code id="verhulst.model_+3A_y0">Y0</code></td>
<td>
<p>: initial condition</p>
</td></tr>
<tr><td><code id="verhulst.model_+3A_duration">duration</code></td>
<td>
<p>: duration of simulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily Y
</p>


<h3>See Also</h3>

<p><code><a href="#topic+verhulst.update">verhulst.update</a></code> for the update function of the Verhulst model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(verhulst.model(0.08,100,1,100), type="l", ylim=c(0,115),
   xlab="day", ylab="Y, population density",lwd=2)
</code></pre>

<hr>
<h2 id='verhulst.update'>The Verhulst (logistic) model - calculate change for one day</h2><span id='topic+verhulst.update'></span>

<h3>Description</h3>

<p>The Verhulst (logistic) model - calculate change for one day
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verhulst.update(Y, a, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verhulst.update_+3A_y">Y</code></td>
<td>
<p>: state variable Y(t=day)</p>
</td></tr>
<tr><td><code id="verhulst.update_+3A_a">a</code></td>
<td>
<p>: growth rate</p>
</td></tr>
<tr><td><code id="verhulst.update_+3A_k">k</code></td>
<td>
<p>: capacity</p>
</td></tr>
</table>


<h3>Value</h3>

<p>state variable at Y(t=day+1)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+verhulst.model">verhulst.model</a></code> for the integration loop function of the Verhulst model.
</p>

<hr>
<h2 id='watbal.define.param'>Define values of the parameters for the WaterBalance model</h2><span id='topic+watbal.define.param'></span>

<h3>Description</h3>

<p>Define values of the parameters for the WaterBalance model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>watbal.define.param()
</code></pre>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup)
</p>

<hr>
<h2 id='watbal.model'>WaterBalance model - calculate soil water over designated time period</h2><span id='topic+watbal.model'></span>

<h3>Description</h3>

<p>WaterBalance model - calculate soil water over designated time period
</p>


<h3>Usage</h3>

<pre><code class='language-R'>watbal.model(param, weather, WP, FC, WAT0 = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="watbal.model_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="watbal.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="watbal.model_+3A_wp">WP</code></td>
<td>
<p>: Water content at wilting Point (cm^3.cm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model_+3A_fc">FC</code></td>
<td>
<p>: Water content at field capacity (cm^3.cm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model_+3A_wat0">WAT0</code></td>
<td>
<p>: Initial Water content (mm). If NA WAT0=z*FC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily RAIN, ETR, Water at the beginning of the day (absolute : WAT, mm and relative value : WATp, -)
</p>

<hr>
<h2 id='watbal.model.arid'>WaterBalance model - Variant with another order of calculation and ARID index</h2><span id='topic+watbal.model.arid'></span>

<h3>Description</h3>

<p>WaterBalance model - Variant with another order of calculation and ARID index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>watbal.model.arid(WHC, MUF, DC, z, CN, weather, WP, FC, WAT0 = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="watbal.model.arid_+3A_whc">WHC</code></td>
<td>
<p>: Water Holding Capacity of the soil (cm^3 cm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_muf">MUF</code></td>
<td>
<p>: Water Uptake coefficient (mm^3 mm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_dc">DC</code></td>
<td>
<p>: Drainage coefficient (mm^3 mm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_z">z</code></td>
<td>
<p>: root zone depth (mm)</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_cn">CN</code></td>
<td>
<p>: Runoff curve number</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_wp">WP</code></td>
<td>
<p>: Water content at wilting Point (cm^3.cm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_fc">FC</code></td>
<td>
<p>: Water content at field capacity (cm^3.cm^-3)</p>
</td></tr>
<tr><td><code id="watbal.model.arid_+3A_wat0">WAT0</code></td>
<td>
<p>: Initial Water content (mm). If NA WAT0=z*FC</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily RAIN, ETR, Water at the beginning of the day (absolute : WAT, mm and relative value : WATp, -)
</p>

<hr>
<h2 id='watbal.simobsdata'>Soil water content measurements and associated simulations with WaterBalance model</h2><span id='topic+watbal.simobsdata'></span>

<h3>Description</h3>

<p>Data of soil water content from Luc Champolivier (CETIOM), in En Crambade (31, France), on canola without irrigation in 2008.
sonde Diviner 2000 (from Sentek Pty Ltd)
Simulation are from watbal.model, with an initial water content estimated from measurement with Diviner 2000.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>watbal.simobsdata
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per day :
Weather : day / RAIN / ETr /
simulation :  WAT / WATp / ARID
observation :t1_WATp_0_40cm / t2_WATp_0_40cm / t3_WATp_0_40cm / WATp_SF.mean / WATp_SF.var</p>


<h3>Source</h3>

<p>CETIOM, Luc Champolivier (2008).
</p>

<hr>
<h2 id='watbal.update'>WaterBalance model - calculate change in soil water for one day</h2><span id='topic+watbal.update'></span>

<h3>Description</h3>

<p>WaterBalance model - calculate change in soil water for one day
</p>


<h3>Usage</h3>

<pre><code class='language-R'>watbal.update(WAT0, RAIN, ETr, param, WP, FC)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="watbal.update_+3A_wat0">WAT0</code></td>
<td>
<p>: Water at the beginning of the day (mm).</p>
</td></tr>
<tr><td><code id="watbal.update_+3A_rain">RAIN</code></td>
<td>
<p>: Rainfall of day (mm)</p>
</td></tr>
<tr><td><code id="watbal.update_+3A_etr">ETr</code></td>
<td>
<p>: Evapotranspiration of day (mm)</p>
</td></tr>
<tr><td><code id="watbal.update_+3A_param">param</code></td>
<td>
<p>: a vector of parameters</p>
</td></tr>
<tr><td><code id="watbal.update_+3A_wp">WP</code></td>
<td>
<p>: Water content at wilting Point (cm^3.cm^-3)</p>
</td></tr>
<tr><td><code id="watbal.update_+3A_fc">FC</code></td>
<td>
<p>: Water content at field capacity (cm^3.cm^-3)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>WAT1 : Water at the beginning of the day+1 (mm).
</p>

<hr>
<h2 id='watbal.weather'>Read weather data for the WaterBalance model (West of France Weather)</h2><span id='topic+watbal.weather'></span>

<h3>Description</h3>

<p>Read weather data for the WaterBalance model (West of France Weather)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>watbal.weather(working.year = NA, working.site = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="watbal.weather_+3A_working.year">working.year</code></td>
<td>
<p>: year for the subset of weather data (default=NA : all the year)</p>
</td></tr>
<tr><td><code id="watbal.weather_+3A_working.site">working.site</code></td>
<td>
<p>: site for the subset of weather data (default=NA : all the site)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with daily weather data for one or several site(s) and for one or several year(s)
</p>

<hr>
<h2 id='weather_EuropeEU'>Weather serie for Europe EU from NASA POWER agroclimatology</h2><span id='topic+weather_EuropeEU'></span>

<h3>Description</h3>

<p>This contemporary daily climate dataset for Europe
covers the period 1st January 2001 to 31 December 2010
with 10 complete years of data.
It cover a part of Europe) with an elevation less than 500m.
The dataset was was extrated from the NASA Langley Research Center POWER Project which provide
agroclimatology dataset (Chandler et al., 2004). 
It was funded through the NASA Earth Science Directorate Applied Science Program
This climate datasetcontains daily estimates of precipitation, mean, minimum and
maximum temperature, relative humidity, dew point, solar radiation
and wind speed with global coverage at one degree resolution
(approximately 111 km at the equator).
The NASA POWER agroclimatology data are derived from various
sources: solar radiation from satellite observations, meteorological
data from the Goddard Earth Observing System global assimilation
model version 4 (GEOS-4), and precipitation from the Global
Precipitation Climate Project and Topical Rainfall Measurement
Mission. A full description can be found at 
<a href="https://power.larc.nasa.gov/common/php/POWER_AboutAgroclimatology.php">https://power.larc.nasa.gov/common/php/POWER_AboutAgroclimatology.php</a>
Elevation (Altitude) were retrive from Aster Global Digital Elevation Model
by using the Webservice api.geonames.org/astergdem?
Sample are: ca 30m x 30m, between 83N and 65S latitude.
Result : a single number giving the elevation in meters according to aster gdem, ocean areas have been masked as &quot;no data&quot; and have been assigned a value of -9999
Example http://api.geonames.org/astergdem?lat=50.01&amp;lng=10.2&amp;username=demo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weather_EuropeEU
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per day.
SRAD     daily Insolation Incident On A Horizontal Surface (MJ/m^2/day) 
T2M      Average Air Temperature At 2 m Above The Surface Of The Earth (degrees C) 
TMIN     Minimum Air Temperature At 2 m Above The Surface Of The Earth (degrees C) 
TMAX     Maximum Air Temperature At 2 m Above The Surface Of The Earth (degrees C) 
RH2M     Relative Humidity At 2 m (
TDEW     Dew/Frost Point Temperature At 2 m (degrees C) 
RAIN     Average Precipitation (mm/day) 
WIND     Wind Speed At 10 m Above The Surface Of The Earth (m/s)</p>


<h3>Source</h3>

<p><a href="http://power.larc.nasa.gov/">http://power.larc.nasa.gov/</a> and <a href="http://asterweb.jpl.nasa.gov/gdem.asp">http://asterweb.jpl.nasa.gov/gdem.asp</a> and <a href="http://www.geonames.org/about.html">http://www.geonames.org/about.html</a>
</p>

<hr>
<h2 id='weather_FranceWest'>Weather series for western France from NASA POWER agroclimatology</h2><span id='topic+weather_FranceWest'></span>

<h3>Description</h3>

<p>This contemporary daily climate dataset for West of France
covers the period 1st January 1984 to 31 December 2011.
The precipitation data is limited to the period Jan-1997 Aug-2009, thus only 12 complete
years of data were available for  analysis involving precipitation(1997 to 2009).
It cover main part of West part of France defined as a rectangle.
The dataset was extracted from the NASA Langley Research Center POWER Project which provide
agroclimatology dataset (Chandler et al., 2004). 
It was funded through the NASA Earth Science Directorate Applied Science Program
This climate dataset contains daily estimates of precipitation, mean, minimum and
maximum temperature, relative humidity, dew point, solar radiation
and wind speed with global coverage at one degree resolution
(approximately 111 km at the equator).
The NASA POWER agroclimatology data are derived from various
sources: solar radiation from satellite observations, meteorological
data from the Goddard Earth Observing System global assimilation
model version 4 (GEOS-4), and precipitation from the Global
Precipitation Climate Project and Topical Rainfall Measurement
Mission. A full description can be found at 
<a href="https://power.larc.nasa.gov/common/php/POWER_AboutAgroclimatology.php">https://power.larc.nasa.gov/common/php/POWER_AboutAgroclimatology.php</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weather_FranceWest
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per day.</p>


<h3>Source</h3>

<p><a href="http://power.larc.nasa.gov/">http://power.larc.nasa.gov/</a>
</p>

<hr>
<h2 id='weather_GNS'>Weather series for Gainesville (FL, USA) years 1982 and 1983</h2><span id='topic+weather_GNS'></span>

<h3>Description</h3>

<p>This  daily climate dataset for Gainesville (FL, USA) years 1982 and 1983
covers the period 1st January 1982 to 31 December 1983
with 2 complete years of data with precipitation.
The dataset was provide by JJW Jones and the university of Florida to run simulation of the
publication of Muchow et al. (1990). 
This climate dataset contains daily estimates of precipitation (RAIN), minimum (Tmin) and
maximum temperature (Tmax), solar radiation (I), 
photosynthically active radiation (PAR)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weather_GNS
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per day.</p>


<h3>Source</h3>

<p>University of Florida
</p>

<hr>
<h2 id='weather_SouthAsia'>Weather series for southern Asia from NASA POWER agroclimatology</h2><span id='topic+weather_SouthAsia'></span>

<h3>Description</h3>

<p>This contemporary daily climate dataset for South Asia
covers the period 1st January 1997 to 31 December 2008
with 12 complete years of data with precipitation.
It cover a part of South Asia (North-East of India, Bangladesh, Myanmar, Neapal) with an elevation less than 2500m.
The dataset was extracted from the NASA Langley Research Center POWER Project which provide
agroclimatology dataset (Chandler et al., 2004). 
It was funded through the NASA Earth Science Directorate Applied Science Program
This climate dataset contains daily estimates of precipitation, mean, minimum and
maximum temperature, relative humidity, dew point, solar radiation
and wind speed with global coverage at one degree resolution
(approximately 111 km at the equator).
The NASA POWER agroclimatology data are derived from various
sources: solar radiation from satellite observations, meteorological
data from the Goddard Earth Observing System global assimilation
model version 4 (GEOS-4), and precipitation from the Global
Precipitation Climate Project and Topical Rainfall Measurement
Mission. A full description can be found at 
<a href="https://power.larc.nasa.gov/common/php/POWER_AboutAgroclimatology.php">https://power.larc.nasa.gov/common/php/POWER_AboutAgroclimatology.php</a>
Elevation (Altitude) were retrive from Aster Global Digital Elevation Model
by using the Webservice api.geonames.org/astergdem?
Sample are: ca 30m x 30m, between 83N and 65S latitude.
Result : a single number giving the elevation in meters according to aster gdem, ocean areas have been masked as &quot;no data&quot; and have been assigned a value of -9999
Example http://api.geonames.org/astergdem?lat=50.01&amp;lng=10.2&amp;username=demo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weather_SouthAsia
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per day.</p>


<h3>Source</h3>

<p><a href="http://power.larc.nasa.gov/">http://power.larc.nasa.gov/</a> and <a href="http://asterweb.jpl.nasa.gov/gdem.asp">http://asterweb.jpl.nasa.gov/gdem.asp</a> and <a href="http://www.geonames.org/about.html">http://www.geonames.org/about.html</a>
</p>

<hr>
<h2 id='weed.define.param'>Define parameter values of the Weed model</h2><span id='topic+weed.define.param'></span>

<h3>Description</h3>

<p>Define parameter values of the Weed model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weed.define.param()
</code></pre>


<h3>Value</h3>

<p>matrix with parameter values (nominal, binf, bsup)
</p>

<hr>
<h2 id='weed.model'>The Weed model - calculate daily values over designated time period</h2><span id='topic+weed.model'></span>

<h3>Description</h3>

<p>The Weed model - calculate daily values over designated time period
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weed.model(param, weed.deci)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weed.model_+3A_param">param</code></td>
<td>
<p>: vector of the 16 parameters</p>
</td></tr>
<tr><td><code id="weed.model_+3A_weed.deci">weed.deci</code></td>
<td>
<p>: decisions table for Soil, Crop et Herbicide</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with annual values of yield
</p>

<hr>
<h2 id='weed.simule'>Wrapper function to run the Weed model multiple times (for multiple sets of inputs)</h2><span id='topic+weed.simule'></span>

<h3>Description</h3>

<p>Wrapper function to run the Weed model multiple times (for multiple sets of inputs)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weed.simule(X, weed.deci)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weed.simule_+3A_x">X</code></td>
<td>
<p>: parameter matrix</p>
</td></tr>
<tr><td><code id="weed.simule_+3A_weed.deci">weed.deci</code></td>
<td>
<p>: decisions table for Soil, Crop et Herbicide</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix with Yield for year 3 for each parameter vector
</p>

<hr>
<h2 id='weed.update'>The Weed model - calculate change for one year</h2><span id='topic+weed.update'></span>

<h3>Description</h3>

<p>The Weed model - calculate change for one year
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weed.update(d, S, SSBa, DSBa, Soil, Crop, Herb, param)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weed.update_+3A_d">d</code></td>
<td>
<p>: weed density at seed emergence (plants/m2) - value for year</p>
</td></tr>
<tr><td><code id="weed.update_+3A_s">S</code></td>
<td>
<p>: seed production per m2 - value for year</p>
</td></tr>
<tr><td><code id="weed.update_+3A_ssba">SSBa</code></td>
<td>
<p>: surface seedbank after tillage (grains/m2) - value for year</p>
</td></tr>
<tr><td><code id="weed.update_+3A_dsba">DSBa</code></td>
<td>
<p>: deep seedbank after tillage (grains/m2) - value for year</p>
</td></tr>
<tr><td><code id="weed.update_+3A_soil">Soil</code></td>
<td>
<p>: value for soil decision (1 or 0)</p>
</td></tr>
<tr><td><code id="weed.update_+3A_crop">Crop</code></td>
<td>
<p>: value for crop decision (1 or 0)</p>
</td></tr>
<tr><td><code id="weed.update_+3A_herb">Herb</code></td>
<td>
<p>: value for herbicide treatment decision (1 or 0)</p>
</td></tr>
<tr><td><code id="weed.update_+3A_param">param</code></td>
<td>
<p>: vector of the 16 parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with values of state variables for year+1
</p>

<hr>
<h2 id='Wheat_GPC'>Grain Protein Contents in Wheat Grains</h2><span id='topic+Wheat_GPC'></span>

<h3>Description</h3>

<p>This dataset contains data for 43 plots. The column GPC corresponds to
measured data, GPC.model1 and GPC.model2 correspond to the results obtained by two models.
The other columns contain other information for each plot (technical practices) :
number, tillage,	max_water,	preceding_crop,	sow_date,	Nendwinter,	Nfertilizer,	SPAD,	NNI,	Yield.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Wheat_GPC
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per plot.</p>


<h3>Source</h3>

<p>Barbottin et al. 2008
</p>

<hr>
<h2 id='WheatYieldGreece'>National Wheat Yield evolution for Greece from FAO</h2><span id='topic+WheatYieldGreece'></span>

<h3>Description</h3>

<p>Wheat yield time series data in Greece from 1961 to 2010 yield.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WheatYieldGreece
</code></pre>


<h3>Format</h3>

<p>a <code>RangedData</code> instance, 1 row per measurement. Year, Yield : Wheat Yield (hectogram/hectare = 0.0001 ton/hectare)</p>


<h3>Source</h3>

<p>Food And Agricultural Organization of United Nations (FAO), FAOSTAT database, http://faostat.fao.org
</p>

<hr>
<h2 id='zakoks.original.model'>Classical SEIR model for plant diseases from Zadoks (1971)</h2><span id='topic+zakoks.original.model'></span>

<h3>Description</h3>

<p><strong>Model description.</strong>
This model is a classical SEIR model for plant disease. It was written from it description included in the original publication of Zadoks (1971)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zakoks.original.model(nlpd = 4 * 10, nipd = 1 * 10, dmfr = 16,
  SITE0 = 5 * 10^9, weather, sdate = 145, ldate = 145 + 50,
  XLAT0 = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zakoks.original.model_+3A_nlpd">nlpd</code></td>
<td>
<p>: latent period (in degree.day) - default value for Puccinia triticina : ~10 days at 20degC</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_nipd">nipd</code></td>
<td>
<p>: infectious period (in degree.day) - default value for Puccinia triticina : ~20 days at 20degC</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_dmfr">dmfr</code></td>
<td>
<p>: daily multiplication factor - default value number of effective spores produced by lesion</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_site0">SITE0</code></td>
<td>
<p>:</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_weather">weather</code></td>
<td>
<p>: weather data.frame for one single year</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_sdate">sdate</code></td>
<td>
<p>: starting date</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_ldate">ldate</code></td>
<td>
<p>: ending date</p>
</td></tr>
<tr><td><code id="zakoks.original.model_+3A_xlat0">XLAT0</code></td>
<td>
<p>:</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This model is a classical SEIR model proposed by Zadoks (1971) to simulate epidemics of diseases of crops. It is a Susceptible-Exposed-Infectious-Removed (SEIR) model. This simple model of an epidemic is based on the epidemiological concepts &quot;latent period&quot;, &quot;infectious period&quot;, and &quot;multiplication factor&quot;. The crop is considered to consist of a large but finite number of infectious sites. The physical dimensions of an infectious site roughly coincide with the reproductive unit of the parasite studied. Different pathosystems (with different infectious site definitions) can be considered with this model. A full description, is available in the original paper:
The model has four essential state variables representing the number of sites in each state
XVAC for vacant (healthy) sites, XLAT for latent site, XINF for infectant sites and XCTR for the cumulative total of removal (post infectious) sites.
Two supplementary variables based on the state variables are used defined as XTO1 = XLAT+XINF+XCTR and XSEV = XINF+XCTR.
Fluxes or rates between the state variables are defined as rocc for occupation, rapp for apparition and rrem for removal.
The model has a time step of one day (dt=1). The system modeled is one hectare of a wheat crop.
</p>


<h3>Value</h3>

<p>list with a data.frame with daily day, DACE, XVAC, XLAT, XINF, XCTR ,XTO1, XSEV= XSEV, severity
and a vector of parameter value (nlpd, nipd, dmfr, SITE0).
</p>


<h3>Source</h3>

<p>Script written from equation described in Zadoks, J.C. 1971. Systems Analysis and the Dynamics of Epidemics. Phytopathology. 61:441-598.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+epirice.model">epirice.model</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>weather=subset(weather_FranceWest, WEYR==1997 &amp; idsite==39)
out=zakoks.original.model(nlpd=4*10,nipd=1*10,dmfr=16,SITE0 = 5*10^9,
weather, sdate = 145, ldate = 145+50 , XLAT0=1)
plot(out$sim$DACE,out$sim$severity, type="l")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
