<!DOCTYPE html><html><head><title>Help for package ZetaSuite</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ZetaSuite}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#countMat'>
<p>Subsampled data from in-house HTS2 screening for global splicing regulators.</p></a></li>
<li><a href='#countMatSC'>
<p>The cell x gene matrix from single-cell RNA-seq.</p></a></li>
<li><a href='#EventCoverage'><p>Generation of Zeta Plot.</p></a></li>
<li><a href='#FDRcutoff'><p>Find a cut-off according to screen strength.</p></a></li>
<li><a href='#negGene'>
<p>Input negative file.</p></a></li>
<li><a href='#nonExpGene'>
<p>Input internal negative control file.</p></a></li>
<li><a href='#posGene'>
<p>Input positive file.</p></a></li>
<li><a href='#QC'><p>Quality control of input datasets.</p></a></li>
<li><a href='#SVM'><p>Find a SVM curve to separate positive and negative controls.</p></a></li>
<li><a href='#SVMcurve'>
<p>The SVM curve lines in Zeta-plot.</p></a></li>
<li><a href='#Zeta'><p>Calculation of zeta and weighted zeta score.</p></a></li>
<li><a href='#ZetaSuitSC'><p>Calculation of zeta score for single cell RNA-seq.</p></a></li>
<li><a href='#Zscore'><p>Z-transformation for input matrix.</p></a></li>
<li><a href='#ZseqList'>
<p>The bin size for Zeta calculation.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analyze High-Dimensional High-Throughput Dataset and Quality
Control Single-Cell RNA-Seq</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-22</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junhui Li &lt;ljh.biostat@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The advent of genomic technologies has enabled the generation of two-dimensional or even multi-dimensional high-throughput data, e.g., monitoring multiple changes in gene expression in genome-wide siRNA screens across many different cell types (E Robert McDonald 3rd (2017) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2017.07.005">doi:10.1016/j.cell.2017.07.005</a>&gt; and Tsherniak A (2017) &lt;<a href="https://doi.org/10.1016%2Fj.cell.2017.06.010">doi:10.1016/j.cell.2017.06.010</a>&gt;) or single cell transcriptomics under different experimental conditions. We found that simple computational methods based on a single statistical criterion is no longer adequate for analyzing such multi-dimensional data. We herein introduce 'ZetaSuite', a statistical package initially designed to score hits from two-dimensional RNAi screens.We also illustrate a unique utility of 'ZetaSuite' in analyzing single cell transcriptomics to differentiate rare cells from damaged ones (Vento-Tormo R (2018) &lt;<a href="https://doi.org/10.1038%2Fs41586-018-0698-6">doi:10.1038/s41586-018-0698-6</a>&gt;). In 'ZetaSuite', we  have the following steps: QC of input datasets, normalization using Z-transformation, Zeta score calculation and hits selection based on defined Screen Strength.</td>
</tr>
<tr>
<td>Imports:</td>
<td>RColorBrewer, Rtsne, dplyr, e1071, ggplot2, reshape2,
gridExtra, mixtools</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Author:</td>
<td>Yajing Hao <a href="https://orcid.org/0000-0003-1384-4176"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Shuyang Zhang <a href="https://orcid.org/0000-0002-8428-1828"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Junhui Li [cre],
  Guofeng Zhao [ctb],
  Xiang-Dong Fu <a href="https://orcid.org/0000-0001-5499-8732"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cph, fnd]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-23 01:59:24 UTC; junhuili</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-24 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='countMat'>
Subsampled data from in-house HTS2 screening for global splicing regulators.
</h2><span id='topic+countMat'></span>

<h3>Description</h3>

<p>A data frame with 1609 individual screened genes and 100 functional readouts. The data was generated from a siRNA screen for global splicing regulators. In this screen, we interrogated ~400 endogenous alternative splicing (AS) events by using an oligo ligation-based strategy to quantify 18,480 pools of siRNAs against annotated protein-coding genes in the human genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("countMat")</code></pre>


<h3>Format</h3>

<p>A data frame with 1609 observations on the following 100 variables
</p>

<p>A data frame with 1609 observations on the following 100 maker variables.Each row represents gene with specific knocking-down siRNA pool, each column is an AS event. The values in the matrix are the processed foldchange values between included exons and skipping exons read counts. 

</p>


<h3>Details</h3>

<p>This data frame is the raw output data from large-scale screening.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(countMat)
</code></pre>

<hr>
<h2 id='countMatSC'>
The cell x gene matrix from single-cell RNA-seq. 
</h2><span id='topic+countMatSC'></span>

<h3>Description</h3>

<p>A scRNA-seq dataset generated from placenta that has been analyzed with CellRanger and used to develop EmptyDrops. We have subsampled the genes from the real datasets to generated the matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("countMatSC")</code></pre>


<h3>Format</h3>

<p>A data frame with 1090 cells and 10000 genes. This is the subset of data obtained from single-cell RNAseq for package testing.  Each row represents one cell detected in single-cell RNA-seq, each column is one gene in detected cells. The values in the matrix are the raw read counts from single-cell RNAseq. 
</p>

<p>A data frame with 1090 cells and 10000 genes.This is the subset of data obtained from single-cell RNAseq for package testing.

</p>


<h3>Details</h3>

<p>This data frame is the generated by single-cell RNA-seq.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(countMatSC)
</code></pre>

<hr>
<h2 id='EventCoverage'>Generation of Zeta Plot.</h2><span id='topic+EventCoverage'></span>

<h3>Description</h3>

<p>A zeta plot is generated from the input Z-score matrix. Zeta plot labels: x-axis: Z-score cutoffs, y-axis: the percentage of readouts that survived at a given Z-score cutoff over the total scored readouts. In order to generate this plot, the range of Z-scores is determined by ranking the absolute value of Zij (Z-score value in row i and column j) from the smallest to the largest. Z-cutoffs next are selected in the range of (-|Znxmx0.9999|, -2) to (2, |Znxmx0.9999|) to excluded the insignificant changes that may result from experimental noise( |Z| &lt; 2, which equals to p-value &gt;0.05). Then, for all Zij within the selected range (both positive range and negative range), the range is divided equally into x bins (the recommended input of x is 100). Thus, the percentage of readouts scored above the Z-cutoff in each bin is determined.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EventCoverage(ZscoreVal, negGene, posGene, binNum, combine = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EventCoverage_+3A_zscoreval">ZscoreVal</code></td>
<td>
<p>zscore value</p>
</td></tr>
<tr><td><code id="EventCoverage_+3A_neggene">negGene</code></td>
<td>
<p>negative control dataset, the siRNAs/genes used as negative controls in screening.</p>
</td></tr>
<tr><td><code id="EventCoverage_+3A_posgene">posGene</code></td>
<td>
<p>positive control dataset, the siRNAs/genes used as positive controls in screening.</p>
</td></tr>
<tr><td><code id="EventCoverage_+3A_binnum">binNum</code></td>
<td>
<p>bin number</p>
</td></tr>
<tr><td><code id="EventCoverage_+3A_combine">combine</code></td>
<td>
<p>combine two direction zeta together(TRUE or FALSE),default FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data.frames and plots, the data.frame includes 'ZseqList', 'EC_N_I', 'EC_N_D', 'EC_P_I' and 'EC_P_D'. The plot 'EC_jitter_D' and 'EC_jitter_I' are the zeta plot for positive and negative samples.'ZseqList', 'EC_N_I', 'EC_N_D', 'EC_P_I' and 'EC_P_D' are the inputfiles for zeta plot and SVM.R. ZseqList describs the bin size in the zeta plot.
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(countMat)
data(negGene)
data(posGene)
ZscoreVal &lt;- Zscore(countMat,negGene)
ECList &lt;- EventCoverage(ZscoreVal,negGene,posGene,binNum=100,combine=TRUE)

</code></pre>

<hr>
<h2 id='FDRcutoff'>Find a cut-off according to screen strength.</h2><span id='topic+FDRcutoff'></span>

<h3>Description</h3>

<p>Find a cutoff according to the Screen Strength (SS) and graph the Screen Strength plot.
Zeta score is used to rank genes, and then, SS is calculated to define a suitable cutoff so that the cutoff can define hits at different confidence intervals.
Formula of SS: SS = 1 - aFDR/bFDR, where aFDR (apparent FDR)  = number of non-expressors identified at hits divided by the total number of hits, bFDR (baseline FDR) = total number of non-expressors divided by all screened genes.
SS plot labels: x-axis: zeta score, y-axis: Screen Strength,
SS value is determined at each bin (m bin in total), then connect individual SS value to generate a simulated SS curve based on balance points. Users may choose one or multiple balance point as the different SS intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FDRcutoff(zetaData, negGene, posGene, nonExpGene, combine = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FDRcutoff_+3A_zetadata">zetaData</code></td>
<td>
<p>ZetaScore file calculated by ZetaSuite.</p>
</td></tr>
<tr><td><code id="FDRcutoff_+3A_neggene">negGene</code></td>
<td>
<p>negative control dataset, the siRNAs/genes used as negative controls in screening.</p>
</td></tr>
<tr><td><code id="FDRcutoff_+3A_posgene">posGene</code></td>
<td>
<p>positive control dataset, the siRNAs/genes used as positive controls in screening.</p>
</td></tr>
<tr><td><code id="FDRcutoff_+3A_nonexpgene">nonExpGene</code></td>
<td>
<p>non-expressed gene</p>
</td></tr>
<tr><td><code id="FDRcutoff_+3A_combine">combine</code></td>
<td>
<p>combine two direction zeta together(TRUE or FALSE),default FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data.frame and plots, the data.frame is cut off matrix with 6 columns including &quot;Cut_Off&quot;,&quot;aFDR&quot;, &quot;SS&quot;,&quot;TotalHits&quot;,&quot;Num_nonExp&quot; and &quot;Type&quot;. Plots includes 'Zeta_type' and 'SS_cutOff'.
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nonExpGene)
data(negGene)
data(posGene)
data(ZseqList)
data(countMat)
ZscoreVal &lt;- Zscore(countMat,negGene)
zetaData &lt;- Zeta(ZscoreVal,ZseqList,SVM=FALSE)
cutoffval &lt;- FDRcutoff(zetaData,negGene,posGene,nonExpGene,combine=TRUE)

</code></pre>

<hr>
<h2 id='negGene'>
Input negative file. 
</h2><span id='topic+negGene'></span>

<h3>Description</h3>

<p>A data frame with 510 different well IDs in which the cells treated with non-specific siRNAs.If users did not have the build-in negative controls, the non-expressed genes should be provided here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("negGene")</code></pre>


<h3>Format</h3>

<p>A data frame with 510 different well IDs in which the cells treated with non-specific siRNAs.
</p>

<p>A data frame with 510 different well IDs in which the cells treated with non-specific siRNAs. These wells were served as negative control.

</p>


<h3>Details</h3>

<p>These wells were designed by the authors in the large-scale screen.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(negGene)
</code></pre>

<hr>
<h2 id='nonExpGene'>
Input internal negative control file. 
</h2><span id='topic+nonExpGene'></span>

<h3>Description</h3>

<p>A data frame with 722 different well IDs in which the cells treated with siRNAs targeting to non-expressed genes in HeLa cells.It the subset of total non-expressed 
genes in HeLa cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("nonExpGene")</code></pre>


<h3>Format</h3>

<p>A data frame with 722 different well IDs in which the cells treated with siRNAs targeting to non-expressed genes in HeLa cells.
</p>

<p>A data frame with 722 different well IDs in which the cells treated with siRNAs targeting to non-expressed genes in HeLa cells. These wells were served as internal negative controls.

</p>


<h3>Details</h3>

<p>These non-expressed genes can be obtained from a prior expression profile.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(nonExpGene)
</code></pre>

<hr>
<h2 id='posGene'>
Input positive file. 
</h2><span id='topic+posGene'></span>

<h3>Description</h3>

<p>A data frame with 299 different well IDs in which the cells treated with siRNAs targeting to PTB.If users didn't have the build-in positive controls, choose the parameters -withoutsvm and the filename can use any name such as 'NA'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("negGene")</code></pre>


<h3>Format</h3>

<p>A data frame with 299 different well IDs in which the cells treated with siRNAs targeting to PTB.
</p>

<p>A data frame with 299 different well IDs in which the cells treated with siRNAs targeting to PTB. These wells were served as positive control.

</p>


<h3>Details</h3>

<p>These wells were designed by the authors in the large-scale screen.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(posGene)
</code></pre>

<hr>
<h2 id='QC'>Quality control of input datasets.</h2><span id='topic+QC'></span>

<h3>Description</h3>

<p>Quality Control (QC) is a step in evaluating the experiment design. For all two-dimension high throughput data, the t-SNE plot is firstly used to evaluate whether features are sufficient to separate positive and negative controls.
The SSMD score (See reference Zhang) is further generated for each readout to evaluate the percentage of high-quality readouts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC(countMat, negGene, posGene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_+3A_countmat">countMat</code></td>
<td>
<p>input data set. The siRNA/gene x readouts matrix from HTS2 or large-scale RNAi screens</p>
</td></tr>
<tr><td><code id="QC_+3A_neggene">negGene</code></td>
<td>
<p>negative control data set, the siRNAs/genes used as negative controls in screening.</p>
</td></tr>
<tr><td><code id="QC_+3A_posgene">posGene</code></td>
<td>
<p>positive control data set, the siRNAs/genes used as positive controls in screening.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of plots, and their names are 'score_q', 'tSNE_QC', 'QC_box' and 'QC_SSMD'. 'tSNE_QC' is the global evaluation based on all the readouts. This figure can evaluate whether the positive and negative samples are well separated based on current all readouts. And the other 3 plots are the quality evaluation of the individual readouts.
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>References</h3>

<p>Laurens van der Maaten GH: Visualizing Data using t-SNE. JournalofMachineLearningResearch 2008,9(2008):2579-2605.
</p>
<p>Zhang XD: A pair of new statistical parameters for quality control in RNA interference high-throughput screening assays. Genomics 2007, 89:552-561.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(countMat)
data(negGene)
data(posGene)
QC(countMat,negGene,posGene)

</code></pre>

<hr>
<h2 id='SVM'>Find a SVM curve to separate positive and negative controls.</h2><span id='topic+SVM'></span>

<h3>Description</h3>

<p>Radical kernel SVM is constructed to maximally separate positive controls from negative controls in the prior defined Z range using e1071 packages of R, and therefore, the SVM curve is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SVM(ECdataList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SVM_+3A_ecdatalist">ECdataList</code></td>
<td>
<p>data list of output EventCoverage, names of list shoule be 'EC_N_D', 'EC_P_D', 'EC_N_I', 'EC_P_I' and 'ZseqList'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data.frame, including 'cutOffD' and 'cutOffI'.cutOffD and cutOffI are the deduced SVM.
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(countMat)
data(negGene)
data(posGene)
ZscoreVal &lt;- Zscore(countMat,negGene)
ECdataList &lt;- EventCoverage(ZscoreVal,negGene,posGene,binNum=10,combine=TRUE)
SVM(ECdataList)

</code></pre>

<hr>
<h2 id='SVMcurve'>
The SVM curve lines in Zeta-plot. 
</h2><span id='topic+SVMcurve'></span>

<h3>Description</h3>

<p>The SVM curves were calculated from raw input matrix files. They were designed to maximally seperate the positive and negative genes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SVMcurve")</code></pre>


<h3>Format</h3>

<p>A data frame with 24 rows and 4 features.
</p>

<p>A data frame with 24 rows and 4 features.The first column is the bins cut-offs for decresed direction. The second column is the values of percentage with different cut-offs in column 1.
The third column is the bins cut-offs for increased direction. The fourth column is the values of percentage with different cut-offs in column 3.

</p>


<h3>Details</h3>

<p>This data frame is the generated by SVM.R.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(SVMcurve)
</code></pre>

<hr>
<h2 id='Zeta'>Calculation of zeta and weighted zeta score.</h2><span id='topic+Zeta'></span>

<h3>Description</h3>

<p>This step calculates the Zeta Score based on the two curvecs. Firstly, this step provides another curve above the SVM curve to set a value to represent the regulatory function of gene i. Then, the area between the two curves (the one mentioned above and the SVM curve) is calculated as the Zeta score for this gene. Since the graph of the curves is divided into m bins, then the Zeta score can be calculated as the sum of all the bins' areas that exist between the two curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zeta(ZscoreVal, ZseqList, SVMcurve = NULL, SVM = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zeta_+3A_zscoreval">ZscoreVal</code></td>
<td>
<p>input file name.</p>
</td></tr>
<tr><td><code id="Zeta_+3A_zseqlist">ZseqList</code></td>
<td>
<p>the list of bins.</p>
</td></tr>
<tr><td><code id="Zeta_+3A_svmcurve">SVMcurve</code></td>
<td>
<p>SVM curves for decrease and increase direction.###not always use</p>
</td></tr>
<tr><td><code id="Zeta_+3A_svm">SVM</code></td>
<td>
<p>do SVM or not, default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame where zeta values for all tested knockding-down genes including positive and negative controls. The first column is the direction which knockding-down gene will lead to exon inclusion, whereas the second column is the knock-down genes will lead to exon skipping.
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ZseqList)
data(SVMcurve)
data(countMat)
data(negGene)
ZscoreVal &lt;- Zscore(countMat,negGene)
zetaData &lt;- Zeta(ZscoreVal,ZseqList,SVM=FALSE)

</code></pre>

<hr>
<h2 id='ZetaSuitSC'>Calculation of zeta score for single cell RNA-seq.</h2><span id='topic+ZetaSuitSC'></span>

<h3>Description</h3>

<p>This tool is used to evalucate the quality of cells detected in the single-cell RNA-seq. A zeta score will be assigned to each cell. And a cut-off for low quality and broken cells will be provided. The users can based on the selected cut-off to select the high quality cells for further analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZetaSuitSC(countMatSC, binNum = 10, filter = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZetaSuitSC_+3A_countmatsc">countMatSC</code></td>
<td>
<p>Shalek input matrix</p>
</td></tr>
<tr><td><code id="ZetaSuitSC_+3A_binnum">binNum</code></td>
<td>
<p>bin number for ZetaScore calculation.</p>
</td></tr>
<tr><td><code id="ZetaSuitSC_+3A_filter">filter</code></td>
<td>
<p>Whether to filter the extreme low read counts cells with nCount &lt;100. default is TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data.frame and plots. The data.frame is the Cell matrix with column name 'Cell' and 'Zeta'. The plot is the distribution of Zeta score for the detected cells and including a cut-off for removing the broken and empty cells.
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(countMatSC)
zetaDataSC &lt;- ZetaSuitSC(countMatSC,binNum=50,filter=TRUE)

</code></pre>

<hr>
<h2 id='Zscore'>Z-transformation for input matrix.</h2><span id='topic+Zscore'></span>

<h3>Description</h3>

<p>In this step, the input matrix is transformed to Z-score matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zscore(countMat, negGene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Zscore_+3A_countmat">countMat</code></td>
<td>
<p>input data set. The siRNA/gene x readouts matrix from HTS2 or large-scale RNAi screens.</p>
</td></tr>
<tr><td><code id="Zscore_+3A_neggene">negGene</code></td>
<td>
<p>negative control dataset, the siRNAs/genes used as negative controls in screening. Z-transfromation according to thses negative control siRNAs/genes for each readout.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The initial input matrix is arranged in N x M dimension, where each row contains individual functional readouts against a siRNA pool and each column corresponds to individually siRNA pools tested on a given functional readout. Readouts in each column may be thus considered as the data from one-dimensional screen (many-to-one), and thus, the typical Z statistic can be used to evaluate the relative function of individual genes in such column. The conversion is repeated on all columns, thereby converting the raw activity matrix into a matrix. Suppose Nij are the values in the original matrix i (1 &lt;= i &lt;= N siRNA pool) row and j ( 1 &lt;= j &lt;= M readout) column, then Zij = (Nij - uj) / sigma(j), where uj and sigma(j) are the mean and standard deviation of negative control samples in column j.
</p>


<h3>Value</h3>

<p>A Z-transformated matrix, where each row represents each knocking-down condition and each column is a specific readout (AS event). The values in the matrix are the normalized values(Z-scores).
</p>


<h3>Author(s)</h3>

<p>Yajing Hao, Shuyang Zhang, Junhui Li, Guofeng Zhao, Xiang-Dong Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(countMat)
data(negGene)
ZscoreVal &lt;- Zscore(countMat,negGene)
ZscoreVal[1:5,1:5]

</code></pre>

<hr>
<h2 id='ZseqList'>
The bin size for Zeta calculation. 
</h2><span id='topic+ZseqList'></span>

<h3>Description</h3>

<p>A data frame with 11 different cut-offs and 2 directions. We divided the ranges of input values into bins. The number of bins is determined by the users.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("ZseqList")</code></pre>


<h3>Format</h3>

<p>A data frame with 11 different cut-offs and 2 directions.
</p>

<p>A data frame with 11 different cut-offs and 2 directions.We divided the ranges of input values into bins. The number of bins is determined by the users.

</p>


<h3>Details</h3>

<p>This data frame is the generated by EventCoverage.R.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(ZseqList)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
