<!DOCTYPE html><html lang="en"><head><title>Help for package ZIM4rv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ZIM4rv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cauchyp'><p>cauchyp</p></a></li>
<li><a href='#Ex1_genedata'><p>An example dataset of genedata</p></a></li>
<li><a href='#Ex1_phenodata'><p>An example dataset of phenodata</p></a></li>
<li><a href='#Ex2_covar'><p>An example dataset of covariate file</p></a></li>
<li><a href='#Ex2_dosage'><p>An example dataset of dosage file</p></a></li>
<li><a href='#Ex2_fam'><p>An example dataset of .fam file</p></a></li>
<li><a href='#Ex2_pheno'><p>An example dataset of pheno file</p></a></li>
<li><a href='#Ex2_region'><p>An example dataset of genetic region file</p></a></li>
<li><a href='#p_burden_single'><p>p_burden_single</p></a></li>
<li><a href='#p_kernel_single'><p>p_kernel_single</p></a></li>
<li><a href='#phi_lambda_hat'><p>phi_lambda_hat</p></a></li>
<li><a href='#phi_mu_hat4zinb'><p>phi_mu_hat4zinb</p></a></li>
<li><a href='#preprocess_genedata'><p>preprocess_genedata</p></a></li>
<li><a href='#preprocess_phenodata'><p>preprocess_phenodata</p></a></li>
<li><a href='#U_fi_lmd'><p>U_fi_lmd</p></a></li>
<li><a href='#U_phi_mu4zinb'><p>U_phi_mu4zinb</p></a></li>
<li><a href='#vuong_test'><p>vuong_test</p></a></li>
<li><a href='#zimfrv'><p>zimfrv</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene‐based Association Tests of Zero‐inflated Count Phenotype
for Rare Variants</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xiaomin Liu &lt;e0717571@u.nus.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene‐based association tests to model count data with excessive zeros and rare variants using zero-inflated Poisson/zero-inflated negative Binomial regression framework. This method was originally described by Fan, Sun, and Li in Genetic Epidemiology 46(1):73-86 &lt;<a href="https://doi.org/10.1002%2Fgepi.22438">doi:10.1002/gepi.22438</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/fanx0037/ZIM4rv">https://github.com/fanx0037/ZIM4rv</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>data.table, pscl, CompQuadForm, stats, SKAT, RNOmni, R (&ge;
3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fanx0037/ZIM4rv/issues">https://github.com/fanx0037/ZIM4rv/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-18 13:34:34 UTC; liuxiaomin</td>
</tr>
<tr>
<td>Author:</td>
<td>Xiaomin Liu [aut, cre, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-19 16:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='cauchyp'>cauchyp</h2><span id='topic+cauchyp'></span>

<h3>Description</h3>

<p>Cauchy combination test (Cauchy‐p)
</p>
<p>This function combines p-values using Cauchy combination test
for testing the joint genetic effect.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cauchyp(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cauchyp_+3A_x">x</code></td>
<td>
<p>a numeric vector containing p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a combined p-value indicating the joint effect
</p>

<hr>
<h2 id='Ex1_genedata'>An example dataset of genedata</h2><span id='topic+Ex1_genedata'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of genotypes for 200 individuals
on 3 rs locations to illustrate the analysis with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex1_genedata)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>
</p>

<dl>
<dt>FID</dt><dd><p>Family IDs</p>
</dd>
<dt>IID</dt><dd><p>Individual IDs</p>
</dd>
<dt>rs1</dt><dd><p>Genotype code for rs1</p>
</dd>
<dt>rs2</dt><dd><p>Genotype code for rs2</p>
</dd>
<dt>rs3</dt><dd><p>Genotype code for rs3</p>
</dd>
</dl>



<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex1_genedata)
head(Ex1_genedata)

</code></pre>

<hr>
<h2 id='Ex1_phenodata'>An example dataset of phenodata</h2><span id='topic+Ex1_phenodata'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of count phenotypes and covariates for 200 individuals
to illustrate the analysis with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex1_phenodata)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>
</p>

<dl>
<dt>FID</dt><dd><p>Family IDs</p>
</dd>
<dt>IID</dt><dd><p>Individual IDs</p>
</dd>
<dt>count</dt><dd><p>Zero-inflated count phenotypes</p>
</dd>
<dt>educ</dt><dd><p>Covariate education years</p>
</dd>
<dt>sex</dt><dd><p>Covariate sex</p>
</dd>
<dt>PC1</dt><dd><p>The first principal component</p>
</dd>
<dt>PC2</dt><dd><p>The second principal component</p>
</dd>
<dt>PC3</dt><dd><p>The third principal component</p>
</dd>
</dl>



<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex1_phenodata)
head(Ex1_phenodata)

</code></pre>

<hr>
<h2 id='Ex2_covar'>An example dataset of covariate file</h2><span id='topic+Ex2_covar'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of covariates for 15 individuals
to illustrate the pre-processing with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex2_covar)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code> listing IDs and covariates separately
</p>


<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex2_covar)
head(Ex2_covar)

</code></pre>

<hr>
<h2 id='Ex2_dosage'>An example dataset of dosage file</h2><span id='topic+Ex2_dosage'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of dosage for 15 individuals
to illustrate the pre-processing with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex2_dosage)
</code></pre>


<h3>Format</h3>

<p>An object of .dosage file
</p>


<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex2_dosage)
head(Ex2_dosage)

</code></pre>

<hr>
<h2 id='Ex2_fam'>An example dataset of .fam file</h2><span id='topic+Ex2_fam'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of .fam file for 15 individuals
to illustrate the pre-processing with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex2_fam)
</code></pre>


<h3>Format</h3>

<p>An object of standard .fam file
</p>


<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex2_fam)
head(Ex2_fam)

</code></pre>

<hr>
<h2 id='Ex2_pheno'>An example dataset of pheno file</h2><span id='topic+Ex2_pheno'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of phenotypes for 15 individuals
to illustrate the pre-processing with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex2_pheno)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code> listing IDs and phenotypes separately
</p>


<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex2_pheno)
head(Ex2_pheno)

</code></pre>

<hr>
<h2 id='Ex2_region'>An example dataset of genetic region file</h2><span id='topic+Ex2_region'></span>

<h3>Description</h3>

<p>Small, artificially generated toy data set that
provides artificial information of 3 genetic regions
to illustrate the pre-processing with the use of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Ex2_region)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code> listing genetic regions
where each row contains chromosome, basepairs and the name of genetic region respectively
</p>


<h3>References</h3>

<p>This data set was artificially created and modified for the ZIM4rv package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Ex2_region)
head(Ex2_region)

</code></pre>

<hr>
<h2 id='p_burden_single'>p_burden_single</h2><span id='topic+p_burden_single'></span>

<h3>Description</h3>

<p>Compute the p-value for the burden test
</p>
<p>This function takes a vector of weights, a data frame of rare variants
and a matrix of Score statistics produced by <code>U_fi_lmd</code> for ZIP model
or <code>U_phi_mu4zinb</code> for ZINB model to compute the p-value for the burden test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_burden_single(wt, G_rare, s)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_burden_single_+3A_wt">wt</code></td>
<td>
<p>a numeric vector containing weights for all variants</p>
</td></tr>
<tr><td><code id="p_burden_single_+3A_g_rare">G_rare</code></td>
<td>
<p>a data frame containing data of rare variants</p>
</td></tr>
<tr><td><code id="p_burden_single_+3A_s">s</code></td>
<td>
<p>a matrix of the score statistics for each variant from each subject</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the p-value for the burden test
</p>

<hr>
<h2 id='p_kernel_single'>p_kernel_single</h2><span id='topic+p_kernel_single'></span>

<h3>Description</h3>

<p>Compute the p-value for the kernel test
</p>
<p>This function takes a diagonal matrix of weights, a data frame of rare variants
and a matrix of Score statistics produced by <code>U_fi_lmd</code> for ZIP model
or <code>U_phi_mu4zinb</code> for ZINB model to compute the p-value for the kernel test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p_kernel_single(wt_matrix2, G_rare, s)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="p_kernel_single_+3A_wt_matrix2">wt_matrix2</code></td>
<td>
<p>a diagonal matrix containing the squared weights for all variants</p>
</td></tr>
<tr><td><code id="p_kernel_single_+3A_g_rare">G_rare</code></td>
<td>
<p>a data frame containing data of rare variants</p>
</td></tr>
<tr><td><code id="p_kernel_single_+3A_s">s</code></td>
<td>
<p>a matrix of the score statistics for each variant from each subject</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the p-value for the kernel test (ZIP-k)
</p>

<hr>
<h2 id='phi_lambda_hat'>phi_lambda_hat</h2><span id='topic+phi_lambda_hat'></span>

<h3>Description</h3>

<p>Estimation of phi_hat and lambda_hat for ZIP model
</p>
<p>This function gives the estimation of 2 parameters phi and lambda for each subject
under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phi_lambda_hat(simud)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phi_lambda_hat_+3A_simud">simud</code></td>
<td>
<p>a data frame containing a phenotype named y and covariates</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function first fits zero‐inflated Poisson regression of phenotype y
on the covariates only to obtain the estimates of regression coefficients
and then compute the estimations of phi and lambda.
</p>


<h3>Value</h3>

<p>a list of 2 estimations of parameters for each subject
</p>


<h3>See Also</h3>

<p><code>zeroinfl</code>
</p>

<hr>
<h2 id='phi_mu_hat4zinb'>phi_mu_hat4zinb</h2><span id='topic+phi_mu_hat4zinb'></span>

<h3>Description</h3>

<p>Estimation of phi_hat, mu_hat and alpha_hat for ZINB model
</p>
<p>This function gives the estimation of three parameters phi, mu and alpha in ZINB model
for each subject under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phi_mu_hat4zinb(simud)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phi_mu_hat4zinb_+3A_simud">simud</code></td>
<td>
<p>a data frame containing a phenotype named y and covariates</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function first fits zero‐inflated negative binomial regression of phenotype y
on the covariates only to obtain the estimates of regression coefficients
and inverse dispersion
and then compute the estimations of phi, mu and alpha.
</p>


<h3>Value</h3>

<p>a list of 3 estimations of parameters for each subject
</p>


<h3>See Also</h3>

<p><code>zeroinfl</code>
</p>

<hr>
<h2 id='preprocess_genedata'>preprocess_genedata</h2><span id='topic+preprocess_genedata'></span>

<h3>Description</h3>

<p>Preprocess genotype files in PLINK format
</p>
<p>This function converts PLINK format files into data frames containing genotypes information in proper format
for the model fitting and testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocess_genedata(fam_file, dosage_file, region_file, gene_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="preprocess_genedata_+3A_fam_file">fam_file</code></td>
<td>
<p>.fam file in PLINK format</p>
</td></tr>
<tr><td><code id="preprocess_genedata_+3A_dosage_file">dosage_file</code></td>
<td>
<p>a dosage file includes dosage information of each variant for all individuals</p>
</td></tr>
<tr><td><code id="preprocess_genedata_+3A_region_file">region_file</code></td>
<td>
<p>a file listing genetic regions where each row contains chromosome, basepairs and the name of genetic region respectively</p>
</td></tr>
<tr><td><code id="preprocess_genedata_+3A_gene_name">gene_name</code></td>
<td>
<p>a character string of the name of a gene, e.g.&quot;CEPT&quot;. The name is case-sensitive.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing genotypes for all individuals in the required format for model fitting and testing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Ex2_fam)
data(Ex2_dosage)
data(Ex2_region)
preprocess_genedata(Ex2_fam,Ex2_dosage,Ex2_region,"r2")

</code></pre>

<hr>
<h2 id='preprocess_phenodata'>preprocess_phenodata</h2><span id='topic+preprocess_phenodata'></span>

<h3>Description</h3>

<p>Preprocess phenotype files in PLINK format
</p>
<p>This function converts PLINK format files into data frames containing phenotypes and covariates information in proper format
for the model fitting and testing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocess_phenodata(pheno_file, cov_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="preprocess_phenodata_+3A_pheno_file">pheno_file</code></td>
<td>
<p>phenotype file in PLINK format</p>
</td></tr>
<tr><td><code id="preprocess_phenodata_+3A_cov_file">cov_file</code></td>
<td>
<p>covariate file in PLINK format</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing phenotypes and covariates respectively for all individuals in the required format for model fitting and testing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Ex2_pheno)
data(Ex2_covar)
preprocess_phenodata(Ex2_pheno,Ex2_covar)

</code></pre>

<hr>
<h2 id='U_fi_lmd'>U_fi_lmd</h2><span id='topic+U_fi_lmd'></span>

<h3>Description</h3>

<p>Compute Score statistics for ZIP model
</p>
<p>This function takes the estimations of phi and lambda produced by the <code>phi_lambda_hat</code>
and computes the score statistics under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>U_fi_lmd(simudata, G_rare)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="U_fi_lmd_+3A_simudata">simudata</code></td>
<td>
<p>a data frame containing a phenotype named y and covariates</p>
</td></tr>
<tr><td><code id="U_fi_lmd_+3A_g_rare">G_rare</code></td>
<td>
<p>a data frame containing data of rare variants with the same subject order as in simudata</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of 2 matrice of the score statistics for each variant from each subject
</p>

<hr>
<h2 id='U_phi_mu4zinb'>U_phi_mu4zinb</h2><span id='topic+U_phi_mu4zinb'></span>

<h3>Description</h3>

<p>Compute score statistics for ZINB model
</p>
<p>This function takes the estimations of phi and lambda produced by the <code>phi_lambda_hat4negbin</code>
and computes the score statistics for ZINB model under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>U_phi_mu4zinb(simudata, G_rare)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="U_phi_mu4zinb_+3A_simudata">simudata</code></td>
<td>
<p>a data frame containing a phenotype named y and covariates</p>
</td></tr>
<tr><td><code id="U_phi_mu4zinb_+3A_g_rare">G_rare</code></td>
<td>
<p>a data frame containing data of rare variants with the same subject order as in simudata</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of 2 matrice of the score statistics for each variant from each subject
</p>

<hr>
<h2 id='vuong_test'>vuong_test</h2><span id='topic+vuong_test'></span>

<h3>Description</h3>

<p>Vuong's test
</p>
<p>This function performs Vuong's test, a likelihood ratio test for model selection and
non-nested hypotheses.
This function is for model selection between zero-inflated Poisson model and zero-inflated negative binomial model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vuong_test(phenodata)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vuong_test_+3A_phenodata">phenodata</code></td>
<td>
<p>a data frame containing family and individual IDs for all objects as well as
zero-inflated counts as a phenotype and a set of covariates.
Each row represents a different individual.
The first two columns are Family ID (FID) and Individual ID (IID) respectively.
There must be one and only one phenotype in the third column and
the phenotype have to be zero-inflated count data which should be non-negative integers, e.g. neuritic plaque counts.
Each of the rest of columns represents a different covariate, e.g. age, sex, etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing returned, prints a table of 3 test statistics and p values, and exits silently.
</p>

<hr>
<h2 id='zimfrv'>zimfrv</h2><span id='topic+zimfrv'></span>

<h3>Description</h3>

<p>Gene‐based association tests to model zero-inflated count data
</p>
<p>This function performs gene‐based association tests
between a set of SNPs/genes and zero-inflated count data
using ZIP regression or ZINB regression or two-stage SKAT model framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zimfrv(
  phenodata,
  genedata,
  genename = "NA",
  weights = "Equal",
  missing_cutoff = 0.15,
  max_maf = 1,
  model = "zip"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="zimfrv_+3A_phenodata">phenodata</code></td>
<td>
<p>a data frame containing family and individual IDs for all objects as well as
zero-inflated counts as a phenotype and a set of covariates.
Each row represents a different individual.
The first two columns are Family ID (FID) and Individual ID (IID) respectively.
There must be one and only one phenotype in the third column and
the phenotype have to be zero-inflated count data which should be non-negative integers, e.g. neuritic plaque counts.
Each of the rest of columns represents a different covariate, e.g. age, sex, etc.</p>
</td></tr>
<tr><td><code id="zimfrv_+3A_genedata">genedata</code></td>
<td>
<p>a data frame containing family and individual IDs for all objects as well as numeric genotype data.
Each row represents a different individual.
The first two columns are Family ID (FID) and Individual ID (IID) respectively.
Each of the rest columns represents a seperate gene/SNP marker.
The genotype should be coded as 0, 1, 2 and NA for AA, Aa, aa and missing.
Both of Family ID (FID) and Individual ID (IID) for each row in the 'genedata'
derived from the PLINK formatted files should be in the same order as in the 'phenodata'.
The number of rows in 'genedata' should be equal to the number of rows in 'phenodata'.</p>
</td></tr>
<tr><td><code id="zimfrv_+3A_genename">genename</code></td>
<td>
<p>a character string of the name of a gene, e.g. &quot;CETP&quot;. The name is case-sensitive.</p>
</td></tr>
<tr><td><code id="zimfrv_+3A_weights">weights</code></td>
<td>
<p>a character string of pre-specified variant weighting schemes (default=&quot;Equal&quot;).
&quot;Equal&quot; represents no weight,
&quot;MadsenBrowning&quot; represents the Madsen and Browning (2009) weight,
&quot;Beta&quot; represents the Beta weight.</p>
</td></tr>
<tr><td><code id="zimfrv_+3A_missing_cutoff">missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15).
Any SNPs with missing rates higher than the cutoff will be excluded from the analysis.</p>
</td></tr>
<tr><td><code id="zimfrv_+3A_max_maf">max_maf</code></td>
<td>
<p>a cutoff of the maximum minor allele frequencies (MAF) (default=1, no cutoff).
Any SNPs with MAF &gt; cutoff will be excluded from the analysis.</p>
</td></tr>
<tr><td><code id="zimfrv_+3A_model">model</code></td>
<td>
<p>character specification of zero-inflated count model family (default=&quot;zip&quot;).
&quot;zip&quot; represents Zero-Inflated Poisson model,
&quot;zinb&quot; represents Zero-Inflated Negative Binomial model,
&quot;skat&quot; represents the two-stage Sequence Kernel Association Test method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of 10 items including the name of gene,
the number of rare variants in the genetic region,
the kind of method used for modeling,
and individual p-values of gene‐based association tests (burden test and kernel test for both parameters)
and combined p-values using Cauchy combination test.
</p>
<table role = "presentation">
<tr><td><code>GeneName</code></td>
<td>
<p>the name of gene.</p>
</td></tr>
<tr><td><code>No.Var</code></td>
<td>
<p>the number of rare variants in the gene.</p>
</td></tr>
<tr><td><code>Method</code></td>
<td>
<p>the method used to compute the p-values.</p>
</td></tr>
<tr><td><code>p.value_pi_burden</code></td>
<td>
<p>single p-value for parameter <code class="reqn">\pi</code> using burden test.</p>
</td></tr>
<tr><td><code>p.value_lambda_burden / p.value_mu_burden</code></td>
<td>
<p>single p-value for parameter <code class="reqn">\lambda</code> or <code class="reqn">\mu</code> using burden test.</p>
</td></tr>
<tr><td><code>p.value_pi_kernel</code></td>
<td>
<p>single p-value for parameter <code class="reqn">\pi</code> using kernel test.</p>
</td></tr>
<tr><td><code>p.value_lambda_kernel / p.value_mu_kernel</code></td>
<td>
<p>single p-value for parameter <code class="reqn">\lambda</code> or <code class="reqn">\mu</code> using kernel test.</p>
</td></tr>
<tr><td><code>p.value_pi_combined</code></td>
<td>
<p>Combined p-value of testing parameter <code class="reqn">\pi</code> from both burden and kernel test using Cauchy combination test.</p>
</td></tr>
<tr><td><code>p.value_lambda_combined / p.value_mu_combined</code></td>
<td>
<p>Combined p-value of testing parameter <code class="reqn">\lambda</code> or <code class="reqn">\mu</code> from both burden and kernel test using Cauchy combination test.</p>
</td></tr>
<tr><td><code>p.value_overall</code></td>
<td>
<p>Combined p-value of testing the overall association using Cauchy combination test.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Fan, Q., Sun, S., &amp; Li, Y.‐J. (2021). Precisely modeling zero‐inflated count phenotype for rare variants. Genetic Epidemiology, 1–14.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Ex1_phenodata)
data(Ex1_genedata)
zimfrv(Ex1_phenodata,Ex1_genedata,weights = "Beta",max_maf = 0.02,model="zinb")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
