<!DOCTYPE html><html><head><title>Help for package ZIPG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ZIPG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Dietary'><p>A diet-microbiome data</p></a></li>
<li><a href='#ZIPG_CI'><p>Get confidence interval from ZIPG model</p></a></li>
<li><a href='#ZIPG_main'><p>Fit zero-inflated poisson-gamma model via EM Algorithm</p></a></li>
<li><a href='#ZIPG_simulate'><p>Simulate W from ZIPG model</p></a></li>
<li><a href='#ZIPG_summary'><p>Summary for ZIPG_main() result.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Zero-Inflated Poisson-Gamma Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Roulan Jiang [aut, cre],
  Tianying Wang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Roulan Jiang &lt;roulan2000@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We provide a flexible Zero-inflated Poisson-Gamma Model (ZIPG) by connecting both the mean abundance and the variability to different covariates, and build valid statistical inference procedures for both parameter estimation and hypothesis testing. These functions can be used to analyze microbiome count data with zero-inflation and overdispersion. The model is discussed in Jiang et al (2023) &lt;<a href="https://doi.org/10.1080%2F01621459.2022.2151447">doi:10.1080/01621459.2022.2151447</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/roulan2000/ZIPG">https://github.com/roulan2000/ZIPG</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/roulan2000/ZIPG/issues">https://github.com/roulan2000/ZIPG/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>pscl, MASS</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-05 18:44:12 UTC; roulan2000</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-05 22:40:18 UTC</td>
</tr>
</table>
<hr>
<h2 id='Dietary'>A diet-microbiome data</h2><span id='topic+Dietary'></span>

<h3>Description</h3>

<p>A diet-microbiome data with shotgun metagenomic sequencing results of fecal samples and daily dietary records of 34 subjects in 17 consecutive days.
See original data from article &quot;Daily Sampling Reveals Personalized Diet-Microbiome Associations in Humans&quot; (Johnson et al. 2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dietary
</code></pre>


<h3>Format</h3>

<p>A list with OTU table and other covariates
</p>

<dl>
<dt>OTU</dt><dd><p>OTU table</p>
</dd>
<dt>COV</dt><dd><p>Major covariates extracted from original data</p>
</dd>
<dt>M</dt><dd><p>Sequencing depth</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://github.com/knights-lab/dietstudy_analyses">https://github.com/knights-lab/dietstudy_analyses</a>
</p>

<hr>
<h2 id='ZIPG_CI'>Get confidence interval from ZIPG model</h2><span id='topic+ZIPG_CI'></span>

<h3>Description</h3>

<p>Get confidence interval from ZIPG model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZIPG_CI(ZIPG_res, type = "Wald", CI_type = "normal", alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZIPG_CI_+3A_zipg_res">ZIPG_res</code></td>
<td>
<p>Result from ZIPG_main()</p>
</td></tr>
<tr><td><code id="ZIPG_CI_+3A_type">type</code></td>
<td>
<p>Type of hypothesis testing method, 'Wald' or 'bWald'.</p>
</td></tr>
<tr><td><code id="ZIPG_CI_+3A_ci_type">CI_type</code></td>
<td>
<p>Type of confidence interval, 'Wald','bWald' or 'pbWald'.</p>
</td></tr>
<tr><td><code id="ZIPG_CI_+3A_alpha">alpha</code></td>
<td>
<p>We construct (1- alpha)% confidence interval by alpha/2 and (1-alpha/2).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table of confidence interval
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Dietary)
dat = Dietary
ZIPG_res &lt;- ZIPG_main(data = dat$COV,
X = ~ALC01+nutrPC1+nutrPC2, X_star = ~ ALC01,
W = dat$OTU[,100], M = dat$M )
ZIPG_CI(ZIPG_res)
</code></pre>

<hr>
<h2 id='ZIPG_main'>Fit zero-inflated poisson-gamma model via EM Algorithm</h2><span id='topic+ZIPG_main'></span>

<h3>Description</h3>

<p>Fit zero-inflated poisson-gamma model via EM Algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZIPG_main(
  data,
  W,
  M,
  X,
  X_star,
  return_model = TRUE,
  pbWald_list = NULL,
  bWald_list = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZIPG_main_+3A_data">data</code></td>
<td>
<p>Data.frame for covariates of interest</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_w">W</code></td>
<td>
<p>Count data</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_m">M</code></td>
<td>
<p>Sequencing depth, ZIPG use log(M) as offset by default</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_x">X</code></td>
<td>
<p>Formula of covariates of differential abundance</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_x_star">X_star</code></td>
<td>
<p>Formula of covariates of differential varibility</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_return_model">return_model</code></td>
<td>
<p>whether return full complete imfomation for fitted model</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_pbwald_list">pbWald_list</code></td>
<td>
<p>A list of arguments for parameteric bootstrap Wald test,
B for bootstrap sample size, X0 and X_star0 for formula of covariates included in H0</p>
</td></tr>
<tr><td><code id="ZIPG_main_+3A_bwald_list">bWald_list</code></td>
<td>
<p>A list of arguments for non-parameteric bootstrap Wald test, B for bootstrap sample size,</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of ZIPG fitted model. Use ZIPG_summary() for a quick look at the results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Dietary)
dat = Dietary
ZIPG_res &lt;- ZIPG_main(data = dat$COV,
X = ~ALC01+nutrPC1+nutrPC2, X_star = ~ ALC01,
W = dat$OTU[,100], M = dat$M )
ZIPG_summary(ZIPG_res)
</code></pre>

<hr>
<h2 id='ZIPG_simulate'>Simulate W from ZIPG model</h2><span id='topic+ZIPG_simulate'></span>

<h3>Description</h3>

<p>Simulate W from ZIPG model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZIPG_simulate(
  M,
  X,
  X_star,
  A = 1,
  d,
  d_star,
  parms,
  N,
  zi = TRUE,
  returnU = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZIPG_simulate_+3A_m">M</code></td>
<td>
<p>Sequencing depth</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_x">X</code></td>
<td>
<p>Covariates matrix with intercept, n * (d+1)</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_x_star">X_star</code></td>
<td>
<p>Covariates matrix with intercept, n * (d_star+1)</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_a">A</code></td>
<td>
<p>no use, reserved for multi-taxa</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_d">d</code></td>
<td>
<p>number of covariates in X</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_d_star">d_star</code></td>
<td>
<p>number of covariates in X_star</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_parms">parms</code></td>
<td>
<p>model paraneters, input c(beta,beta*,gamma)</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_n">N</code></td>
<td>
<p>repetition times</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_zi">zi</code></td>
<td>
<p>whether generate zero-inflated distribution</p>
</td></tr>
<tr><td><code id="ZIPG_simulate_+3A_returnu">returnU</code></td>
<td>
<p>whether return fluctuation factor U</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of W generated from ZIPG model with input parameter
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Dietary)
dat = Dietary
sim_M = sample(dat$M,100,replace = TRUE)
sim_pre = rep(sample(rep(c(0,1),each = 10)),each = 5)
sim_PC1_mean = rep(rnorm(20,mean = 0,sd = 1),each = 5)
sim_PC1_error = rnorm(100,0,0.1)
sim_PC1 = sim_PC1_mean + sim_PC1_error
X = as.matrix(cbind(1,data.frame(X1 = sim_pre,X2 = sim_PC1)))
parms = c(-4.23,1,0.45,0.6,1,0,0) #p = 0.5
W_sim &lt;- ZIPG_simulate(M = sim_M,X=X,X_star=X,d=2,d_star=2,parms = parms,N=100)
hist(W_sim$W_list[[1]])
ZIPG_res &lt;- ZIPG_main(data = data.frame(X1 = sim_pre,X2 = sim_PC1),
X = ~X1+X2, X_star = ~ X1,W = W_sim$W_list[[2]], M = sim_M )
ZIPG_summary(ZIPG_res)
</code></pre>

<hr>
<h2 id='ZIPG_summary'>Summary for ZIPG_main() result.</h2><span id='topic+ZIPG_summary'></span>

<h3>Description</h3>

<p>Summary for ZIPG_main() result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZIPG_summary(ZIPG_res, type = "Wald")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZIPG_summary_+3A_zipg_res">ZIPG_res</code></td>
<td>
<p>Result from ZIPG_main()</p>
</td></tr>
<tr><td><code id="ZIPG_summary_+3A_type">type</code></td>
<td>
<p>Type of hypothesis testing method, 'Wald','bWald' or 'pbWald'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pvalue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Dietary)
dat = Dietary
ZIPG_res &lt;- ZIPG_main(data = dat$COV,
X = ~ALC01+nutrPC1+nutrPC2, X_star = ~ ALC01,
W = dat$OTU[,100], M = dat$M )
ZIPG_summary(ZIPG_res)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
